<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Dockstore</title>
</head>
<body>
<h1>Sentences that contain Dockstore</h1>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36732505">36732505</a> <b>(1)</b> High-depth sequencing characterization of viral dynamics across tissues in fatal COVID-19 reveals compartmentalized infection<ul><li>Viral genomic analyses were performed through use of viral-ngs pipelines (<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs), as implemented on Terra platform (app.terra.bio).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36711673">36711673</a> <b>(18)</b> Scalable Nanopore sequencing of human genomes provides a comprehensive view of haplotype-resolved variation and methylation<ul><li>We publish the complete open-source pipeline written in WDL in the <b>Dockstore</b> repository (<ext-link xlink:href="https://<b>dockstore</b>.org/organizations/NIHCARD" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/NIHCARD</ext-link>; <xref rid="S14" ref-type="sec"><u>Methods</u></xref>) to encourage easy reuse.<li> We used the alignment WDL available in the Flagger repository on <b>Dockstore</b> (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/LongReadAlignerScattered" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/LongReadAlignerScattered</ext-link>).<li>For running Secphase, we used the workflow available in the Secphase repository on <b>Dockstore</b>.<li> (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/secphase/Secphase" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/secphase/Secphase</ext-link>).<li>For running the preprocessing steps, we used the workflow available in the Flagger repository (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/FlaggerPreprocess" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/FlaggerPreprocess</ext-link>) The workflow parameters were set as following: <italic toggle="yes">minReadLength = 5000; minAlignmentLength = 5000;</italic> maxDivergence = 0.02; variantCaller = &#x201C;dv&#x201D; for HiFi; maxDivergence = 0.09, variantCaller = &#x201C;pmdv&#x201D; for ONT.<li>To project the blocks from CHM13 coordinates to the assembly coordinates, we used another workflow provided by Flagger (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/ProjectBlocksForFlagger" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/ProjectBlocksForFlagger</ext-link>).<li>For running the Flagger core pipeline, we used the workflow available in the Flagger repository on <b>Dockstore</b> (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/Flagger" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/Flagger</ext-link>).<li>The output bed file of Flagger with the projected annotations (including sex, centromere and SD) was then given to the FlaggerStats workflow to produce tables of statistics (<ext-link xlink:href="https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/FlaggerStats" ext-link-type="uri">https://<b>dockstore</b>.org/my-workflows/github.com/mobinasri/flagger/FlaggerStats</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36198312">36198312</a> <b>(2)</b> Multimodal surveillance of SARS-CoV-2 at a university enables development of a robust outbreak response framework<ul><li><hr/></td></tr><tr><td>Study-specific analyses</td><td>GitHub (Broad Institute)</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/sc2-cmu-study" id="intref0075">https://github.com/broadinstitute/sc2-cmu-study</ext-link></td></tr><tr><td>LoFreq version 2.1.5</td><td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1093/nar/gks918" id="intref0080">https://doi.org/10.1093/nar/gks918</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://csb5.github.io/lofreq/" id="intref0085">https://csb5.github.io/lofreq/</ext-link></td></tr><tr><td>viral-ngs 2.1.28</td><td>Broad Institute</td><td><ext-link ext-link-type="uri" xlink:href="http://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs" id="intref0090"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35132411">35132411</a> <b>(1)</b> Outbreak.info Research Library: A standardized, searchable platform to discover and explore COVID-19 resources<ul><li> Using this schema we ingested and harmonized metadata from an initial set of sixteen key resources: LitCovid (Publications), bioRxiv and medRxiv (Publications), COVID-19 Literature Surveillance Team (COVID-19 LST) (Publications), <ext-link xlink:href="https://ClinicalTrials.gov" ext-link-type="uri">ClinicalTrials.gov</ext-link> (NCT) (ClinicalTrials), WHO International Clinical Trials Registry Platform (WHO ICTRP) (ClinicalTrials), Figshare (Datasets, Publications, and more), Zenodo (Datasets, ComputationalTools, Publications, and more), MRC Centre for Global Infectious Disease Analysis (Analyses, Publications, and More), Protocols.io (Protocols), Protein Data Bank (PDB) (Datasets), Data Discovery Engine (DDE) (Datasets), Harvard Dataverse (Datasets), ImmPort (Datasets), bio.tools (ComputationalTools), and <b>Dockstore</b> (ComputationalTools) (<xref rid="F1" ref-type="fig"><u>Figure 1a</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36380755">36380755</a> <b>(1)</b> Towards increased accuracy and reproducibility in SARS-CoV-2 next generation sequence analysis for public health surveillance<ul><li> Variant calling was then performed with platform-specific Galaxy workflows, which have previously been described<sup><xref rid="R38" ref-type="bibr"><u>38</u></xref></sup> and are publicly available from <b>Dockstore</b><sup><xref rid="R39" ref-type="bibr"><u>39</u></xref></sup> and the WorkflowHub<sup><xref rid="R40" ref-type="bibr"><u>40</u></xref></sup> against NCBI Reference Sequence NC_045512.2.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34588690">34588690</a>) [39](<name><surname>O&#x2019;Connor</surname><given-names>B. D.</given-names></name><etal/><article-title>The Dockstore: enabling modular, community-focused sharing of Docker-based genomics tools and workflows</article-title>. <source>F1000Research</source><volume>6</volume>, (<year>2017</year>).) [40](<source>Freely accessible ready to use global infrastructure and workflows for SARS-CoV-2 monitoring</source>. <comment><ext-link xlink:href="https://workflowhub.eu/collections/2" ext-link-type="uri">https://workflowhub.eu/collections/2</ext-link>.</comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36302863">36302863</a> <b>(4)</b> A living biobank of canine mammary tumor organoids as a comparative model for human breast cancer<ul><li> Individual lanelets (pairs of fastqs) were mapped to canine canFam3.1 genome, coordinate sorted merged, and duplicates-marked using dockerized cgpmap v3.0.4: <ext-link ext-link-type="uri" xlink:href="https://github.com/cancerit/<b>dockstore</b>-cgpmap">https://github.com/cancerit/<b>dockstore</b>-cgpmap</ext-link>.<li> Copy-number variants were called using ASCAT, which was executed as part of the dockerized cgpwgs v2.0.1: <ext-link ext-link-type="uri" xlink:href="https://github.com/cancerit/<b>dockstore</b>-cgpwgs">https://github.com/cancerit/<b>dockstore</b>-cgpwgs</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36261518">36261518</a> <b>(2)</b> Semi-automated assembly of high-quality diploid human reference genomes<ul><li> Here we automated the Mercury took kit, for more rapid analyses of assemblies (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/human-pangenomics/hpp_production_workflows/Merqury:master?tab=info">https://<b>dockstore</b>.org/workflows/github.com/human-pangenomics/hpp_production_workflows/Merqury:master?tab=info</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36131292">36131292</a> <b>(8)</b> The architecture of clonal expansions in morphologically normal tissue from cancerous and non-cancerous prostates<ul><li> An updated version of this pipeline is available as a Docker image (Alignment: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap</ext-link>; Variant-calling: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs</ext-link>)&#xFEFF;.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35915661">35915661</a> <b>(3)</b> Genomic evidence for divergent co-infections of co-circulating SARS-CoV-2 lineages<ul><li> Since the samples were sampled and sequenced following the ARTIC version 3 protocol, all the sra files were treated with a recommended workflow (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA%3amain?tab=info" id="ir010">https://<b>dockstore</b>.org/workflows/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA:main?tab=info</ext-link>) to detect intra-host single nucleotide variants (iSNVs).<li>The workflow for calling iSNVs can be retrieved from (https://<b>dockstore</b>.org/workflows/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA:main?tab=info).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35843112">35843112</a> <b>(3)</b> Rapid repeat infection of SARS-CoV-2 by two highly distinct delta-lineage viruses<ul><li>Analysis of SARS-CoV-2 genomic data was performed using <italic>Titan ClearLabs</italic> and <italic>Titan_Augur_Run</italic>, 2 workflow description language [WDL] workflows within the Theiagen Public Health Viral Genomics [PHVG] <b>DockStore</b> collection [<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/Theiagen/collections/PublicHealthViralGenomics" id="interref0001">https://<b>dockstore</b>.org/organizations/Theiagen/collections/PublicHealthViralGenomics</ext-link>].</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35446428">35446428</a> <b>(1)</b> The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2022 update<ul><li> The workflows are also automatically published to <b>Dockstore</b> (<xref rid="B12" ref-type="bibr"><u>12</u></xref>) and WorkflowHub (<xref rid="B13" ref-type="bibr"><u>13</u></xref>) and optionally synchronized to a list of Galaxy servers.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [13](<person-group person-group-type="author"><string-name><surname>Goble</surname><given-names>C.</given-names></string-name>, <string-name><surname>Soiland-Reyes</surname><given-names>S.</given-names></string-name>, <string-name><surname>Bacall</surname><given-names>F.</given-names></string-name>, <string-name><surname>Owen</surname><given-names>S.</given-names></string-name>, <string-name><surname>Williams</surname><given-names>A.</given-names></string-name>, <string-name><surname>Eguinoa</surname><given-names>I.</given-names></string-name>, <string-name><surname>Droesbeke</surname><given-names>B.</given-names></string-name>, <string-name><surname>Leo</surname><given-names>S.</given-names></string-name>, <string-name><surname>Pireddu</surname><given-names>L.</given-names></string-name>, <string-name><surname>Rodr&#xED;guez-Navas</surname><given-names>L.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Implementing FAIR digital objects in the EOSC-Life workflow collaboratory</article-title>. <year>2021</year>; <comment>Extended abstract, Zenodo doi:</comment><comment>12 March 2021, preprint: not peer reviewed</comment><pub-id pub-id-type="doi">10.5281/zenodo.4605654</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35414112">35414112</a> <b>(1)</b> Galaxy workflows for fragment-based virtual screening: a case study on the SARS-CoV-2 main protease<ul><li> If tests are successful and the submission is approved by an IWC review, the submitted workflows are deployed to <b>Dockstore</b> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] and WorkflowHub [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], two recently developed platforms for sharing scientific workflows.<li> [19](          <person-group person-group-type="author">            <name>              <surname>Yuen</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Cabansay</surname>              <given-names>L</given-names>            </name>            <name>              <surname>Duncan</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Luu</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Hogue</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Overbeck</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Perez</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Shands</surname>              <given-names>W</given-names>            </name>            <name>              <surname>Steinberg</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Reid</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Olunwa</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Hansen</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Sheets</surname>              <given-names>E</given-names>            </name>            <name>              <surname>O&#x2019;Farrell</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Cullion</surname>              <given-names>K</given-names>            </name>            <name>              <surname>O&#x2019;Connor</surname>              <given-names>BD</given-names>            </name>            <name>              <surname>Paten</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Stein</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>The Dockstore: enhancing a community platform for sharing reproducible and accessible computational protocols</article-title>          <source>Nucleic Acids Res</source>          <year>2021</year>          <volume>49</volume>          <issue>W1</issue>          <fpage>624</fpage>          <lpage>632</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkab346</pub-id>        ) [20](Goble C, Soiland-Reyes S, Bacall F, Owen S, Williams A, Eguinoa I, Droesbeke B, Leo S, Pireddu L, Rodr&#xED;guez-Navas L, Fern&#xE1;ndez JM, Capella-Gutierrez S, M&#xE9;nager H, Gr&#xFC;ning B, Serrano-Solano B, Ewels P, Coppens F (2021) Implementing FAIR Digital Objects in the EOSC-Life Workflow Collaboratory. Extended abstract, submitted to Special issue on Canonical Workflow Frameworks for Research in the journal Data Intelligence. 10.5281/zenodo.4605654) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35368031">35368031</a> <b>(2)</b> A suspension technique for efficient large-scale cancer organoid culturing and perturbation screens<ul><li> Germline variants and technology-specific artefacts were removed by filtering against a matched normal blood sample and the panel of 100 unrelated normal samples (<ext-link ext-link-type="uri" xlink:href="ftp://ftp.sanger.ac.uk/pub/cancer/<b>dockstore</b>/human/SNV_INDEL_ref_GRCh37d5.tar.gz">ftp://ftp.sanger.ac.uk/pub/cancer/<b>dockstore</b>/human/SNV_INDEL_ref_GRCh37d5.tar.gz</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35480627">35480627</a> <b>(2)</b> Comparative transmissibility of SARS-CoV-2 variants Delta and Alpha in New England, USA<ul><li> All of the workflows named below are publicly available via the <b>Dockstore</b> Tool Registry Service (<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35373174">35373174</a> <b>(3)</b> Federated analysis of BRCA1 and BRCA2 variation in a Japanese cohort<ul><li> This can yield valuable evidence for variant interpretation without the sensitive data leaving the home institution.<sup><xref rid="R12" ref-type="bibr"><u>12</u></xref></sup> Container technologies support this approach by bundling the software and all its dependencies into a single module for straightforward installation and deployment on a collaborator&#x2019;s system.<sup><xref rid="R13" ref-type="bibr"><u>13</u></xref></sup> These technologies include Docker,<sup><xref rid="R14" ref-type="bibr"><u>14</u></xref></sup> Singularity,<sup><xref rid="R15" ref-type="bibr"><u>15</u></xref></sup> and Jupyter.<sup><xref rid="R16" ref-type="bibr"><u>16</u></xref></sup> Containers and workflows can be shared on the <b>Dockstore</b> platform<sup><xref rid="R17" ref-type="bibr"><u>17</u></xref></sup> so that multiple institutions can execute the same software, promoting reproducibility.<li> and publishing these certification results could fit well within the framework of container libraries such as <b>Dockstore</b>.<li>All original code has been deposited at GitHub and <b>Dockstore</b> and is publicly available as of the date of publication.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32540846">32540846</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28163975">28163975</a>) [14](<name><surname>Matthias</surname><given-names>K</given-names></name>, and <name><surname>Kane</surname><given-names>SP</given-names></name> (<year>2015</year>). <source>Docker: up and running</source> (<publisher-name>O&#x2019;Reilly Media, Inc</publisher-name>).) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [16](<name><surname>Toomey</surname><given-names>D</given-names></name> (<year>2017</year>). <source>Jupyter for Data Science: exploratory analysis, statistical modeling, machine learning, and data visualization with Jupyter (Packt)</source>.) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35300209">35300209</a> <b>(1)</b> AFA: Ancestry-specific allele frequency estimation in admixed populations: The Hispanic Community Health Study/Study of Latinos<ul><li> CWL workflows for GAFA and LAFA are also available via <b>dockstore</b> and <ext-link ext-link-type="uri" xlink:href="https://github.com/cwl-apps/ancestral-maf-admixed-population" id="intref0020">https://github.com/cwl-apps/ancestral-maf-admixed-population</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35130561">35130561</a> <b>(1)</b> Multiplexed CRISPR-based microfluidic platform for clinical testing of respiratory viruses and identification of SARS-CoV-2 variants<ul><li> all workflows are publicly available on the <b>Dockstore</b> tool registry service.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35051367">35051367</a> <b>(7)</b> Transmission from vaccinated individuals in a large SARS-CoV-2 Delta variant outbreak<ul><li><hr/></td></tr><tr><td>LoFreq version 2.1.5</td><td><xref rid="bib27" ref-type="bibr"><u>Wilm et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://csb5.github.io/lofreq/" id="intref0025">https://csb5.github.io/lofreq/</ext-link></td></tr><tr><td>viral-ngs 2.1.28</td><td>Broad Institute</td><td><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_illumina_full:master" id="intref0030">https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_illumina_full:master</ext-link>.<li><hr/></td></tr><tr><td>LoFreq version 2.1.5</td><td><xref rid="bib27" ref-type="bibr"><u>Wilm et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://csb5.github.io/lofreq/" id="intref0025">https://csb5.github.io/lofreq/</ext-link></td></tr><tr><td>viral-ngs 2.1.28</td><td>Broad Institute</td><td><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_illumina_full:master" id="intref0030">https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_illumina_full:master</ext-link></td></tr><tr><td>Nextstrain v3.0.3</td><td><xref rid="bib12" ref-type="bibr"><u>Hadfield et&#xA0;al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_nextstrain_aligned_input:master" id="intref0035">https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/viral-pipelines/sarscov2_nextstrain_aligned_input:master</ext-link>.<li> All of the workflows named below are publicly available via the <b>Dockstore</b> Tool Registry Service (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs" id="intref0070"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>).<li> [Wilm et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23066108">23066108</a>) [Wilm et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23066108">23066108</a>) [Hadfield et&#xA0;al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35199087">35199087</a> <b>(14)</b> Inverting the model of genomics data sharing with the NHGRI Genomic Data Science Analysis, Visualization, and Informatics Lab-space<ul><li> The analysis components are broadly characterized into 3 major categories: (1) those supporting data management querying, especially Gen3, (2) those supporting batch computing, especially through the use of the WDL on Terra and the closely related <b>Dockstore</b><sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup> for sharing and distributing workflows, and (3) interactive computing using popular analysis suites such as R/Bioconductor, Jupyter Notebooks, and Galaxy.<li> Additionally, these components support reproducibility and reusability as methods and workflows that are deposited in <b>Dockstore</b> are assigned DOIs, cohorts (synthetic or designed) can be referenced via Terra workspace URLs, and we are developing technologies for versioning and publishing workspaces with DOIs that can scale to millions of files and petabytes of information.<li> WDLs can be stored, shared, and described in <b>Dockstore</b> and executed in Terra using the Cromwell compute engine (<ext-link xlink:href="https://cromwell.readthedocs.io/" ext-link-type="uri">https://cromwell.readthedocs.io</ext-link>), allowing for a reproducible analysis of even the largest cohorts with tens of thousands of samples.<li><b>Dockstore</b>: Registry of tools and workflows.<li>The <b>Dockstore</b> (<ext-link xlink:href="https://<b>dockstore</b>.org/" ext-link-type="uri">https://<b>dockstore</b>.org</ext-link>) is another widely used platform where users can find, share, and use curated tools and workflows.<li> The use of Docker makes workflows in <b>Dockstore</b> reproducible by making them easy to run without user installation.<li> <b>Dockstore</b> enables scientists to share analytical tools in a way that makes them machine-readable and -runnable in a variety of environments.<li> <b>Dockstore</b> currently supports 4 workflow languages: the WDL, Common Workflow Language (CWL), Nextflow, and Galaxy Workflows (GWs).<li> <b>Dockstore</b> currently contains 745 workflows in WDL that can be launched in Terra within a few clicks.<li> As such, <b>Dockstore</b> provides one of the most straightforward entry points for users to add batch workflows to the AnVIL as it can work with any tool/workflow that can be encapsulated into a Docker container and executed on the command line.<li> Furthermore, the Terra, Gen3, and <b>Dockstore</b> components of the AnVIL directly support multiple cloud platforms.<li> These workflows are shared through <b>Dockstore</b>, which itself supports the GA4GH Tool Registry Service (TRS), making it possible to share workflows among many different systems beyond the AnVIL.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978761">33978761</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34972102">34972102</a> <b>(1)</b> Facilitating cancer systems epidemiology research<ul><li>Biosphere is an open-source platform developed by the Broad Institute that can operate across several different platforms (e.g., Terra, Gen3, and <b>Dockstore</b>) to create an interoperable data environment for the biomedical community.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34836952">34836952</a> <b>(4)</b> Whole-genome analysis of Nigerian patients with breast cancer reveals ethnic-driven somatic evolution and distinct genomic subtypes<ul><li> <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs:2.0.1?tab=info">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs:2.0.1?tab=info</ext-link>) with default filters implemented.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737383">34737383</a> <b>(5)</b> Design considerations for workflow management systems use in production genomics research and the clinic<ul><li> Containerization technologies, e.g., docker and singularity, and their repositories (e.g <b>Dockstore</b><sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup> and quay.io) facilitate the reproducibility of computational pipelines.<li> <b>Dockstore</b><sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>, the standard implementation of the GA4GH TRS, is now hosting docker-based tools described in Nextflow, besides CWL and WDL.<li>We chose WfMSs (Table <xref rid="Tab4" ref-type="table"><u>4</u></xref>) that support the widely adopted Global Alliance for Genomics and Health (GA4GH) APIs: the Workflow Execution Service Schema (WES) for describing how a user submits a workflow to an execution engine in a standardized way, the Task Execution Schema (TES) for describing batch execution tasks (implemented primarily as Funnel: <ext-link ext-link-type="uri" xlink:href="https://github.com/ohsu-comp-bio/funnel">https://github.com/ohsu-comp-bio/funnel</ext-link>), and the Tool Registry Service (TRS) for sharing code (implemented primarily as <b>dockstore</b>: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link>).<li> [50](O&#x2019;Connor, B.&#xA0;D. <italic>et&#xA0;al.</italic> The dockstore: Enabling modular, community-focused sharing of docker-based genomics tools and workflows. <italic>F1000Research</italic><bold>6</bold> (2017).) [50](O&#x2019;Connor, B.&#xA0;D. <italic>et&#xA0;al.</italic> The dockstore: Enabling modular, community-focused sharing of docker-based genomics tools and workflows. <italic>F1000Research</italic><bold>6</bold> (2017).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34721839">34721839</a> <b>(1)</b> BOSC 2021, the 22nd Annual Bioinformatics Open Source Conference<ul><li>The session on Workflow Management Systems included updates on widely used workflow systems such as Nextflow, <b>Dockstore</b> and Sapporo, and introductions to newer resources such as WDL Analysis Research Pipelines (WARP) and WorkFlow Package Manager (WFPM).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34643507">34643507</a> <b>(1)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li>Multiple online repositories provide CWL documents, including the <b>Dockstore</b> tool registry (<xref rid="bib38" ref-type="bibr"><u>O&#x2019;Connor et al., 2017</u></xref>) and GitHub.<li> [O&#x2019;Connor et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34642698">34642698</a> <b>(3)</b> Comparative transmissibility of SARS-CoV-2 variants Delta and Alpha in New England, USA<ul><li> All of the workflows named below are publicly available via the <b>Dockstore</b> Tool Registry Service (<ext-link xlink:href="http://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs" ext-link-type="uri"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34553169">34553169</a> <b>(3)</b> The role of metadata in reproducible computational research<ul><li> Biotoolsxsd, DOAP, ontosoft, SWO</td><td><b>Dockstore</b>, Biocontainers.<li> An example of such a database is the GA4GH <b>Dockstore</b>,<xref rid="bib98" ref-type="bibr"><sup>140</sup></xref> a hub and associated website that allows for a standardized means of describing and invoking Dockerized tools as well as sharing workflows based on them.<li>Using this metadata, an important aspect of CWL is the focus on richly describing tool invocations both for reproducibility and documentation purposes, with tools referenced as retrievable Docker images or Conda packages, and identifiers to EDAM,<xref rid="bib72" ref-type="bibr"><sup>63</sup></xref> ELIXIR's bio.tools<xref rid="bib118" ref-type="bibr"><sup>59</sup></xref> registry and Research Resource Identifiers (RRIDs).<xref rid="bib119" ref-type="bibr"><sup>161</sup></xref> This wrapping of command line tool interfaces is used by GA4GH <b>Dockstore</b><xref rid="bib98" ref-type="bibr"><sup>140</sup></xref> for providing a uniform executable interface to a large variety of computational tools even outside workflows.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34804501">34804501</a> <b>(1)</b> Perspectives on automated composition of workflows in the life sciences<ul><li>Existing workflows from repositories like myExperiment,<sup><xref rid="ref28" ref-type="bibr"><u>28</u></xref>,<xref rid="ref29" ref-type="bibr"><u>29</u></xref></sup> <b>Dockstore</b><sup><xref rid="ref47" ref-type="bibr"><u>47</u></xref></sup> or the WorkflowHub<sup><xref rid="ref26" ref-type="bibr"><u>26</u></xref></sup> can be reused at different stages, preceding stages in the principal life cycle to be either skipped or shortened.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20501605">20501605</a>) [29](<person-group person-group-type="author"><name><surname>De Roure</surname><given-names>D</given-names></name><name><surname>Goble</surname><given-names>C</given-names></name><name><surname>Stevens</surname><given-names>R</given-names></name></person-group>:<article-title>The design and realisation of the myExperiment Virtual Research Environment for social sharing of workflows.</article-title><source><italic toggle="yes">Future Gener. Comput. Syst.</italic></source><year>2009</year>;<volume>25</volume>:<fpage>561</fpage>&#x2013;<lpage>567</lpage>.<pub-id pub-id-type="doi">10.1016/j.future.2008.06.010</pub-id>) [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [26](<collab>The WorkflowHub</collab>. Last accessed 2020/10/25.<ext-link xlink:href="https://workflowhub.eu/" ext-link-type="uri">Reference Source</ext-link>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<collab>The WorkflowHub</collab>. Last accessed 2020/10/25.<ext-link xlink:href="https://workflowhub.eu/" ext-link-type="uri">Reference Source</ext-link>"><collab>The WorkflowHub</collab>. Last accessed 2020/10/25.<ext-link xlink:href="https://workflowhub.eu/" ext-link-type="uri">Reference Source</ext-link></a> <b>(3)</b> Perspectives on automated composition of workflows in the life sciences<ul><li>The <b>Dockstore</b>.<li>              <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/" ext-link-type="uri">https://<b>dockstore</b>.org/</ext-link>            .</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34185678">34185678</a> <b>(1)</b> NCI Imaging Data Commons<ul><li> Repositories of reusable image analysis tools that are easily accessible from cloud workflows [with the relevant existing platforms including <b>Dockstore</b> (<xref rid="bib29" ref-type="bibr"><u>29</u></xref>) and ModelHub.AI (<xref rid="bib30" ref-type="bibr"><u>30</u></xref>) ]need to be established.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [30](<person-group person-group-type="author"><string-name><surname>Hosny</surname><given-names>A</given-names></string-name>, <string-name><surname>Schwier</surname><given-names>M</given-names></string-name>, <string-name><surname>Berger</surname><given-names>C</given-names></string-name>, <string-name><surname>&#xD6;rnek</surname><given-names>EP</given-names></string-name>, <string-name><surname>Turan</surname><given-names>M</given-names></string-name>, <string-name><surname>Tran</surname><given-names>PV</given-names></string-name>, <etal/></person-group>. <article-title>ModelHub.AI: dissemination platform for deep learning models</article-title>.<source>arXiv [csLG]</source><year>2019</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34097189">34097189</a> <b>(3)</b> Decoding human cancer with whole genome sequencing: a review of PCAWG Project studies published in February 2020<ul><li> It enables queries across different tissues, cell types, developmental stages, and experimental conditions, across thousands of publicly available RNA-seq, microarray, and proteomics datasets</td><td><ext-link ext-link-type="uri" xlink:href="https://www.ebi.ac.uk/gxa/experiments?experimentSet=Pan-Cancer">https://www.ebi.ac.uk/gxa/experiments?experimentSet=Pan-Cancer</ext-link></td></tr><tr><td>PCAWG-Scout</td><td>PCAWG-Scout is a data portal that provides a framework to make on-demand, in-depth analyses over the open access PCAWG data</td><td><ext-link ext-link-type="uri" xlink:href="http://pcawgscout.bsc.es/">http://pcawgscout.bsc.es/</ext-link></td></tr><tr><td>Chromothripsis Explorer</td><td>The Chromothripsis Explorer portal enables exploration of patterns of chromothripsis in the PCAWG dataset</td><td><ext-link ext-link-type="uri" xlink:href="http://compbio.med.harvard.edu/chromothripsis/">http://compbio.med.harvard.edu/chromothripsis/</ext-link></td></tr><tr><td>Cancer LncRNA Census</td><td>The Cancer LncRNA Census is an ongoing effort to identify and catalogue lncRNA genes which have been causally implicated in cancer</td><td><ext-link ext-link-type="uri" xlink:href="https://www.gold-lab.org/clc">https://www.gold-lab.org/clc</ext-link></td></tr><tr><td>PCAWG Core Pipelines</td><td>This <b>Dockstore</b> site contains binaries, source code, and documentation for the open source software tool for all core alignment, QC, and variation-calling pipelines used by PCAWG packaged as portable binaries using Docker and described using workflow description languages.<li> It enables queries across different tissues, cell types, developmental stages, and experimental conditions, across thousands of publicly available RNA-seq, microarray, and proteomics datasets</td><td><ext-link ext-link-type="uri" xlink:href="https://www.ebi.ac.uk/gxa/experiments?experimentSet=Pan-Cancer">https://www.ebi.ac.uk/gxa/experiments?experimentSet=Pan-Cancer</ext-link></td></tr><tr><td>PCAWG-Scout</td><td>PCAWG-Scout is a data portal that provides a framework to make on-demand, in-depth analyses over the open access PCAWG data</td><td><ext-link ext-link-type="uri" xlink:href="http://pcawgscout.bsc.es/">http://pcawgscout.bsc.es/</ext-link></td></tr><tr><td>Chromothripsis Explorer</td><td>The Chromothripsis Explorer portal enables exploration of patterns of chromothripsis in the PCAWG dataset</td><td><ext-link ext-link-type="uri" xlink:href="http://compbio.med.harvard.edu/chromothripsis/">http://compbio.med.harvard.edu/chromothripsis/</ext-link></td></tr><tr><td>Cancer LncRNA Census</td><td>The Cancer LncRNA Census is an ongoing effort to identify and catalogue lncRNA genes which have been causally implicated in cancer</td><td><ext-link ext-link-type="uri" xlink:href="https://www.gold-lab.org/clc">https://www.gold-lab.org/clc</ext-link></td></tr><tr><td>PCAWG Core Pipelines</td><td>This <b>Dockstore</b> site contains binaries, source code, and documentation for the open source software tool for all core alignment, QC, and variation-calling pipelines used by PCAWG packaged as portable binaries using Docker and described using workflow description languages</td><td><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/PCAWG/collections/PCAWG">https://<b>dockstore</b>.org/organizations/PCAWG/collections/PCAWG</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34695175">34695175</a> <b>(1)</b> GEM: scalable and flexible gene&#x2013;environment interaction analysis in millions of samples<ul><li> Docker images and cloud-computing workflows written in Workflow Description Language (WDL) are distributed on github.com/large-scale-gxe-methods and the <b>Dockstore</b> platform (<xref rid="btab223-B15" ref-type="bibr">O&#x2019;Connor <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33990632">33990632</a> <b>(3)</b> Genomic investigation of a dengue virus outbreak in Thi&#xE8;s, Senegal, in 2018<ul><li> Computational workflows used to analyze viral sequencing data are have been made publicly available through <b>dockstore</b>: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs">https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33990632">33990632</a> <b>(135)</b> Genomic investigation of a dengue virus outbreak in Thi&#xE8;s, Senegal, in 2018<ul><li>We originally created <b>Dockstore</b> in response to similar challenges faced by the Pan-Cancer Analysis of Whole Genomes (PCAWG) study (<xref rid="B2" ref-type="bibr"><u>2</u></xref>), which ran between 2014 and 2019.<li> Thus was born <b>Dockstore</b>, a standardized way of packaging, registering, finding and executing analysis workflows that provides improved reproducibility across multiple computing environments.<li> <b>Dockstore</b> is comparable to single language workflow registries such as Agora and the Galaxy Toolshed (<xref rid="B3" ref-type="bibr"><u>3</u></xref>) which are associated with a specific workflow platform, or with container registries such as BioContainers (<xref rid="B4" ref-type="bibr"><u>4</u></xref>).<li> <b>Dockstore</b>'s main distinguishing characteristic is that it is a general solution not tied to any particular workflow architecture, language, or platform.<li>In this way, <b>Dockstore</b> aims to serve as a centralized library of computational methods for the growing variety of technologies that use workflow, container, and cloud solutions for the reproducibility, scalability, and portability of computational analysis in bioinformatics (<xref rid="B5" ref-type="bibr"><u>5</u></xref>,<xref rid="B6" ref-type="bibr"><u>6</u></xref>).<li> <b>Dockstore</b> integrates with multiple workflow languages and &#x2018;<italic toggle="yes">Launch with</italic>&#x2019; partners that can import and run workflows from <b>Dockstore</b> as a service, simplifying analysis and reducing technical barriers for end users.To date, &gt;250 (<xref rid="B7" ref-type="bibr"><u>7</u></xref>) workflow engines have been tracked and many require difficult configuration to run outside their home institutions.<li> The workflow languages that we chose to support in <b>Dockstore</b> were selected due to the robust communities of these languages and like-minded ideas about reproducible workflows, provenance, and best practices in developing bioinformatics software.<li>To share a computational method on <b>Dockstore</b>, a developer first encapsulates the steps of a workflow's environment in a container (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) and then programmatically outlines the analysis steps using a workflow language (also referred to as a &#x2018;descriptor&#x2019;).<li> Together these are registered into <b>Dockstore</b> via popular source control sites.<li>The initial version of <b>Dockstore</b> is described in our 2017 publication (<xref rid="B9" ref-type="bibr"><u>9</u></xref>).<li> Over the past four years, <b>Dockstore</b> has continued building features that increase the Findability, Accessibility, Interoperability, and Reusability (FAIR) (<xref rid="B10" ref-type="bibr"><u>10</u></xref>) of computational analysis resources with our ambition being to make the creation, sharing, and reproduction of scientific analyses as easy as the sharing of scientific publications.<li> In the next sections we provide a brief summary of the main components of <b>Dockstore</b> and then continue with a detailed overview of our new and enhanced features.<li><b>Dockstore</b> core functionality.<li><b>Dockstore</b> aims at being a home for reproducible workflows, exchanged using standardized Global Alliance for Genomics and Health (<xref rid="B11" ref-type="bibr"><u>11</u></xref>) (GA4GH) APIs, and developed in a well-integrated way with common software engineering principles and practices while facilitating their use by users across cloud execution platforms.<li> We have designed <b>Dockstore</b> with two main user types: (i) The developer or author user is typically a bioinformatician or software engineer that will package up their methods using containers and descriptor languages to share tools and workflows with the scientific community.<li> With this in mind, the core of <b>Dockstore</b> includes:.<li><b>Dockstore</b> main site.<li><b>Dockstore</b>'s keystone, and the focus of this paper, is the <b>Dockstore</b> website at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/" ext-link-type="uri">https://<b>dockstore</b>.org/</ext-link>.<li><b>Dockstore</b> library.<li>A workflow or tool in <b>Dockstore</b> is composed of three essential elements as in Figure <xref rid="F1" ref-type="fig"><u>1</u></xref>, including (i) links to the underlying open-source code repository, such as Github or Bitbucket, (ii) links to the containers for the tools in a container repository, such as Docker Hub or Quay.io and critically (ii) the metadata and workflow language descriptions that describe how the tool is configured and parameterized to operate as a complete scientific analysis.<li> By following these best practices, an entry on <b>Dockstore</b> can more easily be incorporated into larger workflows or even decomposed into smaller parts for custom use cases.<li> Once a workflow or tool is registered with <b>Dockstore</b>, it is published in a searchable index where it can be found by other researchers.<li><b>Dockstore</b> makes computational analysis accessible and reproducible by combining containers, descriptor languages, and test parameter files to simplify software reuse and dependency management.<li><b>Dockstore</b> ecosystem.<li>The <b>Dockstore</b> ecosystem consists of three separate important groups: (i) the workflow contributors, who provide valuable tools and workflows to <b>Dockstore</b>; (ii) the <italic toggle="yes">Launch with</italic> partners, who can run workflows with a few clicks and (ii) our development teams based at OICR and UCSC, which maintain the site and work on new features.<li><b>Dockstore</b> provides numerous tutorials, tools, and gentle nudges that help workflow authors create reproducible workflows.We also provide many resources to help orient end users to the <b>Dockstore</b> ecosystem and further training for newcomers to container, cloud, and workflow technologies.<li>How to launch tools and workflows to Launch with platforms or through the <b>Dockstore</b> CLI.<li><b>Dockstore</b> development over the past four years has focused on expanding support for workflow languages and execution platforms, adding integrations with several open source software development environments (making it easier to write reproducible, well tested workflows), and enabling the creation of immutable workflows for publication referencing.<li><b>Dockstore</b>'s support for FAIR principles.<li><b>Dockstore</b> never requires a user to login to search and inspect contents for workflows and tools.<li>One of the most distinguishing features introduced to <b>Dockstore</b> has been its integration with a variety of cloud platforms to directly launch and run workflows over a browser interface.<li> In most cases, integration works by allowing users to click-through from the <b>Dockstore</b> site to the <italic toggle="yes">Launch with</italic> partner site.<li> It should be noted that as these platforms evolve to support newer versions of CWL (<xref rid="B14" ref-type="bibr"><u>14</u></xref>) and WDL (<xref rid="B15" ref-type="bibr"><u>15</u></xref>), <b>Dockstore</b> has updated its validation and testing to match with support for CWL 1.1 and WDL 1.0 as of publication.<li>For the WDL (<xref rid="B10" ref-type="bibr"><u>10</u></xref>) workflow language, <b>Dockstore</b> offers <italic toggle="yes">Launch with</italic> DNAstack (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://www.dnastack.com/" ext-link-type="uri">https://www.dnastack.com/</ext-link>), DNAnexus (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://www.dnanexus.com/" ext-link-type="uri">https://www.dnanexus.com/</ext-link>), Terra (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://terra.bio/" ext-link-type="uri">https://terra.bio/</ext-link>), FireCloud (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://firecloud.terra.bio" ext-link-type="uri">http://firecloud.terra.bio</ext-link>) through Terra's integration, NHLBI Biodata Catalyst (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://biodatacatalyst.nhlbi.nih.gov/" ext-link-type="uri">https://biodatacatalyst.nhlbi.nih.gov/</ext-link>), and AnVIL (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://anvilproject.org/" ext-link-type="uri">https://anvilproject.org/</ext-link>).<li> For the CWL (<xref rid="B11" ref-type="bibr"><u>11</u></xref>) workflow language, <b>Dockstore</b> offers <italic toggle="yes">Launch with</italic> the Cancer Genomics Cloud (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://www.cancergenomicscloud.org/" ext-link-type="uri">https://www.cancergenomicscloud.org/</ext-link>), Cavatica(<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://cavatica.squarespace.com/" ext-link-type="uri">https://cavatica.squarespace.com/</ext-link>) powered by Seven Bridges Genomics, and NHLBI Biodata Catalyst.<li>Browser launch from <b>Dockstore</b></th>                <th rowspan="1" colspan="1">Launch from within Cloud Platform</th>              </tr>            </thead>            <tbody>              <tr>                <td rowspan="1" colspan="1">DNAstack.<li>Recently, <b>Dockstore</b> has added support for the Nextflow (<xref rid="B16" ref-type="bibr"><u>16</u></xref>) and Galaxy (<xref rid="B17" ref-type="bibr"><u>17</u></xref>) workflow languages.<li> Support for a workflow language on <b>Dockstore</b> means that at minimum, we can do some light parsing of workflow content as a sanity check while registering workflows in that language.<li> Details on our level of support for a workflow language is kept up to date on our documentation site (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://docs.<b>dockstore</b>.org/en/develop/end-user-topics/language-support.html" ext-link-type="uri">https://docs.<b>dockstore</b>.org/en/develop/end-user-topics/language-support.html</ext-link>).<li>In addition to supporting the Galaxy workflow language, <b>Dockstore</b> has also started using a plugin architecture that has allowed the project to solicit and quickly integrate code contributions from the Galaxy team.<li> In the future, broader use of this interpreter design pattern (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) via plugins will allow for a more streamlined and rapid expansion of workflow language support on <b>Dockstore</b>.<li>One of the features that distinguishes <b>Dockstore</b> from other platforms is the ability to synchronize workflows from source control repositories like GitHub, BitBucket, and Gitlab.<li><b>Dockstore</b> can register workflows in three main ways, from source control, stored on <b>dockstore</b>.org directly, or tools with descriptors found via quay.io.<li>A recent addition to the platform is the ability to automatically synchronise new versions of workflows from GitHub without requiring developers to visit the <b>Dockstore</b> site.<li> In the traditional method of registering workflows on <b>Dockstore</b>, authors sync their workflows from source control initially, but have to return to <b>Dockstore</b> to pull new versions or releases of their workflows.<li> By using the <b>Dockstore</b> GitHub app (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://docs.<b>dockstore</b>.org/en/develop/getting-started/github-apps/github-apps.html" ext-link-type="uri">https://docs.<b>dockstore</b>.org/en/develop/getting-started/github-apps/github-apps.html</ext-link>), users are able to produce new releases of their software and have these updates automagically pushed to <b>Dockstore</b>.<li> <b>Dockstore</b>'s initial approach to this challenge was incomplete since it relied solely on source control version numbers, which allowed users to delete workflows with one version number and then reuse the same version number for a different workflow.<li> Similarly, the original version of <b>Dockstore</b> allowed users to delete the Docker images that a particular workflow relied on and re-upload something different.<li><b>Dockstore</b> now collects information that detects and prevents sources of inconsistency while also enhancing security.<li> Workflow authors can trigger a feature known as a snapshot that makes a version of a workflow collected by <b>Dockstore</b> immutable, meaning it is no longer subject to change when synchronising with source control.<li> This functionality was introduced and exposed in <b>Dockstore</b>'s implementation of version 2.0 of the GA4GH TRS standard.<li>While implementing this feature, <b>Dockstore</b> added an integration that allows workflow authors to upload their immutable snapshot to Zenodo (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://about.zenodo.org/" ext-link-type="uri">https://about.zenodo.org/</ext-link>) which generates a digital object identifier (DOI), allowing users to cite workflows in publications (<xref rid="B19" ref-type="bibr"><u>19</u></xref>).<li>The <b>Dockstore</b> team contributes to a number of GA4GH standards and is the leading implementation for the Tool Registry Service (TRS) standard to aid in interoperability with other software projects in the genomics and health community.<li> As a part of the Cloud Work Stream, <b>Dockstore</b>'s implementation of TRS allows the platform to provide listings and search for tool and workflow information to its <italic toggle="yes">Launch with</italic> partners and, potentially, collaborating workflow registries (Figure <xref rid="F3" ref-type="fig"><u>3</u></xref>).<li> This standard became an official GA4GH standard in October 2019 and <b>Dockstore</b> implements two draft standards and the official 2.0.0 version of TRS.<li><b>Dockstore</b> contributed to the development of GA4GH TRS, an API that it uses for distributing workflows to <italic toggle="yes">Launch with</italic> partners and tools such as CWL&#x2019;s cwltool.<li>The <b>Dockstore</b> team also helped organize the GA4GH-DREAM Workflow Execution (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="10.6084/m9.figshare.6716063.v1" ext-link-type="doi">https://doi.org/10.6084/m9.figshare.6716063.v1</ext-link>) challenge in 2017.<li> As a part of this challenge participants contributed <b>Dockstore</b>-style workflows and manually ran them across a variety of environments.<li> <b>Dockstore</b> integrated the ability to register and search for checker workflows (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://docs.<b>dockstore</b>.org/en/develop/advanced-topics/checker-workflows.html" ext-link-type="uri">https://docs.<b>dockstore</b>.org/en/develop/advanced-topics/checker-workflows.html</ext-link>), workflows that test themselves across a variety of platforms.<li><b>Dockstore</b> owes much to its user community.<li>Organizations and Collections are new features that allow institutions, companies, grant funding agencies, and collaborations to highlight the tools and workflows that they either create themselves or find on <b>Dockstore</b> for re-use.<li>To create an Organization on <b>Dockstore</b>, a representative submits a request on behalf of the group for the <b>Dockstore</b> team to verify and review.<li>Another social process that the <b>Dockstore</b> team manages in order to provide more trust in the quality of workflows is verification.<li> Workflows are considered verified when the <b>Dockstore</b> team has contacted a third party (independent of the original authors) that can confirm that a workflow runs as intended with a provided parameter file and public input data.<li> For example, at the time this was written, a collection of COVID-19 workflows (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>) was highlighted in this fashion.<li> We have also upgraded to newer versions of Angular and Java (<xref rid="B22" ref-type="bibr"><u>22</u></xref>) while also implementing a robust suite of automated testing frameworks for continuous integration across all parts of <b>Dockstore</b>'s development stack.<li> This will include increasing the robustness and availability of our infrastructure as a whole while allowing <b>Dockstore</b> to interoperate with a wide variety of secure computing environments within North America that handle personal health information (PHI).<li>At the time of publication there have been 705 workflows and 240 tools published to the <b>Dockstore</b> library along with &gt;20 Organizations.<li> As a result, we wish to highlight a few partner collaborations and community groups that have contributed workflows or integrations with <b>Dockstore</b>.<li> As one of our <italic toggle="yes">Launch with</italic> partners, the NHLBI BioData Catalyst project uses <b>Dockstore</b> as its official repository for their created and preferred workflows.<li> Users can launch any valid CWL or WDL workflow, including those maintained directly by&#xA0;the&#xA0;BioData Catalyst community found on their Organization page at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/bdcatalyst" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/bdcatalyst</ext-link>.<li><b>Dockstore</b> is one of the platform components for the National Human Genome Research Institute's (NHGRI) Genomic Data Science Analysis, Visualization, and Informatics Lab-Space (AnVIL), a cloud resource also working on bringing users to a unified cloud computing environment with rich datasets, workflows, and scalable, shared computing resources (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://anvilproject.org" ext-link-type="uri">anvilproject.org</ext-link>).<li> <b>Dockstore</b> serves as the official repository for the tools and workflows that can be directly launched into AnVIL&#x2019;s Terra powered compute environment which also nests a Galaxy platform.<li>Since 2019, <b>Dockstore</b> and AnVIL has also been working to support the sharing and launching of Galaxy workflows into AnVIL.<li> AnVIL is compiling relevant workflows for their community using the <b>Dockstore</b> organization feature (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/anvil" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/anvil</ext-link>).<li><b>Dockstore</b>'s new Organizations and Collections features have enabled groups to showcase both their own workflows and the workflows that they use from others.<li> We would like to highlight a few that demonstrate a high degree of utility for researchers, are of topical interest, or demonstrate a special degree of integration with <b>Dockstore</b>.<li> These workflows and organizations are created by the community and are not the result of direct partnerships with <b>Dockstore</b> staff.<li> These workflows are automatically updated from their underlying source repositories using the <b>Dockstore</b> GitHub app.<li> Data Sciences Platform and can be found on <b>Dockstore</b> on the Broad Institute organization page.<li>Other prominent collections shared by the Broad Institute on <b>Dockstore</b> include the GATK best practices (<xref rid="B25" ref-type="bibr"><u>25</u></xref>) workflows (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/GATKWorkflows" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/GATKWorkflows</ext-link>).<li>Uniformly registered onto <b>Dockstore</b> using GitHub apps, the nf-core workflows (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) are another good example of a set of workflows that are automatically updated on <b>Dockstore</b>.The nf-core organization (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/nfcore" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/nfcore</ext-link>) of workflows represents a community effort to build a curated set of high quality workflows in the Nextflow language.<li> The nf-core workflows are a particularly good fit for the goals of <b>Dockstore</b> since all are validated using continuous integration to ensure reproducibility, are uniformly documented from a shared template, and encourage benchmarks on cloud environments.<li> The latter challenge was a particularly good fit since the goal of the exercise was to test that <b>Dockstore</b>-style workflows would function across multiple cloud platforms.<li> Subsequently, the <b>Dockstore</b> team has included 24 of these workflows in an automated testbed for evaluating new versions of the <b>Dockstore</b> platform.<li>As the <b>Dockstore</b> site has grown in popularity, the development team has also increased the ways in which we provide support for workflow developers, users of workflows and integration contributors.<li> The team has created a large corpus of publicly available tutorials, FAQs, and training videos, available at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://docs.<b>dockstore</b>.org/en/develop/posters-and-talks.html" ext-link-type="uri">https://docs.<b>dockstore</b>.org/en/develop/posters-and-talks.html</ext-link>.<li> This includes documentation covering all the core features of the site, as well as developer documentation and user-focused tutorials for authoring workflows with <b>Dockstore</b>.<li>Users can also post their questions to our built-in discussion board (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://discuss.<b>dockstore</b>.org/" ext-link-type="uri">https://discuss.<b>dockstore</b>.org/</ext-link>) or via issues opened on our open-source code repository on GitHub (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/<b>dockstore</b>/<b>dockstore</b>" ext-link-type="uri">https://github.com/<b>dockstore</b>/<b>dockstore</b></ext-link>).<li> The public forum provided by these venues allows both the <b>Dockstore</b> team and the community at large to collaborate towards and benefit from shared solutions.<li>An important part of <b>Dockstore</b> is not just to build the site, but to provide live tutorials about best practices while also keeping in touch with the community to understand how the field is developing.<li> The <b>Dockstore</b> team has given presentations at a large number of North American and International conferences such as our BOSC talks in 2017 and 2019.<li>In these trainings, we aim to provide users with a basic understanding of Docker, cloud technologies, and sharing workflows on <b>Dockstore</b>.<li>The <b>Dockstore</b> team has also conducted training sessions for supported research groups that have added workflows to <b>Dockstore</b>.<li> This includes groups sharing their tools and workflows with the community through <b>Dockstore</b> organization pages, such as the Large-Scale Gene by Environment analyses (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/LSGxE" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/LSGxE</ext-link>) and the VG team at UC Santa Cruz who are pioneering tools for analyzing graph genomes (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/organizations/UCSCGI/collections/variationgraphs" ext-link-type="uri">https://<b>dockstore</b>.org/organizations/UCSCGI/collections/variationgraphs</ext-link>).<li>We see a number of promising avenues for <b>Dockstore</b> to grow and evolve in the future while continuing to break down silos between communities.<li> Many of the new developments discussed here have not only enhanced the current <b>Dockstore</b> experience, but also lay the foundation to more robustly support and rapidly integrate new collaboration features, workflow languages and partner platforms.<li> <b>Dockstore</b> is also using the evolving GA4GH Workflow Execution Service (WES) standard to provide command-line users with a simplified way of launching workflows remotely onto novel cloud platforms via the <b>Dockstore</b> CLI (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="10.5281/zenodo.4536482" ext-link-type="doi">https://doi.org/10.5281/zenodo.4536482</ext-link>).<li><b>Dockstore</b> also plans to expand its support of reproducible, portable software into new areas to give users the ability to package and launch full fledged applications and services.<li>Importantly, <b>Dockstore</b> is prioritizing efforts to harden the security and trust of our platform and the content shared within it.<li> This, in turn, will provide information to users and platforms running the workflows that will enable choices to be made at the level of trust ascribed to content from <b>Dockstore</b>.<li><b>Dockstore</b> is an open source collaborative initiative available in the GitHub repository (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/<b>dockstore</b>/<b>dockstore</b>" ext-link-type="uri">https://github.com/<b>dockstore</b>/<b>dockstore</b></ext-link>) with supporting code nested under that GitHub organization.<li> The production site is hosted at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://<b>dockstore</b>.org/" ext-link-type="uri">https://<b>dockstore</b>.org/</ext-link> All published workflows and tools can be exported using the GA4GH TRS API.<li> [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32025007">32025007</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25001293">25001293</a>) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28379341">28379341</a>) [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29379135">29379135</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27401684">27401684</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31477884">31477884</a>) [8](<person-group person-group-type="author"><string-name><surname>Gruening</surname><given-names>B.</given-names></string-name>, <string-name><surname>Sallou</surname><given-names>O.</given-names></string-name>, <string-name><surname>Moreno</surname><given-names>P.</given-names></string-name>, <string-name><surname>da&#xA0;Veiga&#xA0;Leprevost</surname><given-names>F.</given-names></string-name>, <string-name><surname>M&#xE9;nager</surname><given-names>H.</given-names></string-name>, <string-name><surname>S&#xF8;ndergaard</surname><given-names>D.</given-names></string-name>, <string-name><surname>R&#xF6;st</surname><given-names>H.</given-names></string-name>, <string-name><surname>Sachsenberg</surname><given-names>T.</given-names></string-name>, <string-name><surname>O&#x2019;Connor</surname><given-names>B.</given-names></string-name>, <string-name><surname>Madeira</surname><given-names>F.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Recommendations for the packaging and containerizing of bioinformatics software</article-title>. <source>F1000Research</source>. <year>2018</year>; <volume>7</volume>:<fpage>742</fpage>.) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26978244">26978244</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27284183">27284183</a>) [14](<person-group person-group-type="author"><string-name><surname>Peter</surname><given-names>A.</given-names></string-name>, <string-name><surname>Michael&#xA0;R.</surname><given-names>C.</given-names></string-name>, <string-name><surname>Neboj&#x161;a</surname><given-names>T.</given-names></string-name>, <string-name><surname>Brad</surname><given-names>C.</given-names></string-name>, <string-name><surname>John</surname><given-names>C.</given-names></string-name>, <string-name><surname>Michael</surname><given-names>H.</given-names></string-name>, <string-name><surname>Andrey</surname><given-names>K.</given-names></string-name>, <string-name><surname>Dan</surname><given-names>L.</given-names></string-name>, <string-name><surname>Herv&#xE9;</surname><given-names>M.</given-names></string-name>, <string-name><surname>Maya</surname><given-names>N.</given-names></string-name></person-group><source>Common Workflow Language, v1.0</source>. <year>2016</year>; ) [15](<person-group person-group-type="author"><string-name><surname>Voss</surname><given-names>K.</given-names></string-name>, <string-name><surname>Gentry</surname><given-names>J.</given-names></string-name>, <string-name><surname>Auwera</surname><given-names>G.V.d.</given-names></string-name></person-group><article-title>Full-stack genomics pipelining with GATK4 + WDL + Cromwell</article-title>. <source>F1000Research</source>. <year>2017</year>; <volume>6</volume>:<fpage>1379</fpage>.) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26978244">26978244</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27284183">27284183</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [18](<person-group person-group-type="author"><string-name><surname>Gamma</surname><given-names>E.</given-names></string-name></person-group><source>Design Patterns: Elements of Reusable Object-oriented Software</source>. <year>1995</year>; <publisher-name>Pearson Education</publisher-name>.) [19](<person-group person-group-type="author"><string-name><surname>Smith</surname><given-names>A.M.</given-names></string-name>, <string-name><surname>Katz</surname><given-names>D.S.</given-names></string-name>, <string-name><surname>Niemeyer</surname><given-names>K.E.</given-names></string-name></person-group><article-title>Software citation principles</article-title>. <source>PeerJ Comput. Sci.</source><year>2016</year>; <volume>2</volume>:<fpage>e86</fpage>.) [22](<person-group person-group-type="author"><string-name><surname>Arnold</surname><given-names>K.</given-names></string-name>, <string-name><surname>Gosling</surname><given-names>J.</given-names></string-name></person-group><source>The Java Programming Language</source>. <year>1996</year>; <publisher-name>Addison-Wesley</publisher-name>.) [25](<person-group person-group-type="author"><string-name><surname>Van&#xA0;derAuwera</surname><given-names>G.</given-names></string-name>, <string-name><surname>Carneiro</surname><given-names>M.O.</given-names></string-name>, <string-name><surname>Hartl</surname><given-names>C.</given-names></string-name>, <string-name><surname>Poplin</surname><given-names>R.</given-names></string-name>, <string-name><surname>del&#xA0;Angel</surname><given-names>G.</given-names></string-name>, <string-name><surname>Levy-Moonshine</surname><given-names>A.</given-names></string-name>, <string-name><surname>Jordan</surname><given-names>T.</given-names></string-name>, <string-name><surname>Shakir</surname><given-names>K.</given-names></string-name>, <string-name><surname>Roazen</surname><given-names>D.</given-names></string-name>, <string-name><surname>Thibault</surname><given-names>J.</given-names></string-name><etal>et al</etal>.</person-group><article-title>From FastQ data to high-confidence variant calls: the genome analysis toolkit best practices pipeline</article-title>. <source>Curr. Protoc. Bioinformatics</source>. <year>2018</year>; <volume>43</volume>:<comment>doi:10.1002/0471250953.bi1110s43</comment>.) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33791701">33791701</a> <b>(2)</b> Freely accessible ready to use global infrastructure for SARS-CoV-2 monitoring<ul><li> To facilitate this process the workflows are readily available from the global repositories, <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link> and <ext-link ext-link-type="uri" xlink:href="https://workflowhub.eu">https://workflowhub.eu</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33758855">33758855</a> <b>(6)</b> DNA spike-ins enable confident interpretation of SARS-CoV-2 genomic data from amplicon-based sequencing<ul><li>We analyzed sequencing data on the Terra platform (app.terra.bio) using viral-ngs 2.1.1 with workflows that are publicly available on the <b>Dockstore</b> Tool Repository Service (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>).<li>Viral genomes were processed using the Terra platform (app.terra.bio) using viral-ngs 2.1.1 with workflows that are publicly available on the <b>Dockstore</b> Tool Repository Service (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33748785">33748785</a> <b>(4)</b> Analysis workflow to assess <italic>de novo</italic> genetic variants from human whole-exome sequencing<ul><li><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/warp/ExomeGermlineSingleSample:ExomeGermlineSingleSample_v2.2.0?tab=info" id="intref0020">https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/warp/ExomeGermlineSingleSample:ExomeGermlineSingleSample_v2.2.0?tab=info</ext-link>.<li><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/warp/JointGenotyping:JointGenotyping_v1.3.0?tab=info" id="intref0030">https://<b>dockstore</b>.org/workflows/github.com/broadinstitute/warp/JointGenotyping:JointGenotyping_v1.3.0?tab=info</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33655247">33655247</a> <b>(3)</b> A single-cell and spatial atlas of autopsy tissues reveals pathology and cellular targets of SARS-CoV-2<ul><li>SARS-CoV-2 genomes were assembled using the open-source software viral-ngs version 2.0.21, implemented on the Terra platform (app.terra.bio), using workflows that are publicly available via the <b>Dockstore</b> Tool Registry Service (<ext-link ext-link-type="uri" xlink:href="http://<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs"><b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33625229">33625229</a> <b>(4)</b> BioContainers Registry: searching bioinformatics and proteomics tools, packages, and containers<ul><li> Multiple services in Genomics such as <b>Dockstore</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/">https://<b>dockstore</b>.org/</ext-link>) or Agora (<ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/agora">https://github.com/broadinstitute/agora</ext-link>) distributed their tools and workflows using the specification <sup><xref rid="R16" ref-type="bibr"><u>16</u></xref></sup>.<li> The Restful API is an implementation of the GA4HG standard which enables compatibility with other resources and initiatives like <b>Dockstore</b>.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33325479">33325479</a> <b>(1)</b> aCLImatise: automated generation of tool definitions for bioinformatics workflows<ul><li> Such repositories exist for WDL (<ext-link ext-link-type="uri" xlink:href="https://github.com/biowdl/tasks">https://github.com/biowdl/tasks</ext-link>), Snakemake (<ext-link ext-link-type="uri" xlink:href="https://snakemake-wrappers.readthedocs.io/en/stable/">https://snakemake-wrappers.readthedocs.io/en/stable/</ext-link>), Nextflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/modules">https://github.com/nf-core/modules</ext-link>) and CWL (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-library/bio-cwl-tools">https://github.com/common-workflow-library/bio-cwl-tools</ext-link>), while registries such as <b>Dockstore</b> cater to multiple workflow languages simultaneously (<xref rid="btaa1033-B13" ref-type="bibr">O&#x2019;Connor <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33303686">33303686</a> <b>(2)</b> Phylogenetic analysis of SARS-CoV-2 in Boston highlights the impact of superspreading events<ul><li> All of the workflows named below are publicly available via the <b>Dockstore</b> Tool Registry Service (<b>dockstore</b>.org/organizations/BroadInstitute/collections/pgs).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33110075">33110075</a> <b>(2)</b> Globally altered epigenetic landscape and delayed osteogenic differentiation in H3.3-G34W-mutant giant cell tumor of bone<ul><li> Somatic SNVs and indels in matched tumor normal pairs were identified using the DKFZ core variant calling workflows of the ICGC Pan-cancer Analysis of Whole Genomes (PCAWG) project (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33097812">33097812</a> <b>(3)</b> Practical guide for managing large-scale human genome data in research<ul><li> A curated set of tool and workflow descriptions can be found at nf-core [<xref ref-type="bibr" rid="CR56"><u>56</u></xref>] and <b>DockStore</b> [<xref ref-type="bibr" rid="CR67"><u>67</u></xref>].<li> We can find a curated set of tool and workflow descriptions at BioWDL (<ext-link ext-link-type="uri" xlink:href="https://github.com/biowdl">https://github.com/biowdl</ext-link>) and <b>DockStore</b> [<xref ref-type="bibr" rid="CR67"><u>67</u></xref>].<li> A curated set of tool and workflow descriptions can be found at Common Workflow Library (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-library">https://github.com/common-workflow-library</ext-link>) and <b>DockStore</b> [<xref ref-type="bibr" rid="CR67"><u>67</u></xref>].<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33028839">33028839</a> <b>(4)</b> Framework for quality assessment of whole genome cancer sequences<ul><li> To enable others to use our approach, there is a Docker container, which can be accessed at <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/jwerner_dkfz/pancanqc:1.2.2">https://<b>dockstore</b>.org/containers/quay.io/jwerner_dkfz/pancanqc:1.2.2</ext-link>.<li>We have collected the custom made scripts and published software together in a Docker container, which can be found at: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/jwerner_dkfz/pancanqc:1.2.2">https://<b>dockstore</b>.org/containers/quay.io/jwerner_dkfz/pancanqc:1.2.2</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32616555">32616555</a> <b>(2)</b> A phase II single-arm study of pembrolizumab with enzalutamide in men with metastatic castration-resistant prostate cancer progressing on enzalutamide alone<ul><li>WES reads were aligned against the GrCH37d5 genome using the Sanger cppmap workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/cancerit/<b>dockstore</b>-cgpmap">https://github.com/cancerit/<b>dockstore</b>-cgpmap</ext-link>) with realignment around indels and base recalibration performed using the Open Genomics GATK cocleaning workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/OpenGenomics/gatk-cocleaning-tool">https://github.com/OpenGenomics/gatk-cocleaning-tool</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32453635">32453635</a> <b>(11)</b> Global Alliance for Genomics and Health Meets Bioconductor: Toward Reproducible and Agile Cancer Genomics at Cloud Scale<ul><li>In this article, we examine an approach to combining the GA4GH Tool Registry Services (TRS) and Workflow Execution Services (WES) concepts, as implemented in <b>dockstore</b>.org<sup><xref rid="B2" ref-type="bibr"><u>2</u></xref></sup> and the Broad Institute Cromwell workflow execution engine, with data, annotation, and software resources and practices developed in the Bioconductor project.<li> <b>Dockstore</b> is used to register code and container for durable public access and manage the conveyance of the workflow to the Terra platform, where the Cromwell workflow execution service manages the creation of the computational and storage environments where the analysis is carried out.<li> The GitHub repository for this package<sup><xref rid="B14" ref-type="bibr"><u>14</u></xref></sup> is used to register the WDL programs in <b>dockstore</b>.org.<li> Upon publishing the workflow at <b>dockstore</b>.org, a user can activate a control labeled Run in Terra.<li> This archive was registered in <b>dockstore</b>.org and made available for use in the Broad Institute Terra environment via the <b>Dockstore</b> Application Programming Interface.<li> A thorough tutorial on using <b>Dockstore</b> for general workflow management and publication is provided in the video at <ext-link ext-link-type="uri" xlink:href="https://youtu.be/RYHUX9jGx24">https://youtu.be/RYHUX9jGx24</ext-link>.<li>This analysis could have been carried out completely in R with Bioconductor or in any resource supporting statistical analysis of TCGA data, such as any of the Cancer Genomics Cloud projects.<sup><xref rid="B16" ref-type="bibr"><u>16</u></xref></sup> Adoption of the container plus workflow protocol of <b>dockstore</b>.org provided convenient access to Google Compute Platform facilities mediated through the Broad Institute app.terra.bio.<li>The conveniences of <b>dockstore</b> plus Terra for expressing and executing complex analytic workflows, fully reproducibly, are compelling.<li> By design, the container plus workflow registered on <b>dockstore</b>.org will have predictable behaviors and thus would not seem to need continuous testing.<li> We describe experimentation with the GA4GH TRS and WES frameworks, with deployments at <b>dockstore</b>.org and execution in the Broad Institute app.terra.bio.<li> [2](          <person-group person-group-type="author">            <name>              <surname>O&#x2019;Connor</surname>              <given-names>BD</given-names>            </name>            <name>              <surname>Yuen</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Chung</surname>              <given-names>V</given-names>            </name>            <etal>et al</etal>          </person-group>          <article-title>The Dockstore: Enabling modular, community-focused sharing of Docker-based genomics tools and workflows</article-title>          <source>F1000 Res</source>          <volume>6</volume>          <fpage>52</fpage>          <year>2017</year>        ) [14](          <comment>Carey VJ: BiocOncoTK: Bioconductor components for general cancer genomics. <ext-link ext-link-type="uri" xlink:href="https://github.com/vjcitn/BiocOncoTK">https://github.com/vjcitn/BiocOncoTK</ext-link></comment>        ) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30691868">30691868</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32024823">32024823</a> <b>(1)</b> High-coverage whole-genome analysis of 1220 cancers reveals hundreds of genes deregulated by rearrangement-mediated <italic>cis</italic>-regulatory alterations<ul><li> The PCAWG workflows are also available as Docker images through <b>Dockstore</b> enabling researchers to replicate the steps involved in the data assembly<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> [11](The ICGC/TCGA Pan-Cancer Analysis of Whole Genomes Consortium. Pan-cancer analysis of whole genomes. <italic>Nature</italic>10.1038/s41586-020-1969-6 (2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32025007">32025007</a> <b>(4)</b> Pan-cancer analysis of whole genomes<ul><li>The core alignment, somatic variant-calling, quality-control and variant consensus-generation pipelines used by PCAWG have each been packaged into portable cross-platform images using the <b>Dockstore</b> system<sup><xref ref-type="bibr" rid="CR84"><u>84</u></xref></sup> and released under an Open Source licence that enables unrestricted use and redistribution.<li> All PCAWG <b>Dockstore</b> images are available to the public at <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/organizations/PCAWG/collections/PCAWG">https://<b>dockstore</b>.org/organizations/PCAWG/collections/PCAWG</ext-link>.<li> [84](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32019919">32019919</a> <b>(1)</b> Organizing genome engineering for the gigabase scale<ul><li> Current workflow tools include both general tools, such as the Common Workflow Language (CWL)<sup><xref ref-type="bibr" rid="CR100"><u>100</u></xref></sup>, the <b>Dockstore</b><sup><xref ref-type="bibr" rid="CR101"><u>101</u></xref></sup> and MyExperiment<sup><xref ref-type="bibr" rid="CR102"><u>102</u></xref></sup> sharing environments, and the PROV ontology for tracking information provenance<sup><xref ref-type="bibr" rid="CR103"><u>103</u></xref></sup> (which is already being applied to link design&#x2013;build&#x2013;test&#x2013;learn cycles in SBOL<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>).<li> [100](Amstutz, P.&#xA0; et al.&#xA0; Common Workflow Language, v1.0 <ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/</ext-link> (2016).) [101](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [102](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20501605">20501605</a>) [103](Missier, P., &#xA0; Belhajjame,&#xA0;K. &amp;&#xA0; Cheney, J.&#xA0; The W3C PROV family of specifications for modelling provenance metadata. In <italic>Proc. 16th International Conference on Extending Database Technology</italic> 773&#x2013;776 (ACM, 2013).) [50](Cox, R. S.&#xA0; et al.&#xA0; Synthetic biology open language (SBOL) version 2.2. 0. <italic>J. Integr. Bioinform.</italic><bold>15</bold>, 20180001 (2018).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31609649">31609649</a> <b>(2)</b> Germline <italic>GPR161</italic> Mutations Predispose to Pediatric Medulloblastoma<ul><li>Sequencing data were processed using a standardized alignment and somatic variant calling pipeline, which was developed in the context of the ICGC Pan-Cancer Analysis of Whole Genomes (PCAWG) project (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31597973">31597973</a> <b>(1)</b> The human body at cellular resolution: the NIH Human Biomolecular Atlas Program<ul><li> These open source portals will use recognized standards and be interoperable with other platforms, such as the HCA Data Coordination Platform, making it possible to readily add, update, and use new software modules (for example, as with <b>Dockstore</b><sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref></sup> and Toil<sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup>).<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32382696">32382696</a> <b>(5)</b> Organizing and running bioinformatics hackathons within Africa: The H3ABioNet cloud computing experience<ul><li>** Corresponding docker containers are available at:<ext-link ext-link-type="uri" xlink:href="https://quay.io/organization/h3abionet_org">https://quay.io/organization/h3abionet_org</ext-link> and<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/h3abionet/h3agatk">https://<b>dockstore</b>.org/workflows/h3abionet/h3agatk</ext-link>.<li>The four pipelines are available publicly via H3ABioNet&#x2019;s GitHub organization page<ext-link ext-link-type="uri" xlink:href="https://github.com/h3abionet">https://github.com/h3abionet</ext-link> in the following repositories: (h3agwas, chipimputation, h3agatk and h3abionet16S) as well as container images on quay.io at<ext-link ext-link-type="uri" xlink:href="https://quay.io/organization/h3abionet_org">quay.io/organization/h3abionet_org</ext-link> and <b>dockstore</b> at:<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/h3abionet/h3agatk">https://<b>dockstore</b>.org/workflows/h3abionet/h3agatk</ext-link>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31321430">31321430</a> <b>(3)</b> CWL-Airflow: a lightweight pipeline manager supporting Common Workflow Language<ul><li> Researchers using CWL are then able to deposit descriptions of their tools and workflows into a repository (e.g., <ext-link ext-link-type="uri" xlink:href="http://<b>dockstore</b>.org/"><b>dockstore</b>.org</ext-link>) upon publication, thus making their analyses reusable by others.<li> Additional collections of tools are available in Rabix Composer [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], a graphical CWL editor from Seven Bridges and at the <b>Dockstore</b> [<xref rid="bib25" ref-type="bibr"><u>25</u></xref>].<li> [7](<person-group person-group-type="author"><name name-style="western"><surname>Kaushik</surname><given-names>G</given-names></name>, <name name-style="western"><surname>Ivkovic</surname><given-names>S</given-names></name>, <name name-style="western"><surname>Simonovic</surname><given-names>J</given-names></name>, <etal>et al</etal>.</person-group><article-title>RABIX: an open-source workflow executor supporting recomputability and interoperability of workflow descriptions</article-title>. <source>Pac Symp Biocomput</source>. <year>2016</year>;<volume>22</volume>:<fpage>154</fpage>&#x2013;<lpage>65</lpage>.) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31222200">31222200</a> <b>(2)</b> Bioportainer Workbench: a versatile and user-friendly system that integrates implementation, management, and use of bioinformatics resources in Docker environments<ul><li>Thus, several bioinformatics-related PaaS and SaaS initiatives based on Docker virtualization systems have been recently developed, such as BioShaDock [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], AlgoRUN [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], GUIdock [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], <b>Dockstore</b> [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], and BioContainers [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], among others.<li> For example, Docker image repositories such as BioShaDock [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>] and <b>Dockstore</b> [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] provide several bioinformatics software programs within Docker containers, but local installation of images still requires adjustments to ensure their full operation, such as the export of network service ports and configuration of data volumes, among other procedures, which are often unclear to the final user owing to lack of proper documentation in such repositories.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26913191">26913191</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27153722">27153722</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27045593">27045593</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28379341">28379341</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26913191">26913191</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30926794">30926794</a> <b>(2)</b> Genomic and transcriptomic changes complement each other in the pathogenesis of sporadic Burkitt lymphoma<ul><li>Somatic SNVs and indels in matched tumor normal pairs were identified using the DKFZ core variant calling workflows of the ICGC PCAWG project (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30849372">30849372</a> <b>(24)</b> Characterizing Mutational Signatures in Human Cancer Cell Lines Reveals Episodic APOBEC Mutagenesis<ul><li>                    <hr/>                  </td>                </tr>                <tr>                  <td>Mapping workflow</td>                  <td>Cancer, Aging and Somatic Mutation group, Wellcome Sanger Institute</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap" id="intref0015">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap</ext-link>                  .<li>                    <hr/>                  </td>                </tr>                <tr>                  <td>Mapping workflow</td>                  <td>Cancer, Aging and Somatic Mutation group, Wellcome Sanger Institute</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap" id="intref0015">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap</ext-link>                  </td>                </tr>                <tr>                  <td>Mutation calling workflows</td>                  <td>Cancer, Aging and Somatic Mutation group, Wellcome Sanger Institute</td>                  <td>Whole-exome: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwxs" id="intref0020">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwxs</ext-link><break/>Whole-genome: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs" id="intref0025">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs</ext-link>.<li><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap" id="intref0155">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap</ext-link>.<li><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwxs" id="intref0160">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwxs</ext-link>.<li><ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs" id="intref0165">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs</ext-link>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30664630">30664630</a> <b>(3)</b> Enhancer hijacking activates oncogenic transcription factor NR4A3 in acinic cell carcinomas of the salivary glands<ul><li> The workflow can be accessed via the <b>Dockstore</b> webpage: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30486782">30486782</a> <b>(5)</b> Developing reproducible bioinformatics analysis workflows for heterogeneous computing environments to support African genomics<ul><li> The workflow is available on the CWL workflows library on GitHub: <ext-link ext-link-type="uri" xlink:href="https://github.com/h3abionet/h3agatk">https://github.com/h3abionet/h3agatk</ext-link>and the container at <b>Dockstore</b>: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/h3abionet/h3agatk">https://<b>dockstore</b>.org/workflows/github.com/h3abionet/h3agatk</ext-link>.<li>We have registered the Docker images for the pipelines at: <ext-link ext-link-type="uri" xlink:href="https://quay.io/organization/h3abionet_org">https://quay.io/organization/h3abionet_org</ext-link> and <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/workflows/github.com/h3abionet/h3agatk">https://<b>dockstore</b>.org/workflows/github.com/h3abionet/h3agatk</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30429477">30429477</a> <b>(4)</b> Whole genome sequencing puts forward hypotheses on metastasis evolution and therapy in colorectal cancer<ul><li> Sequencing reads were mapped and aligned using the DKFZ alignment workflow from ICGC Pan-Cancer Analysis of Whole Genome projects [<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-bwa-mem-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-bwa-mem-workflow</ext-link>].<li> SNVs were initially called using the DKFZ SNV and indel calling workflow from ICGC Pan-Cancer Analysis of Whole Genome projects [<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>]<sup><xref ref-type="bibr" rid="CR54"><u>54</u></xref>,<xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup>.<li> [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23817572">23817572</a>) [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22832583">22832583</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30103702">30103702</a> <b>(4)</b> Unsupervised correction of gene-independent cell responses to CRISPR-Cas9 targeting<ul><li> The Python implementation is dockerized making it platform independent and usable in cloud environments (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-pycrisprcleanr">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-pycrisprcleanr</ext-link>).</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28945760">28945760</a> <b>(8)</b> Appraising the relevance of DNA copy number loss and gain in prostate cancer using whole genome DNA sequence data<ul><li> An updated version of this pipeline is available as a Docker image (Alignment: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpmap</ext-link>; Variant-calling: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs">https://<b>dockstore</b>.org/containers/quay.io/wtsicgp/<b>dockstore</b>-cgpwgs</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28810143">28810143</a> <b>(4)</b> Genomic Evolution of Breast Cancer Metastasis and Relapse<ul><li>                    <hr/>                  </td>                </tr>                <tr>                  <td>The Sanger&#x2019;s Cancer Genome Project core somatic calling workflow from<break/>the ICGC PanCancer Analysis of Whole Genomes (PCAWG) project</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-sanger-cgp-workflow" id="intref0020">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-sanger-cgp-workflow</ext-link>                  .<li> This, together with documentation, can be accessed from <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-sanger-cgp-workflow" id="intref0075">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-sanger-cgp-workflow</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28726821">28726821</a> <b>(3)</b> The whole-genome landscape of medulloblastoma subtypes<ul><li> The workflow is available on the <b>Dockstore</b> webpage: <ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28659176">28659176</a> <b>(2)</b> ISOWN: accurate somatic mutation identification in the absence of normal tissue controls<ul><li> We used variants called using the DKFZ variant-calling pipeline (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow">https://<b>dockstore</b>.org/containers/quay.io/pancancer/pcawg-dkfz-workflow</ext-link>) for the ICGC/TCGA PanCancer Analysis of Whole Genomes Project (<ext-link ext-link-type="uri" xlink:href="https://dcc.icgc.org/pcawg">https://dcc.icgc.org/pcawg</ext-link>).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a> <b>(112)</b> The <b>Dockstore</b>: enabling modular, community-focused sharing of Docker-based genomics tools and workflows<ul><li>The <b>Dockstore</b> project has its roots in the large-scale ICGC PanCancer Analysis of Whole Genomes (PCAWG;<ext-link ext-link-type="uri" xlink:href="https://dcc.icgc.org/pcawg">https://dcc.icgc.org/pcawg</ext-link>) cancer genomics project, which necessitated the creation of highly portable and self-contained computational tools<sup><xref rid="ref-1" ref-type="bibr"><u>1</u></xref></sup>.<li> Our creation, the <b>Dockstore</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link>), generalizes the PCAWG approach in an easy-to-use web application that any tool developer or tool end user can utilize.<li><b>Dockstore</b>&#x2019;s key innovation is its bridging of Docker image registries with a new, standardized approach to describing tools inside images.<li> <b>Dockstore</b> also supports linking multiple tools together using CWL or WDL workflows; these multi-image workflows can then be registered on the site and used as building blocks to create more complex systems.<li> The result is that <b>Dockstore</b>-based tools and workflows can be programmatically addressed and executed, enabling a new level of modularity, automation and integration.<li>In addition to providing a mechanism to bring together Docker-based tools and their corresponding machine-readable descriptors, the <b>Dockstore</b> provides a compelling and useful web-based interface, an instance of which is hosted at<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link>.<li> This allows it to serve two communities: developers who want to register and share their tools through <b>Dockstore</b>, and users wishing to find genomics tools packaged in Docker and ready to execute in their own systems (<xref ref-type="fig" rid="f1"><u>Figure 1</u></xref>).<li> The <b>Dockstore</b> web application provides a full host of capabilities for these two types of users, including registering new Docker images and descriptors, searching for tools others have registered, and assisting users in executing tools on any platform that supports Docker.<li> The <b>Dockstore</b> also provides a command line interface for power users who want to script and automate their use of <b>Dockstore</b>.<li>Use cases for <b>Dockstore</b>.<li>Developers can use <b>Dockstore</b> to register Docker images built by, or uploaded to, Quay.io and DockerHub with CWL/WDL machine- and human-readable descriptors from GitHub or Bitbucket.<li>Finally, <b>Dockstore</b> is supported by the Global Alliance for Genomics and Health (GA4GH) organization<sup><xref rid="ref-3" ref-type="bibr"><u>3</u></xref></sup>.<li> Its implementation in <b>Dockstore</b>, and other sites, is a key goal of the standards effort and will allow for federated searches across tool registries that implement the GA4GH API.<li>The <b>Dockstore</b> implementation can be divided into four facets: a tool and workflow registration process aimed at authors, a RESTful web API used to power the site, a web application that uses this API, and, finally, a command line utility that interacts with, and launches, tools and workflows present on <b>Dockstore</b>.<li><bold><italic>Tool registration process.</italic></bold> The <b>Dockstore</b> does not itself act as a Docker image host or provide services to build Docker images automatically from source.<li> Instead, <b>Dockstore</b> provides a registry to link Docker-based tools hosted on Quay.io or DockerHub with tool metadata described in CWL or WDL and checked into a source control repository at GitHub or Bitbucket.<li> This allows <b>Dockstore</b> to be lightweight and focus on the utility of presenting tools and workflows to the community through a searchable web application.<li>For developers adding tools to <b>Dockstore</b>, we recommend a method in which Docker-based tools are built automatically from public source repositories to maximize transparency and utility to the community.<li> the <b>Dockstore</b> web API allows tool developers to register one or more releases of a particular tool with a simple click in the web application.<li> The <b>Dockstore</b> web API gathers descriptors and Dockerfiles via delegated OAuth authorization<sup><xref rid="ref-4" ref-type="bibr"><u>4</u></xref></sup>.<li> While it is possible to use DockerHub in place of Quay.io, the lack of a public DockerHub API makes integration into <b>Dockstore</b> less streamlined and introduces manual steps.<li>Docker images and tool descriptors or workflows in WDL/CWL are registered with <b>Dockstore</b>.<li>In addition to the recommended automated build process, <b>Dockstore</b> offers alternative manual processes that give developers greater control over how their tools are registered.<li> For example, <b>Dockstore</b> supports tools built outside of the normal DockerHub/Quay.io automated build process (<xref ref-type="fig" rid="f2"><u>Figure 2C</u></xref>).<li> This allows developers to build Docker-based tools themselves, possibly for performance reasons, and then push the finished image to DockerHub or Quay.io for inclusion in <b>Dockstore</b>.<li> For these reasons we recommend the fully automated approach to developers sharing tools on <b>Dockstore</b>.<li> Instead, they reference multiple tools (ideally registered using the <b>Dockstore</b> process).<li> It can then be found and registered in the <b>Dockstore</b> (<xref ref-type="fig" rid="f2"><u>Figure 2B</u></xref>).<li><bold><italic>RESTful application programming interface (API).</italic></bold> The <b>Dockstore</b> web and command line interfaces are driven by a RESTful web API (DOI:<ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.5281/zenodo.154185">10.5281/zenodo.154185</ext-link>).<li> This API includes endpoints that conform to the emerging GA4GH Tool Registry API standard (<xref ref-type="fig" rid="f3"><u>Figure 3</u></xref>), allowing for multiple tools to interoperate with <b>Dockstore</b> and other sites that implement the standard.<li> In addition, the <b>Dockstore</b> API includes extended, non-standard endpoints that are used for additional features implemented on the site, such as user authentication, integration with third-party services, like Github, and tool labelling.<li><bold><italic>Web application interface.</italic></bold> The <b>Dockstore</b> provides a simple-to-use web application that allows developers to register and manage tools and workflows while enabling end users to find and execute them.<li> The <b>Dockstore</b> web application also provides social features.<li>The web interface for the<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link> site.<li>(<bold>A</bold>) The main page lists the most recent additions to <b>Dockstore</b> and allows for users to search and login.<li> (<bold>B</bold>) A developer can easily publish their tools in <b>Dockstore</b> after logging in and linking to accounts.<li>Developers wishing to share their tools on <b>Dockstore</b> can log in using GitHub as an identity provider.<li> Once linked, the developer is prompted to download and configure the <b>Dockstore</b> command line tool and is presented with an API token to be used with the underlying <b>Dockstore</b> web service.<li> Developers wishing to build on top of <b>Dockstore</b> can use this token to authenticate against the <b>Dockstore</b> API and use it to make secure requests to GitHub, Bitbucket, and Quay.io.<li> and not publicly visible in <b>Dockstore</b>.<li> Valid images (images that can be linked to a WDL/CWL descriptor) can be toggled to &#x201C;published&#x201D;, making them visible to any <b>Dockstore</b> user (<xref ref-type="fig" rid="f4"><u>Figure 4B</u></xref>).<li> the particular Docker tool, causing <b>Dockstore</b> to re-query Quay.io and GithHub/Bitbucket to ensure the latest build image and associated descriptors are present in the system.<li> For Docker images hosted in DockerHub, a more labor intensive process is needed to manually register in <b>Dockstore</b> given the current lack of publically available DockerHub API.<li><bold><italic>Command line interface.</italic></bold> The <b>Dockstore</b> command line utility provides the registration and search functionality offered by the web interface, and additionally provides assistance for file provisioning and local execution of tools and workflows registered within the system.<li> This functionality allows <b>Dockstore</b> users to find workflows and tools of interest and quickly execute them using a completely standardized approach.<li> Since every tool and workflow in <b>Dockstore</b> is described with CWL or WDL, the local execution of these tools is always done using the same command line and same parameterization process, greatly simplifying the learning curve for using any particular tool or workflow from <b>Dockstore</b>.<li> The parameterization of the Docker-based tool or workflow is encoded in a JSON document, a template of which can be created with the <b>Dockstore</b> command line.<li> The <b>Dockstore</b> command line supports file provisioning, since provisioning of files is beyond the scope of the specifications for CWL and WDL.<li> The ability to execute tools from <b>Dockstore</b> is of particular value for development and user evaluation purposes and the command line supports a batch processing mode as well.<li> We anticipate that other systems, both open source and commercial, and through a standard API, will ultimately enable larger-scale concurrent analysis with <b>Dockstore</b>-registered workflows and tools.<li><b>Dockstore</b> follows best practices for software development, including using source control through GitHub, continuous integration testing with Travis CI (<ext-link ext-link-type="uri" xlink:href="https://travis-ci.org/">https://travis-ci.org/</ext-link>), testing coverage prediction with Coveralls (<ext-link ext-link-type="uri" xlink:href="https://coveralls.io/">https://coveralls.io/</ext-link>), and community engagement with Gitter (<ext-link ext-link-type="uri" xlink:href="https://gitter.im/ga4gh/<b>dockstore</b>">https://gitter.im/ga4gh/<b>dockstore</b></ext-link>).<li> The <b>Dockstore</b> web application,<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org">https://<b>dockstore</b>.org</ext-link>, will remain an open and free site for users to register their public tool images and workflows.<li> As an open source project<sup><xref rid="ref-5" ref-type="bibr"><u>5</u></xref></sup>, we also encourage others to customize and install instances of <b>Dockstore</b> (both the UI and RESTful web API) at their own sites.<li>Since <b>Dockstore</b> is designed to use Quay.io and Dockerhub as a backend, the server resources necessary for running it are modest.<li> <b>Dockstore</b> has been successfully installed on Ubuntu 14.04 and, while other distributions are possible, we currently only recommend this one.<li><b>Dockstore</b> is a general platform for sharing tools and workflows, so the potential use cases the site supports are quite varied.<li> Using <b>Dockstore</b> necessitates that a tool or workflow author uses source control, leverages a Docker build/hosting service, and provides a standardized description of how to invoke the tool/workflow.<li> As an outcome of registering their tools/workflows on <b>Dockstore</b>, developers can take advantage of the underlying GA4GH Tool Registry API standard.<li> This means a growing number of services can find and launch tools from <b>Dockstore</b>, providing additional motivation for developers to redistribute tools and workflows using the site.<li>For individual users, <b>Dockstore</b> is a catalogue of available tools and workflows that all work in a consistent and reliable way.<li> A user can use <b>Dockstore</b> to find tools and workflows of interest to their research and leverage the standard descriptor format, in either WDL or CWL, to provide clear documentation on how to execute the tool/workflow.<li> Furthermore, the inclusion of known-good test JSON documents on <b>Dockstore</b> provide key examples of inputs and expected outputs, something of importance in the bioinformatics community given the variability in file standards (<xref ref-type="other" rid="SM1"><u>Supplementary File 3</u></xref>).<li> In addition to providing clear usage information and example inputs/outputs, individual users can leverage <b>Dockstore</b>-based tools and workflows in a growing collection of execution environments that understand the GA4GH Tool Registry API standard supported by <b>Dockstore</b>.<li> Since <b>Dockstore</b> was inspired directly from the lessons learned in the highly-distributed PCAWG project, we feel other large-scale, distributed analysis efforts, such as the upcoming ICGCmed (<ext-link ext-link-type="uri" xlink:href="https://icgcmed.org">https://icgcmed.org</ext-link>) project, will be able to benefit from <b>Dockstore</b> infrastructure.<li> In these projects, <b>Dockstore</b>, or sites supporting the GA4GH Tool Registry standard, provide a standardized way to develop and share portable tools and workflows.<li> Developers and researchers creating analytical tools and workflows for these projects can build, test, and distribute these tools/workflows using <b>Dockstore</b>.<li> For those tasked with executing <b>Dockstore</b>-based tools and workflows at scale, their inherent consistency means execution environments shown to run a given <b>Dockstore</b>-based tool or workflow are very likely to be able to run any other <b>Dockstore</b>-based tool or workflow.<li> This separation of concerns, through the consistency provided by <b>Dockstore</b> and portability provided by Docker and standards like CWL and WDL, mean large-scale projects are much more likely to be successful in their distributed computing goals than a model where every tool and workflow needs to be validated across all compute environments used by the distributed project.<li>The <b>Dockstore</b> is unique in its synthesis of programmatically friendly tool descriptors (WDL or CWL) with Docker images hosted on high-quality commercial services.<li>In contrast with generic Docker repositories, such as DockerHub, the <b>Dockstore</b> provides mechanisms to interpret the contents of one or more Docker images, link them together, and execute them on a variety of HPC and cloud environments without modification.<li> Hence, the <b>Dockstore</b> approach combines the cloud-based flexibility and elasticity of Docker with the modularity of tool repositories like Galaxy Toolshed.<li>A number of existing projects, such as BioShaDock<sup><xref rid="ref-7" ref-type="bibr"><u>7</u></xref></sup>, Bioboxes<sup><xref rid="ref-8" ref-type="bibr"><u>8</u></xref></sup>, and BioDocker (<ext-link ext-link-type="uri" xlink:href="http://biodocker.org">http://biodocker.org</ext-link>), focus on encapsulating bioinformatics tools in Docker images in a way similar to <b>Dockstore</b>.<li> BioShaDock is the most similar to <b>Dockstore</b> and provides a fully controlled environment to build and publish bioinformatics software.<li> <b>Dockstore</b>, like these existing efforts, encourages the use of Docker as a technology for packaging and distributing bioinformatics tools.<li> However, unlike Bioboxes and BioDocker, <b>Dockstore</b> has a heavy focus on CWL/WDL in order to collect Docker images that can be used as part of larger workflows.<li> Unlike BioShaDock, <b>Dockstore</b> is a lightweight registry that focuses on deep integration with commercial source code providers and the Quay.io Docker image registry.<li>In the future, it should be possible to leverage multiple open source user interfaces (such as Galaxy) and commercial platforms (such as Seven Bridges Genomics, DNAnexus, DNAstack, and others) to provide a friendly environment for finding, combining, and executing <b>Dockstore</b>-based tools and workflows.<li> The <b>Dockstore</b> is the first implementation of this emerging standard.<li>Future features of <b>Dockstore</b> will include the support of testing frameworks and execution environments.<li> <b>Dockstore</b> entries that are validated to come from trusted sources.<li> This will complement private registry support in <b>Dockstore</b> in order to facilitate sharing Docker-based tools and workflows with a select set of collaborators.<li> A long term evolution of the <b>Dockstore</b> site will include a central registry index, complete with faceted search, for querying across the network of GA4GH-compliant tool registries described previously.<li> <b>Dockstore</b> will also integrate with the related and complementary GA4GH Workflow and Task Execution API standards currently in development, enabling the use of compute resources to run <b>Dockstore</b>-based tools and workflows through standardized APIs.<li> <b>Dockstore</b>&#x2019;s support of these features, and emerging standards, will support future successors to large scale, distributed analysis projects such as PCAWG.<li> This may include efforts, such as the ICGCmed (<ext-link ext-link-type="uri" xlink:href="https://icgcmed.org/">https://icgcmed.org/</ext-link>) and future DREAM challenges (<ext-link ext-link-type="uri" xlink:href="http://dreamchallenges.org/">http://dreamchallenges.org/</ext-link>), where <b>Dockstore</b> can enable the seamless interchange and execution of software tools across a variety of computer environments.<li>Software available at:<ext-link ext-link-type="uri" xlink:href="https://<b>dockstore</b>.org/">https://<b>dockstore</b>.org/</ext-link>.<li><b>Dockstore</b> source code available from the Global Alliance for Genomics and Health (GitHub):<ext-link ext-link-type="uri" xlink:href="https://github.com/ga4gh/<b>dockstore</b>">https://github.com/ga4gh/<b>dockstore</b></ext-link> (web UI:<ext-link ext-link-type="uri" xlink:href="https://github.com/ga4gh/<b>dockstore</b>-ui">https://github.com/ga4gh/<b>dockstore</b>-ui</ext-link>).<li>Archived source code for <b>Dockstore</b> 1.0 release:<ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/154185">https://zenodo.org/record/154185</ext-link>, DOI:<ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.5281/zenodo.154185">10.5281/zenodo.154185</ext-link><sup><xref rid="ref-9" ref-type="bibr"><u>9</u></xref></sup>.<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26156357">26156357</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26526696">26526696</a>) [4](<person-group person-group-type="author"><name><surname>Barry</surname><given-names>L</given-names></name></person-group>:<article-title>Oauth web authorization protocol.</article-title><source><italic>IEEE Internet Computing.</italic></source><year>2012</year>;<volume>16</volume>(<issue>1</issue>):<fpage>74</fpage>&#x2013;<fpage>77</fpage><pub-id pub-id-type="doi">10.1109/MIC.2012.11</pub-id>) [5](<person-group person-group-type="author"><name><surname>Thomas</surname><given-names>FR</given-names></name></person-group>:<article-title>Architectural styles and the design of network-based software architectures.</article-title>University of California, Irvine.<year>2000</year><ext-link ext-link-type="uri" xlink:href="http://www.ics.uci.edu/~fielding/pubs/dissertation/fielding_dissertation.pdf">Reference Source</ext-link>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26913191">26913191</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26473029">26473029</a>) [9](<person-group person-group-type="author"><name><surname>Yuen</surname><given-names>D</given-names></name><name><surname>Duncan</surname><given-names>A</given-names></name><name><surname>Liu</surname><given-names>V</given-names></name><etal/></person-group>:<article-title>ga4gh/dockstore: 1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.5281/zenodo.154185</pub-id>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><name><surname>Yuen</surname><given-names>D</given-names></name><name><surname>Duncan</surname><given-names>A</given-names></name><name><surname>Liu</surname><given-names>V</given-names></name><etal/></person-group>:<article-title>ga4gh/dockstore: 1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.5281/zenodo.154185</pub-id>"><person-group person-group-type="author"><name><surname>Yuen</surname><given-names>D</given-names></name><name><surname>Duncan</surname><given-names>A</given-names></name><name><surname>Liu</surname><given-names>V</given-names></name><etal/></person-group>:<article-title>ga4gh/dockstore: 1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.5281/zenodo.154185</pub-id></a> <b>(2)</b> The <b>Dockstore</b>: enabling modular, community-focused sharing of Docker-based genomics tools and workflows<ul><li> I have a minor comment for the on-line tool and the presented workflows (https://<b>dockstore</b>.org/search-workflows).<li> https://<b>dockstore</b>.org/workflows/ICGC-TCGA-PanCancer/wdl-pcawg-bwa-mem-workflow.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><name><surname>Yuen</surname><given-names>D</given-names></name><name><surname>Duncan</surname><given-names>A</given-names></name><name><surname>Liu</surname><given-names>V</given-names></name><etal/></person-group>:<article-title>ga4gh/dockstore: 1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.5281/zenodo.154185</pub-id>"><person-group person-group-type="author"><name><surname>Yuen</surname><given-names>D</given-names></name><name><surname>Duncan</surname><given-names>A</given-names></name><name><surname>Liu</surname><given-names>V</given-names></name><etal/></person-group>:<article-title>ga4gh/dockstore: 1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.5281/zenodo.154185</pub-id></a> <b>(4)</b> The <b>Dockstore</b>: enabling modular, community-focused sharing of Docker-based genomics tools and workflows<ul><li>In this manuscript, the authors describe the motivation, design, architecture, and merit of <b>Dockstore</b>.org, a community-focused utility for sharing Docker-based tools and workflows for the sciences.<li> The description of <b>Dockstore</b> architecture is thorough and each design decision is justified and informative to the reader.<li> For example, how does <b>Dockstore</b> handle tagging of Dockerfiles and how should users make use of them? This can be brief, but it may be helpful to better describe how to augment the value add that <b>Dockstore</b> brings to reproducibility with good git practices.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26913191">26913191</a> <b>(1)</b> BioShaDock: a community driven bioinformatics shared Docker-based tools registry<ul><li> <b>Dockstore</b><sup><xref rid="ref-32" ref-type="bibr"><u>32</u></xref></sup> is an open platform that enables the registration of Docker images described using CWL.<li> [32](<person-group person-group-type="author"><name><surname>Connor</surname><given-names>BO</given-names></name><name><surname>Kartashov</surname><given-names>A</given-names></name><name><surname>Yuen</surname><given-names>D</given-names></name><etal/></person-group>:<article-title>DockStore</article-title>.<year>2015</year><ext-link ext-link-type="uri" xlink:href="http://dockstore.org">Reference Source</ext-link>) </body>
