<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Makefile</title>
</head>
<body>
<h1>Sentences that contain Makefile</h1>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36642719">36642719</a> <b>(3)</b> A first look at sea-lavenders genomics &#x2013; can genome wide SNP information tip the scales of controversy in the <italic>Limonium vulgare</italic> species complex?<ul><li> docker image [<xref ref-type="bibr" rid="CR72"><u>72</u></xref>] with all the software, configuration files, parameters, and the <italic><b>Makefile</b></italic>, is also provided.<li> Furthermore, a gitlab repository [<xref ref-type="bibr" rid="CR71"><u>71</u></xref>] was created to host all files (and change history) necessary for analyses reproduction, such as the <italic><b>Makefile</b></italic>, static analysis parameter files, and a <italic>Dockerfile</italic>, (containing instructions on how to build the static analysis environment).<li> Those not supported by PGDSpider were performed using scripts available in <ext-link ext-link-type="uri" xlink:href="https://github.com/CoBiG2/RAD_Tools">https://github.com/CoBiG2/RAD_Tools/</ext-link> (technical details on these conversions can be found in the <italic><b>Makefile</b></italic>).<li> [72](Pina-Martins F. limonium_data_analyses Docker image. limonium_data_analyses. 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/repository/docker/stunts/limonium_data_analyses">https://hub.docker.com/repository/docker/stunts/limonium_data_analyses</ext-link>. Cited 2022 Sep 14.) [71](Pina-Martins F. Francisco Pina-Martins / Limonium_GBS_data_analyses GitLab. GitLab. 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://gitlab.com/StuntsPT/Limonium_GBS_data_analyses">https://gitlab.com/StuntsPT/Limonium_GBS_data_analyses</ext-link>. Cited 2022 Sep 14.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36583961">36583961</a> <b>(2)</b> A computational pipeline to learn gene expression predictive models from epigenetic information at enhancers or promoters<ul><li><bold><italic>Note:</italic></bold> SeqCode full distribution contains the source code of the applications, the <b>Makefile</b> to generate the binaries, and a set of automatical tests to check all programs work well in small real datasets.<li>bin include LICENSE <b>Makefile</b> objects README.md src tests.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36467580">36467580</a> <b>(1)</b> The dynamic landscape of peptide activity prediction<ul><li> via <b>makefile</b>, snakemake, drake or targets).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36344928">36344928</a> <b>(2)</b> &#xFEFF;SparkEC: speeding up alignment-based DNA error correction tools<ul><li>Delving into MSA-based parallel tools, the ALLPATHS-LG assembler&#xA0;[<xref ref-type="bibr" rid="CR25"><u>25</u></xref>] provides a built-in error corrector implemented using a set of tasks that are executed through a <b>Makefile</b>.<li> Therefore, it easily allows to set up a multi-threaded execution by taking advantage of the support provided by <b>Makefile</b> for such goal.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21187386">21187386</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36208225">36208225</a> <b>(6)</b> Ontology Development Kit: a toolkit for building, maintaining and standardizing biomedical ontologies<ul><li>The executable ontology-engineering workflows are delivered as an ODK-generated <b>Makefile</b>.<li> Targets in that <b>Makefile</b> can roughly be divided into those that provide the recipe for generating a specific file (such as the release file of an ontology) and those that provide simple workflows, such as &#x2018;clean&#x2019; to delete temporary files, &#x2018;prepare_release&#x2019; to execute the release workflow or &#x2018;refresh-imports&#x2019; to update the terms in all imported ontologies.<li> Unlike the other workflows, it is launched not from a <b>Makefile</b> but from a small wrapper script that uses the ODK Docker image to:.<li>create a new directory which contains all files necessary for editing and managing the new ontology (importantly, this includes the automatically generated <b>Makefile</b> that will pilot all the other standard workflows described in the rest of this section);.<li> The QC tests are on the ODK Docker image via a <b>Makefile</b> target, &#x2018;make test&#x2019;.<li> builds all release targets (release file variants which are configured as <b>Makefile</b> targets) by loading the editors file and performing (mostly ROBOT-based) transformation pipelines such as merging, reasoning and adding version information.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36226245">36226245</a> <b>(1)</b> Analysis of Diabetes Disease Risk Prediction and Diabetes Medication Pattern Based on Data Mining<ul><li> The original version of WEKA consisted of a Tcl/Tk front-end model algorithm and <b>Makefile</b> system to run machine learning and data mining experiments.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36236456">36236456</a> <b>(1)</b> Cross-World Covert Channel on ARM Trustzone through PMU<ul><li> We first write and edit the TA and <b>Makefile</b> code in the OP-TEE TA root directory.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36159175">36159175</a> <b>(1)</b> RiboTaxa: combined approaches for rRNA genes taxonomic resolution down to the species level from metagenomics data revealing novelties<ul><li> Similarly, for SPINGO, the SILVA 138.1 SSU database was indexed using the provided <b>makefile</b> (<ext-link xlink:href="https://github.com/tiendu/SPINGO_updated/tree/master/SPINGO_SILVA/database" ext-link-type="uri">https://github.com/tiendu/SPINGO_updated/tree/master/SPINGO_SILVA/database</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36159898">36159898</a> <b>(1)</b> One-off events? An empirical study of hackathon code creation and reuse<ul><li> While most reused blobs are related to web/mobile apps/frameworks, we also found the relatively uncommon Gherkin being the second most common language for <italic>Medium</italic> projects, and the <italic>Small</italic> projects reused a lot of blobs related to D/DTrace/<b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36119886">36119886</a> <b>(4)</b> Drying temperature-dependent profile of bioactive compounds and prediction of antioxidant capacity of cashew apple pomace using coupled Gaussian Process Regression and Support Vector Regression (GPR&#x2013;SVR) model<ul><li> Source code was first generated using MATLAB and involved allocating function to the generated model file &#x2018;model.c&#x2019;, the codes were then compiled (for each model) into a <b>makefile</b>.<li> The individual <b>makefile</b>s were then combined into a single <b>makefile</b>.<li> The <b>makefile</b> was then run to compute the coupled model based on the supplied residual coupling formulation and the generated predicted output from GPR and SVR.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36008829">36008829</a> <b>(1)</b> FPocketWeb: protein pocket hunting in a web browser<ul><li>In compiling <italic>fpocket</italic> to Wasm, we also compiled the related programs specified in the same <b>makefile</b>, including <italic>mdpocket</italic> (a tool for analyzing pocket conformations sampled by molecular dynamics simulations), <italic>dpocket</italic> (a tool for extracting physicochemical pocket descriptors), and <italic>tpocket</italic> (a tool for evaluating cavity scoring functions).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36042888">36042888</a> <b>(1)</b> Detection of m<sup>6</sup>A in single cultured cells using&#xA0;scDART-seq<ul><li>$ perl <b>Makefile</b>.PL EXPATLIBPATH=$CONDA_PREFIX/lib EXPATINCPATH=$CONDA_PREFIX/include.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35963929">35963929</a> <b>(1)</b> Baseline proteomics characterisation of the emerging host biomanufacturing organism <italic>Halomonas bluephagenesis</italic><ul><li>Documented Halomonas Expressed Protein Index, Quantified Protein Profile and Spectral Library Producing <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35884206">35884206</a> <b>(2)</b> Mining Amphibian and Insect Transcriptomes for Antimicrobial Peptide Sequences with rAMPage<ul><li> The pipeline is implemented as a <b>Makefile</b> with each step as a separate target, making the pipeline modular and providing analysis checkpoints.<li>rAMPage is implemented as a <b>Makefile</b> and written in bash, Python3, and R.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35567771">35567771</a> <b>(12)</b> ntEdit+Sealer: Efficient Targeted Error Resolution and Automated Finishing of Long&#x2010;Read Genome Assemblies<ul><li> The protocol is implemented as a <b>Makefile</b> (GNU Make) pipeline available on Github at <ext-link xlink:href="https://github.com/bcgsc/ntedit_sealer_protocol" ext-link-type="uri">https://github.com/bcgsc/ntedit_sealer_protocol</ext-link>.<li> The ntEdit+Sealer protocol is implemented as a <b>Makefile</b> pipeline, which is a wrapper for the ntHits, ntEdit, ABySS&#x2010;Bloom, and Sealer commands.<li>This <xref rid="cpz1442-prot-0001" ref-type="sec"><u>Basic Protocol</u></xref> describes how to run ntEdit+Sealer with the protocol <b>Makefile</b> to polish and correct errors in a long&#x2010;read genome assembly using short reads.<li>The pipeline is invoked with a single <b>Makefile</b> command.<li>Run the ntEdit+Sealer <b>Makefile</b> with the &#x201C;finish&#x201D; command to polish and fill the draft assembly <monospace>ecoli_shasta.fa</monospace> with the two reads files <monospace>SRR15859208_1.fastq</monospace> and <monospace>SRR15859208_2.fastq</monospace>.<li> Successful completion will result in the <b>Makefile</b> reporting &#x201C;<monospace>ntEdit and Sealer polishing steps complete! Polished assembly can be found in: ecoli_shasta.ntedit_edited.prepd.sealer_scaffold.fa</monospace>&#x201D;.<li> This parameter is controlled by the <monospace>b</monospace> parameter when invoking the protocol <b>Makefile</b>.<li> The <b>Makefile</b> will exit with a non&#x2010;zero code if an error occurs during any of the intermediate steps.<li> The ntEdit+Sealer <b>Makefile</b> has these behaviors configured automatically.<li>The <b>Makefile</b> should not finish running (i.e., will exit with a non&#x2010;zero status) if an error occurs during any of the intermediate steps.<li>The ntEdit+Sealer <b>Makefile</b> gives the error &#x201C;No rule to make target&#x2026;&#x201D;.<li>If other problems arise related to the ntEdit+Sealer protocol <b>Makefile</b> and scripts, create a Github Issue on the ntEdit+Sealer Github page at <ext-link xlink:href="https://github.com/bcgsc/ntedit_sealer_protocol/issues" ext-link-type="uri">https://github.com/bcgsc/ntedit_sealer_protocol/issues</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35436299">35436299</a> <b>(1)</b> Understanding signaling and metabolic paths using semantified and harmonized information about biological interactions<ul><li> A GNU/Linux <b>Makefile</b> on GitHub demonstrates the combination of SPARQL to list all resource IRIs of a certain interaction type and the JSON query tool jq (<ext-link xlink:href="https://stedolan.github.io/jq/" ext-link-type="uri">https://stedolan.github.io/jq/</ext-link>) to process the ShEx module output to count the number of errors for each interaction.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35496797">35496797</a> <b>(2)</b> Detecting archaic introgression and modeling multiple-wave admixture with ArchaicSeeker 2.0<ul><li>First modify the parameters specified by CFLAG (-I and -L) in the <b>makefile</b> into the path of installed libraries, and then type the following commands to compile:<boxed-text id="dtbox1"><p id="p0100">&gt; cd ArchaicSeeker2.0.<li>Modify the parameters specified by CFLAG (-I and -L) in the <b>makefile</b> into the path of installed libraries.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35413944">35413944</a> <b>(1)</b> TE Density: a tool to investigate the biology of transposable elements<ul><li> There is a <b>Makefile</b> within each genome&#x2019;s directory that can be used as a reference to see how analyses and scripts were executed.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35399833">35399833</a> <b>(1)</b> Knowledge-Based Discovery of the Role and Mechanism of Resveratrol in Improving Glomerular Tether Cell Proliferation and Apoptosis in Diabetic Nephropathy<ul><li>In order to make the program run on the laris platform, a <b>Makefile</b> file for the Solaris platform was written in addition to the ANSI C implementation.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35584662">35584662</a> <b>(1)</b> Xist-mediated silencing requires additive functions of SPEN and Polycomb together with differentiation-dependent recruitment of SmcHD1<ul><li>The processed 3D-SIM image files were analyzed using an in-house adapted <b>makefile</b> script for masking of the signal and centroid determination by watershed algorithm (<xref rid="bib61" ref-type="bibr"><u>Rodermund et&#xA0;al., 2021</u></xref>).<li> [Rodermund et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34112668">34112668</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35271204">35271204</a> <b>(1)</b> Multi-Unit Serial Polynomial Multiplier to Accelerate NTRU-Based Cryptographic Schemes in IoT Embedded Systems<ul><li> Finally, to facilitate migration to different development platforms and embedded systems, the <b>makefile</b> used to compile and install the functions and test programs was conveniently updated to incorporate the new functionality.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35242924">35242924</a> <b>(1)</b> Dataset of network simulator related-question posts in stack overflow<ul><li>[&#x2018;vein&#x2019;, &#x2018;omnet&#x2019;, &#x2018;vehicle&#x2019;, &#x2018;sumo&#x2019;, &#x2018;change&#x2019;, &#x2018;rsu&#x2019;, &#x2018;car&#x2019;, &#x2018;scenario&#x2019;, &#x2018;simulation&#x2019;, &#x2018;speed&#x2019;, &#x2018;map&#x2019;, &#x2018;traci&#x2019;, &#x2018;current&#x2019;, &#x2018;model&#x2019;, &#x2018;flow&#x2019;, &#x2018;accident&#x2019;, &#x2018;exist&#x2019;, &#x2018;<b>makefile</b>&#x2019;, &#x2018;edge&#x2019;, &#x2018;tutorial&#x2019;].</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35113859">35113859</a> <b>(1)</b> A field-based general framework to simulate fluids in parallel and the framework&#x2019;s application to a matrix acidization simulation<ul><li><b>Makefile</b>.mac.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35071706">35071706</a> <b>(2)</b> Dataset of exponential growth rate values corresponding non-spherical bubble oscillations under dual-frequency acoustic irradiation<ul><li><b>makefile</b>.<li> The <b>makefile</b> manages the compilation process under a Linux environment.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34992211">34992211</a> <b>(1)</b> Addendum: Transcriptomics and proteomics reveal two waves of translational repression during the maturation of malaria parasite sporozoites<ul><li> All commands that were used in this analysis are included in a <b>Makefile</b> provided in Supplementary Code&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35111904">35111904</a> <b>(1)</b> Microservice security: a systematic literature review<ul><li><b>Makefile</b> to automate the LaTeX build process.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34940723">34940723</a> <b>(1)</b> Evaluation of 2D-/3D-Feet-Detection Methods for Semi-Autonomous Powered Wheelchair Navigation<ul><li> This is achieved by disabling the OPENCV option in the <b>makefile</b> and removing the limitation to 3 channels in the functions <italic toggle="yes">hsv_rgb</italic> and <italic toggle="yes">rgb_hsv</italic> from the file <italic toggle="yes">image.c</italic>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34755882">34755882</a> <b>(1)</b> The Human Disease Ontology 2022 update<ul><li> The pipeline has been added to the DO&#x2019;s <b>Makefile</b> to run each time we build a new DO release, thus creating British English synonyms for new DO terms.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34770655">34770655</a> <b>(1)</b> LC-DEX: Lightweight and Efficient Compressed Authentication Based Elliptic Curve Cryptography in Multi-Hop 6LoWPAN Wireless Sensor Networks in HIP-Based Internet of Things <xref rid="fn1-sensors-21-07348" ref-type="author-notes">&#x2020;</xref><ul><li> Another central reason for this hardware choice is that our code implementation of the HIP DEX protocol is written in C++ and executed with the <b>Makefile</b> method for Linux operating system.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34717540">34717540</a> <b>(1)</b> LongStitch: high-quality genome assembly correction and scaffolding using long reads<ul><li>LongStitch is implemented as a pipeline using a <b>Makefile</b>, and consists of assembly correction and scaffolding stages using long reads (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>a).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34673764">34673764</a> <b>(2)</b> The <italic toggle="yes">Plasmodium</italic> NOT1-G paralogue is an essential regulator of sexual stage maturation and parasite transmission<ul><li> A <b>MakeFile</b> describing the complete bioinformatics workflow (including all steps noted above) and other scripts used are also provided in <xref rid="pbio.3001434.s013" ref-type="supplementary-material"><u>S3 File</u></xref>.<li>A <b>Makefile</b> log of complete RNA-seq bioinformatics workflow, along with all custom R and python scripts used in this work.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34754812">34754812</a> <b>(2)</b> A novel method for utilizing RF information from IEEE 802.11 frames in Software Defined Networks<ul><li> Modifications were made to the driver <b>makefile</b> to allow the WP4 kernel module to be compiled together with the other Openwifi driver modules and loaded as part of the startup process.<li> A new folder has been created in the <italic>user_space</italic> directory titled <italic>wp4_control,</italic> which contains the control plane source files and required <b>makefile</b> for compilation on the device.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34599234">34599234</a> <b>(1)</b> Productive visualization of high-throughput sequencing data using the SeqCode open portable platform<ul><li> We provide a <b>Makefile</b> to generate the binaries from the standalone version, which can be immediately used in the working path of the user machine.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34219973">34219973</a> <b>(1)</b> Mitigate Wormhole Attack and Blackhole Attack Using Elliptic Curve Cryptography in MANET<ul><li>The directory NS2.35 has been changed and the algorithm is used for the files like the ns-lib.TCL, Cmu-trace.cc, Cmu-trace.h, packet.h, ns-packet.TCL, <b>Makefile</b>, priqueue.cc, ns-agent.tcl, ns-mobilenode.tcl.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a> <b>(2)</b> Using prototyping to choose a bioinformatics workflow management system<ul><li> However, traditional shell scripts and even <b>Makefile</b>s have limited error reporting, are hard to debug, can be hard to restart after they go wrong, and can be challenging to move from one computer architecture to another.<li>A Nextflow workflow has a structure analogous to a <b>Makefile</b> or Snakefile&#x2014;it consists of a set of processes which define inputs, outputs, and commands describing how to create the outputs from the inputs.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33552143">33552143</a> <b>(1)</b> Research on Components Assembly Platform of Biological Sequences Alignment Algorithm<ul><li>After all the Apla component conversion and assembly code conversion, the <italic><b>makefile</b></italic> script file for compilation is generated automatically, and is executed to compile and link the C++ components.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33465147">33465147</a> <b>(1)</b> Intra-host changes in Kaposi sarcoma-associated herpesvirus genomes in Ugandan adults with Kaposi sarcoma<ul><li>Paired-end reads, including their dUMI sequence tags, were mapped by <italic>bwa</italic> to sample-consensus genomes (<bold><xref ref-type="fig" rid="ppat.1008594.g002"><u>Fig 2</u></xref></bold>) using a <b>Makefile</b> adapted from [<xref rid="ppat.1008594.ref061" ref-type="bibr"><u>61</u></xref>] (<ext-link ext-link-type="uri" xlink:href="https://github.com/MullinsLab/Duplex-Sequencing">https://github.com/MullinsLab/Duplex-Sequencing</ext-link>).<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25299156">25299156</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33446750">33446750</a> <b>(1)</b> OpenEP: an open-source simulator for electroporation-based tumor treatments<ul><li> <b>Makefile</b> or scalar_field.h) were reused from a recently published simulation code: LibreGrowth<sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup>.<li> [23](          <person-group person-group-type="author">            <name>              <surname>Luj&#xE1;n</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Rosito</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Soba</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Su&#xE1;rez</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Libregrowth: a tumor growth code based on reaction&#x2013;diffusion equations using shared memory</article-title>          <source>Comput. Phys. Commun.</source>          <year>2019</year>          <volume>243</volume>          <fpage>97</fpage>          <lpage>105</lpage>          <pub-id pub-id-type="doi">10.1016/j.cpc.2019.05.005</pub-id>        ) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30271584">30271584</a> <b>(1)</b> A systematic performance evaluation of clustering methods for single-cell RNA-seq data<ul><li> The current system uses a <b>Makefile</b> to run a set of R scripts for clustering, summarization and visualization of the results.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33198787">33198787</a> <b>(1)</b> PathoSPOT genomic epidemiology reveals under-the-radar nosocomial outbreaks<ul><li> The pipeline is implemented as a Rakefile (a <b>Makefile</b> for the Ruby language) that calculates dependencies and executes all necessary subtasks to reach desired outputs.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33202905">33202905</a> <b>(4)</b> A Benchmark of Data Stream Classification for Human Activity Recognition on Connected Objects<ul><li> We&#xA0;generate three synthetic datasets (MOA commands available <uri xlink:href="https://github.com/big-data-lab-team/benchmark-har-data-stream/blob/f314bf4a258e96e418e249228897d269c59cd522/<b>Makefile</b>#L104">https://github.com/big-data-lab-team/benchmark-har-data-stream/blob/f314bf4a258e96e418e249228897d269c59cd522/<b>Makefile</b>#L104</uri>): a hyperplane, a RandomRBF, and a RandomTree dataset.<li> The repository contains the source code and data we used to run the experiment, including the <b>Makefile</b> to compile the benchmark, the&#xA0;Python script to organize the execution and plot the results, and the datasets.<li>The plotting script (<monospace><b>makefile</b>.py</monospace>) extracts data from three CSV files: <monospace>models.csv</monospace>, <monospace>output_runs</monospace>, and <monospace>output</monospace>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33170857">33170857</a> <b>(5)</b> Ten simple rules for writing Dockerfiles for reproducible data science<ul><li> However, a script or <monospace><b>Makefile</b></monospace> might be required so that you do not forget that you set the argument, or how you set it (see <xref ref-type="sec" rid="sec023"><u>Rule 10</u></xref>).<li> These instructions are not limited to <monospace>docker &lt;command&gt;</monospace> but include the usage of bespoke scripts, a <monospace><b>Makefile</b></monospace>, or <monospace>docker-compose</monospace> (see <xref ref-type="sec" rid="sec021"><u>Rule 8</u></xref> and <xref ref-type="sec" rid="sec023"><u>Rule 10</u></xref>).<li> In any case, you should document different variants very well and potentially capture build and run commands in a <monospace><b>Makefile</b></monospace> [<xref rid="pcbi.1008316.ref027" ref-type="bibr"><u>27</u></xref>].<li> If you use a <monospace><b>Makefile</b></monospace>, then keep it in the same repository (see <xref ref-type="sec" rid="sec020"><u>Rule 7</u></xref>), and include instructions for its usage (see <xref ref-type="sec" rid="sec010"><u>Rule 4</u></xref>).<li>In case you need a setup or configuration for the first 2 habits, it is good practice to provide a <b>Makefile</b> alongside your Dockerfile, which can capture the specific commands.<li> [27](Wikipedia contributors. Make (software) [Internet]. Wikipedia. 2019. <ext-link ext-link-type="uri" xlink:href="https://en.wikipedia.org/w/index.php?title=Make_(software)&amp;oldid=929976465">https://en.wikipedia.org/w/index.php?title=Make_(software)&amp;oldid=929976465</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33125376">33125376</a> <b>(4)</b> RNA structure prediction using positive and negative evolutionary information<ul><li>config.status: creating src/<b>Makefile</b>.<li>config.status: creating lib/FastTree/src/<b>Makefile</b>.<li>config.status: creating lib/R2R/R2R-current/src/<b>Makefile</b>.<li>config.status: error: cannot find input file: `lib/RNAVIEW/src/<b>Makefile</b>.in'.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33020189">33020189</a> <b>(1)</b> A Model of Indel Evolution by Finite-State, Continuous-Time Machines<ul><li>File S1 (gzipped tarball) contains JSON files describing the state machines in <xref ref-type="fig" rid="fig2"><u>Figure 2</u></xref>, a <b>Makefile</b> and short script for manipulating these state machines, a Mathematica notebook deriving the equations in this paper, and a text file listing the contents of the tarball.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32647314">32647314</a> <b>(1)</b> The FLUXNET2015 dataset and the ONEFlux processing pipeline for eddy covariance data<ul><li> The source codes and the compiled executables are provided for steps implemented in C, as well as build procedures in make/<b>Makefile</b> format.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32640986">32640986</a> <b>(1)</b> Bioinformatics recipes: creating, executing and distributing reproducible data analysis workflows<ul><li> A recipe may be a collection of several command-line tools, it may be a <b>Makefile</b>, a SnakeMake [<xref ref-type="bibr" rid="CR3"><u>3</u></xref>] file, a Nextflow [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>] pipeline, an R script or any command line oriented computer program.<li> [3](Federico A, et al. Pipeliner: a Nextflow-based framework for the definition of sequencing data processing pipelines, Front Genet. 2019;10:614.&#xA0;10.3389/fgene.2019.00614.) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><u>No PMID</u> <b>(7)</b> Enabling Manual Intervention for Otherwise Automated Registration of Large Image Series<ul><li>Fiji&#xA0;[<xref ref-type="bibr" rid="CR6"><u>6</u></xref>] was used to roughly mark the centre of each section in thumbnail images of the gigapixel scans (<ext-link ext-link-type="uri" xlink:href="http://github.com/romangrothausmann/CZIto3D/blob/4da67edd107303b23ed00a6230cc38775a3b9007/czi2stack/<b>Makefile</b>L96-L104">czi2stack/<b>Makefile</b></ext-link>).<li> These centres were then used to automatically extract the region of each slice as its own image (3000&#xA0;x&#xA0;3000 pixel, PNG) with bfconvert&#xA0;[<xref ref-type="bibr" rid="CR23"><u>23</u></xref>] (<ext-link ext-link-type="uri" xlink:href="http://github.com/romangrothausmann/CZIto3D/blob/4da67edd107303b23ed00a6230cc38775a3b9007/czi2stack/<b>Makefile</b>L179-L192">czi2stack/<b>Makefile</b></ext-link>).<li> This order was then used to register the consecutive slices (<ext-link ext-link-type="uri" xlink:href="http://gitlab.com/romangrothausmann/k2_fibrosis/blob/ba87b312a8047e58dc0fcd0ebcdd000465c902fe/morph-fib/czi2stack/<b>Makefile</b>L111-115">czi2stack/<b>Makefile</b></ext-link>).<li>While the current implementation with <b>Makefile</b>s serves as a proof of principle, it can be further improved to a more intuitive and user-friendly program.<li> [6](Fiji/ImageJ development team: Fiji (ImageJ), see [25, 26]. <ext-link ext-link-type="uri" xlink:href="http://fiji.sc">http://fiji.sc</ext-link>) [23](Open Microscopy Environment: bftools (scripts using Bio-Formats), see [14]. <ext-link ext-link-type="uri" xlink:href="https://docs.openmicroscopy.org/bio-formats/5.7.2/users/comlinetools/%20index.html">https://docs.openmicroscopy.org/bio-formats/5.7.2/users/comlinetools/ index.html</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32385287">32385287</a> <b>(1)</b> Budding yeast complete DNA synthesis after chromosome segregation begins<ul><li> Detailed and documented pipeline can be found in the <b>Makefile</b> on github and in the Supplementary Information.</ul></p>
<p><u>No PMID</u> <b>(1)</b> Tensor-Based CUDA Optimization for ANN Inferencing Using Parallel Acceleration on Embedded GPU<ul><li> This is done by setting their respective flags to 1 in the <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32357829">32357829</a> <b>(1)</b> wg-blimp: an end-to-end analysis pipeline for whole genome bisulfite sequencing data<ul><li><b>Makefile</b>.</ul></p>
<p><u>No PMID</u> <b>(1)</b> A Verified Low-Level Implementation of the Adaptive Exterior Light and Speed Control System<ul><li> Of 5223 source code lines (which also contain a <b>Makefile</b> and code for state (graph) visualization), 3786 lines are test code.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35495212">35495212</a> <b>(2)</b> Microinjection support system for small biological subjects<ul><li>$ make -f <b>Makefile</b>.osx.<li>Do not forget to copy libump.h to system include dir(/usr/local/include) because <b>Makefile</b> included in the SDK only copies built libraries.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32169095">32169095</a> <b>(1)</b> A framework for assessing 16S rRNA marker-gene survey data analysis methods using mixtures.<ul><li> The <b>Makefile</b> and scripts used to run the Mothur pipeline are available <ext-link ext-link-type="uri" xlink:href="https://github.com/nate-d-olson/mgtst_pipelines/blob/master/code/mothur">https://github.com/nate-d-olson/mgtst_pipelines/blob/master/code/mothur</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35903557">35903557</a> <b>(1)</b> TreeSwift: A massively scalable Python tree package<ul><li> This is essential to contrast against the current state-of-the-art, ETE Toolkit, which requires the Six and NumPy Python libraries to install if the user has administrative privileges or Anaconda/Miniconda to install if the user does not, and BioPython, which requires a C compiler and the NumPy Python library as well as the computer fluency to compile tools from source using a <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31897481">31897481</a> <b>(10)</b> Data detectives, self-love, and humility: a research parasite's perspective<ul><li>The Life-Changing Magic of <b>Makefile</b>s.<li>In my opinion, the most revolutionary parasitic tactic of all is pipelining workflows with <b>Makefile</b>s.<li> Broadly, <b>Makefile</b>s contain all of the instructions to reproduce every part of an analysis and are usually paired with a software program that reads the instructions in the <b>Makefile</b> to automatically rerun all parts of the analysis that need to be updated (Figure <xref ref-type="fig" rid="fig4"><u>4</u></xref>) [<xref rid="bib4" ref-type="bibr"><u>7</u></xref>, <xref rid="bib10" ref-type="bibr"><u>8</u></xref>].<li> <b>Makefile</b>s are incredibly useful because they enable the remaking of an entire analysis and&#x2014;more importantly&#x2014;they are by definition a documentation of all the steps taken.<li><b>Makefile</b>s also make the review process much easier, saving time and freeing research parasites from tedious time-consuming details.<li> With a <b>Makefile</b>, it is easy to go back, find, and double-check individual steps in an entire analysis, as opposed to digging through folders of spaghetti code without guidance.<li> Adding new analyses also becomes easier, as new scripts can be seamlessly plugged into the <b>Makefile</b>, using intermediate analysis files already created from other steps in the workflow.<li> And most importantly, <b>Makefile</b>s take care of the details, automatically updating figures and tables when their underlying inputs change, saving research parasites many hours of tedious detail-checking.<li> Dependencies between input data, intermediate files, and scripts are encoded in the <b>Makefile</b>, which automatically reruns any necessary steps when input data or files are updated.<li> [7](<person-group person-group-type="author"><name name-style="western"><surname>Stallman</surname><given-names>RM</given-names></name>, <name name-style="western"><surname>McGrath</surname><given-names>R</given-names></name></person-group><article-title>GNU Make&#x2014;A program for directing recompilation</article-title>. <year>1991</year><comment><ext-link ext-link-type="uri" xlink:href="http://www.gnu.org/software/make/">http://www.gnu.org/software/make/</ext-link>. Accessed September 28, 2019.</comment>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32133422">32133422</a> <b>(1)</b> PYLFIRE: Python implementation of likelihood-free inference by ratio estimation<ul><li> PYLFIRE is provided with a <b>makefile</b> and installation options as follows.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31780762">31780762</a> <b>(1)</b> Bevacizumab for the treatment of non-small cell lung cancer patients with synchronous brain metastases<ul><li>All code written to perform analysis is available online as a series of scripts, along with a <b>Makefile</b> to make the entirety of analysis explicit and enable others to replicate this work using data available from SEER-Medicare (Ascha 2019)<sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup>.<li> [23](Ascha, M.S. 0.0.9000 mustafaascha/seerm-bevacizumab: Repository init, 10.5281/zenodo.2544421 (2019).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31803040">31803040</a> <b>(1)</b> Enabling Large-Scale Simulations With the GENESIS Neuronal Simulator<ul><li> However, the BDW-GC is easily deactivated at compile time in PGENESIS 2.4&#x2212;2019, simply by omitting the GCMALLOC flags in the well-documented <b>Makefile</b>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31357927">31357927</a> <b>(7)</b> ROBOT: A Tool for Automating Ontology Workflows<ul><li> A <b>Makefile</b> consists of a set of rules used to make &#x201C;targets&#x201D;.<li> In ontology development, the <b>Makefile</b> is used for automated tasks, such as preparing an ontology for release.<li>Creating a <b>Makefile</b> to coordinate all these steps can be challenging.<li> This can be used to create a GitHub repository following a standard layout, with a standard <b>Makefile</b> following the workflow detailed above.<li> OBI currently uses a <b>Makefile</b> [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>] that defines a range of tasks for managing imports, converting templates, merging, reasoning, testing, and releasing new versions of OBI.<li> The <b>Makefile</b> specifies various target files, and most target files are generated from a single ROBOT command or a single chain of ROBOT commands.<li>In 2018, DO switched their entire automated build process (contained in the <b>Makefile</b>) from OWLTools [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] and OORT [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] to ROBOT.<li> [39](OBI Makefile as of 2019-02-22. <ext-link ext-link-type="uri" xlink:href="https://github.com/obi-ontology/obi/blob/5fc065729fc7eb013dafd14690559621b5606057/Makefile">https://github.com/obi-ontology/obi/blob/5fc065729fc7eb013dafd14690559621b5606057/Makefile</ext-link>) [14](owlcollab/owltools: OWLTools. <ext-link ext-link-type="uri" xlink:href="https://github.com/owlcollab/owltools">https://github.com/owlcollab/owltools</ext-link> (n.d.). Accessed 21 May 2019.) [15](Oort Intro. <ext-link ext-link-type="uri" xlink:href="https://github.com/owlcollab/owltools/wiki/Oort-Intro">https://github.com/owlcollab/owltools/wiki/Oort-Intro</ext-link> (n.d.). Accessed 21 May 2019.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31344036">31344036</a> <b>(1)</b> Ten simple rules for writing and sharing computational analyses in Jupyter Notebooks<ul><li> Such notebooks can be used not only interactively but also as command-line tools that can be executed automatically&#x2014;a great boon for pipelines! Consider linking your analysis pipeline steps via a <b>Makefile</b> or similar tool that allows for complete noninteractive execution of the entire pipeline, either in full or partial steps.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31328028">31328028</a> <b>(1)</b> A new comprehensive eye-tracking test battery concurrently evaluating the Pupil Labs glasses and the EyeLink 1000<ul><li> To facilitate research with Pupil Labs, we offer our own <b>makefile</b> to automatically compile most dependencies to run Pupil Labs from source, without the need for root-rights.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31265455">31265455</a> <b>(1)</b> Closing target trimming and CTTdocker programs for discovering hidden superfamily loci in genomes<ul><li> function in <italic><b>makefile</b></italic> under the directory <italic>/ctt/dependencies</italic> should help to install the Pfam-A database and the remaining program dependencies smoothly.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31248380">31248380</a> <b>(1)</b> A catalog of CasX genome editing sites in common model organisms<ul><li> The <b>Makefile</b> downloaded the gene annotations (GTF format) and genome sequences (FASTA format) for each of the specified genomes from release 95 of Ensembl [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21045057">21045057</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30971873">30971873</a> <b>(1)</b> SpiNNTools: The Execution Engine for the SpiNNaker Platform<ul><li><b>Makefile</b>, a file to be processed by GNU Make to build the C code.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30898165">30898165</a> <b>(1)</b> QRNAS: software tool for refinement of nucleic acid structures<ul><li> A <b>Makefile</b> is provided for the compilation of the package.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30811390">30811390</a> <b>(2)</b> Script of Scripts: A pragmatic workflow system for daily computational research<ul><li>Another workflow style that SoS supports is the outcome-oriented or <b>Makefile</b> style, which relies on implicit wild-card idioms introduced with the Make utility, and adopted by some pipeline tools [<xref rid="pcbi.1006843.ref015" ref-type="bibr"><u>15</u></xref>, <xref rid="pcbi.1006843.ref023" ref-type="bibr"><u>23</u></xref>].<li> Although such workflows lack the pattern-matching capacity of typical <b>Makefile</b> style workflows, they are easier to design and are especially suitable for composing data flow based workflows.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28486579">28486579</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30740268">30740268</a> <b>(1)</b> LiBiNorm: an htseq-count analogue with improved normalisation of Smart-seq2 data and library preparation diagnostics<ul><li> The associated <b>makefile</b> supports compilation using gcc or clang in order to generate executables for Linux or macOS.</ul></p>
</ul></p>
</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30532258">30532258</a> <b>(1)</b> QTM: Computational package using MPI protocol for Quantum Trajectories Method<ul><li> This should not be a problem because of used <b>makefile</b> mechanism for simplifying this process.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30482799">30482799</a> <b>(1)</b> OCMA: Fast, Memory-Efficient Factorization of Prohibitively Large Relationship Matrices<ul><li> When C compiler and Intel MKL are available in the system, one can enter the src folder and modify the <b>makefile</b> based on the C compiler and the Intel MKL paths; then type make or nmake to compile.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30458008">30458008</a> <b>(1)</b> Population sequencing reveals clonal diversity and ancestral inbreeding in the grapevine cultivar Chardonnay<ul><li> <bold><b>Makefile</b></bold>, The GNU Make pipeline for marker discovery.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30407550">30407550</a> <b>(1)</b> Human Disease Ontology 2018 update: classification, content and workflow expansion<ul><li> A series of commands are contained in a &#x2018;<b>Makefile</b>&#x2019; which automatically builds target products (DO&#x2019;s production OWL and OBO files, slims, subsets) in under 10 minutes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30407590">30407590</a> <b>(2)</b> ECO, the Evidence &amp; Conclusion Ontology: community standard for evidence information<ul><li> in our <b>Makefile</b>.<li> The <b>Makefile</b> contains a set of &#x201C;rules" to make &#x201C;targets" as part of automated build processes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30455637">30455637</a> <b>(4)</b> Code Generation in Computational Neuroscience: A Review of Tools and Techniques<ul><li> This consists of writing both a Python class, a C code block that can update the state of the new neuron model or synapse on a time-step basis, and finally a <italic><b>Makefile</b></italic> that joins the components together to represent the new neuron model.<li>Finally, the end user needs to fill in a template <italic><b>Makefile</b></italic> which compiles the C components into executable C code that can run on the SpiNNaker platform.<li>The IfCurExp <b>Makefile</b> for SpiNNaker.<li> Once the user has written their components, they then write a <b>Makefile</b> which combines these with the rest of the provided neuron executable, as shown in Listing 5; this handles the rest of the processing required to execute the neuron model, such as the sending and receiving of spikes, the recording of variables and spikes, as well as handling any plasticity.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30405329">30405329</a> <b>(7)</b> Teaching Computational Reproducibility for Neuroimaging<ul><li><b>Makefile</b>s are machine-readable text files consisting of rules specifying the sequence of commands necessary for generating certain files and for tracking dependencies between files.<li> Consider the following <b>Makefile</b> rule:.<li>In the example above, the recipe was a single command, but <b>Makefile</b> recipes often involve several commands.<li> The dependencies of a rule can be filenames (e.g., <monospace>progress.md</monospace>), but may also be other rules in the <b>Makefile</b>.<li> These features allow <b>Makefile</b>s to chain together complex sequences of commands necessary for generating multiple target files, which may depend on other files or steps; such chaining is important for building and maintaining reproducible workflows.<li><b>Makefile</b> recipes fetch data, validates fetched data, generates all results and figures in report.<li><b>Makefile</b>s generate EDA work and supplementary analysis.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30192951">30192951</a> <b>(1)</b> Sequana coverage: detection and characterization of genomic variations using running median and mixture models<ul><li> We provide an implementation within the <monospace>Sequana</monospace> project [<xref rid="bib32" ref-type="bibr"><u>34</u></xref>], which is a Python library that also provides HTS pipelines based on the workflow management system called Snakemake [<xref rid="bib33" ref-type="bibr"><u>35</u></xref>] (<b>Makefile</b>-like with a Python syntax).<li> [34](<person-group person-group-type="author"><name name-style="western"><surname>Cokelaer</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Desvillechabrol</surname><given-names>D</given-names></name>, <name name-style="western"><surname>Legendre</surname><given-names>R</given-names></name>, <etal/></person-group><article-title>Sequana: a set of Snakemake NGS pipelines</article-title>. <source>Journal of Open Source Software</source>. <year>2017</year>;<volume>2</volume>, <fpage>16</fpage><pub-id pub-id-type="doi">10.21105/joss.00352</pub-id>.) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30225327">30225327</a> <b>(1)</b> Performance output data and configurations of stencil compilers experiments run through PROVA!<ul><li><b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30083482">30083482</a> <b>(1)</b> The Oyster River Protocol: a multi-assembler and kmer approach for de novo transcriptome assembly<ul><li> The software is implemented as a stand-alone <b>makefile</b> which coordinates all steps described below.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30175248">30175248</a> <b>(1)</b> SPEKcheck &#x2014; fluorescence microscopy spectral visualisation and optimisation: a web application, javascript library, and data resource<ul><li> The git repository in github is public and includes the source code to prepare new releases in a <b>Makefile</b>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30271960">30271960</a> <b>(1)</b> RAiSD detects positive selection based on multiple signatures of a selective sweep and SNP vectors<ul><li> The tool can be compiled using the provided <italic><b>Makefile</b></italic>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29890644">29890644</a> <b>(1)</b> FPGA-Based High-Performance Embedded Systems for Adaptive Edge Computing in Cyber-Physical Systems: The ARTICo<sup>3</sup> Framework<ul><li> In parallel, the toolchain also generates the required software project by combining the user files (where the main application is defined) with the API to access the underlying ARTICo<sup>3</sup> runtime library, creating a customized <b>Makefile</b> to transparently build the application executable.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29979152">29979152</a> <b>(1)</b> Propagation of uncertainty in experiment: structures of Ni (II) coordination complexes<ul><li> We provide the (general) code, manual and <b>makefile</b> in the <ext-link ext-link-type="uri" xlink:href="http://scripts.iucr.org/cgi-bin/sendsup?xj5011">supporting information</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29704459">29704459</a> <b>(1)</b> A reference genome of the Chinese hamster based on a hybrid assembly strategy<ul><li> For better control over the workflow, we used the customizable <b>makefile</b>&#x2010;based smrtmake workflow (smr, <xref rid="bit26722-bib-0048" ref-type="ref"><u>2016</u></xref>).<li> [2016](<collab collab-type="authors">smr</collab>, <year>2016</year>, <article-title>Smrtmake</article-title>: <article-title>Hackable smrtpipe workflows using makefiles instead of smrtpipe.py (commit 29a9c75)</article-title>. Original&#x2010;date: 2014&#x2010;06&#x2010;13T22:32:12Z.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29782489">29782489</a> <b>(1)</b> Systematic identification of factors mediating accelerated mRNA degradation in response to changes in environmental nitrogen<ul><li> With these archives, all analyses downstream of alignments can be re-run completely using the provided GNU <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29777317">29777317</a> <b>(2)</b> Using SMILES strings for the description of chemical connectivity in the Crystallography Open Database<ul><li> atom with at least three halogen atoms attached to it, and place it into the allcod-hal.smi file using obgrep governed by a GNU Make <b>Makefile</b> script.<li> The <b>Makefile</b> used for this conversion is available in the COD repository smi subdirectory, and the cdkrecharge program is included in the <italic>smiles-scripts</italic> package.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29670589">29670589</a> <b>(1)</b> Gene Enrichment Analysis Reveals Major Regulators of <italic>Mycobacterium tuberculosis</italic> Gene Expression in Two Models of Antibiotic Tolerance<ul><li> Code was written in a combination of R, Python3, bash, and <b>makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29531124">29531124</a> <b>(1)</b> Fast Ordered Sampling of DNA Sequence Variants<ul><li> The whole set of programs can be trivially compiled using the included <b>Makefile</b>, with no external dependencies required.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29474446">29474446</a> <b>(4)</b> PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems<ul><li> The sample projects are structured to mirror the intended design of user projects: a project-tailored <monospace><b>Makefile</b></monospace> and <monospace>main.cpp</monospace> in the main directory, and most custom code in the project-tailored functions in the <monospace>custom_modules</monospace> directory.<li>This and the following examples also demonstrate the intended arrangement of projects: users do not modify the contents of <monospace>./core/</monospace>, but instead place their codes in <monospace>./custom_modules/</monospace>, include these in a main project <monospace>main.cpp</monospace>, and modify the <monospace><b>Makefile</b></monospace> to compile and link the components.<li>Users need a working <monospace>g++</monospace> development environment with support for OpenMP and <b>Makefile</b>s (or a reasonably compatible 64-bit C++11 compliant compiler).<li> In the future, PhysiCell should switch from <b>Makefile</b>s to CMake to facilitate simpler cross-platform compiling.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29167739">29167739</a> <b>(5)</b> Measuring evolutionary rates of proteins in a structural context<ul><li> Analyses presented here utilize the<monospace>raxmlHPC-SSE3</monospace> executable, which can be compiled with<monospace><b>Makefile</b>.SSE3.gcc</monospace> or<monospace><b>Makefile</b>.SSE3.mac</monospace>.<li>To thread RAxML, compile the<monospace>raxmlHPC-PTHREADS-SSE3</monospace> executable with<monospace><b>Makefile</b>.SSE3.PTHREADS.gcc or <b>Makefile</b>.SSE3.PTHREADS.mac</monospace>.<li> Analyses presented here use Rate4Site downloaded as<monospace>rate4site.3.2.source.zip</monospace> and compiled with the<monospace><b>Makefile</b>_slow file</monospace>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29471873">29471873</a> <b>(1)</b> miRCancerdb: a database for correlation analysis between microRNA and gene expression in cancer<ul><li> The commands to build the application are wrapped in a <b>makefile</b> that makes it easy to trigger the local build with only a single command.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29503757">29503757</a> <b>(2)</b> Applying Deep Learning in Medical Images: The Case of Bone Age Estimation<ul><li>cp <b>Makefile</b>.config.example <b>Makefile</b>.config.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29138947">29138947</a> <b>(1)</b> A posteriori metadata from automated provenance tracking: integration of AiiDA and TCOD<ul><li> It may also be implemented in other databases and frameworks, such as GNU <b>Makefile</b>-based replication-ready systems that have been around since as early as 1990&#xA0;[<xref ref-type="bibr" rid="CR63"><u>63</u></xref>].<li> [63](          <person-group person-group-type="author">            <name>              <surname>Schwab</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Karrenbach</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Claerbout</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Making scientific computations reproducible</article-title>          <source>Comput Sci Eng</source>          <year>2000</year>          <volume>2</volume>          <fpage>61</fpage>          <lpage>67</lpage>          <pub-id pub-id-type="doi">10.1109/5992.881708</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29065136">29065136</a> <b>(1)</b> Collective motion of predictive swarms<ul><li>The program can be compiled by either intel icpc compiler or GNU g++ compiler by changing two lines in the <b>Makefile</b> marked &#x201C;CHOSE COMPILER.&#x201D; For intel, use &#x201C;CC = $(ICC)&#x201D; and &#x201C;COMP = $(ICOMP)&#x201D;, for GNU, use &#x201C;CC = $(GCC)&#x201D; and &#x201C;COMP = $(GCOMP).&#x201D; Currently, the intel version is faster than the GNU version.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32704488">32704488</a> <b>(3)</b> A Workflow for Automated Satellite Image Processing: from Raw VHSR Data to Object-Based Spectral Information for Smallholder Agriculture<ul><li>Alternative Workflow Implementation through GNU <b>Makefile</b>.<li>We achieved the above using a GNU <b>Makefile</b>, which is a facility to define file&#x2014;oriented input/output processes and associated file dependencies through rule&#x2014;based specifications [<xref rid="cit0063" ref-type="bibr"><u>63</u></xref>].<li># Atmospheric correction: add one <b>Makefile</b> rule per src/dst file pair.<li> [63](<person-group person-group-type="author"><name name-style="western"><surname>Stallman</surname><given-names>R.M.</given-names></name>; <name name-style="western"><surname>McGrath</surname><given-names>R.</given-names></name>; <name name-style="western"><surname>Smith</surname><given-names>P.</given-names></name></person-group><source><italic toggle="yes">GNU Make: A Program for Directing Recompilation, GNU make Version 3.80</italic></source>; <publisher-name>Free Software Foundation</publisher-name>: <publisher-loc>Boston, MA, USA</publisher-loc>, <year>2016</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29075258">29075258</a> <b>(2)</b> Quantitative Characterization of the T Cell Receptor Repertoire of Na&#xEF;ve and Memory Subsets Using an Integrated Experimental and Computational Pipeline Which Is Robust, Economical, and Versatile<ul><li> The processing is broken down into four steps, which can be run independently (with separate scripts) or combined using a <b>Makefile</b>.<li> Each module can be run independently, or the full pipeline can be run simultaneously using a <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29372115">29372115</a> <b>(2)</b> Benchmark datasets for phylogenomic pipeline validation, applications for foodborne pathogen surveillance<ul><li> the sha256sha256 checksums depend on all reads being downloaded), GG creates a <b>Makefile</b>, which is then executed.<li> That <b>Makefile</b> creates a dependency tree such that all files will be downloaded in the order they are needed.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28809849">28809849</a> <b>(1)</b> Spatiotemporal database of US congressional elections, 1896&#x2013;2014<ul><li> In addition, a <b>makefile</b> is provided for convenience, to ensure the build process executes in the correct order.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28768689">28768689</a> <b>(2)</b> Accelerating Wright&#x2013;Fisher Forward Simulations on the Graphics Processing Unit<ul><li> The companion serial Wright&#x2013;Fisher simulation, serial_SFS_code.cpp, is provided in examples/examples_speed, as is a help file, serial_SFS_code_help.txt, and <b>makefile</b>, <b>makefile</b>_serial.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28649474">28649474</a> <b>(1)</b> Using high-throughput sequencing of ITS2 to describe <italic>Symbiodinium</italic> metacommunities in St. John, US Virgin Islands<ul><li> These scripts are provided along with all raw data and a <b>Makefile</b>, which can be executed to fully reproduce the present analysis.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28557998">28557998</a> <b>(2)</b> Hard real-time closed-loop electrophysiology with the Real-Time eXperiment Interface (RTXI)<ul><li> (E) The module after it is compiled with the provided <b>Makefile</b> and loaded into RTXI.<li> The Plugin Template files include a <b>Makefile</b>, which is used to define compilation rules on how files should be compiled, which external libraries are to be included, and how to package the individual pieces together to form a shared library object that is RTXI-compatible.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28523006">28523006</a> <b>(1)</b> Virtual Plant Tissue: Building Blocks for Next-Generation Plant Growth Simulation<ul><li> For those more familiar with the Make build system, in the top level directory, a front-end <b>Makefile</b> is provided to trigger a build.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28232478">28232478</a> <b>(1)</b> ABySS 2.0: resource-efficient assembly of large genomes using a Bloom filter<ul><li>We provide the commands and configuration files used to run the various assemblers and scaffolding tools in <ext-link ext-link-type="uri" xlink:href="http://genome.cshlp.org/lookup/suppl/doi:10.1101/gr.214346.116/-/DC1">Supplemental Listings S1 through S16</ext-link> and as <b>Makefile</b> scripts in the <ext-link ext-link-type="uri" xlink:href="http://genome.cshlp.org/lookup/suppl/doi:10.1101/gr.214346.116/-/DC1">Supplemental Archive 2</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28317012">28317012</a> <b>(1)</b> Using 3dRPC for RNA&#x2013;protein complex structure prediction<ul><li> Open the file &#x201C;${HOME_3dRPC}/source/<b>Makefile</b>&#x201D;, find the line starting with &#x201C;LAPACK_LIBS&#x201D; and &#x201C;BLAS_LIBS&#x201D;, make sure that the paths of the libraries are correctly assigned.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28141825">28141825</a> <b>(2)</b> Using Triplet Ordering Preferences for Estimating Causal Effects in the Analysis of Gene Expression Data<ul><li> Also a standard <b>Makefile</b> is included.<li> In the <b>Makefile</b> it is assumed that it is installed in <monospace>/opt/local/lib</monospace> and <monospace>/opt/local/include</monospace>, please change accordingly.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27711123">27711123</a> <b>(2)</b> Language-Agnostic Reproducible Data Analysis Using Literate Programming<ul><li>In a first step, <monospace>lir-tangle</monospace> extracts the data transformations defined in the source file to executable programs, and uses the declared data objects and rules to build a valid <b>makefile</b>.<li> Generating all results with <monospace>lir-make</monospace> is implemented as an invocation of Make with a <b>makefile</b> generated by <monospace>lir-tangle</monospace> and the appropriate command-line options.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27632164">27632164</a> <b>(3)</b> Assembly of the Complete Sitka Spruce Chloroplast Genome Using 10X Genomics&#x2019; GemCode Sequencing Data<ul><li> The <b>Makefile</b> script to perform these analyses, including the command line parameters used for each program, are available online at <ext-link ext-link-type="uri" xlink:href="https://github.com/sjackman/picea-sitchensis-plastid/blob/1.0.0/<b>Makefile</b>">https://github.com/sjackman/picea-sitchensis-plastid/blob/1.0.0/<b>Makefile</b></ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27624717">27624717</a> <b>(1)</b> Fast, Accurate and Automatic Ancient Nucleosome and Methylation Maps with epiPALEOMIX<ul><li> It supports multi-threading and requires a simple <b>makefile</b> as input indicating the location of sequence alignment files (BAM format) and the parameters of the analyses to be run.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27437028">27437028</a> <b>(1)</b> A representation of a compressed de Bruijn graph for pan-genome analysis that enables search<ul><li> All programs were compiled with g++ (version 4.8.2) using the provided <b>makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27075578">27075578</a> <b>(11)</b> Miscellaneous Topics in Computer-Aided Drug Design: Synthetic Accessibility and GPU Computing, and Other Topics<ul><li> To compile this program, we must modify <b>Makefile</b> for our software environment.<li> There are two <b>Makefile</b>s.<li> One is for the MPI parallel program part (src/<b>Makefile</b>) and the other is for the CUDA program part (src/cuda/<b>Makefile</b>).<li> Users should modify the <b>Makefile</b> of each application program in a manner similar to the following example.<li>Modification of <b>Makefile</b> for the MPI program (src/<b>Makefile</b>) to GPU in this example:.<li>Modification of <b>Makefile</b> for the CUDA program (src/cuda/<b>Makefile</b>) to GPU in this example.<li>[ src/cuda/<b>Makefile</b> ].<li>If both <b>Makefile</b>s are correctly modified, psygene-G should be compiled by the make command in the form &#x201C;$ make&#x201D;.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27504078">27504078</a> <b>(1)</b> <italic>Dragonfly</italic>: an implementation of the expand&#x2013;maximize&#x2013;compress algorithm for single-particle imaging<xref ref-type="fn" rid="fn1">1</xref><ul><li> This utility calls the included <monospace><b>Makefile</b></monospace> that users can modify to customize this compilation.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27195107">27195107</a> <b>(1)</b> Recommendations for open data science<ul><li> Another option is to provide a script or <b>makefile</b> that can regenerate all analyses of a paper with a single command.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27199908">27199908</a> <b>(2)</b> Proteomic Stable Isotope Probing Reveals Biosynthesis Dynamics of Slow Growing Methane Based Microbial Communities<ul><li> The full protocol is contained in a GNU <b>makefile</b> (Supplemental Data Sheet 1); the <b>makefile</b> must be in the same directory as the raw sequencing reads, and qiime 1.8 and usearch must be installed.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26984618">26984618</a> <b>(1)</b> Towards an open, collaborative, reusable framework for sharing hands-on bioinformatics training workshops<ul><li>A <monospace><b>Makefile</b></monospace> together with the <monospace>template.tex</monospace> file provides a simple way to build a trainer and trainee version of the workshop handout in PDF format.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26862394">26862394</a> <b>(2)</b> Scalable linkage-disequilibrium-based selective sweep detection: a performance guide<ul><li>To generate the OmegaPlus-G executable that implements the generic parallel algorithm, one can use the respective <b>makefile</b> (<b>Makefile</b>.OPENMP.GENERIC.gcc).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26869916">26869916</a> <b>(44)</b> Using Make for Reproducible and Parallel Neuroimaging Workflow and Quality-Assurance<ul><li>This paper has an accompanying manual with lab practicals and examples (see <xref ref-type="supplementary-material" rid="SM1"><u>Supplemental Materials</u></xref>) and all data, scripts and <b>Makefile</b>s necessary to run the practicals and examples are available in the &#x201C;makepipelines&#x201D; project at NITRC.<li> Anything that can be called from the shell can be included in a <b>makefile</b>.<li> A <b>makefile</b> (or a file of instructions for the progam <monospace>make</monospace>) consists of a set of rules that create or update target files if they have not been modified since their dependencies were last modified.<li> We create a file, named <monospace><b>Makefile</b></monospace>, which contains the following lines (Example <xref ref-type="fig" rid="E1"><u>1A</u></xref>):.<li><bold><b>Makefile</b> for skull stripping</bold>.<li> Although, this is a simple example, multiple rules can be included in a <b>makefile</b> that, if chained together, form a directed acyclic dependency graph that can be arbitrarily complex.<li>Example <xref ref-type="fig" rid="E4"><u>2</u></xref> shows a slightly more complicated <b>makefile</b> that uses the skull stripped T1 to perform hippocampal segmentation (using FSL's FIRST) and to generate a quality assurance image using FSL utilities (<monospace>overlay</monospace> and <monospace>slices</monospace>).<li><bold>A more complicated <b>Makefile</b> for hippocampal segmentation</bold>.<li> target defined in the <b>Makefile</b> in Example <xref ref-type="fig" rid="E4"><u>2</u></xref></bold>.<li> FreeSurfer supplies a -make option to run via a standard <b>makefile</b>.<li> We typically create a directory for each project that contains (1) subdirectories for scripts and programs specific to the analysis (bin), (2) masks, templates and <b>makefile</b>s (lib), (3) auxiliary behavioral data (data), and (4) one or more sessions of subject imaging data.<li> By creating links to subject and session-level <b>makefile</b>s, it is possible to build targets for both a single subject/timepoint, and across all subjects at a single timepoint (by calling Make recursively within each subject directory).<li> In this case it is possible to copy subject and session-level <b>makefile</b>s and avoid additional complexity.<li> Because a dependency graph expresses inherent parallel structure, a single <b>makefile</b> is a parallel program that can be easily executed on either a multicore machine (using the <monospace>-j</monospace> flag to specify the number of cores to use) or on a cluster using <monospace>qmake</monospace>, an Open Grid Scheduler-supplied utility.<li>In contrast, a <b>makefile</b> to do this task would be written as follows:.<li> The same <b>makefile</b> can be used to execute in parallel (with any degree of parallelism required to allow sufficient memory for the jobs) or sequentially.<li>Writing self-documenting <b>makefile</b>s.<li>One of the advantages of using <b>makefile</b>s is that phony targets can be defined that allow underlying neuroimaging calls to be conceptually named or grouped.<li> Writing <b>makefile</b>s in this way makes it easier for people who are new to neuroimaging to understand and work with the pipelines.<li> message for the <b>makefile</b>.<li> In this way, the help message and the target are close together in the <b>makefile</b>, so that it is easier to keep the help up to date.<li> This has been implemented in a help system described by Graham-Cumming (<xref rid="B12" ref-type="bibr"><u>2015</u></xref>), that produces a list of targets and their corresponding line numbers in specific <b>makefile</b>s.<li> Each of these can pipelines is cleanly written in its own <b>makefile</b>.<li><bold>Example 4</bold> (discussed in detail later) is a <b>makefile</b> to perform registrations.<li> If <bold>Example 4</bold> is located in the <b>Makefile</b> <monospace>xfm.mk</monospace>, the help message for target <monospace>transforms</monospace> (highlighted) would appear along with other target help messages written for other targets (to execute several subject-level processing pipelines) as shown below.<li> This helps people find the targets that are defined by multiple <b>makefile</b>s in separate files.<li> This helps people find the targets that are defined by multiple <b>makefile</b>s in separate files.</p>        <preformat>&#xA0;&#xA0;resting.mk:20:&#xA0;rest&#xA0;--&#xA0;"Run&#xA0;resting&#xA0;&#xA0;state&#xA0;preprocessing&#xA0;pipeline"&#xA0;&#xA0;xfm.mk:3:&#xA0;transforms&#xA0;--&#xA0;Create&#xA0;resting&#xA0;&#xA0;state&#xA0;to&#xA0;MNI&#xA0;transformations&#xA0;&#xA0;fcconnectivity.mk:6:&#xA0;Connectivity&#xA0;--&#xA0;&#xA0;"Perform&#xA0;subject-level&#xA0;seed-based&#xA0;&#xA0;connectivity&#xA0;analysis"&#xA0;&#xA0;QA.mk:10:&#xA0;&#xA0;qa&#xA0;--&#xA0;&#xA0;Create&#xA0;QA&#xA0;report&#xA0;&#xA0;<b>Makefile</b>:42:&#xA0;all&#xA0;--&#xA0;Do&#xA0;skull&#xA0;stripping&#xA0;&#xA0;&#xA0;etiv&#xA0;&amp;&#xA0;HC&#xA0;volume&#xA0;calculation&#xA0;&#xA0;<b>Makefile</b>:55:&#xA0;robex&#xA0;--&#xA0;Alternate&#xA0;skull&#xA0;&#xA0;stripping&#xA0;with&#xA0;ROBEX&#xA0;&#xA0;<b>Makefile</b>:61:&#xA0;freesurferskullstrip&#xA0;--&#xA0;&#xA0;Alternate&#xA0;skull&#xA0;stripping&#xA0;with&#xA0;FreeSurfer&#xA0;&#xA0;<b>Makefile</b>:71:&#xA0;etiv&#xA0;--&#xA0;&#xA0;Estimation&#xA0;of&#xA0;ICV&#xA0;&#xA0;using&#xA0;enigma&#xA0;protocol&#xA0;&#xA0;<b>Makefile</b>:100:&#xA0;flex&#xA0;--&#xA0;&#xA0;Run&#xA0;flex&#xA0;for&#xA0;white&#xA0;&#xA0;matter&#xA0;hyperintensity&#xA0;quantification&#xA0;&#xA0;<b>Makefile</b>:142:&#xA0;clean&#xA0;--&#xA0;Clean&#xA0;up&#xA0;everything&#xA0;&#xA0;from&#xA0;all&#xA0;<b>makefile</b>s.<li>Debugging <b>makefile</b>s has a reputation for being rather difficult.<li> It has a very different model of execution than scripts (variables are, by default, evaluated at time of use, and targets within a <b>makefile</b> are not executed sequentially).<li>On the positive side, <b>makefile</b>s are naturally broken down into targets that can be tested individually.<li> We encourage building of a <b>makefile</b> by adding and testing one target at a time.<li>Our methods generator <b>makefile</b> acquires raw MRI acquisition parameters (e.g., repetition time TR, echo time TE) pertinent to reporting by calling a bash script that automatically pulls these parameters from the raw PAR/REC (a Philips-specific file format) data and stores them in a comma separated value (CSV) file.<li> Similarly, the versions of tools (e.g., FSL) as indicated by their installation directories and options for the implemented MRI preprocessing tools (e.g., smoothing kernel size), are pulled from the <b>makefile</b>s for specific processing streams and additionally stored in this CSV file.<li> The <b>makefile</b> renders this R Markdown file, automatically filling in the parameter values and generating an HTML output report.<li>Registrations are especially suited for execution with <b>makefile</b>s, because a registration process across several types of images involves multiple steps that can normally be calculated individually (in parallel) and concatenated.<li> The syntax of registration commands and the order of arguments can be difficult to remember, but these are abstracted by targets in the <b>makefile</b>.<li> Example <xref ref-type="fig" rid="E6"><u>4</u></xref> (See <xref ref-type="supplementary-material" rid="SM1"><u>Supplemental Materials</u></xref>: Testsubject Transformations for a detailed description) is an example of a <b>makefile</b> to create resting state to standard MNI space transformations.<li> When the <b>makefile</b> is read, the conditional will be evaluated and the dependency graph will include only the necessary components.<li> [2015](<person-group person-group-type="author"><name><surname>Graham-Cumming</surname><given-names>J.</given-names></name></person-group> (<year>2015</year>). <source>The GNU Make Book</source>. <publisher-loc>San Francisco, CA</publisher-loc>: <publisher-name>No Starch Press</publisher-name>.) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26423047">26423047</a> <b>(1)</b> Detrimental effects of duplicate reads and low complexity regions on RNA- and ChIP-seq data<ul><li> We provide a <b>Makefile</b> that automates generation of BED files containing genomic coordinates of all regions identified by RepeatMasker for GRCh37/hg19 and NCBI37/mm9 genomes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26282399">26282399</a> <b>(17)</b> Experiences with workflows for automating data-intensive bioinformatics<ul><li> We started with shell scripts and then moved to standard <b>Makefile</b>s and as an alternative, our own Python-based bioinformatics workflow system.<li> In our bioinformatics projects, the use of <b>Makefile</b>s alone is not enough.<li>While easy to use and construct, <b>Makefile</b>s are often not flexible enough - their support for parallel jobs cannot take multi-threaded or multi-process jobs into account, and they do not provide any usable means to describe a recursive flow, such as progressive application of multiple alignment for large datasets.<li> Some of the shortcomings can be overcome by using sub-<b>Makefile</b>s, however we thought it would be useful to develop a YAML-based workflow description system inspired by <b>Makefile</b>s.<li>Our aim is to build a workflow tool that is as simple as <b>Makefile</b>s, yet one that can make use of more complex functionality.<li> The major challenge is making the system feature-complete and as expressive as <b>Makefile</b>s without sacrificing the simplicity that is inherent in the alternatives.<li>The approaches for automating bioinformatics analysis by the organizations at the SeqAhead hackathons and workshops roughly fall into the following categories: scripting (usually in languages such as Bash, Perl, or Python), <b>Makefile</b>s (Make, CMake, etc.), and other workflow systems (such as Snakemake, Luigi, Galax, Taverna, and BcBio).<li><b>Makefile</b>.<li><b>Makefile</b>s.<li>The standard Unix/Linux solution for automating compilation and other tasks that follow a dependency graph are <b>Makefile</b>s [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>].<li> Drawbacks of <b>Makefile</b>s are their inability to describe dependencies between multiple output files per input file and a lack of support for multiple wildcards in I/O names.<li> These tools try to move beyond <b>Makefile</b>s while retaining the simplicity of GNU Make [<xref ref-type="bibr" rid="CR12"><u>12</u></xref>].<li>As <b>Makefile</b>s grow they tend to become very complex.<li> experience, <b>Makefile</b>s are good for simpler use cases, but have shortcomings when it comes to more complex workflows with multiple steps and branches.<li> While all acknowledge the power and importance of scientific workflow systems to enable reproducible data analysis and simplified integration with HPC systems, in practice it turns out that many projects have started using workflow tools and frameworks but later switched back to custom scripting and <b>Makefile</b>s (or similar) since they discovered limitations of the systems, especially with the pressure to deliver results faster in an internationally competitive environment.<li><b>Makefile</b>s are a quick way to get the work done in a seemingly efficient manner, but the standard Make tool can become limiting when more advanced features are required.<li> [11](          <person-group person-group-type="author">            <name>              <surname>Feldman</surname>              <given-names>SI</given-names>            </name>          </person-group>          <article-title>Make - a program for maintaining computer programs a program for maintaining computer programs</article-title>          <source>Softw Pract Experience</source>          <year>1979</year>          <volume>9</volume>          <issue>4</issue>          <fpage>255</fpage>          <lpage>65</lpage>          <pub-id pub-id-type="doi">10.1002/spe.4380090402</pub-id>        ) [12](Schwab M, Schroeder J. Reproducible research documents using gnumake. In: Stanford Exploration Project: 1995. p. 217&#x2013;26.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26268243">26268243</a> <b>(2)</b> XWAS: A Software Toolset for Genetic Data Analysis and Association Studies of the X Chromosome<ul><li> The entire package is freely available for download from <uri xlink:href="http://keinanlab.cb.bscb.cornell.edu">http://keinanlab.cb.bscb.cornell.edu</uri> and includes 1) scripts, 2) the binary executable XWAS, 3) all source code with a <b>Makefile</b>, 4) a user manual, and 5) example data and examples of running the different options offered by the package.<li> A <b>Makefile</b> is also provided to facilitate local compilation on Linux environments, and can also be adjusted for Windows and MAC OS by revising a few lines indicated therein.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26229594">26229594</a> <b>(2)</b> Deeply sequenced metagenome and metatranscriptome of a biogas-producing microbial community from an agricultural production-scale biogas plant<ul><li>The complete workflow is organized in a single GNU <b>Makefile</b> and available on GitHub [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> <italic>docker run</italic> executes the aforementioned <b>Makefile</b> inside the container.<li> [18](Bremges A, Belmann P, Sczyrba A. GitHub Repository. <ext-link ext-link-type="uri" xlink:href="https://github.com/metagenomics/2015-biogas-cebitec">https://github.com/metagenomics/2015-biogas-cebitec</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26020645">26020645</a> <b>(1)</b> UniqTag: Content-Derived Unique and Stable Identifiers for Gene Annotation<ul><li> The <b>Makefile</b> script that calculates the data used to plot Figs.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26042145">26042145</a> <b>(1)</b> FAST: FAST Analysis of Sequences Toolbox<ul><li> After downloading and unpacking the source directory, change into that directory and execute: <monospace>perl <b>Makefile</b>.PL; make; make test; (sudo) make install</monospace>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25901029">25901029</a> <b>(1)</b> Sequencing the cap-snatching repertoire of H1N1 influenza provides insight into the mechanism of viral transcription initiation<ul><li> The full pipeline is implemented as a <b>Makefile</b>, and all source code is available on the Bartel Lab website (<ext-link ext-link-type="uri" xlink:href="http://bartellab.wi.mit.edu/publication.html">http://bartellab.wi.mit.edu/publication.html</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25860802">25860802</a> <b>(1)</b> Strategies for Achieving High Sequencing Accuracy for Low Diversity Samples and Avoiding Sample Bleeding Using Illumina Platform<ul><li> Cycles used for template generation can be increased simply by editing the <b>Makefile</b> and adding --nd &lt;number of cycles for template&gt; and --first-detection-cycle &lt;the first cycle from which template will be generated&gt; where the image analysis executable is called.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25398475">25398475</a> <b>(1)</b> MaxSSmap: a GPU program for mapping divergent short reads to genomes with the maximum scoring subsequence<ul><li> Our OpenCL implementation has a <b>Makefile</b> to compile on AMD and Intel GPUs as well.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24532722">24532722</a> <b>(1)</b> fast_protein_cluster: parallel and optimized clustering of large-scale protein modeling data<ul><li> The source code, <b>Makefile</b> for Linux compilation, test set and documentation can be downloaded (<ext-link ext-link-type="uri" xlink:href="http://software.compbio.washington.edu/fast_protein_cluster">http://software.compbio.washington.edu/fast_protein_cluster</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24211736">24211736</a> <b>(8)</b> Hyb: A bioinformatics pipeline for the analysis of CLASH (crosslinking, ligation and sequencing of hybrids) data<sup><xref ref-type="fn" rid="d32e1012">&#x2606;</xref></sup><ul><li>Hyb is implemented as an executable &#x2018;<b>Makefile</b>&#x2019; that orchestrates the execution of a number of internal and external programs such as the read aligners and RNA folding software.<li> Many bioinformaticians are familiar with software development and routinely use <b>Makefile</b>s, which capture the provenance of their analysis.<li> <b>Makefile</b>s consist of goals (or tasks) and rules about how to create one file from another according to its dependencies or prerequisites.<li> The hyb pipeline was developed as a <b>Makefile</b> and could be invoked using the GNU &#x201C;make&#x201D; command:<disp-quote><p>make -f hyb.<li>To create an executable <b>Makefile</b>, the following line was used as the first line in a plain-text file:<disp-quote><p>#!/usr/bin/make -rRf.<li> The rest of the script is a conventional <b>Makefile</b>, but it is not necessary to invoke make explicitly on the command-line to run it.<li> Although a minor change from the way that <b>Makefile</b>s are conventionally used, this makes hyb considerably more flexible and easier to parameterise.<li> The ABySS paired-end assembler is, in fact, and executable <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24443382">24443382</a> <b>(2)</b> Exploring genome characteristics and sequence quality without a reference<ul><li> The computations performed in this article are fully reproducible by downloading and running the following <b>Makefile</b>:.<li>The <b>Makefile</b> will download the input data from public repositories, run SGA, and then generate the final reports.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24401750">24401750</a> <b>(1)</b> MOSAL: software tools for multiobjective sequence alignment<ul><li> A <b>makefile</b> is available for compilation under GNU/Linux.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24261882">24261882</a> <b>(2)</b> RaTrav: a tool for calculating mean first-passage times on biochemical networks<ul><li>There is a <b>Makefile</b> attached so under Linux it is sufficient to simply call <monospace>make</monospace>.<li> Alternatively the RaTrav tool may be compiled without the <b>Makefile</b>, e.g.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24270787">24270787</a> <b>(1)</b> The UCSC Genome Browser database: 2014 update<ul><li> To better support our Genome Browser mirror sites and source code users, we have rearchitected the software <b>makefile</b> system for our utilities and command-line tools to allow the compilation of specific tools independent of a full Browser installation.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24027569">24027569</a> <b>(1)</b> Inferring HIV Escape Rates from Multi-Locus Genotype Data<ul><li>The part of our method that is implemented in C and the python bindings can be built using <monospace>make</monospace> and the <b>Makefile</b> provided in the <monospace>src</monospace> directory.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23874174">23874174</a> <b>(1)</b> Hierarchical Modeling for Rare Event Detection and Cell Subset Alignment across Flow Cytometry Samples<ul><li><xref ref-type="supplementary-material" rid="pcbi.1003130.s003"><u>Software S1</u></xref> contains a <b>Makefile</b>, source code and scripts to generate the figures shown in the manuscript.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23762400">23762400</a> <b>(2)</b> diffReps: Detecting Differential Chromatin Modification Sites from ChIP-seq Data with Biological Replicates<ul><li>perl <b>Makefile</b>.PL (Optional, PREFIX&#x200A;=&#x200A;your_perl_directory).<li> If user specified PREFIX in <b>Makefile</b>, it will be installed in user_perl_directory/.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23587322">23587322</a> <b>(1)</b> <italic>AXIOME</italic>: automated exploration of microbial diversity<ul><li>All user-defined workflow analyses and parameters are outlined in a single XML configuration file, which is processed by AXIOME, generating a <b>Makefile</b> that controls and manages analyses.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23390549">23390549</a> <b>(1)</b> Robust Optimal Design of Experiments for Model Discrimination Using an Interactive Software Tool<ul><li> In addition to detailed installation instructions, getting started tutorials, Xcode project files and a Unix <b>makefile</b>, a template model file is provided to built and test the ModelDiscriminationToolkitGUI on different Unix based platforms.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23329413">23329413</a> <b>(1)</b> Grape RNA-Seq analysis pipeline environment<ul><li> through a template file (similar to a <b>Makefile</b>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25540463">25540463</a> <b>(1)</b> CrasyDSE: A framework for solving Dyson&#x2013;Schwinger equations<sup><xref ref-type="fn" rid="d32e1680">&#x2606;</xref></sup><ul><li> A very basic <b>makefile</b> for <italic>Unix</italic> using the <italic>g++</italic> GNU compiler is provided with each of the examples.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22937144">22937144</a> <b>(1)</b> GPU-FS-<italic>k</italic>NN: A Software Tool for Fast and Scalable <italic>k</italic>NN Computation Using GPUs<ul><li> For convenience, we provided a <b>makefile</b> and a sample data set in the source code directory.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22916000">22916000</a> <b>(2)</b> Evolution of Stress Response in the Face of Unreliable Environmental Signals<ul><li> file that specifies the arguments for main(), and <b>Makefile</b>), as well as R scripts for producing the final plots (<sup>*</sup>.R files).<li> needs to be specified in each <sup>*</sup>.cpp file as well as in each <b>Makefile</b> (variable MAIN).</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23012505">23012505</a> <b>(3)</b> Configuring a Context-Aware Middleware for Wireless Sensor Networks<ul><li> For TinyOS, it generates the high level configuration file (*.nc) with declarations and connections between the selected services and the <b>makefile</b> used to control the build process of a project.<li> <xref ref-type="fig" rid="f10-sensors-12-08544"><u>Figure 10(a)</u></xref> shows the configuration nesC file and the associated <b>makefile</b> is shown in <xref ref-type="fig" rid="f10-sensors-12-08544"><u>Figure 10(b)</u></xref>.<li> This configuration file (<italic>ConfigApp.nc</italic>) is called in the <b>Makefile</b> (<xref ref-type="fig" rid="f10-sensors-12-08544"><u>Figure 10(b)</u></xref>, line 2).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22727185">22727185</a> <b>(10)</b> URDME: a modular framework for stochastic simulation of reaction-transport processes in complex geometries<ul><li> URDME plug-in solvers have three main components: a <b>Makefile</b>, the solver source files, and (optionally) a pre-execution script intended to be invoked by the middle-level scripting interface.<li> The solver <b>Makefile</b> is used for compiling and building the solver automatically from the Matlab interface.<li> when urdme is invoked with the option to run a simulation using a specific solver, it will look for a <b>Makefile</b> with the correct naming pattern.<li> This <b>Makefile</b> then compiles the solver along with the propensity functions associated with the model being simulated into a stand-alone binary executable.<li>The source code of the solver itself can formally consist of any number of files in any language as long as the <b>Makefile</b> can create the final executable called by the middle-level interface.<li>All current solvers are written in ANSI-C and use GNU-style <b>Makefile</b>s.<li> URDME solvers have three main components: the solver source code, a <b>Makefile</b>, and an optional pre-execution script.<li> The <b>Makefile</b> creates a standalone Unix executable from the source code.<li><b>Makefile</b>.dfsp.<li>Solver <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22693624">22693624</a> <b>(2)</b> Developing and Applying Heterogeneous Phylogenetic Models with XRate<ul><li> The alignment of <italic>poliovirus</italic> genomes along with the grammars used to produce <xref ref-type="fig" rid="pone-0036898-g003"><u>Figure 3</u></xref> are also included along with a <b>Makefile</b> indicating how the data was analyzed.<li> Typing make help in the directory containing the <b>Makefile</b> will display the demonstrations available to users.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22559072">22559072</a> <b>(1)</b> Readjoiner: a fast and memory efficient string graph-based sequence assembler<ul><li>For our benchmarks, the 64bit <italic>GenomeTools</italic> binary was compiled by <italic>gcc</italic> version 4.3.2 using the provided <b>Makefile</b> with the option &#x201C;64bit&#x2009;=&#x2009;yes assert&#x2009;=&#x2009;no amalgamation&#x2009;=&#x2009;yes&#x201D;.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22540977">22540977</a> <b>(1)</b> Comprehensive comparison of graph based multiple protein sequence alignment strategies<ul><li><bold>Source code, <b>Makefile</b>, installation instructions and test alignments</bold>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21899774">21899774</a> <b>(2)</b> Agile parallel bioinformatics workflow management using Pwrake<ul><li> The <monospace>make </monospace>command interprets a <monospace><b>Makefile</b></monospace>, which defines dependencies between files in a declarative programming manner, and then generates the final target by resolving dependencies, by only executing out-of-date steps.<li> [<xref ref-type="bibr" rid="B1"><u>1</u></xref>], in <b>makefile</b>s without external tools.<li> [1](          <name>            <surname>Van der Aalst</surname>            <given-names>WMP</given-names>          </name>          <name>            <surname>Ter Hofstede</surname>            <given-names>AHM</given-names>          </name>          <name>            <surname>Kiepuszewski</surname>            <given-names>B</given-names>          </name>          <name>            <surname>Barros</surname>            <given-names>AP</given-names>          </name>          <article-title>Workflow patterns</article-title>          <source>Distrib Parallel Dat</source>          <year>2003</year>          <volume>14</volume>          <fpage>5</fpage>          <lpage>51</lpage>          <pub-id pub-id-type="doi">10.1023/A:1022883727209</pub-id>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21631914">21631914</a> <b>(2)</b> Faster Smith-Waterman database searches with inter-sequence SIMD parallelisation<ul><li>The source code for Farrar's STRIPED software was downloaded from the author's website and compiled with the GNU gcc compiler as specified in the supplied <b>Makefile</b> [<xref ref-type="bibr" rid="B11"><u>11</u></xref>].<li> The <b>Makefile</b> was also modified to compile STRIPED using the Intel compiler, to see if there were any differences in performance.<li> [11](          <name>            <surname>Farrar</surname>            <given-names>MS</given-names>          </name>          <article-title>Optimizing Smith-Waterman for the Cell Broadband Engine</article-title>          <ext-link ext-link-type="uri" xlink:href="http://sites.google.com/site/farrarmichael/SW-CellBE.pdf">http://sites.google.com/site/farrarmichael/SW-CellBE.pdf</ext-link>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21143816">21143816</a> <b>(3)</b> NGSQC: cross-platform quality analysis pipeline for deep sequencing data<ul><li> Users can also tweak some run parameters in the 'Environment Section' of the <b>makefile</b>.<li> The default port can be specified in <b>makefile</b> too.<li> To run the pipeline on cluster, a user needs to modify the 'CLUSTER' parameter to either 'SGE' or 'TORQUE' in <b>Makefile</b>, and run 'make'.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21085574">21085574</a> <b>(10)</b> MPI-PHYLIP: Parallelizing Computationally Intensive Phylogenetic Analysis Routines for the Analysis of Large Protein Families<ul><li> A <b>Makefile</b> and instructions are included in the distribution to compile and link the software on platforms where the standard MPI compiler wrappers (ie.<li> Next, copy the <b>Makefile</b>.mpicc file to a file called <b>Makefile</b> (&#x201C;cp <b>Makefile</b>.mpicc <b>Makefile</b>&#x201D;).<li> In general we have found that only minor modifications to the compile and link lines in the <b>Makefile</b> are required to run the code on parallel platforms that do not use the standard MPI wrappers (including the Altix platform used to run the test cases.) To compile on the Altix platform simply copy the <b>Makefile</b>.Altix file to a file called <b>Makefile</b> (&#x201C;cp <b>Makefile</b>.Altix <b>Makefile</b>&#x201D;).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20927408">20927408</a> <b>(2)</b> Neuroimaging Study Designs, Computational Analyses and Data Provenance Using the LONI Pipeline<ul><li> Finally, a key difference between the LONI Pipeline and most other environments is its management of distributed resources via its client-to-server infrastructure and its ability to export automated <b>makefile</b>s/scripts.<li> Currently, the LONI Pipeline allows exporting of any workflow from XML (*.pipe) format to a <b>makefile</b> or a shell script for direct or queuing execution.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20529898">20529898</a> <b>(1)</b> Time and memory efficient likelihood-based tree searches on phylogenomic alignments with missing data<ul><li> The program was compiled with <monospace>gcc v4.3.2</monospace> and the standard <b>Makefile</b> for the sequential SSE3 code that is distributed with the source files.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20515496">20515496</a> <b>(1)</b> Genome-wide identification of new Wnt/&#x3B2;-catenin target genes in the human genome using CART method<ul><li> Finally, <b>Makefile</b> is used to coordinate all computations using GNU make.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20406503">20406503</a> <b>(1)</b> Unsupervised binning of environmental genomic fragments based on an error robust selection of <italic>l</italic>-mers<ul><li> Decompress the package under Ubuntu or Debian environment, then run &#x201C;<b>Makefile</b>&#x201D; to install the software.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20230624">20230624</a> <b>(5)</b> RNAstructure: software for RNA secondary structure prediction and analysis<ul><li> Binaries for the text interfaces are available for Microsoft Windows and source code, including <b>Makefile</b>s, are available for download.<li> An included <b>Makefile</b> provides facility for this.<li> <b>Makefile</b>s are included for compiling the programs in a Unix/Linux/OS X environment.<li> A <b>Makefile</b> is provided to facilitate local compilation on Unix/Linux environments.<li> A <b>Makefile</b> automates testing of the text interfaces and comparison with calculation standards.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20041221">20041221</a> <b>(1)</b> A Quick Guide for Developing Effective Bioinformatics Programming Skills<ul><li> will read and execute any number of tasks from a <b>Makefile</b>, which are defined using a structured macro language.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19737395">19737395</a> <b>(3)</b> PROCOV: maximum likelihood estimation of protein phylogeny under covarion models and site-specific covarion pattern analysis<ul><li> </sup>Math Kernel Library) are included in the <b>Makefile</b>.<li> The <b>Makefile</b> should then be edited to match the type of the compiler and the path and library names of the BLAS and LAPACK libraries used.<li> The <b>Makefile</b> of the PROCOV source code gives some instances of the BLAS installation on a few commonly-used unix systems.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19668382">19668382</a> <b>(1)</b> Evolutionary Modeling and Prediction of Non-Coding RNAs in <italic>Drosophila</italic><ul><li> We break the analysis into a series of discrete steps, explicitly identifying dependencies using <b>Makefile</b> rules, in order to easily run on new data such as different alignments or genomes from other clades.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19649168">19649168</a> <b>(4)</b> Efficient, Distributed and Interactive Neuroimaging Data Analysis Using the LONI <italic>Pipeline</italic><ul><li> Finally, a key difference between the LONI <italic>Pipeline</italic> and some other environments is its management of distributed resources via its client-to-server infrastructure and its ability to export automated <b>makefile</b>s/scripts.<li> These include a 3-tier failover mechanism for Grid hardware, Sun Grid Engine (SGE)/Distributed Resource Management Application API (DRMAA) middleware, and the <italic>Pipeline</italic> server, as well as client-server communication, <b>makefile</b>/script export and data provenance model.<li> In addition, we need a flexible export of available pipeline workflows into <b>makefile</b>s and bash scripts that can be submitted virtually to any computational architecture.<li> Currently, the LONI <italic>Pipeline</italic> allows exporting of any workflow from XML (*.pipe) format to a <italic><b>makefile</b></italic> or a <italic>bash</italic> script for direct or queuing execution.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19014540">19014540</a> <b>(1)</b> A Web-based and Grid-enabled dChip version for the analysis of large sets of gene expression data<ul><li>By modifying the <b>Makefile</b> with the appropriate options it is very easy to obtain different versions of the application depending on the kind of infrastructure chosen for the analysis: standalone Linux, MPI or Grid-enabled versions.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18593718">18593718</a> <b>(1)</b> BioMAJ: a flexible framework for databanks synchronization and processing<ul><li> For SRS indexation, the parallelization has been directly included into the wrapper script using a parallel <b>makefile</b> program.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18577227">18577227</a> <b>(2)</b> CRANKITE: A fast polypeptide backbone conformation sampler<ul><li>The package provides a <b>Makefile</b> recognized by GNU <italic>make </italic>(or <italic>gmake</italic>), which builds the suite.<li> A <b>Makefile</b> to be used with GNU make was tested with GNU, Sun Studio, and MIPS Pro C compilers.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=17217507">17217507</a> <b>(1)</b> Parallelization of multicategory support vector machines (PMC-SVM) for classifying microarray data<ul><li># make the <b>makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=17137519">17137519</a> <b>(3)</b> Improvement in accuracy of multiple sequence alignment using novel group-to-group sequence alignment algorithm with piecewise linear gap cost<ul><li> To compile PRIME, one can check 'INSTALL' and '<b>Makefile</b>' in the archive.<li> Although '<b>Makefile</b>' basically assumes GNU make and g+ +, another compiler can be used with a few modification of '<b>Makefile</b>'.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16961919">16961919</a> <b>(1)</b> Scanning sequences after Gibbs sampling to find multiple occurrences of functional elements<ul><li> <b>Makefile</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16638137">16638137</a> <b>(2)</b> RNAmute: RNA secondary structure mutation analysis tool<ul><li> All files from the directory "RNAMute\RNAMute_progs" should be copied to "ViennaRNA-1.4\Progs" and compiled with "<b>makefile</b>".<li> "<b>Makefile</b>" that appears in the "ViennaRNA-1.4\Progs" directory should be overwritten.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15784138">15784138</a> <b>(2)</b> Complete reannotation of the <italic>Arabidopsis </italic>genome: methods, tools, protocols and the final release<ul><li> The <italic>Arabidopsis </italic>EGC pipeline consists of a single <b>Makefile</b> run nightly on a Linux server.<li> The <b>Makefile</b> runs a series of Perl scripts, each a wrapper around a bioinformatics tool responsible for launching an analysis (e.g.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15458579">15458579</a> <b>(1)</b> GeneXplorer: an interactive web application for microarray data visualization and analysis<ul><li>perl <b>Makefile</b>.PL.</ul></p>
<p><u>No PMID</u> <b>(1)</b> Computational Biology: Unix/Linux, Data Processing and Programming<ul><li> In a similar vein, the instructions for ClustalW include compilation using a <b>makefile</b>.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=11178259">11178259</a> <b>(2)</b> Accessing and distributing EMBL data using CORBA (common object request broker architecture)<ul><li><b>Makefile</b> is provided; it can be used to compile the IDL using one of three commonly used ORBs, i.e., ORBacus, Visibroker or OrbixWeb (save as <b>Makefile</b>).</ul></p>
<p><u>No PMID</u> <b>(3)</b> Manufacturer&#x2019;s CORBA Interface Testing Toolkit: Overview<ul><li> This is only one of the tools in MCITT, which also contains <bold>idlmkmf</bold> (&#x201C;IDL make <b>makefile</b>s&#x201D;) and <bold>TEd</bold> (the NIST Test Editor).<li><bold>Idlmkmf</bold> generates <b>makefile</b> rules for all of the IDL files in the working (current) directory, plus a few useful variable definitions.<li> Then it is only necessary to add rules specific to the clients and servers being built to produce a complete <b>makefile</b> for the CORBA application.</ul></p>
</ul></p>
</ul></p>
</ul></p>
</body>
