<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Nextflow</title>
</head>
<body>
<h1>Sentences that contain Nextflow</h1>
<p>2023 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36469327">36469327</a> RatesTools: a <b>Nextflow</b> pipeline for detecting de novo germline mutations in pedigree sequence data.<ul><li>The pipeline is implemented in <b>Nextflow</b> (Di Tommaso et al., 2017), Ruby (http://www.ruby-lang.org), Bash (https://www.gnu.org/software/bash/) and R (R Core Team, 2020) with reliance upon several other freely available tools. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36472574">36472574</a> annotate_my_genomes: an easy-to-use pipeline to improve genome annotation and uncover neglected genes by hybrid RNA sequencing.<ul><li>Our pipeline, implemented on Anaconda/<b>Nextflow</b> and Docker is an easy-to-use package that can be applied to a broad range of species, tissues, and research areas helping to improve and reconcile current annotations. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36553546">36553546</a> Comparison of Metagenomics and Metatranscriptomics Tools: A Guide to Making the Right Choice.<ul><li>Finally, we have developed two easy-to-use pipelines leveraging <b>Nextflow</b> technology. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36498841">36498841</a> hgtseq: A Standard Pipeline to Study Horizontal Gene Transfer.<ul><li>Thanks to the adoption of <b>Nextflow</b> for life sciences workflows, and to the standards and best practices curated by communities such as nf-core, fully automated, portable, and scalable pipelines can now be developed. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36399058">36399058</a> What the Phage: a scalable workflow for the identification and analysis of phage sequences.<ul><li>WtP is reproducible and scales to thousands of datasets through a workflow manager (<b>Nextflow</b>). </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36386613">36386613</a> MetaGT: A pipeline for <i>de novo</i> assembly of metatranscriptomes with the aid of metagenomic data.<ul><li>The pipeline is implemented in <b>NextFlow</b> and is freely available from https://github.com/ablab/metaGT.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36069454">36069454</a> MetaPhage: an Automated Pipeline for Analyzing, Annotating, and Classifying Bacteriophages in Metagenomics Sequencing Data.<ul><li>The pipeline is implemented in <b>Nextflow</b>, a widely adopted workflow manager that enables an optimized parallelization of tasks in different locations, from local server to the cloud; this ensures reproducible results from containerized packages. <li>The pipeline is implemented in <b>Nextflow</b>, a widely adopted workflow manager that enables an optimized parallelization of tasks. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36238687">36238687</a> scATACpipe: A <b>nextflow</b> pipeline for comprehensive and reproducible analyses of single cell ATAC-seq data.<ul><li>To fill this gap, we developed scATACpipe, a <b>Nextflow</b> pipeline, for performing comprehensive analyses of scATAC-seq data including extensive quality assessment, preprocessing, dimension reduction, clustering, peak calling, differential accessibility inference, integration with scRNA-seq data, transcription factor activity and footprinting analysis, co-accessibility inference, and cell trajectory prediction. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36212537">36212537</a> NanoRTax, a real-time pipeline for taxonomic and diversity analysis of nanopore 16S rRNA amplicon sequencing data.<ul><li>Here we present NanoRTax, a <b>Nextflow</b> pipeline for nanopore 16S rRNA gene amplicon data that features state-of-the-art taxonomic classification tools and real-time capability. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36071521">36071521</a> TMBur: a distributable tumor mutation burden approach for whole genome sequencing.<ul><li>TMBur is implemented in <b>Nextflow</b> and performs all analysis steps to generate TMB estimates directly from fastq files, incorporating somatic variant calling with Manta, Strelka2, and Mutect2, and microsatellite instability profiling with MSISensor. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35979446">35979446</a> A bioinformatic-assisted workflow for genome-wide identification of ncRNAs.<ul><li>Pinc uses <b>Nextflow</b> as a framework and is built with robust and well-established analysis tools. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35781332">35781332</a> HiC-TE: a computational pipeline for Hi-C data analysis to study the role of repeat family interactions in the genome 3D organization.<ul><li>To achieve this, we combined Hi-C data, reference genome data and tools for repeat analysis into a <b>Nextflow</b> pipeline identifying and quantifying the contacts of specific repeat families. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35758608">35758608</a> TIMSCONVERT: a workflow to convert trapped ion mobility data to open data formats.<ul><li>TIMSCONVERT and its documentation can be found at https://github.com/gtluu/timsconvert and is available as a standalone command-line interface tool for Windows and Linux, <b>NextFlow</b> workflow and online in the Global Natural Products Social (GNPS) platform.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35893066">35893066</a> A Comparison of Bioinformatics Pipelines for Enrichment Illumina Next Generation Sequencing Systems in Detecting SARS-CoV-2 Virus Strains.<ul><li>Certain limitations were identified in terms of pipeline algorithm, whereas it is highly recommended in future studies to design a pipeline in an integrated framework, for instance, by using <b>NextFlow</b>, a workflow framework to combine all scripts into one fully integrated pipeline.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35701420">35701420</a> Implementing the reuse of public DIA proteomics datasets: from the PRIDE database to Expression Atlas.<ul><li>Automation is orchestrated with <b>Nextflow</b>, using containerised open analysis software tools, rendering the pipeline readily available and reproducible. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35536253">35536253</a> BioUML-towards a universal research platform.<ul><li>For example, powerful genome browsers as well as viewers for molecular 3D structure are integrated in this way; (v) BioUML supports data analyses using workflows (own format, Galaxy, CWL, BPMN, <b>nextFlow</b>). </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35501696">35501696</a> GEMmaker: process massive RNA-seq datasets on heterogeneous computational infrastructure.<ul><li>GEMmaker, is a nf-core compliant, <b>Nextflow</b> workflow, that quantifies gene expression from small to massive RNA-seq datasets. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35241129">35241129</a> Enhanced protein isoform characterization through long-read proteogenomics.<ul><li>We have released an open-source <b>Nextflow</b> pipeline that integrates long-read sequencing in a proteomic workflow for isoform-resolved analysis.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35171290">35171290</a> Phylodynamic Inference of Bacterial Outbreak Parameters Using Nanopore Sequencing.<ul><li><b>Nextflow</b> pipelines implement SNP polisher training, evaluation, and outbreak alignments, enabling reconstruction of within-lineage transmission dynamics for infection control of bacterial disease outbreaks on portable nanopore platforms. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304334">36304334</a> CIRCprimerXL: Convenient and High-Throughput PCR Primer Design for Circular RNA Quantification.<ul><li>Our tool is both available as a user-friendly web tool and as a stand-alone pipeline based on Docker and <b>Nextflow</b>, allowing users to run the pipeline on a wide range of computer infrastructures. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35118380">35118380</a> nf-core/mag: a best-practice pipeline for metagenome hybrid assembly and binning.<ul><li>It is written in <b>Nextflow</b> and developed as part of the nf-core initiative for best-practice pipeline development. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35157051">35157051</a> riboviz 2: A flexible and robust ribosome profiling data analysis and visualization workflow.<ul><li>riboviz 2 includes an analysis workflow built on the <b>Nextflow</b> workflow management system for end-to-end processing of Ribo-seq data. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35207751">35207751</a> SWAAT Bioinformatics Workflow for Protein Structure-Based Annotation of ADME Gene Variants.<ul><li>The workflow consists of a set of Python codes of which the execution is managed within <b>Nextflow</b> to annotate coding variants based on 37 criteria. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35205129">35205129</a> Side-by-Side Comparison of Post-Entry Quarantine and High Throughput Sequencing Methods for Virus and Viroid Diagnosis.<ul><li>We implemented VirusReport, a new portable, scalable, and reproducible <b>nextflow</b> pipeline that analyses sRNA datasets to detect and identify viruses and viroids. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35081137">35081137</a> ASPICov: An automated pipeline for identification of SARS-Cov2 nucleotidic variants.<ul><li>To ensure FAIR data analysis, this <b>Nextflow</b> pipeline follows nf-core guidelines and use Singularity containers. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35022687">35022687</a> MeRIPseqPipe: An integrated analysis pipeline for MeRIP-seq data based on <b>Nextflow</b>.<ul><li>All the processes are embedded in <b>Nextflow</b> with Docker support, which ensures high reproducibility and scalability of the analysis. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35574063">35574063</a> nf-rnaSeqCount: A <b>Nextflow</b> pipeline for obtaining raw read counts from RNA-seq data.<ul><li>Using <b>Nextflow</b> as a workflow management system and Singularity for application containerisation, the nf-rnaSeqCount pipeline was developed for mapping raw RNA-seq reads to a reference genome and quantifying abundance of identified genomic features for differential gene expression analyses. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850839">34850839</a> Overcoming Data Bottlenecks in Genomic Pathogen Surveillance.<ul><li>The approaches taken include the use of reproducible data parsing pipelines and genome sequence analysis workflows, using technologies such as Data-flo, the <b>Nextflow</b> workflow manager, and containerization of software dependencies. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850837">34850837</a> Clones and Clusters of Antimicrobial-Resistant Klebsiella From Southwestern Nigeria.<ul><li>We conducted in silico multilocus sequence typing; AMR gene, virulence gene, plasmid, and K and O loci profiling; as well as phylogenetic analyses, using publicly available tools and <b>Nextflow</b> pipelines.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34946814">34946814</a> Comprehensive Analysis of Large-Scale Transcriptomes from Multiple Cancer Types.<ul><li>It was developed based on <b>Nextflow</b>, a reproducible workflow management system. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737383">34737383</a> Design considerations for workflow management systems use in production genomics research and the clinic.<ul><li>This work provides an approach and systematic evaluation of key features of popular bioinformatics WfMSs in use today: <b>Nextflow</b>, CWL, and WDL and some of their executors, along with Swift/T, a workflow manager commonly used in high-scale physics applications. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34736398">34736398</a> MSA: reproducible mutational signature attribution with confidence based on simulations.<ul><li>It comprises a set of Python scripts unified in a single <b>Nextflow</b> pipeline with containerisation for cross-platform reproducibility and scalability in high-performance computing environments. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34828348">34828348</a> ORPER: A Workflow for Constrained SSU rRNA Phylogenies.<ul><li>Therefore, we developed the <b>Nextflow</b> workflow "ORPER", for "ORganism PlacER", containerized in Singularity, which allows the determination the phylogenetic position of a collection of organisms in the genomic landscape. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34681040">34681040</a> FA-nf: A Functional Annotation Pipeline for Proteins from Non-Model Organisms Implemented in <b>Nextflow</b>.<ul><li>Here, we introduce FA-nf, a pipeline implemented in <b>Nextflow</b>, a versatile computational workflow management engine. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34625563">34625563</a> Characterisation of tumour microenvironment remodelling following oncogene inhibition in preclinical studies with imaging mass cytometry.<ul><li>We optimise and validate image segmentation strategies and automate the process in a <b>Nextflow</b>-based pipeline (imcyto) that is scalable and portable, allowing for parallelised segmentation of large multi-image datasets. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34383887">34383887</a> nf-LO: A Scalable, Containerized Workflow for Genome-to-Genome Lift Over.<ul><li>Here we present nf-LO (<b>nextflow</b>-LiftOver), a containerized and scalable <b>Nextflow</b> pipeline that enables lift overs within and between any species for which assemblies are available. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33492672">33492672</a> StellarPGx: A <b>Nextflow</b> Pipeline for Calling Star Alleles in Cytochrome P450 Genes.<ul><li>We present StellarPGx, a <b>Nextflow</b> pipeline for accurately genotyping CYP genes by combining genome graph-based variant detection, read coverage information from the original reference-based alignments, and combinatorial diplotype assignments. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34416914">34416914</a> ExOrthist: a tool to infer exon orthologies at any evolutionary distance.<ul><li>Here, we present ExOrthist, a fully reproducible <b>Nextflow</b>-based software enabling inference of exon homologs and orthogroups, visualization of evolution of exon-intron structures, and assessment of conservation of alternative splicing patterns. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34406368">34406368</a> ksrates: positioning whole-genome duplications relative to speciation events in KS distributions.<ul><li>ksrates is implemented in Python 3 and as a <b>Nextflow</b> pipeline. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35136576">35136576</a> Large-scale quality assessment of prokaryotic genomes with metashot/prok-quality.<ul><li>We present metashot/prok-quality, a container-enabled <b>Nextflow</b> pipeline for quality assessment and genome dereplication. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34394197">34394197</a> poreCov-An Easy to Use, Fast, and Robust Workflow for SARS-CoV-2 Genome Reconstruction <i>via</i> Nanopore Sequencing.<ul><li>Here we present poreCov, a highly parallel workflow written in <b>Nextflow</b>, using containers to wrap all the tools necessary for a routine SARS-CoV-2 sequencing lab into one program. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34900230">34900230</a> snpQT: flexible, reproducible, and comprehensive quality control and imputation of genomic data.<ul><li>To address these challenges, we have developed snpQT: a scalable, stand-alone software pipeline using <b>nextflow</b> and BioContainers, for comprehensive, reproducible and interactive quality control of human genomic data. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34255836">34255836</a> tsRNAsearch: A pipeline for the identification of tRNA and ncRNA fragments from small RNA-sequencing data.<ul><li>Here we introduce tsRNAsearch, a <b>Nextflow</b> pipeline for the identification of differentially abundant tRNA fragments and other non-coding RNAs from small RNA-sequencing data. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34153189">34153189</a> DIAproteomics: A Multifunctional Data Analysis Pipeline for Data-Independent Acquisition Proteomics and Peptidomics.<ul><li>Here we present DIAproteomics, a multifunctional, automated, high-throughput pipeline implemented in the <b>Nextflow</b> workflow management system that allows one to easily process proteomics and peptidomics DIA data sets on diverse compute infrastructures. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978761">33978761</a> The Dockstore: enhancing a community platform for sharing reproducible and accessible computational protocols.<ul><li>Dockstore is widely used, more than twenty-five high-profile organizations share analysis collections through the platform in a variety of workflow languages, including the Broad Institute's GATK best practice and COVID-19 workflows (WDL), nf-core workflows (<b>Nextflow</b>), the Intergalactic Workflow Commission tools (Galaxy), and workflows from Seven Bridges (CWL) to highlight just a few. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33580619">33580619</a> eDNAFlow, an automated, reproducible and scalable workflow for analysis of environmental DNA sequences exploiting <b>Nextflow</b> and Singularity.<ul><li>This pipeline is based on <b>Nextflow</b> and Singularity which enable a scalable, portable and reproducible workflow using software containers on a local computer, clouds and high-performance computing (HPC) clusters. <li>To our knowledge, this is the first automated bioinformatic pipeline for eDNA analysis using two powerful tools: <b>Nextflow</b> and Singularity. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34184051">34184051</a> BIGwas: Single-command quality control and association testing for multi-cohort and biobank-scale GWAS/PheWAS data.<ul><li>By using <b>Nextflow</b> workflow and Singularity software container technology, BIGwas performs resource-efficient and reproducible analyses on a local computer or any high-performance compute (HPC) system with just 1 command, with no need to manually install a software execution environment or various software packages. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34172000">34172000</a> MicroPIPE: validating an end-to-end workflow for high-quality complete bacterial genome construction.<ul><li>MicroPIPE is built in modules using Singularity container images and the bioinformatics workflow manager <b>Nextflow</b>, allowing changes and adjustments to be made in response to future tool development. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34020539">34020539</a> Comparison of high-throughput single-cell RNA sequencing data processing pipelines.<ul><li>In this study, we encapsulated seven existing high-throughput scRNA-seq data processing pipelines with <b>Nextflow</b>, a general integrative workflow management framework, and evaluated their performance in terms of running time, computational resource consumption and data analysis consistency using eight public datasets generated from five different high-throughput scRNA-seq platforms. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32735310">32735310</a> PoSeiDon: a <b>Nextflow</b> pipeline for the detection of evolutionary recombination events and positive selection.<ul><li>The pipeline is implemented in <b>Nextflow</b> with Docker support and processes the output of various tools.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978747">33978747</a> DetectIS: a pipeline to rapidly detect exogenous DNA integration sites using DNA or RNA paired-end sequencing data.<ul><li>The pipeline is based on a <b>Nextflow</b> workflow combined with a Singularity image containing all the necessary software, ensuring high reproducibility and scalability of the analysis. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978747">33978747</a> DetectIS: a pipeline to rapidly detect exogenous DNA integration sites using DNA or RNA paired-end sequencing data.<ul><li>The <b>Nextflow</b> workflow, the Singularity image and a test dataset are available at https://github.com/AstraZeneca/detectIS.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33772539">33772539</a> Testcrosses are an efficient strategy for identifying cis-regulatory variation: Bayesian analysis of allele-specific expression (BayesASE).<ul><li>The modular structure of BayesASE has been packaged in Galaxy, made available in <b>Nextflow</b> and as a collection of scripts for the SLURM workload manager on github (https://github.com/McIntyre-Lab/BayesASE).</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33777521">33777521</a> Reproducible, portable, and efficient ancient genome reconstruction with nf-core/eager.<ul><li>This <b>Nextflow</b> pipeline aims to address three main themes: accessibility and adaptability to different computing configurations, reproducibility to ensure robust analytical standards, and updating the pipeline to the latest routine ancient genomic practices. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a> Using prototyping to choose a bioinformatics workflow management system.<ul><li>To select a workflow management system, a rapid survey of available systems was undertaken, and candidates were shortlisted: Snakemake, cwltool, Toil, and <b>Nextflow</b>. <li>The selection process took 10 person-days, a small cost for the assurance that <b>Nextflow</b> satisfied the authors' requirements. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33563213">33563213</a> lncEvo: automated identification and conservation study of long noncoding RNAs.<ul><li>lncEvo is an all-in-one tool built with the <b>Nextflow</b> framework, utilizing state-of-the-art software and algorithms with customizable trade-offs between speed and sensitivity, ease of use and built-in reporting functionalities. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33561126">33561126</a> Metagenomics workflow for hybrid assembly, differential coverage binning, metatranscriptomics and pathway analysis (MUFFIN).<ul><li>The workflow is written by using <b>Nextflow</b>, a workflow orchestration software, to achieve high reproducibility and fast and straightforward use. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33416868">33416868</a> Diamond: A Multi-Modal DIA Mass Spectrometry Data Processing Pipeline.<ul><li>We developed Diamond, a <b>Nextflow</b>-based, containerized, multi-modal data-independent acquisition (DIA) mass spectrometry (MS) data processing pipeline for peptide identification and quantification. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35647481">35647481</a> nf-gwas-pipeline: A <b>Nextflow</b> Genome-Wide Association Study Pipeline.<ul><li>We developed an automated GWAS pipeline by combining multiple analysis tools - including bcftools, vcftools, the R packages SNPRelate/GENESIS/GMMAT and ANNOVAR - through <b>Nextflow</b>, which is a portable, flexible, and reproducible reactive workflow framework for developing pipelines. <li>The pipeline is implemented in <b>Nextflow</b>, dependencies are distributed through Docker, and the code is publicly available on Github.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33322033">33322033</a> RNAflow: An Effective and Simple RNA-Seq Differential Gene Expression Pipeline Using <b>Nextflow</b>.<ul><li>Here, we present a portable, scalable, and parallelizable <b>Nextflow</b> RNA-Seq pipeline to detect DEGs, which assures a high level of reproducibility. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33324401">33324401</a> Accurate and Efficient KIR Gene and Haplotype Inference From Genome Sequencing Reads With Novel K-mer Signatures.<ul><li>KPI is free, open, and easily executable as a <b>Nextflow</b> workflow supported by a Docker environment at https://github.com/droeatumn/kpi.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32864883">32864883</a> DeepRescore: Leveraging Deep Learning to Improve Peptide Identification in Immunopeptidomics.<ul><li>DeepRescore is developed using <b>NextFlow</b> and Docker and is available at https://github.com/bzhanglab/DeepRescore.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33193602">33193602</a> GenomeChronicler: The Personal Genome Project UK Genomic Report Generator Pipeline.<ul><li>GenomeChronicler can be run for one sample at a time, or in parallel making use of the <b>Nextflow</b> workflow manager. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32447016">32447016</a> TractoFlow: A robust, efficient and reproducible diffusion MRI pipeline leveraging <b>Nextflow</b> &amp; Singularity.<ul><li>TractoFlow supports the recent Brain Imaging Data Structure (BIDS) format as input and is based on two engines: <b>Nextflow</b> and Singularity. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32866155">32866155</a> GeneTEFlow: A <b>Nextflow</b>-based pipeline for analysing gene and transposable elements expression from RNA-Seq data.<ul><li>Here, we present GeneTEFlow, a fully automated, reproducible and platform-independent workflow, for the comprehensive analysis of gene and locus-specific TEs expression from RNA-Seq data employing <b>Nextflow</b> and Docker technologies. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32753501">32753501</a> Bactopia: a Flexible Pipeline for Complete Analysis of Bacterial Genomes.<ul><li>We created a new series of pipelines called Bactopia, built using <b>Nextflow</b> workflow software, to provide efficient comparative genomic analyses for bacterial species or genera. <li>Because the pipeline is written in the <b>Nextflow</b> language, analyses can be scaled from individual genomes on a local computer to thousands of genomes using cloud resources. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32641802">32641802</a> lentiMPRA and MPRAflow for high-throughput functional characterization of gene regulatory elements.<ul><li>Here, we provide a detailed protocol for lentiMPRA, along with a user-friendly <b>Nextflow</b>-based computational pipeline-MPRAflow-for quantifying CRS activity from different MPRA designs. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32442274">32442274</a> NanoSPC: a scalable, portable, cloud compatible viral nanopore metagenomic data processing pipeline.<ul><li>We implement NanoSPC using <b>Nextflow</b> manager within Docker images to allow reproducibility and portability of the analysis. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32561888">32561888</a> A scalable SCENIC workflow for single-cell gene regulatory network analysis.<ul><li>This protocol explains how to perform a fast SCENIC analysis alongside standard best practices steps on single-cell RNA-sequencing data using software containers and <b>Nextflow</b> pipelines. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31657527">31657527</a> Scalable Data Analysis in Proteomics and Metabolomics Using BioContainers and Workflows Engines.<ul><li>Finally, a new approach for reproducible and large-scale data analysis based on BioContainers and two of the most popular workflow environments, Galaxy and <b>Nextflow</b>, is introduced to the proteomics and metabolomics communities.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32306927">32306927</a> DolphinNext: a distributed data processing platform for high throughput genomics.<ul><li>DolphinNext facilitates building and deployment of complex pipelines using a modular approach implemented in a graphical interface that relies on the powerful <b>Nextflow</b> workflow framework by providing 1. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32306927">32306927</a> DolphinNext: a distributed data processing platform for high throughput genomics.<ul><li>DolphinNext is a flexible, intuitive, web-based data processing and analysis platform that enables creating, deploying, sharing, and executing complex <b>Nextflow</b> pipelines with extensive revisioning and interactive reporting to enhance reproducible results.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31882996">31882996</a> Bioinformatics pipeline using JUDI: Just Do It!<ul><li>Snakemake, <b>Nextflow</b>, Common Workflow Language, Galaxy, etc. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a> MasterOfPores: A Workflow for the Analysis of Oxford Nanopore Direct RNA Sequencing Datasets.<ul><li>The software is written using the <b>NextFlow</b> framework for parallelization and portability, and relies on Linux containers such as Docker and Singularity for achieving better reproducibility. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31647547">31647547</a> PhyloMagnet: fast and accurate screening of short-read meta-omics data using gene-centric phylogenetics.<ul><li>PhyloMagnet is built using <b>Nextflow</b>, available at github.com/maxemil/PhyloMagnet and is developed and tested on Linux. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants.<ul><li>Sarek is implemented in the <b>Nextflow</b> workflow language and supports both Docker and Singularity containers as well as Conda environments, making it ideal for easy deployment on any POSIX-compatible computers and cloud compute environments. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31755270">31755270</a> ThermoRawFileParser: Modular, Scalable, and Cross-Platform RAW File Conversion.<ul><li>To ensure the broadest possible availability and to increase integration capabilities with popular workflow systems such as Galaxy or <b>Nextflow</b>, we have also built Conda package and BioContainers container around ThermoRawFileParser. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31909794">31909794</a> CAMITAX: Taxon labels for microbial genomes.<ul><li>It uses <b>Nextflow</b> to orchestrate reference databases and software containers and thus combines ease of installation and use with computational reproducibility. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31316552">31316552</a> Pipeliner: A <b>Nextflow</b>-Based Framework for the Definition of Sequencing Data Processing Pipelines.<ul><li>It combines the <b>Nextflow</b> scripting language and Anaconda package manager to generate modular computational workflows. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> Scalable Workflows and Reproducible Data Analysis for Genomics.<ul><li>We show how to create a scalable, reusable, and shareable workflow using four different workflow engines: the Common Workflow Language (CWL), Guix Workflow Language (GWL), Snakemake, and <b>Nextflow</b>. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30419504">30419504</a> Pergola: Boosting Visualization and Analysis of Longitudinal Data by Unlocking Genomic Analysis Tools.<ul><li>They are also easy to reproduce thanks to their integration within <b>Nextflow</b>, a workflow manager using containerized software.</ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30486782">30486782</a> Developing reproducible bioinformatics analysis workflows for heterogeneous computing environments to support African genomics.<ul><li>Two of the workflows are built using the Common Workflow Language framework (CWL) and two using <b>Nextflow</b>. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30462158">30462158</a> The application of Hadoop in structural bioinformatics.<ul><li>Workflow Definition Language, Common Workflow Language and <b>Nextflow</b>) are taken up.</ul></p><p>2018 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a> [<b>Nextflow</b>, an efficient tool to improve computation numerical stability in genomic analysis].<ul><li><b>Nextflow</b> is a flow management framework that uses container technology to insure efficient deployment and reproducibility of computational analysis pipelines. <li>We used RNA-Seq quantification, genome annotation and phylogeny reconstruction examples to show how two seemingly irreproducible analyzes can be made stable across platforms when ported into <b>Nextflow</b>.</ul></p></body>
