<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Nextflow</title>
</head>
<body>
<h1>Sentences that contain Nextflow</h1>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36705549">36705549</a> <b>(10)</b> A molecular phenotypic map of malignant pleural mesothelioma<ul><li> using our <b>nextflow</b> [<xref rid="bib16" ref-type="bibr"><u>16</u></xref>] pipeline IARCbioinfo/fastqc-nf [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>]) to determine the reads base quality, adapter content, and duplication levels.<li>Paired-end read mapping was performed with our <b>nextflow</b> pipeline IARCbioinfo/alignment-nf v1.0 [<xref rid="bib18" ref-type="bibr"><u>18</u></xref>].<li>The resulting raw reads files were processed using our <b>nextflow</b> RNA-seq processing pipeline IARCbioinfo/RNAseq-nf v2.3 [<xref rid="bib19" ref-type="bibr"><u>19</u></xref>], as described previously [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>, <xref rid="bib20" ref-type="bibr"><u>20</u></xref>], that performs reads trimming (Trim Galore v0.6.5; <ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_011847" ext-link-type="uri">RRID:SCR_011847</ext-link>), and mapping to reference genome GRCh38 (gencode version 33) with STAR (v2.7.3a; <ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_004463" ext-link-type="uri">RRID:SCR_004463</ext-link>) [<xref rid="bib21" ref-type="bibr"><u>21</u></xref>].<li> We also improve the alignments as described previously by performing assembly-based realignment (<b>nextflow</b> pipeline IARCbioinfo/abra-nf v3.0 [<xref rid="bib22" ref-type="bibr"><u>22</u></xref>]), using software ABRA2 [<xref rid="bib23" ref-type="bibr"><u>23</u></xref>] (<ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_003277" ext-link-type="uri">RRID:SCR_003277</ext-link>) and base quality score recalibration (<b>nextflow</b> pipeline IARCbioinfo/BQSR-nf v1.1 [<xref rid="bib24" ref-type="bibr"><u>24</u></xref>]), using GATK v4.1.7.0 [<xref rid="bib25" ref-type="bibr"><u>25</u></xref>] (<ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_001876" ext-link-type="uri">RRID:SCR_001876</ext-link>).<li> <ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_016323" ext-link-type="uri">RRID:SCR_016323</ext-link>) (<b>nextflow</b> pipeline IARCbioinfo/RNAseq-transcript-nf v2.2 [<xref rid="bib26" ref-type="bibr"><u>26</u></xref>]).<li>Somatic copy number alterations were called using our <b>nextflow</b> workflow IARCbioinfo/purple-nf v1.0 [<xref rid="bib30" ref-type="bibr"><u>30</u></xref>] that implements the PURPLE [<xref rid="bib31" ref-type="bibr"><u>31</u></xref>, <xref rid="bib32" ref-type="bibr"><u>32</u></xref>] (<ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_022999" ext-link-type="uri">RRID:SCR_022999</ext-link>) software for matched and tumor-only WGS samples.<li>Point mutations were called using Mutect2 (<ext-link xlink:href="https://scicrunch.org/resolver/RRID:SCR_000559" ext-link-type="uri">RRID:SCR_000559</ext-link>) using our <b>nextflow</b> pipeline IARCbioinfo/mutect-nf v2.2b [<xref rid="bib35" ref-type="bibr"><u>35</u></xref>].<li> Our <b>nextflow</b> pipeline implementing the consensus variant calling approach for matched WGS is available in our IARCbioinfo/sv_somatic_cns GitHub repository [<xref rid="bib43" ref-type="bibr"><u>43</u></xref>].<li> NGSCheckMate was run using our <b>nextflow</b> implementation IARCbioinfo/NGSCheckMate-nf v1.1a [<xref rid="bib49" ref-type="bibr"><u>49</u></xref>].<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<comment>IARCbioinfo/fastqc-nf GitHub repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/fastqc-nf" ext-link-type="uri">https://github.com/IARCbioinfo/fastqc-nf</ext-link></comment>, (<comment>last accessed date: 12 January 2020</comment>).) [18](<comment>IARCbioinfo/alignment-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/alignment-nf" ext-link-type="uri">https://github.com/IARCbioinfo/alignment-nf</ext-link></comment>, (<comment>last accessed date: 12 January 2020</comment>).) [19](<comment>IARCbioinfo/RNAseq-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/RNAseq-nf" ext-link-type="uri">https://github.com/IARCbioinfo/RNAseq-nf</ext-link></comment>, (<comment>last accessed date:&#xA0;2 June 2020</comment>).) [10](<person-group person-group-type="author"><string-name><surname>Mangiante</surname><given-names>L</given-names></string-name>, <string-name><surname>Alcala</surname><given-names>N</given-names></string-name>, <string-name><surname>Di&#xA0;Genova</surname><given-names>A</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Multi-omic analysis of malignant pleural mesothelioma identifies molecular axes and specialized tumor profiles driving inter-tumor heterogeneity</article-title>. <source>Nat Genet</source>. <comment>In press</comment>.) [20](<person-group person-group-type="author"><string-name><surname>Gabriel</surname><given-names>AAG</given-names></string-name>, <string-name><surname>Mathian</surname><given-names>E</given-names></string-name>, <string-name><surname>Mangiante</surname><given-names>L</given-names></string-name>, <etal>et al.</etal></person-group><article-title>A molecular map of lung neuroendocrine neoplasms</article-title>. <source>Gigascience</source>. <year>2020</year>;<volume>9</volume>:<comment>giaa112</comment>.) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [22](<comment>IARCbioinfo/abra-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/abra-nf" ext-link-type="uri">https://github.com/IARCbioinfo/abra-nf</ext-link></comment>, (<comment>last accessed date: 2 June 2020</comment>).) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30649250">30649250</a>) [24](<comment>IARCbioinfo/BQSR-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/BQSR-nf" ext-link-type="uri">https://github.com/IARCbioinfo/BQSR-nf</ext-link></comment>, (<comment>last accessed date: 2 June 2020</comment>).) [25](<person-group person-group-type="author"><string-name><surname>Van&#xA0;der&#xA0;Auwera</surname><given-names>GA</given-names></string-name>, <string-name><surname>O'Connor</surname><given-names>BD</given-names></string-name></person-group>. <source>Genomics in the Cloud: Using Docker, GATK, and WDL in Terra</source>. <publisher-loc>Sebastopol, CA (USA)</publisher-loc>: <publisher-name>O'Reilly Media, Inc</publisher-name>; <year>2020</year>.) [26](<comment>IARCbioinfo/RNAseq-transcript-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/RNAseq-transcript-nf" ext-link-type="uri">https://github.com/IARCbioinfo/RNAseq-transcript-nf</ext-link></comment>, (<comment>last accessed date: 2 June 2020</comment>).) [30](<comment>IARCbioinfo/purple-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/purple-nf" ext-link-type="uri">https://github.com/IARCbioinfo/purple-nf</ext-link></comment>, (<comment>last accessed date: May 2021</comment>).) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31645765">31645765</a>) [32](<person-group person-group-type="author"><string-name><surname>Cameron</surname><given-names>DL</given-names></string-name>, <string-name><surname>Baber</surname><given-names>J</given-names></string-name>, <string-name><surname>Shale</surname><given-names>C</given-names></string-name>, <etal>et al.</etal></person-group><article-title>GRIDSS, PURPLE, LINX: unscrambling the tumor genome via integrated analysis of structural variation and copy number</article-title>. <source>bioRxiv</source>. <fpage>781013</fpage>. <pub-id pub-id-type="doi">10.1101/781013</pub-id>.) [35](<comment>IARCbioinfo/mutect-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/mutect-nf" ext-link-type="uri">https://github.com/IARCbioinfo/mutect-nf</ext-link></comment>, (last accessed date: <comment>6 October 2020)</comment>.) [43](<comment>IARCbioinfo/sv_somatic_cns-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/sv_somatic_cns-nf" ext-link-type="uri">https://github.com/IARCbioinfo/sv_somatic_cns-nf</ext-link></comment>, (last accessed date: <comment>3 November 2021</comment>).) [49](<comment>IARCbioinfo/NGSCheckMate-nf Github repository</comment>. <comment><ext-link xlink:href="https://github.com/IARCbioinfo/NGSCheckMate-nf" ext-link-type="uri">https://github.com/IARCbioinfo/NGSCheckMate-nf</ext-link></comment>, (last accessed date: <comment>12 July 2021)</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36721781">36721781</a> <b>(1)</b> Performance of methods for SARS-CoV-2 variant detection and abundance estimation within mixed population samples<ul><li> It is a collection of python and shell scripts and open-source software dependencies that are managed <italic toggle="yes">via</italic> Conda and <b>NextFlow</b> (<xref rid="ref-8" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author"><name><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>          <year iso-8601-date="2017">2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36694127">36694127</a> <b>(6)</b> nf-core/circrna: a portable workflow for the quantification, miRNA target prediction and differential expression analysis of circular RNAs<ul><li><b>Nextflow</b>, java,Docker/Apptainer.<li>To address these shortcomings, we present nf-core/circrna, a workflow for the quantification, miRNA target prediction and differential expression analysis of circRNAs in RNA-Seq data developed using <b>nextflow</b> [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>].<li> <b>Nextflow</b> automatically distributes each process in an isolated environment where task execution is performed within a container, ensuring efficient utilization and scaling of computing resources user-defined via configuration files.<li> Three dependencies are required to run nf-core/circrna: Java (<inline-formula id="IEq32"><alternatives><tex-math id="M61">\documentclass[12pt]{minimal}				\usepackage{amsmath}				\usepackage{wasysym} 				\usepackage{amsfonts} 				\usepackage{amssymb} 				\usepackage{amsbsy}				\usepackage{mathrsfs}				\usepackage{upgreek}				\setlength{\oddsidemargin}{-69pt}				\begin{document}$$&gt;=8$$\end{document}</tex-math><mml:math id="M62"><mml:mrow><mml:mo>&gt;</mml:mo><mml:mo>=</mml:mo><mml:mn>8</mml:mn></mml:mrow></mml:math><inline-graphic xlink:href="12859_2022_5125_Article_IEq32.gif"/></alternatives></inline-formula>), the latest version of <b>nextflow</b> [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>] and a software container client such as Docker [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>] or Apptainer (formerly Singularity) [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>].<li>Programming language: <b>Nextflow</b> DSL2.<li>Other requirements: Java (<inline-formula id="IEq42"><alternatives><tex-math id="M81">\documentclass[12pt]{minimal}				\usepackage{amsmath}				\usepackage{wasysym} 				\usepackage{amsfonts} 				\usepackage{amssymb} 				\usepackage{amsbsy}				\usepackage{mathrsfs}				\usepackage{upgreek}				\setlength{\oddsidemargin}{-69pt}				\begin{document}$$&gt;=8$$\end{document}</tex-math><mml:math id="M82"><mml:mrow><mml:mo>&gt;</mml:mo><mml:mo>=</mml:mo><mml:mn>8</mml:mn></mml:mrow></mml:math><inline-graphic xlink:href="12859_2022_5125_Article_IEq42.gif"/></alternatives></inline-formula>), <b>nextflow</b> v22.10.4,&#xA0;Docker/Apptainer.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [30](Docker. <ext-link ext-link-type="uri" xlink:href="https://www.docker.com/">https://www.docker.com/</ext-link>) [31](Apptainer. <ext-link ext-link-type="uri" xlink:href="https://apptainer.org/">https://apptainer.org/</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36639551">36639551</a> <b>(1)</b> Mechanical compression creates a quiescent muscle stem cell niche<ul><li> We analyzed the data using nf-co RNA-seq v 1.4.2 bioinformatics pipeline, built in <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/rnaseq/1.4.2">https://nf-co.re/rnaseq/1.4.2</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36299420">36299420</a> <b>(1)</b> Development of an amplicon-based sequencing approach in response to the global emergence of human monkeypox virus<ul><li> Primers were trimmed and consensus genomes were generated at a minimum depth of coverage of 5X using the Public Health Wales <b>Nextflow</b> nCoV-2019 pipeline which utilizes iVar [<xref rid="R47" ref-type="bibr"><u>47</u></xref>].<li> [47](<collab>Connor Lab</collab>. <source>ncov2019-artic-nf: A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools, with a focus on ncov2019</source>. [<date-in-citation>cited 4 Sep 2022</date-in-citation>]. <comment>Available</comment>: <comment><ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link></comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36639381">36639381</a> <b>(1)</b> ChIATAC is an efficient strategy for multi-omics mapping of 3D epigenomes from low-cell inputs<ul><li> The raw data were processed using the <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/hicar">https://nf-co.re/hicar</ext-link>) to get the 1D coverage.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36607439">36607439</a> <b>(1)</b> MethylC-analyzer: a comprehensive downstream pipeline for the analysis of genome-wide DNA methylation<ul><li> nf-core/methylseq is a workflow management system built by <b>Nextflow</b> that can run tasks across multiple computational infrastructures in a portable manner.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36609152">36609152</a> <b>(16)</b> A network-guided protocol to discover susceptibility genes in genome-wide association studies using stability selection<ul><li>All gwas-tools workflows are written in <b>Nextflow</b>,<xref rid="bib2" ref-type="bibr"><sup>7</sup></xref> a platform to handle scientific workflows in a platform-agnostic manner.<li>Install <b>Nextflow</b> following <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/getstarted.html" id="intref0030">its official documentation</ext-link>.<li>Configure it by creating a file called <b>nextflow</b>.config inside the directory from which the pipeline will run.<li> By setting it to local, <b>Nextflow</b> will perform the computations in the same computer in which <b>Nextflow</b> is launched.<li> However, by altering this value, <b>Nextflow</b> can handle common platforms to launch computations remotely, like SGE, SLURM or AWS (see the complete list in <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/executor.html" id="intref0035"><b>Nextflow</b>&#x2019;s documentation</ext-link>).<li><bold>Software and algorithms</bold></td></tr><tr><td>Docker</td><td><ext-link ext-link-type="uri" xlink:href="https://www.docker.com/" id="intref0120">https://www.docker.com</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://www.docker.com/" id="intref0125">https://www.docker.com</ext-link></td></tr><tr><td>gwas-tools</td><td>GitHub</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/hclimente/gwas-tools" id="intref0130">https://github.com/hclimente/gwas-tools</ext-link></td></tr><tr><td>gwas-tools Docker image</td><td>Docker Hub</td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hclimente/gwas-tools" id="intref0135">https://hub.docker.com/r/hclimente/gwas-tools</ext-link></td></tr><tr><td><b>Nextflow</b>.<li> (2017)<xref rid="bib2" ref-type="bibr"><sup>7</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>nextflow</b>.io/" id="intref0140">https://<b>nextflow</b>.io</ext-link>.<li>The workflow presented below can run on most computational environments, efficiently using the available resources if <b>Nextflow</b> is configured appropriately (see step 5.b in <xref rid="sec1.1" ref-type="sec"><u>install gwas-tools and its prerequisites</u></xref>).<li> (full list in <xref rid="mmc1" ref-type="supplementary-material"><u>Table&#xA0;S1</u></xref>) need to be preceded by two dashes (&#x2018;--&#x2019;), while <b>Nextflow</b> arguments like &#x2018;-with-docker&#x2019; or &#x2018;-resume&#x2019; (full list in <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/index.html" id="intref0075"><b>Nextflow</b>&#x2019;s documentation</ext-link>) are preceded by only one dash (&#x2018;-&#x2019;).<li>In step 5.b, replace docker.enabled&#xA0;= true by singularity.enabled&#xA0;= true in the <b>nextflow</b>.config file.<li> <b>Nextflow</b> automatically caches the intermediate results and tries to reuse them whenever possible.<li> You can find more information in <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/cli.html?highlight=resume" id="intref0100"><b>Nextflow</b>&#x2019;s documentation</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36684426">36684426</a> <b>(3)</b> Differentiation of human induced pluripotent stem cells into cortical neural stem cells<ul><li> The data pre-processing, including sequence QC, and STAR alignments were made with a custom <b>Nextflow</b> pipeline, which is publicly available at <ext-link xlink:href="https://github.com/wtsi-hgi/<b>nextflow</b>-pipelines/blob/rna_seq_5607/pipelines/rna_seq.nf" ext-link-type="uri">https://github.com/wtsi-hgi/<b>nextflow</b>-pipelines/blob/rna_seq_5607/pipelines/rna_seq.nf</ext-link>, including the specific aligner parameters.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36691616">36691616</a> <b>(1)</b> A wide foodomics approach coupled with metagenomics elucidates the environmental signature of potatoes<ul><li>Bacterial communities were identified through the same software (EPI2ME), which is based on <b>Nextflow</b>,<xref rid="bib41" ref-type="bibr"><sup>41</sup></xref> that enables scalable and flexible scientific analysis.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36602970">36602970</a> <b>(2)</b> Ten simple rules for using public biological data for your research<ul><li>Depending on the number of files and tool functionality and design, customized scripts or workflows (e.g., Snakemake [<xref rid="pcbi.1010749.ref038" ref-type="bibr"><u>38</u></xref>], <b>NextFlow</b> [<xref rid="pcbi.1010749.ref039" ref-type="bibr"><u>39</u></xref>]) may be needed for scalability.<li> This <b>Nextflow</b> pipeline [<xref rid="pcbi.1010749.ref039" ref-type="bibr"><u>39</u></xref>] uses tools such as SRA-tools [<xref rid="pcbi.1010749.ref036" ref-type="bibr"><u>36</u></xref>] to download FASTQ files and metadata based on a user-provided accession ID list.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [36](Sequence Read Archive Toolkit. [cited 2022 Jun 6]. Available from: <ext-link xlink:href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software" ext-link-type="uri">https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36598988">36598988</a> <b>(1)</b> In vitro cellular reprogramming to model gonad development and its disorders<ul><li> A copy of the code used for this step (<xref rid="R72" ref-type="bibr"><italic toggle="yes">72</italic></xref>) is freely available as a <b>Nextflow</b> pipeline and is available at Zenodo (<ext-link xlink:href="https://zenodo.org/record/4270402#.Y3zrhOzP3a0" ext-link-type="uri">https://zenodo.org/record/4270402#.Y3zrhOzP3a0</ext-link>) and the Github repository (<ext-link xlink:href="https://github.com/crickbabs/BABS-RNASeq.git" ext-link-type="uri">https://github.com/crickbabs/BABS-RNASeq.git</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36660474">36660474</a> <b>(4)</b> Perturbation of placental protein glycosylation by endoplasmic reticulum stress promotes maladaptation of maternal hepatic glucose metabolism<ul><li> 2008<xref rid="bib60" ref-type="bibr"><sup>60</sup></xref></td><td/></tr><tr><td><b>Nextflow</b>, nf-core/rnaseq.<li><hr/></td></tr><tr><td><b>Nextflow</b> corresponding software and versions.<li> The alignment and QC were processed using the <b>nextflow</b> (version 21.05.0.edge)<xref rid="bib72" ref-type="bibr"><sup>72</sup></xref> pipeline nf-core/rnaseq (version 3.2, https://nf-co.re/rnaseq; Ewels et&#xA0;al., 2020)<xref rid="bib61" ref-type="bibr"><sup>61</sup></xref> with the option &#x201C;--aligner star_salmon&#x201D; and Ensembl reference genome and annotation for human GRCh38.104.<li>There are 60504 genes identified using the <b>nextflow</b> pipeline in total with Ensembl Gene ID annotation.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36668762">36668762</a> <b>(1)</b> The Influence of Photodynamic Antimicrobial Chemotherapy on the Microbiome, Neuroendocrine and Immune System of Crustacean Post Larvae<ul><li> A quantitative assessment of taxonomic representation within each sample was done by a wf&#x2013;metagenomic <b>Nextflow</b> workflow in EPI2ME Labs provided by ONT using the NCBI 16S + 18S rRNA database.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36576131">36576131</a> <b>(4)</b> LMAS: evaluating metagenomic short <italic toggle="yes">de novo</italic> assembly methods through defined communities<ul><li> LMAS was implemented in <b>Nextflow</b> [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] to provide flexibility and ensure the transparency and reproducibility of the results.<li> The only 2 requirements for the addition of a new assembler are the execution command for the assembler for paired-end short-read data and a <b>Nextflow</b>-compatible container with the assembler and any dependencies.<li> As also suggested by CAMI, LMAS tracks the exact program version and command-line calls through its implementation in <b>Nextflow</b>.<li><bold>Programming languages:</bold> <b>Nextflow</b>, Python, Bash, Javascript.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36613910">36613910</a> <b>(1)</b> Exosome-Based Cell Homing and Angiogenic Differentiation for Dental Pulp Regeneration<ul><li> To quantitate known small RNAs and to identify novel miRNAs, we used the <b>Nextflow</b>-based workflow nf-core/smrnaseq (version v1.1.0, <uri xlink:href="https://github.com/nf-core/smrnaseq">https://github.com/nf-core/smrnaseq</uri>) (Accessed on 27 April 2022) as a small-RNA sequencing analysis pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36572683">36572683</a> <b>(1)</b> Transcriptional reprogramming from innate immune functions to a pro-thrombotic signature by monocytes in COVID-19<ul><li>Raw data was processed using the Sanger <b>Nextflow</b> RNA-seq pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35136669">35136669</a> <b>(9)</b> SCHOOL: Software for Clinical Health in Oncology for Omics Laboratories<ul><li> Furthermore, these workflows can execute on a local cluster using <b>Nextflow</b>,<xref rid="bb0025" ref-type="bibr"><sup>5</sup></xref> a command-line workflow manager, or on the cloud, <ext-link ext-link-type="uri" xlink:href="https://platform.dnanexus.com/panx/projects/FvPKK200Y9g81KqkKjJ9X818/data/" id="ir0005">https://platform.dnanexus.com/panx/projects/FvPKK200Y9g81KqkKjJ9X818/data/</ext-link>, using the DNANexus applets and workflows code.<li> Pipelines can be controlled on local computing resources using languages such as <b>Nextflow</b>, WDL, and Snakemate.<li> The workflow script and configuration were implemented for execution on a local high-performance computing cluster, using the workflow management program <b>Nextflow</b>, and on the cloud, using the DNANexus Toolkit.<li>For users with bioinformatics and computing expertise, <b>Nextflow</b> can be configured to run on a variety of platforms locally and on the cloud.<li> <b>Nextflow</b> readily submits jobs to commonly used compute cluster scheduling software such as SGE, PBS, and SLURM but also can be configured to submit jobs to cloud systems on Amazon Web Services (AWS) and Google Cloud.<li> Users can create a <b>Nextflow</b> configuration file to customize the workflow for their hardware.<li> Additionally, <b>Nextflow</b> allows for users to resume failed jobs and has extensive logging of each step, making troubleshooting easy to document.<li> Finally, these <b>Nextflow</b> workflows can be configured to run on individual tumor samples or tumor/normal sample pairs or in a batch mode for processing the data for an entire sequencing run.<li> SCHOOL can be easily installed on a local computing cluster that uses a queueing system such as SLURM or SGE with <b>Nextflow</b> with minimal dependency on pre-installed packages or runs on the cloud for laboratories that lack local resources and expertise with a user-friendly point-and-click graphic interface.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36550219">36550219</a> <b>(1)</b> Diverse silent chromatin states modulate genome compartmentalization and loop extrusion barriers<ul><li> Hi-C datasets were processed using the distiller pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/open2c/distiller-nf">https://github.com/open2c/distiller-nf</ext-link>) written for <b>nextflow</b><sup><xref ref-type="bibr" rid="CR80"><u>80</u></xref></sup>.<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36536091">36536091</a> <b>(1)</b> Nano3P-seq: transcriptome-wide analysis of gene expression and tail dynamics using end-capture nanopore cDNA sequencing<ul><li>For dRNA-seq reads, polyA tail length estimation was performed using NanoTail, a module from Master of Pores<sup><xref ref-type="bibr" rid="CR63"><u>63</u></xref></sup>, a <b>nextflow</b> workflow for the analysis of direct RNA datasets, which uses internally Nanopolish v0.11.1<sup><xref ref-type="bibr" rid="CR26"><u>26</u></xref></sup>.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26076426">26076426</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36527688">36527688</a> <b>(1)</b> Co-expression Gene Networks and Machine-learning Algorithms Unveil a Core Genetic Toolkit for Reproductive Division of Labour in Rudimentary Insect Societies<ul><li>We reprocessed all RNAseq raw reads in a standardized way using the publicly available <b>Nextflow</b> wrapper nf-core/rnaseq v.1.4.2 (<xref rid="evac174-B17" ref-type="bibr">di Tommaso et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36522333">36522333</a> <b>(1)</b> Multi-omics identify falling LRRC15 as a COVID-19 severity marker and persistent pro-thrombotic signals in convalescence<ul><li>Initial quality control and alignment was performed using the nf-core RNA-seq v3.2 pipeline<sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup> based on <b>nextflow</b><sup><xref ref-type="bibr" rid="CR76"><u>76</u></xref></sup>, a workflow management system.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36520686">36520686</a> <b>(20)</b> Ten simple rules and a template for creating workflows-as-applications<ul><li> Workflow managers, such as Snakemake [<xref rid="pcbi.1010705.ref004" ref-type="bibr"><u>4</u></xref>], <b>Nextflow</b> [<xref rid="pcbi.1010705.ref005" ref-type="bibr"><u>5</u></xref>], and Cromwell [<xref rid="pcbi.1010705.ref006" ref-type="bibr"><u>6</u></xref>] with WDL or CWL [<xref rid="pcbi.1010705.ref007" ref-type="bibr"><u>7</u></xref>], empower researchers to build robust pipelines that call a series of tools and scripts to perform a bespoke analysis.<li> However, the integration of cluster and cloud computing support in Snakemake, <b>Nextflow</b>, and Cromwell helped drive their current popularity.<li>Snakemake and <b>Nextflow</b> are two increasingly popular workflow management systems.<li>We present these ten simple rules for developers to follow which will facilitate their user communities in adopting and executing their workflows with fewer barriers, using examples with Snakemake and <b>Nextflow</b>.<li> Our templates also double as fully functioning examples of our ten simple rules and are available for both Snakemake (<ext-link xlink:href="http://github.com/beardymcjohnface/Snaketool" ext-link-type="uri">github.com/beardymcjohnface/Snaketool</ext-link>) and <b>Nextflow</b> (<ext-link xlink:href="http://github.com/beardymcjohnface/Nektool" ext-link-type="uri">github.com/beardymcjohnface/Nektool</ext-link>).<li><italic toggle="yes">// <b>Nextflow</b> example</italic>.<li><b>nextflow</b> run &lt;pipeline name&gt; <bold>-with-report -with-trace</bold>.<li> While user-customised configurations can be passed via the command line, immutable settings should be read directly by the workflow script (this is possible in both Snakemake and <b>Nextflow</b>).<li> In our example, we implement this functionality in Snakemake as additional bareword arguments, and in <b>Nextflow</b> by using the -entry flag.<li><italic toggle="yes"># Run preprocessing only (<b>Nextflow</b>)</italic>.<li><italic toggle="yes">// <b>Nextflow</b> example</italic>.<li> flags that are available for both Snakemake and <b>Nextflow</b>.<li> This feature is easy to implement in both Snakemake and <b>Nextflow</b>.<li><italic toggle="yes">// <b>Nextflow</b> example</italic>.<li> <b>Nextflow</b>: <ext-link xlink:href="http://nf-co.re/docs/usage/tutorials/step_by_step_institutional_profile" ext-link-type="uri">nf-co.re/docs/usage/tutorials/step_by_step_institutional_profile</ext-link>).<li> In our examples, the launcher achieves this by simply forwarding any unrecognised options to Snakemake or <b>Nextflow</b>.<li> Snakemake and <b>Nextflow</b> have many command line options and users should be encouraged to investigate these options for their situations.<li>&#xA0;&#xA0;<b>Nextflow</b> command:.<li>&#xA0;&#xA0;<b>nextflow</b> run&#x2026; <bold>-with-docker</bold>.<li> The templates will save even more time in this process as well as offering complete working examples of these ten simple rules for both Snakemake and <b>Nextflow</b>.<li> [4](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name>, <name><surname>Rahmann</surname><given-names>S</given-names></name>. <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source>. <year>2012</year>. Available from: <ext-link xlink:href="https://academic.oup.com/bioinformatics/article-abstract/28/19/2520/290322" ext-link-type="uri">https://academic.oup.com/bioinformatics/article-abstract/28/19/2520/290322</ext-link>.) [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [6](<name><surname>Van der Auwera</surname><given-names>GA</given-names></name>, <name><surname>O&#x2019;Connor</surname><given-names>BD</given-names></name>. <source>Genomics in the Cloud: Using Docker, GATK, and WDL in Terra</source>. <publisher-name>O&#x2019;Reilly Media, Inc.</publisher-name>; <year>2020</year>.) [7](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, <etal>et al</etal>. <source>Common workflow language, v1. 0</source>. <year>2016</year>. Available from: <ext-link xlink:href="https://www.research.manchester.ac.uk/portal/en/publications/common-workflow-language-v10(741919f5-d0ab-4557-9763-b811e911423b).html" ext-link-type="uri">https://www.research.manchester.ac.uk/portal/en/publications/common-workflow-language-v10(741919f5-d0ab-4557-9763-b811e911423b).html</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36517892">36517892</a> <b>(2)</b> STRling: a k-mer counting approach that detects short tandem repeat expansions at known and novel loci<ul><li> We provide workflows in three languages, <b>Nextflow</b> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>], Bpipe [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>], and WDL [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], for compatibility with cloud services such as AWS, Google Cloud, and Terra (<ext-link ext-link-type="uri" xlink:href="https://strling.readthedocs.io/en/latest/workflows.html">https://strling.readthedocs.io/en/latest/workflows.html</ext-link>).<li> Workflows are available in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>], Bpipe [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>], and WDL [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>] (<ext-link ext-link-type="uri" xlink:href="https://strling.readthedocs.io/en/latest/workflows.html">https://strling.readthedocs.io/en/latest/workflows.html</ext-link>).<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [39](Voss K, Van der Auwera G, Gentry J. Full-stack genomics pipelining with GATK4+ WDL+ Cromwell. F1000Res. 2017;6:1&#x2013;4.) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [39](Voss K, Van der Auwera G, Gentry J. Full-stack genomics pipelining with GATK4+ WDL+ Cromwell. F1000Res. 2017;6:1&#x2013;4.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36368318">36368318</a> <b>(4)</b> High-resolution single-cell atlas reveals diversity and plasticity of tissue-resident neutrophils in non-small cell lung cancer<ul><li> WTA Analysis Pipeline</td><td>Seven Bridges Genomics</td><td>v1.7.1</td></tr><tr><td>FlowJo&#x2122;</td><td>BD Biosciences</td><td>v10.7</td></tr><tr><td>Mantra Snap</td><td>Akoya Biosciences</td><td>v1.0.4</td></tr><tr><td>inForm Tissue Analysis</td><td>Akoya Biosciences</td><td>v2.4.10</td></tr><tr><td>GraphPad Prism</td><td>Graphpad</td><td>v9</td></tr><tr><td>Cellranger v5.0.0</td><td>10x Genomics</td><td><ext-link ext-link-type="uri" xlink:href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" id="intref0185">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger</ext-link></td></tr><tr><td>nf-core RNA-seq pipeline v3.0</td><td>(Philip et&#xA0;al., 2022)<xref rid="bib131" ref-type="bibr"><sup>131</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/rnaseq" id="intref0190">https://github.com/nf-core/rnaseq</ext-link></td></tr><tr><td><b>Nextflow</b> v22.04.5.<li> WTA Analysis Pipeline</td><td>Seven Bridges Genomics</td><td>v1.7.1</td></tr><tr><td>FlowJo&#x2122;</td><td>BD Biosciences</td><td>v10.7</td></tr><tr><td>Mantra Snap</td><td>Akoya Biosciences</td><td>v1.0.4</td></tr><tr><td>inForm Tissue Analysis</td><td>Akoya Biosciences</td><td>v2.4.10</td></tr><tr><td>GraphPad Prism</td><td>Graphpad</td><td>v9</td></tr><tr><td>Cellranger v5.0.0</td><td>10x Genomics</td><td><ext-link ext-link-type="uri" xlink:href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" id="intref0185">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger</ext-link></td></tr><tr><td>nf-core RNA-seq pipeline v3.0</td><td>(Philip et&#xA0;al., 2022)<xref rid="bib131" ref-type="bibr"><sup>131</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/rnaseq" id="intref0190">https://github.com/nf-core/rnaseq</ext-link></td></tr><tr><td><b>Nextflow</b> v22.04.5</td><td>(Di Tommaso et&#xA0;al., 2017)<xref rid="bib132" ref-type="bibr"><sup>132</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>nextflow</b>.io" id="intref0195">https://<b>nextflow</b>.io</ext-link>.<li> WTA Analysis Pipeline</td><td>Seven Bridges Genomics</td><td>v1.7.1</td></tr><tr><td>FlowJo&#x2122;</td><td>BD Biosciences</td><td>v10.7</td></tr><tr><td>Mantra Snap</td><td>Akoya Biosciences</td><td>v1.0.4</td></tr><tr><td>inForm Tissue Analysis</td><td>Akoya Biosciences</td><td>v2.4.10</td></tr><tr><td>GraphPad Prism</td><td>Graphpad</td><td>v9</td></tr><tr><td>Cellranger v5.0.0</td><td>10x Genomics</td><td><ext-link ext-link-type="uri" xlink:href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" id="intref0185">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger</ext-link></td></tr><tr><td>nf-core RNA-seq pipeline v3.0</td><td>(Philip et&#xA0;al., 2022)<xref rid="bib131" ref-type="bibr"><sup>131</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/rnaseq" id="intref0190">https://github.com/nf-core/rnaseq</ext-link></td></tr><tr><td><b>Nextflow</b> v22.04.5</td><td>(Di Tommaso et&#xA0;al., 2017)<xref rid="bib132" ref-type="bibr"><sup>132</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>nextflow</b>.io" id="intref0195">https://<b>nextflow</b>.io</ext-link></td></tr><tr><td>Singularity/Apptainer v3.7.0&#x2013;1.el7</td><td>(Kurtzer et&#xA0;al., 2017)<xref rid="bib133" ref-type="bibr"><sup>133</sup></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://apptainer.org/" id="intref0200">https://apptainer.org/</ext-link></td></tr><tr><td><b>Nextflow</b> workflow to reproduce this study.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36482348">36482348</a> <b>(1)</b> The genomic basis of copper tolerance in <italic>Drosophila</italic> is shaped by a complex interplay of regulatory and environmental factors<ul><li>RNA-seq analysis was performed using the <italic>rnaseq</italic> pipeline (v.1.2) from the <italic>nf-core</italic> community, a <italic><b>nextflow</b></italic> collection of curated bioinformatic pipelines [<xref ref-type="bibr" rid="CR92"><u>92</u></xref>, <xref ref-type="bibr" rid="CR93"><u>93</u></xref>].<li> [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36569062">36569062</a> <b>(1)</b> Prokaryotic and eukaryotic microbial diversity from three soda lakes in the East African Rift Valley determined by amplicon sequencing<ul><li>The 16S rRNA and ITS sequences were analyzed using the <b>Nextflow</b> computational pipeline ampliseq v1.1.2.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36480496">36480496</a> <b>(1)</b> Ten simple rules for investigating (meta)genomic data from environmental ecosystems<ul><li> Understanding the I/O files can help you extract, filter, or organize content within files through scripts, as well as help set up efficient workflows through the use of workflow management systems like Snakemake (<ext-link xlink:href="https://snakemake.readthedocs.io/" ext-link-type="uri">https://snakemake.readthedocs.io/</ext-link>) or <b>Nextflow</b> (<ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36472574">36472574</a> <b>(4)</b> annotate_my_genomes: an easy-to-use pipeline to improve genome annotation and uncover neglected genes by hybrid RNA sequencing<ul><li> This pipeline is implemented as an easy-to-use package on Anaconda/<b>NextFlow</b>/Docker platforms that integrates gold standard methods associated with transcriptome annotation.<li> The installation and execution of annotate_my_genomes pipeline can be achieved through the Conda package manager (<ext-link xlink:href="https://conda.io" ext-link-type="uri">https://conda.io</ext-link>) [<xref rid="bib63" ref-type="bibr"><u>63</u></xref>], as a self-contained pipeline via the <b>Nextflow</b> workflow framework (<ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>) [<xref rid="bib64" ref-type="bibr"><u>64</u></xref>], or via Docker (<ext-link xlink:href="https://www.docker.com/" ext-link-type="uri">https://www.docker.com/</ext-link>), using a ready-to-use image that contains all requirements to implement the pipeline.<li>Programming language: BASH, Python, <b>NextFlow</b>, Docker.<li> CF, AM and MGO implemented the pipeline in both Anaconda and <b>NextFlow</b> environments.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36473871">36473871</a> <b>(2)</b> The dynamic clustering of insulin receptor underlies its signaling and is disrupted in insulin resistance<ul><li>ChIP-seq bioinformatics analysis for insulin receptor was performed on the Whitehead High-Performance Computing Facility using the nf-core ChIP-seq pipeline v1.2.1<sup><xref ref-type="bibr" rid="CR112"><u>112</u></xref></sup> with <b>Nextflow</b> v20.04.1.<li>RNA-sequencing (RNA-seq) bioinformatics analysis was performed on the Whitehead High-Performance Computing Facility using the nf-core RNA-seq pipeline v1.4.2<sup><xref ref-type="bibr" rid="CR112"><u>112</u></xref></sup> with <b>Nextflow</b> v20.04.1.<li> [112](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [112](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36553546">36553546</a> <b>(6)</b> Comparison of Metagenomics and Metatranscriptomics Tools: A Guide to Making the Right Choice<ul><li> Both are running under <b>Nextflow</b> [<xref rid="B163-genes-13-02280" ref-type="bibr"><u>163</u></xref>], which allows parallel execution and for the software requirements to be tailored.<li> <b>Nextflow</b>.<li><b>Nextflow</b> is a reactive workflow framework and a programming domain-specific language (DSL) that eases the writing of data-intensive computational pipelines [<xref rid="B164-genes-13-02280" ref-type="bibr"><u>164</u></xref>].<li> The command-line tool, documentation and <b>Nextflow</b> source code are available at the GitHub repository <uri xlink:href="https://github.com/BioinfoIPBLN/16S-Metatranscriptomic-Analysis">https://github.com/BioinfoIPBLN/16S-Metatranscriptomic-Analysis</uri> (accessed on 17 October 2022).<li> In addition, this study was not conducted with our QIIME2 <b>Nextflow</b> tool because the reads do not come from rRNA.<li>In this work we have developed two pipelines based on <b>Nextflow</b> technology, which incorporate widely used software.<li> [163](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [164](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22384016">22384016</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36453861">36453861</a> <b>(1)</b> Pharokka: a fast scalable bacteriophage annotation tool<ul><li> With accessible bioconda installation, Pharokka is easily integrated into complex bioinformatic pipelines such as those created with Snakemake (<xref rid="btac776-B23" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</italic>, 2021</xref>) or <b>Nextflow</b> (<xref rid="btac776-B10" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36453996">36453996</a> <b>(5)</b> MirDIP 5.2: tissue context annotation and novel microRNA curation<ul><li> * indicates datasets analyzed using <b>Nextflow</b> pipeline.<li> * indicates datasets analyzed using <b>Nextflow</b> pipeline.<li> 14 remaining paired datasets were submitted to the <b>Nextflow</b> pipelines for processing, see Figures <xref rid="F1" ref-type="fig"><u>1B</u></xref> and&#xA0;<xref rid="F1" ref-type="fig"><u>C</u></xref>, respectively.<li> The raw data sets were processed using the <b>Nextflow</b> pipeline RNA-seq v.<li>For the datasets not run through the <b>Nextflow</b> pipeline, pre-processed data were downloaded from GEO, except for:.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36455041">36455041</a> <b>(3)</b> ERK1/2 signalling dynamics promote neural differentiation by regulating chromatin accessibility and&#xA0;the polycomb repressive complex<ul><li> <ext-link xlink:href="https://github.com/luslab/briscoe-nf-tobias" ext-link-type="uri">https://github.com/luslab/briscoe-nf-tobias</ext-link>) written in the <b>Nextflow</b> domain-specific language (version 21.04.0; [<xref rid="pbio.3000221.ref131" ref-type="bibr"><u>131</u></xref>]).<li> [<xref rid="pbio.3000221.ref132" ref-type="bibr"><u>132</u></xref>]) (<ext-link xlink:href="https://doi.org/10.5281/zenodo.3966161" ext-link-type="uri">https://doi.org/10.5281/zenodo.3966161</ext-link>), written in the <b>Nextflow</b> domain-specific language (version 19.10.0; [<xref rid="pbio.3000221.ref131" ref-type="bibr"><u>131</u></xref>]) was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0; [<xref rid="pbio.3000221.ref133" ref-type="bibr"><u>133</u></xref>]).<li> The command used was "<b>nextflow</b> run nf-core/chipseq&#x2014;input design.csv&#x2014;genome hg19&#x2014;gtf refseq_genes.gtf&#x2014;single_end&#x2014;min_reps_consensus 2 -profile crick -r 1.2.1".<li> [131](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [132](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [131](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [133](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36457147">36457147</a> <b>(11)</b> Characterization of protein isoform diversity in human umbilical vein endothelial cells via long-read proteogenomics<ul><li>Long-read RNA-seq analysis, MS searching, and proteogenomic analysis conducted using a <b>Nextflow</b> pipeline.<li>The long-read proteogenomics pipeline was implemented with <b>Nextflow</b>, a workflow framework which allows for scalable and reproducible computational analysis.<li> The <b>Nextflow</b> pipeline developed and described previously was used to process HUVEC collected PacBio data, translate the resulting transcripts into the protein database (see Deriving a HUVEC sample-specific protein isoform database below), and perform proteomics database searches [<xref rid="cit0022" ref-type="bibr"><u>22</u></xref>].<li> Further information on the workflow including individual modules of the <b>Nextflow</b> pipeline can be found at <ext-link xlink:href="https://github.com/sheynkman-lab/Long-Read-Proteogenomics" ext-link-type="uri">https://github.com/sheynkman-lab/Long-Read-Proteogenomics</ext-link> [<xref rid="cit0022" ref-type="bibr"><u>22</u></xref>].<li>After deriving a high confidence set of full-length transcript isoforms, within the <b>Nextflow</b> pipeline we select the most biologically plausible ORF for each of the Iso-Seq transcripts.<li> modules in the <b>Nextflow</b> pipeline.<li> module of the <b>Nextflow</b> pipeline.<li> module in the <b>Nextflow</b>).<li> A custom branch and Docker image were made as part of the <b>Nextflow</b> pipeline (GitHub: <ext-link xlink:href="https://github.com/smith-chem-wisc/MetaMorpheus/tree/LongReadProteogenomics" ext-link-type="uri">https://github.com/smith-chem-wisc/MetaMorpheus/tree/LongReadProteogenomics</ext-link>,Docker: <ext-link xlink:href="https://hub.docker.com/r/smithchemwisc/metamorpheus/tags?page=1%26ordering=last_updated" ext-link-type="uri">https://hub.docker.com/r/smithchemwisc/metamorpheus/tags?page=1&amp;ordering=last_updated</ext-link> tag: lrproteogenomics) based on MetaMorpheus version 0.0.316.<li>All RAW spectra files were first converted to mzML format with MSConvert prior to analysis with MetaMorpheus (see `mass_spec_raw_convert` module in the <b>Nextflow</b> pipeline).<li> The output of the data analysis including the long-read proteogenomics <b>Nextflow</b> workflow results generated using the mass spectrometry and long-read RNA-sequencing data as well as the post pipeline analysis results are available on Zenodo (<ext-link xlink:href="https://zenodo.org/record/7117445#.Y2FQE-wpD0o" ext-link-type="uri">https://zenodo.org/record/7117445#.Y2FQE-wpD0o</ext-link>).<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34980209">34980209</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34980209">34980209</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36449104">36449104</a> <b>(1)</b> Chemical and mechanical activation of resident cardiac macrophages in the living myocardial slice ex vivo model<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool which pre-processes raw data from FastQ inputs, aligns the reads and performs extensive quality control on the results.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36451166">36451166</a> <b>(6)</b> MoDLE: high-performance stochastic modeling of DNA loop extrusion interactions<ul><li> <xref rid="Fig2" ref-type="fig"><u>2</u></xref> were generated using the heatmap_comparison_pt1 <b>Nextflow</b> [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>] workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li> <xref rid="Fig2" ref-type="fig"><u>2</u></xref> were generated using the heatmap_comparison_pt1 <b>Nextflow</b> workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li>Benchmarks were run on a computing cluster using the run_benchmarks <b>Nextflow</b> workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li>The genome-wide optimization of parameters affecting extrusion barrier occupancies was carried out using the gw_param_optimization <b>Nextflow</b> workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li>The local extrusion barrier parameter optimization was carried out using the extrusion_barrier_param_optimization <b>Nextflow</b> workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li>Data for this section was generated using the comparison_with_mut <b>Nextflow</b> workflow available on GitHub [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and archived on Zenodo [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>].<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) [48](2021-modle-paper-001-data-analysis: Data analysis code for the first paper about MoDLE (preprint available soon) [Internet]. Github. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/paulsengroup/2021-modle-paper-001-data-analysis">https://github.com/paulsengroup/2021-modle-paper-001-data-analysis</ext-link>. [Cited 2022 Apr 11].) [49](Rossini R, Kumar V, Mathelier A, Rognes T, Paulsen J. Data analysis code for: &#x201C;MoDLE: High-performance stochastic modeling of DNA loop extrusion interactions.&#x201D; 2022. Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/7072939">https://zenodo.org/record/7072939</ext-link>. [Cited 2022 Sep 13].) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36451879">36451879</a> <b>(1)</b> SARS-CoV-2 evolves increased infection elicited cell death and fusion in an immunosuppressed individual<ul><li> In addition, we also used the wf_artic (ARTIC SARS-CoV-2) pipeline as built using the <b>nextflow</b> workflow framework57.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36450758">36450758</a> <b>(1)</b> Using coding and non-coding rare variants to target candidate genes in patients with severe tinnitus<ul><li> The main calling protocol was streamlined using <b>Nextflow</b> pipeline Sarek v.2.6.1<sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup> to ensure both portability and reproducibility.<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36446526">36446526</a> <b>(1)</b> Transcriptome analyses in infertile men reveal germ cell&#x2013;specific expression and splicing patterns<ul><li>We processed the raw sequence data with the <b>Nextflow</b> analysis pipeline nf-core/rnaseq 2.0 (<xref rid="bib10" ref-type="bibr"><u>Ewels et al, 2020</u></xref>) and annotated the transcripts with GENCODE release 36 genome annotation based on the GRCh38.p13 genome reference (<xref rid="bib11" ref-type="bibr"><u>Frankish et al, 2019</u></xref>).<li> [Ewels et al, 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Frankish et al, 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30357393">30357393</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36699347">36699347</a> <b>(1)</b> QuasiFlow: a <b>Nextflow</b> pipeline for analysis of NGS-based HIV-1 drug resistance data<ul><li> The tools are assembled into an automatic workflow using <b>Nextflow</b> (<xref rid="vbac089-B16" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36435939">36435939</a> <b>(1)</b> Single-cell multiomics reveals the complexity of TGF&#x3B2; signalling to chromatin in iPSC-derived kidney organoids<ul><li>SCENIC<sup><xref ref-type="bibr" rid="CR102"><u>102</u></xref></sup> was used to determine potential regulatory transcription factors through the VSN <b>nextflow</b> pipeline (v0.26.1)<sup><xref ref-type="bibr" rid="CR103"><u>103</u></xref></sup> using default settings according to the developer vignette (<ext-link ext-link-type="uri" xlink:href="https://github.com/aertslab/SCENIC">https://github.com/aertslab/SCENIC</ext-link>).<li> [102](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28991892">28991892</a>) [103](De Waegeneer, M., Flerin, C., Davie, K. &amp; Hulselmans, G. vib-singlecell-nf/vsn-pipelines: v0.26.1 (v0.26.1). <italic>Zenodo</italic>10.5281/zenodo.5070473 (2021).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36547892">36547892</a> <b>(1)</b> Functional Diversification of Oyster Big Defensins Generates Antimicrobial Specificity and Synergy against Members of the Microbiota<ul><li> SAMBA is developed using the <b>NextFlow</b> workflow manager [<xref rid="B53-marinedrugs-20-00745" ref-type="bibr"><u>53</u></xref>].<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36433966">36433966</a> <b>(2)</b> <italic>TractoInferno</italic> - A large-scale, open-source, multi-site database for machine learning dMRI tractography<ul><li> To make sure that every processing step was traceable and reproducible, a Singularity<sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref></sup> image was used along with the <b>Nextflow</b> pipeline<sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup>.<li>The whole pipeline was run using a Singularity container<sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref></sup> and <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup> for reproducibility.<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36420889">36420889</a> <b>(1)</b> DNA Data Bank of Japan (DDBJ) update report 2022<ul><li> It is based on Sapporo (<xref rid="B22" ref-type="bibr"><u>22</u></xref>), which is an implementation of the Global Alliance for Genomics and Health (GA4GH) WES standard (<ext-link xlink:href="https://ga4gh.github.io/workflow-execution-service-schemas/docs/" ext-link-type="uri">https://ga4gh.github.io/workflow-execution-service-schemas/docs/</ext-link>) and provides graphical interfaces to execute analytical pipelines described in a workflow language, such as <b>Nextflow</b>, Workflow Description Language, and Common Workflow Language, on the NIG supercomputer.<li> [22](<person-group person-group-type="author"><string-name><surname>Suetake</surname><given-names>H.</given-names></string-name>, <string-name><surname>Tanjo</surname><given-names>T.</given-names></string-name>, <string-name><surname>Ishii</surname><given-names>M.</given-names></string-name>, <string-name><surname>Kinoshita</surname><given-names>B.P.</given-names></string-name>, <string-name><surname>Fujino</surname><given-names>T.</given-names></string-name>, <string-name><surname>Hachiya</surname><given-names>T.</given-names></string-name>, <string-name><surname>Kodama</surname><given-names>Y.</given-names></string-name>, <string-name><surname>Fujisawa</surname><given-names>T.</given-names></string-name>, <string-name><surname>Ogasawara</surname><given-names>O.</given-names></string-name>, <string-name><surname>Shimizu</surname><given-names>A.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Sapporo: a workflow execution service that encourages the reuse of workflows in various languages in bioinformatics</article-title>. <source>F1000Res.</source><year>2022</year>; <volume>11</volume>:<fpage>889</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36498841">36498841</a> <b>(9)</b> hgtseq: A Standard Pipeline to Study Horizontal Gene Transfer<ul><li>The introduction of <b>Nextflow</b> [<xref rid="B40-ijms-23-14512" ref-type="bibr"><u>40</u></xref>], an increasingly popular workflow framework, allows the implementation of the above-mentioned values.<li> <b>Nextflow</b> is now largely adopted within research and industry contexts in life science, because of its features and flexibility.<li> Most importantly, one community of <b>Nextflow</b> users, called nf-core [<xref rid="B41-ijms-23-14512" ref-type="bibr"><u>41</u></xref>], offered the opportunity to converge around specific standards both in terms of coding style and conventions, as well as best practices for the use of bioinformatic tools widely adopted in life sciences and beyond.<li> Alternatively, the launch feature allows the user to access the <b>Nextflow</b> Tower launchpad (<uri xlink:href="https://cloud.tower.nf">https://cloud.tower.nf</uri> accessed on 24 October 2022) and manage the execution on the cloud through a graphical interface.<li>The execution on AWS, using the AWS-batch API which <b>Nextflow</b> supports, allowed confirmation of the portability and reproducibility of the code.<li>The first critical choice we made in order to achieve this goal was to develop the code using <b>Nextflow</b>.<li> <b>Nextflow</b> supports most schedulers for on-premise high-performance computing clusters, and all cloud infrastructure vendors: a pipeline developed with this workflow engine can therefore be run virtually anywhere, with just little changes to the configuration files.<li>In order to compose the pipeline, we have used Version 2 of the domain-specific language (DSL2) <b>Nextflow</b> [<xref rid="B40-ijms-23-14512" ref-type="bibr"><u>40</u></xref>].<li> This approach makes it possible to share <b>Nextflow</b> modules within the community across multiple users and pipelines.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36431075">36431075</a> <b>(1)</b> From Samples to Germline and Somatic Sequence Variation: A Focus on Next-Generation Sequencing in Melanoma Research<ul><li> In this way, for workflow standardization and automation, several managers are available, such as Snakemake [<xref rid="B107-life-12-01939" ref-type="bibr"><u>107</u></xref>], WDL [<xref rid="B108-life-12-01939" ref-type="bibr"><u>108</u></xref>], or <b>Nextflow</b> [<xref rid="B109-life-12-01939" ref-type="bibr"><u>109</u></xref>].<li> [107](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [108](          <person-group person-group-type="author"><name><surname>Voss</surname><given-names>K.</given-names></name><name><surname>Van der Auwera</surname><given-names>G.</given-names></name><name><surname>Gentry</surname><given-names>J.</given-names></name></person-group>          <article-title>Full-stack genomics pipelining with GATK4 + WDL + Cromwell</article-title>          <source>F1000Research</source>          <year>2017</year>          <volume>6</volume>          <pub-id pub-id-type="doi">10.7490/f1000research.1114634.1</pub-id>        ) [109](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36411286">36411286</a> <b>(1)</b> Regulome analysis in B-acute lymphoblastic leukemia exposes Core Binding Factor addiction as a therapeutic vulnerability<ul><li>ChIP and RNA-seq data were subjected to QC, mapped and counts matrices generated using <b>Nextflow</b> pipelines.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36414691">36414691</a> <b>(2)</b> Highly multiplexed spatial profiling with CODEX: bioinformatic analysis and application in human disease<ul><li><b>Nextflow</b>, Python.<li><b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36402995">36402995</a> <b>(8)</b> Analysis and Interpretation of metagenomics data: an approach<ul><li> YAMP is implemented in <b>NextFlow</b> [<xref ref-type="bibr" rid="CR87"><u>87</u></xref>] and is accompanied by a Docker [<xref ref-type="bibr" rid="CR88"><u>88</u></xref>] and a Singularity [<xref ref-type="bibr" rid="CR89"><u>89</u></xref>] container.<li>A workflow management system, <b>NextFlow</b>, was utilized to create YAMP, which has been used in numerous life-science initiatives [<xref ref-type="bibr" rid="CR92"><u>92</u></xref>, <xref ref-type="bibr" rid="CR93"><u>93</u></xref>].<li> Because of <b>NextFlow</b>'s user-transparent high-level parallelization, big applications are assured of scalability.<li><b>NextFlow</b> and metaflow|mics pipelines for microbiome marker data analysis.<li> To optimize the deployment of processes across different platforms, <b>NextFlow</b> was developed.<li> In addition to <b>Nextflow</b>, R, Python, Docker, and Singularly containers are used to spread the analytic pipelines [<xref ref-type="bibr" rid="CR97"><u>97</u></xref>].<li> Using <b>NextFlow</b>'s domain-specific language, the demultiplexing parallelization of pipelines is possible regardless of infrastructure.<li> By seamlessly moving data across computers and gathering outcomes from several actions performed simultaneously (or linearly if resources are limited), <b>NextFlow</b> simplifies parallelization and makes it easier to use [<xref ref-type="bibr" rid="CR87"><u>87</u></xref>].<li> [87](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [88](Forde JZ, Bussonnier M, Fortin F-A, Granger BE, Head TD, Holdgraf C, et al. Reproducing machine learning research on binder. NIPS Workshop 2018, Dec 3-8, Montreal, Canada.) [89](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29106441">29106441</a>) [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29073143">29073143</a>) [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=3224833">3224833</a>) [87](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36402955">36402955</a> <b>(12)</b> H3AGWAS: a portable workflow for genome wide association studies<ul><li> Using <b>Nextflow</b> and containerisation promotes scalability and portability.<li>The workflow has been developed in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>], with Python [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], bash and R scripts [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] and uses well-known bioinformatics tools.<li>List and description of <b>Nextflow</b> scripts by phase of GWAS.<li>The H3A<sc>GWAS</sc> workflow requires Java 8 or later and <b>Nextflow</b>, and can either be cloned from GitHub explicitly or run directly using <b>Nextflow</b>.<li> Since it uses <b>Nextflow</b> and containers, it can run on any environment that <b>Nextflow</b> supports such as PBS/Torque.<li>The QC and association workflows have been tested on Amazon Web Services (AWS) as well as Microsoft Azure using batch processing through <b>Nextflow</b>.<li>QC, Association testing integrating R packages SNPRelate/GENESIS/GMMAT and ANNOVAR using <b>Nextflow</b>.<li> Using <b>Nextflow</b> and containers, supports easy installation of the workflow and makes it portable and scalable&#x2014;from laptop to server to cloud (AWS and Azure).<li> Furthermore, different <b>Nextflow</b> scripts for each step allows the user to run analyses with different parameters and customise the analysis to their needs.<li>Program language: <b>Nextflow</b>, Python, R, bash.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [9](          <person-group person-group-type="author">            <name>              <surname>Van Rossum</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Drake</surname>              <given-names>FL</given-names>            </name>          </person-group>          <source>Python 3 reference manual</source>          <year>2009</year>          <publisher-loc>Scotts Valley</publisher-loc>          <publisher-name>CreateSpace</publisher-name>        ) [10](R Core Team. R: a language and environment for statistical computing. Vienna, Austria; 2020. <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/">https://www.R-project.org/</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36399058">36399058</a> <b>(9)</b> What the Phage: a scalable workflow for the identification and analysis of phage sequences<ul><li>WtP was implemented in <b>Nextflow</b>, a portable, scalable, and parallelizable workflow manager [<xref rid="bib27" ref-type="bibr"><u>27</u></xref>].<li> The modular code structure and functionalities of <b>Nextflow</b> and Docker/Singularity (Apptainer) allow easy integration of other phage prediction tools and additional analysis steps in future releases of the pipeline.<li>WtP requires the workflow management software <b>Nextflow</b> [<xref rid="bib27" ref-type="bibr"><u>27</u></xref>] and either Docker [<xref rid="bib39" ref-type="bibr"><u>39</u></xref>] or Singularity (Apptainer) [<xref rid="bib40" ref-type="bibr"><u>40</u></xref>] installed and configured on the system.<li> Each workflow release specifies the <b>Nextflow</b> version to avoid any version conflicts between the workflow code and the workflow manager.<li> A specific <b>Nextflow</b> version can be directly downloaded as an executable file from the <b>Nextflow</b> website.<li> The modular nature of the workflow using Docker and <b>Nextflow</b> allows the integration of new phage prediction tools by request of users, thus allowing WtP to keep up with the fast-developing field of bioinformatic tools.<li>Programming language: <b>Nextflow</b>, Bash, Python, R.<li>Other requirements: Ubuntu 18.04 LTS, Docker-version 20.10.12, <b>Nextflow</b>-version 21.10.6.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [39](<person-group person-group-type="author"><string-name><surname>Boettiger</surname><given-names>C</given-names></string-name></person-group>. <article-title>An introduction to Docker for reproducible research</article-title>. <source>ACM SIGOPS Oper Syst Rev</source>. <year>2015</year>;<volume>49</volume>(<issue>1</issue>):<fpage>71</fpage>&#x2013;<lpage>9</lpage>.) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36400796">36400796</a> <b>(1)</b> A comprehensive update to the M<italic>ycobacterium tuberculosis</italic> H37Rv reference genome<ul><li>A full reproducible pipeline encompassing all these steps herein was built within <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR44"><u>44</u></xref></sup> and can be found at github.com/alemenze/bact-builder.<li> [44](          <person-group person-group-type="author">            <name>              <surname>di Tommaso</surname>              <given-names>P</given-names>            </name>            <etal/>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nat. Biotechnol. 2017 35:4</source>          <year>2017</year>          <volume>35</volume>          <fpage>316</fpage>          <lpage>319</lpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36383075">36383075</a> <b>(1)</b> Genetic architecture of natural variation of cardiac performance from flies to humans<ul><li> FASTQ files were processed using nf-core/rnaseq (<xref rid="bib17" ref-type="bibr"><u>Ewels et al., 2021</u></xref>) with <b>Nextflow</b> Version: 21.03.0.edge.<li> [Ewels et al., 2021](          <person-group person-group-type="author"><name><surname>Ewels</surname><given-names>PA</given-names></name><name><surname>Peltzer</surname><given-names>A</given-names></name><name><surname>Fillinger</surname><given-names>S</given-names></name><name><surname>Patel</surname><given-names>H</given-names></name><name><surname>Alneberg</surname><given-names>J</given-names></name><name><surname>Wilm</surname><given-names>A</given-names></name><name><surname>Garcia</surname><given-names>MU</given-names></name><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Nahnsen</surname><given-names>S</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <data-title>nf-core/rnaseq</data-title>          <version designator="3">3</version>          <source>Zenodo</source>          <pub-id pub-id-type="doi">10.5281/zenodo.7130678</pub-id>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36458259">36458259</a> <b>(1)</b> Mutation accumulation in mtDNA of cancers resembles mutagenesis in normal stem cells<ul><li> Finally, read alignment was performed using the <b>Nextflow</b> Illumina Analysis Pipeline (NF-IAP, v1.2).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36382407">36382407</a> <b>(5)</b> MEGARes and AMR++, v3.0: an updated comprehensive database of antimicrobial resistance determinants and an improved software pipeline for classification using high-throughput sequencing<ul><li>Updated AMR++ 3.0: <b>NextFlow</b> and SnakeMake pipelines.<li>The <b>nextflow</b> scripting language (<xref rid="B34" ref-type="bibr"><u>34</u></xref>) was used to write AMR++ and this was transitioned to use of the DSL2 syntax extension for this 3.0 update, allowing for the pipeline to be organized into modules that can be strung together into workflows.<li> Using <b>nextflow</b> to write the AMR++ pipeline has numerous benefits, including the ability to use the &#x2018;-resume&#x2019; flag to restart a failed pipeline run.<li> and we have expanded the usability of AMR++ 3.0 by adding modulizable <b>NextFlow</b> workflows and a SnakeMake distribution.<li> All AMR++ <b>NextFlow</b> pipeline source files are freely available on GitHub at <ext-link xlink:href="https://github.com/Microbial-Ecology-Group/AMRplusplus" ext-link-type="uri">https://github.com/Microbial-Ecology-Group/AMRplusplus</ext-link>.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36452456">36452456</a> <b>(1)</b> Computational approaches for network-based integrative multi-omics analysis<ul><li> The report and code could be hosted in repositories (e.g., GitHub, Bitbucket, GitLab), reproducible scientific workflow management systems (e.g., <b>Nextflow</b>, Galaxy), environment sharing avenues (e.g., Conda, Docker), or packaged as libraries for programming languages (<xref rid="B11" ref-type="bibr"><u>Canzler et al., 2020</u></xref>).<li> [Canzler et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32034435">32034435</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36368923">36368923</a> <b>(1)</b> Wochenende&#xA0;&#x2014;&#xA0;modular and flexible alignment-based shotgun metagenome analysis<ul><li>We implemented a portable <b>Nextflow</b> version of Wochenende.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36514661">36514661</a> <b>(1)</b> An Early and Preliminary Assessment of the Clinical Severity of the Emerging SARS-CoV-2 Omicron Variants in Maharashtra, India<ul><li> The data were further processed using the wf-ARTIC workflow, a repository that contains the <b>nextflow</b> workflow for running the ARTIC SARS-CoV-2 workflow on multiplexed GridION runs, installed in MinKNOW software.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36357713">36357713</a> <b>(2)</b> Divergent SARS-CoV-2 variant emerges in white-tailed deer with deer-to-human transmission<ul><li>Paired-end illumina reads from ARTIC V4 and Fusion Genomics sequencing were initially analysed separately with the nf-core/viralrecon <b>Nextflow</b> workflow (v2.3) (refs.<li> Phylogenetic analyses were performed using CFIA-NCFAD/scovtree <b>Nextflow</b> workflow (v1.6.0) (<ext-link ext-link-type="uri" xlink:href="https://github.com/CFIA-NCFAD/scovtree/">https://github.com/CFIA-NCFAD/scovtree/</ext-link>) with the consensus sequences contextualized with closely related sequences identified by UShER and randomly sampled representative sequences from major WHO SARS-CoV-2 clades from GISAID<sup><xref ref-type="bibr" rid="CR12"><u>12</u></xref></sup> (downloaded 10 February 2022).<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28382917">28382917</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36439843">36439843</a> <b>(1)</b> The novel oligopeptide utilizing species <italic>Anaeropeptidivorans aminofermentans</italic> M3/9<sup>T</sup>, its role in anaerobic digestion and occurrence as deduced from large-scale fragment recruitment analyses<ul><li> Corresponding computations were scaled-up and parallelized in the de.NBI Cloud by using <b>Nextflow</b> 21.10.5 (<xref rid="ref11" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and the SimpleVM Cluster Mode which is based on Bibigrid.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36351451">36351451</a> <b>(1)</b> Multiple pathways for SARS-CoV-2 resistance to nirmatrelvir<ul><li> Consensus sequence generation was performed using the ONT Epi2Me ARTIC <b>Nextflow</b> pipeline v.0.3.16 (<ext-link ext-link-type="uri" xlink:href="https://github.com/epi2me-labs/wf-artic">https://github.com/epi2me-labs/wf-artic</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36389425">36389425</a> <b>(2)</b> The seeker R package: simplified fetching and processing of transcriptome data<ul><li> pyrpipe uses Python, whereas nf-core/rnaseq uses <b>Nextflow</b>.<li> In addition, unlike pyrpipe, seeker can run in parallel natively, without requiring a workflow manager such as <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36477201">36477201</a> <b>(8)</b> Extensive Recombination-driven Coronavirus Diversification Expands the Pool of Potential Pandemic Pathogens<ul><li>To investigate possible recombination-driven diversity among these viruses we developed IDPlot, which incorporates several distinct analysis steps into a single <b>Nextflow</b> workflow (<xref rid="evac161-B6" ref-type="bibr">Di Tommaso et al.<li>The development of IDPlot, a series of steps implemented in <b>Nextflow</b>, was motivated by reviewing the pain points of installing and running a common series of tools for sequence comparison.<li> In doing so, we created a <b>Nextflow</b> pipeline that requires just two FASTA files&#x2014;one FASTA file containing a sequence the user has identified as the reference sequence (the sequence that will be most prominent in the report) and another FASTA file containing one or many additional sequences to compare to the &#x201C;reference&#x201D; sequence.<li> With <b>Nextflow</b> and Docker installed on the user's system, the IDPlot workflow can be run on these two FASTA files using the following:.<li><b>nextflow</b> run brwnj/idplot -latest -with-docker \.<li><b>nextflow</b> run brwnj/idplot -latest -with-docker.<li><b>nextflow</b> run brwnj/idplot -latest -with-docker \.<li>Since IDPlot utilizes <b>Nextflow</b>, the workflow and dependency deployment are handled entirely in the background.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36366544">36366544</a> <b>(1)</b> Vaccination History, Body Mass Index, Age, and Baseline Gene Expression Predict Influenza Vaccination Outcomes<ul><li>Sequence reads were aligned to the human transcriptome using the nf-core/rnaseq <b>nextflow</b> pipeline [<xref rid="B17-viruses-14-02446" ref-type="bibr"><u>17</u></xref>] with default parameters.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36318584">36318584</a> <b>(2)</b> RNase L activation in the cytoplasm induces aberrant processing of mRNAs in the nucleus<ul><li>Our previously published RNA-seq data [<xref rid="ppat.1010930.ref013" ref-type="bibr"><u>13</u></xref>] were re-processed using a <b>Nextflow</b> pipeline (<ext-link xlink:href="https://github.com/Dowell-Lab/RNAseq-Flow" ext-link-type="uri">https://github.com/Dowell-Lab/RNAseq-Flow</ext-link>) as follows.<li> In particular, ERCC Spike-Ins reads were mapped using the same <b>Nextflow</b> pipeline to the ERCC sequences (<ext-link xlink:href="https://tools.thermofisher.com/content/sfs/manuals/ERCC92.zip" ext-link-type="uri">https://tools.thermofisher.com/content/sfs/manuals/ERCC92.zip</ext-link>), counted using featureCounts and the ERCC Spike-Ins size factors derived by DESeq2.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31494035">31494035</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36314806">36314806</a> <b>(1)</b> Analytical Sensitivity of Lateral Flow Devices against SARS-CoV-2 Omicron Subvariants BA.4, BA.5, and BA.2.75<ul><li> SARS-CoV-2 consensus sequences were generated using the ARTIC Sequencing report generated by the inbuilt <b>Nextflow</b> wf-artic supplied by Oxford Nanopore on their GridION instrument.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36366515">36366515</a> <b>(1)</b> Characterizing Longitudinal Antibody Responses in Recovered Individuals Following COVID-19 Infection and Single-Dose Vaccination: A Prospective Cohort Study<ul><li> Samples were sequenced on an Illumina NextSeq instrument (San Diego, CA, USA) using a tiled 1200bp amplicon scheme and analyzed using a modified ARCTIC <b>Nextflow</b> pipeline (<uri xlink:href="https://github.com/BCCDC-PHL/ncov2019-artic-nf">https://github.com/BCCDC-PHL/ncov2019-artic-nf</uri>, accessed on 15 September 2022).</ul></p>
<p><u>No PMID</u> <b>(1)</b> Single-cell Raman-activated sorting and cultivation (scRACS-Culture) for assessing and mining in situ phosphate-solubilizing microbes from nature<ul><li>A computational pipeline for single-microbial-cell genome was applied (<ext-link ext-link-type="uri" xlink:href="https://github.com/gongyh/nf-core-scgs">https://github.com/gongyh/nf-core-scgs</ext-link>) to efficiently analyze single-cell amplified bacterial genomes (SAGs) datasets, by integrating various tools with <b>Nextflow</b> [<xref ref-type="bibr" rid="CR78"><u>78</u></xref>].<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36386798">36386798</a> <b>(1)</b> Modification mapping by nanopore sequencing<ul><li> MasterofPores is a <b>NextFlow</b> pipeline developed by the Novoa lab that can both pre-process direct RNA sequencing data, and perform RNA modification detection using the tools Tombo and EpiNano (<xref rid="B17" ref-type="bibr"><u>Cozzuto et al., 2020</u></xref>).<li> [Cozzuto et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36386613">36386613</a> <b>(1)</b> MetaGT: A pipeline for <italic>de novo</italic> assembly of metatranscriptomes with the aid of metagenomic data<ul><li> The pipeline is implemented in <b>NextFlow</b> and is freely available at <ext-link xlink:href="https://github.com/ablab/metaGT" ext-link-type="uri">https://github.com/ablab/metaGT</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36300629">36300629</a> <b>(1)</b> IAnimal: a cross-species omics knowledgebase for animals<ul><li> First, an automatic download, analysis, and storage system for omics data was developed using technologies such as Docker, <b>Nextflow</b> (<xref rid="B15" ref-type="bibr"><u>15</u></xref>)&#xA0;and PostgreSQL.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36286237">36286237</a> <b>(3)</b> Rapid reconstruction of neural circuits using tissue expansion and light sheet microscopy<ul><li>We assembled these open-source tools into easy-to-use computational workflows via <b>Nextflow</b> (<xref rid="bib21" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>; <xref rid="fig2s1" ref-type="fig"><u>Figure 2&#x2014;figure supplements 1</u></xref>&#x2013;<xref rid="fig2s3" ref-type="fig"><u>3</u></xref>).<li> The use of <b>Nextflow</b> further reinforces that goal by allowing our computational workflows to execute on any compute cluster or cloud, including but not limited to IBM Platform LSF, SLURM, and AWS Batch.<li> By assembling the workflows into <b>Nextflow</b> pipelines, we also minimize the dependencies that are necessary for the user to install and provide a consistent command-line interface for invoking workflows and specifying runtime options.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36284317">36284317</a> <b>(6)</b> ReadZS detects cell type-specific and developmentally regulated RNA processing programs in single-cell RNA-seq<ul><li> and (5) a very efficient, convenient, and reproducible workflow implementation based on <b>Nextflow</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>], with all needed packages and libraries pre-installed.<li> ReadZS GB&#x2022;hours were calculated automatically for each <b>Nextflow</b> run in the <b>Nextflow</b> Tower interface.<li>Pipeline implementation using <b>Nextflow</b>.<li>To allow for reproducible and parallelizable results, the ReadZS pipeline is written in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>].<li> <b>Nextflow</b> is an open-source workflow management system that integrates command-line and scripting tools to analyze large-scale datasets.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36284361">36284361</a> <b>(3)</b> INSERT-seq enables high-resolution mapping of genomically integrated DNA using Nanopore sequencing<ul><li> We complement the experimental protocol compatible with longer read length with a computational peak calling pipeline implemented in <b>nextflow</b> and available through a web server.<li> Pipeline Implementation through <b>Nextflow</b> and Docker containers ensures easy installation, handling and high reproducibility.<li>The INSERT-seq pipeline has been implemented with <b>Nextflow</b> [<xref ref-type="bibr" rid="CR52"><u>52</u></xref>] and Docker [<xref ref-type="bibr" rid="CR53"><u>53</u></xref>] containerisation to allow an easy and reproducible analysis of the results.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [53](          <person-group person-group-type="author">            <name>              <surname>Merkel</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Docker: lightweight linux containers for consistent development and deployment</article-title>          <source>Linux J</source>          <year>2014</year>          <volume>2014</volume>          <fpage>2</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36483184">36483184</a> <b>(1)</b> Comparison of SARS-CoV-2 Viral Loads in the Nasal Mucosa of Patients Infected With BA.1, BA.2, or BA.5 Omicron Lineages<ul><li> The analysis of the WGS data, including dehosting, generation of a consensus sequence, and variant calling with freebayes, was performed using a <b>Nextflow</b> pipeline for running the ARTIC field bioinformatics tools (<ext-link xlink:href="https://github.com/jts/ncov2019-artic-nf" ext-link-type="uri">https://github.com/jts/ncov2019-artic-nf</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36298865">36298865</a> <b>(1)</b> SARS-CoV-2 from COVID-19 Patients in the Republic of Moldova: Whole-Genome Sequencing Results<ul><li> The demultiplexed sequences were then analyzed using the EPI2ME wf-artic pipeline running on the <b>Nextflow</b> software platform (version 21.10.4.5656) [<xref rid="B10-viruses-14-02310" ref-type="bibr"><u>10</u></xref>].<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36264838">36264838</a> <b>(1)</b> Ten simple rules for a successful international consortium in big data omics<ul><li> Workflow managers like Jip (<ext-link xlink:href="https://pyjip.readthedocs.io/en/latest/index.html" ext-link-type="uri">https://pyjip.readthedocs.io/en/latest/index.html</ext-link>), <b>Nextflow</b> [<xref rid="pcbi.1010546.ref014" ref-type="bibr"><u>14</u></xref>], and Snakemake [<xref rid="pcbi.1010546.ref015" ref-type="bibr"><u>15</u></xref>] can provide a way to describe the steps and share with others.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36627902">36627902</a> <b>(1)</b> Discovery of the inhibitor of DNA binding 1 as a novel marker for radioresistance in pancreatic cancer using genome-wide RNA-seq<ul><li>RNA-seq reads were quality-checked, trimmed, and aligned to the hg38 reference genome (accession: GCA_000001405.15) using the <b>Nextflow</b> RNAseq pipeline, nf-core/rnaseq (version 3.4) available at DOI 10.5281/zenodo.1400710.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36339268">36339268</a> <b>(1)</b> Mapping-friendly sequence reductions: Going beyond homopolymer compression<ul><li>All experiments performed for this article are implemented and documented as <b>nextflow</b> workflows available in this project&#x2019;s repository (<ext-link ext-link-type="uri" xlink:href="http://github.com/lucblassel/MSR_discovery" id="intref0075">github.com/lucblassel/MSR_discovery</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36213965">36213965</a> <b>(2)</b> Human Papillomavirus 42 Drives Digital Papillary Adenocarcinoma and Elicits a Germ Cell&#x2013;like Program Conserved in HPV-Positive Cancers<ul><li>Unprocessed reads from the virus sequence capture assay were processed with the <b>Nextflow</b> nf-core/viralrecon pipeline (<ext-link xlink:href="https://nf-co.re/viralrecon" ext-link-type="uri" xlink:show="new">https://nf-co.re/viralrecon</ext-link>, v2.4.1).<li> All subsequent processing was performed using the <b>Nextflow</b> pipeline virus-detection-nf found under <ext-link xlink:href="https://github.com/ObenaufLab/virus-detection-nf" ext-link-type="uri" xlink:show="new">https://github.com/ObenaufLab/virus-detection-nf</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36268023">36268023</a> <b>(1)</b> Evidence for the heterologous benefits of prior BCG vaccination on COVISHIELD&#x2122; vaccine-induced immune responses in SARS-CoV-2 seronegative young Indian adults<ul><li> Bulk ATACseq reads were pre-processed using the publicly available nfcore/atacseq pipeline [(<xref rid="B47" ref-type="bibr"><u>47</u></xref>), v1.2.1] implemented in <b>Nextflow</b> [(<xref rid="B48" ref-type="bibr"><u>48</u></xref>), v21.04.3] using default settings and the human GRCh38 genome.<li> [47](<person-group person-group-type="author"><name><surname>Ewels</surname><given-names>PA</given-names></name><name><surname>Peltzer</surname><given-names>A</given-names></name><name><surname>Fillinger</surname><given-names>S</given-names></name><name><surname>Patel</surname><given-names>H</given-names></name><name><surname>Alneberg</surname><given-names>J</given-names></name><name><surname>Wilm</surname><given-names>A</given-names></name><etal/></person-group>. <article-title>The nf-core framework for community-curated bioinformatics pipelines</article-title>. <source>Nat Biotech</source> (<year>2020</year>) <volume>38</volume>:<page-range>276&#x2013;8</page-range>. doi: <pub-id pub-id-type="doi">10.1038/s41587-020-0439-x</pub-id>) [48](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source> (<year>2017</year>) <volume>35</volume>:<page-range>316&#x2013;9</page-range>. doi: <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36194133">36194133</a> <b>(1)</b> SARS-CoV-2 3CL<sup>pro</sup> mutations selected in a VSV-based system confer resistance to nirmatrelvir, ensitrelvir, and GC376<ul><li> Sequences between 200 and 1200 bp were assembled with the algorithm epi2me-labs/wf-artic v0.3.18 in <b>Nextflow</b> 22.04.4.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36192798">36192798</a> <b>(3)</b> H3K18 lactylation marks tissue-specific active enhancers<ul><li>All genomic data were processed using pipelines built in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR72"><u>72</u></xref>] v21.04.3, adapted from the Babraham Institute GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/s-andrews/<b>nextflow</b>_pipelines">https://github.com/s-andrews/<b>nextflow</b>_pipelines</ext-link>) for reproducible data analysis.<li> [72](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36181613">36181613</a> <b>(2)</b> Progression of prostate cancer reprograms MYC-mediated lipid metabolism via lysine methyltransferase 2A<ul><li> Downloaded FASTQ files were reprocessed through the <b>nextflow</b> [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>] nf-core/chipseq (v1.2.2) pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/chipseq">https://github.com/nf-core/chipseq</ext-link>).<li> Downloaded FASTQ files were processed through the <b>nextflow</b> nf-core/rnaseq (v3.5) pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/rnaseq">https://github.com/nf-core/rnaseq</ext-link>).<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><u>No PMID</u> <b>(1)</b> Rapid evolution of SARS-CoV-2 in domestic cats<ul><li>Raw sequencing data in FASTQ file format were analyzed with a custom <b>Nextflow</b> (<xref rid="R14" ref-type="bibr">Di Tommaso et&#xA0;al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36238687">36238687</a> <b>(9)</b> scATACpipe: A <b>nextflow</b> pipeline for comprehensive and reproducible analyses of single cell ATAC-seq data<ul><li> The pipeline is powered by a state-of-art workflow management engine, <b>Nextflow</b> (<xref rid="B18" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), making it easy to be deployed to a variety of computing environments.<li>Powered by <b>Nextflow</b>, which was developed with a strong focus on portability, reproducibility, scalability, and usability (<xref rid="B18" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), scATACpipe integrates many carefully selected open-source software tools and custom scripts written in R, Python, or Bash for a comprehensive analysis of scATAC-seq data with up to millions of cells.<li> In accordance with the best practices of <b>Nextflow</b>, individual major processes focusing on specific tasks are modularized in scATACpipe.<li>Each major task in scATACpipe has been modularized in accordance with the best practices suggested by the <b>Nextflow</b> community.<li> An initial analysis of the scATAC-seq data was performed with paths to the reference genome sequence file and the GTF file being specified via <b>Nextflow</b>&#x2019;s command-line parameters and the modified configuration file.<li>To facilitate comprehensive and reproducible analysis of the most popular 10x Genomics scATAC data, we developed a <b>Nextflow</b> pipeline, scATACpipe (<xref rid="F1" ref-type="fig"><u>Figure 1</u></xref>).<li> To overcome these challenges, we have built a scalable, portable, and comprehensive pipeline, scATACpipe, using the <b>Nextflow</b> workflow management system.<li> Our scATACpipe allows users to configure nearly every possible parameter, including those for specifying computing resources, by setting command-line parameters, and by editing configuration files including workflow-level configuration file (<b>nextflow</b>.config) as well as module-level configuration files (conf/base.config and conf/modules.config).<li>In conclusion, scATACpipe is an open-source <b>Nextflow</b>-based pipeline for performing end-to-end analysis of large-scale scATAC-seq data.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36212537">36212537</a> <b>(4)</b> NanoRTax, a real-time pipeline for taxonomic and diversity analysis of nanopore 16S rRNA amplicon sequencing data<ul><li> Here we present NanoRTax, a <b>nextflow</b>-based pipeline for bacterial taxonomy classification and sample diversity analysis of nanopore full-length 16S rRNA amplicon reads.<li>NanoRTax is implemented in the <b>Nextflow</b> <xref rid="b0065" ref-type="bibr"><u>[13]</u></xref> workflow management system to enable efficient parallel execution and built-in integration of software dependencies using Docker containers and conda environments.<li><italic>Programming language</italic>: <b>Nextflow</b>, Python.<li>NanoRTax <b>Nextflow</b> pipeline and Python Dash web application are freely available under MIT license on Github <xref rid="b0195" ref-type="bibr"><u>[39]</u></xref> (<ext-link ext-link-type="uri" xlink:href="https://github.com/genomicsITER/NanoRTax" id="ir010">https://github.com/genomicsITER/NanoRTax</ext-link>).<li> [[13]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [[39]](Rodr&#xED;guez-P&#xE9;rez H, Ciuffreda L, Flores C. NanoRTax source code. 2021. <ext-link ext-link-type="uri" xlink:href="https://github.com/genomicsITER/NanoRTax" id="ir015">https://github.com/genomicsITER/NanoRTax</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36211387">36211387</a> <b>(2)</b> Cilia-related gene signature in the nasal mucosa correlates with disease severity and outcomes in critical respiratory syncytial virus bronchiolitis<ul><li> To facilitate reproducible analysis, samples were processed using the publicly available nf-core/RNA-seq pipeline version 1.4.2 implemented in <b>Nextflow</b> 19.10.0 using Singularity 3.2.1-1 with the minimal command <b>nextflow</b> run nf-core/rnaseq -r 1.4.2 &#x2013;singleEnd -profile singularity &#x2013;reverseStranded &#x2013;three_prime_clip_r2 3 (<xref rid="B26" ref-type="bibr"><u>26</u></xref>&#x2013;<xref rid="B28" ref-type="bibr"><u>28</u></xref>).<li> [26](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>(<issue>4</issue>):<page-range>316&#x2013;9</page-range>. doi: <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36086948">36086948</a> <b>(1)</b> Rapid genome sequencing for pediatrics<ul><li> the use of sophisticated workflow languages, such as <b>nextflow</b> (<ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>) and snakemake (Molder et al.,&#xA0;<xref rid="humu24466-bib-0048" ref-type="bibr"><u>2021</u></xref>); and simple solutions such as networking the sequencers to allow for direct saving of the data to the compute cluster, removing the need for lengthy transfer of raw data, which can also lead to corruption or loss of data.<li> [2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35544603">35544603</a> <b>(2)</b> Functional, structural, and molecular characterizations of the leukemogenic driver MEF2D-HNRNPUL1 fusion<ul><li> For data analysis, we used the <b>nextflow</b>-based workflow nf-core/chipseq (version 1.2.1) to analyze raw ChIP-seq reads with the command &#x201C;<b>nextflow</b> run nf-core/chipseq &#x2013;input sampleinfo.csv -profile singularity &#x2013;fasta GRCh38.primary_assembly.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36130276">36130276</a> <b>(1)</b> Highly significant improvement of protein sequence alignments with AlphaFold2<ul><li> The whole computational pipeline was implemented in the <b>Nextflow</b> language (<xref rid="btac625-B8" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36129445">36129445</a> <b>(1)</b> Impaired immune response drives age-dependent severity of COVID-19<ul><li> The nf-core/rnaseq pipeline (version 3.0 [<xref rid="bib19" ref-type="bibr"><u>Ewels et al., 2020</u></xref>]) written in the <b>Nextflow</b> domain-specific language (version 19.10.0 [<xref rid="bib16" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>]) was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0 [<xref rid="bib35" ref-type="bibr"><u>Kurtzer et al., 2017</u></xref>]).<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Kurtzer et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36129291">36129291</a> <b>(1)</b> Whole-Genome Sequence of <named-content content-type="genus-species">Cervid atadenovirus A</named-content> from the Initial Cases of an Adenovirus Hemorrhagic Disease Epizootic of Black-Tailed Deer in Canada<ul><li> The initial exploratory metagenomic analysis was carried out using a custom <b>Nextflow</b> workflow, nf-villumina v2.0.0 (<ext-link xlink:href="https://github.com/CFIA-NCFAD/nf-villumina" ext-link-type="uri">https://github.com/CFIA-NCFAD/nf-villumina</ext-link>), which automates low-quality read removal, read classification, <italic toggle="yes">de novo</italic> assembly, and BLAST homology searches of assembled contigs, as detailed in a previous publication (<xref rid="B8" ref-type="bibr"><u>8</u></xref>).<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33177604">33177604</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36125295">36125295</a> <b>(1)</b> Genome Sequencing of Methicillin-Resistant and Methicillin-Susceptible <italic toggle="yes">Mammaliicoccus sciuri</italic> from Diseased Animals<ul><li>Genomes were assembled using open-source scripts from ONT, including the EPI2MELABS wf-bacterial-genomes pipeline, and ran on the <b>nextflow</b> platform (<xref rid="B10" ref-type="bibr"><u>10</u></xref>).<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36123620">36123620</a> <b>(5)</b> msPIPE: a pipeline for the analysis and visualization of whole-genome bisulfite sequencing data<ul><li><b>Nextflow</b> methylseq (Bismark) [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>].<li><b>Nextflow</b>.<li><b>Nextflow</b> methylseq (bwa-meth) [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>].<li><b>Nextflow</b>.<li> <b>Nextflow</b> methylseq and msPIPE are the only pipelines that can automatically prepare and set the required genome sequences and annotation information of a reference based on the iGenomes [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and UCSC Genome Browser database [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>] respectively.<li> [36](Phil Ewels RH, Alexander P, Phue SF, Di Tommaso P, Garcia M, Alneberg J. Andreas Wilm, &amp; Alessia. nf-core/methylseq: nf-core/methylseq version 1.3 (1.3). Zenodo; 2019.) [36](Phil Ewels RH, Alexander P, Phue SF, Di Tommaso P, Garcia M, Alneberg J. Andreas Wilm, &amp; Alessia. nf-core/methylseq: nf-core/methylseq version 1.3 (1.3). Zenodo; 2019.) [48](iGenomes Database. <ext-link ext-link-type="uri" xlink:href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">https://support.illumina.com/sequencing/sequencing_software/igenome.html</ext-link>. Accessed 23 May 2022.) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33221922">33221922</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36109741">36109741</a> <b>(1)</b> Congenital tremor and splay leg in piglets &#x2013; insights into the virome, local cytokine response, and histology<ul><li>To analyse the produced datasets a <b>NextFlow</b> pipeline was implemented.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36089194">36089194</a> <b>(1)</b> Ionizing Radiation Drives Key Regulators of Antigen Presentation and a Global Expansion of the Immunopeptidome<ul><li>The <b>Nextflow</b> (v20.10.0) nf-core pipeline Sarek (v2.7) performed data processing and variant calling (<xref rid="bib16" ref-type="bibr"><u>16</u></xref>, <xref rid="bib17" ref-type="bibr"><u>17</u></xref>, <xref rid="bib18" ref-type="bibr"><u>18</u></xref>).<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36073358">36073358</a> <b>(1)</b> Deleterious Mutations and the Rare Allele Burden on Rice Gene Expression<ul><li> SNP/indel calling was performed using GATK v.4.0.1.2 implemented in a <b>Nextflow</b> pipeline (<ext-link xlink:href="https://github.com/zlye/RVE" ext-link-type="uri">https://github.com/zlye/RVE</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36071521">36071521</a> <b>(2)</b> TMBur: a distributable tumor mutation burden approach for whole genome sequencing<ul><li>The TMBur pipeline, implemented in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>], performs all analysis steps to generate TMB estimates from fastq files, including adapter trimming with fastp [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>], alignment with BWA mem 0.7.17 [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], and alignment sorting and duplicate marking with Samtools 1.9 [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>].<li>Schematic diagram of the <b>Nextflow</b> workflow for the singularity container used in the TMBur pipeline.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](          <person-group person-group-type="author">            <name>              <surname>Chen</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Zhou</surname>              <given-names>Y</given-names>            </name>            <name>              <surname>Chen</surname>              <given-names>Y</given-names>            </name>            <name>              <surname>Gu</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>fastp: an ultra-fast all-in-one FASTQ preprocessor</article-title>          <source>Bioinforma Oxf Engl</source>          <year>2018</year>          <volume>34</volume>          <issue>17</issue>          <fpage>i884</fpage>          <lpage>i890</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bty560</pub-id>        ) [16](Li H. Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. ArXiv13033997 Q-Bio [Internet]. 2013 May 26 [cited 2022 Feb 3]. <ext-link ext-link-type="uri" xlink:href="http://arxiv.org/abs/1303.3997">http://arxiv.org/abs/1303.3997</ext-link>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36142206">36142206</a> <b>(1)</b> Histamine Activates Human Eosinophils via H<sub>2</sub>R and H<sub>4</sub>R Predominantly in Atopic Dermatitis Patients<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36069454">36069454</a> <b>(3)</b> MetaPhage: an Automated Pipeline for Analyzing, Annotating, and Classifying Bacteriophages in Metagenomics Sequencing Data<ul><li> To guarantee scalability and reproducibility, MetaPhage was implemented in <b>Nextflow</b> (NF) (<xref rid="B8" ref-type="bibr"><u>8</u></xref>), a workflow manager that uses software containers to allow easy installation.<li> The pipeline is implemented in <b>Nextflow</b>, a scalable and reproducible scientific workflow manager that uses software containers to enable easy installation.<li>The pipeline was implemented in <b>Nextflow</b> (NF) [v21.04.0], a portable, scalable, and parallelizable workflow manager.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36065132">36065132</a> <b>(1)</b> On the way to plant data commons &#x2013; a genotyping use case<ul><li>To reduce the work the user has to put in to start the imputation, one can decide to use workflow languages like Common Workflow Language (CWL) [<xref rid="j_jib-2022-0033_ref_043" ref-type="bibr"><u>43</u></xref>] or <b>nextflow</b>.io [<xref rid="j_jib-2022-0033_ref_044" ref-type="bibr"><u>44</u></xref>].<li> [43](          <comment>Language (CWL). CW. Home</comment>          <article-title>Common work lang CWL</article-title>          <comment>[Internet]; n.d. Available from:</comment>          <ext-link xlink:href="https://www.commonwl.org/" ext-link-type="uri">https://www.commonwl.org/</ext-link>          <comment>[Accessed 15 June 2022]</comment>          <source>.</source>        ) [44](          <article-title>A DSL for parallel and scalable computational pipelines | Nextflow</article-title>          <comment>[Internet]; n.d. Available from:</comment>          <ext-link xlink:href="https://nextflow.io/" ext-link-type="uri">https://nextflow.io/</ext-link>          <comment>[Accessed 15 June 2022]</comment>          <source>.</source>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36140751">36140751</a> <b>(1)</b> Evaluation of the Available Variant Calling Tools for Oxford Nanopore Sequencing in Breast Cancer<ul><li> argument, which was formerly implemented inside the medaka package, it has been exceeded in accuracy and computing performance by alternative approaches and is, thus, deprecated and it is advised to utilize Clair3 either directly or through the Oxford Nanopore Technologies offered <b>Nextflow</b> implementation (Human-SNP-wf) (<uri xlink:href="https://github.com/nanoporetech/medaka">https://github.com/nanoporetech/medaka</uri> (accessed on 8 August 2022)) and that may explain the low performance.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36118859">36118859</a> <b>(1)</b> A high scale SARS-CoV-2 profiling by its whole-genome sequencing using Oxford Nanopore Technology in Kazakhstan<ul><li> Wf-artic is managed by <b>Nextflow</b> and is run using Docker (<xref rid="B3" ref-type="bibr"><u>Ewels et al., 2020</u></xref>).<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36110936">36110936</a> <b>(1)</b> Endocytosis of the thrombopoietin receptor Mpl regulates megakaryocyte and erythroid maturation in mice<ul><li> Bulk RNA-seq data were aligned to the Mus musculus mm10 genome and quality control was performed using <b>Nextflow</b> pipeline (nf-core/rnaseq 1.4.2) (DOI:10.5281/zenodo.1400710) (<xref rid="B52" ref-type="bibr"><u>52</u></xref>).<li> [52](<person-group person-group-type="author"><name><surname>Ewels</surname><given-names>PA</given-names></name><name><surname>Peltzer</surname><given-names>A</given-names></name><name><surname>Fillinger</surname><given-names>S</given-names></name><name><surname>Patel</surname><given-names>H</given-names></name><name><surname>Alneberg</surname><given-names>J</given-names></name><name><surname>Wilm</surname><given-names>A</given-names></name><etal/></person-group>. <article-title>The nf-core framework for community-curated bioinformatics pipelines</article-title>. <source>Nat Biotechnol</source> (<year>2020</year>) <volume>38</volume>(<issue>3</issue>):<page-range>276&#x2013;8</page-range>. doi: <pub-id pub-id-type="doi">10.1038/s41587-020-0439-x</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36039747">36039747</a> <b>(1)</b> RNA helicase-dependent gene looping impacts messenger RNA processing<ul><li>The analysis of sequencing data was based on a previous protocol (<xref rid="B76" ref-type="bibr"><u>76</u></xref>), details of the <b>Nextflow</b> (<xref rid="B77" ref-type="bibr"><u>77</u></xref>) pipeline are available in the following GitHub repository: <ext-link xlink:href="https://gitbio.ens-lyon.fr/LBMC/Bernard/quantitative-nucleosome-analysis/" ext-link-type="uri">https://gitbio.ens-lyon.fr/LBMC/Bernard/quantitative-nucleosome-analysis/</ext-link> and instructions in README_quantitativ-chip.md to reproduce our analysis).<li> [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26130708">26130708</a>) [77](<person-group person-group-type="author"><string-name><surname>Di&#xA0;Tommaso</surname><given-names>P.</given-names></string-name>, <string-name><surname>Chatzou</surname><given-names>M.</given-names></string-name>, <string-name><surname>Floden</surname><given-names>E.W.</given-names></string-name>, <string-name><surname>Barja</surname><given-names>P.P.</given-names></string-name>, <string-name><surname>Palumbo</surname><given-names>E.</given-names></string-name>, <string-name><surname>Notredame</surname><given-names>C.</given-names></string-name></person-group><article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat. Biotech.</source><year>2017</year>; <volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>319</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36040963">36040963</a> <b>(1)</b> Reconstructing the course of the COVID-19 epidemic over 2020 for US states and counties: Results of a Bayesian evidence synthesis model<ul><li> Finally, the <italic toggle="yes">dailyFlow</italic> repository uses the <b>Nextflow</b> workflow engine [<xref rid="pcbi.1010465.ref043" ref-type="bibr"><u>43</u></xref>] to clean the data, orchestrate 3200+ model runs within three supported execution environments (local, HPC, cloud), and export the results for research use and for web consumption.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36040985">36040985</a> <b>(2)</b> Study of the colonic epithelial-mesenchymal dialogue through establishment of two activated or not mesenchymal cell lines: Activated and resting ones differentially modulate colonocytes in co-culture<ul><li> coli, Yeast and PhiX.Reads mapping and variants calling were performed using <b>Nextflow</b> v20.10.0 and Sarek pipeline v2.6.1.<li> Out of <b>nextflow</b> pipeline the variants were annotated with VEP version 103.1 with assembly C57BL_6NJ_v1.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36038949">36038949</a> <b>(1)</b> Multiple genome alignment in the telomere-to-telomere assembly era<ul><li> While GPUs and large compute clusters, whose use has been made easier by workflow engines such as Toil [<xref ref-type="bibr" rid="CR133"><u>133</u></xref>] and <b>Nextflow</b> [<xref ref-type="bibr" rid="CR134"><u>134</u></xref>], may provide significant wall-clock time speedups in proportion to the resources available, they are only beneficial to those with the necessary resources.<li> [133](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [134](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36072757">36072757</a> <b>(1)</b> High-throughput scNMT protocol for multiomics profiling of single cells from mouse brain and pancreatic organoids<ul><li> The following steps are required for every cell, so it&#x2019;s advisable to use a workflow manager like Snakemake or <b>Nextflow</b> to process multiple cells in parallel.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36026470">36026470</a> <b>(2)</b> Genomic characterization of invasive typhoidal and non-typhoidal <italic toggle="yes">Salmonella</italic> in southwestern Nigeria<ul><li>Generated sequence reads from Illumina runs were <italic toggle="yes">de novo</italic> assembled following GHRU protocols (<ext-link xlink:href="https://gitlab.com/cgps/ghru/pipelines/dsl2/pipelines/assembly" ext-link-type="uri">https://gitlab.com/cgps/ghru/pipelines/dsl2/pipelines/assembly</ext-link>) using a <b>Nextflow</b> workflow which inclusively comprises of adapter trimming (trimmomatic v0.38), contamination detection (ConFindr v0.7.2), assembly (SPAdes v3.12.0), Quality Control (multiqc v1.7, qualifyr v1.4.4) and Bactinspector (v 0.1.3).<li> enterica serovar Enteritidis strain 18569 (assembly accession: GCF_000335875.2), respectively, to determine evolutionary relationship amongst the strains following GHRU <b>nextflow</b> SNP phylogeny protocols (<ext-link xlink:href="https://gitlab.com/cgps/ghru/pipelines/snp_phylogeny" ext-link-type="uri">https://gitlab.com/cgps/ghru/pipelines/snp_phylogeny</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35987708">35987708</a> <b>(1)</b> Comparison of SARS-CoV-2 spike RNA sequences in feces and nasopharynx indicates intestinal replication<ul><li> For strains in samples sequenced with Illumina, assembly and determination of SARS-CoV-2 lineages were made with the Arctic Networks nCoV-2019 novel coronavirus bioinformatics protocol running on a <b>nextflow</b> pipeline developed by Connor Lab [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>].<li> [40](ConnorLab. ncov2019-artic-nf: GitHub repository; 2021. [<ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>]) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35995124">35995124</a> <b>(1)</b> <italic toggle="yes">C. elegans</italic> toxicant responses vary among genetically diverse individuals<ul><li> We then wrote a <b>Nextflow</b> pipeline (<xref rid="R11" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) to run command-line instances of CellProfiler in parallel on the Quest High Performance Computing Cluster (Northwestern University) because each experimental block in this study produced many thousands of well images.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36090817">36090817</a> <b>(1)</b> Brain transcriptomic profiling reveals common alterations across neurodegenerative and psychiatric disorders<ul><li> For further sample processing, the Grape pipeline <xref rid="b0640" ref-type="bibr"><u>[128]</u></xref> was used for RNA-Seq analysis, with <b>Nextflow</b> <xref rid="b0645" ref-type="bibr"><u>[129]</u></xref> as the execution backend, the STAR aligner v.2.6.0a tool <xref rid="b0650" ref-type="bibr"><u>[130]</u></xref> for mapping reads to the human genome build hg19 with GENCODE v.28 annotations, and the RSEM tool <xref rid="b0655" ref-type="bibr"><u>[131]</u></xref> for isoform quantification (using default options).<li> [[128]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23329413">23329413</a>) [[129]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [[130]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [[131]](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Dewey</surname>              <given-names>C.N.</given-names>            </name>          </person-group>          <article-title>RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome</article-title>          <source>BMC Bioinf</source>          <volume>12</volume>          <year>2011</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35978378">35978378</a> <b>(2)</b> Transcriptome deregulation of peripheral monocytes and whole blood in <italic>GBA</italic>-related Parkinson&#x2019;s disease<ul><li>FASTQ files were processed with the RAPiD pipeline [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] implemented in the <b>NextFlow</b> framework (RAPiD-nf) (&#x201C;<b>Nextflow</b> - A DSL for Parallel and Scalable Computational Pipelines&#x201D; n.d.) providing automated alignment, quantification, and quality control of each RNA-seq sample.<li> [48](Shah, H. PgmNr 1856: RAPiD&#x2014;An Agile and Dependable RNA-Seq Framework (American Society of Human Genetics, 2015).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35972352">35972352</a> <b>(2)</b> Stitching and registering highly multiplexed whole-slide images of tissues and tumors using ASHLAR<ul><li> ASHLAR is available as a Docker or Singularity container and has been incorporated into MCMICRO (<xref rid="btac544-B33" ref-type="bibr">Schapiro <italic toggle="yes">et al.</italic>, 2022a</xref>), the <b>Nextflow</b>-based image processing pipeline developed by HTAN; as part of MCMICRO, ASHLAR has been tested with several hundred CyCIF, CODEX and mxIF images acquired from 12 types of mouse and human tissues at seven different institutions on five different microscopes and slide scanner platforms (<xref rid="sup1" ref-type="supplementary-material"><u>Supplementary Table S1</u></xref>).<li> Merkel, <xref rid="btac544-B29" ref-type="bibr"><u>2014</u></xref>) and implemented it in the workflow systems <b>Nextflow</b> (<xref rid="btac544-B9" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li> [2014](<person-group person-group-type="author"><string-name><surname>Merkel</surname><given-names>D.</given-names></string-name></person-group> (<year>2014</year>) <article-title>Docker: lightweight Linux containers for consistent development and deployment</article-title>. <source>Linux J</source>., <volume>2</volume>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35973991">35973991</a> <b>(1)</b> Single cell spatial analysis reveals the topology of immunomodulatory purinergic signaling in glioblastoma<ul><li>Image analysis was performed with the Docker-based <b>NextFlow</b> pipeline MCMICRO<sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup> and with customized scripts in Python (v3.6), ImageJ (v1.53r), and MATLAB (v2017, 2018a, 2019b).<li> [66](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34824477">34824477</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35979446">35979446</a> <b>(3)</b> A bioinformatic-assisted workflow for genome-wide identification of ncRNAs<ul><li> Pinc is embedded in a <b>Nextflow</b> framework which allows building very intuitive pipelines.<li> In addition, the stand-alone <b>Nextflow</b> pipeline will also be available at GitHub when all necessary programs are already installed on the system.<li>Pinc will be accessible as a <b>Nextflow</b> pipeline on GitHub if the necessary programs are already installed (<ext-link xlink:href="https://github.com/brummetheus/pinc" ext-link-type="uri">https://github.com/brummetheus/pinc</ext-link>) or as a Docker image to be able to run it independently on the user's preferred operating system.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35962405">35962405</a> <b>(1)</b> Improved SARS-CoV-2 sequencing surveillance allows the identification of new variants and signatures in infected patients<ul><li> It was further parallelized and automated to process 100 samples/h using <b>Nextflow</b> [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35948557">35948557</a> <b>(1)</b> Omicron BA.4/BA.5 escape neutralizing immunity elicited by BA.1 infection<ul><li> In addition, we also used the wf_artic (ARTIC SARS-CoV-2 pipeline, v0.3.18) as built using the <b>nextflow</b> workflow framework57.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36017133">36017133</a> <b>(1)</b> Exploring breast and prostate cancer RNA-seq derived radiosensitivity with the Genomic Adjusted Radiation Dose (GARD) model<ul><li> All processing was implemented using <b>Nextflow</b> scripting <xref rid="b0095" ref-type="bibr"><u>[19]</u></xref> curated within a Docker/Singularity container permitting ease of reproducibility <xref rid="b0100" ref-type="bibr"><u>[20]</u></xref>.<li> [[19]](P. DI Tommaso, M. Chatzou, E.W. Floden, P.P. Barja, E. Palumbo, C. Notredame, Nextflow enables reproducible computational workflows, doi:10.1038/nbt.3820, URL: https://doi.org/10.1093/, 2017.) [[20]](G.M. Kurtzer, V. Sochat, M.W. Bauer, Singularity: Scientific containers for mobility of compute, PLoS One 12 (5) (2017) e0177459, ISSN 19326203, doi:10.1371/journal.pone.0177459, URL: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0177459.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35941262">35941262</a> <b>(1)</b> cfDNA methylome profiling for detection and subtyping of small cell lung cancers<ul><li>A <b>nextflow</b><sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup> (v.20.11.0) pipeline was generated to take the FASTQ files to analysis-ready quantitative sequencing enrichment analysis (QSEA) objects and is provided in the supplementary code.<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35992074">35992074</a> <b>(1)</b> A systems biology approach to study non-alcoholic fatty liver (NAFL) in women with obesity<ul><li> The pipeline was built using <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36204205">36204205</a> <b>(1)</b> DNA methylation dynamics during stress response in woodland strawberry (<italic toggle="yes">Fragaria vesca</italic>)<ul><li> The pipeline used <b>Nextflow</b> v20.07.1 to run multitask in parallel.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35918713">35918713</a> <b>(2)</b> Hmga2 protein loss alters nuclear envelope and 3D chromatin structure<ul><li> pipeline (ver.: 1.2.2) run within Dockers [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>] container through <b>Nextflow</b> process scheduler [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>].<li> pipeline (ver.: 1.2.2) run within Docker [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>] container through <b>Nextflow</b> process scheduler.<li> [47](          <person-group person-group-type="author">            <name>              <surname>Merkel</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Docker: lightweight linux containers for consistent development and deployment</article-title>          <source>Linux J</source>          <year>2014</year>          <volume>239</volume>          <fpage>2</fpage>        ) [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [47](          <person-group person-group-type="author">            <name>              <surname>Merkel</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Docker: lightweight linux containers for consistent development and deployment</article-title>          <source>Linux J</source>          <year>2014</year>          <volume>239</volume>          <fpage>2</fpage>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35996577">35996577</a> <b>(1)</b> Characterization of hemocytes and hematopoietic cells of a freshwater crayfish based on single-cell transcriptome analysis<ul><li>All bioinformatic analyses have been put into a <b>Nextflow</b>-based workflow to facilitate reproducibility, which can be found on <ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/SMS-21-5568-cray" id="intref0055">GitHub</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35795910">35795910</a> <b>(1)</b> Using <italic toggle="yes">in vivo</italic> transcriptomics and RNA enrichment to identify genes involved in virulence of <italic toggle="yes">Candida glabrata</italic><ul><li> ANAPURNAseq (Automated <b>Nextflow</b> Alignment Pipeline for Unprocessed RNAseq) performs various read correction steps, aligns corrected reads to one or two reference genomes and finally counts the number of genes that can be associated with a chosen feature (often gene or CDS).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35893066">35893066</a> <b>(1)</b> A Comparison of Bioinformatics Pipelines for Enrichment Illumina Next Generation Sequencing Systems in Detecting SARS-CoV-2 Virus Strains<ul><li> It is recommended in future studies to design a pipeline in an integrated framework, for instance, by using <b>NextFlow</b>, a workflow framework to combine all scripts into one fully integrated pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35899080">35899080</a> <b>(1)</b> iCOMIC: a graphical interface-driven bioinformatics pipeline for analyzing cancer omics data<ul><li><b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35879309">35879309</a> <b>(1)</b> KSTAR: An algorithm to predict patient-specific kinase activities from phosphoproteomic data<ul><li> For cases where datasets were large, such as for the two global phosphoproteomics studies<sup><xref ref-type="bibr" rid="CR35"><u>35</u></xref>,<xref ref-type="bibr" rid="CR37"><u>37</u></xref></sup> or for large serine/threonine networks, the KSTAR activity predictions were performed using a highly parallel version of the original code, implemented using the <b>Nextflow</b> software package<sup><xref ref-type="bibr" rid="CR58"><u>58</u></xref></sup>.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33755436">33755436</a>) [37](          <person-group person-group-type="author">            <name>              <surname>Kubiniok</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Lavoie</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Therrien</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Thibault</surname>              <given-names>P</given-names>            </name>          </person-group>          <article-title>Time-resolved phosphoproteome analysis of paradoxical RAF activation reveals novel targets of ERK</article-title>          <source>Mol. Cell. Proteom.</source>          <year>2017</year>          <volume>16</volume>          <fpage>663</fpage>          <lpage>679</lpage>          <pub-id pub-id-type="doi">10.1074/mcp.M116.065128</pub-id>        ) [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35897706">35897706</a> <b>(1)</b> HPV Integration Site Mapping: A Rapid Method of Viral Integration Site (VIS) Analysis and Visualization Using Automated Workflows in CLC Microbial Genomics<ul><li> Both manual and developer-based, viral-host mapping workflows, i.e., SearcHPV, and nf-VIF (<b>nextflow</b>-based Virus Insertion Finder) have been reported [<xref rid="B22-ijms-23-08132" ref-type="bibr"><u>22</u></xref>,<xref rid="B23-ijms-23-08132" ref-type="bibr"><u>23</u></xref>,<xref rid="B24-ijms-23-08132" ref-type="bibr"><u>24</u></xref>].<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30728408">30728408</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34160069">34160069</a>) [24](          <person-group person-group-type="author"><name><surname>Mainguen&#xE9;</surname><given-names>J.</given-names></name><name><surname>Vacher</surname><given-names>S.</given-names></name><name><surname>Kamal</surname><given-names>M.</given-names></name><name><surname>Hamza</surname><given-names>A.</given-names></name><name><surname>Masliah-Planchon</surname><given-names>J.</given-names></name><name><surname>Baulande</surname><given-names>S.</given-names></name><name><surname>Ibadioune</surname><given-names>S.</given-names></name><name><surname>Borcoman</surname><given-names>E.</given-names></name><name><surname>Cacheux</surname><given-names>W.</given-names></name><name><surname>Calugaru</surname><given-names>V.</given-names></name><etal/></person-group>          <article-title>Human papilloma virus integration sites and genomic signatures in head and neck squamous cell carcinoma</article-title>          <source>Mol. Oncol.</source>          <year>2022</year>          <pub-id pub-id-type="doi">10.1002/1878-0261.13219</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35893239">35893239</a> <b>(2)</b> Systematic Analysis of Long Non-Coding RNA Genes in Nonalcoholic Fatty Liver Disease<ul><li> In addition, the commands and <b>nextflow</b> [<xref rid="B64-ncrna-08-00056" ref-type="bibr"><u>64</u></xref>] pipelines used for the content of LiverDB are available via the GitHub repository to allow for a further analysis of similar RNA-seq data.<li> RNA-Seq data preprocessing pipeline was managed via <b>Nextflow</b> [<xref rid="B64-ncrna-08-00056" ref-type="bibr"><u>64</u></xref>] (Version 21.10.6.5660).<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35850574">35850574</a> <b>(1)</b> Genomic surveillance of Rift Valley fever virus: from sequencing to lineage assignment<ul><li> The web-based method is built on top of a Genome Detective software tool [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] while the command line is implemented in <b>Nextflow</b> language [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], that is both scalable and reproducible.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30124794">30124794</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35855323">35855323</a> <b>(1)</b> High-throughput method for the hybridisation-based targeted enrichment of long genomic fragments for PacBio third-generation sequencing<ul><li> VCF concordance check according to the PrecisionFDA truth challenge was done using a <italic toggle="yes"><b>nextflow</b></italic> (v20.10.0) pipeline involving <italic toggle="yes">bedtools</italic> (v2.30.0), <italic toggle="yes">GATK</italic> (v4.2.4.1)&#xA0;and <italic toggle="yes">hap.py</italic> (v0.3.14).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35831404">35831404</a> <b>(1)</b> Single-cell transcriptional pharmacodynamics of trifluridine in a tumor-immune model<ul><li> All bioinformatic analyses post-CellRanger have been put into a <b>Nextflow</b>-based workflow to facilitate reproducibility, which can be found on GitHub.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35830908">35830908</a> <b>(1)</b> Monkeypox outbreak in Madrid (Spain): Clinical and virological aspects<ul><li> Sequencing samples were analysed for viral genomes reconstruction using the viralrecon pipeline version 2.4.1 (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/viralrecon" id="interref0001">https://github.com/nf-core/viralrecon</ext-link>) (<ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/6320980#.YpB1IZ9BxhE" id="interref0002">https://zenodo.org/record/6320980#.YpB1IZ9BxhE</ext-link>), written in <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/" id="interref0003">https://www.<b>nextflow</b>.io/</ext-link>) in collaboration between the nf-core community (<ext-link ext-link-type="uri" xlink:href="https://www.nature.com/articles/s41587-020-0439-x" id="interref0004">https://www.nature.com/articles/s41587-020-0439-x</ext-link>) and the Bioinformatics Unit of the Institute of Health Carlos III (<ext-link ext-link-type="uri" xlink:href="https://github.com/BU-ISCIII" id="interref0005">https://github.com/BU-ISCIII</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35891789">35891789</a> <b>(1)</b> Proteotranscriptomics &#x2013; A facilitator in omics research<ul><li> <b>Nextflow</b> <xref rid="b0245" ref-type="bibr"><u>[49]</u></xref>, Snakemake <xref rid="b0250" ref-type="bibr"><u>[50]</u></xref> or bpipe <xref rid="b0255" ref-type="bibr"><u>[51]</u></xref> to automate all relevant programming steps in a parallelized preferably portable pipeline.<li> [[49]](          <person-group person-group-type="author">            <name>              <surname>di Tommaso</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>E.W.</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>P.P.</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E.</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nat Biotechnol</source>          <volume>2017</volume>          <issue>354</issue>          <year>2017</year>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [[50]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [[51]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35867760">35867760</a> <b>(4)</b> Sox8 remodels the cranial ectoderm to generate the ear<ul><li>All data alignment and downstream analysis was carried out using NF-core and custom <b>Nextflow</b> pipelines to allow full reproducibility.<li> All code used, including <b>Nextflow</b> pipelines and downstream analysis, can be found at <ext-link xlink:href="https://github.com/alexthiery/otic-reprogramming" ext-link-type="uri">https://github.com/alexthiery/otic-reprogramming</ext-link>.<li>SmartSeq2 scRNAseq reads were aligned and processed using a custom <b>Nextflow</b> DSL2 pipeline.<li>Putative enhancers were identified using a custom <b>Nextflow</b> DSL2 pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35794592">35794592</a> <b>(1)</b> Evidence for close molecular proximity between reverting and undifferentiated cells<ul><li>Fastq files were pre-processed through a bio-informatic pipeline developed in the team on the <b>Nextflow</b> platform [<xref ref-type="bibr" rid="CR54"><u>54</u></xref>].<li> [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35794110">35794110</a> <b>(2)</b> Deep learning from phylogenies to uncover the epidemiological dynamics of outbreaks<ul><li>For FFNN-SS and CNN-CBLV, we reported the average CPU time of encoding a tree (average over 10,000 trees), as reported by <b>NextFlow</b><sup><xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup>.<li>For BEAST2, we reported the CPU time averaged over 100 analyses with BEAST2 as reported by <b>NextFlow</b>.<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36212867">36212867</a> <b>(1)</b> Seasonal Dynamics in Carbon Cycling of Marine Bacterioplankton Are Lifestyle Dependent<ul><li> 1.1.3, a <b>Nextflow</b> (<xref rid="ref23" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) workflow that uses cutadapt (v.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35791022">35791022</a> <b>(6)</b> Minos: variant adjudication and joint genotyping of cohorts of bacterial genomes<ul><li> Additionally, Minos includes a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>] pipeline to joint genotype large numbers (tens of thousands) of samples, producing a set of calls at the same variant sites across all samples.<li>We wrote a <b>Nextflow</b> pipeline, included in the Minos code repository, to easily joint genotype large numbers of samples.<li>The Minos <b>Nextflow</b> pipeline for joint genotyping large sets of samples is conceptually very similar to running on a per-sample basis and proceeds as follows.<li>The <b>nextflow</b> pipeline included in the Minos github repository was used to run joint genotyping on all three data sets.<li> A <b>nextflow</b> configuration file is included that contains preset profiles to set sensible default parameters for &#x201C;medium&#x201D; and &#x201C;large&#x201D; size data sets.<li> Values were taken from the LSF reports for each <b>Nextflow</b> task (&#x201C;Max Memory&#x201D; and &#x201C;Run time&#x201D;).<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35860283">35860283</a> <b>(1)</b> Trained Immunity in Primary Sj&#xF6;gren&#x2019;s Syndrome: Linking Type I Interferons to a Pro-Atherogenic Phenotype<ul><li>Paired-end raw FASTQ files were downloaded from the GEO database using GEO Series accession number GSE173670 (<xref rid="B18" ref-type="bibr"><u>18</u></xref>), and were analyzed with the nf-core/RNA-seq pipeline (v3.1) using <b>Nextflow</b> (21.05.0.edge) and its default settings (<xref rid="B33" ref-type="bibr"><u>33</u></xref>, <xref rid="B34" ref-type="bibr"><u>34</u></xref>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34413853">34413853</a>) [33](<person-group person-group-type="author"><name><surname>Ewels</surname><given-names>PA</given-names></name><name><surname>Peltzer</surname><given-names>A</given-names></name><name><surname>Fillinger</surname><given-names>S</given-names></name><name><surname>Patel</surname><given-names>H</given-names></name><name><surname>Alneberg</surname><given-names>J</given-names></name><name><surname>Wilm</surname><given-names>A</given-names></name><etal/></person-group>. <article-title>The Nf-Core Framework for Community-Curated Bioinformatics Pipelines</article-title>. <source>Nat Biotechnol</source> (<year>2020</year>) <volume>38</volume>:<page-range>276&#x2013;8</page-range>. doi: <pub-id pub-id-type="doi">10.1038/s41587-020-0439-x</pub-id>) [34](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow Enables Reproducible Computational Workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>:<page-range>316&#x2013;9</page-range>. doi: <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35860739">35860739</a> <b>(1)</b> Advances and Trends in Omics Technology Development<ul><li> Domain specific languages such as <b>Nextflow</b> (<xref rid="B259" ref-type="bibr"><u>259</u></xref>) have been established to help partially resolve the reproducibility issue by sharing the codes and workflows used for data analysis.<li> [259](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35889056">35889056</a> <b>(1)</b> Sharing Vitamin B<sub>12</sub> between Bacteria and Microalgae Does Not Systematically Occur: Case Study of the Haptophyte <italic toggle="yes">Tisochrysis lutea</italic><ul><li> SAMBA was developed using the <b>NextFlow</b> workflow manager [<xref rid="B50-microorganisms-10-01337" ref-type="bibr"><u>50</u></xref>] and built around three main parts: data integrity checking, bioinformatics processes, and statistical analyses.<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35760778">35760778</a> <b>(1)</b> Combinatorial immunotherapies overcome MYC-driven immune evasion in triple negative breast cancer<ul><li> RNA sequencing was aligned with STAR and abundance quantified by RSEM as implemented in the <b>nextflow</b> (v 20.12.0) pipeline nf-core/rnaseq (v 3.0)<sup><xref ref-type="bibr" rid="CR69"><u>69</u></xref></sup>.<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35760080">35760080</a> <b>(1)</b> Emergence of SARS-CoV-2 Omicron lineages BA.4 and BA.5 in South Africa<ul><li> We also used the wf_artic (ARTIC SARS-CoV-2) pipeline as built using the <b>Nextflow</b> workflow framework.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35891376">35891376</a> <b>(1)</b> Gut Microbiome Changes Occurring with Norovirus Infection and Recovery in Infants Enrolled in a Longitudinal Birth Cohort in Leon, Nicaragua<ul><li>Metagenomes were assembled using the <b>Nextflow</b> Core MAG pipeline (URL: nf-co.re/mag) [<xref rid="B35-viruses-14-01395" ref-type="bibr"><u>35</u></xref>,<xref rid="B36-viruses-14-01395" ref-type="bibr"><u>36</u></xref>,<xref rid="B37-viruses-14-01395" ref-type="bibr"><u>37</u></xref>].<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35118380">35118380</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35568036">35568036</a> <b>(2)</b> A family of conserved bacterial virulence factors dampens interferon responses by blocking calcium signaling<ul><li> (2021)</xref></td><td>N/A</td></tr><tr><td>Nfcore/rnaseq pipeline</td><td>V 3.5 (<xref rid="bib20" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>)</td><td>N/A (<xref rid="bib20" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>)</td></tr><tr><td><b>Nextflow</b> domain specific language.<li> <xref rid="bib20" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>) written in the <b>Nextflow</b> domain specific language (version 19.10.0; <xref rid="bib18" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2017</u></xref>) was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0; <xref rid="bib32" ref-type="bibr"><u>Kurtzer et&#xA0;al., 2017</u></xref>).<li> [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Di Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Kurtzer et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35888710">35888710</a> <b>(1)</b> A Current Encyclopedia of Bioinformatics Tools, Data Formats and Resources for Mass Spectrometry Lipidomics<ul><li> Using graphical workflow management systems, such as Galaxy [<xref rid="B139-metabolites-12-00584" ref-type="bibr"><u>139</u></xref>] and KNIME [<xref rid="B140-metabolites-12-00584" ref-type="bibr"><u>140</u></xref>], or programmatic/declarative workflow systems, such as Snakemake [<xref rid="B141-metabolites-12-00584" ref-type="bibr"><u>141</u></xref>], <b>Nextflow</b> [<xref rid="B142-metabolites-12-00584" ref-type="bibr"><u>142</u></xref>] or CWL [<xref rid="B143-metabolites-12-00584" ref-type="bibr"><u>143</u></xref>], enable the creation and sharing of reproducible data workflows that have all parameters to individual processing steps defined and documented.<li> [139](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [140](          <person-group person-group-type="author"><name><surname>Berthold</surname><given-names>M.R.</given-names></name><name><surname>Cebron</surname><given-names>N.</given-names></name><name><surname>Dill</surname><given-names>F.</given-names></name><name><surname>Gabriel</surname><given-names>T.R.</given-names></name><name><surname>K&#xF6;tter</surname><given-names>T.</given-names></name><name><surname>Meinl</surname><given-names>T.</given-names></name><name><surname>Ohl</surname><given-names>P.</given-names></name><name><surname>Sieb</surname><given-names>C.</given-names></name><name><surname>Thiel</surname><given-names>K.</given-names></name><name><surname>Wiswedel</surname><given-names>B.</given-names></name></person-group>          <article-title>KNIME: The Konstanz Information Miner</article-title>          <source>Data Analysis, Machine Learning and Applications</source>          <person-group person-group-type="editor"><name><surname>Preisach</surname><given-names>C.</given-names></name><name><surname>Burkhardt</surname><given-names>H.</given-names></name><name><surname>Schmidt-Thieme</surname><given-names>L.</given-names></name><name><surname>Decker</surname><given-names>R.</given-names></name></person-group>          <publisher-name>Springer</publisher-name>          <publisher-loc>Berlin/Heidelberg, Germany</publisher-loc>          <year>2008</year>          <fpage>319</fpage>          <lpage>326</lpage>        ) [141](          <person-group person-group-type="author"><name><surname>M&#xF6;lder</surname><given-names>F.</given-names></name><name><surname>Jablonski</surname><given-names>K.P.</given-names></name><name><surname>Letcher</surname><given-names>B.</given-names></name><name><surname>Hall</surname><given-names>M.B.</given-names></name><name><surname>Tomkins-Tinch</surname><given-names>C.H.</given-names></name><name><surname>Sochat</surname><given-names>V.</given-names></name><name><surname>Forster</surname><given-names>J.</given-names></name><name><surname>Lee</surname><given-names>S.</given-names></name><name><surname>Twardziok</surname><given-names>S.O.</given-names></name><name><surname>Kanitz</surname><given-names>A.</given-names></name><etal/></person-group>          <article-title>Sustainable Data Analysis with Snakemake</article-title>          <source>F1000Research</source>          <year>2021</year>          <volume>10</volume>          <fpage>PMC8114187</fpage>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) [142](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [143](          <person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P.</given-names></name><name><surname>Crusoe</surname><given-names>M.R.</given-names></name><name><surname>Tijani&#x107;</surname><given-names>N.</given-names></name><name><surname>Chapman</surname><given-names>B.</given-names></name><name><surname>Chilton</surname><given-names>J.</given-names></name><name><surname>Heuer</surname><given-names>M.</given-names></name><name><surname>Kartashov</surname><given-names>A.</given-names></name><name><surname>Leehr</surname><given-names>D.</given-names></name><name><surname>M&#xE9;nager</surname><given-names>H.</given-names></name><name><surname>Nedeljkovich</surname><given-names>M.</given-names></name><etal/></person-group>          <source>Common Workflow Language</source>          <comment>version 1.0; Figshare</comment>          <publisher-name>Digital Science</publisher-name>          <publisher-loc>London, UK</publisher-loc>          <year>2016</year>          <pub-id pub-id-type="doi">10.6084/M9.FIGSHARE.3115156.V2</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35814690">35814690</a> <b>(3)</b> Prediction of Antibiotic Susceptibility Profiles of <italic>Vibrio cholerae</italic> Isolates From Whole Genome Illumina and Nanopore Sequencing Data: CholerAegon<ul><li> The pipeline is implemented in <b>Nextflow</b> (Tommaso et al., <xref rid="B42" ref-type="bibr"><u>2017</u></xref>) and Python3, and carries out short read assembly, long read assembly, and/or hybrid assembly, followed by assembly polishing, and finally assessment of resistance gene presence.<li> CholerAegon&#x2014;A Fully Automated <b>Nextflow</b> Pipeline for Antibiotic Resistance Prediction.<li><bold>(A)</bold> The CholerAegon pipeline is implemented in <b>Nextflow</b>, Download at: <ext-link xlink:href="https://github.com/RaverJay/CholerAegon" ext-link-type="uri">https://github.com/RaverJay/CholerAegon</ext-link>.<li> [2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35729113">35729113</a> <b>(9)</b> A scalable, open-source implementation of a large-scale mechanistic model for single cell proliferation and death signaling<ul><li>SPARCED-nf (<b>nextflow</b>, model version running on Nautilus-Kubernetes cluster).<li><b>Nextflow</b> (nf): In this cloud-scalable version of the model, the Jupyter notebooks have been converted into python source code and re-modularized for greater parallel-simulation efficiency.<li> The process of creating and executing the model is handled entirely by <b>Nextflow</b>, a workflow-management application and language for building resilient pipelines.<li> When SPARCED-nf is launched, <b>Nextflow</b> begins by creating a head pod on the cluster to coordinate each of the jobs needed to run the model (Supplementary Fig.&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref>).<li> The <b>nextflow</b>.config configuration file is used to define the specifics of simulation scenarios.<li><b>nextflow</b>.config: This configuration file has two main sections.<li> in SPARCED-jupyter model, this phase of the <b>Nextflow</b> pipeline constructs all necessary files for the model simulation.<li> Having received the relevant model files from the last step, the workflow ingests the <italic>speciesVals</italic>, <italic>ratelawVals</italic>, and <italic>numCells</italic> arguments set by the user in the <b>nextflow</b>.config.<li>SPARCED-nf:model_run: This final step of the <b>Nextflow</b> workflow is responsible for model execution and output generation.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35725628">35725628</a> <b>(3)</b> SeqScreen: accurate and sensitive functional screening of pathogenic sequences via ensemble learning<ul><li>The SeqScreen pipeline was built using <b>Nextflow</b> [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>], a domain-specific language for creating scalable and portable workflows.<li> SeqScreen combines various stages in separate <b>Nextflow</b> modules and is available as an open-source tool on bioconda (<ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/seqscreen">https://anaconda.org/bioconda/seqscreen</ext-link>).<li> The pipeline is implemented using <b>Nextflow</b> for scalable and reproducible deployment and the scripts are written in Perl and Python.<li> [35](di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol. Nature Publishing Group; 2017. p. 316&#x2013;319. 10.1038/nbt.3820.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35723286">35723286</a> <b>(5)</b> Automated Library Construction and Analysis for High-Throughput Nanopore Sequencing of SARS-CoV-2<ul><li> Both Nimbus and Janus methods and related data are available at <ext-link xlink:href="https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/artic_<b>nextflow</b>/browse" ext-link-type="uri">https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/artic_<b>nextflow</b>/browse</ext-link>.<li> The guppyplex and minion commands were combined with Nextclade assignment and quality control (<xref rid="jfac054-B21" ref-type="bibr"><u>21</u></xref>) in a <b>nextflow</b> 0.1.1 (<xref rid="jfac054-B22" ref-type="bibr"><u>22</u></xref>) workflow to allow easy parallelization on our network.<li> Our analysis workflow is available for download at <ext-link xlink:href="https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/artic_<b>nextflow</b>/browse?%20at=refs%2Ftags%2F0.1.1" ext-link-type="uri">https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/artic_<b>nextflow</b>/browse? at=refs%2Ftags%2F0.1.1</ext-link>.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35725577">35725577</a> <b>(4)</b> NETISCE: a network-based tool for cell fate reprogramming<ul><li> NETISCE is a user-friendly tool implemented as a command-line <b>Nextflow</b> pipeline and Galaxy Project workflow that non-experts can use in modeling or computational approaches to analyze the biological systems of their interest.<li> NETISCE is implemented as a <b>Nextflow</b> workflow<sup><xref ref-type="bibr" rid="CR92"><u>92</u></xref></sup>.<li> <b>Nextflow</b> is a state-of-the-art workflow manager tool that is language agnostic and designed for parallel processing as a dataflow manager.<li> We provide <b>Nextflow</b> pipelines for local machine use and high-performance cluster implementations.<li> [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35782725">35782725</a> <b>(1)</b> DESSO-DB: A web database for sequence and shape motif analyses and identification<ul><li> The backend server was organized and deployed using <b>Nextflow</b> to ensure flexibility and scales efficient scaling on the cloud.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35704623">35704623</a> <b>(1)</b> Genome skimming approach reveals the gene arrangements in the chloroplast genomes of the highly endangered <italic toggle="yes">Crocus</italic> L. species: <italic toggle="yes">Crocus istanbulensis</italic> (B.Mathew) Ruk&#x161;&#x101;ns<ul><li>For the accurate detection of site-specific positive selection in the protein-coding sequences of Iridaceae, a <b>Nextflow</b> pipeline, which is a scalable and reproducible scientific workflow designed for positive selection analysis, called &#x201C;PoSeiDon&#x201D; [<xref rid="pone.0269747.ref025" ref-type="bibr"><u>25</u></xref>] was employed using default parameters.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32735310">32735310</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35705638">35705638</a> <b>(1)</b> Datasets for benchmarking antimicrobial resistance genes in bacterial metagenomic and whole genome sequencing<ul><li>To generate a simulated metagenomic benchmarking dataset, a reproducible <b>nextflow</b><sup><xref ref-type="bibr" rid="CR25"><u>25</u></xref></sup> simulation workflow was used.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><u>No PMID</u> <b>(1)</b> Abstracts from the 24th Annual Meeting of the Irish Society of Human Genetics<ul><li> a scalable, automatic pipeline using <b>Nextflow</b> and BioContainers, which offers reproducible and interactive quality control and imputation pipelines for genomic data.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35701420">35701420</a> <b>(16)</b> Implementing the reuse of public DIA proteomics datasets: from the PRIDE database to Expression Atlas<ul><li>The reanalysis pipeline represents a harmonised combination of metadata annotation protocols, open proteomics software tools for SWATH-MS data in computational workflows orchestrated by <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR25"><u>25</u></xref></sup> which can be run in multiple compute environments, and data integration procedures.<li> (<bold>b</bold>) SWATH-MS data analysis: <b>Nextflow</b> workflow including steps ranging from data conversion, SWATH-MS window management, data quality assessment and control (QA/QC), OpenSWATH analysis, FDR calculation to measurement alignment.<li> (<bold>c</bold>) Statistical analysis: <b>Nextflow</b> workflow for MSstats analysis, normalisation and result filtering.<li> The first (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1a</u></xref>), includes dataset curation and acquisition, and the second (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1b</u></xref>) contains the upstream analysis of the SWATH-MS raw data, automated with a <b>Nextflow</b> workflow.<li> The third (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1c</u></xref>), provides downstream statistical analysis, and is also automated with <b>Nextflow</b>, and a fourth part (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1d</u></xref>) contains an internal workflow for the integration of the results into Expression Atlas.<li> From a technical perspective, the additionally implemented entry points to the first <b>Nextflow</b> workflow (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1b</u></xref>) for pre-processed data (e.g., OpenSWATH output files) proved to be advantageous, since analyses that follow the initial SWATH-MS data analysis step often needed to be tested with different parameter settings.<li> The <b>Nextflow</b> configuration of iteratively repeating failed jobs with an increased memory request was also found to be advantageous.<li> It should be noted in this context that additional open analysis pipelines for DIA datasets have been recently developed using <b>NextFlow</b><sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref></sup> and Galaxy<sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup> as the workflow management systems.<li>The data analysis workflows were constructed using <b>Nextflow</b>.<li> The <b>Nextflow</b> workflows were split after the multi-run alignment step and equipped with optional entry points for pre-processed data from OpenSWATH<sup><xref ref-type="bibr" rid="CR12"><u>12</u></xref></sup> (<italic>.osw</italic> files) to enable more flexible compute settings.<li>The inputs (Table&#xA0;<xref rid="Tab4" ref-type="table"><u>4</u></xref>) for the <b>Nextflow</b> pipeline consisted of the SWATH-MS runs as a collection of<italic>&#xA0;.wiff/.scan</italic> files, together with a descriptor file in<italic>&#xA0;.TraML</italic> format, detailing the iRT peptides used.<li> If a target library in&#xA0;<italic>.pqp</italic> format had already been created or the analysis of processed&#xA0;<italic>.osw</italic> files had already been conducted with different downstream parameters (FDR thresholds), these could be used as alternative entry points in the <b>Nextflow</b> workflow.<li> For the last <b>Nextflow</b> workflow, the study design in an MSstats conformant format must also be provided.<li>Types of input to the DIA reanalysis pipeline implemented in <b>Nextflow</b>.<li>Types of input to the DIA reanalysis pipeline implemented in <b>Nextflow</b>.</p></caption><table frame="hsides" rules="groups"><tbody><tr><td><bold>Analysis pipeline inputs</bold></td></tr><tr><td>An iRT file in OpenSWATH conformant<italic>&#xA0;.TraML</italic> format</td></tr><tr><td>A collection of&#xA0;<italic>.wiff/.scan</italic> files (alternatively pre-processed&#xA0;<italic>.osw</italic> files)</td></tr><tr><td>Target library file in OpenSWATH conformant<italic>.tsv</italic> format (alternatively a prepared&#xA0;<italic>.pqp</italic> file)</td></tr><tr><td>Study design annotation in MSstats conformant format and SDRF format (<italic>.txt/.sdrf</italic>)</td></tr><tr><td>Parameter<italic>&#xA0;.yaml</italic> file (in case the defaults need to be changed)</td></tr><tr><td><bold>Output upstream <b>Nextflow</b></bold>.<li>Types of input to the DIA reanalysis pipeline implemented in <b>Nextflow</b>.</p></caption><table frame="hsides" rules="groups"><tbody><tr><td><bold>Analysis pipeline inputs</bold></td></tr><tr><td>An iRT file in OpenSWATH conformant<italic>&#xA0;.TraML</italic> format</td></tr><tr><td>A collection of&#xA0;<italic>.wiff/.scan</italic> files (alternatively pre-processed&#xA0;<italic>.osw</italic> files)</td></tr><tr><td>Target library file in OpenSWATH conformant<italic>.tsv</italic> format (alternatively a prepared&#xA0;<italic>.pqp</italic> file)</td></tr><tr><td>Study design annotation in MSstats conformant format and SDRF format (<italic>.txt/.sdrf</italic>)</td></tr><tr><td>Parameter<italic>&#xA0;.yaml</italic> file (in case the defaults need to be changed)</td></tr><tr><td><bold>Output upstream <b>Nextflow</b></bold></td></tr><tr><td>FDR filtered alignment results of transition quantifications (<italic>.tsv</italic>)</td></tr><tr><td>QC records for each MS run (<italic>.json</italic>)</td></tr><tr><td><bold>Output downstream <b>Nextflow</b></bold>.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34153189">34153189</a>) [33](Fahrner, M. <italic>et al</italic>. Democratizing data-independent acquisition proteomics analysis on public cloud infrastructures via the galaxy framework. <italic>GigaScience</italic><bold>11</bold>, 10.1093/gigascience/giac005 (2022).) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24727770">24727770</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35695379">35695379</a> <b>(4)</b> Multiplexed Assembly and Annotation of Synthetic BiologyConstructs Using Long-Read Nanopore Sequencing<ul><li> These assemblies can then be annotated withepigenetic base modifications, and our long-read data can be usedto estimate input contamination levels, providing a comprehensivesequence characterization for a wide variety of downstream applications.We have packed the full computational process into a publicly available<b>Nextflow</b> pipeline, which is available at <uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mckennalab/Circuitseq">https://github.com/mckennalab/Circuitseq</uri>.<li> The generated data are thenrun through a custom <b>NextFlow</b> pipeline, producing the final assembledsequences.<li> We then demultiplexed and processed individual plasmids with acustom <b>NextFlow</b> pipeline.<li>We establisheda <b>NextFlow</b> pipeline using a prepackaged Docker container to facilitateadoption and reproducibility.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35695839">35695839</a> <b>(1)</b> RNA-binding proteins direct myogenic cell fate decisions<ul><li>All RNA-sequencing data preprocessing was carried using standardized <b>Nextflow</b> RNA-sequencing (nf-core/rnaseq) with STAR alignment (genome GRCm39), transcriptome mapping with Salmon, and in silico ribosomal depletion using SortMeRNA.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35649353">35649353</a> <b>(1)</b> JAZF1-SUZ12 dysregulates PRC2 function and gene expression during cell differentiation<ul><li>RNA-seq data were processed using the <b>Nextflow</b> core RNA-seq pipeline with default settings (<xref rid="bib23" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>).<li> [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35634739">35634739</a> <b>(1)</b> A large outbreak of COVID-19 in a UK prison, October 2020 to April 2021<ul><li> Samples were processed using the ncov2019 ARTIC <b>Nextflow</b> pipeline [<xref rid="ref12" ref-type="bibr"><u>12</u></xref>] and were analysed using the CLIMB COVID analysis platform [<xref rid="ref13" ref-type="bibr"><u>13</u></xref>] making use of the civet [<xref rid="ref14" ref-type="bibr"><u>14</u></xref>] and MircoReact [<xref rid="ref15" ref-type="bibr"><u>15</u></xref>] tools for analysis and visualisation of the outbreak.<li> [12](<string-name><surname>Anson</surname><given-names>M</given-names></string-name><etal/> (<year>2021</year>) <uri xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</uri> (Accessed <date-in-citation content-type="access-date">13 September 2021</date-in-citation>.) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33397451">33397451</a>) [14](<string-name><surname>O'Toole</surname><given-names>AN</given-names></string-name><etal/> (<year>2021</year>) <article-title>Genomics-informed outbreak investigations of SARS-CoV-2 using civet</article-title>. <source>MedRXiv - Preprint</source>. Published online 14 December 2021. <uri xlink:href="https://www.medrxiv.org/content/10.1101/2021.12.13.21267267v1">https://www.medrxiv.org/content/10.1101/2021.12.13.21267267v1</uri>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28348833">28348833</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35664802">35664802</a> <b>(2)</b> ePeak: from replicated chromatin profiling data to epigenomic dynamics<ul><li> The nf-core (<b>nextflow</b>) pipeline (<xref rid="B34" ref-type="bibr"><u>34</u></xref>) comprises both quality control metrics calculation and differential analysis using the R package DESeq2 and can be applied on narrow and broad peaks.<li>Regarding CUT&amp;RUN and CUT&amp;Tag, the nf-core (<b>nextflow</b>) pipeline (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) and the CUT&amp;RUNTools 2.0 (<xref rid="B9" ref-type="bibr"><u>9</u></xref>) are currently the most complete protocols.<li> [34](<person-group person-group-type="author"><string-name><surname>Patel</surname><given-names>H.</given-names></string-name>, <string-name><surname>Wang</surname><given-names>C.</given-names></string-name>, <string-name><surname>Ewels</surname><given-names>P.</given-names></string-name>, <string-name><surname>Peltzer</surname><given-names>A.</given-names></string-name>, <string-name><surname>Silva</surname><given-names>T.C.</given-names></string-name>, <string-name><surname>Behrens</surname><given-names>D.</given-names></string-name>, <string-name><surname>Garcia</surname><given-names>M.</given-names></string-name>, <string-name><surname>Mashehu</surname><given-names>R.</given-names></string-name>, <string-name><surname>Haglund</surname><given-names>S.</given-names></string-name>, <string-name><surname>Kretzschmar</surname><given-names>W.</given-names></string-name></person-group><article-title>nf-core/chipseq: nf-core/chipseq v1.2.1 - platinum mole</article-title>. <year>2020</year>; <uri xlink:href="https://nf-co.re/chipseq/1.2.1/usage">https://nf-co.re/chipseq/1.2.1/usage</uri>.) [8](<person-group person-group-type="author"><string-name><surname>Cheshire</surname><given-names>C.</given-names></string-name>, <string-name><surname>Charlotte-West&#xA0;Bot</surname><given-names>N.-C.</given-names></string-name>, <string-name><surname>Ladd</surname><given-names>D.</given-names></string-name>, <string-name><surname>Fields</surname><given-names>C.</given-names></string-name>, <string-name><surname>Deu-Pons</surname><given-names>J.</given-names></string-name>, <string-name><surname>Ewels</surname><given-names>P.</given-names></string-name>, <string-name><surname>Menden</surname><given-names>K.</given-names></string-name></person-group><article-title>nf-core/cutandrun: nf-core/cutandrun v1.1 Brass Viper</article-title>. <year>2022</year>; <publisher-name>Zenodo</publisher-name><uri xlink:href="https://zenodo.org/record/5882854#.YoY7hlRByUk">https://zenodo.org/record/5882854#.YoY7hlRByUk</uri>.) [9](<person-group person-group-type="author"><string-name><surname>Yu</surname><given-names>F.</given-names></string-name>, <string-name><surname>Sankaran</surname><given-names>V.G.</given-names></string-name>, <string-name><surname>Yuan</surname><given-names>G.-C.</given-names></string-name></person-group><article-title>CUT&amp;RUNTools 2.0: a pipeline for single-cell and bulk-level CUT&amp;RUN and CUT&amp;Tag data analysis</article-title>. <source>Bioinformatics</source>. <year>2021</year>; <volume>38</volume>:<fpage>252</fpage>&#x2013;<lpage>254</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35610562">35610562</a> <b>(2)</b> Characterising genome architectures using genome decomposition analysis<ul><li> The pipeline consists of a master script that is written in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR32"><u>32</u></xref>].<li> The <b>Nextflow</b> script triggers multiple third party software tools that are used to detect genomic features.<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35799708">35799708</a> <b>(1)</b> Integration and Reanalysis of Four RNA-Seq Datasets Including BALF, Nasopharyngeal Swabs, Lung Biopsy, and Mouse Models Reveals Common Immune Features of COVID-19<ul><li>For the Zhou and Israelow data, we used the <b>Nextflow</b> nf-core RNA-sequencing pipeline (<xref rid="B11" ref-type="bibr"><u>11</u></xref>) to perform adapter trimming using Cutadapt (Martin et al., 2011), alignment of reads to the genome using STAR (<xref rid="B12" ref-type="bibr"><u>12</u></xref>) and quantification of read counts per gene using featureCounts (<xref rid="B13" ref-type="bibr"><u>13</u></xref>).<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35582905">35582905</a> <b>(1)</b> Partial ORF1ab Gene Target Failure with Omicron BA.2.12.1<ul><li> Variant calling and consensus genome generation were performed using the Oxford <b>Nextflow</b> ARTIC pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35585235">35585235</a> <b>(1)</b> MCM complexes are barriers that restrict cohesin-mediated loop extrusion<ul><li>Hi-C and Micro-C data processing was performed using distiller&#x2014;a <b>nextflow</b>-based pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/open2c/distiller-nf">https://github.com/open2c/distiller-nf</ext-link>)<sup><xref ref-type="bibr" rid="CR55"><u>55</u></xref></sup>.<li> [55](Goloborodko, A. et al. mirnylab/distiller-nf: v0.3.3 10.5281/zenodo.3350937 (Zenodo, 2019).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35579325">35579325</a> <b>(2)</b> Multi-tract multi-symptom relationships in pediatric concussion<ul><li> Tractoflow is a novel diffusion MRI processing pipeline, incorporating state-of-the-art functions from FSL, Dipy, and MRtrix into <b>NextFlow</b>.<li> <xref rid="bib50" ref-type="bibr"><u>Sherbrooke Connectivity Imaging Lab, 2022</u></xref>) (wrappers of <italic toggle="yes">Dipy</italic>) implemented in <b>Nextflow</b> to split parcellations into individual labels, apply them to tractograms to create individual bundles, and then extract diffusion measures across them.<li> [Sherbrooke Connectivity Imaging Lab, 2022](          <person-group person-group-type="author"><collab>Sherbrooke Connectivity Imaging Lab</collab></person-group>          <year iso-8601-date="2022">2022</year>          <data-title>Scilpy</data-title>          <version designator="1.0.0">1.0.0</version>          <source>GitHub</source>          <ext-link xlink:href="https://github.com/scilus/scilpy" ext-link-type="uri">https://github.com/scilus/scilpy</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35591887">35591887</a> <b>(1)</b> Accuracy of multiple sequence alignment methods in the reconstruction of transposable element families<ul><li>TEForwardEvolve, genRandomTETrees, analysis scripts, <b>Nextflow</b> (<xref rid="B80" ref-type="bibr"><u>80</u></xref>) job management scripts, along with simulated data generated as part of this study, are available on github at: <ext-link xlink:href="https://github.com/Dfam-consortium/TEForwardEvolve" ext-link-type="uri">https://github.com/Dfam-consortium/TEForwardEvolve</ext-link>, release tag &#x2018;1.0-paper&#x2019;.<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35576466">35576466</a> <b>(1)</b> A male pheromone that improves the quality of the oogenic germline<ul><li>We used a publicly available <b>Nextflow</b> (<xref rid="r75" ref-type="bibr"><u>75</u></xref>) pipeline (<ext-link xlink:href="https://github.com/nf-core/rnaseq" ext-link-type="uri">https://github.com/nf-core/rnaseq</ext-link>) to carry out quality control (FASTQC; ref.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35605437">35605437</a> <b>(1)</b> Advancing genomic epidemiology by addressing the bioinformatics bottleneck: Challenges, design principles, and a Swiss example<ul><li> Another complementary approach to achieving analysis modularity would be to use scientific workflow systems, such as Snakemake&#xA0;(<xref rid="b17" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) or <b>Nextflow</b>&#xA0;(<xref rid="b8" ref-type="bibr"><u>Di&#xA0;Tommaso et al., 2017</u></xref>).<li> [M&#xF6;lder et al., 2021](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F.</given-names>            </name>            <etal/>          </person-group>          <article-title>Sustainable data analysis with snakemake [version 2; peer review: 2 approved]</article-title>          <source>F1000Research</source>          <volume>10</volume>          <issue>33</issue>          <year>2021</year>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) [Di&#xA0;Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35549218">35549218</a> <b>(1)</b> A ComprehensiveEvaluation of Consensus Spectrum GenerationMethods in Proteomics<ul><li>The datamanipulation steps were implemented as reproducible<b>Nextflow</b> workflows (<xref rid="fig1" ref-type="fig">Figure <xref rid="fig1" ref-type="fig"><u>1</u></xref></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35550614">35550614</a> <b>(4)</b> Sox2 levels regulate the chromatin occupancy of WNT mediators in epiblast progenitors responsible for vertebrate body formation<ul><li> 10.5281/zenodo.3529400)<sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup> written in the <b>Nextflow</b> domain specific language (version 19.10.0)<sup><xref ref-type="bibr" rid="CR76"><u>76</u></xref></sup> was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0)<sup><xref ref-type="bibr" rid="CR77"><u>77</u></xref></sup>.<li> <b>nextflow</b> run nf-core/ChIP-seq &#x2013;input design.csv &#x2013;genome mm10 &#x2013;gtf refseq_genes.gtf &#x2013;single_end &#x2013;narrow_peak &#x2013;min_reps_consensus 2 -profile crick -r 1.1.0&#x2019;.<li> 10.5281/zenodo.2634133)<sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup> written in the <b>Nextflow</b> domain specific language (version 19.10.0)<sup><xref ref-type="bibr" rid="CR76"><u>76</u></xref></sup> was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0)<sup><xref ref-type="bibr" rid="CR77"><u>77</u></xref></sup>.<li> <b>nextflow</b> run nf-core/ATAC-seq &#x2013;design design.csv &#x2013;genome mm10 &#x2013;gtf refseq_genes.gtf -profile crick -r 1.0.0&#x2019;.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304282">36304282</a> <b>(1)</b> MIntO: A Modular and Scalable Pipeline For Microbiome Metagenomic and Metatranscriptomic Data Integration<ul><li><b>Nextflow</b> and Conda or Docker/Singularity.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35536194">35536194</a> <b>(1)</b> Evaluating the power and limitations of genome-wide association studies in <italic toggle="yes">Caenorhabditis elegans</italic><ul><li>The NemaScan pipeline is written in <b>Nextflow</b> (<xref rid="jkac114-B19" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35536253">35536253</a> <b>(3)</b> BioUML&#x2014;towards a universal research platform<ul><li><b>Nextflow</b> (<ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>, (<xref rid="B7" ref-type="bibr"><u>7</u></xref>))&#x2014;is a domain specific language and execution environment for scalable and reproducible scientific workflows using software containers.<li> The BioUML platform may use <b>Nextflow</b> to launch programs for data analyses on clouds and clusters.<li> In comparison to common formats (CWL, <b>Nextflow</b>), it provides new workflow elements: logical gateways, events, compensatory action in case of any error.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35523247">35523247</a> <b>(1)</b> Omicron infection enhances Delta antibody immunity in vaccinated persons<ul><li> In addition, we also used the wf_artic (ARTIC SARS-CoV-2) pipeline as built using the <b>nextflow</b> workflow framework.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35573477">35573477</a> <b>(2)</b> Whole-genome sequencing and mutational analysis of human cord-blood derived stem and progenitor cells<ul><li><hr/></td></tr><tr><td><b>Nextflow</b> Illumina Analysis Pipeline (NF-IAP).<li>$ <b>nextflow</b> run /path/to/scripts/NF-IAP/nf-iap.nf \.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35501696">35501696</a> <b>(26)</b> GEMmaker: process massive RNA-seq datasets on heterogeneous computational infrastructure<ul><li> Popular examples include Galaxy [<xref ref-type="bibr" rid="CR1"><u>1</u></xref>], Kepler [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>], <b>Nextflow</b> [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] and Snakemake [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>].<li> For example, <b>Nextflow</b> can manage execution of workflows on desktop computers or HPC systems such as Grid Engine [<xref ref-type="bibr" rid="CR12"><u>12</u></xref>], Portable Batch System (PBS) [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>], HTCondor [<xref ref-type="bibr" rid="CR33"><u>33</u></xref>], SLURM [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>], Kubernetes [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>], popular commercial cloud platforms, and others.<li> <b>Nextflow</b> also uses containers, such as Docker [<xref ref-type="bibr" rid="CR22"><u>22</u></xref>] and Singularity [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] to encapsulate dependent software for the workflow, eliminating the need for installation of software and managing interdependencies.<li>To assist bioinformaticians in the development of portable, standards-based reproducible workflows, the nf-core framework [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] was developed which provides workflow construction standards, peer-review and best-practice recommendations for workflows constructed using <b>Nextflow</b>.<li>The inability to scale without overrunning user data storage is a limitation of <b>Nextflow</b> rather than the nf-core/rnaseq workflow, which could overrun user storage&#x2014;especially for large datasets.<li> First, <b>Nextflow</b> does not currently support cleanup of intermediate files.<li> Second, <b>Nextflow</b> tends to execute all instances of the same step (e.g., downloading of SRAs from NCBI) before moving to the next step (e.g., quantification with kallisto) compounding the challenge of cleanup of intermediate files since cleanup cannot occur until later steps are completed.<li>Until the time that <b>Nextflow</b> supports a file cleanup strategy, a solution is needed to support RNA-seq workflows that need to scale without overrunning storage.<li> Ideally, the solution would be to contribute code to the nf-core/rnaseq workflow to support file cleanup, but the nf-core standards require that workflows only support native <b>Nextflow</b> functionality.<li> It provides much of the functionality of the nf-core/rnaseq workflow as well as the portability and reproducibility benefits inherit with <b>Nextflow</b> and nf-core workflows.<li>GEMmaker uses <b>Nextflow</b> and is a combination of Groovy scripts for interfacing with <b>Nextflow</b>, Python scripts for wrangling intermediate data, and Bash scripts for execution of each software tool in the workflow.<li> <b>Nextflow</b> was selected as the framework because it is widely used, is well supported, has a robust community of workflow creators in the life sciences, supports multiple computing platforms and supports containerization systems such as Docker and Singularity.<li> <b>Nextflow</b> allows for execution of workflows from a command-line interface, which is common with most HPC platforms.<li> <b>Nextflow</b> is then instructed to automatically begin processing any samples that appear in the &#x201C;processing&#x201D; folder.<li> As usual, <b>Nextflow</b> will process samples in parallel, using all CPUs, by first executing the first step for all samples, then the second for all samples, and so forth.<li> However, because GEMmaker limits the number of samples to the number of CPUs, when a sample completes a step, it will move to the next step because <b>Nextflow</b> does not see any samples waiting.<li> <b>Nextflow</b> sees this new sample in the &#x201C;processing&#x201D; folder and immediately begins processing that sample through each step.<li> There is no lag between the time one sample finishes, and another begins and <b>Nextflow</b> should keep all CPUs consistently busy processing samples in parallel.<li> GEMmaker retrieves this Docker image from Docker Hub the first time it is run&#x2014;users need not install any software other than <b>Nextflow</b> and a containerization software (Singularity or Docker).<li> Because GEMmaker is a <b>Nextflow</b> workflow, it can resume execution where it left off.<li> However, <b>Nextflow</b> creates new working directories for each step of the workflow for each sample and when it is resumed it creates new working folders&#x2014;the folders with failed steps remain.<li> When a sample completes a step, then GEMmaker can clean up the working directories that were successful but there is not a mechanism in <b>Nextflow</b> to know about the directories with failed results so that they can be cleaned.<li> As a result, if a high <italic>&#x2013;max_cpus</italic> is used (e.g., 120) and <b>Nextflow</b> is preempted this may result in higher storage usage from directories with failed jobs.<li>Programming language: <b>Nextflow</b> Groovy, Python and bash.<li>Other requirements: <b>Nextflow</b> and Java.<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [21](          <person-group person-group-type="author">            <name>              <surname>Lud&#xE4;scher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Altintas</surname>              <given-names>I</given-names>            </name>            <name>              <surname>Berkley</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Higgins</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Jaeger</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Jones</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Tao</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Zhao</surname>              <given-names>Y</given-names>            </name>          </person-group>          <article-title>Scientific workflow management and the Kepler system: research articles</article-title>          <source>Concurr Comput Pract Exp</source>          <year>2006</year>          <volume>18</volume>          <fpage>1039</fpage>          <lpage>1065</lpage>          <pub-id pub-id-type="doi">10.1002/cpe.v18:10</pub-id>        ) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [12](Gentzsch W. Sun grid engine: Towards creating a compute power grid. In: Proceedings of 1st IEEE/ACM International Symposium on Cluster Computing Grid, CCGrid 2001 35&#x2013;36. 10.1109/CCGRID.2001.923173) [11](Feng H, Misra V, Rubenstein D, Feng H, Misra V, Rubenstein D. PBS: a unified priority-based scheduler. In: Proceedings of the 2007 ACM SIGMETRICS international conference on measurement and modeling of computer systems&#x2014;SIGMETRICS &#x2019;07. ACM Press, New York, New York, USA, p. 203. 10.1145/1254882.1254906) [33](Thain D, Tannenbaum T, Livny M. Distributed computing in practice: the Condor experience: research articles. Concurr Comput Pract Exp 2005;17:323&#x2013;356. 10.1002/cpe.938) [13](          <person-group person-group-type="author">            <name>              <surname>Jette</surname>              <given-names>MA</given-names>            </name>            <name>              <surname>Yoo</surname>              <given-names>AB</given-names>            </name>            <name>              <surname>Grondona</surname>              <given-names>M</given-names>            </name>          </person-group>          <article-title>SLURM: simple linux utility for resource management. job sched</article-title>          <source>Strateg Parallel Process Lect Notes Comput Sci</source>          <year>2003</year>          <volume>2862</volume>          <fpage>44</fpage>          <lpage>60</lpage>          <pub-id pub-id-type="doi">10.1007/10968987_3</pub-id>        ) [35](VMware. DEMYSTIFYING KUBERNETES Overcoming Misconceptions About Container Orchestration 2017.) [22](          <person-group person-group-type="author">            <name>              <surname>Merkel</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Docker: lightweight Linux containers for consistent development and deployment</article-title>          <source>Linux J</source>          <year>2014</year>          <volume>2014</volume>          <fpage>2</fpage>          <pub-id pub-id-type="doi">10.1097/01.NND.0000320699.47006.a3</pub-id>        ) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35585986">35585986</a> <b>(2)</b> Serial Analysis of the T-Cell Receptor &#x3B2;-Chain Repertoire in People Living With HIV Reveals Incomplete Recovery After Long-Term Antiretroviral Therapy<ul><li> A reproducible workflow of all the steps in data QC, data transformation, and analysis was developed using the <b>Nextflow</b> workflow management system (<xref rid="B16" ref-type="bibr"><u>16</u></xref>).<li> All R scripts and <b>Nextflow</b> workflows to analyze and visualize the data used in the study is available on GitHub (<uri xlink:href="https://github.com/shashidhar22/airr_seq_pipelines">https://github.com/shashidhar22/airr_seq_pipelines</uri>).<li> [16](<person-group person-group-type="author"><name><surname>Tommaso</surname><given-names>PD</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow Enables Reproducible Computational Workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>:<page-range>316&#x2013;9</page-range>. doi:&#xA0;<pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35489067">35489067</a> <b>(1)</b> BusyBee Web: towards comprehensive and differential composition-based metagenomic binning<ul><li> In combination with workflow managing tools such as <b>Nextflow</b> or Snakemake, the API increases experiment throughput and reproducibility of results (<xref rid="B39" ref-type="bibr"><u>39</u></xref>,<xref rid="B40" ref-type="bibr"><u>40</u></xref>).<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35571902">35571902</a> <b>(1)</b> Probiotic Mixture Containing <italic>Lactobacillus helveticus</italic>, <italic>Bifidobacterium longum</italic> and <italic>Lactiplantibacillus plantarum</italic> Affects Brain Responses Toward an Emotional Task in Healthy Subjects: A Randomized Clinical Trial<ul><li>The CTMR sequence processing pipeline is based on DADA2 version 1.8 (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) and implemented in <b>Nextflow</b> (<xref rid="B28" ref-type="bibr"><u>28</u></xref>).<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27214047">27214047</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35574438">35574438</a> <b>(1)</b> Mechanisms of Isothiocyanate Detoxification in Larvae of Two Belowground Herbivores, <italic>Delia radicum</italic> and <italic>D. floralis</italic> (Diptera: Anthomyiidae)<ul><li>Gene expression was assessed for both species using the nf-core/rnaseq pipeline in version 3.0 (<xref rid="B17" ref-type="bibr"><u>Ewels et al., 2020</u></xref>), which is a standard <b>nextflow</b> pipeline for RNA-seq analysis, relying on STAR v2.6.1d (<xref rid="B16" ref-type="bibr"><u>Dobin et al., 2012</u></xref>) for the alignment and Salmon 1.4.0 (<xref rid="B47" ref-type="bibr"><u>Patro et al., 2017</u></xref>) for the quantification.<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Dobin et al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [Patro et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28263959">28263959</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35484572">35484572</a> <b>(1)</b> SvAnna: efficient and accurate pathogenicity prediction of coding and regulatory structural variants in long-read genome sequencing<ul><li> The benchmark was performed using the pipeline engine <b>Nextflow</b> [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>].<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35559261">35559261</a> <b>(2)</b> Drugsniffer: An Open Source Workflow for Virtually Screening Billions of Molecules for Binding Affinity to Protein Targets<ul><li><italic>Drugsniffer</italic> is implemented as a <b>Nextflow</b> workflow (<xref rid="B18" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) that orchestrates the activity of a curated set of open source tools, and supports analysis in cluster (SLURM) and cloud (AWS) environments.<li> Docker container files, <b>NextFlow</b> scripts, and tool code are all available <italic>via</italic> GitHub (<ext-link xlink:href="https://github.com/TravisWheelerLab/drug-sniffer" ext-link-type="uri">https://github.com/TravisWheelerLab/drug-sniffer</ext-link>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35440634">35440634</a> <b>(1)</b> Loss of vascular endothelial notch signaling promotes spontaneous formation of tertiary lymphoid structures<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35429300">35429300</a> <b>(1)</b> Validating the knowledge bank approach for personalized prediction of survival in acute myeloid leukemia: a reproducibility study<ul><li> For future authors, workflow managers such as <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</ext-link>) and Snakemake (<ext-link ext-link-type="uri" xlink:href="https://snakemake.readthedocs.io/">https://snakemake.readthedocs.io/</ext-link>) could present a solution, as they provide an abstraction to separate analysis pipelines from computing environments, and hence allow seamless execution of the pipelines across different platforms (Di Tommaso et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35418104">35418104</a> <b>(3)</b> Analysis of human brain tissue derived from DBS surgery<ul><li> The pipeline is implemented in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>]&#x2060;.<li> <b>Nextflow</b> allows the portability and scalability of the pipeline and supports major cloud computing and batch processing technologies.<li> The Conda environment can be created manually, or created by <b>Nextflow</b>.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35414112">35414112</a> <b>(1)</b> Galaxy workflows for fragment-based virtual screening: a case study on the SARS-CoV-2 main protease<ul><li> As a result, workflow management systems such as KNIME [<xref ref-type="bibr" rid="CR1"><u>1</u></xref>], CWL [<xref ref-type="bibr" rid="CR2"><u>2</u></xref>], <b>Nextflow</b> [<xref ref-type="bibr" rid="CR3"><u>3</u></xref>] or Galaxy [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>] prove useful to organize analyses, allowing automation and parallelization of commonly used steps and avoiding tedious manual repetition.<li> [1](          <person-group person-group-type="author">            <name>              <surname>Berthold</surname>              <given-names>MR</given-names>            </name>            <name>              <surname>Cebron</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Dill</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Gabriel</surname>              <given-names>TR</given-names>            </name>            <name>              <surname>K&#xF6;tter</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Meinl</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Ohl</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Thiel</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Wiswedel</surname>              <given-names>B</given-names>            </name>          </person-group>          <article-title>KNIME&#x2014;the Konstanz Information Miner: version 2.0 and beyond</article-title>          <source>AcM SIGKDD Explor Newsl</source>          <year>2009</year>          <volume>11</volume>          <issue>1</issue>          <fpage>26</fpage>          <lpage>31</lpage>          <pub-id pub-id-type="doi">10.1145/1656274.1656280</pub-id>        ) [2](Crusoe MR, Abeln S, Iosup A, Amstutz P, Chilton J, Tijanic N, M&#xE9;nager H, Soiland-Reyes S, Goble CA (2021) Methods included: Standardizing computational reuse and portability with the Common Workflow Language. CoRR abs/2105.07028. <ext-link ext-link-type="uri" xlink:href="http://arxiv.org/abs/2105.07028">arXiv:2105.07028</ext-link>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [4](          <person-group person-group-type="author">            <name>              <surname>Afgan</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Baker</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Batut</surname>              <given-names>B</given-names>            </name>            <name>              <surname>van den Beek</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Bouvier</surname>              <given-names>D</given-names>            </name>            <name>              <surname>&#x10C;ech</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Chilton</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Clements</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Coraor</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Gr&#xFC;ning</surname>              <given-names>BA</given-names>            </name>            <name>              <surname>Guerler</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Hillman-Jackson</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Hiltemann</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Jalili</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Rasche</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Soranzo</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Goecks</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Taylor</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Nekrutenko</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Blankenberg</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update</article-title>          <source>Nucleic Acids Res</source>          <year>2018</year>          <volume>46</volume>          <issue>W1</issue>          <fpage>537</fpage>          <lpage>544</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gky379</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35404959">35404959</a> <b>(1)</b> Detailed analysis of antibody responses to SARS-CoV-2 vaccination and infection in macaques<ul><li> The phip-flow tools perform read alignment using Bowtie2 [<xref rid="ppat.1010155.ref093" ref-type="bibr"><u>93</u></xref>] in a <b>Nextflow</b> [<xref rid="ppat.1010155.ref094" ref-type="bibr"><u>94</u></xref>] pipeline script.<li> [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [94](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35393570">35393570</a> <b>(1)</b> TMPRSS2-ERG promotes the initiation of prostate cancer by suppressing oncogene-induced senescence<ul><li>Reads were aligned using the <b>nextflow</b> core RNA seq pipeline version 1.4.2 as described in <sup><xref rid="R45" ref-type="bibr"><u>45</u></xref></sup>.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35383166">35383166</a> <b>(2)</b> Cell-intrinsic Aryl Hydrocarbon Receptor signalling is required for the resolution of injury-induced colonic stem cells<ul><li>The nf-core/atacseq pipeline (version 1.2.1)<sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup> written in the <b>Nextflow</b> domain specific language (version 19.10.0)<sup><xref ref-type="bibr" rid="CR76"><u>76</u></xref></sup> was used to perform the primary analysis of the fastq samples in conjunction with Singularity (version 2.6.0)<sup><xref ref-type="bibr" rid="CR77"><u>77</u></xref></sup>.<li> The command used was &#x201C;<b>nextflow</b> run nf-core/atacseq -profile crick --input /Path_to_desing/design.csv --genome GRCm38&#x201D;.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35369709">35369709</a> <b>(1)</b> SARS-CoV-2 Testing in the Community: Testing Positive Samples with the TaqMan SARS-CoV-2 Mutation Panel To Find Variants in Real Time<ul><li> The genotyping was then performed using a <b>nextflow</b> pipeline (<ext-link xlink:href="https://github.com/BioWilko/genotyping-pipeline" ext-link-type="uri">https://github.com/BioWilko/genotyping-pipeline</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35377884">35377884</a> <b>(1)</b> Comparison of different sequencing techniques for identification of SARS-CoV-2 variants of concern with multiplex real-time PCR<ul><li> Resultant fastq files were analyzed using <b>Nextflow</b> wf-artic version v0.3.2 (<ext-link xlink:href="https://github.com/epi2me-labs/wf-artic" ext-link-type="uri">https://github.com/epi2me-labs/wf-artic</ext-link>) with &#x201C;fast_min_variant_c507&#x201D; model.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35354286">35354286</a> <b>(1)</b> Clinical Metagenomic Sequencing for Species Identification and Antimicrobial Resistance Prediction in Orthopedic Device Infection<ul><li> The <b>Nextflow</b> (<xref rid="B26" ref-type="bibr"><u>26</u></xref>) workflow used is provided at <ext-link xlink:href="https://gitlab.com/ModernisingMedicalMicrobiology/genericbugontworkflow" ext-link-type="uri">https://gitlab.com/ModernisingMedicalMicrobiology/genericbugontworkflow</ext-link>.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35361860">35361860</a> <b>(1)</b> Transcriptome-wide analysis of glioma stem cell specific m6A modifications in long-non-coding RNAs<ul><li> pipeline (ver.: 1.2.2) run within Singularity<sup><xref ref-type="bibr" rid="CR54"><u>54</u></xref></sup> container through <b>Nextflow</b> process scheduler<sup><xref ref-type="bibr" rid="CR55"><u>55</u></xref></sup>.<li> [54](Kurtzer, G. M. <italic>et al.</italic> hpcng/singularity: Singularity 3.7.3. (2021) 10.5281/ZENODO.4667718.) [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35348588">35348588</a> <b>(1)</b> RNA Sequencing of Lens Capsular Epithelium Implicates Novel Pathways in Pseudoexfoliation Syndrome<ul><li>High-throughput sequencing analysis was carried out through a South Australian Genomics Centre developed workflow initialized using the <b>nextflow</b> workflow language (<ext-link xlink:href="https://github.com/sagc-bioinformatics/nf-rnaseq-sagc" ext-link-type="uri">https://github.com/sagc-bioinformatics/nf-rnaseq-sagc</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35348662">35348662</a> <b>(1)</b> Chromosome-Level Reference Genomes for Two Strains of <italic toggle="yes">Caenorhabditis briggsae</italic>: An Improved Platform for Comparative Genomics<ul><li>Once sequenced, raw reads for the 99 lines were processed for genotyping using the Andersen Lab's nil-ril <b>nextflow</b> pipeline (<ext-link xlink:href="https://github.com/AndersenLab/nil-ril-nf" ext-link-type="uri">https://github.com/AndersenLab/nil-ril-nf</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35343797">35343797</a> <b>(1)</b> Global Composition of the Bacteriophage Community in Honey Bees<ul><li> What the Phage was run on contigs of &gt;1,000&#x2009;bp in length (<b>nextflow</b> run replikation/What_the_Phage -r v0.9.0 &#x2013;cores 8 &#x2013;fasta polished_1.fasta -profile local,docker).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35338232">35338232</a> <b>(8)</b> PowerBacGWAS: a computational pipeline to perform power calculations for bacterial genome-wide association studies<ul><li>Docker and <b>Nextflow</b> implementation of PowerBacGWAS.<li> We have additionally implemented a <b>Nextflow</b> pipeline that uses this Docker image, to automate the multiple computational steps and parallelise the GWAS runs.<li> Using the <b>Nextflow</b> implementation, the pipeline is reduced to only three steps: (1) preparation of input files, (2) ancestral state reconstruction and (3) GWAS runs.<li> The GitHub page (<ext-link ext-link-type="uri" xlink:href="https://github.com/francesccoll/powerbacgwas">https://github.com/francesccoll/powerbacgwas</ext-link>)<sup><xref ref-type="bibr" rid="CR61"><u>61</u></xref></sup> includes instructions on how to install PowerBacGWAS via Docker and <b>Nextflow</b>, or locally.<li> The Usage Wikipage (<ext-link ext-link-type="uri" xlink:href="https://github.com/francesccoll/powerbacgwas/wiki#usage">https://github.com/francesccoll/powerbacgwas/wiki#usage</ext-link>) includes sections on how to run the pipeline using the Docker/<b>Nextflow</b> installation (&#x2018;<b>Nextflow</b> commands&#x2019;) or how to run individual scripts using the locally installation (&#x2018;Individual commands&#x2019;).<li> We recommend to run the <b>Nextflow</b> pipeline with the LSF executor, if an LSF cluster is available, for faster running times, wherein each process is submitted as a separate job.<li> Overall, for the six bacterial datasets used in this work, PowerBacGWAS used a median of 1,670 CPU hours (interquartile range: 548 to 3542) and a median duration of 3.1&#x2009;h (interquartile range: 2&#x2013;7.6&#x2009;h) when using the <b>Nextflow</b> LSF executor.<li> [61](Coll, F. PowerBacGWAS v1.0.0. <italic>Zenodo</italic>10.5281/zenodo.5950535 (2022).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35456751">35456751</a> <b>(1)</b> IntegronFinder 2.0: Identification and Analysis of Integrons across Bacteria, with a Focus on Antibiotic Resistance in Klebsiella<ul><li>IntegronFinder can now be easily run in parallel with a <b>Nextflow</b> script [<xref rid="B27-microorganisms-10-00700" ref-type="bibr"><u>27</u></xref>] that we provide ready to use.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35332213">35332213</a> <b>(1)</b> The ViReflow pipeline enables user friendly large scale viral consensus genome reconstruction<ul><li><b>Nextflow</b><sup><xref ref-type="bibr" rid="CR14"><u>14</u></xref></sup>.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35336849">35336849</a> <b>(1)</b> Age Estimate of <italic toggle="yes">GJB2</italic>-p.(Arg143Trp) Founder Variant in Hearing Impairment in Ghana, Suggests Multiple Independent Origins across Populations<ul><li> (Arg143Trp) (c.427C&gt;T) was estimated in the phased unimputed as well as the phased imputed datasets using a <b>nextflow</b>-based [<xref rid="B26-biology-11-00476" ref-type="bibr"><u>26</u></xref>] workflow (<uri xlink:href="https://github.com/esohkevin/mutationAge/tree/main">https://github.com/esohkevin/mutationAge/tree/main</uri>) (Accessed 25 November 2021) that utilized the Disequilibrium Mapping Likelihood Estimation (DMLE + 2.3) algorithm [<xref rid="B27-biology-11-00476" ref-type="bibr"><u>27</u></xref>].<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](          <article-title>DMLE+ Disease Mapping Using Linkage Disequilibrium</article-title>          <comment>Available online: <ext-link xlink:href="http://www.dmle.org/" ext-link-type="uri">http://www.dmle.org/</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2021-08-17">(accessed on 17 August 2021)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35306876">35306876</a> <b>(1)</b> Machine Learning of All <named-content content-type="genus-species">Mycobacterium tuberculosis</named-content> H37Rv RNA-seq Data Reveals a Structured Interplay between Metabolism, Stress Response, and Infection<ul><li> tuberculosis</named-content> RNA-seq compendium, we used Amazon Web Services (AWS) Batch to run a <b>Nextflow</b> pipeline (<xref rid="B14" ref-type="bibr"><u>14</u></xref>, <xref rid="B65" ref-type="bibr"><u>65</u></xref>).<li> [14](<person-group person-group-type="author"><string-name><surname>Sastry</surname><given-names>AV</given-names></string-name>, <string-name><surname>Poudel</surname><given-names>S</given-names></string-name>, <string-name><surname>Rychel</surname><given-names>K</given-names></string-name>, <string-name><surname>Yoo</surname><given-names>R</given-names></string-name>, <string-name><surname>Lamoureux</surname><given-names>CR</given-names></string-name>, <string-name><surname>Chauhan</surname><given-names>S</given-names></string-name>, <etal>et al</etal></person-group>. <year>2021</year>. <article-title>Mining all publicly available expression data to compute dynamic microbial transcriptional regulatory networks</article-title>. <source>bioRxiv</source><comment>2021.07.01.450581</comment>. doi:<pub-id pub-id-type="doi">10.1101/2021.07.01.450581</pub-id>.) [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35304452">35304452</a> <b>(1)</b> CD4 expression in effector T cells depends on DNA demethylation over a developmentally established stimulus-responsive element<ul><li>We used our computational pipeline developed with <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io">https://www.<b>nextflow</b>.io</ext-link>) to integrate gene-expression data with ChIP-Seq and ATAC-Seq data in order to generate a prioritized list of target genes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35276009">35276009</a> <b>(1)</b> Gene Regulatory Networks of Epidermal and Neural Fate Choice in a Chordate<ul><li> 2020</xref>), which runs <b>Nextflow</b> (v19.10.0) (<xref rid="msac055-B17" ref-type="bibr">Di Tommaso et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35277707">35277707</a> <b>(1)</b> Alevin-fry unlocks rapid, accurate, and memory-frugal quantification of single-cell RNA-seq data<ul><li> Samples were processed using a <b>Nextflow</b> (<xref rid="R53" ref-type="bibr"><u>26</u></xref>) workflow.<li> [26](R53) <li> [26](<name><surname>Tommaso</surname><given-names>Paolo Di</given-names></name>, <name><surname>Chatzou</surname><given-names>Maria</given-names></name>, <name><surname>Floden</surname><given-names>Evan W</given-names></name>, <name><surname>Barja</surname><given-names>Pablo Prieto</given-names></name>, <name><surname>Palumbo</surname><given-names>Emilio</given-names></name>, and <name><surname>Notredame</surname><given-names>Cedric</given-names></name>. <article-title>Nextflow enables reproducible computational workflows.</article-title><source>Nature Biotechnology</source>, <volume>35</volume> (<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>319</lpage>, <year>2017</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35255806">35255806</a> <b>(1)</b> Protocol variations in run-on transcription dataset preparation produce detectable signatures in sequencing libraries<ul><li> For further version information and specific input information, see <b>NextFlow</b> pipeline found at <ext-link ext-link-type="uri" xlink:href="https://github.com/Dowell-Lab/Nascent-Flow.git">https://github.com/Dowell-Lab/Nascent-Flow.git</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35255493">35255493</a> <b>(1)</b> Bibliometric review of ATAC-Seq and its application in gene expression<ul><li><b>Nextflow</b>, R.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35321470">35321470</a> <b>(1)</b> Rectal Colonization and Nosocomial Transmission of Carbapenem-Resistant <italic>Acinetobacter baumannii</italic> in an Intensive Care Unit, Southwest Nigeria<ul><li>All sequence analyses, except where otherwise stated, were carried out as described in the Global Health Research Unit (GHRU) protocol<sup><xref rid="footnote1" ref-type="fn"><u>1</u></xref></sup> using publicly available <b>Nextflow</b> pipelines.<li> [1](<a href="https://www.protocols.io/view/ghru-genomic-surveillance-of-antimicrobial-resista-bpn6mmhe">https://www.protocols.io/view/ghru-genomic-surveillance-of-antimicrobial-resista-bpn6mmhe</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35336136">35336136</a> <b>(2)</b> Transcriptomic Stress Response in <italic toggle="yes">Streptococcus mutans</italic> following Treatment with a Sublethal Concentration of Chlorhexidine Digluconate<ul><li> A <b>Nextflow</b>-based nf-core pipeline nf-core/rnaseq (version 1.4.2; <uri xlink:href="https://github.com/nf-core/rnaseq">https://github.com/nf-core/rnaseq</uri>, accessed on 22 January 2022) was used for the RNA-seq bioinformatic analysis.<li>The analysis of the differential gene expression was performed in R language (version 3.5.1) using DESeq2 (version 1.22) through the <b>Nextflow</b>-based workflow qbic-pipelines/rnadeseq (<uri xlink:href="https://github.com/qbic-pipelines/rnadeseq">https://github.com/qbic-pipelines/rnadeseq</uri>, accessed on 22 January 2022, version 1.3.2).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35241129">35241129</a> <b>(13)</b> Enhanced protein isoform characterization through long-read proteogenomics<ul><li> The entire pipeline and workflow is freely available as an open-source and extensible computational resource, using the community-based workflow language, <b>Nextflow</b>.<li> A <b>Nextflow</b> pipeline processes PacBio data, converts full-length transcripts into a protein database, and performs proteomics database searching (Fig.<li> modules of the <b>Nextflow</b> pipeline.<li> files&#x2014;were subjected to additional analysis using custom python scripts, which are part of the <b>Nextflow</b> pipeline.<li> module in the <b>Nextflow</b> pipeline).<li> module in the <b>Nextflow</b> pipeline).<li> module in the <b>Nextflow</b> pipeline).<li> module in the <b>Nextflow</b> pipeline).<li> module in the <b>Nextflow</b> pipeline).<li>Computational pipeline with <b>Nextflow</b>We implemented the long-read proteogenomic pipeline in <b>Nextflow</b>, a domain-specific language allowing for the highly flexible development of bioinformatic pipelines capable of being deployed on local machines, servers, or cloud environments [<xref ref-type="bibr" rid="CR73"><u>73</u></xref>].<li>We developed a <b>Nextflow</b> pipeline to process PacBio data, convert resulting transcripts into a protein database, and perform proteomics database searching.<li> </bold>Detailed schematic of the <b>Nextflow</b> computational pipeline forlong-read proteogenomics.&#xA0;<bold>Figure S2.&#xA0;</bold>Generation and characterization of candidate protein isoform sequencesfrom long-read RNA-seq data.&#xA0;<bold>Figure S3.<li> [73](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35238865">35238865</a> <b>(1)</b> The aryl hydrocarbon receptor contributes to tissue adaptation of intestinal eosinophils in mice<ul><li> Fastq files were aligned to the GRCm38 Ensembl Release 86 mouse genome using the <b>nextflow</b> package nf-core/rnaseq to generate gene counts.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35241105">35241105</a> <b>(2)</b> Mini-XT, a miniaturized tagmentation-based protocol for efficient sequencing of SARS-CoV-2<ul><li> <b>nextflow</b>-based open-source pipeline [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>], as used by the COG-UK consortium, was then used with its &#x201C;illumina&#x201D; and &#x201C;profile conda&#x201D; options, and otherwise default options.<li>To analyse Nanopore data, the ncov2019-artic <b>nextflow</b> pipeline [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>] was used to automate the ARTIC network nCoV-2019 novel coronavirus bioinformatics protocol [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> [30](Connor-Lab, Nextflow pipeline for running the ARTIC network's field bioinformatics tools. GitHub: <ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>) [30](Connor-Lab, Nextflow pipeline for running the ARTIC network's field bioinformatics tools. GitHub: <ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>) [34](Loman NJ, Will R, Andrew R. NCoV-2019 novel coronavirus bioinformatics protocol&#x2019;. v1.1.0. 2020, URL: <ext-link ext-link-type="uri" xlink:href="https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html">https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35234913">35234913</a> <b>(2)</b> GREEN-DB: a framework for the annotation and prioritization of non-coding regulatory variants from whole-genome sequencing data<ul><li> Additional pre-processed datasets useful for annotation and a <b>Nextflow</b> (<xref rid="B51" ref-type="bibr"><u>51</u></xref>) workflow are distributed together with GREEN-DB and can be used to generate a fully annotated VCF for small non-coding variants.<li> We provide a complete annotation workflow implemented in <b>Nextflow</b> that uses our GREEN-VARAN tool to prioritize regulatory variants in VCF files from whole genome sequencing data.<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35133957">35133957</a> <b>(1)</b> Spatiotemporal Analyses of 2 Co-Circulating SARS-CoV-2 Variants, New York State, USA<ul><li> We processed Illumina libraries with the ARTIC <b>nextflow</b> pipeline (<ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>) as previously described (<xref rid="R14" ref-type="bibr"><italic>14</italic></xref>) (<xref rid="SD1" ref-type="supplementary-material"><u>Appendix</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304334">36304334</a> <b>(3)</b> CIRCprimerXL: Convenient and High-Throughput PCR Primer Design for Circular RNA Quantification<ul><li> The pipeline is written using <b>Nextflow</b> (version 20.10.0.5430) and runs in a Docker container (version 20.10.11).<li>The <b>Nextflow</b> pipeline allows the user even more flexibility in primer design in addition to the options described for the web interface.<li>Due to the use of <b>Nextflow</b> and Docker, CIRCprimerXL runs fast and is highly scalable (<xref rid="T3" ref-type="table"><u>Table 3</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35300010">35300010</a> <b>(1)</b> A High-Quality Reference Genome Sequence and Genetic Transformation System of <italic>Aralia elata</italic><ul><li> The gene expression analysis was performed using the nf-core/rnaseq v3.2 (<xref rid="ref16" ref-type="bibr"><u>Ewels et al., 2020</u></xref>) pipeline in <b>nextflow</b> v21.04.1 (<xref rid="ref13" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35250265">35250265</a> <b>(3)</b> Decoding Human Genome Regulatory Features That Influence HIV-1 Proviral Expression and Fate Through an Integrated Genomics Approach<ul><li>The <b>Nextflow</b><sup><xref rid="bibr28-11779322211072333" ref-type="bibr"><u>28</u></xref></sup> BICF ChIP-seq Analysis Workflow version 1.0.0<sup><xref rid="bibr29-11779322211072333" ref-type="bibr"><u>29</u></xref></sup> processed all ChIP-seq files, merging separate experiments as technical replicates.<li>FASTQ files were processed with a modified <b>Nextflow</b>,<sup><xref rid="bibr28-11779322211072333" ref-type="bibr"><u>28</u></xref></sup> BICF ChIP-seq Analysis Workflow version 1.0.0.<li>FASTQ files were processed with a modified <b>Nextflow</b>,<sup><xref rid="bibr28-11779322211072333" ref-type="bibr"><u>28</u></xref></sup> BICF ChIP-seq Analysis Workflow version 1.0.0.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [29](<person-group person-group-type="author"><name><surname>Barnes</surname><given-names>SD</given-names></name><name><surname>Ruess</surname><given-names>H</given-names></name><name><surname>Mathews</surname><given-names>JA</given-names></name><name><surname>Cheng</surname><given-names>B</given-names></name><name><surname>Malladi</surname><given-names>VS.</given-names></name></person-group><source>BICF ChIP-seq Analysis Workflow</source> (publish_1.0.5). <publisher-loc>Geneva, Switzerland</publisher-loc>: <publisher-name>Zenodo</publisher-name>; <year>2019</year>. doi:<pub-id pub-id-type="doi">10.5281/zenodo.2648844.</pub-id>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35310944">35310944</a> <b>(1)</b> Increasing transparency and reproducibility in stroke-microbiota research: A toolbox for microbiota analysis<ul><li> QIIME2 and containerized pipelines, such as the <b>nextflow</b>-based nf-core/ampliseq, extend this concept further by including provenance tracking and enabling fully reproducible analysis across various computing infrastructures in the case of nf-core (<xref rid="bib15" ref-type="bibr"><u>Bolyen et&#xA0;al., 2019</u></xref>; <xref rid="bib103" ref-type="bibr"><u>Straub et&#xA0;al., 2020</u></xref>).<li> [Bolyen et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31341288">31341288</a>) [Straub et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33193131">33193131</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35269430">35269430</a> <b>(2)</b> Chromatin Remodeler Smarca5 Is Required for Cancer-Related Processes of Primary Cell Fitness and Immortalization<ul><li> The <b>Nextflow</b> pipeline used is available at <uri xlink:href="https://github.com/IARCbioinfo/alignment-nf">https://github.com/IARCbioinfo/alignment-nf</uri>.<li> then the read number per gene per sample was counted with htseq-count (using the <b>Nextflow</b> pipeline described in <uri xlink:href="https://github.com/IARCbioinfo/RNAseq-nf">https://github.com/IARCbioinfo/RNAseq-nf</uri>, accessed on 18 February 2020).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35217695">35217695</a> <b>(4)</b> PacBio long-read amplicon sequencing enables scalable high-resolution population allele typing of the complex <italic>CYP2D6</italic> locus<ul><li> Our pipeline, Phased Long Allele Sequence Typing with Error Removal (PLASTER), is implemented in <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR28"><u>28</u></xref></sup> and available at <ext-link ext-link-type="uri" xlink:href="https://github.com/bahlolab/long-amplicon-typing">https://github.com/bahlolab/PLASTER</ext-link>.<li>The entire data processing pipeline, PLASTER, has been implemented in <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR28"><u>28</u></xref></sup> and is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/bahlolab/long-amplicon-typing">https://github.com/bahlolab/PLASTER</ext-link>.<li> This stage is implemented in a custom R script (&#x201C;fusion_call.R&#x201D;), using the DECIPHER<sup><xref ref-type="bibr" rid="CR51"><u>51</u></xref></sup>, RSamtools<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>, and Biostrings<sup><xref ref-type="bibr" rid="CR53"><u>53</u></xref></sup> R packages, and is available as part of the <b>Nextflow</b> pipeline.<li> Nonetheless, we implemented a <b>Nextflow</b> pipeline matching the methods used as closely as possible, using a current version of PacBio&#x2019;s long-amplicon analysis (2.4.2) and the amplicon splitting stage from PLASTER, and compared the resulting CYP2D6 star alleles assignments of the control samples.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [51](Wright, E. S. Using DECIPHER v2. 0 to analyze big biological sequence data in R. <italic>R J.</italic><bold>8</bold>, 352 (2016).) [52](Morgan, M., Pag&#xE8;s, H., Obenchain, V. &amp; Hayden, N. <italic>Rsamtools: binary alignment (BAM), FASTA, variant call (BCF), and tabix file import</italic>. <ext-link ext-link-type="uri" xlink:href="https://bioconductor.org/packages/Rsamtools">https://bioconductor.org/packages/Rsamtools</ext-link> (2021).) [53](Pag&#xE8;s, H., Aboyoun, P., Gentleman, R. &amp; DebRoy, S. <italic>Biostrings: efficient manipulation of biological strings</italic>, <ext-link ext-link-type="uri" xlink:href="https://bioconductor.org/packages/Biostrings">https://bioconductor.org/packages/Biostrings</ext-link> (2021).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35217577">35217577</a> <b>(1)</b> Tumor-specific T cells support chemokine-driven spatial organization of intratumoral immune microaggregates needed for long survival<ul><li> Quality control and downstream analysis was performed using Scanpy V.1.6.0<xref rid="R18" ref-type="bibr"><u>18</u></xref> and Scirpy V.0.6.1.<xref rid="R19" ref-type="bibr"><u>19</u></xref> All scripts used to perform the analysis are combined in one fully reproducible <b>Nextflow</b> pipeline and publicly available from GitHub (<ext-link xlink:href="https://github.com/icbi-lab/abdulrahman2021_paper" ext-link-type="uri">https://github.com/icbi-lab/abdulrahman2021_paper</ext-link>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29301551">29301551</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32614448">32614448</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35213550">35213550</a> <b>(1)</b> Proteomic analysis of the umbilical cord in fetal growth restriction and preeclampsia<ul><li> Briefly, raw.d files generated from the timsTOF Pro were converted to mzML with OpenMS (v 2.5.0) and tdf2mzml in-house <b>nextflow</b> script.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35281844">35281844</a> <b>(1)</b> Epigenetic Modifier Supplementation Improves Mitochondrial Respiration and Growth Rates and Alters DNA Methylation of Bovine Embryonic Fibroblast Cells Cultured in Divergent Energy Supply<ul><li> <b>Nextflow</b> methylseq pipeline v1.5 (<xref rid="B18" ref-type="bibr"><u>Ewels et al., 2020</u></xref>) was used to perform read mapping and methylation call through Bismark v0.22.3 software.<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35281056">35281056</a> <b>(1)</b> Extracellular Vesicles From Kidney Allografts Express miR-218-5p and Alter Th17/Treg Ratios<ul><li> Read processing was coordinated using custom <b>Nextflow</b> pipeline.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35194141">35194141</a> <b>(1)</b> BugSplit enables genome-resolved metagenomics through highly accurate taxonomic binning of metagenomic assemblies<ul><li>BugSplit uses <b>Nextflow</b> running on AWS Batch to orchestrate processing of sequencing data in the cloud.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35184734">35184734</a> <b>(7)</b> TC-hunter: identification of the insertion site of a transgenic gene within the host genome<ul><li>TC-hunter is a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] pipeline that scans NGS data to report predicted TIS within a sufficiently contiguous host genome making use of chimeric reads and discordant read pairs.<li> Figure&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref> illustrates how these steps are interconnected through channels that are handled in a main <b>Nextflow</b> script allowing for parallelization of data, scripts and tools.<li> Moreover, TC-hunter can be resumed at any stage since <b>Nextflow</b> provides this capability.<li>TC-hunter is an open source pipeline available on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/bcfgothenburg/SSF">https://github.com/bcfgothenburg/SSF</ext-link>) and it requires the installation of the following tools: R 3.5 or higher [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>], python 2.7 [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], samtools 1.10 [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>], <b>Nextflow</b> 19.01.0 [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] and BWA 0.7 [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> If specified, TC-hunter will create a second html report containing the <b>Nextflow</b> workflow execution information.<li> While it uses common bioinformatics software in the analysis of sequencing data, the different steps are collected in a main <b>Nextflow</b> script that simplifies its use.<li> The <b>Nextflow</b> workflow reports can be accessed at <ext-link ext-link-type="uri" xlink:href="https://github.com/bcfgothenburg/SSF">https://github.com/bcfgothenburg/SSF</ext-link>.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](          <person-group person-group-type="author">            <collab>Team RC</collab>          </person-group>          <source>R: A Language and Environment for Statistical Computing</source>          <year>2016</year>          <publisher-loc>Vienna</publisher-loc>          <publisher-name>Austria. In</publisher-name>        ) [16](Van Rossum G, Drake Jr FL: Python reference manual. In., 3.6 edn: Centrum voor Wiskunde en Informatica Amsterdam; 1995.) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H</given-names>            </name>          </person-group>          <source>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</source>          <year>2013</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35171290">35171290</a> <b>(4)</b> Phylodynamic Inference of Bacterial Outbreak Parameters Using Nanopore Sequencing<ul><li> 2017</xref>) (contiguous and highly accurate) were assembled using a standardized <b>Nextflow</b> (<xref rid="msac040-B7" ref-type="bibr">Di Tommaso et al.<li> Finally, we implemented training, evaluation, and deployment of SNP polishers for within-lineage transmission modeling in <b>Nextflow</b>.<li> Although we chose <italic toggle="yes">S.&#xA0;aureus</italic> as a model organism for this work mainly due to our interest in sequencing the outbreaks in PNG and FNQ, as well as because of its relatively small genome (2.8 Mbp), core principles and methods used in this study are immediately applicable to other bacterial pathogens and all steps of the pipelines are implemented in replicable <b>Nextflow</b> workflows (Materials and Methods).<li>Genome assemblies for genotyping were constructed using our <b>Nextflow</b> assembly pipeline (<ext-link xlink:href="https://github.com/np-core/np-assembly" ext-link-type="uri">https://github.com/np-core/np-assembly</ext-link>), which first randomly subsamples reads to a maximum of 200&#xD7; coverage with <italic toggle="yes">rasusa</italic> v0.3.0 (<xref rid="msac040-B19" ref-type="bibr"><u>Hall 2022</u></xref>) and filtered Q&#x2009;&gt;&#x2009;7 with minimum read length of 100&#x2009;bp using <italic toggle="yes">nanoq v0.8.0</italic> (<xref rid="msac040-B40" ref-type="bibr"><u>Steinig and Coin 2022</u></xref>).<li> [Hall 2022](<person-group person-group-type="author"><string-name><surname>Hall</surname><given-names>M.</given-names></string-name></person-group><year>2022</year>. <article-title>Rasusa: randomly subsample sequencing reads to a specified coverage</article-title>. <source>J Open Source Softw</source>. <volume>7</volume>(<issue>69</issue>):<fpage>3941</fpage>.) [Steinig and Coin 2022](<person-group person-group-type="author"><string-name><surname>Steinig</surname><given-names>E</given-names></string-name>, <string-name><surname>Coin</surname><given-names>L.</given-names></string-name></person-group><year>2022</year>. <article-title>Nanoq: ultra-fast quality control for nanopore reads</article-title>. <source>J Open Source Softw</source>. <volume>7</volume>(<issue>69</issue>):<fpage>2991</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34990404">34990404</a> <b>(4)</b> Rapid idiosyncratic mechanisms of clinical resistance to KRAS G12C inhibition<ul><li> The somatic variant&#x2013;calling workflow is documented at <ext-link xlink:href="https://sc.unc.edu/lbg/workflows/<b>nextflow</b>/somatic," ext-link-type="uri">https://sc.unc.edu/lbg/workflows/<b>nextflow</b>/somatic,</ext-link> and the &#x201C;lbg_hg38_SureSelect_AllExonV6&#x201D; profile was used.<li> The expressed variant calling workflow can be found at <ext-link xlink:href="https://sc.unc.edu/lbg/workflows/<b>nextflow</b>/expressed_variants" ext-link-type="uri">https://sc.unc.edu/lbg/workflows/<b>nextflow</b>/expressed_variants</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35157051">35157051</a> <b>(2)</b> riboviz 2: a flexible and robust ribosome profiling data analysis and visualization workflow<ul><li>The riboviz 2 pipeline is implemented via <b>Nextflow</b> (<xref rid="btac093-B3" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li> As riboviz 2 is implemented as a <b>Nextflow</b> workflow going from raw data to visualization while requiring only a single configuration YAML file, reproducing an analysis does not require independently running various tools.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35317427">35317427</a> <b>(1)</b> Atrophic gastritis and gastric cancer tissue miRNome analysis reveals hsa-miR-129-1 and hsa-miR-196a as potential early diagnostic biomarkers<ul><li>Analysis of raw small RNA-seq data was performed by nf-core/smrnaseq pipeline v.1.0.0 including <b>Nextflow</b> v.20.07.1[<xref rid="B15" ref-type="bibr"><u>15</u></xref>], Java v.11.0.7, and Docker v.19.03.12.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30423142">30423142</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35685574">35685574</a> <b>(1)</b> Genomic Analysis of AZD1222 (ChAdOx1) Vaccine Breakthrough Infections in the City of Mumbai<ul><li> A custom, reproducible <b>nextflow</b> workflow was constructed for further data processing.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35207753">35207753</a> <b>(1)</b> African Genomic Medicine Portal: A Web Portal for Biomedical Applications<ul><li> The pipeline is implemented in the <b>Nextflow</b> workflow manager (<uri xlink:href="https://github.com/hothman/text_mining/tree/master/workflow">https://github.com/hothman/text_mining/tree/master/workflow</uri>, accessed on the 10 January 2022).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35207751">35207751</a> <b>(3)</b> SWAAT Bioinformatics Workflow for Protein Structure-Based Annotation of ADME Gene Variants<ul><li>SWAAT was built and tested on&#xA0;version 20.10.0 of&#xA0;<b>Nextflow</b> [<xref rid="B41-jpm-12-00263" ref-type="bibr"><u>41</u></xref>].<li>SWAAT consists of&#xA0;a set of&#xA0;Python scripts and Bash instructions whose series of&#xA0;execution is managed within <b>Nextflow</b>&#x2019;s DSL [<xref rid="B41-jpm-12-00263" ref-type="bibr"><u>41</u></xref>].<li><monospace>&#xA0;&#xA0;<b>nextflow</b>&#xA0;run&#xA0;main.nf&#xA0;--dbhome&#xA0;/home/hothman/SWAAT/database/&#xA0;\</monospace>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35143491">35143491</a> <b>(1)</b> Ten simple rules for large-scale data processing<ul><li>There are workflow tools that can support many use cases including workflow languages and systems such as Workflow Description Language (WDL) [<xref rid="pcbi.1009757.ref012" ref-type="bibr"><u>12</u></xref>], Common Workflow Language (CWL) [<xref rid="pcbi.1009757.ref013" ref-type="bibr"><u>13</u></xref>], Snakemake [<xref rid="pcbi.1009757.ref014" ref-type="bibr"><u>14</u></xref>], or <b>Nextflow</b> [<xref rid="pcbi.1009757.ref015" ref-type="bibr"><u>15</u></xref>].<li> [12](Workflow Description Language (WDL) [Internet]. OpenWDL; 2021 [cited 2021 Jul 29]. Available from: <ext-link xlink:href="https://github.com/openwdl/wdl" ext-link-type="uri">https://github.com/openwdl/wdl</ext-link>.) [13](<name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, <name><surname>Kartashov</surname><given-names>A</given-names></name>, <name><surname>Leehr</surname><given-names>D</given-names></name>, <name><surname>M&#xE9;nager</surname><given-names>H</given-names></name>, <etal>et al</etal>. <article-title>Common Workflow Language, v1.0.</article-title><name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, editors. United States: figshare; <year>2016</year>.) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35138128">35138128</a> <b>(2)</b> The COMBAT-TB Workbench: Making Powerful <named-content content-type="genus-species">Mycobacterium tuberculosis</named-content> Bioinformatics Accessible<ul><li><b>Nextflow</b> workflows.<li>Innuendo is a web interface to sequence storage and <b>Nextflow</b> (<xref rid="B21" ref-type="bibr"><u>21</u></xref>) workflow execution aimed at analysis of foodborne pathogens.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35140207">35140207</a> <b>(2)</b> A SIMPLI (Single-cell Identification from MultiPLexed Images) approach for spatially-resolved tissue phenotyping at single-cell resolution<ul><li>SIMPLI is implemented as a <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup> pipeline employing Singularity containers<sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup> hosted on Singularity Hub<sup><xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup> to manage all the libraries and software tools.<li> <b>Nextflow</b> also manages automatic parallelisation of all processes while still allowing the selection of parts of the analysis to execute.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29186161">29186161</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35205129">35205129</a> <b>(1)</b> Side-by-Side Comparison of Post-Entry Quarantine and High Throughput Sequencing Methods for Virus and Viroid Diagnosis<ul><li>We compared the detection of viruses and viroids by the sRNA-Seq technology using two methods: VirusDetect 1.7 [<xref rid="B40-biology-11-00263" ref-type="bibr"><u>40</u></xref>] and VirReport (<uri xlink:href="https://github.com/eresearchqut/virreport.git">https://github.com/eresearchqut/virreport.git</uri>, accessed on 2 February 2022), a portable, scalable and reproducible <b>nextflow</b> [<xref rid="B41-biology-11-00263" ref-type="bibr"><u>41</u></xref>] pipeline which uses a similar framework to the Yabi web-based Virus Surveillance and Diagnostics pipeline [<xref rid="B42-biology-11-00263" ref-type="bibr"><u>42</u></xref>] (<xref rid="app1-biology-11-00263" ref-type="app"><u>Supplementary Materials Figure S2</u></xref>).<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27825033">27825033</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [42](          <person-group person-group-type="author"><name><surname>Barrero</surname><given-names>R.A.</given-names></name><name><surname>Napier</surname><given-names>K.R.</given-names></name><name><surname>Cunnington</surname><given-names>J.</given-names></name><name><surname>Liefting</surname><given-names>L.</given-names></name><name><surname>Keenan</surname><given-names>S.</given-names></name><name><surname>Frampton</surname><given-names>R.A.</given-names></name><name><surname>Szab&#xF3;</surname><given-names>T.O.</given-names></name><name><surname>Bulman</surname><given-names>S.</given-names></name><name><surname>Hunter</surname><given-names>A.</given-names></name><name><surname>Ward</surname><given-names>L.</given-names></name><etal/></person-group>          <article-title>An internet-based bioinformatics toolkit for plant biosecurity diagnosis and surveillance of viruses and viroids</article-title>          <source>BMC Bioinform.</source>          <year>2017</year>          <volume>18</volume>          <elocation-id>26</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12859-016-1428-4</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35130839">35130839</a> <b>(1)</b> A data management infrastructure for the integration of imaging and omics data in life sciences<ul><li> Moreover, workflow systems such as gUSE [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>] and <b>Nextflow</b> [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] have been successfully connected to the qPortal platform.<li> [30](          <person-group person-group-type="author">            <name>              <surname>Kacsuk</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Farkas</surname>              <given-names>Z</given-names>            </name>            <name>              <surname>Kozlovszky</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Hermann</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Balasko</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Karoczkai</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Marton</surname>              <given-names>I</given-names>            </name>          </person-group>          <article-title>WS-PGRADE/gUSE generic DCI gateway framework for a large variety of user communities</article-title>          <source>Int J Grid Util Comput</source>          <year>2012</year>          <volume>10</volume>          <issue>4</issue>          <fpage>601</fpage>          <lpage>630</lpage>          <pub-id pub-id-type="doi">10.1007/s10723-012-9240-5</pub-id>        ) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35202261">35202261</a> <b>(1)</b> Isolation of DiNP-Degrading Microbes from the Mouse Colon and the Influence DiNP Exposure Has on the Microbiota, Intestinal Integrity, and Immune Status of the Colon<ul><li> The 16S rRNA genes were processed using the TADA <b>Nextflow</b>-based workflow (<uri xlink:href="https://github.com/h3abionet/TADA">https://github.com/h3abionet/TADA</uri>, accessed date 25 February 2021), which implements the DADA2 workflow for denoising and dereplicating reads to generate amplicon sequence variants (ASVs) [<xref rid="B29-toxics-10-00075" ref-type="bibr"><u>29</u></xref>].<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27214047">27214047</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35129076">35129076</a> <b>(1)</b> Differential expression of long non-coding RNAs under <italic toggle="yes">Peste des petits ruminants virus</italic> (PPRV) infection in goats<ul><li> LncEvo is an all-in-one <b>Nextflow</b> based pipeline for transcriptome assembly and lncRNA identification, which takes paired FASTQ reads and genome assembly IDs as the input and uses customized scripts to essentially automate the process.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35134928">35134928</a> <b>(2)</b> Interpretable network-guided epistasis detection<ul><li>Programming language:&#xA0;<b>nextflow</b>.<li>Other requirements: Bash, <b>nextflow</b>, PLINK 1.9 and R 4.0 or higher, with the following packages: clusterProfiler, data.table, igraph, msigdbr, snpStats, tidyverse.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35118380">35118380</a> <b>(3)</b> nf-core/mag: a best-practice pipeline for metagenome hybrid assembly and binning<ul><li> However,&#xA0;only a few pipelines such as Muffin (<xref rid="B7" ref-type="bibr"><u>7</u></xref>) and ATLAS (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) make use of workflow management systems, such as Snakemake (<xref rid="B9" ref-type="bibr"><u>9</u></xref>) or <b>Nextflow</b> (<xref rid="B10" ref-type="bibr"><u>10</u></xref>), which facilitate scalability, portability, reproducibility and ease of application.<li>Here we introduce nf-core/mag, a <b>Nextflow</b> pipeline for hybrid assembly of metagenomes, binning and taxonomic classification of MAGs.<li>nf-core/mag is written in <b>Nextflow</b>, making use of the new DSL2 syntax (see Supplementary Data: Section S2).<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33561126">33561126</a>) [8](<person-group person-group-type="author"><string-name><surname>Kieser</surname><given-names>S.</given-names></string-name>, <string-name><surname>Brown</surname><given-names>J.</given-names></string-name>, <string-name><surname>Zdobnov</surname><given-names>E.M.</given-names></string-name>, <string-name><surname>Trajkovski</surname><given-names>M.</given-names></string-name>, <string-name><surname>McCue</surname><given-names>L.A.</given-names></string-name></person-group><article-title>ATLAS: a snakemake workflow for assembly, annotation, and genomic binning of metagenome sequence data</article-title>. <source>BMC Bioinf.</source><year>2020</year>; <volume>21</volume>:<fpage>257</fpage>.) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35108259">35108259</a> <b>(1)</b> Global diversity and balancing selection of 23 leading <italic toggle="yes">Plasmodium falciparum</italic> candidate vaccine antigens<ul><li> We followed the standard best practice from <italic toggle="yes">Genome Analysis Toolkit (GATK)</italic> version 4.0.12.0 implemented in <italic toggle="yes"><b>nextflow</b></italic> (<ext-link xlink:href="https://github.com/gatk-workflows/gatk4-germline-snps-indels" ext-link-type="uri">https://github.com/gatk-workflows/gatk4-germline-snps-indels</ext-link>) with some minor changes as follows: (i) running the pipeline twice&#x2014;the first without base quality score recalibration (BQSR) and (ii) using the pass variants from the first run for BQSR and hard filtering of variants instead of variant quality score recalibration (VQSR) as there is no external variant call set available.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36699395">36699395</a> <b>(4)</b> SangeR: the high-throughput Sanger sequencing analysis pipeline<ul><li>The <monospace>SangeR</monospace> pipeline uses <monospace>R</monospace>, <monospace><b>Nextflow</b></monospace> (<xref rid="vbac009-B2" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li><italic toggle="yes"><monospace><b>Nextflow</b></monospace> pipeline</italic>: To process large quantities of files, we provide <monospace>SangeR</monospace> as a <monospace><b>Nextflow</b></monospace> script.<li> <monospace><b>Nextflow</b></monospace> is a reactive workflow framework for orchestrating scripts to enable high data throughput.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35163597">35163597</a> <b>(1)</b> Effects of Yeast Species and Processing on Intestinal Health and Transcriptomic Profiles of Atlantic Salmon (<italic toggle="yes">Salmo salar</italic>) Fed Soybean Meal-Based Diets in Seawater<ul><li>RNA sequence data analysis was performed using the publicly available nf-core/RNA-seq pipeline version 3.3 implemented in <b>Nextflow</b> 21.04.0 [<xref rid="B50-ijms-23-01675" ref-type="bibr"><u>50</u></xref>].<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34981076">34981076</a> <b>(1)</b> Omicron infection of vaccinated individuals enhances neutralizing immunity against the Delta variant<ul><li> In addition, we also used the wf_artic (ARTIC SARS-CoV-2) pipeline as built using the <b>nextflow</b> workflow framework57.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35082388">35082388</a> <b>(1)</b> Utilizing museomics to trace the complex history and species boundaries in an avian-study system of conservation concern<ul><li> xanthornus</italic> draft genome, the <b>NextFlow</b>-core pipeline Neutronstar was used.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35081137">35081137</a> <b>(3)</b> ASPICov: An automated pipeline for identification of SARS-Cov2 nucleotidic variants<ul><li> This pipeline is a multistep <b>Nextflow</b> [<xref rid="pone.0262953.ref003" ref-type="bibr"><u>3</u></xref>] pipeline able to process raw-reads sequences into usable information such as quality reports, VCF files, sequence consensus and plots (variants and coverage).<li>ASPICov workflow was created as a <b>Nextflow</b> pipeline following some of the nf-core standards requirements to setup a portable pipeline.<li> Project name, technology, method, path to data, references, Trimmomatic adapter and bedpe files information will be completed by users in a custom configuration file (supported by profile), according to standard <b>Nextflow</b> principles.<li> [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36699378">36699378</a> <b>(4)</b> MODalyseR&#x2014;a novel software for inference of disease module hub regulators identified a putative multiple sclerosis regulator supported by independent eQTL data<ul><li> To increase its usability for a great variety of researchers, we have included it as a R shiny web application in a docker image to run on all operating systems, with the option to export computationally demanding methods as an automated <b>nextflow</b> (<xref rid="vbac006-B17" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li> For computationally demanding methods and datasets a self-contained <b>nextflow</b> script that automatically downloads a singularity image (<xref rid="vbac006-B28" ref-type="bibr">Kurtzer <italic toggle="yes">et al.<li> The output of this <b>nextflow</b> analysis can then be uploaded to MODalyseR.<li> The option to download and run module and network inference methods as a <b>nextflow</b> pipeline, locally or on a cluster, we believe increases the applicability of the software by making it possible to use the tool on large datasets.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35076693">35076693</a> <b>(10)</b> A simple guide to <italic toggle="yes">de novo</italic> transcriptome assembly and annotation<ul><li><monospace>transXpress</monospace> - <ext-link xlink:href="https://github.com/transXpress/transXpress-<b>nextflow</b>" ext-link-type="uri">https://github.com/transXpress/transXpress-<b>nextflow</b></ext-link> (<bold><b>Nextflow</b></bold> version), <ext-link xlink:href="https://github.com/transXpress/transXpress-snakemake" ext-link-type="uri">https://github.com/transXpress/transXpress-snakemake</ext-link> (<bold>Snakeake</bold> version).<li>Recent publications [<xref rid="ref217" ref-type="bibr"><u>217</u></xref>, <xref rid="ref222" ref-type="bibr"><u>222</u></xref>, <xref rid="ref223" ref-type="bibr"><u>223</u></xref>] indicate that the most popular WfMS today include <monospace><b>Nextflow</b></monospace> [<xref rid="ref224" ref-type="bibr"><u>224</u></xref>], <monospace>Snakmake</monospace> [<xref rid="ref221" ref-type="bibr"><u>221</u></xref>] and <monospace>Common Workflow Language</monospace> (<monospace>CWL</monospace>) [<xref rid="ref225" ref-type="bibr"><u>225</u></xref>].<li><monospace><b>Nextflow</b></monospace> is a powerful WfMS based on the <monospace>Groovy</monospace> programming language.<li> Consequently, <monospace><b>Nextflow</b></monospace> permits chaining together scripts (and tools) written in different languages as long as they can be executed on a <monospace>Unix</monospace>-like operating system [<xref rid="ref228" ref-type="bibr"><u>228</u></xref>].<li> Like <monospace>Snakemake</monospace> &#xA0;<monospace><b>Nextflow</b></monospace> is also scalable and platform-agnostic with regards to execution capabilities.<li> A salient feature of <monospace><b>Nextflow</b></monospace> is <monospace>nf-core</monospace> (<ext-link xlink:href="https://nf-co.re/" ext-link-type="uri">https://nf-co.re/</ext-link>) [<xref rid="ref229" ref-type="bibr"><u>229</u></xref>].<li> This is a curated repository of bioinformatics pipelines written in <monospace><b>Nextflow</b></monospace> that cover a range of use cases including RNA-seq, sequence assembly, phylogenetics and sequence annotation.<li><monospace><b>Nextflow</b></monospace> - <ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>.<li> [217](<!--<QUERY id="Q64" name="Bhardwaj, Punit"><p>Please provide the volume, first page for reference 217.</p></QUERY>--><person-group person-group-type="author"><string-name><surname>Leipzig</surname> &#xA0;<given-names>J</given-names></string-name></person-group>. <article-title>A review of bioinformatic pipeline frameworks</article-title>. <source>Brief Bioinform</source> &#xA0;<year>March 2016</year>;<elocation-id>bbw020</elocation-id>.) [222](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a>) [223](<person-group person-group-type="author"><string-name><surname>Reiter</surname> &#xA0;<given-names>T</given-names></string-name>, <string-name><surname>Brooks</surname> &#xA0;<given-names>PT</given-names></string-name>, <string-name><surname>Irber</surname> &#xA0;<given-names>L</given-names></string-name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Streamlining data-intensive biology with workflow systems</article-title>. <source>Gigascience</source> &#xA0;<year>January 2021</year>;<volume>10</volume>(<issue>1</issue>).) [224](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [221](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [225](<!--<QUERY id="Q68" name="Bhardwaj, Punit"><p>Please provide the publisher name, publisher location, editor name for reference 225.</p></QUERY>--><person-group person-group-type="author"><string-name><given-names>Peter</given-names> &#xA0;<surname>Amstutz</surname></string-name>, <string-name><given-names>Michael R</given-names> &#xA0;<surname>Crusoe</surname></string-name>, <string-name><given-names>Neboj&#x161;a</given-names> &#xA0;<surname>Tijani&#x107;</surname></string-name>, <string-name><given-names>Brad</given-names> &#xA0;<surname>Chapman</surname></string-name>, <string-name><given-names>John</given-names> &#xA0;<surname>Chilton</surname></string-name>, <string-name><given-names>Michael</given-names> &#xA0;<surname>Heuer</surname></string-name>, <string-name><given-names>Andrey</given-names> &#xA0;<surname>Kartashov</surname></string-name>, <string-name><given-names>Dan</given-names> &#xA0;<surname>Leehr</surname></string-name>, <string-name><given-names>Herv&#xE9;</given-names> &#xA0;<surname>M&#xE9;nager</surname></string-name>, <string-name><given-names>Maya</given-names> &#xA0;<surname>Nedeljkovich</surname></string-name>, <string-name><given-names>Matt</given-names> &#xA0;<surname>Scales</surname></string-name>, <string-name><given-names>Stian</given-names> &#xA0;<surname>Soiland-Reyes</surname></string-name>, and <string-name><given-names>Luka</given-names> &#xA0;<surname>Stojanovic</surname></string-name></person-group>. <source>Common workflow language</source>, v1.0, <year>July 2016</year>.) [228](<person-group person-group-type="author"><collab>nextflow</collab></person-group>. <comment>Basic concepts - Nextflow 21.04.1 documentation</comment>.) [229](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35072628">35072628</a> <b>(2)</b> Comprehensive characterization of the antibody responses to SARS-CoV-2 Spike protein finds additional vaccine-induced epitopes beyond those for mild infection<ul><li> Post sample demultiplexing, we used a <italic toggle="yes"><b>Nextflow</b></italic> pipeline to process the peptide counts as well as alignment stats for all samples (<xref rid="bib5" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> The source code, <b>Nextflow</b> script, software dependencies, and instructions for rerunning the analysis can be found at <ext-link xlink:href="https://github.com/matsengrp/phage-dms-vacc-analysis" ext-link-type="uri">https://github.com/matsengrp/phage-dms-vacc-analysis</ext-link> (copy archived at <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:e087c50fd0a4f182f15ee6fe5b014058ceb2c8e3;origin=https://github.com/matsengrp/phage-dms-vacc-analysis;visit=swh:1:snp:c9917cbeef868e0898ef8a51b3349f4a2fd310e7;anchor=swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86" ext-link-type="uri">swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86</ext-link>, <xref rid="bib30" ref-type="bibr"><u>Matsen, 2022</u></xref>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Matsen, 2022](          <person-group person-group-type="author"><name><surname>Matsen</surname><given-names>E</given-names></name></person-group>          <year iso-8601-date="2022">2022</year>          <data-title>Differences in Spike epitope targeting between antibodies elicited by SARS-CoV-2 mRNA vaccines versus infection</data-title>          <version designator="swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86">swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86</version>          <source>Software Heritage</source>          <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:e087c50fd0a4f182f15ee6fe5b014058ceb2c8e3;origin=https://github.com/matsengrp/phage-dms-vacc-analysis;visit=swh:1:snp:c9917cbeef868e0898ef8a51b3349f4a2fd310e7;anchor=swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:e087c50fd0a4f182f15ee6fe5b014058ceb2c8e3;origin=https://github.com/matsengrp/phage-dms-vacc-analysis;visit=swh:1:snp:c9917cbeef868e0898ef8a51b3349f4a2fd310e7;anchor=swh:1:rev:d4c770ad49ed2f8ab31e499265dd02273cff6f86</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35072601">35072601</a> <b>(1)</b> ResFinder &#x2013; an open online resource for identification of antimicrobial resistance genes in next-generation sequencing data and prediction of phenotypes from genotypes<ul><li> SnakeMake, <b>Nextflow</b>, Cromwell, etc.).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35042790">35042790</a> <b>(1)</b> Rapid radiation in a highly diverse marine environment<ul><li> Data analysis was managed using <b>nextflow</b> (<xref rid="r71" ref-type="bibr"><u>71</u></xref>).<li> [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35039463">35039463</a> <b>(1)</b> CD161 expression and regulation defines rapidly responding effector CD4+ T cells associated with improved survival in HPV16-associated tumors<ul><li> A <b>Nextflow</b> pipeline<xref rid="R27" ref-type="bibr"><u>27</u></xref> to reproduce the analyses is publicly available from GitHub (<ext-link xlink:href="https://github.com/icbi-lab/duurland2021_paper" ext-link-type="uri">https://github.com/icbi-lab/duurland2021_paper</ext-link>).<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35050209">35050209</a> <b>(4)</b> A Checklist for Reproducible Computational Analysis in Clinical Metabolomics Research<ul><li> Galaxy and <b>Nextflow</b> are two workflow automation platforms that were used in metabolomics field.<li> <b>Nextflow</b> is a workflow automation platform written with Groovy programming language [<xref rid="B120-metabolites-12-00087" ref-type="bibr"><u>120</u></xref>].<li> <b>Nextflow</b> also supports several workflow schedulers, making it suitable for high-performance computing and large-scale data analysis.<li> However, <b>Nextflow</b> is not friendly to a scientist with few programming experiences.<li> [120](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35055061">35055061</a> <b>(1)</b> Transcriptome Analysis of an <italic toggle="yes">Aedes albopictus</italic> Cell Line Single- and Dual-Infected with Lammi Virus and WNV<ul><li>The raw transcriptomic FASTQ files were analyzed with bioinformatic <b>Nextflow</b> pipeline nf-core/rnaseq (v3.0) [<xref rid="B51-ijms-23-00875" ref-type="bibr"><u>51</u></xref>].<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35199087">35199087</a> <b>(1)</b> Inverting the model of genomics data sharing with the NHGRI Genomic Data Science Analysis, Visualization, and Informatics Lab-space<ul><li> Dockstore currently supports 4 workflow languages: the WDL, Common Workflow Language (CWL), <b>Nextflow</b>, and Galaxy Workflows (GWs).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35025877">35025877</a> <b>(2)</b> A short plus long-amplicon based sequencing approach improves genomic coverage and variant detection in the SARS-CoV-2 genome<ul><li> To facilitate reproducibility, analysis samples were processed using the publicly available nf-core/viralrecon pipeline version 1.1.0 implemented in <b>Nextflow</b> 20.01.0 using Singularity 3.3.0 (10.5281/zenodo.3901628) [<xref rid="pone.0261014.ref018" ref-type="bibr"><u>18</u></xref>&#x2013;<xref rid="pone.0261014.ref020" ref-type="bibr"><u>20</u></xref>].<li> Pipeline automation was done by creating <b>Nextflow</b> workflows (v20.01.0).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35022441">35022441</a> <b>(1)</b> The formation of avian montane diversity across barriers and along elevational gradients<ul><li>All genomes were assembled de novo using Supernova v2.0.1<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup> as incorporated in the <b>nextflow</b>-core Neutronstar workflow<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>.<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28381613">28381613</a>) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35021077">35021077</a> <b>(2)</b> Stride-level analysis of mouse open field behavior using deep-learning-based pose estimation<ul><li> The GWAS can be reproduced using the command:<preformat position="float" xml:space="preserve">export G=<ext-link xlink:href="https://raw.githubusercontent.com/TheJacksonLaboratory/mousegwas/" ext-link-type="uri">https://raw.githubusercontent.com/TheJacksonLaboratory/mousegwas/</ext-link> master <b>nextflow</b> run TheJacksonLaboratory/mousegwas -r gait \ &#x2013;yaml $G/example/gait_nowild_withBL.yaml \&#x2013;shufyaml $G/example/gait_shuffle_withBL.yaml \&#x2013;input $G/example/gait_paper_strain_survey_2019_11_12.csv \ &#x2013;outdir gait_output_withBL &#x2013;clusters 1 \&#x2013;addpostp &#x201C;&#x2013;colorgroup &#x2013;meanvariance &#x2013;set3 &#x2013;minherit 0.25 \ &#x2013;loddrop 1.5&#x201D; &#x2013;addheatmap &#x201C;&#x2013;meanvariance -p 0.1&#x201D;<b>nextflow</b> run TheJacksonLaboratory/mousegwas -r gait \&#x2013;yaml $G/example/gait_nowild_noBL.yaml \&#x2013;shufyaml $G/example/gait_shuffle_noBL.yaml \&#x2013;input $G/example/gait_paper_strain_survey_2019_11_12.csv \ &#x2013;outdir gait_output_noBL &#x2013;clusters 1 \&#x2013;addpostp &#x201C;&#x2013;colorgroup &#x2013;meanvariance &#x2013;set3 &#x2013;minherit 0.25 \ &#x2013;loddrop 1.5&#x201D; &#x2013;addheatmap &#x201C;&#x2013;meanvariance -p 0.1&#x201D;.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35007277">35007277</a> <b>(1)</b> Population-level genome-wide STR discovery and validation for population structure and genetic diversity assessment of <italic toggle="yes">Plasmodium</italic> species<ul><li>SNPs and deletions/insertions (Indels) were called using the standard best practice from Genome Analysis Toolkit (GATK) version 4.0.12.0 implemented in <b>nextflow</b> [<xref rid="pgen.1009604.ref020" ref-type="bibr"><u>20</u></xref>,<xref rid="pgen.1009604.ref068" ref-type="bibr"><u>68</u></xref>] (<ext-link xlink:href="https://www.google.com/url?q=https://github.com/gatk-workflows/gatk4-germline-snps-indels&amp;sa=D&amp;source=editors&amp;ust=1615116994818000&amp;usg=AOvVaw0zjNFN0dsvX08nlFEBLzPx" ext-link-type="uri">https://github.com/gatk-workflows/gatk4-germline-snps-indels</ext-link>).<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20644199">20644199</a>) [68](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31249686">31249686</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35042229">35042229</a> <b>(1)</b> Rapid epidemic expansion of the SARS-CoV-2 Omicron variant in southern Africa<ul><li> We also used the wf_artic (ARTIC SARS-CoV-2) pipeline as built using the <b>Nextflow</b> workflow framework<sup><xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup>.<li> [56](Wright, C. &amp; Parker, M. epi2me-labs/wf-artic: ARTIC SARS-CoV-2 workflow and reporting (GitHub); <ext-link ext-link-type="uri" xlink:href="https://github.com/epi2me-labs/wf-artic#readme">https://github.com/epi2me-labs/wf-artic#readme</ext-link> (2021).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35018222">35018222</a> <b>(1)</b> First detection of SARS-CoV-2 variant B.1.1.529 (Omicron) in Ecuador<ul><li> The genome was processed using the Artic bioinformatic pipeline, assembled using two methodologies: 1) Manual: [rampart]-[contigs]-[bwa-mem]-[samtools]-[mosdepth]-[zcat]-[bedtools]-[bcftools]-[bgzip]-[tabix]-[consensus]-[renameHeader] [<xref rid="bib6" ref-type="bibr"><u>6</u></xref>] and 2) Automated: poreCov-An, through <b>Nextflow</b> and either Docker [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>].<li> [6](          <person-group person-group-type="author">            <collab>Artic Network</collab>          </person-group>          <article-title>SARS-Cov-2. SARS-CoV-2 2021:1</article-title>          <ext-link ext-link-type="uri" xlink:href="https://artic.network/ncov-2019" id="intref0040">https://artic.network/ncov-2019</ext-link>        ) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34394197">34394197</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35111923">35111923</a> <b>(1)</b> Improving bioinformatics software quality through incorporation of software engineering practices<ul><li> EDAM (<xref rid="ref-43" ref-type="bibr"><u>Ison et al., 2013</u></xref>)), workflow systems including a great number in life sciences ranging from analytic platforms like KNIME (<xref rid="ref-28" ref-type="bibr"><u>Fillbrunn et al., 2017</u></xref>) to Jupyter Notebooks (<xref rid="ref-54" ref-type="bibr"><u>Kluyver et al., 2016</u></xref>), command-line workflow tools like <b>Nextflow</b> (<xref rid="ref-24" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), and dependency systems like SnakeMake (<xref rid="ref-57" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) have all aided the bioinformatics community.<li> [Ison et al., 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23479348">23479348</a>) [Fillbrunn et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28757290">28757290</a>) [Kluyver et al., 2016](          <person-group person-group-type="author"><name><surname>Kluyver</surname><given-names>T</given-names></name><name><surname>Ragan-Kelley</surname><given-names>B</given-names></name><name><surname>P&#xE9;rez</surname><given-names>F</given-names></name><name><surname>Granger</surname><given-names>BE</given-names></name><name><surname>Bussonnier</surname><given-names>M</given-names></name><name><surname>Frederic</surname><given-names>J</given-names></name><name><surname>Kelley</surname><given-names>K</given-names></name><name><surname>Hamrick</surname><given-names>JB</given-names></name><name><surname>Grout</surname><given-names>J</given-names></name><name><surname>Corlay</surname><given-names>S</given-names></name><name><surname>Ivanov</surname><given-names>P</given-names></name></person-group>          <article-title>Jupyter Notebooks-a publishing format for reproducible computational workflows</article-title>          <year iso-8601-date="2016">2016</year>          <conf-name>20th International Conference on Electronic Publishing</conf-name>          <volume>2016</volume>          <fpage>87</fpage>          <lpage>90</lpage>        ) [Di Tommaso et al., 2017](          <person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year iso-8601-date="2017">2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34890118">34890118</a> <b>(1)</b> Safety study of Rift Valley Fever human vaccine candidate (DDVax) in mosquitoes<ul><li> These tools were run as part of a reproducible <b>Nextflow</b> pipeline, available at <ext-link xlink:href="https://github.com/stenglein-lab/viral_variant_caller/releases/tag/DDVax_paper_release" ext-link-type="uri">https://github.com/stenglein&#x2010;lab/viral_variant_caller/releases/tag/DDVax_paper_release</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34965197">34965197</a> <b>(9)</b> sRNARFTarget: a fast machine-learning-based approach for transcriptome-wide sRNA target prediction<ul><li> To get the sRNA and mRNA sequences, we wrote two <b>Nextflow</b> (version 0.32.0) [<xref rid="cit0030" ref-type="bibr"><u>30</u></xref>] pipelines.<li> All <b>Nextflow</b> pipelines and training/validation data used are available at <ext-link xlink:href="https://github.com/BioinformaticsLabAtMUN/sRNARFTarget" ext-link-type="uri">https://github.com/BioinformaticsLabAtMUN/sRNARFTarget</ext-link>.<li> To obtain <italic toggle="yes">k</italic>-mer frequency and then <italic toggle="yes">k</italic>-mer frequency difference, we ran another <b>Nextflow</b> pipeline using scikit-bio (version 0.5.5) [<xref rid="cit0039" ref-type="bibr"><u>39</u></xref>] in Python (version 3.7.4).<li> We selected the model with the highest AUROC, and saved this model to be used by the <b>Nextflow</b> pipeline implementing sRNARFTarget.<li>sRNARFTarget <b>nextflow</b> pipeline.<li>We wrote a <b>Nextflow</b> pipeline that uses our best model for sRNA target prediction.<li> coli</italic> using our <b>Nextflow</b> pipeline as described above.<li> We then obtained the sequences for these mRNAs using our <b>Nextflow</b> pipeline.<li> To obtain a total execution time for IntaRNA, we created a <b>Nextflow</b> pipeline to run IntaRNA&#x2019;s two steps: 1) getting the interaction energy and 2) calculating the p-values for the interaction energy.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [39](<article-title>scikit-bio</article-title>. [<date-in-citation>cited 2020 <month>Jun</month><day>04</day></date-in-citation>]. <comment>Available from:</comment><ext-link xlink:href="http://scikit-bio.org" ext-link-type="uri">http://scikit-bio.org</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34965454">34965454</a> <b>(1)</b> What&#x2019;s new and what&#x2019;s next in diffusion MRI preprocessing<ul><li> (2020)</xref> mention the complexity of adding and maintaining packages in Nipype as a challenge for creating dMRI preprocessing pipelines, and instead opt for <b>Nextflow</b> (<xref rid="R79" ref-type="bibr"><u>DI Tommaso et al., 2017</u></xref>).<li> [DI Tommaso et al., 2017](<name><surname>DI Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <name><surname>Palumbo</surname><given-names>E</given-names></name>, <name><surname>Notredame</surname><given-names>C</given-names></name>, <year>2017</year>. <source>Nextflow enables reproducible computational workflows</source>. <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35059606">35059606</a> <b>(2)</b> Cantaloupe melon genome reveals 3D chromatin features and structural relationship with the ancestral cucurbitaceae karyotype<ul><li>Protein-coding genes were annotated through integration of different databases (<ext-link ext-link-type="uri" xlink:href="https://lipm-gitlab.toulouse.inra.fr/LIPM-BIOINFO/<b>nextflow</b>-functionnalannotation%20%5f%20commit%2017f5efb9dcf7c67f74769cf6a1930cf9ca986e47" id="intref0230">https://lipm-gitlab.toulouse.inra.fr/LIPM-BIOINFO/<b>nextflow</b>-functionnalannotation&#x2013; commit 17f5efb9dcf7c67f74769cf6a1930cf9ca986e47</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35052387">35052387</a> <b>(1)</b> A Nonsense Variant in <italic toggle="yes">CCDC65</italic> Gene Causes Respiratory Failure Associated with Increased Lamb Mortality in French Lacaune Dairy Sheep<ul><li>Reads mapping and variants calling were performed using <b>Nextflow</b> v20.10.0 and Sarek pipeline v2.6.1 [<xref rid="B29-genes-13-00045" ref-type="bibr"><u>29</u></xref>].<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34936673">34936673</a> <b>(1)</b> NGS read classification using AI<ul><li>In order to make the benchmark reproducible, we have implemented it as a <b>nextflow</b> [<xref rid="pone.0261548.ref047" ref-type="bibr"><u>47</u></xref>] pipeline (see <ext-link xlink:href="https://gitlab.com/dabrowskiw/cdsfinderbenchmark" ext-link-type="uri">https://gitlab.com/dabrowskiw/cdsfinderbenchmark</ext-link>).<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34928370">34928370</a> <b>(7)</b> GIP: an open-source computational pipeline for mapping genomic instability from protists to cancer cells<ul><li> GIP is implemented in <b>Nextflow</b> (<xref rid="B30" ref-type="bibr"><u>30</u></xref>), a workflow language that allows to execute GIP seamlessly in local workstation, on an HPC or remotely in the cloud.<li> <b>Nextflow</b> option was used to render HTML timelines for all GIP executed processes.<li> <b>Nextflow</b> option was used to summarize GIP computational resources.<li> A detailed description of each of the two options is available from <b>Nextflow</b> documentation at <ext-link xlink:href="https://www.<b>nextflow</b>.io/docs/latest/tracing.html#" ext-link-type="uri">https://www.<b>nextflow</b>.io/docs/latest/tracing.html#</ext-link>.<li> GIP consists of three files: the <b>Nextflow</b> pipeline code, the configuration and the Singularity container files.<li>We used the <b>Nextflow</b> built-in options to render the process execution time and the computational resources of each task executed by GIP for each sample considered in this study.<li> Because of its <b>Nextflow</b> implementation GIP can be executed on a local machine, on cluster resource manager or the cloud.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35574063">35574063</a> <b>(27)</b> nf-rnaSeqCount: A <b>Nextflow</b> pipeline for obtaining raw read counts from RNA-seq data<ul><li> This paper presents nf-rnaSeqCount, a portable, reproducible and scalable <b>Nextflow</b> pipeline that addresses the first two steps of RNA-seq data analyses, i.e., (1) identification of genes from RNA-seq data and (2) quantifying their abundance.<li><b>Nextflow</b> (<xref rid="R6" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and Singularity (<xref rid="R17" ref-type="bibr"><u>Kurtzer et al., 2017</u></xref>) were used to implement the pipeline into a portable and reproducible pipeline.<li> <b>Nextflow</b> is a Groovy-based domain-specific language (DSL) specifically designed for bioinformaticists with strong programming knowledge to solve many of the challenges of the inability to reproduce data analysis.<li> The advantages of <b>Nextflow</b> as a workflow management system are that it can handle input and output as channels between processes and reduce the need of having to create intermediate directories to store intermediate results.<li><b>Nextflow</b> also has a number of features that promote pipeline reproducibility and portability.<li>The nf-rnaSeqCount pipeline depends on <b>Nextflow</b> and Singularity to run.<li> The nf-rnaSeqCount pipeline can be obtained using the following command:</p>        <preformat position="float" xml:space="preserve"><b>nextflow</b> pull phelelani/nf-rnaSeqCount.<li>The help menu for the pipeline can be accessed with the following command:</p>        <preformat position="float" xml:space="preserve"><b>nextflow</b> run nf-rnaSeqCount --<styled-content style="color:#0000FF">help</styled-content>.<li> This step is crucial as all processes in the pipeline depend on the applications that are packaged in these containers:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Download Singularity containers</styled-content><b>nextflow</b> run nf-rnaSeqCount --mode getContainers.<li> Before indexing, the location for the reference genome (FASTA), annotation file (GTF), input FASTQ files and output directory can be provided in a configuration file (as in <xref rid="F2" ref-type="fig"><u>Figure 2</u></xref>), and passed to the <b>Nextflow</b> command using the -c option.<li> To index the reference genome using STAR and Bowtie, the following commands can be used:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Generate STAR and Bowtie2 indexes</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode prep.Indexes.<li> To assess the quality of the RNA-seq reads, the following command can be executed:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Perform QC on reads</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode run.ReadQC.<li> The &#x2013;trim option can be used to pass Trimmomatic arguments to the pipeline:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Trim low quality reads</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode run.ReadTrimming --trim &#x2019;TRAILING:28 MINLEN:40&#x2019;.<li> To align the reads to the reference genome and quantify the abundance of the genomic features identified, the following commands can be used:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Align reads to reference genome</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode run.ReadAlignment.<li> To align the reads to the reference genome and quantify the abundance of the genomic features identified, the following commands can be used:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Align reads to reference genome</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode run.ReadAlignment</preformat>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Quantify the abundance of identified features</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode run.ReadCounting.<li> To obtain the summary statistics of the workflow, the following command can be used:</p>          <preformat position="float" xml:space="preserve"><styled-content style="color:#FF0000">## Obtain summary statistics from all tools</styled-content><b>nextflow</b> run nf-rnaSeqCount -c main.conf --mode MultiQC.<li>For AWS execution of the pipeline, the <b>Nextflow</b> supported Amazon Machine Image (AMI), ami-4b7daa32, was used to deploy an Amazon Elastic Block Store (EBS) of 1000GB using the Elastic Compute Cloud (EC2), m4.10xlarge, with 40 virtual CPUs and 160 GB of memory.<li> All AWS analyses were performed on the European (Ireland) region since the <b>Nextflow</b> AMI was only available for this region.<li>The nf-rnaSeqCount pipeline has been successfully implemented in <b>Nextflow</b> and Singularity, and can be executed on any UNIX-based OS with <b>Nextflow</b> and Singularity installed.<li> In addition to running the pipeline locally, the nf-rnaSeqCount pipeline also supports the PBS and SLURM job schedulers on HPCs, and this information can be passed to the -profile option of <b>Nextflow</b> when executing the pipeline.<li> However, nf-rnaSeqCount has added advantages over Rsubread: (1) <bold>Parallelisation:</bold> in addition to applications being multi-threaded (with the exception of htseq-count) within the workflow, the implementation of nf-rnaSeqCount on <b>Nextflow</b> also allows processes to be run across multiple nodes on HPC, which drastically improves performance when working with large datasets; (2) <bold>Installation:</bold> there is no need for installation of packages and dependencies for the nf-rnaSeqCount pipeline, e.g., only <b>Nextflow</b> and Singularity are required; and (3) <bold>Usability:</bold> there is no need for writing tedious scripts for performing RNA-seq analysis with nf-rnaSeqCount, i.e., all inputs, outputs and parameters can be put into a config file which will be used to execute each step of the analysis.<li> nf-rnaSeqCount is designed on <b>Nextflow</b> and all its application dependencies are packaged in Singularity containers.<li> This makes it possible to run the pipeline on any machine, from desktop to HPC, with both <b>Nextflow</b> and Singularity installed.<li> <b>Nextflow</b> supports a wide variety of job schedulers, and the nf-rnaSeqCount pipeline comes packaged with support for PBS and SLURM schedulers.<li> Advanced users can add their own scheduler support using the <monospace><b>nextflow</b>.config</monospace> file.<li> [Di Tommaso et al., 2017](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <name><surname>Palumbo</surname><given-names>E</given-names></name>, &amp; <name><surname>Notredame</surname><given-names>C</given-names></name> (<year>2017</year>). <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nature Biotechnology</source>, <volume>35</volume>(<issue>4</issue>), <fpage>316</fpage>&#x2013;<lpage>319</lpage>. <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [Kurtzer et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34919405">34919405</a> <b>(1)</b> Skyline Batch:An Intuitive User Interface for BatchProcessing with Skyline<ul><li>Generalized workflow managers such as Cromwell, Snakemake,and<b>Nextflow</b> have been created to improve research reproducibility.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34904638">34904638</a> <b>(3)</b> Generation of ENSEMBL-based proteogenomics databases boosts the identification of non-canonical peptides<ul><li>To make progress in high throughput proteogenomics analysis, we present a <italic toggle="yes">Python</italic> application integrated into a <italic toggle="yes"><b>Nextflow</b></italic> workflow to facilitate the generation of proteogenomics databases from sample-specific and public resources under varying conditions (e.g.<li>To enable the generation of ENSEMBL-based proteogenomic databases, we have also built the Proteomics-Genomics DataBase (<italic toggle="yes">pgdb</italic>&#x2014;<ext-link xlink:href="https://nf-co.re/pgdb" ext-link-type="uri">https://nf-co.re/pgdb</ext-link>) workflow in <italic toggle="yes"><b>Nextflow</b> using bioconda and BioContainers</italic>.<li> The <italic toggle="yes">pgdb</italic> is the first <italic toggle="yes"><b>Nextflow</b></italic> workflow for proteogenomics database generation and its development within the nf-core community will ensure its stability, continued development and community support.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34970617">34970617</a> <b>(1)</b> Performance and Agreement Between WGS Variant Calling Pipelines Used for Bovine Tuberculosis Control: Toward International Standardization<ul><li><b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34893601">34893601</a> <b>(3)</b> RNA modifications detection by comparative Nanopore direct RNA sequencing<ul><li> We have developed an automated <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/tleonardi/nanocompore_pipeline">https://github.com/tleonardi/nanocompore_pipeline</ext-link>) that automatically runs the entire analysis from preprocessing of raw Nanopore data (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1A</u></xref>), to modified-base detection with Nanocompore (<ext-link ext-link-type="uri" xlink:href="https://github.com/tleonardi/nanocompore">https://github.com/tleonardi/nanocompore</ext-link>, Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1B</u></xref>).<li> Finally, we implemented analysis pipelines in the <b>Nextflow</b> and Snakemake&#xA0;Domain Specific Languages, allowing automatic execution of all processing steps, from raw data up to the execution of Nanocompore and other RNA modification tools, thus greatly simplifying the bioinformatics work.<li>All the datasets were preprocessed using an automated analysis <b>NextFlow</b> pipeline, before running Nanocompore (<ext-link ext-link-type="uri" xlink:href="https://github.com/tleonardi/nanocompore_pipeline">https://github.com/tleonardi/nanocompore_pipeline</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34950899">34950899</a> <b>(1)</b> Common-sense approaches to sharing tabular data alongside publication<ul><li> Depending on whether the workflow is GUI driven (e.g., Galaxy) or text driven (e.g., <b>Nextflow</b>, Snakemake), the workflow itself can be checked into the version control repository.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34890448">34890448</a> <b>(1)</b> Linking big biomedical datasets to modular analysis with Portable Encapsulated Projects<ul><li> These pipelines are increasingly written using pipeline frameworks, leading to now dozens of such frameworks that simplify developing reusable computational pipelines [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], such as the common workflow language [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], Snakemake [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], Galaxy [<xref rid="bib13" ref-type="bibr"><u>13</u></xref>], and <b>Nextflow</b> [<xref rid="bib14" ref-type="bibr"><u>14</u></xref>].<li> [10](<person-group person-group-type="author"><string-name><surname>Leipzig</surname><given-names>J</given-names></string-name></person-group>. <article-title>A review of bioinformatic pipeline frameworks</article-title>. <source>Brief Bioinform</source>. <year>2016</year>;doi:<pub-id pub-id-type="doi">10.1093/bib/bbw020</pub-id>.) [11](<person-group person-group-type="author"><string-name><surname>Amstutz</surname><given-names>P</given-names></string-name>, <string-name><surname>Crusoe</surname><given-names>MR</given-names></string-name>, <string-name><surname>Tijani&#xE9;</surname><given-names>N</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Common Workflow Language, v1.0</article-title>. <source>figshare</source>. <year>2016</year>;<comment>doi:10.6084/m9.figshare.3115156.v2</comment>.) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34909774">34909774</a> <b>(1)</b> Macaque-human differences in SARS-CoV-2 Spike antibody response elicited by vaccination or infection<ul><li> The phip-flow tools perform read alignment using Bowtie2 [<xref rid="R97" ref-type="bibr"><u>97</u></xref>] in a <b>Nextflow</b> [<xref rid="R98" ref-type="bibr"><u>98</u></xref>] pipeline script.<li> [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [98](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850832">34850832</a> <b>(1)</b> High-Resolution Genomic Profiling of Carbapenem-Resistant <italic toggle="yes">Klebsiella pneumoniae</italic> Isolates: A Multicentric Retrospective Indian Study<ul><li>Bioinformatic analysis was performed using <b>Nextflow</b> pipelines developed as a part of Genomic Surveillance of Antimicrobial Resistance-AMR as detailed in <ext-link xlink:href="http://www.protocols.io" ext-link-type="uri">www.protocols.io</ext-link> [<xref rid="CIT0014" ref-type="bibr"><u>14</u></xref>].<li> [14](<person-group person-group-type="author"><collab>GHRU (Genomic Surveillance of Antimicrobial Resistance)</collab></person-group>. <comment>Retrospective 1 Bioinformatics Methods V.4, 2020</comment>. <ext-link xlink:href="https://gitlab.com/antunderwood/bactinspector" ext-link-type="uri">https://www.protocols.io/view/ghru-genomic-surveillance-of-antimicrobial-resista-bpn6mmhe</ext-link>. <comment>Accessed 13 May 2021</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35028332">35028332</a> <b>(1)</b> Variation in Severe Acute Respiratory Syndrome Coronavirus 2 Bioaerosol Production in Exhaled Breath<ul><li>We used the nfcore/viralrecon bioinformatic pipeline containerized on <b>Nextflow</b> to perform variant calling and generate consensus sequences from raw reads [<xref rid="CIT0007" ref-type="bibr"><u>7</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34917052">34917052</a> <b>(1)</b> Human and Animal RNA Virus Diversity Detected by Metagenomics in Cameroonian Clams<ul><li>Bioinformatic analysis was performed using <b>nextflow</b> to monitor the run on the computing cluster.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34900230">34900230</a> <b>(18)</b> snpQT: flexible, reproducible, and comprehensive quality control and imputation of genomic data<ul><li> Software architecture tools such as <b>nextflow</b> and BioContainers can address these issues<sup><xref rid="ref-13" ref-type="bibr"><u>13</u></xref></sup> and have been proposed as automated solutions<sup><xref rid="ref-14" ref-type="bibr"><u>14</u></xref></sup>; however restrictions exist in terms of limited and relatively rigid QC analysis, lacking such steps as imputation, limited variety of threshold choice and plot outputs, and the requirement for users to have extensive knowledge of the software in order to tailor their analysis.<li>Outline of the snpQT architecture, which includes nine core workflows (A&#x2013;I) that are implemented using <b>nextflow</b>.<li>snpQT was developed as a set of nine core workflow components implemented with the <b>nextflow</b> workflow management system<sup><xref rid="ref-13" ref-type="bibr"><u>13</u></xref></sup>.<li> Combining independent containerised modules into workflows, and providing multiple workflow combinations, using the <b>nextflow</b> architecture, enables snpQT to be a reproducible and uniquely versatile tool for the analysis of human variant genomic data.<li> In addition, <b>nextflow</b> enables caching at continuous checkpoints, so users can alter thresholds without needing to rerun earlier parts of the analysis.<li>snpQT is implemented in <b>nextflow</b>, R and Unix command line utilities.<li> The minimum software requirements to run snpQT are Java 8, <b>nextflow</b> v21.04.3, and a POSIX-compatible operating system (tested on CentOS 8).<li>Before downloading and running snpQT, depending on their needs, the user should have downloaded <b>nextflow</b> (v21.04.3 or later) and any of the following software including Anaconda, Singularity, Docker or Environment Modules.<li> <b>Nextflow</b> stores these in the<monospace>snpQT/work/</monospace> directory by default.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --vcf data/data/toy.vcf.gz \  --fam data/data/toy.fam \  --convert_build \  --input_build 37 \  --output_build 38 \  --results convert_build_toy/</styled-content>.<li><b>Nextflow</b> options:.<li>snpQT&#x2019;s main Quality Control analysis is divided into two distinct <b>nextflow</b> workflows: Sample and Variant QC.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --bed data/toy.bed \  --bim data/toy.bim \  --fam data/toy.fam \  --qc \  --results results_toy/ \  --sexcheck false</styled-content>.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --bed data/toy.bed \  --bim data/toy.bim \  --fam data/toy.fam \  --qc \  --pop_strat \  --results results_toy/ \  --sexcheck false</styled-content>.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --bed data/toy.bed \  --bim data/toy.bim \  --fam data/toy.fam \  --qc \  --pop_strat \  --gwas \  --results results_toy/ \  --sexcheck false</styled-content>.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --bed data/toy.bed \  --bim data/toy.bim \  --fam data/toy.fam \  --qc \  --pre_impute \  --results results_toy/ \  --sexcheck false</styled-content>.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --bed data/toy.bed \  --bim data/toy.bim \  --fam data/toy.fam \  --qc \  --pop_strat \  --impute \  --gwas \  --results results_toy/ \  --sexcheck false</styled-content>.<li><preformat preformat-type="computer code" position="float" xml:space="preserve"><styled-content style="font-size:15px;color:#230000;"><b>nextflow</b> run main.nf \  -profile standard,singularity \  -resume \  -params-file parameters.yaml \  --vcf merged_imputed.vcf.gz \  --fam results_toy/qc/bfiles/E11.fam \  --post_impute \  --qc_false \  --results results_postImpute/</styled-content>.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](<person-group person-group-type="author"><name><surname>Song</surname><given-names>Z</given-names></name><name><surname>Gurinovich</surname><given-names>A</given-names></name><name><surname>Federico</surname><given-names>A</given-names></name><etal/></person-group>:<article-title>nf-gwas-pipeline: A Nextflow Genome-Wide Association Study Pipeline.</article-title><source><italic toggle="yes">J Open Source Softw.</italic></source><year>2021</year>;<volume>6</volume>(<issue>59</issue>):<fpage>2957</fpage>.<pub-id pub-id-type="doi">10.21105/joss.02957</pub-id>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a> <b>(1)</b> snpQT: flexible, reproducible, and comprehensive quality control and imputation of genomic data<ul><li> Furthermore, to give an indication of performance for datasets of more relevant size, we have now used <b>nextflow</b>&#x2019;s built-in reporting functionality to generate performance metrics for two cohort sizes from the ALS dataset, and added this to the manuscript text:.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a> <b>(1)</b> snpQT: flexible, reproducible, and comprehensive quality control and imputation of genomic data<ul><li> The second type is responsible to set up successfully all the required libraries and software, assuming that the user has not installed anything except <b>nextflow</b> and the profile environment software of choice (e.g.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34843238">34843238</a> <b>(1)</b> Automated Determination of Nuclear Magnetic ResonanceChemical Shift Perturbations in Ligand Screening Experiments: ThePICASSO Web Server<ul><li> There are two main lines for future development: (i) integrationwith automated procedures for HSQC spectra processing and peak picking,for example, using NMRpipe<sup><xref ref-type="bibr" rid="ref36"><u>36</u></xref></sup> as the mainsoftware package and <b>Nextflow</b><sup><xref ref-type="bibr" rid="ref37"><u>37</u></xref></sup> as a lightweightworkflow management system and (ii) incorporating 3D structural information.Our tool is entirely written in Python, which permits easy integrationwith other packages in the same language, such as nmrglue,<sup><xref ref-type="bibr" rid="ref38"><u>38</u></xref></sup> a Python-based module for working with NMR data.Currently, users must produce the input peak lists for PICASSO independently;to facilitate this task, our tool requires input in the csv format,which can be produced by most software used to handle NMR data.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=8520220">8520220</a>) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23456039">23456039</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34824391">34824391</a> <b>(1)</b> Coordination between cell proliferation and apoptosis after DNA damage in <italic>Drosophila</italic><ul><li>ATAC-seq data analysis was performed using the nf-core/atacseq pipeline (v1.2.1), which runs <b>Nextflow</b> (v20.10.0), for quality controls, read alignment against Dm6 assembly using BWA-MEM (v0.7.17), filtering for blacklisted regions, data visualization, peak calling using MACS2 (v2.2.7.1), read count using featureCounts (v2.0.1), and differential accessibility analysis using the DESeq2 R library.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34943818">34943818</a> <b>(1)</b> Ascorbic Acid/Retinol and/or Inflammatory Stimuli&#x2019;s Effect on Proliferation/Differentiation Properties and Transcriptomics of Gingival Stem/Progenitor Cells<ul><li> Raw FastQ files were aligned, quality controlled, and transformed into read counts, using <b>Nextflow</b> nfcore/RNAseq pipeline <uri xlink:href="https://nf-co.re/rnaseq">https://nf-co.re/rnaseq</uri> (accessed on 22 February 2021) [<xref rid="B27-cells-10-03310" ref-type="bibr"><u>27</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850831">34850831</a> <b>(1)</b> Train-the-Trainer as an Effective Approach to Building Global Networks of Experts in Genomic Surveillance of Antimicrobial Resistance (AMR)<ul><li> Pipelines: <b>Nextflow</b> [<xref rid="CIT0023" ref-type="bibr"><u>23</u></xref>] <break/>&#x2022; Phylogeny: RAxML [<xref rid="CIT0024" ref-type="bibr"><u>24</u></xref>] <break/>&#x2022; Genome Assembly: SPAdes [<xref rid="CIT0025" ref-type="bibr"><u>25</u></xref>] <break/>&#x2022; QC tools: FastQC [<xref rid="CIT0026" ref-type="bibr"><u>26</u></xref>], Quast [<xref rid="CIT0027" ref-type="bibr"><u>27</u></xref>], Bactinspector [<xref rid="CIT0028" ref-type="bibr"><u>28</u></xref>], Qualifyr [<xref rid="CIT0029" ref-type="bibr"><u>29</u></xref>] <break/>&#x2022; Typing tools: PubMed MLST (www.pubmlst.org) <break/>&#x2022; The Training and Learning Toolkit.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31070718">31070718</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32559359">32559359</a>) [26](<collab>FastQC.</collab> Babraham Bioinformatics. Available at: <ext-link xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" ext-link-type="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>. Accessed <comment>16 June 2021</comment>.) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23422339">23422339</a>) [28](<person-group person-group-type="author"><string-name><surname>Underwood</surname>&#xA0;<given-names>A</given-names></string-name></person-group>. <comment>Bactinspector</comment>., Available at: <ext-link xlink:href="https://gitlab.com/antunderwood/bactinspector" ext-link-type="uri">https://gitlab.com/antunderwood/bactinspector</ext-link>. Accessed: <comment>16 June 2021</comment>.) [29](<person-group person-group-type="author"><string-name><surname>Underwood</surname>&#xA0;<given-names>A</given-names></string-name></person-group>. <comment>Qualifyr</comment>. Available at: <ext-link xlink:href="https://gitlab.com/cgps/qualifyr" ext-link-type="uri">https://gitlab.com/cgps/qualifyr</ext-link>. Accessed: <comment>16 June 2021</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850835">34850835</a> <b>(1)</b> Complexity of Genomic Epidemiology of Carbapenem-Resistant <italic toggle="yes">Klebsiella pneumoniae</italic> Isolates in Colombia Urges the Reinforcement of Whole Genome Sequencing-Based Surveillance Programs<ul><li>All genome sequence data were processed using versioned <b>Nextflow</b> workflows and associated Docker containers covering de novo assembly, mapping-based SNP phylogeny, MLST assignment, and AMR determinant detection.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850839">34850839</a> <b>(3)</b> Overcoming Data Bottlenecks in Genomic Pathogen Surveillance<ul><li> For the GHRU project we took both these approaches, using workflows written for the <b>Nextflow</b> workflow manager, and Docker or Singularity containers to analyze the raw sequence data and produce de novo assemblies, AMR predictions, the 7-locus multilocus sequence type, and mapping-based single-nucleotide polymorphism phylogenies (<xref rid="F1" ref-type="fig"><u>Figures 1</u></xref> and <xref rid="F2" ref-type="fig"><u>2</u></xref>, bottleneck &#x2463;; <xref rid="sup1" ref-type="supplementary-material"><u>Supplementary Table 1</u></xref>) [<xref rid="CIT0023" ref-type="bibr"><u>23&#x2013;25</u></xref>].<li> In each partner center, samples were processed on either high-specification laptops, or a workstation only requiring the installation of Ubuntu Linux, Java RE, <b>Nextflow</b>, and Docker, with either a simple command-line recipe or automated with ansible [<xref rid="CIT0029" ref-type="bibr"><u>29</u></xref>].<li> The workflow manager <b>Nextflow</b> that was used in the project already has an interface to launch pipelines in cloud computing environments [<xref rid="CIT0042" ref-type="bibr"><u>42</u></xref>].<li> [23&#x2013;25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [29](<collab>Center for Genomic Pathogen Surveillance.</collab>&#xA0;<comment>CGPS protocols</comment>. Available at: <ext-link xlink:href="https://www.pathogensurveillance.net/resources/protocols/" ext-link-type="uri">https://www.pathogensurveillance.net/resources/protocols/</ext-link>. Accessed <comment>15 December 2020</comment>.) [42](<collab>Nextflow Tower.</collab>&#xA0;<comment>Get started</comment>. Available at: <ext-link xlink:href="https://tower.nf/" ext-link-type="uri">https://tower.nf/</ext-link>. Accessed <comment>8 October 2020</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34823519">34823519</a> <b>(1)</b> Analysis of <italic>pir</italic> gene expression across the <italic>Plasmodium</italic> life cycle<ul><li> chabaudi</italic> RNAseq were processed through the rna-seq pipeline v1.4.2 from nf-core repository in the <b>Nextflow</b> pipeline software v20.07.1 [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>, <xref ref-type="bibr" rid="CR27"><u>27</u></xref>], using Trim Galore! v0.6.4 to trim sequences, HISAT2 v.2.1.0 as the aligner, and featurecounts v.1.6.4 as the gene count program [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>&#x2013;<xref ref-type="bibr" rid="CR30"><u>30</u></xref>].<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) [30](Krueger F. Trim Galore! 2015. <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34946031">34946031</a> <b>(1)</b> Identification of Bull Semen Microbiome by 16S Sequencing and Possible Relationships with Fertility<ul><li>The analysis of the 16S rRNA sequencing data was performed using <b>Nextflow</b> pipeline ampliseq v.1.1.2 (<uri xlink:href="https://github.com/nf-core/ampliseq">https://github.com/nf-core/ampliseq</uri>, accessed on 24 November 2021).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34824477">34824477</a> <b>(6)</b> MCMICRO: a scalable, modular image-processing pipeline for multiplexed tissue imaging<ul><li> Analogous challenges in genomics have been addressed using computational pipelines such as Seurat, Scanpy and Cumulus (see Supplementary Table <xref rid="MOESM1" ref-type="media"><u>2</u></xref> for references), platforms such as Galaxy<sup><xref ref-type="bibr" rid="CR9"><u>9</u></xref></sup> that make use of software containers (for example, Docker<sup><xref ref-type="bibr" rid="CR10"><u>10</u></xref></sup>) and formal domain-specific workflow languages such as <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li>In this paper, we describe MCMICRO (Multiple Choice MICROscopy), a scalable, modular and open-source image-processing pipeline that leverages Docker/Singularity containers<sup><xref ref-type="bibr" rid="CR10"><u>10</u></xref>,<xref ref-type="bibr" rid="CR12"><u>12</u></xref></sup> and is implemented in <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup> and Galaxy<sup><xref ref-type="bibr" rid="CR9"><u>9</u></xref></sup>.<li> The <b>Nextflow</b> implementation uses a plain-text configuration file to simplify addition, management and execution of modules and a command line interface; Galaxy uses Conda environments for package management and a graphical user interface (GUI).<li>To demonstrate the technology-agnostic capabilities of MCMICRO implementations in Galaxy and <b>Nextflow</b> (Supplementary Fig.<li> Our experience also suggests that new users can master either the <b>NextFlow</b> command line interface or the Galaxy GUI with a day of training.<li>MCMICRO was implemented in <b>Nextflow</b>, which was chosen for its natural integration with container technologies such as Docker and Singularity, its automatic provenance tracking and parallelization of image-processing tasks and its ability to specify module dependencies that may change at runtime<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [10](Merkel, D. Docker: lightweight LINUX containers for consistent development and deployment. <italic>Linux J.</italic><ext-link ext-link-type="uri" xlink:href="https://www.linuxjournal.com/content/docker-lightweight-linux-containers-consistent-development-and-deployment">https://www.linuxjournal.com/content/docker-lightweight-linux-containers-consistent-development-and-deployment</ext-link> (2014).) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](Merkel, D. Docker: lightweight LINUX containers for consistent development and deployment. <italic>Linux J.</italic><ext-link ext-link-type="uri" xlink:href="https://www.linuxjournal.com/content/docker-lightweight-linux-containers-consistent-development-and-deployment">https://www.linuxjournal.com/content/docker-lightweight-linux-containers-consistent-development-and-deployment</ext-link> (2014).) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34946814">34946814</a> <b>(3)</b> Comprehensive Analysis of Large-Scale Transcriptomes from Multiple Cancer Types<ul><li> The pipeline is based on <b>Nextflow</b> [<xref rid="B55-genes-12-01865" ref-type="bibr"><u>55</u></xref>], a reproducible workflow management system, and integrates eight different tools to extract different information from RNA-seq data in the first module (<xref rid="genes-12-01865-f001" ref-type="fig"><u>Figure 1</u></xref>B).<li> PipeOne was built based on Docker and <b>Nextflow</b> [<xref rid="B55-genes-12-01865" ref-type="bibr"><u>55</u></xref>], making installation and management of workflow easy.<li> By using the <b>Nextflow</b> management system, advanced users could modify PipeOne to adapt their own research purposes.<li> [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34815497">34815497</a> <b>(1)</b> Epitachophoresis is a novel versatile total nucleic acid extraction method<ul><li>We have used the publicly available <b>nextflow</b> pipeline to process the miRNA sequencing data.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35755950">35755950</a> <b>(1)</b> Computational cancer neoantigen prediction: current status and recent advances<ul><li> Workflow management systems such as <b>Nextflow</b>,<xref rid="bib152" ref-type="bibr"><sup>152</sup></xref> Snakemake,<xref rid="bib153" ref-type="bibr"><sup>153</sup></xref> Airflow (<ext-link ext-link-type="uri" xlink:href="https://airflow.apache.org/" id="intref0025">https://airflow.apache.org/</ext-link>), and CWL<xref rid="bib154" ref-type="bibr"><sup>154</sup></xref> allow the development of scalable and reproducible data analysis workflows.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34550806">34550806</a> <b>(5)</b> Whole-Genome Sequencing of SARS-CoV-2: Assessment of the Ion Torrent AmpliSeq Panel and Comparison with the Illumina MiSeq ARTIC Protocol<ul><li>For data generated on the Illumina MiSeq system, reads were processed by the ARTIC <b>Nextflow</b> pipeline (<ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf/tree/illumina" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf/tree/illumina</ext-link>).<li>The <b>Nextflow</b>-adapted ARTIC bioinformatics pipeline employed to process Illumina-generated reads in the cloud is freely available and is designed to be highly parallelized.<li> In addition to consensus sequences, the <b>Nextflow</b> pipeline provides quality-trimmed fastq files, primer-trimmed BAM alignment files, and VCF files with information on variants (SNVs and indels), including minor variants.<li> However, establishing the necessary accounts in the cloud, properly configuring VMs, and installing and running <b>Nextflow</b> software and the ARTIC pipeline require both information technology and bioinformatics expertise, which might be prohibitive for some laboratories.<li> Alternatively, the individual components that constitute the <b>Nextflow</b> ARTIC pipeline, such as TrimGalore, BWA, and iVAR, can be freely and locally installed.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34789306">34789306</a> <b>(1)</b> Promotech: a general tool for bacterial promoter recognition<ul><li>A <b>Nextflow</b> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] pipeline was designed to obtain the promoter (positive) sequences from the TSS coordinates by taking the genome FASTA file and the TSS coordinates as input and obtaining the promoter coordinates as 40 bp upstream from the TSS to the TSS using BEDTools&#x2019; slop command.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34788790">34788790</a> <b>(3)</b> nextNEOpi: a comprehensive pipeline for computational neoantigen prediction<ul><li>We implemented nextNEOpi in the <b>Nextflow</b> workflow language (<xref rid="btab759-B2" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li> It is implemented in <b>Nextflow</b> to ensure easy installation and usage, as well as high portability, scalability (see example computational times in <xref rid="sup1" ref-type="supplementary-material"><u>Supplementary Table S7</u></xref>), and reproducibility.<li> In the near future, we plan to extend nextNEOpi to other classes of noncanonical neoantigens and to introduce DSL 2 <b>Nextflow</b> syntax to facilitate the integration of additional features relevant for neoantigen prioritization.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34805989">34805989</a> <b>(3)</b> EpiDiverse Toolkit: a pipeline suite for the analysis of bisulfite sequencing data in ecological plant epigenetics<ul><li>The tools presented herein (Figure <xref rid="F1" ref-type="fig"><u>1</u></xref>) are implemented with <b>Nextflow</b> (<xref rid="B11" ref-type="bibr"><u>11</u></xref>), building on best-practice concepts outlined by nf-core (<xref rid="B12" ref-type="bibr"><u>12</u></xref>).<li> Each resource allocation is fully customisable under the <b>Nextflow</b> framework to suit integration under different systems and scheduling software.<li> Dependencies are as simple as installing <b>Nextflow</b> alongside one of either Bioconda (<xref rid="B13" ref-type="bibr"><u>13</u></xref>), Docker (<xref rid="B14" ref-type="bibr"><u>14</u></xref>)&#xA0;or Singularity (<xref rid="B15" ref-type="bibr"><u>15</u></xref>) on a POSIX compatible system, facilitating a high level of flexibility and reproducibility through the use of portable software containers and environments.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [14](<person-group person-group-type="author"><string-name><surname>Merkel</surname><given-names>D.</given-names></string-name></person-group><article-title>Docker: lightweight linux containers for consistent development and deployment</article-title>. <source>Linux journal</source>. <year>2014</year>; <volume>2014</volume>:<fpage>1</fpage>&#x2013;<lpage>5</lpage>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34762647">34762647</a> <b>(1)</b> Positive correlation between transcriptomic stemness and PI3K/AKT/mTOR signaling scores in breast cancer, and a counterintuitive relationship with <italic toggle="yes">PIK3CA</italic> genotype<ul><li>Raw read processing was performed with the <b>Nextflow</b> (version 20.07.1) nf-core RNAseq pipeline (version 1.4.2) [<xref rid="pgen.1009876.ref067" ref-type="bibr"><u>67</u></xref>], with Spliced Transcripts Aligment to a Reference (STAR) [<xref rid="pgen.1009876.ref068" ref-type="bibr"><u>68</u></xref>] for read alignment to the human genome (Homo_sapiens.GRCh38.96.gtf) and featureCounts [<xref rid="pgen.1009876.ref069" ref-type="bibr"><u>69</u></xref>] for counting of mapped reads (multimapped reads were discarded).<li> [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [68](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34827932">34827932</a> <b>(1)</b> An Official Outbreak Investigation of Acute Haemorrhagic Diarrhoea in Dogs in Norway Points to <italic toggle="yes">Providencia alcalifaciens</italic> as a Likely Cause<ul><li>Initial quality control and assembly of isolates was performed using an in-house <b>Nextflow</b> Bifrost pipeline [<xref rid="B14-animals-11-03201" ref-type="bibr"><u>14</u></xref>].<li> [14](          <person-group person-group-type="author"><name><surname>Lagesen</surname><given-names>K.</given-names></name></person-group>          <article-title>Nextflow Bifrost Pipeline</article-title>          <comment>Available online: </comment>          <pub-id pub-id-type="doi">10.5281/zenodo.4020511</pub-id>          <date-in-citation content-type="access-date" iso-8601-date="2021-11-01">(accessed on 1 November 2021)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34651188">34651188</a> <b>(1)</b> Characterization of primary models of human trophoblast<ul><li> The alignment and quality control were processed using the <b>Nextflow</b> (<xref rid="DEV199749C13" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>; <xref rid="DEV199749C16" ref-type="bibr"><u>Ewels et al., 2020</u></xref>) pipeline (version 20.01.0, <uri xlink:href="https://nf-co.re/rnaseq">https://nf-co.re/rnaseq</uri>) with the option &#x2018;--aligner hisat2&#x2019;.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34736398">34736398</a> <b>(9)</b> MSA: reproducible mutational signature attribution with confidence based on simulations<ul><li> We resolve this issue by utilising <b>Nextflow</b>&#xA0;[<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], a domain-specific language (DSL) designed to primarily address computational irreproducibility and efficient parallel execution in a large number of computing environments, from individual workstations to server clusters and cloud computing services.<li>The MSA tool is implemented as a set of scripts in Python language, with simulation, optimisation and final attribution steps fully automised using <b>Nextflow</b> workflow management system&#xA0;[<xref ref-type="bibr" rid="CR10"><u>10</u></xref>].<li> All dependencies, including packages such as <italic>pandas, numpy, scipy, matplotlib</italic> and <italic>seaborn</italic>, are automatically handled by <b>Nextflow</b> via containerisation using Docker technology, as well as Singularity where Docker is not available.<li> Users not willing to use Docker or Singularity may opt to use a Conda environment that is also automatically handled by <b>Nextflow</b>, yet this approach has inferior reproducibility compared to container technology.<li><bold><italic><b>nextflow</b> run https://gitlab.com/s.senkin/MSA -profile docker --dataset test --SP_matrix_generator_output_path path/to/SP_ME/ --SP_extractor_output_path path/to/SP_extractor/</italic></bold>.<li>Execution of the MSA tool using <b>Nextflow</b> automatically produces various output files, including tables with absolute and relative signature attributions, statistical information with goodness-of-fit measures, optimisation plots and tables, fitted mutation spectra, residuals and all bootstrap output including confidence intervals of attribution.<li> Therefore, as a default approach fully automised within the MSA <b>Nextflow</b> pipeline, we use data-driven simulations based on a simple bootstrap of signature activities derived without regularisation (i.e.&#xA0;with zero penalties).<li> The tool comprises a set of Python scripts unified in a single <b>Nextflow</b> pipeline with containerisation, specifically designed for cross-platform reproducibility and scalability in high-performance computing environments.<li><bold>Programming language:</bold> Python, <b>Nextflow</b> DSL.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34803875">34803875</a> <b>(1)</b> Fixel-Based Analysis and Free Water Corrected DTI Evaluation of HIV-Associated Neurocognitive Disorders<ul><li> Free water corrected DTI (fwcDTI) metrics (FA<sub>T</sub> and MD<sub>T</sub>) were computed with a bi-tensor model from the DWI using a previously described algorithm (<xref rid="B58" ref-type="bibr"><u>58</u></xref>) and the processing was performed using <b>Nextflow</b> pipeline (<xref rid="B59" ref-type="bibr"><u>59</u></xref>) with all software dependencies bundled in a Singularity Container (<xref rid="B60" ref-type="bibr"><u>60</u></xref>).<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31632265">31632265</a>) [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737283">34737283</a> <b>(1)</b> Smad4 controls signaling robustness and morphogenesis by differentially contributing to the Nodal and BMP pathways<ul><li>Biological replicate datasets were analyzed using the Francis Crick Institute &#x201C;BABS-RNASeq <b>Nextflow</b> pipeline [<ext-link ext-link-type="uri" xlink:href="https://github.com/crickbabs/BABS-RNASeq">https://github.com/crickbabs/BABS-RNASeq</ext-link>]&#x201D;<sup><xref ref-type="bibr" rid="CR102"><u>102</u></xref></sup>.<li> [102](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737324">34737324</a> <b>(2)</b> High-throughput sequencing for species authentication and contamination detection of 63 cell lines<ul><li> To determine the species of the cell line, metagenomic analysis was performed using the nf-villumina<sup><xref ref-type="bibr" rid="CR8"><u>8</u></xref></sup> (v2.0.0) <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR9"><u>9</u></xref></sup> workflow on the concatenated DNA and cDNA sequencing data.<li> The processed reads for each cell line were mapped against the top matching <italic>COX1</italic> sequence identified by BLAST analysis using Snippy (v4.6.0)<sup><xref ref-type="bibr" rid="CR18"><u>18</u></xref></sup> as part of the nf-illmap <b>Nextflow</b> workflow (v1.0.0)<sup><xref ref-type="bibr" rid="CR19"><u>19</u></xref></sup>.<li> [8](Kruczkiewicz, P. peterk87/nf-villumina. <ext-link ext-link-type="uri" xlink:href="https://github.com/peterk87/nf-villumina">https://github.com/peterk87/nf-villumina</ext-link> (2020).) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](Seemann, T. <italic>Snippy (v4.4.1): Rapid haploid variant calling and core genome alignment</italic>. (2019).) [19](Kruczkiewicz, P. <italic>peterk87/nf-illmap</italic>. (2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737383">34737383</a> <b>(66)</b> Design considerations for workflow management systems use in production genomics research and the clinic<ul><li> <b>Nextflow</b> is a complete system that combines the workflow language and execution engine, and is perhaps one of the most mature WfMSs to-date.<li> The Groovy-based <b>Nextflow</b> is similarly powerful, though easier to work with, providing the object-oriented look and feel of Java and access to any library written for JVM.<li> Like Swift/T, <b>Nextflow</b> treats functions as <italic>first class objects</italic><sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup> that can be used in the same ways as variables, enabling the programmer to create easily extensible pipelines, which is a very important feature in the world of ever-changing bioinformatics analyses.<li>Summary of language-level differences among Swift/T, <b>Nextflow</b>, CWL and WDL.<li><b>Nextflow</b></th><th align="left">CWL</th><th align="left">WDL</th></tr></thead><tbody><tr><td align="left">Parent language.<li><b>Nextflow</b>Workbench<sup><xref ref-type="bibr" rid="CR27"><u>27</u></xref></sup>, DolphinNext<sup><xref ref-type="bibr" rid="CR28"><u>28</u></xref></sup>.<li> <b>Nextflow</b> is similar, but not quite as permissive as Swift/T, where it defines processes to wrap user&#x2019;s scripts written in other languages and considers a workflow to be a series of those process definitions.<li> <b>Nextflow</b> lacked the ability to import and reuse processes, until recently with <b>Nextflow</b> DSL-2.<li>In contrast, WDL and <b>Nextflow</b> stylistically separate the areas where sequential execution of multiple commands is permitted.<li> command blocks inside WDL tasks are equivalent to the script blocks inside <b>Nextflow</b> processes.<li> Uniquely, <b>Nextflow</b> defines both process dependencies and parallelization via input/output variables, where a multi-valued queue channel signals parallelization over its elements.<li>In Swift/T and <b>Nextflow</b>, the workflow language and its executor are packaged together, and therefore co-evolve without compatibility issues.<li>Summary of executor-level differences among Swift/T, <b>Nextflow</b>, CWL and WDL.<li><italic><b>Nextflow</b></italic> (DSL-1, DSL-2).<li> <b>Nextflow</b> produces the DAG upon completion of a workflow execution, in static or interactive format, using Cytoscape.js<sup><xref ref-type="bibr" rid="CR37"><u>37</u></xref></sup> and Dagre graph visualization libraries (Fig.<li> Alternatively, <b>Nextflow</b>Workbench<sup><xref ref-type="bibr" rid="CR27"><u>27</u></xref></sup> and DolphinNext<sup><xref ref-type="bibr" rid="CR28"><u>28</u></xref></sup> provide convenient graphical and web interfaces developed outside the <b>Nextflow</b> core team for visualizing, creating, deploying and executing <b>Nextflow</b> pipelines.<li> section), including docker images specifications, memory, queue and/or cloud resources is readily possible in <b>Nextflow</b>, CWL and WDL.<li><b>Nextflow</b> and executors of CWL and WDL all provide this staging capability via a canonical hierarchy of execution folders.<li> <b>Nextflow</b> supports this out of the box: the analyst need only specify a URL to enable reading of inputs from AWS S3 storage or Google cloud buckets.<li> <b>Nextflow</b>, CWL, and WDL made this a requirement.<li>We compared the performance of <b>Nextflow</b> , toil (running CWL), and Cromwell (running WDL and CWL code) against these criteria by designing very simple one-step and two-step workflows&#xA0;(Fig.<li>On AWS, <b>Nextflow</b> and Cromwell+WDL both showed excellent performance up to about 100 tasks: the elapsed workflow run time did not increase significantly with the number of tasks, suggesting that they were properly parallelized with minimal management overhead (Fig.<li> <b>Nextflow</b> performed particularly well, finishing the runs about 4 times faster than Cromwell.<li> <b>Nextflow</b> could not be used at all on more than 512 tasks: it quickly stopped the run and cleaned-up.<li>Scaling a one-step (solid line) and two-step (dashed line) workflow in Cromwell+WDL (black) and <b>Nextflow</b> (yellow)&#xA0;on AWS&#xA0;Parallel cluster.<li> Again,&#xA0;<b>Nextflow</b> always ran much faster than the others, up to 50x at times.<li> In this experiment, while we were controlling <b>Nextflow</b>'s&#xA0;maxForks directive, we used the default value of 100 for the queueSize parameter, which defines &#x201C;the number of tasks the executor will handle in a parallel manner&#x201D;.<li> Yet, this inefficiency did not prevent <b>Nextflow</b> from outperforming Cromwell in terms of run time, despite Cromwell spreading the processes across nodes in near-perfect alliance with theoretical expectation.<li> While Swift/T, CWL, and WDL provide the typical String, Integer, Float, and Boolean types, <b>Nextflow</b> does not distinguish between these but rather uses <italic>qualifiers</italic> to indicate how variables are to be handled.<li> While the four WfMSs studied provide this, <b>Nextflow</b> goes above and beyond generic functions for reading and writing.<li> <b>Nextflow</b> provides refinements to handle especially large, binary and compressed files.<li> <b>Nextflow</b> workflows can use nf-core schema commands<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup>.<li> Perhaps for these reasons, data streaming is only directly possible with <b>Nextflow</b> DSL-2 syntax, as of the time of writing (March 2021).<li> <b>Nextflow</b> couples the maxRetries, maxErrors, and errorStrategy process directives to allow the user to retry a failed process, ignore the error, finish the run, or totally terminate the workflow effectively killing submitted processes.<li> <b>Nextflow</b> is on the opposite end of the spectrum, permitting very granular access to workflow stages for restart purposes.<li> section), <b>Nextflow</b> and the runners of CWL and WDL produce a canonical hierarchy of execution folders, with logs capturing the status of each workflow step saved into the same subfolder as the actual bash script being executed, along with the corresponding input and output data.<li><b>Nextflow</b> supports several levels of detailed monitoring upon executing a workflow: (1) a crude trace report, (2) an html timeline, and (3) a complete execution report, including information about resources usage and processes runtime metrics (e.g., status, hash, command).<li> Additionally, <b>Nextflow</b> is adding the Research Object (RO) model<sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>, and thus adding greater transparency by uniquely identifying, collecting, and linking all provenance metadata of workflow runs<sup><xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup>.<li> Additionally, <b>Nextflow</b> has support for email notifications of workflow events like onComplete and onError, independent of the usual notifications from the CRM.<li> <b>Nextflow</b> has the Tower platform (<ext-link ext-link-type="uri" xlink:href="https://tower.nf/">https://tower.nf/</ext-link>) for efficient monitoring and deployment; whereas WDL workflows can be submitted to a Cromwell server and examined via cromshell<sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> and Oliver<sup><xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup>.<li> <b>Nextflow</b> utilizes a conda directive to specify packages needed by a given process and supports a container directive that allows processes to specify the docker or singularity images in which execution occurs.<li> CWL and WDL each provide the code annotation capability via metadata blocks, and <b>Nextflow</b> via both the manifest scope of the configuration file and the directives of processes.<li> Likewise, curating <b>Nextflow</b> pipelines in nf-core<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup> requires using an MIT license, docker bundled software, stable release tags, a common pipeline structure, and continuous integration testing, in addition to passing their nf-core lint tests.<li><b>Nextflow</b>, CWL and WDL are fruits of practicing bioinformaticians and computer scientists collaborating with biologists, who needed practical solutions to the problem of reliably performing their own large-scale analyses.<li> AWS provides the iGenomes database (<ext-link ext-link-type="uri" xlink:href="https://registry.opendata.aws/aws-igenomes/">https://registry.opendata.aws/aws-igenomes/</ext-link>) with the ability to directly use <b>Nextflow</b> and Cromwell on AWS Batch.<li><b>Nextflow</b>.<li> Assess: is there a need to build a new pipeline, or there is an existing reasonable pipeline in the <b>Nextflow</b>, CWL,or WDL repos?.<li> If a quick development cycle is important, <b>Nextflow</b> is optimal.<li> Our recommended free, production-scale runners for these are: Cromwell (for both WDL and CWL), and <b>Nextflow</b> (for <b>Nextflow</b> workflows).<li> Within the bioinformatics community, <b>Nextflow</b>, CWL, and WDL seem to be among the most adopted.<li> It is also not clear how comparable the implementation was in the systems compared: Snakemake (python-based), Pegasus, <b>Nextflow</b> (java-based), Toil+CWL (python-based), and Cromwell+WDL (java-based).<li> By rapid prototyping, they quickly evaluated Snakemake, CWL+CWLtool, CWL+Toil, and <b>Nextflow</b> on a subset of RiboVis<sup><xref ref-type="bibr" rid="CR90"><u>90</u></xref></sup> workflows.<li> On the other, for the command-line frameworks examined here, many supporting tools exist that allow more friendly interfaces to the creation and deployment of workflow: <b>Nextflow</b>Workbench<sup><xref ref-type="bibr" rid="CR27"><u>27</u></xref></sup> &amp; DolphinNext<sup><xref ref-type="bibr" rid="CR28"><u>28</u></xref></sup> for <b>Nextflow</b>; Rabix composer &amp; CWL-Experimental (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-language/cwl-ex">https://github.com/common-workflow-language/cwl-ex</ext-link>) for CWL; and womtool for WDL.<li> Dockstore<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>, the standard implementation of the GA4GH TRS, is now hosting docker-based tools described in <b>Nextflow</b>, besides CWL and WDL.<li> In some WfMS nomenclatures this is called a <italic>process</italic> (in <b>Nextflow</b>), <italic>task</italic> (in WDL), or <italic>leaf app</italic> (in Swift/T).<li> These WfMS languages are Common Workflow Languagen (CWL)<sup><xref ref-type="bibr" rid="CR100"><u>100</u></xref></sup>, Workflow Description Language (WDL)<sup><xref ref-type="bibr" rid="CR26"><u>26</u></xref></sup> and <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR101"><u>101</u></xref></sup>.<li><b>Nextflow</b>.<li> <ext-link ext-link-type="uri" xlink:href="https://github.com/ncsa/Genomics_MGC_VariantCalling_<b>Nextflow</b>/tree/dev-gatk">https://github.com/ncsa/Genomics_MGC_VariantCalling_<b>Nextflow</b>/tree/dev-gatk</ext-link><sup>&#x2021;</sup>.<li> (2) Cloud cluster, which is built by <b>Nextflow</b> at run time and uses Apache Ignite for resource management; and (3) a dedicated, fixed-size Slurm Parallel cluster that we constructed out of 100 worker nodes and a head node, all m5a.24xlarge instances with 96 cores and <inline-formula id="IEq6"><alternatives><tex-math id="M3">\documentclass[12pt]{minimal}				\usepackage{amsmath}				\usepackage{wasysym} 				\usepackage{amsfonts} 				\usepackage{amssymb} 				\usepackage{amsbsy}				\usepackage{mathrsfs}				\usepackage{upgreek}				\setlength{\oddsidemargin}{-69pt}				\begin{document}$$\sim$$\end{document}</tex-math><mml:math id="M4"><mml:mo>&#x223C;</mml:mo></mml:math><inline-graphic xlink:href="41598_2021_99288_Article_IEq6.gif"/></alternatives></inline-formula>412 GB RAM each, configured using default AWS settings.<li><b>Nextflow</b>.<li> [20](Scott, M.&#xA0;L. <italic>Programming Language Pragmatics</italic> (Morgan Kaufmann, 2009), third edition edn.) [27](Kurs, J.&#xA0;P., Simi, M. &amp; Campagne, F. NextflowWorkbench: Reproducible and reusable workflows for beginners and experts. <italic>bioRxiv</italic> (2016).) [28](          <person-group person-group-type="author">            <name>              <surname>Yukselen</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Turkyilmaz</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Ozturk</surname>              <given-names>AR</given-names>            </name>            <name>              <surname>Garber</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kucukural</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>Dolphinnext: A distributed data processing platform for high throughput genomics</article-title>          <source>BMC Genom.</source>          <year>2020</year>          <volume>21</volume>          <fpage>1</fpage>          <lpage>16</lpage>          <pub-id pub-id-type="doi">10.1186/s12864-020-6714-x</pub-id>        ) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26415722">26415722</a>) [27](Kurs, J.&#xA0;P., Simi, M. &amp; Campagne, F. NextflowWorkbench: Reproducible and reusable workflows for beginners and experts. <italic>bioRxiv</italic> (2016).) [28](          <person-group person-group-type="author">            <name>              <surname>Yukselen</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Turkyilmaz</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Ozturk</surname>              <given-names>AR</given-names>            </name>            <name>              <surname>Garber</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kucukural</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>Dolphinnext: A distributed data processing platform for high throughput genomics</article-title>          <source>BMC Genom.</source>          <year>2020</year>          <volume>21</volume>          <fpage>1</fpage>          <lpage>16</lpage>          <pub-id pub-id-type="doi">10.1186/s12864-020-6714-x</pub-id>        ) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [41](          <person-group person-group-type="author">            <name>              <surname>Belhajjame</surname>              <given-names>K</given-names>            </name>            <etal/>          </person-group>          <article-title>Using a suite of ontologies for preserving workflow-centric research objects</article-title>          <source>J. Web Seman.</source>          <year>2015</year>          <volume>32</volume>          <fpage>16</fpage>          <lpage>42</lpage>          <pub-id pub-id-type="doi">10.1016/j.websem.2015.01.003</pub-id>        ) [42](          <person-group person-group-type="author">            <name>              <surname>Garriga Nogales</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow integration for the Research Object Specification</article-title>          <source>Zenodo</source>          <year>2018</year>          <pub-id pub-id-type="doi">10.5281/zenodo.1323830</pub-id>        ) [45](Smith, J. <italic>et&#xA0;al.</italic> broadinstitute /cromshell (2020).) [46](St. Jude Cloud Team. Oliver (2020).) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [90](          <person-group person-group-type="author">            <name>              <surname>Carja</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Xing</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Wallace</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Plotkin</surname>              <given-names>JB</given-names>            </name>            <name>              <surname>Shah</surname>              <given-names>P</given-names>            </name>          </person-group>          <article-title>riboviz: Analysis and visualization of ribosome profiling datasets</article-title>          <source>BMC Bioinf.</source>          <year>2017</year>          <volume>18</volume>          <fpage>1</fpage>          <lpage>4</lpage>          <pub-id pub-id-type="doi">10.1186/s12859-017-1873-8</pub-id>        ) [27](Kurs, J.&#xA0;P., Simi, M. &amp; Campagne, F. NextflowWorkbench: Reproducible and reusable workflows for beginners and experts. <italic>bioRxiv</italic> (2016).) [28](          <person-group person-group-type="author">            <name>              <surname>Yukselen</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Turkyilmaz</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Ozturk</surname>              <given-names>AR</given-names>            </name>            <name>              <surname>Garber</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kucukural</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>Dolphinnext: A distributed data processing platform for high throughput genomics</article-title>          <source>BMC Genom.</source>          <year>2020</year>          <volume>21</volume>          <fpage>1</fpage>          <lpage>16</lpage>          <pub-id pub-id-type="doi">10.1186/s12864-020-6714-x</pub-id>        ) [50](O&#x2019;Connor, B.&#xA0;D. <italic>et&#xA0;al.</italic> The dockstore: Enabling modular, community-focused sharing of docker-based genomics tools and workflows. <italic>F1000Research</italic><bold>6</bold> (2017).) [100](Amstutz, P. <italic>et&#xA0;al.</italic> Common workflow language, v1.0 (2016).) [26](Voss, K., Gentry, J. &amp; Van&#xA0;der Auwera, G. Full-stack genomics pipelining with gatk4+ wdl+ cromwell [version 1; not peer reviewed] (2017).) [101](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34731594">34731594</a> <b>(2)</b> Age-related Differences in the Nasal Mucosal Immune Response to SARS-CoV-2<ul><li> To facilitate reproducible analysis, samples were processed using the publicly available nf-core/RNA-seq pipeline version 1.4.2 implemented in <b>Nextflow</b> 19.10.0 using Singularity 3.2.1&#x2013;1 with the minimal command <b>nextflow</b> run nf-core/rnaseq -r 1.4.2 &#x2013;singleEnd -profile singularity &#x2013;reverseStranded &#x2013;three_prime_clip_r2 3 (<xref rid="bib28" ref-type="bibr"><u>28</u></xref>&#x2013;<xref rid="bib30" ref-type="bibr"><u>30</u></xref>).<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34731634">34731634</a> <b>(2)</b> Global characterization of macrophage polarization mechanisms and identification of M2-type polarization inhibitors<ul><li> This analytical pipeline is available on the <b>Nextflow</b> platform (<xref rid="R16" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> The pipeline is built upon <b>Nextflow</b>, which is a workflow management platform for arranging multiple tasks across various computational infrastructures in a portable manner.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34724897">34724897</a> <b>(1)</b> FASTAFS: file system virtualisation of random access compressed FASTA files<ul><li> This will allow the researcher to integrate FASTAFS with workflow management systems such as Snakemake&#xA0;[<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] or <b>Nextflow</b>&#xA0;[<xref ref-type="bibr" rid="CR21"><u>21</u></xref>] as well as software dependencies by including dependency management configurations.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34827843">34827843</a> <b>(1)</b> In Vitro Incubations Do Not Reflect In Vivo Differences Based on Ranking of Low and High Methane Emitters in Dairy Cows<ul><li>Analysis of 16S sequencing data was performed using the <b>Nextflow</b> computational pipeline ampliseq v1.1.2 (<uri xlink:href="https://github.com/nf-core/ampliseq">https://github.com/nf-core/ampliseq</uri>, accessed on 21 September 2020).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34828348">34828348</a> <b>(4)</b> ORPER: A Workflow for Constrained SSU rRNA Phylogenies<ul><li> All steps are embedded in a <b>Nextflow</b> script [<xref rid="B18-genes-12-01741" ref-type="bibr"><u>18</u></xref>], and a Singularity definition file is provided for containerization [<xref rid="B19-genes-12-01741" ref-type="bibr"><u>19</u></xref>].<li><b>Nextflow</b> is the latest workflow system.<li> <b>Nextflow</b> further presents the advantage of exploiting Singularity containers as an operating system [<xref rid="B18-genes-12-01741" ref-type="bibr"><u>18</u></xref>], which ensures the sustainability of future analyses.<li> Owing to these advantages, we chose the combination <b>Nextflow</b>-Singularity for ORPER.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34715899">34715899</a> <b>(1)</b> Over 1000 tools reveal trends in the single-cell RNA-seq analysis landscape<ul><li> In most cases, the platform is a programming language but a minority of tools is built around an existing framework including workflow managers such as Snakemake [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>, <xref ref-type="bibr" rid="CR26"><u>26</u></xref>] and <b>Nextflow</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34716263">34716263</a> <b>(2)</b> SARS-CoV-2 evolution in animals suggests mechanisms for rapid variant selection<ul><li>Raw sequencing data were input into a comprehensive <b>Nextflow</b> pipeline for processing next-generation sequencing data and SNV and SV calling.<li> <b>Nextflow</b> is a bioinformatics workflow manager that facilitates the development of complex and reproducible computational pipelines (<xref rid="r62" ref-type="bibr"><u>62</u></xref>).<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34834987">34834987</a> <b>(1)</b> The Evolutionary Landscape of SARS-CoV-2 Variant B.1.1.519 and Its Clinical Impact in Mexico City<ul><li> Reads were processed according to the ARTIC Network protocols for COVID-19 [<xref rid="B18-viruses-13-02182" ref-type="bibr"><u>18</u></xref>] using a <b>nextflow</b> pipeline (<uri xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</uri>, accessed on 1 November 2020).<li> [18](          <person-group person-group-type="author"><name><surname>Loman</surname><given-names>N.</given-names></name><name><surname>Rambaut</surname><given-names>A.</given-names></name><name><surname>Rowe</surname><given-names>W.</given-names></name></person-group>          <article-title>NCoV-2019 Novel Coronavirus Bioinformatics Protocol</article-title>          <year>2020</year>          <comment>Available online: <ext-link xlink:href="https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html" ext-link-type="uri">https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-11-01">(accessed on 1 November 2020)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34777307">34777307</a> <b>(1)</b> Machine Learning Uncovers a Data-Driven Transcriptional Regulatory Network for the Crenarchaeal Thermoacidophile <italic toggle="yes">Sulfolobus acidocaldarius</italic><ul><li> acidocaldarius</italic> RNA-seq compendium, we used Amazon Web Services (AWS) Batch to run the <b>Nextflow</b> pipeline (<xref rid="B12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>)<sup><xref rid="footnote5" ref-type="fn"><u>5</u></xref></sup>.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [5](<a href="https://github.com/avsastry/modulome-workflow/tree/main/2_process_data">https://github.com/avsastry/modulome-workflow/tree/main/2_process_data</a>) </ul></p>
<p><u>No PMID</u> <b>(31)</b> VirMutSig: Discovery and assignment of viral mutational signaturesfrom sequencing data<ul><li>Part 2) DATA PREPROCESSING PIPELINE (via <b>Nextflow</b>).<li> 20)</td></tr><tr><td>Docker Image</td><td>Docker Hub</td><td>dcblab/virmutsig_img:latest (<ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/dcblab/virmutsig_img" id="intref0160">https://hub.docker.com/r/dcblab/virmutsig_img</ext-link>)</td></tr><tr><td><b>Nextflow</b>.<li> (2017)</xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/" id="intref0165">https://www.<b>nextflow</b>.io/</ext-link> (v.21).<li><b>Nextflow</b> (version&#xA0;= 21) (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/" id="intref0035">https://www.<b>nextflow</b>.io/</ext-link>).<li>1&#x2013;2)INSTALLATION and DATA PREPROCESSINGPIPELINE</td><td><b>Nextflow</b> (<xref rid="bib22" ref-type="bibr"><u>Di Tommaso et&#xA0;al.,2017</u></xref>).<li>SRR_to_SNVlist.nf: the<b>Nextflow</b> pipeline file.<li><b>nextflow</b>.config: the configfile with the preprocessing settings andparameters.<li>The data preprocessing pipeline included in the VirMutSigprotocol is provided as a <b>Nextflow</b> pipeline file, named &#x201C;SRR_to_SNVlist.nf&#x201D;and included in the &#x201C;preprocessing&#x201D; folder of the GitHub repository:<ext-link ext-link-type="uri" xlink:href="https://github.com/BIMIB-DISCo/VirMutSig" id="intref0065">https://github.com/BIMIB-DISCo/VirMutSig</ext-link>.<li>The folder also includes a file named: &#x201C;<b>nextflow</b>.config&#x201D;,which must be opportunely modified according to the specific experimentalsettings (see below for the parameter description).<li>&gt;<b>nextflow</b> runSRR_to_SNVlist.nf.<li> The obtained listof SRR IDs (one per line) can be passed as input todownload the files via SRA-toolkit, by editing thefollowing parameter of the &#x201C;<b>nextflow</b>.config&#x201D;:<boxed-text id="dtbox6"><p id="p0660">params.FASTQ_input&#xA0;='../example/SRAlist_paired.txt'.<li>Also in this case, the librarypreparation layout (single-end or paired-end) mustbe specified by editing the following parameter in&#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox7"><p id="p0670">params.library_preparation&#xA0;=&#x2018;paired&#x2019; // or &#x2018;single&#x2019;.<li> It possible to change the directory byediting the following parameter in in&#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox8"><p id="p0680">params.FASTQdir&#xA0;='intermediate/FASTQ'.<li>In this case, the user must indicate the directory where theFASTQ files are located, editing the following parameter of the &#x201C;<b>nextflow</b>.config&#x201D;:<boxed-text id="dtbox12"><p id="p0710">params.FASTQ_input&#xA0;='/Path/To/Directory/'.<li>The library preparation layout (single-end or paired-end)must be specified by editing the following parameter in &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox13"><p id="p0720">params.library_preparation&#xA0;= &#x2018;paired&#x2019;//or &#x2018;single&#x2019;.<li> To perform this step, the <b>Nextflow</b> pipelineexploits the tool Trimmomatic (<ext-link ext-link-type="uri" xlink:href="http://www.usadellab.org/cms/?page=trimmomatic" id="intref0080">http://www.usadellab.org/cms/?page=trimmomatic</ext-link>).<li>To tune the additional parameters used by Trimmomatic pleaseedit the following parameter in &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox14"><p id="p0750">&#xA0;params.trimmomatic_setting&#xA0;= 'LEADING:20TRAILING:20.<li>The user can specify the reference genome file by editingfollowing parameter in &#x201C;<b>nextflow</b>.config&#x201D;&#xA0;file:<boxed-text id="dtbox15"><p id="p0810">params.fasta&#xA0;='reference/SARS-CoV-2-ANC.fasta'.<li> It ispossible to change the directory by editing the following parameter in in&#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox16"><p id="p0830">params.BAMdir&#xA0;='intermediate/BAM'.<li>To include or skip this step in the preprocessing pipeline,please set accordingly the following parameter in &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox17"><p id="p0850">params.remove_duplicates&#xA0;=&#x201C;true&#x201D; // or &#x201C;false&#x201D;.<li> It is possible to change the directory by editing the followingparameter in in &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox18"><p id="p0875">params.COVERAGEdir&#xA0;='intermediate/COVERAGE'.<li> The VarScan setting can be adjusted byediting the following parameter included in &#x201C;<b>nextflow</b>.config&#x201D; file.<li> It ispossible to change the directory by editing the following parameter in in&#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox20"><p id="p0905">params.VCFdir&#xA0;='intermediate/VCF'.<li> The followingparameters determine the filtering threshold, andthey can be set by changing the correspondingnumeric value included into a string assigned toVirMutSig_QCfilter parameter present in the&#x201C;<b>nextflow</b>.config&#x201D; file.<li> It ispossible to change the directory by editing thefollowing parameter in in &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox23"><p id="p0965">params.SNVlistdir&#xA0;='../example.<li> Instead,if VirMutSig preprocessing steps have been used,the script is automatically executed via<b>Nextflow</b>.<li> For this reason, it ispossible to specify the maximum number of cores to be used, by changing thecpus value in the following setting of the &#x201C;<b>nextflow</b>.config&#x201D; file:<boxed-text id="dtbox25"><p id="p1005">process{.<li> To do so, please specify the correspondingparameter name (the string following &#x201C;params.&#x201D;) and specify the opportuneargument of the <b>Nextflow</b> command.<li>In the example, the SRR list file path and the outputdirectory path of the SNVs list file will be specified without editing the&#x201C;<b>nextflow</b>.config&#x201D; file.<li>&gt;<b>nextflow</b> runSRR_to_SNVlist.nf \.<li> The related processes areautomatically executed via <b>Nextflow</b>.<li> For example,the config files (i.e., &#x201C;<b>nextflow</b>.config&#x201D;, &#x201C;denovo_config.yalm&#x201D;, and&#x201C;assignment_config.yalm&#x201D;) can be modified using the &#x201C;nano&#x201D; editor with thefollowing command:<boxed-text id="dtbox34"><p id="p1805">root@d55f578b7306:/# nano/VirMutSig/denovo/denovo_config.yaml.<li> [Di Tommaso et&#xA0;al.,2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34675338">34675338</a> <b>(1)</b> Repositioning of a novel GABA-B receptor agonist, AZD3355 (Lesogaberan), for the treatment of non-alcoholic steatohepatitis<ul><li> RNASeq raw data were processed using FAST QC software (<ext-link ext-link-type="uri" xlink:href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>) and aligned with STAR aligner<sup><xref ref-type="bibr" rid="CR62"><u>62</u></xref></sup> via a computational pipeline implemented in <b>NextFlow</b><sup><xref ref-type="bibr" rid="CR63"><u>63</u></xref></sup>.<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34669013">34669013</a> <b>(1)</b> Anthracycline-free tumor elimination in mice leads to&#xA0;functional and molecular cardiac recovery from cancer-induced alterations in contrast to long-lasting doxorubicin treatment effects<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34668728">34668728</a> <b>(2)</b> SARS-CoV-2 Antibody Binding and Neutralization in Dried Blood Spot Eluates and Paired Plasma<ul><li> We then used a <b>Nextflow</b> data processing pipeline to align the demultiplexed sample reads to the reference peptide library in parallel, allowing for up to 2 mismatches.<li>The <b>Nextflow</b> pipeline, used to align epitope-mapping sample reads to the reference library, is available at <ext-link xlink:href="https://github.com/matsengrp/phip-flow" ext-link-type="uri">https://github.com/matsengrp/phip-flow</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34681040">34681040</a> <b>(14)</b> FA-nf: A Functional Annotation Pipeline for Proteins from Non-Model Organisms Implemented in <b>Nextflow</b><ul><li>Here, we present a scalable and parallelizable workflow for functional annotation, named FA-nf, implemented in the <b>Nextflow</b> framework [<xref rid="B21-genes-12-01645" ref-type="bibr"><u>21</u></xref>] with Docker container images generated for every process.<li> Importantly, the <b>Nextflow</b> framework also avoids the problem of installing software dependencies by the user and can, therefore, be used by researchers with less bioinformatic expertise [<xref rid="B22-genes-12-01645" ref-type="bibr"><u>22</u></xref>] than is required by other existing pipelines (<xref rid="genes-12-01645-f001" ref-type="fig"><u>Figure 1</u></xref>).<li> During recent years, the bioinformatics community has been progressively adopting various management systems, such as Snakemake [<xref rid="B24-genes-12-01645" ref-type="bibr"><u>24</u></xref>], Galaxy [<xref rid="B25-genes-12-01645" ref-type="bibr"><u>25</u></xref>], <b>Nextflow</b> [<xref rid="B21-genes-12-01645" ref-type="bibr"><u>21</u></xref>].<li> For the FA-nf pipeline, we selected <b>Nextflow</b> because it has provided support for container technologies, different HPC queue schedulers and cloud providers from its inception.<li>Every <b>Nextflow</b> process (that is, a pipeline step) is associated with a specific container image.<li> Within a configuration file, customarily named <b>nextflow</b>.config, as long as processes are properly labelled, the user can adjust the minimum assigned CPU cores and available memory to be demanded from a computation node.<li>Install <b>Nextflow</b> (version 20.10.0 tested).<li> curl -s <uri xlink:href="https://get.<b>nextflow</b>.io">https://get.<b>nextflow</b>.io</uri> | bash.<li>If your system does not have internet connection, you can generate Singularity files in advance and modify accordingly the container tag values in the <b>nextflow</b>.config file.<li>A <b>Nextflow</b> pipeline script for downloading necessary datasets (download.nf) is available.<li>$ ./<b>nextflow</b> run -bg download.nf --config params.download.config &amp;&gt; download.logfile .<li>Check <b>nextflow</b>.config file and adjust it according to the characteristics of your HPC queue system by replacing queue names and increasing or decreasing aspects such as CPU or RAM.<li> More details at: <uri xlink:href="https://www.<b>nextflow</b>.io/docs/latest/config.html">https://www.<b>nextflow</b>.io/docs/latest/config.html</uri>.<li>$ ./<b>nextflow</b> run -bg main.nf --config params.config &amp;&gt; logfile.<li> Indeed, with <b>Nextflow</b>, it is possible to define a specific release of its engine with the NXF_VER environment variable.<li> Thanks to that, different versions of the pipeline would continue working, even if non-backwards compatible syntax changes are introduced in future versions of <b>Nextflow</b> software.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33600404">33600404</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34721839">34721839</a> <b>(1)</b> BOSC 2021, the 22nd Annual Bioinformatics Open Source Conference<ul><li>The session on Workflow Management Systems included updates on widely used workflow systems such as <b>Nextflow</b>, Dockstore and Sapporo, and introductions to newer resources such as WDL Analysis Research Pipelines (WARP) and WorkFlow Package Manager (WFPM).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34649268">34649268</a> <b>(1)</b> Genomic reconstruction of the SARS-CoV-2 epidemic in England<ul><li> We sequenced 384-sample pools on Illumina NovaSeq, and produced consensus fasta sequences according to the ARTIC <b>nextflow</b> processing pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34643507">34643507</a> <b>(2)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> For example, the Snakemake and <b>Nextflow</b> workflow engines facilitate execution of workflows and are used widely for scientific research (<xref rid="bib15" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>; <xref rid="bib28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> For example, to use Snakemake or <b>Nextflow</b>, a researcher can specify a workflow using a custom programming language that is specific to either workflow engine.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> A reviewer notes, "I think it would be useful to beginners to see more information on the benefits of using CWL over some of the alternatives (<b>Nextflow</b>, Snakemake).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> A reviewer notes, "I think it would be useful to beginners to see more information on the benefits of using CWL over some of the alternatives (<b>Nextflow</b>, Snakemake).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34636840">34636840</a> <b>(1)</b> <italic toggle="yes">vcf2gwas</italic>: Python API for comprehensive GWAS analysis using GEMMA<ul><li> <italic toggle="yes">vcf2gwas</italic> avoids steps such as setting up a configuration file that are common in <b>Nextflow</b>, Snakemake-based pipelines, and also avoids commonly experienced issues with root privileges necessary for running an application inside a Docker container.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34634030">34634030</a> <b>(1)</b> qc3C: Reference-free quality control for Hi-C sequencing data<ul><li>To validate qc3C&#x2019;s reference-free method, a simulated parameter sweep was implemented in <b>Nextflow</b> [<xref rid="pcbi.1008839.ref016" ref-type="bibr"><u>16</u></xref>], where reads were generated using the Hi-C read simulator sim3C (v0.3.2_py3) [<xref rid="pcbi.1008839.ref017" ref-type="bibr"><u>17</u></xref>] with E.coli MG1655 (acc: GCF_000005845.2) as the genomic source.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29149264">29149264</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34625563">34625563</a> <b>(5)</b> Characterisation of tumour microenvironment remodelling following oncogene inhibition in preclinical studies with imaging mass cytometry<ul><li>imcyto: a <b>Nextflow</b>-based automated pipeline for segmentation on a high-performance computing cluster.<li>For the automated processing of our complex sequential segmentation method on large datasets, we developed a <b>Nextflow</b>-based segmentation pipeline, optimised to run on a high-performance computing cluster.<li> <b>Nextflow</b> is based on the concept that software does not need to be installed on a server for it to be executed.<li> For automation of our workflow, we combined IMCtools, Ilastik and CellProfiler into our <b>Nextflow</b>-based &#x201C;imcyto&#x201D; pipeline (Supplementary Fig.&#xA0;<xref rid="MOESM1" ref-type="media"><u>2</u></xref>).<li>An automated <b>Nextflow</b>-based pipeline that sequentially preprocesses and segments imaging data and extracts single cell expression information (see Supplementary Fig.&#xA0;<xref rid="MOESM1" ref-type="media"><u>2</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34690959">34690959</a> <b>(10)</b> Graph-Based Approaches Significantly Improve the Recovery of Antibiotic Resistance Genes From Complex Metagenomic Datasets<ul><li>The pipeline is implemented using the <b>Nextflow</b> framework (Di Tommaso et al., <xref rid="B5" ref-type="bibr"><u>2017</u></xref>; Ewels et al., <xref rid="B6" ref-type="bibr"><u>2020</u></xref>) that enables scalable, reproducible and efficient computational workflow.<li>The pipeline is implemented in <b>Nextflow</b> and therefore requires <b>Nextflow</b> to be installed in order to be used.<li> For the full reproducibility, the use of <b>Nextflow</b>-supported package manager such as Conda is advised.<li>Install <b>nextflow</b> (<ext-link xlink:href="https://nf-co.re/usage/installation" ext-link-type="uri">https://nf-co.re/usage/installation</ext-link>).<li>Install any <b>Nextflow</b>-supported container engines, such as conda (<ext-link xlink:href="https://conda.io/miniconda.html" ext-link-type="uri">https://conda.io/miniconda.html</ext-link>).<li>Download the pipeline and test it on a minimal dataset with a single command: <b>nextflow</b> run ablab/graphamr -profile test, conda.<li><monospace><b>nextflow</b> run ablab/graphamr -profile conda --reads &#x2018;<sup>*</sup>_R{1,2}.fastq.gz&#x2019;</monospace>.<li><monospace><b>nextflow</b> run ablab/graphamr -profile conda --graph &#x2018;assembly_graph_with_scaffolds.gfa&#x2019;</monospace>.<li><monospace><b>nextflow</b> run ablab/graphamr -profile conda --graph &#x2018;assembly_graph_with_scaffolds.gfa&#x2019; --db [&#x2018;ncbi_AMR_HMM&#x2019;, &#x2018;card_AA&#x2019;]</monospace>.<li> [2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34615891">34615891</a> <b>(2)</b> A scalable open-source MATLAB toolbox for reconstruction and analysis of multispectral optoacoustic tomography data<ul><li> A <b>Nextflow</b> wrapper around the MATLAB systems allows multiple pipelines to be run on a single computational cluster simultaneously.<li> To this end, we implemented a <b>Nextflow</b> wrapper around the toolbox to illustrate a possible scenario of deploying the toolbox on a computational cluster.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34642694">34642694</a> <b>(2)</b> Comprehensive characterization of the antibody responses to SARS-CoV-2 Spike protein after infection and/or vaccination<ul><li> Post sample demultiplexing, we used a <italic toggle="yes"><b>Nextflow</b></italic> pipeline to process the peptide counts as well as alignment stats for all samples<sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup>.<li> The source code, <b>Nextflow</b> script, software dependencies, and instructions for re-running the analysis can be found at <ext-link xlink:href="https://github.com/matsengrp/phage-dms-vacc-analysis" ext-link-type="uri">https://github.com/matsengrp/phage-dms-vacc-analysis</ext-link>.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34611252">34611252</a> <b>(1)</b> An automated and combinative method for the predictive ranking of candidate effector proteins of fungal plant pathogens<ul><li> The pipeline is implemented in <b>Nextflow</b> (version&#x2009;&gt;&#x2009;20)<sup><xref ref-type="bibr" rid="CR64"><u>64</u></xref></sup>, and a conda environment and Docker container are available for easy installation of dependencies, with scripts to integrate user-downloaded proprietary software into these environments.<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34608132">34608132</a> <b>(1)</b> Orchestrating and sharing large multimodal data for transparent and reproducible research<ul><li> There have been multiple community efforts in creating standardized workflow languages, such as the Common Workflow Language (CWL) and the Workflow Definition Language (WDL), along with associated workflow management systems such as Snakemake<sup><xref ref-type="bibr" rid="CR8"><u>8</u></xref></sup> and <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR9"><u>9</u></xref></sup>, in order to promote reproducibility<sup><xref ref-type="bibr" rid="CR10"><u>10</u></xref>,<xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](          <person-group person-group-type="author">            <name>              <surname>Goble</surname>              <given-names>C</given-names>            </name>            <etal/>          </person-group>          <article-title>FAIR computational workflows</article-title>          <source>Data Intell.</source>          <year>2020</year>          <volume>2</volume>          <fpage>108</fpage>          <lpage>121</lpage>          <pub-id pub-id-type="doi">10.1162/dint_a_00033</pub-id>        ) [11](          <person-group person-group-type="author">            <name>              <surname>Kulkarni</surname>              <given-names>N</given-names>            </name>            <etal/>          </person-group>          <article-title>Reproducible bioinformatics project: a community for reproducible bioinformatics analysis pipelines</article-title>          <source>BMC Bioinforma.</source>          <year>2018</year>          <volume>19</volume>          <fpage>349</fpage>          <pub-id pub-id-type="doi">10.1186/s12859-018-2296-x</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34599327">34599327</a> <b>(1)</b> A Minimally Morphologically Destructive Approach for DNA Retrieval and Whole-Genome Shotgun Sequencing of Pinned Historic Dipteran Vector Species<ul><li> 2021</xref>) (<b>nextflow</b> version 20.10.0, EAGER last modified December 2020).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34608459">34608459</a> <b>(1)</b> The role of viral genomics in understanding COVID-19 outbreaks in long-term care facilities<ul><li> Convergence on a small number of open-source bioinformatics workflows using best practices should mitigate future issues in this regard (eg, a <ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" id="interrefs20"><b>Nextflow</b> pipeline</ext-link> with a focus on COVID-19, one global resource on the <ext-link ext-link-type="uri" xlink:href="https://covid19.galaxyproject.org" id="interrefs30">Galaxy platform</ext-link> for the analysis of SARS-CoV-2 data, and a workspace on the <ext-link ext-link-type="uri" xlink:href="https://app.terra.bio/#workspaces/pathogen-genomic-surveillance/COVID-19" id="interrefs40">Terra app</ext-link> with COVID-19 genomic data and workflows.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34608459">34608459</a> <b>(1)</b> The role of viral genomics in understanding COVID-19 outbreaks in long-term care facilities<ul><li> This analysis workflow was implemented in the <b>Nextflow</b> framework and open sourced under the MIT license as the CoVigator NGS pipeline (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/TRON-Bioinformatics/covigator-ngs-pipeline" ext-link-type="uri">https://github.com/TRON-Bioinformatics/covigator-ngs-pipeline</ext-link>) [<xref rid="pone.0249254.ref020" ref-type="bibr"><u>20</u></xref>].<li> [20](<name><surname>Pablo Riesgo-Ferreiro</surname><given-names>Patrick Sorn</given-names></name>, <name><surname>Bukur</surname><given-names>Thomas</given-names></name> (<year>2021</year>) <article-title>TRON-Bioinformatics/covigator-ngs-pipeline.</article-title><source>Release Covigator NGS pipeline v0.4.0</source>.) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34559589">34559589</a> <b>(2)</b> Computational methods for RNA modification detection from nanopore direct RNA sequencing data<ul><li> Users can choose to run these steps manually or they can be executed automatically using the Nanocompore pipeline, an automated workflow implemented in <b>Nextflow</b> that runs the entire analysis.<li> EpiNano source code can be downloaded from GitHub, but can be also executed directly as part of the MasterOfPores [<xref rid="cit0059" ref-type="bibr"><u>59</u></xref>] <b>NextFlow</b> workflow, which is aimed at facilitating the analysis of direct RNA nanopore sequencing data, and performs data pre-processing, mapping, RNA modification detection and poly(A) tail length estimation.<li> [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34632220">34632220</a> <b>(1)</b> Amplification of Femtograms of Bacterial DNA Within3 h Using a Digital Microfluidics Platform for MinION Sequencing<ul><li> The taxonomy pipeline was implemented using<b>Nextflow</b> v19.10.0.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34554082">34554082</a> <b>(1)</b> Rapid and accurate SNP genotyping of clonal bacterial pathogens with BioHansel<ul><li> A <b>Nextflow</b> workflow has been developed for performing the computational resource benchmarking and is available online at: <ext-link xlink:href="https://github.com/peterk87/nf-bh-typing-comparison" ext-link-type="uri">https://github.com/peterk87/nf-bh-typing-comparison</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35865756">35865756</a> <b>(1)</b> Detecting microstructural deviations in individuals with deep diffusion MRI tractometry<ul><li> Tractometry<sup><xref rid="R16" ref-type="bibr"><u>16</u></xref></sup> was performed (sampling fractional anisotropy, mean diffusivity, and RISH0 and RISH2 at 20 locations along the tracts<sup><xref rid="R19" ref-type="bibr"><u>19</u></xref>,<xref rid="R23" ref-type="bibr"><u>23</u></xref>,</sup><sup><xref rid="R25" ref-type="bibr"><u>25</u></xref></sup>) using a <b>Nextflow</b> architecture provided by SCILPY (<ext-link xlink:href="https://github.com/scilus/scilpy" ext-link-type="uri">github.com/scilus/scilpy</ext-link>).<li> [16](          <person-group person-group-type="author"><name><surname>Bells</surname><given-names>S</given-names></name><etal/></person-group>          <source>Tractometry-comprehensive multi-modal quantitative assessment of white matter along specific tracts</source>          <conf-name>Proc ISMRM</conf-name>          <year>2011</year>          <volume>678</volume>          <fpage>0678</fpage>        ) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23166771">23166771</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28794981">28794981</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31228638">31228638</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36700091">36700091</a> <b>(1)</b> <italic toggle="yes">Omics Notebook</italic>: robust, reproducible and flexible automated multiomics exploratory analysis and reporting<ul><li> KNIME or <b>Nextflow</b>) are powerful tools for constructing pipelines but add system dependencies (e.g.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34548573">34548573</a> <b>(2)</b> HTSQualC is a flexible and one-step quality control software for high-throughput sequencing data analysis<ul><li> Additionally, we have also provided the docker image and <b>Nextflow</b> template for running the HTSQualC.<li> distributed mode with 18 CPUs per computing node using <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34646384">34646384</a> <b>(1)</b> Glycoproteome remodeling in MLL-rearranged B-cell precursor acute lymphoblastic leukemia<ul><li> For the latter analysis, reads were quality-checked and processed using the nf-core/rnaseq analysis workflow v1.3 pipeline consisting of <b>Nextflow</b> v19.04.1, FastQC v0.11.8 Cutadapt v2.4, Trim Galore! v0.5.0, STAR vSTAR_2.6.1d, HISAT2 v2.1.0, Picard MarkDuplicates v2.18.27, Samtools v1.9, featureCounts v1.6.4, StringTie v1.3.5, Preseq v2.0.3, deepTools, v3.2.0, RSeQC v3.0.0, MultiQCv1.7 software.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34611536">34611536</a> <b>(1)</b> RNA sequencing data for responses to drought stress and/or clubroot infection in developing seeds of <italic>Brassica napus</italic><ul><li> Briefly, the pipeline was based on <b>Nextflow</b> v19.07.0 <xref rid="bib0010" ref-type="bibr"><u>[10]</u></xref> and processed data using the default options..<li> [[10]](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>E.W.</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>P.P.</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E.</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nat. Biotechnol</source>          <volume>11</volume>          <year>2017</year>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34631327">34631327</a> <b>(1)</b> Software-Defined Workflows for Distributed Interoperable Closed-Loop Neuromodulation Control Systems<ul><li> Popular workflow languages such as CWL [<xref rid="R2" ref-type="bibr"><u>2</u></xref>], WDL [<xref rid="R24" ref-type="bibr"><u>24</u></xref>], and <b>NextFlow</b> [<xref rid="R25" ref-type="bibr"><u>25</u></xref>] help develop scientific applications in a modular fashion from interoperable services.<li> [2](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, and <name><surname>Kartashov</surname><given-names>A</given-names></name>, &#x201C;<article-title>Common workflow language, v1. 0</article-title>,&#x201D; <source>Figshare, Tech. Rep</source>, <year>2016</year>.) [24](<name><surname>Frazer</surname><given-names>S</given-names></name>. (<year>2014</year>). <source>Workflow Description Language</source>. <comment>GitHub. [Online]. Available</comment>: <comment><ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/wdl">https://github.com/broadinstitute/wdl</ext-link></comment>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a> <b>(3)</b> Software-Defined Workflows for Distributed Interoperable Closed-Loop Neuromodulation Control Systems<ul><li> <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>) written in the <b>Nextflow</b> domain specific language (version 20.10.0)<sup><xref rid="R48" ref-type="bibr"><u>48</u></xref></sup> was used to perform the QC, variant calling and consensus sequence generation for the samples.<li> The full command used was &#x201C;<b>nextflow</b> run ncov2019-artic-nf --nanopolish --prefix $PREFIX --basecalled_fastq fastq_pass/ --fast5_pass fast5_pass/ --sequencing_summary sequencing_summary.txt --schemeVersion V3 --minReadsPerBarcode 1 --minReadsArticGuppyPlex 1 -with-singularity articncov2019-nanopore.img -profile singularity,slurm -r v1.1.1&#x201D;.<li> The full command used was &#x201C;<b>nextflow</b> run nf-core/viralrecon --input samplesheet.csv --genome &#x2018;MN908947.3&#x2019; --amplicon_bed nCoV-2019.artic.V3.bed --protocol &#x2018;amplicon&#x2019; --callers ivar --skip_assembly --skip_markduplicates --skip_fastqc --skip_picard_metrics --save_align_intermeds -profile crick -r 1.1.0&#x201D;.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34580592">34580592</a> <b>(1)</b> Intestinal Organoids in Colitis Research: Focusing on Variability and Cryopreservation<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34525337">34525337</a> <b>(1)</b> Adenosine-to-inosine editing of endogenous Z-form RNA by the deaminase ADAR1 prevents spontaneous MAVS-dependent type I interferon responses<ul><li> Sequencing data was processed using a <b>Nextflow</b> v20.07 (<xref rid="bib18" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2017</u></xref>) pipeline automating quality control using FastQC v0.11.8 (bioinformatics.babraham.ac.uk/projects/fastqc/), quality and adaptor trimming using cutadapt v1.18 (<xref rid="bib56" ref-type="bibr"><u>Martin, 2011</u></xref>), contaminant detection using screen.sh (within BBMap v36.20, sourceforge.net/projects/bbmap/), strand-aware alignment using HISAT2 v2.1.0 (<xref rid="bib38" ref-type="bibr"><u>Kim et&#xA0;al., 2019</u></xref>) and STAR v2.7.1a (<xref rid="bib20" ref-type="bibr"><u>Dobin et&#xA0;al., 2013</u></xref>), post-alignment quality-assurance using &#x2018;gene body coverage&#x2019;, &#x2018;transcript integrity&#x2019;, and &#x2018;inner distance&#x2019; metrics from RSeQC v2.6.4 (<xref rid="bib87" ref-type="bibr"><u>Wang et&#xA0;al., 2012</u></xref>), and strand-specific counting of uniquely-mapping reads using featureCounts (within Subread v1.6.4, (<xref rid="bib47" ref-type="bibr"><u>Liao et&#xA0;al., 2014</u></xref>)) against Ensembl GRCm38.100 annotations.<li> [Di Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Martin, 2011](          <person-group person-group-type="author">            <name>              <surname>Martin</surname>              <given-names>M.</given-names>            </name>          </person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet.journal</source>          <volume>17</volume>          <year>2011</year>          <fpage>3</fpage>        ) [Kim et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31375807">31375807</a>) [Dobin et&#xA0;al., 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [Wang et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22743226">22743226</a>) [Liao et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34519531">34519531</a> <b>(1)</b> Baseline Gut Metagenomic Functional Gene Signature Associated with Variable Weight Loss Responses following a Healthy Lifestyle Intervention in Humans<ul><li> The full workflow used to process the metagenomic data is provided as a <b>Nextflow</b> pipeline at <ext-link xlink:href="https://github.com/Gibbons-Lab/pipelines/tree/master/shallow_shotgun" ext-link-type="uri">https://github.com/Gibbons-Lab/pipelines/tree/master/shallow_shotgun</ext-link>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34520756">34520756</a> <b>(1)</b> Deconvolution of multiplexed transcriptional responses to wood smoke particles defines rapid aryl hydrocarbon receptor signaling dynamics<ul><li>PRO-Seq data were processed using a standardized <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/Dowell-Lab/Nascent-Flow" id="intref0010">https://github.com/Dowell-Lab/Nascent-Flow</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34520756">34520756</a> <b>(1)</b> Deconvolution of multiplexed transcriptional responses to wood smoke particles defines rapid aryl hydrocarbon receptor signaling dynamics<ul><li>GeneTEflow [<xref rid="B133-biology-10-00896" ref-type="bibr"><u>133</u></xref>] is a fully automated, reproducible, and platform-independent workflow that allows the comprehensive analysis of both genes and locus-specific TEs expression from RNA-Seq data employing different technologies (<b>Nextflow</b> [<xref rid="B134-biology-10-00896" ref-type="bibr"><u>134</u></xref>] and Docker [<xref rid="B135-biology-10-00896" ref-type="bibr"><u>135</u></xref>]).<li> [133](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32866155">32866155</a>) [134](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [135](          <person-group person-group-type="author"><name><surname>Merkel</surname><given-names>D.</given-names></name></person-group>          <article-title>Docker: Lightweight Linux Containers for Consistent Development and Deployment</article-title>          <source>Linux J.</source>          <year>2014</year>          <volume>2014</volume>          <fpage>2</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author"><name><surname>Merkel</surname><given-names>D.</given-names></name></person-group>          <article-title>Docker: Lightweight Linux Containers for Consistent Development and Deployment</article-title>          <source>Linux J.</source>          <year>2014</year>          <volume>2014</volume>          <fpage>2</fpage>        ">          <person-group person-group-type="author"><name><surname>Merkel</surname><given-names>D.</given-names></name></person-group>          <article-title>Docker: Lightweight Linux Containers for Consistent Development and Deployment</article-title>          <source>Linux J.</source>          <year>2014</year>          <volume>2014</volume>          <fpage>2</fpage>        </a> <b>(2)</b> Deconvolution of multiplexed transcriptional responses to wood smoke particles defines rapid aryl hydrocarbon receptor signaling dynamics<ul><li> The Fluidigm data targeting the V4 region of the 16S rRNA gene were processed using the TADA <b>Nextflow</b>-based workflow (<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/h3abionet/TADA">https://github.com/h3abionet/TADA</uri>) that implements the DADA2 workflow [<xref rid="B19-toxics-09-00215" ref-type="bibr"><u>19</u></xref>] for dereplicating and denoising reads to generate amplicon sequence variants (ASVs).<li>Briefly, sequence data were processed using a <b>Nextflow</b>-based workflow to maintain consistency and reproducibility [<xref rid="B25-toxics-09-00215" ref-type="bibr"><u>25</u></xref>].<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27214047">27214047</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34804501">34804501</a> <b>(1)</b> Perspectives on automated composition of workflows in the life sciences<ul><li> This enables workflows and associated components to be exchanged between the WorkflowHub, workflow management systems like Galaxy, Snakemake and <b>Nextflow</b> and their repositories, and workflow utilities like OpenEBench and LifeMonitor.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34493866">34493866</a> <b>(3)</b> A compendium of uniformly processed human gene expression and splicing quantitative trait loci<ul><li> Quantification was performed using a customized <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR80"><u>80</u></xref></sup> workflow that we developed by adding new quantification methods to an nf-core/rnaseq pipeline<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>.<li> The steps described above are implemented in the eQTL-Catalogue/qtlmap v.21.04.1 <b>Nextflow</b> workflow available from GitHub (see <xref rid="Sec24" ref-type="sec"><u>URLs</u></xref>).<li> The steps described above are implemented in the eQTL-Catalogue/qtlmap v.21.04.1 <b>Nextflow</b> workflow available from GitHub (see <xref rid="Sec24" ref-type="sec"><u>URLs</u></xref>).<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a> <b>(1)</b> A compendium of uniformly processed human gene expression and splicing quantitative trait loci<ul><li><underline><bold>XSVLen:</bold></underline> The XSVLen workflow requires Python3, Minimap2, <b>Nextflow</b>, and R &gt;=3.5.0.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34425749">34425749</a> <b>(1)</b> A graphical, interactive and GPU-enabled workflow to process long-read sequencing data<ul><li> MasterOfPores is a workflow using the <b>NextFlow</b> framework [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>], a script-based engine that requires programming experience to deploy and modify.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34416914">34416914</a> <b>(5)</b> <italic>ExOrthist</italic>: a tool to infer exon orthologies at any evolutionary distance<ul><li> The <italic>main</italic> module is a <italic><b>Nextflow</b></italic>-based pipeline [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>] that infers exon homologies using an innovative approach, namely evaluating three features: conservation of the neighboring intron positions (with respect to the protein-coding sequence), sequence and length conservation of the query exons, and sequence conservation of the upstream/downstream exons.<li> Second, the <italic>exint_plotter</italic> module is also a <italic><b>Nextflow</b></italic>-based pipeline that allows the visualization of evolutionary conservation and of changes in exon-intron structure among gene homologs of interest.<li> Data on the computing performance of <italic>ExOrthist</italic> for each type of run (short, medium, long), as well as for combinations of three or four species, are provided in Additional file <xref rid="MOESM1" ref-type="media"><u>1</u></xref>: Figure S1 and in the respective <italic><b>Nextflow</b></italic> reports [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> To run the program, a user only needs to install <italic><b>Nextflow</b></italic> and one Linux container, either <italic>Docker</italic> or <italic>Singularity</italic>.<li><italic>ExOrthist</italic> efficiently parallelizes all alignments and realignments thanks to <italic><b>Nextflow</b></italic>&#x2019;s management system, and later joins all the best isoform-wise matches for a given species pair.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [34](Marquez Y, Mantica F, Cozzuto L, Burguera D, Hermoso-Pulido A, Ponomarenko J, et al. ExOrthist: a tool to infer exon orthologies at any evolutionary distance. Github. 2021; <ext-link ext-link-type="uri" xlink:href="https://github.com/biocorecrg/ExOrthist">https://github.com/biocorecrg/ExOrthist</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34410361">34410361</a> <b>(1)</b> Post-Vaccination Coronavirus Disease 2019: A Case-Control Study and Genomic Analysis of 119 Breakthrough Infections in Partially Vaccinated Individuals<sup><xref rid="ciab714-FM1" ref-type="author-notes">&#xA0;</xref></sup><ul><li> For Illumina sequencing, raw data were processed with the ARTIC <b>NextFlow</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35136576">35136576</a> <b>(6)</b> Large-scale quality assessment of prokaryotic genomes with metashot/prok-quality<ul><li> Tools and standards such as Galaxy,<sup><xref rid="ref9" ref-type="bibr"><u>9</u></xref></sup> <b>Nextflow</b><sup><xref rid="ref10" ref-type="bibr"><u>10</u></xref></sup> and the Common Workflow Language,<sup><xref rid="ref11" ref-type="bibr"><u>11</u></xref></sup> coupled with container technologies like<ext-link xlink:href="https://www.docker.com/" ext-link-type="uri">Docker</ext-link>, allows researchers to circumvent these issues, providing a way to build, run and share reproducible computational workflows.<li>We present metashot/prok-quality, a comprehensive and easy-to-use <b>Nextflow</b> pipeline for assessing the quality of draft prokaryotic genomes.<li>Metashot/prok-quality is written using the <b>Nextflow</b> domain-specific language.<li> <b>Nextflow</b> is a framework for building scalable scientific workflows using containers, allowing implicit parallelism on a wide range of computing systems.<li>metashot/prok-quality v1.2.0 requires Docker and <b>Nextflow</b> (tested on v20.07.1).<li> directory, the version 1.2.0 of the workflow can be run with the following command line:<preformat preformat-type="computer code" position="float" xml:space="preserve"><b>nextflow</b> run metashot/prok-quality -r 1.2.0\--genomes 'bins/*.fa'\--outdir results.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> <b>(3)</b> Large-scale quality assessment of prokaryotic genomes with metashot/prok-quality<ul><li> # in&#xA0;prok-quality/<b>nextflow</b>.config.<li> platform that supports <b>Nextflow</b>, Docker or Singularity" should read "...<li> platform that supports <b>Nextflow</b> and either Docker or Singularity".<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> <b>(1)</b> Large-scale quality assessment of prokaryotic genomes with metashot/prok-quality<ul><li>The authors present &#x2018;prok-quality&#x2019;, a one-stop-shop <b>Nextflow</b> workflow for prokaryotic genome quality assessment, wrapping several state-of-the-art tools into a user-friendly pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34387545">34387545</a> <b>(1)</b> Patterns of within-host genetic diversity in SARS-CoV-2<ul><li>Alignment was performed using the ARTIC Illumina <b>nextflow</b> pipeline available from <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link> (<xref rid="bib5" ref-type="bibr"><u>Bull, 2020</u></xref>; copy archived at <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:a68e5dd15f72a1308b9549a4801f1e998aec7cf1;origin=https://github.com/connor-lab/ncov2019-artic-nf;visit=swh:1:snp:8abba37d67f611a88adc04cad809c1f3260926be;anchor=swh:1:rev:8af5152cf7107c3a369b996f5bad3473d329050c" ext-link-type="uri">swh:1:rev:8af5152cf7107c3a369b996f5bad3473d329050c</ext-link>).<li> [Bull, 2020](          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        ">          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        </a> <b>(1)</b> Patterns of within-host genetic diversity in SARS-CoV-2<ul><li>'Alignment was performed using the ARTIC Illumina <b>nextflow</b> pipeline available from {https://github.com/connor-lab/ncov2019-artic-nf}.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        ">          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        </a> <b>(1)</b> Patterns of within-host genetic diversity in SARS-CoV-2<ul><li> The <b>Nextflow</b> mapping and assembly pipeline is published on<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mclaugsf/mgc-public/tree/master/AOAC_TYM_ERV" ext-link-type="uri">GitHub</ext-link>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        ">          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        </a> <b>(2)</b> Patterns of within-host genetic diversity in SARS-CoV-2<ul><li> It is written in <b>Nextflow</b>, a domain-specific language and workflow manager that allows easy implementation, redistribution, and scalability of complex workflows across every Unix-based operating system; ranging from a desktop machine to cloud computing and HPC clusters.<li>In conclusion, we provide a transposition of the UCSC lift over pipeline within the <b>Nextflow</b> language, together with the necessary containers to run the analyses, allowing an easy, streamlined implementation in any Unix-based system.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        ">          <person-group person-group-type="author"><name><surname>Bull</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2020">2020</year>          <data-title>A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools with a focus on ncov2019</data-title>          <source>Github</source>          <version designator="AGPL-3.0">AGPL-3.0</version>          <ext-link xlink:href="https://github.com/connor-lab/ncov2019-artic-nf" ext-link-type="uri">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>        </a> <b>(3)</b> Patterns of within-host genetic diversity in SARS-CoV-2<ul><li> <b>Nextflow</b> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) or Snakemake (<xref rid="B19" ref-type="bibr"><u>19</u></xref>)), (iii) is straightforward to install via static binaries available for most platforms and architectures, and (iv) provides a public API accessible to developers wanting to manipulate phylogenetic trees and multiple sequence&#xA0;alignments in Go.<li>The Gotree/Goalign toolkit is already used in several public workflows such as Grapevine (the phylogenetic pipeline for the COG-UK project <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/COG-UK/grapevine" ext-link-type="uri">https://github.com/COG-UK/grapevine</ext-link>), ARTIC-EBOV (The ARTIC Ebola virus phylogenetic analysis protocol <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/artic-network/artic-ebov" ext-link-type="uri">https://github.com/artic-network/artic-ebov</ext-link>), KOVID-TREES-NF (a <b>Nextflow</b> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) workflow for SARS-CoV-2 phylogenetics <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/MDU-PHL/kovid-trees-nf" ext-link-type="uri">https://github.com/MDU-PHL/kovid-trees-nf</ext-link>), bioconvert (a generic bioinformatic file format converter <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/bioconvert/bioconvert" ext-link-type="uri">https://github.com/bioconvert/bioconvert</ext-link>), and in phylogenetic studies (<xref rid="B20" ref-type="bibr"><u>20&#x2013;26</u></xref>).<li>The Gotree/Goalign toolkit has been developed to be easily integrated in phylogenetic workflows, using main workflow managers such as <b>Nextflow</b> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) or Snakemake (<xref rid="B19" ref-type="bibr"><u>19</u></xref>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [20&#x2013;26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31440543">31440543</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34376657">34376657</a> <b>(2)</b> The evolution of hematopoietic cells under cancer therapy<ul><li> Then, the fastq files from phs000159, EGAD00001005028, and the inhouse cohort were processed in a uniform manner using the sarek<sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup> pipeline implemented within <b>nextflow</b>-core (<b>nextflow</b> version 19.10)<sup><xref ref-type="bibr" rid="CR61"><u>61</u></xref></sup>.<li> [60](Garcia, M. et al. Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants. <italic>F1000Research</italic><bold>9</bold>, (2020).) [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34370660">34370660</a> <b>(1)</b> Post-weaning shifts in microbiome composition and metabolism revealed by over 25&#x200A;000 pig gut metagenome-assembled genomes<ul><li>In order to manage the processing of the data in the high performance computing environment (sequence data processing, pooled assembly, binning, MAGs quality assessment), <b>Nextflow</b> [<xref rid="R46" ref-type="bibr"><u>46</u></xref>] was used.<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34439659">34439659</a> <b>(3)</b> Structural Connectivity Alterations in Operculo-Insular Epilepsy<ul><li>The RNA-seq data were processed using a <b>Nextflow</b> pipeline v1.1 (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/Dowell-Lab/RNAseq-Flow" ext-link-type="uri">https://github.com/Dowell-Lab/RNAseq-Flow</ext-link>).<li>All ChIP-seq data were processed (mapped and quality checked) using a <b>Nextflow</b> pipeline, ChIP-Flow v1.0 (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/Dowell-Lab/ChIP-Flow" ext-link-type="uri">https://github.com/Dowell-Lab/ChIP-Flow</ext-link>).<li>Two biological replicates for each treatment (DMSO/Nutlin-3a) were generated for each of the 3 cell lines were processed using a <b>Nextflow</b> pipeline for nascent data (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://doi.org/10.5281/zenodo.2641755" ext-link-type="uri">https://doi.org/10.5281/zenodo.2641755</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34439659">34439659</a> <b>(6)</b> Structural Connectivity Alterations in Operculo-Insular Epilepsy<ul><li> Where a workflow language is used, the execution engine will typically have an option for producing a report on resource usage for each step of the pipeline&#x2014;for example, <b>Nextflow</b> offers both resource reporting, producing a detailed HTML document (using the <italic toggle="yes">-with-report</italic> option) and tracing of the individual workflow steps (using the <italic toggle="yes">-with-trace</italic> option), which can provide invaluable information on bottlenecks, as well as rate-limiting steps.<li> Items reported in the <b>Nextflow</b> execution report are CPU usage, memory usage, job duration, and I/O reads/writes.<li>Workflow languages, such as <b>Nextflow</b>, often allow the explicit specification of a queue for a particular pipeline step (i.e., process/code blocks), as well as the maximum parallelism for the step.<li> For example, when combined with a workflow language, such as <b>Nextflow</b>, the reproducibility of pipelines can be assured by pinning down specific pipeline steps to use specific versions of tools as defined by designated Conda environments.<li> The required packages can then be loaded, or if necessary, downloaded on the fly by <b>Nextflow</b> and cached for subsequent use, thereby allowing workflow pipelines to be shared collaboratively with the confidence that correct packages will be used for execution.<li> A solution for this is to use a workflow language such as Workflow Definition Language (WDL) [<xref rid="pcbi.1009207.ref013" ref-type="bibr"><u>13</u></xref>], Common Workflow Language (CWL) [<xref rid="pcbi.1009207.ref014" ref-type="bibr"><u>14</u></xref>], <b>Nextflow</b> [<xref rid="pcbi.1009207.ref015" ref-type="bibr"><u>15</u></xref>], or Snakemake, an extension to the Python language [<xref rid="pcbi.1009207.ref016" ref-type="bibr"><u>16</u></xref>].<li> [13](Broad Institute. WDL (Workflow Definition Language) specification and documentation. 2016. Available from: <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://software.broadinstitute.org/wdl/documentation/spec" ext-link-type="uri">https://software.broadinstitute.org/wdl/documentation/spec</ext-link>.) [14](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Andeer</surname><given-names>R</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Valls Guimera</surname><given-names>R</given-names></name>, <etal>et al</etal>. <source>Common Workflow Language</source>, Draft 3. <year>2016</year>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](Python. Snakemake documentation. 2021. Available from: <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://snakemake.readthedocs.io/en/stable/index.html" ext-link-type="uri">https://snakemake.readthedocs.io/en/stable/index.html</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34354145">34354145</a> <b>(1)</b> Yippee like 4 (<italic>Ypel4</italic>) is essential for normal mouse red blood cell membrane integrity<ul><li> Fastq files were processed using the nf-core <b>Nextflow</b> pipeline for RNAseq <italic>nf-core/rnaseq</italic> v1.4.2<sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref>,<xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34342226">34342226</a> <b>(1)</b> ppx: Programmatic access to proteomics data repositories<ul><li> The adoption of workflow engines such as Snakemake [<xref rid="R26" ref-type="bibr"><u>26</u></xref>], <b>Nextflow</b> [<xref rid="R27" ref-type="bibr"><u>27</u></xref>], and Cromwell [<xref rid="R28" ref-type="bibr"><u>28</u></xref>] provide frameworks to run analysis pipelines reproducibly and orchestrate the necessary environments or containers.<li> [26](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> and <name><surname>Rahmann</surname><given-names>S</given-names></name> &#x201C;<article-title>Snakemake&#x2013;a Scalable Bioinformatics Workflow Engine</article-title>.&#x201D; eng. In: <source>Bioinformatics (Oxford, England)</source><volume>28</volume>.<issue>19</issue> (<month>10</month>. <year>2012</year>), pp. <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) [27](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <etal/> &#x201C;<article-title>Nextflow Enables Reproducible Computational Workflows</article-title>.&#x201D; eng. In: <source>Nature Biotechnology</source><volume>35</volume>.<issue>4</issue> (<month>4</month>. <year>2017</year>), pp. <fpage>316</fpage>&#x2013;<lpage>319</lpage>.) [28](<name><surname>Voss</surname><given-names>K</given-names></name>, <name><surname>Auwera</surname><given-names>GVD</given-names></name>, and <name><surname>Gentry</surname><given-names>J</given-names></name><article-title>Full-Stack Genomics Pipelining with GATK4 + WDL + Cromwell [Version 1; Not Peer Reviewed]</article-title>. Vol. <volume>6</volume>. <source>F1000Research</source>, <year>2017</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34290042">34290042</a> <b>(3)</b> Enhancers with tissue-specific activity are enriched in intronic regions<ul><li><b>Nextflow</b><sup><xref rid="bibr44-11779322211035921" ref-type="bibr"><u>44</u></xref></sup>.<li> nf-core genomic pipelines arewritten in <b>Nextflow</b>.<li>By containerizing applications, the reproducibility and portability areensured, allowing for sophisticated workflow management solutions, such as<b>Nextflow</b>, to accelerate the generation of portable and scalablepipeline.<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34394197">34394197</a> <b>(5)</b> poreCov-An Easy to Use, Fast, and Robust Workflow for SARS-CoV-2 Genome Reconstruction <italic>via</italic> Nanopore Sequencing<ul><li>We use <b>Nextflow</b> (<xref rid="B4" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) as a workflow manager giving poreCov the capabilities to be executed on most computational infrastructures from a laptop up to a compute cluster and cloud computing solutions.<li> This leaves <b>Nextflow</b> and either Docker or Singularity as the only dependencies needed to be installed for poreCov.<li> If these dependencies are available, an initial installation and further updates of the workflow and the included tools can be performed with a single simple command (&#x201C;<b>nextflow</b> pull replikation/poreCov&#x201D;).<li><bold>Command flag variations (<b>nextflow</b> run poreCov &#x2026;)</bold>.<li> We ensure a high degree of reproducibility and stability by utilizing <b>Nextflow</b> and virtualizing each tool through containers.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34320313">34320313</a> <b>(1)</b> Evaluation ofSpin Columns for Human Plasma Depletionto Facilitate MS-Based Proteomics Analysis of Plasma<ul><li>Orbitrap raw MS/MSfiles were convertedto mzML format using msConvert from the ProteoWizard tool suite.<sup><xref ref-type="bibr" rid="ref17"><u>17</u></xref></sup> Spectra were searched using the ddamsproteomicspipeline (v1.5) (DOI 10.5281/zenodo.3714589), which is a <b>Nextflow</b><sup><xref ref-type="bibr" rid="ref18"><u>18</u></xref></sup> pipeline runningMSGF+ (2020.03.12),<sup><xref ref-type="bibr" rid="ref19"><u>19</u></xref></sup> Percolator (v3.4).<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23051804">23051804</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25358478">25358478</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25358478">25358478</a> <b>(1)</b> Evaluation ofSpin Columns for Human Plasma Depletionto Facilitate MS-Based Proteomics Analysis of Plasma<ul><li> ChIP-Seq and mRNA-seq bioinformatic analysis pipelines were based on <b>Nextflow</b> (<xref rid="bib20" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and the nf-core framework (<xref rid="bib25" ref-type="bibr"><u>Ewels et al., 2020</u></xref>) are available at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/Gregor-Mendel-Institute/RKP2021-CMT3" ext-link-type="uri">https://github.com/Gregor-Mendel-Institute/RKP2021-CMT3</ext-link>, [copy archived at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://archive.softwareheritage.org/swh:1:dir:7589b4e193c639b8e5bf09809d389a39c1e653aa;origin=https://github.com/Gregor-Mendel-Institute/RKP2021-CMT3;visit=swh:1:snp:8e4b9655914607e3abe7868ed585c077572dcaf1;anchor=swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f" ext-link-type="uri">swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f</ext-link>&#xA0;(<xref rid="bib73" ref-type="bibr"><u>Papareddy, 2021</u></xref>)].&#xA0;.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Papareddy, 2021](          <person-group person-group-type="author"><name><surname>Papareddy</surname><given-names>R</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <data-title>RKP2021-CMT3</data-title>          <source>Software Heritage&#xA0;</source>          <version designator="swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f">swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f</version>          <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://archive.softwareheritage.org/swh:1:dir:7589b4e193c639b8e5bf09809d389a39c1e653aa;origin=https://github.com/Gregor-Mendel-Institute/RKP2021-CMT3;visit=swh:1:snp:8e4b9655914607e3abe7868ed585c077572dcaf1;anchor=swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:7589b4e193c639b8e5bf09809d389a39c1e653aa;origin=https://github.com/Gregor-Mendel-Institute/RKP2021-CMT3;visit=swh:1:snp:8e4b9655914607e3abe7868ed585c077572dcaf1;anchor=swh:1:rev:89d7e2ea78af1969bb161640baed09296ed2485f</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34294043">34294043</a> <b>(1)</b> Correction to: SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor&#x2011;powered RNA&#x2011;seq analyses<ul><li> In order of submission, the name of each <b>Nextflow</b> process is printed, along with (1) the working directory: where all relevant files are present, (2) the exit code: a standard indication of whether the process succeeded or how it failed, (3) a list of the specific commands run during the given process.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34294043">34294043</a> <b>(1)</b> Correction to: SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor&#x2011;powered RNA&#x2011;seq analyses<ul><li> Spectra were searched using MSGF+ and Percolator within a proteomics analysis pipeline developed in <b>Nextflow</b> (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/lehtiolab/ddamsproteomics" ext-link-type="uri">https://github.com/lehtiolab/ddamsproteomics</ext-link>, vs1.5) against the human protein database Ensembl 99.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34294043">34294043</a> <b>(1)</b> Correction to: SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor&#x2011;powered RNA&#x2011;seq analyses<ul><li> Finally, the <italic toggle="yes">dailyFlow</italic> repository uses the <b>Nextflow</b> workflow engine<sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup> to clean the data, orchestrate 3200+ model runs within three supported execution environments (local, HPC, cloud), and export the results for research use and for web consumption.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34280125">34280125</a> <b>(1)</b> Whole-genome sequencing of single circulating tumor cells from neuroendocrine neoplasms<ul><li>Bulk sequencing data were processed with the <b>nextflow</b> Sarek v2.3.FIX1 pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/UCL-BLIC/Sarek_v2.3.FIX1">https://github.com/UCL-BLIC/Sarek_v2.3.FIX1</ext-link>) following GATK best practices.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34267305">34267305</a> <b>(1)</b> Function-adaptive clustered nanoparticles reverse <italic>Streptococcus mutans</italic> dental biofilm and maintain microbiota balance<ul><li>Demultiplexed sequence data were submitted to the dada2 high-throughput <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR94"><u>94</u></xref></sup> pipeline on the Biocluster at the Carl R.<li> [94](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34267208">34267208</a> <b>(8)</b> Extensive transcriptional and chromatin changes underlie astrocyte maturation in vivo and in culture<ul><li>Reads from RNA-Seq fastq files were aligned to the mm10 genome and quantified using the crickbabs/BABS-RNASeq <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/crickbabs/BABS-RNASeq">https://github.com/crickbabs/BABS-RNASeq</ext-link>; git commit id: 335ce47db079d6cc2a7f82f4b762620c4f7f27e2).<li> The pipeline was run using <b>nextflow</b> version 0.30.2<sup><xref ref-type="bibr" rid="CR69"><u>69</u></xref></sup> with the command &#x2018;<b>nextflow</b> run main.nf -params-file params.yml&#x2019;.<li> Published in vitro astrocyte gene expression data from other models<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref>,<xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup> was re-analysed with the <b>nextflow</b> pipeline above, then merged with the datasets of this study and analysed as above.<li>Reads from ATAC-Seq fastq files were aligned to the mm10 genome, peaks mapped and quantified using the crickbabs/BABS-ATACSeqPE <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/">https://github.com/</ext-link> crickbabs/BABS-ATACSeqPE; git commit id: 22edccf72855d42e6692a27385cf50666c8f391c; now superseded; a newer version is available as part of the nf-core project: nf-core/atacseq pipeline (10.5281/zenodo.2634132<sup><xref ref-type="bibr" rid="CR72"><u>72</u></xref></sup>).<li> The pipeline was run using <b>nextflow</b> version 0.30.2<sup><xref ref-type="bibr" rid="CR69"><u>69</u></xref></sup> with the command &#x2018;<b>nextflow</b> run main.nf --design../design.csv --genome mm10 -profile conda --outdir../results/&#x2019;.<li> <b>nextflow</b> pipeline<sup><xref ref-type="bibr" rid="CR72"><u>72</u></xref></sup> (version 1.2.1).<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30290178">30290178</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28462931">28462931</a>) [72](Ewels, P. A. et al. The nf-core framework for community-curated bioinformatics pipelines. <italic>Nat. Biotechnol</italic>. 10.1038/s41587-020-0439-x (2020).) [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [72](Ewels, P. A. et al. The nf-core framework for community-curated bioinformatics pipelines. <italic>Nat. Biotechnol</italic>. 10.1038/s41587-020-0439-x (2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34267211">34267211</a> <b>(1)</b> Effective variant filtering and expected candidate variant yield in studies of rare human disease<ul><li>For this study, we used <b>nextflow</b><sup><xref ref-type="bibr" rid="CR27"><u>27</u></xref></sup> to run slivar with and without low-complexity regions for each cohort and tools.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34335529">34335529</a> <b>(2)</b> Comparative Analysis of Fecal Microbiomes From Wild Waterbirds to Poultry, Cattle, Pigs, and Wastewater Treatment Plants for a Microbial Source Tracking Approach<ul><li> SAMBA was developed using the <b>NextFlow</b> workflow manager (<xref rid="B17" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and built around three main parts: data integrity checking, bioinformatics processes, and statistical analyses.<li>Our bioinformatic analysis used the <b>Nextflow</b>-based SAMBA pipeline<sup><xref ref-type="fn" rid="footnote7"><u>7</u></xref></sup>, which is based on the approach of ASVs rather than OTUs.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [7](<a href="https://www.frontiersin.org/articles/10.3389/fmicb.2021.697553/full#supplementary-material">https://www.frontiersin.org/articles/10.3389/fmicb.2021.697553/full#supplementary-material</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34308377">34308377</a> <b>(1)</b> FACS-based isolation of fixed mouse neuronal nuclei for ATAC-seq and Hi-C<ul><li> Sequencing reads were processed with the ATAC-seq <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/atacseq" id="intref0015">https://nf-co.re/atacseq</ext-link>) (<xref rid="bib5" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>) with default parameters, except for the peak calling option.<li> [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a> <b>(1)</b> FACS-based isolation of fixed mouse neuronal nuclei for ATAC-seq and Hi-C<ul><li> Raw sequencing data were processed with the nf-villumina <b>Nextflow</b> (<xref rid="B12" ref-type="bibr"><u>12</u></xref>) workflow (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/CFIA-NCFAD/nf-villumina" ext-link-type="uri">https://github.com/CFIA-NCFAD/nf-villumina</ext-link>) for quality control, taxonomic classification, and <italic toggle="yes">de novo</italic> assembly (<xref rid="B13" ref-type="bibr"><u>13</u></xref>).<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33177604">33177604</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34240169">34240169</a> <b>(1)</b> Genome Analysis Traces Regional Dispersal of Rice in Taiwan and Southeast Asia<ul><li> Briefly, we used a <b>Nextflow</b> v.0.25.1.4460 pipeline (<ext-link xlink:href="https://github.com/grafau/NextGatkSNPs" ext-link-type="uri">https://github.com/grafau/NextGatkSNPs, last accessed July 14, 2021</ext-link>) to produce the final set of SNPs via the following steps outlined below.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34225648">34225648</a> <b>(1)</b> HOME-BIO (sHOtgun MEtagenomic analysis of BIOlogical entities): a specific and comprehensive pipeline for metagenomic shotgun sequencing data analysis<ul><li> Furthermore, this pipeline is implemented in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] and all the applications needed to execute the workflow are containerized in Singularity [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>].<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34734190">34734190</a> <b>(1)</b> Chemotherapy induces canalization of cell state in childhood B-cell precursor acute lymphoblastic leukemia<ul><li>Bulk RNAseq samples were processed using a <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/UCL-BLIC/rnaseq">https://github.com/UCL-BLIC/rnaseq</ext-link>) which runs FastQC, TrimGalore!, STAR and featureCounts to clean, map the reads to the human reference GRCh38.p12 and quantify gene expression using the Ensembl v96 annotations.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34210356">34210356</a> <b>(7)</b> CLIMB-COVID: continuous integration supporting decentralised sequencing for SARS-CoV-2 genomic surveillance<ul><li>To assist with the on-boarding of new sites, including those with limited bioinformatics support we also built a reproducible <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>) that enables the processing of data for sites following the ARTIC sequencing protocols [<xref ref-type="bibr" rid="CR6"><u>6</u></xref>].<li> Elan (<ext-link ext-link-type="uri" xlink:href="https://github.com/SamStudio8/elan-<b>nextflow</b>/">https://github.com/SamStudio8/elan-<b>nextflow</b>/</ext-link>) is an open-source pipeline built with the <b>NextFlow</b> workflow language [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>].<li> Datapipe is a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] pipeline that combines a downsampled set of non-UK SARS-CoV-2 sequences from GISAID with the complete set of COG-UK sequences that have passed basic quality control.<li>We recently developed a mechanism to automate submission of consensus sequences to the INSDC via EMBL-EBI, leveraging the ENA webin client (<ext-link ext-link-type="uri" xlink:href="https://github.com/SamStudio8/elan-ena-<b>nextflow</b>">https://github.com/SamStudio8/elan-ena-<b>nextflow</b></ext-link>).<li> [6](          <person-group person-group-type="author">            <name>              <surname>Tyson</surname>              <given-names>JR</given-names>            </name>            <name>              <surname>James</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Stoddart</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Sparks</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Wickenhagen</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>G</given-names>            </name>            <etal/>          </person-group>          <source>Improvements to the ARTIC multiplex PCR method for SARS-CoV-2 genome sequencing using nanopore</source>          <year>2020</year>          <fpage>2020.09.04.283077</fpage>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34140313">34140313</a> <b>(18)</b> Enzymatic methyl sequencing detects DNA methylation at single-base resolution from picograms of DNA<ul><li> By using <b>Nextflow</b> workflow technology &#xA0;[<xref rid="bib14" ref-type="bibr"><u>13</u></xref>] in combination with Singularity software container technology&#xA0;[<xref rid="bib15" ref-type="bibr"><u>14</u></xref>] (Methods), BIGwas can be started with a single command (Fig.&#xA0;<xref rid="fig1" ref-type="fig"><u>1</u></xref>), with no need to manually install a full software execution environment or various software packages.<li> Using the <b>Nextflow</b> engine, all process executions are further recorded in a single trace file.<li>We use the free and open-source <b>Nextflow</b>&#xA0;[<xref rid="bib14" ref-type="bibr"><u>13</u></xref>] pipeline software for easy parallelization of processes, HPC-based development, and pipeline control.<li><b>Nextflow</b>.<li><b>Nextflow</b>, a tool specifically designed to run bioinformatics applications, therefore provides a suitable environment to facilitate the parallelism of processes in all workflows.<li> mode and <b>Nextflow</b>&#x2019;s job splitting feature to achieve fast processing of large input files.<li> <b>Nextflow</b> takes care of the creation, workspace separation, and merging of all jobs, avoids user-defined error-prone synchronization of all jobs, and oversees the merging of results from complex analysis pipelines.<li>Another major advantage of <b>Nextflow</b> is the already integrated handling and agnostic use of HPC cluster resources (i.e., it is able to work under different environments).<li> With <b>Nextflow</b>, it is possible to write parallel and concurrent pipeline workflows without having knowledge (and thus knowing all dependencies) of a specific cluster environment, such as SGE, SLURM, or PBS/TORQUE.<li> The deep integration of different storage solutions like ECS object storage or S3 buckets and container tools such as Docker and Singularity make <b>Nextflow</b> a first choice for a portable, widely available platform, with the only requirement of a working Java installation.<li>Our general <b>Nextflow</b> script code base is hosted on GitHub.<li> For each benchmark data set consisting of a different number of GWAS samples and genetic variants (Tables&#xA0;<xref rid="tbl2" ref-type="table"><u>2</u></xref> and&#xA0;<xref rid="tbl3" ref-type="table"><u>3</u></xref>), we used <b>Nextflow</b>&#x2019;s built-in reporting function to determine the elapsed wall-clock time.<li>We have introduced BIGwas, a portable (i.e., Singularity container-based), fully automated GWAS QC and association pipeline in <b>Nextflow</b> that can be executed with a single command without manual installation by the user.<li>To enable a reproducible analysis of 1 million GWAS samples on any common compute platform, we have combined various code optimizations of our existing GWAS software with <b>Nextflow</b> workflow and Singularity container technology.<li> Our <b>Nextflow</b> pipelines are inherently parallel and can be transparently scaled up or down, and they provide an abstraction layer between the logic of the pipeline and the execution layer (i.e., they can be executed on multiple platforms without changing the source code).<li> With the Singularity container technology, we have packed all our <b>Nextflow</b> workflows, software, and libraries, as well as publicly available reference data, into a single Singularity container that runs on any standard computer platform without manual download or installation by the user.<li> In the future, we aim to test the usability of further genome-wide analysis approaches, such as gene-gene interaction tests, for use with <b>Nextflow</b> and Singularity technology.<li>Programming language: <b>Nextflow</b>, Bash, Python, Perl, R, Rust.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33564759">33564759</a> <b>(1)</b> Extensive recombination-driven coronavirus diversification expands the pool of potential pandemic pathogens<ul><li>To investigate possible recombination-driven diversity among these viruses we developed IDPlot, which incorporates several distinct analysis steps into a single <b>Nextflow</b> workflow [<xref rid="R35" ref-type="bibr"><u>35</u></xref>] and generates a comprehensive HTML report to facilitate interpretation and downstream analysis.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34180774">34180774</a> <b>(1)</b> Adaptation of the emerging pathogenic yeast <italic>Candida auris</italic> to high caspofungin concentrations correlates with cell wall changes<ul><li>The results were analyzed with a RNA-seq pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/BU-ISCIII/rnaseq-nf">https://github.com/BU-ISCIII/rnaseq-nf</ext-link>) written in <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</ext-link>) based on the nf-core (<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/">https://nf-co.re/</ext-link>) previously written RNA-seq pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/rnaseq">https://github.com/nf-core/rnaseq</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34170846">34170846</a> <b>(1)</b> COVID-19 Transmission among Healthcare Workers at a Quarantine Facility in Thailand: Genomic and Outbreak Investigations<ul><li>Sequencing reads were analyzed using the COG-UK <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34172000">34172000</a> <b>(3)</b> MicroPIPE: validating an end-to-end workflow for high-quality complete bacterial genome construction<ul><li> The pipeline is written in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] and the dependencies are packaged into Singularity [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>] container images available through the Docker Hub and Quay.io BioContainers repositories.<li> The bioinformatics workflow manager <b>Nextflow</b> allows users to run the pipeline locally or using common High-Performance Computing schedulers.<li><bold>Programming language:</bold> <b>Nextflow</b>, Python.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34171097">34171097</a> <b>(2)</b> Whole-genome Sequencing of Follicular Thyroid Carcinomas Reveal Recurrent Mutations in MicroRNA Processing Subunit <italic toggle="yes">DGCR8</italic><ul><li> Sarek uses <b>nextflow</b> DSL, in this study version 19.03.0-SNAPSHOT, as a computational framework.<li> National Genomic Infrastructure performed read alignment and extracted read counts by using <b>nextflow</b>/rnaseq pipeline (available in the GitHub repository <ext-link xlink:href="https://github.com/nf-core/rnaseq" ext-link-type="uri">https://github.com/nf-core/rnaseq</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34171097">34171097</a> <b>(1)</b> Whole-genome Sequencing of Follicular Thyroid Carcinomas Reveal Recurrent Mutations in MicroRNA Processing Subunit <italic toggle="yes">DGCR8</italic><ul><li>All corresponding FASTQ files for ChIP and controls were downloaded and used as input into the <b>nextflow</b> pipeline nf-core/chipseq (using version 1.1.0).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34162869">34162869</a> <b>(1)</b> COVseq is a cost-effective workflow for mass-scale SARS-CoV-2 genomic surveillance<ul><li> Following this, we processed the samples using a <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR24"><u>24</u></xref></sup> (version 20.10.0) based analysis pipeline from nf-core<sup><xref ref-type="bibr" rid="CR25"><u>25</u></xref></sup> called viralrecon<sup><xref ref-type="bibr" rid="CR26"><u>26</u></xref></sup> (version 1.1.0).<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [26](Patel, H. et al. <italic>nf-core/viralrecon: nf-core/viralrecon v1.1.0&#x2014;Steel Pangolin</italic>. (Zenodo, 2020). 10.5281/zenodo.3905178.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34239974">34239974</a> <b>(13)</b> Compi: a framework for portable and reproducible pipelines<ul><li> Furthermore, command-line based applications such as <b>Nextflow</b> (<xref rid="ref-2" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), Snakemake (<xref rid="ref-6" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>), or SciPipe (<xref rid="ref-7" ref-type="bibr"><u>Lampa et al., 2019</u></xref>), provide feature-rich workflow engines oriented to bioinformaticians with medium-to-high programming skills.<li>Unlike Snakemake, the <b>Nextflow</b> and SciPipe tools allow dynamic scheduling, that is, the ability to change the pipeline structure dynamically to schedule a different number of tasks based on the results of a previous step or any other parameter.<li>Compi, Snakemake, and <b>Nextflow</b> are language independent, allowing external scripts written in any programming language to be invoked.<li>Similar to Compi runners, <b>Nextflow</b> defines executors, which are the components that determine where a pipeline process runs and its execution is supervised.<li> In this sense, <b>Nextflow</b> provides built-in support to run individual tasks or complete pipelines using Docker, Singularity or Podman images.<li> In the case of Docker, <b>Nextflow</b> is able to mount the input and output paths automatically, since it is aware of the files needed by tasks.<li> <b>Nextflow</b> provides a log command that returns useful information about a specific pipeline execution, and incorporates a &#x201C;<italic>&#x2013;with-report</italic>&#x201D; option in the run command that instructs <b>Nextflow</b> to generate an HTML execution report that includes many useful metrics about a specific workflow execution.<li><b>Nextflow</b> has nf-core, an environment with a dual purpose: to provide an online repository of <b>Nextflow</b> pipelines and to provide a command-line tool to interact with the repository and manage the execution of the hosted pipelines.<li>After these considerations, and given the choices available, we believe that Compi may be a reasonable choice for researchers with CLI skills looking to create medium complexity pipelines without requiring them to learn new programming languages (e.g., to learn the <b>Nextflow</b> DSL or Go for SciPipe pipeline development).<li> Finally, to illustrate the way of creating Compi pipelines and the main differences with other workflow management systems, we have implemented a <b>Nextflow</b> example pipeline in Compi (<xref ref-type="supplementary-material" rid="supp-2"><u>File S2</u></xref>).<li>Implementation of a <b>Nextflow</b> example pipeline in Compi.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Lampa et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a> <b>(1)</b> Compi: a framework for portable and reproducible pipelines<ul><li> This pipeline was executed using the <b>Nextflow</b> [<xref rid="pgen.1009603.ref100" ref-type="bibr"><u>100</u></xref>] engine for computational workflows and comprises several processing steps [<xref rid="pgen.1009603.ref101" ref-type="bibr"><u>101</u></xref>].<li> [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [101](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22743226">22743226</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22743226">22743226</a> <b>(5)</b> Compi: a framework for portable and reproducible pipelines<ul><li> The final list of annotated variants and the accompanying SnpEff output files are available here (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mclaugsf/mgc-public/tree/master/f1000_10-281/<b>nextflow</b>/annotated-variants" ext-link-type="uri">https://github.com/mclaugsf/mgc-public/tree/master/f1000_10-281/<b>nextflow</b>/annotated-variants</ext-link>).<li> The gff3 file that was used to perform the SnpEff annotation is available for download (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mclaugsf/mgc-public/blob/master/f1000_10-281/gff/P-Envy-05-25-2021.gff3.gz" ext-link-type="uri">https://github.com/mclaugsf/mgc-public/blob/master/f1000_10-281/gff/P-Envy-05-25-2021.gff3.gz</ext-link>) as well as Dockerized workflows written in <b>nextflow</b> used to perform the mapping, variants calling and annotation analysis (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mclaugsf/mgc-public/tree/master/f1000_10-281/<b>nextflow</b>" ext-link-type="uri">https://github.com/mclaugsf/mgc-public/tree/master/f1000_10-281/<b>nextflow</b></ext-link>).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34178042">34178042</a> <b>(7)</b> Omics in a Digital World: The Role of Bioinformatics in Providing New Insights Into Human Aging<ul><li> In our work we have adopted one of such DSL, namely <b>Nextflow</b> (<xref rid="B19" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), which is certainly having a major impact in data analysis for life sciences, and aging research as well.<li> A very original effort has been made around the use of workflow managers, and a community in particular, built around the use of <b>Nextflow</b>, is worth of notice: the nf-core community (<xref rid="B24" ref-type="bibr"><u>Ewels et al., 2020</u></xref>).<li> One could therefore use workflow languages like <b>Nextflow</b>, in order to process raw data, perform initial data integration and network-based analysis, add classical pathway-based analyses (<xref rid="B98" ref-type="bibr"><u>Zhao et al., 2018</u></xref>), and then follow-up with appropriate tools designed for modeling biological systems.<li><b>Nextflow</b>.<li>The best way to approach <b>Nextflow</b> is to look at the extensive material produced or cataloged by the nf-core community.<li>Tutorials page: <ext-link ext-link-type="uri" xlink:href="https://nfco.re/usage/<b>nextflow</b>">https://nfco.re/usage/<b>nextflow</b></ext-link><break/>Bite-sized webinars (under events): <ext-link ext-link-type="uri" xlink:href="https://nf-co.re/events">https://nf-co.re/events</ext-link>.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Zhao et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30643820">30643820</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34110280">34110280</a> <b>(3)</b> rMAP: the Rapid Microbial Analysis Pipeline for ESKAPE bacterial group whole-genome sequence data<ul><li><b>Nextflow</b>.<li> ASA3P [<xref rid="R43" ref-type="bibr"><u>43</u></xref>] and Bactopia [<xref rid="R44" ref-type="bibr"><u>44</u></xref>] required a bioinformatics-competent user for operation, since they are written in complex languages, namely, Groovy and <b>Nextflow</b>, respectively.<li>As a significant limitation, rMAP is coded exclusively in Bash and is not implemented within a modern workflow language manager, such as Snakemake or <b>Nextflow</b>.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32134915">32134915</a>) [44](          <person-group person-group-type="author">            <name>              <surname>Petit</surname>              <given-names>RA</given-names>            </name>            <name>              <surname>Read</surname>              <given-names>TD</given-names>            </name>          </person-group>          <article-title>Bactopia: a flexible pipeline for complete analysis of bacterial genomes</article-title>          <source>bioRxiv</source>          <year>2020:2020.02.28.969394</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34168993">34168993</a> <b>(2)</b> DIVIS: Integrated and Customizable Pipeline for Cancer Genome Sequencing Analysis and Interpretation<ul><li> Workflow management software such as Snakemake (<xref rid="B23" ref-type="bibr"><u>23</u></xref>), Bpipe (<xref rid="B24" ref-type="bibr"><u>24</u></xref>), Ruffus (<xref rid="B25" ref-type="bibr"><u>25</u></xref>), <b>Nextflow</b> (<xref rid="B26" ref-type="bibr"><u>26</u></xref>), and Galaxy (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) sufficiently satisfy this requirement by automatically generating analysis scripts through graphical operation, which simplifies human intervention.<li> <b>Nextflow</b> also describes a workflow by using its DSL and generates scalable and reproducible scientific workflows by using software containers (<xref rid="B26" ref-type="bibr"><u>26</u></xref>).<li> [23](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake&#x2014;a Scalable Bioinformatics Workflow Engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>19</issue>):<page-range>2520&#x2013;2</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) [24](<person-group person-group-type="author"><name><surname>Sadedin</surname><given-names>SP</given-names></name><name><surname>Pope</surname><given-names>B</given-names></name><name><surname>Oshlack</surname><given-names>A</given-names></name></person-group>. <article-title>Bpipe: A Tool for Running and Managing Bioinformatics Pipelines</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>11</issue>):<page-range>1525&#x2013;6</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts167</pub-id>) [25](<person-group person-group-type="author"><name><surname>Goodstadt</surname><given-names>L</given-names></name></person-group>. <article-title>Ruffus: A Lightweight Python Library for Computational Pipelines</article-title>. <source>Bioinformatics</source> (<year>2010</year>) <volume>26</volume>(<issue>21</issue>):<page-range>2778&#x2013;9</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/btq524</pub-id>) [26](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow Enables Reproducible Computational Workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>(<issue>4</issue>):<page-range>316&#x2013;9</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [26](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow Enables Reproducible Computational Workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>(<issue>4</issue>):<page-range>316&#x2013;9</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34149808">34149808</a> <b>(2)</b> A Scalable Strand-Specific Protocol Enabling Full-Length Total RNA Sequencing From Single Cells<ul><li> A <b>Nextflow</b> script orchestrating these operations across folders of fastqs is available at <ext-link ext-link-type="uri" xlink:href="https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/single_cell_rna/browse">https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/single_cell_rna/browse</ext-link> where the R scripts used to make related figures can also be found.<li> Circular RNA detection was performed with CIRIquant (<xref rid="B54" ref-type="bibr"><u>Zhang et al., 2020</u></xref>) for which a <b>Nextflow</b> script and associated R notebook are available at <ext-link ext-link-type="uri" xlink:href="https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/single_cell_rna/browse">https://svn.bcgsc.ca/bitbucket/projects/RCORBETT/repos/single_cell_rna/browse</ext-link>.<li> [Zhang et al., 2020](<person-group person-group-type="author"><name><surname>Zhang</surname><given-names>J.</given-names></name><name><surname>Chen</surname><given-names>S.</given-names></name><name><surname>Yang</surname><given-names>J.</given-names></name><name><surname>Zhao</surname><given-names>F.</given-names></name></person-group> (<year>2020</year>). <article-title>Accurate quantification of circular RNAs identifies extensive circular isoform switching events.</article-title><source><italic>Nat. Commun.</italic></source><volume>1</volume>:<fpage>90</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><name><surname>Zhang</surname><given-names>J.</given-names></name><name><surname>Chen</surname><given-names>S.</given-names></name><name><surname>Yang</surname><given-names>J.</given-names></name><name><surname>Zhao</surname><given-names>F.</given-names></name></person-group> (<year>2020</year>). <article-title>Accurate quantification of circular RNAs identifies extensive circular isoform switching events.</article-title><source><italic>Nat. Commun.</italic></source><volume>1</volume>:<fpage>90</fpage>."><person-group person-group-type="author"><name><surname>Zhang</surname><given-names>J.</given-names></name><name><surname>Chen</surname><given-names>S.</given-names></name><name><surname>Yang</surname><given-names>J.</given-names></name><name><surname>Zhao</surname><given-names>F.</given-names></name></person-group> (<year>2020</year>). <article-title>Accurate quantification of circular RNAs identifies extensive circular isoform switching events.</article-title><source><italic>Nat. Commun.</italic></source><volume>1</volume>:<fpage>90</fpage>.</a> <b>(1)</b> A Scalable Strand-Specific Protocol Enabling Full-Length Total RNA Sequencing From Single Cells<ul><li> <b>Nextflow</b> [<xref rid="bib6" ref-type="bibr"><u>9</u></xref>] (version 18.10.1) was used to manage processing of the data on the HPC.<li> [9](<person-group person-group-type="author"><string-name><surname>Ewels</surname> &#xA0;<given-names>PA</given-names></string-name>, <string-name><surname>Peltzer</surname> &#xA0;<given-names>A</given-names></string-name>, <string-name><surname>Fillinger</surname> &#xA0;<given-names>S</given-names></string-name>, <etal>et al.</etal></person-group> &#xA0;<article-title>nf-core&#x202F;: Community curated bioinformatics pipelines</article-title>. <source>BioRxiv</source>. <year>2019</year>;<comment>610741</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34079046">34079046</a> <b>(1)</b> Transcription factor enrichment analysis (TFEA) quantifies the activity of multiple transcription factors from a single experiment<ul><li> All GRO-Seq and PRO-Seq data were processed using the <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR79"><u>79</u></xref></sup> NascentFlow pipeline (v1.1<sup><xref ref-type="bibr" rid="CR80"><u>80</u></xref></sup>) specifying the &#x201C;&#x2013;tfit&#x201D; flag.<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [80](Tripodi, I. J. &amp; Gruca, M. A. <italic>Nascent-Flow</italic>. <ext-link ext-link-type="uri" xlink:href="https://osf.io/ndhj2/">https://osf.io/ndhj2/</ext-link> (2018).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=Tripodi, I. J. &amp; Gruca, M. A. <italic>Nascent-Flow</italic>. <ext-link ext-link-type="uri" xlink:href="https://osf.io/ndhj2/">https://osf.io/ndhj2/</ext-link> (2018).">Tripodi, I. J. &amp; Gruca, M. A. <italic>Nascent-Flow</italic>. <ext-link ext-link-type="uri" xlink:href="https://osf.io/ndhj2/">https://osf.io/ndhj2/</ext-link> (2018).</a> <b>(5)</b> Transcription factor enrichment analysis (TFEA) quantifies the activity of multiple transcription factors from a single experiment<ul><li> <monospace>pyrpipe</monospace> pipelines can be easily scaled using a workflow manager, including the popular Snakemake (<xref rid="B7" ref-type="bibr"><u>7</u></xref>), <b>NextFlow</b> (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) or Toil (<xref rid="B14" ref-type="bibr"><u>14</u></xref>).<li>We implemented <monospace>pyrpipe</monospace> within two workflow management systems, Snakemake (<xref rid="B7" ref-type="bibr"><u>7</u></xref>) and <b>NextFlow</b> (<xref rid="B8" ref-type="bibr"><u>8</u></xref>), selecting these specifically because they are widely used by the bioinformatics community (<xref rid="B27" ref-type="bibr"><u>27</u></xref>).<li> Snakemake and <b>NextFlow</b> were independently used to implement, manage and execute the pipeline for multiple RNA-Seq samples in parallel on a single cluster.<li> strategy to check for specific output files and schedule jobs accordingly (<xref rid="B6" ref-type="bibr"><u>6</u></xref>,<xref rid="B7" ref-type="bibr"><u>7</u></xref>), whereas <b>Nextflow</b> uses a &#x2018;push-based&#x2019; scheme in which a &#x2018;process&#x2019; defined in the workflow pushes its outputs to downstream &#x2018;processes&#x2019; (<xref rid="B8" ref-type="bibr"><u>8</u></xref>,<xref rid="B38" ref-type="bibr"><u>38</u></xref>).<li> similar to <b>Nextflow</b> it implements dataflow based task scheduling.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<person-group person-group-type="author"><string-name><surname>Strozzi</surname><given-names>F.</given-names></string-name>, <string-name><surname>Janssen</surname><given-names>R.</given-names></string-name>, <string-name><surname>Wurmus</surname><given-names>R.</given-names></string-name>, <string-name><surname>Crusoe</surname><given-names>M.R.</given-names></string-name>, <string-name><surname>Githinji</surname><given-names>G.</given-names></string-name>, <string-name><surname>Di&#xA0;Tommaso</surname><given-names>P.</given-names></string-name>, <string-name><surname>Belhachemi</surname><given-names>D.</given-names></string-name>, <string-name><surname>M&#xF6;ller</surname><given-names>S.</given-names></string-name>, <string-name><surname>Smant</surname><given-names>G.</given-names></string-name>, <string-name><surname>de&#xA0;Ligt</surname><given-names>J.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Scalable workflows and reproducible data analysis for genomics</article-title>. <source>Evolutionary Genomics</source>. <year>2019</year>; <publisher-name>Springer</publisher-name><fpage>723</fpage>&#x2013;<lpage>745</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34051791">34051791</a> <b>(1)</b> Dysregulation of COVID-19 related gene expression in the COPD lung<ul><li> The bioinformatics workflow was organized using <b>Nextflow</b> workflow management system (v19.07) [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>] and Bioconda software management tool [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>].<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34122366">34122366</a> <b>(1)</b> Respiratory Co-Infections: Modulators of SARS-CoV-2 Patients&#x2019; Clinical Sub-Phenotype<ul><li> The downstream analysis was automated using the <b>Nextflow</b> script.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34072237">34072237</a> <b>(1)</b> Multi-Omics Model Applied to Cancer Genetics<ul><li> Examples of such systems include Galaxy [<xref rid="B50-ijms-22-05751" ref-type="bibr"><u>50</u></xref>], Snakemake [<xref rid="B51-ijms-22-05751" ref-type="bibr"><u>51</u></xref>], <b>Nextflow</b> [<xref rid="B52-ijms-22-05751" ref-type="bibr"><u>52</u></xref>], and the general-purpose Common Workflow Language [<xref rid="B53-ijms-22-05751" ref-type="bibr"><u>53</u></xref>].<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [53](          <person-group person-group-type="author">            <name>              <surname>Peter</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>R.C.</given-names>            </name>            <name>              <surname>Neboj&#x161;a</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Brad</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>John</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Andrey</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Dan</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Herv&#xE9;</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Maya</surname>              <given-names>N.</given-names>            </name>            <etal/>          </person-group>          <article-title>Common Workflow Language, v1.0</article-title>          <year>2016</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://escholarship.org/uc/item/25z538jj">https://escholarship.org/uc/item/25z538jj</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-05-30">(accessed on 30 May 2020)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author">            <name>              <surname>Peter</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>R.C.</given-names>            </name>            <name>              <surname>Neboj&#x161;a</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Brad</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>John</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Andrey</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Dan</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Herv&#xE9;</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Maya</surname>              <given-names>N.</given-names>            </name>            <etal/>          </person-group>          <article-title>Common Workflow Language, v1.0</article-title>          <year>2016</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://escholarship.org/uc/item/25z538jj">https://escholarship.org/uc/item/25z538jj</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-05-30">(accessed on 30 May 2020)</date-in-citation>        ">          <person-group person-group-type="author">            <name>              <surname>Peter</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>R.C.</given-names>            </name>            <name>              <surname>Neboj&#x161;a</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Brad</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>John</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Andrey</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Dan</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Herv&#xE9;</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Maya</surname>              <given-names>N.</given-names>            </name>            <etal/>          </person-group>          <article-title>Common Workflow Language, v1.0</article-title>          <year>2016</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://escholarship.org/uc/item/25z538jj">https://escholarship.org/uc/item/25z538jj</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-05-30">(accessed on 30 May 2020)</date-in-citation>        </a> <b>(1)</b> Multi-Omics Model Applied to Cancer Genetics<ul><li>A dedicated computational pipeline for the single-microbial-cell genome was developed (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/gongyh/nf-core-scgs" ext-link-type="uri">https://github.com/gongyh/nf-core-scgs</ext-link>) to efficiently analyze single-cell amplified bacterial genome (SAG) data sets by integrating various tools with <b>Nextflow</b> (<xref rid="B69" ref-type="bibr"><u>69</u></xref>).<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34055480">34055480</a> <b>(6)</b> ProkEvo: an automated, reproducible, and scalable framework for high-throughput bacterial population genomics analyses<ul><li><b>Nextflow</b><xref ref-type="fn" rid="table-1fn5"><u>*****</u></xref>.<li>While <b>Nextflow</b> is more commonly used for bioinformatics applications, Pegasus WMS has the best overall performance for efficiently utilizing the computational resources (<xref rid="ref-52" ref-type="bibr"><u>Heller &amp; Ghahramani, 2005</u></xref>).<li> While Pegasus is a versatile WMS, other WMS such as <b>Nextflow</b> (<xref rid="ref-33" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and Snakemake (<xref rid="ref-60" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>) are more commonly used for bioinformatics applications.<li> Bactopia is one of the most comprehensive pipelines available for analysis of bacterial genomes using <b>Nextflow</b> workflow manager (<xref rid="ref-79" ref-type="bibr"><u>Petit &amp; Read, 2020</u></xref>).<li> The documentations for Pegasus and <b>Nextflow</b> provide examples of how to do this.<li> <b>Nextflow</b>, the WMS used for Bactopia, generates many intermediate files that can exceed the available storage quotas on the standard computational platforms.<li> [*****](table-1fn5) [Heller &amp; Ghahramani, 2005](          <person-group person-group-type="author">            <name>              <surname>Heller</surname>              <given-names>KA</given-names>            </name>            <name>              <surname>Ghahramani</surname>              <given-names>Z</given-names>            </name>          </person-group>          <article-title>Bayesian hierarchical clustering</article-title>          <year>2005</year>          <conf-name>Proceedings of the 22nd International Conference on Machine Learning</conf-name>          <publisher-loc>New York</publisher-loc>          <publisher-name>ACM</publisher-name>          <fpage>297</fpage>          <lpage>304</lpage>        ) [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Petit &amp; Read, 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32753501">32753501</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34016747">34016747</a> <b>(1)</b> Nonparametric coalescent inference of mutation spectrum history and demography<ul><li>All of the analyses and figures for this paper can be reproduced using <b>Nextflow</b> pipelines (<xref rid="r76" ref-type="bibr"><u>76</u></xref>) and Jupyter notebooks (<ext-link ext-link-type="uri" xlink:href="https://jupyter.org" xlink:show="new">https://jupyter.org</ext-link>) available in ref.<li> [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34069990">34069990</a> <b>(1)</b> Dadaist2: A Toolkit to Automate and Simplify Statistical Analysis and Plotting of Metabarcoding Experiments<ul><li> To show this applicability on our website, we provide a <b>NextFlow</b> [<xref rid="B23-ijms-22-05309" ref-type="bibr"><u>23</u></xref>] example of a pipeline that uses various Dadaist2 modules.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33993283">33993283</a> <b>(1)</b> An epidemiological investigation of COVID-19 outbreaks in a group of care homes in Wales, UK: a retrospective cohort study<ul><li> This process is provided as a <b>Nextflow</b> workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</ext-link>).Consensus fasta sequences were submitted to the COVID-19 Genomics UK Consortium (COG-UK) analysis environment hosted on MRC CLIMB, and each sequence was assigned a global lineage.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33993283">33993283</a> <b>(1)</b> An epidemiological investigation of COVID-19 outbreaks in a group of care homes in Wales, UK: a retrospective cohort study<ul><li> TractoFlow-ABS is an extension of the recent TractoFlow pipeline (<xref rid="bib93" ref-type="bibr"><u>Theaud et al., 2020a</u></xref>) publicly available for academic research purposes (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/scilus/tractoflow-ABS" ext-link-type="uri">https://github.com/scilus/tractoflow-ABS</ext-link>;&#xA0;<xref rid="bib94" ref-type="bibr"><u>Theaud, 2020b</u></xref>) that uses <b>Nextflow</b> (<xref rid="bib28" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and Singularity (<xref rid="bib56" ref-type="bibr"><u>Kurtzer et al., 2017</u></xref>) to ensure efficient and reproducible diffusion processing.<li> [Theaud et al., 2020a](          <person-group person-group-type="author">            <name>              <surname>Theaud</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Houde</surname>              <given-names>J-C</given-names>            </name>            <name>              <surname>Bor&#xE9;</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Rheault</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Morency</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Descoteaux</surname>              <given-names>M</given-names>            </name>          </person-group>          <year iso-8601-date="2020">2020a</year>          <article-title>TractoFlow: a robust, efficient and reproducible diffusion MRIpipeline leveraging nextflow &amp; singularity</article-title>          <source>bioRxiv</source>          <pub-id pub-id-type="doi">10.1101/631952</pub-id>        ) [Theaud, 2020b](          <person-group person-group-type="author">            <name>              <surname>Theaud</surname>              <given-names>G</given-names>            </name>          </person-group>          <year iso-8601-date="2020">2020b</year>          <data-title>TractoFlow-ABS</data-title>          <source>GitHub</source>          <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/scilus/tractoflow-ABS" ext-link-type="uri">https://github.com/scilus/tractoflow-ABS</ext-link>        ) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Kurtzer et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978747">33978747</a> <b>(2)</b> DetectIS: a pipeline to rapidly detect exogenous DNA integration sites using DNA or RNA paired-end sequencing data<ul><li> All the steps of the detectIS pipeline are embedded in a <b>Nextflow</b> (<xref rid="btab366-B4" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.<li> A bash script is also given to analyze a test dataset without <b>Nextflow</b> and can be used as a template for analysis in local environments.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33978747">33978747</a> <b>(4)</b> DetectIS: a pipeline to rapidly detect exogenous DNA integration sites using DNA or RNA paired-end sequencing data<ul><li><b>Nextflow</b> and galaxy support.<li>Recently, Dockstore has added support for the <b>Nextflow</b> (<xref rid="B16" ref-type="bibr"><u>16</u></xref>) and Galaxy (<xref rid="B17" ref-type="bibr"><u>17</u></xref>) workflow languages.<li>Uniformly registered onto Dockstore using GitHub apps, the nf-core workflows (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) are another good example of a set of workflows that are automatically updated on Dockstore.The nf-core organization (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://dockstore.org/organizations/nfcore" ext-link-type="uri">https://dockstore.org/organizations/nfcore</ext-link>) of workflows represents a community effort to build a curated set of high quality workflows in the <b>Nextflow</b> language.<li> These trainings have the added benefit of allowing the team to connect and collaborate with many like minded communities including <b>Nextflow</b> and Biocontainers (Elixir Europe).<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34067022">34067022</a> <b>(1)</b> The PI3K/mTOR Pathway Is Targeted by Rare Germline Variants in Patients with Both Melanoma and Renal Cell Carcinoma<ul><li><b>Nextflow</b>-based [<xref rid="B37-cancers-13-02243" ref-type="bibr"><u>37</u></xref>] exome processing pipelines are available through GitHub (<uri xlink:href="https://github.com/IARCbioinfo">https://github.com/IARCbioinfo</uri> accessed on 25 January 2021).<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34036290">34036290</a> <b>(3)</b> A community effort to identify and correct mislabeled samples in proteogenomic studies<ul><li>Implementation of COSMO using <b>Nextflow</b> and Docker.<li>The COSMO workflow was implemented using <b>Nextflow</b><xref rid="bib31" ref-type="bibr"><sup>31</sup></xref> and Docker.<li> Different components of COSMO were integrated using <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33991511">33991511</a> <b>(3)</b> Epitope profiling reveals binding signatures of SARS-CoV-2 immune response in natural infection and cross-reactivity with endemic human CoVs<ul><li><hr/></td></tr><tr><td>Geneious R11</td><td><ext-link ext-link-type="uri" xlink:href="https://www.geneious.com" id="intref0010">https://www.geneious.com</ext-link></td><td>N/A</td></tr><tr><td>Prism 9</td><td><ext-link ext-link-type="uri" xlink:href="https://www.graphpad.com" id="intref0015">https://www.graphpad.com</ext-link></td><td>N/A</td></tr><tr><td><b>Nextflow</b>.<li> The <b>Nextflow</b> pipeline used to align sequencing reads to the pan-CoV reference library and conduct downstream sample curation and analysis is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/pan-CoV-manuscript" id="intref0040">https://github.com/matsengrp/pan-CoV-manuscript</ext-link>.<li>An enrichment matrix was created by aligning all sequences to the pan-CoV reference library using a <b>Nextflow</b> data processing pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/pan-CoV-manuscript" id="intref0055">https://github.com/matsengrp/pan-CoV-manuscript</ext-link>) (<xref rid="bib11" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2017</u></xref>).<li> [Di Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33952321">33952321</a> <b>(10)</b> <italic>geneshot</italic>: gene-level metagenomics identifies genome islands associated with immunotherapy response<ul><li>Finally, by wrapping together the set of tasks required for comprehensive gene-level metagenomic analysis using the <b>Nextflow</b> workflow management system [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], we provide a convenient mechanism for microbiome analysis which can be implemented across a variety of high-performance computing infrastructures with minimal configuration required for each user.<li>The process of performing gene-level metagenomic analysis is implemented by <italic>geneshot</italic> using a series of bioinformatic processes which are all coordinated into an overarching workflow using the free and open-source <b>Nextflow</b> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] workflow management system.<li> to the workflow, users can modify the details of how the analysis is executed, and a record of the parameters used to execute the workflow can be saved (along with a summary of the computational resources which were used) with the reporting functionality provided by <b>Nextflow</b>.<li>Setting up <b>Nextflow</b>.<li>In order to run <italic>geneshot</italic> on the available computational resources, start by installing the workflow management tool <b>Nextflow</b> and configuring it.<li> The complete documentation for <b>Nextflow</b> can be found at <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/getstarted.html">https://www.<b>nextflow</b>.io/docs/latest/getstarted.html</ext-link>.<li> The process of configuring <b>Nextflow</b> is accomplished by saving a &#x201C;<b>nextflow</b>.config&#x201D; file in the user&#x2019;s home directory which describes the compute resources which should be used (local execution, SLURM, PBS, AWS, etc.), and which will be referenced when running <italic>geneshot</italic>.<li> In order to run <italic>geneshot</italic>, the user starts with the command &#x201C;<b>nextflow</b> run Golob-Minot/geneshot&#x201D; and appends any of the flags shown in Additional file <xref rid="MOESM2" ref-type="media"><u>2</u></xref>: Table S1 (e.g., --manifest manifest_file.csv).<li><bold>Additional file 2: Table S1.</bold> In order to run <italic>geneshot</italic>, the user starts with the command &#x201C;<b>nextflow</b> run Golob-Minot/geneshot&#x201D; and appends any of the flags.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34010620">34010620</a> <b>(3)</b> High-resolution profiling of pathways of escape for SARS-CoV-2 spike-binding antibodies<ul><li><hr/></td></tr><tr><td>Python</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org/" id="intref0055">https://www.python.org/</ext-link></td></tr><tr><td>DNAChisel (version 3.2.2)</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://github.com/Edinburgh-Genome-Foundry/DnaChisel" id="intref0060">https://github.com/Edinburgh-Genome-Foundry/DnaChisel</ext-link></td></tr><tr><td>phip-flow</td><td>Matsen Lab</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/phip-flow" id="intref0065">https://github.com/matsengrp/phip-flow</ext-link></td></tr><tr><td><b>Nextflow</b>.<li><hr/></td></tr><tr><td>Python</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org/" id="intref0055">https://www.python.org/</ext-link></td></tr><tr><td>DNAChisel (version 3.2.2)</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://github.com/Edinburgh-Genome-Foundry/DnaChisel" id="intref0060">https://github.com/Edinburgh-Genome-Foundry/DnaChisel</ext-link></td></tr><tr><td>phip-flow</td><td>Matsen Lab</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/phip-flow" id="intref0065">https://github.com/matsengrp/phip-flow</ext-link></td></tr><tr><td><b>Nextflow</b></td><td/><td><ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/" id="intref0070">https://www.<b>nextflow</b>.io/</ext-link>.<li> The <b>Nextflow</b> pipeline, used to align all sample reads to the reference library, is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/phip-flow" id="intref0150">https://github.com/matsengrp/phip-flow</ext-link>.<li> Demultiplexed sample reads were aligned to the reference library in parallel using a <b>Nextflow</b> data processing pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34968299">34968299</a> <b>(2)</b> The EpiDiverse Plant Epigenome-Wide Association Studies (EWAS) Pipeline<ul><li>The pipeline was built using <b>Nextflow</b> [<xref rid="B54-epigenomes-05-00012" ref-type="bibr"><u>54</u></xref>], a workflow tool for running tasks across multicompute infrastructures in a portable manner.<li> The pipeline was set up using <b>Nextflow</b> 20.07.1 revisions [<xref rid="B54-epigenomes-05-00012" ref-type="bibr"><u>54</u></xref>] and is based on the GEM R package [<xref rid="B50-epigenomes-05-00012" ref-type="bibr"><u>50</u></xref>].<li> [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27480116">27480116</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33932985">33932985</a> <b>(15)</b> SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor-powered RNA-seq analyses<ul><li> We then improved the usability and portability of this pipeline thanks to the <b>Nextflow</b> framework [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> Other key features of SPEAQeasy are that it produces the information that coupled with DNA genotyping information can be used for detecting and fixing sample swaps [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>&#x2013;<xref ref-type="bibr" rid="CR33"><u>33</u></xref>], RNA-seq processing quality metrics that are helpful for statistically adjusting for quality differences across samples [<xref ref-type="bibr" rid="CR5"><u>5</u></xref>], data that powers the exploration of the un-annotated transcriptome, and that it can be used in several computational frameworks thanks to <b>Nextflow</b>&#x2019;s configuration flexibility [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> SPEAQeasy is a <b>Nextflow</b>-powered [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] pipeline that starts from a set of FASTQ files [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>], performs quality assessment and other processing steps (Implementation: overview), and produces easy-to-use R objects [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>].<li> Furthermore, the <b>Nextflow</b>-based implementation allows for more experienced developers to quickly add additional steps or switch out software, creating a flexible and scalable RNA-seq processing pipeline.<li>SPEAQeasy, through <b>Nextflow</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>], can be deployed in a variety of high-throughput computational environments such as: local machines, Sun/Son Grid Engine (SGE) compute clusters, servers that enable Docker [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>], and cloud computing environments [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>] such as Amazon AWS.<li> <b>Nextflow</b> provides the ability to run the same code using configuration files that are specific to the computing environment at hand.<li> The preferred installation method relies on <b>Nextflow</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] to automatically pull pre-specified docker images at run-time unless they were previously downloaded; this approach is used in some currently-available pipelines [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>].<li> In the case of <b>Nextflow</b> and snakemake-based workflows, individual jobs are internally submitted for each pipeline component, and typically it is implicitly left to the user to worry about time specification for every component.<li> We were motivated to implement this feature after observing how as the pipeline grew in complexity, it became increasingly necessary to understand <b>Nextflow</b>&#x2019;s implementation details to debug execution errors.<li> Given the open-source nature of SPEAQeasy and <b>Nextflow</b>, the SPEAQeasy code can be adapted if users are interested in switching processing tools or want to expand support to other genome references beyond mm10, rn6, hg19 and hg38.<li> As a pipeline based on <b>Nextflow</b>, SPEAQeasy can isolate individual components of the workflow, called <italic>processes</italic>, inside docker containers.<li> This reflects a general feature of SPEAQeasy, provided by its <b>Nextflow</b> base- processes are never &#x201C;repeated&#x201D; if their outputs already exist.<li> These syntaxes come from <b>Nextflow</b>, which manages how to translate these simple user-defined options into a syntax recognized by the cluster (if applicable).<li> However, <b>Nextflow</b> also makes it simple to explicitly specify cluster-specific options.<li> In order of submission, the name of each <b>Nextflow</b> process is printed, along with (1) the working directory: where all relevant files are present, (2) the exit code: a standard indication of whether the process succeeded or how it failed, (3) a list of the specific commands run during the given process.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23559639">23559639</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25954321">25954321</a>) [5](K. Van den Berge <italic>et al.</italic>, RNA sequencing data: hitchhiker&#x2019;s guide to expression analysis. <italic>Annu. Rev. Biomed. Data Sci.</italic><bold>2</bold> (2019). 10.1146/annurev-biodatasci-072018-021255.) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20015970">20015970</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25633503">25633503</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [48](D. Merkel, Docker: Lightweight Linux Containers for Consistent Development and Deployment. <italic>Linux J.</italic><bold>2014</bold> (2014).) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29379135">29379135</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](P. Ewels et al. nf-core/rnaseq: nf-core/rnaseq version 1.4.2. Zenodo 2019;10.5281/zenodo.3503887.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33914880">33914880</a> <b>(1)</b> Application of a bioinformatic pipeline to RNA-seq data identifies novel virus-like sequence in human blood<ul><li>Our novel pipeline, Mystery Miner, is written as a <b>Nextflow</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33914880">33914880</a> <b>(1)</b> Application of a bioinformatic pipeline to RNA-seq data identifies novel virus-like sequence in human blood<ul><li><b>NextFlow</b>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33879786">33879786</a> <b>(1)</b> Epigenomic landscape of human colorectal cancer unveils an aberrant core of pan-cancer enhancers orchestrated by YAP/TAZ<ul><li> RNA-seq and ChIP-seq primary analyses were executed by a custom pipeline managed by <b>Nextflow</b> (v20.04.1.5335)<sup><xref ref-type="bibr" rid="CR48"><u>48</u></xref></sup>.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(3)</b> Sustainable data analysis with Snakemake<ul><li>Third, with systems like <b>Nextflow</b><sup><xref rid="ref-17" ref-type="bibr"><u>17</u></xref></sup>, Snakemake<sup><xref rid="ref-18" ref-type="bibr"><u>18</u></xref></sup>, BioQueue<sup><xref rid="ref-19" ref-type="bibr"><u>19</u></xref></sup>, Bpipe<sup><xref rid="ref-20" ref-type="bibr"><u>20</u></xref></sup>, ClusterFlow<sup><xref rid="ref-21" ref-type="bibr"><u>21</u></xref></sup>, Cylc<sup><xref rid="ref-22" ref-type="bibr"><u>22</u></xref></sup>, and BigDataScript<sup><xref rid="ref-23" ref-type="bibr"><u>23</u></xref></sup>, workflows are specified using a domain specific language (DSL).<li> For <b>Nextflow</b> and Snakemake, since the DSL is implemented as an extension to a generic programming language (Groovy and Python), access to the full power of the underlying programming language is maintained (e.g.<li> Galaxy, <b>Nextflow</b>, Snakemake, WDL, CWL), by allowing the definition and scalable execution of each involved step, including deployment of the software stack needed for each step (e.g.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28633441">28633441</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28299179">28299179</a>) [22](<person-group person-group-type="author"><name><surname>Oliver</surname><given-names>HJ</given-names></name><name><surname>Shin</surname><given-names>M</given-names></name><name><surname>Sanders</surname><given-names>O</given-names></name></person-group>:<article-title>Cylc: A Work&#xFB02;ow Engine for Cycling Systems.</article-title><source><italic toggle="yes">J Open Source Softw.</italic></source><year>2018</year>;<volume>3</volume>(<issue>27</issue>):<fpage>737</fpage>.<pub-id pub-id-type="doi">10.21105/joss.00737</pub-id>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33875000">33875000</a> <b>(1)</b> A community-driven resource for genomic epidemiology and antimicrobial resistance prediction of <italic>Neisseria gonorrhoeae</italic> at Pathogenwatch<ul><li> This pipeline runs a <b>Nextflow</b> workflow to quality-check (QC) paired-end short-read fastq files before and after filtering and trimming, assembles the data and quality-checks the resulting assembly.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33859326">33859326</a> <b>(1)</b> A longitudinal analysis of brain extracellular free water in HIV infected individuals<ul><li> A bi-tensor model was used to reconstruct FW maps at each voxel from the DWI data using a previously described algorithm<sup><xref ref-type="bibr" rid="CR21"><u>21</u></xref></sup> and the processing was carried out using the <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup> pipeline with all software dependencies bundled in a Singularity Container<sup><xref ref-type="bibr" rid="CR34"><u>34</u></xref></sup>.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31632265">31632265</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33846313">33846313</a> <b>(1)</b> Go Get Data (GGD) is a framework that facilitates reproducible access to genomic data<ul><li> Furthermore, GGD can easily be integrated into reproducible processing and analysis workflow systems such as <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup> and Snakemake<sup><xref ref-type="bibr" rid="CR24"><u>24</u></xref></sup>.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33831079">33831079</a> <b>(1)</b> Whole genome variation in 27 Mexican indigenous populations, demographic and biomedical insights<ul><li>In general, the pipelines we provide in Methods use <b>Nextflow</b> [<xref rid="pone.0249773.ref060" ref-type="bibr"><u>60</u></xref>] for controlling data flow, custom bash and R scripts (mainly using tidyverse, data table, and Circlize packages) [<xref rid="pone.0249773.ref061" ref-type="bibr"><u>61</u></xref>], and bcftools [<xref rid="pone.0249773.ref062" ref-type="bibr"><u>62</u></xref>] for data handling and analysis.<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24930139">24930139</a>) [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21653522">21653522</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33891875">33891875</a> <b>(1)</b> Early introductions and transmission of SARS-CoV-2 variant B.1.1.7 in the United States<ul><li>Illumina libraries were processed with ARTIC <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="http://github.com/connor-lab/ncov2019-artic-nf/tree/illumina" id="intref0200">http://github.com/connor-lab/ncov2019-artic-nf/tree/illumina</ext-link>, last updated April 2020).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33850640">33850640</a> <b>(1)</b> Bioinformatic strategies for the analysis of genomic aberrations detected by targeted NGS panels with clinical application<ul><li> Oshlack, 2012</xref>), reflow (<uri xlink:href="https://github.com/grailbio/reflow">https://github.com/grailbio/reflow</uri>) and <b>nextflow</b> (<xref rid="ref-33" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), which also offers a set of curated pipelines under the nf-core project (<xref rid="ref-37" ref-type="bibr"><u>Ewels et al., 2020</u></xref>).<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Ewels et al., 2020](          <person-group person-group-type="author">            <name>              <surname>Ewels</surname>              <given-names>PA</given-names>            </name>            <name>              <surname>Peltzer</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Fillinger</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Patel</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Alneberg</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Garcia</surname>              <given-names>MU</given-names>            </name>            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>The nf-core framework for community-curated bioinformatics pipelines</article-title>          <source>Nature Biotechnology</source>          <year>2020</year>          <volume>38</volume>          <issue>3</issue>          <fpage>276</fpage>          <lpage>278</lpage>          <pub-id pub-id-type="doi">10.1038/s41587-020-0439-x</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33772596">33772596</a> <b>(1)</b> FUNGI: FUsioN Gene Integration toolset<ul><li> FUNGI is similar to <b>nextflow</b> based rnafusion (<ext-link xlink:href="https://github.com/nf-core/rnafusion" ext-link-type="uri">https://github.com/nf-core/rnafusion</ext-link>) in the capacity to run several fusion calling algorithms, integrating them and generating a report, but rnafusion lacks oncogenic scoring methods.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33772539">33772539</a> <b>(10)</b> Testcrosses are an efficient strategy for identifying <italic>cis</italic>-regulatory variation: Bayesian analysis of allele-specific expression (BayesASE)<ul><li> Workflow templates have been constructed for SLURM schedulers, for <b>Nextflow</b> (<xref rid="jkab096-B11" ref-type="bibr">Di Tommaso <italic>et al.<li> Workflows for each module are coded according to SLURM workflow manager, <b>Nextflow</b> and Galaxy specifications, and all scripts are available on Github (<ext-link ext-link-type="uri" xlink:href="https://github.com/McIntyre-Lab/BayesASE">https://github.com/McIntyre-Lab/BayesASE</ext-link>).<li>To facilitate its use, BayesASE is available for SLURM schedulers, as workflows on the <b>Nextflow</b> platform, and as a Galaxy Tool.<li> Names are consistent with the flow charts and standard naming conventions for sbatch and <b>Nextflow</b> are used.<li> <b>Nextflow</b> (<xref rid="jkab096-B11" ref-type="bibr">Di Tommaso <italic>et al.<li> <b>Nextflow</b> pipelines consist of a configuration file and a series of processes that define the major steps in the pipeline.<li> <b>Nextflow</b> also provides an automatic caching mechanism for identifying and skipping successfully completed tasks and using previously cached results for downstream tasks.<li> Each BasyesASE module has been coded in <b>Nextflow</b> and is available on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/McIntyre-Lab/BayesASE">https://github.com/McIntyre-Lab/BayesASE</ext-link>).<li> The BayesASE framework lays out each step to testing AI transparently, and enables researchers to perform analysis using their preferred approach (galaxy, <b>Nextflow</b>, or SBATCH queuing).<li> We provide a set of well-documented python scripts organized into modules and available with examples as SLURM bash jobs, using <b>Nextflow</b> or Galaxy.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33765910">33765910</a> <b>(2)</b> BugSeq: a highly accurate cloud platform for long-read metagenomic analyses<ul><li> This pipeline, which we call BugSeq, has been packaged with <b>Nextflow</b> (v20.07.1) and made available as an online service (<ext-link ext-link-type="uri" xlink:href="https://bugseq.com/free">https://bugseq.com/free</ext-link>) for easy cloud analyses.<li>Programming language: <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33841452">33841452</a> <b>(1)</b> Toward Systematic Understanding of Flower Bud Induction in Apple: A Multi-Omics Approach<ul><li> The resulting fastq files were demultiplexed and transferred to QBiC and analyzed on an HPC cluster of the University of T&#xFC;bingen in a fully automated way using a <b>Nextflow</b>-based RNA-Seq pipeline<sup><xref ref-type="fn" rid="footnote1"><u>1</u></xref></sup> (release 1.3).<li> [1](<a href="https://github.com/nf-core/rnaseq">https://github.com/nf-core/rnaseq</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33735170">33735170</a> <b>(1)</b> Boosting GWAS using biological networks: A study on susceptibility to familial breast cancer<ul><li> To make it easier for others to apply these methods to new datasets and aggregate their solutions, we built six <monospace><b>nextflow</b></monospace> pipelines [<xref rid="pcbi.1008819.ref069" ref-type="bibr"><u>69</u></xref>] with a consistent interface and, whenever possible, parallelized computation.<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33735276">33735276</a> <b>(1)</b> Application of a bioinformatics training delivery method for reaching dispersed and distant trainees<ul><li>Introduction to Snakemake and <b>Nextflow</b> (June 2019).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33735208">33735208</a> <b>(1)</b> Principles for data analysis workflows<ul><li> These inform project workflow, and tools, such as Snakemake and <b>Nextflow</b>, can begin to be used to improve the flexibility and reproducibility of the project [<xref rid="pcbi.1008770.ref021" ref-type="bibr"><u>21</u></xref>&#x2013;<xref rid="pcbi.1008770.ref023" ref-type="bibr"><u>23</u></xref>].<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33729153">33729153</a> <b>(2)</b> Action detection using a neural network elucidates the genetics of mouse grooming behavior<ul><li> copy archived at <ext-link ext-link-type="uri" xlink:href="https://archive.softwareheritage.org/swh:1:dir:07f01d98248b9603c2824c6f947f819eeaad3b18;origin=https://github.com/TheJacksonLaboratory/mousegwas;visit=swh:1:snp:ef547bbb13017e0b6df06750bf7ebd5867e100b0;anchor=swh:1:rev:5d2caac2637da442f4b9648ac1eb1f35bd1136cf/">swh:1:rev:5d2caac2637da442f4b9648ac1eb1f35bd1136cf</ext-link> it also includes a singularity container definition file and a <b>nextflow</b> pipeline for regenerating the results.<li> The command used for generating the results is:<code xml:space="preserve" position="float" orientation="portrait">export GH=https://raw.githubusercontent.com/TheJacksonLaboratory/mousegwas/grooming <b>nextflow</b> run TheJacksonLaboratory/mousegwas -r grooming \ --yaml $GH/example/grooming_nowild.yaml \ --shufyaml $GH/example/grooming_shuffle.yaml \ --input $GH/example/grooming_paper_strain_survey_2019_11_21.csv \ --outdir grooming_output --addpostp="--loddrop 0" -profile slurm,singularity.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33777521">33777521</a> <b>(16)</b> Reproducible, portable, and efficient ancient genome reconstruction with nf-core/eager<ul><li>To address these shortcomings, we have completely re-implemented the latest version of the EAGER pipeline in <b>Nextflow</b>&#xA0;(<xref rid="ref-12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>)&#xA0;(a domain-specific-language or &#x2018;DSL&#x2019;, specifically designed for the construction of omics analysis pipelines), and introduced new features and more flexible pipeline configuration.<li>nf-core/eager can be run on POSIX-family operating systems (e.g., Linux and macOS) and has at minimum three dependencies, Java (&gt;=8), <b>Nextflow</b> (<xref rid="ref-12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), and a software container or environment system, e.g., Conda (<ext-link ext-link-type="uri" xlink:href="https://conda.io">https://conda.io</ext-link>), Docker (<ext-link ext-link-type="uri" xlink:href="https://www.docker.com">https://www.docker.com</ext-link>) or Singularity (<ext-link ext-link-type="uri" xlink:href="https://sylabs.io">https://sylabs.io</ext-link>), and can be run on both local machines as well as high performance computing (HPC) clusters.<li> Once installed, <b>Nextflow</b> handles all subsequent pipeline-related software dependencies.<li> Routinely used parameters or institutional configurations can be specified using <b>Nextflow</b>&#x2019;s in-built profile system, an example of which is provided in the basic test data, which can also be used for training and evaluation by new users.<li> If running on a HPC cluster utilising a scheduling system, <b>Nextflow</b> will generate and carry out efficient submission strategies of jobs for the user.<li> (2020)</xref> using nf-core/eager v2.2.0 (commit: e7471a7 and <b>Nextflow</b> version: 20.04.1) by simultaneously aligning publicly available shotgun-sequencing reads against the human reference genome (hs37d5) using bwa aln and the off-target reads against the NCBI Nucleotide (nt) database (October 2017 - uploaded here to Zenodo under DOI: <ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.5281/zenodo.4382153">10.5281/zenodo.4382153</ext-link>) with MALT.<li>The re-implementation of EAGER into <b>Nextflow</b> offers a range of benefits over the original custom pipeline framework.<li> Typically the output of the launch functionality is a JSON file that can be used with a nf-core/tools launch command as a single parameter (similar to the original EAGER), however integration with <b>Nextflow</b>&#x2019;s companion monitoring tool tower.nf (<ext-link ext-link-type="uri" xlink:href="https://tower.nf">https://tower.nf</ext-link>) also allows direct submission of pipelines without any command line usage.<li>nf-core/eager provides improved efficiency over the original EAGER pipeline by replacing sample-by-sample sequential processing with <b>Nextflow</b>&#x2019;s asynchronous job parallelisation, whereby multiple pipeline steps and samples are run in parallel (in addition to natively parallelised pipeline steps).<li> This is similar to the approach taken by PALEOMIX, however nf-core/eager expands this by utilising <b>Nextflow</b>&#x2019;s ability to customise the resource parameters for every job in the pipeline; reducing unnecessary resource allocation that can occur with unfamiliar users to each step of a high-throughput short-read data processing pipeline.<li>The development of nf-core/eager in <b>Nextflow</b> and the nf-core initiative will also improve open-source development, while ensuring the high quality of community contributions to the pipeline.<li> While <b>Nextflow</b> is written primarily in Groovy, the <b>Nextflow</b> DSL simplifies a number of concepts to an intermediate level that bioinformaticians without Java/Groovy experience can easily access (regardless of own programming language experience).<li> Furthermore, <b>Nextflow</b> places ubiquitous and more widely known command-line interfaces, such as bash, in a prominent position within the code, rather than custom Java code and classes (as in EAGER).<li> This will also be supported by the open and active nf-core community who provide general guidance and advice on developing <b>Nextflow</b> and nf-core pipelines.<li> This re-implementation of EAGER into <b>Nextflow</b> and nf-core will improve reproducibility and scalability of rapidly increasing aDNA datasets, for both large and small laboratories.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33594373">33594373</a> <b>(1)</b> Early introductions and community transmission of SARS-CoV-2 variant B.1.1.7 in the United States<ul><li>Illumina libraries were processed with ARTIC <b>nextflow</b> pipeline (<ext-link xlink:href="https://github.com/connor-lab/ncov2019-articnf/tree/illumina" ext-link-type="uri">github.com/connor-lab/ncov2019-articnf/tree/illumina</ext-link>, last updated April 2020).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33705383">33705383</a> <b>(1)</b> Ten simple rules for quick and dirty scientific programming<ul><li> Such scripts can be made using shell programming (Bash), computational notebooks (e.g., Jupyter [<xref rid="pcbi.1008549.ref030" ref-type="bibr"><u>30</u></xref>]), or workflow description languages such as <b>NextFlow</b> [<xref rid="pcbi.1008549.ref031" ref-type="bibr"><u>31</u></xref>] or Snakemake [<xref rid="pcbi.1008549.ref032" ref-type="bibr"><u>32</u></xref>].<li> [30](<name><surname>Kluyver</surname><given-names>T</given-names></name>, <name><surname>Ragan-Kelley</surname><given-names>B</given-names></name>, <name><surname>P&#xE9;rez</surname><given-names>F</given-names></name>, <name><surname>Granger</surname><given-names>BE</given-names></name>, <name><surname>Bussonnier</surname><given-names>M</given-names></name>, <name><surname>Frederic</surname><given-names>J</given-names></name>, <etal>et al</etal>. <source>Jupyter Notebooks-a publishing format for reproducible computational workflows</source>. In: <publisher-name>ELPUB</publisher-name>; <year>2016</year>. p. <fpage>87</fpage>&#x2013;<lpage>90</lpage>.) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33748783">33748783</a> <b>(7)</b> Co-fractionation/mass spectrometry to identify protein complexes<ul><li>                  <hr/>                </td>              </tr>              <tr>                <td>Cfmsflow</td>                <td>This paper, <ext-link ext-link-type="uri" xlink:href="https://github.com/marcottelab/cfmsflow" id="intref0030">https://github.com/marcottelab/cfmsflow</ext-link></td>                <td/>              </tr>              <tr>                <td>TPOT</td>                <td>                  <xref rid="bib6" ref-type="bibr"><u>Olson and Moore, 2016</u></xref>                </td>                <td>Version &gt;= 0.10.0</td>              </tr>              <tr>                <td><b>Nextflow</b>.<li>To simplify the computational process of detecting protein interactions from CF/MS data, we provide cfmsflow, a <b>Nextflow</b> pipeline for Linux systems.<li><inline-graphic xlink:href="fx2.gif"/><bold>CRITICAL:</bold> Install <b>nextflow</b> onto a Linux-based operating system (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/docs/latest/getstarted.html" id="intref0010">https://www.<b>nextflow</b>.io/docs/latest/getstarted.html</ext-link>).<li><b>nextflow</b> main.nf &#x2013;params-file user_parameters.json<disp-quote><p><inline-graphic xlink:href="fx2.gif"/><bold>CRITICAL:</bold> Before running cfmsflow on real data, test the pipeline by running the provided example.<li><b>nextflow</b> main.nf &#x2013;params-file example_params/example_wholepipeline.json<disp-quote><p><bold><italic>Note:</italic></bold> Examples of all input file formats are provided in the example_input directory of the github repository.<li> After modifying this parameter, rerun the <b>nextflow</b> command with an additional &#x2013;resume flag to avoid rerunning already complete portions of the pipeline.<li> To rerun the TPOT pipeline limiting the maximum number of features, modify the max_features_to_select value in your user parameters json to a smaller number than the total number of features and resume the pipeline with <b>nextflow</b> main.nf &#x2013;params-file user_parameters.json &#x2013;resume or create a new parameters file that begins at step 4.<li> [Olson and Moore, 2016](          <person-group person-group-type="author">            <name>              <surname>Olson</surname>              <given-names>R.S.</given-names>            </name>            <name>              <surname>Moore</surname>              <given-names>J.H.</given-names>            </name>          </person-group>          <chapter-title>TPOT: A tree-based pipeline optimization tool for automating machine learning</chapter-title>          <person-group person-group-type="editor">            <name>              <surname>Hutter</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Kotthoff</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Vanschoren</surname>              <given-names>J.</given-names>            </name>          </person-group>          <source>Automated Machine Learning</source>          <year>2016</year>          <publisher-name>Springer</publisher-name>          <fpage>66</fpage>          <lpage>74</lpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33758844">33758844</a> <b>(2)</b> SARS-CoV-2 evolution in animals suggests mechanisms for rapid variant selection<ul><li>Raw sequencing data were input into a comprehensive <b>Nextflow</b> pipeline for processing next generation sequencing data and single nucleotide and structural variant calling.<li> <b>Nextflow</b> is a bioinformatics workflow manager that facilitates the development of complex and reproducible computational pipelines (<xref rid="R47" ref-type="bibr"><u>47</u></xref>).<li> [47](<name><surname>di Tommaso</surname><given-names>P.</given-names></name>, <name><surname>Chatzou</surname><given-names>M.</given-names></name>, <name><surname>Floden</surname><given-names>E. W.</given-names></name>, <name><surname>Barja</surname><given-names>P. P.</given-names></name>, <name><surname>Palumbo</surname><given-names>E.</given-names></name>, <name><surname>Notredame</surname><given-names>C.</given-names></name>, <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nature Biotechnology</source><volume>35</volume><fpage>316</fpage>&#x2013;<lpage>319</lpage> (<year>2017</year>). doi:<pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33746987">33746987</a> <b>(1)</b> Antiretroviral Treatment-Induced Decrease in Immune Activation Contributes to Reduced Susceptibility to Tuberculosis in HIV-1/Mtb Co-infected Persons<ul><li> Datasets were analyzed using the BABS-RNASeq <b>Nextflow</b> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) pipeline developed at the Francis Crick Institute.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33664263">33664263</a> <b>(1)</b> A chromosome-level genome of <italic>Astyanax mexicanus</italic> surface fish for comparing population-specific genetic differences contributing to trait evolution<ul><li> <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR86"><u>86</u></xref></sup> workflows for our running of both SV callers can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/esrice/workflows">https://github.com/esrice/workflows</ext-link>.<li> [86](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35647481">35647481</a> <b>(6)</b> nf-gwas-pipeline: A <b>Nextflow</b> Genome-Wide Association Study Pipeline<ul><li>The pipeline is implemented in <b>Nextflow</b>, a portable, scalable and parallelizable reactive workflow framework for data-intensive pipelines (<xref rid="R4" ref-type="bibr"><u>Di Tommaso P et al., 2017</u></xref>).<li> Through <b>Nextflow</b>, the GWAS pipeline is designed to automatically parallelize workflow steps, denoted as processes, and does not require users to manually handle intermediate data between steps, thus improving traceability and reproducibility.<li> Adding processes (i.e., additional analytical steps) in <b>Nextflow</b> framework is standardized and simple.<li> Advanced <b>Nextflow</b> users can define and incorporate additional processes in the pipeline framework, which will then be executed as part of the customized pipeline.<li> In addition, <b>Nextflow</b> creates its own work folder to hold intermediate output files, which serves as a record of a run.<li> While three types of analyses are currently available, the <b>Nextflow</b>-based pipeline is easy to extend with new tools and methods.<li> [Di Tommaso P et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630830">33630830</a> <b>(3)</b> Using a multiple-delivery-mode training approach to develop local capacity and infrastructure for advanced bioinformatics in Africa<ul><li> <b>Nextflow</b> (version 20.07.1) and Java (version 1.8 or higher); and hosting of an R-Studio (version 1.2.5042) Singularity container with the required R (version 3.6.3) analysis and visualisation packages installed with the ability for users to interactively log into a compute node to execute the container (<ext-link ext-link-type="uri" xlink:href="https://github.com/ilifu/bioinformatics/blob/master/containers/R/RStudio/bionic-R3.6.3-RStudio1.2.5042-bio.def">https://github.com/ilifu/bioinformatics/blob/master/containers/R/RStudio/bionic-R3.6.3-RStudio1.2.5042-bio.def</ext-link>) for further analysis.<li> An initial 16S rRNA Microbiome workflow was redeveloped to version 2 using <b>Nextflow</b> (version 20.07.1) and nf-core design principles [<xref rid="pcbi.1008640.ref009" ref-type="bibr"><u>9</u></xref>] consisting of 2 Docker containers, one for NGS QC (version 17.09.0-ce) and the second Docker container (version 19.03.11) incorporating the DADA2 pipeline (version 1.14.0) and its various dependencies that are built, tested, and hosted on Quay.IO (docker pull quay.io/h3abionet_org/h3a16s-fastqc; docker pull quay.io/h3abionet_org/16s-rdna-dada2-pipeline).<li><bold>Reproducibility&#x2013;</bold><b>Nextflow</b> raw outputs were requested from all sites (and included results from random students) to test reproducibility and all these results were identical.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a> <b>(51)</b> Using prototyping to choose a bioinformatics workflow management system<ul><li> and <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>nextflow</b>-io/<b>nextflow</b>">https://github.com/<b>nextflow</b>-io/<b>nextflow</b></ext-link>) [<xref rid="pcbi.1008622.ref009" ref-type="bibr"><u>9</u></xref>].<li> From our evaluation, via development of these prototypes, <b>Nextflow</b> was chosen.<li> repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/riboviz/workflows">https://github.com/riboviz/workflows</ext-link>) [<xref rid="pcbi.1008622.ref010" ref-type="bibr"><u>10</u></xref>] contains notes made during the selection and the Snakemake, <b>Nextflow</b>, and CWL workflows that were prototyped.<li>To avoid being rendered indecisive through over-choice, and, as we did not have time to review every popular option, even for shortlisting, we decided to choose only 2 systems: Snakemake and <b>Nextflow</b>.<li><b>Nextflow</b>.<li><b>Nextflow</b></th>                </tr>              </thead>              <tbody>                <tr>                  <td align="left" rowspan="1" colspan="1">Installation method used.<li> This ease of implementation accounts for the discrepancy in the Number of RiboViz processing steps implemented, as compared to cwltool, Toil, and <b>Nextflow</b>, shown in <xref rid="pcbi.1008622.t005" ref-type="table"><u>Table 5</u></xref>.<li><b>Nextflow</b>.<li><b>Nextflow</b> was easy to download and install, via the conda package manager, and had a simple tutorial.<li>A <b>Nextflow</b> workflow has a structure analogous to a Makefile or Snakefile&#x2014;it consists of a set of processes which define inputs, outputs, and commands describing how to create the outputs from the inputs.<li> However, <b>Nextflow</b> adopts a dataflow programming model whereby the processes are connected via their outputs and inputs to other processes, and processes run as soon as they receive an input.<li> Unlike Snakemake, a user does not specify the files they want to create, rather, they declare their input files and related configuration, and <b>Nextflow</b> continues to invoke processes until no process has any outstanding inputs.<li><b>Nextflow</b> is implemented in Java.<li> Code written in Groovy, which runs under Java, can be embedded within a <b>Nextflow</b> workflow.<li> It was not necessary to implement any additional code beyond the <b>Nextflow</b> workflow itself.<li> <b>Nextflow</b>&#x2019;s documentation was comprehensive, but additional concrete examples of how to implement common use cases, and an expanded tutorial how to implement a complex multistep workflow, would have been helpful.<li> Although we overlooked it during prototyping, the <b>Nextflow</b> web site does have a repository of common implementation patterns [<xref rid="pcbi.1008622.ref031" ref-type="bibr"><u>31</u></xref>].<li> Similarly, we later found a <b>Nextflow</b> 2017 workshop tutorial [<xref rid="pcbi.1008622.ref032" ref-type="bibr"><u>32</u></xref>], but our understanding of <b>Nextflow</b> had, at that point, passed beyond its content.<li> Despite this, writing a <b>Nextflow</b> workflow was easier than writing a CWL workflow.<li><b>Nextflow</b> provided all the required and most of the useful functionality listed in our evaluation criteria.<li> <b>Nextflow</b>, like Snakemake, supports both incremental build and re-entrancy, via a &#x201C;resume&#x201D; option.<li> These directories have auto-generated names, but <b>Nextflow</b> allows the contents of these directories to be written into known locations with more readable names.<li> analogous to that supported by Snakemake and Make, has been suggested in a <b>Nextflow</b> issue [<xref rid="pcbi.1008622.ref033" ref-type="bibr"><u>33</u></xref>], but has not progressed due to challenges in implementing such a feature within a dataflow model.<li> The <b>Nextflow</b> authors instead recommend using small datasets to validate scripts.<li><b>Nextflow</b> has support for running its jobs within containers, HPC systems, and cloud.<li>We decided to adopt <b>Nextflow</b> for the following reasons.<li> While writing <b>Nextflow</b> workflows does require knowledge of Groovy, the authors, familiar with Python and R, did not find learning Groovy challenging.<li> The fact that <b>Nextflow</b> was based on Java incurs no additional installation overhead for either users or developers compared to Snakemake&#x2014;each can be installed using the conda package manager using a single command.<li> Based on our impressions of their documentation, <b>Nextflow</b>&#x2019;s built-in support for, and documentation around, containers, HPC systems, and cloud, seemed more thorough than that of Snakemake (although we appreciate that this may change as both tools evolve).<li> It was our subjective impression that <b>Nextflow</b> felt far richer than Snakemake both in terms of features and expressivity, and it was felt that these outweighed its lack of a dry run feature.<li> Finally, we had to pick one, and we just liked <b>Nextflow</b> better.<li>Completing the RiboViz workflow implementation in <b>Nextflow</b>.<li>It took approximately 5 person-days to complete an implementation of the RiboViz workflow (including support for multiplexed files) within <b>Nextflow</b>.<li> Our existing regression test framework for our Python script was used to validate the implementation of our <b>Nextflow</b> script.<li>The <b>Nextflow</b> implementation has been tested by the RiboViz development team on their own development platforms and also on EDDIE, The University of Edinburgh&#x2019;s HPC cluster (<ext-link ext-link-type="uri" xlink:href="https://www.ed.ac.uk/information-services/research-support/research-computing/ecdf/high-performance-computing">https://www.ed.ac.uk/information-services/research-support/research-computing/ecdf/high-performance-computing</ext-link>).<li>Release 2.0 of RiboViz [<xref rid="pcbi.1008622.ref034" ref-type="bibr"><u>34</u></xref>] includes the <b>Nextflow</b> implementation of the RiboViz workflow.<li><b>Nextflow</b> has the nf-core collection of bioinformatics pipelines, a resource of open-source, reviewed, and validated <b>Nextflow</b> scripts implementing common data analyses [<xref rid="pcbi.1008622.ref035" ref-type="bibr"><u>35</u></xref>].<li> <ext-link ext-link-type="uri" xlink:href="https://nf-co.re/community">https://nf-co.re/community</ext-link>) has some overlap with the <b>Nextflow</b> developers, but is primarily composed of bioinformaticians.<li> Our positive experiences with Snakemake, and the small effort that would be required to complete the implementation of RiboViz into Snakemake, give us confidence that if we need to migrate from <b>Nextflow</b> to Snakemake in future, then this would be a relatively straightforward migration to undertake.<li> The use of <b>Nextflow</b> provides us with an implementation of RiboViz that is more maintainable, more portable, and which will allow us to exploit the power of distributed computing resources in the analysis of large-scale datasets.<li> Our prototype of the RiboViz workflow in <b>Nextflow</b>, created as part of our comparative evaluation of our shortlisted systems, provided a sound basis for completing our implementation once we had decided upon <b>Nextflow</b> as our chosen workflow management system.<li> <b>Nextflow</b>, like all the workflow management systems described in this article, is under active development and is constantly evolving, as is the RiboViz workflow.<li> What <b>Nextflow</b> offers today may be deprecated, or nonexistent, 6 months or a year from now.<li> Similarly, the nature of the RiboViz workflow may evolve so that what it requires from <b>Nextflow</b> is no longer being provided, and we need to migrate to another workflow management system.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<name><surname>Jackson</surname><given-names>M</given-names></name>, <name><surname>Wallace</surname><given-names>E</given-names></name>, <name><surname>Kavoussanakis</surname><given-names>K</given-names></name>, <name><surname>Anderson</surname><given-names>F</given-names></name>. <article-title>Options for RiboViz workflow management</article-title>. [cited 2020 Oct 28] <source>Database: figshare</source> [Internet]. <year>2020</year>. <pub-id pub-id-type="doi">10.6084/m9.figshare.13147979</pub-id>) [31](Nextflow. Nextflow patterns. GitHub [Internet]. [cited 2020 Oct 19]. Available from: <ext-link ext-link-type="uri" xlink:href="http://nextflow-io.github.io/patterns/index.html">http://nextflow-io.github.io/patterns/index.html</ext-link>.) [32](<collab>Nextflow</collab>. <article-title>Nextflow tutorial</article-title>. <source>GitHub</source> [Internet]. <year>2018</year> [cited 2020 Aug 4]. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/nextflow-io/nf-hack17-tutorial">https://github.com/nextflow-io/nf-hack17-tutorial</ext-link>.) [33](<collab>Nextflow</collab>. <article-title>suggestion: run -dry #31</article-title>. <source>GitHub</source> [Internet]. <year>2015</year> [updated 2020 Jan 14; cited 2020 Aug 4]. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/nextflow-io/nextflow/issues/31">https://github.com/nextflow-io/nextflow/issues/31</ext-link>.) [34](<name><surname>Wallace</surname><given-names>E</given-names></name>, <name><surname>Anderson</surname><given-names>F</given-names></name>, <name><surname>Kavoussanakis</surname><given-names>K</given-names></name>, <name><surname>Jackson</surname><given-names>M</given-names></name>, <name><surname>Shah</surname><given-names>P</given-names></name>, <name><surname>Lareau</surname><given-names>L</given-names></name>. <article-title>riboviz: software for analysis and visualization of ribosome profiling datasets</article-title>. <year>2020</year><month>7</month><day>8</day> [cited 2020 Aug 4]. <source>Database: figshare</source> [Internet]. <pub-id pub-id-type="doi">10.6084/m9.figshare.12624200</pub-id>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33625229">33625229</a> <b>(1)</b> BioContainers Registry: searching bioinformatics and proteomics tools, packages, and containers<ul><li> These tools are used in combinations with workflows such as Galaxy <sup><xref rid="R13" ref-type="bibr"><u>13</u></xref></sup>, <b>Nextflow</b> <sup><xref rid="R14" ref-type="bibr"><u>14</u></xref></sup>, or Snakemake <sup><xref rid="R15" ref-type="bibr"><u>15</u></xref></sup> which probably are the reason why they are more used than other more windows-based tools such as MaxQuant or Skyline.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33627512">33627512</a> <b>(1)</b> Primer, Pipelines, Parameters: Issues in 16S rRNA Gene Sequencing<ul><li>The three mock communities and human data sets were analyzed using the nfcore/ampliseq <b>nextflow</b> pipeline (<xref rid="B111" ref-type="bibr"><u>111</u></xref>, <xref rid="B112" ref-type="bibr"><u>112</u></xref>).<li> [111](<person-group person-group-type="author"><string-name><surname>Peltzer</surname><given-names>A</given-names></string-name>, <string-name><surname>Straub</surname><given-names>D</given-names></string-name>, <string-name><surname>Patel</surname><given-names>H</given-names></string-name></person-group>. <year>2019</year>. <article-title>nf-core/ampliseq: Ampliseq version 1.1.2</article-title>. <pub-id pub-id-type="doi">10.5281/zenodo.3585924</pub-id>.) [112](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33592074">33592074</a> <b>(1)</b> Prospective observational study and serosurvey of SARS-CoV-2 infection in asymptomatic healthcare workers at a Canadian tertiary care center<ul><li> Reads were aligned to a SARS-CoV-2 reference genome (GenBank: MN908947.3) using a <b>Nextflow</b> workflow [<xref rid="pone.0247258.ref015" ref-type="bibr"><u>15</u></xref>] that generates a consensus sequence from Illumina reads using the ARTIC network nCoV-2019 novel coronavirus bioinformatics protocol [<xref rid="pone.0247258.ref014" ref-type="bibr"><u>14</u></xref>].<li> [15](Connor Lab. ncov2019-artic-nf&#x2014;A Nextflow pipeline for running the ARTIC network&#x2019;s fieldbioinformatics tools 2020. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/connor-lab/ncov2019-artic-nf/tree/9a91994b25adcb6b1ac6b5af48321fad37dcce2a">https://github.com/connor-lab/ncov2019-artic-nf/tree/9a91994b25adcb6b1ac6b5af48321fad37dcce2a</ext-link>.) [14](Artic Network. nCoV-2019 novel coronavirus bioinformatics protocol 2020. Available from: <ext-link ext-link-type="uri" xlink:href="https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html">https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html</ext-link>.) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33748749">33748749</a> <b>(15)</b> Named Data Networking for Genomics Data Management and Integrated Workflows<ul><li> GEMmaker is regulated by the <b>Nextflow</b> (<xref rid="B35" ref-type="bibr"><b>NextFlow</b>, 2019</xref>) (<xref rid="B12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) workflow manager, which automates the fluid execution of each step in GEMmaker, which are defined as <b>Nextflow</b> processes.<li> <b>Nextflow</b> has a variety of executors that enable the deployment of processes to a number of HPC and cloud environments, including Kubernetes (K8s).<li> <b>Nextflow</b> is able to mount and unmount GEMmaker pods directly to this PVC, which stores all input, intermediate, and output data.<li> When the user executes GEMmaker, <b>Nextflow</b> deploys pods that automatically retrieve the RNA-seq data directly from NCBI using Aspera or FTP services.<li><b>Nextflow</b>.<li><ext-link ext-link-type="uri" xlink:href="https://github.com/<b>nextflow</b>-io/<b>nextflow</b>">https://github.com/<b>nextflow</b>-io/<b>nextflow</b></ext-link>.<li><b>Nextflow</b> is a general purpose workflow manager.<li> GEMmaker is a genomics workflow adpated for <b>Nextflow</b> (see <xref ref-type="sec" rid="s4-3"><u>Section 4.3</u></xref> above for more details).<li><b>Nextflow</b> managed GEMmaker workflow on a Kubernetes Cluster.<li>GEMmaker is able to run <b>nextflow</b> processes inside specified containers.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33563213">33563213</a> <b>(6)</b> lncEvo: automated identification and conservation study of long noncoding RNAs<ul><li>The <b>Nextflow</b> script is transparently executed in a Docker container.<li> <b>Nextflow</b> automatically manages the computing environment, allocates resources, and initiates and terminates nodes for each task according to the executed directed acyclic graph (DAG), job queues and computing environments.<li> <italic><b>Nextflow</b></italic> retrieves the credentials from the user ~<italic>/.aws/credentials</italic> or ~<italic>/.aws/config</italic> files; alternatively, <italic>AWS_ACCESS_KEY_ID</italic> and <italic>AWS_SECRET_ACCESS_KEY</italic> may be specified in the environment.<li><b>Nextflow</b> generates an HTML execution report with a number of metrics regarding workflow execution.<li>Programming language: <b>Nextflow</b> DSL.<li>Other requirements:&#x2009;&gt;=&#x2009;Java 8,&#x2009;&gt;=&#x2009;<b>Nextflow</b> 2.20,&#x2009;&gt;=&#x2009;Docker 19.03.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33561126">33561126</a> <b>(15)</b> Metagenomics workflow for hybrid assembly, differential coverage binning, metatranscriptomics and pathway analysis (MUFFIN)<ul><li> We combined the workflow manager <b>Nextflow</b> [<xref rid="pcbi.1008716.ref014" ref-type="bibr"><u>14</u></xref>] with virtualization software (so-called &#x2019;containers&#x2019;) to generate reproducible results in various working environments and allow full parallelization of the workload to a higher degree.<li>MUFFIN integrates state-of-the-art bioinformatic tools via Conda recipes or Docker/Singularity containers for the processing of metagenomic sequences in a <b>Nextflow</b> workflow environment (<xref ref-type="fig" rid="pcbi.1008716.g001"><u>Fig 1</u></xref>).<li> One of them is the workflow management system <b>Nextflow</b> [<xref rid="pcbi.1008716.ref014" ref-type="bibr"><u>14</u></xref>] (version 20.07+), and the other can be either Conda <sup>20</sup>[<xref rid="pcbi.1008716.ref022" ref-type="bibr"><u>22</u></xref>] as a package manager or Docker [<xref rid="pcbi.1008716.ref023" ref-type="bibr"><u>23</u></xref>] / Singularity to use containerized tools.<li> Each MUFFIN release specifies the <b>Nextflow</b> version it was tested on, but any version of MUFFIN V1.0.2+ will work with <b>nextflow</b> version 20.07+.<li> A <b>Nextflow</b>-specific version can always be directly downloaded as an executable file from <ext-link ext-link-type="uri" xlink:href="https://github.com/<b>nextflow</b>-io/<b>nextflow</b>/releases">https://github.com/<b>nextflow</b>-io/<b>nextflow</b>/releases</ext-link>, which can then be paired with a compatible MUFFIN version via the -r flag.<li>We chose <b>Nextflow</b> for the development of our metagenomic workflow because of its direct cloud computing support (Amazon AWS, Google Life Science, Kubernetes), various ready-to-use batch schedulers (SGE, SLURM, LSF), state-of-the-art container support (Docker, Singularity), and accessibility of a widely used software package manager (Conda).<li> Moreover, <b>Nextflow</b> [<xref rid="pcbi.1008716.ref014" ref-type="bibr"><u>14</u></xref>] provides a practical and straightforward intermediary file handling with process-specific work directories and the possibility to resume failed executions where the work ceased.<li> The MUFFIN workflow gathers the different steps of a metagenomics analysis in an easy-to-install, highly reproducible, and scalable workflow using <b>Nextflow</b>, which makes them easily accessible to researchers.<li> Due to the modular coding structure of <b>nextflow</b> DSL2 language, MUFFIN can quickly adapt towards better tools or improved versions if necessary, in the future.<li> The modular workflow setup of MUFFIN using <b>Nextflow</b> allows for fast adjustments as soon as future developments in hybrid metagenomics arise, including the pre-configuration for other workload managers.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [22](Anaconda Software distribution. Anaconda | The World&#x2019;s Most Popular Data Science Platform. In: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link> [Internet]. 5 Feb 2020 [cited 5 Feb 2020]. Available: <ext-link ext-link-type="uri" xlink:href="https://www.anaconda.com/">https://www.anaconda.com/</ext-link>) [23](<name><surname>Boettiger</surname><given-names>C</given-names></name>. <article-title>An introduction to Docker for reproducible research</article-title>. <source>ACM SIGOPS Oper Syst Rev</source>. <year>2015</year>;<volume>49</volume>: <fpage>71</fpage>&#x2013;<lpage>79</lpage>. <pub-id pub-id-type="doi">10.1145/2723872.2723882</pub-id>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a> <b>(5)</b> Metagenomics workflow for hybrid assembly, differential coverage binning, metatranscriptomics and pathway analysis (MUFFIN)<ul><li> How can a user make sure that they pick the correct version of <b>nextflow</b> or Muffin? Isn't there another way where it's not up to the user to read the release notes to exclude version conflicts e.g.<li> a conda package providing the tested version of muffin and <b>nextflow</b>?.<li>Submitted filename: <named-content content-type="submitted-filename"><b>nextflow</b>.log.3</named-content>.<li>Submitted filename: <named-content content-type="submitted-filename"><b>nextflow</b>.log</named-content>.<li>Submitted filename: <named-content content-type="submitted-filename"><b>nextflow</b>_local.log</named-content>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33526779">33526779</a> <b>(4)</b> Identification and analysis of splicing quantitative trait loci across multiple tissues in the human genome<ul><li> To ensure highly parallel, portable and reproducible sQTL mapping, we embedded sQTLseekeR2 in a <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup> (plus Docker, <ext-link ext-link-type="uri" xlink:href="https://www.docker.com">https://www.docker.com</ext-link>) computational workflow named sqtlseeker2-nf, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/guigolab/sqtlseeker2-nf">https://github.com/guigolab/sqtlseeker2-nf</ext-link>.<li>In summary, our implementation of an enhanced pipeline for sQTL mapping based on sQTLseekeR2, <b>Nextflow</b> and Docker will help sQTL discovery in multiple datasets, across different platforms, in a highly parallel and reproducible manner.<li> sQTLseekeR2 was embedded in sqtlseeker2-nf, a highly parallel, portable and reproducible pipeline for sQTL mapping developed using <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup>, a framework for computational workflows, and Docker container technology.<li> A <b>Nextflow</b> implementation of the Integrative Pipeline for Splicing Analyses (IPSA), developed in-house (<ext-link ext-link-type="uri" xlink:href="https://github.com/guigolab/ipsa-nf">https://github.com/guigolab/ipsa-nf</ext-link>), was employed to determine the number of reads supporting splicing completion and splicing incompletion, for each intron annotated in GENCODE v19.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33524142">33524142</a> <b>(1)</b> Flexible comparison of batch correction methods for single-cell RNA-seq using BatchBench<ul><li> BatchBench is implemented in <b>Nextflow</b> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) and it is freely available at <ext-link ext-link-type="uri" xlink:href="https://github.com/cellgeni/batchbench">https://github.com/cellgeni/batchbench</ext-link> under the MIT Licence.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33514375">33514375</a> <b>(1)</b> XenoCell: classification of cellular barcodes in single cell experiments from xenograft samples<ul><li>XenoCell is implemented in Python and made available as a Docker image, which contains all third-party software dependencies as a pre-configured system in order to facilitate portability across platforms and easier integration into existing workflows, such as those built with the Snakemake [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] or <b>Nextflow</b> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>] frameworks.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33572758">33572758</a> <b>(1)</b> The Expression of the Cancer-Associated lncRNA <italic>Snhg15</italic> Is Modulated by EphrinA5-Induced Signaling<ul><li>The paired-end FASTA files have been processed by the <b>nextflow</b> pipeline nfcore/rnaseq v1.4.2 mainly using the standard settings [<xref rid="B70-ijms-22-01332" ref-type="bibr"><u>70</u></xref>].<li> [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33532801">33532801</a> <b>(2)</b> Immune response to SARS-CoV-2 in the nasal mucosa in children and adults<ul><li> To facilitate reproducible analysis, samples were processed using the publicly available nf-core/RNA-seq pipeline version 1.4.2 implemented in <b>Nextflow</b> 19.10.0 using Singularity 3.2.1&#x2013;1 with the minimal command <b>nextflow</b> run nf-core/rnaseq -r 1.4.2 &#x2013;singleEnd &#x2013;profile singularity &#x2013;reverseStranded --three_prime_clip_r2 3 (<xref rid="R27" ref-type="bibr"><u>27</u></xref>&#x2013;<xref rid="R29" ref-type="bibr"><u>29</u></xref>).<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33320836">33320836</a> <b>(2)</b> The MUC5B-associated variant rs35705950 resides within an enhancer subject to lineage- and disease-dependent epigenetic remodeling<ul><li>PRO-seq data were processed using a standardized <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/Dowell-Lab/Nascent-Flow">https://github.com/Dowell-Lab/Nascent-Flow</ext-link>; master; commit d5946017c7e33bda3e990b10c177a693a3164dc1).<li>ATAC-seq data were processed using a standardized <b>Nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/Dowell-Lab/ChIP-Flow">https://github.com/Dowell-Lab/ChIP-Flow</ext-link>; master; commit 44fd202d9fa2355366e1ac20f41cf67fd4a6ebc4).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33469076">33469076</a> <b>(1)</b> Blood-based protein profiling identifies serum protein c-KIT as a novel biomarker for hypertrophic cardiomyopathy<ul><li> The pipeline uses <b>Nextflow</b>, a bioinformatics workflow tool.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33584814">33584814</a> <b>(1)</b> ATACgraph: Profiling Genome-Wide Chromatin Accessibility From ATAC-seq<ul><li> nf-core/atacseq (<xref rid="B5" ref-type="bibr"><u>Ewels et al., 2020</u></xref>) is a pipeline built by <b>Nextflow</b>, a workflow tool, to run multiple tasks across multiple computing infrastructures.<li> [Ewels et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33438730">33438730</a> <b>(2)</b> Streamlining data-intensive biology with workflow systems<ul><li> Snakemake and <b>Nextflow</b> [<xref rid="bib25" ref-type="bibr"><u>25</u></xref>] are commonly used for developing new research pipelines, where flexibility and iterative, branching development are key features.<li><b>Nextflow</b>.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33429418">33429418</a> <b>(2)</b> Circuits between infected macrophages and T cells in SARS-CoV-2 pneumonia<ul><li>To facilitate reproducible analysis, samples were processed using the publicly available nf-core/RNA-seq pipeline version 1.4.2 implemented in <b>Nextflow</b> 19.10.0 using Singularity 3.2.1&#x2013;1 with the minimal command <b>nextflow</b> run nf-core/rnaseq -r 1.4.2 &#x2013;singleEnd -profile singularity &#x2013;reverseStranded --three_prime_clip_r2 3<sup><xref ref-type="supplementary-material" rid="SD13"><u>53</u></xref>&#x2013;<xref ref-type="supplementary-material" rid="SD13"><u>55</u></xref></sup>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33420778">33420778</a> <b>(1)</b> Significantly improving the quality of genome assemblies through curation<ul><li> Before being loaded into gEVAL, all assemblies are run through a <b>nextflow</b> [<xref rid="bib39" ref-type="bibr"><u>38</u></xref>] pipeline that performs contamination detection and separation or removal as described in Table&#xA0;<xref rid="tbl1" ref-type="table"><u>1</u></xref>, combined with removal of trailing Ns [<xref rid="bib39" ref-type="bibr"><u>38</u></xref>].<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33490975">33490975</a> <b>(6)</b> Extensible combinatorial CRISPR screening in mammalian cells<ul><li> The pipeline is written in <b>nextflow</b>, and running the pipeline requires <b>nextflow</b> installation (<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/usage/installation" id="intref0020">https://nf-co.re/usage/installation</ext-link>).<li>The <b>nextflow</b>.config file contains all arguments used when running CombiPIPE.<li>The CombiGEM.nf file is the <b>nextflow</b> pipeline that connects the processes in CombiPIPE.<li>Run the pipeline with <b>nextflow</b>:<list list-type="simple" id="ulist0055"><list-item id="u0130"><p id="p0700">$ <b>nextflow</b> CombiGEM.nf --fastq test-dataset/OVARIAN_testdataset.fastq --sampinfo test-dataset/sampleInfo.csv --barcodes test-dataset/barcode_list.csv --pattern @D00691 --dimensions 3 --linker CAATTC --dummysgs 1,2.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33275900">33275900</a> <b>(1)</b> Evaluating the Effects of SARS-CoV-2 Spike Mutation D614G on Transmissibility and Pathogenicity<ul><li> <xref rid="bib34" ref-type="bibr"><u>Rambaut et&#xA0;al., 2020</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://www.iqtree.org/" id="intref0085">http://www.iqtree.org/</ext-link>                  </td>                </tr>                <tr>                  <td>MRC-CLIMB</td>                  <td>(<xref rid="bib4" ref-type="bibr"><u>Connor et&#xA0;al., 2016</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.climb.ac.uk/" id="intref0090">https://www.climb.ac.uk/</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b> pipeline for processing/assembly of ARTIC protocol amplicons.<li> [Rambaut et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32669681">32669681</a>) [Connor et&#xA0;al., 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28785418">28785418</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33404053">33404053</a> <b>(1)</b> Tool recommender system in Galaxy using deep learning<ul><li> To simplify the data analysis process and to ensure a reproducible analysis, several workflow systems such as Bcbio-nextgen, Omics Pipe, <b>Nextflow</b>, Luigi, Toil, and many others have emerged [<xref rid="bib1" ref-type="bibr"><u>1&#x2013;3</u></xref>].<li> [1&#x2013;3](<person-group person-group-type="author"><name name-style="western"><surname>Ewels</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Krueger</surname><given-names>F</given-names></name>, <name name-style="western"><surname>K&#xE4;ller</surname><given-names>M</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Cluster Flow: A user-friendly bioinformatics workflow tool</article-title>. <source>F1000Res</source>. <year>2017</year>;<volume>5</volume>:<fpage>2824</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33350904">33350904</a> <b>(1)</b> Susceptibility of Domestic Swine to Experimental Infection with Severe Acute Respiratory Syndrome Coronavirus 2<ul><li> Data analysis also were performed by the CFIA NCFAD Genomics Unit using nf-villumina version 2.0.0 (<ext-link xlink:href="https://github.com/peterk87/nf-villumina" ext-link-type="uri">https://github.com/peterk87/nf-villumina</ext-link>), an in-house workflow developed by using <b>Nextflow</b> (<xref rid="R26" ref-type="bibr"><italic>26</italic></xref>), which performed read quality filtering with fastp (<xref rid="R27" ref-type="bibr"><italic>27</italic></xref>); Centrifuge version 1.0.4-beta (<xref rid="R28" ref-type="bibr"><italic>28</italic></xref>) and Kraken2 version 2.0.8 (<xref rid="R29" ref-type="bibr"><italic>29</italic></xref>) read taxonomic classification using an index of the National Center for Biotechnology Information (NCBI) nucleotide database (downloaded 2020 Feb 4); a Kraken2 index of NCBI RefSeq (<ext-link xlink:href="https://www.ncbi.nlm.nih.gov/refseq" ext-link-type="uri">https://www.ncbi.nlm.nih.gov/refseq</ext-link>) sequences of archaea, bacterial, viral, and human genomes GRCh38 (downloaded and built on 2019 Mar 22); removal of nonviral reads (i.e., not classified as belonging to superkingdom &#x201C;Viruses&#x201D; (NCBI taxonomic identification 10239) by using Kraken2 and Centrifuge taxonomic classification results; de novo metagenomics assembly of taxonomically filtered reads by Shovill version 1.0.9 (<xref rid="R30" ref-type="bibr"><italic>30</italic></xref>), Unicycler version 1.0.9 (<xref rid="R31" ref-type="bibr"><italic>31</italic></xref>), and Megahit version 1.2.9 (<xref rid="R32" ref-type="bibr"><italic>32</italic></xref>); and nucleotide BLAST+ version 2.9.0 (<xref rid="R33" ref-type="bibr"><italic>33</italic></xref>,<xref rid="R34" ref-type="bibr"><italic>34</italic></xref>) search of all assembled contigs against the NCBI nucleotide BLAST database (downloaded 2020 April 10) using the &#x201C;update_blastb.pl&#x201D; script as part of the blast Bioconda package (<xref rid="R35" ref-type="bibr"><italic>35</italic></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33357438">33357438</a> <b>(5)</b> RTEL1 Regulates G4/R-Loops to Avert Replication-Transcription Collisions<ul><li>                    <hr/>                  </td>                </tr>                <tr>                  <td>Adobe Photoshop CC</td>                  <td>Adobe</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.adobe.com/es/products/photoshop.html" id="intref0035">https://www.adobe.com/es/products/photoshop.html</ext-link>                  </td>                </tr>                <tr>                  <td>ImageJ</td>                  <td>NIH</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0040">https://imagej.nih.gov/ij/</ext-link>                  </td>                </tr>                <tr>                  <td>Volocity 6.3</td>                  <td>PerkinElmer</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.perkinelmer.com/uk/lab-products-and-services/resources/cellular-imaging-software-downloads.html" id="interref0015">https://www.perkinelmer.com/uk/lab-products-and-services/resources/cellular-imaging-software-downloads.html</ext-link>                  </td>                </tr>                <tr>                  <td>GraphPad Prism 7</td>                  <td>GraphPad</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.graphpad.com/" id="intref0050">https://www.graphpad.com/</ext-link>                  </td>                </tr>                <tr>                  <td>Flow Jo 10</td>                  <td>FlowJo</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.flowjo.com/" id="intref0055">https://www.flowjo.com/</ext-link>                  </td>                </tr>                <tr>                  <td>FastQC</td>                  <td>                    <xref rid="bib11" ref-type="bibr"><u>Daley and Smith, 2013</u></xref>                  </td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" id="intref0060">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>                  </td>                </tr>                <tr>                  <td>nf-core/chipseq</td>                  <td>                    <xref rid="bib17" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>                  </td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/3966161" id="intref0065">https://zenodo.org/record/3966161</ext-link>                  </td>                </tr>                <tr>                  <td>picard-tools</td>                  <td>Broad Institute</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://broadinstitute.github.io/picard" id="intref0070">http://broadinstitute.github.io/picard</ext-link>                  </td>                </tr>                <tr>                  <td>Pysam</td>                  <td>GitHub</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://github.com/pysam-developers/pysam" id="intref0075">https://github.com/pysam-developers/pysam</ext-link>                  </td>                </tr>                <tr>                  <td>ggplot2</td>                  <td>(<xref rid="bib59" ref-type="bibr"><u>Wickham et al., 2016</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://ggplot2-book.org/" id="intref0080">https://ggplot2-book.org/</ext-link>                  </td>                </tr>                <tr>                  <td>cutadapt</td>                  <td>(<xref rid="bib60" ref-type="bibr"><u>Martin, 2011</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://journal.embnet.org/index.php/embnetjournal/article/view/200" id="intref0085">https://journal.embnet.org/index.php/embnetjournal/article/view/200</ext-link>                  </td>                </tr>                <tr>                  <td>Fgsea</td>                  <td>(<xref rid="bib62" ref-type="bibr"><u>Korotkevich et al., 2019</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/10.1101/060012v2.full" id="intref0090">https://www.biorxiv.org/content/10.1101/060012v2.full</ext-link>                  </td>                </tr>                <tr>                  <td>Trim Galore!</td>                  <td>Babraham Bioinformatics</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" id="intref0095">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b>.<li>                    <hr/>                  </td>                </tr>                <tr>                  <td>Adobe Photoshop CC</td>                  <td>Adobe</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.adobe.com/es/products/photoshop.html" id="intref0035">https://www.adobe.com/es/products/photoshop.html</ext-link>                  </td>                </tr>                <tr>                  <td>ImageJ</td>                  <td>NIH</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0040">https://imagej.nih.gov/ij/</ext-link>                  </td>                </tr>                <tr>                  <td>Volocity 6.3</td>                  <td>PerkinElmer</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.perkinelmer.com/uk/lab-products-and-services/resources/cellular-imaging-software-downloads.html" id="interref0015">https://www.perkinelmer.com/uk/lab-products-and-services/resources/cellular-imaging-software-downloads.html</ext-link>                  </td>                </tr>                <tr>                  <td>GraphPad Prism 7</td>                  <td>GraphPad</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.graphpad.com/" id="intref0050">https://www.graphpad.com/</ext-link>                  </td>                </tr>                <tr>                  <td>Flow Jo 10</td>                  <td>FlowJo</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.flowjo.com/" id="intref0055">https://www.flowjo.com/</ext-link>                  </td>                </tr>                <tr>                  <td>FastQC</td>                  <td>                    <xref rid="bib11" ref-type="bibr"><u>Daley and Smith, 2013</u></xref>                  </td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" id="intref0060">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>                  </td>                </tr>                <tr>                  <td>nf-core/chipseq</td>                  <td>                    <xref rid="bib17" ref-type="bibr"><u>Ewels et&#xA0;al., 2020</u></xref>                  </td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/3966161" id="intref0065">https://zenodo.org/record/3966161</ext-link>                  </td>                </tr>                <tr>                  <td>picard-tools</td>                  <td>Broad Institute</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://broadinstitute.github.io/picard" id="intref0070">http://broadinstitute.github.io/picard</ext-link>                  </td>                </tr>                <tr>                  <td>Pysam</td>                  <td>GitHub</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://github.com/pysam-developers/pysam" id="intref0075">https://github.com/pysam-developers/pysam</ext-link>                  </td>                </tr>                <tr>                  <td>ggplot2</td>                  <td>(<xref rid="bib59" ref-type="bibr"><u>Wickham et al., 2016</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://ggplot2-book.org/" id="intref0080">https://ggplot2-book.org/</ext-link>                  </td>                </tr>                <tr>                  <td>cutadapt</td>                  <td>(<xref rid="bib60" ref-type="bibr"><u>Martin, 2011</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://journal.embnet.org/index.php/embnetjournal/article/view/200" id="intref0085">https://journal.embnet.org/index.php/embnetjournal/article/view/200</ext-link>                  </td>                </tr>                <tr>                  <td>Fgsea</td>                  <td>(<xref rid="bib62" ref-type="bibr"><u>Korotkevich et al., 2019</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/10.1101/060012v2.full" id="intref0090">https://www.biorxiv.org/content/10.1101/060012v2.full</ext-link>                  </td>                </tr>                <tr>                  <td>Trim Galore!</td>                  <td>Babraham Bioinformatics</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" id="intref0095">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b></td>                  <td>                    <xref rid="bib50" ref-type="bibr"><u>Tommaso et&#xA0;al., 2017</u></xref>                  </td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://<b>nextflow</b>.io/" id="intref0100">https://<b>nextflow</b>.io/</ext-link>                  .<li> <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/3966161" id="intref0265">https://zenodo.org/record/3966161</ext-link>) written in the <b>Nextflow</b> domain specific language (version 19.10.0; (<xref rid="bib50" ref-type="bibr"><u>Tommaso et&#xA0;al., 2017</u></xref>)) was used to perform the primary analysis of the samples in conjunction with Singularity (version 2.6.0; (<xref rid="bib26" ref-type="bibr"><u>Kurtzer et&#xA0;al., 2017</u></xref>)).<li> The command used was &#x201C;<b>nextflow</b> run nf-core/chipseq&#x2013;input design.csv&#x2013;genome mm10&#x2013;gtf refseq_genes.gtf&#x2013;single_end&#x2013;narrow_peak -profile crick -r 1.2.1.&#x201D; To summarize, the pipeline performs adaptor trimming (Trim Galore! - <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" id="intref0270">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</ext-link>), read alignment (BWA - (<xref rid="bib29" ref-type="bibr"><u>Li and Durbin, 2009</u></xref>)) and filtering (SAMtools - (<xref rid="bib30" ref-type="bibr"><u>Li et&#xA0;al., 2009</u></xref>); BEDTools - (<xref rid="bib36" ref-type="bibr"><u>Quinlan and Hall, 2010</u></xref>); BamTools - (<xref rid="bib4" ref-type="bibr"><u>Barnett et&#xA0;al., 2011</u></xref>); pysam - <ext-link ext-link-type="uri" xlink:href="https://github.com/pysam-developers/pysam" id="intref0275">https://github.com/pysam-developers/pysam</ext-link>; picard-tools - <ext-link ext-link-type="uri" xlink:href="http://broadinstitute.github.io/picard" id="intref0280">http://broadinstitute.github.io/picard</ext-link>), normalized coverage track generation (BEDTools - (<xref rid="bib36" ref-type="bibr"><u>Quinlan and Hall, 2010</u></xref>); bedGraphToBigWig - (<xref rid="bib24" ref-type="bibr"><u>Kent et&#xA0;al., 2010</u></xref>)), peak calling (MACS2 - (<xref rid="bib58" ref-type="bibr"><u>Zhang et&#xA0;al., 2008</u></xref>)) and annotation relative to gene features (HOMER - (<xref rid="bib21" ref-type="bibr"><u>Heinz et&#xA0;al., 2010</u></xref>)), consensus peak set creation (BEDTools -(<xref rid="bib36" ref-type="bibr"><u>Quinlan and Hall, 2010</u></xref>)), differential binding analysis (featureCounts - (<xref rid="bib31" ref-type="bibr"><u>Liao et&#xA0;al., 2014</u></xref>); R - R Core Team; DESeq2 - (<xref rid="bib32" ref-type="bibr"><u>Love et&#xA0;al., 2014</u></xref>)) and extensive QC and version reporting (MultiQC - (<xref rid="bib16" ref-type="bibr"><u>Ewels et&#xA0;al., 2016</u></xref>); FastQC - <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" id="intref0285">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>; preseq - (<xref rid="bib11" ref-type="bibr"><u>Daley and Smith, 2013</u></xref>); deepTools - (<xref rid="bib37" ref-type="bibr"><u>Ram&#xED;rez et&#xA0;al., 2016</u></xref>); phantompeakqualtools - (<xref rid="bib27" ref-type="bibr"><u>Landt et&#xA0;al., 2012</u></xref>)).<li> [Daley and Smith, 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23435259">23435259</a>) [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Wickham et al., 2016](          <person-group person-group-type="author">            <name>              <surname>Wickham</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Navarro</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Pedersen</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Lin</surname>            </name>          </person-group>          <source>ggplot2: elegant graphics for data analysi</source>          <year>2016</year>          <publisher-name>Springer</publisher-name>          <comment>In preparation</comment>        ) [Martin, 2011](          <person-group person-group-type="author">            <name>              <surname>Martin</surname>              <given-names>M.</given-names>            </name>          </person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet.journal</source>          <year>2011</year>          <pub-id pub-id-type="doi">10.14806/ej.17.1.200</pub-id>        ) [Korotkevich et al., 2019](Korotkevich. 2019. <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/10.1101/060012v2.full" id="interref0550">https://www.biorxiv.org/content/10.1101/060012v2.full</ext-link>.) [Daley and Smith, 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23435259">23435259</a>) [Ewels et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [Wickham et al., 2016](          <person-group person-group-type="author">            <name>              <surname>Wickham</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Navarro</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Pedersen</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Lin</surname>            </name>          </person-group>          <source>ggplot2: elegant graphics for data analysi</source>          <year>2016</year>          <publisher-name>Springer</publisher-name>          <comment>In preparation</comment>        ) [Martin, 2011](          <person-group person-group-type="author">            <name>              <surname>Martin</surname>              <given-names>M.</given-names>            </name>          </person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet.journal</source>          <year>2011</year>          <pub-id pub-id-type="doi">10.14806/ej.17.1.200</pub-id>        ) [Korotkevich et al., 2019](Korotkevich. 2019. <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/10.1101/060012v2.full" id="interref0550">https://www.biorxiv.org/content/10.1101/060012v2.full</ext-link>.) [Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a>) [Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a>) [Kurtzer et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [Li and Durbin, 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19451168">19451168</a>) [Li et&#xA0;al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [Quinlan and Hall, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [Barnett et&#xA0;al., 2011](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21493652">21493652</a>) [Quinlan and Hall, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [Kent et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20639541">20639541</a>) [Zhang et&#xA0;al., 2008](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18798982">18798982</a>) [Heinz et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20513432">20513432</a>) [Quinlan and Hall, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [Liao et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) [Love et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25516281">25516281</a>) [Ewels et&#xA0;al., 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27312411">27312411</a>) [Daley and Smith, 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23435259">23435259</a>) [Ram&#xED;rez et&#xA0;al., 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27079975">27079975</a>) [Landt et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22955991">22955991</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33125867">33125867</a> <b>(5)</b> Two MicroRNAs Are Sufficient for Embryonic Patterning in <italic>C.&#xA0;elegans</italic><ul><li>                    <hr/>                  </td>                </tr>                <tr>                  <td>RNAfold Two-State-Folding algorithm</td>                  <td>[<xref rid="bib53" ref-type="bibr"><u>53</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://unafold.rna.albany.edu/" id="intref0035">http://unafold.rna.albany.edu/</ext-link>                  </td>                </tr>                <tr>                  <td>miRNA analysis pipeline for sRNaseq data</td>                  <td>[<xref rid="bib28" ref-type="bibr"><u>28</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1038/nmeth.4610" id="intref0040">https://doi.org/10.1038/nmeth.4610</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b> pipeline.<li>                    <hr/>                  </td>                </tr>                <tr>                  <td>RNAfold Two-State-Folding algorithm</td>                  <td>[<xref rid="bib53" ref-type="bibr"><u>53</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://unafold.rna.albany.edu/" id="intref0035">http://unafold.rna.albany.edu/</ext-link>                  </td>                </tr>                <tr>                  <td>miRNA analysis pipeline for sRNaseq data</td>                  <td>[<xref rid="bib28" ref-type="bibr"><u>28</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1038/nmeth.4610" id="intref0040">https://doi.org/10.1038/nmeth.4610</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b> pipeline</td>                  <td>[<xref rid="bib54" ref-type="bibr"><u>54</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/about-us.html" id="intref0045">https://www.<b>nextflow</b>.io/about-us.html</ext-link>                  .<li>                    <hr/>                  </td>                </tr>                <tr>                  <td>RNAfold Two-State-Folding algorithm</td>                  <td>[<xref rid="bib53" ref-type="bibr"><u>53</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://unafold.rna.albany.edu/" id="intref0035">http://unafold.rna.albany.edu/</ext-link>                  </td>                </tr>                <tr>                  <td>miRNA analysis pipeline for sRNaseq data</td>                  <td>[<xref rid="bib28" ref-type="bibr"><u>28</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1038/nmeth.4610" id="intref0040">https://doi.org/10.1038/nmeth.4610</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b> pipeline</td>                  <td>[<xref rid="bib54" ref-type="bibr"><u>54</u></xref>]</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/about-us.html" id="intref0045">https://www.<b>nextflow</b>.io/about-us.html</ext-link>                  </td>                </tr>                <tr>                  <td><b>Nextflow</b> pipeline modifications.<li> Modifications of the <b>Nextflow</b> pipeline for can be found under: <ext-link ext-link-type="uri" xlink:href="https://github.com/lengfei5/smallRNA_nf/tree/master/dev_sRBC" id="intref0065">https://github.com/lengfei5/smallRNA_nf/tree/master/dev_sRBC</ext-link>.<li> adaptor-barcode demultiplexing, which was added after trimming adaptor sequence in the <b>Nextflow</b>-based pipeline [<xref rid="bib54" ref-type="bibr"><u>54</u></xref>].<li> The modified <b>Nextflow</b> pipeline can be found at: <ext-link ext-link-type="uri" xlink:href="https://github.com/lengfei5/smallRNA_nf/tree/master/dev_sRBC" id="intref0075">https://github.com/lengfei5/smallRNA_nf/tree/master/dev_sRBC</ext-link>.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15980540">15980540</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29481550">29481550</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15980540">15980540</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29481550">29481550</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15980540">15980540</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29481550">29481550</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33325479">33325479</a> <b>(3)</b> aCLImatise: automated generation of tool definitions for bioinformatics workflows<ul><li> Four popular workflow languages in bioinformatics are <b>Nextflow</b>, Snakemake, Workflow Definition Language (WDL) and Common Workflow Language (CWL) (<xref rid="btaa1033-B2" ref-type="bibr"><u>Bed&#x151;, 2019</u></xref>).<li> Such repositories exist for WDL (<ext-link ext-link-type="uri" xlink:href="https://github.com/biowdl/tasks">https://github.com/biowdl/tasks</ext-link>), Snakemake (<ext-link ext-link-type="uri" xlink:href="https://snakemake-wrappers.readthedocs.io/en/stable/">https://snakemake-wrappers.readthedocs.io/en/stable/</ext-link>), <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/modules">https://github.com/nf-core/modules</ext-link>) and CWL (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-library/bio-cwl-tools">https://github.com/common-workflow-library/bio-cwl-tools</ext-link>), while registries such as Dockstore cater to multiple workflow languages simultaneously (<xref rid="btaa1033-B13" ref-type="bibr">O&#x2019;Connor <italic>et al.<li> We envisage that Galaxy (<xref rid="btaa1033-B1" ref-type="bibr">Afgan <italic>et al.</italic>, 2018</xref>), <b>Nextflow</b> and Snakemake could be supported in the future.<li> [Bed&#x151;, 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31328031">31328031</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33325750">33325750</a> <b>(1)</b> Genomic analysis of SARS-CoV-2 reveals local viral evolution in Ghana<ul><li>A <b>Nextflow</b> pipeline that automates the ARTIC network nCoV-2019 novel coronavirus bioinformatics protocol was used in assembling the Illumina sequencing data into consensus genomes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33319904">33319904</a> <b>(7)</b> Can reproducibility be improved in clinical natural language processing? A study of 7 clinical NLP suites<ul><li> <b>Nextflow</b>,<xref rid="ocaa261-B16" ref-type="bibr"><sup>16</sup></xref> SnakeMake,<xref rid="ocaa261-B17" ref-type="bibr"><sup>17</sup></xref> and Galaxy<xref rid="ocaa261-B18" ref-type="bibr"><sup>18</sup></xref> are the most used workflow management systems within the bioinformatics community.<li> For example, QuickUMLS<xref rid="ocaa261-B54" ref-type="bibr"><sup>54</sup></xref> can easily be integrated into the WMS <b>Nextflow</b><xref rid="ocaa261-B16" ref-type="bibr"><sup>16</sup></xref> (eg, <ext-link ext-link-type="uri" xlink:href="https://github.com/equipe22/QuickUMLS_<b>Nextflow</b>">https://github.com/equipe22/QuickUMLS_<b>Nextflow</b></ext-link>).<li> Today, Snakemake, Galaxy, and <b>Nextflow</b> are adopted by the bioinformatics community.<li> The developer community uses all WMS with a preference for Snakemake and <b>Nextflow</b>.<li> Recently, the community of <b>Nextflow</b> users released nf-core,<xref rid="ocaa261-B60" ref-type="bibr"><sup>60</sup></xref> a shared repository of curated workflows.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33362872">33362872</a> <b>(1)</b> Editorial: Computational Epitranscriptomics: Bioinformatic Approaches for the Analysis of RNA Modifications<ul><li> A second study introduced MasterOfPores, a <b>NextFlow</b> workflow that facilitates the analysis of these data, allowing the prediction of RNA modifications and the estimation of polyA tail lengths (<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.3389/fgene.2020.00211">Cozzuto et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33362867">33362867</a> <b>(2)</b> State of the Field in Multi-Omics Research: From Computational Needs to Data Mining and Sharing<ul><li><b>Nextflow</b>.<li>                <ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</ext-link>              .<li>As multi-omic analyses are often multi-step with each output being the input of another, in order to increase the efficiency workflows can be written with Domain Specific Languages (DSL) such as: (a) Common Workflow Language (CWL) (<xref rid="B2" ref-type="bibr"><u>Amstutz et al., 2016</u></xref>), (b) <b>Nextflow</b> (<xref rid="B23" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), (c) Snakemake (<xref rid="B48" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), and (d) Galaxy-workflows (<xref rid="B58" ref-type="bibr"><u>McGowan et al., 2020</u></xref>).<li> [Amstutz et al., 2016](<person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P.</given-names></name><name><surname>Chapman</surname><given-names>B.</given-names></name><name><surname>Chilton</surname><given-names>J.</given-names></name><name><surname>Heuer</surname><given-names>M.</given-names></name><name><surname>Stojanovic</surname><given-names>E.</given-names></name></person-group> (<year>2016</year>). <source><italic>Common Workflow Language, v1.0 Common Workflow Language (CWL) Command Line Tool Description, v1.0.</italic></source><pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [McGowan et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32236523">32236523</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33322033">33322033</a> <b>(12)</b> RNAflow: An Effective and Simple RNA-Seq Differential Gene Expression Pipeline Using <b>Nextflow</b><ul><li>Our pipeline is implemented in a reactive workflow manager <b>Nextflow</b>&#xA0;v20.10.0 [<xref rid="B11-genes-11-01487" ref-type="bibr"><u>11</u></xref>], which&#xA0;makes it portable, scalable, parallelizable, and ensures a high level of reproducibility.<li> Only <b>Nextflow</b> and Conda or Docker or Singularity need to be installed in order to run the pipeline.<li> Because&#xA0;<b>Nextflow</b> offers out-of-the-box execution for many schedulers and cloud computing platforms, it can be easily customized to meet individual needs.<li> A&#xA0;recent study compared the currently most commonly used workflow management systems (Snakemake, <b>Nextflow</b>, CWL) with a focus on rapid prototyping for the analysis of ribosome profiling data and, finally, chose <b>Nextflow</b>, which best met the authors&#x2019; requirements [<xref rid="B14-genes-11-01487" ref-type="bibr"><u>14</u></xref>].<li> <b>Nextflow</b> and the containerization software can be run on Linux, Mac OS, and Windows supported through WSL2.<li> Thus, <b>Nextflow</b> is able to check whether the resource is already available and it skips the download step the next time the pipeline is executed.<li> Because of its implementation in the <b>Nextflow</b> framework, <monospace>RNAflow</monospace> comes with out-of-the-box support for various batch schedulers (SGE, LSF, SLURM, PBS, and HTCondor) and cloud platforms (Kubernetes, Amazon AWS, and Google Cloud).<li>Here, we present <monospace>RNAflow</monospace> (publicly available at <uri xlink:href="https://github.com/hoelzer-lab/rnaflow">https://github.com/hoelzer-lab/rnaflow</uri>), another RNA-Seq-Pipeline, which, however, was implemented within the workflow management system <b>Nextflow</b> and, therefore, aims at a very easy handling.<li> Not only do we automatically address common pitfalls in RNA-Seq expression analysis, such as rRNA removal and low abundance feature filtering, but&#xA0;we also offer an in-depth differential expression analysis with <monospace>DESeq2</monospace>, in&#xA0;contrast to other <b>Nextflow</b> RNA-Seq pipelines, such as <monospace>nf-core/rnaseq</monospace> [<xref rid="B9-genes-11-01487" ref-type="bibr"><u>9</u></xref>].<li> However, since container technologies, like Docker and Singularity, are still not available for all users&#x2014;e.g., due to security standards or slow introduction of new software&#x2014;we also allow the monitoring of software and database versions, as&#xA0;well as version tracking of the pipeline itself, by&#xA0;direct deployment from GitHub with <b>Nextflow</b> while using Conda environments.<li> In&#xA0;the future, we will provide stable release versions of the pipeline that can be directly pulled via <b>Nextflow</b> in order to allow full reproducibility of the DEG results.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](          <person-group person-group-type="author">            <name>              <surname>Jackson</surname>              <given-names>M.J.</given-names>            </name>            <name>              <surname>Wallace</surname>              <given-names>E.</given-names>            </name>            <name>              <surname>Kavoussanakis</surname>              <given-names>K.</given-names>            </name>          </person-group>          <article-title>Using rapid prototyping to choose a bioinformatics workflow management system</article-title>          <source>bioRxiv</source>          <year>2020</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/early/2020/08/05/2020.08.04.236208.full.pdf">https://www.biorxiv.org/content/early/2020/08/05/2020.08.04.236208.full.pdf</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-12-09">(accessed on 9 December 2020)</date-in-citation>        ) [9](          <person-group person-group-type="author">            <name>              <surname>Ewels</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Hammar&#xE9;n</surname>              <given-names>R.</given-names>            </name>            <name>              <surname>Peltzer</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Moreno</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Garcia</surname>              <given-names>M.</given-names>            </name>            <collab>rfenouil</collab>            <collab>marchoeppner</collab>            <name>              <surname>Panneerselvam</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Sven</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>jun-wan</surname>            </name>            <etal/>          </person-group>          <article-title>nf-core/rnaseq: Nf-core/rnaseq version 1.4.2</article-title>          <year>2019</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/3503887#.X9Hk0LMRVPY">https://zenodo.org/record/3503887#.X9Hk0LMRVPY</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-12-09">(accessed on 9 December 2020)</date-in-citation>          <pub-id pub-id-type="doi">10.5281/zenodo.3503887</pub-id>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33291220">33291220</a> <b>(4)</b> The International Virus Bioinformatics Meeting 2020<ul><li>WtP is written in <b>Nextflow</b> and utilizes Docker and Singularity containers for a simplistic workflow execution in any Linux environment.<li> WtP is freely available via GitHub (<uri xlink:href="https://github.com/replikation/What_the_Phage">https://github.com/replikation/What_the_Phage</uri>), which includes a simple installation routine for <b>Nextflow</b>/Docker.<li>Here, we present the <b>Nextflow</b> [<xref rid="B57-viruses-12-01398" ref-type="bibr"><u>57</u></xref>] pipeline <monospace>ViralClust</monospace> that can deal with a vast amount of viral sequences and assigns each virus to a cluster.Four approaches are currently implemented as modules, one of them being a novel approach of clustering viral genomes based on kmer frequencies.<li> <b>Nextflow</b> is used to ensure reproducible and transparent analyses and results.<li> [57](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33262485">33262485</a> <b>(2)</b> Abstracts from the 53rd European Society of Human Genetics (ESHG) Conference: Interactive e-Posters<ul><li> We have designed the analysis pipelines using <b>Nextflow</b> on a Slurm scheduler, with Singularity containers to maximise portability and reproducibility.<li> We propose Sarek, an open-source container based bioinformatics workflow for germline or matching tumor-normal pairs, written in <b>Nextflow</b>, to process WGS, whole-exome or gene-panel samples.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33262379">33262379</a> <b>(1)</b> DNA aptamers against bacterial cells can be efficiently selected by a SELEX process using state-of-the art qPCR and ultra-deep sequencing<ul><li>Analysis of the raw NGS-data was accomplished with a workflow implemented in <b>Nextflow</b> 19.10.0<sup><xref ref-type="bibr" rid="CR44"><u>44</u></xref></sup> (illustrated in Supplementary Figure <xref rid="MOESM1" ref-type="media"><u>S8</u></xref>).<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33324401">33324401</a> <b>(2)</b> Accurate and Efficient KIR Gene and Haplotype Inference From Genome Sequencing Reads With Novel K-mer Signatures<ul><li>KMC 3, with workflows implemented in <b>Nextflow</b> (<xref rid="B12" ref-type="bibr"><u>12</u></xref>) and Apache Groovy (<xref rid="B13" ref-type="bibr"><u>13</u></xref>) and a software environment implemented as a Docker container, is used to create 25mer databases from sequence or short-read data and match the markers across the datasets.<li> KPI is free software with a GPL3 license and is implemented as a <b>Nextflow</b> workflow backed with an optional Docker environment.<li> [12](<source>dwgsim</source>. Available at: <uri xlink:type="simple" xlink:href="https://github.com/nh13/dwgsim">https://github.com/nh13/dwgsim</uri>.) [13](<person-group person-group-type="author"><name><surname>Boomsma</surname><given-names>DI</given-names></name><name><surname>Wijmenga</surname><given-names>C</given-names></name><name><surname>Slagboom</surname><given-names>EP</given-names></name><name><surname>Swertz</surname><given-names>MA</given-names></name><name><surname>Karssen</surname><given-names>LC</given-names></name><name><surname>Abdellaoui</surname><given-names>A</given-names></name><etal/></person-group><article-title>The Genome of the Netherlands: design, and project goals</article-title>. <source>Eur J Hum Genet</source> (<year>2014</year>) <volume>22</volume>:<page-range>221&#x2013;7</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1038/ejhg.2013.118</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33239336">33239336</a> <b>(1)</b> Whole-genome sequencing as part of national and international surveillance programmes for antimicrobial resistance: a roadmap<ul><li> For example, the same bioinformatics pipelines for genome assembly, phylogenetic tree construction, Multilocus Sequence Typing (MLST) calling and AMR prediction have been implemented in partner laboratories in Colombia, India, Nigeria, the Philippines and the UK, using <b>Nextflow</b><xref rid="R78" ref-type="bibr"><u>78</u></xref> as a workflow manager and Docker<xref rid="R79" ref-type="bibr"><u>79</u></xref> or Singularity<xref rid="R80" ref-type="bibr"><u>80</u></xref> containers to ensure that pipelines are compatible with local operating systems.<li> [78](<person-group person-group-type="author"><collab>Nextflow</collab></person-group><article-title>A DSL for parallel and scalable computational pipelines</article-title>. Available: <uri xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</uri> [Accessed <comment>3 Mar 2020</comment>].) [79](<person-group person-group-type="author"><collab>Docker</collab></person-group><article-title>Empowering app development for developers</article-title>. Available: <uri xlink:href="https://www.docker.com/">https://www.docker.com/</uri> [Accessed <comment>3 Mar 2020</comment>].) [80](<person-group person-group-type="author"><collab>Singularity</collab></person-group><article-title>Sylabs.io</article-title>. Available: <uri xlink:href="https://sylabs.io/singularity/">https://sylabs.io/singularity/</uri> [Accessed <comment>3 Mar 2020</comment>].) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33232347">33232347</a> <b>(1)</b> Data analysis and modeling pipelines for controlled networked social science experiments<ul><li> <b>NextFlow</b> [<xref rid="pone.0242453.ref089" ref-type="bibr"><u>89</u></xref>] is a domain specific language for computational workflow management systems.<li> [89](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33205815">33205815</a> <b>(5)</b> Unifying package managers, workflow engines, and containers: Computational reproducibility with BioNix<ul><li> This is commonly handled with &#x201C;workflow engines" (e.g., Toil&#xA0;[<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], SnakeMake&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], WDL&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], Cromwell&#xA0;[<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], <b>NextFlow</b>&#xA0;[<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], Ruffus&#xA0;[<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], and Rubra&#xA0;[<xref rid="bib13" ref-type="bibr"><u>13</u></xref>]), which manage stages (we define a &#x201C;stage" as the concrete execution of &#x2265;1 executable on &#x2265;1 input file, producing &#x2265;1 output file) and their execution, providing features such as parallelism, remote building, resumability, and logging.<li> Examples of implicit workflows are SnakeMake&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>] and <b>Nextflow</b>&#xA0;[<xref rid="bib11" ref-type="bibr"><u>11</u></xref>].<li> Examples 9 and 10 illustrate 2 simple examples from the documentation of WDL&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>] and <b>NextFlow</b>&#xA0;[<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] alongside the equivalent BioNix expression.<li> By contrast, the WDL and <b>NextFlow</b> examples presented are valid workflow specifications without software definitions, although they do support software environment management via containers (e.g., Docker) or package managers (e.g., Conda).<li><b>Nextflow</b> basic pipeline example&#xA0;[<xref rid="bib49" ref-type="bibr"><u>47</u></xref>] (<italic>right</italic>) taken verbatim from the documentation and translated to BioNix (<italic>left</italic>).<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<comment>WDL Documentation. <ext-link ext-link-type="uri" xlink:href="https://software.broadinstitute.org/wdl/">https://software.broadinstitute.org/wdl/</ext-link>. Accessed 23 January 2019</comment>.) [10](<comment>Cromwell. <ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/cromwell">https://github.com/broadinstitute/cromwell</ext-link>. Accessed 17 June 2020</comment>.) [11](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechno</source>. <year>2017</year>;<volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [13](Rubra. <comment><ext-link ext-link-type="uri" xlink:href="https://github.com/bjpop/rubra">https://github.com/bjpop/rubra</ext-link>. Accessed 17 June 2020</comment>.) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechno</source>. <year>2017</year>;<volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [9](<comment>WDL Documentation. <ext-link ext-link-type="uri" xlink:href="https://software.broadinstitute.org/wdl/">https://software.broadinstitute.org/wdl/</ext-link>. Accessed 23 January 2019</comment>.) [11](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechno</source>. <year>2017</year>;<volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [47](<source>Nextflow - Basic pipeline</source>. <year>2019</year><comment><ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/example1.html">https://www.nextflow.io/example1.html</ext-link></comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33564394">33564394</a> <b>(1)</b> gprofiler2 -- an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler<ul><li> A <b>Nextflow</b> differential gene expression analysis<ext-link ext-link-type="uri" xlink:href="https://github.com/qbic-pipelines/rnadeseq">pipeline</ext-link> includes<bold>gprofiler2</bold> for pathway analysis.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33236010">33236010</a> <b>(2)</b> High resolution profiling of pathways of escape for SARS-CoV-2 spike-binding antibodies<ul><li> The <b>Nextflow</b> pipeline, used to align all sample reads to the reference library, is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/matsengrp/phip-flow">https://github.com/matsengrp/phip-flow</ext-link>.<li> Demultiplexed sample reads were aligned to the reference library in parallel using a <b>Nextflow</b> data processing pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33191407">33191407</a> <b>(1)</b> Human papilloma virus (HPV) integration signature in Cervical Cancer: identification of <italic>MACROD2</italic> gene as HPV hot spot integration site<ul><li> Nf-VIF is implemented through the <b>Nextflow</b> workflow management system, ensuring a high portability, reproducibility, and scalability (see <xref rid="MOESM1" ref-type="media"><u>Supplementary materials</u></xref> for details).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33196647">33196647</a> <b>(1)</b> Long-read RNA sequencing of human and animal filarial parasites improves gene models and discovers operons<ul><li> The RNA-seq pipeline was implemented using <b>Nextflow</b> [<xref rid="pntd.0008869.ref052" ref-type="bibr"><u>52</u></xref>] and is publicly available (<ext-link ext-link-type="uri" xlink:href="https://github.com/zamanianlab/BmalayiRNAseq-nf">https://github.com/zamanianlab/BmalayiRNAseq-nf</ext-link>).<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33177604">33177604</a> <b>(1)</b> Discovery and comparative genomic analysis of elk circovirus (ElkCV), a novel circovirus species and the first reported from a cervid host<ul><li>Initial exploratory metagenomic analysis was carried out using the in-house nf-villumina <b>Nextflow</b> workflow (v2.0.0)<sup><xref ref-type="bibr" rid="CR13"><u>13</u></xref></sup>.<li> [13](Kruczkiewicz, P. peterk87/nf-villumina. <ext-link ext-link-type="uri" xlink:href="https://github.com/peterk87/nf-villumina">https://github.com/peterk87/nf-villumina</ext-link> (2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33152046">33152046</a> <b>(1)</b> A machine learning-based treatment prediction model using whole genome variants of hepatitis C virus<ul><li>Mapping the sequence reads to the HCV genome and identification of variants were performed using an in-house computational pipeline based on the <b>Nextflow</b> scripting, which included nucleotide base correction by the k-mer method using the Pollux program, HCV-1b mapping (Con1 [<xref rid="pone.0242028.ref007" ref-type="bibr"><u>7</u></xref>], HCV-J [<xref rid="pone.0242028.ref008" ref-type="bibr"><u>8</u></xref>], HCV-BK [<xref rid="pone.0242028.ref009" ref-type="bibr"><u>9</u></xref>]) by the Burrows-Wheeler Transform algorithm using the BWA and Bowtie2 programs, genome variant detection by the haplotype-based method using FreeBayes program, variant normalization using BCFtools, and variant filtering using VCFtools.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=11907240">11907240</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=1849488">1849488</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=12615931">12615931</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33147627">33147627</a> <b>(2)</b> Computational strategies to combat COVID-19: useful tools to accelerate SARS-CoV-2 and coronavirus research<ul><li>The workflow poreCov is implemented in <b>nextflow</b> [<xref rid="ref100" ref-type="bibr"><u>100</u></xref>] for full parallelization of the workload and stable sample processing (see Figure <xref ref-type="fig" rid="f2"><u>2</u></xref>).<li> Workflow management systems such as SnakeMake or <b>Nextflow</b> [<xref rid="ref19" ref-type="bibr"><u>19</u></xref>, <xref rid="ref100" ref-type="bibr"><u>100</u></xref>] allow easy installation and reproducible execution on various platforms.<li> [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33124659">33124659</a> <b>(4)</b> A molecular map of lung neuroendocrine neoplasms<ul><li> The sequencing reads from the 16 atypical carcinoids' whole exomes and the 3 carcinoids' whole genomes were processed using the in-house <b>Nextflow</b> [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>] workflow available at the IARCbioinfo/alignment-nf [<xref rid="bib16" ref-type="bibr"><u>16</u></xref>] GitHub repository, revision No.<li>The QCs of the WES and WGS data were performed using FastQC v0.11.8 (FastQC, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_014583">RRID:SCR_014583</ext-link>) [<xref rid="bib21" ref-type="bibr"><u>21</u></xref>] and QualiMap v2.2.1 (QualiMap, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_001209">RRID:SCR_001209</ext-link>) [<xref rid="bib22" ref-type="bibr"><u>22</u></xref>] using the in-house <b>Nextflow</b> [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>] workflows available at IARCbioinfo/fastqc-nf [<xref rid="bib23" ref-type="bibr"><u>23</u></xref>] and IARCbioinfo/qualimap-nf [<xref rid="bib24" ref-type="bibr"><u>24</u></xref>] repositories, respectively, and the results aggregated using MultiQC v1.7 (MultiQC, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_014982">RRID:SCR_014982</ext-link>) [<xref rid="bib25" ref-type="bibr"><u>25</u></xref>] (Fig.&#xA0;<xref ref-type="fig" rid="fig1"><u>1</u></xref>, left panel).<li> These steps were performed using our in-house <b>Nextflow</b> [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>] pipelines available at the following GitHub repositories: IARCbioinfo/RNAseq-nf [<xref rid="bib35" ref-type="bibr"><u>35</u></xref>] release v2.3, IARCbioinfo/abra-nf [<xref rid="bib36" ref-type="bibr"><u>36</u></xref>] release v3.0, IARCbioinfo/BQSR-nf [<xref rid="bib37" ref-type="bibr"><u>37</u></xref>] release v1.1, and IARCbioinfo/RNAseq-transcript-nf [<xref rid="bib38" ref-type="bibr"><u>38</u></xref>] release v2.1.<li> We also provide in the Nextjournal notebook, the <b>Nextflow</b> command lines allowing to obtain the read counts.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a>) [16](<person-group person-group-type="author"><collab>IARCbioinfo/alignment-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/alignment-nf">https://github.com/IARCbioinfo/alignment-nf.</ext-link> Accessed March 2018</comment>.) [21](<person-group person-group-type="author"><name name-style="western"><surname>Andrews</surname><given-names>S</given-names></name>, <name name-style="western"><surname>Krueger</surname><given-names>F</given-names></name>, <name name-style="western"><surname>Segonds-Pichon</surname><given-names>A</given-names></name>, <etal>et al.</etal></person-group><source>FastQC</source>. <publisher-name>Babraham</publisher-name><year>2012</year><comment><ext-link ext-link-type="uri" xlink:href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link></comment>. Accessed August 2019.) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26428292">26428292</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a>) [23](<person-group person-group-type="author"><collab>IARCbioinfo/fastqc-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/fastqc-nf">https://github.com/IARCbioinfo/fastqc-nf</ext-link>. Accessed August 2019</comment>.) [24](<person-group person-group-type="author"><collab>IARCbioinfo/qualimap-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/qualimap-nf">https://github.com/IARCbioinfo/qualimap-nf</ext-link>. Accessed August 2019</comment>.) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27312411">27312411</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29412134">29412134</a>) [35](<person-group person-group-type="author"><collab>IARCbioinfo/RNAseq-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/RNAseq-nf">https://github.com/IARCbioinfo/RNAseq-nf.</ext-link> Accessed May 2020</comment>.) [36](<person-group person-group-type="author"><collab>IARCbioinfo/abra-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/abra-nf">https://github.com/IARCbioinfo/abra-nf</ext-link>. Accessed May 2020</comment>.) [37](<person-group person-group-type="author"><collab>IARCbioinfo/BQSR-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/BQSR-nf">https://github.com/IARCbioinfo/BQSR-nf</ext-link>. Accessed May 2020</comment>.) [38](<person-group person-group-type="author"><collab>IARCbioinfo/RNAseq-transcript-nf GitHub repository</collab>.</person-group><comment><ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/RNAseq-transcript-nf">https://github.com/IARCbioinfo/RNAseq-transcript-nf</ext-link>. Accessed May 2020</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33126635">33126635</a> <b>(3)</b> Comparative Genome Analysis of 33 <italic>Chlamydia</italic> Strains Reveals Characteristic Features of <italic>Chlamydia Psittaci</italic> and Closely Related Species<ul><li>The Roary ILP Bacterial Annotation Pipeline (RIBAP) is a newly developed pipeline freely available at github.com/hoelzer-lab/ribap and implemented using the workflow management system <b>Nextflow</b> [<xref rid="B76-pathogens-09-00899" ref-type="bibr"><u>76</u></xref>].<li> Each step of the pipeline is encapsulated in its own Docker container, thus, only <b>Nextflow</b> and Docker are required to run RIBAP.<li> For the present study, we ran the pipeline in v0.4 using the following command: <b>nextflow</b> run hoelzer-lab/ribap -r v0.4 --fasta &#x2018;*.fasta&#x2019; --tree --tmlim 240.<li> [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33106560">33106560</a> <b>(2)</b> Raw transcriptomics data to gene specific SSRs: a validated free bioinformatics workflow for biologists<ul><li> <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup> is another command line application that can be used for various RNA-seq analyses.<li> However, the transcriptome assembly process of <b>Nextflow</b> is reference-based and cannot be used for de novo assembly.<li> [66](Nextflow - A DSL for parallel and scalable computational pipelines. <ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33193131">33193131</a> <b>(2)</b> Interpretations of Environmental Microbial Community Studies Are Biased by the Selected 16S rRNA (Gene) Amplicon Sequencing Pipeline<ul><li>The pipeline nf-core/ampliseq v1.1.0 (<xref rid="B89" ref-type="bibr"><u>Straub and Peltzer, 2019</u></xref>) was executed on environmental data sets using <b>nextflow</b> v19.10.0.5170, Java v1.8.0_112, and singularity v3.0.1 with optional parameters &#x201C;- -multiple Sequencing Runs&#x201D; (because the sequencing data originated from three MiSeq runs), &#x201C;- -trunclenf 180&#x201D;, &#x201C;- -trunclenr 180&#x201D; (to resemble truncation values of QIIME2 with DADA2), &#x201C;- -classifier_removeHash&#x201D; (because hash sign in some taxa names lets QIIME2 v2018.6 fail), and a metadata sheet was specified as indicated by the documentation.<li> nf-core/ampliseq will prove valuable because it has minimal software requirements (<b>nextflow</b>, Java, Unix), is easy to use (minimal parameter input: folder containing raw data and primer sequences), and uses computational infrastructure optimally (e.g., hpc job schedulers or cloud computing).<li> [Straub and Peltzer, 2019](<person-group person-group-type="author"><name><surname>Straub</surname><given-names>D.</given-names></name><name><surname>Peltzer</surname><given-names>A.</given-names></name></person-group> (<year>2019</year>). <source><italic>nf-core/Ampliseq.</italic></source><publisher-loc>London</publisher-loc>: <publisher-name>Zenodo</publisher-name>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33097812">33097812</a> <b>(4)</b> Practical guide for managing large-scale human genome data in research<ul><li>The <b>Nextflow</b> [<xref ref-type="bibr" rid="CR65"><u>65</u></xref>] is another workflow engine as well as a domain-specific language (DSL).<li> <b>Nextflow</b> has a Groovy-based DSL, as shown in Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1a</u></xref>, and is easy to understand if users are already familiar with some programming languages.<li> <b>Nextflow</b> also has the GUI frontend [<xref ref-type="bibr" rid="CR66"><u>66</u></xref>].<li>The simple hello world example by using workflow description languages: (<bold>a</bold>) <b>Nextflow</b>, (<bold>b</bold>) WDL, and (<bold>c</bold>) CWL.<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [66](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32306927">32306927</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33091012">33091012</a> <b>(1)</b> Ten simple rules for biologists initiating a collaboration with computer scientists<ul><li> iPython, <ext-link ext-link-type="uri" xlink:href="https://ipython.org/">https://ipython.org</ext-link>) notebooks, Rmarkdown (<ext-link ext-link-type="uri" xlink:href="https://rmarkdown.rstudio.com/">https://rmarkdown.rstudio.com</ext-link>), Galaxy (<ext-link ext-link-type="uri" xlink:href="https://galaxyproject.org/">https://galaxyproject.org</ext-link>), snakemake (<ext-link ext-link-type="uri" xlink:href="https://snakemake.readthedocs.io/">https://snakemake.readthedocs.io</ext-link>), <b>nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io</ext-link>), and LaTeX (<ext-link ext-link-type="uri" xlink:href="https://www.latex-project.org/">https://www.latex-project.org</ext-link>) (and Overleaf; <ext-link ext-link-type="uri" xlink:href="https://www.overleaf.com/">https://www.overleaf.com</ext-link>).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33068114">33068114</a> <b>(1)</b> NanoGalaxy: Nanopore long-read sequencing data analysis in Galaxy<ul><li> Command line&#x2013;based workflow managers such as Snakemake and <b>Nextflow</b>&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>,<xref rid="bib10" ref-type="bibr"><u>10</u></xref>] can be used for analysing the data.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33059593">33059593</a> <b>(6)</b> In silico benchmarking of metagenomic tools for coding sequence detection reveals the limits of sensitivity and precision<ul><li> We address this core challenge of benchmarking by implementing our analysis within a workflow management tool, <b>Nextflow</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], which achieves a high degree of reproducibility by executing each component task within Docker containers, a portable and fixed computational environment.<li>All of the analytical steps for analyzing computational tools for CDS detection from metagenomes were executed within a single analytical workflow (&#x2018;evaluate-gene-detection.nf&#x2019;) which can be downloaded from <ext-link ext-link-type="uri" xlink:href="https://github.com/FredHutch/evaluate-gene-level-metagenomics-tools">https://github.com/FredHutch/evaluate-gene-level-metagenomics-tools</ext-link> and executed via <b>Nextflow</b>.<li> Our goal is that other researchers will be able to use this <b>Nextflow</b>-based workflow to reproduce and expand upon the benchmarking results presented in this manuscript.<li> While this total number of organisms may be lower than what can be observed in some environments using metagenomic sequencing, we encourage other researchers to repeat this benchmarking effort using the provided <b>Nextflow</b>-based workflow if they have a particular interest in a specific level of community complexity.<li> This benchmarking approach is implemented within a reproducible <b>Nextflow</b> workflow, and therefore should be relatively straightforward for other researchers to reproduce and augment as additional tools for CDS detection become available.<li> In order to enable other researchers to test a wider range of parameters, or to evaluate a particular set of reference genomes, we have provided the complete benchmarking computational workflow in the community supported <b>Nextflow</b> syntax.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33066701">33066701</a> <b>(1)</b> Amplicon-Based Detection and Sequencing of SARS-CoV-2 in Nasopharyngeal Swabs from Patients With COVID-19 and Identification of Deletions in the Viral Genome That Encode Proteins Involved in Interferon Antagonism<ul><li> For the identification of deletions within the SARS-CoV-2 genome, ONT RSLA library reads were aligned to the coronavirus 2 isolate Wuhan-Hu-1 reference genome using Minimap2, as part of the <b>Nextflow</b> ARTIC analysis pipeline (<uri xlink:href="https://github.com/connor-lab/ncov2019-artic-nf">https://github.com/connor-lab/ncov2019-artic-nf</uri>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33066701">33066701</a> <b>(1)</b> Amplicon-Based Detection and Sequencing of SARS-CoV-2 in Nasopharyngeal Swabs from Patients With COVID-19 and Identification of Deletions in the Viral Genome That Encode Proteins Involved in Interferon Antagonism<ul><li>The resulting profile was implemented in a <b>Nextflow</b> (<xref rid="btaa871-B3" ref-type="bibr">Di Tommaso <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33046894">33046894</a> <b>(2)</b> A systematic evaluation of the design and context dependencies of massively parallel reporter assays<ul><li> The sections below document the various components of the pipeline, which borrow heavily from our earlier work<sup><xref rid="R18" ref-type="bibr"><u>18</u></xref></sup>, and were implemented into a reproducible <b>Nextflow</b>-based codebase named MPRAflow<sup><xref rid="R44" ref-type="bibr"><u>44</u></xref></sup>.<li>A reproducible processing pipeline for MPRA data is available as a <b>Nextflow</b>-based MPRA processing pipeline named MPRAflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/shendurelab/MPRAflow)44">https://github.com/shendurelab/MPRAflow</ext-link>)<sup><xref rid="R44" ref-type="bibr"><u>44</u></xref></sup>.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27831498">27831498</a>) [44](R44) [44](R44) <li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27831498">27831498</a>) [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32641802">32641802</a>) [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32641802">32641802</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33193602">33193602</a> <b>(9)</b> GenomeChronicler: The Personal Genome Project UK Genomic Report Generator Pipeline<ul><li> Either entry point of the pipeline can be run locally in a single machine, as a <b>Nextflow</b> workflow or in the Cloud.<li> The stand-alone GenomeChronicler is available at (<ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler">https://github.com/PGP-UK/GenomeChronicler</ext-link>), the integration of GenomeChronicler with <b>Nextflow</b> is available at (<ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler-nf">https://github.com/PGP-UK/GenomeChronicler-nf</ext-link>) and the combined GenomeChronicler with Sarek variant calling is available at (<ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler-Sarek-nf">https://github.com/PGP-UK/GenomeChronicler-Sarek-nf</ext-link>).<li>To simplify this entire process and to make the tool more accessible to users who may not know how to run a germline variant calling pipeline, GenomeChronicler can also be run in a fully automated mode from the raw sequencing data, where the germline variant calling pipeline is also run and the whole process is managed by the <b>Nextflow</b> workflow management system (<xref rid="B4" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li>The combined version of Sarek + GenomeChronicler written using the <b>Nextflow</b> workflow manager (<xref rid="B4" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) is available both on Github<sup><xref ref-type="fn" rid="footnote3"><u>3</u></xref></sup> and on Lifebit CloudOS.<li>To enable reproducible, massively parallel, cloud native analyses, GenomeChronicler has also been implemented as a <b>Nextflow</b> pipeline.<li> The implementation abstracts the installation overhead from the end user, as all the dependencies are already available via pre-built containers, integrated seamlessly in the <b>Nextflow</b> pipeline.<li> The source code for this implementation is available on GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler-nf">https://github.com/PGP-UK/GenomeChronicler-nf</ext-link>, as a standalone <b>Nextflow</b> process.<li> The <b>Nextflow</b> integrated version is available at [<ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler-nf">https://github.com/PGP-UK/GenomeChronicler-nf</ext-link>] and finally, the version also containing the Sarek variant calling pipeline is available at [<ext-link ext-link-type="uri" xlink:href="https://github.com/PGP-UK/GenomeChronicler-Sarek-nf">https://github.com/PGP-UK/GenomeChronicler-Sarek-nf</ext-link>].<li> CC contributed the <b>Nextflow</b> and Lifebit CloudOS integrations with support from PP.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [3](<a href="https://github.com/PGP-UK/GenomeChronicler-Sarek-nf">https://github.com/PGP-UK/GenomeChronicler-Sarek-nf</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32968280">32968280</a> <b>(2)</b> Conformation of sister chromatids in the replicated human genome<ul><li>Hi-C samples were preprocessed using a custom <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/gerlichlab/scshic_pipeline">https://github.com/gerlichlab/scshic_pipeline</ext-link>).<li> After completion of the <b>nextflow</b> pipeline, cis sister and trans sister contacts were merged, and the resulting file balanced using cooler<sup><xref rid="R58" ref-type="bibr"><u>58</u></xref></sup>.<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22941365">22941365</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32983415">32983415</a> <b>(1)</b> The 21st annual Bioinformatics Open Source Conference (BOSC 2020, part of BCC2020)<ul><li> Some training workshops focused on specific tools or analysis ecosystems (e.g., Reactome, <b>Nextflow</b>, Terra, and of course Galaxy); some discussed a field such as machine learning or RNA-Seq analysis; some introduced useful general-purpose tools such as Git and Docker; and some were data-focused.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33033490">33033490</a> <b>(1)</b> Editorial: User-Friendly Tools Applied to Genetics or Systems Biology<ul><li> By combining the Anaconda package manager with <b>Nextflow</b> scripting language, Pipeliner enable users to generate modular computational workflows for processing various types of sequencing data, including single-cell expression data (<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.3389/fgene.2019.00614">Federico et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> <b>(12)</b> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants<ul><li><b>Nextflow</b>.<li><ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/index.html">https://www.<b>nextflow</b>.io/index.html</ext-link>.<li>Sarek is implemented in <b>Nextflow</b> (<xref rid="ref-9" ref-type="bibr">Di Tommaso<italic>et al</italic>., 2017</xref>), a workflow language designed specifically for bioinformatics applications.<li> <b>Nextflow</b> has a transparent design, making the Sarek code easy to read, adjust and extend.<li> Compared to the Bpipe workflow language (used in for example DNAp), <b>Nextflow</b> offers superior support for different execution environments, like Slurm, Sun Grid Engine, LSF and Kubernetes, and includes native support for cloud compute environments including Google Cloud and AWS.<li> Sarek is part of a rapidly growing community effort of well documented and community-tested<ext-link ext-link-type="uri" xlink:href="https://nf-co.re/"><b>Nextflow</b> pipelines</ext-link>, and adheres to the nf-core portability and documentation guidelines (<xref rid="ref-12" ref-type="bibr">Ewels<italic>et al</italic>., 2019</xref>).<li>Sarek is run from a computer system with a local installation of <b>Nextflow</b> and support for either Conda environments, Docker or Singularity containers.<li> <b>Nextflow</b> can automatically fetch the Sarek source code from GitHub.<li>Running Sarek with Singularity container on a computer system supporting Java 8 requires only installation of <b>Nextflow</b> and Singularity.<li> A full analysis run starting from FASTQ files including mapping, recalibration, variant calling and annotation, as well as generating a full QC report can be invoked by a single <b>Nextflow</b> command:.<li> <b>nextflow</b> run nf-core/sarek -r 2.5.2 -profile singularity --input samples.tsv --tools Mutect2,Strelka,Manta,TIDDIT,ASCAT,ControlFREEC,snpEff,VEP</styled-content>.<li><b>Nextflow</b> will recognize the workflow name and will download the specified version (2.5.2) of the pipeline from GitHub, including the corresponding container, as well as fetching the required reference files from<ext-link ext-link-type="uri" xlink:href="https://ewels.github.io/AWS-iGenomes/">AWS-iGenomes</ext-link>.<li> Sarek builds on a philosophy of reasonably narrow, independent workflows, written in the domain-specific language <b>Nextflow</b>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> <b>(2)</b> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants<ul><li> Sarek is based on <b>Nextflow</b>, a popular tool for defining computational workflows.<li> Sarek aims to minimize the installation and management time overhead by building a NGS workflow on top of <b>Nextflow</b>, automatically installing the required software components.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> <b>(3)</b> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants<ul><li> Sarek is based on <b>Nextflow</b>, a popular tool for defining computational workflows.<li> Sarek aims to minimize the installation and management time overhead by building a NGS workflow on top of <b>Nextflow</b>, automatically installing the required software components.<li> How does Sarek support run diagnostics and relaunching failed jobs?</p></list-item></list><bold>This is a good comment, and <b>NextFlow</b> reports the failed process and the error-code from the underlying software, and Sarek is designed to make it very easy to resume failed jobs from the point of failure.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> <b>(3)</b> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants<ul><li>Sarek is open source and part of the nf-core community effort which builds well-curated analysis pipelines in the <b>Nextflow</b> pipeline framework.<li>Sarek can use different <b>Nextflow</b> configuration profiles.<li>Whereas the choice of <b>Nextflow</b> is justified, there is little argumentation for why the different tools (variant callers in particular) are selected other than that they represent state-of-the-art.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269765">32269765</a> <b>(3)</b> Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants<ul><li>Sarek is open source and part of the nf-core community effort which builds well-curated analysis pipelines in the <b>Nextflow</b> pipeline framework.<li>Sarek can use different <b>Nextflow</b> configuration profiles.<li>Whereas the choice of <b>Nextflow</b> is justified, there is little argumentation for why the different tools (variant callers in particular) are selected other than that they represent state-of-the-art.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32880574">32880574</a> <b>(4)</b> Co-expression analysis reveals interpretable gene modules controlled by <italic>trans</italic>-acting genetic variants<ul><li> The <b>Nextflow</b> pipelines for genotype processing and quality control are available from GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/eQTL-Catalogue/genotype_qc">https://github.com/eQTL-Catalogue/genotype_qc</ext-link>).<li>We performed <italic>cis</italic>-eQTL analysis using the qtlmap (<ext-link ext-link-type="uri" xlink:href="https://github.com/eQTL-Catalogue/qtlmap">https://github.com/eQTL-Catalogue/qtlmap</ext-link>) <b>Nextflow</b> (<xref rid="bib14" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) workflow developed for the eQTL Catalogue project (<xref rid="bib27" ref-type="bibr"><u>Kerimov et al., 2020</u></xref>).<li> The fine mapping <b>Nextflow</b> workflow for <italic>cis</italic>-eQTLs is available from GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/eQTL-Catalogue/susie-workflow">https://github.com/eQTL-Catalogue/susie-workflow</ext-link>).<li> The <italic>cis</italic>-eQTL colocalisation <b>Nextflow</b> workflow is available from GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/kauralasoo/colocWrapper">https://github.com/kauralasoo/colocWrapper</ext-link>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Kerimov et al., 2020](          <person-group person-group-type="author">            <name>              <surname>Kerimov</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Hayhurst</surname>              <given-names>JD</given-names>            </name>            <name>              <surname>Manning</surname>              <given-names>JR</given-names>            </name>            <name>              <surname>Walter</surname>              <given-names>P</given-names>            </name>          </person-group>          <year>2020</year>          <article-title>eQTL catalogue: a compendium of uniformly processed human gene expression and splicing QTLs</article-title>          <source>bioRxiv</source>          <pub-id pub-id-type="doi">10.1101/2020.01.29.924266</pub-id>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32953263">32953263</a> <b>(1)</b> DATMA: Distributed AuTomatic Metagenomic Assembly and annotation framework<ul><li> Rahmann, 2012</xref>), <b>Nextflow</b>&#xA0;(<xref rid="ref-16" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), Ibis&#xA0;(<xref rid="ref-5" ref-type="bibr"><u>Bal et al., 2010</u></xref>), and Swift&#xA0;(<xref rid="ref-50" ref-type="bibr"><u>Wilde et al., 2011</u></xref>)) we selected COMPSs&#xA0;(<xref rid="ref-4" ref-type="bibr"><u>Badia et al., 2015</u></xref>) which provided us with the tools that we required: simple python interface and automatic parallel task distribution and synchronization.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Bal et al., 2010](          <person-group person-group-type="author">            <name>              <surname>Bal</surname>              <given-names>HE</given-names>            </name>            <name>              <surname>Maassen</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Van&#xA0;Nieuwpoort</surname>              <given-names>RV</given-names>            </name>            <name>              <surname>Drost</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Kemp</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Van&#xA0;Kessel</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Palmer</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Wrzesinska</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Kielmann</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Van&#xA0;Reeuwijk</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Seinstra</surname>              <given-names>FJ</given-names>            </name>            <name>              <surname>Jacobs</surname>              <given-names>CJH</given-names>            </name>            <name>              <surname>Verstoep</surname>              <given-names>K</given-names>            </name>          </person-group>          <year>2010</year>          <article-title>Real-world distributed computer with Ibis</article-title>          <source>Computer</source>          <volume>43</volume>          <issue>8</issue>          <fpage>54</fpage>          <lpage>62</lpage>          <pub-id pub-id-type="doi">10.1109/mc.2010.184</pub-id>        ) [Wilde et al., 2011](          <person-group person-group-type="author">            <name>              <surname>Wilde</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Hategan</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Wozniak</surname>              <given-names>JM</given-names>            </name>            <name>              <surname>Clifford</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Katz</surname>              <given-names>DS</given-names>            </name>            <name>              <surname>Foster</surname>              <given-names>I</given-names>            </name>          </person-group>          <year>2011</year>          <article-title>Swift: a language for distributed parallel scripting</article-title>          <source>Parallel Computing</source>          <volume>37</volume>          <issue>9</issue>          <fpage>633</fpage>          <lpage>652</lpage>          <pub-id pub-id-type="doi">10.1016/j.parco.2011.05.005</pub-id>        ) [Badia et al., 2015](          <person-group person-group-type="author">            <name>              <surname>Badia</surname>              <given-names>RM</given-names>            </name>            <name>              <surname>Conejero</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Diaz</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Ejarque</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lezzi</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Lordan</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Ramon-Cortes</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Sirvent</surname>              <given-names>R</given-names>            </name>          </person-group>          <year>2015</year>          <article-title>COMP Superscalar, an interoperable programming framework</article-title>          <source>SoftwareX</source>          <volume>3&#x2013;4</volume>          <fpage>32</fpage>          <lpage>36</lpage>          <pub-id pub-id-type="doi">10.1016/j.softx.2015.10.004</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32952600">32952600</a> <b>(2)</b> DeepVariant-on-Spark: Small-Scale Genome Analysis Using a Cloud-Based Computing Framework<ul><li> To the best of our knowledge, substantial efforts, such as <b>Nextflow</b>, DNAnexus, DNAstack, and Parabricks, have been devoted to accelerating the DeepVariant pipeline.<li> <b>Nextflow</b> offers parallel processing of multiple samples for DeepVariant at a time [<xref rid="B26" ref-type="bibr"><u>26</u></xref>], producing the results in a convenient and reproducible manner; however, the total wall-clock time for each sample is unchanged.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32866155">32866155</a> <b>(7)</b> GeneTEFlow: A <b>Nextflow</b>-based pipeline for analysing gene and transposable elements expression from RNA-Seq data<ul><li>To address this need, we developed GeneTEFlow, a reproducible and platform-independent workflow, for the comprehensive analysis of gene and locus-specific TEs expression from RNA-Seq data using <b>Nextflow</b> [<xref rid="pone.0232994.ref026" ref-type="bibr"><u>26</u></xref>] and Docker [<xref rid="pone.0232994.ref027" ref-type="bibr"><u>27</u></xref>] technologies.<li> Second, GeneTEflow uses <b>Nextflow</b> to define the computational workflows, not only enabling parallelization and complete automation of the analysis, but also providing capability to track analysis parameters.<li> Thus, GeneTEFlow allows users to generate reproducible analysis results through utilization of both Docker and <b>Nextflow</b> in a platform independent manner.<li>The GeneTEFlow pipeline was developed using <b>Nextflow</b>, a portable, flexible, and reproducible workflow management system, and Docker technology, a solution to securely build and run applications on multiple platforms.<li> Data processing and analysis steps were implemented by modules using <b>Nextflow</b>.<li> The workflow includes four major inputs: raw sequence files in fastq format, a sample meta data file in excel format, reference genome and gene annotation files, and a <b>Nextflow</b> configuration file.<li> Taking advantage of the advanced functionalities provided by <b>Nextflow</b> and Docker, GeneTEFlow allows users to run analysis reproducibly on different computing platforms without the need for individual tool installation and manual version tracking.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<name><surname>Merkel</surname><given-names>D</given-names></name>. <article-title>Docker: lightweight Linux containers for consistent development and deployment</article-title>. <source>Linux J</source>. <year>2014</year>;<volume>2014</volume>(<issue>239</issue>):Article 2.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32817379">32817379</a> <b>(1)</b> Tracking of Antibiotic Resistance Transfer and Rapid Plasmid Evolution in a Hospital Setting by Nanopore Sequencing<ul><li> The <italic toggle="yes">pathoLogic</italic> platform was created using the <italic toggle="yes"><b>Nextflow</b></italic> (<xref rid="B39" ref-type="bibr"><u>39</u></xref>) environment to chain different tools and scripts, represented here as circular nodes.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32824272">32824272</a> <b>(1)</b> A Comparison of Whole Genome Sequencing of SARS-CoV-2 Using Amplicon-Based Sequencing, Random Hexamers, and Bait Capture<ul><li> Yet our standardized iVar-based pipeline (github.com/jaleezyy/covid-19-signal), compatible with and extending the Connor lab ARTIC <b>nextflow</b> pipeline (github.com/connor-lab/ncov2019-artic-nf), was able to overcome these regions of low coverage, favoring the majority reads to generate a final genome sequence.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32790776">32790776</a> <b>(1)</b> No more business as usual: Agile and effective responses to emerging pathogen threats require open data and open analytics<ul><li> They can be combined into workflows using systems like the Common Workflow Language (CWL) [<xref rid="ppat.1008643.ref019" ref-type="bibr"><u>19</u></xref>], <b>Nextflow</b> [<xref rid="ppat.1008643.ref020" ref-type="bibr"><u>20</u></xref>], or Snakemake [<xref rid="ppat.1008643.ref021" ref-type="bibr"><u>21</u></xref>].<li> [19](Common Workflow Language [Internet]. Common Workflow Language. [cited 2020 Feb 21]. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/</ext-link>.) [20](Nextflow&#x2014;A DSL for parallel and scalable computational pipelines [Internet]. Nextflow. [cited 2020 Feb 21]. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</ext-link>.) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32764349">32764349</a> <b>(1)</b> Distinct miRNA Profile of Cellular and Extracellular Vesicles Released from Chicken Tracheal Cells Following Avian Influenza Virus Infection<ul><li> The MirDeep2 package, which maps reads against a library of known miRNAs from miRBase, was used for miRNA quantification from the reads, coordinated by a <b>Nextflow</b> workflow [<xref rid="B43-vaccines-08-00438" ref-type="bibr"><u>43</u></xref>,<xref rid="B44-vaccines-08-00438" ref-type="bibr"><u>44</u></xref>].<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21911355">21911355</a>) [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32753501">32753501</a> <b>(18)</b> Bactopia: a Flexible Pipeline for Complete Analysis of Bacterial Genomes<ul><li> Bactopia is based on <b>Nextflow</b> workflow software (<xref rid="B3" ref-type="bibr"><u>3</u></xref>) and is designed to be scalable, allowing projects as small as a single genome to be run on a local desktop or projects including many thousands of genomes to be run as a batch on a cloud infrastructure.<li> The underlying <b>Nextflow</b> structure ensures reproducibility.<li>Bactopia links together open-source bioinformatics software, available from Bioconda (<xref rid="B1" ref-type="bibr"><u>1</u></xref>), using <b>Nextflow</b> (<xref rid="B3" ref-type="bibr"><u>3</u></xref>).<li> <b>Nextflow</b> was chosen for its flexibility: Bactopia can be run locally, on clusters, or on cloud platforms with simple parameter changes.<li> <b>Nextflow</b> automates installation of the component software of the workflow through integration with Bioconda.<li><b>Nextflow</b>.<li>                    <ext-link ext-link-type="uri" xlink:href="https://github.com/<b>nextflow</b>-io/<b>nextflow</b>">https://github.com/<b>nextflow</b>-io/<b>nextflow</b></ext-link>                  .<li><b>Nextflow</b>.<li> <b>Nextflow</b> (<xref rid="B3" ref-type="bibr"><u>3</u></xref>) recorded detailed statistics on resource usage, including CPU, memory, job duration, and input-output (I/O).<li><b>Nextflow</b> runtime report for <italic>Lactobacillus</italic> genomes processed by Bactopia.<li> It can be run on a laptop for a single bacterial sample, but, critically, the underlying <b>Nextflow</b> framework allows it to make efficient use of large clusters and cloud-computing environments to process the many thousands of genomes that are currently being generated.<li> For multiple samples, this method does not make efficient use of <b>Nextflow</b>&#x2019;s queue system.<li> By using the FOFN method, <b>Nextflow</b> queues each sample and makes efficient use of available resources.<li> Bactopia Tools (BaTs) are a set of comparative-analysis workflows written using <b>Nextflow</b> that take advantage of the predictable output structure from BaAP.<li> Each Bactopia tool has a separate <b>Nextflow</b> workflow with its own conda environment, Docker image, and Singularity image.<li> [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [1](<person-group person-group-type="author"><name name-style="western"><surname>Gr&#xFC;ning</surname><given-names>B</given-names></name>, <name name-style="western"><surname>Dale</surname><given-names>R</given-names></name>, <name name-style="western"><surname>Sj&#xF6;din</surname><given-names>A</given-names></name>, <name name-style="western"><surname>Rowe</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Chapman</surname><given-names>BA</given-names></name>, <name name-style="western"><surname>Tomkins-Tinch</surname><given-names>CH</given-names></name>, <name name-style="western"><surname>Valieris</surname><given-names>R</given-names></name>, <name name-style="western"><surname>K&#xF6;ster</surname><given-names>J</given-names></name></person-group>, <collab>The Bioconda Team</collab>. <year>2017</year><article-title>Bioconda: a sustainable and comprehensive software distribution for the life sciences</article-title>. <source>bioRxiv</source><ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/10.1101/207092v2">https://www.biorxiv.org/content/10.1101/207092v2</ext-link>.) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32723480">32723480</a> <b>(1)</b> Notch and TLR signaling coordinate monocyte cell fate and inflammation<ul><li> Pre-processing of FASTQ inputs, alignment of the reads and quality control was conducted by nfcore/rnaseq (version 1.5dev) analysis pipeline (The National Genomics Infrastructure, SciLifeLab Stockholm, Sweden) using <b>Nextflow</b> tool.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32689929">32689929</a> <b>(2)</b> Performance evaluation of lossy quality compression algorithms for RNA-seq data<ul><li> All the processing pipelines were organized using <b>Nextflow</b> [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> For RNA-seq data analysis tools, we collected the CPU run-time and peak memory consumption directly from <b>Nextflow</b> with the number of CPU threads limited to six for each container.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32680854">32680854</a> <b>(1)</b> rdmc: An Open Source R Package Implementing Convergent Adaptation Models of Lee and Coop (2017)<ul><li> Making <italic>rdmc</italic> a portable and easy to install R package simplifies running separate genomic regions as independent jobs in parallel using workflows such as Snakemake (<xref rid="bib14" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) or <b>Nextflow</b> (<xref rid="bib5" ref-type="bibr">Di Tommaso <italic>et al.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32660416">32660416</a> <b>(1)</b> MethylStar: A fast and robust pre-processing pipeline for bulk or single-cell whole-genome bisulfite sequencing data<ul><li> The nf-core/methylseq framework is built using <b>Nextflow</b> [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>], and aims to provide reproducible pipeline templates that can be easily adapted by both developers as well as experimentalists.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32646465">32646465</a> <b>(1)</b> Transcriptional responses in <italic>Parascaris univalens</italic> after <italic>in vitro</italic> exposure to ivermectin, pyrantel citrate and thiabendazole<ul><li>Read quality assessment, adaptor removal, filtering and removal of duplicates was performed using fastp [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] (Pipeline with <b>nextflow</b> and docker support available at <ext-link ext-link-type="uri" xlink:href="https://github.com/SLUBioinformaticsInfrastructure/RNAseq_nf">https://github.com/SLUBioinformaticsInfrastructure/RNAseq_nf</ext-link>.).<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30423086">30423086</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32640986">32640986</a> <b>(1)</b> Bioinformatics recipes: creating, executing and distributing reproducible data analysis workflows<ul><li> A recipe may be a collection of several command-line tools, it may be a Makefile, a SnakeMake [<xref ref-type="bibr" rid="CR3"><u>3</u></xref>] file, a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>] pipeline, an R script or any command line oriented computer program.<li> [3](Federico A, et al. Pipeliner: a Nextflow-based framework for the definition of sequencing data processing pipelines, Front Genet. 2019;10:614.&#xA0;10.3389/fgene.2019.00614.) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32621602">32621602</a> <b>(1)</b> Tripal and Galaxy: supporting reproducible scientific workflows for community biological databases<ul><li> Some of these include Kepler (<xref rid="ref7" ref-type="bibr"><u>7</u></xref>), SnakeMake (<xref rid="ref8" ref-type="bibr"><u>8</u></xref>), <b>Nextflow</b> (<xref rid="ref9" ref-type="bibr"><u>9</u></xref>), the Galaxy Project (<xref rid="ref10" ref-type="bibr"><u>10</u></xref>), Pegasus (<xref rid="ref11" ref-type="bibr"><u>11</u></xref>) and others.<li> [7](<person-group person-group-type="author"><name name-style="western"><surname>Lud&#xE4;scher</surname><given-names>B.</given-names></name>, <name name-style="western"><surname>Altintas</surname><given-names>I.</given-names></name>, <name name-style="western"><surname>Berkley</surname><given-names>C.</given-names></name><etal>et&#xA0;al.</etal></person-group> (<year>2006</year>) <article-title>Scientific workflow management and the Kepler system</article-title>. <source>Concurr. Comp-Pract Exp</source>. <volume>18</volume>, <fpage>1039</fpage>&#x2013;<lpage>1065</lpage>.) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [11](<person-group person-group-type="author"><name name-style="western"><surname>Deelman</surname><given-names>E.</given-names></name>, <name name-style="western"><surname>Vahi</surname><given-names>K.</given-names></name>, <name name-style="western"><surname>Juve</surname><given-names>G.</given-names></name><etal>et&#xA0;al.</etal></person-group> (<year>2015</year>) <article-title>Pegasus, a workflow management system for science automation</article-title>. <source>Futur. Gener. Comput. Syst.</source>, <volume>46</volume>, <fpage>17</fpage>&#x2013;<lpage>35</lpage>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32630016">32630016</a> <b>(1)</b> An Economic Dilemma between Molecular Weapon Systems May Explain an Arachno-Atypical Venom in Wasp Spiders (<italic>Argiope bruennichi</italic>)<ul><li> All containers (<xref ref-type="app" rid="app1-biomolecules-10-00978"><u>Table S1</u></xref>) were established using <b>Nextflow</b> v19.01.0 (<uri xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</uri>).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32556167">32556167</a> <b>(14)</b> Watchdog 2.0: New developments for reusability, reproducibility, and workflow execution<ul><li> Popular WMSs are Galaxy [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], KNIME [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], Snakemake [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], and <b>Nextflow</b> [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] and differ in the implemented set of features, target audience, the required training period, usage fees, and more (for more details see the comparison in the first article on Watchdog [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] and at the end of this article).<li> The previously published version of Watchdog [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] was already extensively compared against the most popular WMSs for biological analyses, i.e., Galaxy (Galaxy, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_006281">RRID:SCR_006281</ext-link>)&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], KNIME (KNIME, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_006164">RRID:SCR_006164</ext-link>)&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], Snakemake (Snakemake, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_003475">RRID:SCR_003475</ext-link>)&#xA0;[<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], and <b>Nextflow</b> [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] (see Fig.<li> First, we provide a brief description of Galaxy, KNIME, Snakemake, and <b>Nextflow</b>.<li><b>Nextflow</b> extends the Unix pipes model to transfer complex data between consecutive processes as shared data streams.<li> Individual analysis steps are defined as processes in the <b>Nextflow</b> workflow itself; thus, no actual encapsulation of tools into reusable components is supported.<li><b>Nextflow</b></th>              </tr>            </thead>            <tbody>              <tr>                <td rowspan="1" colspan="1">Reusability.<li>For this part of the comparison, we focused on features that support development and sharing of tools (modules in Watchdog, tools in Galaxy, nodes in KNIME, rules in Snakemake, processes in <b>Nextflow</b>) for (re-)use in multiple analysis workflows as well as sharing and repurposing of existing workflows (F1&#x2013;F7 in Table&#xA0;<xref rid="tbl1" ref-type="table"><u>1</u></xref>).<li> Because there is no real encapsulation of tools in <b>Nextflow</b>, most of these features are not applicable to it.<li> Furthermore, dedicated sharing platforms are operated by the KNIME and <b>Nextflow</b> community [<xref rid="bib27" ref-type="bibr"><u>30</u></xref>,<xref rid="bib28" ref-type="bibr"><u>33</u></xref>].<li> For <b>Nextflow</b>, input is provided via command-line parameters.<li> However, Galaxy, Snakemake, and <b>Nextflow</b> support controlling external software dependencies and versions with the Conda package manager or using Docker containers (F9).<li>A description of all performed analysis steps (F10) can be obtained in Snakemake and <b>Nextflow</b> through generation of HTML reports, in which individual steps are listed in a table format and in the case of Snakemake also visualized as a graph.<li> With Snakemake and <b>Nextflow</b>, new samples (e.g., additional replicates) can be included in an analysis workflow without having to reprocess all samples (F14), but this option has to be forcibly triggered in Snakemake.<li>Finally, similar execution modes to the detach/reattach mode of Watchdog 2.0 (F15) are at least partly supported by all compared WMSs apart from <b>Nextflow</b>.<li> [8](<person-group person-group-type="author"><name name-style="western"><surname>Taylor</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Schenck</surname><given-names>I</given-names></name>, <name name-style="western"><surname>Blankenberg</surname><given-names>D</given-names></name>, <etal>et al</etal>.</person-group><article-title>Using Galaxy to perform large-scale interactive data analyses</article-title>. <source>Curr Protoc Bioinformatics</source>. <year>2007</year>;<volume>Chapter 10</volume>:<fpage>Unit 10.5</fpage>.) [9](<person-group person-group-type="author"><name name-style="western"><surname>Berthold</surname><given-names>MR</given-names></name>, <name name-style="western"><surname>Cebron</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Dill</surname><given-names>F</given-names></name>, <etal>et al</etal>.</person-group><article-title>KNIME: The Konstanz Information Miner</article-title>. In: <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>. <publisher-loc>Heidelberg-Berlin</publisher-loc>: <publisher-name>Springer</publisher-name>; <year>2007</year>:<fpage>319</fpage>&#x2013;<lpage>26</lpage>.) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a>) [8](<person-group person-group-type="author"><name name-style="western"><surname>Taylor</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Schenck</surname><given-names>I</given-names></name>, <name name-style="western"><surname>Blankenberg</surname><given-names>D</given-names></name>, <etal>et al</etal>.</person-group><article-title>Using Galaxy to perform large-scale interactive data analyses</article-title>. <source>Curr Protoc Bioinformatics</source>. <year>2007</year>;<volume>Chapter 10</volume>:<fpage>Unit 10.5</fpage>.) [9](<person-group person-group-type="author"><name name-style="western"><surname>Berthold</surname><given-names>MR</given-names></name>, <name name-style="western"><surname>Cebron</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Dill</surname><given-names>F</given-names></name>, <etal>et al</etal>.</person-group><article-title>KNIME: The Konstanz Information Miner</article-title>. In: <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>. <publisher-loc>Heidelberg-Berlin</publisher-loc>: <publisher-name>Springer</publisher-name>; <year>2007</year>:<fpage>319</fpage>&#x2013;<lpage>26</lpage>.) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [30](<comment>NodePit. <ext-link ext-link-type="uri" xlink:href="https://nodepit.com">https://nodepit.com</ext-link>. Accessed 11 November 2019</comment>.) [33](<comment>nf-core. <ext-link ext-link-type="uri" xlink:href="https://nf-co.re">https://nf-co.re</ext-link>. Accessed 11 November 2019</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32511224">32511224</a> <b>(2)</b> Genetic and functional diversification of chemosensory pathway receptors in mosquito-borne filarial nematodes<ul><li> The RNA-seq pipeline was implemented using <b>Nextflow</b> [<xref rid="pbio.3000723.ref111" ref-type="bibr"><u>111</u></xref>] and is publicly available (<ext-link ext-link-type="uri" xlink:href="https://github.com/zamanianlab/Bmalayi_HTRNAseq-nf">https://github.com/zamanianlab/Bmalayi_HTRNAseq-nf</ext-link>).<li> The RNA-seq pipeline was implemented using <b>Nextflow</b> [<xref rid="pbio.3000723.ref111" ref-type="bibr"><u>111</u></xref>] and is publicly available (<ext-link ext-link-type="uri" xlink:href="https://github.com/zamanianlab/BmalayiRNAseq-nf">https://github.com/zamanianlab/BmalayiRNAseq-nf</ext-link>).<li> [111](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [111](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><u>No PMID</u> <b>(2)</b> APE: A Command-Line Tool and API for Automated Workflow Composition<ul><li> Numerous scientific workflow management systems exist&#xA0;[<xref ref-type="bibr" rid="CR1"><u>1</u></xref>, <xref ref-type="bibr" rid="CR29"><u>29</u></xref>], some of the currently most popular being Galaxy&#xA0;[<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], KNIME&#xA0;[<xref ref-type="bibr" rid="CR6"><u>6</u></xref>] and <b>Nextflow</b>&#xA0;[<xref ref-type="bibr" rid="CR7"><u>7</u></xref>].<li> The mapping process described in this paper can furthermore serve as a template for the translation of APE results to other workflow formats, such as <b>NextFlow</b>&#xA0;[<xref ref-type="bibr" rid="CR7"><u>7</u></xref>], SnakeMake&#xA0;[<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] or the Workflow Description Language (WDL)&#xA0;[<xref ref-type="bibr" rid="CR3"><u>3</u></xref>].<li> [1](Existing Workflow systems. <ext-link ext-link-type="uri" xlink:href="https://s.apache.org/existing-workflow-systems">https://s.apache.org/existing-workflow-systems</ext-link>) [29](Wikipedia contributors: scientific workflow system &#x2013; Wikipedia, the free encyclopedia (2019). <ext-link ext-link-type="uri" xlink:href="https://en.wikipedia.org/w/index.php?title=Scientific_workflow_system&amp;oldid=928001704">https://en.wikipedia.org/w/index.php?title=Scientific_workflow_system&amp;oldid=928001704</ext-link>. Accessed 3 Feb 2020) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [6](          <person-group person-group-type="author">            <name>              <surname>Berthold</surname>              <given-names>MR</given-names>            </name>            <etal/>          </person-group>          <article-title>Knime-the konstanz information miner: version 2.0 and beyond</article-title>          <source>AcM SIGKDD Explor. Newslett.</source>          <year>2009</year>          <volume>11</volume>          <issue>1</issue>          <fpage>26</fpage>          <lpage>31</lpage>          <pub-id pub-id-type="doi">10.1145/1656274.1656280</pub-id>        ) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [3](Workflow Description Language (WDL), April 2020. <ext-link ext-link-type="uri" xlink:href="https://github.com/openwdl/wdl">https://github.com/openwdl/wdl</ext-link>, original-date: 2012&#x2013;08-01T03:12:48Z) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32442274">32442274</a> <b>(4)</b> NanoSPC: a scalable, portable, cloud compatible viral nanopore metagenomic data processing pipeline<ul><li> We implement NanoSPC using <b>Nextflow</b> pipeline manager within Docker images to allow reproducibility and portability of the analysis.<li> NanoSPC uses <b>Nextflow</b> pipeline manager and packs all the software dependencies within Docker images (red).<li>In order to perform reproducible and portable analysis of sequencing data, we implement NanoSPC using <b>Nextflow</b> pipeline manager (<xref rid="B31" ref-type="bibr"><u>31</u></xref>) that enables parallel processing of multiple datasets (Figure <xref ref-type="fig" rid="F1"><u>1</u></xref>).<li> All SP<sup>3</sup> software are deployed to a head node equipped with Ubuntu 18.04 and <b>Nextflow</b>.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32414140">32414140</a> <b>(1)</b> Antimicrobial Resistance Profiles of Adherent Invasive <italic>Escherichia coli</italic> Show Increased Resistance to &#x3B2;-Lactams<ul><li> The analysis pipelines and workflows were executed using <b>Nextflow</b> (SeqEra Labs, Barcelona, Spain) [<xref rid="B50-antibiotics-09-00251" ref-type="bibr"><u>50</u></xref>].<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33575589">33575589</a> <b>(2)</b> Comparative performance of the BGI and Illumina sequencing technology for single-cell RNA-sequencing<ul><li> This is further supported by data generated by the NextSeq 500 using the <b>NextFlow</b> version 2.0 sequencing reagent kit, which yielded lower quality RNA reads (<xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Table S1</u></xref>) and subsequently identified 3,085 less correctly assigned cells, in comparison to the <b>NextFlow</b> version 2.5 kit (<xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Table S3</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32397992">32397992</a> <b>(1)</b> Optimizing 16S rRNA gene profile analysis from low biomass nasopharyngeal and induced sputum specimens<ul><li> The <b>Nextflow</b> tool [<xref ref-type="bibr" rid="CR80"><u>80</u></xref>] was used to loop the bioinformatics processing workflow.<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32374845">32374845</a> <b>(1)</b> The <italic>Quest for Orthologs</italic> benchmark service and consensus calls in 2020<ul><li> Indeed, the QfO benchmark service now uses software containers within <b>Nextflow</b> workflows (<xref rid="B20" ref-type="bibr"><u>20</u></xref>).<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32374865">32374865</a> <b>(1)</b> miEAA 2.0: integrating multi-species microRNA enrichment analysis and workflow management systems<ul><li> First, reproducibility of statistical experiments can be improved, because usage of the versioned API in the context of a workflow manager such as Snakemake (<xref rid="B26" ref-type="bibr"><u>26</u></xref>) or <b>Nextflow</b> fosters self-documenting research setups (<xref rid="B43" ref-type="bibr"><u>43</u></xref>).<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32369166">32369166</a> <b>(1)</b> MaRe: Processing Big Data with application containers on Apache Spark<ul><li> Cutting-edge workflow systems, such as Luigi [<xref rid="bib14" ref-type="bibr"><u>14</u></xref>], <b>NextFlow</b> [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>], Galaxy [<xref rid="bib16" ref-type="bibr"><u>16</u></xref>], and Pachyderm [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>], allow for running processing tools as application containers.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27942268">27942268</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](<person-group person-group-type="author"><name name-style="western"><surname>Moreno</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Pireddu</surname><given-names>L</given-names></name>, <name name-style="western"><surname>Roger</surname><given-names>P</given-names></name>, <etal>et al</etal>.</person-group><article-title>Galaxy-Kubernetes integration: scaling bioinformatics workflows in the cloud</article-title>. <source>BioRxiv</source>. <year>2018</year>:<fpage>488643</fpage>.) [17](<person-group person-group-type="author"><name name-style="western"><surname>Novella</surname><given-names>JA</given-names></name>, <name name-style="western"><surname>Emami&#xA0;Khoonsari</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Herman</surname><given-names>S</given-names></name>, <etal>et al</etal>.</person-group><article-title>Container-based bioinformatics with Pachyderm</article-title>. <source>Bioinformatics</source>. <year>2018</year>;<volume>35</volume>(<issue>5</issue>):<fpage>839</fpage>&#x2013;<lpage>46</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32357829">32357829</a> <b>(9)</b> wg-blimp: an end-to-end analysis pipeline for whole genome bisulfite sequencing data<ul><li> We compared wg-blimp to BAT [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], bicycle [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>], CpG_Me/DMRichR [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>, <xref ref-type="bibr" rid="CR22"><u>22</u></xref>&#x2013;<xref ref-type="bibr" rid="CR24"><u>24</u></xref>], ENCODE-DCC&#x2019;s WGBS pipeline [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>], Methy-Pipe [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>], <b>Nextflow</b> methylseq (two available workflows) [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>], PiGx [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>] and snakePipes [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>].<li> Using package managers such as Bioconda or workflow environments like <b>Nextflow</b> [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] not only simplifies installation for users but also provides straightforward update processes of both the pipeline itself as well as its dependencies.<li> For workflow management, we prefer using dedicated workflow management systems such as Snakemake or <b>Nextflow</b> over plain shell scripts, as these allow more scalable and robust execution.<li><b>Nextflow</b> methylseq (Bismark).<li><b>Nextflow</b>.<li><b>Nextflow</b>.<li><b>Nextflow</b> methylseq (bwa-meth).<li><b>Nextflow</b>.<li><b>Nextflow</b>.<li> [20](Kretzmer H, Otto C, Hoffmann S. BAT: Bisulfite analysis toolkit [version 1; peer review: 3 approved]. F1000Research. 2017; 6(1490). 10.12688/f1000research.12302.1.) [21](          <person-group person-group-type="author">            <name>              <surname>Gra&#xF1;a</surname>              <given-names>O</given-names>            </name>            <name>              <surname>L&#xF3;pez-Fern&#xE1;ndez</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Fdez-Riverola</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Gonz&#xE1;lez Pisano</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Glez-Pe&#xF1;a</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Bicycle: a bioinformatics pipeline to analyze bisulfite sequencing data</article-title>          <source>Bioinformatics</source>          <year>2017</year>          <volume>34</volume>          <issue>8</issue>          <fpage>1414</fpage>          <lpage>5</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btx778</pub-id>        ) [10](          <person-group person-group-type="author">            <name>              <surname>Hansen</surname>              <given-names>KD</given-names>            </name>            <name>              <surname>Langmead</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Irizarry</surname>              <given-names>RA</given-names>            </name>          </person-group>          <article-title>BSmooth: from whole genome bisulfite sequencing reads to differentially methylated regions</article-title>          <source>Genome Biol</source>          <year>2012</year>          <volume>13</volume>          <issue>10</issue>          <fpage>83</fpage>          <pub-id pub-id-type="doi">10.1186/gb-2012-13-10-r83</pub-id>        ) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31010359">31010359</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21493656">21493656</a>) [25](          <person-group person-group-type="author">            <name>              <surname>Davis</surname>              <given-names>CA</given-names>            </name>            <name>              <surname>Hitz</surname>              <given-names>BC</given-names>            </name>            <name>              <surname>Sloan</surname>              <given-names>CA</given-names>            </name>            <name>              <surname>Chan</surname>              <given-names>ET</given-names>            </name>            <name>              <surname>Davidson</surname>              <given-names>JM</given-names>            </name>            <name>              <surname>Gabdank</surname>              <given-names>I</given-names>            </name>            <name>              <surname>Hilton</surname>              <given-names>JA</given-names>            </name>            <name>              <surname>Jain</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Baymuradov</surname>              <given-names>UK</given-names>            </name>            <name>              <surname>Narayanan</surname>              <given-names>AK</given-names>            </name>            <name>              <surname>Onate</surname>              <given-names>KC</given-names>            </name>            <name>              <surname>Graham</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Miyasato</surname>              <given-names>SR</given-names>            </name>            <name>              <surname>Dreszer</surname>              <given-names>TR</given-names>            </name>            <name>              <surname>Strattan</surname>              <given-names>JS</given-names>            </name>            <name>              <surname>Jolanki</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Tanaka</surname>              <given-names>FY</given-names>            </name>            <name>              <surname>Cherry</surname>              <given-names>JM</given-names>            </name>          </person-group>          <article-title>The Encyclopedia of DNA elements (ENCODE): data portal update</article-title>          <source>Nucleic Acids Res</source>          <year>2017</year>          <volume>46</volume>          <issue>D1</issue>          <fpage>794</fpage>          <lpage>801</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkx1081</pub-id>        ) [26](          <person-group person-group-type="author">            <name>              <surname>Jiang</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Sun</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Lun</surname>              <given-names>FMF</given-names>            </name>            <name>              <surname>Guo</surname>              <given-names>AM</given-names>            </name>            <name>              <surname>Wang</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Chan</surname>              <given-names>KCA</given-names>            </name>            <name>              <surname>Chiu</surname>              <given-names>RWK</given-names>            </name>            <name>              <surname>Lo</surname>              <given-names>YMD</given-names>            </name>            <name>              <surname>Sun</surname>              <given-names>H</given-names>            </name>          </person-group>          <article-title>Methy-pipe: An integrated bioinformatics pipeline for whole genome bisulfite sequencing data analysis</article-title>          <source>PLOS ONE</source>          <year>2014</year>          <volume>9</volume>          <issue>6</issue>          <fpage>1</fpage>          <lpage>11</lpage>        ) [27](Ewels P., Hammar&#xE9;n R., Peltzer A., H&#xFC;ther P., F. S., Tommaso P. D., Garcia M., Alneberg J., Wilm A.Alessia nf-core/methylseq: nf-core/methylseq version 1.3. Zenodo. 2019. 10.5281/zenodo.2555454.) [28](Gosdschan A, Wreczycka K, Osberg B, Wurmus R. PiGx. 2019. <ext-link ext-link-type="uri" xlink:href="https://github.com/BIMSBbioinfo/pigx_bsseq">https://github.com/BIMSBbioinfo/pigx_bsseq</ext-link>. Accessed 13 Nov 2019.) [19](Bhardwaj V, Heyne S, Sikora K, Rabbani L, Rauer M, Kilpert F, Richter AS, Ryan DP, Manke T. snakePipes: facilitating flexible, scalable and integrative epigenomic analysis. Bioinformatics. 2019. 10.1093/bioinformatics/btz436.) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32353042">32353042</a> <b>(1)</b> Combining signal and sequence to detect RNA polymerase initiation in ATAC-seq data<ul><li>Both ATAC-seq SRRs and nascent transcription (GRO/PRO-seq) SRRs were processed using <b>Nextflow</b>-based [<xref rid="pone.0232332.ref017" ref-type="bibr"><u>17</u></xref>] pipelines [<xref rid="pone.0232332.ref018" ref-type="bibr"><u>18</u></xref>, <xref rid="pone.0232332.ref019" ref-type="bibr"><u>19</u></xref>].<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](Tripodi IJ, Gruca M. Nascent-Flow v1.0. 2018.) [19](Gruca M. ChIP-Flow v1.3, Nextflow implementation of the Dowell Lab ChIP-seq pipeline; 2019. <ext-link ext-link-type="uri" xlink:href="https://github.com/Dowell-Lab/ChIP-Flow">https://github.com/Dowell-Lab/ChIP-Flow</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32345704">32345704</a> <b>(1)</b> Reference Genome Assembly for Australian <italic>Ascochyta rabiei</italic> Isolate ArME14<ul><li> Computational steps were coordinated using <b>Nextflow</b> (<xref rid="bib68" ref-type="bibr">Di Tommaso <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32363341">32363341</a> <b>(2)</b> Needlestack: an ultra-sensitive variant caller for multi-sample next generation sequencing data<ul><li> We deployed our pipeline focusing on efficiency and robustness using the domain-specific language (DSL) <b>nextflow</b> (<xref rid="B12" ref-type="bibr"><u>12</u></xref>), and on reproducibility by providing Docker and Singularity images.<li>Our pipeline is implemented using <b>nextflow</b> (<xref rid="B12" ref-type="bibr"><u>12</u></xref>), to facilitate its scientific reproducibility but also efficient parallel computations.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32306927">32306927</a> <b>(37)</b> DolphinNext: a distributed data processing platform for high throughput genomics<ul><li><b>Nextflow</b> is a recently developed workflow engine built to address many of these needs [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>].<li> The <b>Nextflow</b> engine can be configured to use a variety of executors (e.g.<li> Further, <b>Nextflow</b> allows in-line process definition that simplifies the incorporation of small processes that implement new functionality.<li> Not surprisingly, <b>Nextflow</b> has quickly gained popularity, as reflected by several efforts to provide curated and revisioned <b>Nextflow</b>-based pipelines such as nf-core [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], Pipeliner [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] and CHIPER [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], which are available from a public repository.<li>In spite of its simplicity, <b>Nextflow</b> can get unwieldy when pipelines become complex, and maintaining them becomes taxing.<li> DolphinNext builds on <b>Nextflow</b> as shown in Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>.<li> Both advanced <b>Nextflow</b> users as well as users with no prior experience benefit from the ability to visualize dependencies, branch points, and parallel processing opportunities.<li> DolphinNext goes beyond providing a <b>Nextflow</b> graphical design environment and addresses many of the needs of high-throughput data processing: First, DolphinNext helps with reproducibility enabling the easy distribution and running of pipelines.<li>DolphinNext builds on <b>Nextflow</b> and simplifies creating complex workflows.<li>Third, as with <b>Nextflow</b>, DolphinNext is implemented as a generic workflow design and execution environment.<li> This focus is driven by our current use of DolphinNext, but its architecture is designed to support any workflow that can be supported by <b>Nextflow</b>.<li> At its core, it provides a drag-and-drop user interface for creating and modifying <b>Nextflow</b> pipelines.<li> <b>Nextflow</b> [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] is a language to create scalable and reproducible scientific workflows.<li> In creating DolphinNext, we aim to simplify <b>Nextflow</b> pipeline building by shifting the focus from software engineering to bioinformatics processes using a graphical interface that requires no programming experience.<li>To facilitate the reproducibility of data processing and the execution of pipelines in any computing environment, DolphinNext leverages <b>Nextflow</b>&#x2019;s support for Singularity and Docker container technologies [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>, <xref ref-type="bibr" rid="CR26"><u>26</u></xref>].<li> This allows the execution of a pipeline created by DolphinNext to require only <b>Nextflow</b> and a container software (Singularity or Docker) to be installed in the host machine.<li> Alternatively, DolphinNext pipelines can be exported as <b>Nextflow</b> code and distributed in publications.<li> To facilitate re-running of a pipeline, DolphinNext builds on <b>Nextflow</b>&#x2019;s ability to record a pipeline execution state, enabling the ability to re-execute or resume a pipeline from any of its steps, even after correcting parameters or correcting a process.<li> DolphinNext readily integrates with Amazon AWS and Google GCP where, a new, dedicated computer cluster can easily be set up within DolphinNext with <b>Nextflow</b>&#x2019;s Amazon and Google cloud support.<li> DolphinNext relies on <b>Nextflow</b> [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] to encapsulate computing environment settings and allows administrators to rely on a single configuration file that enables users to run the pipelines on diverse environments with minimal impact on user experience.<li>While <b>Nextflow</b> provides a powerful platform to build pipelines, it requires advanced programming skills to define pipelines as it requires users to use a programming language to specify processes, dependencies, and the execution order.<li>DolphinNext facilitates pipeline building, maintenance, and execution by providing a graphical user interface to create and modify <b>Nextflow</b> pipelines.<li> Once a pipeline is created, users can track revisions, edit, delete and share either as a stand-alone container <b>Nextflow</b> program, or in PDF format for documentation purposes.<li> Unlike nf-core or other <b>Nextflow</b> based pipeline repositories [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>&#x2013;<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], DolphinNext keeps track of revisions for each of the processes within a pipeline rather than keeping revisions for each pipeline.<li>Although integration with GitHub makes sharing and executing possible, pipelines can also be downloaded in <b>Nextflow</b> format for documentation, distribution and execution outside of DolphinNext.<li> Further, it also gives access to <b>Nextflow</b>&#x2019;s execution reports, which include the execution time, resource utilization (e.g.<li>On the other hand, using <b>Nextflow</b>&#x2019;s cloud support and containerization, DolphinNext supports all these environments through configurations stored within DolphinNext.<li> If a pipeline is shared through GitHub, investigators can also directly execute the pipeline from the command line using <b>Nextflow</b> without a DolphinNext instance installed.<li> However, it is an easy task in DolphinNext, once input parameters are changed and the run resumed, <b>Nextflow</b> only executes the affected processes with this change to produce the new reports.<li>Comparison of related applications</p></caption><table frame="hsides" rules="groups"><thead><tr><th/><th>DolphinNext</th><th>Galaxy [<xref ref-type="bibr" rid="CR5"><u>5</u></xref>]</th><th>Sequanix [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>]</th><th>Taverna [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>]</th><th>Arvados [<xref ref-type="bibr" rid="CR63"><u>63</u></xref>]</th></tr></thead><tbody><tr><td>Platform<sup>a</sup></td><td>JS/PHP</td><td>Python</td><td>Python</td><td>Java</td><td>Go</td></tr><tr><td>Workflow management system</td><td><b>Nextflow</b>.<li>DolphinNext currently relies on <b>Nextflow</b>.<li> Although <b>Nextflow</b> is a very successful and widely used pipeline engine, supporting other platforms like SnakeMake [<xref ref-type="bibr" rid="CR64"><u>64</u></xref>] and WDL based execution engines [<xref ref-type="bibr" rid="CR65"><u>65</u></xref>] would increase the usage of DolphinNext.<li> This approach required more structured syntax, which limits the ability to import publicly available <b>Nextflow</b> pipelines such as those available from nf-core [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], Pipeliner [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] or CHIPER [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>].<li> It builds on top of recent technological advances such as software containerization and the <b>Nextflow</b> engine to address current data processing needs in high-throughput biological data analysis.<li> Further, DolphinNext supports pipeline versioning where users can create and tag a pipeline version with a unique identifier (UID) that includes the general pipeline &#x201C;graphical representation&#x201D;, the executable <b>Nextflow</b> script, as well as the exact software versions and parameters used when run on a specific dataset.<li><bold>Other requirements:</bold> <b>Nextflow</b> and Java 8 or higher, docker or singularity for containerization.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, et al. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020;38:276&#x2013;8.) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31316552">31316552</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28789639">28789639</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [26](          <person-group person-group-type="author">            <name>              <surname>Hale</surname>              <given-names>JS</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>L</given-names>            </name>            <name>              <surname>Richardson</surname>              <given-names>CN</given-names>            </name>            <name>              <surname>Wells</surname>              <given-names>GN</given-names>            </name>          </person-group>          <article-title>Containers for Portable, Productive, and Performant Scientific Computing</article-title>          <source>Computing in Science Engineering.</source>          <year>2017</year>          <volume>19</volume>          <fpage>40</fpage>          <lpage>50</lpage>          <pub-id pub-id-type="doi">10.1109/MCSE.2017.2421459</pub-id>        ) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, et al. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020;38:276&#x2013;8.) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28789639">28789639</a>) [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29361152">29361152</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [63](Amstutz P. Portable, Reproducible Analysis with Arvados. F1000Res. 2015;4 <ext-link ext-link-type="uri" xlink:href="https://f1000research.com/assets/download/1110114">https://f1000research.com/assets/download/1110114</ext-link>.) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [65](          <person-group person-group-type="author">            <name>              <surname>Chapman</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Gentry</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lin</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Magee</surname>              <given-names>P</given-names>            </name>            <name>              <surname>O&#x2019;Connor</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Prabhakaran</surname>              <given-names>A</given-names>            </name>            <etal/>          </person-group>          <source>OpenWDL</source>          <year>2019</year>        ) [18](Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, et al. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020;38:276&#x2013;8.) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31316552">31316552</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28789639">28789639</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32337106">32337106</a> <b>(3)</b> CoproID predicts the source of coprolites and paleofeces using microbiome composition and host DNA content<ul><li> 2</xref>) (<uri xlink:href="http://dx.doi.org.10.5281/zenodo.2653757">DOI 10.5281/zenodo.2653757</uri>) written using <b>Nextflow</b> (<xref rid="ref-15" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and made available through nf-core (<xref rid="ref-16" ref-type="bibr"><u>Ewels et al., 2019</u></xref>).<li> <b>Nextflow</b> is a Domain Specific Language designed to ensure reproducibility and scalability for scientific pipelines, and nf-core is a community-developed set of guidelines and tools to promote standardization and maximum usability of <b>Nextflow</b> pipelines.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Ewels et al., 2019](          <person-group person-group-type="author">            <name>              <surname>Ewels</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Peltzer</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Fillinger</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Alneberg</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Patel</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Garcia</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>nf-core: Community curated bioinformatics pipelines</article-title>          <source>bioRxiv</source>          <year>2019</year>          <fpage>610741</fpage>          <pub-id pub-id-type="doi">10.1101/610741</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32273506">32273506</a> <b>(1)</b> Cancer neoantigen prioritization through sensitive and reliable proteogenomics analysis<ul><li> NeoFlow was implemented into a workflow using <b>nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</ext-link>)<sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup> and is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/bzhanglab/neoflow">https://github.com/bzhanglab/neoflow</ext-link>.<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269230">32269230</a> <b>(1)</b> DNA punch cards for storing data on native DNA sequences via enzymatic nicking<ul><li>Data was processed using a <b>Nextflow</b>-based workflow<sup><xref ref-type="bibr" rid="CR26"><u>26</u></xref></sup>, implemented as follows.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32318338">32318338</a> <b>(1)</b> Computational Oncology in the Multi-Omics Era: State of the Art<ul><li> Examples of these systems include Galaxy (<xref rid="B21" ref-type="bibr"><u>21</u></xref>), Snakemake (<xref rid="B22" ref-type="bibr"><u>22</u></xref>), <b>Nextflow</b> (<xref rid="B23" ref-type="bibr"><u>23</u></xref>), and the general purpose Common Workflow Language (<xref rid="B24" ref-type="bibr"><u>24</u></xref>).<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P</given-names></name><name><surname>Crusoe</surname><given-names>MR</given-names></name><name><surname>Tijani&#x107;</surname><given-names>N</given-names></name><name><surname>Chapman</surname><given-names>B</given-names></name><name><surname>Chilton</surname><given-names>J</given-names></name><name><surname>Heuer</surname><given-names>M</given-names></name><etal/></person-group><article-title>Common Workflow Language, v1.0</article-title>. (<year>2016</year>). Available online at: <ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32241255">32241255</a> <b>(1)</b> EpiMOLAS: an intuitive web-based framework for genome-wide DNA methylation analysis<ul><li>Docker|Galaxy web server</td><td>Docker</td><td>Shell script</td><td>Bioconda Snakemake</td><td>Docker <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31907277">31907277</a> <b>(1)</b> Tumoral and immune heterogeneity in an anti-PD-1-responsive glioblastoma: a case study<ul><li>Data processing for WES samples was performed using a custom <b>Nextflow</b> pipeline that is available on GitHub (<uri xlink:href="https://github.com/losiclab/exoseq">https://github.com/losiclab/exoseq</uri>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a> <b>(23)</b> MasterOfPores: A Workflow for the Analysis of Oxford Nanopore Direct RNA Sequencing Datasets<ul><li><italic>MasterOfPores</italic> simplifies the analysis of direct RNA sequencing data by providing a containerized pipeline implemented in the <b>NextFlow</b> framework.<li> During the last decade, an increasing interest in the field has led to the development of different programs such as Snakemake (<xref rid="B20" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), <b>NextFlow</b> (<xref rid="B12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), Galaxy (<xref rid="B1" ref-type="bibr"><u>Afgan et al., 2018</u></xref>), SciPipe (<xref rid="B23" ref-type="bibr"><u>Lampa et al., 2019</u></xref>) or GenPipes (<xref rid="B6" ref-type="bibr"><u>Bourgey et al., 2019</u></xref>), among others.<li>Here we chose the workflow framework <b>NextFlow</b> (<xref rid="B12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) because of its native support of different batch schedulers (SGE, LSF, SLURM, PBS, and HTCondor), cloud platforms (Kubernetes, Amazon AWS, and Google Cloud) and GPU computing, which is crucial for processing huge volumes of data produced by nanopore sequencers.<li> <b>NextFlow</b> has tight integration with lightweight Linux containers, such as Docker and Singularity.<li> Automatic organization of intermediate results produced during the <b>NextFlow</b> pipeline execution allows reducing the complexity of intermediary file names and the possibility of name clashing.<li> Continuous check-pointing with the possibility of resuming failed executions, interoperability and meticulous monitoring and reporting of resource usage are among other thought-after features of <b>NextFlow</b>.<li>Install <b>NextFlow</b> (version 19.10.0):.<li>$ <italic>curl -s <ext-link ext-link-type="uri" xlink:href="https://get.<b>nextflow</b>.io">https://get.<b>nextflow</b>.io</ext-link> | bash</italic>.<li>$ <b>nextflow</b> run nanopreprocess.nf-with-singularity.<li>$ <italic><b>nextflow</b> run nanopreprocess.nf-with-docker</italic>.<li>$ <italic><b>nextflow</b> run nanotail.nf-bg-with-singularity &#x2013;input_folders &#x201C;.NanoPreprocess/output/RNA<sup>&#x2217;</sup>&#x201D;</italic>.<li>$ <b>nextflow</b> run nanomod.nf <italic>-with-singularityinput_path &#x201C;.NanoPreprocess/output/&#x201D;</italic>.<li><italic>$ cd master_of_pores/NanoPreprocess</italic><italic>$ <b>nextflow</b> run nanopreprocess.nf &#x2013;basecaller guppy &#x2013;seqtype RNA \</italic><italic>&#x2013;fast5 &#x201C;FOLDERNAME/<sup>&#x2217;</sup>.fast5&#x201D; &#x2013;demultiplexing &#x201C;OFF&#x201D; \</italic><italic>&#x2013;map_type &#x201C;spliced&#x201D; &#x2013;mapper_opt &#x201C;-uf -k14&#x201D; \</italic><italic>&#x2013;reference genome.fa.gz &#x2013;mapper minimap2 &#x2013;ref_type &#x201C;genome&#x201D;\</italic><italic>&#x2013;filter nanofilt &#x2013;filter_opt &#x201C;-q 0 &#x2013;headcrop 5 &#x2013;tailcrop 3 &#x2013;readtype 1D</italic>&#x201D;.<li><italic>$ <b>nextflow</b> run nanotail.nf &#x2013;input_folders &#x201C;./NanoPreprocess/output/<sup>&#x2217;</sup>&#x201D; \</italic><italic>&#x2013;nanopolish_opt &#x201C;&#x201D; &#x2013;tailfindr_opt &#x201C;&#x201D; &#x2013;reference &#x201C;genome.fa.gz&#x201D;</italic>.<li><italic>$ <b>nextflow</b> run nanomod.nf &#x2013;input_path &#x201C;./NanoPreprocess/output/&#x201D; \</italic><italic>&#x2013;comparison &#x201C;./comparison.tsv&#x201D; &#x2013;reference &#x201C;genome.fa.gz&#x201D; \</italic><italic>&#x2013;tombo_opt &#x201C;&#x2013;num-bases 5&#x201D; &#x2013;epinano_opt &#x201C;&#x201D;</italic>.<li>Taking advantage of the <b>NextFlow</b> reporting functions, the pipeline can produce detailed reports on the time and resources consumed by each process (<xref ref-type="fig" rid="F4"><u>Figure 4</u></xref>), in addition to the output files (bam, fastq) and final report (html), if the workflow is executed with parameters <italic>-with-report</italic> (formatted report) or<italic>-with-trace</italic> (plain text report).<li>Snapshot of the <b>NextFlow</b> resources report.<li>Finally, we should note that the latest (19.10.0) version of <b>NextFlow</b> allows the user to control the execution of a pipeline remotely.<li> To enable this feature, the user needs to login to the <ext-link ext-link-type="uri" xlink:href="https://tower.nf/">https://tower.nf/</ext-link>website developed by the <b>NextFlow</b> authors and retrieve a token for communicating with the pipeline.<li><italic>$ export TOWER_ACCESS_TOKEN</italic> = <italic>YOUR_TOKEN</italic><italic>$ cd master_of_pores/NanoPreprocess</italic><italic>$ <b>nextflow</b> run nanopreprocess.nf -with-docker -with-report -bg -with-tower</italic>.<li> To overcome these issues and facilitate the data analysis of direct RNA sequencing to the general user, we propose the use of <b>NextFlow</b> workflows.<li>Specifically, we propose the use of <italic>MasterOfPores</italic> workflow for the analysis of direct RNA sequencing datasets, which is a containerized pipeline implemented in the <b>NextFlow</b> framework.<li> <italic>MasterOfPores</italic> can handle both single- and multi-FAST5 reads as input, is highly customizable by the user (<xref rid="T2" ref-type="table"><u>Table 2</u></xref>) and produces informative detailed reports on both the FAST5 data processing and analysis (MultiQC report, <xref ref-type="fig" rid="F3"><u>Figure 3</u></xref>) as well as on the computing resources used to perform each step (<b>NextFlow</b> report, see <xref ref-type="fig" rid="F4"><u>Figure 4</u></xref>).<li> Thus, the current outputs of the <italic>MasterOfPores</italic> workflow include: (i) base-called FAST5 files, (ii) base-called fastq file, (iii) sorted BAM file containing mapped reads, (iv) per-gene or per-transcript counts (depending on algorithm choice), (v) MultiQC report, (vi) <b>NextFlow</b> report, (vii) per-read polyA tail length estimations, including the correlation of predictions using two distinct algorithms, and (viii) per-site RNA modification predictions, including a final plain text file containing the consensus sites predicted by two distinct algorithms.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Afgan et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [Lampa et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) [Bourgey et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31185495">31185495</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32134380">32134380</a> <b>(4)</b> DEN-IM: dengue virus genotyping from amplicon and shotgun metagenomic sequencing<ul><li> DEN-IM is implemented in <b>Nextflow</b> [<xref rid="R14" ref-type="bibr"><u>14</u></xref>], workflow management software that uses Docker (<ext-link ext-link-type="uri" xlink:href="https://www.docker.com">https://www.docker.com</ext-link>) containers with preinstalled software for all the workflow tools.<li>It is implemented in <b>Nextflow</b>, a workflow management system that allows the effortless deployment and execution of complex distributed computational workflows in any UNIX-based system, from local machines to high-performance computing clusters (HPCs) with a container engine installation, such as Docker (<ext-link ext-link-type="uri" xlink:href="https://www.docker.com/">https://www.docker.com/</ext-link>), Shifter [<xref rid="R15" ref-type="bibr"><u>15</u></xref>] or Singularity [<xref rid="R16" ref-type="bibr"><u>16</u></xref>].<li> The software versions are also described in the <b>Nextflow</b> script and configuration files, and in the Docker files for each container, allowing the traceability of each step of data processing.<li>Having been developed in <b>Nextflow</b>, DEN-IM runs on any UNIX-like system and provides out-of-the-box support for several job schedulers (e.g.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](          <person-group person-group-type="author">            <name>              <surname>Gerhardt</surname>              <given-names>L</given-names>            </name>            <name>              <surname>Bhimji</surname>              <given-names>W</given-names>            </name>            <name>              <surname>Canon</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Fasel</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Jacobsen</surname>              <given-names>D</given-names>            </name>            <etal>et al</etal>          </person-group>          <article-title>Shifter: containers for HPC</article-title>          <source>J Phys Conf Ser</source>          <year>2017</year>          <volume>898</volume>          <fpage>082021</fpage>          <pub-id pub-id-type="doi">10.1088/1742-6596/898/8/082021</pub-id>        ) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32024987">32024987</a> <b>(1)</b> Butler enables rapid cloud-based analysis of thousands of human genomes<ul><li> The recently developed cloud-based scientific workflow frameworks <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR2"><u>2</u></xref></sup>, Toil<sup><xref ref-type="bibr" rid="CR3"><u>3</u></xref></sup> and GenomeVIP<sup><xref ref-type="bibr" rid="CR4"><u>4</u></xref></sup> focus their support largely on individual commercial cloud computing environments&#x2014;mostly Amazon Web Services&#x2014;and lack complete functionality for other major providers.<li> [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28522612">28522612</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32019919">32019919</a> <b>(1)</b> Organizing genome engineering for the gigabase scale<ul><li> There are also a number of bioinformatics-focused engines, including Cromwell<sup><xref ref-type="bibr" rid="CR104"><u>104</u></xref></sup>, Galaxy<sup><xref ref-type="bibr" rid="CR105"><u>105</u></xref></sup>, <b>NextFlow</b><sup><xref ref-type="bibr" rid="CR106"><u>106</u></xref></sup>, and Toil<sup><xref ref-type="bibr" rid="CR107"><u>107</u></xref></sup>.<li> [104](Broad Institute. The Workflow Description Language and cromwell, <ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/cromwell">https://github.com/broadinstitute/cromwell</ext-link> (2019).) [105](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [106](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [107](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33430992">33430992</a> <b>(1)</b> Towards reproducible computational drug discovery<ul><li> Prominent examples of this trends include tools like <b>Nextflow</b> [<xref ref-type="bibr" rid="CR207"><u>207</u></xref>], Snakemake&#xA0;[<xref ref-type="bibr" rid="CR208"><u>208</u></xref>], Ruffus&#xA0;[<xref ref-type="bibr" rid="CR209"><u>209</u></xref>], BPipe&#xA0;[<xref ref-type="bibr" rid="CR210"><u>210</u></xref>], Cuneiform [<xref ref-type="bibr" rid="CR211"><u>211</u></xref>] and Luigi [<xref ref-type="bibr" rid="CR212"><u>212</u></xref>].<li> [207](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [208](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [209](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [210](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [211](          <person-group person-group-type="author">            <name>              <surname>Brandt</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Reisig</surname>              <given-names>W</given-names>            </name>            <name>              <surname>Leser</surname>              <given-names>ULF</given-names>            </name>          </person-group>          <article-title>Computation semantics of the functional scientific workflow language cuneiform</article-title>          <source>J Funct Program</source>          <year>2017</year>          <pub-id pub-id-type="doi">10.1017/S0956796817000119</pub-id>        ) [212](Bernhardsson E, Freider E, Rouhani A (2012) Luigi GitHub repository.&#xA0;<ext-link ext-link-type="uri" xlink:href="https://github.com/spotify/luigi">https://github.com/spotify/luigi</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32002517">32002517</a> <b>(4)</b> JASS: command line and web interface for the joint analysis of GWAS results<ul><li> and (v) a <b>nextflow</b> (<xref rid="B24" ref-type="bibr"><u>24</u></xref>) pipeline integrating GWAS cleaning, imputation and analyses into one tool.<li> Note that we integrated the LDscore <italic>H</italic><sub>0</sub> matrix computation in our <b>nextflow</b> pipeline (see the&#xA0;&#x2018;Data pre- and postprocessing&#x2019; section).<li>We integrated these preprocessing steps into a <b>nextflow</b> (<xref rid="B24" ref-type="bibr"><u>24</u></xref>) pipeline, which provides several advantages for the user: (i) automated transmission of preprocessed data to subsequent analysis steps that greatly limit the number of input data left to the user; (ii) if executed on an HPC cluster, automatically parallelized subprocesses when possible; and (iii) a stable computing environment by assigning a docker container to each process.<li> The <b>nextflow</b> pipeline integrating all JASS preprocessing steps (GWAS harmonization, GWAS imputation and <italic>H</italic><sub>0</sub> covariance matrix computation) is available at <ext-link ext-link-type="uri" xlink:href="https://gitlab.pasteur.fr/statistical-genetics/jass_suite_pipeline">https://gitlab.pasteur.fr/statistical-genetics/jass_suite_pipeline</ext-link>.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31941899">31941899</a> <b>(1)</b> Intratumoral heterogeneity and clonal evolution in liver cancer<ul><li> WES data processing was performed using a custom <b>nextflow</b> pipeline that is available on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/losiclab/exoseq">https://github.com/losiclab/exoseq</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31930375">31930375</a> <b>(2)</b> RiboFlow, RiboR and RiboPy: an ecosystem for analyzing ribosome profiling data at read length resolution<ul><li> RiboFlow is a <b>Nextflow</b> (<xref rid="btaa028-B8" ref-type="bibr">Di Tommaso <italic>et al.<li>RiboFlow is a <b>Nextflow</b> (<xref rid="btaa028-B8" ref-type="bibr">Di Tommaso <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32161811">32161811</a> <b>(1)</b> Guidelines for systematic reporting of sequence alignments<ul><li> Big data poses special requirements, to handle such cases, methods like <b>Nextflow</b> [<xref rid="bpaa001-B8" ref-type="bibr"><u>8</u></xref>] can be used.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31909794">31909794</a> <b>(4)</b> CAMITAX: Taxon labels for microbial genomes<ul><li>CAMITAX incorporates many state-of-the-art pieces of software, and automatically resolves all software and database dependencies with <b>Nextflow</b> [<xref rid="bib56" ref-type="bibr"><u>56</u></xref>] in a containerized environment (Table&#xA0;<xref rid="tbl1" ref-type="table"><u>1</u></xref>).<li>CAMITAX automatically resolves all software dependencies with <b>Nextflow</b> using BioContainers in a containerized environment.<li> It uses <b>Nextflow</b> to orchestrate reference databases and software containers.<li>CAMITAX is implemented in <b>Nextflow</b> and Python&#xA0;3 and is freely available under Apache License 2.0 at <ext-link ext-link-type="uri" xlink:href="https://github.com/CAMI-challenge/CAMITAX">https://github.com/CAMI-challenge/CAMITAX</ext-link>.<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32175479">32175479</a> <b>(1)</b> Variant calling on the GRCh38 assembly with the data from phase three of the 1000 Genomes Project<ul><li> Our comparison is restricted to the regions in the autosomes and in the PAR region of chromosome X where GIAB considers calls to be high confidence (on average 77.9%, standard deviation 12.1%, of the bases for each of the chromosomes are in high confidence regions) and was performed using the <b>Nextflow</b><sup><xref rid="ref-21" ref-type="bibr"><u>21</u></xref></sup> workflow accessible from the link in the software availability section.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32025286">32025286</a> <b>(1)</b> BOSC 2019, the 20th annual Bioinformatics Open Source Conference<ul><li> Participants worked together to fix bugs, add new features, write tests, and improve documentation in new and ongoing projects including<ext-link ext-link-type="uri" xlink:href="https://biopython.org/">Biopython</ext-link>,<ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">Common Workflow Language (CWL)</ext-link>,<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/"><b>Nextflow</b></ext-link>,<ext-link ext-link-type="uri" xlink:href="https://github.com/bigdatagenomics/cannoli">Cannoli</ext-link>,<ext-link ext-link-type="uri" xlink:href="https://github.com/codeisscience/software-usage-reporting">CodeCite</ext-link>, the<ext-link ext-link-type="uri" xlink:href="https://github.com/terra-workflows/synthetic-seq-data">Synthetic Sequence Data Project</ext-link>, and the OBF&#x2019;s<ext-link ext-link-type="uri" xlink:href="https://github.com/OBF/obf-docs/pull/44">affiliated project policy</ext-link>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31830916">31830916</a> <b>(10)</b> uap: reproducible and robust HTS data analysis<ul><li> bcbio-nextgen [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], Bpipe [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>], or <b>nextflow</b> [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> Based on the comparison of tools in Additional file&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref>: Table S1, the tools most dedicated to reproducible research and providing the most similar set of features compared to uap are Galaxy and <b>Nextflow</b>.<li><b>Nextflow</b> is a powerful WMS dedicated to scalability and reproducibility [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> <b>Nextflow</b> and uap share several concepts, e.g.<li> uap therefore provides a means for volatilization of intermediate results without breaking dependencies in the DAG - a feature which does not seem to be available in <b>Nextflow</b>.<li> Logging is somewhat limited in <b>Nextflow</b> compared to uap, but <b>Nextflow</b> provides a broader support for HPC environments including also support for cloud computing.<li> <b>Nextflow</b>&#x2019;s approach to reproducibility is powerful when software from Github is used, as it enables the user to request a specific commit, or when the tools used are publicly available as a container.<li> Where uap and <b>Nextflow</b> differ most clearly regarding reproducibility is linking data and code, as to our understanding based on publications and the online documentation this is not available in <b>Nextflow</b>.<li> [16](          <person-group person-group-type="author">            <name>              <surname>Guimera</surname>              <given-names>RV</given-names>            </name>          </person-group>          <article-title>bcbio-nextgen: Automated, distributed next-gen sequencing pipeline</article-title>          <source>EMBnet J</source>          <year>2012</year>          <volume>17</volume>          <fpage>30</fpage>          <pub-id pub-id-type="doi">10.14806/ej.17.B.286</pub-id>        ) [17](          <person-group person-group-type="author">            <name>              <surname>Sadedin</surname>              <given-names>SP</given-names>            </name>            <name>              <surname>Pope</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Oshlack</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>Bpipe: a tool for running and managing bioinformatics pipelines,</article-title>          <source>Bioinformatics (Oxford Engl)</source>          <year>2012</year>          <volume>28</volume>          <issue>11</issue>          <fpage>1525</fpage>          <lpage>6</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts167</pub-id>        ) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31823712">31823712</a> <b>(4)</b> Vertical and horizontal integration of multi-omics data with miodin<ul><li> This has been addressed by <b>Nextflow</b> [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] and related software, which are used to construct workflows and support Docker technology [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>] for deployment.<li> Scalability is supported by translating a miodin workflow into a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] script, which can be deployed as a job on high-performance computer clusters.<li> This is enabled by simply setting deployAsJob&#x2009;=&#x2009;TRUE when calling execute on the workflow, provided that <b>Nextflow</b> is installed (see the user manual for details).<li> <b>Nextflow</b> supports Docker, so the user only needs to specify the name of the image with jobContainer when executing the workflow.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [30](          <person-group person-group-type="author">            <name>              <surname>Merkel</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Docker: lightweight linux containers for consistent development and deployment</article-title>          <source>Linux J</source>          <year>2014</year>          <volume>2014</volume>          <issue>239</issue>          <fpage>2</fpage>        ) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31813948">31813948</a> <b>(1)</b> Current RNA-seq methodology reporting limits reproducibility<ul><li> In this view, we advocate the use of workflow management systems, such as <b>Nextflow</b> [<xref rid="ref22" ref-type="bibr"><u>22</u></xref>] and Snakemake [<xref rid="ref23" ref-type="bibr"><u>23</u></xref>].<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31800587">31800587</a> <b>(2)</b> Enhanced translation expands the endo-lysosome size and promotes antigen presentation during phagocyte activation<ul><li> The resulting RNAseq reads were processed using the <b>nextflow</b> RNAseq pipeline (version 1.3; <ext-link ext-link-type="uri" xlink:href="https://nf-co.re/">https://nf-co.re/</ext-link>) using default settings.<li> Within the <b>nextflow</b> pipeline, high quality of sequencing reads was assured using fastQC (<ext-link ext-link-type="uri" xlink:href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">http://www.bioinformatics.babraham.ac.uk/projects/fastqc</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31800587">31800587</a> <b>(7)</b> Enhanced translation expands the endo-lysosome size and promotes antigen presentation during phagocyte activation<ul><li><b>Nextflow</b>.<li>                <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/<b>nextflow</b>-io/<b>nextflow</b>" ext-link-type="uri">https://github.com/<b>nextflow</b>-io/<b>nextflow</b></ext-link>              .<li>To ensure reproducible data analysis, all genomic analyses were performed using pipelines generated in the <b>Nextflow</b> workflow management system framework (<xref rid="bib26" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> Each <b>Nextflow</b> pipeline used in this study is briefly described below (<xref rid="supp5" ref-type="supplementary-material"><u>Supplementary file 5</u></xref>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31792410">31792410</a> <b>(1)</b> Large multiple sequence alignments with a root-to-leaf regressive method<ul><li> The computational pipeline (<italic>c.f.</italic><xref ref-type="fn" rid="FN2"><u>Code Availability</u></xref>) was implemented in the <b>Nextflow</b> language<sup><xref rid="R25" ref-type="bibr"><u>25</u></xref></sup> and was deployed in a containerized form using Singularity.<li> [Code Availability](<a href="http://www.tcoffee.org">http://www.tcoffee.org</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055708">32055708</a> <b>(1)</b> Antibiotic resistance prediction for<italic>Mycobacterium tuberculosis</italic> from genome sequence data with Mykrobe<ul><li>The benchmarking pipeline was implemented using <b>Nextflow</b><sup><xref rid="ref-48" ref-type="bibr"><u>48</u></xref></sup>, Singularity<sup><xref rid="ref-49" ref-type="bibr"><u>49</u></xref></sup> and Python scripts, and is freely available under the MIT licence (code available here:<ext-link ext-link-type="uri" xlink:href="https://github.com/iqbal-lab-org/tb-amr-benchmarking">https://github.com/iqbal-lab-org/tb-amr-benchmarking</ext-link> ; commit version used was 3e768c025ee54f154116dc6b2f7100a844081609).<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31772845">31772845</a> <b>(7)</b> From command-line bioinformatics to bioGUI<ul><li> In addition, workflow specification languages such as the common workflow language (CWL) or <b>Nextflow</b> exist.<li><b>Nextflow</b> and DolphinNext.<li>The combination of <b>Nextflow</b> <uri xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</uri> and DolphinNext <uri xlink:href="https://github.com/UMMS-Biocore/dolphinnext">https://github.com/UMMS-Biocore/dolphinnext</uri> is similar to Galaxy or Yabi.<li> While <b>Nextflow</b> is a DSL for describing general workflows (lacking a GUI definition), DolphinNext provides the web-based user interface (UI) which enables a convenient usage of <b>Nextflow</b> workflows.<li> <b>Nextflow</b> requires a POSIX system architecture and may or may not run on Microsoft Windows using <xref rid="ref-9" ref-type="bibr"><u>Cygwin (2019)</u></xref>.<li><italic>bioGUI</italic> is not a classical workflow system like Galaxy, CWL or DolphinNext with <b>Nextflow</b>.<li> [Cygwin (2019)](          <person-group person-group-type="author">            <collab>              <institution>Cygwin</institution>            </collab>          </person-group>          <article-title>Cygwin</article-title>          <year>2019</year>          <uri xlink:href="https://www.cygwin.com/">https://www.cygwin.com/</uri>          <date-in-citation content-type="access-date" iso-8601-date="2019-09-22">22 September 2019</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31722416">31722416</a> <b>(2)</b> MEGARes 2.0: a database for classification of antimicrobial drug, biocide and metal resistance determinants in metagenomic sequence data<ul><li> AMR++ 1.0 was implemented via <b>nextflow</b> scripting language (<xref rid="B41" ref-type="bibr"><u>41</u></xref>) and made publicly available via Galaxy (<xref rid="B42" ref-type="bibr"><u>42</u></xref>), an easy-to-use web-based platform.<li>In addition, we improved the usability of AMR++ 2.0 on non-AWS computing clusters by providing the configuration file templates that <b>nextflow</b> uses to interface with SLURM Workload Manager, a common job scheduler.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31703553">31703553</a> <b>(1)</b> PyGMQL: scalable data extraction and analysis for heterogeneous genomic datasets<ul><li> This has brought to the development of very successful workflow management software like Galaxy [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>], Snakemake [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>], <b>Nextflow</b> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] and FireCloud [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] (evolving to a new system named Terra, in May 2019).<li> [13](          <person-group person-group-type="author">            <name>              <surname>Afgan</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Baker</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Van den Beek</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Blankenberg</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Bouvier</surname>              <given-names>D</given-names>            </name>            <name>              <surname>&#x10C;ech</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Chilton</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Clements</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Coraor</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Eberhard</surname>              <given-names>C</given-names>            </name>            <etal/>          </person-group>          <article-title>The galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2016 update</article-title>          <source>Nucleic Acids Res</source>          <year>2016</year>          <volume>44</volume>          <issue>W1</issue>          <fpage>3</fpage>          <lpage>10</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkw343</pub-id>        ) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](Birger C, Hanna M, Salinas E, Neff J, Saksena G, Livitz D, Rosebrock D, Stewart C, Leshchiner I, Baumann A, Voet D, Cibulskis K, Banks E, Philippakis A, Getz G. Firecloud, a scalable cloud-based platform for collaborative genome analysis: Strategies for reducing and controlling costs. bioRxiv. 2017. 10.1101/209494. <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/early/2017/11/03/209494.full.pdf">https://www.biorxiv.org/content/early/2017/11/03/209494.full.pdf</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31704961">31704961</a> <b>(2)</b> Rapid, multiplexed, whole genome and plasmid sequencing of foodborne pathogens using long-read nanopore technology<ul><li>The full <b>NextFlow</b> workflow, Conda environment configuration, and other associated code used in the analyses are publicly-available on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/jvolkening/minion_bacterial">https://github.com/jvolkening/minion_bacterial</ext-link>).<li>The MinION sequencing data was assembled using a custom <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR39"><u>39</u></xref></sup> workflow that utilized publicly-available tools.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31699036">31699036</a> <b>(1)</b> Hecaton: reliably detecting copy number variation in plant genomes using short read sequencing data<ul><li> To this end, Hecaton is run with a single command using the <b>Nextflow</b> [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>] framework, which provides a unified method to chain together and parallelize the different processes that are executed.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31675414">31675414</a> <b>(2)</b> Sharing interoperable workflow provenance: A review of best practices and their practical application in CWLProv<ul><li> For instance, <b>Nextflow</b> [<xref rid="bib111" ref-type="bibr"><u>112</u></xref>] may fulfill <italic>R18-resource-use</italic> at Provenance Level 2 because it can produce trace reports with hardware resource usage per task execution [<xref rid="bib112" ref-type="bibr"><u>113</u></xref>], but not for the overall workflow.<li> While a <b>Nextflow</b> trace report is a separate CSV file with implementation-specific columns, our planned <italic>R18-resource-use</italic> approach for CWL is to combine <italic>CWL-metrics</italic> [<xref rid="bib113" ref-type="bibr"><u>114</u></xref>], permalinks, and the standard <italic>GFD.204</italic> [<xref rid="bib114" ref-type="bibr"><u>115</u></xref>] to further relate resource use with Level 1 and Level 2 provenance within the CWLProv RO.<li> [112](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [113](<collab>Centre for Genomic Regulation (CRG)</collab>, <source>Nextflow: Tracing and visualization</source>. <year>2018</year><ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">https://www.nextflow.io/docs/latest/tracing.html#trace-report</ext-link>. Acces<comment>sed 28 November 2018.</comment>) [114](<person-group person-group-type="author"><name name-style="western"><surname>Ohta</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Tanjo</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Ogasawara</surname><given-names>O</given-names></name></person-group><article-title>Accumulating computational resource usage of genomic data analysis workflow to optimize cloud computing instance selection</article-title>. <source>Gigascience</source>. <year>2019</year>;<volume>8</volume>(<issue>4</issue>), <pub-id pub-id-type="doi">10.1093/gigascience/giz052</pub-id>.) [115](<person-group person-group-type="author"><name name-style="western"><surname>Cristofori</surname><given-names>A</given-names></name>, <name name-style="western"><surname>Bologna</surname><given-names>I</given-names></name>, <name name-style="western"><surname>Gordon</surname><given-names>J</given-names></name>, <etal>et al</etal>.</person-group><article-title>Usage record&#x2013;format recommendation</article-title>. In: <source>Open Grid Forum</source>. <comment>Document GFD-R-P.204</comment><year>2013</year>, <ext-link ext-link-type="uri" xlink:href="https://www.ogf.org/documents/GFD.204.pdf">https://www.ogf.org/documents/GFD.204.pdf</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31672996">31672996</a> <b>(1)</b> The Personal Genome Project-UK, an open access resource of human multi-omics data<ul><li>All of the RNA-seq samples were processed with a modified version of the <b>nextflow</b><sup><xref ref-type="bibr" rid="CR18"><u>18</u></xref></sup> nf-core RNA-seq pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/UCL-BLIC/rnaseq">https://github.com/UCL-BLIC/rnaseq</ext-link>).<li> [18](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <etal/>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31665515">31665515</a> <b>(2)</b> Expression Atlas update: from tissues to single cells<ul><li> This pipeline was originally implemented in iRAP (<ext-link ext-link-type="doi" xlink:href="10.1101/005991">https://doi.org/10.1101/005991</ext-link>), but has recently been re-implemented using <b>Nextflow</b> (<xref rid="B11" ref-type="bibr"><u>11</u></xref>).<li> We currently employ <b>Nextflow</b> at the quantification stage, and Galaxy (<xref rid="B19" ref-type="bibr"><u>19</u></xref>) for downstream analysis in a reproducible way.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31647547">31647547</a> <b>(1)</b> PhyloMagnet: fast and accurate screening of short-read meta-omics data using gene-centric phylogenetics<ul><li> It is written as a <b>Nextflow</b> (<xref rid="btz799-B15" ref-type="bibr">Di Tommaso <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33902704">33902704</a> <b>(2)</b> The impact of sequencing depth on the inferred taxonomic composition and AMR gene content of metagenomic samples<ul><li>Implementation into a <b>Nextflow</b> pipeline.<li>The entire workflow (both taxonomic and AMR gene profiling) has been implemented into a <b>Nextflow</b> [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>] pipeline complying with POSIX standards, written in Python: ResPipe (<ext-link ext-link-type="uri" xlink:href="https://gitlab.com/hsgweon/ResPipe">https://gitlab.com/hsgweon/ResPipe</ext-link>).<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31622330">31622330</a> <b>(1)</b> Patient Dossier: Healthcare queries over distributed resources<ul><li> Some popular projects in workflow community are <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/">https://www.<b>nextflow</b>.io/</ext-link>), Galaxy, common workflow language (CWL), and workflow definition language (WDL).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31615395">31615395</a> <b>(1)</b> <italic>doepipeline</italic>: a systematic approach to optimizing multi-level and multi-step data processing workflows<ul><li> Given the recent development and consolidation of workflow managing systems [<xref ref-type="bibr" rid="CR44"><u>44</u></xref>] it would be possible to integrate <italic>doepipeline</italic> with for example SnakeMake [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>] or <b>NextFlow</b> [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>], similar to other implementations [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>, <xref ref-type="bibr" rid="CR48"><u>48</u></xref>].<li> [44](Karim MR, Michel A, Zappa A, Baranov P, Sahay R, Rebholz-Schuhmann D. Improving data workflow systems with cloud services and use of open data for bioinformatics research. Brief Bioinform [Internet]. 2018 [cited 2019 Jun 20];19(5):1035&#x2013;1050. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bib/article/19/5/1035/3737318">https://academic.oup.com/bib/article/19/5/1035/3737318</ext-link>) [45](Koster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics [Internet]. 2012 [cited 2019 Jun 20];28(19):2520&#x2013;2522. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts480">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts480</ext-link>) [46](Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol [Internet]. 2017 [cited 2019 Jun 20];35(4):316&#x2013;319. Available from: <ext-link ext-link-type="uri" xlink:href="http://www.nature.com/articles/nbt.3820">http://www.nature.com/articles/nbt.3820</ext-link>) [47](Holl S, Mohammed Y, Zimmermann O, Palmblad M. Scientific workflow optimization for improved peptide and protein identification. BMC Bioinformatics [Internet]. 2015 [cited 2019 Jun 20];16(1):284. Available from: <ext-link ext-link-type="uri" xlink:href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0714-x">http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0714-x</ext-link>) [48](Palmblad M, Lamprecht A-L, Ison J, Schw&#xE4;mmle V. Automated workflow composition in mass spectrometry-based proteomics. Wren J, editor. Bioinformatics [Internet]. 2019 [cited 2019 Jun 20];35(4):656&#x2013;664. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bioinformatics/article/35/4/656/5060940">https://academic.oup.com/bioinformatics/article/35/4/656/5060940</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31632265">31632265</a> <b>(1)</b> Free Water in White Matter Differentiates MCI and AD From Control Subjects<ul><li> All processing was done using a <b>Nextflow</b> (Tommaso et al., <xref rid="B50" ref-type="bibr"><u>2017</u></xref>) pipeline with all software dependencies bundled in a Singularity container (Kurtzer et al., <xref rid="B24" ref-type="bibr"><u>2017</u></xref>) ensuring quick and easy reproducibility of the results.<li> [2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31506435">31506435</a> <b>(1)</b> BioExcel Building Blocks, a software library for interoperable biomolecular simulation workflows<ul><li> <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup>, PyCOMPSs<sup><xref ref-type="bibr" rid="CR18"><u>18</u></xref></sup>, Snakemake)<sup><xref ref-type="bibr" rid="CR19"><u>19</u></xref></sup>.<li> [17](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <etal/>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <fpage>316</fpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [18](          <person-group person-group-type="author">            <name>              <surname>Tejedor</surname>              <given-names>E</given-names>            </name>            <etal/>          </person-group>          <article-title>PyCOMPSs: Parallel computational workflows in Python</article-title>          <source>The International Journal of High Performance Computing Applications</source>          <year>2015</year>          <volume>31</volume>          <fpage>66</fpage>          <lpage>82</lpage>          <pub-id pub-id-type="doi">10.1177/1094342015594678</pub-id>        ) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31455220">31455220</a> <b>(1)</b> Single Cell Explorer, collaboration-driven tools to leverage large-scale single cell RNA-seq data<ul><li> Alternatively, raw data can be processed using Bash or <b>Nextflow</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31431620">31431620</a> <b>(3)</b> Integrative and comparative genomic analyses identify clinically&#xA0;relevant pulmonary carcinoid&#xA0;groups and unveil the supra-carcinoids<ul><li> This workflow is based on the <b>nextflow</b> domain-specific language<sup><xref ref-type="bibr" rid="CR47"><u>47</u></xref></sup> and consists of three steps: reads mapping (software bwa version 0.7.12-r1044)<sup><xref ref-type="bibr" rid="CR48"><u>48</u></xref></sup>, duplicate marking (software samblaster, version 0.1.22)<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup>, and reads sorting (software sambamba, version 0.5.9)<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>.<li>The 210 raw reads files (89 carcinoids, 69 LCNEC, 52 SCLC) were processed in three steps using the RNA-seq processing workflow based on the <b>nextflow</b> language<sup><xref ref-type="bibr" rid="CR47"><u>47</u></xref></sup> and accessible at <ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/RNAseq-nf">https://github.com/IARCbioinfo/RNAseq-nf</ext-link> (revision da7240d).<li> Gene fragments per kilobase million (FPKM) of all genes from the gencode gene annotation file were computed using software StringTie (v1.3.3b)<sup><xref ref-type="bibr" rid="CR57"><u>57</u></xref></sup> in single pass mode (no new transcript discovery), using the protocols from Pertea et al.<sup><xref ref-type="bibr" rid="CR57"><u>57</u></xref></sup> (<b>nextflow</b> pipeline accessible at <ext-link ext-link-type="uri" xlink:href="https://github.com/IARCbioinfo/RNAseq-transcript-nf">https://github.com/IARCbioinfo/RNAseq-transcript-nf</ext-link>; revision c5d114e42d).<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19451168">19451168</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24812344">24812344</a>) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25697820">25697820</a>) [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [57](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27560171">27560171</a>) [57](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27560171">27560171</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32382696">32382696</a> <b>(8)</b> Organizing and running bioinformatics hackathons within Africa: The H3ABioNet cloud computing experience<ul><li>CWL and<break/><b>Nextflow</b>.<li><b>Nextflow</b>.<li> These were containerization technology such as<ext-link ext-link-type="uri" xlink:href="https://www.docker.com/">Docker</ext-link>, and the writing of genomic pipelines in popularly used workflow languages and newly emerging community-standards like <b>Nextflow</b><sup><xref rid="ref-12" ref-type="bibr"><u>12</u></xref></sup> and the Common Workflow Language (CWL)<sup><xref rid="ref-13" ref-type="bibr"><u>13</u></xref></sup>, respectively.<li> While expertise for <b>Nextflow</b> already existed within the network, two collaborators from outside Africa were interested to join the project given their expertise in cloud environments, containerization of code<sup><xref rid="ref-14" ref-type="bibr"><u>14</u></xref></sup> and developing CWL<sup><xref rid="ref-13" ref-type="bibr"><u>13</u></xref></sup>.<li> Streams A and B decided to use CWL for their pipeline development, whereas Streams C and D opted to use <b>Nextflow</b> due to their prior experience using <b>Nextflow</b>.<li>To maximize the learning experience, upon selection, participants were given prerequisite tutorials and materials (Github, <b>Nextflow</b>, CWL, Docker and the SOPs) to go through.<li>The initial day of the H3ABioNet Cloud Computing hackathon was dedicated to introductions, expectations by the participants and practical tutorials covering the use of CWL, <b>Nextflow</b> and creation of Docker containers to ensure all participants had the same basic level of knowledge.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [13](<person-group person-group-type="author"><name><surname>Peter</surname><given-names>A</given-names></name><name><surname>Crusoe</surname><given-names>MR</given-names></name><name><surname>Neboj&#x161;a</surname><given-names>T</given-names></name><etal/></person-group>:<article-title>Common Workflow Language, v1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28344774">28344774</a>) [13](<person-group person-group-type="author"><name><surname>Peter</surname><given-names>A</given-names></name><name><surname>Crusoe</surname><given-names>MR</given-names></name><name><surname>Neboj&#x161;a</surname><given-names>T</given-names></name><etal/></person-group>:<article-title>Common Workflow Language, v1.0</article-title>.<year>2016</year><pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31375735">31375735</a> <b>(2)</b> Assessing genetic diversity and similarity of 435 KPC-carrying plasmids<ul><li>Schematic illustration of the workflow for the clustering and annotation of the KPC-plasmids in <b>Nextflow</b>.<li>The whole workflow is illustrated in Fig.&#xA0;<xref rid="Fig3" ref-type="fig"><u>3</u></xref>, written in <b>Nextflow</b><sup><xref ref-type="bibr" rid="CR63"><u>63</u></xref></sup> (<ext-link ext-link-type="uri" xlink:href="http://www.<b>nextflow</b>.io">www.<b>nextflow</b>.io</ext-link>) and stored at <ext-link ext-link-type="uri" xlink:href="https://github.com/replication/plasmid_analysis">https://github.com/replication/plasmid_analysis</ext-link>.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31328031">31328031</a> <b>(2)</b> BioShake: a Haskell EDSL for bioinformatics workflows<ul><li>High level feature comparison of BioShake with other execution engines (Toil, Cromwell), specification languages (WDL, CWL), and DSLs (<b>NextFlow</b>, Snakemake).<li><b>NextFlow</b></th>                <th rowspan="1" colspan="1">Toil</th>                <th rowspan="1" colspan="1">Cromwell</th>                <th rowspan="1" colspan="1">WDL</th>                <th rowspan="1" colspan="1">CWL</th>                <th rowspan="1" colspan="1">BioShake</th>              </tr>            </thead>            <tbody>              <tr>                <td rowspan="1" colspan="1">DSL.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31287816">31287816</a> <b>(3)</b> Managing genomic variant calling workflows with Swift/T<ul><li> Coding and debugging can require a more substantial effort than say, <b>Nextflow</b> [<xref rid="pone.0211608.ref067" ref-type="bibr"><u>67</u></xref>], and that can be a barrier for biologists.<li><b>NextFlow</b> [<xref rid="pone.0211608.ref067" ref-type="bibr"><u>67</u></xref>]</th>                  <th align="center" rowspan="1" colspan="1">Galaxy [<xref rid="pone.0211608.ref069" ref-type="bibr"><u>69</u></xref>]</th>                  <th align="center" rowspan="1" colspan="1">Kepler [<xref rid="pone.0211608.ref070" ref-type="bibr"><u>70</u></xref>]</th>                </tr>              </thead>              <tbody>                <tr>                  <td align="center" rowspan="1" colspan="1">Nature.<li> a user-produced XML file in Pegasus [<xref rid="pone.0211608.ref088" ref-type="bibr"><u>88</u></xref>]), language-specific syntax (BigDataScript [<xref rid="pone.0211608.ref089" ref-type="bibr"><u>89</u></xref>]), automatic construction of workflow graphs (Swift [<xref rid="pone.0211608.ref016" ref-type="bibr"><u>16</u></xref>], WDL [<xref rid="pone.0211608.ref014" ref-type="bibr"><u>14</u></xref>], <b>Nextflow</b> [<xref rid="pone.0211608.ref067" ref-type="bibr"><u>67</u></xref>]), rule-based approaches (Ruffus [<xref rid="pone.0211608.ref090" ref-type="bibr"><u>90</u></xref>] and bpipe [<xref rid="pone.0211608.ref074" ref-type="bibr"><u>74</u></xref>]) or implicit conventions, while abstracting away from HPC cluster management concerns (Job Management System [<xref rid="pone.0211608.ref091" ref-type="bibr"><u>91</u></xref>]);.<li> [67](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <name><surname>Palumbo</surname><given-names>E</given-names></name>, <name><surname>Notredame</surname><given-names>C</given-names></name>. <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source>. <year>2017</year>;<volume>35</volume>(<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>319</lpage>. <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [67](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <name><surname>Palumbo</surname><given-names>E</given-names></name>, <name><surname>Notredame</surname><given-names>C</given-names></name>. <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source>. <year>2017</year>;<volume>35</volume>(<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>319</lpage>. <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [70](Altintas I, Berkley C, Jaeger E, Jones M, Ludascher B, Mock S. Kepler: an extensible system for design and execution of scientific workflows. In: Scientific and Statistical Database Management, 2004. Proceedings. 16th International Conference on. IEEE; 2004. p. 423&#x2013;424.) [88](<name><surname>Deelman</surname><given-names>E</given-names></name>, <name><surname>Vahi</surname><given-names>K</given-names></name>, <name><surname>Juve</surname><given-names>G</given-names></name>, <name><surname>Rynge</surname><given-names>M</given-names></name>, <name><surname>Callaghan</surname><given-names>S</given-names></name>, <name><surname>Maechling</surname><given-names>PJ</given-names></name>, <etal>et al</etal><article-title>Pegasus: a Workflow Management System for Science Automation</article-title>. <source>Future Generation Computer Systems</source>. <year>2015</year>;<volume>46</volume>:<fpage>17</fpage>&#x2013;<lpage>35</lpage>. <pub-id pub-id-type="doi">10.1016/j.future.2014.10.008</pub-id>) [89](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a>) [16](<name><surname>Wilde</surname><given-names>M</given-names></name>, <name><surname>Hategan</surname><given-names>M</given-names></name>, <name><surname>Wozniak</surname><given-names>JM</given-names></name>, <name><surname>Clifford</surname><given-names>B</given-names></name>, <name><surname>Katz</surname><given-names>DS</given-names></name>, <name><surname>Foster</surname><given-names>I</given-names></name>. <article-title>Swift: A language for distributed parallel scripting</article-title>. <source>Parallel Computing</source>. <year>2011</year>;<volume>37</volume>(<issue>9</issue>):<fpage>633</fpage>&#x2013;<lpage>652</lpage>. <pub-id pub-id-type="doi">10.1016/j.parco.2011.05.005</pub-id>) [14](<name><surname>Voss</surname><given-names>K</given-names></name>, <name><surname>Gentry</surname><given-names>J</given-names></name>, <name><surname>der Auwera</surname><given-names>GV</given-names></name>, <name><surname>Voss</surname><given-names>K</given-names></name>, <name><surname>Gentry</surname><given-names>J</given-names></name>, <name><surname>Van der Auwera</surname><given-names>G</given-names></name>. <article-title>Full-stack genomics pipelining with GATK4 + WDL + Cromwell</article-title>. <source>F1000Research</source>. <year>2017</year>;<volume>6</volume>.) [67](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <name><surname>Palumbo</surname><given-names>E</given-names></name>, <name><surname>Notredame</surname><given-names>C</given-names></name>. <article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source>. <year>2017</year>;<volume>35</volume>(<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>319</lpage>. <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [90](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [91](<name><surname>Brown</surname><given-names>DK</given-names></name>, <name><surname>Penkler</surname><given-names>DL</given-names></name>, <name><surname>Musyoka</surname><given-names>TM</given-names></name>, <name><surname>Bishop</surname><given-names>OT</given-names></name>. <article-title>JMS: An Open Source Workflow Management System and Web-Based Cluster Front-End for High Performance Computing</article-title>. <source>PLOS ONE</source>. <year>2015</year>;<volume>10</volume>(<issue>8</issue>):<fpage>1</fpage>&#x2013;<lpage>25</lpage>. <pub-id pub-id-type="doi">10.1371/journal.pone.0134273</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31269897">31269897</a> <b>(2)</b> iMAP: an integrated bioinformatics and visualization pipeline for microbiome data analysis<ul><li> The pre-release version of iMAP described in this manuscript (iMAP v1.0) is at the preliminary phase, and perhaps it lacks significant reproducibility aspects compared to the modern bioinformatics workflow management systems such as <b>Nextflow</b> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], <b>Nextflow</b>Workbench [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], or Snakemake [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>].<li> [19](Di Tommaso P, Chatzou M, Prieto P, Palumbo E, Notredame C. Nextflow: A tool for deploying reproducible computational pipelines [version 1; not peer reviewed]. F1000Research 2015, 4(ISCB Comm J):430 (poster) (10.7490/f1000research.1110183.1).) [20](          <person-group person-group-type="author">            <name>              <surname>Thomer</surname>              <given-names>AK</given-names>            </name>            <name>              <surname>Twidale</surname>              <given-names>MB</given-names>            </name>            <name>              <surname>Guo</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Yoder</surname>              <given-names>MJ</given-names>            </name>          </person-group>          <article-title>NextflowWorkbench: reproducible and reusable workflows for beginners and experts</article-title>          <source>Conference on human factors in computing systems - proceedings</source>          <year>2016</year>        ) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31316552">31316552</a> <b>(15)</b> Pipeliner: A <b>Nextflow</b>-Based Framework for the Definition of Sequencing Data Processing Pipelines<ul><li> It uses <b>Nextflow</b>, a portable, scalable, and parallelizable domain-specific language, to define data workflows (<xref rid="B4" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> Using <b>Nextflow</b>, each pipeline is modularized, consisting of a configuration file as well as a series of processes.<li> <b>Nextflow</b> processes are connected through channels&#x2014;asynchronous first in, first out queues&#x2014;which allow data to be passed between the different steps in each pipeline using a dataflow programming model.<li>The Pipeliner framework employs reusable template processes strung together <italic>via</italic> <b>Nextflow</b>&#x2019;s scripting language to create workflows in addition to developer tools such as toy datasets and testing modules.<li> An abstraction layer between <b>Nextflow</b> and Pipeliner logic enables platform independence and seamless compatibility with high-performance cloud computing executors such as Amazon Web Services.<li> If a pipeline can inherit all of its processes with predefined templates, the user is only required to link these processes<italic> via </italic><b>Nextflow</b>&#x2019;s scripting language and create a basic configuration file.<li>The Pipeliner framework requires <b>Nextflow</b> and Anaconda.<li> <b>Nextflow</b> requires Java 8 (or higher) to be installed and can be used on Linux and OS X machines.<li>The final step is to download a <b>Nextflow</b> executable package in the same directory as the available pipelines.<li>The final step is to download a <b>Nextflow</b> executable package in the same directory as the available pipelines.</p>        <preformat>$ cd pipeliner/pipelines$ curl -s https://get.<b>nextflow</b>.io | bash.<li>With the setup complete, any of the available pipelines can be executed with their respective toy datasets with the following commands.</p>        <preformat>$./<b>nextflow</b> rnaseq.nf -c rnaseq.config$./<b>nextflow</b> scrnaseq.nf -c scrnaseq.config$./<b>nextflow</b> dge.nf -c dge.config.<li>Together with <b>Nextflow</b> and Anaconda, Pipeliner enables users to process large and complex sequencing datasets with pipelines that are customizable, reproducible, and extensible.<li>Pipeliner is implemented in <b>Nextflow</b>, Python, R, and Bash and released under a General Public License 3.0 license.<li> We will continue to develop the Pipeliner framework as the <b>Nextflow</b> programming language matures, and we plan to provide additional pipelines for other types of sequencing data and analysis workflows in the future.<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31226873">31226873</a> <b>(1)</b> Influence of Socio-Economic and Psychosocial Profiles on the Human Breast Milk Bacteriome of South African Women<ul><li> <b>Nextflow</b> [<xref rid="B30-nutrients-11-01390" ref-type="bibr"><u>30</u></xref>] was used to loop the entire processing workflow.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31185495">31185495</a> <b>(3)</b> GenPipes: an open-source framework for distributed and scalable genomic analyses<ul><li> In the second type, WMSs such as Snakemake [<xref rid="bib6" ref-type="bibr"><u>6</u></xref>], <b>Nextflow</b> [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], BPipe [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], and BigDataScript [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>] and declarative workflow description languages such as CWL or WDL are dedicated to providing a customizable framework to build bioinformatics pipelines.<li> It is important to note that GenPipes, as well as several other WMSs, like <b>Nextflow</b> [<xref rid="bib64" ref-type="bibr"><u>65</u></xref>] and SnakeMake [<xref rid="bib65" ref-type="bibr"><u>66</u></xref>], support community-developed pipelines; however, those have not been included in the comparison.<li><b>Nextflow</b>.<li> [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a>) [65](<comment>NextFlow Community Pipelines. <ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core">https://github.com/nf-core</ext-link></comment>. Accessed Septmeber 2018.) [66](<comment>SnakeMake Community Pipelines. <ext-link ext-link-type="uri" xlink:href="https://github.com/snakemake-workflows">https://github.com/snakemake-workflows</ext-link></comment>. Accessed Septmeber 2018.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31159881">31159881</a> <b>(1)</b> NG-meta-profiler: fast processing of metagenomes using NGLess, a domain-specific language<ul><li> The classical Make tool, which was originally designed for compiling software, but has been used as the basis of a scientific pipeline system [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], uses its own internal language, as do the more modern pipelining tools Snakemake [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>] and <b>nextflow</b> [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>].<li> [39](          <person-group person-group-type="author">            <name>              <surname>Schwab</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Karrenbach</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Claerbout</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Making scientific computations reproducible</article-title>          <source>Comput Sci Eng</source>          <year>2000</year>          <volume>2</volume>          <fpage>61</fpage>          <lpage>67</lpage>          <pub-id pub-id-type="doi">10.1109/5992.881708</pub-id>        ) [40](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31134269">31134269</a> <b>(1)</b> snakePipes: facilitating flexible, scalable and integrative epigenomic analysis<ul><li> Computational frameworks, such as Galaxy (<xref rid="btz436-B8" ref-type="bibr">Goecks <italic>et al.</italic>, 2010</xref>), <b>Nextflow</b> (<xref rid="btz436-B4" ref-type="bibr">Di Tommaso <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31122257">31122257</a> <b>(6)</b> Exploiting orthology and de novo transcriptome assembly to refine target sequence information<ul><li>Schematic overview of the <b>Nextflow</b> sequence refinement pipeline, the a&amp;o-tool.<li>To automate the workflow described above it was implemented in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>].<li> Apart from BinPacker, BLAST+ and MUSCLE, the <b>Nextflow</b> script contains bash and Python code.<li> <xref rid="Fig2" ref-type="fig"><u>2</u></xref>), called a&amp;o-tool, in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>].<li>The <b>Nextflow</b> pipeline is available.<li><bold>Programming language:</bold> <b>Nextflow</b>.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31026367">31026367</a> <b>(3)</b> eDiVA&#x2014;Classification and prioritization of pathogenic variants for clinical diagnostics<ul><li> eDiVA can be run using the <b>NextFlow</b> (Di Tommaso et al., <xref rid="humu23772-bib-0010" ref-type="ref"><u>2017</u></xref>) pipeline management system to ensure its compatibility with most standalone or cloud&#x2010;computing platforms as well as to guarantee reproducibility on any system.<li> eDiVA is available as an open&#x2010;source repository, <ext-link ext-link-type="uri" xlink:href="https://github.com/mbosio85/ediva">https://github.com/mbosio85/ediva</ext-link>, with a Docker container composition wrapped within a <b>NextFlow</b> (Di Tommaso et al., <xref rid="humu23772-bib-0010" ref-type="ref"><u>2017</u></xref>) interface to guarantee exact reproducibility on the most common computing platforms (including several cloud platforms) and as a freely accessible web server: <ext-link ext-link-type="uri" xlink:href="http://www.eDiVA.crg.eu">http://www.eDiVA.crg.eu</ext-link>.<li> Integration with <b>NextFlow</b>, moreover, guarantees reproducibility of results at later time points and on most computing platforms.<li> [2017](<string-name><surname>Di Tommaso</surname>, <given-names>P.</given-names></string-name>, <string-name><surname>Chatzou</surname>, <given-names>M.</given-names></string-name>, <string-name><surname>Floden</surname>, <given-names>E. W.</given-names></string-name>, <string-name><surname>Barja</surname>, <given-names>P. P.</given-names></string-name>, <string-name><surname>Palumbo</surname>, <given-names>E.</given-names></string-name>, &amp; <string-name><surname>Notredame</surname>, <given-names>C.</given-names></string-name> (<year>2017</year>). <article-title>Nextflow enables reproducible computational workflows</article-title>. <source xml:lang="en">Nature Biotechnology</source>, <volume>35</volume>, <fpage>316</fpage>&#x2013;<lpage>319</lpage>.) [2017](<string-name><surname>Di Tommaso</surname>, <given-names>P.</given-names></string-name>, <string-name><surname>Chatzou</surname>, <given-names>M.</given-names></string-name>, <string-name><surname>Floden</surname>, <given-names>E. W.</given-names></string-name>, <string-name><surname>Barja</surname>, <given-names>P. P.</given-names></string-name>, <string-name><surname>Palumbo</surname>, <given-names>E.</given-names></string-name>, &amp; <string-name><surname>Notredame</surname>, <given-names>C.</given-names></string-name> (<year>2017</year>). <article-title>Nextflow enables reproducible computational workflows</article-title>. <source xml:lang="en">Nature Biotechnology</source>, <volume>35</volume>, <fpage>316</fpage>&#x2013;<lpage>319</lpage>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31193391">31193391</a> <b>(3)</b> Computational Methods for Mapping, Assembly and Quantification for Coding and Non-coding Transcripts<ul><li> Snakemake (<ext-link ext-link-type="uri" xlink:href="https://snakemake.readthedocs.io/en/stable/" id="ir0025">https://snakemake.readthedocs.io/en/stable/</ext-link>), <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>nextflow</b>-io" id="ir0030">https://github.com/<b>nextflow</b>-io</ext-link>) and Docker (<ext-link ext-link-type="uri" xlink:href="https://github.com/ngs-docs/2015-nov-docker/" id="ir0035">https://github.com/ngs-docs/2015-nov-docker/</ext-link>)are different platforms with pipelines for reproducible transcriptome analysis.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31069047">31069047</a> <b>(1)</b> META-pipe cloud setup and execution<ul><li> Several pipeline managers support containers, including <b>Nextflow</b><sup><xref rid="ref-9" ref-type="bibr"><u>9</u></xref></sup>, Toil,<ext-link ext-link-type="uri" xlink:href="http://www.pachyderm.io/">Pachyderm</ext-link>, and<ext-link ext-link-type="uri" xlink:href="https://github.com/spotify/luigi">Luigi</ext-link><sup><xref rid="ref-10" ref-type="bibr"><u>10</u></xref></sup>.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28163975">28163975</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a> <b>(4)</b> SciPipe: A workflow library for agile development of complex and dynamic bioinformatics pipelines<ul><li> Text-based tools such as Snakemake&#xA0;[<xref rid="bib33" ref-type="bibr"><u>9</u></xref>], <b>Nextflow</b>&#xA0;[<xref rid="bib17" ref-type="bibr"><u>10</u></xref>], BPipe&#xA0;[<xref rid="bib59" ref-type="bibr"><u>11</u></xref>], Cuneiform&#xA0;[<xref rid="bib7" ref-type="bibr"><u>12</u></xref>], and Pachyderm&#xA0;[<xref rid="bib55" ref-type="bibr"><u>13</u></xref>], on the other hand, are implemented as domain-specific languages (DSLs), which can often provide a higher level of flexibility, at the expense of the ease of use of a graphical user interface.<li> Text-based tools implemented as DSLs, such as Snakemake, <b>Nextflow</b>, BPipe, Pachyderm, and Cuneiform, do not have this limitation but have other characteristics that might be problematic for for complex workflows.<li><b>Nextflow</b> has push-based scheduling and supports dynamic scheduling via the dataflow paradigm and does not suffer from this limitation.<li> Similarly to <b>Nextflow</b>, SciPipe leverages the dataflow paradigm to achieve dynamic scheduling of tasks based on input data, allowing many workflow constructs not easily coded in many other tools.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al</etal>.</person-group><article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source>. <year>2017</year>;<volume>35</volume>(<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [12](<person-group person-group-type="author"><name name-style="western"><surname>Brandt</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Bux</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Leser</surname><given-names>U</given-names></name></person-group><article-title>Cuneiform: A functional language for large scale scientific data analysis</article-title>. In: <source>Proceedings of the Workshops of the EDBT/ICDT, Brussels, Belgium, 2015</source>. <fpage>17</fpage>&#x2013;<lpage>26</lpage>.) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30101309">30101309</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31222200">31222200</a> <b>(11)</b> Bioportainer Workbench: a versatile and user-friendly system that integrates implementation, management, and use of bioinformatics resources in Docker environments<ul><li> Other projects, such as Snakemake [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>], Common Workflow Language [<xref rid="bib16" ref-type="bibr"><u>16</u></xref>], and <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] have provided frameworks for the implementation, administration, and execution of complex pipelines and workflows within Docker containers, allowing simultaneous management and coordinated launching of the various software involved in such analyses.<li> The Dockerfile of the BioPortainer Pipeline Runner, on the other hand, has been developed independently and presents a greater level of complexity in its structure, as it carries all the software and library prerequisites necessary for the execution of <b>NextFlow</b> scripts [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] in a Docker-in-Docker environment.<li>The BioPortainer Pipeline Runner is accessed through the BioPortainer Panel main menu (see Fig.&#xA0;S3 in the manual), providing users with a graphical interface that allows the download, through a GIT protocol, of bioinformatics pipelines available at the <b>NextFlow</b> and NF-Core repositories.<li> Both repositories are developed by collaborative projects that use <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] (a type of domain-specific language) to develop and adapt scalable and reproducible scientific workflows using software containers.<li> Overall, &gt;80 high-quality bioinformatics pipelines are available from the <b>NextFlow</b> and NF-Core repositories.<li> Once activated, the BioPortainer Pipeline Runner GUI provides different user options through its menu, which allow users to download, execute, monitor, update, and delete <b>NextFlow</b>-based pipelines from the abovementioned repositories, or harnessed from alternative Github sources (details regarding the options to run the BioPortainer Pipeline Runner can be found in the BioPortainer Workbench User Manual).<li>The incorporation of <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] as the main framework for the Bioportainer Pipeline Runner was facilitated by its easy integration with the Docker project.<li> Moreover, the <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] community is extremely active, which should help to expand the number and scope of available pipelines over the next years.<li> By using such resources, the BioPortainer Pipeline Runner becomes an extremely flexible tool, allowing the construction and execution of complex pipelines using libraries and tools developed in both <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] and Python languages.<li>To demonstrate the full functionality of the BioPortainer Pipeline Runner, we ran 2 pipelines obtained from the <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] project home page.<li> Thus, <b>NextFlow</b> [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] executions based on the Singularity container system [<xref rid="bib45" ref-type="bibr"><u>47</u></xref>] cannot be performed through this version of the software.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<person-group person-group-type="author"><name name-style="western"><surname>Amstutz</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Crusoe</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <etal>et al</etal>.</person-group><comment>Common Workflow Language, v1.0.</comment><year>2016</year><ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/.</ext-link><comment>Accessed 19 June 2018.</comment>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [47](<person-group person-group-type="author"><name name-style="western"><surname>Kurtzer</surname><given-names>GM</given-names></name>, <name name-style="western"><surname>Sochat</surname><given-names>V</given-names></name>, <name name-style="western"><surname>Bauer</surname><given-names>MW</given-names></name></person-group><article-title>Singularity: Scientific containers for mobility of compute</article-title>. <source>PLOS One</source>. <year>2017</year>;<volume>12</volume>(<issue>5</issue>):<fpage>1</fpage>&#x2013;<lpage>20</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31222199">31222199</a> <b>(1)</b> Accumulating computational resource usage of genomic data analysis workflow to optimize cloud computing instance selection<ul><li>Other workflow operation frameworks exist that are able to capture runtime metrics, including Galaxy [<xref rid="bib18" ref-type="bibr"><u>25</u></xref>], Toil [<xref rid="bib19" ref-type="bibr"><u>26</u></xref>], Cromwell [<xref rid="bib20" ref-type="bibr"><u>27</u></xref>], and <b>Nextflow</b> [<xref rid="bib21" ref-type="bibr"><u>28</u></xref>].<li> [25](<person-group person-group-type="author"><name name-style="western"><surname>Tyryshkina</surname><given-names>A</given-names></name>, <name name-style="western"><surname>Coraor</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Nekrutenko</surname><given-names>A</given-names></name></person-group><article-title>Predicting runtimes of bioinformatics tools based on historical data: Five years of Galaxy usage</article-title>. <source>Bioinformatics</source>. <year>2019</year>, doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/btz054</pub-id>.) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [27](<person-group person-group-type="author"><name name-style="western"><surname>Voss</surname><given-names>K</given-names></name>, <name name-style="western"><surname>Van&#xA0;der&#xA0;Auwera</surname><given-names>G</given-names></name>, <name name-style="western"><surname>Gentry</surname><given-names>J</given-names></name></person-group><article-title>Full-stack genomics pipelining with GATK4 + WDL + Cromwell [version 1; not peer reviewed]</article-title>. <source>F1000Res</source>. <year>2017</year>;<volume>6</volume>(<issue>ISCB Comm J</issue>):<fpage>1381</fpage>.) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30958262">30958262</a> <b>(1)</b> In-depth human plasma proteome analysis captures tissue proteins and transfer of protein variants across the placenta<ul><li> Spectra were then searched by MSGF+ (<xref rid="bib32" ref-type="bibr"><u>Kim and Pevzner, 2014</u></xref>), and post processed with Percolator (<xref rid="bib26" ref-type="bibr"><u>K&#xE4;ll et al., 2007</u></xref>) in a <b>Nextflow</b> pipeline (<xref rid="bib16" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), using a concatenated target-decoy strategy.<li> [Kim and Pevzner, 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25358478">25358478</a>) [K&#xE4;ll et al., 2007](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=17952086">17952086</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31543945">31543945</a> <b>(1)</b> Recommendations for the packaging and containerizing of bioinformatics software<ul><li> Workflow environments such as <b>Nextflow</b> and Galaxy will execute 1 (task, process) per tool, each tool being one container.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31543945">31543945</a> <b>(1)</b> Recommendations for the packaging and containerizing of bioinformatics software<ul><li> The widespread use of packages and containers has been made possible through the use of workflow managers such as <b>Nextflow</b> and Galaxy that wrap the standard commands into Docker or Singularity commands, mount the required volumes in the container or even build the Conda environment at run time.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31543945">31543945</a> <b>(1)</b> Recommendations for the packaging and containerizing of bioinformatics software<ul><li> Workflow tools (<b>nextflow</b>, cwl based or others) will execute 1 task per tool, each tool being one container.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30859831">30859831</a> <b>(2)</b> Spectral Clustering Improves Label-Free Quantificationof Low-Abundant Proteins<ul><li>The complete workflow used to generate the spectralcounting based data is freely available as a <b>nextflow</b> (<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://www.<b>nextflow</b>.io">http://www.<b>nextflow</b>.io</uri>) workflowat <uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/bigbio/nf-workflows">https://github.com/bigbio/nf-workflows</uri>.<li> The <b>nextflow</b> (<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://www.<b>nextflow</b>.io">https://www.<b>nextflow</b>.io</uri>) workflows containing the completepipeline are available at <uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/bigbio/nf-workflows">https://github.com/bigbio/nf-workflows</uri>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30851093">30851093</a> <b>(1)</b> Interoperable and scalable data analysis with microservices: applications in metabolomics<ul><li> In fact, our microservice architecture supports other major workflow engines such as <b>Nextflow</b> (<xref rid="btz160-B13" ref-type="bibr">Di Tommaso <italic>et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30811390">30811390</a> <b>(2)</b> Script of Scripts: A pragmatic workflow system for daily computational research<ul><li>To evaluate SoS in terms of workflow features with leading bioinformatics workflow systems, we created an online table [<xref rid="pcbi.1006843.ref025" ref-type="bibr"><u>25</u></xref>] that compares SoS with Snakemake [<xref rid="pcbi.1006843.ref015" ref-type="bibr"><u>15</u></xref>], <b>Nextflow</b> [<xref rid="pcbi.1006843.ref014" ref-type="bibr"><u>14</u></xref>], Galaxy [<xref rid="pcbi.1006843.ref008" ref-type="bibr"><u>8</u></xref>], CWL [<xref rid="pcbi.1006843.ref009" ref-type="bibr"><u>9</u></xref>] and Bpipe [<xref rid="pcbi.1006843.ref022" ref-type="bibr"><u>22</u></xref>].<li>The table compare SoS with several popular bioinformatics workflow systems including <b>Nextflow</b>, Snakemake, Bpipe, CWL, and Galaxy, in three broad aspects: 1) basic features (syntax, file format, user interface, etc), 2) workflow features (workflow specification, dependency handling, execution and monitoring, etc), and built-in support for external tools and services (container support, HPC systems, distributed systems and cloud support).<li> [25](Peng B, Wang G. How does SoS compare with other workflow engines 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://vatlab.github.io/blog/post/comparison/">https://vatlab.github.io/blog/post/comparison/</ext-link>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [9](Amstutz P, Crusoe MR, Tijani&#x107; N, Chapman B, Chilton J, Heuer M, et al. Common Workflow Language, v1.0. Specification, Common Workflow Language working group. 2016.) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30796295">30796295</a> <b>(1)</b> Effects of a <italic>Saccharomyces cerevisiae</italic> fermentation product on liver abscesses, fecal microbiome, and resistome in feedlot cattle raised without antibiotics<ul><li> The pipeline was executed using <b>Nextflow</b> (v.0.26.0)<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>.<li> [52](Tommaso, P. D. <italic>et al</italic>. Nextflow enables reproducible computational workflows. <italic>Nature Biotechnology</italic>, 10.1038/nbt.3820 (2017).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30697489">30697489</a> <b>(1)</b> Prioritizing bona fide bacterial small RNAs with machine learning classifiers<ul><li> sRNACharP is implemented in <b>Nextflow</b> (<xref rid="ref-13" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and available at <ext-link ext-link-type="uri" xlink:href="https://github.com/BioinformaticsLabAtMUN/sRNACharP">https://github.com/BioinformaticsLabAtMUN/sRNACharP</ext-link>.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> <b>(17)</b> Scalable Workflows and Reproducible Data Analysis for Genomics<ul><li>In this chapter, we show how to create a bioinformatics pipeline using four workflow systems: CWL, GWL, Snakemake, and <b>Nextflow</b>.<li>A third workflow language and orchestrator, <b>Nextflow</b>, allows scalable and reproducible scientific workflows to run seamlessly across multiple platforms from local computers to HPC clusters and the cloud, offering a concise and expressive DSL to describe complex workflows.<li> <b>Nextflow</b> is routinely used in organizations and institutes, such as the Roche Sequencing, the Wellcome Trust Sanger Institute, and the Center for Genomic Regulation (CRG) <ext-link xlink:href="https://www.crg.eu/en/event/coursescrg-<b>nextflow</b>-reproducible-silico-genomics" ext-link-type="uri"><b>Nextflow</b> workshop</ext-link>.<li><b>Nextflow</b>.<li><b>Nextflow</b> [<xref rid="R43" ref-type="bibr"><u>43</u></xref>] is a framework and an orchestration tool that enables scalable and reproducible scientific workflows using software containers (<ext-link xlink:href="https://www.<b>nextflow</b>.io/" ext-link-type="uri">https://www.<b>nextflow</b>.io/</ext-link>).<li>A <b>Nextflow</b> pipeline is described as a series of processes, where each process can be written in any language that can be executed or interpreted on Unix-like operating systems (e.g., Bash, Perl, Ruby, Python, etc.).<li> A key component of <b>Nextflow</b> is the dataflow programming model, which is a message-based abstraction for parallel programming similar to the CSP paradigm (<italic toggle="yes">see</italic> [<xref rid="R23" ref-type="bibr"><u>23</u></xref>]).<li> But <b>Nextflow</b> can handle this data flow in a parallel and asynchronous manner, so a process can operate on multiple inputs and emit multiple outputs at the same time.<li> To allow workflow portability, <b>Nextflow</b> supports multiple container technologies such as Docker and Singularity and integrates natively with Git and popular code sharing platforms, such as GitHub.<li> <b>Nextflow</b> allows executing workflows across different computing platforms by supporting several cluster schedulers (e.g., SLURM, PBS, LSF and SGE) and allowing direct execution on the Amazon cloud (AWS), using services, such as AWS Batch or automating the creation of a compute cluster in the cloud for the user.<li>To run the <b>Nextflow</b> example, you need to have Java 8 and a Docker engine (1.10 or higher) installed.<li> Next install <b>Nextflow</b> with:.<li> Next install <b>Nextflow</b> with:</p>        <preformat preformat-type="computer code" position="float" xml:space="preserve">curl -s https://get.<b>nextflow</b>.io | bash.<li>Run the example from the repository tree:</p>        <preformat preformat-type="computer code" position="float" xml:space="preserve">./<b>nextflow</b> run <b>Nextflow</b>/workflow.nf -with-docker evolutionar-ygenomics/scalability.<li> To run the example on a compute cluster or in the cloud, it is sufficient to specify a different executor (e.g., sge or awsbatch) in the <b>Nextflow</b> configuration file and ensure that those environments are configured to properly work with the Docker container.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](          <person-group person-group-type="author"><name><surname>Welch</surname><given-names>P</given-names></name><name><surname>Aldous</surname><given-names>J</given-names></name><name><surname>Foster</surname><given-names>J</given-names></name></person-group>          <article-title>Csp networking for java (jcsp. net)</article-title>          <source>Comput Sci</source>          <year>2002</year>          <volume>2002</volume>          <fpage>695</fpage>          <lpage>708</lpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30543621">30543621</a> <b>(1)</b> Full-Length Envelope Analyzer (<monospace>FLEA</monospace>): A tool for longitudinal analysis of viral amplicons<ul><li> The logic of <monospace>FLEA</monospace> is implemented in <monospace><b>Nextflow</b></monospace> [<xref rid="pcbi.1006498.ref040" ref-type="bibr"><u>40</u></xref>], a workflow framework for deploying parallel pipelines to clusters and clouds.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30486782">30486782</a> <b>(38)</b> Developing reproducible bioinformatics analysis workflows for heterogeneous computing environments to support African genomics<ul><li>There are various workflow systems and languages such as <b>Nextflow</b>, and open community standards for expressing workflows such as Common Workflow Language (CWL) [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>].<li><b>Nextflow</b>.<li><b>Nextflow</b> [<xref ref-type="bibr" rid="CR12"><u>12</u></xref>] is a workflow language and system developed at the Centre for Genome Regulation in Barcelona.<li> <b>Nextflow</b> supports: execution of workflows, with partial resumption; containerisation with Docker and Singularity; and multiple execution modes including local execution, execution on clusters, Amazon EC2 (cloud-init and AWS Batch), Kubernetes, and OpenStack.<li> The <b>Nextflow</b> workflows are highly portable, e.g.<li>distributed across a compute cluster (again assuming all underlying application software has been installed): the <b>Nextflow</b> program submits the necessary jobs to the cluster job manager on the scientist&#x2019;s behalf, considering dependencies between tasks; the <b>Nextflow</b> monitoring process itself will run on the head node;.<li>without any of the bioinformatics software installed on the system, and they are set up using <b>Nextflow</b>&#x2019;s Docker support (locally, on a Docker Swarm or on Amazon EC2).<li>In all cases <b>Nextflow</b> adapts its execution strategy to the environment.<li> Workflows C and D below were implemented using <b>Nextflow</b>.<li> We provide documentation on what bioinformatics software needs to be installed, as well as a set of Docker images so that the workflows can be run without installing any software other than <b>Nextflow</b> and Java.<li> A GWAS data analysis workflow (Workflow C) was therefore implemented using <b>Nextflow</b>.<li> If required, these parameters may be modified in the <b>Nextflow</b>.config file.<li>The workflow was developed using <b>Nextflow</b>, and identifies regions to be imputed on the basis of an input file in PLINK format and produces output in IMPUTE haplotype format (Fig.&#xA0;<xref rid="Fig4" ref-type="fig"><u>4</u></xref>).<li>Both CWL and <b>Nextflow</b> based workflows offer high levels of reproducibility and portability.<li> <b>Nextflow</b> has direct support for Amazon EC2 which allows the dynamic creation of EC2 clusters using <b>Nextflow</b> commands.<li> The workflow setup requires installation of <b>Nextflow</b> (a one line command) and Java 8, and either a Docker engine or the installation of the underlying application software.<li> Community driven projects such as CWL offer a more robust approach towards standardisation of workflow definitions, whereas <b>Nextflow</b> addresses more specialised needs such as the partial resumption of a workflow without the need to restart the whole analysis.<li> The use of containers makes installation of the software <italic>much</italic> easier across a range of systems, and the use of <b>Nextflow</b> as a language also aids portability in allowing execution in a range of execution environments, and most importantly the scalability of execution as <b>Nextflow</b> can transparently map the workflow to the available computational resources.<li> Discussions of the possibility of standardising on one workflow language were had, but there was a desire by different groups to use either CWL and <b>Nextflow</b>.<li> CWL and <b>Nextflow</b> are different in their philosophy and syntax which partly reflects their very different origins.<li> <b>Nextflow</b> comes from one research group which drives the <b>Nextflow</b> language and system development using its community as a sounding board.<li> Our tentative experience was that <b>Nextflow</b> is easier to learn and may fit the needs of smaller and mid-size groups better; however, these findings are based upon subjective experiences and may simply reflect that different programmers have personal preferences beyond objective justification.<li>Four members of the Hackathon group attended a <b>Nextflow</b> Hackathon organised by CRG in Barcelona.<li> As part of their task for the hackathon it was decided to convert the h3abionet16S CWL workflow running on Docker containers to <b>Nextflow</b> using Singularity containers.<li> Since two of the CRG hackathon members were originally part of Stream B that had developed the h3abionet16S CWL workflow, they were able to compare CWL and <b>Nextflow</b> based on their own experience.<li> They felt that it was quicker to develop the h3abionet16S workflow in <b>Nextflow</b>, possibly because they already had a CWL template to work from.<li> The members also consider <b>Nextflow</b> to be the best choice for getting a bioinformatics pipeline up and running quickly, but will consider setting up CWL workflows if the workflows need to be integrated into frameworks such as Galaxy.<li> <b>Nextflow</b>, similar to major workflow systems, integrates seamlessly into common resource managers and provides an advantage of having clearly standardized code blocks that are implicitly intended for parallelisation, and code blocks that are executed sequentially which is common in designing bioinformatics workflows where inputs are dependent on outputs of prior processes.<li> Other nice user features of <b>Nextflow</b> include embedded support for user notification on the progress saving the user from explicitly coding mailx commands.<li> The <b>Nextflow</b> version of the h3abionet16S package is now also available in the GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/h3abionet/h3abionet16S/tree/master/workflows-nxf">https://github.com/h3abionet/h3abionet16S/tree/master/workflows-nxf</ext-link>).<li> <b>Nextflow</b>&#x2019;s error messages are intended for people comfortable with <b>Nextflow</b>.<li> Within these constraints we try to print meaningful error messages, and have requested modification to <b>Nextflow</b> to allow greater control of what errors appear.<li>Project name: H3ABioNetProject home page: <ext-link ext-link-type="uri" xlink:href="https://github.com/h3abionet">https://github.com/h3abionet</ext-link>Operating system(s): Linux/Unix/Mac Programming language: <b>Nextflow</b>, CWLOther requirements: DockerLicense: All code for Workflows A, B, C and D are available under MIT license.<li> [4](Leipzig J. A review of bioinformatic pipeline frameworks. Brief Bioinform. 2017; 18(3):530&#x2013;6. 10.1093/bib/bbw020.) [12](DI Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol. 2017; 35:316&#x2013;9. Nature Publishing Group; Nature Biotechnology, <ext-link ext-link-type="uri" xlink:href="https://www.nature.com/articles/nbt.3820">https://www.nature.com/articles/nbt.3820</ext-link>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30271886">30271886</a> <b>(1)</b> ANIMA: Association network integration for multiscale analysis<ul><li> This is exemplified by <b>Nextflow</b><sup><xref rid="ref-5" ref-type="bibr"><u>5</u></xref></sup>, a workflow management system using Docker.<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a> <b>(1)</b> ANIMA: Association network integration for multiscale analysis<ul><li><b>nextflow</b> (Di Tomasso et al., 2017, Nat Biotech).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30395323">30395323</a> <b>(1)</b> Plasmid ATLAS: plasmid visual analytics and identification in high-throughput sequencing data<ul><li>To create the default pATLAS database from the current NCBI RefSeq release, the user can take advantage of the scripts available in pATLAS github (<ext-link ext-link-type="uri" xlink:href="https://github.com/B-UMMI/pATLAS">https://github.com/B-UMMI/pATLAS</ext-link>) or the <b>nextflow</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/tiagofilipe12/pATLAS-db-creation">https://github.com/tiagofilipe12/pATLAS-db-creation</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30371888">30371888</a> <b>(2)</b> HmtVar: a new resource for human mitochondrial variations and pathogenicity data<ul><li> The web interface was developed using Bootstrap (<ext-link ext-link-type="uri" xlink:href="https://getbootstrap.com/">https://getbootstrap.com/</ext-link>) and the update procedure takes advantage of the <b>Nextflow</b> pipeline management system (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io/index.html">https://www.<b>nextflow</b>.io/index.html</ext-link>).<li> this set of scripts was then aggregated using the <b>Nextflow</b> pipeline manager, which is capable of parallelizing the different retrieval and parsing tasks, speeding up the entire updating process considerably.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30419504">30419504</a> <b>(5)</b> Pergola: Boosting Visualization and Analysis of Longitudinal Data by Unlocking Genomic Analysis Tools<ul><li>To ensure computational reproducibility, we ran Pergola using <b>Nextflow</b> (<xref rid="bib11" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2017</u></xref>), a workflow manager that allows entire pipelines to be deployed using software containers.<li> Containers can also be handled by workflow managers (<xref rid="bib12" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2015</u></xref>) like <b>Nextflow</b> that provide increased scalability through optimized parallelization.<li> Thanks to the integration of these resources within <b>Nextflow</b>, all the analyses shown here can be repeated by entering a single command line (<xref rid="mmc1" ref-type="supplementary-material"><u>Transparent Methods</u></xref>) on an adequately configured computer (i.e., <b>Nextflow</b> and Docker must be installed so as to allow software to be automatically gathered and deployed).<li>We also demonstrate here how high-throughput behavioral analysis can become more systematic and reproducible by adopting workflow managers such as <b>Nextflow</b> (<xref rid="bib11" ref-type="bibr"><u>Di Tommaso et&#xA0;al., 2017</u></xref>).<li> [Di Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Di Tommaso et&#xA0;al., 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26421241">26421241</a>) [Di Tommaso et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30367595">30367595</a> <b>(3)</b> Reproducible bioinformatics project: a community for reproducible bioinformatics analysis pipelines<ul><li> Instead, projects like (<ext-link ext-link-type="uri" xlink:href="https://snakemake.bitbucket.io">https://snakemake.bitbucket.io</ext-link>) or <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="https://www.<b>nextflow</b>.io">https://www.<b>nextflow</b>.io</ext-link>) allow users to create reproducible and scalable data analyses specifying their own pipeline through a powerful metalanguage for workflow specification.<li> snakemake and <b>nextflow</b>, is specifically designed for the R community.<li> service, and (ii) to provide an automatic translation of our workflow specified in R into an equivalent workflow specified in snakemake format or in <b>nextflow</b> format.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30356428">30356428</a> <b>(1)</b> Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification<ul><li> Such a command can be automated in a bash loop using bash variables, or one can use more advanced workflow management systems such as Snakemake<sup><xref rid="ref-34" ref-type="bibr"><u>34</u></xref></sup> or <b>Nextflow</b><sup><xref rid="ref-35" ref-type="bibr"><u>35</u></xref></sup>.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30261842">30261842</a> <b>(7)</b> Real-time analysis of nanopore-based metagenomic sequencing from infected orthopaedic devices<ul><li>The analysis was performed within a <b>Nextflow</b> workflow [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] with the software contained within a Singularity [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] image generated from a Docker repository [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> The second receives the fast5 files and uses a <b>Nextflow</b> workflow that basecalls data to be classified and aligns them to specific reference sequences with results pushed to a database, Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>(<xref rid="Fig1" ref-type="fig"><u>c</u></xref>).<li> <bold>b</bold> Fast5 files are detected and submitted as batches for the <b>Nextflow</b> workflow.<li> <bold>c</bold> <b>Nextflow</b> workflow which is contained within a singularity image and can be distributed across a cluster (SLURM used here) or on a local machine.<li>The entire workflow was run in <b>Nextflow</b> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] with the software contained inside a Singularity [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] image.<li> Each batch analysed within the <b>Nextflow</b> workflow took between four and fifteen minutes to process using a single core, depending on which node the job was submitted to, Additional file <xref rid="MOESM5" ref-type="media"><u>5</u></xref>: Figure S5a.<li><bold>Figure S5.</bold> Batch job duration times in minutes sample report taken from <b>Nextflow</b>.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](          <person-group person-group-type="author">            <name>              <surname>Kurtzer</surname>              <given-names>GM</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Bauer</surname>              <given-names>MW</given-names>            </name>          </person-group>          <article-title>Singularity: scientific containers for mobility of compute</article-title>          <source>PLoS One</source>          <year>2017</year>          <volume>12</volume>          <fpage>1</fpage>          <lpage>20</lpage>          <pub-id pub-id-type="doi">10.1371/journal.pone.0177459</pub-id>        ) [18](What is docker. 2017. <ext-link ext-link-type="uri" xlink:href="https://www.docker.com/what-docker">https://www.docker.com/what-docker</ext-link>. Accessed Nov 2017.) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [17](          <person-group person-group-type="author">            <name>              <surname>Kurtzer</surname>              <given-names>GM</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Bauer</surname>              <given-names>MW</given-names>            </name>          </person-group>          <article-title>Singularity: scientific containers for mobility of compute</article-title>          <source>PLoS One</source>          <year>2017</year>          <volume>12</volume>          <fpage>1</fpage>          <lpage>20</lpage>          <pub-id pub-id-type="doi">10.1371/journal.pone.0177459</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30186700">30186700</a> <b>(2)</b> BioWorkbench: a high-performance framework for managing and analyzing bioinformatics experiments<ul><li> <b>Nextflow</b> (<xref rid="ref-12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) is a framework comprised by a domain-specific language for the development of workflows.<li> <b>Nextflow</b> allows the user to enable the provenance record, or trace report, at the moment of the workflow execution and there is currently an effort to standardize this information to Research Object and W3C PROV.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30101309">30101309</a> <b>(3)</b> Container-based bioinformatics with Pachyderm<ul><li> One appealing option is to use workflow systems such as Snakemake (<xref rid="bty699-B18" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), Galaxy (<xref rid="bty699-B1" ref-type="bibr">Afgan <italic>et al.</italic>, 2016</xref>) or <b>Nextflow</b> (<xref rid="bty699-B11" ref-type="bibr">Di Tommaso <italic>et al.<li> Indeed, containers have been successfully employed in existing workflow systems such as Snakemake and <b>Nextflow</b> for life science applications.<li> Regarding the way workflows are actually specified, Pachyderm pipeline stages are defined in JSON/YAML format that is common within the Kubernetes community, while other workflow tools such as <b>Nextflow</b> implement their own domain specific language.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30013858">30013858</a> <b>(2)</b> <italic>Staphylococcus aureus</italic> viewed from the perspective of 40,000+ genomes<ul><li> StAP (<uri xlink:href="https://doi.org/10.5281/zenodo.1255310">DOI 10.5281/zenodo.1255310</uri>) consisted of custom Python3 scripts and open source software organized by the <b>Nextflow</b> (<xref rid="ref-10" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) (v0.28.2) workflow management platform (<xref ref-type="fig" rid="fig-2">Fig.<li> The pipeline was designed as a series of modules running individual software packages, organized by the <b>Nextflow</b> (<xref rid="ref-10" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) workflow language, which made it possible to run the entire pipeline or individual components as needed.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29917068">29917068</a> <b>(6)</b> YAMP: a containerized workflow enabling reproducibility in metagenomics research<ul><li> YAMP is implemented in <b>Nextflow</b>&#xA0;[<xref rid="bib6" ref-type="bibr"><u>6</u></xref>] and is accompanied by a Docker&#xA0;[<xref rid="bib30" ref-type="bibr"><u>30</u></xref>] and a Singularity&#xA0;[<xref rid="bib31" ref-type="bibr"><u>31</u></xref>] container.<li>YAMP is developed in <b>Nextflow</b>, a workflow management system that allows the effortless development, deployment, and execution of complex distributed computational workflows&#xA0;[<xref rid="bib6" ref-type="bibr"><u>6</u></xref>]; it has been used in several life-science projects (e.g.,[<xref rid="bib32" ref-type="bibr"><u>32</u></xref>&#x2013;<xref rid="bib34" ref-type="bibr"><u>34</u></xref>]).<li> <b>Nextflow</b> allows for user-transparent high-level parallelization and offers out-of-the-box support for distributed computational environments, ensuring the scalability of large projects.<li> In fact, while software versions are described in the Docker/Singularity container, the <b>Nextflow</b> script and configuration file capture all the details needed to fully track each step of data processing, therefore, satisfying the prospective provenance requirements, while the very detailed YAMP log file ensures retrospective provenance.<li> Being based on <b>Nextflow</b>, YAMP runs on any UNIX-like system, provides out-of-the-box support for several job schedulers (e.g., PBS, SGE, SLURM) and for Amazon AWS cloud, and its integration with Docker/Singularity is completely user transparent.<li>YAMP is made available as a <b>Nextflow</b> script that allows a user-friendly execution via the command line.<li> [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [30](<collab>Docker</collab>. <ext-link ext-link-type="uri" xlink:href="https://www.docker.com/">https://www.docker.com/</ext-link>, <comment>Accessed 21st June 2018</comment>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [32](<person-group person-group-type="author"><name name-style="western"><surname>Guzman</surname><given-names>C</given-names></name>, <name name-style="western"><surname>D&#x2019;Orso</surname><given-names>I</given-names></name></person-group><article-title>CIPHER: a flexible and extensive workflow platform for integrative next-generation sequencing data analysis and genomic regulatory element prediction</article-title>. <source>BMC Biology</source>. <year>2017</year>;<volume>18</volume>(<issue>1</issue>):<fpage>363</fpage>.) [34](<person-group person-group-type="author"><name name-style="western"><surname>Sanderson</surname><given-names>ND</given-names></name>, <name name-style="western"><surname>Street</surname><given-names>TL</given-names></name>, <name name-style="western"><surname>Foster</surname><given-names>D</given-names></name><etal/></person-group><article-title>Real-time analysis of nanopore-based metagenomic sequencing from orthopaedic device infection</article-title>. <source>bioRxiv</source>. <year>2017</year>; <fpage>220616</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29720997">29720997</a> <b>(1)</b> Transposable Element Genomic Fissuring in <italic>Pyrenophora teres</italic> Is Associated With Genome Expansion and Dynamics of Host&#x2013;Pathogen Genetic Interactions<ul><li> To ensure comparability of results between assemblies and reproducibility of results more generally, the method was formalized as a reproducible <b>Nextflow</b> pipeline (<xref rid="B25" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29670290">29670290</a> <b>(1)</b> Renewing Felsenstein&#x2019;s Phylogenetic Bootstrap in the Era ofBig Data<ul><li>For the sake of reproducibility, all analyses described in this articlewere implemented in the <b>NextFlow</b> workflow manager<sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup>, and are accessible along with all our data at<ext-link ext-link-type="uri" xlink:href="https://github.com/evolbioinfo/booster-workflows">https://github.com/evolbioinfo/booster-workflows</ext-link>.<li> [42](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <etal/>          </person-group>          <article-title>Nextflow enables reproducible computationalworkflows</article-title>          <source>Nat. Biotech</source>          <volume>35</volume>          <fpage>316</fpage>          <lpage>319</lpage>          <year>2017</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29601575">29601575</a> <b>(1)</b> Discovery of genomic intervals that underlie nematode responses to benzimidazoles<ul><li> The complete RNA-seq pipeline was implemented with <b>Nextflow</b> [<xref rid="pntd.0006368.ref035" ref-type="bibr"><u>35</u></xref>] and is publicly available through GitHub (<ext-link ext-link-type="uri" xlink:href="http://www.github.com/AndersenLab/BZRNA-seq-nf">www.github.com/AndersenLab/BZRNA-seq-nf</ext-link>).<li> [35](Kurs JP, Simi M, Campagne F. NextflowWorkbench: Reproducible and Reusable Workflows for Beginners and Experts. bioRxiv. 2016; p. 041236.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a> <b>(13)</b> Watchdog &#x2013; a workflow management system for the distributed analysis of large-scale experimental data<ul><li> As representative WMSs <italic>Galaxy</italic> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>], <italic>KNIME</italic> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], <italic>Snakemake</italic> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], and <italic><b>Nextflow</b></italic> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>] were chosen.<li> <sup>7</sup>other supported executors: <italic>Watchdog</italic>: new executors can be added with the plugin system, <italic>Galaxy</italic>: PBS/Torque, Open Grid Engine, Univa Grid Engine, Platform LSF, HTCondor, Slurm, Galaxy Pulsar, <italic>Snakemake</italic>: can also use cluster engines with access to a common file system and a submit command that accepts shell scripts as first argument, <italic><b>Nextflow</b></italic>: SGE, LSF, Slurm, PBS/Torque, NQSII, HTCondor, Ignite; <sup>8</sup>non-free extensions for SGE or dedicated server support are available; <sup>9</sup>custom executors for cloud computing services can be created using the plugin system; <sup>10</sup><italic>Watchdog</italic>: HTTP/S, FTP/S and SFTP by default, can be extended to any remote file system with an implementation of the FileProvider interface from the Commons Virtual File System project, <italic>Galaxy</italic>: Object Store plugins for S3, Azure, iRODS, <italic>Snakemake</italic>: S3, GS, SFTP, HTTP, FTP, Dropbox, XRootD, NCBI, WebDAV, GFAL, GridFTP.<li> <italic><b>Nextflow</b></italic>: HTTP/S, FTP, S3; <sup>11</sup>a hard-coded error checker triggered on keywords &#x2018;exception&#x2019; and &#x2018;error&#x2019; in standard output and error is provided; <sup>12</sup>depends on the node implementation and left to developer; <sup>13</sup>explanation: usage of local storage during distributed execution in order to avoid unnecessary load on the shared storage system; <sup>14</sup>direct integration of python code is possible; <sup>15</sup>own scripting language available; <sup>16</sup>explanation: describes the concept used to separate workflow definition and functionality (e.g.<li><b>Nextflow</b>.<li>The idea behind the WMS <italic><b>Nextflow</b></italic> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>] is to use pipes to transfer information from one task to subsequent tasks.<li> In <italic><b>Nextflow</b></italic>, different tasks communicate through channels, which are equivalent to pipes, by using them as input and output.<li> A workflow consists of several tasks, which are denoted as processes and are defined using <italic><b>Nextflow</b></italic>&#x2019;s own language.<li> The commands that are executed by processes can be either bash commands or defined in <italic><b>Nextflow</b></italic>&#x2019;s own scripting language.<li> <italic><b>Nextflow</b></italic> also provides the possibility to apply a task on a set of input files that follow a specific filename pattern using a channel that is filled with the filenames at runtime.<li>By default, all running processes are killed by <italic><b>Nextflow</b></italic> if a single process causes an error.<li>In <italic><b>Nextflow</b></italic>, there is no encapsulation of integrated tools at all since the commands to execute are defined in the file containing the workflow.<li> Furthermore, <italic><b>Nextflow</b></italic> also does not offer a GUI for workflow design, which makes it hard for beginners to create their own workflows as they must be written in <italic><b>Nextflow</b></italic>&#x2019;s own very comprehensive programming language.<li> [8](Taylor J, Schenck I, Blankenberg D, Nekrutenko A. Using galaxy to perform large-scale interactive data analyses. Curr Protoc Bioinforma. 2007. Chapter 10:Unit 10.5.) [9](          <person-group person-group-type="author">            <name>              <surname>Berthold</surname>              <given-names>MR</given-names>            </name>            <name>              <surname>Cebron</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Dill</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Gabriel</surname>              <given-names>TR</given-names>            </name>            <name>              <surname>K&#xF6;tter</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Meinl</surname>              <given-names>T</given-names>            </name>            <etal/>          </person-group>          <article-title>KNIME: The Konstanz Information Miner</article-title>          <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>          <year>2007</year>          <publisher-loc>Heidelberg-Berlin</publisher-loc>          <publisher-name>Springer</publisher-name>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29481550">29481550</a> <b>(2)</b> Cell-type specific sequencing of microRNAs from complex animal tissues<ul><li>The <b>Nextflow</b><xref rid="R66" ref-type="bibr"><u>66</u></xref> pipeline developed for this analysis, as well as the R script used for further analysis (see below) can be found at: <ext-link ext-link-type="uri" xlink:href="https://gitlab.com/tburk/smallRNA-meth">https://gitlab.com/tburk/smallRNA-meth</ext-link>.<li> The <b>Nextflow</b> pipeline and the R scripts used for data analysis can be found at: <ext-link ext-link-type="uri" xlink:href="https://gitlab.com/tburk/smallRNA-meth">https://gitlab.com/tburk/smallRNA-meth</ext-link>.<li> [66](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29568489">29568489</a> <b>(2)</b> Ten steps to get started in Genome Assembly and Annotation<ul><li>Several workflow management systems, such as <b>Nextflow</b>, Toll and Galaxy, have recently been reported as having the capacity to use and deploy containers.<li> Authors should be encouraged to develop their pipelines in commonly used workflow managers (Galaxy<sup><xref rid="ref-69" ref-type="bibr"><u>69</u></xref></sup>, <b>Nextflow</b><sup><xref rid="ref-70" ref-type="bibr"><u>70</u></xref></sup>, Snakemake<sup><xref rid="ref-71" ref-type="bibr"><u>71</u></xref></sup>).<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29352322">29352322</a> <b>(3)</b> qPortal: A platform for data-driven biomedical research<ul><li> Additionally, by creating an interface in the Workflow Handler java library, the gUSE workflow system can be replaced by other workflow systems like <b>Nextflow</b> [<xref rid="pone.0191603.ref029" ref-type="bibr"><u>29</u></xref>].<li> While the current production instances interface with gUSE/WS-PGRADE, the extension of the workflow system with interfaces to SnakeMake and <b>NextFlow</b> is possible.<li> Therefore, users will be able to create SnakeMake or <b>NextFlow</b> workflows and use them through qPortal provided that the corresponding tools are available on the cluster instance.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29342134">29342134</a> <b>(1)</b> Chromosomal instability drives metastasis through a cytosolic DNA response<ul><li>All custom code, statistical analysis, and visualizations were performed in Python or R, and used <b>Nextflow</b> to manage some of the computational pipelines<sup><xref rid="R59" ref-type="bibr"><u>59</u></xref></sup>.<li> [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29272324">29272324</a> <b>(1)</b> Bioinformatics for precision oncology<ul><li> Multiple workflow managers are available such as Snakemake [<xref rid="bbx143-B27" ref-type="bibr"><u>27</u></xref>], <b>Nextflow</b> [<xref rid="bbx143-B28" ref-type="bibr"><u>28</u></xref>], Toil [<xref rid="bbx143-B29" ref-type="bibr"><u>29</u></xref>], Bpipe [<xref rid="bbx143-B30" ref-type="bibr"><u>30</u></xref>] and to some extent also the Galaxy framework [<xref rid="bbx143-B31" ref-type="bibr"><u>31</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29048555">29048555</a> <b>(1)</b> An architecture for genomics analysis in a clinical setting using Galaxy and Docker<ul><li> In addition, workflow management systems (such as Taverna [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], Galaxy [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], SnakeMake [<xref rid="bib13" ref-type="bibr"><u>13</u></xref>], <b>NextFlow</b> [<xref rid="bib14" ref-type="bibr"><u>14</u></xref>]) also ensure the traceability and reproducibility of the analytical process.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](<person-group person-group-type="author"><name name-style="western"><surname>Tommaso</surname><given-names>PD</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Prieto</surname><given-names>P</given-names></name><etal/></person-group><article-title>Nextflow: a tool for deploying reproducible computational pipelines</article-title>. <source>F1000Res</source><year>2015</year><comment><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.7490/f1000research.1110183.1">https://doi.org/10.7490/f1000research.1110183.1</ext-link></comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28875074">28875074</a> <b>(2)</b> Atropos: specific, sensitive, and speedy trimming of sequencing reads<ul><li>Second, we implemented our benchmark workflow using the <b>Nextflow</b> (<xref rid="ref-6" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) framework.<li> Importantly, <b>Nextflow</b> enables workflows to be run either locally or in most cluster environments, and supports running containerized software via either a Docker or Singularity (<xref rid="ref-15" ref-type="bibr"><u>Kurtzer, 2016</u></xref>) client (depending on the operating system).<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Kurtzer, 2016](          <person-group person-group-type="author">            <name>              <surname>Kurtzer</surname>              <given-names>GM</given-names>            </name>          </person-group>          <article-title>Singularity linux application and environment containers for science</article-title>          <source>Zenodo</source>          <year>2016</year>          <version designator="2.1.2">Version: 2.1.2</version>          <pub-id pub-id-type="doi">10.5281/zenodo.60736</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28789639">28789639</a> <b>(4)</b> CIPHER: a flexible and extensive workflow platform for integrative next-generation sequencing data analysis and genomic regulatory element prediction<ul><li> By combining <b>Nextflow</b> [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>, <xref ref-type="bibr" rid="CR12"><u>12</u></xref>] - a powerful workflow language based on the Unix pipe concept, Docker [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>] - a container-based virtualization technology, open source software and custom scripts, we provide a robust, and powerful toolkit that simplifies NGS data analysis and provides a significant improvement over currently available pipelines in terms of flexibility, speed and ease of use.<li> In contrast to these approaches, CIPHER was designed using <b>Nextflow</b>, a specialized, and new workflow language that is built around the Unix pipe concept [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>].<li> By using <b>Nextflow</b> as the underlying language for the CIPHER platform, we gain access to several useful features, including automatic parallelization, Docker and GitHub support, the capacity to run locally on a desktop or on a cluster, and the ability to seamlessly integrate custom scripts in a variety of programming languages.<li>In comparison, CIPHER only requires the manual installation of <b>Nextflow</b> and Docker, greatly reducing the number of obstacles a new user may encounter during their setup.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [12](Silver, A. 2017. Software simplified. Nature. Available at: <ext-link ext-link-type="uri" xlink:href="http://www.nature.com/news/software-simplified-1.22059?WT.ec_id=NATURE-20170601&amp;spMailingID=54179123&amp;spUserID=NzMxODM3Mjg2MQS2&amp;spJobID=1165210174&amp;spReportId=MTE2NTIxMDE3NAS2">http://www.nature.com/news/software-simplified-1.22059?WT.ec_id=NATURE-20170601&amp;spMailingID=54179123&amp;spUserID=NzMxODM3Mjg2MQS2&amp;spJobID=1165210174&amp;spReportId=MTE2NTIxMDE3NAS2</ext-link>) [13](Anon, 2015. Docker Available at: <ext-link ext-link-type="uri" xlink:href="https://www.docker.com/">https://www.docker.com/</ext-link> [Accessed January 5, 2017].) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28695896">28695896</a> <b>(1)</b> R-Spondin chromosome rearrangements drive Wnt-dependent tumour initiation and maintenance in the intestine<ul><li> We used DESeq2 to perform all differential expression analyses (v1.12.3)<xref ref-type="bibr" rid="b40"><u>40</u></xref>, and utilized <b>Nextflow</b> (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" ext-link-type="uri" xlink:href="http://dx.doi.org/10.6084/m9.figshare.1254958">http://dx.doi.org/10.6084/m9.figshare.1254958</ext-link>) to implement our computational pipelines.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25516281">25516281</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28854616">28854616</a> <b>(1)</b> Bio-Docklets: virtualization containers for single-step execution of NGS pipelines<ul><li> As an alternative, we have also considered lightweight workflow engines such as <b>NextFlow</b> [<xref rid="bib39" ref-type="bibr"><u>39</u></xref>], but settled on Galaxy given that the ToolShed allows us to perform easy installations for some of the tools we included in the pipelines.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28609295">28609295</a> <b>(2)</b> <italic>clubber</italic>: removing the bioinformatics bottleneck in big data analyses<ul><li>Existing workflow frameworks like Galaxy [<xref rid="j_jib-2017-0020_ref_010" ref-type="bibr"><u>10</u></xref>] and <b>Nextflow</b> [<xref rid="j_jib-2017-0020_ref_011" ref-type="bibr"><u>11</u></xref>] allow users to create computational pipelines to process and analyze biological data.<li> <b>Nextflow</b>, on the other hand, has limited data filtering and visualization capabilities.<li> [10](          <source>Galaxy: The Galaxy Project</source>          <year>2017</year>          <comment>Available from: <ext-link ext-link-type="uri" xlink:href="https://galaxyproject.org/">https://galaxyproject.org/</ext-link></comment>          <date-in-citation content-type="access">Accessed on: April 12th, 2017</date-in-citation>        ) [11](          <source>Nextflow: Comparative Bioinformatics group, Barcelona Center for Genomic Regulation (CRG)</source>          <year>2016</year>          <comment>Available from: <ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</ext-link></comment>          <date-in-citation content-type="access">Accessed on: April 12th, 2017</date-in-citation>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28299179">28299179</a> <b>(1)</b> Cluster Flow: A user-friendly bioinformatics workflow tool<ul><li> Many of the popular tools allow the user to create analysis pipelines using specialised domain specific languages (<italic>Snakemake</italic><sup><xref rid="ref-1" ref-type="bibr"><u>1</u></xref></sup>,<italic><b>NextFlow</b></italic><sup><xref rid="ref-2" ref-type="bibr"><u>2</u></xref></sup>,<italic>Bpipe</italic><sup><xref rid="ref-3" ref-type="bibr"><u>3</u></xref></sup>).<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28327945">28327945</a> <b>(1)</b> Enhancing knowledge discovery from cancer genomics data with Galaxy<ul><li> Tools that automate pipeline development such as Kronos [<xref rid="bib1" ref-type="bibr"><u>1</u></xref>], <b>Nextflow</b> [<xref rid="bib2" ref-type="bibr"><u>2</u></xref>], and Snakemake [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>] can satisfy the needs of more technically savvy users.<li> [1](<person-group person-group-type="author"><name name-style="western"><surname>Jafar Taghiyar</surname><given-names>M</given-names></name><etal/></person-group><article-title>Kronos: a workflow assembler for genome analytics and informatics</article-title>. <source>bioRxiv</source><year>2016</year>:<fpage>040352</fpage><comment>doi:10.1101/040352</comment>.) [2](<person-group person-group-type="author"><name name-style="western"><surname>Kurs</surname><given-names>JP</given-names></name>, <name name-style="western"><surname>Simi</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Campagne</surname><given-names>F.</given-names></name></person-group>, <source>Nextflow Workbench Documentation Booklet</source>. <source>Fabien Campagne</source>, <year>2016</year><comment><ext-link ext-link-type="uri" xlink:href="http://campagnelab.org/download/nextflowworkbench-documentation-1-5-1/?wpdmdl=6826">http://campagnelab.org/download/nextflowworkbench-documentation-1-5-1/?wpdmdl=6826</ext-link></comment>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28327935">28327935</a> <b>(2)</b> Science in the cloud (SIC): A use case in MRI connectomics<ul><li><bold>Batch/ECS</bold>, Kubernetes [<xref rid="bib21" ref-type="bibr"><u>21</u></xref>], MyBinder [<xref rid="bib22" ref-type="bibr"><u>22</u></xref>], CBRAIN [<xref rid="bib23" ref-type="bibr"><u>23</u></xref>], <b>Nextflow</b> [<xref rid="bib24" ref-type="bibr"><u>24</u></xref>].<li> Middleware such as Kubernetes [<xref rid="bib21" ref-type="bibr"><u>21</u></xref>], Tutum, or <b>Nextflow</b> [<xref rid="bib24" ref-type="bibr"><u>24</u></xref>] can enable a user to distribute their jobs across a cluster existing in different computing environments (i.e., separate clouds).<li> [21](<person-group person-group-type="author"><name name-style="western"><surname>Brewer</surname><given-names>EA</given-names></name></person-group><article-title>Kubernetes and the path to cloud native</article-title>. In: <source>Proceedings of the Sixth ACM Symposium on Cloud Computing</source>. <publisher-name>ACM</publisher-name>; <year>2015</year>;<fpage>167</fpage>.) [22](<article-title>Binder</article-title>. <comment><ext-link ext-link-type="uri" xlink:href="http://mybinder.org/">http://mybinder.org/</ext-link> (September 10 2016, date last accessed</comment>).) [23](<person-group person-group-type="author"><name name-style="western"><surname>Sherif</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Rioux</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Rousseau</surname><given-names>ME</given-names></name><etal/></person-group><article-title>Cbrain: a web-based, distributed computing platform for collaborative neuroimaging research</article-title>. <source>Front Neuroinform</source><year>2015</year>;<fpage>102</fpage>.) [24](<person-group person-group-type="author"><name name-style="western"><surname>DiTommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Baraja</surname><given-names>PP</given-names></name><etal/></person-group><article-title>A novel tool for highly scalable computational pipelines</article-title>. <year>2014</year>.) [21](<person-group person-group-type="author"><name name-style="western"><surname>Brewer</surname><given-names>EA</given-names></name></person-group><article-title>Kubernetes and the path to cloud native</article-title>. In: <source>Proceedings of the Sixth ACM Symposium on Cloud Computing</source>. <publisher-name>ACM</publisher-name>; <year>2015</year>;<fpage>167</fpage>.) [24](<person-group person-group-type="author"><name name-style="western"><surname>DiTommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Baraja</surname><given-names>PP</given-names></name><etal/></person-group><article-title>A novel tool for highly scalable computational pipelines</article-title>. <year>2014</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28049418">28049418</a> <b>(1)</b> ChimPipe: accurate detection of fusion genes and transcription-induced chimeras from RNA-seq data<ul><li>Despite these advantages, ChimPipe could be improved in at least 2 aspects: (1) it could provide all the chimeric transcripts compatible with the chimeric junction (module for which we already have a tested code) as additional information, (2) it could be made more robust by being reimplemented in a pipeline specific language such as <b>nextflow</b> (<ext-link ext-link-type="uri" xlink:href="http://www.<b>nextflow</b>.io/">http://www.<b>nextflow</b>.io/</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27994937">27994937</a> <b>(1)</b> The growing need for microservices in bioinformatics<ul><li> <b>NextFlow</b>,[<xref rid="ref11" ref-type="bibr"><u>11</u></xref>] the PanCancer Launcher[<xref rid="ref12" ref-type="bibr"><u>12</u></xref>] from the International Cancer Genome Consoritum, and BaseSpace[<xref rid="ref13" ref-type="bibr"><u>13</u></xref>] from Illumina, are a few examples of contemporary tools that take advantage of Docker to help manage associated pipelines.<li> [11](          <source>Nextflow. Io Documentation</source>          <date-in-citation>Last cited on 2016 Aug 12</date-in-citation>          <comment>Available from:<uri xlink:type="simple" xlink:href="https://www.nextflow.io/docs/latest/docker.html">https://www.nextflow.io/docs/latest/docker.html</uri></comment>        ) [12](          <source>Working with PanCancer Data on AWS. International Cancer Genome Consortium</source>          <date-in-citation>Last updated on 2015 Nov 18; Last cited on 2016 Aug 12</date-in-citation>          <comment>Available from:<uri xlink:type="simple" xlink:href="https://www.icgc.org/working-pancancer-data-aws">https://www.icgc.org/working-pancancer-data-aws</uri></comment>        ) [13](          <source>BaseSpace Native App Engine Overview. Illumina, Inc</source>          <date-in-citation>Last cited on 2016 Aug 12</date-in-citation>          <comment>Available from:<uri xlink:type="simple" xlink:href="https://developer.basespace.illumina.com/docs/content/documentation/native-apps/native-app-overview">https://developer.basespace.illumina.com/docs/content/documentation/native-apps/native-app-overview</uri></comment>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27942268">27942268</a> <b>(3)</b> Towards agile large-scale predictive modelling in drug discovery with flow-based programming design principles<ul><li>Snakemake&#xA0;[<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], <b>NextFlow</b>&#xA0;[<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] and BPipe&#xA0;[<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] are text-based tools implemented as Domain Specific Languages (DSL).<li> <b>NextFlow</b>&#x2019;s DSL allows more flexibility due to its dataflow-based implementation, but does not support creating a library of reusable component definitions.<li> Instead, <b>NextFlow</b> requires components to be defined in conjunction with the workflow definition&#xA0;[<xref ref-type="bibr" rid="CR24"><u>24</u></xref>].<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](          <person-group person-group-type="author">            <name>              <surname>Tommaso</surname>              <given-names>PD</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Baraja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>A novel tool for highly scalable computational pipelines</article-title>          <source>Figshare</source>          <year>2014</year>        ) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [24](Discussion on blog post. <ext-link ext-link-type="uri" xlink:href="http://bionics.it/posts/fbp-data-flow-syntax%23comment-2141038801">http://bionics.it/posts/fbp-data-flow-syntax#comment-2141038801</ext-link>. Accessed 18 April 2016) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27781083">27781083</a> <b>(1)</b> The 2016 Bioinformatics Open Source Conference (BOSC)<ul><li> Forty community members worked on projects including improvements to the<ext-link ext-link-type="uri" xlink:href="http://www.commonwl.org/">Common Workflow Language</ext-link>, integration between the<ext-link ext-link-type="uri" xlink:href="https://github.com/bigdatagenomics/adam">ADAM</ext-link> and<ext-link ext-link-type="uri" xlink:href="https://www.<b>Nextflow</b>.io/"><b>Nextflow</b></ext-link> analysis environments, additions to the<ext-link ext-link-type="uri" xlink:href="https://github.com/ewels/MultiQC">MultiQC</ext-link> reporting framework, and enhancements to existing<ext-link ext-link-type="uri" xlink:href="https://www.open-bio.org/">Open Bioinformatics</ext-link> projects including BioPerl, BioJava and Biopython,.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27013646">27013646</a> <b>(3)</b> A review of bioinformatic pipeline frameworks<ul><li> Some implicit frameworks, such as <b>Nextflow</b> (<ext-link ext-link-type="uri" xlink:href="http://<b>nextflow</b>.io">http://<b>nextflow</b>.io</ext-link>), provide tools to abstract and manage filenaming into global variables to reduce ambiguity.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26869916">26869916</a> <b>(1)</b> Using Make for Reproducible and Parallel Neuroimaging Workflow and Quality-Assurance<ul><li> This has driven development of several Make-like systems geared toward specific types of data analysis (e.g., Drake, Factual, <xref rid="B7" ref-type="bibr"><u>2015</u></xref>, <b>Nextflow</b>, Tommaso, <xref rid="B32" ref-type="bibr"><u>2015</u></xref>).<li> [2015](<person-group person-group-type="author"><name><surname>Factual</surname><given-names>I.</given-names></name></person-group> (<year>2015</year>). <source>Drake.</source> Available online at: <ext-link ext-link-type="uri" xlink:href="https://github.com/Factual/drake">https://github.com/Factual/drake</ext-link>) [2015](<person-group person-group-type="author"><name><surname>Tommaso</surname><given-names>P. D.</given-names></name></person-group> (<year>2015</year>). <source>Nextflow.</source><publisher-name>Barcelona Center for Genomic Regulation (CRG); Comparative Bioinformatics group</publisher-name> Available online at: <ext-link ext-link-type="uri" xlink:href="http://www.nextflow.io/">http://www.nextflow.io/</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26421241">26421241</a> <b>(4)</b> The impact of Docker containers on the performance of genomic pipelines<ul><li>All three pipelines are developed with <b>Nextflow</b>, a tool that is designed to simplify the deployment of computational pipelines across different platforms in a reproducible manner (<xref rid="ref-3" ref-type="bibr"><u>Di Tommaso et al., 2014</u></xref>).<li> <b>Nextflow</b> provides built-in support for Docker, allowing pipeline tasks to be executed transparently in Docker containers; this allowed us to execute the same pipeline natively or run it with Docker without having to modify the pipeline code, but by simply specifying the Docker image to be used in the <b>Nextflow</b> configuration file.<li> In this case, a tool like <b>Nextflow</b>, which handles the execution of Docker containers transparently, can greatly simplify the resulting application.<li> [Di Tommaso et al., 2014](          <person-group>            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Baraja</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow: a novel tool for highly scalable computational pipelines</article-title>          <year>2014</year>          <comment>            <italic>Available at <uri xlink:href="http://dx.doi.org/10.6084/m9.figshare.1254958">http://dx.doi.org/10.6084/m9.figshare.1254958</uri></italic>          </comment>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a> <b>(2)</b> BigDataScript: a scripting language for data pipelines<ul><li> (v) <b>NextFlow</b> (<ext-link ext-link-type="uri" xlink:href="http://www.<b>nextflow</b>.io">www.<b>nextflow</b>.io</ext-link>), another Groovy-based DSL, is based on data flow programming paradigm.<li> <b>NextFlow</b>&#x2019;s data-flow programming model or Leaf&#x2019;s pipeline drawings) that can be powerful, but require programming outside the traditional imperative model, and thus might create a steep learning curve.</ul></p>
</body>
