<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Snakemake</title>
</head>
<body>
<h1>Sentences that contain Snakemake</h1>
<p>2023 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36637197">36637197</a> Mapache: a flexible pipeline to map ancient DNA.<ul><li>Mapache is implemented in the workflow manager <b>Snakemake</b> and is optimized for low-space consumption, allowing to efficiently (re)map large data sets-such as reference panels and multiple extracts and libraries per sample-to one or several genomes. </ul></p><p>2023 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36469333">36469333</a> TransFlow: a <b>Snakemake</b> workflow for transmission analysis of Mycobacterium tuberculosis whole-genome sequencing data.<ul><li>TransFlow relies on <b>Snakemake</b> and Conda to resolve dependencies among consecutive processing steps and can be easily adapted to any computation environment.</ul></p><p>2023 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36352329">36352329</a> Whole-genome sequencing of artificial single-nucleotide variants induced by DNA degradation in biological crime scene traces.<ul><li>In a preliminary study, blood samples were stored for up to 120&#xA0;days and whole-genome sequencing was performed using the <b>Snakemake</b> workflow dna-seq-gatk-variant-calling to identify positions that vary between the time point 0 sample and the aged samples. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36340405">36340405</a> <i>Fusarium oxysporum</i> effector clustering version 2: An updated pipeline to infer host range.<ul><li>It is designed as a <b>Snakemake</b> pipeline and incorporates a new interactive visualization app. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36053180">36053180</a> grenepipe: a flexible, scalable and reproducible pipeline to automate variant calling from sequence reads.<ul><li>We developed grenepipe, an all-in-one <b>Snakemake</b> workflow to streamline the data processing from raw high-throughput sequencing data of individuals or populations to genotype variant calls. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36174014">36174014</a> MetaWorks: A flexible, scalable bioinformatic pipeline for high-throughput multi-marker biodiversity assessments.<ul><li>Pipelines are automated using <b>Snakemake</b> to minimize user intervention and facilitate scalability. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35899080">35899080</a> iCOMIC: a graphical interface-driven bioinformatics pipeline for analyzing cancer omics data.<ul><li>Our iCOMIC toolkit pipeline featuring many independent workflows is embedded in the popular <b>Snakemake</b> workflow management system. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35703559">35703559</a> MAGNETO: An Automated Workflow for Genome-Resolved Metagenomics.<ul><li>MAGNETO is implemented as a <b>Snakemake</b> workflow and is available at: https://gitlab.univ-nantes.fr/bird_pipeline_registry/magneto. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35731209">35731209</a> riboCleaner: a pipeline to identify and quantify rRNA read contamination from RNA-seq data in plants.<ul><li>We have implemented riboCleaner as a containerized <b>Snakemake</b> workflow. </ul></p><p>2023 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36636106">36636106</a> Hex Me If You Can.<ul><li>The mesh generation pipeline is defined with the help of <b>Snakemake</b>, a modern workflow management system, which allows us to specify a fully automated, extensible, and sustainable workflow. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35902092">35902092</a> Tourmaline: A containerized workflow for rapid and iterable amplicon sequence analysis using QIIME 2 and <b>Snakemake</b>.<ul><li>We developed Tourmaline, a Python-based workflow that implements QIIME 2 and is built using the <b>Snakemake</b> workflow management system. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35852420">35852420</a> Spacemake: processing and analysis of large-scale spatial transcriptomics data.<ul><li>Here, we present spacemake, a modular, robust, and scalable spatial transcriptomics pipeline built in <b>Snakemake</b> and Python. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304319">36304319</a> Snaq: A Dynamic <b>Snakemake</b> Pipeline for Microbiome Data Analysis With QIIME2.<ul><li>In this article, we describe Snaq, a <b>snakemake</b> pipeline that helps automate and optimize 16S data analysis using QIIME2. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35178872">35178872</a> epiGBS2: Improvements and evaluation of highly multiplexed, epiGBS-based reduced representation bisulfite sequencing.<ul><li>The bioinformatics pipeline was integrated in the <b>Snakemake</b> workflow management system, which makes the pipeline easy to execute and modular, and parameter settings for important computational steps flexible. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35698034">35698034</a> GenErode: a bioinformatics pipeline to investigate genome erosion in endangered and extinct species.<ul><li>GenErode is written in <b>Snakemake</b> and Python3 and uses Conda and Singularity containers to achieve reproducibility on high-performance compute clusters. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35551337">35551337</a> Exodus: sequencing-based pipeline for quantification of pooled variants.<ul><li>Exodus is implemented in Python within a <b>Snakemake</b> framework. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35664802">35664802</a> ePeak: from replicated chromatin profiling data to epigenomic dynamics.<ul><li>We present ePeak, a <b>Snakemake</b>-based pipeline for the identification and quantification of reproducible peaks from raw ChIP-seq, CUT&amp;RUN and CUT&amp;Tag epigenomic profiling techniques. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35496474">35496474</a> A bioinformatic pipeline for simulating viral integration data.<ul><li>Our method [1] takes the form of a <b>snakemake</b> [2] pipeline, consisting of a Python [3] script using the Biopython [4] module that simulates integrations of a viral reference into a host reference. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35711922">35711922</a> Detection and Validation of Circular DNA Fragments Using Nanopore Sequencing.<ul><li>The workflow is implemented based on the <b>Snakemake</b> workflow management system. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35191485">35191485</a> Topological approximate Bayesian computation for parameter inference of an angiogenesis model.<ul><li>All code used to produce our results is available as a <b>Snakemake</b> workflow from github.com/tt104/tabc_angio.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35477344">35477344</a> Findzx: an automated pipeline for detecting and visualising sex chromosomes using whole-genome sequencing data.<ul><li>Here we present findZX, an automated <b>Snakemake</b>-based computational pipeline for detecting and visualizing sex chromosomes through differences in genome coverage and heterozygosity between any number of males and females. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35020798">35020798</a> StainedGlass: interactive visualization of massive tandem repeat structures with identity heatmaps.<ul><li>StainedGlass is implemented using <b>Snakemake</b> and available open source under the MIT license at https://mrvollger.github.io/StainedGlass/.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35330728">35330728</a> MEDUSA: A Pipeline for Sensitive Taxonomic Classification and Flexible Functional Annotation of Metagenomic Shotgun Sequences.<ul><li>These tools are installed via Conda, and the workflow is managed by <b>Snakemake</b>, easing the setup and execution. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35120462">35120462</a> Assembly-free rapid differential gene expression analysis in non-model organisms using DNA-protein alignment.<ul><li>A <b>Snakemake</b> implementation of our workflow is available at: https://bitbucket.org/project_samar/samar .</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35077539">35077539</a> DENTIST-using long reads for closing assembly gaps at high accuracy.<ul><li>DENTIST's source code including a <b>Snakemake</b> workflow, conda package, and Docker container is available at https://github.com/a-ludi/dentist. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35037206">35037206</a> Pipeline Automation via <b>Snakemake</b>.<ul><li>Here, we describe the setup and use of <b>Snakemake</b>, a pipeline automation tool derived from GNU MAKE.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34927666">34927666</a> Identifying cancer pathway dysregulations using differential causal effects.<ul><li>The GitHub repository also contains the <b>Snakemake</b> workflows needed to reproduce all results presented here.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34936221">34936221</a> BUSCO: Assessing Genomic Data Quality and Beyond.<ul><li>Basic Protocol 1: Assessing an input sequence with a BUSCO dataset specified manually Basic Protocol 2: Assessing an input sequence with a dataset automatically selected by BUSCO Basic Protocol 3: Assessing multiple inputs Alternate Protocol: Decreasing analysis runtime when assessing a large number of small genomes with BUSCO auto-lineage workflow and <b>Snakemake</b> Support Protocol 1: BUSCO setup Support Protocol 2: Visualizing BUSCO results Support Protocol 3: Building phylogenomic trees.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34788788">34788788</a> bollito: a flexible pipeline for comprehensive single-cell RNA-seq analyses.<ul><li>bollito is built using the <b>Snakemake</b> workflow management system, which easily connects each execution step and facilitates the reproducibility of results. </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34586393">34586393</a> MuWU: Mutant-seq library analysis and annotation.<ul><li>MuWU is a <b>Snakemake</b> based workflow and freely available at https://github.com/tgstoecker/MuWU.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34553212">34553212</a> Genetic demultiplexing of pooled single-cell RNA-sequencing samples in cancer facilitates effective experimental design.<ul><li>To facilitate similar analyses at the experimental design phase, we provide freely accessible code and a reproducible <b>Snakemake</b> workflow built around the best-performing tools found in our in silico benchmark evaluations, available at https://github.com/lmweber/snp-dmx-cancer.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34567846">34567846</a> VGEA: an RNA viral assembly toolkit.<ul><li>Here we present (<b>V</b>iral <b>G</b>enomes <b>E</b>asily <b>A</b>nalyzed), a <b>Snakemake</b> workflow for analyzing RNA viral genomes. <li>VGEA was built on the <b>Snakemake</b> workflow management system and utilizes existing tools for each step: <b>fastp</b> (Chen et&#xA0;al., 2018) for read trimming and read-level quality control, <b>BWA</b> (Li &amp; Durbin, 2009) for mapping sequencing reads to the human reference genome, <b>SAMtools</b> (Li et&#xA0;al., 2009) for extracting unmapped reads and also for splitting bam files into fastq files, <b>IVA</b> (Hunt et&#xA0;al., 2015) for <i>de novo</i> assembly to generate contigs, <b>shiver</b> (Wymant et&#xA0;al., 2018) to pre-process reads for quality and contamination, then map to a reference tailored to the sample using corrected contigs supplemented with the user's choice of existing reference sequences, <b>SeqKit</b> (Shen et&#xA0;al., 2016) for cleaning shiver assembly for QUAST, <b>QUAST</b> (Gurevich et&#xA0;al., 2013) to evaluate/assess the quality of genome assemblies and <b>MultiQC</b> (Ewels et&#xA0;al., 2016) for aggregation of the results from fastp, BWA and QUAST. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34342226">34342226</a> ppx: Programmatic Access to Proteomics Data Repositories.<ul><li>To demonstrate how ppx enhances reproducible research, we used ppx within a <b>Snakemake</b> workflow to reanalyze a published data set with the open modification search tool ANN-SoLo and compared our reanalysis to the original results. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33165513">33165513</a> Evaluating single-cell cluster stability using the Jaccard similarity index.<ul><li>The R package scclusteval and the accompanying <b>Snakemake</b> workflow implement all steps of the pipeline: subsampling the cells, repeating the clustering with Seurat and estimation of cluster stability using the Jaccard similarity index and providing rich visualizations.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33165513">33165513</a> Evaluating single-cell cluster stability using the Jaccard similarity index.<ul><li>R package scclusteval: https://github.com/crazyhottommy/scclusteval <b>Snakemake</b> workflow: https://github.com/crazyhottommy/pyflow_seuratv3_parameter Tutorial: https://crazyhottommy.github.io/EvaluateSingleCellClustering/.</ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33689914">33689914</a> MutVis: Automated framework for analysis and visualization of mutational signatures in pathogenic bacterial strains.<ul><li>The current framework integrates various publicly available packages using <b>Snakemake</b> workflow management software, Python and R scripting. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34103501">34103501</a> Systematic benchmarking of tools for CpG methylation detection from nanopore sequencing.<ul><li><b>Snakemake</b> pipelines are also provided for reproducibility and to enable the systematic application of our analyses to other datasets.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33926025">33926025</a> Species-Specific Quality Control, Assembly and Contamination Detection in Microbial Isolate Sequences with AQUAMIS.<ul><li>Here, we present AQUAMIS, a <b>Snakemake</b> workflow for an extensive quality control and assembly of raw Illumina sequencing data, allowing laboratories to automatize the initial analysis of their microbial whole-genome sequencing data. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32717040">32717040</a> SCORE: Smart Consensus Of RNA Expression-a consensus tool for detecting differentially expressed genes in bacteria.<ul><li>We present the <b>Snakemake</b>-based tool Smart Consensus Of RNA Expression (SCORE) which uses a consensus approach founded on a selection of well-established tools for differential gene expression analysis. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33810805">33810805</a> Fast lightweight accurate xenograft sorting.<ul><li>It is written in numba-compiled Python and comes with sample <b>Snakemake</b> workflows for hash table construction and dataset processing.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33258916">33258916</a> MOSGA: Modular Open-Source Genome Annotator.<ul><li>MOSGA is built on a portable, customizable and easily extendible <b>Snakemake</b> backend, and thus, can be tailored to a wide range of users and projects.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33736596">33736596</a> WASP: a versatile, web-accessible single cell RNA-Seq processing platform.<ul><li>The processing pipeline is realized as a <b>Snakemake</b> workflow, while an R Shiny application is provided for interactive result visualization. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a> Using prototyping to choose a bioinformatics workflow management system.<ul><li>To select a workflow management system, a rapid survey of available systems was undertaken, and candidates were shortlisted: <b>Snakemake</b>, cwltool, Toil, and Nextflow. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> Sustainable data analysis with <b>Snakemake</b>.<ul><li>We show how the popular workflow management system <b>Snakemake</b> can be used to guarantee this, and how it enables an ergonomic, combined, unified representation of all steps involved in data analysis, ranging from raw data processing, to quality control and fine-grained, interactive exploration and plotting of final results.</ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33252655">33252655</a> Dadasnake, a <b>Snakemake</b> implementation of DADA2 to process amplicon sequencing data for microbial ecology.<ul><li>We present dadasnake, a user-friendly, 1-command <b>Snakemake</b> pipeline that wraps the preprocessing of sequencing reads and the delineation of exact sequence variants by using the favorably benchmarked and widely used DADA2 algorithm with a taxonomic classification and the post-processing of the resultant tables, including hand-off in standard formats. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33252655">33252655</a> Dadasnake, a <b>Snakemake</b> implementation of DADA2 to process amplicon sequencing data for microbial ecology.<ul><li>By use of <b>Snakemake</b>, dadasnake makes efficient use of high-performance computing infrastructures. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33329686">33329686</a> <i>Cascabel</i>: A Scalable and Versatile Amplicon Sequence Data Analysis Pipeline Delivering Reproducible and Documented Results.<ul><li>We have developed <i>Cascabel</i>, a scalable, flexible, and easy-to-use amplicon sequence data analysis pipeline, which uses <b>Snakemake</b> and a combination of existing and newly developed solutions for its computational steps. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33198651">33198651</a> Natrix: a <b>Snakemake</b>-based workflow for processing, clustering, and taxonomically assigning amplicon sequencing reads.<ul><li>The workflow is written using <b>Snakemake</b>, a workflow management engine for developing data analysis workflows. <li>Thus, <b>Snakemake</b> ensures reproducibility and Conda offers version control of the utilized programs. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33194387">33194387</a> The Bacteria Genome Pipeline (BAGEP): an automated, scalable workflow for bacteria genomes with <b>Snakemake</b>.<ul><li>The pipeline was built on the <b>Snakemake</b> framework and utilizes existing tools for each processing step: fastp for quality trimming, snippy for variant calling, Centrifuge for taxonomic classification, Abricate for AMR gene detection, snippy-core for generating whole and core genome alignments, IQ-TREE for phylogenetic tree construction and vcfR for an interactive heatmap visualization which shows SNPs at specific locations across the genomes. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33087062">33087062</a> MetaLAFFA: a flexible, end-to-end, distributed computing-compatible metagenomic functional annotation pipeline.<ul><li>MetaLAFFA is implemented as a <b>Snakemake</b> pipeline, which enables convenient integration with distributed computing clusters, allowing users to take full advantage of available computing resources. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32993491">32993491</a> Hypercluster: a flexible tool for parallelized unsupervised clustering optimization.<ul><li>We present hypercluster, a python package and <b>SnakeMake</b> pipeline for flexible and parallelized clustering evaluation and selection. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33575609">33575609</a> gNOMO: a multi-omics pipeline for integrated host and microbiome analysis of non-model organisms.<ul><li>The pipeline has been developed using the workflow management framework <b>Snakemake</b> in order to obtain an automated and reproducible pipeline. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32703166">32703166</a> CDSnake: <b>Snakemake</b> pipeline for retrieval of annotated OTUs from paired-end reads using CD-HIT utilities.<ul><li>To solve this issue we created CDSnake - <b>Snakemake</b> pipeline utilizing CD-HIT tools for easier consecutive launch of CD-HIT-OTU-Miseq tools for complete processing of paired end reads in metagenomic studies. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32703166">32703166</a> CDSnake: <b>Snakemake</b> pipeline for retrieval of annotated OTUs from paired-end reads using CD-HIT utilities.<ul><li>Inheriting this properties from original CD-HIT-OTU-MiSeq utilities, CDSnake made their usage handier due to simple scalability, easier automated runs and other <b>Snakemake</b> benefits.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32703166">32703166</a> CDSnake: <b>Snakemake</b> pipeline for retrieval of annotated OTUs from paired-end reads using CD-HIT utilities.<ul><li>We developed <b>Snakemake</b> pipeline for OTU-MiSeq utilities, which simplified and automated data analysis. </ul></p><p>2021 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33500774">33500774</a> Reproducibly sampling SARS-CoV-2 genomes across time, geography, and viral diversity.<ul><li>A complete default pipeline is available through a <b>Snakemake</b> workflow, so subsampling can be achieved using a single command. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32571209">32571209</a> ATLAS: a <b>Snakemake</b> workflow for assembly, annotation, and genomic binning of metagenome sequence data.<ul><li>ATLAS is written in Python and the workflow implemented in <b>Snakemake</b>; it operates in a Linux environment, and is compatible with Python 3.5+ and Anaconda 3+ versions. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32571209">32571209</a> ATLAS: a <b>Snakemake</b> workflow for assembly, annotation, and genomic binning of metagenome sequence data.<ul><li>ATLAS provides a user-friendly, modular and customizable <b>Snakemake</b> workflow for metagenome data processing; it is easily installable with conda and maintained as open-source on GitHub at https://github.com/metagenome-atlas/atlas.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32497118">32497118</a> HemoMIPs-Automated analysis and result reporting pipeline for targeted sequencing data.<ul><li>Our <b>Snakemake</b> pipeline performs sample demultiplexing, overlap paired-end merging, alignment, MIP-arm trimming, variant calling, coverage analysis and report generation. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32271863">32271863</a> gplas: a comprehensive tool for plasmid analysis using short-read graphs.<ul><li>Gplas is written in R, Bash and uses a <b>Snakemake</b> pipeline as a workflow management system. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31882996">31882996</a> Bioinformatics pipeline using JUDI: Just Do It!<ul><li><b>Snakemake</b>, Nextflow, Common Workflow Language, Galaxy, etc. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32183729">32183729</a> RASflow: an RNA-Seq analysis workflow with <b>Snakemake</b>.<ul><li>Utilizing the workflow management system <b>Snakemake</b> and the package management system Conda, we have developed a modular, flexible and user-friendly RNA-Seq analysis workflow: RNA-Seq Analysis <b>Snakemake</b> Workflow (RASflow). </ul></p><p>2022 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32082406">32082406</a> Can-Seq: a PCR and DNA sequencing strategy for identifying new alleles of known and candidate genes.<ul><li>Subsequently, mutations in the known and candidate genes were identified using a custom <b>Snakemake</b> script (https://github.com/Carroll-Lab/can_seq), and PCR zygosity tests were then designed and used to identify the individual mutants carrying each mutation. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31928673">31928673</a> Detection of significant SNP associated with production and oil quality traits in interspecific oil palm hybrids using RARSeq.<ul><li>The obtained sequences were processed with a "<b>snakemake</b>" pipeline, filtered and missing values were imputed. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31829904">31829904</a> Generation of a Transcriptional Radiation Exposure Signature in Human Blood Using Long-Read Nanopore Sequencing.<ul><li>The data generated was analyzed using a <b>Snakemake</b> pipeline modified to handle paired samples. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31768265">31768265</a> SNAPPy: A <b>snakemake</b> pipeline for scalable HIV-1 subtyping by phylogenetic pairing.<ul><li>We designed SNAPPy: a <b>snakemake</b> pipeline for scalable HIV-1 subtyping by phylogenetic pairing. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31727118">31727118</a> mPies: a novel metaproteomics tool for the creation of relevant protein databases and automatized protein annotation.<ul><li>mPies is implemented in Python 3/<b>Snakemake</b> and freely available on GitHub: https://github.com/johanneswerner/mPies/. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31718539">31718539</a> PyBDA: a command line tool for automated analysis of big biological data sets.<ul><li>It uses <b>Snakemake</b> in order to automatically schedule jobs to a high-performance computing cluster. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31513641">31513641</a> uORF-Tools-Workflow for the determination of translation-regulatory upstream open reading frames.<ul><li>uORF-Tools is available as a free and open <b>Snakemake</b> workflow at https://github.com/Biochemistry1-FFM/uORF-Tools. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31500660">31500660</a> A comparison of automatic cell identification methods for single-cell RNA sequencing data.<ul><li>Additionally, we provide a <b>Snakemake</b> workflow to facilitate the benchmarking and to support the extension of new methods and new datasets.</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31510660">31510660</a> Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology.<ul><li>A <b>snakemake</b> pipeline to reproduce the benchmark is available at https://github.com/grst/immune_deconvolution_benchmark. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31088905">31088905</a> ARMOR: An <u>A</u>utomated <u>R</u>eproducible <u>MO</u>dular Workflow for Preprocessing and Differential Analysis of <u>R</u>NA-seq Data.<ul><li>ARMOR is implemented using the <b>Snakemake</b> workflow management system and leverages conda environments; Bioconductor objects are generated to facilitate downstream analysis, ensuring seamless integration with many R packages. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31343654">31343654</a> Eye in a Disk: eyeIntegration Human Pan-Eye and Body Transcriptome Database Version 1.0.<ul><li>The 916 eye and 1375 GTEx samples were sent into a <b>Snakemake</b>-based reproducible pipeline we wrote to quantify all known transcripts and genes, removes samples with poor sequence quality and mislabels, normalizes expression values across each tissue, perform 882 differential expression tests, calculate GO term enrichment, and output all as a single SQLite database file: the Eye in a Disk (EiaD) dataset. </ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30786165">30786165</a> Integrating Bacterial ChIP-seq and RNA-seq Data With SnakeChunks.<ul><li>The protocols here describe a modular system that enable a user to compose and fine-tune workflows based on SnakeChunks, a library of rules for the <b>Snakemake</b> workflow engine. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30902113">30902113</a> Sunbeam: an extensible pipeline for analyzing metagenomic sequencing experiments.<ul><li>Sunbeam is written in Python using the <b>Snakemake</b> workflow management software and is freely available at github.com/sunbeam-labs/sunbeam under the GPLv3.</ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30875429">30875429</a> A Bayesian model for single cell transcript expression analysis on MERFISH data.<ul><li>The entire analysis performed in this paper is provided as a fully reproducible <b>Snakemake</b> (K&#xF6;ster and Rahmann, 2012) workflow via Zenodo (https://doi.org/10.5281/zenodo.752340).</ul></p><p>2020 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> Scalable Workflows and Reproducible Data Analysis for Genomics.<ul><li>We show how to create a scalable, reusable, and shareable workflow using four different workflow engines: the Common Workflow Language (CWL), Guix Workflow Language (GWL), <b>Snakemake</b>, and Nextflow. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30295032">30295032</a> ProteomeGenerator: A Framework for Comprehensive Proteomics Based on de Novo Transcriptome Assembly and High-Accuracy Peptide Mass Spectral Matching.<ul><li>Finally, ProteomeGenerator is implemented as a <b>Snakemake</b> workflow within a Singularity container for one-step installation in diverse computing environments, thereby enabling open, scalable, and facile discovery of sample-specific, non-canonical, and neomorphic biological proteomes.</ul></p><p>2018 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30064353">30064353</a> ATAC2GRN: optimized ATAC-seq and DNase1-seq pipelines for rapid and accurate genome regulatory network inference.<ul><li>We present optimized ATAC-seq and DNase-seq pipelines in both <b>Snakemake</b> and bash formats as well as optimal sequencing depths for ATAC-seq and DNase-seq projects. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28096075">28096075</a> hppRNA-a <b>Snakemake</b>-based handy parameter-free pipeline for RNA-Seq analysis of numerous samples.<ul><li>It features six independent core-workflows comprising the state-of-the-art technology with dozens of popular cutting-edge tools such as Tophat-Cufflink-Cuffdiff, Subread-featureCounts-DESeq2, STAR-RSEM-EBSeq, Bowtie-eXpress-edgeR, kallisto-sleuth, HISAT-StringTie-Ballgown, and embeds itself in <b>Snakemake</b>, which is a modern pipeline management system. </ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29361152">29361152</a> Sequanix: a dynamic graphical interface for <b>Snakemake</b> workflows.<ul><li>We designed a PyQt graphical user interface-Sequanix-aimed at democratizing the use of <b>Snakemake</b> pipelines in the NGS space and beyond. <li>Sequanix will be useful to both <b>Snakemake</b> developers in exposing their pipelines and to a wide audience of users.</ul></p><p>2019 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29340599">29340599</a> A benchmark study of scoring methods for non-coding mutations.<ul><li>The <b>Snakemake</b>, C++&#x2009;and R codes are freely available from https://github.com/Oncostat/BenchmarkNCVTools and supported on Linux.</ul></p><p>2018 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a> VIPER: Visualization Pipeline for RNA-seq, a <b>Snakemake</b> workflow for efficient and complete RNA-seq analysis.<ul><li>Using the workflow management system <b>Snakemake</b> we have developed a user friendly, fast, efficient, and comprehensive pipeline for RNA-seq analysis. </ul></p><p>2018 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29037173">29037173</a> TaxMapper: an analysis tool, reference database and workflow for metatranscriptome analysis of eukaryotic microorganisms.<ul><li>Additionally, TaxMapper is part of a metatranscriptomic <b>Snakemake</b> workflow developed to perform quality assessment, functional and taxonomic annotation and (multivariate) statistical analysis including environmental data. </ul></p><p>2018 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28807044">28807044</a> MetaMeta: integrating metagenome analysis tools to improve taxonomic profiling.<ul><li>MetaMeta uses <b>Snakemake</b> and has six pre-configured tools, all available at BioConda channel for easy installation (conda install -c bioconda metameta). </ul></p><p>2017 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26628585">26628585</a> An automated workflow for parallel processing of large multiview SPIM recordings.<ul><li>The pipeline relies on <b>snakemake</b> to resolve dependencies among consecutive processing steps and can be easily adapted to any cluster environment for processing SPIM data in a fraction of the time required to collect it.</ul></p><p>2013 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>Snakemake</b>--a scalable bioinformatics workflow engine.<ul><li><b>Snakemake</b> is a workflow engine that provides a readable Python-based workflow definition language and a powerful execution environment that scales from single-core workstations to compute clusters without modifying the workflow. </ul></p><p>2013 <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>Snakemake</b>--a scalable bioinformatics workflow engine.<ul><li>http://<b>snakemake</b>.googlecode.com.</ul></p></body>
