<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>Sentences that contain Snakemake</title>
</head>
<body>
<h1>Sentences that contain Snakemake</h1>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36707685">36707685</a> <b>(2)</b> A chromosome-level genome assembly of <italic>Plantago ovata</italic><ul><li> All scripts were written in <b>Snakemake</b> v5.26.1 (<b>Snakemake</b>, RRID:SCR_003475).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36681680">36681680</a> <b>(1)</b> Epigenomic charting and functional annotation of risk loci in renal cell carcinoma<ul><li>Sample-sample clustering, principal component analysis, and identification of lineage-enriched peaks were performed using Mapmaker (<ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/cfce/mapmaker">https://bitbucket.org/cfce/mapmaker</ext-link>), a ChIP-seq analysis pipeline implemented with <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR97"><u>97</u></xref></sup>.<li> [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36668672">36668672</a> <b>(2)</b> Multienzyme deep learning models improve peptide <italic toggle="yes">de novo</italic> sequencing by mass spectrometry proteomics<ul><li>A <b>snakemake</b> [<xref rid="pcbi.1010457.ref053" ref-type="bibr"><u>53</u></xref>] was created for the DDA search.<li> For the second step, a <b>snakemake</b> workflow was created for the extraction of the selected spectra and generation of the features and MGF files.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36662691">36662691</a> <b>(2)</b> Omnicrobe, an open-access database of microbial habitats and phenotypes using a comprehensive text mining and data fusion approach<ul><li>For integrating the heterogeneous processing steps and automating their execution we use <b>Snakemake</b> [<xref rid="pone.0272473.ref032" ref-type="bibr"><u>32</u></xref>], a workflow management system popular in bioinformatics.<li> <b>Snakemake</b> manages the tool dependencies and orchestrates the execution of the full workflows in cluster-based environments.<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36662851">36662851</a> <b>(1)</b> ESPRESSO: Robust discovery and quantification of transcript isoforms from error-prone long-read RNA-seq data<ul><li> Moreover, to facilitate easy use of ESPRESSO, we also provide a <b>Snakemake</b> pipeline that covers all of the main steps in ESPRESSO and upstream preprocessing steps, such as basecalling and read alignment.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36647171">36647171</a> <b>(1)</b> Bacterial topography of the upper and lower respiratory tract in pigs<ul><li>Shotgun metagenomic data were processed using a custom pipeline built on <b>Snakemake</b> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>].<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36639222">36639222</a> <b>(1)</b> Lineage tracing reveals fate bias and transcriptional memory in human B cells<ul><li>We used <b>Snakemake</b> (<xref rid="bib25" ref-type="bibr"><u>M&#xF6;lder et al, 2021</u></xref>) to manage the computational workflow.<li> [M&#xF6;lder et al, 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36639735">36639735</a> <b>(1)</b> HA-tag CD63 is a novel conditional transgenic approach to track extracellular vesicle interactions with sperm and their transfer at conception<ul><li>Our small non-coding RNA sequencing (sncRNASeq) analytical pipeline was designed using the <b>snakemake</b> framework and is available via GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/acshetty/sncRNA-seq-analysis">https://github.com/acshetty/sncRNA-seq-analysis</ext-link><sup><xref ref-type="bibr" rid="CR65"><u>65</u></xref></sup>.<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36687728">36687728</a> <b>(1)</b> Association between inflammation, lipopolysaccharide binding protein, and gut microbiota composition in a New Hampshire Bhutanese refugee population with a high burden of type 2 diabetes<ul><li>Raw sequences were processed using the Metagenome-atlas <b>snakemake</b> workflow (<xref rid="B16" ref-type="bibr"><u>16</u></xref>).<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32571209">32571209</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36602970">36602970</a> <b>(1)</b> Ten simple rules for using public biological data for your research<ul><li>Depending on the number of files and tool functionality and design, customized scripts or workflows (e.g., <b>Snakemake</b> [<xref rid="pcbi.1010749.ref038" ref-type="bibr"><u>38</u></xref>], NextFlow [<xref rid="pcbi.1010749.ref039" ref-type="bibr"><u>39</u></xref>]) may be needed for scalability.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36593369">36593369</a> <b>(1)</b> Genomic evidence of SARS-CoV-2 reinfection cases in southern Brazil<ul><li>A variant-calling pipeline, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/fhsantanna/variant_calling_pipeline">https://github.com/fhsantanna/variant_calling_pipeline</ext-link>, was built using the <b>Snakemake</b> scheme.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36711979">36711979</a> <b>(2)</b> RNA-seq preprocessing and sample size considerations for gene network inference<ul><li>A pipeline was developed to implement this workflow, which is based on the workflow management system <b>Snakemake</b> [<xref rid="R29" ref-type="bibr"><u>29</u></xref>].<li> It is based on <b>Snakemake</b> [<xref rid="R29" ref-type="bibr"><u>29</u></xref>], which is an efficient workflow management system.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36677413">36677413</a> <b>(1)</b> Metagenomic Strain-Typing Combined with Isolate Sequencing Provides Increased Resolution of the Genetic Diversity of <italic toggle="yes">Campylobacter jejuni</italic> Carriage in Wild Birds<ul><li>A computational pipeline for analysis of metagenomic sequencing data was implemented with <b>snakemake</b> [<xref rid="B34-microorganisms-11-00121" ref-type="bibr"><u>34</u></xref>].<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36700121">36700121</a> <b>(1)</b> Characterization of the humanized FRG mouse model and development of an AAV-LK03 variant with improved liver lobular biodistribution<ul><li> A workflow was written in <b>Snakemake</b> (5.6)42 to process reads and count barcodes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36636341">36636341</a> <b>(1)</b> Alternative telomere maintenance mechanism in <italic>Alligator sinensis</italic> provides insights into aging evolution<ul><li> Phylogenetic trees of mammals and non-mammal species were constructed using the Maximum Likelihood method with RAxML.<xref rid="bib130" ref-type="bibr"><sup>130</sup></xref> Positive selection analysis was carried out using the Codeml software in PAML package<xref rid="bib131" ref-type="bibr"><sup>131</sup></xref> implemented in the python pipeline "OH-SNAP" (Optimized High-throughput <b>Snakemake</b> Automation of PAML; available at <ext-link ext-link-type="uri" xlink:href="https://github.com/batlabucd/OHSNAP" id="intref0020">https://github.com/batlabucd/OHSNAP</ext-link>), which required inputs of phylogenetic trees, alignments, leading branches labeled in taxa, and set-up models.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36564496">36564496</a> <b>(1)</b> A metagenomic insight into the microbiomes of geothermal springs in the Subantarctic Kerguelen Islands<ul><li> Sequences were then processed with the <b>snakemake</b> of Anvi&#x2019;o<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref>&#x2013;<xref ref-type="bibr" rid="CR54"><u>54</u></xref></sup> (v7&#x2014;<ext-link ext-link-type="uri" xlink:href="https://github.com/merenlab/anvio">https://github.com/merenlab/anvio</ext-link>), filtered by integrated minoche script (v2.8&#x2014;<ext-link ext-link-type="uri" xlink:href="https://github.com/merenlab/illumina-utils">https://github.com/merenlab/illumina-utils</ext-link>).<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33349678">33349678</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36550219">36550219</a> <b>(1)</b> Diverse silent chromatin states modulate genome compartmentalization and loop extrusion barriers<ul><li> <xref ref-type="bibr" rid="CR61"><u>61</u></xref> but excluding those obtained from the ENCODE portal, were processed following the steps of the ENCODE ChIP&#x2013;seq pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/ENCODE-DCC/chip-seq-pipeline2">https://github.com/ENCODE-DCC/chip-seq-pipeline2</ext-link>) with slight modifications using a simplified custom <b>snakemake</b> workflow.<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26257180">26257180</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36522653">36522653</a> <b>(2)</b> Oncogenic KRAS alters splicing factor phosphorylation and alternative splicing in lung cancer<ul><li> The analysis pipeline is available as a <b>Snakemake</b> workflow at <ext-link ext-link-type="uri" xlink:href="https://github.com/bergerbio/RNA-splicing-screen">https://github.com/bergerbio/RNA-splicing-screen</ext-link>.<li> Analysis pipeline is available as a <b>Snakemake</b> workflow along with the in-house R scripts used to aggregate rMATS results and perform additional analyses and interpretations (<ext-link ext-link-type="uri" xlink:href="https://github.com/bergerbio/RNA-splicing-screen">https://github.com/bergerbio/RNA-splicing-screen</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36519725">36519725</a> <b>(2)</b> Automated hippocampal unfolding for morphometry and subfield segmentation with HippUnfold<ul><li> This is achieved via Snakebids, a tool designed to interface between BIDS datasets and <b>Snakemake</b> (<xref rid="bib34" ref-type="bibr"><u>Khan and Haast, 2021</u></xref>).<li> All aspects of HippUnfold use <b>Snakemake</b> (<xref rid="bib40" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>), a workflow management system based on Python which is reproducible, scalable, and seamlessly combines shell commands, Python code, and external dependencies in a human-readable workflow.<li> [Khan and Haast, 2021](          <person-group person-group-type="author"><name><surname>Khan</surname><given-names>A</given-names></name><name><surname>Haast</surname><given-names>R</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <data-title>Snakebids - BIDS integration into snakemake workflows</data-title>          <version designator="0.2.1">0.2.1</version>          <source>Zenodo</source>          <pub-id pub-id-type="doi">10.5281/zenodo.4488249</pub-id>        ) [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(1)</b> Automated hippocampal unfolding for morphometry and subfield segmentation with HippUnfold<ul><li> but with <b>Snakemake</b> handling these additional dependencies via Singularity.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36555644">36555644</a> <b>(1)</b> Gain or Loss? Evidence for Legume Predisposition to Symbiotic Interactions with Rhizobia via Loss of Pathogen-Resistance-Related Gene Families<ul><li> <b>Snakemake</b> 5.19.3 [<xref rid="B40-ijms-23-16003" ref-type="bibr"><u>40</u></xref>] workflow was used to control the execution of all steps of quality control, reconstruction and annotation.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36520686">36520686</a> <b>(22)</b> Ten simple rules and a template for creating workflows-as-applications<ul><li> Workflow managers, such as <b>Snakemake</b> [<xref rid="pcbi.1010705.ref004" ref-type="bibr"><u>4</u></xref>], Nextflow [<xref rid="pcbi.1010705.ref005" ref-type="bibr"><u>5</u></xref>], and Cromwell [<xref rid="pcbi.1010705.ref006" ref-type="bibr"><u>6</u></xref>] with WDL or CWL [<xref rid="pcbi.1010705.ref007" ref-type="bibr"><u>7</u></xref>], empower researchers to build robust pipelines that call a series of tools and scripts to perform a bespoke analysis.<li> However, the integration of cluster and cloud computing support in <b>Snakemake</b>, Nextflow, and Cromwell helped drive their current popularity.<li><b>snakemake</b> --cores 16 --configfile config.yaml \.<li><b>Snakemake</b> and Nextflow are two increasingly popular workflow management systems.<li>We present these ten simple rules for developers to follow which will facilitate their user communities in adopting and executing their workflows with fewer barriers, using examples with <b>Snakemake</b> and Nextflow.<li> Our templates also double as fully functioning examples of our ten simple rules and are available for both <b>Snakemake</b> (<ext-link xlink:href="http://github.com/beardymcjohnface/Snaketool" ext-link-type="uri">github.com/beardymcjohnface/Snaketool</ext-link>) and Nextflow (<ext-link xlink:href="http://github.com/beardymcjohnface/Nektool" ext-link-type="uri">github.com/beardymcjohnface/Nektool</ext-link>).<li><italic toggle="yes"># <b>Snakemake</b> example</italic>.<li> While user-customised configurations can be passed via the command line, immutable settings should be read directly by the workflow script (this is possible in both <b>Snakemake</b> and Nextflow).<li> In our example, we implement this functionality in <b>Snakemake</b> as additional bareword arguments, and in Nextflow by using the -entry flag.<li><italic toggle="yes"># Run preprocessing only (<b>Snakemake</b>)</italic>.<li> flags that are available for both <b>Snakemake</b> and Nextflow.<li>&#xA0;&#xA0;<b>Snakemake</b> command:.<li>&#xA0;&#xA0;<b>Snakemake</b> -j 8 --configfile my_tool_output/config.yaml \.<li> This feature is easy to implement in both <b>Snakemake</b> and Nextflow.<li> While profiles can be difficult to set up, there are official examples and templates for running workflow managers across different schedulers (<b>Snakemake</b>: <ext-link xlink:href="http://github.com/<b>snakemake</b>-profiles/doc" ext-link-type="uri">github.com/<b>snakemake</b>-profiles/doc</ext-link>; Nextflow: <ext-link xlink:href="http://nf-co.re/docs/usage/tutorials/step_by_step_institutional_profile" ext-link-type="uri">nf-co.re/docs/usage/tutorials/step_by_step_institutional_profile</ext-link>).<li> In our examples, the launcher achieves this by simply forwarding any unrecognised options to <b>Snakemake</b> or Nextflow.<li> <b>Snakemake</b> and Nextflow have many command line options and users should be encouraged to investigate these options for their situations.<li>&#xA0;&#xA0;<b>Snakemake</b> command:.<li>&#xA0;&#xA0;<b>snakemake</b>&#x2026; --<bold>dry-run</bold>.<li> The templates will save even more time in this process as well as offering complete working examples of these ten simple rules for both <b>Snakemake</b> and Nextflow.<li> [4](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name>, <name><surname>Rahmann</surname><given-names>S</given-names></name>. <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source>. <year>2012</year>. Available from: <ext-link xlink:href="https://academic.oup.com/bioinformatics/article-abstract/28/19/2520/290322" ext-link-type="uri">https://academic.oup.com/bioinformatics/article-abstract/28/19/2520/290322</ext-link>.) [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [6](<name><surname>Van der Auwera</surname><given-names>GA</given-names></name>, <name><surname>O&#x2019;Connor</surname><given-names>BD</given-names></name>. <source>Genomics in the Cloud: Using Docker, GATK, and WDL in Terra</source>. <publisher-name>O&#x2019;Reilly Media, Inc.</publisher-name>; <year>2020</year>.) [7](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, <etal>et al</etal>. <source>Common workflow language, v1. 0</source>. <year>2016</year>. Available from: <ext-link xlink:href="https://www.research.manchester.ac.uk/portal/en/publications/common-workflow-language-v10(741919f5-d0ab-4557-9763-b811e911423b).html" ext-link-type="uri">https://www.research.manchester.ac.uk/portal/en/publications/common-workflow-language-v10(741919f5-d0ab-4557-9763-b811e911423b).html</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36517866">36517866</a> <b>(1)</b> Genomic, transcriptomic and RNA editing analysis of human MM1 and VV2 sporadic Creutzfeldt-Jakob disease<ul><li>Bioinformatic analysis of sequencing data was performed with in-house pipelines using the <b>Snakemake</b> [<xref ref-type="bibr" rid="CR56"><u>56</u></xref>] workflow management system for process optimization [<xref ref-type="bibr" rid="CR57"><u>57</u></xref>].<li> [56](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <year>2012</year>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) [57](          <person-group person-group-type="author">            <name>              <surname>Dall&#x2019;Olio</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Curti</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Fonzi</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Sala</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Remondini</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Castellani</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Giampieri</surname>              <given-names>E</given-names>            </name>          </person-group>          <article-title>Impact of concurrency on the performance of a whole exome sequencing pipeline</article-title>          <source>BMC Bioinform</source>          <year>2021</year>          <volume>22</volume>          <fpage>1</fpage>          <lpage>15</lpage>          <pub-id pub-id-type="doi">10.1186/S12859-020-03780-3</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36513983">36513983</a> <b>(6)</b> Evaluation of taxonomic classification and profiling methods for long-read shotgun metagenomic sequencing datasets<ul><li> The pipeline is provided as a configurable <b>snakemake</b> workflow.<li> The pipeline is provided as a configurable <b>snakemake</b> workflow.<li> The information for the sample fasta files was added to the sample configuration file (Sample-Config.yaml), and the <b>snakemake</b> (Snakefile-taxprot) was executed.<li> The pipeline is provided as a configurable <b>snakemake</b> workflow.<li> The information for the sample fasta files was added to the sample configuration file (Sample-Config.yaml), and the <b>snakemake</b> (Snakefile-taxnuc) was executed.<li>We then edited the minimap2 command in the <b>snakemake</b> file to include the ONT recommended settings:.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36548615">36548615</a> <b>(1)</b> Effects of Moss-Dominated Biocrusts on Soil Microbial Community Structure in an Ionic Rare Earth Tailings Area of Southern China<ul><li> All steps were implemented via the <b>Snakemake</b> workflow [<xref rid="B34-toxics-10-00782" ref-type="bibr"><u>34</u></xref>].<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21651976">21651976</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36482253">36482253</a> <b>(1)</b> A critical period of translational control during brain development at codon resolution<ul><li>A <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR88"><u>88</u></xref></sup> file automating the above workflow is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/ohlerlab/cortexomics">https://github.com/ohlerlab/cortexomics</ext-link>.<li> [88](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36478271">36478271</a> <b>(1)</b> Freddie: annotation-independent detection and discovery of transcriptomic alternative splicing isoforms using long-read sequencing<ul><li> Note that the computational benchmarking pipeline can be fully reproduced using our <b>Snakemake</b>&#xA0;(<xref rid="B26" ref-type="bibr"><u>26</u></xref>) scripts at <ext-link xlink:href="https://github.com/vpc-ccg/freddie/tree/benchmarking" ext-link-type="uri">https://github.com/vpc-ccg/freddie/tree/benchmarking</ext-link>.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36480496">36480496</a> <b>(3)</b> Ten simple rules for investigating (meta)genomic data from environmental ecosystems<ul><li> Understanding the I/O files can help you extract, filter, or organize content within files through scripts, as well as help set up efficient workflows through the use of workflow management systems like <b>Snakemake</b> (<ext-link xlink:href="https://<b>snakemake</b>.readthedocs.io/" ext-link-type="uri">https://<b>snakemake</b>.readthedocs.io/</ext-link>) or Nextflow (<ext-link xlink:href="https://www.nextflow.io/" ext-link-type="uri">https://www.nextflow.io/</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36472514">36472514</a> <b>(1)</b> Genetic Drift Shapes the Evolution of a Highly Dynamic Metapopulation<ul><li> These analyses were conducted at the sciCORE (<ext-link xlink:href="http://scicore.unibas.ch/" ext-link-type="uri">http://scicore.unibas.ch/</ext-link>) scientific computing center at the University of Basel using a <b>snakemake</b> (<xref rid="msac264-B71" ref-type="bibr">M&#xF6;lder et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36469333">36469333</a> <b>(7)</b> TransFlow: a <b>Snakemake</b> workflow for transmission analysis of <italic toggle="yes">Mycobacterium tuberculosis</italic> whole-genome sequencing data<ul><li>In this article, we present a novel workflow named TransFlow which uses a modern computational workflow management system, <b>Snakemake</b> (<xref rid="btac785-B29" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), to combine many of the state-of-the-art tools currently employed in WGS-based MTBC transmission analysis into a single, fast, easy-to-use pipeline.<li> Upon execution, <b>Snakemake</b> infers the combination of rules necessary to achieve a &#x2018;target&#x2019; or specific output, in our case, the final summary report (referred to as the report file).<li> Secondly, TransFlow is developed based on <b>Snakemake</b> to acquire both efficiency and customizability.<li>4.3 <b>Snakemake</b> as a framework.<li>TransFlow is built upon <b>Snakemake</b> (<xref rid="btac785-B29" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), a scalable workflow engine that helps to manage workflows easily.<li> Additionally, <b>Snakemake</b> infers which rules are independent of each other and can be run in parallel, thus reducing idle CPU time to speed up workflow completions.<li> Next, modify some contents of the <b>Snakemake</b> rule which uses BWA for reads mapping, including the format of output files and shell commands, to meet the requirements of Bowtie2.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36556234">36556234</a> <b>(1)</b> Pharmacogenetics of <italic toggle="yes">CYP2A6</italic>, <italic toggle="yes">CYP2B6</italic>, and <italic toggle="yes">UGT2B7</italic> in the Context of HIV Treatments in African Populations<ul><li>The analysis was conducted using <b>Snakemake</b>, a python-based workflow management system developed for scalable and reproducible data analyses [<xref rid="B18-jpm-12-02013" ref-type="bibr"><u>18</u></xref>].<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36497174">36497174</a> <b>(1)</b> Deciphering Transcriptional Networks during Human Cardiac Development<ul><li>Demultiplexing, alignment on the GRCh38 reference genome and counting steps were conducted on each sequencing run with the <b>Snakemake</b> pipeline developed by the GenoBiRD core facility [<xref rid="B16-cells-11-03915" ref-type="bibr"><u>16</u></xref>].<li> [16](          <person-group person-group-type="author"><name><surname>Charpentier</surname><given-names>E.</given-names></name><name><surname>Cornec</surname><given-names>M.</given-names></name><name><surname>Dumont</surname><given-names>S.</given-names></name><name><surname>Meistermann</surname><given-names>D.</given-names></name><name><surname>Bordron</surname><given-names>P.</given-names></name><name><surname>David</surname><given-names>L.</given-names></name><name><surname>Redon</surname><given-names>R.</given-names></name><name><surname>Bonnaud</surname><given-names>S.</given-names></name><name><surname>Bihou&#xE9;e</surname><given-names>A.</given-names></name></person-group>          <article-title>3&#x2032;RNA Sequencing for Robust and Low-Cost Gene Expression Profiling</article-title>          <source>Protoc. Exch.</source>          <year>2021</year>          <pub-id pub-id-type="doi">10.21203/rs.3.pex-1336/v1</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36453861">36453861</a> <b>(1)</b> Pharokka: a fast scalable bacteriophage annotation tool<ul><li> With accessible bioconda installation, Pharokka is easily integrated into complex bioinformatic pipelines such as those created with <b>Snakemake</b> (<xref rid="btac776-B23" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36451239">36451239</a> <b>(1)</b> SIEVE: joint inference of single-nucleotide variants and cell phylogeny from single-cell DNA sequencing data<ul><li> The benchmarking framework was built using <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR55"><u>55</u></xref>].<li> [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36450762">36450762</a> <b>(1)</b> Self-cleaving guide RNAs enable pharmacological selection of precise gene editing events in vivo<ul><li> Quality checks, read trimming, pseudoalignment, and quantitation were performed via a reproducible <b>snakemake</b> pipeline, and all dependencies for these steps were deployed within the anaconda package management system<sup><xref ref-type="bibr" rid="CR57"><u>57</u></xref>&#x2013;<xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup>.<li> [57](Anon. Anaconda. <italic>Anaconda Software Distribution</italic><ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link> (2020).) [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27043002">27043002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36445112">36445112</a> <b>(1)</b> Long-Read Metagenome-Assembled Genomes Improve Identification of Novel Complete Biosynthetic Gene Clusters in a Complex Microbial Activated Sludge Ecosystem<ul><li> All steps in the annotation, genome mining, and GCF detection were managed using <b>Snakemake</b> v7.6.1 to ensure reproducibility (<xref rid="B74" ref-type="bibr"><u>74</u></xref>).<li> [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36443289">36443289</a> <b>(1)</b> Altered tRNA processing is linked to a distinct and unusual La protein in <italic>Tetrahymena thermophila</italic><ul><li>Details about tools and parameters of bioinformatic analyses are provided in a reproducible <b>Snakemake</b> workflow that can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/etiennefc/t_thermophila_RNA_Seq.git">https://github.com/etiennefc/t_thermophila_RNA_Seq.git</ext-link> and are also described below.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36441764">36441764</a> <b>(1)</b> A single N6-methyladenosine site regulates lncRNA HOTAIR function in breast cancer cells<ul><li> Briefly, a custom <b>Snakemake</b> workflow was generated based on the original eCLIP analysis strategies [<xref rid="pbio.3001885.ref058" ref-type="bibr"><u>58</u></xref>] to map reads to the human genome.<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27018577">27018577</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36506320">36506320</a> <b>(1)</b> Transcriptional mapping of the macaque retina and RPE-choroid reveals conserved inter-tissue transcription drivers and signaling pathways<ul><li>For the exact commands used in the analysis of this data, we make our <b>Snakemake</b> (<xref rid="B17" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) reproducible workflow available at <ext-link xlink:href="https://github.com/davemcg/macaque_macula_RNA-seq" ext-link-type="uri">https://github.com/davemcg/macaque_macula_RNA-seq</ext-link><underline>.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36434275">36434275</a> <b>(1)</b> A small bacteriophage protein determines the hierarchy over co-residential jumbo phage in <italic>Bacillus thuringiensis</italic> serovar <italic>israelensis</italic><ul><li>Samples were initially processed using a custom <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup> workflow (available at <ext-link ext-link-type="uri" xlink:href="https://github.com/AnzeLovse/mag">https://github.com/AnzeLovse/mag</ext-link>).<li> [40](K&#xF6;ster, J. et al. Sustainable data analysis with Snakemake. <italic>F1000Research</italic><bold>10</bold>, 33 (2021).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36467580">36467580</a> <b>(2)</b> The dynamic landscape of peptide activity prediction<ul><li> via docker, conda, <b>snakemake</b> or renv) the tool is granted a silver category.<li> via makefile, <b>snakemake</b>, drake or targets).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36422864">36422864</a> <b>(1)</b> <italic toggle="yes">N6</italic>-methyladenosine (m6A) reader Pho92 is recruited co-transcriptionally and couples translation to mRNA decay to promote meiotic fitness in yeast<ul><li> Due to the high ncRNA content in miCLIP libraries a sequential mapping strategy was used for all libraries, which is available as a <b>Snakemake</b> (<xref rid="bib50" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) pipeline from <ext-link xlink:href="https://www.github.com/ulelab/ncawareclip/" ext-link-type="uri">https://www.github.com/ulelab/ncawareclip/</ext-link>, (<xref rid="bib96" ref-type="bibr"><u>Varier, 2022</u></xref> copy archived at <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:13e5dca9d4008e8ec4a577e86d9dd07e19bce185;origin=https://www.github.com/ulelab/ncawareclip/;visit=swh:1:snp:183ab017bb9216823a1ce0e187e7dace3872d1f8;anchor=swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375" ext-link-type="uri">swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375</ext-link>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Varier, 2022](          <person-group person-group-type="author"><name><surname>Varier</surname><given-names>RA</given-names></name></person-group>          <year iso-8601-date="2022">2022</year>          <data-title>Ncawareclip</data-title>          <version designator="swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375">swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375</version>          <source>Software Heritage</source>          <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:13e5dca9d4008e8ec4a577e86d9dd07e19bce185;origin=https://www.github.com/ulelab/ncawareclip/;visit=swh:1:snp:183ab017bb9216823a1ce0e187e7dace3872d1f8;anchor=swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:13e5dca9d4008e8ec4a577e86d9dd07e19bce185;origin=https://www.github.com/ulelab/ncawareclip/;visit=swh:1:snp:183ab017bb9216823a1ce0e187e7dace3872d1f8;anchor=swh:1:rev:11b92a6302c5738f1c8c6da1f8b4aeb405082375</ext-link>        ) <p><u>No PMID</u> <b>(4)</b> A step forward for Shiga toxin-producing <italic toggle="yes">Escherichia coli</italic> identification and characterization in raw milk using long-read metagenomics<ul><li>A <b>snakemake</b> [<xref rid="R38" ref-type="bibr"><u>38</u></xref>] pipeline, STECmetadetector (<ext-link xlink:href="https://gitlab.com/bfr_bioinformatics/STECmetadetector" ext-link-type="uri">https://gitlab.com/bfr_bioinformatics/STECmetadetector</ext-link>), was developed in this study for user-friendly characterization of STEC strains from long-read metagenomics data.<li> STECmetadetector may be called by either using the Python <italic toggle="yes">STECmetadetector.py</italic> wrapper or by using the <b>snakemake</b> command line and editing the <italic toggle="yes">config.yaml</italic> file.<li>Generation of a <b>snakemake</b> pipeline for STEC characterization from long-read metagenomics data.<li>We generated a complete and open-source <b>snakemake</b> pipeline (<ext-link xlink:href="https://gitlab.com/bfr_bioinformatics/STECmetadetector" ext-link-type="uri">https://gitlab.com/bfr_bioinformatics/STECmetadetector</ext-link>), which orchestrates appropriate state-of-the-art tools to consistently and reproducibly characterize STEC strains from metagenomics data using an assembly-based approach (<xref rid="F2" ref-type="fig">Fig.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36557203">36557203</a> <b>(1)</b> Human Gut Microbiota in Coronary Artery Disease: A Systematic Review and Meta-Analysis <xref rid="fn1-metabolites-12-01165" ref-type="author-notes">&#x2020;</xref><ul><li>The meta-analytical pipeline has been run using <b>Snakemake</b> version 6.6.1.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36416261">36416261</a> <b>(2)</b> GPSAdb: a comprehensive web resource for interactive exploration of genetic perturbation RNA-seq datasets<ul><li> The curated RNA-seq data was processed by a <b>snakemake</b>-based pipeline (<ext-link xlink:href="https://github.com/xuzhougeng/auto_sra_rnaseq_pipeline" ext-link-type="uri">https://github.com/xuzhougeng/auto_sra_rnaseq_pipeline</ext-link>).<li> The RNA-seq data was processed by a <b>snakemake</b>-based pipeline (<ext-link xlink:href="https://github.com/xuzhougeng/auto_sra_rnaseq_pipeline" ext-link-type="uri">https://github.com/xuzhougeng/auto_sra_rnaseq_pipeline</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36482891">36482891</a> <b>(1)</b> Transcriptomic meta-analysis reveals unannotated long non-coding RNAs related to the immune response in sheep<ul><li>We downloaded and analysed the 422 RNA-seq samples with a uniform workflow using custom <b>Snakemake</b> v.6.15.1 (<xref rid="B44" ref-type="bibr"><u>K&#xF6;ster et al., 2021</u></xref>) pipelines (<xref rid="F1" ref-type="fig"><u>Figure 1</u></xref>).<li> [K&#xF6;ster et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36409063">36409063</a> <b>(2)</b> m<sup>6</sup>A modification of U6 snRNA modulates usage of two major classes of pre-mRNA 5&#x2019; splice site<ul><li> Pipelines for processing of nanopore DRS data were built and executed using <b>Snakemake</b> version 6.15.3 (<xref rid="bib55" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> Pipelines were written and executed using <b>Snakemake</b> version 6.15.3 (<xref rid="bib55" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36479125">36479125</a> <b>(1)</b> Human gut microbiota-reactive DP8&#x3B1; regulatory T cells, signature and related emerging functions<ul><li> Bioinformatics steps are performed using a <b>snakemake</b> pipeline (<uri xlink:href="https://bio.tools/3SRP">https://bio.tools/3SRP</uri>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36431075">36431075</a> <b>(1)</b> From Samples to Germline and Somatic Sequence Variation: A Focus on Next-Generation Sequencing in Melanoma Research<ul><li> In this way, for workflow standardization and automation, several managers are available, such as <b>Snakemake</b> [<xref rid="B107-life-12-01939" ref-type="bibr"><u>107</u></xref>], WDL [<xref rid="B108-life-12-01939" ref-type="bibr"><u>108</u></xref>], or Nextflow [<xref rid="B109-life-12-01939" ref-type="bibr"><u>109</u></xref>].<li> [107](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [108](          <person-group person-group-type="author"><name><surname>Voss</surname><given-names>K.</given-names></name><name><surname>Van der Auwera</surname><given-names>G.</given-names></name><name><surname>Gentry</surname><given-names>J.</given-names></name></person-group>          <article-title>Full-stack genomics pipelining with GATK4 + WDL + Cromwell</article-title>          <source>F1000Research</source>          <year>2017</year>          <volume>6</volume>          <pub-id pub-id-type="doi">10.7490/f1000research.1114634.1</pub-id>        ) [109](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36395816">36395816</a> <b>(1)</b> ViLoN&#x2014;a multi-layer network approach to data integration demonstrated for patient stratification<ul><li>An implementation of the algorithm introduced in this manuscript (using <b>Snakemake</b> (<xref rid="B74" ref-type="bibr"><u>74</u></xref>) workflows) together with documentation, tutorials, and pre-processed datasets for benchmarking and files supporting the analysis are available at <ext-link xlink:href="https://github.com/data-int/vilon/" ext-link-type="uri">https://github.com/data-int/vilon/</ext-link>.<li> [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36395091">36395091</a> <b>(1)</b> From Easy to Hopeless&#x2014;Predicting the Difficulty of Phylogenetic Analyses<ul><li> We implement this pipeline using the <b>Snakemake</b> workflow management system (<xref rid="msac254-B15" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) and Python 3.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36384919">36384919</a> <b>(1)</b> Compensatory epistasis maintains ACE2 affinity in SARS-CoV-2 Omicron BA.1<ul><li> To do so, we developed a <b>snakemake</b> pipeline<sup><xref ref-type="bibr" rid="CR39"><u>39</u></xref></sup> that first parsed through all fastq files and separated the reads according to inline indices, UMIs, and sequence reads using Python library regex<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup>.<li> [39](Moulana, A. et al. <italic>desai-lab/compensatory_epistasis_omicron</italic>. (Zenodo, 2022). 10.5281/ZENODO.7235104.) [40](Barnett, M. Regex. Preprint at (2013).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36382407">36382407</a> <b>(4)</b> MEGARes and AMR++, v3.0: an updated comprehensive database of antimicrobial resistance determinants and an improved software pipeline for classification using high-throughput sequencing<ul><li>Updated AMR++ 3.0: NextFlow and <b>SnakeMake</b> pipelines.<li> Alternatively, we present AMR++ written in <b>Snakemake</b> (<xref rid="B37" ref-type="bibr"><u>37</u></xref>), a python-based workflow engine.<li> and we have expanded the usability of AMR++ 3.0 by adding modulizable NextFlow workflows and a <b>SnakeMake</b> distribution.<li> The <b>SnakeMake</b> AMR++ pipeline version files are available under the MIT license on <ext-link xlink:href="https://github.com/jonathan-bravo/amrplusplus_v2" ext-link-type="uri">https://github.com/jonathan-bravo/amrplusplus_v2</ext-link>.<li> [37](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J.</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S.</given-names></string-name></person-group><article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>. <source>Bioinforma. Oxf. Engl.</source><year>2012</year>; <volume>28</volume>:<fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36383565">36383565</a> <b>(1)</b> Predictable and stable epimutations induced during clonal plant propagation with embryonic transcription factor<ul><li>The genome was assembled into 120 contigs of total length of 123.5 Mb with an N50 of 6.57 Mb using a custom <b>Snakemake</b> pipeline, &#x201C;auto-asm&#x201D; (<ext-link xlink:href="https://github.com/weigelworld/auto-asm" ext-link-type="uri">https://github.com/weigelworld/auto-asm</ext-link>).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36421248">36421248</a> <b>(1)</b> Novel Alligator Cathelicidin As-CATH8 Demonstrates Anti-Infective Activity against Clinically Relevant and Crocodylian Bacterial Pathogens<ul><li>An automated workflow was established using <b>Snakemake</b> version 5.3.0 [<xref rid="B57-antibiotics-11-01603" ref-type="bibr"><u>57</u></xref>].<li> [57](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36369503">36369503</a> <b>(1)</b> A nascent peptide code for translational control of mRNA stability in human cells<ul><li>Pre-processing steps for high-throughput sequencing were implemented as <b>Snakemake</b> workflows<sup><xref ref-type="bibr" rid="CR100"><u>100</u></xref></sup>.<li> [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36357860">36357860</a> <b>(1)</b> Genomic trajectories of a near-extinction event in the Chatham Island black robin<ul><li>To identify Z- and W-linked scaffolds in the black robin assembly, we used the <b>snakemake</b> workflow findZX (<ext-link ext-link-type="uri" xlink:href="https://github.com/hsigeman/findZX/">https://github.com/hsigeman/findZX/</ext-link>; [<xref ref-type="bibr" rid="CR81"><u>81</u></xref>]) and processed whole genome data from three male and three female black robin.<li> [81](Sigeman H, Sinclair B, Hansson B. Findzx: an automated pipeline for detecting and visualising sex chromosomes using whole-genome sequencing data. BMC Genomics. 2022;23:1-14.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36352329">36352329</a> <b>(1)</b> Whole-genome sequencing of artificial single-nucleotide variants induced by DNA degradation in biological crime scene traces<ul><li>Raw data analysis consisted of quality control and paired-end read alignment to the GRCh38 reference with Burrows-Wheeler Alignment (BWA) [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>] subsequent variant calling, and was performed by the Core Facility Genomics of the University (Westf&#xE4;lische-Wilhelms Universit&#xE4;t M&#xFC;nster) based on the Genome Analysis Toolkit (GATK) best-practice workflow [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] implemented by the <b>Snakemake</b> workflow dna-seq-gatk-variant-calling by K&#xF6;sters et al.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20080505">20080505</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21478889">21478889</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36428589">36428589</a> <b>(1)</b> Plasma Sequencing for Patients with GIST&#x2014;Limitations and Opportunities in an Academic Setting<ul><li>For research purposes a <b>Snakemake</b> [<xref rid="B12-cancers-14-05496" ref-type="bibr"><u>12</u></xref>] pipeline: dna-seq-varlociraptor [<xref rid="B13-cancers-14-05496" ref-type="bibr"><u>13</u></xref>] was adapted.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [13](          <person-group person-group-type="author"><collab>Forster</collab></person-group>          <article-title>snakemake-workflows/dna-seq-varlociraptor: v3.14.1</article-title>          <year>2022</year>          <comment>Available online: <ext-link xlink:href="https://github.com/snakemake-workflows/dna-seq-varlociraptor" ext-link-type="uri">https://github.com/snakemake-workflows/dna-seq-varlociraptor</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2022-07-25">(accessed on 25 July 2022)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36439243">36439243</a> <b>(2)</b> Streamlined single-molecule RNA-FISH of core clock mRNAs in clock neurons in whole mount <italic>Drosophila</italic> brains<ul><li> Finally, we developed an integrated pipeline in which all data analysis workflows are automated by the workflow manager <b>Snakemake</b> (<xref rid="B25" ref-type="bibr"><u>Molder et al., 2021</u></xref>) (<xref rid="F1" ref-type="fig"><u>Figure 1D</u></xref>).<li>To ensure reproducibility and aid optimization of analysis parameters, we developed an automated analysis pipeline using the <b>Snakemake</b> workflow management system (<xref rid="B25" ref-type="bibr"><u>Molder et al., 2021</u></xref>) (<xref rid="F1" ref-type="fig"><u>Figure 1F</u></xref>).<li> [Molder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [Molder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36342196">36342196</a> <b>(3)</b> A framework for group-wise summarization and comparison of chromatin state annotations<ul><li>The model is implemented using Python&#x2019;s sklearn, pybedtools packages and <b>snakemake</b> (<xref rid="btac722-B4" ref-type="bibr">Dale <italic toggle="yes">et al.<li>To facilitate the use of CSREP, we provide an implementation of CSREP as a <b>snakemake</b> pipeline (<xref rid="btac722-B18" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>; <xref rid="btac722-B20" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.<li> The program can be run either on local computers or on computing clusters, in which case <b>snakemake</b> will optimize the workflow for execution.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36389425">36389425</a> <b>(1)</b> The seeker R package: simplified fetching and processing of transcriptome data<ul><li> installSysDeps sets environmental variables so the CLIs are accessible within R, and installs <b>snakemake</b> (<xref rid="ref-13" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) to make it easier to use seeker in reproducible analyses.<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36333777">36333777</a> <b>(1)</b> A novel and dual digestive symbiosis scales up the nutrition and immune system of the holobiont <italic>Rimicaris exoculata</italic><ul><li>We removed adapter sequences from the reads using bbduk from bbmap v38.57 (<ext-link ext-link-type="uri" xlink:href="https://sourceforge.net/projects/bbmap/">https://sourceforge.net/projects/bbmap/</ext-link>) and performed the following steps using the metagenomic <b>snakemake</b> workflow implemented in anvi&#x2019;o v6 [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>].<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36334099">36334099</a> <b>(1)</b> The Dynamics of Adaptation to Stress from Standing Genetic Variation and de novo Mutations<ul><li> The workflow is available at <ext-link xlink:href="https://github.com/SLAment/AdaptationDynamics" ext-link-type="uri">https://github.com/SLAment/AdaptationDynamics</ext-link> as <b>Snakemake</b> v.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36335101">36335101</a> <b>(1)</b> Learning the histone codes with large genomic windows and three-dimensional chromatin interactions using transformer<ul><li> The whole data collection and preprocessing pipeline is implemented in <b>snakemake</b> workflow management system v6.5.3<sup><xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup>.<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36362329">36362329</a> <b>(1)</b> Transcriptomic Studies of Antidepressant Action in Rodent Models of Depression: A First Meta-Analysis<ul><li>The analyses were produced using <b>Snakemake</b> (<uri xlink:href="https://doi.org/10.12688/f1000research.29032.1">https://doi.org/10.12688/f1000research.29032.1</uri>, accessed on 19 April 2022) as workflow manager and R Bookdown for reporting.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36333300">36333300</a> <b>(1)</b> Mitochondrial Fission Process 1 controls inner membrane integrity and protects against heart failure<ul><li> The RNA-seq analysis was performed with Sequana 0.8.5<sup><xref ref-type="bibr" rid="CR91"><u>91</u></xref></sup> using an RNA-seq pipeline 0.9.13 (<ext-link ext-link-type="uri" xlink:href="https://github.com/sequana/sequana_rnaseq">https://github.com/sequana/sequana_rnaseq</ext-link>) built on top of <b>Snakemake</b> 5.8.1<sup><xref ref-type="bibr" rid="CR92"><u>92</u></xref></sup>.<li> [91](          <person-group person-group-type="author">            <name>              <surname>Cokelaer</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Desvillechabrol</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Legendre</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Cardon</surname>              <given-names>M</given-names>            </name>          </person-group>          <article-title>&#x201C;Sequana&#x201D;: a Set of Snakemake NGS pipelines</article-title>          <source>J Open Source Softw</source>          <year>2017</year>          <volume>2</volume>          <fpage>352</fpage>        ) [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36407635">36407635</a> <b>(8)</b> Plastaumatic: Automating plastome assembly and annotation<ul><li>The automation of the pipeline was achieved through <b>Snakemake</b>, a specification language built on Python (<xref rid="B16" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>).<li> Upon executing the snakefile, <b>Snakemake</b> can then run through all described steps in the workflow at once by taking the output files from an upstream rule and automatically feed them into the next rule.<li> The pipeline can be run using <b>Snakemake</b> or a shell script, both of which produce the same output.<li>In order to execute the pipeline, the computing systems used need to have <b>Snakemake</b> installed (<uri xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/getting_started/installation.html">https://<b>snakemake</b>.readthedocs.io/en/stable/getting_started/installation.html</uri>).<li> An additional wrapper script written in shell is also available for running the <italic>Plastaumatic</italic> pipeline and it does not require to have <b>Snakemake</b> installed.<li>The <italic>Plastaumatic</italic> pipeline can be executed either as a <b>snakemake</b> pipeline or as a shell script.<li> [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36321867">36321867</a> <b>(2)</b> GAVISUNK: genome assembly validation via inter-SUNK distances in Oxford Nanopore reads<ul><li>GAVISUNK can run on consumer hardware, research clusters, or in cloud computing environments, as its steps are automated as part of a configurable <b>Snakemake</b> workflow (<xref rid="btac714-B13" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.<li><monospace><b>snakemake</b> &#x2013;use-conda &#x2013;cores {thread.count}</monospace>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36322581">36322581</a> <b>(2)</b> RASCL: Rapid Assessment of Selection in CLades through molecular sequence analysis<ul><li>As a standalone pipeline (<ext-link xlink:href="https://github.com/veg/RASCL" ext-link-type="uri">https://github.com/veg/RASCL</ext-link>) in <b>Snakemake</b> [<xref rid="pone.0275623.ref037" ref-type="bibr"><u>37</u></xref>].<li>RASCL is available in two forms, as a standalone pipeline that uses <b>Snakemake</b>, as well as a web application integrated as a workflow in the Galaxy framework.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(2)</b> RASCL: Rapid Assessment of Selection in CLades through molecular sequence analysis<ul><li> It should be possible to run RASCL via the <b>Snakemake</b> file on a single Linux workstation with a sufficient number of cores and RAM.<li>* There is no file named `<b>snakemake</b>_config.json` in the RASCL directory.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(2)</b> RASCL: Rapid Assessment of Selection in CLades through molecular sequence analysis<ul><li> It should be possible to run RASCL via the <b>Snakemake</b> file on a single Linux workstation with a sufficient number of cores and RAM.<li>* There is no file named `<b>snakemake</b>_config.json` in the RASCL directory.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36386798">36386798</a> <b>(1)</b> Modification mapping by nanopore sequencing<ul><li> The <b>Snakemake</b> pipeline MetaCompore enables RNA modification detection using six different software tools (<xref rid="B57" ref-type="bibr"><u>Leger and Leonardi, 2021</u></xref>), and the METEORE pipeline takes an analogous approach to DNA methylation detection (<xref rid="B117" ref-type="bibr"><u>Yuen et al., 2021</u></xref>), enabling users to run multiple analyses simultaneously and compare their outputs.<li> [Leger and Leonardi, 2021](<person-group person-group-type="author"><name><surname>Leger</surname><given-names>A.</given-names></name><name><surname>Leonardi</surname><given-names>T.</given-names></name></person-group> (<year>2021</year>). <article-title>a-slide/MetaCompore: Release 0.1.3</article-title>. <source>Zenodo</source>. <pub-id pub-id-type="doi">10.5281/zenodo.4726175</pub-id>) [Yuen et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34103501">34103501</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36307828">36307828</a> <b>(1)</b> scTAM-seq enables targeted high-confidence analysis of DNA methylation in single cells<ul><li> The pipeline has been implemented in <italic>bash</italic> and <italic><b>snakemake</b></italic> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] and is available from the GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/veltenlab/scTAM-seq-scripts">https://github.com/veltenlab/scTAM-seq-scripts</ext-link>) [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>].<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [38](Scherer M, et al. scTAM-seq-scripts. GitHub. 2022. <ext-link ext-link-type="uri" xlink:href="https://github.com/veltenlab/scTAM-seq-scripts">https://github.com/veltenlab/scTAM-seq-scripts</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36300789">36300789</a> <b>(1)</b> Covalent disruptor of YAP-TEAD association suppresses defective Hippo signaling<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline <xref rid="bib10" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>.<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36300797">36300797</a> <b>(1)</b> Wide-ranging consequences of priority effects governed by an overarching factor<ul><li>Variants in genomic data were identified with <italic toggle="yes">grenepipe</italic> (0.1.0) (<xref rid="bib17" ref-type="bibr"><u>Czech and Exposito-Alonso, 2021</u></xref>), an automated variant calling pipeline for <italic toggle="yes"><b>snakemake</b></italic> (<xref rid="bib51" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [Czech and Exposito-Alonso, 2021](          <person-group person-group-type="author"><name><surname>Czech</surname><given-names>L</given-names></name><name><surname>Exposito-Alonso</surname><given-names>M</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <article-title>Grenepipe: A Flexible, Scalable, and Reproducible Pipeline to Automate Variant and Frequency Calling from Sequence Reads</article-title>          <source>arXiv</source>          <ext-link xlink:href="http://arxiv.org/abs/2103.15167" ext-link-type="uri">http://arxiv.org/abs/2103.15167</ext-link>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36286492">36286492</a> <b>(1)</b> Elimination of Foreign Sequences in Eukaryotic Viral Reference Genomes Improves the Accuracy of Virome Analysis<ul><li> Thus, we established a consecutive scrutiny pipeline, which is composed of five parts and managed using <b>Snakemake</b> (see <xref rid="figS1" ref-type="supplementary-material"><u>Fig.&#xA0;S1</u></xref> in the supplemental material).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36285285">36285285</a> <b>(1)</b> ICEscreen: a tool to detect Firmicute ICEs and IMEs, isolated or enclosed in composite structures<ul><li> The pipeline was managed with <b>Snakemake</b> (<xref rid="B42" ref-type="bibr"><u>42</u></xref>).<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36268892">36268892</a> <b>(1)</b> Establishing <italic toggle="yes">Physalis</italic> as a <italic toggle="yes">Solanaceae</italic> model system enables genetic reevaluation of the inflated calyx syndrome<ul><li> Structural gene annotations were then generated using the Mikado v2.0rc2 (<xref rid="koac305-B90" ref-type="bibr"><u>Venturini et al., 2018</u></xref>) framework using the evidence set mentioned above following the <b>Snakemake</b>-based pipeline [Daijin].<li> [Venturini et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30052957">30052957</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36264838">36264838</a> <b>(1)</b> Ten simple rules for a successful international consortium in big data omics<ul><li> Workflow managers like Jip (<ext-link xlink:href="https://pyjip.readthedocs.io/en/latest/index.html" ext-link-type="uri">https://pyjip.readthedocs.io/en/latest/index.html</ext-link>), Nextflow [<xref rid="pcbi.1010546.ref014" ref-type="bibr"><u>14</u></xref>], and <b>Snakemake</b> [<xref rid="pcbi.1010546.ref015" ref-type="bibr"><u>15</u></xref>] can provide a way to describe the steps and share with others.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36340405">36340405</a> <b>(5)</b> <italic>Fusarium oxysporum</italic> effector clustering version 2: An updated pipeline to infer host range<ul><li> FoEC2 is implemented in Python3 and based in <b>Snakemake</b> (<xref rid="B21" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>) to help manage external tools and internal scripts, as well as improve scalability with multithreading, i.<li> It is based on Python3 and <b>Snakemake</b> (v.<li>Several configuration files are used in the <b>Snakemake</b> pipeline.<li> The first three parts are executed by a <b>Snakemake</b> pipeline, which produces the output needed for the fourth part which performs clustering and visualization of presence/absence patterns.<li> The increased run time caused by using nhmmer searching is compensated by the use of multithreading facilitated by <b>Snakemake</b>, the chosen framework for FoEC2.<li> [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36261518">36261518</a> <b>(3)</b> Semi-automated assembly of high-quality diploid human reference genomes<ul><li> To improve reproducibility and transparency, <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>)<sup><xref ref-type="bibr" rid="CR81"><u>81</u></xref></sup> was used for pipeline construction and execution (10.18434/mds2-2578).<li> [81](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36293264">36293264</a> <b>(1)</b> Identification of Candidate mRNA Isoforms for Prostate Cancer-Risk SNPs Utilizing Iso-eQTL and sQTL Methods<ul><li> For RNA-Seq Analysis, <b>Snakemake</b> Workflow (RASflow) was used [<xref rid="B38-ijms-23-12406" ref-type="bibr"><u>38</u></xref>].<li> [38](          <person-group person-group-type="author"><name><surname>Zhang</surname><given-names>X.</given-names></name><name><surname>Jonassen</surname><given-names>I.</given-names></name></person-group>          <article-title>RASflow: An RNA-Seq analysis workflow with Snakemake</article-title>          <source>BMC Bioinform.</source>          <year>2020</year>          <volume>21</volume>          <elocation-id>110</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12859-020-3433-x</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33159858">33159858</a> <b>(1)</b> Single-Cell Profiling of Ebola Virus Disease <italic>In&#xA0;Vivo</italic> Reveals Viral and Host Dynamics<ul><li> We used a <b>Snakemake</b> wrapper around Dropseq-tools that is available in the open source Github repository <ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe" id="intref0085">https://github.com/Hoohm/dropSeqPipe</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33239365">33239365</a> <b>(4)</b> Whole-Genome Sequence Analysis of an Extensively Drug-Resistant <named-content content-type="genus-species">Salmonella enterica</named-content> Serovar Agona Isolate from an Australian Silver Gull (<italic>Chroicocephalus novaehollandiae</italic>) Reveals the Acquisition of Multidrug Resistance Plasmids<ul><li>Genotyping of SG17-135 and other <italic>S.</italic> Agona genomic assemblies was performed using a custom <b>Snakemake</b>-based workflow which utilized ABRicate (<ext-link ext-link-type="uri" xlink:href="https://github.com/tseemann/Abricate">https://github.com/tseemann/Abricate</ext-link>) version 0.9.3 and the nucleotide databases CARD (<xref rid="B54" ref-type="bibr"><u>54</u></xref>), VFDB (<xref rid="B55" ref-type="bibr"><u>55</u></xref>), and PlasmidFinder (<xref rid="B56" ref-type="bibr"><u>56</u></xref>).<li> The presence of pathogenicity islands was also detected using this <b>Snakemake</b> workflow; however, strains were considered positive for a given pathogenicity island in cases where they exhibited&#x2009;&#x2265;60% coverage and&#x2009;&#x2265;95% nucleotide identity threshold, as is the default setting of SPIfinder (<xref rid="B58" ref-type="bibr"><u>58</u></xref>) (from which SPI nucleotide sequences were sourced).<li> Specific versions of databases and dependencies are available, along with <b>Snakemake</b> workflows, at <ext-link ext-link-type="uri" xlink:href="https://github.com/maxlcummins/SG17-135">https://github.com/maxlcummins/SG17-135</ext-link>.<li>Phylogenetic analysis was performed using a custom <b>Snakemake</b>-based workflow which performed a full genome alignment of all <italic>S</italic>.<li> [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23650175">23650175</a>) [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15608208">15608208</a>) [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24777092">24777092</a>) [58](<person-group person-group-type="author"><name name-style="western"><surname>Roer</surname><given-names>L</given-names></name>, <name name-style="western"><surname>Hendriksen</surname><given-names>RS</given-names></name>, <name name-style="western"><surname>Leekitcharoenphon</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Lukjancenko</surname><given-names>O</given-names></name>, <name name-style="western"><surname>Kaas</surname><given-names>RS</given-names></name>, <name name-style="western"><surname>Hasman</surname><given-names>H</given-names></name>, <name name-style="western"><surname>Aarestrup</surname><given-names>FM</given-names></name></person-group><year>2016</year><article-title>Is the evolution of <italic>Salmonella enterica</italic> subsp. <italic>enterica</italic> linked to restriction-modification systems?</article-title><source>mSystems</source><volume>1</volume>:<elocation-id>e00009-16</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/mSystems.00009-16</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33281477">33281477</a> <b>(1)</b> The taxonomy of the model filamentous fungus <italic><named-content content-type="taxon-name"><named-content content-type="genus">Podospora</named-content><named-content content-type="species">anserina</named-content></named-content></italic><ul><li> 2017</xref>) and additional custom scripts available as a full <b>Snakemake</b> (<xref rid="B16" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2018</u></xref>) pipeline at <ext-link ext-link-type="uri" xlink:href="https://github.com/SLAment/Podosporaceae">https://github.com/SLAment/Podosporaceae</ext-link>.<li> [K&#xF6;ster and Rahmann 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28961734">28961734</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33230554">33230554</a> <b>(1)</b> RepeatFS: a file system providing reproducibility through provenance and automation<ul><li> Generic pipeline frameworks such as Ruffus (<xref rid="btaa950-B14" ref-type="bibr"><u>Goodstadt, 2010</u></xref>), Bpipe (<xref rid="btaa950-B25" ref-type="bibr"><u>Sadedin et al., 2012</u></xref>), <b>Snakemake</b> (<xref rid="btaa950-B19" ref-type="bibr"><u>K&#xF6;ster et al., 2012</u></xref>) and SoS (<xref rid="btaa950-B29" ref-type="bibr"><u>Wang et al., 2019</u></xref>), while designed to work with any program, require the researcher to write scripts to migrate their workflow into the framework.<li> [Goodstadt, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [Sadedin et al., 2012](<person-group person-group-type="author"><string-name><surname>Sadedin</surname><given-names>S.P.</given-names></string-name></person-group><etal>et al</etal> (<year>2012</year>) <article-title>Bpipe: a tool for running and managing bioinformatics pipelines</article-title>. <source>Bioinf. Oxf. Engl</source>., <volume>28</volume>, <fpage>1525</fpage>&#x2013;<lpage>1526</lpage>.) [K&#xF6;ster et al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Wang et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30811390">30811390</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33235280">33235280</a> <b>(1)</b> An integrative atlas of chicken long non-coding genes and their annotations across 25 tissues<ul><li> This workflow is part of a <b>snakemake</b><sup><xref ref-type="bibr" rid="CR98"><u>98</u></xref></sup> pipeline, available at Ref.<li> [98](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Dewey</surname>              <given-names>CN</given-names>            </name>          </person-group>          <article-title>RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome</article-title>          <source>BMC Bioinform.</source>          <year>2011</year>          <volume>12</volume>          <fpage>323</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33226342">33226342</a> <b>(1)</b> Combined transient ablation and single-cell RNA-sequencing reveals the development of medullary thymic epithelial cells<ul><li> The full <b>Snakemake</b> pipeline, our personal analysis package, and scripts to recreate all figures are available on GitHub: <ext-link ext-link-type="uri" xlink:href="https://github.com/kwells4/mtec_analysis">https://github.com/kwells4/mtec_analysis</ext-link>&#xA0;(<xref rid="bib63" ref-type="bibr"><u>Wells, 2020</u></xref>; copy archived at <ext-link ext-link-type="uri" xlink:href="https://archive.softwareheritage.org/swh:1:dir:bad3bd778d1b7fba04a3a33f35e83a60e04fe489;origin=https://github.com/kwells4/mtec_analysis;visit=swh:1:snp:87044609633d1421b6b619cbc7c3498667bcd583;anchor=swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe/">swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe</ext-link>) with the companion R package available: <ext-link ext-link-type="uri" xlink:href="https://github.com/kwells4/mtec.10x.pipeline">https://github.com/kwells4/mtec.10x.pipeline</ext-link>.<li> [Wells, 2020](          <person-group person-group-type="author">            <name>              <surname>Wells</surname>              <given-names>K</given-names>            </name>          </person-group>          <year>2020</year>          <data-title>mtec_analysis</data-title>          <source>Software Heritage</source>          <version designator="swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe">swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe</version>          <ext-link ext-link-type="uri" xlink:href="https://archive.softwareheritage.org/swh:1:dir:bad3bd778d1b7fba04a3a33f35e83a60e04fe489;origin=https://github.com/kwells4/mtec_analysis;visit=swh:1:snp:87044609633d1421b6b619cbc7c3498667bcd583;anchor=swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe/">https://archive.softwareheritage.org/swh:1:dir:bad3bd778d1b7fba04a3a33f35e83a60e04fe489;origin=https://github.com/kwells4/mtec_analysis;visit=swh:1:snp:87044609633d1421b6b619cbc7c3498667bcd583;anchor=swh:1:rev:d3955fad0d73dc404a93fc2d81b84141c9c79efe/</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33329686">33329686</a> <b>(8)</b> <italic>Cascabel</italic>: A Scalable and Versatile Amplicon Sequence Data Analysis Pipeline Delivering Reproducible and Documented Results<ul><li> All scripts of <italic>Cascabel</italic> are located within the project folder, and together with the <b>Snakemake</b> and configuration file, every run of the pipeline is completely reproducible at any time.<li>We here provide <italic>Cascabel</italic>, a <b>Snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B40" ref-type="bibr"><u>2012</u></xref>) pipeline for the analysis of community marker gene sequence data which is easy to use for people with little bioinformatics background, and both flexible and powerful enough to be attractive for people with bioinformatics training.<li>Our pipeline makes use of the workflow management engine <b>Snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B40" ref-type="bibr"><u>2012</u></xref>), which scales from personal workstations to computer clusters.<li> <b>Snakemake</b> takes care of executing the rules in a computationally efficient manner, making optimal use of available resources, e.g., distributing jobs over several nodes.<li> <b>Snakemake</b> will not re-run a rule if the output file of that rule already exists, unless &#x2013;&#x2013;<monospace>forcerun</monospace> is used or the input file has a more recent date than the existing output file.<li> Because <italic>Cascabel</italic> is a <b>Snakemake</b> workflow, it is flexible and easy to extend to encompass more or alternative rules.<li> With <italic>Cascabel</italic> being invoked from the command line, the user can make use of the full potential that <b>Snakemake</b> has to offer, e.g., &#x2013;&#x2013;<monospace>prioritize</monospace> to force the execution of specific rules prior to others when distributing tasks across computing resources, &#x2013;&#x2013;<monospace>until</monospace> to run the pipeline up to a specific rule, &#x2013;&#x2013;<monospace>summary</monospace>, which shows the rules executed so far and &#x2013;&#x2013;<monospace>dag</monospace> which shows the rules executed and the ones yet to be done in a directed acyclic graph.<li>We have presented <italic>Cascabel</italic>, an open source pipeline to analyze amplicon sequence data based on the workflow engine <b>Snakemake</b>.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33214309">33214309</a> <b>(1)</b> Draft Genome Sequences of 59 Endospore-Forming Gram-Positive Bacteria Associated with Crop Plants Grown in Vietnam<ul><li> The complete pipeline results were saved as a <b>Snakemake</b> file (<xref rid="B4" ref-type="bibr"><u>4</u></xref>) and uploaded on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/CptChiler/snakeGenome">https://github.com/CptChiler/snakeGenome</ext-link>).<li> [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33205815">33205815</a> <b>(2)</b> Unifying package managers, workflow engines, and containers: Computational reproducibility with BioNix<ul><li> This is commonly handled with &#x201C;workflow engines" (e.g., Toil&#xA0;[<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], <b>SnakeMake</b>&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], WDL&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], Cromwell&#xA0;[<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], NextFlow&#xA0;[<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], Ruffus&#xA0;[<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], and Rubra&#xA0;[<xref rid="bib13" ref-type="bibr"><u>13</u></xref>]), which manage stages (we define a &#x201C;stage" as the concrete execution of &#x2265;1 executable on &#x2265;1 input file, producing &#x2265;1 output file) and their execution, providing features such as parallelism, remote building, resumability, and logging.<li> Examples of implicit workflows are <b>SnakeMake</b>&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>] and Nextflow&#xA0;[<xref rid="bib11" ref-type="bibr"><u>11</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<comment>WDL Documentation. <ext-link ext-link-type="uri" xlink:href="https://software.broadinstitute.org/wdl/">https://software.broadinstitute.org/wdl/</ext-link>. Accessed 23 January 2019</comment>.) [10](<comment>Cromwell. <ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/cromwell">https://github.com/broadinstitute/cromwell</ext-link>. Accessed 17 June 2020</comment>.) [11](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechno</source>. <year>2017</year>;<volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [13](Rubra. <comment><ext-link ext-link-type="uri" xlink:href="https://github.com/bjpop/rubra">https://github.com/bjpop/rubra</ext-link>. Accessed 17 June 2020</comment>.) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotechno</source>. <year>2017</year>;<volume>35</volume>:<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33206700">33206700</a> <b>(1)</b> Freshwater diatom biomonitoring through benthic kick-net metabarcoding<ul><li> SCVURL is an automated <b>snakemake</b> [<xref rid="pone.0242143.ref047" ref-type="bibr"><u>47</u></xref>] bioinformatic pipeline that runs in a conda [<xref rid="pone.0242143.ref048" ref-type="bibr"><u>48</u></xref>] environment.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [48](Anaconda. Conda. 2016. <ext-link ext-link-type="uri" xlink:href="https://docs.conda.io/projects/conda/en/latest/index.html">https://docs.conda.io/projects/conda/en/latest/index.html</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33208761">33208761</a> <b>(1)</b> PRMT1 inhibition induces differentiation of colon cancer cells<ul><li> Bioinformatics Pipeline was run using <b>snakemake</b><sup><xref ref-type="bibr" rid="CR65"><u>65</u></xref></sup>.<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33203900">33203900</a> <b>(1)</b> XPC deficiency increases risk of hematologic malignancies through mutator phenotype and characteristic mutational signature<ul><li> All processing steps were combined in a pipeline built with <b>snakemake</b><sup><xref ref-type="bibr" rid="CR44"><u>44</u></xref></sup> (v5.4.0).<li> [44](K&#xF6;ster, J. &amp; Rahmann, S. Snakemake-a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic>10.1093/bioinformatics/bts480 (2012).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33198651">33198651</a> <b>(15)</b> Natrix: a <b>Snakemake</b>-based workflow for processing, clustering, and taxonomically assigning amplicon sequencing reads<ul><li> The workflow is written using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>], a workflow management engine for developing data analysis workflows.<li> <b>Snakemake</b> ensures reproducibility of a workflow by automatically deploying dependencies of workflow steps (rules) and scales seamlessly to different computing environments like servers, computer clusters, or cloud services.<li> Natrix is written in <b>Snakemake</b> [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>], a workflow management engine inspired by GNU Make using a Python-like language.<li> The generation of output files from input files in <b>Snakemake</b> is defined in a rule.<li> The order in which the rules are executed and which rules can be run in parallel is automatically inferred by <b>Snakemake</b> when a workflow is started by computing a directed acyclic graph (DAG).<li> If a target file is not specified during the start of the workflow, <b>Snakemake</b> will infer the order based on the input required by the first rule, which creates an <italic>all</italic> rule that exclusively contains the desired target files of the workflow as input.<li> Similar to the QIIME2 plugin system and in contrast to mothur and USEARCH, Natrix is easily extensible using <b>Snakemake</b> wrappers.<li> The workflow can further be extended or modified using the rule-based <b>Snakemake</b> syntax and Python, R scripts, or shell commands.<li> A standalone script can be incorporated into Natrix by specifying the input and output of the script in a <b>Snakemake</b> rule, the path to the script, and in the script itself, the only necessary adjustments are replacing the input and output file paths by a reference to <b>Snakemake</b>.<li> A further advantage is the automatic parallelization of rules by <b>Snakemake</b>.<li> While in other workflows the processing steps have to be run sequentially, <b>Snakemake</b> infers which rules are independent of each other and can be run in parallel.<li> The rule-based <b>Snakemake</b> syntax allows easy modification and extension of Natrix or the incorporation of parts into other workflows.<li>Programming environment: <b>Snakemake</b>, Python, R, Bash.<li> Natrix depends on either Docker or on <b>Snakemake</b> and the Conda package manager.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33203179">33203179</a> <b>(1)</b> Quick, Coordinated and Authentic Reprogramming of Ribosome Biogenesis during iPSC Reprogramming<ul><li> The FASTQ files were uploaded to the campus high performance computer cluster at the author&#x2019;s university in order to analyze the RNA-seq data with the following custom pipeline built in the <b>Snakemake</b> system (v5.2.2) [<xref rid="B62-cells-09-02484" ref-type="bibr"><u>62</u></xref>]: First, the author conducted quality control of the reads using FastQC, and trim the bases with quality scores of less than 20 using Trim_Galore! (v0.4.5).<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33202985">33202985</a> <b>(1)</b> The Role of Omega-3 Polyunsaturated Fatty Acids from Different Sources in Bone Development<ul><li> Pipeline was run using <b>Snakemake</b> [<xref rid="B27-nutrients-12-03494" ref-type="bibr"><u>27</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> The Role of Omega-3 Polyunsaturated Fatty Acids from Different Sources in Bone Development<ul><li> We implemented HRIBO based on <b>snakemake</b> (<xref rid="btaa959-B12" ref-type="bibr"><u>K&#xF6;ster et&#xA0;al., 2012</u></xref>), which allows highly reproducible and fully automatic data analysis.<li> [K&#xF6;ster et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33166284">33166284</a> <b>(2)</b> Investigating hospital <italic>Mycobacterium chelonae</italic> infection using whole genome sequencing and hybrid assembly<ul><li>We used the Sunbeam v1.3 [<xref rid="pone.0236533.ref015" ref-type="bibr"><u>15</u></xref>] pipeline to process the short reads and a custom <b>snakemake</b> v.5.6.0 pipeline, Nanoflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/zhoac1/nanoflow">https://github.com/zhoac1/nanoflow</ext-link>), to process the long reads and perform the hybrid assembly.<li>We used another custom <b>snakemake</b> pipeline, CoreSNPs (<ext-link ext-link-type="uri" xlink:href="https://github.com/chrgu/coreSNPs">https://github.com/chrgu/coreSNPs</ext-link>) to investigate how related the clinical isolates were to each other and to other whole genome sequenced references collected from GenBank.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30902113">30902113</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33158916">33158916</a> <b>(1)</b> Analysis of tumor-infiltrating NK and T cells highlights IL-15 stimulation and TIGIT blockade as a combination immunotherapy strategy for soft tissue sarcomas<ul><li> Raw fastq files underwent preprocessing using <italic><b>snakemake</b></italic>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33147627">33147627</a> <b>(3)</b> Computational strategies to combat COVID-19: useful tools to accelerate SARS-CoV-2 and coronavirus research<ul><li> The workflow itself is implemented using <b>Snakemake</b> [<xref rid="ref48" ref-type="bibr"><u>48</u></xref>], which provides advanced job balancing and input/output control mechanisms, and uses conda [<xref rid="ref28" ref-type="bibr"><u>28</u></xref>] to provide well-defined and harmonized software environments.<li> V-pipe uses the workflow management system <b>Snakemake</b> [<xref rid="ref48" ref-type="bibr"><u>48</u></xref>] to organize the order of required computational steps, and it supports cluster computing environments.<li> Workflow management systems such as <b>SnakeMake</b> or Nextflow [<xref rid="ref19" ref-type="bibr"><u>19</u></xref>, <xref rid="ref100" ref-type="bibr"><u>100</u></xref>] allow easy installation and reproducible execution on various platforms.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33139880">33139880</a> <b>(1)</b> Roles of bacteriophages, plasmids and CRISPR immunity in microbial community dynamics revealed using time-series integrated meta-omics<ul><li>Bioinformatics workflow automation was achieved using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR91"><u>91</u></xref></sup> (v.3.10.2 to v.5.1.4).<li> [91](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33500774">33500774</a> <b>(9)</b> Reproducibly sampling SARS-CoV-2 genomes across time, geography, and viral diversity<ul><li> The complete workflow can be applied in one step using the included <b>Snakemake</b><sup><xref rid="ref-7" ref-type="bibr"><u>7</u></xref></sup> workflow, or the steps can be applied individually.<li> The surrounding dashed box represents the <b>Snakemake</b> workflow which automates execution of the contained steps.<li> Given context metadata, context sequences, and focal sequences, the <b>Snakemake</b> workflow will produce a fasta file which is ready for alignment and a summary of the sampling procedure as a QIIME 2 visualization.<li> We hope the similarity of our interfaces (both of which require the same input and output, are accessible through <b>Snakemake</b>, and use the same terminology to describe data) will allow for independent comparison of these and other approaches.<li>The retrospective data provenance tracking system implemented in QIIME 2 differs from other systems such as <b>Snakemake</b><sup><xref rid="ref-7" ref-type="bibr"><u>7</u></xref></sup> or Galaxy<sup><xref rid="ref-11" ref-type="bibr"><u>11</u></xref></sup>, which we view as providing prospective data provenance tracking.<li> For example, when a <b>Snakemake</b> file is used to run a workflow, that workflow is documented for reproducibility by the <b>Snakemake</b> file.<li> This is not necessary with QIIME 2&#x2019;s retrospective data provenance tracking system, which records steps regardless of whether the workflow is run using a tool like <b>Snakemake</b> or Galaxy, or whether individual components are run independently.<li> QIIME 2 is therefore fully compatible with workflow engines such as <b>Snakemake</b> or Galaxy, but provides additional information which further ensures reproducibility.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a> <b>(2)</b> Reproducibly sampling SARS-CoV-2 genomes across time, geography, and viral diversity<ul><li>Note that <b>snakemake</b> supports config files that let you put the overridable configuration parameters in a YAML file.<li> Then you'd just need to install <b>snakemake</b> and run it with -- use-conda to get it all done.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a> <b>(5)</b> Reproducibly sampling SARS-CoV-2 genomes across time, geography, and viral diversity<ul><li><italic>Note that <b>snakemake</b> supports config files that let you put the overridable configuration parameters in a YAML file.<li> We now use a <b>Snakemake</b> config file to configure these parameters.<li> Then you'd just need to install <b>snakemake</b> and run it with -- use-conda to get it all done.<li> We have simplified our installation workflow (both for <b>Snakemake</b> users and non-<b>Snakemake</b> users) and hope that our provided workflow can serve as a template for more sophisticated use-cases, such as rule-specific conda or singularity containers.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33112851">33112851</a> <b>(1)</b> Major role of iron uptake systems in the intrinsic extra-intestinal virulence of the genus <italic>Escherichia</italic> revealed by a genome-wide association study<ul><li>Code is mostly based on the Python programming language and the following libraries: numpy v1.17.3 [<xref rid="pgen.1009065.ref095" ref-type="bibr"><u>95</u></xref>], scipy v1.4.0 [<xref rid="pgen.1009065.ref096" ref-type="bibr"><u>96</u></xref>], biopython v1.75 [<xref rid="pgen.1009065.ref097" ref-type="bibr"><u>97</u></xref>,<xref rid="pgen.1009065.ref098" ref-type="bibr"><u>98</u></xref>], pandas v0.25.3 [<xref rid="pgen.1009065.ref099" ref-type="bibr"><u>99</u></xref>], pybedtools v0.8.0 [<xref rid="pgen.1009065.ref100" ref-type="bibr"><u>100</u></xref>], dendropy 4.4.0 [<xref rid="pgen.1009065.ref101" ref-type="bibr"><u>101</u></xref>], ete3 v3.1.1 [<xref rid="pgen.1009065.ref102" ref-type="bibr"><u>102</u></xref>], statsmodels v0.10.2 [<xref rid="pgen.1009065.ref103" ref-type="bibr"><u>103</u></xref>], matplotlib v3.1.2 [<xref rid="pgen.1009065.ref104" ref-type="bibr"><u>104</u></xref>], seaborn v0.9.0 [<xref rid="pgen.1009065.ref105" ref-type="bibr"><u>105</u></xref>], jupyterlab v1.2.4 [<xref rid="pgen.1009065.ref106" ref-type="bibr"><u>106</u></xref>] and <b>snakemake</b> v5.8.2 [<xref rid="pgen.1009065.ref107" ref-type="bibr"><u>107</u></xref>].<li> [95](<name><surname>Van Der Walt</surname><given-names>S</given-names></name>, <name><surname>Colbert</surname><given-names>SC</given-names></name>, <name><surname>Varoquaux</surname><given-names>G</given-names></name>. <article-title>The NumPy array: a structure for efficient numerical computation</article-title>. <source>Comput. Sci. Eng</source>. <year>2011</year>;<volume>13</volume>:<fpage>22</fpage>&#x2013;<lpage>30</lpage>.) [96](<name><surname>Jones</surname><given-names>E</given-names></name>, <name><surname>Oliphant</surname><given-names>T</given-names></name>, <name><surname>Peterson</surname><given-names>P</given-names></name>. <article-title>SciPy: Open source scientific tools for Python.</article-title><year>2001</year><ext-link ext-link-type="uri" xlink:href="https://www.scipy.org/">http://www.scipy.org/</ext-link>.) [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19304878">19304878</a>) [98](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22909249">22909249</a>) [99](McKinney W, Others. Data structures for statistical computing in Python. in <italic>Proceedings of the 9th Python in Science Conference</italic> vol. 2010;445:51&#x2013;56.) [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21949271">21949271</a>) [101](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20421198">20421198</a>) [102](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26921390">26921390</a>) [103](Seabold S, Perktold J. Statsmodels: Econometric and statistical modeling with python. in Proceedings of the 9th Python in Science Conference vol. 57 61; SciPy society Austin, 2010.) [104](<name><surname>Hunter</surname><given-names>JD</given-names></name>. <article-title>Matplotlib: A 2D Graphics Environment</article-title>. <source>Computing in Science Engineering</source>. <year>2007</year>;<volume>9</volume>:<fpage>90</fpage>&#x2013;<lpage>95</lpage>.) [105](Waskom M, Botvinnik O, O'Kane D, Hobson P, Ostblom J, Lukauskas S, et al. mwaskom/seaborn: v0.9.0 (July 2018). 2018. <pub-id pub-id-type="doi">10.5281/zenodo.1313201</pub-id>) [106](<name><surname>Kluyver</surname><given-names>T</given-names></name>, <name><surname>Ragan-Kelley</surname><given-names>B</given-names></name>, <name><surname>P&#xE9;rez</surname><given-names>F</given-names></name>, <name><surname>Granger</surname><given-names>B</given-names></name>, <name><surname>Bussonnier</surname><given-names>M</given-names></name>, <name><surname>Frederic</surname><given-names>J</given-names></name>, <etal>et al</etal><article-title>Jupyter Notebooks-a publishing format for reproducible computational workflows.</article-title> in ELPUB 87&#x2013;90. <year>2016</year>.) [107](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33194387">33194387</a> <b>(6)</b> The Bacteria Genome Pipeline (BAGEP): an automated, scalable workflow for bacteria genomes with <b>Snakemake</b><ul><li> It uses existing pipelines and bioinformatics tools and an advanced workflow management system called <b>Snakemake</b> (<xref rid="ref-15" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>).<li>Upon execution of the pipeline, <b>Snakemake</b> organizes the correct combination of tasks in order to generate the desired output and also leverages on the wildcard feature to analyze multiple genomes that have similar format allowed in the rule (<xref rid="ref-15" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>).<li> BAGEP capitalizes on <b>Snakemake</b>&#x2019;s multi-threading feature which implies that it can be deployed on laptops with greater performance or a computing server to improve its speed.<li>BAGEP was built around the core concepts of <b>Snakemake</b> which offers parallelization of several tasks in an orderly manner.<li>As the backbone of BAGEP is <b>Snakemake</b>, each rule is run in its own environment and allows the combination of other programming languages such as Python, R and Bash.<li>We present BAGEP, a fully automated and scalable pipeline that is built on <italic><b>Snakemake</b></italic> framework.<li> [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33105476">33105476</a> <b>(2)</b> Defining the sediment prokaryotic communities of the Indian River Lagoon, FL, USA, an Estuary of National Significance<ul><li> Quantitative Insights Into Microbial Ecology 2&#x2019;s (QIIME2) (version 2018.11) [<xref rid="pone.0236305.ref040" ref-type="bibr"><u>40</u></xref>] standard 16S workflow was used for analysis, and a <b>Snakemake</b> file was used for the orchestration for reproducibility [<xref rid="pone.0236305.ref041" ref-type="bibr"><u>41</u></xref>, <xref rid="pone.0236305.ref042" ref-type="bibr"><u>42</u></xref>].<li> The <b>Snakemake</b> file used for QIIME2 analysis and any subsequent scripts used in statistical analysis can be found on Github (<ext-link ext-link-type="uri" xlink:href="https://github.com/djbradshaw2/General_16S_Amplicon_Sequencing_Analysis">https://github.com/djbradshaw2/General_16S_Amplicon_Sequencing_Analysis</ext-link>) [<xref rid="pone.0236305.ref069" ref-type="bibr"><u>69</u></xref>].<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31341288">31341288</a>) [41](<collab>QIIME 2 Development Team</collab>. <article-title>&#x201C;Moving Pictures&#x201D; tutorial</article-title>. <year>2020</year> [cited 6 Sep 2020]. Available: <ext-link ext-link-type="uri" xlink:href="https://docs.qiime2.org/2020.8/tutorials/moving-pictures/">https://docs.qiime2.org/2020.8/tutorials/moving-pictures/</ext-link>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [69](<name><surname>Bradshaw</surname><given-names>DJ</given-names></name>. <source>General_16S_Amplicon_Sequencing_Analysis</source>. <year>2020</year> [cited 6 Sep 2020]. Available: <ext-link ext-link-type="uri" xlink:href="https://github.com/djbradshaw2/General_16S_Amplicon_Sequencing_Analysis">https://github.com/djbradshaw2/General_16S_Amplicon_Sequencing_Analysis</ext-link>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33113898">33113898</a> <b>(1)</b> In Vivo Validation of Alternative FDXR Transcripts in Human Blood in Response to Ionizing Radiation<ul><li>Nanopore cDNA reads were analysed using a <b>snakemake</b> pipeline modified to handle paired samples [<xref rid="B63-ijms-21-07851" ref-type="bibr"><u>63</u></xref>].<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30356428">30356428</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33092528">33092528</a> <b>(5)</b> PyGNA: a unified framework for geneset network analysis<ul><li> PyGNA facilitates the integration with workflow systems, such as <b>Snakemake</b> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], thus lowering the barrier to introduce network analysis in existing pipelines.<li> Using a CLI interface facilitates integration with workflow analysis systems, such as <sc><b>Snakemake</b></sc> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>].<li> We have developed <sc><b>Snakemake</b></sc> pipelines to perform network analysis, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/stracquadaniolab/workflow-pygna">https://github.com/stracquadaniolab/workflow-pygna</ext-link>, which can be readily integrated into existing workflows.<li> <xref rid="Fig1" ref-type="fig"><u>1</u></xref> and <sc><b>Snakemake</b></sc> pipelines are available at: <ext-link ext-link-type="uri" xlink:href="https://github.com/stracquadaniolab/workflow-pygna">https://github.com/stracquadaniolab/workflow-pygna</ext-link>.<li> Moreover, the support for multi-core processing and easy integration with <sc><b>Snakemake</b></sc> allows to run PyGNA on multiple datasets and experiments at a glance.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33091012">33091012</a> <b>(3)</b> Ten simple rules for biologists initiating a collaboration with computer scientists<ul><li> iPython, <ext-link ext-link-type="uri" xlink:href="https://ipython.org/">https://ipython.org</ext-link>) notebooks, Rmarkdown (<ext-link ext-link-type="uri" xlink:href="https://rmarkdown.rstudio.com/">https://rmarkdown.rstudio.com</ext-link>), Galaxy (<ext-link ext-link-type="uri" xlink:href="https://galaxyproject.org/">https://galaxyproject.org</ext-link>), <b>snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/">https://<b>snakemake</b>.readthedocs.io</ext-link>), nextflow (<ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io</ext-link>), and LaTeX (<ext-link ext-link-type="uri" xlink:href="https://www.latex-project.org/">https://www.latex-project.org</ext-link>) (and Overleaf; <ext-link ext-link-type="uri" xlink:href="https://www.overleaf.com/">https://www.overleaf.com</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33087062">33087062</a> <b>(14)</b> MetaLAFFA: a flexible, end-to-end, distributed computing-compatible metagenomic functional annotation pipeline<ul><li>Workflow management and distributed computing integration via <b>Snakemake</b>.<li>MetaLAFFA is implemented in <b>Snakemake</b> [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>], a Python-based workflow management framework that is specifically designed for bioinformatics analysis pipelines.<li> <b>Snakemake</b> determines dependencies between different steps of a pipeline based on the expected inputs and outputs of each step, and ensures that later steps are only run once their inputs become available.<li> One benefit of this approach is that <b>Snakemake</b> automatically detects when the expected outputs of a step are missing, halts the pipeline, and allows the user to simply resume operation from the most recent successfully completed steps after they address any issues that led to the failure.<li>Another useful feature of <b>Snakemake</b> is its ability to compartmentalize pipeline operations.<li> <b>Snakemake</b> can split steps in the pipeline into independent jobs where, for example, each job processes a separate sample.<li>Finally, <b>Snakemake</b> has built-in capabilities to take advantage of parallel processing and distributed computing, which allows MetaLAFFA users to make the best use of their available computational resources.<li> Specifically, <b>Snakemake</b> automatically determines which steps (and individual jobs when steps are split into separate jobs) are independent and can be run in parallel.<li> Then <b>Snakemake</b> will use available cores if being run locally, or an available distributed computing environment (e.g.<li> Additionally, this system leverages <b>Snakemake</b>&#x2019;s usage of workflow specification through input and output naming patterns to enable users to more quickly experiment with different operating parameters.<li> This is supported thanks to <b>Snakemake</b>&#x2019;s ability to identify, based on output folder and file names, which intermediate outputs need to be newly generated.<li> Implemented using <b>Snakemake</b>, MetaLAFFA can take advantage of extensive parallelization, making use of either local or distributed computing resources.<li><italic>Programming language</italic> Python and <b>Snakemake</b>.<li><italic>Other requirements</italic> Python 3.6 or greater, Conda 4.8 or greater, and <b>Snakemake</b> 3.13.3 or greater.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33086062">33086062</a> <b>(2)</b> Enhancer Reprogramming Confers Dependence on Glycolysis and IGF Signaling in KMT2D Mutant Melanoma<ul><li> RNaseq data were processed by pyflow-RNaseq (<xref rid="R86" ref-type="bibr"><u>Tang, 2017b</u></xref>), a <b>snakemake</b> based RNaseq pipeline.<li> ChIP-seq data were quality controlled and processed by pyflow-ChIPseq (<xref rid="R85" ref-type="bibr"><u>Tang, 2017a</u></xref>), a <b>snakemake</b> (<xref rid="R35" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) based ChIPseq pipeline.<li> [Tang, 2017b](<name><surname>Tang</surname><given-names>M</given-names></name> (<year>2017b</year>). <source>pyflow-RNAseq: a snakemake based RNAseq pipeline</source> (<publisher-name>Zenodo</publisher-name>).) [Tang, 2017a](<name><surname>Tang</surname><given-names>M</given-names></name> (<year>2017a</year>). <source>pyflow-ChIPseq: a snakemake based ChIP-seq pipeline</source> (<publisher-name>Zenodo</publisher-name>).) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33077766">33077766</a> <b>(2)</b> Genetic determinism of spontaneous masculinisation in XX female rainbow trout: new insights using medium throughput genotyping and whole-genome sequencing<ul><li>After sequence mapping on the reference genome assembly<sup><xref ref-type="bibr" rid="CR22"><u>22</u></xref>,<xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup> (bwa mem v0.7.17, GCA_002163495.1), 34,064,394 variants were obtained using a homemade pipeline (<ext-link ext-link-type="uri" xlink:href="https://forgemia.inra.fr/bios4biol/workflows/tree/master/<b>Snakemake</b>/IMAGE_calling">https://forgemia.inra.fr/bios4biol/workflows/tree/master/<b>Snakemake</b>/IMAGE_calling</ext-link>) based on three variant calling tools (GATK v3.7.0, FreeBayes v1.2.0 and SAMtools mpileup v1.8.0).<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31768021">31768021</a>) [23](Gao, G. <italic>et al.</italic> A new single nucleotide polymorphism database for rainbow trout generated through whole genome resequencing. <italic>Front. Genet.</italic><bold>9</bold> (2018).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33068114">33068114</a> <b>(1)</b> NanoGalaxy: Nanopore long-read sequencing data analysis in Galaxy<ul><li> Command line&#x2013;based workflow managers such as <b>Snakemake</b> and Nextflow&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>,<xref rid="bib10" ref-type="bibr"><u>10</u></xref>] can be used for analysing the data.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33060642">33060642</a> <b>(1)</b> Repeated sampling facilitates within- and between-subject modeling of the human sperm transcriptome to identify dynamic and stress-responsive sncRNAs<ul><li>The small non-coding RNA sequencing (sncRNASeq) analytical pipeline was designed using the <b>snakemake</b> framework and is available via GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/acshetty/sncRNA-seq-analysis">https://github.com/acshetty/sncRNA-seq-analysis</ext-link><sup><xref ref-type="bibr" rid="CR104"><u>104</u></xref></sup>.<li> [104](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33056985">33056985</a> <b>(2)</b> The structural variation landscape in 492 Atlantic salmon genomes<ul><li>We developed a workflow for SV discovery using paired-end short-read sequencing data aligned to the unmasked ICSASG_V2 reference assembly<sup><xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup>, which can be run in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup> (Supplementary Fig.&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref>).<li> All of the above steps were combined in a <b>Snakemake</b> (v3.11.0)<sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup> workflow, with the input being paired-end sequencing data (FASTQ format), and the output a VCF file with SV locations and genotypes for all individuals in a study (Supplementary Fig.&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref> and Supplementary Note&#xA0;<xref rid="MOESM1" ref-type="media"><u>2</u></xref> provides Snakefile).<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27088604">27088604</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33051688">33051688</a> <b>(1)</b> dbGuide: a database of functionally validated guide RNAs for genome editing in human and mouse cells<ul><li> A custom python <b>snakemake</b> (<xref rid="B37" ref-type="bibr"><u>37</u></xref>) pipeline was made to calculate non-homologous end joining (NHEJ) mutation frequencies.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33035207">33035207</a> <b>(2)</b> Genome Complexity Browser: Visualization and quantification of genome variability<ul><li> <b>snakemake</b> pipeline for these steps is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/paraslonic/orthosnake/blob/tree/Snakefile_tree">https://github.com/paraslonic/orthosnake/blob/tree/Snakefile_tree</ext-link>.<li> <b>Snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/paraslonic/orthosnake">https://github.com/paraslonic/orthosnake</ext-link>) is provided to infer orthogroups.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33035207">33035207</a> <b>(1)</b> Genome Complexity Browser: Visualization and quantification of genome variability<ul><li> to the <b>Snakemake</b> script to infer homology groups (name, place?).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33009951">33009951</a> <b>(1)</b> Patient-derived organoids and orthotopic xenografts of primary and recurrent gliomas represent relevant patient avatars for precision oncology<ul><li> Workflow automation was performed using the workflow manager <b>snakemake</b> (v.5.6.0) under conda (v.4.7.12) [<xref ref-type="bibr" rid="CR58"><u>58</u></xref>].<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33023248">33023248</a> <b>(3)</b> OneStopRNAseq: A Web Application for Comprehensive and Efficient Analyses of RNA-Seq Data<ul><li> The <b>Snakemake</b> workflow-management system [<xref rid="B36-genes-11-01165" ref-type="bibr"><u>36</u></xref>] was used to build the robust, reproducible, and scalable analysis pipeline.<li> The <b>Snakemake</b> workflow is available for downloading or contributing at <uri xlink:href="https://github.com/radio1988/OneStopRNAseq">https://github.com/radio1988/OneStopRNAseq</uri>.<li>At the back end, the <b>Snakemake</b> workflow management system is used for reproducible and scalable data analyses.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32993491">32993491</a> <b>(12)</b> Hypercluster: a flexible tool for parallelized unsupervised clustering optimization<ul><li>Here we present hypercluster, a python package and <b>SnakeMake</b> pipeline for rigorous, reproducible and parallelized clustering calculation and evaluation.<li> The <b>SnakeMake</b> pipeline allows parallelization, greatly reducing wall-clock time for users [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> Hypercluster requires python3, pandas [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], numpy [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>], scipy [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>], matplotlib [<xref ref-type="bibr" rid="CR42"><u>42</u></xref>], seaborn [<xref ref-type="bibr" rid="CR43"><u>43</u></xref>], scikit-learn [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>], python-igraph [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>], leidenalg [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>], louvain-igraph [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>] and <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> <bold>b</bold> <b>Snakemake</b> is then used to distribute each clustering calculation into different jobs.<li>Hypercluster can be run independently of <b>SnakeMake</b>, as a standalone python package.<li>Configuring the <b>SnakeMake</b> pipeline.<li>The <b>SnakeMake</b> pipeline allows users to parallelize clustering calculations on multiple threads on a single computer, multiple compute nodes on a high performance cluster or in a cloud cluster [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li>Parameters in <b>SnakeMake</b> configuration file.<li> Users can then simply run &#x201C;<b>snakemake</b> -s hypercluster.smk -configfile config.yml&#x201D; in the command line, with any additional <b>SnakeMake</b> flags appropriate for their system.<li> Applying the same configuration to new files or testing new algorithms on old data simply requires editing the inputs in the config.yml file and rerunning the <b>SnakeMake</b> command.<li>Other requirements: Hypercluster runs with the following versions or higher: python 3.7, pandas 0.24.2, numpy 1.16.4, scipy 1.2.1, matplotlib 3.1.0, seaborn 0.9.0, scikit-learn 0.22.0, hdbscan 0.8.24, <b>snakemake</b> 5.8.2, python-igraph 0.7.1, leidenalg 0.7.0, louvain 0.6.1.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [39](McKinney W, Others. Data structures for statistical computing in python. In: Proceedings of the 9th Python in Science Conference. Austin, TX; 2010. p. 51&#x2013;6.) [40](          <person-group person-group-type="author">            <name>              <surname>van der Walt</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Colbert</surname>              <given-names>SC</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G</given-names>            </name>          </person-group>          <article-title>The NumPy array: a structure for efficient numerical computation</article-title>          <source>Comput Sci Eng</source>          <year>2011</year>          <volume>13</volume>          <issue>2</issue>          <fpage>22</fpage>          <lpage>30</lpage>        ) [41](Virtanen P, Gommers R, Oliphant TE, Haberland M, Reddy T, Cournapeau D et al. SciPy 1.0--Fundamental Algorithms for Scientific Computing in Python [Internet]. arXiv [cs.MS]. 2019. <ext-link ext-link-type="uri" xlink:href="http://arxiv.org/abs/1907.10121">https://arxiv.org/abs/1907.10121</ext-link>) [42](          <person-group person-group-type="author">            <name>              <surname>Hunter</surname>              <given-names>JD</given-names>            </name>          </person-group>          <article-title>Matplotlib: a 2D graphics environment</article-title>          <source>Comput Sci Eng</source>          <year>2007</year>          <volume>9</volume>          <issue>3</issue>          <fpage>90</fpage>          <lpage>95</lpage>        ) [43](Waskom M, Botvinnik O, O&#x2019;Kane D, Hobson P, Lukauskas S, Gemperline DC et al. mwaskom/seaborn: v0.8.1 (September 2017) [Internet]. 2017. <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/883859">https://zenodo.org/record/883859</ext-link>.) [35](          <person-group person-group-type="author">            <name>              <surname>Pedregosa</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Gramfort</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Michel</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Thirion</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Grisel</surname>              <given-names>O</given-names>            </name>            <etal/>          </person-group>          <article-title>Scikit-learn: machine learning in Python</article-title>          <source>J Mach Learn Res</source>          <year>2011</year>          <volume>12</volume>          <fpage>2825</fpage>          <lpage>2830</lpage>        ) [36](          <person-group person-group-type="author">            <name>              <surname>Csardi</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Nepusz</surname>              <given-names>T</given-names>            </name>            <etal/>          </person-group>          <article-title>The igraph software package for complex network research</article-title>          <source>Int J Complex Syst</source>          <year>2006</year>          <volume>1695</volume>          <issue>5</issue>          <fpage>1</fpage>          <lpage>9</lpage>        ) [37](Traag V. leidenalg [Internet]. Github; [cited 2020 Jan 27]. <ext-link ext-link-type="uri" xlink:href="https://github.com/vtraag/leidenalg">https://github.com/vtraag/leidenalg</ext-link>) [38](Traag V. louvain-igraph [Internet]. Github; [cited 2020 Jan 27]. <ext-link ext-link-type="uri" xlink:href="https://github.com/vtraag/louvain-igraph">https://github.com/vtraag/louvain-igraph</ext-link>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32994293">32994293</a> <b>(1)</b> Strain-Level Analysis of <italic toggle="yes">Bifidobacterium</italic> spp. from Gut Microbiomes of Adults with Differing Lactase Persistence Genotypes<ul><li>All software was run parallelized on a high-performance computing cluster at the Max Planck Institute for Developmental Biology via <b>Snakemake</b> (<xref rid="B46" ref-type="bibr"><u>46</u></xref>).<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32995794">32995794</a> <b>(1)</b> COVID-19 CG: Tracking SARS-CoV-2 mutations by locations and dates of interest<ul><li>Our data processing pipeline is written with the <b>Snakemake</b> scalable bioinformatics workflow engine (<xref rid="R9" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), which modularizes our workflow and enables reproducibility and compatibility with cloud-computing.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32986754">32986754</a> <b>(6)</b> Genomic characterization of a diazotrophic microbiota associated with maize aerial root mucilage<ul><li> Code for the <b>Snakemake</b> [<xref rid="pone.0239677.ref018" ref-type="bibr"><u>18</u></xref>] workflow used to conduct the computational analysis is available at: (<ext-link ext-link-type="uri" xlink:href="https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates">https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates</ext-link>).<li> Code used to generate, compare and classify MinHash genome sketches is included in the <b>Snakemake</b> workflow hosted at: (<ext-link ext-link-type="uri" xlink:href="https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates">https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates</ext-link>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32978264">32978264</a> <b>(1)</b> Genomic Prediction Informed by Biological Processes Expands Our Understanding of the Genetic Architecture Underlying Free Amino Acid Traits in Dry <italic>Arabidopsis</italic> Seeds<ul><li>Genotype data were previously published (<xref rid="bib8" ref-type="bibr">Atwell <italic>et al.</italic> 2010</xref>) and were accessed from <ext-link ext-link-type="uri" xlink:href="https://github.com/Gregor-Mendel-Institute/atpolydb/wiki">https://github.com/Gregor-Mendel-Institute/atpolydb/wiki</ext-link> The scripts and phenotypic data supporting the conclusions of this article are publicly available as a <b>Snakemake</b> workflow (v5.4.2, <xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) on GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/mishaploid/aa-genomicprediction">https://github.com/mishaploid/aa-genomicprediction</ext-link> (archived at <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.4048850">https://doi.org/10.5281/zenodo.4048850</ext-link>).<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32955438">32955438</a> <b>(1)</b> The effect of hybridization on transposable element accumulation in an undomesticated fungal species<ul><li>GNU Parallel version 20170922 (<xref rid="bib120" ref-type="bibr"><u>Tange, 2011</u></xref>) and <b>SnakeMake</b> version 5.5.4 (<xref rid="bib60" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) were used extensively in the analyses.<li> [Tange, 2011](          <person-group person-group-type="author">            <name>              <surname>Tange</surname>              <given-names>O</given-names>            </name>          </person-group>          <year>2011</year>          <source>GNU Parallel: The Command-Line Power Tool</source>          <publisher-name>Login: The USENIX Magazine</publisher-name>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32941642">32941642</a> <b>(2)</b> The conserved elongation factor Spn1 is required for normal transcription, histone modifications, and splicing in <italic>Saccharomyces cerevisiae</italic><ul><li> All data analyses were managed using the <b>Snakemake</b> workflow management system (<xref rid="B103" ref-type="bibr"><u>103</u></xref>).<li> Additionally, updated versions of the <b>Snakemake</b> pipelines used are available at github.com/winston-lab.<li> [103](<person-group person-group-type="author"><name name-style="western"><surname>Koster</surname><given-names>J.</given-names></name>, <name name-style="western"><surname>Rahmann</surname><given-names>S.</given-names></name></person-group><article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source>. <year>2018</year>; <volume>28</volume>:<fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32999994">32999994</a> <b>(1)</b> Trade-Offs between Geographic Scale, Cost, and Infrastructure Requirements for Fully Renewable Electricity in Europe<ul><li>The model code and all analysis steps are publicly available as a reproducible <b>Snakemake</b><xref rid="bib34" ref-type="bibr"><sup>34</sup></xref> workflow on Zenodo <ext-link ext-link-type="doi" xlink:href="10.5281/zenodo.3949794" id="PC_linkLJLur1sn5Q">https://doi.org/10.5281/zenodo.3949794</ext-link> and <ext-link ext-link-type="doi" xlink:href="10.5281/zenodo.3950775" id="PC_linkYZxa3k5P3u">https://doi.org/10.5281/zenodo.3950775</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32926128">32926128</a> <b>(1)</b> Shark: fishing relevant reads in an RNA-Seq sample<ul><li> Information on how to reproduce the experiments are available at <ext-link ext-link-type="uri" xlink:href="https://github.com/AlgoLab/shark_experiments">https://github.com/AlgoLab/shark_experiments</ext-link> in the form of a Conda environment and several <b>Snakemake</b> workflows (<xref rid="btaa779-B21" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32925972">32925972</a> <b>(3)</b> Diazotrophic bacteria from maize exhibit multifaceted plant growth promotion traits in multiple hosts<ul><li> Code used to generate MinHash sketches of isolate WGS assemblies is provided within the <b>Snakemake</b> workflow [<xref rid="pone.0239081.ref018" ref-type="bibr"><u>18</u></xref>] hosted on Github: (<ext-link ext-link-type="uri" xlink:href="https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates">https://github.com/shigdon/<b>snakemake</b>_mucilage-isolates</ext-link>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32908143">32908143</a> <b>(1)</b> The neuromuscular junction is a focal point of mTORC1 signaling in sarcopenia<ul><li> adapter, indexing the reference transcriptome, mapping RNA-Seq libraries, and assigning mapped reads to transcripts was performed with <b>Snakemake</b> workflow framework<sup><xref ref-type="bibr" rid="CR89"><u>89</u></xref></sup>.<li> [89](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32983411">32983411</a> <b>(3)</b> RVDB-prot, a reference viral protein database and its HMM profiles<ul><li>The HMM generation rationale was inspired from vFam (the database of HMM profiles built from all the viral proteins present in RefSeq, discontinued from 2014)<sup><xref rid="ref-5" ref-type="bibr"><u>5</u></xref></sup>, but was entirely re-coded as a <b>Snakemake</b> pipeline<sup><xref rid="ref-6" ref-type="bibr"><u>6</u></xref></sup>, using different tools for some key steps (clustering, alignment).<li>The different steps explained above are performed using a <b>Snakemake</b> pipeline<sup><xref rid="ref-6" ref-type="bibr"><u>6</u></xref></sup>, available at Institut Pasteur&#x2019;s Gitlab.<li>Several tools are needed to run the pipeline, including: Python, Mafft, Golden, HMMER, <b>Snakemake</b>, Silix, Blast+.<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25140992">25140992</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32893715">32893715</a> <b>(1)</b> Impact and consequences of intensive chemotherapy on intestinal barrier and microbiota in acute myeloid leukemia: the role of mucosal strengthening<ul><li> The next steps were realized with <b>Snakemake</b> to manage QIIME (2019.1) software.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32883364">32883364</a> <b>(1)</b> 1200 high-quality metagenome-assembled genomes from the rumen of African cattle and their relevance in the context of sub-optimal feeding<ul><li>For taxonomic classification of the 850 dereplicated MAGs, the <b>Snakemake</b> pipeline MAGpy was employed [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>].<li> [20](Stewart RD, Auffret M, Snelling TJ, Roehe R, Watson M. MAGpy: a reproducible pipeline for the downstream analysis of metagenome-assembled genomes (MAGs). Bioinformatics. 2018;35(12):2150&#x2013;2.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32953263">32953263</a> <b>(1)</b> DATMA: Distributed AuTomatic Metagenomic Assembly and annotation framework<ul><li>Although there are several workflow engines (e.g., <b>Snakemake</b>&#xA0;(<xref rid="ref-24" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>), Nextflow&#xA0;(<xref rid="ref-16" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), Ibis&#xA0;(<xref rid="ref-5" ref-type="bibr"><u>Bal et al., 2010</u></xref>), and Swift&#xA0;(<xref rid="ref-50" ref-type="bibr"><u>Wilde et al., 2011</u></xref>)) we selected COMPSs&#xA0;(<xref rid="ref-4" ref-type="bibr"><u>Badia et al., 2015</u></xref>) which provided us with the tools that we required: simple python interface and automatic parallel task distribution and synchronization.<li> [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Bal et al., 2010](          <person-group person-group-type="author">            <name>              <surname>Bal</surname>              <given-names>HE</given-names>            </name>            <name>              <surname>Maassen</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Van&#xA0;Nieuwpoort</surname>              <given-names>RV</given-names>            </name>            <name>              <surname>Drost</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Kemp</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Van&#xA0;Kessel</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Palmer</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Wrzesinska</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Kielmann</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Van&#xA0;Reeuwijk</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Seinstra</surname>              <given-names>FJ</given-names>            </name>            <name>              <surname>Jacobs</surname>              <given-names>CJH</given-names>            </name>            <name>              <surname>Verstoep</surname>              <given-names>K</given-names>            </name>          </person-group>          <year>2010</year>          <article-title>Real-world distributed computer with Ibis</article-title>          <source>Computer</source>          <volume>43</volume>          <issue>8</issue>          <fpage>54</fpage>          <lpage>62</lpage>          <pub-id pub-id-type="doi">10.1109/mc.2010.184</pub-id>        ) [Wilde et al., 2011](          <person-group person-group-type="author">            <name>              <surname>Wilde</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Hategan</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Wozniak</surname>              <given-names>JM</given-names>            </name>            <name>              <surname>Clifford</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Katz</surname>              <given-names>DS</given-names>            </name>            <name>              <surname>Foster</surname>              <given-names>I</given-names>            </name>          </person-group>          <year>2011</year>          <article-title>Swift: a language for distributed parallel scripting</article-title>          <source>Parallel Computing</source>          <volume>37</volume>          <issue>9</issue>          <fpage>633</fpage>          <lpage>652</lpage>          <pub-id pub-id-type="doi">10.1016/j.parco.2011.05.005</pub-id>        ) [Badia et al., 2015](          <person-group person-group-type="author">            <name>              <surname>Badia</surname>              <given-names>RM</given-names>            </name>            <name>              <surname>Conejero</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Diaz</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Ejarque</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lezzi</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Lordan</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Ramon-Cortes</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Sirvent</surname>              <given-names>R</given-names>            </name>          </person-group>          <year>2015</year>          <article-title>COMP Superscalar, an interoperable programming framework</article-title>          <source>SoftwareX</source>          <volume>3&#x2013;4</volume>          <fpage>32</fpage>          <lpage>36</lpage>          <pub-id pub-id-type="doi">10.1016/j.softx.2015.10.004</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32876050">32876050</a> <b>(1)</b> Integrating genotypes and phenotypes improves long-term forecasts of seasonal influenza A/H3N2 evolution<ul><li>The entire workflow for our analyses was implemented with <b>Snakemake</b> (<xref rid="bib21" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32869739">32869739</a> <b>(1)</b> Intra-species differences in population size shape life history and genome evolution<ul><li> To generate a pipeline for data processing we used <b>Snakemake</b> (<xref rid="bib40" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32887686">32887686</a> <b>(1)</b> Comprehensive analysis of structural variants in breast cancer genomes using single-molecule sequencing<ul><li>The SV inference and comparison workflow is implemented with <b>Snakemake</b> (<xref rid="GR260497AGAC041" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) v 5.5.4 and is available at GitHub (<uri xlink:href="https://github.com/aganezov/EnsembleSV">https://github.com/aganezov/EnsembleSV</uri>).<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33367265">33367265</a> <b>(1)</b> Maintenance of Species Differences in Closely Related Tetraploid Parasitic <italic>Euphrasia</italic> (Orobanchaceae) on an Isolated Island<ul><li> These computations were automated using <b>Snakemake</b> pipelines (<xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32859925">32859925</a> <b>(1)</b> The archives are half-empty: an assessment of the availability of microbial community sequencing data<ul><li> Including extensive primer and file formatting information, as well as documentation of computational processing steps, which is automatically provided by state-of-the-art pipelines such as QIIME2<sup><xref ref-type="bibr" rid="CR25"><u>25</u></xref></sup> or <b>Snakemake</b> workflows<sup><xref ref-type="bibr" rid="CR26"><u>26</u></xref></sup> in the sequence metadata, may greatly facilitate data re-use.<li> [25](Bolyen, E. et al. QIIME 2: Reproducible, interactive, scalable, and extensible microbiome data science PeerJ Preprints (2018).) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32853198">32853198</a> <b>(4)</b> Ten simple rules to increase computational skills among biologists with Code Clubs<ul><li><b>Snakemake</b>.<li>Given a bash script that contains an analysis pipeline, convert it to a <b>Snakemake</b> workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/SchlossLab/<b>snakemake</b>_riffomonas_tutorial">https://github.com/SchlossLab/<b>snakemake</b>_riffomonas_tutorial</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32817965">32817965</a> <b>(1)</b> Genomic Diversity of SARS-CoV-2 During Early Introduction into the United States National Capital Region<ul><li>We used a customized Nextstrain <b>snakemake</b> pipeline<sup><xref rid="R12" ref-type="bibr"><u>12</u></xref></sup> using augur v9.0.0 on the final dataset, which included the 886 Global 1K dataset plus all removed regional groups (including all 114 JHHS sequences and all published Virginia sequences), resulting in 1279 sequences used in our phylogenetic analysis.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33343922">33343922</a> <b>(1)</b> Efficient inference, potential, and limitations of site-specific substitution models<ul><li> The different steps were assembled into a pipeline using the workflow manager <b>Snakemake</b> (<xref rid="veaa066-B16" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>).<li> [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32810208">32810208</a> <b>(2)</b> Enhancer RNAs predict enhancer&#x2013;gene regulatory links and are critical for enhancer function in neuronal systems<ul><li>Paired-end FASTQ files were aligned and trimmed using a custom bioinformatic pipeline initialized in <b>snakemake</b> v5.3.0 (<xref rid="B49" ref-type="bibr"><u>49</u></xref>).<li>Paired-end FASTQ files were uploaded to the University of Alabama at Birmingham's High-Performance Computer cluster for custom bioinformatics analysis using a pipeline built with <b>snakemake</b> (<xref rid="B49" ref-type="bibr"><u>49</u></xref>).<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32790776">32790776</a> <b>(1)</b> No more business as usual: Agile and effective responses to emerging pathogen threats require open data and open analytics<ul><li> They can be combined into workflows using systems like the Common Workflow Language (CWL) [<xref rid="ppat.1008643.ref019" ref-type="bibr"><u>19</u></xref>], Nextflow [<xref rid="ppat.1008643.ref020" ref-type="bibr"><u>20</u></xref>], or <b>Snakemake</b> [<xref rid="ppat.1008643.ref021" ref-type="bibr"><u>21</u></xref>].<li> [19](Common Workflow Language [Internet]. Common Workflow Language. [cited 2020 Feb 21]. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/</ext-link>.) [20](Nextflow&#x2014;A DSL for parallel and scalable computational pipelines [Internet]. Nextflow. [cited 2020 Feb 21]. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</ext-link>.) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32780016">32780016</a> <b>(1)</b> Irisin directly stimulates osteoclastogenesis and bone resorption in vitro and in vivo<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="bib7" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32767996">32767996</a> <b>(5)</b> Integrative analyses of single-cell transcriptome and regulome using MAESTRO<ul><li>We implemented the MAESTRO using the <b>Snakemake</b> workflow management system [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>], which brings three advantages.<li> First, MAESTRO utilizes <b>Snakemake</b> to deploy and parallelize jobs on most computing platforms from high-performance servers, clusters, to the cloud.<li> Second, MAESTRO retrieves job descriptions and parameters through the <b>Snakemake</b> configuration files, so the pipeline can be easily customized for data from different technologies.<li> Last, the <b>Snakemake</b> workflow keeps track of the parameters and log files in each step, so it is easy to reproduce the result or fine-tune the MAESTRO pipeline.<li> We implemented MAESTRO using the <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>] and deposited the package under the Conda environment, which allowed MAESTRO to be installed and executed with simple commands.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32760117">32760117</a> <b>(1)</b> Supply-side options to reduce land requirements of fully renewable electricity in Europe<ul><li> All data procedures and all analysis steps are publicly available [<xref rid="pone.0236958.ref029" ref-type="bibr"><u>29</u></xref>] as a <b>Snakemake</b> workflow [<xref rid="pone.0236958.ref030" ref-type="bibr"><u>30</u></xref>].<li> [29](<name><surname>Tr&#xF6;ndle</surname><given-names>T.</given-names></name><source>Code related to "Tr&#xF6;ndle (2020)&#x2013;Supply-side options to reduce land requirements of fully renewable electricity in Europe"</source>. <publisher-loc>Potsdam</publisher-loc>: <publisher-name>Institute for Advanced Sustainability Studies</publisher-name>; <year>2020</year><pub-id pub-id-type="doi">10.5281/zenodo.3956531</pub-id>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33575609">33575609</a> <b>(11)</b> gNOMO: a multi-omics pipeline for integrated host and microbiome analysis of non-model organisms<ul><li> The pipeline has been implemented using the Python-based <b>Snakemake</b> (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) framework to perform fully automated and reproducible multi-omics analyses of host and microbiome samples.<li>The complete gNOMO pipeline is built in <b>Snakemake</b> (<xref rid="B27" ref-type="bibr"><u>27</u></xref>), a management system for bioinformatic workflows, that allows obtaining standardized and reproducible output data.<li> The input data and parameters of programs that are used in <b>Snakemake</b> are defined by editing a single configuration file.<li> Tools added to BioConda provide a user-friendly installation because the required tools and libraries are easily incorporated and automatically installed with the use of <b>Snakemake</b> environments.<li> The runtime highly depends on the number of available cores because <b>Snakemake</b> is able to parallelize non-dependent tasks and decreases the runtime this way substantially.<li> In addition, it should be stated that the <b>Snakemake</b> workflow engine is compatible and scalable in cluster environments (e.g.<li> This software is not implemented in BioConda but will be automatically installed locally with the <b>snakemake</b> script and allows a TIGRFAM (<xref rid="B37" ref-type="bibr"><u>37</u></xref>) functional annotation.<li>The pipeline is developed using the modular <b>Snakemake</b> framework that allows to incorporate software tools and libraries with different requirements.<li> <b>Snakemake</b> makes use of programming languages Python and Bash, which are commonly used in bioinformatics.<li> Parameters can be specified in the configuration file provided to <b>Snakemake</b>, so it can be adapted to any kind of host or microbiome analyzed.<li> The use of <b>Snakemake</b> makes gNOMO fully automated, efficient and reproducible.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=12520025">12520025</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32744506">32744506</a> <b>(1)</b> ZCWPW1 is recruited to recombination hotspots by PRDM9 and is essential for meiotic double strand break repair<ul><li>Computational analysis was performed using R (<xref rid="bib82" ref-type="bibr"><u>R Development Core Team, 2018</u></xref>) and <b>snakemake</b> (<xref rid="bib46" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [R Development Core Team, 2018](          <person-group person-group-type="author">            <collab>R Development Core Team</collab>          </person-group>          <year>2018</year>          <data-title>R: A Language and Environment for Statistical Computing</data-title>          <publisher-name>R Foundation for Statistical Computing</publisher-name>          <publisher-loc>Vienna, Austria</publisher-loc>          <ext-link ext-link-type="uri" xlink:href="http://www.r-project.org">http://www.r-project.org</ext-link>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32953934">32953934</a> <b>(1)</b> CERE-120 Prevents Irradiation-Induced Hypofunction and Restores Immune Homeostasis in Porcine Salivary Glands<ul><li> FASTQ files were extracted using bcl2fastq, and the FASTQ files were further processed using the <italic><b>snakemake</b>/3.8</italic> utility in the Biowulf2 cluster.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32718320">32718320</a> <b>(1)</b> Bovine breed-specific augmented reference graphs facilitate accurate sequence read mapping and unbiased variant discovery<ul><li><italic><b>Snakemake</b></italic> workflows [<xref ref-type="bibr" rid="CR80"><u>80</u></xref>] for whole-genome graph construction, read mapping, and variant discovery are available in the <italic>Github</italic> repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/danangcrysnanto/bovine-graphs-mapping">https://github.com/danangcrysnanto/bovine-graphs-mapping</ext-link>).<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32722451">32722451</a> <b>(1)</b> A Transposon Story: From TE Content to TE Dynamic Invasion of <italic>Drosophila</italic> Genomes Using the Single-Molecule Sequencing Technology from Oxford Nanopore<ul><li> All codes are available in a <b>snakemake</b> file at <uri xlink:href="https://github.com/DrosophilaGenomeEvolution/TrEMOLO">https://github.com/DrosophilaGenomeEvolution/TrEMOLO</uri>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32722369">32722369</a> <b>(1)</b> Parental Selenium Nutrition Affects the One-Carbon Metabolism and the Hepatic DNA Methylation Pattern of Rainbow Trout (<italic>Oncorhynchus mykiss</italic>) in the Progeny<ul><li>In-house R and Python scripts were coordinated in a pipeline by using <b>Snakemake</b> [<xref rid="B69-life-10-00121" ref-type="bibr"><u>69</u></xref>].<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32703166">32703166</a> <b>(6)</b> CDSnake: <b>Snakemake</b> pipeline for retrieval of annotated OTUs from paired-end reads using CD-HIT utilities<ul><li> Here we combined CD-HIT-OTU-Miseq utilities into pipeline using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>] workflow.<li> <b>Snakemake</b> is fully portable, as only a Python installation is required to run Snakefiles, and does not require tight integration of tools into the workflow system.<li> Usage of <b>Snakemake</b> makes application of CD-HIT tools easier due to one command launch of pipeline and provides better reproducibility of the results.<li> <b>Snakemake</b> also allows to resume interrupted work and reports percentage of accomplished tasks, making large-scale data processing handier.<li> This can be explained by usage of additional python components which are necessary to run <b>Snakemake</b> pipelines.<li> We developed <b>Snakemake</b> pipeline for OTU-MiSeq utilities, which can be helpful for easier automated runs.<li> [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32792932">32792932</a> <b>(1)</b> Experimental Pipeline (Expipe): A Lightweight Data Management Platform to Simplify the Steps From Experiment to Data Analysis<ul><li> For instance, workflows written in <b>Snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B7" ref-type="bibr"><u>2012</u></xref>) can depend on files in an Expipe structure to define an automated analysis workflow.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33575604">33575604</a> <b>(1)</b> Untangling biological factors influencing trajectory inference from single cell data<ul><li> Analysis scripts can be found as a <b>Snakemake</b> pipeline in <ext-link ext-link-type="uri" xlink:href="https://github.com/mochar/cell_variation">https://github.com/mochar/cell_variation</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32680854">32680854</a> <b>(1)</b> rdmc: An Open Source R Package Implementing Convergent Adaptation Models of Lee and Coop (2017)<ul><li> Making <italic>rdmc</italic> a portable and easy to install R package simplifies running separate genomic regions as independent jobs in parallel using workflows such as <b>Snakemake</b> (<xref rid="bib14" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) or Nextflow (<xref rid="bib5" ref-type="bibr">Di Tommaso <italic>et al.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32663247">32663247</a> <b>(1)</b> TE-greedy-nester: structure-based detection of LTR retrotransposons and their nesting<ul><li> A <b>Snakemake</b> pipeline was created to run TE-greedy-nester on larger datasets and store results of the analysis in GFF files and also in a relational database <ext-link ext-link-type="uri" xlink:href="http://hedron.fi.muni.cz/TEDb/index.html">http://hedron.fi.muni.cz/TEDb/index.html</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32668255">32668255</a> <b>(1)</b> Meta-Analysis of the Alzheimer&#x2019;s Disease Human Brain Transcriptome and Functional Dissection in Mouse Models<ul><li> Next, each computing node was assigned to process one sample using our customized RNA-seq pipeline, as implemented using <b>Snakemake</b> v.4.8.0 (<xref rid="R60" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32674400">32674400</a> <b>(2)</b> Genetic Diversity in Invasive Populations of Argentine Stem Weevil Associated with Adaptation to Biocontrol<ul><li> We used <b>Snakemake</b> [<xref rid="B48-insects-11-00441" ref-type="bibr"><u>48</u></xref>] to arrange analysis steps into workflows and monitor dependencies, and Singularity [<xref rid="B49-insects-11-00441" ref-type="bibr"><u>49</u></xref>] to capture the computing environment.<li> Using the code repositories listed in each methods section, the final results can be reproduced from the raw data with a single command using <b>Snakemake</b> and Singularity.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32660534">32660534</a> <b>(1)</b> Cell type-specific genome scans of DNA methylation divergence indicate an important role for transposable elements<ul><li> The complete pipeline was implemented in <b>Snakemake</b> [<xref ref-type="bibr" rid="CR88"><u>88</u></xref>], mainly using the scientific python stack [<xref ref-type="bibr" rid="CR89"><u>89</u></xref>&#x2013;<xref ref-type="bibr" rid="CR92"><u>92</u></xref>], R visualization libraries [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>, <xref ref-type="bibr" rid="CR93"><u>93</u></xref>, <xref ref-type="bibr" rid="CR94"><u>94</u></xref>], and tools for genome analysis [<xref ref-type="bibr" rid="CR43"><u>43</u></xref>, <xref ref-type="bibr" rid="CR95"><u>95</u></xref>, <xref ref-type="bibr" rid="CR96"><u>96</u></xref>].<li> [88](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [89](          <person-group person-group-type="author">            <name>              <surname>Hunter</surname>              <given-names>JD</given-names>            </name>          </person-group>          <article-title>Matplotlib: a 2D graphics environment</article-title>          <source>Comput Sci Eng</source>          <year>2007</year>          <volume>9</volume>          <issue>3</issue>          <fpage>90</fpage>          <lpage>5</lpage>        ) [92](Waskom M, Botvinnik O, O&#x2019;Kane D, Hobson P, Ostblom J, Lukauskas S, et al.mwaskom/seaborn: Version v0.9.0 Zenodo. 2018. <pub-id pub-id-type="doi">10.5281/zenodo.1313201</pub-id>. Accessed July 2018.) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28645171">28645171</a>) [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24930139">24930139</a>) [94](          <person-group person-group-type="author">            <name>              <surname>Hahne</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Ivanek</surname>              <given-names>R</given-names>            </name>          </person-group>          <article-title>Visualizing genomic data using Gviz and Bioconductor</article-title>          <source>Statistical genomics methods in molecular biology</source>          <year>2016</year>          <publisher-loc>New York</publisher-loc>          <publisher-name>Humana Press</publisher-name>        ) [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [95](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21949271">21949271</a>) [96](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24799436">24799436</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32650716">32650716</a> <b>(1)</b> A database for ITS2 sequences from nematodes<ul><li> markerDB is implemented in the R programming language [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>] and run as a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>] pipeline.<li> [8](          <person-group person-group-type="author">            <collab>R Core Team</collab>          </person-group>          <source>R: A Language and Environment for Statistical Computing</source>          <year>2019</year>          <publisher-loc>Vienna</publisher-loc>          <publisher-name>R Foundation for Statistical Computing</publisher-name>        ) [9](K&#xF6;ster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine; 28(19):2520&#x2013;2. 10.1093/bioinformatics/bts480.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32659381">32659381</a> <b>(1)</b> Muc2 Mucin and Nonmucin Microbiota Confer Distinct Innate Host Defense in Disease Susceptibility and Colonic Injury<ul><li> The entire procedure was run as a <b>Snakemake</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32647059">32647059</a> <b>(1)</b> Prototypical pacemaker neurons interact with the resident microbiota<ul><li>Raw data from scRNA-seq were processed using a <b>snakemake</b> scRNA-seq pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32640986">32640986</a> <b>(1)</b> Bioinformatics recipes: creating, executing and distributing reproducible data analysis workflows<ul><li> A recipe may be a collection of several command-line tools, it may be a Makefile, a <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR3"><u>3</u></xref>] file, a Nextflow [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>] pipeline, an R script or any command line oriented computer program.<li> [3](Federico A, et al. Pipeliner: a Nextflow-based framework for the definition of sequencing data processing pipelines, Front Genet. 2019;10:614.&#xA0;10.3389/fgene.2019.00614.) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34020538">34020538</a> <b>(1)</b> Evaluating assembly and variant calling software for strain-resolved analysis of large DNA viruses<ul><li>Input/output (I/O), memory consumption and run time of assembly programs measured with the <b>snakemake</b> benchmarking function for all samples.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32636332">32636332</a> <b>(1)</b> Genomic Analysis of Bovine <named-content content-type="genus-species">Staphylococcus aureus</named-content> Isolates from Milk To Elucidate Diversity and Determine the Distributions of Antimicrobial and Virulence Genes and Their Association with Mastitis<ul><li> The entire genome assembly process was automated using the <b>Snakemake</b> workflow engine (<xref rid="B109" ref-type="bibr"><u>109</u></xref>).<li> [109](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32621602">32621602</a> <b>(1)</b> Tripal and Galaxy: supporting reproducible scientific workflows for community biological databases<ul><li> Some of these include Kepler (<xref rid="ref7" ref-type="bibr"><u>7</u></xref>), <b>SnakeMake</b> (<xref rid="ref8" ref-type="bibr"><u>8</u></xref>), Nextflow (<xref rid="ref9" ref-type="bibr"><u>9</u></xref>), the Galaxy Project (<xref rid="ref10" ref-type="bibr"><u>10</u></xref>), Pegasus (<xref rid="ref11" ref-type="bibr"><u>11</u></xref>) and others.<li> [7](<person-group person-group-type="author"><name name-style="western"><surname>Lud&#xE4;scher</surname><given-names>B.</given-names></name>, <name name-style="western"><surname>Altintas</surname><given-names>I.</given-names></name>, <name name-style="western"><surname>Berkley</surname><given-names>C.</given-names></name><etal>et&#xA0;al.</etal></person-group> (<year>2006</year>) <article-title>Scientific workflow management and the Kepler system</article-title>. <source>Concurr. Comp-Pract Exp</source>. <volume>18</volume>, <fpage>1039</fpage>&#x2013;<lpage>1065</lpage>.) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [11](<person-group person-group-type="author"><name name-style="western"><surname>Deelman</surname><given-names>E.</given-names></name>, <name name-style="western"><surname>Vahi</surname><given-names>K.</given-names></name>, <name name-style="western"><surname>Juve</surname><given-names>G.</given-names></name><etal>et&#xA0;al.</etal></person-group> (<year>2015</year>) <article-title>Pegasus, a workflow management system for science automation</article-title>. <source>Futur. Gener. Comput. Syst.</source>, <volume>46</volume>, <fpage>17</fpage>&#x2013;<lpage>35</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32719671">32719671</a> <b>(1)</b> First Genome Sequences of Two Multidrug-Resistant <italic>Candida haemulonii</italic> var. <italic>vulnera</italic> Isolates From Pediatric Patients With Candidemia<ul><li> The pipeline was built using <b>Snakemake</b> (<xref rid="B45" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>).<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32620171">32620171</a> <b>(3)</b> Syntrophic acetate oxidation replaces acetoclastic methanogenesis during thermophilic digestion of biowaste<ul><li> Therefore, we built an automated metagenome analysis pipeline embedded in a <b>Snakemake</b> workflow.<li> Along with the qualities of <b>Snakemake</b> [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>], this pipeline is transferable, easily customizable, and scalable for much more complex datasets.<li> In brief, all tools from raw read processing to phylogenetic analysis of final MAGs were embedded and automated in a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>] workflow.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32714870">32714870</a> <b>(1)</b> Integrative Network Fusion: A Multi-Omics Approach in Molecular Profiling<ul><li>The complete INF pipeline is implemented through the workflow management tool <b>Snakemake</b> (<xref rid="B27" ref-type="bibr"><u>27</u></xref>, <xref rid="B28" ref-type="bibr"><u>28</u></xref>), which allows automatic handling of all dependencies required to generate the INF output.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32605926">32605926</a> <b>(1)</b> MutantHuntWGS: A Pipeline for Identifying <italic>Saccharomyces cerevisiae</italic> Mutations<ul><li> It is built in a <b>Snakemake</b> framework (<xref rid="bib17" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) that runs in a Conda environment (<ext-link ext-link-type="uri" xlink:href="https://docs.conda.io/en/latest/">https://docs.conda.io/en/latest/</ext-link>), similar to the container-based analysis environment we used for MutantHuntWGS.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33575596">33575596</a> <b>(2)</b> Factorial study of the RNA-seq computational workflow identifies biases as technical gene signatures<ul><li> The dependencies and parameters for each pipeline steps are accessible in a <b>Snakemake</b> (<xref rid="B36" ref-type="bibr"><u>36</u></xref>) project.<li>All codes used for the analyses described in this work are available as a <b>Snakemake</b> workflow (<ext-link ext-link-type="uri" xlink:href="http://gitlabscottgroup.med.usherbrooke.ca/Joel/permutations-ica">http://gitlabscottgroup.med.usherbrooke.ca/Joel/permutations-ica</ext-link>).<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32553182">32553182</a> <b>(4)</b> Genomic Rewiring of SOX2 Chromatin Interaction Network during Differentiation of ESCs to Postmitotic Neurons<ul><li> RRID:SCR_011848</td>                </tr>                <tr>                  <td><b>Snakemake</b> 5.0.<li> RRID:SCR_011848</td>                </tr>                <tr>                  <td><b>Snakemake</b> 5.0</td>                  <td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1093/bioinformatics/bts480" id="intref0225">https://doi.org/10.1093/bioinformatics/bts480</ext-link> (<xref rid="bib36" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>)</td>                  <td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/" id="intref0230">https://<b>snakemake</b>.readthedocs.io/</ext-link>; RRID:SCR_003475.<li>ATAC-seq data were processed following the steps in (<xref rid="bib15" ref-type="bibr"><u>Buenrostro et&#xA0;al., 2013</u></xref>) using the custom <b>Snakemake</b> (<xref rid="bib36" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) pipeline.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Buenrostro et&#xA0;al., 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24097267">24097267</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32573438">32573438</a> <b>(3)</b> A community-maintained standard library of population genetic models<ul><li> To reproducibly evaluate and compare the performance of inference methods, we developed workflows using <monospace><b>snakemake</b></monospace> (<xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), available from <ext-link ext-link-type="uri" xlink:href="https://github.com/popsim-consortium/analysis">https://github.com/popsim-consortium/analysis</ext-link>, that allow efficient computing in multicore or cluster environments.<li>To demonstrate the utility of <monospace>stdpopsim</monospace> we created <monospace><b>Snakemake</b></monospace> workflows (<xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) that perform demographic inference on tree sequence output from our package using a few common software packages (see <xref ref-type="fig" rid="app1fig10"><u>Appendix 1&#x2014;figure 10</u></xref> for an example workflow).<li> Our choice of <monospace><b>Snakemake</b></monospace> allows complete reproducibility of the analyses shown, and all code is available from <ext-link ext-link-type="uri" xlink:href="https://github.com/popsim-consortium/analysis">https://github.com/popsim-consortium/analysis</ext-link>.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> A community-maintained standard library of population genetic models<ul><li> I did however notice that the authors have provided their analyses as <b>Snakemake</b> files in the interest of replication.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> A community-maintained standard library of population genetic models<ul><li> I did however notice that the authors have provided their analyses as <b>Snakemake</b> files in the interest of replication.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32571209">32571209</a> <b>(4)</b> ATLAS: a <b>Snakemake</b> workflow for assembly, annotation, and genomic binning of metagenome sequence data<ul><li> The internal use of <b>Snakemake</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] allows efficient and automated deployment on a computing cluster.<li> ATLAS is implemented in Python and uses the <b>Snakemake</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] workflow manager for extensive control of external tools, including versioning of configurations and environments, provenance capabilities, and scalability on high-performance computing clusters.<li> Definition of which samples are likely to contain the same bacterial species, via a group attribute in the <b>Snakemake</b> configuration file, supports binning based on co-abundance patterns across samples.<li>Programming language: <b>Snakemake</b>/Python.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32571206">32571206</a> <b>(1)</b> Whole genome characterization of autochthonous <italic>Bos taurus brachyceros</italic> and introduced <italic>Bos indicus indicus</italic> cattle breeds in Cameroon regarding their adaptive phenotypic traits and pathogen resistance<ul><li> To have many of these processes parallelized and automated, a workflow written in the workflow language <b>Snakemake</b> from QbiC was used which is freely available at Github (https://github.com/qbicsoftware/exomseq).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32555605">32555605</a> <b>(1)</b> BiomMiner: An advanced exploratory microbiome analysis and visualization pipeline<ul><li>BiomMiner uses <b>Snakemake</b> [<xref rid="pone.0234860.ref010" ref-type="bibr"><u>10</u></xref>] as the primary workflow management language for scalability and reproducible execution of various wrapper scripts developed in python and R for existing software tools.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32556167">32556167</a> <b>(25)</b> Watchdog 2.0: New developments for reusability, reproducibility, and workflow execution<ul><li> Popular WMSs are Galaxy [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], KNIME [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], <b>Snakemake</b> [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], and Nextflow [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] and differ in the implemented set of features, target audience, the required training period, usage fees, and more (for more details see the comparison in the first article on Watchdog [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] and at the end of this article).<li> The previously published version of Watchdog [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] was already extensively compared against the most popular WMSs for biological analyses, i.e., Galaxy (Galaxy, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_006281">RRID:SCR_006281</ext-link>)&#xA0;[<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], KNIME (KNIME, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_006164">RRID:SCR_006164</ext-link>)&#xA0;[<xref rid="bib9" ref-type="bibr"><u>9</u></xref>], <b>Snakemake</b> (<b>Snakemake</b>, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_003475">RRID:SCR_003475</ext-link>)&#xA0;[<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], and Nextflow [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] (see Fig.<li> First, we provide a brief description of Galaxy, KNIME, <b>Snakemake</b>, and Nextflow.<li><b>Snakemake</b> uses a Python-based language to define workflows in a so-called Snakefile as a set of rules that describe how output files are created from input files.<li> Writing workflows and wrappers requires knowledge of the <b>Snakemake</b> syntax and some degree of programming skills.<li> Similar to <b>Snakemake</b>, programming experience is required to define workflows and no GUI is provided.<li><b>Snakemake</b></th>                <th align="left" rowspan="1" colspan="1">Nextflow</th>              </tr>            </thead>            <tbody>              <tr>                <td rowspan="1" colspan="1">Reusability.<li>For this part of the comparison, we focused on features that support development and sharing of tools (modules in Watchdog, tools in Galaxy, nodes in KNIME, rules in <b>Snakemake</b>, processes in Nextflow) for (re-)use in multiple analysis workflows as well as sharing and repurposing of existing workflows (F1&#x2013;F7 in Table&#xA0;<xref rid="tbl1" ref-type="table"><u>1</u></xref>).<li> <b>Snakemake</b> does not provide any software or script for defining wrappers.<li> In the case of <b>Snakemake</b>, the specification does not require explicit documentation of parameters and input and output.<li> A reference book containing information on all available tools (F3) can be generated for <b>Snakemake</b> wrappers as a separate web page.<li>For sharing tools (F5) or complete workflows (F6) with other users, Galaxy and KNIME operate dedicated sharing platforms [<xref rid="bib23" ref-type="bibr"><u>28</u></xref>,<xref rid="bib24" ref-type="bibr"><u>29</u></xref>], while <b>Snakemake</b> provides source code repositories similar to Watchdog 2.0 [<xref rid="bib25" ref-type="bibr"><u>31</u></xref>,<xref rid="bib26" ref-type="bibr"><u>32</u></xref>].<li> For <b>Snakemake</b>, relative paths to input files are hard-coded in the Snakefile.<li> Thus, repurposing a <b>Snakemake</b> workflow only requires the Snakefile to be copied to a directory in which input files are stored or linked in the subdirectory structure used in the Snakefile.<li> This is not more effort than required by other WMSs, with the exception of <b>Snakemake</b>.<li> However, it provides more flexibility than <b>Snakemake</b> regarding how input data are distributed in the file system, and workflows can be stored anywhere, e.g., in a directory containing all previously developed workflows.<li> However, Galaxy, <b>Snakemake</b>, and Nextflow support controlling external software dependencies and versions with the Conda package manager or using Docker containers (F9).<li> Furthermore, <b>Snakemake</b> reports on executed workflows (see next paragraph) display the Conda environment for each task, including software versions.<li>A description of all performed analysis steps (F10) can be obtained in <b>Snakemake</b> and Nextflow through generation of HTML reports, in which individual steps are listed in a table format and in the case of <b>Snakemake</b> also visualized as a graph.<li> With <b>Snakemake</b> and Nextflow, new samples (e.g., additional replicates) can be included in an analysis workflow without having to reprocess all samples (F14), but this option has to be forcibly triggered in <b>Snakemake</b>.<li> Finally, <b>Snakemake</b> provides the option to stop scheduling by sending the TERM signal and waiting for all jobs to be finished before terminating.<li> Alternatively, Ctrl+C kills the main <b>Snakemake</b> process and all jobs running on the local computer, but jobs already running on a computing cluster keep running.<li> [8](<person-group person-group-type="author"><name name-style="western"><surname>Taylor</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Schenck</surname><given-names>I</given-names></name>, <name name-style="western"><surname>Blankenberg</surname><given-names>D</given-names></name>, <etal>et al</etal>.</person-group><article-title>Using Galaxy to perform large-scale interactive data analyses</article-title>. <source>Curr Protoc Bioinformatics</source>. <year>2007</year>;<volume>Chapter 10</volume>:<fpage>Unit 10.5</fpage>.) [9](<person-group person-group-type="author"><name name-style="western"><surname>Berthold</surname><given-names>MR</given-names></name>, <name name-style="western"><surname>Cebron</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Dill</surname><given-names>F</given-names></name>, <etal>et al</etal>.</person-group><article-title>KNIME: The Konstanz Information Miner</article-title>. In: <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>. <publisher-loc>Heidelberg-Berlin</publisher-loc>: <publisher-name>Springer</publisher-name>; <year>2007</year>:<fpage>319</fpage>&#x2013;<lpage>26</lpage>.) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a>) [8](<person-group person-group-type="author"><name name-style="western"><surname>Taylor</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Schenck</surname><given-names>I</given-names></name>, <name name-style="western"><surname>Blankenberg</surname><given-names>D</given-names></name>, <etal>et al</etal>.</person-group><article-title>Using Galaxy to perform large-scale interactive data analyses</article-title>. <source>Curr Protoc Bioinformatics</source>. <year>2007</year>;<volume>Chapter 10</volume>:<fpage>Unit 10.5</fpage>.) [9](<person-group person-group-type="author"><name name-style="western"><surname>Berthold</surname><given-names>MR</given-names></name>, <name name-style="western"><surname>Cebron</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Dill</surname><given-names>F</given-names></name>, <etal>et al</etal>.</person-group><article-title>KNIME: The Konstanz Information Miner</article-title>. In: <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>. <publisher-loc>Heidelberg-Berlin</publisher-loc>: <publisher-name>Springer</publisher-name>; <year>2007</year>:<fpage>319</fpage>&#x2013;<lpage>26</lpage>.) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [28](<comment>Galaxy Tool Shed. <ext-link ext-link-type="uri" xlink:href="https://toolshed.g2.bx.psu.edu">https://toolshed.g2.bx.psu.edu</ext-link>. Accessed 11 November 2019</comment>.) [29](<comment>KNIME Hub. <ext-link ext-link-type="uri" xlink:href="https://hub.knime.com">https://hub.knime.com</ext-link>. Accessed 11 November 2019</comment>.) [31](<comment>SnakeMake Wrappers repository. <ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/snakemake/snakemake-wrappers">https://bitbucket.org/snakemake/snakemake-wrappers</ext-link>. Accessed 11 November 2019</comment>.) [32](<comment>SnakeMake Workflows repository. <ext-link ext-link-type="uri" xlink:href="https://github.com/snakemake-workflows">https://github.com/snakemake-workflows</ext-link>. Accessed 11 November 2019</comment>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32545904">32545904</a> <b>(1)</b> Cotargeting of XPO1 Enhances the Antileukemic Activity of Midostaurin and Gilteritinib in Acute Myeloid Leukemia<ul><li>Analysis of resultant FASTQ utilized <b>Snakemake</b> [<xref rid="B28-cancers-12-01574" ref-type="bibr"><u>28</u></xref>] to organize a pipeline for analysis, skewer [<xref rid="B29-cancers-12-01574" ref-type="bibr"><u>29</u></xref>] to trim, and MAGeCK [<xref rid="B13-cancers-12-01574" ref-type="bibr"><u>13</u></xref>,<xref rid="B30-cancers-12-01574" ref-type="bibr"><u>30</u></xref>,<xref rid="B31-cancers-12-01574" ref-type="bibr"><u>31</u></xref>] for count and robust rank algorithm analysis.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [29](          <person-group person-group-type="author">            <name>              <surname>Jiang</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Lei</surname>              <given-names>R.</given-names>            </name>            <name>              <surname>Ding</surname>              <given-names>S.-W.</given-names>            </name>            <name>              <surname>Zhu</surname>              <given-names>S.</given-names>            </name>          </person-group>          <article-title>Skewer: A fast and accurate adapter trimmer for next-generation sequencing paired-end reads</article-title>          <source>BMC Bioinform.</source>          <year>2014</year>          <volume>15</volume>          <elocation-id>182</elocation-id>          <pub-id pub-id-type="doi">10.1186/1471-2105-15-182</pub-id>        ) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30710114">30710114</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25476604">25476604</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26673418">26673418</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32514171">32514171</a> <b>(1)</b> Cartography of opportunistic pathogens and antibiotic resistance genes in a tertiary hospital environment<ul><li>Illumina shotgun metagenomic sequencing reads were processed using a <b>Snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/gis-rpd/pipelines/tree/master/metagenomics/shotgun-metagenomics">https://github.com/gis-rpd/pipelines/tree/master/metagenomics/shotgun-metagenomics</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32517321">32517321</a> <b>(1)</b> Comparative Transcriptomics Reveals Gene Families Associated with Predatory Behavior in <italic>Photuris femme fatale</italic> Fireflies<ul><li> Briefly, FUSTr uses the engine workflow <b>SnakeMake</b> [<xref rid="B37-genes-11-00627" ref-type="bibr"><u>37</u></xref>] to automate the following pipeline: it first predicts ORFs using Transdecoder [<xref rid="B38-genes-11-00627" ref-type="bibr"><u>38</u></xref>], retaining the longest isoform.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23845962">23845962</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32512857">32512857</a> <b>(1)</b> Whole Genome Sequencing Analysis of Porcine Faecal Commensal <italic>Escherichia coli</italic> Carrying Class 1 Integrons from Sows and Their Offspring<ul><li>The short read alignment of our strains to K12-MG1655 (ST10) as a reference genome was performed as previously described using the custom <b>snakemake</b> workflow snplord, available at <uri xlink:href="https://github.com/CJREID/snplord">https://github.com/CJREID/snplord</uri>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32497118">32497118</a> <b>(3)</b> HemoMIPs&#x2014;Automated analysis and result reporting pipeline for targeted sequencing data<ul><li> Using an easily adapted <b>Snakemake</b> workflow [<xref rid="pcbi.1007956.ref007" ref-type="bibr"><u>7</u></xref>], hemoMIPs performs sample demultiplexing, overlap paired-end merging, alignment using BWA, MIP-arm trimming, variant calling using GATK, coverage analysis and HTML report generation for single end and paired end sequencing datasets.<li> All steps are implemented in the workflow management software <b>Snakemake</b> [<xref rid="pcbi.1007956.ref007" ref-type="bibr"><u>7</u></xref>] and rely on Conda predefined environments to manage software dependencies and easy deployment.<li> The hemoMIPs workflow can be easily adapted to other MIP designs due to the use of <b>Snakemake</b> as workflow management and Conda for managing software dependencies.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32497070">32497070</a> <b>(2)</b> A cost-effective approach to DNA methylation detection by Methyl Sensitive DArT sequencing<ul><li> The analytical steps are coordinated and executed by the <b>Snakemake</b> workflow management system [<xref rid="pone.0233800.ref072" ref-type="bibr"><u>72</u></xref>].<li> Scripts and the <b>snakemake</b> configuration files are available in GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/wendelljpereira/ms-dart-seq">https://github.com/wendelljpereira/ms-dart-seq</ext-link>.<li> [72](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32484158">32484158</a> <b>(2)</b> Diverse homeostatic and immunomodulatory roles of immune cells in the developing mouse lung at single cell resolution<ul><li> To coordinate mapping and counting on Stanford&#x2019;s high-performance computing cluster, <ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/"><b>snakemake</b></ext-link> was used (<xref rid="bib51" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479592">32479592</a> <b>(4)</b> Recommendations to enhance rigor and reproducibility in biomedical research<ul><li><b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_003475">RRID:SCR_003475</ext-link>) is a tool to create reproducible and scalable data analysis workflows, with a language based on Python.<li> <b>Snakemake</b> makes it easier to execute data analyses on different environments without modification of the workflow definition.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32490244">32490244</a> <b>(1)</b> Profiling and quantification of pluripotency reprogramming reveal that WNT pathways and cell morphology have to be reprogramed extensively<ul><li> The FASTQ files were uploaded to the UAB High Performance Computer cluster for bioinformatics analysis with the following custom pipeline built in the <b>Snakemake</b> workflow system (v5.2.2) (<xref rid="bib18" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>): first, quality and control of the reads were assessed using FastQC, and trimming of the bases with quality scores of less than 20 was performed with Trim_Galore! (v0.4.5).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><u>No PMID</u> <b>(1)</b> APE: A Command-Line Tool and API for Automated Workflow Composition<ul><li> The mapping process described in this paper can furthermore serve as a template for the translation of APE results to other workflow formats, such as NextFlow&#xA0;[<xref ref-type="bibr" rid="CR7"><u>7</u></xref>], <b>SnakeMake</b>&#xA0;[<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] or the Workflow Description Language (WDL)&#xA0;[<xref ref-type="bibr" rid="CR3"><u>3</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [3](Workflow Description Language (WDL), April 2020. <ext-link ext-link-type="uri" xlink:href="https://github.com/openwdl/wdl">https://github.com/openwdl/wdl</ext-link>, original-date: 2012&#x2013;08-01T03:12:48Z) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32451505">32451505</a> <b>(3)</b> GPSeq reveals the radial organization of chromatin in the cell nucleus<ul><li> This analysis was implemented as a <b>snakemake</b> flow<sup><xref rid="R63" ref-type="bibr"><u>63</u></xref></sup>, available at <ext-link ext-link-type="uri" xlink:href="http://github.com/ggirelli/gpseqc-<b>snakemake</b>">http://github.com/ggirelli/gpseqc-<b>snakemake</b></ext-link>.<li> [63](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinforma Oxf Engl</source>          <year>2018</year>          <volume>34</volume>          <fpage>3600</fpage>        ) </ul></p>
<p><u>No PMID</u> <b>(1)</b> Scaling Genomics Data Processing with Memory-Driven Computing to Accelerate Computational Biology<ul><li> This has lead to the development of specialized bioinformatics pipelines and dedicated workflow managers like <b>Snakemake</b> [<xref ref-type="bibr" rid="CR22"><u>22</u></xref>].<li> [22](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <year>2012</year>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32413257">32413257</a> <b>(1)</b> Design and Experimental Evaluation of a Minimal, InnocuousWatermarking Strategy to Distinguish Near-Identical DNA and RNA Sequences<ul><li>The marker discoveryand quantification tools were developed <italic>in scala</italic>,and the entire pipeline is implemented in pythonusing <b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32414048">32414048</a> <b>(1)</b> Sorghum Growth Promotion by <italic>Paraburkholderia tropica</italic> and <italic>Herbaspirillum frisingense</italic>: Putative Mechanisms Revealed by Genomics and Metagenomics<ul><li> The pipeline is available as a <b>Snakemake</b> v4.7.0 pipeline on request from the NIOO-KNAW GitLab.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32490035">32490035</a> <b>(1)</b> Attenuation of Heparan Sulfate Proteoglycan Binding Enhances <italic>In&#xA0;Vivo</italic> Transduction of Human Primary Hepatocytes with AAV2<ul><li> To process reads and count barcodes we used a <b>Snakemake</b> (5.6) pipeline.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32374865">32374865</a> <b>(2)</b> miEAA 2.0: integrating multi-species microRNA enrichment analysis and workflow management systems<ul><li> To unify data preprocessing, we implemented an automated pipeline using <b>Snakemake</b> (<xref rid="B26" ref-type="bibr"><u>26</u></xref>), Python 3.6, and the pandas (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) Python package facilitating data collection and filtering steps.<li> First, reproducibility of statistical experiments can be improved, because usage of the versioned API in the context of a workflow manager such as <b>Snakemake</b> (<xref rid="B26" ref-type="bibr"><u>26</u></xref>) or Nextflow fosters self-documenting research setups (<xref rid="B43" ref-type="bibr"><u>43</u></xref>).<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32370214">32370214</a> <b>(1)</b> A PIANO (Proper, Insufficient, Aberrant, and NO Reprogramming) Response to the Yamanaka Factors in the Initial Stages of Human iPSC Reprogramming<ul><li> The FASTQ files were uploaded to the UAB High-Performance Computer cluster for bioinformatics analysis with the following custom pipeline built in the <b>Snakemake</b> workflow system (v5.2.2) [<xref rid="B23-ijms-21-03229" ref-type="bibr"><u>23</u></xref>]: initial quality and control of the reads were quantified with FastQC, and trimming of the bases with quality scores of less than 20 was executed with Trim_Galore! (v0.4.5).<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32357829">32357829</a> <b>(6)</b> wg-blimp: an end-to-end analysis pipeline for whole genome bisulfite sequencing data<ul><li>We base the wg-blimp pipeline on the workflow execution system <b>Snakemake</b> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] as it enables robust and scalable execution of analysis pipelines and prevents generation of faulty results in case of failure.<li> <b>Snakemake</b> also provides run-time and memory usage logging, thus easing the search for bottlenecks and performance optimization.<li> For workflow management, we prefer using dedicated workflow management systems such as <b>Snakemake</b> or Nextflow over plain shell scripts, as these allow more scalable and robust execution.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li> [16](K&#xF6;ster J, Rahmann S. Snakemake&#x2013;a scalable bioinformatics workflow engine. Bioinformatics (Oxford, England). 2012; 28(19):2520&#x2013;2. 10.1093/bioinformatics/bts480.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32345333">32345333</a> <b>(5)</b> Varlociraptor: enhancing sensitivity and controlling false discovery rate in somatic indel discovery<ul><li>The evaluation performed in this paper is available as a reproducible <b>Snakemake</b> [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] workflow archive (10.5281/zenodo.3361700).<li> In addition, we provide a <italic><b>Snakemake</b> report</italic> (Additional file&#xA0;<xref rid="MOESM2" ref-type="media"><u>2</u></xref>) that allows to interactively explore all figures shown in this article in the context of the workflow, the parameters, and the code used to generate them.<li> We provide a reproducible <b>Snakemake</b> workflow conducting all necessary steps to generate the data (10.5281/zenodo.3630241).<li><bold>Additional file 2</bold> <b>Snakemake</b> Report: A <italic><b>Snakemake</b> report</italic> that allows to interactively explore all figures shown in this article in the context of workflow, parameters, software versions, and code.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32368312">32368312</a> <b>(1)</b> Development and validation of next generation sequencing based 35-gene hereditary cancer panel<ul><li> <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], a workflow management framework, were implemented to control the execution of the pipeline.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32338743">32338743</a> <b>(1)</b> VarFish: comprehensive DNA variant analysis for diagnostics and research<ul><li> We provide the full <b>Snakemake</b> (<xref rid="B30" ref-type="bibr"><u>30</u></xref>) workflow for downloading the data from open and free sources for reproducibility.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32426295">32426295</a> <b>(1)</b> Cystic Fibrosis Patients Infected With Epidemic <italic>Pseudomonas aeruginosa</italic> Strains Have Unique Microbial Communities<ul><li> Sequences were processed using a custom <b>Snakemake</b> pipeline (Timsit et al., <xref rid="B64" ref-type="bibr"><u>2016</u></xref>).<li> [2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27066712">27066712</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32483558">32483558</a> <b>(4)</b> Clonal Evolution and Heterogeneity of Osimertinib Acquired Resistance Mechanisms in EGFR Mutant Lung Cancer<ul><li>                    <hr/>                  </td>                </tr>                <tr>                  <td>Processed data and code used</td>                  <td>This paper</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://github.com/aleighbrown/pwgs_<b>snakemake</b>" id="intref0020">https://github.com/aleighbrown/pwgs_<b>snakemake</b></ext-link>                  .<li>The accession number for the sequencing data reported in this paper is dbGaP: phs002001 Data analysis pipeline is available at: <ext-link ext-link-type="uri" xlink:href="https://github.com/aleighbrown/pwgs_<b>snakemake</b>" id="intref0090">https://github.com/aleighbrown/pwgs_<b>snakemake</b></ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32306927">32306927</a> <b>(2)</b> DolphinNext: a distributed data processing platform for high throughput genomics<ul><li>Comparison of related applications</p></caption><table frame="hsides" rules="groups"><thead><tr><th/><th>DolphinNext</th><th>Galaxy [<xref ref-type="bibr" rid="CR5"><u>5</u></xref>]</th><th>Sequanix [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>]</th><th>Taverna [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>]</th><th>Arvados [<xref ref-type="bibr" rid="CR63"><u>63</u></xref>]</th></tr></thead><tbody><tr><td>Platform<sup>a</sup></td><td>JS/PHP</td><td>Python</td><td>Python</td><td>Java</td><td>Go</td></tr><tr><td>Workflow management system</td><td>Nextflow</td><td>Galaxy</td><td><b>Snakemake</b>.<li> Although Nextflow is a very successful and widely used pipeline engine, supporting other platforms like <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR64"><u>64</u></xref>] and WDL based execution engines [<xref ref-type="bibr" rid="CR65"><u>65</u></xref>] would increase the usage of DolphinNext.<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29361152">29361152</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [63](Amstutz P. Portable, Reproducible Analysis with Arvados. F1000Res. 2015;4 <ext-link ext-link-type="uri" xlink:href="https://f1000research.com/assets/download/1110114">https://f1000research.com/assets/download/1110114</ext-link>.) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [65](          <person-group person-group-type="author">            <name>              <surname>Chapman</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Gentry</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lin</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Magee</surname>              <given-names>P</given-names>            </name>            <name>              <surname>O&#x2019;Connor</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Prabhakaran</surname>              <given-names>A</given-names>            </name>            <etal/>          </person-group>          <source>OpenWDL</source>          <year>2019</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015097">32015097</a> <b>(1)</b> Concentration-Dependent Recruitment of Mammalian Odorant Receptors<ul><li> <b>Snakemake</b> v3.5.5 (<xref rid="B36" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) was used to process read files through alignment and quantification.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32295881">32295881</a> <b>(2)</b> Robust landscapes of ribosome dwell times and aminoacyl-tRNAs in response to nutrient stress in liver<ul><li>Codes to model DTs and flux from ribosome profiling are available on GitHub: <ext-link ext-link-type="uri" xlink:href="https://github.com/cgob/codonDT_<b>snakemake</b>" xlink:show="new">https://github.com/cgob/codonDT_<b>snakemake</b></ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32269224">32269224</a> <b>(1)</b> Cysteine synthases CYSL-1 and CYSL-2 mediate <italic>C. elegans</italic> heritable adaptation to <italic>P. vranovensis</italic> infection<ul><li> A <b>snakemake</b> workflow<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup> was created for the RNA-seq analysis and can it be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/cristianriccio/celegans-pathogen-adaptation">https://github.com/cristianriccio/celegans-pathogen-adaptation</ext-link>.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32318338">32318338</a> <b>(1)</b> Computational Oncology in the Multi-Omics Era: State of the Art<ul><li> Examples of these systems include Galaxy (<xref rid="B21" ref-type="bibr"><u>21</u></xref>), <b>Snakemake</b> (<xref rid="B22" ref-type="bibr"><u>22</u></xref>), Nextflow (<xref rid="B23" ref-type="bibr"><u>23</u></xref>), and the general purpose Common Workflow Language (<xref rid="B24" ref-type="bibr"><u>24</u></xref>).<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P</given-names></name><name><surname>Crusoe</surname><given-names>MR</given-names></name><name><surname>Tijani&#x107;</surname><given-names>N</given-names></name><name><surname>Chapman</surname><given-names>B</given-names></name><name><surname>Chilton</surname><given-names>J</given-names></name><name><surname>Heuer</surname><given-names>M</given-names></name><etal/></person-group><article-title>Common Workflow Language, v1.0</article-title>. (<year>2016</year>). Available online at: <ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32241255">32241255</a> <b>(1)</b> EpiMOLAS: an intuitive web-based framework for genome-wide DNA methylation analysis<ul><li>Docker|Galaxy web server</td><td>Docker</td><td>Shell script</td><td>Bioconda <b>Snakemake</b>.</ul></p>
</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32296416">32296416</a> <b>(1)</b> JunB Controls Intestinal Effector Programs in Regulatory T Cells<ul><li> All steps of this pipeline were performed using custom scripts written for the <b>Snakemake</b> workflow manager (<xref rid="B62" ref-type="bibr"><u>62</u></xref>).<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32223898">32223898</a> <b>(1)</b> Measuring the distribution of fitness effects in somatic evolution by combining clonal dynamics with dN/dS ratios<ul><li>Code to reproduce all the figures in the manuscript (using a <b>snakemake</b> [<xref rid="bib20" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>] workflow) is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/marcjwilliams1/dnds-clonesize">github.com/marcjwilliams1/dnds-clonesize</ext-link> (<xref rid="bib52" ref-type="bibr"><u>Williams, 2020</u></xref>; copy archived at <ext-link ext-link-type="uri" xlink:href="https://elifesciences-publications/dnds-clonesize">https://elifesciences-publications/dnds-clonesize</ext-link>).&#xA0;We also created a singularity image with all software dependencies which is available at shub://marcjwilliams1/dnds-clonesize-R-container.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Williams, 2020](          <person-group person-group-type="author">            <name>              <surname>Williams</surname>              <given-names>MJ</given-names>            </name>          </person-group>          <year>2020</year>          <data-title>dnds-clonesize</data-title>          <version designator="4b7a694">4b7a694</version>          <publisher-name>Github</publisher-name>          <ext-link ext-link-type="uri" xlink:href="https://github.com/marcjwilliams1/dnds-clonesize">https://github.com/marcjwilliams1/dnds-clonesize</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32214316">32214316</a> <b>(2)</b> Seven quick tips for analysis scripts in neuroimaging<ul><li> For example, build systems that are optimized for creating data analysis pipelines include <ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/"><b>snakemake</b></ext-link>, <ext-link ext-link-type="uri" xlink:href="https://pydoit.org/">pydoit</ext-link>, <ext-link ext-link-type="uri" xlink:href="https://luigi.readthedocs.io/">luigi</ext-link>, and <ext-link ext-link-type="uri" xlink:href="https://nipype.readthedocs.io/en/latest">nipype</ext-link> [<xref rid="pcbi.1007358.ref026" ref-type="bibr"><u>26</u></xref>].<li> [26](<name><surname>Gorgolewski</surname><given-names>KJ</given-names></name>, <name><surname>Esteban</surname><given-names>O</given-names></name>, <name><surname>Ellis</surname><given-names>DG</given-names></name>, <name><surname>Notter</surname><given-names>MP</given-names></name>, <name><surname>Ziegler</surname><given-names>E</given-names></name>, <name><surname>Johnson</surname><given-names>H</given-names></name>, <etal>et al</etal><article-title>Nipy/Nipype: Release 0.13.1</article-title>. <year>2017</year>; <pub-id pub-id-type="doi">10.5281/zenodo.581704</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32183729">32183729</a> <b>(8)</b> RASflow: an RNA-Seq analysis workflow with <b>Snakemake</b><ul><li>In this article, we present RNA-Seq Analysis <b>Snakemake</b> Workflow (RASflow) that is usable for a wide range of applications.<li> Using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR23"><u>23</u></xref>], the whole analysis is completely reproducible and highly user-friendly also for users with limited programming skills.<li>Overview of the steps performed by RNA-Seq Analysis <b>Snakemake</b> Workflow (RASflow).<li><b>Snakemake</b> as framework.<li><b>Snakemake</b> is a scalable workflow engine that helps to manage workflows in an easy way.<li> While the remaining workflows do not provide graphical interfaces, they use <b>Snakemake</b> to manage all the steps in the workflow, making them easy to use also for those with limited programming skills.<li>RASflow is built on the basis of Conda and <b>Snakemake</b>, making installation and management very easy.<li> The whole workflow is defined by rules managed by <b>Snakemake</b>, which makes it highly modular.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256520">32256520</a> <b>(1)</b> MasterOfPores: A Workflow for the Analysis of Oxford Nanopore Direct RNA Sequencing Datasets<ul><li> During the last decade, an increasing interest in the field has led to the development of different programs such as <b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), NextFlow (<xref rid="B12" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), Galaxy (<xref rid="B1" ref-type="bibr"><u>Afgan et al., 2018</u></xref>), SciPipe (<xref rid="B23" ref-type="bibr"><u>Lampa et al., 2019</u></xref>) or GenPipes (<xref rid="B6" ref-type="bibr"><u>Bourgey et al., 2019</u></xref>), among others.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Afgan et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [Lampa et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) [Bourgey et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31185495">31185495</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32156824">32156824</a> <b>(1)</b> Humans Are Selectively Exposed to <named-content content-type="genus-species">Pneumocystis jirovecii</named-content><ul><li> Pipelines were written using <b>Snakemake</b> (<xref rid="B39" ref-type="bibr"><u>39</u></xref>).<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32144496">32144496</a> <b>(3)</b> CBPtools: a Python package for regional connectivity-based parcellation<ul><li><italic>CBPtools</italic> makes use of <b>snakemake</b> (K&#xF6;ster and Rahmann <xref ref-type="bibr" rid="CR42"><u>2012</u></xref>), an easy-to-use and well documented workflow management system with parallel processing capabilities that allows the workflow execution to be scaled to various processing environments (i.e., server, cluster, grid, or cloud environments).<li> Through <b>snakemake</b>, <italic>CBPtools</italic> is compatible with job schedulers that support shell script (such as SLURM and qsub).<li> The combination of <b>snakemake</b>&#x2019;s command line execution and an easily modifiable configuration file make it possible to set up and run the software without any programming knowledge.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32140724">32140724</a> <b>(1)</b> Successful field performance in warm and dry environments of soybean expressing the sunflower transcription factor HB4<ul><li> This analysis pipeline was run with the aid of the <b>Snakemake</b> workflow engine (<xref rid="CIT0035" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>).<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32257349">32257349</a> <b>(1)</b> On the relations of phase separation and Hi-C maps to epigenetics<ul><li> The entire pipeline was deployed using <b>Snakemake</b> [<xref rid="RSOS191976C251" ref-type="bibr"><u>251</u></xref>].<li> [251](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31953333">31953333</a> <b>(1)</b> Metagenomic Shotgun Analyses Reveal Complex Patterns of Intra- and Interspecific Variation in the Intestinal Microbiomes of Codfishes<ul><li> Paired-end reads from each individual fish were mapped to the reference genomes (Table S3) using the <b>Snakemake</b> workflow (<xref rid="B95" ref-type="bibr"><u>95</u></xref>) of anvi&#x2019;o (ver.<li> [95](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32256504">32256504</a> <b>(1)</b> Performance of Mapping Approaches for Whole-Genome Bisulfite Sequencing Data in Crop Plants<ul><li> For automatization we implemented a <b>snakemake</b> pipeline (<xref rid="B14" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), shown in <xref ref-type="fig" rid="f3"><bold>Figure 3</bold></xref>.<li> [K&#xF6;ster and Rahmann, 2012](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J.</given-names></name><name><surname>Rahmann</surname><given-names>S.</given-names></name></person-group> (<year>2012</year>). <article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>. <source>Bioinf. (Oxford England)</source><volume>28</volume> (<issue>19</issue>), <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32098056">32098056</a> <b>(1)</b> Comparative Transcriptome Analysis of Two Cucumber Cultivars with Different Sensitivity to Cucumber Mosaic Virus Infection<ul><li> Data analysis processing was automated using pipelines implemented in the SnakeLines framework (manuscript in preparation) running on the <b>Snakemake</b> workflow engine (version 5.2.2) [<xref rid="B110-pathogens-09-00145" ref-type="bibr"><u>110</u></xref>].<li> [110](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32084206">32084206</a> <b>(1)</b> The stochastic nature of errors in next-generation sequencing of circulating cell-free DNA<ul><li>To generate uncorrected nonconsensus alignments, paired end fastq datasets were aligned to GRCh37 using a standard BWA mem, Picard MarkDuplicate, and GATK polishing <b>snakemake</b> workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/HuntsmanCancerInstitute/Workflows/blob/master/Alignment/alignQC_1.3.svgsee">https://github.com/HuntsmanCancerInstitute/Workflows/blob/master/Alignment/alignQC_1.3.svgsee</ext-link> also alignQC_1.3.sm and alignQC_1.3.sh).</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32076265">32076265</a> <b>(1)</b> &#x3B3;&#x3B4;T cells and adipocyte IL-17RC control fat innervation and thermogenesis<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline<sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup>.<li> [42](R42) <li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32071071">32071071</a> <b>(4)</b> BlobToolKit &#x2013; Interactive Quality Assessment of Genome Assemblies<ul><li>BlobToolKit is comprised of four distinct components: BlobTools2 (command line tools to create and filter datasets), Specification (a formal specification and validator for the JSON-based data format), Viewer (interactive dataset visualization), and INSDC-pipeline (a <b>Snakemake</b> pipeline to run the BlobToolKit workflow on publicly available datasets).<li>INSDC-pipeline is a reusable <b>Snakemake</b> (<xref rid="bib21" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2018</u></xref>) pipeline to run analyses on publicly available, International Nucleotide Sequence Database Collaboration (INSDC; <ext-link ext-link-type="uri" xlink:href="http://www.insdc.org/">http://www.insdc.org/</ext-link>) public eukaryotic genome assemblies.<li>Depiction of the <b>snakemake</b> workflow used to analyze publicly available (INSDC-registered) eukaryotic genome assemblies.<li> Each box represents a <b>Snakemake</b> rule that may be run one or more times during workflow execution.<li> [K&#xF6;ster and Rahmann 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32117244">32117244</a> <b>(1)</b> Fc Receptor-Like 6 (FCRL6) Discloses Progenitor B Cell Heterogeneity That Correlates With Pre-BCR Dependent and Independent Pathways of Natural Antibody Selection<ul><li> The FASTQ files were uploaded to the UAB High Performance Computer cluster for bioinformatics analysis with the following custom pipeline built in the <b>Snakemake</b> workflow system (v4.8.0) (<xref rid="B60" ref-type="bibr"><u>60</u></xref>): first, quality and control of the reads were assessed using FastQC, and trimming of the bases with quality scores of &lt;20 was performed with Trim_Galore! (v0.4.5).<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32053166">32053166</a> <b>(1)</b> Methplotlib: analysis of modified nucleotides from nanopore sequencing<ul><li> Together with the tool, we have also developed a <b>snakemake</b> workflow (<xref rid="btaa093-B12" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) to facilitate the processing of multiple datasets and multiple regions of interest.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32082406">32082406</a> <b>(1)</b> Can-Seq: a PCR and DNA sequencing strategy for identifying new alleles of known and candidate genes<ul><li>A workflow for identification of candidate gene mutations and annotation of GenBank format sequence files was developed as a custom <b>Snakemake</b> script (<ext-link ext-link-type="uri" xlink:href="https://github.com/Carroll-Lab/can_seq">https://github.com/Carroll-Lab/can_seq</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32528659">32528659</a> <b>(1)</b> Fluent genomics with&#xA0;<italic>plyranges&#xA0;</italic>and&#xA0;<italic>tximeta</italic><ul><li> The package also contains the<monospace><b>Snakemake</b></monospace> file that was used to distribute the<italic>Salmon</italic> quantification jobs on a cluster (<xref rid="ref-7" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32033573">32033573</a> <b>(1)</b> Lisa: inferring transcriptional regulators through integrative modeling of public chromatin accessibility and ChIP-seq data<ul><li>The Lisa pipeline is implemented with <b>Snakemake</b> [<xref ref-type="bibr" rid="CR73"><u>73</u></xref>].<li> [73](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32034234">32034234</a> <b>(1)</b> Quantifying Hematopoietic Stem Cell Clonal Diversity by Selecting Informative Amplicon Barcodes<ul><li> SABER is Linux/Mac compatible and requires only installation of <b>Snakemake</b> (via conda) which deploys all other software dependencies.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32029569">32029569</a> <b>(1)</b> Five Complete Genome Sequences Spanning the Dutch <named-content content-type="genus-species">Streptococcus suis</named-content> Serotype 2 and Serotype 9 Populations<ul><li> For workflow management, <b>Snakemake</b> version 5.7.1 (<xref rid="B14" ref-type="bibr"><u>14</u></xref>) was used.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32022861">32022861</a> <b>(1)</b> Census of halide-binding sites in protein structures<ul><li> <b>Snakemake</b> pipeline was constructed by using Python programming language to provide the ability to reproduce the results of this study.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32001747">32001747</a> <b>(1)</b> Systematic Comparison of High-throughput Single-Cell and Single-Nucleus Transcriptomes during Cardiomyocyte Differentiation<ul><li> We developed a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR22"><u>22</u></xref></sup> protocol that takes a FASTQ file with such paired-end reads as input and produces an expression matrix corresponding to the UMI of each gene in each cell.<li> [22](K&#xF6;ster, J. &amp; Rahmann, S. Snakemake-a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic>, 10.1093/bioinformatics/bts480 (2012).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33430992">33430992</a> <b>(1)</b> Towards reproducible computational drug discovery<ul><li> Prominent examples of this trends include tools like Nextflow [<xref ref-type="bibr" rid="CR207"><u>207</u></xref>], <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR208"><u>208</u></xref>], Ruffus&#xA0;[<xref ref-type="bibr" rid="CR209"><u>209</u></xref>], BPipe&#xA0;[<xref ref-type="bibr" rid="CR210"><u>210</u></xref>], Cuneiform [<xref ref-type="bibr" rid="CR211"><u>211</u></xref>] and Luigi [<xref ref-type="bibr" rid="CR212"><u>212</u></xref>].<li> [207](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [208](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [209](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [210](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [211](          <person-group person-group-type="author">            <name>              <surname>Brandt</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Reisig</surname>              <given-names>W</given-names>            </name>            <name>              <surname>Leser</surname>              <given-names>ULF</given-names>            </name>          </person-group>          <article-title>Computation semantics of the functional scientific workflow language cuneiform</article-title>          <source>J Funct Program</source>          <year>2017</year>          <pub-id pub-id-type="doi">10.1017/S0956796817000119</pub-id>        ) [212](Bernhardsson E, Freider E, Rouhani A (2012) Luigi GitHub repository.&#xA0;<ext-link ext-link-type="uri" xlink:href="https://github.com/spotify/luigi">https://github.com/spotify/luigi</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32025367">32025367</a> <b>(1)</b> Conserved novel ORFs in the mitochondrial genome of the ctenophore <italic>Beroe forskalii</italic><ul><li>A script written with <b>Snakemake</b> (<xref rid="ref-50" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) that reproduces many of the plots found in this manuscript is available in the Supplemental GitHub/Zenodo Data.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31980822">31980822</a> <b>(3)</b> Reducing the structure bias of RNA-Seq reveals a large number of non-annotated non-coding RNA<ul><li> The read cluster detection pipeline was implemented with the workflow management system <b>Snakemake</b> (<xref rid="B28" ref-type="bibr"><u>28</u></xref>) and is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/GaspardR/<b>snakemake</b>_blockbuster">https://github.com/GaspardR/<b>snakemake</b>_blockbuster</ext-link>.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33790984">33790984</a> <b>(1)</b> Bioelectrical domain walls in homogeneous tissues<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31963239">31963239</a> <b>(1)</b> Trimethylamine-<italic>N</italic>-Oxide Postprandial Response in Plasma and Urine Is Lower After Fermented Compared to Non-Fermented Dairy Consumption in Healthy Adults<ul><li> Raw reads were processed using the DADA2 R package (1.8.0) [<xref rid="B31-nutrients-12-00234" ref-type="bibr"><u>31</u></xref>], integrated in a <b>Snakemake</b> (5.3.0) [<xref rid="B32-nutrients-12-00234" ref-type="bibr"><u>32</u></xref>] pipeline to identify amplicon sequence variants (ASV), which confer a higher resolution than traditional operational taxonomic unit assignment.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27214047">27214047</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31941956">31941956</a> <b>(1)</b> The absence of intact polar lipid-derived GDGTs in marine waters dominated by Marine Group II: Implications for lipid biosynthesis in Archaea<ul><li> The 16S rRNA gene amplicon sequences were analyzed by Cascabel<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup>, an amplicon sequence data analysis pipeline based on <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>.<li> [49](Alejandro, A. A. <italic>et al</italic>. Cascabel: a flexible, scalable and easy-to-use amplicon sequence data analysis pipeline. <italic>bioRxiv</italic> 1&#x2013;8 (2019).) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31930382">31930382</a> <b>(1)</b> FilTar: using RNA-Seq data to improve microRNA target prediction accuracy in animals<ul><li> FilTar utilizes <b>Snakemake</b> (v5.4.0) (<xref rid="btaa007-B27" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) for workflow management.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31929557">31929557</a> <b>(1)</b> Cell-free DNA donor fraction analysis in pediatric and adult heart transplant patients by multiplexed allele-specific quantitative PCR: Validation of a rapid and highly sensitive clinical test for stratification of rejection probability<ul><li> Sequence processing was performed using CentOS 7.4, bwa 0.7, Trimmomatic-0.38, samtools 0.1, bcftools/htslib 1.9, GRCh37/hs37d5, bam-readcount 0.8, <b>Snakemake</b> 5.3, Python 3.6, RStudio 1.0, R 3.4, R-xlsx 0.6, and plyr 1.8.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31936409">31936409</a> <b>(1)</b> Whole Genome Sequencing for Studying <italic>Bacillus anthracis</italic> from an Outbreak in the Abruzzo Region of Italy<ul><li> For scalability and fast run-time, the pipeline is implemented within the <b>Snakemake</b> workflow management system [<xref rid="B20-microorganisms-08-00087" ref-type="bibr"><u>20</u></xref>].<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31934500">31934500</a> <b>(8)</b> sv-callers: a highly portable parallel workflow for structural variant detection in whole-genome sequence data<ul><li> For example, <b>Snakemake</b> is an established generic workflow system inspired by the GNU Make build automation tool (<xref rid="ref-21" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> Moreover, <b>Snakemake</b> supports Conda package manager and light-weight containers such as Docker (<uri xlink:href="https://www.docker.com/">https://www.docker.com/</uri>) or Singularity (<xref rid="ref-17" ref-type="bibr"><u>Kurtzer, Sochat &amp; Bauer, 2017</u></xref>), Common Workflow Language (<xref rid="ref-3" ref-type="bibr"><u>Amstutz et al., 2016</u></xref>) and parallel execution in HPC cluster (via DRMAA, <uri xlink:href="http://www.drmaa.org/">http://www.drmaa.org/</uri>) or cloud environments (via Kubernetes, <uri xlink:href="https://kubernetes.io/">https://kubernetes.io/</uri>).<li>To alleviate these problems, we developed <italic>sv-callers</italic>, a user-friendly, portable and scalable workflow based on the <b>Snakemake</b> and Xenon (middleware) software.<li> <xref rid="ref-18" ref-type="bibr"><u>Kuzniar, 2019</u></xref>) using the <b>Snakemake</b> workflow system (v4.8.0; <xref rid="ref-21" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) and the <italic>Xenon</italic> software suite (v3.0.0).<li> Moreover, the workflow parameters are also accessible through the command-line interface (see <monospace><b>snakemake</b> -C [params]</monospace> argument).<li> For this reason, <italic>xenon-cli</italic> was configured as the cluster submission command for our <b>Snakemake</b>-based workflow, which improved the portability by making the workflow independent of the system-specific solutions (see <monospace><b>snakemake</b> --cluster</monospace> &#x2018;<monospace>xenon scheduler [type]</monospace>&#x2019; argument).<li> We aim to extend cloud support in Xenon with schedulers such as Microsoft Azure Batch and Amazon AWS Batch as well as to improve the coupling between Xenon and <b>Snakemake</b> software.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Kurtzer, Sochat &amp; Bauer, 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [Amstutz et al., 2016](          <person-group person-group-type="author">            <name>              <surname>Amstutz</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Crusoe</surname>              <given-names>MR</given-names>            </name>            <name>              <surname>Tijani&#x107;</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Chapman</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Chilton</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Heuer</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kartashov</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Leehr</surname>              <given-names>D</given-names>            </name>            <name>              <surname>M&#xE9;nager</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Nedeljkovich</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Scales</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Soiland-Reyes</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Stojanovic</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>Common workflow language, v1.0</article-title>          <year>2016</year>          <source>figshare</source>          <pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>        ) [Kuzniar, 2019](          <person-group person-group-type="author">            <name>              <surname>Kuzniar</surname>              <given-names>A</given-names>            </name>          </person-group>          <data-title>sv-callers</data-title>          <year>2019</year>          <version designator="1.1.0">Version 1.1.0</version>          <comment>Zenodo</comment>          <uri xlink:href="http://doi.org/10.5281/zenodo.1217111">http://doi.org/10.5281/zenodo.1217111</uri>        ) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31872320">31872320</a> <b>(1)</b> Rice Stress-Resistant SNP Database<ul><li> The processing workflow was created using the <b>snakemake</b> workflow system (K&#xF6;ster and Rahmann <xref ref-type="bibr" rid="CR10"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31830916">31830916</a> <b>(1)</b> uap: reproducible and robust HTS data analysis<ul><li> WMS approaches like Ruffus [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] or <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>] allow the implementation of highly individual analyses, either via domain-specific programming languages or a general-purpose programming language.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31813948">31813948</a> <b>(1)</b> Current RNA-seq methodology reporting limits reproducibility<ul><li> In this view, we advocate the use of workflow management systems, such as Nextflow [<xref rid="ref22" ref-type="bibr"><u>22</u></xref>] and <b>Snakemake</b> [<xref rid="ref23" ref-type="bibr"><u>23</u></xref>].<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31778174">31778174</a> <b>(2)</b> Genetic Landscapes Reveal How Human Genetic Diversity Aligns with Geography<ul><li> Processing was done using a reproducible <b>snakemake</b> pipeline (<xref rid="msz280-B34" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) available under <ext-link ext-link-type="uri" xlink:href="http://github.com/NovembreLab/eems-merge">http://github.com/NovembreLab/eems-merge</ext-link>, heavily relying on plink 1.9 (<xref rid="msz280-B17" ref-type="bibr">Chang et&#xA0;al.<li>We developed a second pipeline using <b>snakemake</b> (<xref rid="msz280-B34" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) to perform all subsetting and demographic analyses, available under github.com/NovembreLab/eems-around-the-world.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31772058">31772058</a> <b>(1)</b> Evolution and Global Transmission of a Multidrug-Resistant, Community-Associated Methicillin-Resistant <named-content content-type="genus-species">Staphylococcus aureus</named-content> Lineage from the Indian Subcontinent<ul><li> The workflow is implemented in <b>Snakemake</b> (<xref rid="B94" ref-type="bibr"><u>94</u></xref>) and runs in virtual environments that include software distributed in the Bioconda (<xref rid="B95" ref-type="bibr"><u>95</u></xref>) channel.<li> [94](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [95](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31768606">31768606</a> <b>(1)</b> Modeling population heterogeneity from microbial communities to immune response in cells<ul><li><b>Snakemake</b> [<xref ref-type="bibr" rid="CR106"><u>106</u></xref>].<li> [106](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31748746">31748746</a> <b>(1)</b> LONGITUDINAL MOLECULAR TRAJECTORIES OF DIFFUSE GLIOMA IN ADULTS<ul><li>All pipelines were developed using <b>snakemake</b> 5.2.2 <sup><xref rid="R47" ref-type="bibr"><u>47</u></xref></sup>.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31768265">31768265</a> <b>(8)</b> SNAPPy: A <b>snakemake</b> pipeline for scalable HIV-1 subtyping by phylogenetic pairing<ul><li> To answer these limitations, we used the <b>Snakemake</b> workflow management system (<xref rid="vez050-B24" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>) to create a reproducible and scalable HIV-1 subtyping method based on phylogenetic pairing (SNAPPy).<li>The SNAPPy pipeline was built on the <b>Snakemake</b> workflow management system (<xref rid="vez050-B24" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>).<li>Alternatively, as any other <b>Snakemake</b> (<xref rid="vez050-B24" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>) pipeline, intermediate tasks can be performed without the execution of the entire pipeline, making SNAPPy extremely useful for HIV-1 MSA (<xref ref-type="fig" rid="vez050-F1"><u>Fig.&#xA0;1</u></xref>, Point 7).<li> At the end of each SNAPPy run, a <b>snakemake</b> hidden folder named &#x2018;.<b>snakemake</b>&#x2019; is deleted because it occupies a substantial amount of space.<li> The usage of <b>Snakemake</b> (<xref rid="vez050-B24" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>) as a workflow manager allows the construction of a directed acyclic graph of jobs, inferring which tasks need to be performed sequential and which can run in parallel.<li> As an indicator, a SNAPPy run of 50,000 HIV-1 sequences occupied at the peak 59 GB and &lt;4 GB after the depletion of the <b>snakemake</b> hidden logs folder.<li> <b>Snakemake</b> (<xref rid="vez050-B24" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>), the pipeline workflow management systems that SNAPPy is built upon, allows an almost linear scaling in the ratio of computational time/number of CPU cores used.<li> [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31727118">31727118</a> <b>(4)</b> mPies: a novel metaproteomics tool for the creation of relevant protein databases and automatized protein annotation<ul><li> mPies is written in Python 3.6, uses the workflow management system <b>Snakemake</b> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] and relies on Bioconda [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] to ensure reproducibility.<li> The utilization of <b>Snakemake</b> allows easy adjustment of the assembly and gene calling parameters.<li>Authors response: <italic>The value for maximum target sequences is adaptable, as are most parameters in the <b>Snakemake</b> workflow.<li><bold>Other requirements:</bold> <b>Snakemake</b>, bioconda.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31718539">31718539</a> <b>(4)</b> PyBDA: a command line tool for automated analysis of big biological data sets<ul><li> PyBDA uses <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] to automatically execute these workflows of methods.<li> Every operation in the Petri net, i.e., every triple of input file, method and output file, is then executed by <b>Snakemake</b>.<li> It then uses <b>Snakemake</b> for execution of each triple.<li> PyBDA automatically parses a user defined pipeline of multiple machine learning and data analysis tasks from a config file and distributes jobs to compute nodes using <b>Snakemake</b> and Apache Spark.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31703553">31703553</a> <b>(1)</b> PyGMQL: scalable data extraction and analysis for heterogeneous genomic datasets<ul><li> This has brought to the development of very successful workflow management software like Galaxy [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>], <b>Snakemake</b> [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>], Nextflow [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] and FireCloud [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] (evolving to a new system named Terra, in May 2019).<li> [13](          <person-group person-group-type="author">            <name>              <surname>Afgan</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Baker</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Van den Beek</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Blankenberg</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Bouvier</surname>              <given-names>D</given-names>            </name>            <name>              <surname>&#x10C;ech</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Chilton</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Clements</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Coraor</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Eberhard</surname>              <given-names>C</given-names>            </name>            <etal/>          </person-group>          <article-title>The galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2016 update</article-title>          <source>Nucleic Acids Res</source>          <year>2016</year>          <volume>44</volume>          <issue>W1</issue>          <fpage>3</fpage>          <lpage>10</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkw343</pub-id>        ) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](Birger C, Hanna M, Salinas E, Neff J, Saksena G, Livitz D, Rosebrock D, Stewart C, Leshchiner I, Baumann A, Voet D, Cibulskis K, Banks E, Philippakis A, Getz G. Firecloud, a scalable cloud-based platform for collaborative genome analysis: Strategies for reducing and controlling costs. bioRxiv. 2017. 10.1101/209494. <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/early/2017/11/03/209494.full.pdf">https://www.biorxiv.org/content/early/2017/11/03/209494.full.pdf</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31699023">31699023</a> <b>(3)</b> Third international conference &#x201C;Bioinformatics: from Algorithms to Applications&#x201D; (BiATA 2019)<ul><li>P6 CDSnake: <b>Snakemake</b> pipeline for retrieval of annotated OTUs from paired-end reads using CD-HIT utilities.<li> Here we assembled CD-HIT-OTU-Miseq utilities into pipeline using <b>Snakemake</b> workflow.<li> We developed <b>Snakemake</b> pipeline for OTU-MiSeq utilities, which can be useful for easier automated runs.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31695094">31695094</a> <b>(1)</b> Empirical design of a variant quality control pipeline for whole genome sequencing data using replicate discordance<ul><li>The biallelic pipeline was written in a series of shell scripts (Supplementary Text&#xA0;<xref rid="MOESM1" ref-type="media"><u>S3</u></xref>), using bcftools and vcftools, which are adaptable for workflow environments such as <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref>,<xref ref-type="bibr" rid="CR45"><u>45</u></xref>,<xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup>.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21653522">21653522</a>) [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21903627">21903627</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31675414">31675414</a> <b>(1)</b> Sharing interoperable workflow provenance: A review of best practices and their practical application in CWLProv<ul><li> Some existing make-like systems such as <b>Snakemake</b> support partial re-enactments but typically rely on fixed file paths for input data and require manual intervention to provide the specific directory structure.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31672156">31672156</a> <b>(10)</b> Optimizing sequencing protocols for leaderboard metagenomics by combining long and short reads<ul><li>Assembled TSLR libraries for each of the 8 samples sequenced with TruSeqNano and NexteraXT libraries were processed into contig databases using a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR50"><u>50</u></xref>] pipeline adaptation of the recommended workflow for the Anvi&#x2019;o analysis and visualization platform [<xref ref-type="bibr" rid="CR33"><u>33</u></xref>].<li> This workflow can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/tanaes/<b>snakemake</b>_anvio">https://github.com/tanaes/<b>snakemake</b>_anvio</ext-link>.<li>We constructed an automated pipeline using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR50"><u>50</u></xref>] to assemble samples and compare them to the reference bins, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/RNAer/assembly_<b>snakemake</b>_pipeline">https://github.com/RNAer/assembly_<b>snakemake</b>_pipeline</ext-link>.<li> Once initiated, the intelligent workflow tracking provided by <b>Snakemake</b> allows test sequences from additional library prep protocols to be sequentially added and compared to prior results, using the minimum necessary re-computation.<li>Demultiplexed sequences were quality filtered and adapter trimmed using Atropos [<xref ref-type="bibr" rid="CR63"><u>63</u></xref>], assembled using SPAdes [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] or metaSPAdes [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>], and quality metrics summarized using Quast [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] and MultiQC [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>, <xref ref-type="bibr" rid="CR64"><u>64</u></xref>], all implemented in a custom <b>Snakemake</b> [<xref ref-type="bibr" rid="CR50"><u>50</u></xref>] workflow, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/tanaes/<b>snakemake</b>_assemble">https://github.com/tanaes/<b>snakemake</b>_assemble</ext-link>.<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28875074">28875074</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22506599">22506599</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28298430">28298430</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23422339">23422339</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23422339">23422339</a>) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27312411">27312411</a>) [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31665499">31665499</a> <b>(1)</b> ReMap 2020: a database of regulatory regions from an integrative analysis of Human and Arabidopsis DNA-binding sequencing experiments<ul><li> Briefly, our pipelines uses <b>SnakeMake</b> either in a Conda or Singularity environment depending on the HPC resources, where Torque and Slurm managers are both supported.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32072135">32072135</a> <b>(1)</b> IDOL regulates systemic energy balance through control of neuronal VLDLR expression<ul><li>The demultiplexed fastq files from the Drop-seq sequencing data were processed into a digital gene expression matrix using the dropSeqPipe (<ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe">https://github.com/Hoohm/dropSeqPipe</ext-link>) <b>SnakeMake</b> wrapper for Drop-seq tools version 1.13.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31644425">31644425</a> <b>(1)</b> Single cell analysis reveals immune cell&#x2013;adipocyte crosstalk regulating the transcription of thermogenic adipocytes<ul><li> The workflow is available as modified version of the <b>snakemake</b>-based dropSeqPipe (<ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe">https://github.com/Hoohm/dropSeqPipe</ext-link>) workflow and is available on github (<xref rid="bib1" ref-type="bibr"><u>Arneson, 2019</u></xref>; copy archived at <ext-link ext-link-type="uri" xlink:href="https://github.com/elifesciences-publications/dropSeqPipeDropEST">https://github.com/elifesciences-publications/dropSeqPipeDropEST</ext-link>).<li> [Arneson, 2019](          <person-group person-group-type="author">            <name>              <surname>Arneson</surname>              <given-names>D</given-names>            </name>          </person-group>          <year>2019</year>          <data-title>dropSeqPipeDropEST</data-title>          <source>GitHub</source>          <version designator="3d357a0">3d357a0</version>          <ext-link ext-link-type="uri" xlink:href="https://github.com/darneson/dropSeqPipeDropEST">https://github.com/darneson/dropSeqPipeDropEST</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31646766">31646766</a> <b>(1)</b> Transcriptomic and barrier responses of human airway epithelial cells exposed to cannabis smoke<ul><li> A <b>snakemake</b>&#x2010;based pipeline called hppRNA (v.1.3.3) was used to combine the above steps into a streamlined work&#x2010;flow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31640560">31640560</a> <b>(1)</b> Transcriptomic analysis of <italic>Macrobrachium rosenbergii</italic> (giant fresh water prawn) post-larvae in response to <italic>M. rosenbergii</italic> nodavirus (<italic>Mr</italic>NV) infection: de novo assembly and functional annotation<ul><li>The automated data analysis pipeline was written in python using the <b>Snakemake</b> tool [<xref ref-type="bibr" rid="CR108"><u>108</u></xref>].<li> [108](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31639030">31639030</a> <b>(2)</b> SEPATH: benchmarking the search for pathogens in human tissue whole genome sequence data leads to template pipelines<ul><li>The top-performing algorithms and parameters for each of the stages have been combined in a unifying template pipeline implemented in <b>snakemake</b> [<xref ref-type="bibr" rid="CR53"><u>53</u></xref>]: SEPATH (Fig.&#xA0;<xref rid="Fig8" ref-type="fig"><u>8</u></xref>, <ext-link ext-link-type="uri" xlink:href="https://github.com/UEA-Cancer-Genetics-Lab/sepath_tool_UEA">https://github.com/UEA-Cancer-Genetics-Lab/sepath_tool_UEA</ext-link>).<li>Project name: SEPATHProject home page: <ext-link ext-link-type="uri" xlink:href="https://github.com/UEA-Cancer-Genetics-Lab/sepath_tool_UEA">https://github.com/UEA-Cancer-Genetics-Lab/sepath_tool_UEA</ext-link>Operating system(s): Linux-based high performance computing cluster environmentsProgramming language: Python 3, BashOther requirements: Python v3.5, <b>Snakemake</b> v3.13.3, Trimmomatic v0.36, Java v.8.0_51, bbmap v37.28, mOTUs2 v2.0.1, Kraken 1, Spades v3.11.1, Pysam v0.15.1License: GPL version 3 or later.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31622330">31622330</a> <b>(3)</b> Patient Dossier: Healthcare queries over distributed resources<ul><li> tool and general workflows has been noted elsewhere, for instance, in the <b>Snakemake</b> tool (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31615395">31615395</a> <b>(1)</b> <italic>doepipeline</italic>: a systematic approach to optimizing multi-level and multi-step data processing workflows<ul><li> Given the recent development and consolidation of workflow managing systems [<xref ref-type="bibr" rid="CR44"><u>44</u></xref>] it would be possible to integrate <italic>doepipeline</italic> with for example <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>] or NextFlow [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>], similar to other implementations [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>, <xref ref-type="bibr" rid="CR48"><u>48</u></xref>].<li> [44](Karim MR, Michel A, Zappa A, Baranov P, Sahay R, Rebholz-Schuhmann D. Improving data workflow systems with cloud services and use of open data for bioinformatics research. Brief Bioinform [Internet]. 2018 [cited 2019 Jun 20];19(5):1035&#x2013;1050. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bib/article/19/5/1035/3737318">https://academic.oup.com/bib/article/19/5/1035/3737318</ext-link>) [45](Koster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics [Internet]. 2012 [cited 2019 Jun 20];28(19):2520&#x2013;2522. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts480">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts480</ext-link>) [46](Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol [Internet]. 2017 [cited 2019 Jun 20];35(4):316&#x2013;319. Available from: <ext-link ext-link-type="uri" xlink:href="http://www.nature.com/articles/nbt.3820">http://www.nature.com/articles/nbt.3820</ext-link>) [47](Holl S, Mohammed Y, Zimmermann O, Palmblad M. Scientific workflow optimization for improved peptide and protein identification. BMC Bioinformatics [Internet]. 2015 [cited 2019 Jun 20];16(1):284. Available from: <ext-link ext-link-type="uri" xlink:href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0714-x">http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0714-x</ext-link>) [48](Palmblad M, Lamprecht A-L, Ison J, Schw&#xE4;mmle V. Automated workflow composition in mass spectrometry-based proteomics. Wren J, editor. Bioinformatics [Internet]. 2019 [cited 2019 Jun 20];35(4):656&#x2013;664. Available from: <ext-link ext-link-type="uri" xlink:href="https://academic.oup.com/bioinformatics/article/35/4/656/5060940">https://academic.oup.com/bioinformatics/article/35/4/656/5060940</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31613890">31613890</a> <b>(2)</b> HaTSPiL: A modular pipeline for high-throughput sequencing data analysis<ul><li> BigDataScript [<xref rid="pone.0222512.ref008" ref-type="bibr"><u>8</u></xref>], Bpipe [<xref rid="pone.0222512.ref009" ref-type="bibr"><u>9</u></xref>] and <b>SnakeMake</b> [<xref rid="pone.0222512.ref010" ref-type="bibr"><u>10</u></xref>] are examples of languages specifically designed for the development of bioinformatics pipelines, aiming at the maximum flexibility and powerfulness without the steeper learning curve of a general-purpose programming language.<li> It is worth noting that Sequanix is a GUI to <b>SnakeMake</b>, demonstrating that it is possible to obtain high accessibility without sacrificing flexibility and powerfulness.<li> [8](<name><surname>Cingolani</surname><given-names>P</given-names></name>, <name><surname>Sladek</surname><given-names>R</given-names></name>, <name><surname>Blanchette</surname><given-names>M</given-names></name>. <article-title>BigDataScript: a scripting language for data pipelines</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2015</year>;<volume>31</volume>:<fpage>10</fpage>&#x2013;<lpage>16</lpage>. <pub-id pub-id-type="doi">10.1093/bioinformatics/btu595</pub-id>) [9](<name><surname>Sadedin</surname><given-names>SP</given-names></name>, <name><surname>Pope</surname><given-names>B</given-names></name>, <name><surname>Oshlack</surname><given-names>A</given-names></name>. <article-title>Bpipe: a tool for running and managing bioinformatics pipelines</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2012</year>;<volume>28</volume>:<fpage>1525</fpage>&#x2013;<lpage>1526</lpage>. <pub-id pub-id-type="doi">10.1093/bioinformatics/bts167</pub-id>) [10](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name>, <name><surname>Rahmann</surname><given-names>S</given-names></name>. <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2018</year>;<volume>34</volume>:<fpage>3600</fpage><pub-id pub-id-type="doi">10.1093/bioinformatics/bty350</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31723427">31723427</a> <b>(1)</b> Creating and sharing reproducible research code the workflowr way<ul><li> development and management of computational pipelines (e.g.,<bold>GNU Make</bold>,<bold><b>Snakemake</b></bold><sup><xref rid="ref-35" ref-type="bibr"><u>35</u></xref></sup>,<bold>drake</bold><sup><xref rid="ref-36" ref-type="bibr"><u>36</u></xref></sup>); management and archiving of data objects (e.g.,<bold>archivist</bold><sup><xref rid="ref-37" ref-type="bibr"><u>37</u></xref></sup>, Dryad<sup><xref rid="ref-38" ref-type="bibr"><u>38</u></xref></sup>, Zenodo); and distribution of open source software (e.g., CRAN, Bioconductor<sup><xref rid="ref-39" ref-type="bibr"><u>39</u></xref></sup>, Bioconda<sup><xref rid="ref-40" ref-type="bibr"><u>40</u></xref></sup>).<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [36](<person-group person-group-type="author"><name><surname>Landau</surname><given-names>WM</given-names></name></person-group>:<article-title>The drake R package: a pipeline toolkit for reproducibility and high-performance computing.</article-title><source><italic toggle="yes">J Open Source Softw.</italic></source><year>2018</year>;<volume>3</volume>(<issue>21</issue>):<fpage>550</fpage><pub-id pub-id-type="doi">10.21105/joss.00550</pub-id>) [37](<person-group person-group-type="author"><name><surname>Biecek</surname><given-names>P</given-names></name><name><surname>Kosinski</surname><given-names>M</given-names></name></person-group>:<article-title>archivist: an R package for managing, recording and restoring data analysis results.</article-title><source><italic toggle="yes">J Stat Softw.</italic></source><year>2017</year>;<volume>82</volume>(<issue>11</issue>):<fpage>1</fpage>&#x2013;<lpage>28</lpage>.<pub-id pub-id-type="doi">10.18637/jss.v082.i11</pub-id>) [38](<person-group person-group-type="author"><name><surname>Vision</surname><given-names>T</given-names></name></person-group>:<article-title>The dryad digital repository: published evolutionary data as part of the greater data ecosystem</article-title>.<year>2010</year><ext-link ext-link-type="uri" xlink:href="http://precedings.nature.com/documents/4595/version/1/files/npre20104595-1.pdf">Reference Source</ext-link>) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15461798">15461798</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31568525">31568525</a> <b>(1)</b> Telescope: Characterization of the retrotranscriptome by accurate estimation of transposable element expression<ul><li>A complete <b>snakemake</b>[<xref rid="pcbi.1006453.ref078" ref-type="bibr"><u>78</u></xref>] pipeline for reproducing the ENCODE analysis is available from <ext-link ext-link-type="uri" xlink:href="https://github.com/mlbendall/TelescopeEncode">https://github.com/mlbendall/TelescopeEncode</ext-link>.<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31561627">31561627</a> <b>(2)</b> Association Mapping between Candidate Gene SNP and Production and Oil Quality Traits in Interspecific Oil Palm Hybrids<ul><li> The <b>Snakemake</b>-capture workflow identified initially 12,200 potential SNP.<li> The cleaned, genotype specific sequences were processed using the &#x201C;<b>Snakemake</b>-capture&#x201D; script [<xref rid="B66-plants-08-00377" ref-type="bibr"><u>66</u></xref>] of the South Green bioinformatics platform to map the reads using BWA v0.7.15 [<xref rid="B67-plants-08-00377" ref-type="bibr"><u>67</u></xref>], to clean the alignments with Samtools v1.3 [<xref rid="B68-plants-08-00377" ref-type="bibr"><u>68</u></xref>], to sort the reads with Picard-tools v2.7.0 [<xref rid="B69-plants-08-00377" ref-type="bibr"><u>69</u></xref>] and to call the SNP using GATK haplotype caller v3.7-0 [<xref rid="B70-plants-08-00377" ref-type="bibr"><u>70</u></xref>].<li> [66](          <person-group person-group-type="author">            <name>              <surname>Soriano</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Guitton</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Mournet</surname>              <given-names>P.</given-names>            </name>          </person-group>          <article-title>Workflow-Snakemake-Capture</article-title>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://github.com/SouthGreenPlatform/Workflow-snakemake-capture">https://github.com/SouthGreenPlatform/Workflow-snakemake-capture</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2019-07-22">(accessed on 22 July 2019)</date-in-citation>        ) [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20080505">20080505</a>) [68](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [69](          <article-title>Picard Tools&#x2014;By Broad Institute</article-title>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://broadinstitute.github.io/picard/index.html">https://broadinstitute.github.io/picard/index.html</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2018-08-30">(accessed on 30 August 2018)</date-in-citation>        ) [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20644199">20644199</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31554283">31554283</a> <b>(1)</b> Causes and Consequences of A Glutamine Induced Normoxic HIF1 Activity for the Tumor Metabolism<ul><li>We set up a <b>Snakemake</b> [<xref rid="B84-ijms-20-04742" ref-type="bibr"><u>84</u></xref>] workflow, called &#x2018;VipeR_HIF1alpha&#x2019;, for the analysis of the RNA-Seq and qPCR data in a reproducible, automated, and partially contained manner.<li> [84](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S.</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <year>2018</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31532761">31532761</a> <b>(1)</b> Inverted translational control of eukaryotic gene expression by ribosome collisions<ul><li> All the simulation steps are implemented as <b>Snakemake</b> workflows [<xref rid="pbio.3000396.ref087" ref-type="bibr"><u>87</u></xref>] in the Snakefile script.<li> [87](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31534146">31534146</a> <b>(1)</b> Conventional and organic soil management as divergent drivers of resident and active fractions of major soil food web constituents<ul><li>Sequences were processed with Hydra pipeline version 1.3.3<sup><xref ref-type="bibr" rid="CR81"><u>81</u></xref></sup> implemented in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR82"><u>82</u></xref></sup>.<li> [81](de Hollander, M. NIOO-KNAW/hydra: Version 1.3. 3. Zenodo., 10.5281/zenodo.1434147 (2017).) [82](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31513641">31513641</a> <b>(2)</b> <italic>uORF-Tools</italic>&#x2014;Workflow for the determination of translation-regulatory upstream open reading frames<ul><li> It is based on <italic><b>Snakemake</b></italic> [<xref rid="pone.0222459.ref022" ref-type="bibr"><u>22</u></xref>] and automatically installs all tool dependencies in a version-controlled manner via <italic>bioconda</italic> [<xref rid="pone.0222459.ref023" ref-type="bibr"><u>23</u></xref>].<li><italic>uORF-Tools</italic> is provided as a readily deployable <italic><b>Snakemake</b></italic> workflow, which comes with extensive documentation (<xref ref-type="supplementary-material" rid="pone.0222459.s001"><u>S1 File</u></xref>).<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31506435">31506435</a> <b>(1)</b> BioExcel Building Blocks, a software library for interoperable biomolecular simulation workflows<ul><li> Nextflow<sup><xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup>, PyCOMPSs<sup><xref ref-type="bibr" rid="CR18"><u>18</u></xref></sup>, <b>Snakemake</b>)<sup><xref ref-type="bibr" rid="CR19"><u>19</u></xref></sup>.<li> [17](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <etal/>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <fpage>316</fpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [18](          <person-group person-group-type="author">            <name>              <surname>Tejedor</surname>              <given-names>E</given-names>            </name>            <etal/>          </person-group>          <article-title>PyCOMPSs: Parallel computational workflows in Python</article-title>          <source>The International Journal of High Performance Computing Applications</source>          <year>2015</year>          <volume>31</volume>          <fpage>66</fpage>          <lpage>82</lpage>          <pub-id pub-id-type="doi">10.1177/1094342015594678</pub-id>        ) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31500660">31500660</a> <b>(5)</b> A comparison of automatic cell identification methods for single-cell RNA sequencing data<ul><li> To support the future extension of this benchmarking work with new classifiers and datasets, we provide a <b>Snakemake</b> workflow to automate the performed benchmarking analyses (<ext-link ext-link-type="uri" xlink:href="https://github.com/tabdelaal/scRNAseq_Benchmark/tree/<b>snakemake</b>_and_docker">https://github.com/tabdelaal/scRNAseq_Benchmark/</ext-link>).<li>In order to ensure reproducibility and support the future extension of this benchmarking work with new classification methods and benchmarking datasets, a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR51"><u>51</u></xref>] workflow for automating the performed benchmarking analyses was developed with an MIT license (<ext-link ext-link-type="uri" xlink:href="https://github.com/tabdelaal/scRNAseq_Benchmark/tree/<b>snakemake</b>_and_docker">https://github.com/tabdelaal/scRNAseq_Benchmark/</ext-link>).<li> These images will be used through <b>Snakemake</b>&#x2019;s singularity integration to allow the workflow to be run without the requirement to install specific methods and to ensure reproducibility.<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31505865">31505865</a> <b>(3)</b> <italic>Drosophila melanogaster</italic> Mutated in its <italic>GBA1b</italic> Ortholog Recapitulates Neuronopathic Gaucher Disease<ul><li> Pipeline was constructed using <b>Snakemake</b> (<uri xlink:href="https://<b>snakemake</b>.readthedocs.io/en/v3.9.1/">https://<b>snakemake</b>.readthedocs.io/en/v3.9.1/</uri>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31500242">31500242</a> <b>(1)</b> Identification of Structural Variation from NGS-Based Non-Invasive Prenatal Testing<ul><li>All computational steps were executed using <b>Snakemake</b> workflow engine [<xref rid="B42-ijms-20-04403" ref-type="bibr"><u>42</u></xref>].<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31467286">31467286</a> <b>(1)</b> Identification of somatic mutations in single cell DNA-seq using a spatial model of allelic imbalance<ul><li>SCAN-SNV is implemented using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR24"><u>24</u></xref></sup> and distributed as a Conda<sup><xref ref-type="bibr" rid="CR25"><u>25</u></xref></sup> package.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [25](Conda, <ext-link ext-link-type="uri" xlink:href="http://conda.io">http://conda.io</ext-link>.) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31438611">31438611</a> <b>(2)</b> WiPP: Workflow for Improved Peak Picking for Gas Chromatography-Mass Spectrometry (GC-MS) Data<ul><li> The modular architecture of WiPP, based on the python workflow framework <b>Snakemake</b>, enables new peak picking algorithms integration with little effort.<li>The pipeline was implemented in python 3 using <b>Snakemake</b> [<xref rid="B24-metabolites-09-00171" ref-type="bibr"><u>24</u></xref>], a reproducible and scalable workflow management system.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31427603">31427603</a> <b>(1)</b> Mitogenic and progenitor gene programmes in single pilocytic astrocytoma cells<ul><li> Sequenced reads were aligned to the UCSC hg19 reference genome assembly using STAR v2.5.1b<sup><xref ref-type="bibr" rid="CR48"><u>48</u></xref></sup> and gene expression was quantified using RSEM v1.2.28<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup> as part of the VIPER <b>snakemake</b> pipeline<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21884820">21884820</a>) [50](          <person-group person-group-type="author">            <name>              <surname>Cornwell</surname>              <given-names>M</given-names>            </name>            <etal/>          </person-group>          <article-title>VIPER: Visualization Pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis</article-title>          <source>BMC Bioinforma.</source>          <year>2018</year>          <volume>19</volume>          <fpage>135</fpage>          <pub-id pub-id-type="doi">10.1186/s12859-018-2139-9</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31415589">31415589</a> <b>(1)</b> Advances in gene ontology utilization improve statistical power of annotation enrichment<ul><li>Scripts and <b>snakemake</b> [<xref rid="pone.0220728.ref029" ref-type="bibr"><u>29</u></xref>] workflows for performing these analyses can be found in Scripts Directory 4 in the FigShare directory available at.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31412777">31412777</a> <b>(1)</b> Genotyping by low-coverage whole-genome sequencing in intercross pedigrees from outbred founders: a cost-efficient approach<ul><li>The complete analysis pipeline outlined in Steps 2 to 4 was implemented in a user-friendly pipeline manager <b>Snakemake</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>].<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31412927">31412927</a> <b>(1)</b> Deciphering rhizosphere microbiome assembly of wild and modern common bean (<italic>Phaseolus vulgaris</italic>) in native and agricultural soils from Colombia<ul><li> All steps were implemented in a <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR32"><u>32</u></xref>].<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31399544">31399544</a> <b>(1)</b> Long noncoding RNAs are involved in multiple immunological pathways in response to vaccination<ul><li>We developed a <b>Snakemake</b> pipeline to update probe annotations from microarray platforms.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31387524">31387524</a> <b>(1)</b> <italic>Rpv3&#x2013;1</italic> mediated resistance to grapevine downy mildew is associated with specific host transcriptional responses and the accumulation of stilbenes<ul><li>Initial steps from raw data quality control to mapping and eventually read counting was undertaken by QBiC on the High Performance cluster (HPC) of the University of T&#xFC;bingen using a fully automated workflow written in <b>Snakemake</b> [<xref ref-type="bibr" rid="CR81"><u>81</u></xref>].<li> [81](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31387612">31387612</a> <b>(1)</b> BART-Seq: cost-effective massively parallelized targeted sequencing for genomics, transcriptomics, and single-cell analysis<ul><li> Given the acceptable quality, <b>Snakemake</b> workflow engine [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>] was used for automatic or step-by-step analysis of raw reads, sets of primers, linkers, barcodes, and expected amplicons.<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31384914">31384914</a> <b>(1)</b> Intraspecific Variation in Protists: Clues for Microevolution from <italic>Poteriospumella lacustris</italic> (Chrysophyceae)<ul><li> A <b>Snakemake</b> (<xref rid="evz171-B45" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>) automated workflow was created to process the sequencing data.<li> [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31375809">31375809</a> <b>(1)</b> Compendium of 4,941 rumen metagenome-assembled genomes for rumen microbiome biology and enzyme discovery<ul><li> The 4,941 best bins were analyzed using MAGpy<sup><xref ref-type="bibr" rid="CR10"><u>10</u></xref></sup>, a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR55"><u>55</u></xref></sup> pipeline that runs a series of analyses on the bins, including CheckM (v.1.0.5); prodigal<sup><xref ref-type="bibr" rid="CR34"><u>34</u></xref></sup> (v2.6.3) protein prediction; Pfam_Scan<sup><xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup> (v.1.6); a DIAMOND<sup><xref ref-type="bibr" rid="CR12"><u>12</u></xref></sup> (v.0.9.22.123) search against UniProt TrEMBL; PhyloPhlAn<sup><xref ref-type="bibr" rid="CR13"><u>13</u></xref></sup> (v.0.99); and a sourmash (v.2.0.0) search against all public bacterial genomes.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30418481">30418481</a>) [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20211023">20211023</a>) [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24288371">24288371</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25402007">25402007</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23942190">23942190</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31350555">31350555</a> <b>(1)</b> amplimap: a versatile tool to process and analyze targeted NGS data<ul><li> Internally, amplimap uses the <b>Snakemake</b> work flow management package (<xref rid="btz582-B6" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) to automate the efficient execution of external software as well as custom code.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31347500">31347500</a> <b>(3)</b> Combinations of <italic>Spok</italic> genes create multiple meiotic drivers in <italic>Podospora</italic><ul><li> copy&#xA0;archived&#xA0;at&#xA0;<ext-link ext-link-type="uri" xlink:href="https://github.com/elifesciences-publications/SpokPaper">https://github.com/elifesciences-publications/SpokPaper</ext-link>)&#xA0;for a <b>Snakemake</b> pipeline and Circos configuration files.<li> A full <b>Snakemake</b> (<xref rid="bib54" ref-type="bibr"><u>Johannes and Rahmann, 2018</u></xref>) pipeline can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/johannessonlab/SpokPaper">https://github.com/johannessonlab/SpokPaper</ext-link>.<li> A full <b>Snakemake</b> pipeline is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/johannessonlab/SpokPaper">https://github.com/johannessonlab/SpokPaper</ext-link>.<li> [Johannes and Rahmann, 2018](          <person-group person-group-type="author">            <name>              <surname>Johannes</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <year>2018</year>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <volume>34</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31377530">31377530</a> <b>(1)</b> Kevlar: A Mapping-Free Framework for Accurate Discovery of <italic>De Novo</italic> Variants<ul><li>Kevlar is available as an open source software project and can be invoked via a Python API, a command-line interface, or a standard <b>Snakemake</b> workflow (<xref rid="bib17" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31335917">31335917</a> <b>(3)</b> Where did you come from, where did you go: Refining metagenomic analysis tools for horizontal gene transfer characterisation<ul><li>Daisy integration and integration with <b>Snakemake</b>.<li><b>Snakemake</b> is a common workflow management system [<xref rid="pcbi.1007208.ref027" ref-type="bibr"><u>27</u></xref>] which we used to implement the different steps of DaisyGPS.<li> To ensure compatibility, we reimplemented the Daisy workflow in <b>Snakemake</b> as well, and integrated both into a combined suite (called DaisySuite, see also <xref ref-type="fig" rid="pcbi.1007208.g002"><u>Fig 2</u></xref>).<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31824661">31824661</a> <b>(3)</b> Iron Hack - A symposium/hackathon focused on porphyrias, Friedreich&#x2019;s ataxia, and other rare iron-related diseases<ul><li> The primary work takes place in a custom <b>snakemake</b> script that aligns sequences with Hisat2 and then quantifies transcripts with Stringtie in a parallelized fashion across available machines and cores.<li>We planned the main<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/"><b>snakemake</b></ext-link> (version 5.4.2) to automate dispatching of jobs depending on the available cores and memory of a machine<sup><xref rid="ref-91" ref-type="bibr"><u>91</u></xref></sup>.<li> [91](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31297530">31297530</a> <b>(1)</b> De Novo Mutation Rate Estimation in Wolves of Known Pedigree<ul><li> The pipeline described above was implemented using <b>Snakemake</b> (<xref rid="msz159-B25" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) and is available on github (<ext-link ext-link-type="uri" xlink:href="https://github.com/emkoch/wolf-dnm-pipeline">https://github.com/emkoch/wolf-dnm-pipeline</ext-link>; last accessed July 8, 2019).<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31354646">31354646</a> <b>(1)</b> Interaction-Specific Changes in the Transcriptome of <italic>Polynucleobacter asymbioticus</italic> Caused by Varying Protistan Communities<ul><li> All steps performed in the analysis were run using the <b>Snakemake</b> workflow management system (v4.0.0, K&#xF6;ster and Rahmann, <xref rid="B46" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31289836">31289836</a> <b>(1)</b> Identifying, understanding, and correcting technical artifacts on the sex chromosomes in next-generation sequencing data<ul><li> We further provide links to data and <b>Snakemake</b> [<xref rid="bib51" ref-type="bibr"><u>51</u></xref>] workflows for all assembly and analysis steps on Github [<xref rid="bib38" ref-type="bibr"><u>38</u></xref>] and Zenodo [<xref rid="bib52" ref-type="bibr"><u>52</u></xref>].<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [38](<comment>XYalign. <ext-link ext-link-type="uri" xlink:href="https://github.com/SexChrLab/XYalign">https://github.com/SexChrLab/XYalign</ext-link>. Accessed 20 April 2019.</comment>) [52](<person-group person-group-type="author"><name name-style="western"><surname>Webster</surname><given-names>TH</given-names></name>, <name name-style="western"><surname>Couse</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Grande</surname><given-names>BM</given-names></name>, <etal>et al</etal>.</person-group><comment>XYalign: Version 1.1.6</comment><source>Zenodo</source>. <year>2019</year>, doi:<pub-id pub-id-type="doi">10.5281/zenodo.2635885.</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31328031">31328031</a> <b>(2)</b> BioShake: a Haskell EDSL for bioinformatics workflows<ul><li>High level feature comparison of BioShake with other execution engines (Toil, Cromwell), specification languages (WDL, CWL), and DSLs (NextFlow, <b>Snakemake</b>).<li><b>Snakemake</b></th>                <th rowspan="1" colspan="1">NextFlow</th>                <th rowspan="1" colspan="1">Toil</th>                <th rowspan="1" colspan="1">Cromwell</th>                <th rowspan="1" colspan="1">WDL</th>                <th rowspan="1" colspan="1">CWL</th>                <th rowspan="1" colspan="1">BioShake</th>              </tr>            </thead>            <tbody>              <tr>                <td rowspan="1" colspan="1">DSL.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31287816">31287816</a> <b>(1)</b> Managing genomic variant calling workflows with Swift/T<ul><li> In contrast, lower level systems with a command-line interface (CLI), such as <b>Snakemake</b> [<xref rid="pone.0211608.ref071" ref-type="bibr"><u>71</u></xref>], Luigi [<xref rid="pone.0211608.ref072" ref-type="bibr"><u>72</u></xref>], BcBio [<xref rid="pone.0211608.ref073" ref-type="bibr"><u>73</u></xref>], Bpipe [<xref rid="pone.0211608.ref074" ref-type="bibr"><u>74</u></xref>], are easier to maintain and share, provide good documentation and reproducibility, fault tolerance, and task automation; however, they require a lot more programming expertise.<li> [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [72](GitHub&#x2014;spotify/luigi; 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/spotify/luigi">https://github.com/spotify/luigi</ext-link>.) [73](<name><surname>Guimera</surname><given-names>RV</given-names></name>. <article-title>bcbio-nextgen: Automated, distributed next-gen sequencing pipeline</article-title>. <source>EMBnet j</source>. <year>2012</year>;<volume>17</volume>(<issue>B</issue>):<fpage>30</fpage><pub-id pub-id-type="doi">10.14806/ej.17.B.286</pub-id>) [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31510664">31510664</a> <b>(1)</b> Large-scale mammalian genome rearrangements coincide with chromatin interactions<ul><li>All results from this article can be reproduced by downloading the Hi-C data and running <b>Snakemake</b> (<xref rid="btz343-B21" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) as described at <ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/thekswenson/mammal_rearrange_interactions">https://bitbucket.org/thekswenson/mammal_rearrange_interactions</ext-link>.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31510660">31510660</a> <b>(3)</b> Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology<ul><li> In a separate repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/grst/immune_deconvolution_benchmark">https://github.com/grst/immune_deconvolution_benchmark</ext-link>), we provide a <b>snakemake</b> pipeline (<xref rid="btz363-B19" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) that allows to fully reproduce this benchmark and includes step-by-step instructions on how to test additional methods.<li> We use <b>Snakemake</b> 5.2.0 (<xref rid="btz363-B19" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), conda and bioconda (<xref rid="btz363-B13" ref-type="bibr">Gr&#xFC;ning <italic>et al.<li> A ready-to-run <b>Snakemake</b> pipeline, including preprocessed data, to reproduce the results is available from <ext-link ext-link-type="uri" xlink:href="https://github.com/grst/immune_deconvolution_benchmark">https://github.com/grst/immune_deconvolution_benchmark</ext-link>.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31269897">31269897</a> <b>(1)</b> iMAP: an integrated bioinformatics and visualization pipeline for microbiome data analysis<ul><li> The pre-release version of iMAP described in this manuscript (iMAP v1.0) is at the preliminary phase, and perhaps it lacks significant reproducibility aspects compared to the modern bioinformatics workflow management systems such as Nextflow [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], NextflowWorkbench [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], or <b>Snakemake</b> [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>].<li> [19](Di Tommaso P, Chatzou M, Prieto P, Palumbo E, Notredame C. Nextflow: A tool for deploying reproducible computational pipelines [version 1; not peer reviewed]. F1000Research 2015, 4(ISCB Comm J):430 (poster) (10.7490/f1000research.1110183.1).) [20](          <person-group person-group-type="author">            <name>              <surname>Thomer</surname>              <given-names>AK</given-names>            </name>            <name>              <surname>Twidale</surname>              <given-names>MB</given-names>            </name>            <name>              <surname>Guo</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Yoder</surname>              <given-names>MJ</given-names>            </name>          </person-group>          <article-title>NextflowWorkbench: reproducible and reusable workflows for beginners and experts</article-title>          <source>Conference on human factors in computing systems - proceedings</source>          <year>2016</year>        ) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31479207">31479207</a> <b>(8)</b> Insertion Pool Sequencing for Insertional Mutant Analysis in Complex Host&#x2010;Microbe Interactions<ul><li><italic toggle="yes">NOTE</italic>: iPool&#x2010;Seq pipeline is based on the workflow engine &#x201C;<b>snakemake</b>&#x201D; (K&#xF6;ster &amp; Rahmann, <xref rid="cppb20097-bib-0013" ref-type="bibr"><u>2012</u></xref>).<li><monospace><b>snakemake</b> data/Uhse_et_al.2018/expA.r1.dv.tab</monospace>.<li><monospace><b>snakemake</b> data/<italic toggle="yes">your_design</italic>/<italic toggle="yes">your_lib</italic>.trim.1.fq.gz</monospace>.<li><monospace><b>snakemake</b> data/<italic toggle="yes">your_design</italic>/<italic toggle="yes">your_lib</italic>.fastqc.1.html</monospace>.<li><monospace><b>snakemake</b> data/<italic toggle="yes">your_design</italic>/<italic toggle="yes">your_lib</italic>.fastqc.2.html</monospace>.<li><monospace><b>snakemake</b> data/<italic toggle="yes">your_design</italic>/<italic toggle="yes">your_lib</italic>.assign.bam</monospace>.<li><monospace><b>snakemake</b> data/<italic toggle="yes">your_design</italic>/<italic toggle="yes">your_lib</italic>.count.tab</monospace>.<li><monospace><b>snakemake</b> data/your_design/your_replicate.dv.tab</monospace>.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31343654">31343654</a> <b>(5)</b> Eye in a Disk: eyeIntegration Human Pan-Eye and Body Transcriptome Database Version 1.0<ul><li><b>Snakemake</b> Reproducible Pipeline.<li>While the sample search and metadata parsing in a semicurated process, the processing from the raw data to the creation of the SQLite EiaD database underlying eyeIntegration is wrapped in a <b>Snakemake</b> pipeline, which ensures full reproducibility of the results.<li>We built an automated pipeline for processing and analyzing all data for the web app using the program <b>Snakemake</b>, a python-based workflow management system that allows for efficient parallel execution of the analysis, facilitates reproduction by others, and simplifies long-term maintenance of the EiaD data (<xref ref-type="fig" rid="i1552-5783-60-8-3236-f02">Fig.<li> Instead run our <b>Snakemake</b> pipeline (available in the public domain at <ext-link ext-link-type="uri" xlink:href="https://www.github.com/davemcg/EiaD_build">https://www.github.com/davemcg/EiaD_build</ext-link>), adding your samples.<li> Wrapping all of the data processing steps in a <b>Snakemake</b> pipeline has several important advantages for the community: our code is publicly available for review, our analyses are reproducible, future sample updates can be streamlined in with less effort, and because all the processing is in modular pieces it is easier to add new analysis steps.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31253090">31253090</a> <b>(1)</b> <italic>Odyssey</italic>: a semi-automated pipeline for phasing, imputation, and analysis of genome-wide genetic data<ul><li>In the future, <italic>Odyssey&#x2019;s</italic> capabilities will be further improved via implementation into domain-specific language (DSL) implicit frameworks such as <b>Snakemake</b> [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] and by continuing to explore routes to optimize the pipeline to save time and space.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31253656">31253656</a> <b>(1)</b> The cancer cell proteome and transcriptome predicts sensitivity to targeted and cytotoxic drugs<ul><li>To parallelize drug sensitivity predictions, we used the workflow tool <b>Snakemake</b> (<xref rid="bib26" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31346594">31346594</a> <b>(2)</b> A set of <italic>AP2</italic>-like genes is associated with inflorescence branching and architecture in domesticated rice<ul><li> We used <italic><b>snakemake</b></italic> (<xref rid="CIT0039" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) to arrange analysis steps into workflows and monitor dependencies, and <italic>Singularity</italic> (<xref rid="CIT0041" ref-type="bibr">Kurtzer <italic>et al.<li> The final results and all intermediate steps can be exactly reproduced from the raw data with a single command using <italic><b>snakemake</b></italic> and <italic>Singularity</italic>.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31236809">31236809</a> <b>(1)</b> Detection of circulating tumor cells and circulating tumor DNA before and after mammographic breast compression in a cohort of breast cancer patients scheduled for neoadjuvant treatment<ul><li> Using a <b>Snakemake</b> workflow, reads in FASTQ format were aligned to the human reference genome GRCh38.p8 (including alternative sequences and decoys, and patched with dbSNP Build 147) using HISAT2 2.0.5 [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>] (with default options except --rna-strandness RF, --rg-id ${ID_NAME}, --rg PL:illumina, --rg PU:${UNIT}, --rg SM:${SAMPLE}), and duplicate reads were marked using SAMBLASTER 0.1.24.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25751142">25751142</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31221194">31221194</a> <b>(1)</b> Essential guidelines for computational method benchmarking<ul><li> More general tools for managing computational workflows, including <b>Snakemake</b> [<xref ref-type="bibr" rid="CR104"><u>104</u></xref>], Make, Bioconda [<xref ref-type="bibr" rid="CR105"><u>105</u></xref>], and conda, can be customized to capture setup information.<li> [104](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [105](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31185495">31185495</a> <b>(3)</b> GenPipes: an open-source framework for distributed and scalable genomic analyses<ul><li> In the second type, WMSs such as <b>Snakemake</b> [<xref rid="bib6" ref-type="bibr"><u>6</u></xref>], Nextflow [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], BPipe [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], and BigDataScript [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>] and declarative workflow description languages such as CWL or WDL are dedicated to providing a customizable framework to build bioinformatics pipelines.<li> It is important to note that GenPipes, as well as several other WMSs, like Nextflow [<xref rid="bib64" ref-type="bibr"><u>65</u></xref>] and <b>SnakeMake</b> [<xref rid="bib65" ref-type="bibr"><u>66</u></xref>], support community-developed pipelines; however, those have not been included in the comparison.<li><b>SnakeMake</b>.<li> [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a>) [65](<comment>NextFlow Community Pipelines. <ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core">https://github.com/nf-core</ext-link></comment>. Accessed Septmeber 2018.) [66](<comment>SnakeMake Community Pipelines. <ext-link ext-link-type="uri" xlink:href="https://github.com/snakemake-workflows">https://github.com/snakemake-workflows</ext-link></comment>. Accessed Septmeber 2018.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31159881">31159881</a> <b>(1)</b> NG-meta-profiler: fast processing of metagenomes using NGLess, a domain-specific language<ul><li> The classical Make tool, which was originally designed for compiling software, but has been used as the basis of a scientific pipeline system [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], uses its own internal language, as do the more modern pipelining tools <b>Snakemake</b> [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>] and nextflow [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>].<li> [39](          <person-group person-group-type="author">            <name>              <surname>Schwab</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Karrenbach</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Claerbout</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Making scientific computations reproducible</article-title>          <source>Comput Sci Eng</source>          <year>2000</year>          <volume>2</volume>          <fpage>61</fpage>          <lpage>67</lpage>          <pub-id pub-id-type="doi">10.1109/5992.881708</pub-id>        ) [40](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31160924">31160924</a> <b>(3)</b> Horizontal transfer of a retrotransposon between parasitic nematodes and the common shrew<ul><li> We used our <b>Snakemake</b> pipeline &#x2018;ConTest&#x2019; to test RTE1_Sar is a contaminant in shrews.<li><b>Snakemake</b> workflows to determine the taxon of origin of RTE encoding sequence reads.<li>We developed <b>Snakemake</b> [<xref ref-type="bibr" rid="CR43"><u>43</u></xref>] workflows to determine whether sequence reads with RTEs originated from the sequenced organism or from contamination.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31136709">31136709</a> <b>(1)</b> Minimally Empirical Double-Hybrid Functionals Trainedagainst the GMTKN55 Database: revDSD-PBEP86-D4, revDOD-PBE-D4, andDOD-SCAN-D4<ul><li>The reference geometries, charge and multiplicity information,and reference data were extracted from the ACCDB database of Morganteand Peverati.<sup><xref ref-type="bibr" rid="ref77"><u>77</u></xref></sup> While initial runs weremade with the help of the <b>Snakemake</b><sup><xref ref-type="bibr" rid="ref78"><u>78</u></xref></sup> workflowsdefined as part of ACCDB, once we had a full set of input files, weelected to use our own scripting.<li> [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30582189">30582189</a>) [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31134269">31134269</a> <b>(2)</b> snakePipes: facilitating flexible, scalable and integrative epigenomic analysis<ul><li> Computational frameworks, such as Galaxy (<xref rid="btz436-B8" ref-type="bibr">Goecks <italic>et al.</italic>, 2010</xref>), Nextflow (<xref rid="btz436-B4" ref-type="bibr">Di Tommaso <italic>et al.</italic>, 2017</xref>) and <b>snakemake</b> (<xref rid="btz436-B11" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) address this issue to some extent by allowing users to create their own workflows, or adopt workflows from public repositories.<li>snakePipes employs <b>snakemake</b> (<xref rid="btz436-B11" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) as its core workflow language, which benefits from easy readability of the code, widespread adoption and scalability to most clusters and cloud platforms.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31127303">31127303</a> <b>(1)</b> A Fast Likelihood Method to Reconstruct and Visualize Ancestral Scenarios<ul><li>All of our data, trees, ACRs, and <b>Snakemake</b> pipelines (<xref rid="msz131-B41" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) used to reconstruct the trees and analyze them are available from <ext-link ext-link-type="uri" xlink:href="https://pastml.pasteur.fr">https://pastml.pasteur.fr</ext-link>; Last accessed June 1, 2019.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31123324">31123324</a> <b>(1)</b> ATAC-seq reveals alterations in open chromatin in pancreatic islets from subjects with type 2 diabetes<ul><li> The whole ATAC-seq workflow was implemented in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup>.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31002208">31002208</a> <b>(1)</b> Latitudinal variation in soil nematode communities under climate warming&#x2010;related range&#x2010;expanding and native plants<ul><li>The obtained raw 18S rDNA sequence reads were curated in the Hydra pipeline (de Hollander, <xref rid="gcb14657-bib-0013" ref-type="ref"><u>2017</u></xref>) implemented in <b>Snakemake</b> (K&#xF6;ster &amp; Rahmann, <xref rid="gcb14657-bib-0032" ref-type="ref"><u>2012</u></xref>); in short, after filtering contaminants and removing barcodes, the forward reads were used for annotation.<li> [2017](<string-name><surname>de Hollander</surname>, <given-names>M.</given-names></string-name> (<year>2017</year>). <source xml:lang="en">nioo&#x2010;knaw/hydra: 1.3.3 (Version 1.3.3)</source>. Retrieved from <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/884028#.XKw9spgzaUk">https://zenodo.org/record/884028#.XKw9spgzaUk</ext-link>: Zenodo) [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31092189">31092189</a> <b>(1)</b> Accurate sequence variant genotyping in cattle using variation-aware genome graphs<ul><li>We implemented all pipelines using <italic><b>Snakemake</b></italic> (version 5.2.0) [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>].<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31089679">31089679</a> <b>(1)</b> Ultra-deep, long-read nanopore sequencing of mock microbial community standards<ul><li> Our pipeline was orchestrated with <b>Snakemake</b> [<xref rid="bib30" ref-type="bibr"><u>36</u></xref>]; the workflow is available from our repository.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31088905">31088905</a> <b>(21)</b> ARMOR: An <underline>A</underline>utomated <underline>R</underline>eproducible <underline>MO</underline>dular Workflow for Preprocessing and Differential Analysis of <underline>R</underline>NA-seq Data<ul><li> The application is implemented using the <b>Snakemake</b> workflow management system (<xref rid="bib15" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>), and allows the user to easily perform quality assessment, adapter trimming, genome alignment, transcript and gene abundance quantification, differential expression analysis and geneset analyses with a simple command, after specifying the required reference files and information about the experimental design in a configuration file.<li> While high-performance computing environments and cloud computing are not specifically targeted, <b>Snakemake</b> enables the usage of a cluster without the need to modify the workflow itself.<li>ARMOR is implemented as a modular <b>Snakemake</b> (<xref rid="bib15" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) workflow, and the execution of the individual steps is controlled by the provided Snakefile.<li> <b>Snakemake</b> will automatically keep track of the dependencies between the different parts of the workflow; rerunning the workflow will thus only regenerate results that are out of date or missing given these dependencies.<li>By default, all software packages that are needed for the analysis will be installed in an auto-generated conda environment, which will be automatically activated by <b>Snakemake</b> before the execution of each rule.<li>First, the user needs to have a recent version of <b>Snakemake</b> and conda installed.<li>          <monospace><b>snakemake</b>&#x2010;&#x2010;use-conda</monospace>        .<li>          <monospace><b>snakemake</b></monospace>        .<li> For this, <b>Snakemake</b> and the tools listed in envs/environment.yaml need to be manually installed, in addition to a recent version of R and the R packages listed in scripts/install_pkgs.R.<li> If the specified R library does not contain the required packages, <b>Snakemake</b> will try to install them (<italic>i.e.<li> A detailed summary of the output files generated by the workflow, including the shell command that was used to generate each of them, the time of creation, and information about whether the associated inputs, code or parameters have since been updated, can be obtained at any time by invoking <b>Snakemake</b> with the flag <monospace>-D</monospace> (or<monospace>&#x2010;&#x2010;detailed-summary</monospace>).<li> Using the benchmark directive of <b>Snakemake</b>, ARMOR also automatically generates additional text files summarizing the run time and peak memory usage of each executed rule.<li> <monospace><b>snakemake</b>&#x2010;&#x2010;use-conda</monospace> will create a separate conda environment in each subdirectory, which means that the installed software may be duplicated.<li> It is also possible to explicitly specify the path to the desired config.yaml configuration file when <b>snakemake</b> is called:.<li>          <monospace><b>snakemake</b>&#x2010;&#x2010;configfile config.yaml</monospace>        .<li>By taking advantage of the <b>Snakemake</b> framework, ARMOR makes file and software organization relatively autonomous.<li>Once everything is set up, running <monospace><b>snakemake</b></monospace>, which operates on the rules in the Snakefile, will construct the hierarchy of instructions to execute, given the specifications in the config.yaml file.<li> <b>Snakemake</b> automatically determines the dependencies between the rules and will invoke the instructions in a logical order.<li>          <monospace><b>snakemake</b>&#x2010;&#x2010;use-conda&#x2010;&#x2010;cores 20 \</monospace>        .<li>The required resources for the generation of each output file (grouped by the <b>Snakemake</b> rule) in the real data walk-through, as reported by the benchmarking directive of <b>Snakemake</b>.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31193391">31193391</a> <b>(5)</b> Computational Methods for Mapping, Assembly and Quantification for Coding and Non-coding Transcripts<ul><li> <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="ir0025">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>), Nextflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/nextflow-io" id="ir0030">https://github.com/nextflow-io</ext-link>) and Docker (<ext-link ext-link-type="uri" xlink:href="https://github.com/ngs-docs/2015-nov-docker/" id="ir0035">https://github.com/ngs-docs/2015-nov-docker/</ext-link>)are different platforms with pipelines for reproducible transcriptome analysis.<li> For example, Visualization Pipeline for RNA-Seq analysis (VIPER) is a user-friendly and comprehensive analysis workflow that uses <b>Snakemake</b> [<xref rid="bb0870" ref-type="bibr"><u>174</u></xref>].<li> Another <b>Snakemake</b>-based pipeline, hppRNA [<xref rid="bb0875" ref-type="bibr"><u>175</u></xref>], is a parameter-free pipeline that can be used for numerous samples.<li> [174](          <person-group person-group-type="author">            <name>              <surname>Cornwell</surname>              <given-names>M.</given-names>            </name>          </person-group>          <article-title>VIPER: visualization pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis</article-title>          <source>BMC Bioinform</source>          <volume>19</volume>          <issue>1</issue>          <year>Dec. 2018</year>          <fpage>135</fpage>        ) [175](          <person-group person-group-type="author">            <name>              <surname>Wang</surname>              <given-names>D.</given-names>            </name>          </person-group>          <article-title>hppRNA&#x2014;a Snakemake-based handy parameter-free pipeline for RNA-Seq analysis of numerous samples</article-title>          <source>Brief Bioinform</source>          <volume>19</volume>          <issue>4</issue>          <year>Jan. 2017</year>          <fpage>bbw143</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31060495">31060495</a> <b>(1)</b> Tree reconciliation combined with subsampling improves large scale inference of orthologous group hierarchies<ul><li> An open source python implementation using the <b>Snakemake</b> workflow engine [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/meringlab/og_consistency_pipeline">https://github.com/meringlab/og_consistency_pipeline</ext-link>.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31045203">31045203</a> <b>(1)</b> TAGOOS: genome-wide supervised learning of non-coding loci associated to complex phenotypes<ul><li>Training and test pipelines were implemented with <b>Snakemake</b> (<xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Figure S1</u></xref>) (<xref rid="B43" ref-type="bibr"><u>43</u></xref>).<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a> <b>(6)</b> SciPipe: A workflow library for agile development of complex and dynamic bioinformatics pipelines<ul><li> Text-based tools such as <b>Snakemake</b>&#xA0;[<xref rid="bib33" ref-type="bibr"><u>9</u></xref>], Nextflow&#xA0;[<xref rid="bib17" ref-type="bibr"><u>10</u></xref>], BPipe&#xA0;[<xref rid="bib59" ref-type="bibr"><u>11</u></xref>], Cuneiform&#xA0;[<xref rid="bib7" ref-type="bibr"><u>12</u></xref>], and Pachyderm&#xA0;[<xref rid="bib55" ref-type="bibr"><u>13</u></xref>], on the other hand, are implemented as domain-specific languages (DSLs), which can often provide a higher level of flexibility, at the expense of the ease of use of a graphical user interface.<li> Text-based tools implemented as DSLs, such as <b>Snakemake</b>, Nextflow, BPipe, Pachyderm, and Cuneiform, do not have this limitation but have other characteristics that might be problematic for for complex workflows.<li>For example, <b>Snakemake</b> is dependent on file-naming strategies for defining dependencies, which can in some situations be limiting, and also uses a &#x201C;pull-based&#x201D; scheduling strategy (the workflow is invoked by asking for a specific output file, whereafter all tasks required for reproducing the file will be executed).<li> <b>Snakemake</b> also performs scheduling and execution of the workflow graph in separate stages, meaning that it does not support dynamic scheduling.<li>With pull-based workflow tools such as <b>Snakemake</b> or Luigi, it is easy to reproduce a particular output file on-demand because the scheduling mechanism is optimized for the use case of asking for a specific file and calculating all the tasks required to be executed based on that.<li>Furthermore, reproducing specific output files is not as natural and easy as with pull-based tools like <b>Snakemake</b>, although SciPipe provides a mechanism to partly resolve this problem.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<person-group person-group-type="author"><name name-style="western"><surname>Di&#xA0;Tommaso</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Floden</surname><given-names>EW</given-names></name>, <etal>et al</etal>.</person-group><article-title>Nextflow enables reproducible computational workflows</article-title>. <source>Nat Biotech</source>. <year>2017</year>;<volume>35</volume>(<issue>4</issue>):<fpage>316</fpage>&#x2013;<lpage>9</lpage>.) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [12](<person-group person-group-type="author"><name name-style="western"><surname>Brandt</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Bux</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Leser</surname><given-names>U</given-names></name></person-group><article-title>Cuneiform: A functional language for large scale scientific data analysis</article-title>. In: <source>Proceedings of the Workshops of the EDBT/ICDT, Brussels, Belgium, 2015</source>. <fpage>17</fpage>&#x2013;<lpage>26</lpage>.) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30101309">30101309</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31222200">31222200</a> <b>(1)</b> Bioportainer Workbench: a versatile and user-friendly system that integrates implementation, management, and use of bioinformatics resources in Docker environments<ul><li> Other projects, such as <b>Snakemake</b> [<xref rid="bib15" ref-type="bibr"><u>15</u></xref>], Common Workflow Language [<xref rid="bib16" ref-type="bibr"><u>16</u></xref>], and NextFlow [<xref rid="bib17" ref-type="bibr"><u>17</u></xref>] have provided frameworks for the implementation, administration, and execution of complex pipelines and workflows within Docker containers, allowing simultaneous management and coordinated launching of the various software involved in such analyses.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<person-group person-group-type="author"><name name-style="western"><surname>Amstutz</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Crusoe</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <etal>et al</etal>.</person-group><comment>Common Workflow Language, v1.0.</comment><year>2016</year><ext-link ext-link-type="uri" xlink:href="https://www.commonwl.org/">https://www.commonwl.org/.</ext-link><comment>Accessed 19 June 2018.</comment>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31037220">31037220</a> <b>(2)</b> Intra- and interpatient evolution of enterovirus D68 analyzed by whole-genome deep sequencing<ul><li> All these steps were chained using the workflow engine <b>snakemake</b> (<xref rid="vez007-B18" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li> The combined dataset was analyzed using nextstrain (<xref rid="vez007-B11" ref-type="bibr">Hadfield <italic>et al.</italic> 2018</xref>) and <b>snakemake</b> (<xref rid="vez007-B18" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li> [K&#xF6;ster and Rahmann 2012](<person-group person-group-type="author"><name name-style="western"><surname>K&#xF6;ster</surname><given-names>J.</given-names></name>, <name name-style="western"><surname>Rahmann</surname><given-names>S.</given-names></name></person-group> (<year>2012</year>) &#x2018;<article-title>Snakemake-a Scalable Bioinformatics Workflow Engine</article-title>&#x2019;, <source>Bioinformatics (Oxford, England)</source>, <volume>28</volume>: <fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) [K&#xF6;ster and Rahmann 2012](<person-group person-group-type="author"><name name-style="western"><surname>K&#xF6;ster</surname><given-names>J.</given-names></name>, <name name-style="western"><surname>Rahmann</surname><given-names>S.</given-names></name></person-group> (<year>2012</year>) &#x2018;<article-title>Snakemake-a Scalable Bioinformatics Workflow Engine</article-title>&#x2019;, <source>Bioinformatics (Oxford, England)</source>, <volume>28</volume>: <fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31024771">31024771</a> <b>(1)</b> Bacterial community composition and diversity of two different forms of an organic residue of bioenergy crop<ul><li>The sequences were analyzed by Mothur version 1.33.2 (<xref rid="ref-60" ref-type="bibr"><u>Schloss, 2010</u></xref>) combined with <b>Snakemake</b>r (<xref rid="ref-31" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>).<li> [Schloss, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20628621">20628621</a>) [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30967549">30967549</a> <b>(2)</b> iFISH is a publically available resource enabling versatile DNA FISH to study genome architecture<ul><li>To test the possible effect of local DNA accessibility on the quality of the 330 probes, we first downloaded previously published HAP1 ATAC-seq data<sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> from the GEO repository GSE111047 (SRR6766909, SRR6766910, SRR6766911), and processed them using the Parker Lab&#x2019;s ATAC-seq processing pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/ParkerLab/ATACseq-<b>Snakemake</b>">https://github.com/ParkerLab/ATACseq-<b>Snakemake</b></ext-link>).<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30082317">30082317</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30952863">30952863</a> <b>(1)</b> Root traits and belowground herbivores relate to plant&#x2013;soil feedback variation among congeners<ul><li>The obtained raw 16S and 18S rRNA gene sequence reads were curated in the Hydra pipeline (nioo-knaw/hydra (Version 1.3.3)) implemented in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup>; in short, after filtering contaminants and removing barcodes, 16S rDNA reads were merged with the fastq_mergepairs option of vsearch<sup><xref ref-type="bibr" rid="CR57"><u>57</u></xref></sup>, whereas for the 18S rRNA gene data the forward reads were used.<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [57](          <person-group person-group-type="author">            <name>              <surname>Rognes</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Flouri</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Nichols</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Quince</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Mah&#xE9;</surname>              <given-names>F</given-names>            </name>          </person-group>          <article-title>VSEARCH: a versatile open source tool for metagenomics</article-title>          <source>PeerJ Preprints</source>          <year>2016</year>          <volume>4</volume>          <fpage>e2409v2401</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30949696">30949696</a> <b>(1)</b> An integrative method to predict signalling perturbations for cellular transitions<ul><li> It is available as a <b>Snakemake</b> pipeline (<xref rid="B53" ref-type="bibr"><u>53</u></xref>) with all necessary datasets at: <ext-link ext-link-type="uri" xlink:href="https://git-r3lab.uni.lu/gaia.zaffaroni/INCanTeSIMO">https://git-r3lab.uni.lu/gaia.zaffaroni/INCanTeSIMO</ext-link>.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31001414">31001414</a> <b>(1)</b> Correction of gene model annotations improves isoform abundance estimates: the example of ketohexokinase (<italic>Khk</italic>)<ul><li> All workflows were orchestrated using <b>Snakemake</b> v 3.13.3<sup><xref rid="ref-45" ref-type="bibr"><u>45</u></xref></sup>.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30935387">30935387</a> <b>(1)</b> Conbase: a software for unsupervised discovery of clonal somatic mutations in single cells through read phasing<ul><li>For simulation data, a <b>Snakemake</b> pipeline was used to run all the four methods (Conbase, Monovar, SCcaller, and LiRA), see&#xA0;<ext-link ext-link-type="uri" xlink:href="https://github.com/joannahard/Genome_Biology_2019">https://github.com/joannahard/Genome_Biology_2019</ext-link>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30909881">30909881</a> <b>(2)</b> UTAP: User-friendly Transcriptome Analysis Pipeline<ul><li>The UTAP system is composed of a <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>] workflow system backend, and <italic>Python</italic> (v2.7) and a <italic>Django</italic> (v1.11) - based web user interface (WUI) through which users can run analyses.<li><italic><b>Snakemake</b></italic> bundles in-house scripts (written in <italic>Python</italic> and <italic>R</italic>) and public bioinformatics tools for completing stepwise processes.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30911144">30911144</a> <b>(1)</b> Range-expansion effects on the belowground plant microbiome<ul><li> All bioinformatics steps were implemented with a publicly available workflow made with <ext-link ext-link-type="uri" xlink:href="https://gitlab.bioinf.nioo.knaw.nl/amplicon-metagenomics/illumina_paired_end"><b>Snakemake</b></ext-link><xref rid="R88" ref-type="bibr"><u>88</u></xref>.<li> [88](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30902113">30902113</a> <b>(16)</b> Sunbeam: an extensible pipeline for analyzing metagenomic sequencing experiments<ul><li> It features error handling, task resumption, and parallel computing capabilities thanks to its implementation in the <b>Snakemake</b> workflow language [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>].<li> Each step is implemented as a rule in the <b>Snakemake</b> workflow framework [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>].<li> Implementation of the workflow in <b>Snakemake</b> offered several advantages, including workflow assembly, parallelism, and ability to resume following an interruption.<li> <b>Snakemake</b> determines which steps require outputs from other steps in order to form a directed acyclic graph (DAG) of dependencies when the workflow is started, so the order of execution can change to suit available resources.<li> The DAG structure also allows <b>Snakemake</b> to identify steps that can operate in parallel (e.g., on other processors or compute nodes) if requested by the user.<li> <b>Snakemake</b> manages the scheduling and is compatible with job submission systems on shared clusters.<li> This is accomplished in two ways: first, the dependency DAG created by <b>Snakemake</b> allows Sunbeam&#x2019;s execution to continue with other steps executing in parallel if a step fails.<li>Sunbeam also inherits all of <b>Snakemake</b>&#x2019;s error-handling abilities, including the ability to deal with NFS filesystem latency (via the &#x201C;--latency-wait&#x201D; option), and automatically re-running failed rules (via the &#x201C;--restart-times&#x201D; option).<li> option forces <b>Snakemake</b> to wait extra time for the filesystem to catch up.<li> Other workflow systems, such as the Common Workflow Language, allow the workflow to be specified separately from the file paths, and for this reason may offer more safety than the <b>Snakemake</b> framework [<xref ref-type="bibr" rid="CR55"><u>55</u></xref>].<li> In normal use, however, we found that the precautions taken by the <b>Snakemake</b> framework were sufficient to run on shared NFS filesystems at our institutions under typical load.<li> Extensions take the form of supplementary rules written in the <b>Snakemake</b> format and define the additional steps to be executed.<li>Architecture/usage</td></tr><tr><td>&#x2003;Dependency management</td><td>Conda</td><td>Bash</td><td>Pip (partial)</td><td/><td>Conda</td></tr><tr><td>&#x2003;Modularity</td><td><b>Snakemake</b>.<li>Architecture/usage</td></tr><tr><td>&#x2003;Dependency management</td><td>Conda</td><td>Bash</td><td>Pip (partial)</td><td/><td>Conda</td></tr><tr><td>&#x2003;Modularity</td><td><b>Snakemake</b></td><td/><td/><td>Perl modules</td><td><b>Snakemake</b>.<li> Tools used by each pipeline: trimmomatic [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>], cutadapt [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>], tadpole [<xref ref-type="bibr" rid="CR78"><u>78</u></xref>], fastqc [<xref ref-type="bibr" rid="CR47"><u>47</u></xref>], FaQCs [<xref ref-type="bibr" rid="CR79"><u>79</u></xref>], BBDuk2 [<xref ref-type="bibr" rid="CR80"><u>80</u></xref>], DUST [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>], TRF [<xref ref-type="bibr" rid="CR81"><u>81</u></xref>], VSEARCH [<xref ref-type="bibr" rid="CR82"><u>82</u></xref>], bwa [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>], bowtie2 [<xref ref-type="bibr" rid="CR83"><u>83</u></xref>], BBMap [<xref ref-type="bibr" rid="CR84"><u>84</u></xref>], KRAKEN [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>], SNAP [<xref ref-type="bibr" rid="CR85"><u>85</u></xref>], MUMmer [<xref ref-type="bibr" rid="CR86"><u>86</u></xref>], JBrowse [<xref ref-type="bibr" rid="CR87"><u>87</u></xref>], GOTTCHA [<xref ref-type="bibr" rid="CR88"><u>88</u></xref>], MetaPhlAn [<xref ref-type="bibr" rid="CR58"><u>58</u></xref>], DIAMOND [<xref ref-type="bibr" rid="CR89"><u>89</u></xref>], FastTree [<xref ref-type="bibr" rid="CR90"><u>90</u></xref>], MEGAHIT [<xref ref-type="bibr" rid="CR51"><u>51</u></xref>], SPAdes [<xref ref-type="bibr" rid="CR91"><u>91</u></xref>], Minimo [<xref ref-type="bibr" rid="CR92"><u>92</u></xref>], Prodigal [<xref ref-type="bibr" rid="CR52"><u>52</u></xref>], BLASTp [<xref ref-type="bibr" rid="CR53"><u>53</u></xref>], Prokka [<xref ref-type="bibr" rid="CR93"><u>93</u></xref>], BLASTn [<xref ref-type="bibr" rid="CR53"><u>53</u></xref>], eggNOG [<xref ref-type="bibr" rid="CR94"><u>94</u></xref>], ENZYME [<xref ref-type="bibr" rid="CR95"><u>95</u></xref>], dbCAN [<xref ref-type="bibr" rid="CR96"><u>96</u></xref>], Primer3 [<xref ref-type="bibr" rid="CR97"><u>97</u></xref>], RAPSearch [<xref ref-type="bibr" rid="CR98"><u>98</u></xref>], RAxML [<xref ref-type="bibr" rid="CR99"><u>99</u></xref>], conda [<xref ref-type="bibr" rid="CR43"><u>43</u></xref>], PhaME [<xref ref-type="bibr" rid="CR100"><u>100</u></xref>], <b>Snakemake</b> [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>], SAMtools [<xref ref-type="bibr" rid="CR54"><u>54</u></xref>].<li>Here, we introduce Sunbeam, a <b>Snakemake</b>-based pipeline for analyzing shotgun metagenomic data with a focus on reproducible analysis, ease of deployment, and use.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [55](Chapman B, Chilton J, Heuer M, Kartashov A, Leehr D, M&#xE9;nager H, et al. Common workflow language, v1.0. Specification, common workflow language working group. Amstutz P, Crusoe MR, Tijani&#x107; N, editors. 2016.) [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24695404">24695404</a>) [45](Martin M. Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet J. 2015;17:1&#x2013;3.) [78](JGI. Tadpole. 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/tadpole-guide/">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/tadpole-guide/</ext-link>) [47](BabrahamBioinformatics. FastQC. 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>) [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24383880">24383880</a>) [80](JGI. BBDuk. 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbduk-guide/">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbduk-guide/</ext-link>) [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16796549">16796549</a>) [81](Benson G. Tandem repeats finder: a program to analyze DNA sequences. Nucleic Acids Research. 1999;27:573&#x2013;80.) [82](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27781170">27781170</a>) [48](Li H, Li H, Durbin R, Durbin R. Fast and accurate short read alignment with Burrows-Wheeler transform. Bioinformatics. 2009;25:1754&#x2013;60.) [83](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [84](JGI. BBMap. 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbmap-guide/">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbmap-guide/</ext-link>) [49](Wood DE, Salzberg SL. Kraken: ultrafast metagenomic sequence classification using exact alignments. Genome Biol. 2014;15:R46.) [85](Zaharia M, Bolosky WJ, Curtis K, Fox A, Patterson D, Shenker S, et al. Faster and more accurate sequence alignment with SNAP. arXiv. 2011;1111:e5572v1.) [86](          <person-group person-group-type="author">            <name>              <surname>Mar&#xE7;ais</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Delcher</surname>              <given-names>AL</given-names>            </name>            <name>              <surname>Phillippy</surname>              <given-names>AM</given-names>            </name>            <name>              <surname>Coston</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Salzberg</surname>              <given-names>SL</given-names>            </name>            <name>              <surname>Zimin</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>MUMmer4: a fast and versatile genome alignment system</article-title>          <source>PLoS Comput Biol</source>          <year>2018</year>          <volume>14</volume>          <fpage>1</fpage>          <lpage>14</lpage>          <pub-id pub-id-type="doi">10.1371/journal.pcbi.1005944</pub-id>        ) [87](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19570905">19570905</a>) [88](Freitas TAK, Li P-E, Scholz MB, Chain PSG. Accurate read-based metagenome characterization using a hierarchical suite of unique signatures. Nucleic Acids Res. 2015;43:e69.) [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26418763">26418763</a>) [89](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25402007">25402007</a>) [90](Price MN, Dehal PS, Arkin AP. FastTree 2 - approximately maximum-likelihood trees for large alignments. PLoS One. 2010;5:e9490.) [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27012178">27012178</a>) [91](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22506599">22506599</a>) [92](Treangen TJ, Sommer DD, Angly FE, Koren S, Pop M. Next generation sequence assembly with AMOS. Curr Protoc Bioinforma. 2011;33:11.8.1&#x2013;11.8.18.) [52](Hyatt D, Chen G-L, LoCascio PF, Land ML, Larimer FW, Hauser LJ. Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics. 2010;11:119.) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=2231712">2231712</a>) [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24642063">24642063</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=2231712">2231712</a>) [94](          <person-group person-group-type="author">            <name>              <surname>Jensen</surname>              <given-names>LJ</given-names>            </name>            <name>              <surname>Julien</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Kuhn</surname>              <given-names>M</given-names>            </name>            <name>              <surname>von Mering</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Muller</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Doerks</surname>              <given-names>T</given-names>            </name>            <etal/>          </person-group>          <article-title>eggNOG: automated construction and annotation of orthologous groups of genes</article-title>          <source>Nucleic Acids Res</source>          <year>2008</year>          <volume>36</volume>          <fpage>250</fpage>          <lpage>254</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkm796</pub-id>        ) [95](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=10592255">10592255</a>) [96](          <person-group person-group-type="author">            <name>              <surname>Yin</surname>              <given-names>Y</given-names>            </name>            <name>              <surname>Mao</surname>              <given-names>X</given-names>            </name>            <name>              <surname>Yang</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Chen</surname>              <given-names>X</given-names>            </name>            <name>              <surname>Mao</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Xu</surname>              <given-names>Y</given-names>            </name>          </person-group>          <article-title>DbCAN: a web resource for automated carbohydrate-active enzyme annotation</article-title>          <source>Nucleic Acids Res</source>          <year>2012</year>          <volume>40</volume>          <fpage>445</fpage>          <lpage>451</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gks479</pub-id>        ) [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21908400">21908400</a>) [98](Ye Y, Choi JH, Tang H. RAPSearch: a fast protein similarity search tool for short reads. BMC Bioinformatics. 2011;12:159.) [99](          <person-group person-group-type="author">            <name>              <surname>Stamatakis</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Ludwig</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Meier</surname>              <given-names>H</given-names>            </name>          </person-group>          <article-title>RAxML-II: a program for sequential, parallel and distributed inference of large phylogenetic trees</article-title>          <source>Concurr Comput Pract Exp</source>          <year>2005</year>          <volume>17</volume>          <fpage>1705</fpage>          <lpage>1723</lpage>          <pub-id pub-id-type="doi">10.1002/cpe.954</pub-id>        ) [43](Anaconda INC. Conda. 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.org">https://anaconda.org</ext-link>) [100](Ahmed SA, Lo C-C, Li P-E, Davenport KW, Chain PSG. From raw reads to trees: whole genome SNP phylogenetics across the tree of life. bioRxiv. 2015:032250 Available from: <ext-link ext-link-type="uri" xlink:href="http://biorxiv.org/content/early/2015/11/19/032250.abstract">http://biorxiv.org/content/early/2015/11/19/032250.abstract</ext-link>.) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30889192">30889192</a> <b>(1)</b> A novel transcriptome-derived SNPs array for tench (<italic>Tinca tinca</italic> L.)<ul><li> These procedures were performed using custom scripts in Python, R (R Core Team 2016), SAMtools [<xref rid="pone.0213992.ref062" ref-type="bibr"><u>62</u></xref>] and <b>Snakemake</b> [<xref rid="pone.0213992.ref063" ref-type="bibr"><u>63</u></xref>].<li> [62](<name><surname>Li</surname><given-names>H</given-names></name>, <name><surname>Handsaker</surname><given-names>B</given-names></name>, <name><surname>Wysoker</surname><given-names>A</given-names></name>, <name><surname>Fennell</surname><given-names>T</given-names></name>, <name><surname>Ruan</surname><given-names>J</given-names></name>, <name><surname>Homer</surname><given-names>N</given-names></name>, <etal>et al</etal><article-title>The Sequence Alignment/Map format and SAMtools</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2009</year>;<volume>25</volume>(<issue>16</issue>):<fpage>2078</fpage>&#x2013;<lpage>9</lpage>.) [63](<name><surname>Koster</surname><given-names>J</given-names></name>, <name><surname>Rahmann</surname><given-names>S</given-names></name>. <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2012</year>;<volume>28</volume>(<issue>19</issue>):<fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30936858">30936858</a> <b>(1)</b> Belowground Consequences of Intracontinental Range-Expanding Plants and Related Natives in Novel Environments<ul><li> All steps where implemented in a workflow made with <b>Snakemake</b> (<xref rid="B36" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30851093">30851093</a> <b>(1)</b> Interoperable and scalable data analysis with microservices: applications in metabolomics<ul><li> In fact, our microservice architecture supports other major workflow engines such as Nextflow (<xref rid="btz160-B13" ref-type="bibr">Di Tommaso <italic>et al.</italic>, 2017</xref>) or <b>Snakemake</b> (<xref rid="btz160-B24" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30847485">30847485</a> <b>(1)</b> Functional protein representations from biological networks enable diverse cross-species inference<ul><li> We executed software pipelines for performing the experiments in part using the <monospace><b>Snakemake</b></monospace> software (<xref rid="B46" ref-type="bibr"><u>46</u></xref>).<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30863790">30863790</a> <b>(1)</b> A Neuron-Optimized CRISPR/dCas9 Activation System for Robust and Specific Gene Regulation<ul><li>Paired-end FASTQ files were uploaded to the University of Alabama at Birmingham&#x2019;s High Performance Computer cluster for custom bioinformatics analysis using a pipeline built with <b>snakemake</b> (<xref rid="B31" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>; v5.1.4).<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30863792">30863792</a> <b>(1)</b> Comprehensive Virulence Gene Profiling of Bovine Non-<italic>aureus</italic> Staphylococci Based on Whole-Genome Sequencing Data<ul><li> An in-house pipeline, implemented in <b>Snakemake</b> workflow engine (<xref rid="B126" ref-type="bibr"><u>126</u></xref>), was used to assemble and annotate genomes.<li> [126](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30837458">30837458</a> <b>(3)</b> The <italic>Wolbachia</italic> mobilome in <italic>Culex pipiens</italic> includes a putative plasmid<ul><li> Except for the manual refinement step, we automated the processing of our raw sequencing data with a <b>snakemake</b><sup><xref ref-type="bibr" rid="CR79"><u>79</u></xref></sup> workflow anvi&#x2019;o v5 implements (<ext-link ext-link-type="uri" xlink:href="http://merenlab.org/2018/07/09/anvio-<b>snakemake</b>-workflows/">http://merenlab.org/2018/07/09/anvio-<b>snakemake</b>-workflows/</ext-link>), and the Code availablility section reports necessary configuration files to fully reproduce this analysis.<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30821816">30821816</a> <b>(2)</b> Common workflow language (CWL)-based software pipeline for <italic>de novo</italic> genome assembly from long- and short-read data<ul><li>One possible approach would be to employ frameworks, such as <b>SnakeMake</b> [<xref rid="bib47" ref-type="bibr"><u>47</u></xref>], Ruffus/Rubra [<xref rid="bib48" ref-type="bibr"><u>48</u></xref>], Toil [<xref rid="bib49" ref-type="bibr"><u>49</u></xref>], and Rabix [<xref rid="bib50" ref-type="bibr"><u>50</u></xref>], or to use the common workflow language (CWL) [<xref rid="bib51" ref-type="bibr"><u>51</u></xref>] for workflows [<xref rid="bib52" ref-type="bibr"><u>52</u></xref>].<li> For instance, <b>SnakeMake</b> has been used in multiple workflows relating to RNA sequencing analyses [<xref rid="bib53" ref-type="bibr"><u>53</u></xref>], and Rubra is used in workflows such as RedDog [<xref rid="bib54" ref-type="bibr"><u>54</u></xref>] to infer single-nucleotide polymorphism (SNP) datasets derived from bacterial populations for subsequent phylogenetic analyses.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [50](<person-group person-group-type="author"><name name-style="western"><surname>Kaushik</surname><given-names>G</given-names></name>, <name name-style="western"><surname>Ivkovic</surname><given-names>S</given-names></name>, <name name-style="western"><surname>Simonovic</surname><given-names>J</given-names></name>, <etal>et al</etal>.</person-group><article-title>Rabix: an open-source workflow executor supporting recomputability and interoperability of workflow descriptions</article-title>In: <source>Pac Symp Biocomput</source>, <publisher-name>World Scientific Publishing Company</publisher-name>, <publisher-loc>https://psb.stanford.edu/psb-online/proeedings/psb17</publisher-loc><year>2017</year>, pp. <fpage>154</fpage>&#x2013;<lpage>65</lpage>.) [51](<person-group person-group-type="author"><name name-style="western"><surname>Amstutz</surname><given-names>P</given-names></name>, <name name-style="western"><surname>Crusoe</surname><given-names>MR</given-names></name>, <name name-style="western"><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, <etal>et al</etal>.</person-group><article-title>Common Workflow Language v1. 0</article-title>. <source>Figshare</source>. <year>2016</year>; <comment>doi:10.6084/m9.figshare.3115156.v2</comment>.) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27013646">27013646</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28581496">28581496</a>) [54](<person-group person-group-type="author"><name name-style="western"><surname>Edwards</surname><given-names>D</given-names></name>, <name name-style="western"><surname>Pope</surname><given-names>B</given-names></name>, <name name-style="western"><surname>Holt</surname><given-names>K</given-names></name></person-group><source>Reddog tutorial. GitHub</source>: <ext-link ext-link-type="uri" xlink:href="https://githubcom/katholt/RedDog/blob/master/docs/RedDogTutorialpdf">https://githubcom/katholt/RedDog/blob/master/docs/RedDogTutorialpdf</ext-link>. <year>2016</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30811390">30811390</a> <b>(2)</b> Script of Scripts: A pragmatic workflow system for daily computational research<ul><li>To evaluate SoS in terms of workflow features with leading bioinformatics workflow systems, we created an online table [<xref rid="pcbi.1006843.ref025" ref-type="bibr"><u>25</u></xref>] that compares SoS with <b>Snakemake</b> [<xref rid="pcbi.1006843.ref015" ref-type="bibr"><u>15</u></xref>], Nextflow [<xref rid="pcbi.1006843.ref014" ref-type="bibr"><u>14</u></xref>], Galaxy [<xref rid="pcbi.1006843.ref008" ref-type="bibr"><u>8</u></xref>], CWL [<xref rid="pcbi.1006843.ref009" ref-type="bibr"><u>9</u></xref>] and Bpipe [<xref rid="pcbi.1006843.ref022" ref-type="bibr"><u>22</u></xref>].<li>The table compare SoS with several popular bioinformatics workflow systems including Nextflow, <b>Snakemake</b>, Bpipe, CWL, and Galaxy, in three broad aspects: 1) basic features (syntax, file format, user interface, etc), 2) workflow features (workflow specification, dependency handling, execution and monitoring, etc), and built-in support for external tools and services (container support, HPC systems, distributed systems and cloud support).<li> [25](Peng B, Wang G. How does SoS compare with other workflow engines 2018. Available from: <ext-link ext-link-type="uri" xlink:href="https://vatlab.github.io/blog/post/comparison/">https://vatlab.github.io/blog/post/comparison/</ext-link>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [9](Amstutz P, Crusoe MR, Tijani&#x107; N, Chapman B, Chilton J, Heuer M, et al. Common Workflow Language, v1.0. Specification, Common Workflow Language working group. 2016.) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30811422">30811422</a> <b>(1)</b> Comparison of efficiency and specificity of CRISPR-associated (Cas) nucleases in plants: An expanded toolkit for precision genome engineering<ul><li> All scripts and a <b>snakemake</b> pipeline [<xref rid="pone.0211598.ref044" ref-type="bibr"><u>44</u></xref>] containing the whole workflow are available at <ext-link ext-link-type="uri" xlink:href="https://github.com/EI-CoreBioinformatics/CRISPRanto">https://github.com/EI-CoreBioinformatics/CRISPRanto</ext-link>.<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30799494">30799494</a> <b>(2)</b> SPIM workflow manager for HPC<ul><li> As a representative workflow we use a <italic><b>Snakemake</b></italic>-based SPIM data processing pipeline operating on large multi-view SPIM image datasets (<xref rid="btz140-B5" ref-type="bibr">Schmied <italic>et al.<li> The <italic><b>Snakemake</b></italic> workflow engine resolves dependencies between subsequent steps and executes in parallel any tasks appearing to be independent, such as processing of individual time points of a time-lapse acquisition.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30793188">30793188</a> <b>(1)</b> A conserved genetic interaction between Spt6 and Set2 regulates H3K36 methylation<ul><li>A custom ChIP-seq pipeline was generated using the <b>Snakemake</b> workflow manager (<xref rid="B61" ref-type="bibr"><u>61</u></xref>).<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30744683">30744683</a> <b>(1)</b> CellFishing.jl: an ultrafast and scalable cell search method for single-cell RNA sequencing<ul><li> To ensure reproducibility, all experiments were run using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>], a Python-based workflow management tool for bioinformatics.<li> [48](Koster J, Rahmann S. Snakemake&#x2013;a scalable bioinformatics workflow engine. Bioinformatics. 2012; 28(19):2520&#x2013;2. 10.1093/bioinformatics/bts480.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30738204">30738204</a> <b>(1)</b> Human papillomavirus 16 sub-lineage dispersal and cervical cancer risk worldwide: Whole viral genome sequences from 7116 HPV16-positive women<sup><xref ref-type="fn" rid="d31e2497">&#x2606;</xref></sup><ul><li> The pipeline was executed using <b>Snakemake</b> <xref rid="bib34" ref-type="bibr"><u>[34]</u></xref>, and settings and parameters can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/cgrlab/cgrHPV16" id="ir0010">https://github.com/cgrlab/cgrHPV16</ext-link>.<li> [[34]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30689725">30689725</a> <b>(1)</b> Haplotype assembly of autotetraploid potato using integer linear programing<ul><li> To insure reproducibility of the results, we wrote a <b>Snakemake</b> pipeline (<xref rid="btz060-B9" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and deployed it on IBM Cloud<sup>TM</sup> using private single-core instances.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30674894">30674894</a> <b>(5)</b> Comparative oncogenomics identifies combinations of driver genes and drug targets in <italic>BRCA1</italic>-mutated breast cancer<ul><li>The entire analysis of RNA-Seq and CNV-Seq data was implemented using <b>Snakemake</b> (version 4.3.1) and is freely available on GitHub [<ext-link ext-link-type="uri" xlink:href="https://github.com/jrderuiter/<b>snakemake</b>-rnaseq">https://github.com/jrderuiter/<b>snakemake</b>-rnaseq</ext-link> and <ext-link ext-link-type="uri" xlink:href="https://github.com/jrderuiter/<b>snakemake</b>-cnvseq">https://github.com/jrderuiter/<b>snakemake</b>-cnvseq</ext-link>].</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30618377">30618377</a> <b>(1)</b> Genetic effects on promoter usage are highly context-specific and contribute to complex traits<ul><li>The <b>Snakemake</b> (<xref rid="bib41" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) files used for gene and transcript expression quantification, QTL mapping and colocalisaton are available from the project&#x2019;s GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/kauralasoo/macrophage-tuQTLs">https://github.com/kauralasoo/macrophage-tuQTLs</ext-link>; copy archived at <ext-link ext-link-type="uri" xlink:href="https://github.com/elifesciences-publications/macrophage-tuQTLs">https://github.com/elifesciences-publications/macrophage-tuQTLs</ext-link>;&#xA0;<xref rid="bib5" ref-type="bibr"><u>Alasoo, 2018b</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Alasoo, 2018b](          <person-group person-group-type="author">            <name>              <surname>Alasoo</surname>              <given-names>K</given-names>            </name>          </person-group>          <year>2018b</year>          <data-title>macrophage-tuQTLs</data-title>          <source>Github</source>          <version designator="5fafd68">5fafd68</version>          <ext-link ext-link-type="uri" xlink:href="https://github.com/kauralasoo/macrophage-tuQTLs">https://github.com/kauralasoo/macrophage-tuQTLs</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31278683">31278683</a> <b>(19)</b> Scalable Workflows and Reproducible Data Analysis for Genomics<ul><li>In this chapter, we show how to create a bioinformatics pipeline using four workflow systems: CWL, GWL, <b>Snakemake</b>, and Nextflow.<li>Forth there is <b>Snakemake</b>, another widely used workflow manager system, written in Python and inspired by GNU Make.<li><b>Snakemake</b>.<li><b>Snakemake</b> [<xref rid="R42" ref-type="bibr"><u>42</u></xref>] is a workflow management system that takes inspiration from GNU Make [<xref rid="R40" ref-type="bibr"><u>40</u></xref>], a tool to coordinate the compilation of large programs consisting of interdependent source files (<ext-link xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" ext-link-type="uri">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>).<li><b>Snakemake</b> provides a DSL that allows the user to specify generator rules.<li>When a user requests a certain file to be generated, <b>Snakemake</b> matches the file name against concrete or wildcard rules, traverses the graph from the target file upward, and begins processing the steps for every rule for which no new output file is available.<li> In doing so, <b>Snakemake</b> only performs work that has not yet been done or for which the results are out of date, just like GNU Make.<li> <b>Snakemake</b> can be configured to distribute jobs to batch systems or to run jobs on the local system in parallel.<li><b>Snakemake</b> is written in Python and allows users to import Python modules and use them in the definition of rules, for example.<li><b>Snakemake</b> provides native support for the Conda package manager.<li> <b>Snakemake</b> will then invoke Conda to download the required packages as specified in the environment file.<li> Alternatively, <b>Snakemake</b> can interface with an installation of the Singularity container system [<xref rid="R26" ref-type="bibr"><u>26</u></xref>] and execute a rule within the context of a named application bundle, such as a Docker image.<li>To run the <b>Snakemake</b> workflow, you need to install <b>Snakemake</b> (example showed with Conda):.<li>To run the <b>Snakemake</b> workflow, you need to install <b>Snakemake</b> (example showed with Conda):</p>        <preformat preformat-type="computer code" position="float" xml:space="preserve">conda install -y -c bioconda <b>snakemake</b>=4.2.0.<li>And then to run the example from the repository tree:</p>        <preformat preformat-type="computer code" position="float" xml:space="preserve">cd <b>Snakemake</b>.<li>And then to run the example from the repository tree:</p>        <preformat preformat-type="computer code" position="float" xml:space="preserve">cd <b>Snakemake</b></preformat>        <preformat preformat-type="computer code" position="float" xml:space="preserve"><b>snakemake</b>.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](          <person-group person-group-type="author"><name><surname>Stallman</surname><given-names>R</given-names></name><name><surname>Mcgrath</surname><given-names>R</given-names></name></person-group>          <source>Gnu make: a program for directing recompilation</source>          <publisher-name>Free Software Foundation</publisher-name>          <publisher-loc>Boston, MA</publisher-loc>          <year>1989</year>        ) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30566439">30566439</a> <b>(1)</b> Expression reflects population structure<ul><li> We provide a package of tools for computing the projections and estimating gene significance, as well as a <b>Snakemake</b> file [<xref rid="pgen.1007841.ref041" ref-type="bibr"><u>41</u></xref>] that can be used to completely reproduce the analysis, from data acquisition to figure generation.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30558550">30558550</a> <b>(1)</b> DAWN: a resource for yielding insights into the diversity among wheat genomes<ul><li>The processing pipeline underpinning the DAWN data was implemented using <b>Snakemake</b> v5.1.4 [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>].<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30559361">30559361</a> <b>(1)</b> A rapid and robust method for single cell chromatin accessibility profiling<ul><li>All sequencing data were processed using a pipeline written in <b>snakemake</b><sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30559433">30559433</a> <b>(1)</b> Long-read sequence and assembly of segmental duplications<ul><li> We developed a <b>Snakemake</b> pipeline for this analysis ProcessCollapsedAssembly.py, which can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/mvollger/SDA">https://github.com/mvollger/SDA</ext-link>.<li> Specialized parameters are applied such that Canu can execute on such short contigs (<ext-link ext-link-type="uri" xlink:href="https://github.com/mvollger/SDA/blob/master/SDA.2.<b>snakemake</b>.py">https://github.com/mvollger/SDA/blob/master/SDA.2.<b>snakemake</b>.py</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30534606">30534606</a> <b>(2)</b> RELACS nuclei barcoding enables high-throughput ChIP-seq<ul><li> The workflows are freely available at <ext-link ext-link-type="uri" xlink:href="https://github.com/maxplanck-ie/<b>snakemake</b>_workflows">https://github.com/maxplanck-ie/<b>snakemake</b>_workflows</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30518860">30518860</a> <b>(1)</b> Complex Mammalian-like Hematopoietic System Found in a Colonial Chordate<ul><li>Determination of gene counts was performed using a <b>Snakemake</b><sup><xref rid="R46" ref-type="bibr"><u>46</u></xref></sup> pipeline.<li> [46](R46) <li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30514897">30514897</a> <b>(1)</b> Single-cell mutation identification via phylogenetic inference<ul><li>The simulation framework was implemented using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR27"><u>27</u></xref></sup> and can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/cbg-ethz/SCIPhI">https://github.com/cbg-ethz/SCIPhI</ext-link>.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30496484">30496484</a> <b>(1)</b> Recovery and analysis of transcriptome subsets from pooled single-cell RNA-seq libraries<ul><li> A reproducible software pipeline (including <b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) pipeline and R Markdown documents) is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/rnabioco/scrna-subsets">https://github.com/rnabioco/scrna-subsets</ext-link>.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30478041">30478041</a> <b>(1)</b> Integrative approach using <italic>Yersinia pestis</italic> genomes to revisit the historical landscape of plague during the Medieval Period<ul><li> S5</ext-link>) to analyze raw sequencing data from modern and ancient genomes using <b>snakemake</b> as the workflow manager system (<xref rid="r52" ref-type="bibr"><u>52</u></xref>).<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30455465">30455465</a> <b>(1)</b> H3K9 methyltransferases and demethylases control lung tumor-propagating cells and lung cancer progression<ul><li> RNA-seq analysis was performed using the VIPER <b>snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/cfce/viper/">https://bitbucket.org/cfce/viper/</ext-link>).</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30418481">30418481</a> <b>(2)</b> MAGpy: a reproducible pipeline for the downstream analysis of metagenome-assembled genomes (MAGs)<ul><li> MAGpy is implemented as a <b>Snakemake</b> (<xref rid="bty905-B13" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) pipeline, enabling reproducible analyses, extensibility, integration with high-performance-compute clusters and restart capabilities.<li>MAGpy makes use of <b>Snakemake</b> to define an analysis workflow for MAGs based on open source and freely available bioinformatics software.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30425841">30425841</a> <b>(3)</b> Modeling site-specific amino-acid preferences deepens phylogenetic estimates of viral sequence divergence<ul><li><monospace><b>snakemake</b></monospace> (<xref rid="vey033-B35" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) version 3.11.2 (available at <ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>) to run the pipelines.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30389934">30389934</a> <b>(1)</b> Deciphering highly similar multigene family transcripts from Iso-Seq data with IsoCon<ul><li> This repository also include a <b>snakemake</b><sup><xref ref-type="bibr" rid="CR63"><u>63</u></xref></sup> workflow that reproduces intermediate and final data for the ampliconic gene family analysis.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30373609">30373609</a> <b>(2)</b> SwissMTB: establishing comprehensive molecular cancer diagnostics in Swiss clinics<ul><li> All steps from mapping to prioritization are fully automatized using a <b>Snakemake</b>-based pipeline.<li> The pipeline for WES/WGS analysis from reads to unannotated variant calls is based on the framework described in [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>], employing the <b>Snakemake</b> workflow environment [<xref ref-type="bibr" rid="CR42"><u>42</u></xref>].<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28968639">28968639</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30362940">30362940</a> <b>(1)</b> Chromatin accessibility dynamics across <italic>C. elegans</italic> development and ageing<ul><li> Computationally intensive steps were managed and parallelized using <b>snakemake</b> (<xref rid="bib47" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30367288">30367288</a> <b>(1)</b> An automated framework for NMR chemical shift calculations of small organic molecules<ul><li> Future work includes wrapping individual steps of the ISiCLE NMR module into a formal workflow management system such as <b>Snakemake</b>, to include better fault tolerance, modularization, and improved data provenance.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30631441">30631441</a> <b>(2)</b> Increasing workflow development speed and reproducibility with Vectools<ul><li> Workflow-managers [e.g.,<ext-link ext-link-type="uri" xlink:href="https://pypi.org/project/<b>snakemake</b>/"><b>Snakemake</b></ext-link><sup><xref rid="ref-2" ref-type="bibr"><u>2</u></xref></sup>] have greatly simplified many aspects needed for reproducibility.<li> [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30367595">30367595</a> <b>(4)</b> Reproducible bioinformatics project: a community for reproducible bioinformatics analysis pipelines<ul><li> Instead, projects like (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.bitbucket.io">https://<b>snakemake</b>.bitbucket.io</ext-link>) or Nextflow (<ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io">https://www.nextflow.io</ext-link>) allow users to create reproducible and scalable data analyses specifying their own pipeline through a powerful metalanguage for workflow specification.<li> <b>snakemake</b> and nextflow, is specifically designed for the R community.<li> service, and (ii) to provide an automatic translation of our workflow specified in R into an equivalent workflow specified in <b>snakemake</b> format or in nextflow format.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30323331">30323331</a> <b>(2)</b> Precision Identification of Diverse Bloodstream Pathogens in the Gut Microbiome<ul><li>StrainSifter is a pipeline deployed as a <b>Snakemake</b><sup><xref rid="R34" ref-type="bibr"><u>34</u></xref></sup> workflow packaged with conda, available at GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/bhattlab/strainsifter">https://github.com/bhattlab/strainsifter</ext-link>).<li> <b>Snakemake</b> v5.1.4 and conda v4.5.9 were used in this manuscript.<li> [34](R34) <li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30277498">30277498</a> <b>(1)</b> PiGx: reproducible genomics analysis pipelines with GNU Guix<ul><li> For this purpose, we used <b>SnakeMake</b> [<xref rid="bib3" ref-type="bibr"><u>6</u></xref>], which provides target-driven execution infrastructure similar to GNU Make but with Python syntax, along with useful features such as parallel execution on HPC scheduling systems.<li> [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30356428">30356428</a> <b>(1)</b> Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification<ul><li> Such a command can be automated in a bash loop using bash variables, or one can use more advanced workflow management systems such as <b>Snakemake</b><sup><xref rid="ref-34" ref-type="bibr"><u>34</u></xref></sup> or Nextflow<sup><xref rid="ref-35" ref-type="bibr"><u>35</u></xref></sup>.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30254269">30254269</a> <b>(1)</b> Single cell molecular alterations reveal target cells and pathways of concussive brain injury<ul><li>Our analytical workflow was stored and freely accessible as a <b>SnakeMake</b> file [<ext-link ext-link-type="uri" xlink:href="https://github.com/darneson/DropSeq">https://github.com/darneson/DropSeq</ext-link>].</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30202060">30202060</a> <b>(1)</b> Terminal exon characterization with TECtool reveals an abundance of cell-specific isoforms<ul><li> The analysis flows are implemented in the <b>snakemake</b> framework <xref rid="R46" ref-type="bibr"><u>46</u></xref>, and the parameters for each type of analysis are specified in a corresponding configuration file.<li> [46](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <year>2012</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30192951">30192951</a> <b>(3)</b> Sequana coverage: detection and characterization of genomic variations using running median and mixture models<ul><li> We provide an implementation within the <monospace>Sequana</monospace> project [<xref rid="bib32" ref-type="bibr"><u>34</u></xref>], which is a Python library that also provides HTS pipelines based on the workflow management system called <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>35</u></xref>] (Makefile-like with a Python syntax).<li> A <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>35</u></xref>] pipeline was also recently implemented within <monospace>Sequana</monospace> [<xref rid="bib32" ref-type="bibr"><u>34</u></xref>] (named Coverage), allowing each chromosome to be analyzed independently.<li> A graphical interface using Sequanix [<xref rid="bib40" ref-type="bibr"><u>39</u></xref>] (a <b>Snakemake</b> graphical user interface) is also available, making the configuration of the parameters and execution of the analysis on a cluster accessible.<li> [34](<person-group person-group-type="author"><name name-style="western"><surname>Cokelaer</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Desvillechabrol</surname><given-names>D</given-names></name>, <name name-style="western"><surname>Legendre</surname><given-names>R</given-names></name>, <etal/></person-group><article-title>Sequana: a set of Snakemake NGS pipelines</article-title>. <source>Journal of Open Source Software</source>. <year>2017</year>;<volume>2</volume>, <fpage>16</fpage><pub-id pub-id-type="doi">10.21105/joss.00352</pub-id>.) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [34](<person-group person-group-type="author"><name name-style="western"><surname>Cokelaer</surname><given-names>T</given-names></name>, <name name-style="western"><surname>Desvillechabrol</surname><given-names>D</given-names></name>, <name name-style="western"><surname>Legendre</surname><given-names>R</given-names></name>, <etal/></person-group><article-title>Sequana: a set of Snakemake NGS pipelines</article-title>. <source>Journal of Open Source Software</source>. <year>2017</year>;<volume>2</volume>, <fpage>16</fpage><pub-id pub-id-type="doi">10.21105/joss.00352</pub-id>.) [39](<person-group person-group-type="author"><name name-style="western"><surname>Desvillechabrol</surname><given-names>D</given-names></name>, <name name-style="western"><surname>Legendre</surname><given-names>R</given-names></name>, <name name-style="western"><surname>Rioualen</surname><given-names>C</given-names></name><etal/></person-group><article-title>Sequanix: a dynamic graphical interface for Snakemake workflows</article-title>. <source>Bioinformatics</source>, <volume>34</volume>, <issue>11</issue>, <fpage>1934</fpage>&#x2013;<lpage>1936</lpage>. <year>2017</year>; <comment>10.1093/bioinformatics/bty034</comment>.) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30158930">30158930</a> <b>(1)</b> Silent Witness: Dual-Species Transcriptomics Reveals Epithelial Immunological Quiescence to Helminth Larval Encounter and Fostered Larval Development<ul><li> Here, we rely on <b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) as a current bioinformatics community standard to provide workflows at <uri xlink:type="simple" xlink:href="https://gitlab.com/mkuhring/project_asuum/">https://gitlab.com/mkuhring/project_asuum/</uri>.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30096152">30096152</a> <b>(1)</b> scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data<ul><li><b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30101309">30101309</a> <b>(2)</b> Container-based bioinformatics with Pachyderm<ul><li> One appealing option is to use workflow systems such as <b>Snakemake</b> (<xref rid="bty699-B18" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), Galaxy (<xref rid="bty699-B1" ref-type="bibr">Afgan <italic>et al.<li> Indeed, containers have been successfully employed in existing workflow systems such as <b>Snakemake</b> and Nextflow for life science applications.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30123121">30123121</a> <b>(8)</b> Reproducing Polychronization: A Guide to Maximizing the Reproducibility of Spiking Network Models<ul><li> <b>Snakemake</b>.<li><b>Snakemake</b> is a script based workflow management system which allows reproducible and scalable data analysis (K&#xF6;ster and Rahmann, <xref rid="B11" ref-type="bibr"><u>2012</u></xref>).<li> The complexity of our simulations, involving several different versions of neuron, synapse and network models as well as analysis scripts was massively eased by using <b>snakemake</b>.<li> <b>Snakemake</b> links the version of the program to the data it created, such that it can re-run specific sections of a workflow depending on what parts were changed.<li>Example visualization of the <b>snakemake</b> workflow for comparing the bitwise reproduction and the qualitative model.<li> This is used by <b>Snakemake</b> to generate the dependency tree in Figure <xref ref-type="fig" rid="F1"><u>1</u></xref>.<li> This workflow is repeated automatically 10 times for all experiments and 100 times for bitwise reproduction and qualitative reproduction using <b>Snakemake</b>.<li> Additionally, we modified the original to allow the seed for the random number generator to be set as a parameter, thus enabling multiple runs of the model to be carried out in our <b>snakemake</b> workflow (see Section 2.4).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30064353">30064353</a> <b>(4)</b> ATAC2GRN: optimized ATAC-seq and DNase1-seq pipelines for rapid and accurate genome regulatory network inference<ul><li> Additionally, we provide optimized <b>Snakemake</b> and bash pipelines with pre-configured algorithmic parameters capable of maximizing biological reproducibility and network accuracy for DNase1 hypersensitivity profiling and ATAC-seq [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>].<li>Optimized <b>Snakemake</b> pipelines for ATAC and DNase1 footprinting are available for download at the Chiorini Lab github page (<ext-link ext-link-type="uri" xlink:href="http://www.github.com/ChioriniLab">www.github.com/ChioriniLab</ext-link>).<li> Our analyses identified the optimal argument combinations for ATAC-seq and DNase1 analysis and these pre-optimized pipelines were translated into <b>Snakemake</b> pipelines for routine and reproducible deployment to supercomputing clusters.<li> All code was translated to the <b>Snakemake</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] pipeline management platform to streamline scientific supercomputing deployment and maximize data reproducibility.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25177346">25177346</a>) [27](K&#xF6;ster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine. Bioinformatics. 2012;28 10.1093/bioinformatics/bts480.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30063227">30063227</a> <b>(3)</b> BARM and BalticMicrobeDB, a reference metagenome and interface to meta-omic data for the Baltic Sea<ul><li>Code used to preprocess reads, assemble contigs and annotate genes is publicly available at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" ext-link-type="uri" xlink:href="https://github.com/EnvGen/BLUEPRINT_pipeline">https://github.com/EnvGen/BLUEPRINT_pipeline</ext-link>, containing the pipeline definition of the workflows used, <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" ext-link-type="uri" xlink:href="https://github.com/EnvGen/<b>snakemake</b>-workflows">https://github.com/EnvGen/<b>snakemake</b>-workflows</ext-link>, where the <b>snakemake</b> rules are specified in order to build the command used for each step, and the branch BARM_publication of <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" ext-link-type="uri" xlink:href="https://github.com/EnvGen/toolbox">https://github.com/EnvGen/toolbox</ext-link>, for custom scripts.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30044795">30044795</a> <b>(3)</b> Automated size selection for short cell-free DNA fragments enriches for circulating tumor DNA and improves error correction during next generation sequencing<ul><li>A <b>Snakemake</b> workflow was constructed to convert fastq sequencing datasets with unique molecular identifiers to processed alignments.<li> See the <b>snakemake</b> file (<xref ref-type="supplementary-material" rid="pone.0197333.s021"><u>S1 File</u></xref>) and bash script file (<xref ref-type="supplementary-material" rid="pone.0197333.s022"><u>S2 File</u></xref>) for exact program versions and parameter settings.<li><b>Snakemake</b> rule set for executing the consensus alignment workflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30052957">30052957</a> <b>(4)</b> Leveraging multiple transcriptome assembly methods for improved gene structure annotation<ul><li>We also developed a <b>Snakemake</b>-based pipeline, Daijin, in order to drive Mikado, including the calls to external programs to calculate ORFs and protein homology.<li> Details of the process can be found in the two <b>snakemake</b>s &#x201C;compare.snakefile&#x201D; and &#x201C;compare_simulations.snakefile&#x201D; present in the &#x201C;<b>Snakemake</b>&#x201D; directory of the script repository.<li>Other requirements: <b>SnakeMake</b>, BioPython, NumPY, SciPY, Scikit-learn, BLAST+ or DIAMOND, Prodigal or TransDecoder, Portcullis.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29982548">29982548</a> <b>(1)</b> Effect of UV irradiation on&#xA0;<italic>Sulfolobus acidocaldarius</italic> and involvement of the general transcription factor TFB3 in the early UV response<ul><li> All steps were automated via <b>Snakemake</b> (Version: 3.13.3) (<xref rid="B45" ref-type="bibr"><u>45</u></xref>).<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29931311">29931311</a> <b>(1)</b> Patterns of Conservation and Diversification in the Fungal Polarization Network<ul><li>ggMatch is implemented in python using <b>SnakeMake</b> (<xref rid="evy121-B50" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29971009">29971009</a> <b>(4)</b> The CoLoMoTo Interactive Notebook: Accessible and Reproducible Computational Analyses for Qualitative Biological Networks<ul><li> <xref ref-type="supplementary-material" rid="SM1"><u>Supplementary File</u></xref> &#x201C;<italic><b>SnakeMake</b></italic>&#x201D; provides an example of S<sc>nake</sc>M<sc>ake</sc> workflow relying on GIN<sc>sim</sc> and N<sc>u</sc>SMV.<li> As demonstrated in the <xref ref-type="supplementary-material" rid="SM1"><u>Supplementary File</u></xref> &#x201C;<italic><b>SnakeMake</b></italic>&#x201D;, the CoLoMoTo Docker image can also be used in standard workflow systems, such as <b>SnakeMake</b>, to lighten the burden of installing the different software tools and make them accessible on different operating systems.<li> AN and LP edited the notebook tutorials, while CH edited the <b>SnakeMake</b> workflow example.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29895634">29895634</a> <b>(1)</b> Reassessing the Role of Type II Toxin-Antitoxin Systems in Formation of <named-content content-type="genus-species">Escherichia coli</named-content> Type II Persister Cells<ul><li> All genomic analyses were performed using <b>Snakemake</b> (<xref rid="B77" ref-type="bibr"><u>77</u></xref>) as workflow manager together with software installations from Bioconda (<xref rid="B78" ref-type="bibr"><u>78</u></xref>).<li> [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [78](<person-group person-group-type="author"><name name-style="western"><surname>Gr&#xFC;ning</surname><given-names>B</given-names></name>, <name name-style="western"><surname>Dale</surname><given-names>R</given-names></name>, <name name-style="western"><surname>Sj&#xF6;din</surname><given-names>A</given-names></name>, <name name-style="western"><surname>Rowe</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Chapman</surname><given-names>BA</given-names></name>, <name name-style="western"><surname>Tomkins-Tinch</surname><given-names>CH</given-names></name>, <name name-style="western"><surname>Valieris</surname><given-names>R</given-names></name>, <collab>The Bioconda Team</collab>, <name name-style="western"><surname>K&#xF6;ster</surname><given-names>J</given-names></name></person-group><year>2017</year><article-title>Bioconda: a sustainable and comprehensive software distribution for the life sciences</article-title>. <source>bioRxiv</source> doi:<pub-id pub-id-type="doi">10.1101/207092</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29922601">29922601</a> <b>(1)</b> <italic>Galleria mellonella</italic> Reveals Niche Differences Between Highly Pathogenic and Closely Related Strains of <italic>Francisella</italic> spp.<ul><li> All analysis steps except tree generation were performed using <b>snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B25" ref-type="bibr"><u>2012</u></xref>) and Bioconda.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29873797">29873797</a> <b>(2)</b> The translational landscape of <italic>Arabidopsis</italic> mitochondria<ul><li> All the pipeline was automatized through the use of <b>Snakemake</b> workflow engine version 3.5.2 (<xref rid="B31" ref-type="bibr"><u>31</u></xref>).<li> <b>Snakemake</b> workflow engine was also used to automatize the steps of this analysis.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29789628">29789628</a> <b>(1)</b> Targetable vulnerabilities in T- and NK-cell lymphomas identified through preclinical models<ul><li>Analysis of RNASeq data was implemented as a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR64"><u>64</u></xref></sup> workflow.<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29745835">29745835</a> <b>(1)</b> GraphTeams: a method for discovering spatial gene clusters in Hi-C sequencing data<ul><li>We implemented Algorithm 3 in the Python programming language and provide an entirely automated <b>Snakemake</b> [<xref ref-type="bibr" rid="CR23"><u>23</u></xref>] workflow for the identification of spatial gene clusters.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a> <b>(17)</b> VIPER: Visualization Pipeline for RNA-seq, a <b>Snakemake</b> workflow for efficient and complete RNA-seq analysis<ul><li>The system presented here, VIPER (Visualization Pipeline for RNA sequencing analysis), uses a modern computational workflow management system, <b>Snakemake</b> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>], to combine many of the most useful tools currently employed in RNA-seq analysis into a single, fast, easy to use pipeline, that includes alignment steps, quality control, differential gene expression and pathway analyses.<li> (1) Highly modular pipeline exploiting the <b>Snakemake</b> framework that allows for rapid integration of new approaches or replacement of existing algorithms.<li> Upon execution, <b>Snakemake</b> <italic>infers</italic> the combination of rules necessary to achieve a &#x201C;target&#x201D; or specific output, in our case the final report.<li> The STAR aligner is known for its superior speed that integrates very well with <b>Snakemake</b>&#x2019;s underlying ability to allocate resources and execute multithreaded processes.<li> Secondly, using <b>Snakemake</b> offers distinct advantages in both efficiency and customizability.<li><b>Snakemake</b> as a framework.<li>VIPER&#x2019;s <b>Snakemake</b> backbone provides several advantages that set it apart from other sequencing pipelines.<li> As of <b>Snakemake</b> 3.7 each rule is evaluated in its own environment making it even easier to mix tools (e.g.<li><b>Snakemake</b> was built with the concept of parallelization in mind enabling VIPER to make use of its ability to spawn jobs in parallel to maximize its speed and make full use of the provided processing power.<li> Additionally, <b>Snakemake</b> has the capability to scale from single-core workstations over multi-core servers to compute clusters of different architectures, without the need to modify the workflow.<li><b>Snakemake</b>&#x2019;s &#x201C;bottom up&#x201D; method of determination of job execution allows for a number of advantages including crash recovery and specification of subsetted analyses.<li> A <b>Snakemake</b> workflow is composed of individualized rules, each of which takes a specified input and generates a designated output.<li> <b>Snakemake</b> determines the execution of events by checking timestamps, and as long as a rule was properly executed and the input file timestamps have not changed, then it will not attempt to regenerate the output.<li> <b>Snakemake</b> will determine via the file timestamps that the aligned data was not changed (just the metadata describing samples) and will proceed to the analysis downstream of alignment (starting with differential expression) thus skipping the computationally intensive upstream rules.<li>Customization of VIPER requires a baseline understanding of the underlying framework of <b>Snakemake</b>, but will allow users to continually update and modify their instance of VIPER.<li>The RNA-seq pipeline HppRNA employs the same <b>Snakemake</b> workflow management platform as VIPER allowing it to share the benefits of this workflow engine.<li> This should be readily achievable with the capabilities of the <b>snakemake</b> framework [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] and make highly scalable analysis more widely available.<li> [15](          <person-group person-group-type="author">            <name>              <surname>Koster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake--a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <issue>19</issue>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) [15](          <person-group person-group-type="author">            <name>              <surname>Koster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake--a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <issue>19</issue>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29692763">29692763</a> <b>(1)</b> <italic>Nitrosospira</italic> sp. Govern Nitrous Oxide Emissions in a Tropical Soil Amended With Residues of Bioenergy Crop<ul><li> Further, the sequences were processed using the UCLUST pipeline implemented in a <b>Snakemake</b> workflow which is available upon request (<xref rid="B17" ref-type="bibr"><u>Edgar, 2010</u></xref>).<li> [Edgar, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20709691">20709691</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29592812">29592812</a> <b>(1)</b> Discovery of physiological and cancer-related regulators of 3&#x2032; UTR processing with KAPAC<ul><li>All scripts, intermediate steps, and analysis of the TCGA data sets were executed as workflows created with <b>snakemake</b> version 3.13.0 [<xref ref-type="bibr" rid="CR58"><u>58</u></xref>].<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29534677">29534677</a> <b>(21)</b> Watchdog &#x2013; a workflow management system for the distributed analysis of large-scale experimental data<ul><li> Other commonly used WMSs are <italic>KNIME</italic> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], an open-source data analysis platform which allows programmers to extend its basic functionality by adding new Java programs, and <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], a python-based WMS.<li> <italic><b>Snakemake</b></italic> allows definition of tasks based on rules and automatically infers dependencies between tasks by matching filenames.<li>Although explicit dependency definition adds a small manual overhead compared to automatic identification based on in- and output filenames as in <italic><b>Snakemake</b></italic>, it also provides more flexibility as dependencies can be defined that are not obvious from filenames.<li> As representative WMSs <italic>Galaxy</italic> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>], <italic>KNIME</italic> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], and <italic>Nextflow</italic> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>] were chosen.<li> Furthermore, an analysis of the computational overhead of <italic>Watchdog</italic> and <italic><b>Snakemake</b></italic> showed that the computational overhead of using either WMS (and likely any other) is negligible compared to the actual runtime of the executed tasks (see Additional file&#xA0;<xref rid="MOESM4" ref-type="media"><u>4</u></xref>).<li> <sup>7</sup>other supported executors: <italic>Watchdog</italic>: new executors can be added with the plugin system, <italic>Galaxy</italic>: PBS/Torque, Open Grid Engine, Univa Grid Engine, Platform LSF, HTCondor, Slurm, Galaxy Pulsar, <italic><b>Snakemake</b></italic>: can also use cluster engines with access to a common file system and a submit command that accepts shell scripts as first argument, <italic>Nextflow</italic>: SGE, LSF, Slurm, PBS/Torque, NQSII, HTCondor, Ignite; <sup>8</sup>non-free extensions for SGE or dedicated server support are available; <sup>9</sup>custom executors for cloud computing services can be created using the plugin system; <sup>10</sup><italic>Watchdog</italic>: HTTP/S, FTP/S and SFTP by default, can be extended to any remote file system with an implementation of the FileProvider interface from the Commons Virtual File System project, <italic>Galaxy</italic>: Object Store plugins for S3, Azure, iRODS, <italic><b>Snakemake</b></italic>: S3, GS, SFTP, HTTP, FTP, Dropbox, XRootD, NCBI, WebDAV, GFAL, GridFTP.<li><b>Snakemake</b>.<li>A workflow processed by <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] is defined as a set of rules.<li> These rules must be specified in <italic><b>Snakemake</b></italic>&#x2019;s own language in a text file named <italic>Snakefile</italic>.<li> At the beginning of workflow execution, <italic><b>Snakemake</b></italic> automatically infers the rule execution order and dependencies based on the names of the input and output files for each rule.<li>In <italic><b>Snakemake</b></italic>, there is no clear separation between the tool library and workflow definition as the command used to generate output files is defined in the rule definition itself.<li> Starting with version 3.5.5, <italic><b>Snakemake</b></italic> introduced re-usable wrapper scripts e.g.<li> Thus, <italic><b>Snakemake</b></italic> now allows both encapsulation of integrated tools as well as quickly adding commands directly into the workflow.<li>By default, no new jobs are scheduled in <italic><b>Snakemake</b></italic> as soon as one error is detected based on the exit code of the executed command.<li> Finally, <italic><b>Snakemake</b></italic> is the only one of the compared WMSs that does not provide return variables that can be used as parameters in later steps.<li>In summary, <italic><b>Snakemake</b></italic> is a much improved version of <italic>GNU Make</italic>.<li> Programmers will be able to create and execute own workflows using <italic><b>Snakemake</b></italic> once they learned the syntax and semantic of the <italic><b>Snakemake</b></italic> workflow definition language.<li> However, as <italic><b>Snakemake</b></italic> does not offer a GUI or editor for workflow design, most experimentalists without programming skills will not be able to create their own workflows.<li> Contains an analysis of the computational overhead of <italic>Watchdog</italic> and <italic><b>Snakemake</b></italic> for executing a workflow with a variable number of samples.<li> [9](          <person-group person-group-type="author">            <name>              <surname>Berthold</surname>              <given-names>MR</given-names>            </name>            <name>              <surname>Cebron</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Dill</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Gabriel</surname>              <given-names>TR</given-names>            </name>            <name>              <surname>K&#xF6;tter</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Meinl</surname>              <given-names>T</given-names>            </name>            <etal/>          </person-group>          <article-title>KNIME: The Konstanz Information Miner</article-title>          <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>          <year>2007</year>          <publisher-loc>Heidelberg-Berlin</publisher-loc>          <publisher-name>Springer</publisher-name>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](Taylor J, Schenck I, Blankenberg D, Nekrutenko A. Using galaxy to perform large-scale interactive data analyses. Curr Protoc Bioinforma. 2007. Chapter 10:Unit 10.5.) [9](          <person-group person-group-type="author">            <name>              <surname>Berthold</surname>              <given-names>MR</given-names>            </name>            <name>              <surname>Cebron</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Dill</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Gabriel</surname>              <given-names>TR</given-names>            </name>            <name>              <surname>K&#xF6;tter</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Meinl</surname>              <given-names>T</given-names>            </name>            <etal/>          </person-group>          <article-title>KNIME: The Konstanz Information Miner</article-title>          <source>Studies in Classification, Data Analysis, and Knowledge Organization (GfKL 2007)</source>          <year>2007</year>          <publisher-loc>Heidelberg-Berlin</publisher-loc>          <publisher-name>Springer</publisher-name>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29531245">29531245</a> <b>(1)</b> Using massively parallel shotgun sequencing of maternal plasmatic cell-free DNA for cytomegalovirus DNA detection during pregnancy: a proof of concept study<ul><li>All data processing and analysis from reads obtained from sequencer to the final report with the viral load estimation was coded using <b>snakemake</b><sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>, a workflow engine which calls R (<ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/">https://www.r-project.org/</ext-link>) and python (<ext-link ext-link-type="uri" xlink:href="https://www.python.org/">https://www.python.org/</ext-link>) scripts and shell (bash) command lines.<li> [41](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>          <source>Bioinforma. Oxf. Engl.</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29563897">29563897</a> <b>(1)</b> Drought Legacy Effects on the Composition of Soil Fungal and Prokaryote Communities<ul><li> The pipeline was made with <b>Snakemake</b> (<xref rid="B50" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) as available at DOI: <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.597131">https://doi.org/10.5281/zenodo.597131</ext-link> (<xref rid="B15" ref-type="bibr"><u>de Hollander, 2016</u></xref>).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [de Hollander, 2016](<person-group person-group-type="author"><name><surname>de Hollander</surname><given-names>M.</given-names></name></person-group> (<year>2016</year>). <source><italic>GitLab Community Edition: Open Source Software to Collaborate on Code [Online]</italic>.</source><comment>Available at: <ext-link ext-link-type="uri" xlink:href="https://gitlab.bioinf.nioo.knaw.nl/ampliconmetagenomics/illumina_paired_end">https://gitlab.bioinf.nioo.knaw.nl/amplicon metagenomics/illumina_paired_end</ext-link></comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29482639">29482639</a> <b>(3)</b> Improving saliva shotgun metagenomics by chemical host DNA depletion<ul><li>Demultiplexed sequences were processed using an in-house modular workflow employing <b>Snakemake</b> [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>] (<ext-link ext-link-type="uri" xlink:href="https://github.com/tanaes/<b>snakemake</b>_assemble">https://github.com/tanaes/<b>snakemake</b>_assemble</ext-link>, commit 1c393f4).<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29483941">29483941</a> <b>(1)</b> Genome-resolved metagenomics of sugarcane vinasse bacteria<ul><li> Processing was performed in a <b>Snakemake</b> v3.7.1 workflow or with bash or Perl scripts (available upon request).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29422027">29422027</a> <b>(1)</b> OVAS: an open-source variant analysis suite with inheritance modelling<ul><li>Initial development considered the use of pre-existing implicit convention frameworks such as <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR12"><u>12</u></xref>], but a predilection towards coding-flexibility and processing efficiency (especially with respect to extensive use of standard system input/ouput streams) meant that a more unix-driven pipeline framework was required.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29441358">29441358</a> <b>(3)</b> Growing old, yet staying young: The role of telomeres in bats&#x2019; exceptional longevity<ul><li> CodeML selective pressure variation tests were automated using a custom-built workflow, OH-SNAP (Optimised High-throughput <b>SNakemake</b> Automation of PAML; available at <ext-link ext-link-type="uri" xlink:href="https://github.com/batlabucd/OHSNAP">https://github.com/batlabucd/OHSNAP</ext-link>) (see Materials and Methods and fig.<li>OH-SNAP: Automation of CodeML using <b>Snakemake</b>.<li> The workflow was designed using the <b>Snakemake</b> workflow management system (<xref rid="R64" ref-type="bibr"><italic>64</italic></xref>) and was executed on a high-performance compute cluster.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29568489">29568489</a> <b>(1)</b> Ten steps to get started in Genome Assembly and Annotation<ul><li> Authors should be encouraged to develop their pipelines in commonly used workflow managers (Galaxy<sup><xref rid="ref-69" ref-type="bibr"><u>69</u></xref></sup>, Nextflow<sup><xref rid="ref-70" ref-type="bibr"><u>70</u></xref></sup>, <b>Snakemake</b><sup><xref rid="ref-71" ref-type="bibr"><u>71</u></xref></sup>).<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29361152">29361152</a> <b>(24)</b> Sequanix: a dynamic graphical interface for <b>Snakemake</b> workflows<ul><li>Amongst the recent CLI-based workflow managers, <b>Snakemake</b> (<xref rid="bty034-B4" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) has been adopted by a large community of developers.<li> This is especially pronounced in the field of NGS (see <ext-link ext-link-type="uri" xlink:href="http://<b>snakemake</b>.readthedocs.io">http://<b>snakemake</b>.readthedocs.io</ext-link>), where there is an increasing demand for production-ready pipelines to handle massive amounts of data from different technologies.<li> Although <b>Snakemake</b> provides a GUI interface, it is server-oriented, which is a limitation on some distributed clusters.<li> Besides, scientists willing to change the behaviour of the pipelines needs to edit the configuration file or masterize the numerous <b>Snakemake</b> arguments.<li> Although Sequanix was designed to expose Sequana pipelines (<xref rid="bty034-B1" ref-type="bibr"><u>Cokelaer, 2017</u></xref>), it can also load any <b>Snakemake</b> pipelines as demonstrated in the <xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Material</u></xref> with third-party pipelines available in the SnakeChunks library (<xref rid="bty034-B7" ref-type="bibr"><u>Rioualen, 2017</u></xref>).<li><b>Snakemake</b> is a text-based workflow that uses Python and a definition language to define rules and workflow properties.<li> A <b>Snakemake</b> pipeline is defined within a file called <italic>Snakefile</italic> (examples in <xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Material</u></xref>).<li>Sequana project provides a set of <b>Snakemake</b> pipelines dedicated to NGS analysis (e.g.<li> Hereafter, we distinguish the <b>Snakemake</b> pipelines provided in Sequana from other <b>Snakemake</b> pipelines.<li> The user interface consists of a <italic>main</italic> dialog (see <xref ref-type="fig" rid="bty034-F1"><u>Fig.&#xA0;1</u></xref>), a <italic><b>Snakemake</b></italic> dialog and a <italic>Preferences</italic> dialog (See <xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Material</u></xref>).<li> The <b>Snakemake</b> dialog is used to configure the <b>Snakemake</b> framework behavior (e.g.<li> Then, one can select a <b>Snakemake</b> pipeline, or specify the working directory where analysis is performed and results stored (<bold>A</bold>).<li> Finally, a working directory must be set where the project (<b>Snakemake</b> pipeline and configuration file edited by the user) is saved.<li> The only restriction is that the <b>Snakemake</b> pipeline must be executable in the Sequanix&#x2019;s environment.<li>We designed Sequanix to build upon the <b>Snakemake</b> framework and add important functionalities.<li>The <b>Snakemake</b> framework scales without modification, from single and multi-core workstations to cluster engines.<li> This ability is reflected in the <italic>Sequanix</italic> interface (<xref ref-type="fig" rid="bty034-F1"><u>Fig.&#xA0;1B</u></xref>) and in the <b>Snakemake</b> dialog where one can switch between local and cluster mode, set the number of CPUs, or provide specific job scheduler arguments (e.g.<li> Stopping the process (<italic>Stop</italic> button) behaves as a normal <b>Snakemake</b> interruption allowing re-entry.<li><italic>Sequanix</italic> provides a GUI for <b>Snakemake</b> workflows.<li> Its simple interface makes it easy for non-specialists to implement existing production-ready pipelines that have been created by <b>Snakemake</b> developers.<li> Although Sequanix was primarily developed to expose Sequana pipelines, it should also benefit to the community of <b>Snakemake</b> developers willing to provide a graphical interface to their users.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Cokelaer, 2017](<person-group person-group-type="author"><name name-style="western"><surname>Cokelaer</surname><given-names>T.</given-names></name></person-group><etal>et al</etal> (<year>2017</year>) <article-title>Sequana: a set of snakemake NGS pipelines</article-title>. <source>J. Open Source Softw</source>., <volume>2</volume>, <fpage>352</fpage>.) [Rioualen, 2017](<person-group person-group-type="author"><name name-style="western"><surname>Rioualen</surname><given-names>C.</given-names></name></person-group><etal>et al</etal> (<year>2017</year>) SnakeChunks: modular blocks to build Snakemake workflows for reproducible NGS analyses. bioRxiv doi: <pub-id pub-id-type="doi">10.1101/165191</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29352322">29352322</a> <b>(3)</b> qPortal: A platform for data-driven biomedical research<ul><li> For example, some of the available analysis pipelines include <b>Snakemake</b> workflows [<xref rid="pone.0191603.ref037" ref-type="bibr"><u>37</u></xref>].<li> While the current production instances interface with gUSE/WS-PGRADE, the extension of the workflow system with interfaces to <b>SnakeMake</b> and NextFlow is possible.<li> Therefore, users will be able to create <b>SnakeMake</b> or NextFlow workflows and use them through qPortal provided that the corresponding tools are available on the cluster instance.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29150517">29150517</a> <b>(1)</b> Multi-omics Reveals the Lifestyle of the Acidophilic, Mineral-Oxidizing Model Species <named-content content-type="genus-species">Leptospirillum ferriphilum</named-content><sup>T</sup><ul><li>A custom pipeline was written in <b>snakemake</b> (<xref rid="B76" ref-type="bibr"><u>76</u></xref>) for processing and analysis of the transcriptome sequencing (RNA-seq) data (the source code is available at <ext-link ext-link-type="uri" xlink:href="https://git-r3lab.uni.lu/malte.herold/LF_omics_analysis">https://git-r3lab.uni.lu/malte.herold/LF_omics_analysis</ext-link>).<li> [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29372116">29372116</a> <b>(2)</b> FUSTr: a tool to find gene families under selection in transcriptomes<ul><li>FUSTr is written in Python with all data filtration, preparation steps, and command line arguments/parameters for external programs contained in the workflow engine <b>Snakemake</b> (<xref rid="ref-9" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> <b>Snakemake</b> allows FUSTr to operate on high performance computational facilities, while also maintaining ease of reproducibility.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29330469">29330469</a> <b>(1)</b> <italic>Mycobacterium tuberculosis</italic> Exploits a Molecular <italic>Off Switch</italic> of the Immune System for Intracellular Survival<ul><li> This procedure, and all subsequent steps, was automated with a computational pipeline (written in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR61"><u>61</u></xref></sup>, available on request) for reproducibility and consistency (Fig.&#xA0;<xref rid="Fig6" ref-type="fig"><u>6</u></xref>).<li> [61](K&#xF6;ster, J. &amp; Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic><bold>28</bold> (2012).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29295995">29295995</a> <b>(1)</b> Perturbation-response genes reveal signaling footprints in cancer gene expression<ul><li> The preprocessing pipeline was written in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29284518">29284518</a> <b>(1)</b> DE-kupl: exhaustive capture of biological variation in RNA-seq data through <italic>k</italic>-mer decomposition<ul><li>The DE-kupl pipeline (Additional file&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref>: Figure S14) is implemented using the <b>Snakemake</b> [<xref ref-type="bibr" rid="CR42"><u>42</u></xref>] workflow manager (v3.10.1).<li> [42](K&#xF6;ster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine. Bioinformatics. 2012; 28(19):2520&#x2013;2. <ext-link ext-link-type="uri" xlink:href="https://doi.org/10.1093/bioinformatics/bts480">https://doi.org/10.1093/bioinformatics/bts480</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29280973">29280973</a> <b>(1)</b> Transcriptome-Wide Identification and Characterization of Potato Circular RNAs in Response to <italic>Pectobacterium carotovorum</italic> Subspecies <italic>brasiliense</italic> Infection<ul><li> All analysis pipelines were built by <b>snakemake</b> (v4.0.0) [<xref rid="B44-ijms-19-00071" ref-type="bibr"><u>44</u></xref>].<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29272324">29272324</a> <b>(1)</b> Bioinformatics for precision oncology<ul><li> Multiple workflow managers are available such as <b>Snakemake</b> [<xref rid="bbx143-B27" ref-type="bibr"><u>27</u></xref>], Nextflow [<xref rid="bbx143-B28" ref-type="bibr"><u>28</u></xref>], Toil [<xref rid="bbx143-B29" ref-type="bibr"><u>29</u></xref>], Bpipe [<xref rid="bbx143-B30" ref-type="bibr"><u>30</u></xref>] and to some extent also the Galaxy framework [<xref rid="bbx143-B31" ref-type="bibr"><u>31</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29375779">29375779</a> <b>(4)</b> Genome&#x2010;wide single&#x2010;nucleotide polymorphism data reveal cryptic species within cryptic freshwater snail species&#x2014;The case of the <italic>Ancylus fluviatilis</italic> species complex<ul><li> After exporting data with export_sql.pl from Stacks specifying a minimum stack depth of 8, further analyses were conducted using the workflow management tool <b>Snakemake</b> (K&#xF6;ster &amp; Rahmann, <xref rid="ece33706-bib-0033" ref-type="ref"><u>2012</u></xref>), in which different self&#x2010;written scripts for data reformatting, filtering, and population genetic analyses were combined.<li> Through the possibility to use wildcards in the <b>Snakemake</b> workflow, datasets resulting from the different Stacks batches and different filter settings could be easily analyzed simultaneously.<li> The <b>Snakemake</b> workflow included the script stacks2fasta.pl (Macher et&#xA0;al., <xref rid="ece33706-bib-0036" ref-type="ref"><u>2015</u></xref>).<li> Further used scripts and the <b>Snakemake</b> workflow are available in BitBucket (<ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/GeneStream_PhD/ddrad_workflow_af">https://bitbucket.org/GeneStream_PhD/ddrad_workflow_af</ext-link>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25691988">25691988</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29186349">29186349</a> <b>(1)</b> GDSCTools for mining pharmacogenomic interactions in cancer<ul><li> We also provide solutions based on the <b>Snakemake</b> framework (<xref rid="btx744-B5" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) to parallelize the analysis on distributed cluster farm architectures such as LSF or SLURM (<xref ref-type="supplementary-material" rid="sup1"><u>Supplementary Material</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29176637">29176637</a> <b>(1)</b> Coordinated response of the <italic>Desulfovibrio desulfuricans</italic> 27774 transcriptome to nitrate, nitrite and nitric oxide<ul><li>The automated workflow for the RNA-seq analysis was implemented in the <b>snakemake</b> programming language, and is available on the supporting website.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29145803">29145803</a> <b>(1)</b> Unraveling the evolution and coevolution of small regulatory RNAs and coding genes in <italic>Listeria</italic><ul><li>The strategy we developed is implemented in a <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR60"><u>60</u></xref>] that consists in four main steps (see Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>).<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29147973">29147973</a> <b>(1)</b> Co-Variation of Bacterial and Fungal Communities in Different Sorghum Cultivars and Growth Stages is Soil Dependent<ul><li> All procedures were implemented in a <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR33"><u>33</u></xref>].<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29048555">29048555</a> <b>(1)</b> An architecture for genomics analysis in a clinical setting using Galaxy and Docker<ul><li> In addition, workflow management systems (such as Taverna [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], Galaxy [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], <b>SnakeMake</b> [<xref rid="bib13" ref-type="bibr"><u>13</u></xref>], NextFlow [<xref rid="bib14" ref-type="bibr"><u>14</u></xref>]) also ensure the traceability and reproducibility of the analytical process.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](<person-group person-group-type="author"><name name-style="western"><surname>Tommaso</surname><given-names>PD</given-names></name>, <name name-style="western"><surname>Chatzou</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Prieto</surname><given-names>P</given-names></name><etal/></person-group><article-title>Nextflow: a tool for deploying reproducible computational pipelines</article-title>. <source>F1000Res</source><year>2015</year><comment><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.7490/f1000research.1110183.1">https://doi.org/10.7490/f1000research.1110183.1</ext-link></comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29037173">29037173</a> <b>(16)</b> TaxMapper: an analysis tool, reference database and workflow for metatranscriptome analysis of eukaryotic microorganisms<ul><li> It is part of an automated rule-based <b>Snakemake</b> workflow developed to perform quality assessment and both functional and taxonomic annotation, as well as (multivariate) statistical analysis including environmental data.<li>A comprehensive workflow for metatranscriptome analysis was developed and made available in an executable <b>Snakemake</b>-based workflow.<li> <b>Snakemake</b> is a workflow description language and execution environment developed by K&#xF6;ster et al.<li> For the described workflow <b>Snakemake</b> version 3.9.1 was used.<li><b>Snakemake</b> workflow.<li> Rules of the <b>Snakemake</b> workflow processing the FASTQ input files to the final output <italic>all</italic>.<li>The steps of the bioinformatic workflow are specified in the workflow management system <b>Snakemake</b>.<li> <b>Snakemake</b> rules describe how to create output files from input files by executing commands on the input files.<li> The commands can also be run on single files in the terminal, Python or R, but for automation, parallelization and reproducibility of the workflow, <b>Snakemake</b> is used.<li> We briefly explain the <b>Snakemake</b> syntax here on a short example <b>Snakemake</b> file:.<li> <b>Snakemake</b> determines the rule dependencies by matching file names and automatically fills the wildcard dataset with the correct names: dataset1 and dataset2, that are expected as the input of rule all.<li> The compressed FASTQ files are used as input and the <b>snakemake</b> rule runs FastQC as a shell command on the input.<li> We have to get the input folder from the input files and provide an output file from TaxMapper as output for <b>snakemake</b>.<li> To let <b>Snakemake</b> handle parallelization and provide user-defined parameters, the workflow can also be run in five successive steps: search, map, filter, count and plot (see Fig.&#xA0;<xref rid="Fig5" ref-type="fig"><u>5</u></xref> TaxMapper box).<li>TaxMapper can be run straightforwardly on a folder of sequencing data or as part of the <b>Snakemake</b> workflow.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29053696">29053696</a> <b>(3)</b> 3' End Sequencing Library Preparation with A-seq2<ul><li> The analysis includes four main steps: (1) downloading the git repository, (2) installation of a virtual environment, (3) setting specific parameters in the configuration file and (4) launching the analysis through &#x2018;<b>snakemake</b>&#x2019; <xref rid="B9" ref-type="bibr"><u>10</u></xref>.<li> The computational pipeline depends on an anaconda-based <xref rid="B10" ref-type="bibr"><u>11</u></xref> python 3 virtual environment with the <b>snakemake</b> package available <xref rid="B9" ref-type="bibr"><u>10</u></xref>.<li> The following publicly available software tools are required and thereby installed together with the environment: <b>snakemake</b> (v3.9.1) <xref rid="B9" ref-type="bibr"><u>10</u></xref>,fastx toolkit (v0.0.14) <xref rid="B11" ref-type="bibr"><u>12</u></xref>, STAR (v2.5.2a) <xref rid="B12" ref-type="bibr"><u>13</u></xref>, cutadapt (v1.12) <xref rid="B13" ref-type="bibr"><u>14</u></xref>, samtools (v1.3.1) <xref rid="B13" ref-type="bibr"><u>14</u></xref><xref rid="B14" ref-type="bibr"><u>15</u></xref>, bedtools (v2.26.0) <xref rid="B15" ref-type="bibr"><u>16</u></xref><xref rid="B16" ref-type="bibr"><u>17</u></xref>.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](          <person-group person-group-type="author">            <name>              <surname>Analytics</surname>              <given-names>C</given-names>            </name>          </person-group>          <source>Anaconda Software Distribution</source>          <year>2016</year>          <comment>Available from:<ext-link ext-link-type="uri" xlink:href="continuum.io">https://continuum.io</ext-link></comment>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [12](          <person-group person-group-type="author">            <name>              <surname>Lab</surname>              <given-names>H</given-names>            </name>          </person-group>          <source>FASTX-Toolkit - Hannon Lab</source>          <year>2017</year>          <comment>Available from:<ext-link ext-link-type="uri" xlink:href="hannonlab.cshl.edu/fastx_toolkit/index.html">http://hannonlab.cshl.edu/fastx_toolkit/index.html</ext-link></comment>        ) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [14](          <person-group person-group-type="author">            <name>              <surname>Martin</surname>              <given-names>M</given-names>            </name>          </person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet.journal</source>          <year>2011</year>          <volume>17</volume>          <issue>1</issue>          <fpage>10</fpage>          <lpage>12</lpage>        ) [14](          <person-group person-group-type="author">            <name>              <surname>Martin</surname>              <given-names>M</given-names>            </name>          </person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet.journal</source>          <year>2011</year>          <volume>17</volume>          <issue>1</issue>          <fpage>10</fpage>          <lpage>12</lpage>        ) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21221095">21221095</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28969586">28969586</a> <b>(2)</b> SimBA: A methodology and tools for evaluating the performance of RNA-Seq bioinformatic pipelines<ul><li> A benchmarking pipeline implemented with <italic>SimBA</italic> is composed of three components, i/ Simulation of synthetic data using <italic>SimCT</italic>, ii/ Processing of the synthetic data using a pipeline manager (i.e <b>Snakemake</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], iii/ Qualitative evaluation of the results using <italic>BenchCT</italic>.<li> In this paper, we illustrate how this vision applies in genomic medicine by building a set of RNA-Seq data simulations of a somatic condition and propose an automated workflow using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] to compare tools in three qualitative RNA-Seq use cases: SNVs, indels and fusion genes.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29048530">29048530</a> <b>(1)</b> Population Genomic Analysis of a Pitviper Reveals Microevolutionary Forces Underlying Venom Chemistry<ul><li> <b>Snakemake</b> files (<xref rid="evx199-B51" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) for genome assembly and variant calling, as well as a script for R-based analysis and intermediate data are in a MySQL database available at <ext-link ext-link-type="uri" xlink:href="https://github.com/mikheyev/mucrosquamatus-selection">https://github.com/mikheyev/mucrosquamatus-selection</ext-link>, last accessed September 29, 2017.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28928457">28928457</a> <b>(1)</b> Methanogens predominate in natural corrosion protective layers on metal sheet piles<ul><li>The sequence data were processed using a <b>Snakemake</b> workflow<sup><xref ref-type="bibr" rid="CR53"><u>53</u></xref></sup> in mothur version 1.33.2<sup><xref ref-type="bibr" rid="CR54"><u>54</u></xref></sup> following the SOP for 454 data.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19801464">19801464</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28900256">28900256</a> <b>(1)</b> Genomic and transcriptomic analysis of the toluene degrading black yeast Cladophialophora immunda<ul><li> To this aim a multiple genomes alignment of <italic>Aspergillus niger</italic>, <italic>Aspergillus nidulans</italic>, <italic>Aspergillus fumigatus</italic>, <italic>Cladophialophora immunda</italic>, <italic>Exophiala dermatitidis</italic>, <italic>Fusarium oxysporum</italic>, <italic>Fusarium solani</italic>, <italic>Neurospora crassa</italic>, <italic>Saccharomyces cerevisiae</italic>, <italic>Schizosaccharomyces pombe</italic> and <italic>Trichophyton rubrum</italic> was generated with the multiz pipeline<sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup> implemented in <b>snakemake</b><sup><xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15060014">15060014</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28912836">28912836</a> <b>(1)</b> Methods for enhancing the reproducibility of biomedical research findings using electronic health records<ul><li> Given the steady increase in size and complexity of EHR data, workflow management systems used in bioinformatics such as Galaxy [<xref ref-type="bibr" rid="CR69"><u>69</u></xref>], Taverna [<xref ref-type="bibr" rid="CR70"><u>70</u></xref>], and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR71"><u>71</u></xref>] can enable the development of scalable approaches and tools in EHR research.<li> [69](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16169926">16169926</a>) [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28873405">28873405</a> <b>(3)</b> kWIP: The <italic>k</italic>-mer weighted inner product, a <italic>de novo</italic> estimator of genetic similarity<ul><li> Docker images, <b>Snakemake</b> workflows and Jupyter notebooks used to perform all analyses presented here are available online at <ext-link ext-link-type="uri" xlink:href="https://github.com/kdmurray91/kwip-experiments">https://github.com/kdmurray91/kwip-experiments</ext-link>; the respective software versions are noted within the repository.<li>We provide all information necessary to reproduce our work: the <monospace>kWIP</monospace> analyses performed here are implemented in <b>Snakemake</b> workflows [<xref rid="pcbi.1005727.ref049" ref-type="bibr"><u>49</u></xref>], which describe all steps and software parameters; random seeds have been fixed where necessary.<li> Both the <b>Snakemake</b> workflows and Jupyter notebooks are available online at <ext-link ext-link-type="uri" xlink:href="https://github.com/kdmurray91/kwip-experiments">https://github.com/kdmurray91/kwip-experiments</ext-link>; the respective software versions are noted within this repository.<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28968639">28968639</a> <b>(4)</b> NGS-pipe: a flexible, easily extendable and highly configurable framework for NGS analysis<ul><li> NGS-pipe is implemented using the workflow management system <b>Snakemake</b> (<xref rid="btx540-B5" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> Using <b>Snakemake</b> functionality, there are several additional layers of documentation within NGS-pipe, e.g.<li> If one of the steps of the pipeline failed and produced incomplete or no results, the computation of all depending steps is halted and an error message is thrown, using <b>Snakemake</b> intrinsics.<li> NGS-pipe provides a <b>Snakemake</b>-based framework for analyzing such NGS data in a transparent and reproducible manner.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28835202">28835202</a> <b>(1)</b> Upper and lower respiratory tract microbiota in horses: bacterial communities associated with health and mild asthma (inflammatory airway disease) and effects of dexamethasone<ul><li> The entire procedure was run as a <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>] and code for the pipeline (version 1.0.1) is available on Github (<ext-link ext-link-type="uri" xlink:href="https://github.com/ucvm/vmmp">https://github.com/ucvm/vmmp</ext-link>).<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28807044">28807044</a> <b>(4)</b> MetaMeta: integrating metagenome analysis tools to improve taxonomic profiling<ul><li> MetaMeta was built with <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR12"><u>12</u></xref>] and is open-source.<li> In reality, the parallelization provided by <b>Snakemake</b> makes the pipeline run in a reasonable time using most of the computational resources (Table <xref rid="Tab1" ref-type="table"><u>1</u></xref>).<li> The MetaMeta pipeline was built in a way to facilitate the execution in many computational environments using <b>Snakemake</b> and BioConda.<li><bold>Project name:</bold> MetaMeta<bold>Project home page:</bold> https://github.com/pirovc/metameta<bold>Operating systems:</bold> Linux<bold>Programming language:</bold> Python<bold>Other requirements:</bold> <b>Snakemake</b><bold>Licence:</bold> MIT.<li> [12](Koster J, Rahmann S. Snakemake&#x2013;a scalable bioinformatics workflow engine. Bioinformatics. 2012; 28(19):2520&#x2013;2. doi:<ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.1093/bioinformatics/bts480">10.1093/bioinformatics/bts480</ext-link>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28804485">28804485</a> <b>(1)</b> Transcriptional Profiling of Midguts Prepared from <italic>Trypanosoma/T. congolense</italic>-Positive <italic>Glossina palpalis palpalis</italic> Collected from Two Distinct Cameroonian Foci: Coordinated Signatures of the Midguts&#x2019; Remodeling As <italic>T. congolense</italic>-Supportive Niches<ul><li>The successive tasks of the bioinformatics analysis were managed using a <b>Snakemake</b> workflow (<xref rid="B49" ref-type="bibr"><u>49</u></xref>).<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28720766">28720766</a> <b>(1)</b> Predicting susceptibility to tuberculosis based on gene expression profiling in dendritic cells<ul><li>We automated our analysis using Python (<ext-link ext-link-type="uri" xlink:href="https://www.python.org/">https://www.python.org/</ext-link>) and <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR67"><u>67</u></xref></sup>.<li> [67](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28701218">28701218</a> <b>(1)</b> Investigating reproducibility and tracking provenance &#x2013; A genomic workflow case study<ul><li> However, the command line based pipeline frameworks such as bpipe [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], <b>Snakemake</b> [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>] and Ruffus [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>] used to develop these systems are not flexible enough to support integration of new user-defined steps and analysis tools.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28751969">28751969</a> <b>(3)</b> valr: Reproducible genome interval analysis in R<ul><li>Command-line tools like<monospace>BEDtools</monospace> and<monospace>bedops</monospace> can be incorporated into reproducible workflows (e.g., with<monospace><ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/<b>snakemake</b>/<b>snakemake</b>/wiki/Home"><b>snakemake</b></ext-link></monospace><sup><xref rid="ref-11" ref-type="bibr"><u>11</u></xref></sup>), but it is cumbersome to transition from command-line tools to exploratory analysis and plotting software.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28655203">28655203</a> <b>(1)</b> Kronos: a workflow assembler for genome analytics and informatics<ul><li> A number of lightweight workflow management tools have been specifically developed to simplify scripting for these target users, including Ruffus [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>], Bpipe [<xref rid="bib8" ref-type="bibr"><u>8</u></xref>], and <b>Snakemake</b> [<xref rid="bib9" ref-type="bibr"><u>9</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [8](<person-group person-group-type="author"><name name-style="western"><surname>Sadedin</surname><given-names>S</given-names></name>, <name name-style="western"><surname>Pope</surname><given-names>B</given-names></name>, <name name-style="western"><surname>Oshlack</surname><given-names>A</given-names></name></person-group><article-title>Bpipe: A tool for running and managing bioinformatics pipelines</article-title>. <publisher-name>Bioinformatics</publisher-name><year>2012</year>;<volume>28</volume>(<issue>11</issue>):<fpage>1525</fpage>&#x2013;<lpage>6</lpage>.) [9](<person-group person-group-type="author"><name name-style="western"><surname>Koster</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Rahmann</surname><given-names>S</given-names></name></person-group><article-title>Snakemake - A scalable bioinformatics workflow engine</article-title>. <publisher-name>Bioinformatics</publisher-name><year>2012</year>;<volume>28</volume>(<issue>19</issue>):<fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28594382">28594382</a> <b>(1)</b> Coralsnake Venomics: Analyses of Venom Gland Transcriptomes and Proteomes of Six Brazilian Taxa<ul><li> Some parts of the code were written with <b>Snakemake</b> [<xref rid="B269-toxins-09-00187" ref-type="bibr"><u>269</u></xref>] in order to automate it.<li> [269](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28538723">28538723</a> <b>(1)</b> Genomic epidemiology reveals multiple introductions of Zika virus into the United States<ul><li> <b>Snakemake</b> was used as the workflow management system<sup><xref rid="R49" ref-type="bibr"><u>49</u></xref></sup>.<li> [49](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Building and documenting workflows with python-based snakemake</article-title>          <source>OASIcs-OpenAccess Series in Informatics</source>          <volume>26</volume>          <publisher-name>Schloss Dagstuhl-Leibniz-Zentrum fuer Informatik</publisher-name>          <year>2012</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28579987">28579987</a> <b>(1)</b> Killer Immunoglobulin-Like Receptor Allele Determination Using Next-Generation Sequencing Technology<ul><li> Guyet, Genomics and Bioinformatics Core Facility GenoBiRD, Nantes, France) and consists of an analysis pipeline built with <b>Snakemake</b> on the same logic as the manual method.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28468958">28468958</a> <b>(1)</b> An immediate&#x2013;late gene expression module decodes <styled-content style="fixed-case">ERK</styled-content> signal duration<ul><li> The entire preprocessing pipeline was written in <b>Snakemake</b> (K&#xF6;ster &amp; Rahmann, <xref rid="msb177554-bib-0028" ref-type="ref"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28299179">28299179</a> <b>(1)</b> Cluster Flow: A user-friendly bioinformatics workflow tool<ul><li> Many of the popular tools allow the user to create analysis pipelines using specialised domain specific languages (<italic><b>Snakemake</b></italic><sup><xref rid="ref-1" ref-type="bibr"><u>1</u></xref></sup>,<italic>NextFlow</italic><sup><xref rid="ref-2" ref-type="bibr"><u>2</u></xref></sup>,<italic>Bpipe</italic><sup><xref rid="ref-3" ref-type="bibr"><u>3</u></xref></sup>).<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28327945">28327945</a> <b>(1)</b> Enhancing knowledge discovery from cancer genomics data with Galaxy<ul><li> Tools that automate pipeline development such as Kronos [<xref rid="bib1" ref-type="bibr"><u>1</u></xref>], Nextflow [<xref rid="bib2" ref-type="bibr"><u>2</u></xref>], and <b>Snakemake</b> [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>] can satisfy the needs of more technically savvy users.<li> [1](<person-group person-group-type="author"><name name-style="western"><surname>Jafar Taghiyar</surname><given-names>M</given-names></name><etal/></person-group><article-title>Kronos: a workflow assembler for genome analytics and informatics</article-title>. <source>bioRxiv</source><year>2016</year>:<fpage>040352</fpage><comment>doi:10.1101/040352</comment>.) [2](<person-group person-group-type="author"><name name-style="western"><surname>Kurs</surname><given-names>JP</given-names></name>, <name name-style="western"><surname>Simi</surname><given-names>M</given-names></name>, <name name-style="western"><surname>Campagne</surname><given-names>F.</given-names></name></person-group>, <source>Nextflow Workbench Documentation Booklet</source>. <source>Fabien Campagne</source>, <year>2016</year><comment><ext-link ext-link-type="uri" xlink:href="http://campagnelab.org/download/nextflowworkbench-documentation-1-5-1/?wpdmdl=6826">http://campagnelab.org/download/nextflowworkbench-documentation-1-5-1/?wpdmdl=6826</ext-link></comment>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28204568">28204568</a> <b>(1)</b> Differential Gene Expression in the Human Brain Is Associated with Conserved, but Not Accelerated, Noncoding Sequences<ul><li> In addition to specific tools referenced below, these analyses relied on the R language (<xref rid="msx076-B53" ref-type="bibr"><u>R Core Team 2016</u></xref>), <b>Snakemake</b> (<xref rid="msx076-B30" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>), and many components of the SciPy stack, including Matplotlib (<xref rid="msx076-B21" ref-type="bibr"><u>Hunter 2007</u></xref>).<li> [R Core Team 2016](<collab>R Core Team</collab>. <year>2016</year> R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. Available from: <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/">https://www.R-project.org/</ext-link>, last accessed November 6, 2016.) [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Hunter 2007](<person-group person-group-type="author"><name name-style="western"><surname>Hunter</surname><given-names>JD.</given-names></name></person-group><year>2007</year><article-title>Matplotlib: a 2D graphics environment</article-title>. <source>Comput Sci Eng.</source><volume>9</volume>:<fpage>90</fpage>&#x2013;<lpage>95</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27932211">27932211</a> <b>(1)</b> A novel class of somatic mutations in blood detected preferentially in CD8&#xA0;+ cells<ul><li> Execution of the analysis pipeline was performed using the <b>Snakemake</b> <xref rid="bb0185" ref-type="bibr"><u>[37]</u></xref> software.<li> [[37]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28168105">28168105</a> <b>(1)</b> Successive DNA extractions improve characterization of soil microbial communities<ul><li>Sequencing analyses were performed using a <b>Snakemake</b> workflow (<xref rid="ref-24" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>), which follows a standard operating procedure for 454 data in mothur version 1.33.2 (<xref rid="ref-47" ref-type="bibr"><u>Schloss et al., 2009</u></xref>).<li> [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Schloss et al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19801464">19801464</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28097055">28097055</a> <b>(1)</b> Comprehensive transcriptome analysis provides new insights into nutritional strategies and phylogenetic relationships of chrysophytes<ul><li> All analysis steps were performed using the workflow environment <b>Snakemake</b> (v3.2.1;&#xA0;<xref rid="ref-41" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann (2012)</u></xref>).<li> [K&#xF6;ster &amp; Rahmann (2012)](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28056032">28056032</a> <b>(1)</b> Ten Simple Rules for Developing Usable Software in Computational Biology<ul><li> To this end, consider parallelization and make sure your tool can be integrated seamlessly in workflow management systems (e.g., GALAXY [<xref rid="pcbi.1005265.ref007" ref-type="bibr"><u>7</u></xref>] and Taverna [<xref rid="pcbi.1005265.ref008" ref-type="bibr"><u>8</u></xref>]), pipeline frameworks (e.g., Ruffus [<xref rid="pcbi.1005265.ref009" ref-type="bibr"><u>9</u></xref>] and <b>SnakeMake</b> [<xref rid="pcbi.1005265.ref010" ref-type="bibr"><u>10</u></xref>]), or a cluster framework (e.g., Hadoop, <ext-link ext-link-type="uri" xlink:href="http://hadoop.apache.org/">http://hadoop.apache.org/</ext-link>).<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16169926">16169926</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28101080">28101080</a> <b>(1)</b> Soil Microbiome Is More Heterogeneous in Organic Than in Conventional Farming System<ul><li>The 16S rRNA partial gene reads were analyzed using MOTHUR version 1.33.2 (<xref rid="B60" ref-type="bibr"><u>Schloss et al., 2009</u></xref>) combined with the workflow engine <b>Snakemake</b> (<xref rid="B34" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> [Schloss et al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19801464">19801464</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28035947">28035947</a> <b>(1)</b> Non-Coding RNAs in Lung Cancer: Contribution of Bioinformatics Analysis to the Development of Non-Invasive Diagnostic Tools<ul><li> Nevertheless, complete solutions do not yet exist, but have to be developed using custom scripts or using flexible pipelines like Galaxy [<xref rid="B169-genes-08-00008" ref-type="bibr"><u>169</u></xref>], Ruffus [<xref rid="B170-genes-08-00008" ref-type="bibr"><u>170</u></xref>] or <b>Snakemake</b> [<xref rid="B171-genes-08-00008" ref-type="bibr"><u>171</u></xref>].<li> [169](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=16169926">16169926</a>) [170](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [171](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28066335">28066335</a> <b>(1)</b> Comprehensive Phylogenetic Analysis of Bovine Non-<italic>aureus</italic> Staphylococci Species Based on Whole-Genome Sequencing<ul><li> The entire genome assembly process was automated using the <b>Snakemake</b> workflow engine (K&#xF6;ster and Rahmann, <xref rid="B26" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27995928">27995928</a> <b>(1)</b> Production of individualized V gene databases reveals high levels of immunoglobulin genetic diversity<ul><li>IgDiscover consists of a set of command-line scripts written in Python and a workflow implemented in <b>Snakemake</b><xref ref-type="bibr" rid="b37"><u>37</u></xref> that automatically runs the scripts and other tools in the correct order and with the necessary parameters.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27986083">27986083</a> <b>(5)</b> IMP: a pipeline for&#xA0;reproducible reference-independent integrated metagenomic and metatranscriptomic analyses<ul><li> Workflow implementation and automation is achieved using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>, <xref ref-type="bibr" rid="CR51"><u>51</u></xref>].<li> <b>Snakemake</b> allows the seamless integration of Python code and shell (bash) commands and the use of <italic>make</italic> scripting style, which are arguably some of the most widely used bioinformatic scripting languages.<li> <b>Snakemake</b> also supports parallel processing and the ability to interoperate with various tools and/or web services [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>, <xref ref-type="bibr" rid="CR51"><u>51</u></xref>].<li>Automation of the workflow is achieved using <b>Snakemake</b> 3.4.2 [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>, <xref ref-type="bibr" rid="CR51"><u>51</u></xref>], a Python-based make language implemented specifically for building reproducible bioinformatic workflows and pipelines.<li> <b>Snakemake</b> is inherently modular and thus allows various features to be implemented within IMP, including the options of i) executing specific/selected steps within the pipeline, ii) check-pointing, i.e., resuming analysis from a point of possible interruption/termination, iii) analysis of single-omic datasets (MG or MT).<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [51](          <person-group person-group-type="author">            <name>              <surname>Koster</surname>              <given-names>J</given-names>            </name>          </person-group>          <source>Reproducibility in next-generation sequencing analysis</source>          <year>2014</year>        ) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [51](          <person-group person-group-type="author">            <name>              <surname>Koster</surname>              <given-names>J</given-names>            </name>          </person-group>          <source>Reproducibility in next-generation sequencing analysis</source>          <year>2014</year>        ) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [51](          <person-group person-group-type="author">            <name>              <surname>Koster</surname>              <given-names>J</given-names>            </name>          </person-group>          <source>Reproducibility in next-generation sequencing analysis</source>          <year>2014</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27905880">27905880</a> <b>(4)</b> The Lair: a resource for exploratory analysis of published RNA-Seq data<ul><li>The infrastructure for The Lair is based on <b>Snakemake</b> [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], a python-based workflow system.<li> A master <b>Snakemake</b> workflow organizes a series of computations starting with downloading of data to the sequence read archive and ending with deployment of a sleuth analyses to a Shiny server.<li>The config.json file specifies the species name for the RNA-Seq analysis, allowing <b>Snakemake</b> to automatically download and index the appropriate Ensembl [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>] transcriptome for the analysis, the type of reads (single or paired), the design matrix and type of testing to perform, as well as the DOI of the paper and parameters for the kallisto processing.<li>The <b>Snakemake</b> workflow.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24316576">24316576</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27933079">27933079</a> <b>(1)</b> Small RNAs from <italic>Bemisia tabaci</italic> Are Transferred to <italic>Solanum lycopersicum</italic> Phloem during Feeding<ul><li>Bioinformatic analyses were done using the <b>Snakemake</b> workflow management tool (K&#xF6;ster and Rahmann, <xref rid="B33" ref-type="bibr"><u>2012</u></xref>) to generate bioinformatic pipelines.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27942268">27942268</a> <b>(2)</b> Towards agile large-scale predictive modelling in drug discovery with flow-based programming design principles<ul><li><b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], NextFlow&#xA0;[<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] and BPipe&#xA0;[<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] are text-based tools implemented as Domain Specific Languages (DSL).<li> This was perceived to be the case with <b>Snakemake</b> and BPipe.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](          <person-group person-group-type="author">            <name>              <surname>Tommaso</surname>              <given-names>PD</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Baraja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>A novel tool for highly scalable computational pipelines</article-title>          <source>Figshare</source>          <year>2014</year>        ) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27832061">27832061</a> <b>(1)</b> Ten Simple Rules for Developing Public Biological Databases<ul><li> For meta-databases that draw content from other sites, content aggregation should be automated at regular intervals from the beginning of the project using tools like <b>snakemake</b> [<xref rid="pcbi.1005128.ref040" ref-type="bibr"><u>40</u></xref>].<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27822460">27822460</a> <b>(1)</b> The Transcriptome of <italic>Exophiala dermatitidis</italic> during <italic>Ex-vivo</italic> Skin Model Infection<ul><li>The pipelines for the RNAseq mapping, differential expression, chimeric RNA annotation and gene functional annotation were implemented with <b>snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B56" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27466232">27466232</a> <b>(1)</b> Modeling <italic>cis</italic>-regulation with a compendium of genome-wide histone H3K27ac profiles<ul><li> The higher <inline-formula><mml:math id="IL19"><mml:msub><mml:mrow><mml:mover><mml:mi>s</mml:mi><mml:mo>~</mml:mo></mml:mover></mml:mrow><mml:mi>i</mml:mi></mml:msub></mml:math></inline-formula> is, the more likely UDHS region <italic>i</italic> is to be bound by the factors that regulate the input gene set.</p></list-item></list>The workflow engine, <b>Snakemake</b> (<xref rid="WANGGR201574C33" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>), is used to link together subprocesses in the MARGE pipeline.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27650223">27650223</a> <b>(1)</b> systemPipeR: NGS workflow and report generation environment<ul><li> These include <italic>Galaxy</italic> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>, <xref ref-type="bibr" rid="CR38"><u>38</u></xref>], <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>], <italic>Taverna</italic> [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>], <italic>BioBlend</italic> [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>], <italic>bcbio-nextgen</italic> [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>], <italic>Knime</italic> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], <italic>Ruffus</italic> [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>], <italic>Kepler</italic> [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>], Wasp [<xref ref-type="bibr" rid="CR22"><u>22</u></xref>], <italic>ViennaNGS</italic> [<xref ref-type="bibr" rid="CR23"><u>23</u></xref>], <italic>Mercury</italic> [<xref ref-type="bibr" rid="CR24"><u>24</u></xref>], <italic>RAP</italic> [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>], and <italic>LONI</italic> [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>] among others.<li> [15](          <person-group person-group-type="author">            <name>              <surname>Goecks</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Nekrutenko</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Taylor</surname>              <given-names>J</given-names>            </name>            <collab>Galaxy Team</collab>          </person-group>          <article-title>Galaxy: a comprehensive approach for supporting accessible reproducible, and transparent computational research in the life sciences</article-title>          <source>Genome Biol</source>          <year>2010</year>          <volume>11</volume>          <issue>8</issue>          <fpage>86</fpage>          <pub-id pub-id-type="doi">10.1186/gb-2010-11-8-r86</pub-id>        ) [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22068528">22068528</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [17](          <person-group person-group-type="author">            <name>              <surname>Wolstencroft</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Haines</surname>              <given-names>R</given-names>            </name>            <name>              <surname>Fellows</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Williams</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Withers</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Owen</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Soiland-Reyes</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Dunlop</surname>              <given-names>I</given-names>            </name>            <name>              <surname>Nenadic</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Fisher</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Bhagat</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Belhajjame</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Bacall</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Hardisty</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Nieva de la Hidalga</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Balcazar Vargas</surname>              <given-names>MP</given-names>            </name>            <name>              <surname>Sufi</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Goble</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>The taverna workflow suite: designing and executing workflows of web services on the desktop, web or in the cloud</article-title>          <source>Nucleic Acids Res</source>          <year>2013</year>          <volume>41</volume>          <issue>Web Server issue</issue>          <fpage>557</fpage>          <lpage>61</lpage>          <pub-id pub-id-type="doi">10.1093/nar/gkt328</pub-id>        ) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23630176">23630176</a>) [18](          <person-group person-group-type="author">            <name>              <surname>Guimera</surname>              <given-names>RV</given-names>            </name>          </person-group>          <article-title>bcbio-nextgen: Automated, distributed next-gen sequencing pipeline</article-title>          <source>EMBnet J</source>          <year>2012</year>          <volume>17</volume>          <issue>B</issue>          <fpage>30</fpage>          <pub-id pub-id-type="doi">10.14806/ej.17.B.286</pub-id>        ) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22644661">22644661</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22594911">22594911</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22944616">22944616</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26236465">26236465</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24475911">24475911</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25649116">25649116</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23139896">23139896</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27632392">27632392</a> <b>(1)</b> The Contribution of Mosaic Variants to Autism Spectrum Disorder<ul><li> A variant calling and assessment pipeline was written using <b>Snakemake</b> [<xref rid="pgen.1006245.ref038" ref-type="bibr"><u>38</u></xref>].<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27582018">27582018</a> <b>(2)</b> ANGIOGENES: knowledge database for protein-coding and noncoding RNA genes in endothelial cells<ul><li><xref ref-type="fig" rid="f3"><u>Figure 3</u></xref> shows a directed acyclic graph (DAG) describing the <b>Snakemake</b><xref ref-type="bibr" rid="b43"><u>43</u></xref> pipeline used in ANGIOGENES.<li> Specific details regarding the arguments used to run the pipelines can be found in the <b>Snakemake</b> files hosted at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" ext-link-type="uri" xlink:href="https://bitbucket.org/raistlin91/angiogenes_pipeline">https://bitbucket.org/raistlin91/angiogenes_pipeline</ext-link>.<li> [43](<name><surname>Koster</surname><given-names>J.</given-names></name> &amp; <name><surname>Rahmann</surname><given-names>S.</given-names></name><article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source><volume>28</volume>, <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>, <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id> (<year>2012</year>).) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27460335">27460335</a> <b>(1)</b> Nitrous oxide emission related to ammonia-oxidizing bacteria and mitigation options from N fertilization in a tropical soil<ul><li>MOTHUR Version 1.34.2 was used to process the 16S rRNA partial genes sequences, implemented using a <b>Snakemake</b> workflow on a 32-node server running Linux Ubuntu 14.4<xref ref-type="bibr" rid="b40"><u>40</u></xref>.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27401684">27401684</a> <b>(1)</b> Tools and techniques for computational reproducibility<ul><li> <b>Snakemake</b> [<xref ref-type="bibr" rid="CR109"><u>109</u></xref>]: an extension of Make that provides a more flexible syntax and makes it easier to execute tasks in parallel.<li> [109](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27393146">27393146</a> <b>(1)</b> Age-associated hydroxymethylation in human bone-marrow mesenchymal stem cells<ul><li>All the necessary steps for upstream and downstream analyses were defined and implemented using the <b>Snakemake</b> tool [<xref ref-type="bibr" rid="CR42"><u>42</u></xref>], which allows data scientists to generate a reproducible and inherently parallel processing pipeline.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27307617">27307617</a> <b>(1)</b> RapMap: a rapid, sensitive and accurate tool for mapping RNA-seq reads to transcriptomes<ul><li> All experiments were scripted using <b>Snakemake</b> (<xref rid="btw277-B15" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and performed on a 64-bit linux server with 256 GB of RAM and 4 &#xD7; 6-core Intel Xeon E5-4607 v2 CPUs running at 2.60 GHz.<li> [K&#xF6;ster and Rahmann, 2012](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J.</given-names></name><name><surname>Rahmann</surname><given-names>S.</given-names></name></person-group> (<year>2012</year>) <article-title>Building and documenting workflows with python-based snakemake</article-title>. <source>GCB</source>, <volume>26</volume>, <fpage>49</fpage>&#x2013;<lpage>56</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27255077">27255077</a> <b>(3)</b> SUSHI: an exquisite recipe for fully documented, reproducible and reusable NGS data analysis<ul><li> Scripting frameworks like bpipe [<xref ref-type="bibr" rid="CR6"><u>6</u></xref>], Ruffus [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>], nestly [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>], NGSANE [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], Makeflow [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>], and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR11"><u>11</u></xref>], let users build bioinformatics pipelines in a command line fashion.<li>Various types of workflow management systems are compared</p></caption><table frame="hsides" rules="groups"><thead><tr><th>System</th><th>UI</th><th>Language</th><th>Application</th><th>Meta-info.</th><th>Reproducibility</th><th>Documentation</th></tr></thead><tbody><tr><td>Galaxy</td><td>GUI</td><td>Python</td><td>Workflow editor</td><td>Generating</td><td>Workflow</td><td>Galaxy file (.ga)</td></tr><tr><td>Chipster</td><td>GUI</td><td>Java</td><td>Workflow view</td><td>None</td><td>Workflow</td><td>Chipster file (.bsh)</td></tr><tr><td>GeneProf</td><td>GUI</td><td>Java</td><td>Workflow designer</td><td>None</td><td>Workflow</td><td>Image file</td></tr><tr><td>GenePattern</td><td>GUI,CLI</td><td>Java</td><td>Additional module</td><td>None</td><td>Pipeline</td><td>GenePattern library</td></tr><tr><td>Taverna</td><td>GUI,CLI</td><td>Java,Scufl</td><td>Plugin</td><td>Three types</td><td>Workflow</td><td>Workflow file</td></tr><tr><td>TOGGLE</td><td>CLI</td><td>Perl</td><td>Text file</td><td>None</td><td>Perl script</td><td>Text file</td></tr><tr><td>bpipe</td><td>CLI</td><td>Goovy,Java</td><td>bpipe script</td><td>None</td><td>bpipe script</td><td>bpipe script</td></tr><tr><td>NGSANE</td><td>CLI</td><td>Bash</td><td>Text file</td><td>None</td><td>trigger.sh</td><td>Text file</td></tr><tr><td>nestly</td><td>CLI</td><td>Python</td><td>Python script</td><td>None</td><td>nestrun</td><td>Python script</td></tr><tr><td><b>Snakemake</b>.<li>Various types of workflow management systems are compared</p></caption><table frame="hsides" rules="groups"><thead><tr><th>System</th><th>UI</th><th>Language</th><th>Application</th><th>Meta-info.</th><th>Reproducibility</th><th>Documentation</th></tr></thead><tbody><tr><td>Galaxy</td><td>GUI</td><td>Python</td><td>Workflow editor</td><td>Generating</td><td>Workflow</td><td>Galaxy file (.ga)</td></tr><tr><td>Chipster</td><td>GUI</td><td>Java</td><td>Workflow view</td><td>None</td><td>Workflow</td><td>Chipster file (.bsh)</td></tr><tr><td>GeneProf</td><td>GUI</td><td>Java</td><td>Workflow designer</td><td>None</td><td>Workflow</td><td>Image file</td></tr><tr><td>GenePattern</td><td>GUI,CLI</td><td>Java</td><td>Additional module</td><td>None</td><td>Pipeline</td><td>GenePattern library</td></tr><tr><td>Taverna</td><td>GUI,CLI</td><td>Java,Scufl</td><td>Plugin</td><td>Three types</td><td>Workflow</td><td>Workflow file</td></tr><tr><td>TOGGLE</td><td>CLI</td><td>Perl</td><td>Text file</td><td>None</td><td>Perl script</td><td>Text file</td></tr><tr><td>bpipe</td><td>CLI</td><td>Goovy,Java</td><td>bpipe script</td><td>None</td><td>bpipe script</td><td>bpipe script</td></tr><tr><td>NGSANE</td><td>CLI</td><td>Bash</td><td>Text file</td><td>None</td><td>trigger.sh</td><td>Text file</td></tr><tr><td>nestly</td><td>CLI</td><td>Python</td><td>Python script</td><td>None</td><td>nestrun</td><td>Python script</td></tr><tr><td><b>Snakemake</b></td><td>CLI</td><td>Python</td><td>Build file</td><td>None</td><td><b>snakemake</b>.<li> [6](          <person-group person-group-type="author">            <name>              <surname>Sadedin</surname>              <given-names>SP</given-names>            </name>            <name>              <surname>Pope</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Oshlack</surname>              <given-names>A</given-names>            </name>          </person-group>          <article-title>Bpipe: a tool for running and managing bioinformatics pipelines</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <fpage>1525</fpage>          <lpage>1526</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts167</pub-id>        ) [7](          <person-group person-group-type="author">            <name>              <surname>Goodstadt</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>Ruffus: a lightweight Python library for computational pipelines</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2010</year>          <volume>26</volume>          <fpage>2778</fpage>          <lpage>2779</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btq524</pub-id>        ) [8](          <person-group person-group-type="author">            <name>              <surname>McCoy</surname>              <given-names>CO</given-names>            </name>            <name>              <surname>Gallagher</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Hoffman</surname>              <given-names>NG</given-names>            </name>          </person-group>          <article-title>nestly&#x2014;a framework for running software with nested parameter choices and aggregating results</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2013</year>          <volume>29</volume>          <fpage>387</fpage>          <lpage>388</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts696</pub-id>        ) [9](          <person-group person-group-type="author">            <name>              <surname>Buske</surname>              <given-names>FA</given-names>            </name>            <name>              <surname>French</surname>              <given-names>HJ</given-names>            </name>            <name>              <surname>Smith</surname>              <given-names>MA</given-names>            </name>            <name>              <surname>Clark</surname>              <given-names>SJ</given-names>            </name>            <name>              <surname>Bauer</surname>              <given-names>DC</given-names>            </name>          </person-group>          <article-title>NGSANE: a lightweight production informatics framework for high-throughput data analysis</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2014</year>          <volume>30</volume>          <fpage>1471</fpage>          <lpage>1472</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btu036</pub-id>        ) [10](          <person-group person-group-type="author">            <name>              <surname>Yu</surname>              <given-names>L</given-names>            </name>            <name>              <surname>Moretti</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Thrasher</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Emrich</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Judd</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Thain</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Harnessing Parallelism in Multicore Clusters with the All-Pairs, Wavefront, and Makeflow Abstractions</article-title>          <source>Journal of Cluster Computing</source>          <year>2010</year>          <volume>13</volume>          <fpage>243</fpage>          <lpage>256</lpage>          <pub-id pub-id-type="doi">10.1007/s10586-010-0134-7</pub-id>        ) [11](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics (Oxford, England)</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27441714">27441714</a> <b>(1)</b> Integration and Visualization of Translational Medicine Data for Better Understanding of Human Diseases<ul><li> The first group includes <b>Snakemake</b>,<sup><xref rid="B39" ref-type="bibr"><u>39</u></xref></sup> Yabi,<sup><xref rid="B40" ref-type="bibr"><u>40</u></xref></sup> Chipster,<sup><xref rid="B41" ref-type="bibr"><u>41</u></xref></sup> or Mobyle<sup><xref rid="B42" ref-type="bibr"><u>42</u></xref></sup> and relies on textual workflow construction in a script-like format.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22333270">22333270</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21251254">21251254</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19689959">19689959</a>) <li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22333270">22333270</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21251254">21251254</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19689959">19689959</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27228153">27228153</a> <b>(9)</b> TissueMiner: A multiscale analysis toolkit to quantify how cellular processes create tissue dynamics<ul><li> The <italic><b>snakemake</b></italic> automated workflow is described in <xref ref-type="fig" rid="fig7"><u>Figure 7</u></xref>.<li> To do so, we use the <italic><b>snakemake</b></italic> workflow engine developed by Koster and Rahmann (<xref rid="bib22" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> <italic><b>Snakemake</b></italic> automatically determines the execution order, provides means for error recovery and job control, and supports High Performance Computing (HPC) environments.<li> By using <italic><b>snakemake</b></italic> we enable the user to easily run and monitor TissueMiner, while maintaining a proper decoupling of tools as independent executables.<li> <italic><b>Snakemake</b></italic> automatically builds a directed graph from which the execution order of processing steps is inferred.<li>Automated workflow using <b>snakemake</b>.<li>(<bold>A</bold>) The <b>snakemake</b> engine can generate a directed acyclic graph (DAG) where we show an example here.<li>One major advantage of a workflow engine such as <italic><b>snakemake</b></italic> is that it can run the workflow on various architectures - from single-core workstations to multi-core servers and clusters - without the need to modify the rules, thereby facilitating reproducible research.<li> More advanced users can use TissueMiner directly from the command-line with or without <italic><b>snakemake</b></italic> and can&#xA0;thus perform simultaneous analyses of multiple movies.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> TissueMiner: A multiscale analysis toolkit to quantify how cellular processes create tissue dynamics<ul><li> Subsequent analysis steps, such as video creation, can be run on demand by selecting the corresponding <italic><b>snakemake</b></italic> rules, see <xref ref-type="fig" rid="fig7"><u>Figure 7</u></xref> and TM R-User Manual.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27013646">27013646</a> <b>(2)</b> A review of bioinformatic pipeline frameworks<ul><li><b>Snakemake</b> [<xref rid="bbw020-B5" ref-type="bibr"><u>5</u></xref>] builds on the implicit or wildcard-based logic of Make while extending its capabilities by allowing Python to be interspersed through the pipeline in conjunction with a DSL.<li>A <b>Snakemake</b> rule for performing a sequence alignment.<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26869916">26869916</a> <b>(1)</b> Using Make for Reproducible and Parallel Neuroimaging Workflow and Quality-Assurance<ul><li> <b>Snakemake</b> is a Python-compatible workflow system (with a Python-like syntax) based on the logic of Make (K&#xF6;ster and Rahmann, <xref rid="B21" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26812582">26812582</a> <b>(1)</b> Genetic Variation, Not Cell Type of Origin, Underlies the Majority of Identifiable Regulatory Differences in iPSCs<ul><li> The full pipeline was implemented using the Python package <b>Snakemake</b> [<xref rid="pgen.1005793.ref035" ref-type="bibr"><u>35</u></xref>].<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26673418">26673418</a> <b>(6)</b> Quality control, modeling, and visualization of CRISPR screens with MAGeCK-VISPR<ul><li> MAGeCK-VISPR employs a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>] workflow to combine MAGeCK and VISPR in a scalable and reproducible way (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>).<li>Implementation of the MAGeCK-VISPR workflow with <b>Snakemake</b>.<li>We implemented the MAGeCK-VISPR workflow with the workflow management system <b>Snakemake</b> [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>], allowing an automatic execution of some or all of the MAGeCK-VISPR functions: quality control, essential gene analysis, and visualization.<li> Choosing a workflow management system like <b>Snakemake</b> has several advantages.<li> Second, <b>Snakemake</b> tracks metadata (like creation date, input, and log files) for all generated result and intermediate files.<li> Once configured, the Snakefile can be executed with <b>Snakemake</b>.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26628585">26628585</a> <b>(5)</b> An automated workflow for parallel processing of large multiview SPIM recordings<ul><li>To map and dispatch the workflow logic to a single workstation or on a HPC cluster, we use the automated workflow engine <italic><b>snakemake</b></italic> (<xref rid="btv706-B7" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> Upon invocation, the <italic><b>snakemake</b></italic> rule engine resolves the dependencies between individual processing steps based on the input files required and the output files produced during the workflow.<li> Each instance of <italic><b>snakemake</b></italic> for one dataset is independent and thus the workflow can be applied simultaneously to multiple dataset.<li> The workflow is executed by passing the .<italic>yaml</italic> file to <italic><b>snakemake</b></italic> on the command line (<ext-link ext-link-type="uri" xlink:href="http://bioinformatics.oxfordjournals.org/lookup/suppl/doi:10.1093/bioinformatics/btv706/-/DC1">Supplementary Fig.<li><italic><b>Snakemake</b></italic> supports multiple back ends to perform the command dispatch: local, cluster and <italic>Distributed Resource Management Application API</italic> (<italic>DRMAA</italic>) (<xref rid="btv706-B7" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26600436">26600436</a> <b>(1)</b> Reliably Detecting Clinically Important Variants Requires Both Combined Variant Calls and Optimized Filtering Strategies<ul><li> easy to use GUI and web based approaches like GALAXY [<xref rid="pone.0143199.ref005" ref-type="bibr"><u>5</u></xref>] and Taverna [<xref rid="pone.0143199.ref006" ref-type="bibr"><u>6</u></xref>] compared to language specific frameworks such as GATK&#x2019;s Queue (<ext-link ext-link-type="uri" xlink:href="https://github.com/broadgsa/gatk/">https://github.com/broadgsa/gatk/</ext-link>), BPipe [<xref rid="pone.0143199.ref007" ref-type="bibr"><u>7</u></xref>] and <b>Snakemake</b> [<xref rid="pone.0143199.ref008" ref-type="bibr"><u>8</u></xref>].<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15201187">15201187</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26454273">26454273</a> <b>(1)</b> Jflow: a workflow management system for web applications<ul><li> Other WMS such as weaver (<xref rid="btv589-B2" ref-type="bibr">Bui <italic>et&#xA0;al</italic>., 2012</xref>), <b>Snakemake</b> (<xref rid="btv589-B6" ref-type="bibr">Koster <italic>et&#xA0;al</italic>., 2012</xref>), Ruffus (<xref rid="btv589-B5" ref-type="bibr"><u>Goodstadt, 2010</u></xref>) or Cosmos (<xref rid="btv589-B3" ref-type="bibr">Gafni <italic>et&#xA0;al</italic>., 2014</xref>) provide a framework or a domain-specific language to developers wanting to build and run workflows.<li> [Goodstadt, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26282399">26282399</a> <b>(11)</b> Experiences with workflows for automating data-intensive bioinformatics<ul><li> For this user group, a number of lightweight workflow systems have emerged to simplify scripting and parallelizing tasks, which is particular relevant for an efficient exploitation of HPC resources, including Luigi (<ext-link ext-link-type="uri" xlink:href="https://github.com/spotify/luigi">https://github.com/spotify/luigi</ext-link>), Bpipe [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] and BcBio (<ext-link ext-link-type="uri" xlink:href="https://github.com/chapmanb/bcbio-nextgen">https://github.com/chapmanb/bcbio-nextgen</ext-link>).<li>We have initially tested several systems, including, Bpipe [<xref ref-type="bibr" rid="CR9"><u>9</u></xref>], Moa [<ext-link ext-link-type="uri" xlink:href="https://github.com/mfiers/Moa">https://github.com/mfiers/Moa</ext-link>], Ruffus [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>], and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>].<li> We have since focused on exploring <b>Snakemake</b> due to, among other features, its make-like workflow definition, simple integration with Python, Bash code portability, ease of porting workflows to a cluster, intuitive parallelization, and ongoing active development.<li> We are currently working on extending <b>Snakemake</b> with a lightweight modular system for development cycle control and policy-based specification of rules and requirements that supports an in-flow enforcement of consistency constraints.<li>While <b>Snakemake</b> seems to be a promising tool, on its own, similar to currently available alternatives, it does not provide an exhaustive workflow system solution but instead requires external mechanisms to support critical features like revision control and management of multiple workflow instances run with varying parameter sets.<li> We are now working to integrate <b>Snakemake</b> with external tools and our modular code generation system for in-flow enforcement of consistency constraints.<li> We are investigating Piper (<ext-link ext-link-type="uri" xlink:href="https://github.com/johandahlberg/piper">https://github.com/johandahlberg/piper</ext-link>), <b>Snakemake</b>, and Luigi, which seem to be more adaptable with regard to deployment strategies.<li>The approaches for automating bioinformatics analysis by the organizations at the SeqAhead hackathons and workshops roughly fall into the following categories: scripting (usually in languages such as Bash, Perl, or Python), Makefiles (Make, CMake, etc.), and other workflow systems (such as <b>Snakemake</b>, Luigi, Galax, Taverna, and BcBio).<li> To address these issues both general purpose Make implementations, like SCons [<ext-link ext-link-type="uri" xlink:href="http://www.scons.org">http://www.scons.org</ext-link>], PGMake [<xref ref-type="bibr" rid="CR44"><u>44</u></xref>] or GXP make [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>] were developed, and bioinformatrics-dedicated systems, like Makeflow [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>] and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>].<li> <b>Snakemake</b>, Bpipe).<li>Apart from using workflow systems the authors have developed new tools aiding workflow construction (Chipster, CloudGene, SeqPig), contributed to other workflow systems (Galaxy, <b>Snakemake</b>, Chipster) and analyzed frameworks such as Hadoop and Spark.<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [44](Lih A, Zadok E. Pgmake: A portable distributed make system. 1994. Technical report.) [45](          <person-group person-group-type="author">            <name>              <surname>Taura</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Matsuzaki</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Miwa</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kamoshida</surname>              <given-names>Y</given-names>            </name>            <name>              <surname>Yokoyama</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Dun</surname>              <given-names>N</given-names>            </name>            <etal/>          </person-group>          <article-title>Design and implementation of gxp make - a workflow system based on make</article-title>          <source>Future Gener Comput Syst</source>          <year>2013</year>          <volume>29</volume>          <issue>2</issue>          <fpage>662</fpage>          <lpage>72</lpage>          <pub-id pub-id-type="doi">10.1016/j.future.2011.05.026</pub-id>        ) [46](          <person-group person-group-type="author">            <name>              <surname>Albrecht</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Donnelly</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Bui</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Thain</surname>              <given-names>D</given-names>            </name>          </person-group>          <article-title>Makeflow: A portable abstraction for data intensive computing on clusters, clouds, and grids</article-title>          <source>Proceedings of the 1st ACM SIGMOD Workshop on Scalable Workflow Execution Engines and Technologies. SWEET &#x2019;12</source>          <year>2012</year>          <publisher-loc>New York, NY, USA</publisher-loc>          <publisher-name>ACM</publisher-name>        ) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26091036">26091036</a> <b>(1)</b> Ebola Virus Epidemiology, Transmission, and Evolution during Seven Months in Sierra Leone<ul><li> This pipeline includes command-line tools for each of the above steps and optional <b>Snakemake</b> workflows (<xref rid="bib14" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) to automate them either sequentially or in parallel.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25979828">25979828</a> <b>(1)</b> TUT7 controls the fate of precursor microRNAs by using three different uridylation mechanisms<ul><li> The&#xA0;source codes, workflows implemented in <b>Snakemake</b> (Koster &amp; Rahmann, <xref rid="b25" ref-type="bibr"><u>2012</u></xref>), and interactive notebooks in&#xA0;IPython Notebook&#xA0;(Perez &amp; Granger, <xref rid="b45" ref-type="bibr"><u>2007</u></xref>) used for the analyses in this study are&#xA0;freely available from <ext-link ext-link-type="uri" xlink:href="https://github.com/hyeshik/bskim-2015-pre-miRNA">https://github.com/hyeshik/bskim-2015-pre-miRNA</ext-link>.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2007](          <person-group person-group-type="author">            <name>              <surname>Perez</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Granger</surname>              <given-names>BE</given-names>            </name>          </person-group>          <article-title>IPython: a system for interactive scientific computing</article-title>          <source>Comput Sci Eng</source>          <year>2007</year>          <volume>9</volume>          <fpage>21</fpage>          <lpage>29</lpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25723394">25723394</a> <b>(1)</b> N6-Adenosine Methylation in MiRNAs<ul><li>The whole analysis pipeline was implemented as a <b>Snakemake</b> workflow to keep it reproducible [<xref rid="pone.0118438.ref019" ref-type="bibr"><u>19</u></xref>].<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25362363">25362363</a> <b>(1)</b> Coverage recommendations for methylation analysis by whole genome bisulfite sequencing<ul><li>Analysis was performed using Python (<ext-link ext-link-type="uri" xlink:href="http://www.python.org">www.python.org</ext-link>), R/Bioconductor<sup><xref rid="R13" ref-type="bibr"><u>13</u></xref></sup> and <b>Snakemake</b><sup><xref rid="R14" ref-type="bibr"><u>14</u></xref></sup>.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15461798">15461798</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25289191">25289191</a> <b>(1)</b> Massively parallel read mapping on GPUs with the <italic>q</italic>-group index and PEANUT<ul><li> Documentation and installation instructions are available at <uri xlink:href="http://peanut.readthedocs.org">http://peanut.readthedocs.org</uri>, where we further provide a <b>Snakemake</b> (<xref rid="ref-13" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) workflow of all analyses conducted in this work.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a> <b>(2)</b> BigDataScript: a scripting language for data pipelines<ul><li> although a full comparison is beyond the scope of this article, we mention a few that relate to our work: (i) <b>Snakemake</b> (<xref rid="btu595-B6" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) written as a Python domain-specific language (DSL), which has a strong influence from &#x2018;make&#x2019; command.<li> This concept is at the core of many of the pipeline programming tools, such as <b>SnakeMake</b>, Ruffus, Leaf and Bpipe.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24982428">24982428</a> <b>(1)</b> COSMOS: Python library for massively parallel workflows<ul><li> <b>Snakemake</b> (<xref rid="btu385-B5" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) looks to the proven design of GNU Make to describe DAGs for complicated workflows, whereas the Ruffus project (<xref rid="btu385-B4" ref-type="bibr"><u>Goodstadt, 2010</u></xref>) aims to create a DAG by providing a library of Python decorators.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Goodstadt, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20847218">20847218</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24971059">24971059</a> <b>(1)</b> NeuronDepot: keeping your colleagues in sync by combining modern cloud storage services, the local file system, and simple web applications<ul><li> Workflows are created in Python with the help of libraries like <b>Snakemake</b> (K&#xF6;ster and Rahmann, <xref rid="B17" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25077016">25077016</a> <b>(1)</b> An Illumina metabarcoding pipeline for fungi<ul><li> We refrained from providing a wrapper script for the entire pipeline, but all steps presented here can be easily wrapped up in simple bash scripts, or in workflow engines such as <b>Snakemake</b> (K&#xF6;ster and Rahmann <xref rid="b19" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36243721">36243721</a> <b>(1)</b> Genetic variants associated with psychiatric disorders are enriched at epigenetically active sites in lymphoid cells<ul><li> We first lifted over the summary statistics (autosomal chromosomes only) and harmonized variants to the reference strand using the EBI summary statistics <b>snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/EBISPOT/gwas-sumstats-harmoniser">https://github.com/EBISPOT/gwas-sumstats-harmoniser</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36240281">36240281</a> <b>(1)</b> Machine learning on syngeneic mouse tumor profiles to model clinical immunotherapy response<ul><li> RIMA is an automated <b>Snakemake</b> pipeline developed by our group to streamline RNA-seq data processing, including but not limited to read alignment, quality control (QC), expression quantification, and batch effect removal.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36241673">36241673</a> <b>(6)</b> OSeMOSYS Global, an open-source, open data global electricity system model generator<ul><li> A popular open-source tool used to manage these types of flexible workflows is the Python package <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup>.<li><b>Snakemake</b> provides a system to create scalable and flexible workflow pipelines using Python syntax.<li> In <b>Snakemake</b>, rules are defined which have input files, output files, and an associated script or shell command.<li> Based on the defined input and output files, <b>Snakemake</b> will determine how the rules are linked and execute them in the correct order, parallelizing the work where possible.<li> Furthermore, <b>Snakemake</b> will search through the workflow before running to only execute rules downstream of where changes have been made.<li>The high-level overview of the OSeMOSYS Global&#x2019;s <b>Snakemake</b> workflow is shown in Fig.&#xA0;<xref rid="Fig7" ref-type="fig"><u>7</u></xref> below, and later discussed in more detail.<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36263061">36263061</a> <b>(2)</b> A pseudovirus system enables deep mutational scanning of the full SARS-CoV-2 spike<ul><li> The <italic toggle="yes">dms-vep-pipeline</italic> consists of a series of <b>Snakemake</b> (<xref rid="R44" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) rules that run Python scripts or Jupyter notebooks, and specifies a conda environment that provides details on the software used for the analysis.<li> The repository for BA.1 is publicly available at <ext-link xlink:href="https://github.com/dms-vep/SARS-CoV-2_Omicron_BA.1_spike_DMS_mAbs" ext-link-type="uri">https://github.com/dms-vep/SARS-CoV-2_Omicron_BA.1_spike_DMS_mAbs</ext-link> and the repository for Delta is at <ext-link xlink:href="https://github.com/dms-vep/SARS-CoV-2_Delta_spike_DMS_REGN10933" ext-link-type="uri">https://github.com/dms-vep/SARS-CoV-2_Delta_spike_DMS_REGN10933</ext-link> Note how each repository has a configuration file (the <italic toggle="yes">config.yaml</italic> file), project-specific input data (the <italic toggle="yes">data</italic> subdirectory), and a top-level <b>Snakemake</b> file (the <italic toggle="yes">Snakefile</italic>) that runs the analysis.<li> [M&#xF6;lder et al., 2021](<name><surname>M&#xF6;lder</surname><given-names>F.</given-names></name>, <name><surname>Jablonski</surname><given-names>K.P.</given-names></name>, <name><surname>Letcher</surname><given-names>B.</given-names></name>, <name><surname>Hall</surname><given-names>M.B.</given-names></name>, <name><surname>Tomkins-Tinch</surname><given-names>C.H.</given-names></name>, <name><surname>Sochat</surname><given-names>V.</given-names></name>, <name><surname>Forster</surname><given-names>J.</given-names></name>, <name><surname>Lee</surname><given-names>S.</given-names></name>, <name><surname>Twardziok</surname><given-names>S.O.</given-names></name>, <name><surname>Kanitz</surname><given-names>A.</given-names></name>, <name><surname>Wilm</surname><given-names>A.</given-names></name>, <name><surname>Holtgrewe</surname><given-names>M.</given-names></name>, <name><surname>Rahmann</surname><given-names>S.</given-names></name>, <name><surname>Nahnsen</surname><given-names>S.</given-names></name>, <name><surname>K&#xF6;ster</surname><given-names>J.</given-names></name>, <year>2021</year>. <source>Sustainable data analysis with Snakemake</source>. <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36239393">36239393</a> <b>(7)</b> <italic toggle="yes">binny</italic>: an automated binning algorithm to recover high-quality genomes from complex metagenomic datasets<ul><li> <italic toggle="yes">binny</italic> is implemented as a reproducible Python-based workflow using <b>Snakemake</b> [<xref rid="ref35" ref-type="bibr"><u>35</u></xref>].<li><italic toggle="yes">binny</italic> is implemented as a <b>Snakemake</b> [<xref rid="ref35" ref-type="bibr"><u>35</u></xref>] workflow (<xref rid="f1" ref-type="fig"><u>Figure 1</u></xref>).<li><bold><italic toggle="yes">binny</italic> workflow.</bold> Overview of the <b>Snakemake</b> pipeline and of <italic toggle="yes">binny</italic>&#x2019;s binning method.<li>If not provided explicitly, the average depth of coverage calculation can be performed directly from given BAM files within the <b>Snakemake</b> workflow using BEDTools [<xref rid="ref49" ref-type="bibr"><u>49</u></xref>] <italic toggle="yes">genomeCoverageBed</italic> and an in-house Perl script.<li><italic toggle="yes">binny</italic> is implemented as a <b>Snakemake</b> pipeline and an installation script is provided that takes care of the installation of all necessary dependencies and required databases.<li>VAMB (3.0.2) [<xref rid="ref36" ref-type="bibr"><u>36</u></xref>] was run with the default parameters and using the <b>Snakemake</b> pipeline as described in the documentation (<ext-link xlink:href="https://github.com/RasmussenLab/vamb/blob/master/README.md" ext-link-type="uri">https://github.com/RasmussenLab/vamb/blob/master/README.md</ext-link>).<li> Because VAMB is designed to achieve optimal performance through the combination of the data of multiple samples, the samples from each of the six CAMI datasets were concatenated and run together, as described by the authors (README sections <monospace>Recommended workflow</monospace> and <monospace><b>Snakemake</b> workflow</monospace>).<li> [35](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S</given-names></string-name></person-group>. <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source><year>2018</year>;<volume>34</volume>(<issue>20</issue>): <fpage>3600</fpage>.) [35](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S</given-names></string-name></person-group>. <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source><year>2018</year>;<volume>34</volume>(<issue>20</issue>): <fpage>3600</fpage>.) [49](<person-group person-group-type="author"><string-name><surname>Quinlan</surname><given-names>AR</given-names></string-name>, <string-name><surname>Hall</surname><given-names>IM</given-names></string-name></person-group>. <article-title>BEDTools: a flexible suite of utilities for comparing genomic features</article-title>. <source>Bioinformatics (Oxford, England)</source><year>2010</year>;<volume>26</volume>(<issue>6</issue>):<fpage>841</fpage>&#x2013;<lpage>2</lpage>.) [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33398153">33398153</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36313436">36313436</a> <b>(1)</b> Allelic diversity of the pharmacogene CYP2D6 in New Zealand M&#x101;ori and Pacific peoples<ul><li> An end-to-end pipeline for data management was designed in a conda environment using a <b>snakemake</b> workflow (<xref rid="B25" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>)<bold>.<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36226801">36226801</a> <b>(3)</b> Genome of the endangered Guatemalan Beaded Lizard, <italic toggle="yes">Heloderma charlesbogerti</italic>, reveals evolutionary relationships of squamates and declines in effective population sizes<ul><li> A <b>Snakemake</b> workflow implementing the genome annotation can be found at <ext-link xlink:href="https://github.com/LachanceLab/GBL" ext-link-type="uri">https://github.com/LachanceLab/GBL</ext-link> (<xref rid="jkac276-B52" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.<li> <b>Snakemake</b> workflows for these analyses can be found at <ext-link xlink:href="https://github.com/LachanceLab/GBL" ext-link-type="uri">https://github.com/LachanceLab/GBL</ext-link> (<xref rid="jkac276-B52" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.<li> A <b>snakemake</b> workflow of the analysis is available at <ext-link xlink:href="https://github.com/LachanceLab/GBL" ext-link-type="uri">https://github.com/LachanceLab/GBL</ext-link> (<xref rid="jkac276-B52" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36222650">36222650</a> <b>(1)</b> High-resolution species assignment of <italic toggle="yes">Anopheles</italic> mosquitoes using <italic toggle="yes">k</italic>-mer distances on targeted sequences<ul><li> If the genomic coordinates of the primers are not known, we align the samples to the most appropriate reference genome, identify the genomic coordinates of the primer sites if not yet known and follow the steps above.This pipeline is implemented in <b>Snakemake</b> 5.30.2 (<xref rid="bib37" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36281413">36281413</a> <b>(2)</b> Comparison and critical assessment of single-cell Hi-C protocols<ul><li>ONT-data was processed using Pore-C-<b>Snakemake</b>.<li> For bioinformatics analysis of ONT libraries we used the output of the Pore-C-<b>Snakemake</b> program (version 0.2.0), specifically files from merged_contacts and align_table directories.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36223391">36223391</a> <b>(1)</b> Diverse susceptibilities and responses of human and rodent cells to orthohantavirus infection reveal different levels of cellular restriction<ul><li> Then, we used the RNA-seq pipeline (v0.13.0), which is built on top of <b>Snakemake</b> 5.8.1 [<xref rid="pntd.0010844.ref047" ref-type="bibr"><u>47</u></xref>] and is available online (<ext-link xlink:href="https://github.com/sequana/sequana_rnaseq" ext-link-type="uri">https://github.com/sequana/sequana_rnaseq</ext-link>).<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36217016">36217016</a> <b>(1)</b> Identifying a novel role for the master regulator <italic>Tal1</italic> in the Endothelial to Hematopoietic Transition<ul><li>The processing of the ATAC-seq data was done using an in-house <b>Snakemake</b> pipeline starting from the raw fastq files produced following from sequencing.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36228784">36228784</a> <b>(4)</b> SARS-CoV-2 infection dynamics revealed by wastewater sequencing analysis and deconvolution<ul><li>Feature comparison between different available pipelines and analysis tools.</p></caption><alt-text id="al0035">Table 1</alt-text><table frame="hsides" rules="groups"><thead><tr><th/><th>COJAC (+ V-pipe)</th><th>Freyja</th><th>ARTIC bioinformatics pipeline</th><th>PiGx Sars-Cov-2</th></tr></thead><tbody><tr><td>Deployment</td><td>Package available through conda, but execution relies on separate jupyter notebooks</td><td>Package available through conda</td><td>Package available through conda</td><td>Package available through GNU Guix, workflow management using <b>snakemake</b>.<li> PiGx SARS-CoV-2 offers state-of-the-art software bit-by-bit reproducibility thanks to GNU Guix (<xref rid="bb0230" ref-type="bibr"><u>Wurmus et al., 2018</u></xref>) and workflow management using <b>snakemake</b>.<li> PiGx SARS-CoV-2 uses <b>snakemake</b> (<xref rid="bb0130" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) to define and run the workflow.<li> JD contributed to the pipeline by building the whole automation backbone with <b>snakemake</b> and the initial test dataset.<li> [Wurmus et al., 2018](          <person-group person-group-type="author">            <name>              <surname>Wurmus</surname>              <given-names>R.</given-names>            </name>            <name>              <surname>Uyar</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Osberg</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Franke</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Gosdschan</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Wreczycka</surname>              <given-names>K.</given-names>            </name>            <etal/>          </person-group>          <article-title>PiGx: reproducible genomics analysis pipelines with GNU Guix</article-title>          <source>Gigascience</source>          <volume>7</volume>          <year>2018</year>        ) [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36230844">36230844</a> <b>(1)</b> Single-Cell Identification of Melanoma Biomarkers in Circulating Tumor Cells<ul><li>Data analysis was performed using an adapted <b>Snakemake</b> (version 4.3.0, Johannes K&#xF6;ster, North Rhine-Westphalia, German) pipeline found at <uri xlink:href="https://github.com/m-chang3/CTC_Pipeline">https://github.com/m-chang3/CTC_Pipeline</uri> (accessed on 19th May 2021).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36205601">36205601</a> <b>(1)</b> Phytest: quality control for phylogenetic analyses<ul><li> These include Phytest for quality control in a Nextstrain/<b>Snakemake</b> pipeline (<xref rid="btac664-B4" ref-type="bibr">Hadfield <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36207490">36207490</a> <b>(1)</b> Multivalent binding kinetics resolved by fluorescence proximity sensing<ul><li>We employed <b>Snakemake</b> v6.9.1 using Python v3.8.5 to develop the machine learning workflow<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36339261">36339261</a> <b>(16)</b> RetroSnake: A modular pipeline to detect human endogenous retroviruses in genome sequencing data<ul><li>Here, we present RetroSnake, a comprehensive <b>Snakemake</b> pipeline for the detection of HERVs in short-read NGS data.<li>RetroSnake uses a modern computational workflow management system, <b>Snakemake</b> (<xref rid="bib15" ref-type="bibr"><u>Koster and Rahmann, 2018</u></xref>), to combine all steps employed in the detection of transposable elements into a single, fast, and easy to use pipeline.<li>The <b>Snakemake</b> workflows consist of rules that define how to create output files from input files.<li> Upon execution, <b>Snakemake</b> determines the combination of rules necessary to achieve a requested output.<li> The main advantages of RetroSnake over the other available tools are that it is end-to-end, based on an extensively wet-lab validated protocol for HERV-K detection, and developed within the <b>Snakemake</b> framework which provides computational efficiency, scalability, customizability, crash recovery features, and it is easy to use.<li>The <b>Snakemake</b> framework allows the analysis to scale from single samples on personal computers to multiple samples on HPC clusters.<li> When run on a cluster, <b>Snakemake</b> takes care of submitting jobs in parallel in an optimized way.<li>When <b>Snakemake</b> determines which rules are to be executed by building a DAG of jobs, it checks timestamps of input and output files.<li> As long as a rule was properly executed and its input file is older than its output file, <b>Snakemake</b> will not attempt to regenerate the output.<li> Another advantage provided by the <b>Snakemake</b> framework is that in case of a failed rule, intermediate files created by that rule are removed by <b>Snakemake</b> before exiting.<li><hr/></td></tr><tr><td><b>Snakemake</b>.<li><hr/></td></tr><tr><td><b>Snakemake</b></td><td><xref rid="bib15" ref-type="bibr"><u>Koster and Rahmann (2018)</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>" id="intref0030">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>.<li> [Koster and Rahmann, 2018](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S.</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <volume>34</volume>          <year>2018</year>          <object-id pub-id-type="publisher-id">3600</object-id>        ) [Koster and Rahmann (2018)](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S.</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <volume>34</volume>          <year>2018</year>          <object-id pub-id-type="publisher-id">3600</object-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36200773">36200773</a> <b>(1)</b> Emergence of Tn<italic toggle="yes">1999.7</italic>, a New Transposon in <italic toggle="yes">bla</italic><sub>OXA-48</sub>-Harboring Plasmids Associated with Increased Plasmid Stability<ul><li> Sequence data were processed with an in-house pipeline using <b>Snakemake</b> 5.31.1 (<xref rid="B21" ref-type="bibr"><u>21</u></xref>), a python-based script language.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36636106">36636106</a> <b>(8)</b> Hex Me If You Can<ul><li> All meshes have been generated from computer&#x2010;aided design (CAD) models, following a workflow defined with <b>Snakemake</b> [<xref rid="cgf14608-bib-0018" ref-type="bibr"><u>MJL&#x2217;21</u></xref>], using the Gmsh [<xref rid="cgf14608-bib-0008" ref-type="bibr"><u>GR09</u></xref>] API with custom parameters defined in yaml metadata files.<li> HexMe is designed in a way that new models and even meshing definitions can be added conveniently (<b>Snakemake</b> workflow available on a public GitHub).<li>The pipeline handling the mesh generation is orchestrated by <b>Snakemake</b> [<xref rid="cgf14608-bib-0018" ref-type="bibr"><u>MJL&#x2217;21</u></xref>], a popular (currently &#x223C; 7 citations per week) scalable workflow management system.<li> In a few words, <b>Snakemake</b> is a modern version of GNU Make, whose syntax is close to Python.<li><b>Snakemake</b> scans the workflow in a backward fashion, meaning that the input files are determined from the output ones.<li> The use of <b>Snakemake</b> easily yields a sustainable dataset, since a rule is applied only if an output is either missing or older than the corresponding input.<li> Alternatively, it is possible to download meshes <italic toggle="yes">one&#x2010;by&#x2010;one</italic> from the catalog (the catalog is mostly generated by <b>Snakemake</b>, using the report feature) The catalog is split into three categories (i, n, s), that correspond to the model categories (respectively: industrial, <bold>n</bold>asty, simple).<li> The 189 meshes are produced thanks to a workflow that is defined with <b>Snakemake</b>.<li> [MJL&#x2217;21](<string-name><surname>M&#xF6;lder</surname><given-names>F.</given-names></string-name>, <string-name><surname>Jablonski</surname><given-names>K. P.</given-names></string-name>, <string-name><surname>Letcher</surname><given-names>B.</given-names></string-name>, <string-name><surname>Hall</surname><given-names>M. B.</given-names></string-name>, <string-name><surname>Tomkins&#x2010;Tinch</surname><given-names>C. H.</given-names></string-name>, <string-name><surname>Sochat</surname><given-names>V.</given-names></string-name>, <string-name><surname>Forster</surname><given-names>J.</given-names></string-name>, <string-name><surname>Lee</surname><given-names>S.</given-names></string-name>, <string-name><surname>Twardziok</surname><given-names>S. O.</given-names></string-name>, <string-name><surname>Kanitz</surname><given-names>A.</given-names></string-name>, et al.: <article-title>Sustainable data analysis with snakemake</article-title>. <source>F1000Research</source><volume>10</volume> (<year>2021</year>). 2, 5) [GR09](<string-name><surname>Geuzaine</surname><given-names>C.</given-names></string-name>, <string-name><surname>Remacle</surname><given-names>J.&#x2010;F.</given-names></string-name>: <article-title>Gmsh: A 3&#x2010;d finite element mesh generator with built&#x2010;in pre&#x2010;and post&#x2010;processing facilities</article-title>. <source>International journal for numerical methods in engineering</source><volume>79</volume>, <issue>11</issue> (<year>2009</year>), <fpage>1309</fpage>&#x2013;<lpage>1331</lpage>. 2, 3, 5) [MJL&#x2217;21](<string-name><surname>M&#xF6;lder</surname><given-names>F.</given-names></string-name>, <string-name><surname>Jablonski</surname><given-names>K. P.</given-names></string-name>, <string-name><surname>Letcher</surname><given-names>B.</given-names></string-name>, <string-name><surname>Hall</surname><given-names>M. B.</given-names></string-name>, <string-name><surname>Tomkins&#x2010;Tinch</surname><given-names>C. H.</given-names></string-name>, <string-name><surname>Sochat</surname><given-names>V.</given-names></string-name>, <string-name><surname>Forster</surname><given-names>J.</given-names></string-name>, <string-name><surname>Lee</surname><given-names>S.</given-names></string-name>, <string-name><surname>Twardziok</surname><given-names>S. O.</given-names></string-name>, <string-name><surname>Kanitz</surname><given-names>A.</given-names></string-name>, et al.: <article-title>Sustainable data analysis with snakemake</article-title>. <source>F1000Research</source><volume>10</volume> (<year>2021</year>). 2, 5) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36274703">36274703</a> <b>(1)</b> Leguminous cover crops and soya increased soil fungal diversity and suppressed pathotrophs caused by continuous cereal cropping<ul><li>All sequencing data were processed using a Python-based <italic><b>Snakemake</b></italic> pipeline (<xref rid="ref64" ref-type="bibr"><u>Sun et al., 2021</u></xref>).<li> [Sun et al., 2021](<person-group person-group-type="author"><name><surname>Sun</surname><given-names>X.</given-names></name><name><surname>Hu</surname><given-names>Y. H.</given-names></name><name><surname>Wang</surname><given-names>J.</given-names></name><name><surname>Fang</surname><given-names>C.</given-names></name><name><surname>Li</surname><given-names>J. G.</given-names></name><name><surname>Han</surname><given-names>M.</given-names></name><etal/></person-group>. (<year>2021</year>). <article-title>Efficient and stable metabarcoding sequencing data using a DNBSEQ-G400 sequencer validated by comprehensive community analyses</article-title>. <source>Gigabyte</source><volume>2021</volume>, <fpage>1</fpage>&#x2013;<lpage>15</lpage>. doi: <pub-id pub-id-type="doi">10.46471/gigabyte.16</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36200744">36200744</a> <b>(1)</b> Hot-CarrierTransfer across a Nanoparticle&#x2013;MoleculeJunction: The Importance of Orbital Hybridization and Level Alignment<ul><li> The <b>Snakemake</b><sup><xref ref-type="bibr" rid="ref67"><u>67</u></xref></sup> package was used for managing the calculationworkflow.<li> [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36202811">36202811</a> <b>(1)</b> A method for multiplexed full-length single-molecule sequencing of the human mitochondrial genome<ul><li>At the time of the analysis, the pre-processing protocol of the raw Oxford Nanopore reads produced with the Q20+&#x2009;ligation sequencing kit was still under development (<ext-link ext-link-type="uri" xlink:href="https://nanoporetech.com/about-us/news/new-nanopore-sequencing-chemistry-developers-hands-set-deliver-q20-99-raw-read">https://nanoporetech.com/about-us/news/new-nanopore-sequencing-chemistry-developers-hands-set-deliver-q20-99-raw-read</ext-link>), so we used a custom <b>snakemake</b> pipeline<sup><xref ref-type="bibr" rid="CR55"><u>55</u></xref></sup> (<ext-link ext-link-type="uri" xlink:href="https://github.com/marcDabad/q20plus_rebasecall">https://github.com/marcDabad/q20plus_rebasecall</ext-link>), which involves four different tools.<li> [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36292683">36292683</a> <b>(1)</b> Robust Genetic Analysis of the X-Linked Anophthalmic (<italic toggle="yes">Ie</italic>) Mouse<ul><li> Long-read sequencing analysis was performed using nano-<b>snakemake</b> [<xref rid="B14-genes-13-01797" ref-type="bibr"><u>14</u></xref>] where an average read length of ~14 Kb was obtained.<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31186302">31186302</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36195962">36195962</a> <b>(3)</b> KAGE: fast alignment-free graph-based genotyping of SNPs and short indels<ul><li> Further details about these experiments can be found in the Section&#xA0;<xref rid="Sec8" ref-type="sec"><u>5</u></xref>, and a <b>Snakemake</b> pipeline for reproducing all the results can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/ivargr/genotyping-benchmarking">https://github.com/ivargr/genotyping-benchmarking</ext-link>.<li> A <b>Snakemake</b> [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] pipeline for running all the experiments and a conda environment with all the necessary tools can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/ivargr/genotyping-benchmarking">https://github.com/ivargr/genotyping-benchmarking</ext-link>.<li> Exact data URLs can be found in the config file of the <b>Snakemake</b> pipeline.<li> [17](M&#xF6;lder F, Jablonski KP, Letcher B, Hall MB, Tomkins-Tinch CH, Sochat V, et&#xA0;al. Sustainable data analysis with Snakemake. F1000Research. 2021;10.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36195901">36195901</a> <b>(1)</b> Novel trends of genome&#xA0;evolution in highly complex tropical sponge microbiomes<ul><li> This pipeline was run for each replicated MAG using the <b>Snakemake</b> workflow management system v.5.13.0 [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>].<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36263020">36263020</a> <b>(2)</b> The proneural transcription factor ASCL1 regulates cell proliferation and primes for differentiation in neuroblastoma<ul><li> Analysis was conducted using an available <b>Snakemake</b> pipeline (<xref rid="B54" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> Analysis was conducted using an adapted version of an available <b>Snakemake</b> pipeline (<xref rid="B54" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36178955">36178955</a> <b>(2)</b> HAYSTAC: A Bayesian framework for robust and rapid species identification in high-throughput sequencing data<ul><li>HAYSTAC is written in Python and built upon the <b>snakemake</b> workflow engine [<xref rid="pcbi.1010493.ref047" ref-type="bibr"><u>47</u></xref>].<li> Employing <b>snakemake</b>&#x2019;s feature to use conda for installing software dependencies required by a rule in a containerised fashion, allows HAYSTAC to have a minimum list of dependencies as a package.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(2)</b> HAYSTAC: A Bayesian framework for robust and rapid species identification in high-throughput sequencing data<ul><li>Second, more seriously, this distracts from the issue that HAYSTAC is not a single program, but rather a Python script that uses <b>Snakemake</b> to call eight different programs.<li>While the use of bioconda and <b>Snakemake</b> ensure that all necessary versions of the programs are put in place when the tool is used, my worry would be that the HAYSTAC is "frozen in time" and moving forward it will be difficult to benefit from updates to any of the eight ingredient programs.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36175836">36175836</a> <b>(5)</b> RTX-KG2: a system for building a semantically standardized knowledge graph for translational biomedicine<ul><li> [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>, <xref ref-type="bibr" rid="CR37"><u>37</u></xref>, <xref ref-type="bibr" rid="CR41"><u>41</u></xref>, <xref ref-type="bibr" rid="CR42"><u>42</u></xref>, <xref ref-type="bibr" rid="CR44"><u>44</u></xref>, <xref ref-type="bibr" rid="CR59"><u>59</u></xref>, <xref ref-type="bibr" rid="CR60"><u>60</u></xref>], batch frameworks&#xA0;[<xref ref-type="bibr" rid="CR32"><u>32</u></xref>], rule-based build frameworks&#xA0;[<xref ref-type="bibr" rid="CR33"><u>33</u></xref>, <xref ref-type="bibr" rid="CR35"><u>35</u></xref>, <xref ref-type="bibr" rid="CR61"><u>61</u></xref>], semantic web-compliant build frameworks such as PheKnowLator&#xA0;[<xref ref-type="bibr" rid="CR62"><u>62</u></xref>], and parallel-capable systems such as <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR63"><u>63</u></xref>].<li> RTX-KG2 integrates 70&#xA0;sources including the major sources UMLS, SemMedDB, ChEMBL, DrugBank, SMPDB, Reactome, KEGG, and UniProtKB using a modular build system leveraging the parallel-capable workflow framework, <b>Snakemake</b>.<li> RTX-KG2 uses an ETL approach for knowledge graph construction and it automates builds using <b>Snakemake</b>; together, these enable efficient knowledge graph construction.<li> Although ETL-focused, open-source workflow tools such as Pentaho Kettle and Apache Hop are available, we selected <b>Snakemake</b> because of its seamless integration with Python and its parallelization support.<li> In addition, <b>Snakemake</b> functionality along with the ETL-based approach allows for unchanged source databases to be ignored until the Merge step (see Fig.&#xA0;<xref rid="Fig10" ref-type="fig"><u>10</u></xref>), which reduces build time.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28936969">28936969</a>) [37](Baranzini S, Huang S, Israni S, et al. Scalable precision medicine knowledge engine, 2021. Accessed: 2021-06-01. <ext-link ext-link-type="uri" xlink:href="https://spoke.ucsf.edu">https://spoke.ucsf.edu</ext-link>.) [41](Dumontier M, Callahan A, Cruz-Toledo J, et al. Bio2RDF release 3: a larger connected network of linked data for the life sciences. In: Proceedings of the 2014 international conference on posters &amp; demonstrations track, 2014, vol. 1272, pp. 401&#x2013;404.) [42](          <person-group person-group-type="author">            <name>              <surname>Livingston</surname>              <given-names>KM</given-names>            </name>            <name>              <surname>Bada</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Baumgartner</surname>              <given-names>WA</given-names>            </name>            <name>              <surname>Hunter</surname>              <given-names>LE</given-names>            </name>          </person-group>          <article-title>KaBOB: ontology-based semantic integration of biomedical databases</article-title>          <source>BMC Bioinform</source>          <year>2015</year>          <volume>16</volume>          <issue>1</issue>          <fpage>126</fpage>          <pub-id pub-id-type="doi">10.1186/s12859-015-0559-3</pub-id>        ) [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31410449">31410449</a>) [59](Ramsey S, Koslicki D, Yao Y, et al. RTXteam/RTX: Initial proof-of-concept software version from November 2017, 2018. 10.5281/ZENODO.1185486) [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27899636">27899636</a>) [32](Do&#x11F;an T, Atas H, Joshi V, et al. CROssBAR: comprehensive resource of biomedical relations with deep learning applications and knowledge graph representations. bioRxiv, 2020. 10.1101/2020.09.14.296889) [33](Pareja-Tobes P, Tobes R, Manrique M, et al. Bio4j: a high-performance cloud-enabled graph-based data platform. bioRxiv, 2015. 10.1101/016758) [35](Fiannaca A, La Rosa M, La Paglia L, et al. Biographdb: a new graphdb collecting heterogeneous data for bioinformatics analysis. In: Eighth international conference on bioinformatics, biocomputational systems and biotechnologies, Wilmington, IARIA, 2016.) [61](Elsworth B, Epigraph DB. 2021. 10.5281/ZENODO.4534128.) [62](Callahan TJ, Tripodi IJ, Hunter LE, Baumgartner WA. A framework for automated construction of heterogeneous large-scale biomedical knowledge graphs. bioRxiv, 2020. 10.1101/2020.04.30.071407) [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36174014">36174014</a> <b>(7)</b> MetaWorks: A flexible, scalable bioinformatic pipeline for high-throughput multi-marker biodiversity assessments<ul><li> To facilitate reproducibility, scalability, and shareability of workflows we use the Conda package manager to facilitate the download of most programs and dependencies and the <b>Snakemake</b> workflow manager to automate pipelines and utilize computational resources efficiently [<xref rid="pone.0274260.ref031" ref-type="bibr"><u>31</u></xref>&#x2013;<xref rid="pone.0274260.ref033" ref-type="bibr"><u>33</u></xref>].<li> <b>Snakemake</b> supports re-entrancy and automatic deployment of multiple parallel jobs, both ideal for high performance computing environments where many cores are available to speed up the analysis of large datasets.<li> Several workflows are available as <b>Snakemake</b> pipelines such as taxonomic assignment of ESVs (<xref rid="pone.0274260.g001" ref-type="fig"><u>Fig 1</u></xref>), clustering of ESVs into OTUs, or for processing single reads.<li> The tutorial also walks users through the steps necessary to set up their environment to run the pipeline for the first time, assuming the user has never worked with Conda or <b>Snakemake</b> before.<li> Although multiple <b>Snakemake</b> workflows are provided in MetaWorks, here we show the main workflow that generates taxonomically assigned ESVs.<li> The input files in white boxes are required by <b>snakemake</b> to run the appropriate workflow.<li> There is extensive online documentation available including detailed explanations of the pipeline, available workflows, and a tutorial for new users who have never used Conda or <b>Snakemake</b> before.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34556866">34556866</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36167715">36167715</a> <b>(1)</b> Genomic epidemiology of <italic>Mycobacterium abscessus</italic> in a Canadian cystic fibrosis centre<ul><li> <b>Snakemake</b> (v6.0.3)<sup><xref ref-type="bibr" rid="CR24"><u>24</u></xref></sup> was used to develop and run the analysis workflow, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/waglecn/mabs">https://github.com/waglecn/mabs</ext-link>.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36699358">36699358</a> <b>(3)</b> TOSCA: an automated Tumor Only Somatic CAlling workflow for somatic mutation detection without matched normal samples<ul><li>TOSCA is implemented using the <b>Snakemake</b> management system (<xref rid="vbac070-B6" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and conda environments.<li> <b>Snakemake</b> ensures a well-controlled and scalable execution of an end-to-end analysis of whole-exome sequencing (WES) and targeted panel sequencing (TS) data, starting from one or more fastq files with raw sequencing reads representing distinct samples, either single-end or paired-end.<li> Furthermore, implementation of our workflow on <b>Snakemake</b> framework will facilitate users, even those with no coding experience, to process their own data easily.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36086948">36086948</a> <b>(1)</b> Rapid genome sequencing for pediatrics<ul><li> the use of sophisticated workflow languages, such as nextflow (<ext-link xlink:href="https://www.nextflow.io/" ext-link-type="uri">https://www.nextflow.io/</ext-link>) and <b>snakemake</b> (Molder et al.,&#xA0;<xref rid="humu24466-bib-0048" ref-type="bibr"><u>2021</u></xref>); and simple solutions such as networking the sequencers to allow for direct saving of the data to the compute cluster, removing the need for lengthy transfer of raw data, which can also lead to corruption or loss of data.<li> [2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36151076">36151076</a> <b>(1)</b> Accumulation of mutations in antibody and CD8 T cell epitopes in a B cell depleted lymphoma patient with chronic SARS-CoV-2 infection<ul><li>The phylogenetic tree was obtained from a local Nextstrain installation using the supplied <b>Snakemake</b> workflow<sup><xref ref-type="bibr" rid="CR78"><u>78</u></xref></sup>.<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36138446">36138446</a> <b>(1)</b> An initial investigation of accuracy required for the identification of small molecules in complex samples using quantum chemical calculated NMR chemical shifts<ul><li> ISiCLE is an automated pipeline for high-accuracy chemical property calculation, implemented using the <b>Snakemake</b> workflow management system [<xref ref-type="bibr" rid="CR132"><u>132</u></xref>].<li> [132](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36127363">36127363</a> <b>(1)</b> Adar-mediated A-to-I editing is required for embryonic patterning and innate immune response regulation in zebrafish<ul><li> Sequencing data were demultiplexed, and preprocessed with a <b>Snakemake</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36127324">36127324</a> <b>(3)</b> Deciphering multi-way interactions in the human genome<ul><li>Reads which passed Q-score filtering (&#x2013;min_qscore 7, 4.56 million reads) after base calling on the Oxford Nanopore GridION (Guppy, version 4.0.11) were used as input for the Pore-C-<b>Snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/nanoporetech/Pore-C-<b>Snakemake</b>">https://github.com/nanoporetech/Pore-C-<b>Snakemake</b></ext-link>, commit 6b2f762).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36139499">36139499</a> <b>(1)</b> Macrophage-Secreted CSF1 Transmits a Calorie Restriction-Induced Self-Renewal Signal to Mammary Epithelial Stem Cells<ul><li> Pipeline was run using <b>snakemake</b> [<xref rid="B46-cells-11-02923" ref-type="bibr"><u>46</u></xref>].<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36189299">36189299</a> <b>(1)</b> Splicing factor SRSF1 is essential for CD8 T cell function and host antigen-specific viral immunity<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="B34" ref-type="bibr"><u>34</u></xref>).<li> [34](<person-group person-group-type="author"><name><surname>Cornwell</surname><given-names>MI</given-names></name><name><surname>Vangala</surname><given-names>M</given-names></name><name><surname>Taing</surname><given-names>L</given-names></name><name><surname>Herbert</surname><given-names>Z</given-names></name><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Li</surname><given-names>B</given-names></name><etal/></person-group>. <article-title>Viper: Visualization pipeline for RNA-seq, a snakemake workflow for efficient and complete RNA-seq analysis</article-title>. <source>BMC Bioinf</source> (<year>2018</year>) <volume>19</volume>(<issue>1</issue>):<page-range>135</page-range>. doi:&#xA0;<pub-id pub-id-type="doi">10.1186/s12859-018-2139-9</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36107768">36107768</a> <b>(3)</b> sgcocaller and comapr: personalised haplotype assembly and comparative crossover map analysis using single-gamete sequencing data<ul><li> function from <monospace><b>Snakemake</b></monospace> (<xref rid="B47" ref-type="bibr"><u>47</u></xref>).<li> From comparing the reported max_rss (the maximum amount of memory occupied by a process at any time that is held in main memory (RAM)) by <monospace><b>Snakemake</b></monospace> (<xref rid="B47" ref-type="bibr"><u>47</u></xref>), sgcocaller required only 30&#x2013;50% of the memory required by Hapi (Figure&#xA0;<xref rid="F6" ref-type="fig"><u>6F</u></xref>,&#xA0;G).<li> Each program was executed with three repeats and the computational usages were recorded using the benchmark function in <monospace><b>Snakemake</b></monospace>(<xref rid="B47" ref-type="bibr"><u>47</u></xref>).<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36056719">36056719</a> <b>(1)</b> Whole&#x2010;genome sequencing of SARS&#x2010;CoV&#x2010;2: Comparison of target capture and amplicon single molecule real&#x2010;time sequencing protocols<ul><li>Bioinformatic analysis was done with the <b>Snakemake</b> pipeline for complete genome construction.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36146797">36146797</a> <b>(3)</b> Microseek: A Protein-Based Metagenomic Pipeline for Virus Diagnostic and Discovery<ul><li>We present Microseek, a pipeline with a graphical interface for virus identification and discovery to be run with <b>Snakemake</b> [<xref rid="B29-viruses-14-01990" ref-type="bibr"><u>29</u></xref>] (<xref rid="viruses-14-01990-f001" ref-type="fig"><u>Figure 1</u></xref>).<li> <b>Snakemake</b> manages the clusters options.<li>In this work, we introduced Microseek, a <b>Snakemake</b>-based pipeline for virus identification and discovery taking advantage of a reference viral protein database, RVDB-prot [<xref rid="B14-viruses-14-01990" ref-type="bibr"><u>14</u></xref>].<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32983411">32983411</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36070702">36070702</a> <b>(1)</b> Molecular convergence between Down syndrome and fragile X syndrome identified using human pluripotent stem cell models<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="R9" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<name><surname>Cornwell</surname><given-names>M</given-names></name>, <name><surname>Vangala</surname><given-names>M</given-names></name>, <name><surname>Taing</surname><given-names>L</given-names></name>, <name><surname>Herbert</surname><given-names>Z</given-names></name>, <name><surname>K&#xF6;ster</surname><given-names>J</given-names></name>, <name><surname>Li</surname><given-names>B</given-names></name>, <name><surname>Sun</surname><given-names>H</given-names></name>, <name><surname>Li</surname><given-names>T</given-names></name>, <name><surname>Zhang</surname><given-names>J</given-names></name>, <name><surname>Qiu</surname><given-names>X</given-names></name>, <etal/> (<year>2018</year>). <article-title>VIPER: visualization Pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis</article-title>. <source>BMC Bio-inf</source>. <volume>19</volume>, <fpage>135</fpage>. <pub-id pub-id-type="doi">10.1186/s12859-018-2139-9</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36064320">36064320</a> <b>(3)</b> SUBATOMIC: a SUbgraph BAsed mulTi-OMIcs clustering framework to analyze integrated multi-edge networks<ul><li> Most importantly, we automated the workflow and integrated it into a general <b>Snakemake</b> pipeline.<li>The main analysis workflow has been integrated into one <b>Snakemake</b> pipeline.<li> A <b>Snakemake</b> workflow diagram is available in the Additional file <xref rid="MOESM7" ref-type="media"><u>7</u></xref>, and a schematic overview can be found in Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36409181">36409181</a> <b>(1)</b> Pangenomic Genotyping with the Marker Array<ul><li> Both were measured with the <b>Snakemake</b> tool&#x2019;s <monospace>benchmark</monospace> directive [<xref rid="R22" ref-type="bibr"><u>22</u></xref>].<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36048017">36048017</a> <b>(1)</b> EBF1 primes B-lymphoid enhancers and limits the myeloid bias in murine multipotent progenitors<ul><li> Analysis downstream was performed using a custom pipeline (<ext-link xlink:href="https://github.com/maxplanck-ie/ATACofthesnake" ext-link-type="uri">https://github.com/maxplanck-ie/ATACofthesnake</ext-link>) implemented in <b>snakemake</b> (version 6.0.0 [<xref rid="bib59" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>]).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36050313">36050313</a> <b>(1)</b> Decoding the olfactory map through targeted transcriptomics links murine olfactory receptors to glomeruli<ul><li> <b>Snakemake</b> v3.5.5<sup><xref ref-type="bibr" rid="CR92"><u>92</u></xref></sup> was used to process read files through alignment and quantification.<li> [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36139780">36139780</a> <b>(1)</b> Impact of Antioxidant Feed and Growth Manipulation on the Redox Regulation of Atlantic Salmon Smolts<ul><li> The pipelines of the preprocessing as well as the following deferential gene expression analyses were developed and maintained using <b>Snakemake</b> [<xref rid="B53-antioxidants-11-01708" ref-type="bibr"><u>53</u></xref>].<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36038563">36038563</a> <b>(1)</b> Genomic epidemiology and emergence of SARS-CoV-2 variants of concern in the United Arab Emirates<ul><li> The entire workflow is managed with <b>snakemake</b><sup><xref ref-type="bibr" rid="CR35"><u>35</u></xref></sup>.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36072757">36072757</a> <b>(1)</b> High-throughput scNMT protocol for multiomics profiling of single cells from mouse brain and pancreatic organoids<ul><li> The following steps are required for every cell, so it&#x2019;s advisable to use a workflow manager like <b>Snakemake</b> or Nextflow to process multiple cells in parallel.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36176312">36176312</a> <b>(1)</b> Estimating the effect of donor sex on red blood cell transfused patient mortality: A retrospective cohort study using a targeted learning and emulated trials-based approach<ul><li> The data processing pipeline was made in Python (anaconda3/5.3.0) using <b>snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36029318">36029318</a> <b>(1)</b> Structural variants in the barley gene pool: precision and sensitivity to detect them using short-read sequencing and their association with gene expression and phenotypic variation<ul><li> The workflow was implemented in <b>Snakemake</b> (version 5.10.0) (K&#xF6;ster et&#xA0;al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36008753">36008753</a> <b>(2)</b> Revised time estimation of the ancestral human chromosome 2 fusion<ul><li> The principal pipeline was implemented as a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>] workflow to make it reproducible and scalable.<li> All scripts and <b>Snakemake</b> workflow files are publicly available at GitHub Page: <ext-link ext-link-type="uri" xlink:href="https://github.com/bposzewiecka/tytus">https://github.com/bposzewiecka/tytus</ext-link>.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36091392">36091392</a> <b>(1)</b> The shared mother-child epigenetic signature of neglect is related to maternal adverse events<ul><li> <b>Snakemake</b> was used to treat bioinformatics workflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36145267">36145267</a> <b>(1)</b> Potential Coronaviral Inhibitors of the Nucleocapsid Protein Identified In Silico and In Vitro from a Large Natural Product Library<ul><li>Virtual screening and estimation of binding affinities was performed using a HPC &#x201C;<b>snakemake</b> workflow&#x201D; that implemented automatized steps of structural-based screening.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35993719">35993719</a> <b>(1)</b> Using Nanopore Sequencing to Obtain Complete Bacterial Genomes from Saliva Samples<ul><li>The anvi&#x2019;o (v-dev) pangenomics workflow (<xref rid="B38" ref-type="bibr"><u>38</u></xref>, <xref rid="B60" ref-type="bibr"><u>60</u></xref>, <xref rid="B61" ref-type="bibr"><u>61</u></xref>) was implemented using <b>Snakemake</b> (<xref rid="B62" ref-type="bibr"><u>62</u></xref>) and used to perform the pangenomics analysis.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33349678">33349678</a>) [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29423345">29423345</a>) [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35986880">35986880</a> <b>(1)</b> Secondary zoonotic dog-to-human transmission of SARS-CoV-2 suggested by timeline but refuted by viral genome sequencing<ul><li>The phylogenetic tree was obtained from a local Auspice installation using the <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>].<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36093061">36093061</a> <b>(2)</b> Heme oxygenase-1 mitigates liver injury and fibrosis via modulation of LNX1/Notch1 pathway in myeloid cells<ul><li><hr/></td></tr><tr><td>Differential gene expression analysis (DESeq2, v1.22.1)</td><td>This paper</td><td>GEO Submission (GSE206069)</td></tr><tr><td>RNA Sequencing analysis (VIPER <b>snakemake</b> pipeline).<li> RNA-seq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="bib12" ref-type="bibr"><u>Cornwell et&#xA0;al., 2018</u></xref>).<li> [Cornwell et&#xA0;al., 2018](          <person-group person-group-type="author">            <name>              <surname>Cornwell</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Vangala</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Taing</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Herbert</surname>              <given-names>Z.</given-names>            </name>            <name>              <surname>K&#xF6;ster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Sun</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Zhang</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Qiu</surname>              <given-names>X.</given-names>            </name>            <etal/>          </person-group>          <article-title>VIPER: visualization Pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis</article-title>          <source>BMC Bioinf.</source>          <volume>19</volume>          <year>2018</year>          <fpage>135</fpage>          <pub-id pub-id-type="doi">10.1186/s12859-018-2139-9</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36093058">36093058</a> <b>(5)</b> Global chromatin accessibility profiling analysis reveals a chronic activation state in aged muscle stem cells<ul><li> (2008)</xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/taoliu/MACS/" id="intref0080">https://github.com/taoliu/MACS/</ext-link></td></tr><tr><td>Samtools v1.3</td><td><xref rid="bib21" ref-type="bibr"><u>Li and Durbin (2009)</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://samtools.sourceforge.net/" id="intref0085">http://samtools.sourceforge.net/</ext-link></td></tr><tr><td><b>Snakemake</b> v4.6.0.<li> (2008)</xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/taoliu/MACS/" id="intref0080">https://github.com/taoliu/MACS/</ext-link></td></tr><tr><td>Samtools v1.3</td><td><xref rid="bib21" ref-type="bibr"><u>Li and Durbin (2009)</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://samtools.sourceforge.net/" id="intref0085">http://samtools.sourceforge.net/</ext-link></td></tr><tr><td><b>Snakemake</b> v4.6.0</td><td><xref rid="bib57" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/<b>snakemake</b>/<b>snakemake</b>/wiki/Home" id="intref0090">https://bitbucket.org/<b>snakemake</b>/<b>snakemake</b>/wiki/Home</ext-link>.<li> [Li and Durbin (2009)](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19451168">19451168</a>) [Li and Durbin (2009)](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19451168">19451168</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35978191">35978191</a> <b>(1)</b> Spatial profiling of chromatin accessibility in mouse and human tissues<ul><li> A preprocessing pipeline we developed using <b>Snakemake</b> workflow management system (v5.28.0) is available at GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/dyxmvp/Spatial_ATAC-seq">https://github.com/dyxmvp/Spatial_ATAC-seq</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35970983">35970983</a> <b>(1)</b> Regional connectivity drove bidirectional transmission of SARS-CoV-2 in the Middle East during travel restrictions<ul><li> Consensus sequences were assembled using an inhouse <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> pipeline with bwa-mem<sup><xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup> and iVar v1.2.2<sup><xref ref-type="bibr" rid="CR47"><u>47</u></xref></sup>.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [46](Li, H. Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. 10.48550/ARXIV.1303.3997 (arXiv, 2013).) [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30621750">30621750</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35951114">35951114</a> <b>(2)</b> Systems spatiotemporal dynamics of traumatic brain injury at single-cell resolution reveals humanin as a therapeutic target<ul><li> A single-cell digital gene expression matrix was generated using a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] workflow of Drop-seq Tools [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>] and dropEst [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>].<li> The workflow is available as modified version of the <b>snakemake</b>-based dropSeqPipe (<ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe">https://github.com/Hoohm/dropSeqPipe</ext-link>) workflow and is available on github (<ext-link ext-link-type="uri" xlink:href="https://github.com/darneson/dropSeqPipeDropEST">https://github.com/darneson/dropSeqPipeDropEST</ext-link>).<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26000488">26000488</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29921301">29921301</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35953484">35953484</a> <b>(1)</b> SARS-CoV-2 and other respiratory pathogens are detected in continuous air samples from congregate settings<ul><li> The data were analyzed using a custom workflow implemented in <b>Snakemake</b> and is publicly available at <ext-link ext-link-type="uri" xlink:href="https://github.com/dholab/SARS-CoV-2-Spike-RBD-Analysis">https://github.com/dholab/SARS-CoV-2-Spike-RBD-Analysis</ext-link><sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup>.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35945497">35945497</a> <b>(1)</b> Towards reliable whole genome sequencing for outbreak preparedness and response<ul><li> The complete workflow was written as a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] workflow which is, together with the custom python and R scripts, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/dnieuw/platform-comparison-arbovirus">https://github.com/dnieuw/platform-comparison-arbovirus</ext-link>.<li> [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36016925">36016925</a> <b>(1)</b> Spatial organization and early signaling of the B-cell receptor in CLL<ul><li> Pipeline was run using <b>snakemake</b> (<xref rid="B27" ref-type="bibr"><u>27</u></xref>).<li> [27](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>19</issue>):<page-range>2520&#x2013;2</page-range>. doi:&#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35945619">35945619</a> <b>(1)</b> The recombination landscape of the Khoe-San likely represents the upper limits of recombination divergence in humans<ul><li> We developed a <b>Snakemake</b> pipeline (Fig.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35941371">35941371</a> <b>(3)</b> Geographic isolation drives speciation in Nearctic aphids<ul><li> Reads were assembled using the <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR39"><u>39</u></xref></sup> pipeline developed for analysis of target enrichment data (<ext-link ext-link-type="uri" xlink:href="https://github.com/AAFC-BICoE/<b>snakemake</b>-partial-genome-pipeline">https://github.com/AAFC-BICoE/<b>snakemake</b>-partial-genome-pipeline</ext-link>) and described in detail elsewhere<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup>.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [40](          <person-group person-group-type="author">            <name>              <surname>Brunke</surname>              <given-names>AJ</given-names>            </name>            <etal/>          </person-group>          <article-title>The limits of Quediini at last (Staphylinidae: Staphylininae): a rove beetle mega-radiation resolved by comprehensive sampling and anchored phylogenomics</article-title>          <source>Syst. Entomol.</source>          <year>2021</year>          <volume>46</volume>          <fpage>396</fpage>          <lpage>421</lpage>          <pub-id pub-id-type="doi">10.1111/syen.12468</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36533145">36533145</a> <b>(1)</b> covSampler: A subsampling method with balanced genetic diversity for large-scale SARS-CoV-2 genome data sets<ul><li> The workflow is managed using <b>Snakemake</b> (<xref rid="R19" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>).<li> [M&#xF6;lder et&#xA0;al., 2021](<person-group person-group-type="author"><string-name><surname>M&#xF6;lder</surname><given-names>F.</given-names></string-name></person-group><etal>et&#xA0;al.</etal> (<year>2021</year>) &#x2018;<article-title>Sustainable Data Analysis with Snakemake F1000Research</article-title>&#x2019;, <source><italic toggle="yes">F1000Research</italic></source>, <volume>10</volume>: <page-range>33</page-range>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35992681">35992681</a> <b>(1)</b> The impact of PrsA over-expression on the <italic>Bacillus subtilis</italic> transcriptome during fed-batch fermentation of alpha-amylase production<ul><li> (2022)</xref> (<ext-link xlink:href="https://doi.org/10.5281/zenodo.4534403" ext-link-type="uri">doi: 10.5281/zenodo.4534403</ext-link>), which provides a pipeline in a <b>Snakemake</b> framework nested in computational reproducible Anaconda environments (<xref rid="B39" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35924489">35924489</a> <b>(1)</b> Efficient detection and assembly of non-reference DNA sequences with synthetic long reads<ul><li>We measured Novel-X runtime and memory consumption on Intel Xeon Gold 6230 2.1-3.9GHz processors using 32 cores on the HG002 GIAB genome using <b>Snakemake</b> internal tools.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35918316">35918316</a> <b>(1)</b> Transcriptomic diversity in human medullary thymic epithelial cells<ul><li> We re-implemented the pipeline in a compute-system-independent framework with integrated version management (<b>Snakemake</b><sup><xref ref-type="bibr" rid="CR113"><u>113</u></xref></sup>), allowing for the analysis to run on any high-performance compute cluster or on any local computer.<li> [113](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34316353">34316353</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35913189">35913189</a> <b>(1)</b> Comparative Metagenomics Highlight a Widespread Pathway Involved in Catabolism of Phosphonates in Marine and Terrestrial Serpentinizing Ecosystems<ul><li> This pipeline was implemented in <b>Snakemake</b> (<xref rid="B85" ref-type="bibr"><u>85</u></xref>) and deposited on the Github repository at <ext-link xlink:href="https://github.com/elfrouin/MetaGPipeline.git" ext-link-type="uri">https://github.com/elfrouin/MetaGPipeline.git</ext-link>.<li> [85](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35904814">35904814</a> <b>(1)</b> TET1 regulates gene expression and repression of endogenous retroviruses independent of DNA demethylation<ul><li>MINUTE-ChIP multiplexed FASTQ files were processed using minute, a data processing pipeline implemented in <b>Snakemake</b> (<xref rid="B52" ref-type="bibr"><u>52</u></xref>).<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35902092">35902092</a> <b>(30)</b> Tourmaline: A containerized workflow for rapid and iterable amplicon sequence analysis using QIIME 2 and <b>Snakemake</b><ul><li> <b>Snakemake</b> [<xref rid="bib42" ref-type="bibr"><u>42</u></xref>] is a popular workflow management system in the bioinformatics community that manages input and output files in a defined directory structure, with commands defined in a Snakefile as &#x201C;rules&#x201D; and parameters and initial input files set by the user in a configuration file.<li> <b>Snakemake</b> ensures that only the commands required for requested output files not yet generated are run, saving time and computation when rerunning part of a workflow.<li>Here, we present Tourmaline [<xref rid="bib43" ref-type="bibr"><u>43</u></xref>], an amplicon analysis pipeline that uses <b>Snakemake</b> to run QIIME 2 commands for core analysis and interactive visualization&#x2014;plus workflow-specific commands that generate an HTML report of output and summary tables and figures&#xA0;of data and metadata&#x2014;with rapid analysis aided by workflow iterability and scalability, support for multiple cores, a Docker container, and a detailed tutorial.<li>Tourmaline is a <b>Snakemake</b>-based bioinformatics workflow that operates in a defined directory structure (Fig.&#xA0;<xref rid="fig1" ref-type="fig"><u>1</u></xref>).<li> The starting directory structure is then cloned directly from GitHub and is built out through <b>Snakemake</b> commands, defined as &#x201C;rules&#x201D; in <italic toggle="yes">Snakefile</italic>.<li> Run by calling the <b>Snakemake</b> commands for <italic toggle="yes">denoise, taxonomy, diversity</italic>, and <italic toggle="yes">report</italic>&#x2014;or running just the <italic toggle="yes">report</italic> command to generate all output if the parameters do not need to be changed between individual commands.<li> Some experience with the command line, QIIME 2, and <b>Snakemake</b> is helpful to use Tourmaline; basic tutorials for each of these are provided at [<xref rid="bib45" ref-type="bibr"><u>45</u></xref>].<li>As a <b>Snakemake</b> workflow, Tourmaline has as its core files (i) a <italic toggle="yes">Snakefile</italic> that provides all the commands (rules) that comprise the workflow and (ii) a <italic toggle="yes">config.yaml</italic> file that provides the input files and parameters for the workflow.<li>The workflow is run using <b>Snakemake</b> commands.<li> For example, if using DADA2 paired-end method without any filtering (see below), the commands would be (i) <italic toggle="yes"><b>snakemake</b> dada2_pe_denoise</italic>, (ii) <italic toggle="yes"><b>snakemake</b> dada2_pe_taxonomy_unfiltered</italic>, (iii) <italic toggle="yes"><b>snakemake</b> dada2_pe_diversity_unfiltered</italic>, and (iv) <italic toggle="yes"><b>snakemake</b> dada2_pe_report_unfiltered</italic>.<li> Alternatively, the entire workflow can be run at once with the last command, <italic toggle="yes"><b>snakemake</b> dada2_pe_report_unfiltered</italic>.<li>The first command is <italic toggle="yes"><b>snakemake</b> dada2_pe_denoise</italic> (Fig.&#xA0;<xref rid="fig2" ref-type="fig"><u>2</u></xref>), which imports the FASTQ files and reference database (if not already present in directory <italic toggle="yes">01-imported</italic>), summarizes the FASTQ data, runs denoising using DADA2, and summarizes the output.<li>The second command is <italic toggle="yes"><b>snakemake</b> dada2_pe_taxonomy_unfiltered</italic> (Fig.&#xA0;<xref rid="fig2" ref-type="fig"><u>2</u></xref>), which assigns taxonomy to the representative sequences using a naive Bayes classifier or consensus BLAST or VSEARCH method and generates an interactive taxonomy table and an interactive barplot of sample taxonomic composition.<li>The third command is <italic toggle="yes"><b>snakemake</b> dada2_pe_diversity_unfiltered</italic> (Fig.&#xA0;<xref rid="fig2" ref-type="fig"><u>2</u></xref>), which aligns representative sequences using 1 of 3 methods, computes outliers using odseq [<xref rid="bib54" ref-type="bibr"><u>54</u></xref>], and builds a phylogenetic tree.<li>The fourth and final command is <italic toggle="yes"><b>snakemake</b> dada2_pe_report_unfiltered</italic> (Fig.&#xA0;<xref rid="fig2" ref-type="fig"><u>2</u></xref>), which creates a comprehensive HTML report of parameters, metadata, inputs, outputs, and visualizations in a single file.<li> This is done by setting the filtering parameters in <italic toggle="yes">config.yaml</italic> and providing a list of any individual representative sequences to filter, then running the <italic toggle="yes">filtered</italic> commands of the workflow: <italic toggle="yes"><b>snakemake</b> dada2_pe_taxonomy_filtered, <b>snakemake</b> dada2_pe_diversity_filtered</italic>, and <italic toggle="yes"><b>snakemake</b> dada2_pe_report_filtered</italic> (Fig.&#xA0;<xref rid="fig2" ref-type="fig"><u>2</u></xref>).<li> Speed improvements with parallelization were tested by running <b>Snakemake</b> with either 1 or 8 cores (parameter: <italic toggle="yes">--cores</italic>).<li> The <b>Snakemake</b> command used the parameter <italic toggle="yes">--cores 1</italic> or <italic toggle="yes">--cores 8</italic>, and parameters in <italic toggle="yes">config.yaml</italic> specifying the number of threads for individual rules were set to 1 or 8, respectively.<li><bold><b>Snakemake</b></bold>.<li> Managing the workflow with <b>Snakemake</b> provides several benefits:.<li> Future improvements to the workflow will include support for new QIIME 2 releases and plugins, better integration with <b>Snakemake</b>, possibly including Conda integration and connecting <b>Snakemake</b>&#x2019;s reporting ability with QIIME 2&#x2019;s provenance tracking, and enhanced support for cloud computing environments.<li> This figure&#xA0;was generated from the test data that comes with the repository by running the commands (a) <italic toggle="yes"><b>snakemake</b> dada2_pe_report_unfiltered --dag | dot -Tpdf -Grankdir=LR -Gnodesep=0.1 -Granksep=0.1 &gt; dag_pe_report_unfiltered.pdf</italic> and (b) <italic toggle="yes"><b>snakemake</b> dada2_pe_report_filtered --dag | dot -Tpdf -Grankdir=LR -Gnodesep=0.1 -Granksep=0.1 &gt; dag_pe_report_filtered.pdf</italic>.<li> The file format of <italic toggle="yes">config.yaml</italic>, YAML (yet another markup language), is a simple markup language that is used by <b>Snakemake</b> to specify parameters for a workflow.<li> [42](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S.</given-names></string-name></person-group><article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics (Oxford, England)</source>. <year>2012</year>;<volume>28</volume>:<fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) [43](<person-group person-group-type="author"><collab>Tourmaline GitHub repository</collab></person-group>. <comment><ext-link xlink:href="https://github.com/aomlomics/tourmaline" ext-link-type="uri">https://github.com/aomlomics/tourmaline</ext-link>. Accessed 21 April 2022</comment>.) [45](<person-group person-group-type="author"><collab>Tutorials GitHub repository</collab></person-group>. <comment><ext-link xlink:href="https://github.com/aomlomics/tutorials" ext-link-type="uri">https://github.com/aomlomics/tutorials</ext-link>. Accessed 21 April 2022</comment>.) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26303676">26303676</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35896284">35896284</a> <b>(1)</b> Overcoming resistance to &#x3B1;PD-1 of MMR-deficient tumors with high tumor-induced neutrophils levels by combination of &#x3B1;CTLA-4 and &#x3B1;PD-1 blockers<ul><li> PCR duplicates were removed and base quality scores recalibrated using MarkDuplicates and BaseRecalibrator tools (GATK package).<xref rid="R16" ref-type="bibr"><u>16</u></xref> Somatic SNVs and INDELs were called and filtered using GATK tools Mutect2, FilterMutectCalls and FilterByOrientationBias and functional effects of mutations were annotated with ANNOVAR.<xref rid="R17" ref-type="bibr"><u>17</u></xref> Quality control of FASTQ and mapping was done with FastQC, samtools,<xref rid="R18" ref-type="bibr"><u>18</u></xref> GATK HSMetrics and MultiQc.<xref rid="R19" ref-type="bibr"><u>19</u></xref> The processing steps were combined in a pipeline built with <b>snakemake</b> scripts.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21478889">21478889</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20601685">20601685</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27312411">27312411</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35899080">35899080</a> <b>(13)</b> iCOMIC: a graphical interface-driven bioinformatics pipeline for analyzing cancer omics data<ul><li>iCOMIC is embedded in the popular workflow management system, <b>Snakemake</b> (<xref rid="B37" ref-type="bibr"><u>37</u></xref>).<li> The <b>Snakemake</b> wrapper scripts together with the conda environment manage the automated installation of software and their dependencies.<li> Parallelization of workflows is managed by <b>Snakemake</b>.<li><b>Snakemake</b> workflow management system.<li> files specifying the input, output and the shell/wrapper script form the basic units of <b>Snakemake</b>.<li> The GUI retrieves the user input files and the parameters and communicates with the <b>Snakemake</b> rules to set up the analysis.<li> A Python wrapper binds together the PyQt5 GUI and the <b>Snakemake</b> workflow of iCOMIC.<li> The GUI is embedded in a Python wrapper script that connects the <b>Snakemake</b> workflows.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li><b>Snakemake</b>.<li> [37](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J.</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S.</given-names></string-name></person-group><article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>. <source>Bioinforma. Oxf. Engl.</source><year>2012</year>; <volume>28</volume>:<fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35861394">35861394</a> <b>(1)</b> PanTools v3: functional annotation, classification and phylogenomics<ul><li> We created a <b>Snakemake</b> pipeline (<xref rid="btac506-B7" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35927986">35927986</a> <b>(1)</b> High enhancer activity is an epigenetic feature of HPV negative atypical head and neck squamous cell carcinoma<ul><li>Raw fastq reads for all ChIP-seq experiments were processed using a <b>Snakemake</b> based pipeline <ext-link xlink:href="https://github.com/crazyhottommy/pyflow-ChIPseq" ext-link-type="uri">https://github.com/crazyhottommy/pyflow-ChIPseq</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35852420">35852420</a> <b>(3)</b> Spacemake: processing and analysis of large-scale spatial transcriptomics data<ul><li> Spacemake is written in <b>snakemake</b> [<xref rid="bib13" ref-type="bibr"><u>13</u></xref>] with a back-end logic written in Python.<li> Otherwise, <b>snakemake</b> uses the n_beads parameter of the current run-mode to select the top n_beads number of barcodes with the highest read count using the BamTagHistogram function of Dropseq-tools.<li> implemented the pipeline in <b>snakemake</b>.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35834166">35834166</a> <b>(1)</b> Transmission dynamics of SARS-CoV-2 in British Columbia&#x2019;s largest school district during the second half of the 2020&#x2013;2021 school year<ul><li> Seemann, <xref ref-type="bibr" rid="CR13"><u>2021</u></xref>) modified into a <b>snakemake</b> workflow, followed by clade assignment using the Tree Cluster algorithm using a maximum clade size of 6 mutations separating samples with the same Clade number (Balaban et al., <xref ref-type="bibr" rid="CR1"><u>2019</u></xref>).<li> [2021](Goncalves da Silva, A., &amp; Seemann, T. (2021). <italic>Generate fast trees for clustering purposes in SARS-CoV-2</italic> [Nextflow]. Microbiological Diagnostic Unit Public Health Laboratory. <ext-link ext-link-type="uri" xlink:href="https://github.com/MDU-PHL/kovid-trees-nf">https://github.com/MDU-PHL/kovid-trees-nf</ext-link> (Original work published 2021)) [2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31437182">31437182</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35821263">35821263</a> <b>(1)</b> Multi-omics profiling of collagen-induced arthritis mouse model reveals early metabolic dysregulation via SIRT1 axis<ul><li>FASTQ files generated by sequencing mouse paws were pre-processed by RNA-Seq Analysis <b>Snakemake</b> Workflow (RASflow)<sup><xref ref-type="bibr" rid="CR70"><u>70</u></xref></sup>.<li> [70](          <person-group person-group-type="author">            <name>              <surname>Zhang</surname>              <given-names>X</given-names>            </name>            <name>              <surname>Jonassen</surname>              <given-names>I</given-names>            </name>          </person-group>          <article-title>RASflow: An RNA-Seq analysis workflow with Snakemake</article-title>          <source>BMC Bioinform.</source>          <year>2020</year>          <volume>21</volume>          <issue>1</issue>          <fpage>110</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35889112">35889112</a> <b>(2)</b> Co-Occurrence of <italic toggle="yes">Francisella</italic>, Spotted Fever Group <italic toggle="yes">Rickettsia</italic>, and <italic toggle="yes">Midichloria</italic> in Avian-Associated <italic toggle="yes">Hyalomma rufipes</italic><ul><li>Illumina reads from enriched tick samples were analyzed using a pipeline controlled by <b>Snakemake</b> (v.6.2.1) [<xref rid="B52-microorganisms-10-01393" ref-type="bibr"><u>52</u></xref>].<li> Using the workflow manager <b>Snakemake</b> (v.6.2.1) [<xref rid="B52-microorganisms-10-01393" ref-type="bibr"><u>52</u></xref>], the genome assemblies and the public genomes were aligned pairwise with prograssiveMauve (v.2015_02_13) to selected reference genomes: <italic toggle="yes">F.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35891789">35891789</a> <b>(1)</b> Proteotranscriptomics &#x2013; A facilitator in omics research<ul><li> Nextflow <xref rid="b0245" ref-type="bibr"><u>[49]</u></xref>, <b>Snakemake</b> <xref rid="b0250" ref-type="bibr"><u>[50]</u></xref> or bpipe <xref rid="b0255" ref-type="bibr"><u>[51]</u></xref> to automate all relevant programming steps in a parallelized preferably portable pipeline.<li> [[49]](          <person-group person-group-type="author">            <name>              <surname>di Tommaso</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>E.W.</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>P.P.</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E.</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nat Biotechnol</source>          <volume>2017</volume>          <issue>354</issue>          <year>2017</year>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [[50]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [[51]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35802025">35802025</a> <b>(1)</b> FOXR2 Is an Epigenetically Regulated Pan-Cancer Oncogene That Activates ETS Transcriptional Circuits<ul><li> RNA-seq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="bib28" ref-type="bibr"><u>28</u></xref>).<li> [28](<person-group person-group-type="author"><string-name><surname>Cornwell</surname><given-names>M</given-names></string-name>, <string-name><surname>Vangala</surname><given-names>M</given-names></string-name>, <string-name><surname>Taing</surname><given-names>L</given-names></string-name>, <string-name><surname>Herbert</surname><given-names>Z</given-names></string-name>, <string-name><surname>Koster</surname><given-names>J</given-names></string-name>, <string-name><surname>Li</surname><given-names>B</given-names></string-name>, <etal/></person-group>. <article-title>VIPER: visualization pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis</article-title>. <source>BMC Bioinf</source><year>2018</year>;<volume>19</volume>:<fpage>135</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35802654">35802654</a> <b>(2)</b> Development and application of a next-generation sequencing protocol and bioinformatics pipeline for the comprehensive analysis of the canine immunoglobulin repertoire<ul><li>We developed an analysis pipeline based upon a previously reported protocol [<xref rid="pone.0270710.ref013" ref-type="bibr"><u>13</u></xref>] to profile the BCR repertoire using the <b>Snakemake</b> workflow management system [<xref rid="pone.0270710.ref026" ref-type="bibr"><u>26</u></xref>].<li> Developing the pipeline with <b>Snakemake</b> enables scalability while enforcing reproducibility [<xref rid="pone.0270710.ref026" ref-type="bibr"><u>26</u></xref>].<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27490633">27490633</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35594855">35594855</a> <b>(4)</b> Systematic analysis of intrinsic enhancer-promoter compatibility in the mouse genome<ul><li><hr/></td></tr><tr><td>Homer v4.10</td><td><xref rid="bib31" ref-type="bibr"><u>Heinz et&#xA0;al., 2010</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://homer.ucsd.edu/homer/data/software/homer.v4.10.zip" id="intref0085">http://homer.ucsd.edu/homer/data/software/homer.v4.10.zip</ext-link></td></tr><tr><td>BatchPrimer3 v1.0</td><td><xref rid="bib69" ref-type="bibr"><u>You et&#xA0;al., 2008</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://wheat.pw.usda.gov/demos/BatchPrimer3/" id="intref0090">https://wheat.pw.usda.gov/demos/BatchPrimer3/</ext-link></td></tr><tr><td>R version 4.0.5 (2021-03-31)</td><td><xref rid="bib74" ref-type="bibr"><u>R Core Team, 2021</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0095">https://www.r-project.org/</ext-link></td></tr><tr><td>ggplot2</td><td><xref rid="bib68" ref-type="bibr"><u>Wickham, 2016</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://ggplot2.tidyverse.org/" id="intref0100">https://ggplot2.tidyverse.org/</ext-link></td></tr><tr><td>Bowtie2 version 2.3.4</td><td><xref rid="bib47" ref-type="bibr"><u>Langmead and Salzberg, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" id="intref0105">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</ext-link></td></tr><tr><td>Python version 3.6.2</td><td><xref rid="bib56" ref-type="bibr"><u>Rossum and Drake, 2009</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org/downloads/release/python-362/" id="intref0110">https://www.python.org/downloads/release/python-362/</ext-link></td></tr><tr><td><b>Snakemake</b> version 4.4.0.<li><hr/></td></tr><tr><td>Homer v4.10</td><td><xref rid="bib31" ref-type="bibr"><u>Heinz et&#xA0;al., 2010</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://homer.ucsd.edu/homer/data/software/homer.v4.10.zip" id="intref0085">http://homer.ucsd.edu/homer/data/software/homer.v4.10.zip</ext-link></td></tr><tr><td>BatchPrimer3 v1.0</td><td><xref rid="bib69" ref-type="bibr"><u>You et&#xA0;al., 2008</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://wheat.pw.usda.gov/demos/BatchPrimer3/" id="intref0090">https://wheat.pw.usda.gov/demos/BatchPrimer3/</ext-link></td></tr><tr><td>R version 4.0.5 (2021-03-31)</td><td><xref rid="bib74" ref-type="bibr"><u>R Core Team, 2021</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0095">https://www.r-project.org/</ext-link></td></tr><tr><td>ggplot2</td><td><xref rid="bib68" ref-type="bibr"><u>Wickham, 2016</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://ggplot2.tidyverse.org/" id="intref0100">https://ggplot2.tidyverse.org/</ext-link></td></tr><tr><td>Bowtie2 version 2.3.4</td><td><xref rid="bib47" ref-type="bibr"><u>Langmead and Salzberg, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" id="intref0105">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</ext-link></td></tr><tr><td>Python version 3.6.2</td><td><xref rid="bib56" ref-type="bibr"><u>Rossum and Drake, 2009</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org/downloads/release/python-362/" id="intref0110">https://www.python.org/downloads/release/python-362/</ext-link></td></tr><tr><td><b>Snakemake</b> version 4.4.0</td><td><xref rid="bib45" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/<b>snakemake</b>/files?version=4.4.0" id="intref0115">https://anaconda.org/bioconda/<b>snakemake</b>/files?version=4.4.0</ext-link>.<li> The processes described in Linking barcodes to element singlets or duplets and Pre-Processing of cDNA and pDNA reads were performed using a custom <b>Snakemake</b> (<xref rid="bib45" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) pipeline.<li> [Heinz et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20513432">20513432</a>) [You et&#xA0;al., 2008](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18510760">18510760</a>) [R Core Team, 2021](R Core Team (2021). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/" id="intref0010a">https://www.R-project.org/</ext-link>.) [Wickham, 2016](          <person-group person-group-type="author">            <name>              <surname>Wickham</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>ggplot2 Elegant Graphics for Data Analysis</part-title>          <year>2016</year>          <publisher-name>Springer International Publishing</publisher-name>        ) [Langmead and Salzberg, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [Rossum and Drake, 2009](          <person-group person-group-type="author">            <name>              <surname>Rossum</surname>              <given-names>G.V.</given-names>            </name>            <name>              <surname>Drake</surname>              <given-names>F.L.</given-names>            </name>          </person-group>          <part-title>Python 3 Reference Manual</part-title>          <year>2009</year>          <publisher-name>CreateSpace</publisher-name>        ) [Heinz et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20513432">20513432</a>) [You et&#xA0;al., 2008](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18510760">18510760</a>) [R Core Team, 2021](R Core Team (2021). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/" id="intref0010a">https://www.R-project.org/</ext-link>.) [Wickham, 2016](          <person-group person-group-type="author">            <name>              <surname>Wickham</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>ggplot2 Elegant Graphics for Data Analysis</part-title>          <year>2016</year>          <publisher-name>Springer International Publishing</publisher-name>        ) [Langmead and Salzberg, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [Rossum and Drake, 2009](          <person-group person-group-type="author">            <name>              <surname>Rossum</surname>              <given-names>G.V.</given-names>            </name>            <name>              <surname>Drake</surname>              <given-names>F.L.</given-names>            </name>          </person-group>          <part-title>Python 3 Reference Manual</part-title>          <year>2009</year>          <publisher-name>CreateSpace</publisher-name>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35895415">35895415</a> <b>(1)</b> Genomic Sequencing of Dengue Virus Strains Associated with Papua New Guinean Outbreaks in 2016 Reveals Endemic Circulation of DENV-1 and DENV-2<ul><li> <b>Snakemake</b> v.5.26.1<xref rid="b14" ref-type="bibr"><sup>14</sup></xref> was used to run the workflow and the generated newick tree was drawn and annotated using GGTREE v.1.14.6.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304319">36304319</a> <b>(18)</b> Snaq: A Dynamic <b>Snakemake</b> Pipeline for Microbiome Data Analysis With QIIME2<ul><li>By combining the analysis strengths of QIIME2 with the flexibility in the definition of pipelines provided by <b>Snakemake</b>, here we introduce &#x201C;Snaq&#x201D;, a dynamic <b>Snakemake</b> pipeline for microbiome data analysis with QIIME2.<li>In an effort closer to Snaq, Hu and Alexander implemented a <b>Snakemake</b> pipeline for QIIME2 analysis, designed to run with parameters specified through a configuration file (<xref rid="B17" ref-type="bibr"><u>Hu and Alexander, 2020</u></xref>).<li>Dadasnake is another example of a <b>Snakemake</b> pipeline that automates DADA2 analysis outside the setting of the QIIME2 framework (<xref rid="B31" ref-type="bibr"><u>Wei&#xDF;becker et al., 2020</u></xref>).<li><b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>; <xref rid="B23" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) is a Python dialect created for the specification of pipeline workflows.<li> A <b>Snakemake</b> pipeline is specified through the definition of <italic>rules</italic>; where each rule typically has: <italic>input</italic> for the specification of input files; <italic>output</italic> for the specification of output files; and <italic>shell</italic> for the specification of the command used to produce the output based on the input.<li>The execution of a <b>Snakemake</b> pipeline is achieved via the definition of a single target file name.<li> <b>Snakemake</b> will then determine the steps required to produce the target output based on its rules, the file name, and the application of the wildcards concept<xref rid="fn2" ref-type="fn"><sup>2</sup></xref>.<li> This feature of <b>Snakemake</b> is especially suitable for parameter optimization.<li> <monospace>Snakefile</monospace> is the file where all the required <b>snakemake</b> rules are implemented; notice these rules were carefully constructed as not to contend with each other and to make the whole process run smoothly.<li> Then, by incorporating a descriptive name file convention and the rule-based structure of <b>Snakemake</b>, it makes possible the definition and execution of dynamically defined pipelines through a single terminal command.<li> This means that <b>Snakemake</b> will parse the target file name and infer the sequence of steps and the parameter values used.<li>To let <b>Snakemake</b> infer the required steps and their order, we used a predetermined output nomenclature for each stage (<xref rid="T1" ref-type="table"><u>Table 1</u></xref>).<li> The command required to run this step is: <monospace><b>snakemake</b> &#x2013;&#x2013;use-conda &#x2013;&#x2013;cores 10 results/AB/AB.qza</monospace> Hereafter, we will omit the command and options (<monospace><b>snakemake</b> &#x2013;&#x2013;use-conda &#x2013;&#x2013;cores 10</monospace>), and focus only on the target file name.<li> We recommend running Snaq after activating the <b>Snakemake</b> environment installed using Conda.<li> Once input data is available and the first step in the analysis is executed, Snaq (through <b>Snakemake</b>) will automatically build the Conda environment required for that step and download the QIIME2 plugins specified in the corresponding environment YAML file.<li>We have introduced Snaq, a <b>Snakemake</b> pipeline for QIIME2 16S data analysis, including data QC and trimming.<li> [Hu and Alexander, 2020](<person-group person-group-type="author"><name><surname>Hu</surname><given-names>S. K.</given-names></name><name><surname>Alexander</surname><given-names>H.</given-names></name></person-group> (<year>2020</year>). <article-title>Pipeline to Run Qiime2 with Snakemake</article-title>. <comment><italic>Github Repository</italic>. Available at: <ext-link xlink:href="http://Https://github.com/shu251/tagseq-qiime2-snakemake" ext-link-type="uri">Https://github.com/shu251/tagseq-qiime2-snakemake</ext-link> (Accessed on 09 04, 2021).</comment>) [Wei&#xDF;becker et al., 2020](<person-group person-group-type="author"><name><surname>Wei&#xDF;becker</surname><given-names>C.</given-names></name><name><surname>Schnabel</surname><given-names>B.</given-names></name><name><surname>Heintz-Buschart</surname><given-names>A.</given-names></name></person-group> (<year>2020</year>). <article-title>Dadasnake, a Snakemake Implementation of DADA2 to Process Amplicon Sequencing Data for Microbial Ecology</article-title>. <source>GigaScience</source><volume>9</volume>, <fpage>135</fpage>. <pub-id pub-id-type="doi">10.1093/gigascience/giaa135</pub-id>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35846154">35846154</a> <b>(7)</b> RNA-SSNV: A Reliable Somatic Single Nucleotide Variant Identification Framework for Bulk RNA-Seq Data<ul><li>We built our whole framework using <b>Snakemake</b> (<xref rid="B33" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and class-oriented python scripts.<li> <b>Snakemake</b> (<ext-link xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>" ext-link-type="uri">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>) was applied to manage standard bioinformatic workflows involved in this study (co-cleaning, calling, and annotation) and conduct task auto-management without complicating shell scripts.<li> Both <b>Snakemake</b>-based workflows and python scripts were available within our project repository (<ext-link xlink:href="https://github.com/pmglab/RNA-SSNV" ext-link-type="uri">https://github.com/pmglab/RNA-SSNV</ext-link>), which helped create reproducible analysis.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35767012">35767012</a> <b>(1)</b> Evaluation of a Wastewater-Based Epidemiological Approach to Estimate the Prevalence of SARS-CoV-2 Infections and the Detection of Viral Variants in Disparate Oregon Communities at City and Neighborhood Scales<ul><li> The custom software to conduct these calculations was implemented in R and packaged into a <italic toggle="yes"><b>SnakeMake</b></italic> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35836413">35836413</a> <b>(1)</b> Composition and Biogeography of Planktonic Pro- and Eukaryotic Communities in the Atlantic Ocean: Primer Choice Matters<ul><li> MGPrimerEval is an automated <b>snakemake</b> pipeline that identifies binding sites for SSU rRNA oligonucleotide primers from raw, unassembled, and quality-filtered metagenomic reads.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35764313">35764313</a> <b>(1)</b> Long-Read Sequencing Identifies the First Retrotransposon Insertion and Resolves Structural Variants Causing Antithrombin Deficiency<ul><li>We used the <b>Snakemake</b> library to develop an<italic>in-house</italic>multimodal analysis workflow for the sensitive detection of SVs,<xref rid="JR210539-18" ref-type="bibr"><u>18</u></xref>which is publicly available at<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/who-blackbird/magpie">https://github.com/who-blackbird/magpie</uri>.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31186302">31186302</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35760914">35760914</a> <b>(1)</b> High-throughput total RNA sequencing in single cells using VASA-seq<ul><li> To this end, we implemented a custom computational workflow using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR71"><u>71</u></xref></sup> based on Hisat2/StringTie2 (ref.<li> [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35758606">35758606</a> <b>(2)</b> Quality-controlled R-loop meta-analysis reveals the characteristics of R-loop consensus regions<ul><li> For all upstream processing steps, we used <italic toggle="yes">RLPipes</italic> v0.9.0, a CLI tool based on <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>)&#xA0;(see Availability).<li> Then, each down-sampled peak set was processed further using the <italic toggle="yes">rlregions.smk</italic> workflow (built on <italic toggle="yes"><b>snakemake</b></italic>) from the accompanying data generation repository (see Availability).<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35804890">35804890</a> <b>(4)</b> Epithelial-to-Mesenchymal Transition Drives Invasiveness of Breast Cancer Brain Metastases<ul><li> The entire analysis was implemented by Julian de Ruiter using <b>Snakemake</b> (<b>snakemake</b> version 5.19.2; wrapper version 0.63.0) (Amsterdam, Netherlands) [<xref rid="B47-cancers-14-03115" ref-type="bibr"><u>47</u></xref>] and is freely available on GitHub (<uri xlink:href="https://github.com/jrderuiter/<b>snakemake</b>-rnaseq">https://github.com/jrderuiter/<b>snakemake</b>-rnaseq</uri>) (accessed on 4 June 2020).<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35750695">35750695</a> <b>(1)</b> Results of the phase I CCTG IND.231 trial of CX-5461 in patients with advanced solid tumors enriched for DNA-repair deficiencies<ul><li> Pipeline workflow with python 3.6.12, <b>snakemake</b> 3.13.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35482965">35482965</a> <b>(1)</b> Intervening on the storage time of RBC units and its effects on adverse recipient outcomes using real-world data<ul><li>Statistical analyses were performed in R (version 3.6.2) and data processing in python (anaconda3/4.4.0) using <b>Snakemake</b> (version 6.9.1).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35646329">35646329</a> <b>(2)</b> Python for gene expression<ul><li>BioPython<break/><b>Snakemake</b><sup><xref rid="tfn1" ref-type="table-fn"><sup>1</sup></xref></sup>.<li><b>Snakemake</b><sup><xref rid="ref9" ref-type="bibr"><u>9</u></xref></sup> is python based workflow managing system, in other words pipelines organizing software, which is more than a regular package (compared to others mentioned in this table).<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35888710">35888710</a> <b>(1)</b> A Current Encyclopedia of Bioinformatics Tools, Data Formats and Resources for Mass Spectrometry Lipidomics<ul><li> Using graphical workflow management systems, such as Galaxy [<xref rid="B139-metabolites-12-00584" ref-type="bibr"><u>139</u></xref>] and KNIME [<xref rid="B140-metabolites-12-00584" ref-type="bibr"><u>140</u></xref>], or programmatic/declarative workflow systems, such as <b>Snakemake</b> [<xref rid="B141-metabolites-12-00584" ref-type="bibr"><u>141</u></xref>], Nextflow [<xref rid="B142-metabolites-12-00584" ref-type="bibr"><u>142</u></xref>] or CWL [<xref rid="B143-metabolites-12-00584" ref-type="bibr"><u>143</u></xref>], enable the creation and sharing of reproducible data workflows that have all parameters to individual processing steps defined and documented.<li> [139](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [140](          <person-group person-group-type="author"><name><surname>Berthold</surname><given-names>M.R.</given-names></name><name><surname>Cebron</surname><given-names>N.</given-names></name><name><surname>Dill</surname><given-names>F.</given-names></name><name><surname>Gabriel</surname><given-names>T.R.</given-names></name><name><surname>K&#xF6;tter</surname><given-names>T.</given-names></name><name><surname>Meinl</surname><given-names>T.</given-names></name><name><surname>Ohl</surname><given-names>P.</given-names></name><name><surname>Sieb</surname><given-names>C.</given-names></name><name><surname>Thiel</surname><given-names>K.</given-names></name><name><surname>Wiswedel</surname><given-names>B.</given-names></name></person-group>          <article-title>KNIME: The Konstanz Information Miner</article-title>          <source>Data Analysis, Machine Learning and Applications</source>          <person-group person-group-type="editor"><name><surname>Preisach</surname><given-names>C.</given-names></name><name><surname>Burkhardt</surname><given-names>H.</given-names></name><name><surname>Schmidt-Thieme</surname><given-names>L.</given-names></name><name><surname>Decker</surname><given-names>R.</given-names></name></person-group>          <publisher-name>Springer</publisher-name>          <publisher-loc>Berlin/Heidelberg, Germany</publisher-loc>          <year>2008</year>          <fpage>319</fpage>          <lpage>326</lpage>        ) [141](          <person-group person-group-type="author"><name><surname>M&#xF6;lder</surname><given-names>F.</given-names></name><name><surname>Jablonski</surname><given-names>K.P.</given-names></name><name><surname>Letcher</surname><given-names>B.</given-names></name><name><surname>Hall</surname><given-names>M.B.</given-names></name><name><surname>Tomkins-Tinch</surname><given-names>C.H.</given-names></name><name><surname>Sochat</surname><given-names>V.</given-names></name><name><surname>Forster</surname><given-names>J.</given-names></name><name><surname>Lee</surname><given-names>S.</given-names></name><name><surname>Twardziok</surname><given-names>S.O.</given-names></name><name><surname>Kanitz</surname><given-names>A.</given-names></name><etal/></person-group>          <article-title>Sustainable Data Analysis with Snakemake</article-title>          <source>F1000Research</source>          <year>2021</year>          <volume>10</volume>          <fpage>PMC8114187</fpage>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) [142](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [143](          <person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P.</given-names></name><name><surname>Crusoe</surname><given-names>M.R.</given-names></name><name><surname>Tijani&#x107;</surname><given-names>N.</given-names></name><name><surname>Chapman</surname><given-names>B.</given-names></name><name><surname>Chilton</surname><given-names>J.</given-names></name><name><surname>Heuer</surname><given-names>M.</given-names></name><name><surname>Kartashov</surname><given-names>A.</given-names></name><name><surname>Leehr</surname><given-names>D.</given-names></name><name><surname>M&#xE9;nager</surname><given-names>H.</given-names></name><name><surname>Nedeljkovich</surname><given-names>M.</given-names></name><etal/></person-group>          <source>Common Workflow Language</source>          <comment>version 1.0; Figshare</comment>          <publisher-name>Digital Science</publisher-name>          <publisher-loc>London, UK</publisher-loc>          <year>2016</year>          <pub-id pub-id-type="doi">10.6084/M9.FIGSHARE.3115156.V2</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35776646">35776646</a> <b>(9)</b> High-throughput sequencing SELEX for the determination of DNA-binding protein specificities <italic>in&#xA0;vitro</italic><ul><li><hr/></td></tr><tr><td>Every Motif Ever (eme_selex)</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/kashyapchhatbar/eme_selex" id="intref0105">https://github.com/kashyapchhatbar/eme_selex</ext-link> (This paper)</td><td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.6586738" id="intref0110">https://doi.org/10.5281/zenodo.6586738</ext-link></td></tr><tr><td>Flexbar 3.5.0</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/seqan/flexbar" id="intref0115">https://github.com/seqan/flexbar</ext-link></td><td>N/A</td></tr><tr><td><b>Snakemake</b>.<li><hr/></td></tr><tr><td>Every Motif Ever (eme_selex)</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/kashyapchhatbar/eme_selex" id="intref0105">https://github.com/kashyapchhatbar/eme_selex</ext-link> (This paper)</td><td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.6586738" id="intref0110">https://doi.org/10.5281/zenodo.6586738</ext-link></td></tr><tr><td>Flexbar 3.5.0</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/seqan/flexbar" id="intref0115">https://github.com/seqan/flexbar</ext-link></td><td>N/A</td></tr><tr><td><b>Snakemake</b></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>" id="intref0120">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>.<li> conda create -n eme_selex -c bioconda flexbar <b>snakemake</b> pip jupyterlab tqdm pandas seaborn.<li><bold><italic>Note:</italic></bold> Use a workflow manager such as <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io" id="intref0050">https://<b>snakemake</b>.readthedocs.io</ext-link>) to automate this step for all samples.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35731919">35731919</a> <b>(1)</b> The Prostate Cancer Androgen Receptor Cistrome in African American Men Associates with Upregulation of Lipid Metabolism and Immune Response<ul><li>Heatmap clustering, principal component analysis, and identification of subgroup-specific binding sites were performed using Mapmaker (<ext-link xlink:href="https://bitbucket.org/cfce/mapmaker" ext-link-type="uri" xlink:show="new">https://bitbucket.org/cfce/mapmaker</ext-link>), a ChIP-seq analysis pipeline implemented with <b>Snakemake</b> (<xref rid="bib19" ref-type="bibr"><u>19</u></xref>).<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35732665">35732665</a> <b>(1)</b> RNase III-CLASH of multi-drug resistant <italic>Staphylococcus aureus</italic> reveals a regulatory mRNA 3&#x2032;UTR required for intermediate vancomycin resistance<ul><li> The pipeline for CLASH data analysis, called Hyb-CRAC-R, has been implemented in <b>Snakemake</b> workflow management language and is available in the following GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/IgnatiusPang/Hyb-CRAC-R">https://github.com/IgnatiusPang/Hyb-CRAC-R</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35732669">35732669</a> <b>(1)</b> Multi-marker DNA metabarcoding detects suites of environmental gradients from an urban harbour<ul><li> MetaWorks is an automated <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR59"><u>59</u></xref></sup> bioinformatic pipeline that runs in a conda<sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup> environment.<li> [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [60](Anon. <italic>Conda</italic>. (2016).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35727987">35727987</a> <b>(1)</b> Metatranscriptomics captures dynamic shifts in mycorrhizal coordination in boreal forests<ul><li> Preprocessing and analysis of metatranscriptomic data were implemented in a reproducible <b>snakemake</b> workflow available on Bitbucket (<ext-link xlink:href="https://bitbucket.org/scilifelab-lts/n_street_1801/" ext-link-type="uri">https://bitbucket.org/scilifelab-lts/n_street_1801/</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35741082">35741082</a> <b>(1)</b> Population Scale Analysis of Centromeric Satellite DNA Reveals Highly Dynamic Evolutionary Patterns and Genomic Organization in Long-Tailed and Rhesus Macaques<ul><li> The StainedGlass pipeline workflow [<xref rid="B79-cells-11-01953" ref-type="bibr"><u>79</u></xref>] was applied and implemented in <b>Snakemake</b> [<xref rid="B80-cells-11-01953" ref-type="bibr"><u>80</u></xref>,<xref rid="B81-cells-11-01953" ref-type="bibr"><u>81</u></xref>].<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35020798">35020798</a>) [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [81](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35710760">35710760</a> <b>(1)</b> Identifying plant genes shaping microbiota composition in the barley rhizosphere<ul><li>Downstream data pre-processing and analysis for both transcript quantification and variant calling was carried out using <b>snakemake</b><sup><xref ref-type="bibr" rid="CR95"><u>95</u></xref></sup>.<li> [95](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35799885">35799885</a> <b>(1)</b> Narrow transmission bottlenecks and limited within-host viral diversity during a SARS-CoV-2 outbreak on a fishing boat<ul><li>All data processing from the raw unaligned sequencing files onward was handled by our <b>Snakemake</b> pipeline available on Github&#x2014;<ext-link xlink:href="https://github.com/jbloomlab/SARS-CoV-2_bottleneck" ext-link-type="uri">https://github.com/jbloomlab/SARS-CoV-2_bottleneck</ext-link> (<xref rid="R15" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li> [K&#xF6;ster and Rahmann 2012](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J.</given-names></string-name>, and <string-name><surname>Rahmann</surname><given-names>S.</given-names></string-name></person-group> (<year>2012</year>) &#x2018;<article-title>Snakemake&#x2014;A Scalable Bioinformatics Workflow Engine</article-title>&#x2019;, <source><italic toggle="yes">Bioinformatics (Oxford, England)</italic></source>, <volume>28</volume>: <fpage>2520</fpage>&#x2013;<lpage>2</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35769891">35769891</a> <b>(1)</b> Identifying SARS-CoV-2 regional introductions and transmission clusters in real time<ul><li>All code for this simulation is available as a modular and reproducible <b>Snakemake</b> pipeline at <ext-link xlink:href="https://github.com/jmcbroome/pandemic-simulator" ext-link-type="uri">https://github.com/jmcbroome/pandemic-simulator</ext-link>.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35703559">35703559</a> <b>(2)</b> MAGNETO: An Automated Workflow for Genome-Resolved Metagenomics<ul><li>MAGNETO is a <b>Snakemake</b> (<xref rid="B43" ref-type="bibr"><u>43</u></xref>) workflow connecting open-source bioinformatics software, all available from Bioconda and conda-forge.<li> <b>Snakemake</b> was chosen for its flexibility, its capacity to run both locally and on clusters, and its Conda management automating software installation.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35695776">35695776</a> <b>(1)</b> SPEAR: Systematic ProtEin AnnotatoR<ul><li>SPEAR is written in Python (version 3.10) and utilizes <b>Snakemake</b> (<xref rid="btac391-B13" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35698034">35698034</a> <b>(12)</b> GenErode: a bioinformatics pipeline to investigate genome erosion in endangered and extinct species<ul><li> Written in <b>Snakemake</b> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], this pipeline can be run on Linux systems such as high-performance computing (HPC) clusters.<li>For a species with a genome size of around 3 Gbp the GenErode pipeline requires a Linux system with at least 16 cores, each with a minimum of 6&#xA0;GB of RAM (a total of 96&#xA0;GB), and internet access so <b>Snakemake</b> can download the required containers on the fly.<li> file provided in the GitHub repository, which contains <b>Snakemake</b> (version 6.12.1 [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>]), Python3, and all required additional python modules.<li> All custom python scripts used by the pipeline are included in the GitHub repository and all additional software and dependencies are automatically downloaded as Docker images and run as Singularity containers by <b>Snakemake</b>, so there is no need for any additional installations by the user.<li> This pipeline has been developed using the slurm workload manager [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] with the official <b>Snakemake</b> profile for slurm (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>Snakemake</b>-Profiles/">https://github.com/<b>Snakemake</b>-Profiles/</ext-link>).<li> The computational resources for the different jobs that are submitted to the cluster via <b>Snakemake</b> are specified in a cluster configuration file (&#x2018;config/cluster.yaml&#x2019;) that needs to be edited according to the user&#x2019;s system.<li>Programming language: <b>Snakemake</b>, Python3.<li> GenErode depends on <b>Snakemake</b>, the Conda package manager and Singularity.<li> <b>Snakemake</b>) is available on GitHub (&#x2018;environment.yml&#x2019;).<li> All other dependencies will be automatically handled by <b>Snakemake</b> using Singularity when running the pipeline.<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [48](SLURM TEAM, et al. SLURM Workload Manager Version 16.05; Sbatch. Accessed August 2016; 2016.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35822110">35822110</a> <b>(1)</b> Evolutionary traits of Tick-borne encephalitis virus: Pervasive non-codingRNA structure conservation and molecular epidemiology<ul><li>The workflow management system <b>Snakemake</b> v5.32.1 (<xref rid="R41" ref-type="bibr"><u>K&#xF6;sterand Rahmann 2012</u></xref>) was used to build a pipeline for rapid and reproducibledeployment of Nextstrain (<xref rid="R29" ref-type="bibr">Hadfield et&#xA0;al.<li> [K&#xF6;sterand Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35742885">35742885</a> <b>(1)</b> Staphylococcal <italic toggle="yes">saoABC</italic> Operon Codes for a DNA-Binding Protein SaoC Implicated in the Response to Nutrient Deficit<ul><li> 4.0.3) scripts combined together with Salmon in one <b>Snakemake</b> (ver.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35674442">35674442</a> <b>(1)</b> Genomic and Temporal Trends in Canine ExPEC Reflect Those of Human ExPEC<ul><li> A modular analysis pipeline known as pipelord2, implemented with the <b>Snakemake</b> workflow management system, was used to perform primary bioinformatic analysis (<xref rid="B61" ref-type="bibr"><u>61</u></xref>).<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35670539">35670539</a> <b>(1)</b> Microbial Diversity and Interaction Specificity in Kombucha Tea Fermentations<ul><li> These draft genomes were then used by the anvi&#x2019;o (<xref rid="B41" ref-type="bibr"><u>41</u></xref>) <b>snakemake</b> workflows (<xref rid="B42" ref-type="bibr"><u>42</u></xref>, <xref rid="B43" ref-type="bibr"><u>43</u></xref>) to compute the pangenomes of <italic toggle="yes">K.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33349678">33349678</a>) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33323122">33323122</a>) [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35658001">35658001</a> <b>(2)</b> scAmpi&#x2014;A versatile pipeline for single-cell RNA-seq analysis from basics to clinics<ul><li> Utilizing the <b>Snakemake</b> workflow management system [<xref rid="pcbi.1010097.ref011" ref-type="bibr"><u>11</u></xref>], scAmpi is easy to use and offers a high degree of flexibility in the choice of methods, while it can be employed in a highly standardized and reproducible fashion.<li> Finally, because of the modular <b>Snakemake</b> framework, we foresee a continued extension and refinement of the pipeline and its open source code, also by the single-cell community.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35655063">35655063</a> <b>(1)</b> The microbiome of cryospheric ecosystems<ul><li> The entire workflow is setup in a reproducible <italic><b>Snakemake</b></italic> format<sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup>.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35653374">35653374</a> <b>(1)</b> Sensitive and accurate analysis of gene expression signatures enabled by oligonucleotide-labelled cDNA<ul><li>All NGS data analysis workflows were implemented using the <b>Snakemake</b> workflow manager v6.1.0 [<xref rid="cit0027" ref-type="bibr"><u>27</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35650195">35650195</a> <b>(1)</b> The androgen receptor is a therapeutic target in desmoplastic small round cell sarcoma<ul><li>ChIP-seq data were quality controlled and processed by pyflow-ChIP-seq<sup><xref ref-type="bibr" rid="CR71"><u>71</u></xref></sup>, a <b>snakemake</b><sup><xref ref-type="bibr" rid="CR72"><u>72</u></xref></sup> based ChIP-seq pipeline.<li> [71](Tang, M. pyflow-ChIPseq: a snakemake based ChIP-seq pipeline. <italic>Zenodo</italic> (2017).) [72](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35641504">35641504</a> <b>(1)</b> Structural variant-based pangenome construction has low sensitivity to variability of haplotype-resolved bovine assemblies<ul><li>Assembly, validation, and pangenome workflows have been implemented using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR104"><u>104</u></xref></sup>, and are available online (<ext-link ext-link-type="uri" xlink:href="https://github.com/AnimalGenomicsETH/bovine-assembly">https://github.com/AnimalGenomicsETH/bovine-assembly</ext-link>).<li> [104](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35711922">35711922</a> <b>(9)</b> Detection and Validation of Circular DNA Fragments Using Nanopore Sequencing<ul><li> The entire workflow is available at github.com/<b>snakemake</b>-workflows/cyrcular-calling.<li> The easiest way to obtain and use it is via the <b>snakemake</b> workflow catalog: <ext-link xlink:href="https://<b>snakemake</b>.github.io/<b>snakemake</b>-workflow-catalog?usage=<b>snakemake</b>-workflows/cyrcular-calling" ext-link-type="uri">https://<b>snakemake</b>.github.io/<b>snakemake</b>-workflow-catalog?usage=<b>snakemake</b>-workflows/cyrcular-calling</ext-link>.<li>We implemented a <b>Snakemake</b> (<xref rid="B13" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) workflow for detecting circular DNA fragments from nanopore sequencing data using the protocol outlined in <xref rid="s2-1" ref-type="sec"><u>Section 2.1</u></xref>; see <xref rid="F3" ref-type="fig"><u>Figure 3</u></xref>.<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35637409">35637409</a> <b>(1)</b> Polycomb repressive complex 2 shields na&#xEF;ve human pluripotent cells from trophectoderm differentiation<ul><li>MINUTE-ChIP multiplexed FASTQ files were processed using minute<sup><xref ref-type="bibr" rid="CR73"><u>73</u></xref></sup>, a data processing pipeline implemented in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR74"><u>74</u></xref></sup>.<li> [73](Navarro, C., Martin, M. &amp; Els&#xE4;sser, S. minute: A MINUTE-ChIP data analysis workflow. <italic>BioRxiv</italic>10.1101/2022.03.14.484318 (2022).) [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35664802">35664802</a> <b>(1)</b> ePeak: from replicated chromatin profiling data to epigenomic dynamics<ul><li> (<bold>B</bold>) ePeak <b>Snakemake</b> rule graph illustrating the input/output dependencies between steps.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35561686">35561686</a> <b>(5)</b> The genomic origins of the world&#x2019;s first farmers<ul><li> R Foundation for Statistical Computing, Vienna, Austria</td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0215">https://www.R-project.org/</ext-link></td></tr><tr><td>SAMtools - version 1.3</td><td>(<xref rid="bib144" ref-type="bibr"><u>Li et&#xA0;al., 2009</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="http://www.htslib.org/" id="intref0220">http://www.htslib.org/</ext-link></td></tr><tr><td>Samtools 1.9</td><td>(<xref rid="bib69" ref-type="bibr"><u>Danecek et&#xA0;al., 2021</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/samtools/samtools" id="intref0225">https://github.com/samtools/samtools</ext-link></td></tr><tr><td>seqtk - version 1.2</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/lh3/seqtk" id="intref0230">https://github.com/lh3/seqtk</ext-link></td></tr><tr><td>SHAPEIT4 v1.2</td><td>(<xref rid="bib73" ref-type="bibr"><u>Delaneau et&#xA0;al., 2019</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://odelaneau.github.io/shapeit4/" id="intref0235">https://odelaneau.github.io/shapeit4/</ext-link></td></tr><tr><td><b>Snakemake</b> - version 4.0.<li> R Foundation for Statistical Computing, Vienna, Austria</td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0215">https://www.R-project.org/</ext-link></td></tr><tr><td>SAMtools - version 1.3</td><td>(<xref rid="bib144" ref-type="bibr"><u>Li et&#xA0;al., 2009</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="http://www.htslib.org/" id="intref0220">http://www.htslib.org/</ext-link></td></tr><tr><td>Samtools 1.9</td><td>(<xref rid="bib69" ref-type="bibr"><u>Danecek et&#xA0;al., 2021</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/samtools/samtools" id="intref0225">https://github.com/samtools/samtools</ext-link></td></tr><tr><td>seqtk - version 1.2</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/lh3/seqtk" id="intref0230">https://github.com/lh3/seqtk</ext-link></td></tr><tr><td>SHAPEIT4 v1.2</td><td>(<xref rid="bib73" ref-type="bibr"><u>Delaneau et&#xA0;al., 2019</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://odelaneau.github.io/shapeit4/" id="intref0235">https://odelaneau.github.io/shapeit4/</ext-link></td></tr><tr><td><b>Snakemake</b> - version 4.0</td><td>(<xref rid="bib128" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="intref0240">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li><italic><b>Snakemake</b></italic> - version 4.0 (<xref rid="bib128" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li>We used commit 6df90e7 of the <italic><b>snakemake</b></italic> pipeline <italic>ATLAS-Pipeline</italic>, which is available at <ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/wegmannlab/atlas-pipeline/src/master/" id="intref0295">bitbucket.org/wegmannlab/atlas-pipeline</ext-link> and summarised below.<li> [Li et&#xA0;al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [Danecek et&#xA0;al., 2021](          <person-group person-group-type="author">            <name>              <surname>Danecek</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Bonfield</surname>              <given-names>J.K.</given-names>            </name>            <name>              <surname>Liddle</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Marshall</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Ohan</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Pollard</surname>              <given-names>M.O.</given-names>            </name>            <name>              <surname>Whitwham</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Keane</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>McCarthy</surname>              <given-names>S.A.</given-names>            </name>            <name>              <surname>Davies</surname>              <given-names>R.M.</given-names>            </name>            <etal/>          </person-group>          <article-title>Twelve years of SAMtools and BCFtools</article-title>          <source>GigaScience</source>          <volume>10</volume>          <year>2021</year>        ) [Delaneau et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31780650">31780650</a>) [Li et&#xA0;al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [Danecek et&#xA0;al., 2021](          <person-group person-group-type="author">            <name>              <surname>Danecek</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Bonfield</surname>              <given-names>J.K.</given-names>            </name>            <name>              <surname>Liddle</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Marshall</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Ohan</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Pollard</surname>              <given-names>M.O.</given-names>            </name>            <name>              <surname>Whitwham</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Keane</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>McCarthy</surname>              <given-names>S.A.</given-names>            </name>            <name>              <surname>Davies</surname>              <given-names>R.M.</given-names>            </name>            <etal/>          </person-group>          <article-title>Twelve years of SAMtools and BCFtools</article-title>          <source>GigaScience</source>          <volume>10</volume>          <year>2021</year>        ) [Delaneau et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31780650">31780650</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35639618">35639618</a> <b>(1)</b> PHACT: Phylogeny-Aware Computing of Tolerance for Missense Mutations<ul><li> This study effectively used a workflow management system tool, referred to as <b>Snakemake</b> Field (<xref rid="msac114-B26" ref-type="bibr"><u>Koster and Rahmann 2012</u></xref>), which offers high-performance computing cluster-level scalability, to perform reproducible and scalable data analysis.<li> [Koster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35692633">35692633</a> <b>(4)</b> Genetic polyploid phasing from low-depth progeny samples<ul><li><hr/></td></tr><tr><td>Whatshap polyphase genetic (used version)</td><td>This paper</td><td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.6519173" id="intref0010">https://doi.org/10.5281/zenodo.6519173</ext-link></td></tr><tr><td><b>Snakemake</b>.<li><hr/></td></tr><tr><td>Whatshap polyphase genetic (used version)</td><td>This paper</td><td><ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.6519173" id="intref0010">https://doi.org/10.5281/zenodo.6519173</ext-link></td></tr><tr><td><b>Snakemake</b></td><td><xref rid="bib8" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io" id="intref0015">https://<b>snakemake</b>.github.io</ext-link>.<li> Experiments have been organized as pipeline via <b>Snakemake</b> (<xref rid="bib8" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>).<li> [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35744611">35744611</a> <b>(1)</b> PhyloFunDB: A Pipeline to Create and Update Functional Gene Taxonomic Databases<ul><li> The pipeline is implemented in <b>Snakemake</b> [<xref rid="B10-microorganisms-10-01093" ref-type="bibr"><u>10</u></xref>].<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35707723">35707723</a> <b>(3)</b> The genes controlling normal function of citrate and spermine secretion are lost in aggressive prostate cancer and prostate model systems<ul><li>Since GAPGOM only allows us to do annotation prediction one gene at a time, GAPGOM was used in conjunction with <b>Snakemake</b> [<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/project_info/citations.html" id="intref0210">https://<b>snakemake</b>.readthedocs.io/en/stable/project_info/citations.html</ext-link>] to create a software pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35584008">35584008</a> <b>(2)</b> RegulonDB 11.0: Comprehensive high-throughput datasets on transcriptional regulation in <italic toggle="yes">Escherichia coli</italic> K-12<ul><li> A library based on the <b>snakemake</b> workflow management system, which was used to design a generalizable workflow to perform reproducible ChIP-seq analyses [<xref rid="R1" ref-type="bibr"><u>1</u></xref>].<li> The ChIP-seq workflows were implemented in <b>snakemake</b> 6.10.0 [<xref rid="R29" ref-type="bibr"><u>29</u></xref>].<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30786165">30786165</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35585235">35585235</a> <b>(1)</b> MCM complexes are barriers that restrict cohesin-mediated loop extrusion<ul><li>For downstream analysis, we used quaich (<ext-link ext-link-type="uri" xlink:href="https://github.com/open2c/quaich">https://github.com/open2c/quaich</ext-link>), a new <b>snakemake</b> pipeline for Hi-C postprocessing.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35579969">35579969</a> <b>(1)</b> Arginine-mediated gut microbiome remodeling promotes host pulmonary immune defense against nontuberculous mycobacterial infection<ul><li> We used the <b>Snakemake</b><sup><xref rid="cit0067" ref-type="bibr"><u>67</u></xref></sup> workflow to process the raw reads; FastQC (version 0.11.9) was used to preprocess and control the quality of raw reads;<sup><xref rid="cit0068" ref-type="bibr"><u>68</u></xref></sup> Cutadapt-PE (version 2.1) was used to trim the adapter sequence from paired-end reads;<sup><xref rid="cit0069" ref-type="bibr"><u>69</u></xref></sup> Trimmed reads in FASTQ format were quantified at the transcript level using Salmon<sup><xref rid="cit0070" ref-type="bibr"><u>70</u></xref></sup> against a reference mouse genome (GRCm38 March 2020).<li> [67](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [68](<person-group person-group-type="author"><string-name><surname>Andrews</surname><given-names>SF</given-names></string-name></person-group>. <article-title>A quality control tool for high throughput sequence data. In. Fastqc. A quality control tool for high throughput sequence data</article-title>; <year>2010</year>. <ext-link xlink:href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc" ext-link-type="uri">http://www.bioinformatics.babraham.ac.uk/projects/fastqc</ext-link> [<date-in-citation>accessed 2021 <month>May</month><day>3</day>].</date-in-citation>) [69](<person-group person-group-type="author"><string-name><surname>Martin</surname><given-names>M</given-names></string-name></person-group>. <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>. <source>EMBnet Journal</source>. <year>2011</year>;<volume>17</volume>(<issue>1</issue>):<fpage>3</fpage>. doi:<pub-id pub-id-type="doi">10.14806/ej.17.1.200</pub-id>.) [70](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28263959">28263959</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35628390">35628390</a> <b>(1)</b> Meta-Analysis of Two Human RNA-seq Datasets to Determine Periodontitis Diagnostic Biomarkers and Drug Target Candidates<ul><li> Periodontitis and healthy RNA samples were analyzed using the <b>Snakemake</b>-based ARMOR workflow within a dedicated Conda environment, as described by Orjuela [<xref rid="B39-ijms-23-05580" ref-type="bibr"><u>39</u></xref>].<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31088905">31088905</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304283">36304283</a> <b>(2)</b> Scalable Microbial Strain Inference in Metagenomic Data Using StrainFacts<ul><li> For reproducibility, analyses were performed using <b>Snakemake</b> (<xref rid="B25" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) and with a Singularity container (<xref rid="B15" ref-type="bibr"><u>Kurtzer et al., 2017</u></xref>) that can be obtained at <ext-link xlink:href="https://hub.docker.com/repository/docker/bsmith89/compbio" ext-link-type="uri">https://hub.docker.com/repository/docker/bsmith89/compbio</ext-link>.<li>Runtimes were determined using the <b>Snakemake</b> <italic>benchmark:</italic> directive, and memory requirements using the GNU time utility, version 1.8 with all benchmarks run on the Wynton compute cluster at the University of California, San Francisco.<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [Kurtzer et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35663029">35663029</a> <b>(1)</b> Sequence-based pangenomic core detection<ul><li>All experiments throughout this paper are documented as a <b>Snakemake</b> workflow (<xref rid="bib29" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>) at&#xA0;Corer&#x2019;s code repository (<ext-link ext-link-type="uri" xlink:href="https://gitlab.ub.uni-bielefeld.de/gi/corer" id="intref0095">https://gitlab.ub.uni-bielefeld.de/gi/corer</ext-link>).<li> [M&#xF6;lder et&#xA0;al., 2021](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A.</given-names>            </name>            <etal/>          </person-group>          <article-title>Sustainable data analysis with Snakemake [version 1; peer review: 1 approved, 1 approved with reservations]</article-title>          <source>F1000Research</source>          <volume>10</volume>          <year>2021</year>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.1</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35316214">35316214</a> <b>(1)</b> ZFP36L2 suppresses mTORc1 through a P53-dependent pathway to prevent peripartum cardiomyopathy in mice<ul><li> All of the sleuth-CN analysis and plotting was performed in R, version 3.5.1, and the pipeline for analyzing the data from start to finish was done using bioconda and <b>snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35605437">35605437</a> <b>(1)</b> Advancing genomic epidemiology by addressing the bioinformatics bottleneck: Challenges, design principles, and a Swiss example<ul><li> Another complementary approach to achieving analysis modularity would be to use scientific workflow systems, such as <b>Snakemake</b>&#xA0;(<xref rid="b17" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>) or Nextflow&#xA0;(<xref rid="b8" ref-type="bibr"><u>Di&#xA0;Tommaso et al., 2017</u></xref>).<li> [M&#xF6;lder et al., 2021](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F.</given-names>            </name>            <etal/>          </person-group>          <article-title>Sustainable data analysis with snakemake [version 2; peer review: 2 approved]</article-title>          <source>F1000Research</source>          <volume>10</volume>          <issue>33</issue>          <year>2021</year>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) [Di&#xA0;Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35603231">35603231</a> <b>(1)</b> Pollock: fishing for cell states<ul><li> It provides a command line interface (CLI), making it compatible with command line dependent pipelines and workflow management systems, such as Galaxy (<xref rid="vbac028-B2" ref-type="bibr">Afgan <italic toggle="yes">et al.</italic>, 2018</xref>), CWL (<xref rid="vbac028-B6" ref-type="bibr">Crusoe <italic toggle="yes">et al.</italic>, 2021</xref>) and <b>Snakemake</b> (<xref rid="vbac028-B14" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35560159">35560159</a> <b>(1)</b> CrowdGO: Machine learning and semantic similarity guided consensus Gene Ontology annotation<ul><li> <b>Snakemake</b> [<xref rid="pcbi.1010075.ref025" ref-type="bibr"><u>25</u></xref>] workflows are provided to run each of the individual predictors, and to run CrowdGO to produce consensus annotations.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35551337">35551337</a> <b>(1)</b> Exodus: sequencing-based pipeline for quantification of pooled variants<ul><li>The Exodus algorithm is a multistep process (<xref rid="btac319-F1" ref-type="fig"><u>Fig.&#xA0;1A</u></xref>) implemented in Python using a <b>Snakemake</b> workflow (<xref rid="btac319-B7" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>); thus it runs in parallel on the available resources of the machine at hand with no user intervention required.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35551248">35551248</a> <b>(3)</b> Allorecognition genes drive reproductive isolation in <italic>Podospora anserina</italic><ul><li> Most bioinformatics analyses were performed using the workflow manager <b>Snakemake</b> v.5.4.4 (ref.<li> A <b>Snakemake</b> pipeline (SNPpop.smk) is available at 10.5281/zenodo.6323682.<li> A <b>Snakemake</b> pipeline (DiversityStats.smk) is available at 10.5281/zenodo.6323682.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35551306">35551306</a> <b>(1)</b> Genome-wide analysis of cis-regulatory changes underlying metabolic adaptation of cavefish<ul><li> Reads were processed for calling variants with HaplotypeCaller using a GATK best practices pipeline implemented with <b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35546658">35546658</a> <b>(1)</b> B-assembler: a circular bacterial genome assembler<ul><li>Other requirements:&#xA0;python 3.0 or higher, and <b>snakemake</b> (see GitHub page);.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35545626">35545626</a> <b>(3)</b> The effects of fish meal substitution by clam meal on the growth and health of Florida pompano (<italic>Trachinotus carolinus</italic>)<ul><li>Demultiplexed raw sequences were processed through a <b>snakemake</b> protocol<sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref>,<xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup> that uses fastx and trim galore to trim primers and perform quality control<sup><xref ref-type="bibr" rid="CR34"><u>34</u></xref>,<xref ref-type="bibr" rid="CR35"><u>35</u></xref></sup>.<li> Another <b>snakemake</b> protocol based around QIIME2 was used to join and denoise sequences with Dada2<sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref>,<xref ref-type="bibr" rid="CR36"><u>36</u></xref>,<xref ref-type="bibr" rid="CR37"><u>37</u></xref></sup>.<li> The <b>snakemake</b>, conda environment, R scripts, and R session files can be found on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/aqua-omics/clam_meal_study">https://github.com/aqua-omics/clam_meal_study</ext-link>).<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33105476">33105476</a>) [34](Gordon, A. &amp; Hannon, G.J. Fastx-toolkit. <italic>FASTQ/A short-reads pre-processing tools</italic>. <ext-link ext-link-type="uri" xlink:href="http://hannonlab.cshl.edu/fastx_toolkit">http://hannonlab.cshl.edu/fastx_toolkit</ext-link> (2010).) [35](Krueger, F. <italic>Trim Galore</italic>. Babraham Bioinformatics. <ext-link ext-link-type="uri" xlink:href="https://github.com/FelixKrueger/TrimGalore">https://github.com/FelixKrueger/TrimGalore</ext-link> (2020).) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33105476">33105476</a>) [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27214047">27214047</a>) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31341288">31341288</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36304282">36304282</a> <b>(2)</b> MIntO: A Modular and Scalable Pipeline For Microbiome Metagenomic and Metatranscriptomic Data Integration<ul><li> It is encapsulated within a user-friendly framework using <b>Snakemake</b> (<xref rid="B39" ref-type="bibr"><u>M&#xF6;lder, 2021</u></xref>) to facilitate the scalability of the pipeline by optimizing the number of parallel processes from a single-core workstation to compute clusters.<li>MIntO implementation and automation are achieved by <b>Snakemake</b> (<xref rid="B39" ref-type="bibr"><u>M&#xF6;lder, 2021</u></xref>), a user-friendly framework that facilitates the scalability of the pipeline by optimizing the number of parallel processes from a single-core workstation to compute clusters.<li> [M&#xF6;lder, 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [M&#xF6;lder, 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35349481">35349481</a> <b>(1)</b> Specific human endogenous retroviruses predict metastatic potential in uveal melanoma<ul><li>The data processing pipeline was implemented on a high-performance computing cluster using <b>Snakemake</b> (<xref rid="B79" ref-type="bibr"><u>79</u></xref>) with the bioconda package manager (<xref rid="B80" ref-type="bibr"><u>80</u></xref>).<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35512829">35512829</a> <b>(2)</b> miR-486 is essential for muscle function and suppresses a dystrophic transcriptome<ul><li> The FASTQ files were uploaded to the UAB high-performance computer cluster for bioinformatics analysis with the following custom pipeline built in the <b>Snakemake</b> workflow system (v5.2.2) (<xref rid="bib27" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>): first, quality and control of the reads were assessed using FastQC, and trimming of the Illumina adapters and bases with quality scores of less than 20 were performed with Trim Galore! (v0.4.5).<li>The entire processing pipeline was performed by using the program <b>Snakemake</b> to procedurally perform all processing steps in order.<li> [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35632715">35632715</a> <b>(1)</b> The International Virus Bioinformatics Meeting 2022<ul><li> (4) incorporating modern workflow management tool (<b>snakemake</b>) [<xref rid="B43-viruses-14-00973" ref-type="bibr"><u>43</u></xref>] for improved scalability in high performance computing clusters and also overall usability.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35603154">35603154</a> <b>(1)</b> Impact of an Immune Modulator Mycobacterium-w on Adaptive Natural Killer Cells and Protection Against COVID-19<ul><li> This pipeline from nanopore tech uses <b>snakemake</b>, minimap2, salmon, edgeR, DEXSeq, and stageR to automate DGE workflows on long read data.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35504902">35504902</a> <b>(1)</b> Alternative splicing modulation by G-quadruplexes<ul><li>The differential gene expression analysis was performed using a <b>Snakemake</b> pipeline<sup><xref ref-type="bibr" rid="CR107"><u>107</u></xref></sup> that was adapted from a publicly available repository<sup><xref ref-type="bibr" rid="CR108"><u>108</u></xref></sup>.<li> [107](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [108](K&#xF6;ster, J., Forster, J., Schmeier, S., Salazar, V. &amp; matrs. snakemake-workflows/rna-seq-star-deseq2: Version 1.2.0. 10.5281/zenodo.5245549 (2021).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35501696">35501696</a> <b>(1)</b> GEMmaker: process massive RNA-seq datasets on heterogeneous computational infrastructure<ul><li> Popular examples include Galaxy [<xref ref-type="bibr" rid="CR1"><u>1</u></xref>], Kepler [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>], Nextflow [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>].<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [21](          <person-group person-group-type="author">            <name>              <surname>Lud&#xE4;scher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Altintas</surname>              <given-names>I</given-names>            </name>            <name>              <surname>Berkley</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Higgins</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Jaeger</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Jones</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Tao</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Zhao</surname>              <given-names>Y</given-names>            </name>          </person-group>          <article-title>Scientific workflow management and the Kepler system: research articles</article-title>          <source>Concurr Comput Pract Exp</source>          <year>2006</year>          <volume>18</volume>          <fpage>1039</fpage>          <lpage>1065</lpage>          <pub-id pub-id-type="doi">10.1002/cpe.v18:10</pub-id>        ) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35489067">35489067</a> <b>(1)</b> BusyBee Web: towards comprehensive and differential composition-based metagenomic binning<ul><li> In combination with workflow managing tools such as Nextflow or <b>Snakemake</b>, the API increases experiment throughput and reproducibility of results (<xref rid="B39" ref-type="bibr"><u>39</u></xref>,<xref rid="B40" ref-type="bibr"><u>40</u></xref>).<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35572712">35572712</a> <b>(1)</b> Association Between Human Gut Microbiome and N-Glycan Composition of Total Plasma Proteome<ul><li> The pipeline was constructed using <b>Snakemake</b> (<xref rid="B25" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35570946">35570946</a> <b>(1)</b> Impact of SARS-CoV-2 Mutations on PCR Assay Sequence Alignment<ul><li> The PSET workflow itself was implemented using Biopython and <b>Snakemake</b> (<xref rid="B44" ref-type="bibr"><u>44</u></xref>, <xref rid="B45" ref-type="bibr"><u>45</u></xref>).<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19304878">19304878</a>) [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35477344">35477344</a> <b>(3)</b> Findzx: an automated pipeline for detecting and visualising sex chromosomes using whole-genome sequencing data<ul><li>Here we present findZX, a <b>Snakemake</b>-based [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] pipeline that identifies sex chromosomes, or more specifically the non-recombining part of the sex chromosomes, using whole-genome sequencing (WGS) paired-end data from a flexible number of male and female samples.<li>We present a novel <b>Snakemake</b> pipeline and demonstrate its effectiveness in uncovering sex-linked genomic regions using WGS data from a wide range of species and sex chromosome systems.<li>Programming language: R, bash, Python/<b>Snakemake</b>.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35477154">35477154</a> <b>(1)</b> <italic>ABO</italic> genotype alters the gut microbiota by regulating GalNAc levels in pigs<ul><li> The corresponding pipeline was constructed using <b>Snakemake</b> (v.7.0.1)<sup><xref ref-type="bibr" rid="CR103"><u>103</u></xref></sup>.<li> [103](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35467402">35467402</a> <b>(1)</b> Within-Host Genetic Variation in <named-content content-type="genus-species">Neisseria gonorrhoeae</named-content> over the Course of Infection<ul><li> <b>Snakemake</b> v5.31.1 was used for workflow management (<xref rid="B30" ref-type="bibr"><u>30</u></xref>).<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35452306">35452306</a> <b>(1)</b> Estimating bonobo (<italic toggle="yes">Pan</italic><italic toggle="yes">paniscus</italic>) and chimpanzee (<italic toggle="yes">Pan</italic><italic toggle="yes">troglodytes</italic>) evolutionary history from nucleotide site patterns<ul><li> <xref rid="r71" ref-type="bibr"><u>71</u></xref> and contained in an automated <b>Snakemake</b> (<xref rid="r73" ref-type="bibr"><u>73</u></xref>) available on GitHub (<ext-link xlink:href="https://github.com/thw17/Pan_reassembly" ext-link-type="uri">https://github.com/thw17/Pan_reassembly</ext-link>).<li> [71](<person-group person-group-type="author"><string-name><given-names>C. M.</given-names><surname>Brand</surname></string-name>, <string-name><given-names>F. J.</given-names><surname>White</surname></string-name>, <string-name><given-names>N.</given-names><surname>Ting</surname></string-name>, <string-name><given-names>T. H.</given-names><surname>Webster</surname></string-name></person-group>, Soft sweeps predominate recent positive selection in bonobos (Pan paniscus) and chimpanzees (Pan troglodytes). bioRxiv [Preprint] (<year>2021</year>). <pub-id pub-id-type="doi">10.1101/2020.12.14.422788</pub-id> (Accessed 8 January 2021).) [73](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35446881">35446881</a> <b>(1)</b> Development and evaluation of ActSeq: A targeted next-generation sequencing panel for clinical oncology use<ul><li> Raw data from all cell line samples are available from SRA (BioProject PRJNA803819) and the bioinformatics pipeline (<b>snakemake</b> v6.14.0) is available on GitHub (<ext-link xlink:href="https://github.com/harpernd/actseq" ext-link-type="uri">https://github.com/harpernd/actseq</ext-link>, release v1.0.0).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35449230">35449230</a> <b>(1)</b> Genome-wide identification, classification and expression analysis of the JmjC domain-containing histone demethylase gene family in <italic>Jatropha curcas</italic> L.<ul><li> We have integrated our pipeline using <b>snakemake</b><sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35444261">35444261</a> <b>(1)</b> Synthetic bacterial community derived from a desert rhizosphere confers salt stress resilience to tomato in the presence of a soil microbiome<ul><li> All processing steps were implemented in a <b>SnakeMake</b> workflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35439933">35439933</a> <b>(3)</b> MALVIRUS: an integrated application for viral variant analysis<ul><li> However, we believe that thanks to our pipelines, implemented as <b>Snakemake</b> workflows [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], any user can easily build a variant catalog starting from his own private set of assemblies.<li> All pipelines described in this section have been implemented as <b>Snakemake</b> workflows [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>], thus easy to use and fully reproducible.<li>All the experiments can be reproduced (for the second experimental&#xA0;evaluation, after non-freely redistributable data are downloaded from GISAID) using the <b>Snakemake</b> workflows available at&#xA0;<ext-link ext-link-type="uri" xlink:href="https://github.com/AlgoLab/MALVIRUS-repro/">https://github.com/AlgoLab/MALVIRUS-repro/</ext-link>.<li> [19](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Holtgrewe</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Sustainable data analysis with Snakemake</article-title>          <source>F1000Research</source>          <year>2021</year>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) [19](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>C</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Holtgrewe</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Sustainable data analysis with Snakemake</article-title>          <source>F1000Research</source>          <year>2021</year>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35440545">35440545</a> <b>(1)</b> Distinct and additive effects of calorie restriction and rapamycin in aging skeletal muscle<ul><li> adapters and poly(A)/poly(T) stretches, indexing reference transcriptomes, mapping the RNA-Seq reads to transcripts and counting reads assigned to individual transcripts were performed with a <b>Snakemake</b> framework<sup><xref ref-type="bibr" rid="CR72"><u>72</u></xref></sup>.<li> [72](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35430813">35430813</a> <b>(2)</b> DEIMoS: An Open-Source Tool for Processing High-DimensionalMass Spectrometry Data<ul><li>While DEIMoS functionality is implementedas a Python API, a typical workflow has been implemented using the<b>Snakemake</b><sup><xref ref-type="bibr" rid="ref62"><u>62</u></xref></sup> workflow management system<sup><xref ref-type="bibr" rid="ref62"><u>62</u></xref></sup> and made accessible via command line interface(CLI).<li> Moreover, <b>Snakemake</b> can automatically handle scalingto HPC and cloud resources, enabling the high-throughput processingof numerous samples.<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35429300">35429300</a> <b>(3)</b> Validating the knowledge bank approach for personalized prediction of survival in acute myeloid leukemia: a reproducibility study<ul><li> For future authors, workflow managers such as Nextflow (<ext-link ext-link-type="uri" xlink:href="https://www.nextflow.io/">https://www.nextflow.io/</ext-link>) and <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/">https://<b>snakemake</b>.readthedocs.io/</ext-link>) could present a solution, as they provide an abstraction to separate analysis pipelines from computing environments, and hence allow seamless execution of the pipelines across different platforms (Di Tommaso et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35455749">35455749</a> <b>(2)</b> Deciphering of Adult Glioma Vulnerabilities through Expression Pattern Analysis of GABA, Glutamate and Calcium Neurotransmitter Genes<ul><li> <b>Snakemake</b> Pipepeline Creation.<li>The study pipeline was built using <b>snakemake</b> [<xref rid="B28-jpm-12-00633" ref-type="bibr"><u>28</u></xref>] workflow manager v5.32.0.<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35456493">35456493</a> <b>(1)</b> Development and Optimization of a Silica Column-Based Extraction Protocol for Ancient DNA<ul><li> The GenErode pipeline [<xref rid="B41-genes-13-00687" ref-type="bibr"><u>41</u></xref>] written in <b>Snakemake</b> version 4.5.0 [<xref rid="B42-genes-13-00687" ref-type="bibr"><u>42</u></xref>] was used to subsequently process the sequencing files.<li> [41](          <person-group person-group-type="author"><name><surname>Kutschera</surname><given-names>V.E.</given-names></name><name><surname>Kierczak</surname><given-names>M.</given-names></name><name><surname>van der Valk</surname><given-names>T.</given-names></name><name><surname>von Seth</surname><given-names>J.</given-names></name><name><surname>Dussex</surname><given-names>N.</given-names></name><name><surname>Lord</surname><given-names>E.</given-names></name><name><surname>Dehasque</surname><given-names>M.</given-names></name><name><surname>Stanton</surname><given-names>D.W.G.</given-names></name><name><surname>Khoonsari</surname><given-names>P.E.</given-names></name><name><surname>Nystedt</surname><given-names>B.</given-names></name><etal/></person-group>          <article-title>GenErode: A Bioinformatics Pipeline to Investigate Genome Erosion in Endangered and Extinct Species</article-title>          <source>bioRxiv</source>          <year>2022</year>          <pub-id pub-id-type="doi">10.1101/2022.03.04.482637</pub-id>        ) [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35456492">35456492</a> <b>(1)</b> Robust Mutation Profiling of SARS-CoV-2 Variants from Multiple Raw Illumina Sequencing Data with Cloud Workflow<ul><li> Even though, with workflow systems, an understanding of the syntax was still required to set up the environment first before running the entire workflow, for example, COVseq [<xref rid="B53-genes-13-00686" ref-type="bibr"><u>53</u></xref>] and V-pipe [<xref rid="B54-genes-13-00686" ref-type="bibr"><u>54</u></xref>] that required to install the workflow engine of <b>Snakemake</b> [<xref rid="B55-genes-13-00686" ref-type="bibr"><u>55</u></xref>], resulting in an additional time that may delay mutation profiling process.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34162869">34162869</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33471068">33471068</a>) [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35418676">35418676</a> <b>(1)</b> Nonlinear control of transcription through enhancer&#x2013;promoter interactions<ul><li> Nanopore sequencing analysis has been implemented using <b>Snakemake</b> workflow (v.3.13.3).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35437474">35437474</a> <b>(1)</b> PipeCoV: a pipeline for SARS-CoV-2 genome assembly, annotation and variant identification<ul><li> V-pipe (<xref rid="ref-22" ref-type="bibr"><u>Posada-C&#xE9;spedes et al., 2021</u></xref>) uses the workflow management system <b>Snakemake</b> to determine the order in which the steps of the specified pipeline are executed with further verification output files.<li> [Posada-C&#xE9;spedes et al., 2021](          <person-group person-group-type="author"><name><surname>Posada-C&#xE9;spedes</surname><given-names>S</given-names></name><name><surname>Seifert</surname><given-names>D</given-names></name><name><surname>Topolsky</surname><given-names>I</given-names></name><name><surname>Jablonski</surname><given-names>KP</given-names></name><name><surname>Metzner</surname><given-names>KJ</given-names></name><name><surname>Beerenwinkel</surname><given-names>N</given-names></name><name><surname>Xu</surname><given-names>J</given-names></name></person-group>          <article-title>V-pipe: a computational pipeline for assessing viral genetic diversity from high-throughput data</article-title>          <source>Bioinformatics</source>          <year iso-8601-date="2021">2021</year>          <volume>37</volume>          <issue>12</issue>          <fpage>1673</fpage>          <lpage>1680</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btab015</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35418483">35418483</a> <b>(2)</b> Cancer stem-like cells evade CD8<sup>+</sup>CD103<sup>+</sup> tumor-resident memory T (T<sub>RM</sub>) lymphocytes by initiating an epithelial-to-mesenchymal transition program in a human lung tumor model<ul><li>Quality control was performed on raw FastQ files with FastQC (V.0.11.8) and quality reports were gathered with MultiQC (V.1.10.1).<xref rid="R23" ref-type="bibr"><u>23</u></xref> Abundance estimation was performed with Salmon (V.0.9.0) using GENCODE (GRCh38, V.34) annotation.<xref rid="R24" ref-type="bibr"><u>24</u></xref> This pipeline was powered by both <b>Snakemake</b> and <b>Snakemake</b>Wrappers.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27312411">27312411</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30357393">30357393</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35463472">35463472</a> <b>(16)</b> A computational pipeline to visualize DNA-protein binding states using dSMF data<ul><li> The pipeline is built using <b>Snakemake</b> (<xref rid="bib4" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> <b>Snakemake</b> is a scalable bioinformatics workflow engine that enables execution of reproducible pipelines easily in a high-performance cluster.<li><hr/></td></tr><tr><td><b>Snakemake</b>.<li><hr/></td></tr><tr><td><b>Snakemake</b></td><td>(<xref rid="bib4" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>)</td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="intref0100">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li><b>snakemake</b>--snakefile cooperative_binding_analysis.smk.<li> in <b>snakemake</b>s/trim_galore_pe.smk) for adapter trimming and &#x201C;Bismark&#x201D; (see rule &#x201C;Bismark_align_pe&#x201D; in <b>snakemake</b>s/Bismark_align_pe.smk) for bisulfite sequence alignment.<li><b>snakemake</b> -np --snakefile cooperative_binding_analysis.smk.<li>The pipeline failed for my ROI named as &#x201C;roi_peak_x" in my single site bed file when I ran the following <b>snakemake</b> command (step: &#x201C;<xref rid="sec3.6" ref-type="sec"><u>assigning binding states at a pair of binding sites</u></xref>&#x201D;):<boxed-text id="dtbox3"><p id="p0310"><b>snakemake</b> --snakefile cooperative_binding_analysis.smk.<li>It is noteworthy that <b>Snakemake</b> builds workflow based on parsing the output filename.<li> The example peak_id written here as &#x201C;roi_peak_x" overlaps with one of the string literals (&#x201C;roi&#x201D;) used by <b>Snakemake</b> to retrieve wildcards.<li> Please refrain from using names and other parameters that overlaps with <b>Snakemake</b> wildcards.<li><b>Snakemake</b> facilitates visualization of pipeline up to your desired output file.<li><b>snakemake</b> -np --dag --snakefile cooperative_binding_analysis.smk.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35404072">35404072</a> <b>(1)</b> Microbial Activities and Selection from Surface Ocean to Subseafloor on the Namibian Continental Shelf<ul><li> The Anvi&#x2019;o <b>snakemake</b> workflow (<xref rid="B131" ref-type="bibr"><u>131</u></xref>, <xref rid="B132" ref-type="bibr"><u>132</u></xref>) was used for downstream binning analysis, using coassembly mode with default settings (Table S2).<li> [131](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [132](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35404110">35404110</a> <b>(1)</b> <italic toggle="yes">mucG, mucH,</italic> and <italic toggle="yes">mucI</italic> Modulate Production of Mutanocyclin and Reutericyclins in <named-content content-type="genus-species">Streptococcus mutans</named-content> B04Sm5<ul><li>Pangenome analysis was performed using the anvi'o pangenomics <b>Snakemake</b> workflow on 244 <named-content content-type="genus-species">S.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35496474">35496474</a> <b>(7)</b> A bioinformatic pipeline for simulating viral integration data<ul><li>Our simulation pipeline was developed using <b>snakemake</b> 5.27 <xref rid="bib0002" ref-type="bibr"><u>[2]</u></xref>, Python 3.7 <xref rid="bib0003" ref-type="bibr"><u>[3]</u></xref>, biopython 1.76 <xref rid="bib0004" ref-type="bibr"><u>[4]</u></xref>, Pysam 0.16 <xref rid="bib0007" ref-type="bibr"><u>[7</u></xref>,<xref rid="bib0008" ref-type="bibr"><u>8]</u></xref>, NumPy <xref rid="bib0009" ref-type="bibr"><u>[9]</u></xref>, Pandas 1.0 <xref rid="bib0010" ref-type="bibr"><u>[10]</u></xref>, SciPy 1.5 <xref rid="bib0011" ref-type="bibr"><u>[11]</u></xref> and ART 2016.06.05 <xref rid="bib0005" ref-type="bibr"><u>[5]</u></xref>.<li> The user may make use of either Conda <xref rid="bib0012" ref-type="bibr"><u>[12]</u></xref> or Singularity <xref rid="bib0013" ref-type="bibr"><u>[13]</u></xref> to supply these dependencies automatically (via <b>snakemake</b>).<li>Our data consist code to simulate integrations and create tables of their properties, consisting of a <b>snakemake</b> workflow and several Python scripts, as well as an example dataset to illustrate the method <xref rid="bib0001" ref-type="bibr"><u>[1]</u></xref>.<li> This figure was generated using <b>snakemake</b> <xref rid="bib0002" ref-type="bibr"><u>[2]</u></xref> and dot <xref rid="bib0014" ref-type="bibr"><u>[14]</u></xref>.<li>&#x2018;Snakefile&#x2019;: The file specifying the <b>snakemake</b> workflow for simulating integrations and reads.<li>&#x2018;<b>snakemake</b>_rules&#x2019;: A directory containing the rules for running the <b>snakemake</b> workflow.<li> [[2]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [[3]](Python Software Foundation, Python Language Reference, version 3, n.d. <ext-link ext-link-type="uri" xlink:href="http://www.python.org" id="interref0006">http://www.python.org</ext-link>.) [[4]](Biopython team, Biopython, n.d. <ext-link ext-link-type="uri" xlink:href="https://biopython.org/" id="interref0007">https://biopython.org/</ext-link>.) [[7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [8]](Pysam development team, Pysam, p, n.d. <ext-link ext-link-type="uri" xlink:href="https://pysam.readthedocs.io" id="interref0008">https://pysam.readthedocs.io</ext-link>.) [[9]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32939066">32939066</a>) [[10]](The Pandas development team, Pandas, n.d. https://doi.org/10.5281/zenodo.3509134.) [[11]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [[5]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22199392">22199392</a>) [[12]](Anaconda development team, Anaconda Software Distribution, n.d. <ext-link ext-link-type="uri" xlink:href="https://anaconda.com" id="interref0009">https://anaconda.com</ext-link>.) [[13]](          <person-group person-group-type="author">            <name>              <surname>Kurtzer</surname>              <given-names>G.M.</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Bauer</surname>              <given-names>M.W.</given-names>            </name>          </person-group>          <article-title>Singularity: scientific containers for mobility of compute</article-title>          <source>PLoS ONE</source>          <volume>12</volume>          <year>2017</year>          <pub-id pub-id-type="doi">10.1371/journal.pone.0177459</pub-id>        ) [[1]](          <person-group person-group-type="author">            <name>              <surname>Scott</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Wilson</surname>              <given-names>L.O.W.</given-names>            </name>          </person-group>          <part-title>Isling</part-title>          <year>2022</year>          <pub-id pub-id-type="doi">10.5281/zenodo.6403449</pub-id>        ) [[2]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [[14]](S.C. North, E.R. Gansner, J.C. Ellson, Y. Hu, Dot, n.d. <ext-link ext-link-type="uri" xlink:href="https://www.graphviz.org/pdf/dotguide.pdf" id="interref0010">https://www.graphviz.org/pdf/dotguide.pdf</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35239511">35239511</a> <b>(1)</b> Microbiota dynamics in a randomized trial of gut decontamination during allogeneic hematopoietic cell transplantation<ul><li> Bioinformatics workflows were implemented using <b>Snakemake</b> (<xref rid="B64" ref-type="bibr"><u>64</u></xref>).<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35396485">35396485</a> <b>(14)</b> Searching thousands of genomes to classify somatic and novel structural variants using STIX<ul><li>Source code availability and <b>Snakemake</b> pipeline.<li>To improve readability and reproducatiblity, the source code for all experiments and analysis in this paper are part of a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup> pipeline available at <ext-link ext-link-type="uri" xlink:href="https://github.com/ryanlayer/stix_paper/blob/main/Snakefile">https://github.com/ryanlayer/stix_paper/blob/main/Snakefile</ext-link>.<li>Relevant <b>Snakemake</b> rules.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>Relevant <b>Snakemake</b> rules:.<li>The germline filtering analysis pipeline and associated scripts are contained within a <b>Snakemake</b> pipeline located at <ext-link ext-link-type="uri" xlink:href="https://github.com/ryanlayer/stix_paper/tree/main/germline_filtering/stix.smk">https://github.com/ryanlayer/stix_paper/tree/main/germline_filtering/stix.smk</ext-link>.<li>Relevant <b>Snakemake</b> rules:.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35197310">35197310</a> <b>(3)</b> A framework to score the effects of structural variants in health and disease<ul><li> The annotation framework automatically retrieves the features from primary annotation sets using the workflow management system <b>Snakemake</b> (<xref rid="GR275995KLEC35" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li>Novel SVs can be scored with a pipeline implemented in <b>Snakemake</b> (<xref rid="GR275995KLEC35" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>), using conda (<xref rid="GR275995KLEC22" ref-type="bibr">Gr&#xFC;ning et al.<li> Conda and <b>Snakemake</b> guarantee easy installation procedures as well as stability through dependency management.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35365713">35365713</a> <b>(1)</b> Strain-resolved analysis in a randomized trial of antibiotic pretreatment and maintenance dose delivery mode with fecal microbiota transplant for ulcerative colitis<ul><li>Sequence and metabolite data were analyzed using a reproducible pipeline implemented with the <b>Snakemake</b> workflow manager<sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup>.<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36518130">36518130</a> <b>(3)</b> A computational workflow for predicting cancer neo-antigens<ul><li>The workflow is created using <b>snakemake</b> version 6.12.3 [<xref rid="R16" ref-type="bibr"><u>16</u></xref>].<li> This <b>snakemake</b> workflow requires variant annotation results from ANNOVAR[<xref rid="R17" ref-type="bibr"><u>17</u></xref>] and proteomics search results.<li> We developed a proteogenomics data analysis workflow using <b>snakemake</b> package.<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20601685">20601685</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35356891">35356891</a> <b>(3)</b> Unifying the known and unknown microbial coding sequence space<ul><li><b>Snakemake</b>.<li><b>Snakemake</b>.<li>We implemented a computation workflow based on <b>Snakemake</b> (<xref rid="bib63" ref-type="bibr"><u>K&#xF6;ster, 2018</u></xref>) for the easy processing of large datasets in a reproducible manner.<li> [K&#xF6;ster, 2018](          <person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name></person-group>          <year iso-8601-date="2018">2018</year>          <article-title>Reproducible data analysis with Snakemake</article-title>          <source>F1000Research</source>          <volume>7</volume>          <elocation-id>801</elocation-id>          <pub-id pub-id-type="doi">10.7490/f1000research.1115801.1</pub-id>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35378751">35378751</a> <b>(1)</b> SARS-CoV-2 and other respiratory pathogens are detected in continuous air samples from congregate settings<ul><li> The data were analyzed using a custom workflow implemented in <b>Snakemake</b><sup><xref rid="R65" ref-type="bibr"><u>65</u></xref></sup>.<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35351165">35351165</a> <b>(1)</b> Adding hydrogen atoms to molecular models via fragment superimposition<ul><li>polar terminal group.</p></list-item></list>The scripts and <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR23"><u>23</u></xref>] to reproduce this benchmark are available in Additional file <xref rid="MOESM1" ref-type="media"><u>1</u></xref>.<li> [23](M&#xF6;lder F, Jablonski KP, Letcher B, Hall MB, Tomkins-Tinch CH, Sochat V, Forster J, Lee S, Twardziok SO, Kanitz A, Wilm A, Holtgrewe M, Rahmann S, Nahnsen S, K&#xF6;ster J. Sustainable data analysis with Snakemake. F1000Research <bold>10</bold>(33) (2021). 10.12688/f1000research.29032.1) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35351938">35351938</a> <b>(1)</b> Linkage mapping, comparative genome analysis, and QTL detection for growth in a non-model teleost, the meagre <italic>Argyrosomus regius</italic>, using ddRAD sequencing<ul><li>STACKS pipeline<sup><xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup> was performed in the <b>snakemake</b>-based singularity container built previously<sup><xref ref-type="bibr" rid="CR68"><u>68</u></xref></sup>.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23701397">23701397</a>) [68](          <person-group person-group-type="author">            <name>              <surname>Zafeiropoulos</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Gioti</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Ninidakis</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Potirakis</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Paragkamian</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Angelova</surname>              <given-names>N</given-names>            </name>            <etal/>          </person-group>          <article-title>0s and 1s in marine molecular research: A regional HPC perspective</article-title>          <source>GigaScience</source>          <year>2021</year>          <volume>10</volume>          <issue>8</issue>          <fpage>53</fpage>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35331283">35331283</a> <b>(1)</b> Perplexity: evaluating transcript abundance estimation in the absence of ground truth<ul><li>We implement perplexity in Rust and provide <b>snakemake</b> [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>] workflows to (a) set up quantify-validate splits of fragment-sets for K-fold cross-validation, and (b) compute perplexities of salmon abundance estimates with respect to validation fragment sets at: <ext-link ext-link-type="uri" xlink:href="http://github.com/COMBINE-lab/perplexity">github.com/COMBINE-lab/perplexity</ext-link>.<li> [41](M&#xF6;lder F, Jablonski KP, Letcher B, Hall MB, Tomkins-Tinch CH, Sochat V, et al.: Sustainable data analysis with Snakemake. F1000Research 10, 33, 2021. 10.12688/f1000research.29032.1) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35324895">35324895</a> <b>(1)</b> Transcriptome diversity is a systematic source of variation in RNA-sequencing data<ul><li>For reproducibility a <b>snakemake</b> pipeline is provided at this study&#x2019;s github repo (<ext-link xlink:href="https://github.com/pablo-gar/transcriptome_diversity_paper" ext-link-type="uri">https://github.com/pablo-gar/transcriptome_diversity_paper</ext-link>).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35324915">35324915</a> <b>(1)</b> Genome wide association study of <italic toggle="yes">Escherichia coli</italic> bloodstream infection isolates identifies genetic determinants for the portal of entry but not fatal outcome<ul><li> Most of the analysis were incorporated in a reproducible pipeline using <b>snakemake</b> v6.5.0 [<xref rid="pgen.1010112.ref077" ref-type="bibr"><u>77</u></xref>] and conda v4.10.3 [<xref rid="pgen.1010112.ref078" ref-type="bibr"><u>78</u></xref>,<xref rid="pgen.1010112.ref079" ref-type="bibr"><u>79</u></xref>].<li> [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [78](Analytics C. Anaconda software distribution. computer software. vers. 2&#x2013;2.4. 0. 2015.) [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35332213">35332213</a> <b>(1)</b> The ViReflow pipeline enables user friendly large scale viral consensus genome reconstruction<ul><li><b>Snakemake</b><sup><xref ref-type="bibr" rid="CR13"><u>13</u></xref></sup>.<li> [13](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>KP</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>MB</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>CH</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>SO</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Holtgrewe</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>          </person-group>          <article-title>Sustainable data analysis with Snakemake</article-title>          <source>F1000 Res.</source>          <year>2021</year>          <volume>10</volume>          <fpage>33</fpage>          <pub-id pub-id-type="doi">10.12688/f1000research.29032.2</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35707452">35707452</a> <b>(5)</b> The National Ecological Observatory Network&#x2019;s soil metagenomes: assembly and basic analysis<ul><li> metaGEM, and its underlying <b>Snakemake</b> framework (<xref rid="ref38" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>), are designed to address common problems with software versioning and updating, as well as efficient data re-analysis (i.e.<li>To customize or expand on the workflow below, it is helpful to know the basic logic of <b>Snakemake</b>, which is the underlying framework for the metaGEM pipeline.<li> <b>Snakemake</b> relies on a series of rules, which specify input files, output files, and any necessary commands.<li> When a rule is called, <b>Snakemake</b> works backwards from the output files to decide if any input files are missing or outdated, and tries to re-run rules as needed (<xref rid="ref38" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> However, there is currently a limited number of supported software tools within KBase, so the next section presents a <b>Snakemake</b>-based approach for carrying out similar tasks.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> The National Ecological Observatory Network&#x2019;s soil metagenomes: assembly and basic analysis<ul><li> The core approach (using Sunbeam, a <b>snakemake</b> pipeline, to analyze NEON metagenomics data) seems like a good one, and will offer advantages to users who are not yet comfortable enough to develop their own such pipeline from scratch.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(2)</b> The National Ecological Observatory Network&#x2019;s soil metagenomes: assembly and basic analysis<ul><li> The core approach (using Sunbeam, a <b>snakemake</b> pipeline, to analyze NEON metagenomics data) seems like a good one, and will offer advantages to users who are not yet comfortable enough to develop their own such pipeline from scratch.<li> Due to the referenced issues with installing software, we have switched to a similar <b>Snakemake</b> pipeline (metaGEM) that has been tested on various computing systems.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35341048">35341048</a> <b>(2)</b> Chikungunya virus time course infection of human macrophages reveals intracellular signaling pathways relevant to repurposed therapeutics<ul><li>The fastq files containing the RNA-sequencing data were subjected to analysis by the <b>Snakemake</b>-based Automated Reproducible MOdular Workflow for Preprocessing and Differential Analysis of RNA-seq Data (ARMOR) software.<li> This workflow, which uses the python-based <b>Snakemake</b> workflow language (<xref rid="ref-30" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2018</u></xref>), performs the following steps: trimming reads with TrimGalore (<xref rid="ref-29" ref-type="bibr"><u>Krueger et al., 2021</u></xref>), performing quality control with FastQC (<xref rid="ref-55" ref-type="bibr"><u>s-andrews, 2021</u></xref>), mapping and quantifying reads to the human GRCh38 transcriptome with Salmon (<xref rid="ref-48" ref-type="bibr"><u>Patro et al., 2017</u></xref>), generating DEG lists with edgeR (<xref rid="ref-52" ref-type="bibr"><u>Robinson, McCarthy &amp; Smyth, 2010</u></xref>), performing GO enrichment with Camera (<xref rid="ref-71" ref-type="bibr"><u>Wu &amp; Smyth, 2012</u></xref>), and identifying significant splice variants from the detected transcripts with DRIMseq (<xref rid="ref-42" ref-type="bibr"><u>Nowicka &amp; Robinson, 2016</u></xref>).<li> [K&#xF6;ster &amp; Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [Krueger et al., 2021](          <person-group person-group-type="author"><name><surname>Krueger</surname><given-names>F</given-names></name><name><surname>James</surname><given-names>F</given-names></name><name><surname>Ewels</surname><given-names>P</given-names></name><name><surname>Afyounian</surname><given-names>E</given-names></name><name><surname>Schuster-Boeckler</surname><given-names>B</given-names></name></person-group>          <article-title>FelixKrueger/TrimGalore: v0.6.7 - DOI via Zenodo. Zenodo</article-title>          <year iso-8601-date="2021">2021</year>          <uri xlink:href="https://github.com/FelixKrueger/TrimGalore/releases">https://github.com/FelixKrueger/TrimGalore/releases</uri>        ) [s-andrews, 2021](          <person-group person-group-type="author"><collab><institution>s-andrews</institution></collab></person-group>          <article-title>GitHub-s-andrews/FastQC: a quality control analysis tool for high throughput sequencing data</article-title>          <year iso-8601-date="2021">2021</year>          <uri xlink:href="https://github.com/s-andrews/FastQC">https://github.com/s-andrews/FastQC</uri>          <date-in-citation content-type="access-date" iso-8601-date="2021-11-23">23 November 2021</date-in-citation>        ) [Patro et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28263959">28263959</a>) [Robinson, McCarthy &amp; Smyth, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19910308">19910308</a>) [Wu &amp; Smyth, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22638577">22638577</a>) [Nowicka &amp; Robinson, 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28105305">28105305</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35327505">35327505</a> <b>(1)</b> CD5 Deficiency Alters Helper T Cell Metabolic Function and Shifts the Systemic Metabolome<ul><li> RNA-Seq data were analyzed using the BYU high-performance computing environment and processed using the ARMOR <b>Snakemake</b>-based automated workflow within a dedicated Conda environment [<xref rid="B36-biomedicines-10-00704" ref-type="bibr"><u>36</u></xref>].<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31088905">31088905</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35301450">35301450</a> <b>(2)</b> Massively-parallel enrichment of minor alleles for mutational testing via low-depth duplex sequencing<ul><li> Processing of BAM files was automated using a <b>Snakemake</b><sup><xref rid="R52" ref-type="bibr"><u>52</u></xref></sup> workflow (<xref rid="SD1" ref-type="supplementary-material">Supplementary Fig.<li> In the <b>Snakemake</b> workflow, MiredasCollectErrorMetric uses the duplex BAM file to describe the number of errors and calculates errors per base sequenced.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35290758">35290758</a> <b>(1)</b> Benchmarking the topological accuracy of bacterial phylogenomic workflows using <italic toggle="yes">in silico</italic> evolution<ul><li>All code is available as a <b>Snakemake</b> v5.8.1 [<xref rid="R15" ref-type="bibr"><u>15</u></xref>] pipeline at <ext-link xlink:href="https://github.com/niekh-13/phylogenetic_workflows" ext-link-type="uri">https://github.com/niekh-13/phylogenetic_workflows</ext-link>.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35277501">35277501</a> <b>(8)</b> FAIRly big: A framework for computationally reproducible processing of large-scale data<ul><li> Importantly, DataLad itself is not a workflow engine, but can be employed for individual nodes and segments of a processing graph defined by other solutions like HTCondor DAGMan<sup><xref ref-type="bibr" rid="CR20"><u>20</u></xref></sup>, or <b>snakemake</b><sup><xref ref-type="bibr" rid="CR21"><u>21</u></xref></sup>.<li> <italic><b>Snakemake</b></italic><sup><xref ref-type="bibr" rid="CR21"><u>21</u></xref></sup> is a feature-rich and domain-agnostic workflow engine, with support for including software environments in the form of conda-environments or software containers, portable workflows that allow execution on remote resources such as cloud services or batch systems on compute clusters, and provenance capture.<li> DataLad can enhance <b>snakemake</b> workflows by retrieving versioned input data files (<ext-link ext-link-type="uri" xlink:href="http://docs.datalad.org/projects/mihextras/generated/man/datalad-x-<b>snakemake</b>.html">http://docs.datalad.org/projects/mihextras/generated/man/datalad-x-<b>snakemake</b>.html</ext-link>), and <b>snakemake</b>-based compute job orchestration could be employed as an alternative to the custom implementations for SLURM and HTCondor used in the work presented here.<li> Similar to <b>snakemake</b>, it provides an alternative workflow definition and orchestration solution, and could employ the provenance capture approach proposed here.<li> However, like <b>snakemake</b> files, these specifications could be tracked within DataLad datasets in order to combine the capabilities of these systems.<li> [20](          <person-group person-group-type="author">            <name>              <surname>Thain</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Tannenbaum</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Livny</surname>              <given-names>M</given-names>            </name>          </person-group>          <article-title>Distributed computing in practice: the Condor experience</article-title>          <source>Concurrency and Computation: Practice and Experience</source>          <year>2005</year>          <volume>17</volume>          <fpage>323</fpage>          <lpage>356</lpage>          <pub-id pub-id-type="doi">10.1002/cpe.938</pub-id>        ) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35277528">35277528</a> <b>(1)</b> 1024-channel electrophysiological recordings in macaque V1 and V4 during resting state<ul><li> These steps were executed after the recording session and implemented into a Python workflow using the <b>Snakemake</b> workflow management system<sup><xref ref-type="bibr" rid="CR42"><u>42</u></xref></sup>.<li> [42](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35266962">35266962</a> <b>(1)</b> Mapping spatial frequency preferences across human primary visual cortex<ul><li> The following packages were used: <b>snakemake</b> (<xref rid="bib62" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>), Jupyter Lab (<xref rid="bib54" ref-type="bibr"><u>Kluyver et&#xA0;al., 2016</u></xref>), numpy (<xref rid="bib3" ref-type="bibr"><u>Array programming with NumPy, 2020</u></xref>), matplotlib (<xref rid="bib40" ref-type="bibr"><u>Hunter, 2007</u></xref>), scipy (<xref rid="bib82" ref-type="bibr"><u>Virtanen et&#xA0;al., 2020</u></xref>), seaborn (<xref rid="bib85" ref-type="bibr"><u>Waskom, 2021</u></xref>), pandas (<xref rid="bib59" ref-type="bibr"><u>Wes McKinney, 2010</u></xref>; <xref rid="bib65" ref-type="bibr"><u>pandas development team, 2020</u></xref>), nipype (<xref rid="bib26" ref-type="bibr"><u>Gorgolewski et&#xA0;al., 2011</u></xref>, <xref rid="bib27" ref-type="bibr"><u>2018</u></xref>), nibabel (<xref rid="bib11" ref-type="bibr"><u>Brett et&#xA0;al., 2020</u></xref>), scikit-learn (<xref rid="bib67" ref-type="bibr"><u>Pedregosa et&#xA0;al., 2011</u></xref>), neuropythy (<xref rid="bib9" ref-type="bibr"><u>Benson &amp; Winawer, 2018</u></xref>), pytorch (<xref rid="bib66" ref-type="bibr"><u>Paszke et&#xA0;al., 2019</u></xref>), psychopy (<xref rid="bib69" ref-type="bibr"><u>Peirce et&#xA0;al., 2019b</u></xref>), FSL (<xref rid="bib78" ref-type="bibr"><u>Smith et&#xA0;al., 2004</u></xref>), freesurfer (<xref rid="bib16" ref-type="bibr"><u>Dale et&#xA0;al., 1999</u></xref>), <ext-link xlink:href="https://github.com/vistalab/vistasoft" ext-link-type="uri">vistasoft</ext-link>, and GLMdenoise (<xref rid="bib49" ref-type="bibr"><u>Kay et&#xA0;al., 2013a</u></xref>).<li> [M&#xF6;lder et&#xA0;al., 2021](<person-group person-group-type="author"><string-name><surname>Moutsiana</surname>, <given-names>C.</given-names></string-name>, <string-name><surname>de Haas</surname>, <given-names>B.</given-names></string-name>, <string-name><surname>Papageorgiou</surname>, <given-names>A.</given-names></string-name>, <string-name><surname>van Dijk</surname>, <given-names>J. A.</given-names></string-name>, <string-name><surname>Balraj</surname>, <given-names>A.</given-names></string-name>, <string-name><surname>Greenwood</surname>, <given-names>J. A.</given-names></string-name>, &amp; <string-name><surname>Schwarzkopf</surname>, <given-names>D. S.</given-names></string-name></person-group> (<year>2016</year>). <article-title>Cortical idiosyncrasies predict the perception of object size</article-title>. <source><italic toggle="yes">Nature Communications,</italic></source><volume>7</volume>(<issue>1</issue>).) [Kluyver et&#xA0;al., 2016](<person-group person-group-type="author"><string-name><surname>Kluyver</surname>, <given-names>T.</given-names></string-name>, <string-name><surname>Ragan-Kelley</surname>, <given-names>B.</given-names></string-name>, <string-name><surname>Perez</surname>, <given-names>F.</given-names></string-name>, <string-name><surname>Granger</surname>, <given-names>B.</given-names></string-name>, <string-name><surname>Bussonnier</surname>, <given-names>M.</given-names></string-name>, <string-name><surname>Frederic</surname>, <given-names>J.</given-names></string-name>, &amp; <string-name><surname>development team</surname>, <given-names>J.</given-names></string-name></person-group> (<year>2016</year>). <article-title>Jupyter notebooks - A publishing format for reproducible computational workflows</article-title>. In <person-group person-group-type="editor"><string-name><given-names>F.</given-names><surname>Loizides</surname></string-name> &amp; <string-name><given-names>B.</given-names><surname>Scmidt</surname></string-name></person-group> (Eds.), <source><italic toggle="yes">Positioning and power in academic publishing: Players, agents and agendas</italic></source> (pp. <fpage>87</fpage>&#x2013;<lpage>90</lpage>). <publisher-name>Amsterdam, the Netherlands: IOS Press</publisher-name>.) [Array programming with NumPy, 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32939066">32939066</a>) [Hunter, 2007](<person-group person-group-type="author"><string-name><surname>Hunter</surname>, <given-names>J. D.</given-names></string-name></person-group> (<year>2007</year>). <article-title>Matplotlib: A 2d graphics environment</article-title>. <source><italic toggle="yes">Computing in Science &amp; Engineering,</italic></source><volume>9</volume>(<issue>3</issue>), <fpage>90</fpage>&#x2013;<lpage>95</lpage>.) [Virtanen et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [Waskom, 2021](<person-group person-group-type="author"><string-name><surname>Waskom</surname>, <given-names>M. L.</given-names></string-name></person-group> (<year>2021</year>). <article-title>Seaborn: Statistical data visualization</article-title>. <source><italic toggle="yes">Journal of Open Source Software,</italic></source><volume>6</volume>(<issue>60</issue>), <fpage>3021</fpage>.) [Wes McKinney, 2010](<person-group person-group-type="author"><string-name><surname>McKinney</surname>, <given-names>W.</given-names></string-name></person-group> (<year>2010</year>). <article-title>Data structures for statistical computing in Python</article-title>. In <person-group person-group-type="editor"><string-name><given-names>S.</given-names><surname>van der Walt</surname></string-name> &amp; <string-name><given-names>J.</given-names><surname>Millman</surname></string-name></person-group> (Eds.), <source><italic toggle="yes">Proceedings of the 9th Python in Science Conference</italic></source> (pp. <fpage>56</fpage>&#x2013;<lpage>61</lpage>).) [pandas development team, 2020](<collab>Pandas Development Team, T</collab>. (<year>2020</year>). <article-title>Pandas-dev/pandas: Pandas (Version latest)</article-title>. <source><italic toggle="yes">Zenodo</italic></source>.) [Gorgolewski et&#xA0;al., 2011](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21897815">21897815</a>) [2018](<person-group person-group-type="author"><string-name><surname>Gorgolewski</surname>, <given-names>K. J.</given-names></string-name>, <string-name><surname>Esteban</surname>, <given-names>O.</given-names></string-name>, <string-name><surname>Markiewicz</surname>, <given-names>C. J.</given-names></string-name>, <string-name><surname>Ziegler</surname>, <given-names>E.</given-names></string-name>, <string-name><surname>Ellis</surname>, <given-names>D. G.</given-names></string-name>, <string-name><surname>Notter</surname>, <given-names>M. P.</given-names></string-name>, &amp; <string-name><surname>Ghosh</surname>, <given-names>S.</given-names></string-name></person-group> (<year>2018</year>). <source><italic toggle="yes">Nipype. Software</italic></source>. Available: <pub-id pub-id-type="doi">10.5281/zenodo.596855</pub-id>.) [Brett et&#xA0;al., 2020](<person-group person-group-type="author"><string-name><surname>Brett</surname>, <given-names>M.</given-names></string-name>, <string-name><surname>Markiewicz</surname>, <given-names>C. J.</given-names></string-name>, <string-name><surname>Hanke</surname>, <given-names>M.</given-names></string-name>, <string-name><surname>Cote</surname>, <given-names>M.-A.</given-names></string-name>, <string-name><surname>Cipollini</surname>, <given-names>B.</given-names></string-name>, <string-name><surname>McCarthy</surname>, <given-names>P.</given-names></string-name>, &amp; <string-name><surname>Cottaar</surname>, <given-names>M.</given-names></string-name></person-group> (<year>2020</year>). <article-title>Nipy/nibabel: 3.2.1</article-title>, doi:<pub-id pub-id-type="doi">10.5281/zenodo.4295521</pub-id>.) [Pedregosa et&#xA0;al., 2011](<person-group person-group-type="author"><string-name><surname>Pedregosa</surname>, <given-names>F.</given-names></string-name>, <string-name><surname>Varoquaux</surname>, <given-names>G.</given-names></string-name>, <string-name><surname>Gramfort</surname>, <given-names>A.</given-names></string-name>, <string-name><surname>Michel</surname>, <given-names>V.</given-names></string-name>, <string-name><surname>Thirion</surname>, <given-names>B.</given-names></string-name>, <string-name><surname>Grisel</surname>, <given-names>O.</given-names></string-name>, &amp; <string-name><surname>Duchesnay</surname>, <given-names>E.</given-names></string-name></person-group> (<year>2011</year>). <article-title>Scikit-learn: Machine learning in Python</article-title>. <source><italic toggle="yes">Journal of Machine Learning Research,</italic></source><volume>12</volume>, <fpage>2825</fpage>&#x2013;<lpage>2830</lpage>.) [Benson &amp; Winawer, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30520736">30520736</a>) [Paszke et&#xA0;al., 2019](<person-group person-group-type="author"><string-name><surname>Paszke</surname>, <given-names>A.</given-names></string-name>, <string-name><surname>Gross</surname>, <given-names>S.</given-names></string-name>, <string-name><surname>Massa</surname>, <given-names>F.</given-names></string-name>, <string-name><surname>Lerer</surname>, <given-names>A.</given-names></string-name>, <string-name><surname>Bradbury</surname>, <given-names>J.</given-names></string-name>, <string-name><surname>Chanan</surname>, <given-names>G.</given-names></string-name>, &amp; <string-name><surname>Chintala</surname>, <given-names>S.</given-names></string-name></person-group> (<year>2019</year>). <article-title>Pytorch: An imperative style, high-performance deep learning library</article-title>. <source><italic toggle="yes">Proceedings of the 33rd International Conference on Neural Information Processing Systems</italic></source> (pp. <fpage>8024</fpage>&#x2013;<lpage>8035</lpage>). <publisher-name>Red Hook, NY, USA: Curran Associates, Inc.</publisher-name>) [Peirce et&#xA0;al., 2019b](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30734206">30734206</a>) [Smith et&#xA0;al., 2004](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15501092">15501092</a>) [Dale et&#xA0;al., 1999](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=9931268">9931268</a>) [Kay et&#xA0;al., 2013a](<person-group person-group-type="author"><string-name><surname>Kay</surname>, <given-names>K. N.</given-names></string-name>, <string-name><surname>Rokem</surname>, <given-names>A.</given-names></string-name>,<string-name><surname>Winawer</surname>, <given-names>J.</given-names></string-name>, <string-name><surname>Dougherty</surname>, <given-names>R. F.</given-names></string-name>, &amp; <string-name><surname>Wandell</surname>, <given-names>B. A.</given-names></string-name></person-group> (<year>2013</year>). <article-title>Glmdenoise: A fast, automated technique for denoising task-based FMRI data</article-title>. <source><italic toggle="yes">Frontiers in Neuroscience,</italic></source><volume>7</volume>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35264116">35264116</a> <b>(1)</b> Assessment of linkage disequilibrium patterns between structural variants and single nucleotide polymorphisms in three commercial chicken populations<ul><li>The complete pipeline was set up in <b>snakemake</b> 5.3.0 [<xref ref-type="bibr" rid="CR84"><u>84</u></xref>] and the according scripts including the snakefile with all used parameters as well as the dependency analytics graph (DAG) and the rulegraph of the pipeline can be found on Zenodo (<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.5770348">https://doi.org/10.5281/zenodo.5770348</ext-link>).<li> [84](K&#xF6;ster J, Rahmann S. Snakemake&#x2013;a scalable bioinformatics workflow engine. Bioinformatics. 2012;28:2520&#x2013;2. 10.1093/bioinformatics/bts480.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35181971">35181971</a> <b>(3)</b> Genomics4RD: An integrated platform to share Canadian deep&#x2010;phenotype and multiomic data for international rare disease gene discovery<ul><li> The pipelines are built using <b>Snakemake</b>, a modern workflow management system that supports reproducible and scalable data analysis (K&#xF6;ster &amp; Rahmann,&#xA0;<xref rid="humu24354-bib-0018" ref-type="bibr"><u>2012</u></xref>; links to the available pipelines can be found under Web Resources).<li> The bioinformatics analysis pipelines are written in Python using the <b>Snakemake</b> workflow management system.<li>WGS and WES <b>Snakemake</b> pipeline documentation: <ext-link xlink:href="https://github.com/ccmbioinfo/crg2" ext-link-type="uri">https://github.com/ccmbioinfo/crg2</ext-link>.<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35260564">35260564</a> <b>(1)</b> A single-cell transcriptomic landscape of innate and adaptive intratumoral immunity in triple negative breast cancer during chemo- and immunotherapies<ul><li>FASTQ files were converted to digital gene-cell count matrices using a Singularity- dependent <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR24"><u>24</u></xref>] employing the Cell Ranger v4 software.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35258340">35258340</a> <b>(2)</b> Quantifying and Cataloguing Unknown Sequences within Human Microbiomes<ul><li>We have developed a modular metagenomic and unknown sequence analysis framework using the sophisticated pipeline management tool <b>Snakemake</b>.<li> Our analysis pipeline takes advantage of portability and flexibility offered by Python, BioPython, and <b>Snakemake</b> tools which allow reproducible analysis of large meta-omic data on any processing servers and clusters.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35259160">35259160</a> <b>(1)</b> Comprehensive mouse microbiota genome catalog reveals major difference to its human counterpart<ul><li>Based on our curated taxonomy, we build Kraken 2 and bracken [<xref rid="pcbi.1009947.ref022" ref-type="bibr"><u>22</u></xref>] databases for the CMMG and the UHGG using FlexTaxD [<xref rid="pcbi.1009947.ref060" ref-type="bibr"><u>60</u></xref>] with a <b>snakemake</b> pipeline available from: <ext-link xlink:href="https://github.com/SilasK/Kraken" ext-link-type="uri">https://github.com/SilasK/Kraken</ext-link>.<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31779668">31779668</a>) [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34469515">34469515</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35336963">35336963</a> <b>(1)</b> <italic toggle="yes">Alphacoronavirus</italic> in a Daubenton&#x2019;s Myotis Bat (<italic toggle="yes">Myotis daubentonii</italic>) in Sweden<ul><li> All software were installed using Bioconda [<xref rid="B30-viruses-14-00556" ref-type="bibr"><u>30</u></xref>] through the workflow manager <b>Snakemake</b> [<xref rid="B31-viruses-14-00556" ref-type="bibr"><u>31</u></xref>].<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33885785">33885785</a>) [31](          <person-group person-group-type="author"><name><surname>Gr&#xFC;ning</surname><given-names>B.</given-names></name><name><surname>Dale</surname><given-names>R.</given-names></name><name><surname>Sj&#xF6;din</surname><given-names>A.</given-names></name><name><surname>Chapman</surname><given-names>B.A.</given-names></name><name><surname>Rowe</surname><given-names>J.</given-names></name><name><surname>Tomkins-Tinch</surname><given-names>C.H.</given-names></name><name><surname>Valieris</surname><given-names>R.</given-names></name><name><surname>K&#xF6;ster</surname><given-names>J.</given-names></name></person-group>          <article-title>Bioconda: Sustainable and comprehensive software distribution for the life sciences</article-title>          <source>Nat. Met.</source>          <year>2018</year>          <volume>15</volume>          <fpage>475</fpage>          <lpage>476</lpage>          <pub-id pub-id-type="doi">10.1038/s41592-018-0046-7</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35255960">35255960</a> <b>(5)</b> SnakeCube: containerized and automated pipeline for de novo genome assembly in HPC environments<ul><li>The pipeline was constructed using the <b>Snakemake</b> workflow manager [<xref ref-type="bibr" rid="CR4"><u>4</u></xref>].<li> <b>Snakemake</b> splits the workflow into clearly defined individual steps (rules), which are either logical pieces of the procedure (i.e.<li>The memory straining of each individual task was monitored using the <italic>benchmark</italic> directive of <b>Snakemake</b>, and specifically the <italic>max_uss</italic> (Unique Set Size) field, which records the actual RAM used by the process without including the shared memory, and thus highlights the true cost of each step.<li> Based on this benchmarking, we specified the cpu threshold of each rule as a fraction of the overall host system cores, provided as a hyperparameter in the config file and via the <italic>threads</italic> directive of <b>Snakemake</b> (e.g.<li> Taking all benchmarks into account, the final optimization was formed using the <italic>group</italic> directive of <b>Snakemake</b>.<li> [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35255493">35255493</a> <b>(2)</b> Bibliometric review of ATAC-Seq and its application in gene expression<ul><li><b>Snakemake</b>.<li><b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35330728">35330728</a> <b>(6)</b> MEDUSA: A Pipeline for Sensitive Taxonomic Classification and Flexible Functional Annotation of Metagenomic Shotgun Sequences<ul><li> Sunbeam, for example, adopts the use of the <b>Snakemake</b> workflow management software (<xref rid="B21" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) to achieve reproducibility and automation.<li> The pipeline&#x2019;s execution rules were detailed using <b>Snakemake</b>, a workflow management system for scalable and reproducible data analyses.<li>MEDUSA is a pipeline for shotgun metagenomic data deployed by the Conda package manager and managed by <b>Snakemake</b>.<li> The <b>Snakemake</b> rules produce results for the reads with and without performing the assembly, but users can easily change this behavior by editing the rules.<li> The intermediate files stored, that might be further inspected, are used by <b>Snakemake</b> to skip steps previously done when the pipeline is restarted.<li> This way, <b>Snakemake</b> will not perform any rule to create these files, as they are no more present in the targets.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35246634">35246634</a> <b>(1)</b> Region-based analysis of rare genomic variants in whole-genome sequencing datasets reveal two novel Alzheimer&#x2019;s disease-associated genes: <italic>DTNB</italic> and <italic>DLG2</italic><ul><li> We have used FBAT [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] (URLs), R [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>], <b>snakemake</b> [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] and bash commands to implement and run the described analyses.<li> [29](Laird N, Horvath S, Xu X. Implementing a unified approach to family-based tests of association. Genet Epidemiol. 2000;19 Suppl 1:S36-42.) [30](Team RC. R: a language and environment for statistical computing. <ext-link ext-link-type="uri" xlink:href="https://www.r-project.org">https://www.r-project.org</ext-link>.) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35261973">35261973</a> <b>(1)</b> G-quadruplex occurrence and conservation: more than just a question of guanine&#x2013;cytosine content<ul><li>The methodological pipeline was performed using <b>Snakemake</b>, a workflow management system (<xref rid="B50" ref-type="bibr"><u>50</u></xref>).<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35229721">35229721</a> <b>(1)</b> Robust and annotation-free analysis of alternative splicing across diverse cell types in mice<ul><li>The bioinformatic pipeline was implemented using <b>Snakemake</b> (<xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35233021">35233021</a> <b>(1)</b> Optimization of cerebrospinal fluid microbial DNA metagenomic sequencing diagnostics<ul><li>PaRCA is implemented as a fully automated one-click <b>snakemake</b> pipeline<sup><xref ref-type="bibr" rid="CR31"><u>31</u></xref></sup>.<li> [31](PaRCA. <ext-link ext-link-type="uri" xlink:href="https://github.com/ClinicalGenomicsGBG/PARCA">https://github.com/ClinicalGenomicsGBG/PARCA</ext-link>. Accessed 5 January 2022.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35296093">35296093</a> <b>(1)</b> A Specific CD44lo CD25lo Subpopulation of Regulatory T Cells Inhibits Anti-Leukemic Immune Response and Promotes the Progression in a Mouse Model of Chronic Lymphocytic Leukemia<ul><li>After initial QCs using FastQC (<uri xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</uri>) and FastQ Screen (<uri xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/">https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/</uri>), fastq files were processed using a local <b>Snakemake</b> workflow including the following main steps.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35217859">35217859</a> <b>(6)</b> Assessing species coverage and assembly quality of rapidly accumulating sequenced genomes<ul><li> The workflow is implemented using the <b>Snakemake</b> workflow management engine [<xref rid="bib34" ref-type="bibr"><u>34</u></xref>,<xref rid="bib35" ref-type="bibr"><u>35</u></xref>], and all software dependencies are managed by the Conda package manager.<li> The entire workflow is implemented using the <b>Snakemake</b> workflow management engine [<xref rid="bib34" ref-type="bibr"><u>34</u></xref>,<xref rid="bib35" ref-type="bibr"><u>35</u></xref>] and all software dependencies are managed by the Conda package manager; this implementation ensures that the workflow is portable and entirely reproducible.<li> All data-related figures, numbers, and supplementary material were generated with a <b>Snakemake</b> workflow [<xref rid="bib35" ref-type="bibr"><u>35</u></xref>] available from [<xref rid="bib61" ref-type="bibr"><u>61</u></xref>] using <b>Snakemake</b> version 6.3.0.<li>Programming language: <b>Snakemake</b>, Python.<li>Other requirements: <b>Snakemake</b>, Conda.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [61](<person-group person-group-type="author"><string-name><surname>Feron</surname><given-names>R</given-names></string-name></person-group>. <comment>paper-a3cat. GitLab. <ext-link xlink:href="https://gitlab.com/evogenlab/paper-a3cat" ext-link-type="uri">https://gitlab.com/evogenlab/paper-a3cat</ext-link>. Accessed 21 October 2021</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35212724">35212724</a> <b>(1)</b> Transcription Factors Evolve Faster Than Their Structural Gene Targets in the Flavonoid Pigment Pathway<ul><li> (2021</xref>), with the steps unified as a single <b>Snakemake</b> pipeline (<ext-link xlink:href="https://osf.io/b7gcp/" ext-link-type="uri">https://osf.io/b7gcp/</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35281736">35281736</a> <b>(1)</b> Analysis of myocardial cellular gene expression during pressure overload reveals matrix based functional intercellular communication<ul><li>Analyses were performed via the <b>snakemake</b> pipeline dropSeqPipe version 0.4 (<ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe" id="intref0090">https://github.com/Hoohm/dropSeqPipe</ext-link>) and drop-seq-tools version 1.13.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35197628">35197628</a> <b>(6)</b> TDP-43 loss and ALS-risk SNPs drive mis-splicing and depletion of UNC13A<ul><li> Our alignment pipeline is implemented in <b>Snakemake</b> version 5.5.4<sup><xref ref-type="bibr" rid="CR43"><u>43</u></xref></sup> and available at: <ext-link ext-link-type="uri" xlink:href="https://github.com/frattalab/rna_seq_<b>snakemake</b>">https://github.com/frattalab/rna_seq_<b>snakemake</b></ext-link>.<li> Our splicing pipeline is implemented in <b>Snakemake</b> version 5.5.4 and available at: <ext-link ext-link-type="uri" xlink:href="https://github.com/frattalab/splicing">https://github.com/frattalab/splicing</ext-link>.<li> Splice junction parsing pipeline is implemented in <b>Snakemake</b> version 5.5.4 and available at: <ext-link ext-link-type="uri" xlink:href="https://github.com/frattalab/bedops_parse_star_junctions">https://github.com/frattalab/bedops_parse_star_junctions</ext-link>.<li>Cross-linked read files from TDP-43 iCLIP experiments in SH-SY5Y and human neuronal stem cells<sup><xref ref-type="bibr" rid="CR22"><u>22</u></xref></sup> were processed using iCount v2.0.1.dev implemented in <b>Snakemake</b> version 5.5.4, available at <ext-link ext-link-type="uri" xlink:href="https://github.com/frattalab/pipeline_iclip">https://github.com/frattalab/pipeline_iclip</ext-link>.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21358640">21358640</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35196485">35196485</a> <b>(1)</b> Astrocyte-neuron crosstalk through Hedgehog signaling mediates cortical synapse development<ul><li> RNAseq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="R19" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35228820">35228820</a> <b>(1)</b> Using Machine Learning to Identify Patients at High Risk of Inappropriate Drug Dosing in Periods with Renal Dysfunction<ul><li>The full analytical pipeline was built with <b>Snakemake</b><xref rid="cit0033" ref-type="bibr"><u>33</u></xref> (schematic overview in <underline><ext-link xlink:href="https://www.dovepress.com/get_supplementary_file.php?f=344435.pdf" ext-link-type="uri">Supplementary Figure S2</ext-link></underline>) to facilitate transparency and reproducibility; blinding was impractical and so foregone, but all analytic code is available online (DOI: 10.5281/zenodo.4560078).<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35189950">35189950</a> <b>(1)</b> The Medaka Inbred Kiyosu-Karlsruhe (MIKK) panel<ul><li>We developed a <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR82"><u>82</u></xref>] called NanoSnake [<xref ref-type="bibr" rid="CR83"><u>83</u></xref>] to run the entire analysis, including read trimming, mapping, quality control, and transcript abundance estimate.<li> [82](K&#xF6;ster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics. 2012 ;28(19):2520&#x2013;2.) [83](Leger A. a-slide/NanoSnake: v0.0.3.1. 2020 [cited 2021 Feb 17]; Available from: <ext-link ext-link-type="uri" xlink:href="https://zenodo.org/record/3630380">https://zenodo.org/record/3630380</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35189951">35189951</a> <b>(1)</b> Genomic variations and epigenomic landscape of the Medaka Inbred Kiyosu-Karlsruhe (MIKK) panel<ul><li>We developed a <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] called pycoSnake [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>] to run the entire analysis, including mapping, quality control, differential methylation analysis, and structural variation calling.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24502796">24502796</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35188571">35188571</a> <b>(3)</b> Rapid and accurate identification of ribosomal RNA sequences via deep learning<ul><li> The runtime, memory (maximum unique set size), and output size were determined with the <b>Snakemake</b> (<xref rid="B45" ref-type="bibr"><u>45</u></xref>) benchmark functionality on a computer with a NVIDIA Tesla V100 GPU, Intel Xeon processor (2.20GHz) virtualized to 80 cores (40 cores were used in the benchmark) and 500GB of main memory.<li> We evaluated the runtimes and memory uses of all methods with the benchmark functionality of <b>Snakemake</b> (Material and Methods).<li> RiboPicker, rRNAFilter, and RQC_rRNAfilter, which had runtimes of&#xA0;&gt;30 min per 1M reads in the <b>Snakemake</b> runtime benchmark (Figure <xref rid="F1" ref-type="fig"><u>1B</u></xref>), were not included in this evaluation and following analyses.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35179201">35179201</a> <b>(2)</b> The Polarity and Specificity of Antiviral T Lymphocyte Responses Determine Susceptibility to SARS-CoV-2 Infection in Patients with Cancer and Healthy Individuals<ul><li> The whole pipeline was powered by both <b>Snakemake</b> (<xref rid="bib99" ref-type="bibr"><u>99</u></xref>) and <b>Snakemake</b>Wrappers.<li> [99](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35605195">35605195</a> <b>(4)</b> Utility of Triti-Map for bulk-segregated mapping of causal genes and regulatory elements in Triticeae<ul><li> These two computational modules (i.e., the interval mapping module and the <italic>de novo</italic> assembly module) integrate computational steps using <b>Snakemake</b> (<xref rid="bib24" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> All of the analysis modules were developed and integrated based on the <b>Snakemake</b> workflow management system.<li> The workflow of this pipeline based on <b>Snakemake</b> is easy to maintain and expand.<li>The Triti-Map package was developed using <b>Snakemake</b> (<xref rid="bib24" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) and Conda.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35404389">35404389</a> <b>(1)</b> Monocytes transition to macrophages within the inflamed vasculature via monocyte CCR2 and endothelial TNFR2<ul><li> RNA-seq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="bib15" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35168548">35168548</a> <b>(1)</b> Whole-genome analyses reveal a novel prophage and cgSNPs-derived sublineages of <italic>Brachyspira hyodysenteriae</italic> ST196<ul><li>GenBank-formatted public genomes and linked metadata (Additional file <xref rid="MOESM3" ref-type="media"><u>3</u></xref>) were downloaded and processed following a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>] workflow in Anvi&#x2019;o (<ext-link ext-link-type="uri" xlink:href="http://merenlab.org/2019/03/14/ncbi-genome-download-magic/">http://merenlab.org/2019/03/14/ncbi-genome-download-magic/</ext-link>).<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35186498">35186498</a> <b>(2)</b> Wei2GO: weighted sequence similarity-based protein function prediction<ul><li>These files can either be produced separately, or be incorporated with Wei2GO <italic toggle="yes">via</italic> its <b>Snakemake</b> pipeline (<xref rid="ref-13" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> Additionally, <b>Snakemake</b> scripts are provided to run the full annotation pipeline, and update the databases to their latest versions.<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35144563">35144563</a> <b>(1)</b> Micronutrient supplementation affects DNA methylation in male gonads with potential intergenerational epigenetic inheritance involving the embryonic development through glutamate receptor-associated genes<ul><li>Both RNA-seq and RRBS pipelines were organised using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR74"><u>74</u></xref>] to combine multiple bioinformatics tools, R and Python scripts for high-throughput sequence analysis.<li> [74](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35143491">35143491</a> <b>(1)</b> Ten simple rules for large-scale data processing<ul><li>There are workflow tools that can support many use cases including workflow languages and systems such as Workflow Description Language (WDL) [<xref rid="pcbi.1009757.ref012" ref-type="bibr"><u>12</u></xref>], Common Workflow Language (CWL) [<xref rid="pcbi.1009757.ref013" ref-type="bibr"><u>13</u></xref>], <b>Snakemake</b> [<xref rid="pcbi.1009757.ref014" ref-type="bibr"><u>14</u></xref>], or Nextflow [<xref rid="pcbi.1009757.ref015" ref-type="bibr"><u>15</u></xref>].<li> [12](Workflow Description Language (WDL) [Internet]. OpenWDL; 2021 [cited 2021 Jul 29]. Available from: <ext-link xlink:href="https://github.com/openwdl/wdl" ext-link-type="uri">https://github.com/openwdl/wdl</ext-link>.) [13](<name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, <name><surname>Kartashov</surname><given-names>A</given-names></name>, <name><surname>Leehr</surname><given-names>D</given-names></name>, <name><surname>M&#xE9;nager</surname><given-names>H</given-names></name>, <etal>et al</etal>. <article-title>Common Workflow Language, v1.0.</article-title><name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijani&#x107;</surname><given-names>N</given-names></name>, editors. United States: figshare; <year>2016</year>.) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35169803">35169803</a> <b>(1)</b> Narrow transmission bottlenecks and limited within-host viral diversity during a SARS-CoV-2 outbreak on a fishing boat<ul><li>All data processing from the raw unaligned sequencing files onwards was handled by our <b>Snakemake</b> pipeline available on Github &#x2013; <ext-link xlink:href="https://github.com/jbloomlab/SARS-CoV-2_bottleneck" ext-link-type="uri">https://github.com/jbloomlab/SARS-CoV-2_bottleneck</ext-link> [<xref rid="R22" ref-type="bibr"><u>22</u></xref>].<li> [22](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name>, <name><surname>Rahmann</surname><given-names>S</given-names></name>. <article-title>Snakemake--a scalable bioinformatics workflow engine</article-title>. <source>Bioinforma Oxf Engl</source>. <year>2012</year>;<volume>28</volume>: <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>. doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35140405">35140405</a> <b>(1)</b> Early prediction of preeclampsia in pregnancy with cell-free RNA<ul><li>Across samples, the bioinformatic pipeline was managed using <b>Snakemake</b> (v.5.8.1).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35140234">35140234</a> <b>(1)</b> Mass cytometric and transcriptomic profiling of epithelial-mesenchymal transitions in human mammary cell lines<ul><li> The analysis scripts were run via <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>, and all the code is available on GitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/csoneson/WagnerEMT2020">https://github.com/csoneson/WagnerEMT2020</ext-link>).<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35129436">35129436</a> <b>(1)</b> Extensive age-dependent loss of antibody diversity in naturally short-lived turquoise killifish<ul><li> <b>Snakemake</b> (<xref rid="bib32" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) was used to design and run data-processing pipelines.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35132263">35132263</a> <b>(1)</b> Cell types of origin of the cell-free transcriptome<ul><li>The bioinformatic pipeline was managed using <b>snakemake</b> (version 5.8.1).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35120462">35120462</a> <b>(3)</b> Assembly-free rapid differential gene expression analysis in non-model organisms using DNA-protein alignment<ul><li>We have implemented our proposed strategy as a <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR32"><u>32</u></xref>], which is available at <ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/project_samar/samar">https://bitbucket.org/project_samar/samar</ext-link>.<li><bold>Programming language:</bold> <b>Snakemake</b> workflow management system.<li> All dependencies automatically handled by <b>Snakemake</b>.<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35187487">35187487</a> <b>(1)</b> Neurodegenerative and functional signatures of the cerebellar cortex in m.3243A&#x2009;&gt;&#x2009;G patients<ul><li>All automatic anatomical and functional data (pre-)processing steps as detailed above have been implemented in a custom and publicly available <b>Snakemake</b><sup><xref rid="fcac024-B63" ref-type="bibr"><u>63</u></xref></sup> workflow (<ext-link xlink:href="https://github.com/royhaast/smk-melas" ext-link-type="uri">https://github.com/royhaast/smk-melas</ext-link>).<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35107341">35107341</a> <b>(2)</b> OptiFit: an Improved Method for Fitting Amplicon Sequences to Existing OTUs<ul><li> We used the benchmark feature provided by <b>Snakemake</b> to measure the run time of every clustering job.<li>We implemented the analysis workflow in <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>) and wrote scripts in R (<xref rid="B20" ref-type="bibr"><u>20</u></xref>), Python (<xref rid="B21" ref-type="bibr"><u>21</u></xref>), and GNU bash (<xref rid="B22" ref-type="bibr"><u>22</u></xref>).<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<collab>R Core Team</collab>. <year>2020</year>. <source>R: a language and environment for statistical computing.</source><publisher-name>R Foundation for Statistical Computing</publisher-name>, <publisher-loc>Vienna, Austria</publisher-loc>.) [21](<person-group person-group-type="author"><string-name><surname>Van Rossum</surname><given-names>G</given-names></string-name>, <string-name><surname>Drake</surname><given-names>FL</given-names></string-name></person-group>. <year>2009</year>. <source>Python 3 reference manual.</source><publisher-name>CreateSpace</publisher-name>, <publisher-loc>Scotts Valley, CA</publisher-loc>.) [22](<collab>GNU Project</collab>. <article-title>Bash reference manual</article-title>. <ext-link xlink:href="https://www.gnu.org/software/bash/manual/bash.html" ext-link-type="uri">https://www.gnu.org/software/bash/manual/bash.html</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35118380">35118380</a> <b>(1)</b> nf-core/mag: a best-practice pipeline for metagenome hybrid assembly and binning<ul><li> However,&#xA0;only a few pipelines such as Muffin (<xref rid="B7" ref-type="bibr"><u>7</u></xref>) and ATLAS (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) make use of workflow management systems, such as <b>Snakemake</b> (<xref rid="B9" ref-type="bibr"><u>9</u></xref>) or Nextflow (<xref rid="B10" ref-type="bibr"><u>10</u></xref>), which facilitate scalability, portability, reproducibility and ease of application.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33561126">33561126</a>) [8](<person-group person-group-type="author"><string-name><surname>Kieser</surname><given-names>S.</given-names></string-name>, <string-name><surname>Brown</surname><given-names>J.</given-names></string-name>, <string-name><surname>Zdobnov</surname><given-names>E.M.</given-names></string-name>, <string-name><surname>Trajkovski</surname><given-names>M.</given-names></string-name>, <string-name><surname>McCue</surname><given-names>L.A.</given-names></string-name></person-group><article-title>ATLAS: a snakemake workflow for assembly, annotation, and genomic binning of metagenome sequence data</article-title>. <source>BMC Bioinf.</source><year>2020</year>; <volume>21</volume>:<fpage>257</fpage>.) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35173762">35173762</a> <b>(6)</b> Protocol: A Multiplexed Reporter Assay to Study Effects of Chromatin Context on DNA Double-Strand Break Repair<ul><li>2.2.1 Setting up the Environment for the <b>Snakemake</b> Pipeline (<xref rid="F3" ref-type="fig"><u>Figure 3E</u></xref>).<li> To successfully run the <italic><b>Snakemake</b></italic> pipeline we recommend to download the DSB-TRIP <italic>github</italic> repository (<ext-link xlink:href="https://github.com/vansteensellab/DSB_TRIP_protocol" ext-link-type="uri">https://github.com/vansteensellab/DSB_TRIP_protocol</ext-link>):.<li> Running the <b>Snakemake</b> pipeline&#x2014;1&#x2013;4&#xA0;h (depending on the amount of data and number of cores used).<li> Run the DSB_TRIP <italic><b>Snakemake</b></italic> pipeline with 20 cores:.<li><monospace><b>snakemake</b> &#x2013;s src/dsb_trip.snake \--configfile &lt;path/to/config.yaml&gt; \--use-conda -j 20</monospace>.<li>The DSB-TRIP pipeline is written as a <italic><b>Snakemake</b></italic> pipeline (<xref rid="B39" ref-type="bibr"><u>Molder et al., 2021</u></xref>).<li> [Molder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35101114">35101114</a> <b>(1)</b> AGAMEMNON: an Accurate metaGenomics And MEtatranscriptoMics quaNtificatiON analysis suite<ul><li>AGAMEMNON is implemented under a modular approach, where all modules are interconnected using the <b>Snakemake</b> workflow management system [<xref ref-type="bibr" rid="CR59"><u>59</u></xref>].<li> [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35102136">35102136</a> <b>(1)</b> Rare transmission of commensal and pathogenic bacteria in the gut microbiome of hospitalized adults<ul><li> Bioinformatics workflows were implemented with <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR97"><u>97</u></xref></sup>.<li> [97](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35202087">35202087</a> <b>(1)</b> FibroDB: Expression Analysis of Protein-Coding and Long Non-Coding RNA Genes in Fibrosis<ul><li> Since the interest to study fibrosis has increased in recent years, the commands and <b>snakemake</b> [<xref rid="B61-ncrna-08-00013" ref-type="bibr"><u>61</u></xref>] pipelines are available via the GitHub repository to allow further analysis of similar RNA-seq data.<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35161323">35161323</a> <b>(1)</b> Impact of Soil Inoculation with <italic toggle="yes">Bacillus amyloliquefaciens</italic> FZB42 on the Phytoaccumulation of Germanium, Rare Earth Elements, and Potentially Toxic Elements<ul><li> R Core Team 2017) into <b>Snakemake</b> [<xref rid="B73-plants-11-00341" ref-type="bibr"><u>73</u></xref>].<li> [73](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35091090">35091090</a> <b>(1)</b> Proteomics Uncovers Novel Components of an Interactive Protein Network Supporting RNA Export in Trypanosomes<ul><li> We used <ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc" id="intref0040">https://www.bioinformatics.babraham.ac.uk/projects/fastqc</ext-link> and MultiQC for quality control, and analysis was done with <b>Snakemake</b> script.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35145977">35145977</a> <b>(1)</b> Data Integration Challenges for Machine Learning in Precision Medicine<ul><li> A number of general purpose approaches have been developed, such is the case of the suite of <monospace>R</monospace>-based algorithms and programs in the <monospace>Bioconductor</monospace> repositories (<xref rid="B156" ref-type="bibr"><u>156</u></xref>), the pipeline management tools, such as <monospace><b>Snakemake</b></monospace> (<xref rid="B179" ref-type="bibr"><u>179</u></xref>, <xref rid="B180" ref-type="bibr"><u>180</u></xref>) and <monospace>Taverna</monospace> (<xref rid="B181" ref-type="bibr"><u>181</u></xref>) or the cloud-based development suites <monospace>Helastic</monospace> (<xref rid="B182" ref-type="bibr"><u>182</u></xref>) and <monospace>BioNimbus</monospace> (<xref rid="B183" ref-type="bibr"><u>183</u></xref>).<li> [156](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15461798">15461798</a>) [179](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [180](<person-group person-group-type="author"><name><surname>Larsonneur</surname><given-names>E</given-names></name><name><surname>Mercier</surname><given-names>J</given-names></name><name><surname>Wiart</surname><given-names>N</given-names></name><name><surname>Le Floch</surname><given-names>E</given-names></name><name><surname>Delhomme</surname><given-names>O</given-names></name><name><surname>Meyer</surname><given-names>V</given-names></name></person-group>. <article-title>Evaluating workflow management systems: a bioinformatics use case</article-title>. In: <source>2018 IEEE International Conference on Bioinformatics and Biomedicine (BIBM).</source><publisher-loc>Madrid</publisher-loc>: <publisher-name>IEEE</publisher-name> (<year>2018</year>). p. <fpage>2773</fpage>&#x2013;<lpage>5</lpage>. ) [181](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23640334">23640334</a>) [182](<person-group person-group-type="author"><name><surname>Aubin</surname><given-names>MR</given-names></name><name><surname>da Rosa Righi</surname><given-names>R</given-names></name><name><surname>Valiati</surname><given-names>VH</given-names></name><name><surname>da Costa</surname><given-names>CA</given-names></name><name><surname>Antunes</surname><given-names>RS</given-names></name><name><surname>Galante</surname><given-names>G</given-names></name></person-group>. <article-title>Helastic: on combining threshold-based and Serverless elasticity approaches for optimizing the execution of bioinformatics applications</article-title>. <source>J Comput Sci.</source> (<year>2021</year>) <volume>53</volume>:<fpage>101407</fpage>. <pub-id pub-id-type="doi">10.1016/j.jocs.2021.101407</pub-id>) [183](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24464852">24464852</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35077539">35077539</a> <b>(4)</b> DENTIST&#x2014;using long reads for closing assembly gaps at high accuracy<ul><li>For DENTIST, we did not count the runtime of local <b>Snakemake</b> rules, which consume very little resources.<li>To make it easy for users to run DENTIST, we used <b>Snakemake</b> [<xref rid="bib14" ref-type="bibr"><u>15</u></xref>] to automate the entire workflow (Fig.&#xA0;<xref rid="fig1" ref-type="fig"><u>1</u></xref>).<li> Furthermore, to enable easy application on a compute cluster without the necessity of complicated software installation steps, we provide DENTIST and all required tools in a Docker container environment that can be easily used with <b>Snakemake</b>'s Singularity integration [<xref rid="bib15" ref-type="bibr"><u>16</u></xref>].<li>Other requirements: <b>Snakemake</b> 5.32.1 or higher.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35076693">35076693</a> <b>(13)</b> A simple guide to <italic toggle="yes">de novo</italic> transcriptome assembly and annotation<ul><li><monospace>transXpress</monospace> - <ext-link xlink:href="https://github.com/transXpress/transXpress-nextflow" ext-link-type="uri">https://github.com/transXpress/transXpress-nextflow</ext-link> (<bold>Nextflow</bold> version), <ext-link xlink:href="https://github.com/transXpress/transXpress-<b>snakemake</b>" ext-link-type="uri">https://github.com/transXpress/transXpress-<b>snakemake</b></ext-link> (<bold>Snakeake</bold> version).<li><monospace><b>Snakemake</b></monospace> is based on <monospace>Python</monospace> which is among the most popular programming languages [<xref rid="ref226" ref-type="bibr"><u>226</u></xref>].<li> As a result of <monospace>Python</monospace>&#x2019;s user-friendly syntax, workflows written in <monospace><b>Snakemake</b></monospace> are not only very readable but also approachable for beginners.<li> In this regard, some so-called wrapper scripts are offered through the <monospace><b>Snakemake</b> Wrapper Repository</monospace> (<ext-link xlink:href="https://<b>snakemake</b>-wrappers.readthedocs.io/en/stable/" ext-link-type="uri">https://<b>snakemake</b>-wrappers.readthedocs.io/en/stable/</ext-link>) that are templated for common bioinformatics tasks.<li> <monospace><b>Snakemake</b></monospace> pipelines are portable and scalable.<li> The only requirements are <monospace>Python</monospace> and <monospace><b>Snakemake</b></monospace> itself.<li> Like <monospace><b>Snakemake</b></monospace> &#xA0;<monospace>Nextflow</monospace> is also scalable and platform-agnostic with regards to execution capabilities.<li><monospace><b>Snakemake</b></monospace> - <ext-link xlink:href="https://<b>snakemake</b>.github.io/" ext-link-type="uri">https://<b>snakemake</b>.github.io/</ext-link>.<li> [226](<!--<QUERY id="Q69" name="Bhardwaj, Punit"><p>Please provide the publisher name, publisher location, editor name for reference 226.</p></QUERY>--><person-group person-group-type="author"><collab>stackoverflow</collab></person-group>. <source>Stack Overflow Developer Survey</source>, <year>2020</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35078858">35078858</a> <b>(2)</b> Frequency and Genomic Aspects of Intrinsic Resistance to Vismodegib in Locally Advanced Basal Cell Carcinoma<ul><li> All processing steps were combined in a pipeline built with <b>snakemake</b> (v5.4.0; RRID:SCR_003475; ref.<li> All processing steps were combined in a pipeline built with <b>snakemake</b> (v5.4.0; RRID:SCR_003475; ref.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34921763">34921763</a> <b>(1)</b> piRNAs initiate transcriptional silencing of spermatogenic genes during <italic>C.&#xA0;elegans</italic> germline development<ul><li> Unless otherwise stated, computations were done using Python and UNIX utilities, either as standalone scripts or as steps implemented in a <b>Snakemake</b> workflow (<xref rid="bib29" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>; <xref rid="bib34" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35072601">35072601</a> <b>(1)</b> ResFinder &#x2013; an open online resource for identification of antimicrobial resistance genes in next-generation sequencing data and prediction of phenotypes from genotypes<ul><li> <b>SnakeMake</b>, Nextflow, Cromwell, etc.).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35128493">35128493</a> <b>(1)</b> Close genetic linkage between human and companion animal extraintestinal pathogenic <italic>Escherichia coli</italic> ST127<ul><li>A modular analysis pipeline known as pipelord2, implemented with the <b>Snakemake</b> workflow management system was used to perform primary bioinformatic analysis (<xref rid="bib0040" ref-type="bibr"><u>Koster&#xA0;and Rahmann 2012</u></xref>).<li> [Koster&#xA0;and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35074468">35074468</a> <b>(1)</b> Betacoronavirus-specific alternate splicing<ul><li> Except for the nasal swab (SARS-CoV-2-A) dataset [<xref rid="bb0135" ref-type="bibr"><u>27</u></xref>], all datasets were processed using a <b>snakemake</b> pipeline that performs the following steps: samples are downloaded from the SRA, quality-control is performed using fastp [<xref rid="bb0145" ref-type="bibr"><u>[29]</u></xref>, <xref rid="bb0150" ref-type="bibr"><u>[30]</u></xref>], alignment to Genome Reference Consortium Human Build 38 version 91 is done using STAR [<xref rid="bb0155" ref-type="bibr"><u>31</u></xref>], and isoform quantification is carried out by RSEM [<xref rid="bb0160" ref-type="bibr"><u>32</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33712587">33712587</a>) [[29]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34127534">34127534</a>) [[30]](          <person-group person-group-type="author">            <name>              <surname>Chen</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Zhou</surname>              <given-names>Y.</given-names>            </name>            <name>              <surname>Chen</surname>              <given-names>Y.</given-names>            </name>            <name>              <surname>Gu</surname>              <given-names>J.</given-names>            </name>          </person-group>          <article-title>fastp: an ultra-fast all-in-one FASTQ preprocessor</article-title>          <source>Bioinformatics</source>          <volume>1</volume>          <issue>34</issue>          <year>2018</year>          <fpage>i884</fpage>          <lpage>i890</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bty560</pub-id>        ) [31](          <person-group person-group-type="author">            <name>              <surname>Dobin</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Davis</surname>              <given-names>C.A.</given-names>            </name>            <name>              <surname>Schlesinger</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Drenkow</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Zaleski</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Jha</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Batut</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Chaisson</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Gingeras</surname>              <given-names>T.R.</given-names>            </name>          </person-group>          <article-title>STAR: ultrafast universal RNA-seq aligner</article-title>          <source>Bioinformatics (Oxford, England)</source>          <volume>29</volume>          <year>2013</year>          <fpage>15</fpage>          <lpage>21</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts635</pub-id>        ) [32](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Dewey</surname>              <given-names>C.N.</given-names>            </name>          </person-group>          <article-title>Rsem: accurate transcript quantification from rna-seq data with or without a reference genome</article-title>          <source>BMC Bioinformat.</source>          <volume>12</volume>          <year>2011</year>          <fpage>323</fpage>          <pub-id pub-id-type="doi">10.1186/1471-2105-12-323</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35205227">35205227</a> <b>(1)</b> Population Structure and Diversity in European Honey Bees (<italic toggle="yes">Apis</italic><italic toggle="yes">mellifera</italic> L.)&#x2014;An Empirical Comparison of Pool and Individual Whole-Genome Sequencing<ul><li> The entire strategy for pooled sequence analysis described here was successfully applied in previous studies [<xref rid="B17-genes-13-00182" ref-type="bibr"><u>17</u></xref>,<xref rid="B52-genes-13-00182" ref-type="bibr"><u>52</u></xref>], and is available as an automatized <b>Snakemake</b> pipeline [<xref rid="B53-genes-13-00182" ref-type="bibr"><u>53</u></xref>,<xref rid="B54-genes-13-00182" ref-type="bibr"><u>54</u></xref>] at <uri xlink:href="https://github.com/jlanga/smsk_popoolation">https://github.com/jlanga/smsk_popoolation</uri> (accessed on 3 November 2019).<li> [17](          <person-group person-group-type="author"><name><surname>Momeni</surname><given-names>J.</given-names></name><name><surname>Parejo</surname><given-names>M.</given-names></name><name><surname>Nielsen</surname><given-names>R.O.</given-names></name><name><surname>Langa</surname><given-names>J.</given-names></name><name><surname>Montes</surname><given-names>I.</given-names></name><name><surname>Papoutsis</surname><given-names>L.</given-names></name><name><surname>Farajzadeh</surname><given-names>L.</given-names></name><name><surname>Bendixen</surname><given-names>C.</given-names></name><name><surname>Cauia</surname><given-names>E.</given-names></name><name><surname>Charriere</surname><given-names>J.-D.</given-names></name><etal/></person-group>          <article-title>Authoritative subspecies diagnosis tool for European honey bees based on ancestry informative SNPs</article-title>          <source>BMC Genom.</source>          <year>2021</year>          <volume>22</volume>          <elocation-id>101</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12864-021-07379-7</pub-id>        ) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29566643">29566643</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35276782">35276782</a> <b>(1)</b> Co-toxicity of Endotoxin and Indoxyl Sulfate, Gut-Derived Bacterial Metabolites, to Vascular Endothelial Cells in Coronary Arterial Disease Accompanied by Gut Dysbiosis<ul><li> The entire workflow for microbiome analysis was run in <b>Snakemake</b> [<xref rid="B11-nutrients-14-00424" ref-type="bibr"><u>11</u></xref>], cutadapt [<xref rid="B12-nutrients-14-00424" ref-type="bibr"><u>12</u></xref>], USEARCH3 algorithm [<xref rid="B13-nutrients-14-00424" ref-type="bibr"><u>13</u></xref>], phyloseq object [<xref rid="B14-nutrients-14-00424" ref-type="bibr"><u>14</u></xref>], decipher idtaxa [<xref rid="B15-nutrients-14-00424" ref-type="bibr"><u>15</u></xref>], with SILVA as reference database [<xref rid="B16-nutrients-14-00424" ref-type="bibr"><u>16</u></xref>].<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [12](          <person-group person-group-type="author"><name><surname>Martin</surname><given-names>M.</given-names></name></person-group>          <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>          <source>EMBnet J.</source>          <year>2011</year>          <volume>17</volume>          <fpage>10</fpage>          <lpage>12</lpage>          <pub-id pub-id-type="doi">10.14806/ej.17.1.200</pub-id>        ) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20709691">20709691</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23630581">23630581</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29291746">29291746</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23193283">23193283</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35026983">35026983</a> <b>(5)</b> Integrated analysis of long non-coding RNAs and mRNAs reveals the regulatory network of maize seedling root responding to salt stress<ul><li> NLncCirSmk is based on the <b>snakemake</b> workflow management system, an open-source tool for creating reproducible and scalable data analyses [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> To improve the efficiency of bioinformatic analysis, we developed a flexible, user-friendly, and modular RNA analysis workflow, named Novel mRNA, LncRNA, and CircRNA Analysis <b>Snakemake</b> Workflow (NLncCirSmk).<li> It is based on the package management software Conda and the workflow management system <b>Snakemake</b>.<li> The <b>Snakemake</b> workflow management system is a tool to create reproducible and scalable data analysis pipelines.<li> Based on the <b>Snakemake</b>, we developed the NLncCirSmk to build an efficient, flexible, and reproducible bioinformatic analysis pipeline.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35022419">35022419</a> <b>(3)</b> Focal adhesion kinase plays a dual role in TRAIL resistance and metastatic outgrowth of malignant melanoma<ul><li> Reads handling from trimming to feature counting was performed using the workflow manager <b>Snakemake</b> (v5.20.1 [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>]) and our RNA-seq template (release v0.2 freely available at <ext-link ext-link-type="uri" xlink:href="https://git-r3lab.uni.lu/aurelien.ginolhac/<b>snakemake</b>-rna-seq">https://git-r3lab.uni.lu/aurelien.ginolhac/<b>snakemake</b>-rna-seq</ext-link>).<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35057489">35057489</a> <b>(1)</b> Prebiotic Galacto-Oligosaccharides Impact Stool Frequency and Fecal Microbiota in Self-Reported Constipated Adults: A Randomized Clinical Trial<ul><li>For the microbiota data, bioinformatic analyses were done with the NIZO VSEARCH-based <b>Snakemake</b> pipeline (V3-V4 16S rRNA region).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35016721">35016721</a> <b>(3)</b> Contribution of 3D genome topological domains to genetic risk of cancers: a genome-wide computational study<ul><li>The different steps described above have been gathered in an easy-to-execute pipeline, using <italic><b>Snakemake</b> </italic>(<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io">https://<b>snakemake</b>.github.io</ext-link>, [<xref ref-type="bibr" rid="CR59"><u>59</u></xref>]) unifying the analysis of different datasets.<li> [59](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35069532">35069532</a> <b>(1)</b> &#x3B2;-Glucan-Induced Immuno-Modulation: A Role for the Intestinal Microbiota and Short-Chain Fatty Acids in Common Carp<ul><li> All steps were implemented in a <b>Snakemake</b> workflow (<xref rid="B49" ref-type="bibr"><u>49</u></xref>), which is available on GitHub<xref rid="fn1" ref-type="fn"><sup>1</sup></xref>.<li> [49](<person-group person-group-type="author"><name><surname>Koster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake&#x2013;a Scalable Bioinformatics Workflow Engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>:<page-range>2520&#x2013;2</page-range>. doi: <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35069616">35069616</a> <b>(1)</b> Structure and Function of Rhizosphere Soil and Root Endophytic Microbial Communities Associated With Root Rot of <italic toggle="yes">Panax notoginseng</italic><ul><li> The EukDetect pipeline uses a <b>Snakemake</b> workflow engine (<xref rid="B58" ref-type="bibr"><u>Kster and Rahmann, 2018</u></xref>).<li> [Kster and Rahmann, 2018](<person-group person-group-type="author"><name><surname>Kster</surname><given-names>J.</given-names></name><name><surname>Rahmann</surname><given-names>S.</given-names></name></person-group> (<year>2018</year>). <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine.</article-title><source><italic toggle="yes">Bioinformatics</italic></source><volume>28</volume><fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35111923">35111923</a> <b>(1)</b> Improving bioinformatics software quality through incorporation of software engineering practices<ul><li> EDAM (<xref rid="ref-43" ref-type="bibr"><u>Ison et al., 2013</u></xref>)), workflow systems including a great number in life sciences ranging from analytic platforms like KNIME (<xref rid="ref-28" ref-type="bibr"><u>Fillbrunn et al., 2017</u></xref>) to Jupyter Notebooks (<xref rid="ref-54" ref-type="bibr"><u>Kluyver et al., 2016</u></xref>), command-line workflow tools like Nextflow (<xref rid="ref-24" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), and dependency systems like <b>SnakeMake</b> (<xref rid="ref-57" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) have all aided the bioinformatics community.<li> [Ison et al., 2013](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23479348">23479348</a>) [Fillbrunn et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28757290">28757290</a>) [Kluyver et al., 2016](          <person-group person-group-type="author"><name><surname>Kluyver</surname><given-names>T</given-names></name><name><surname>Ragan-Kelley</surname><given-names>B</given-names></name><name><surname>P&#xE9;rez</surname><given-names>F</given-names></name><name><surname>Granger</surname><given-names>BE</given-names></name><name><surname>Bussonnier</surname><given-names>M</given-names></name><name><surname>Frederic</surname><given-names>J</given-names></name><name><surname>Kelley</surname><given-names>K</given-names></name><name><surname>Hamrick</surname><given-names>JB</given-names></name><name><surname>Grout</surname><given-names>J</given-names></name><name><surname>Corlay</surname><given-names>S</given-names></name><name><surname>Ivanov</surname><given-names>P</given-names></name></person-group>          <article-title>Jupyter Notebooks-a publishing format for reproducible computational workflows</article-title>          <year iso-8601-date="2016">2016</year>          <conf-name>20th International Conference on Electronic Publishing</conf-name>          <volume>2016</volume>          <fpage>87</fpage>          <lpage>90</lpage>        ) [Di Tommaso et al., 2017](          <person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year iso-8601-date="2017">2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34979909">34979909</a> <b>(1)</b> Association analysis and functional annotation of imputed sequence data within genomic regions influencing resistance to gastro-intestinal parasites detected by an LDLA approach in a nucleus flock of Sarda dairy sheep<ul><li> Whole-genome sequence (WGS) data was processed with a pipeline implemented with <b>Snakemake</b> [<xref ref-type="bibr" rid="CR30"><u>30</u></xref>], and developed at CRS4 (Center For Advanced Studies, Research and Development in Sardinia <ext-link ext-link-type="uri" xlink:href="https://www.crs4.it/">https://www.crs4.it/</ext-link>)) available at <ext-link ext-link-type="uri" xlink:href="https://github.com/solida-core">https://github.com/solida-core</ext-link>.<li> [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34979329">34979329</a> <b>(1)</b> Dynamic changes in &#x3B2;-cell [Ca<sup>2+</sup>] regulate NFAT activation, gene transcription, and islet gap junction communication<ul><li>A custom <b>snakemake</b> pipeline and R scripts to recreate the motif enrichment analysis can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/kwells4/benniger_tf_regulators" id="intref0015"><italic>https://github.com/kwells4/benniger_tf_regulators</italic></ext-link> and will be made available upon publication.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36699416">36699416</a> <b>(2)</b> The developmentally dynamic microRNA transcriptome of <italic toggle="yes">Glossina pallidipes</italic> tsetse flies, vectors of animal trypanosomiasis<ul><li> An in-house script (<ext-link xlink:href="https://github.com/CareenNaitore/<b>Snakemake</b>_project/blob/master/script.folder/adapter_removal.sh" ext-link-type="uri">https://github.com/CareenNaitore/<b>Snakemake</b>_project/blob/master/script.folder/adapter_removal.sh</ext-link>) was used to remove low-quality sequences, contaminants and adaptors of 70-bp length using the Cutadapt software (v1.18; <xref rid="vbab047-B38" ref-type="bibr"><u>Martin, 2011</u></xref>).<li> [Martin, 2011](<person-group person-group-type="author"><string-name><surname>Martin</surname><given-names>M.</given-names></string-name></person-group> (<year>2011</year>) <article-title>Cutadapt removes adapter sequences from high-throughput sequencing reads</article-title>. <source>EMBnet J</source>., <volume>17</volume>, <fpage>10</fpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35062691">35062691</a> <b>(1)</b> mDrop-Seq: Massively Parallel Single-Cell RNA-Seq of <italic toggle="yes">Saccharomyces cerevisiae</italic> and <italic toggle="yes">Candida albicans</italic><ul><li> We developed a <b>Snakemake</b> [<xref rid="B55-vaccines-10-00030" ref-type="bibr"><u>55</u></xref>] protocol called dropRunner that takes paired-end reads in FASTQ format as input and produces an expression matrix corresponding to the UMI of each gene in each cell.<li> [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34965454">34965454</a> <b>(1)</b> What&#x2019;s new and what&#x2019;s next in diffusion MRI preprocessing<ul><li> (2020)</xref> discuss multiple pipeline-tools including Luigi, <b>Snakemake</b> (<xref rid="R177" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), and Nipype (<xref rid="R112" ref-type="bibr"><u>Gorgolewski et al., 2011</u></xref>), where the latter is a dedicated neuroimaging data processing framework in Python.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Gorgolewski et al., 2011](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21897815">21897815</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35059606">35059606</a> <b>(5)</b> Cantaloupe melon genome reveals 3D chromatin features and structural relationship with the ancestral cucurbitaceae karyotype<ul><li> (2017)</xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.nuget.org/packages/SuperNova/2.0.0" id="intref0060">https://www.nuget.org/packages/SuperNova/2.0.0</ext-link></td></tr><tr><td><b>Snakemake</b> workflow.<li> (2017)</xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.nuget.org/packages/SuperNova/2.0.0" id="intref0060">https://www.nuget.org/packages/SuperNova/2.0.0</ext-link></td></tr><tr><td><b>Snakemake</b> workflow</td><td><xref rid="bib63" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/tutorial/basics.html" id="intref0065">https://<b>snakemake</b>.readthedocs.io/en/stable/tutorial/basics.html</ext-link>.<li> assembly size 334.71 Mb), with a <b>snakemake</b> workflow (<xref rid="bib63" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>) using last 963 (<xref rid="bib45" ref-type="bibr"><u>Kie&#x142;basa et&#xA0;al., 2011</u></xref>) and bedtools 2.27.1 (<xref rid="bib77" ref-type="bibr"><u>Quinlan and Hall, 2010</u></xref>).<li>Transcripts were predicted using a <b>Snakemake</b> pipeline (<xref rid="bib47" ref-type="bibr"><u>Koster and Rahmann, 2018</u></xref>).<li> [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [Kie&#x142;basa et&#xA0;al., 2011](          <person-group person-group-type="author">            <name>              <surname>Kie&#x142;basa</surname>              <given-names>S.M.</given-names>            </name>            <name>              <surname>Wan</surname>              <given-names>R.</given-names>            </name>            <name>              <surname>Sato</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Horton</surname>              <given-names>P.</given-names>            </name>            <name>              <surname>Frith</surname>              <given-names>M.C.</given-names>            </name>          </person-group>          <article-title>Adaptive seeds tame genomic sequence comparison</article-title>          <source>Genome Res.</source>          <year>2011</year>          <pub-id pub-id-type="doi">10.1101/gr.113985.110</pub-id>        ) [Quinlan and Hall, 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [Koster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35070483">35070483</a> <b>(2)</b> Stress Testing Pathology Models with Generated Artifacts<ul><li>Due to the large number of samples and experimental conditions involved, we utilized the <b>Snakemake</b> workflow engine[<xref rid="ref26" ref-type="bibr"><u>26</u></xref>] to reproducibly execute the full prediction and image manipulation pipeline.<li>Schematic of processing steps undertaken by our <b>Snakemake</b> workflow.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34949812">34949812</a> <b>(6)</b> Benchmarking atlas-level data integration in single-cell genomics<ul><li> Furthermore, the generated outputs are visible on our scIB website and our workflow is provided as a reproducible <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR29"><u>29</u></xref></sup> pipeline to allow users to test and evaluate data integration methods in their own setting (Code availability).<li>Monitoring the CPU time and peak memory use reported by our <b>Snakemake</b> pipeline (Extended Data Fig.<li> To enable in-depth characterization of method performance on specific tasks, we have provided the reproducible scIB-pipeline <b>Snakemake</b> pipeline and the scIB python module for users to easily benchmark their particular integration scenario.<li>All integration runs were performed using our <b>Snakemake</b> pipeline.<li> For each run of the <b>Snakemake</b> pipeline, we used the <b>Snakemake</b> benchmarking function to measure time and peak memory use (max PSS).<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35003017">35003017</a> <b>(2)</b> Single-Molecule RNA Sequencing Reveals IFN&#x3B3;-Induced Differential Expression of Immune Escape Genes in Merkel Cell Polyomavirus&#x2013;Positive MCC Cell Lines<ul><li> For raw data processing, we made use of the <b>snakemake</b> pipeline provided by Oxford Nanopore (available at<sup><xref rid="footnote1" ref-type="fn"><u>1</u></xref></sup>).<li>Raw counts of the expressed transcripts were collected from the output of the <b>snakemake</b> pipeline.<li> [1](<a href="https://www.frontiersin.org/articles/10.3389/fmicb.2021.785662/full#supplementary-material">https://www.frontiersin.org/articles/10.3389/fmicb.2021.785662/full#supplementary-material</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34988437">34988437</a> <b>(2)</b> <italic toggle="yes">OLOGRAM-MODL</italic>: mining enriched <italic toggle="yes">n</italic>-wise combinations of genomic features with Monte Carlo and dictionary learning<ul><li>The complete workflow used to produce the Results of this study, along with the full data used, is available as a Snakefile (<b>Snakemake</b> file, (<xref rid="B29" ref-type="bibr"><u>29</u></xref>)) at <ext-link xlink:href="https://github.com/qferre/ologram-modl_supp_mat" ext-link-type="uri">https://github.com/qferre/ologram-modl_supp_mat</ext-link>.<li> As a command line tool whose dependencies can be handled by <italic toggle="yes">conda</italic>, it is convenient to run on clusters with reproducible workflow managers such as <b>Snakemake</b>.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34927666">34927666</a> <b>(1)</b> Identifying cancer pathway dysregulations using differential causal effects<ul><li> The GitHub repository also contains the <b>Snakemake</b> (<xref rid="btab847-B26" ref-type="bibr">M&#xF6;lder <italic toggle="yes">et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34928390">34928390</a> <b>(7)</b> Foster thy young: enhanced prediction of orphan genes in assembled genomes<ul><li>Complete methods, all scripts used in this study, and all result files are documented, and a fully automated version of the direct inference pipeline implemented with pyrpipe and <b>snakemake</b> (<xref rid="B49" ref-type="bibr"><u>49</u></xref>) are available at <ext-link xlink:href="https://github.com/eswlab/orphan-prediction" ext-link-type="uri">https://github.com/eswlab/orphan-prediction</ext-link>.<li>We used the <b>Snakemake</b> workflow manager (<xref rid="B49" ref-type="bibr"><u>49</u></xref>) to integrate the Direct Inference pyrpipe pipeline with the meta-assembly steps.<li>To run Direct Inference with pyrpipe and <b>Snakemake</b> requires a single step - the input of SRA accession IDs; BRAKER requires a user to execute three single command-line operations.<li> <b>Snakemake</b> provides multiple options for executing and scaling the pipeline on different HPC systems.<li> Because we implemented Direct Inference using the <b>Snakemake</b> workflow manager, it can be conveniently managed and scaled for multiple nodes.<li>To enable BIND and MIND in a best practices format, we have implemented all Direct Inference core RNA-Seq processing steps in python using pyrpipe (<xref rid="B45" ref-type="bibr"><u>45</u></xref>) and the pipeline in <b>Snakemake</b> (<xref rid="B49" ref-type="bibr"><u>49</u></xref>) (see methods for details), we provide MAKER and BRAKER in singularity containers, and we have developed full, documented MIND and BIND pipelines in a versatile reproducible open source framework.<li> By automating the Direct Inference pipeline using pyrpipe (<xref rid="B45" ref-type="bibr"><u>45</u></xref>) and <b>Snakemake</b> (<xref rid="B49" ref-type="bibr"><u>49</u></xref>), we provide an end-to-end prediction solution that requires minimal user intervention.This automation, combined with extensive step-by-step documentation, enable a researcher aiming to annotate a novel genome to apply the methods to her/his own dataset.<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [45](<person-group person-group-type="author"><string-name><surname>Singh</surname><given-names>U.</given-names></string-name>, <string-name><surname>Li</surname><given-names>J.</given-names></string-name>, <string-name><surname>Seetharam</surname><given-names>A.</given-names></string-name>, <string-name><surname>Wurtele</surname><given-names>E.S.</given-names></string-name></person-group><article-title>pyrpipe: a Python package for RNA-Seq workflows</article-title>. <source>NAR Genom. Bioinformat.</source><year>2021</year>; <volume>3</volume>:<fpage>lqab049</fpage>.) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [45](<person-group person-group-type="author"><string-name><surname>Singh</surname><given-names>U.</given-names></string-name>, <string-name><surname>Li</surname><given-names>J.</given-names></string-name>, <string-name><surname>Seetharam</surname><given-names>A.</given-names></string-name>, <string-name><surname>Wurtele</surname><given-names>E.S.</given-names></string-name></person-group><article-title>pyrpipe: a Python package for RNA-Seq workflows</article-title>. <source>NAR Genom. Bioinformat.</source><year>2021</year>; <volume>3</volume>:<fpage>lqab049</fpage>.) [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35317023">35317023</a> <b>(1)</b> Community Consensus Guidelines to Support FAIR Data Standards in Clinical Research Studies in Primary Mitochondrial Disease<ul><li> to disclose analytical methods in reusable forms (e.g., a CWL or KNIME, <b>snakemake</b> or BioCompute Object) and to specify unambiguously the terms of use and ownership of the computational protocol which could potentially remedy that concern.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34960813">34960813</a> <b>(1)</b> Prediction of SARS-CoV-2 Variant Lineages Using the S1-Encoding Region Sequence Obtained by PacBio Single-Molecule Real-Time Sequencing<ul><li>Data analysis: the A6 amplicon consensus sequences were built from the PacBio reads using a home-made <b>Snakemake</b> pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34919405">34919405</a> <b>(1)</b> Skyline Batch:An Intuitive User Interface for BatchProcessing with Skyline<ul><li>Generalized workflow managers such as Cromwell, <b>Snakemake</b>,andNextflow have been created to improve research reproducibility.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34921170">34921170</a> <b>(1)</b> An atypical class of non-coding small RNAs is produced in rice leaves upon bacterial infection<ul><li> The latest version of our primary sRNA processing workflow was written into a <b>Snakemake</b> pipeline and is available on gitHub (<ext-link ext-link-type="uri" xlink:href="https://github.com/Aucomte/sRNAmake">https://github.com/Aucomte/sRNAmake</ext-link>).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34806414">34806414</a> <b>(1)</b> Transcriptomic Analysis of Cardiomyocyte Extracellular Vesicles in Hypertrophic Cardiomyopathy Reveals Differential snoRNA Cargo<ul><li> The sequencing data were analyzed using a pipeline consisting of FastQC for quality control, Skewer [<xref rid="B28" ref-type="bibr"><u>28</u></xref>] for read trimming, HiSat2 [<xref rid="B29" ref-type="bibr"><u>29</u></xref>] for read alignment (to GRch38 build), StringTie [<xref rid="B30" ref-type="bibr"><u>30</u></xref>] for transcript quantification, and DESeq2 [<xref rid="B31" ref-type="bibr"><u>31</u></xref>] and <b>Snakemake</b> [<xref rid="B32" ref-type="bibr"><u>32</u></xref>] to calculate differential expression.<li> [28](<person-group person-group-type="author"><string-name><surname>Jiang</surname><given-names>H</given-names></string-name>, <string-name><given-names>R</given-names><surname>Lei</surname></string-name>, <string-name><given-names>SW</given-names><surname>Ding</surname></string-name> and <string-name><given-names>S</given-names><surname>Zhu</surname></string-name></person-group>. (<year>2014</year>). <article-title>Skewer: a fast and accurate adapter trimmer for next-generation sequencing paired-end reads</article-title>. <source>BMC Bioinform</source><volume>15</volume>:<fpage>182</fpage>.) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31375807">31375807</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25690850">25690850</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25516281">25516281</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34911553">34911553</a> <b>(2)</b> A benchmark of structural variation detection by long reads through a realistic simulated model<ul><li> Runtime performance was measured using the Unix time command and <b>Snakemake</b> (v5.7.0) [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>] benchmark function on the custom VCF of 24,600 SVs.<li> Runtime performance was measured using the Unix time command and <b>Snakemake</b> (v5.7.0) benchmark function on the 15x sequencing coverage simulation with chromosome 1 of GRCh38.<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34944901">34944901</a> <b>(2)</b> A Comparison of Tools for Copy-Number Variation Detection in Germline Whole Exome and Whole Genome Sequencing Data<ul><li>All tools were run using <b>Snakemake</b> [<xref rid="B39-cancers-13-06283" ref-type="bibr"><u>39</u></xref>], post-processing of the called CNVs was carried out using Python 3.6 and R 3.5.0 [<xref rid="B33-cancers-13-06283" ref-type="bibr"><u>33</u></xref>].<li><b>Snakemake</b>&#x2019;s built-in capabilities for benchmarking runtime and memory usage were used to measure wall-clock time and peak resident set size for calling CNVs on sample NA12878.<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [33](          <person-group person-group-type="author"><collab>R Core Team R</collab></person-group>          <source>A Language and Environment for Statistical Computing</source>          <publisher-name>R Foundation for Statistical Computing</publisher-name>          <publisher-loc>Vienna, Austria</publisher-loc>          <year>2021</year>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34897455">34897455</a> <b>(1)</b> NLR diversity and candidate fusiform rust resistance genes in loblolly pine<ul><li> The <b>snakemake</b> pipeline used for the variant-calling is available at <ext-link xlink:href="https://doi.org/10.5281/zenodo.5762704" ext-link-type="uri">https://doi.org/10.5281/zenodo.5762704</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34893601">34893601</a> <b>(3)</b> RNA modifications detection by comparative Nanopore direct RNA sequencing<ul><li>In order to compare our results with those obtained through other tools, we developed Metacompore, a software pipeline written in the <b>Snakemake</b> language<sup><xref ref-type="bibr" rid="CR31"><u>31</u></xref></sup> that automatically runs 6 different algorithms for modification detection, namely: Nanocompore, Tombo, Eligos, Diff_err, Epinano and MINES (see <bold>Materials and Methods and</bold> Supplementary Table&#xA0;<xref rid="MOESM1" ref-type="media"><u>1</u></xref> for a comparison of their features).<li> Finally, we implemented analysis pipelines in the Nextflow and <b>Snakemake</b>&#xA0;Domain Specific Languages, allowing automatic execution of all processing steps, from raw data up to the execution of Nanocompore and other RNA modification tools, thus greatly simplifying the bioinformatics work.<li>In order to compare Nanocompore against most of the other tools available for RNA modification detection in a reproducible way, we wrote a <b>snakemake</b> pipeline called MetaCompore (<ext-link ext-link-type="uri" xlink:href="https://github.com/a-slide/MetaCompore">https://github.com/a-slide/MetaCompore</ext-link>).<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34950899">34950899</a> <b>(1)</b> Common-sense approaches to sharing tabular data alongside publication<ul><li> Depending on whether the workflow is GUI driven (e.g., Galaxy) or text driven (e.g., Nextflow, <b>Snakemake</b>), the workflow itself can be checked into the version control repository.</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34891160">34891160</a> <b>(2)</b> A high-throughput multiplexing and selection strategy to complete bacterial genomes<ul><li>The proposed workflow aims at capturing a comprehensive representation of the gene content variation across the lineages in the target population and is available at [<xref rid="bib46" ref-type="bibr"><u>30</u></xref>] as a <b>Snakemake</b> pipeline [<xref rid="bib27" ref-type="bibr"><u>31</u></xref>] that only requires (i) a presence/absence matrix in the same format as created by Roary/Panaroo and (ii) the desired number of long-read isolates.<li> The proposed approach is fully available as a <b>Snakemake</b> pipeline at [<xref rid="bib46" ref-type="bibr"><u>30</u></xref>].<li> [30](<person-group person-group-type="author"><collab>Snakemake pipeline to select isolates for long-read sequencing based on a gene presence/absence matrix.</collab></person-group><comment><ext-link xlink:href="https://gitlab.com/sirarredondo/long_read_selection" ext-link-type="uri">https://gitlab.com/sirarredondo/long_read_selection</ext-link>.&#xA0; Accessed 9 June 2021.</comment>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [30](<person-group person-group-type="author"><collab>Snakemake pipeline to select isolates for long-read sequencing based on a gene presence/absence matrix.</collab></person-group><comment><ext-link xlink:href="https://gitlab.com/sirarredondo/long_read_selection" ext-link-type="uri">https://gitlab.com/sirarredondo/long_read_selection</ext-link>.&#xA0; Accessed 9 June 2021.</comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34960302">34960302</a> <b>(2)</b> Big Data Workflows: Locality-Aware Orchestration Using Software Containers<ul><li><b>Snakemake</b> [<xref rid="B43-sensors-21-08212" ref-type="bibr"><u>43</u></xref>] is a workflow orchestration tool that supports wrapping individual steps in containers, and different data solutions can be integrated into workflows by extending the <b>Snakemake</b> codebase.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34938279">34938279</a> <b>(1)</b> Comparative Genomics of Novel <italic toggle="yes">Agrobacterium</italic> G3 Strains Isolated From the International Space Station and Description of <italic toggle="yes">Agrobacterium tomkonis</italic> sp. nov.<ul><li>(1) On one hand, a pangenomic <b>Snakemake</b> workflow (<xref rid="B28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) was run in anvi&#x2019;o v6.2 (<xref rid="B20" ref-type="bibr"><u>Eren et al., 2015</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Eren et al., 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34890448">34890448</a> <b>(4)</b> Linking big biomedical datasets to modular analysis with Portable Encapsulated Projects<ul><li> These pipelines are increasingly written using pipeline frameworks, leading to now dozens of such frameworks that simplify developing reusable computational pipelines [<xref rid="bib10" ref-type="bibr"><u>10</u></xref>], such as the common workflow language [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>], <b>Snakemake</b> [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>], Galaxy [<xref rid="bib13" ref-type="bibr"><u>13</u></xref>], and Nextflow [<xref rid="bib14" ref-type="bibr"><u>14</u></xref>].<li> e.g., <b>Snakemake</b> includes a directive to import a PEP into a workflow that functions alongside earlier, specialized data formats [<xref rid="bib20" ref-type="bibr"><u>20</u></xref>].<li> For instance, <b>Snakemake</b> relies on the peppy package to handle parsing and reading PEP-formatted project metadata to power a workflow run.<li> We have been reaching out to workflow engine communities, such as the <b>Snakemake</b> [<xref rid="bib12" ref-type="bibr"><u>12</u></xref>] and CWL [<xref rid="bib11" ref-type="bibr"><u>11</u></xref>] communities, which already have some support for PEP-formatted metadata.<li> [10](<person-group person-group-type="author"><string-name><surname>Leipzig</surname><given-names>J</given-names></string-name></person-group>. <article-title>A review of bioinformatic pipeline frameworks</article-title>. <source>Brief Bioinform</source>. <year>2016</year>;doi:<pub-id pub-id-type="doi">10.1093/bib/bbw020</pub-id>.) [11](<person-group person-group-type="author"><string-name><surname>Amstutz</surname><given-names>P</given-names></string-name>, <string-name><surname>Crusoe</surname><given-names>MR</given-names></string-name>, <string-name><surname>Tijani&#xE9;</surname><given-names>N</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Common Workflow Language, v1.0</article-title>. <source>figshare</source>. <year>2016</year>;<comment>doi:10.6084/m9.figshare.3115156.v2</comment>.) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27137889">27137889</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32055031">32055031</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](<person-group person-group-type="author"><string-name><surname>Amstutz</surname><given-names>P</given-names></string-name>, <string-name><surname>Crusoe</surname><given-names>MR</given-names></string-name>, <string-name><surname>Tijani&#xE9;</surname><given-names>N</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Common Workflow Language, v1.0</article-title>. <source>figshare</source>. <year>2016</year>;<comment>doi:10.6084/m9.figshare.3115156.v2</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34873156">34873156</a> <b>(3)</b> The histologic phenotype of lung cancers is associated with transcriptomic features rather than genomic characteristics<ul><li> The process was implemented as a <b>Snakemake</b> pipeline and can be found at <ext-link ext-link-type="uri" xlink:href="https://gitlab.com/tangming2005/<b>snakemake</b>_DNAseq_pipeline/tree/multiRG">https://gitlab.com/tangming2005/<b>snakemake</b>_DNAseq_pipeline/tree/multiRG</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34851167">34851167</a> <b>(1)</b> <italic toggle="yes">Muribaculaceae</italic> Genomes Assembled from Metagenomes Suggest Genetic Drivers of Differential Response to Acarbose Treatment in Mice<ul><li>Bioinformatic processing of metagenomes was implemented as a <b>Snakemake</b> pipeline (<xref rid="B63" ref-type="bibr"><u>63</u></xref>) run with version 5.18.1.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34851967">34851967</a> <b>(2)</b> Identification of individual root-knot nematodes using low coverage long-read sequencing<ul><li>We built a reproducible data analysis workflow using the <b>Snakemake</b> workflow manager [<xref rid="pone.0253248.ref024" ref-type="bibr"><u>24</u></xref>].<li> Bioinformatics analysis steps were performed with the <b>Snakemake</b> workflow manager employing a virtual software environment built from this environment list.<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34844637">34844637</a> <b>(6)</b> recount3: summaries and queries for large-scale RNA-seq expression and splicing<ul><li> Lastly, our <b>Snakemake</b>-based [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] analysis pipeline monorail used to produce the summaries is designed to be easy for users to run on their own local RNA-seq reads across computing environments by way of a single Docker/Singularity image.<li> The workflow is driven by <b>Snakemake</b> and runs within a Singularity container on a cluster node.<li>The Monorail analysis workflow is driven by a <b>Snakemake</b> workflow that runs inside a Singularity container.<li><b>Snakemake</b>: 5.4.0.<li> The unifier consists chiefly of a <b>Snakemake</b> workflow, using the GNU parallel utility [<xref ref-type="bibr" rid="CR54"><u>54</u></xref>] to split work across processors.<li>The new, <b>Snakemake</b>-based [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] analysis pipeline used to produce the summaries is now much easier to use than the previous Rail-RNA software [<xref ref-type="bibr" rid="CR41"><u>41</u></xref>].<li> [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [54](          <person-group person-group-type="author">            <name>              <surname>Tange</surname>              <given-names>O</given-names>            </name>            <etal/>          </person-group>          <article-title>Gnu parallel-the command-line power tool</article-title>          <source>USENIX Mag</source>          <year>2011</year>          <volume>36</volume>          <issue>1</issue>          <fpage>42</fpage>          <lpage>47</lpage>        ) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27592709">27592709</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850137">34850137</a> <b>(1)</b> <monospace>CRISPRroots</monospace>: on- and off-target assessment of RNA-seq data in CRISPR&#x2013;Cas9 edited cells<ul><li> The <monospace>CRISPRroots</monospace> pipeline is made in a user-friendly <monospace><b>Snakemake</b></monospace>&#xA0;(<xref rid="B20" ref-type="bibr"><u>20</u></xref>) workflow that optimizes the handling of computing resources, parallelises tasks, and minimizes software prerequisites via the definition of <monospace>Conda</monospace> environments (<ext-link xlink:href="https://docs.anaconda.com/" ext-link-type="uri">https://docs.anaconda.com/</ext-link>), facilitating re-usability and reproducibility.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34988536">34988536</a> <b>(1)</b> Genomic analysis of <italic>Elizabethkingia</italic> species from aquatic environments: Evidence for potential clinical transmission<ul><li>Maximum-likelihood phylogenetic trees were constructed using PhyloSift (<xref rid="bib0019" ref-type="bibr"><u>Darling&#xA0;et&#xA0;al., 2014</u></xref>), and single nucleotide polymorphism (SNP)-based phylogenetic trees were made using Snplord (github.com/maxlcummins/pipelord/tree/master/snplord), an automated <b>snakemake</b> (<xref rid="bib0055" ref-type="bibr"><u>K&#xF6;ster&#xA0;and Rahmann,&#xA0;2012</u></xref>) pipeline that utilizes snippy (github.com/tseemann/snippy), Gubbins (<xref rid="bib0017" ref-type="bibr"><u>Croucher&#xA0;et&#xA0;al., 2015</u></xref>) and SNP-sites (github.com/sanger-pathogens/snp-sites).<li> [Darling&#xA0;et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24482762">24482762</a>) [K&#xF6;ster&#xA0;and Rahmann,&#xA0;2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Croucher&#xA0;et&#xA0;al., 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25414349">25414349</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34850112">34850112</a> <b>(6)</b> PharmacoDB 2.0: improving scalability and transparency of <italic toggle="yes">in vitro</italic> pharmacogenomics analysis<ul><li> Finally, the <b>Snakemake</b> workflow management tool is used to integrate ORCESTRA, rPharmacoDI, PharmacoDI and PharmacoDB into a fully modular and scalable ETL pipeline (<ext-link xlink:href="https://github.com/bhklab/PharmacoDI_<b>snakemake</b>_pipeline" ext-link-type="uri">https://github.com/bhklab/PharmacoDI_<b>snakemake</b>_pipeline</ext-link>) which ensures that all software dependencies, metadata and code are version controlled, transparent and fully reproducible.<li> and the <b>SnakeMake</b> database creation pipelines: <ext-link xlink:href="https://github.com/bhklab/PharmacoDI_<b>snakemake</b>_pipeline" ext-link-type="uri">https://github.com/bhklab/PharmacoDI_<b>snakemake</b>_pipeline</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34977275">34977275</a> <b>(1)</b> Novel human liver-tropic AAV variants define transferable domains that markedly enhance the human tropism of AAV7 and AAV8<ul><li> A workflow was written in <b>Snakemake</b> (5.6)<xref rid="bib42" ref-type="bibr"><sup>42</sup></xref> to process reads and count barcodes.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34818540">34818540</a> <b>(1)</b> Reprogramming of H3K9bhb at regulatory elements is a key feature of fasting in the small intestine<ul><li>Raw fastq reads for all ChIP-seq experiments were processed using a <b>snakemake</b> pipeline <ext-link ext-link-type="uri" xlink:href="https://github.com/crazyhottommy/pyflow-ChIPseq">https://github.com/crazyhottommy/pyflow-ChIPseq</ext-link> as described in (<xref rid="R55" ref-type="bibr"><u>Terranova et al., 2018</u></xref>).<li> [Terranova et al., 2018](<name><surname>Terranova</surname><given-names>C</given-names></name>, <name><surname>Tang</surname><given-names>M</given-names></name>, <name><surname>Orouji</surname><given-names>E</given-names></name>, <name><surname>Maitituoheti</surname><given-names>M</given-names></name>, <name><surname>Raman</surname><given-names>A</given-names></name>, <name><surname>Amin</surname><given-names>S</given-names></name>, <name><surname>Liu</surname><given-names>Z</given-names></name>, and <name><surname>Rai</surname><given-names>K</given-names></name> (<year>2018</year>). <article-title>An Integrated Platform for Genome-wide Mapping of Chromatin States Using High-throughput ChIP-sequencing in Tumor Tissues</article-title>. <source>J. Vis. Exp</source> (<volume>134</volume>)) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35755950">35755950</a> <b>(1)</b> Computational cancer neoantigen prediction: current status and recent advances<ul><li> Workflow management systems such as Nextflow,<xref rid="bib152" ref-type="bibr"><sup>152</sup></xref> <b>Snakemake</b>,<xref rid="bib153" ref-type="bibr"><sup>153</sup></xref> Airflow (<ext-link ext-link-type="uri" xlink:href="https://airflow.apache.org/" id="intref0025">https://airflow.apache.org/</ext-link>), and CWL<xref rid="bib154" ref-type="bibr"><sup>154</sup></xref> allow the development of scalable and reproducible data analysis workflows.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34799582">34799582</a> <b>(2)</b> Interplay between copy number alterations and immune profiles in the early breast cancer Scandinavian Breast Group 2004-1 randomized phase II trial: results from a feasibility study<ul><li> The preprocessing pipeline used in this study is made available through a <b>snakemake</b> file on <ext-link ext-link-type="uri" xlink:href="https://github.com/ljwharbers/sbg2004-cutseq">https://github.com/ljwharbers/sbg2004-cutseq</ext-link>.<li> The copy number variant pipeline with the exact parameters used is made available through a <b>snakemake</b> file on <ext-link ext-link-type="uri" xlink:href="https://github.com/ljwharbers">https://github.com/ljwharbers/sbg2004-cutseq</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34870271">34870271</a> <b>(2)</b> Reproducible Analysis Pipeline for Data Streams: Open-Source Software to Process Data Collected With Mobile Devices<ul><li>RAPIDS is an open-source collection of Python and R scripts that are executed by the <b>Snakemake</b> workflow manager (<xref rid="B43" ref-type="bibr"><u>43</u></xref>) and organized based on the cookie cutter data science project structure (<xref rid="B44" ref-type="bibr"><u>44</u></xref>).<li>RAPIDS uses the workflow manager <b>Snakemake</b> to organize analysis pipelines into contained, ready-to-use, scalable, auditable steps.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [44](<person-group person-group-type="author"><collab>DrivenData</collab></person-group>. <source>Home - Cookiecutter Data Science.</source> (<year>2020</year>). Available online at: <ext-link xlink:href="http://drivendata.github.io/cookiecutter-data-science/#links-to-related-projects-and-references" ext-link-type="uri">http://drivendata.github.io/cookiecutter-data-science/#links-to-related-projects-and-references</ext-link> (accessed March 30, 2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34868242">34868242</a> <b>(1)</b> Evaluation of Germline Structural Variant Calling Methods for Nanopore Sequencing Data<ul><li> A scalable workflow for SV calling based on the popular workflow language <b>Snakemake</b> (<xref rid="B21" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) is available at <ext-link xlink:href="https://github.com/davidebolo1993/EViNCe" ext-link-type="uri">https://github.com/davidebolo1993/EViNCe</ext-link> and can be used to reproduce findings described in this article and adapted to future nanopore sequencing datasets.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34795237">34795237</a> <b>(1)</b> Accurate and scalable variant calling from single cell DNA sequencing data with ProSolo<ul><li> We used <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR47"><u>47</u></xref></sup> (version 6.3.0) to implement the benchmarking workflows.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34795059">34795059</a> <b>(4)</b> Molecular characterization of Barrett&#x2019;s esophagus at single-cell resolution<ul><li>Sequencing reads were mapped to the human genome using the NlaIII mapping pipeline of SingleCellMultiOmics package (see <ext-link xlink:href="https://github.com/BuysDB/SingleCellMultiOmics/tree/master/singlecellmultiomics/<b>snakemake</b>_workflows/nlaIII" ext-link-type="uri">https://github.com/BuysDB/SingleCellMultiOmics/tree/master/singlecellmultiomics/<b>snakemake</b>_workflows/nlaIII</ext-link>).<li>Sequencing reads were mapped to the human genome using the SingleCellMultiOmics pipeline (see <ext-link xlink:href="https://github.com/BuysDB/SingleCellMultiOmics/tree/master/singlecellmultiomics/<b>snakemake</b>_workflows/cs2_scmo" ext-link-type="uri">https://github.com/BuysDB/SingleCellMultiOmics/tree/master/singlecellmultiomics/<b>snakemake</b>_workflows/cs2_scmo</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34789764">34789764</a> <b>(1)</b> DHX15-independent roles for TFIP11 in U6 snRNA modification, U4/U6.U5 tri-snRNP assembly and pre-mRNA splicing fidelity<ul><li> Mapped reads were analyzed using samtools v1.9, <b>snakemake</b>-minimal v5.24.1 and the R package RNAmodR.RiboMethSeq v1.6.0 and the Score C/RiboMethScore were used as a measurement for 2&#x2019;-O-methylation.</ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34835483">34835483</a> <b>(1)</b> Differing Dietary Nutrients and Diet-Associated Bacteria Has Limited Impact on Spider Gut Microbiota Composition<ul><li> All steps were implemented in a <b>Snakemake</b> workflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34829249">34829249</a> <b>(1)</b> Danish Whole-Genome-Sequenced <italic toggle="yes">Candida albicans</italic> and <italic toggle="yes">Candida glabrata</italic> Samples Fit into Globally Prevalent Clades<ul><li>The pipeline for variant calling was constructed using <b>Snakemake</b> [<xref rid="B36-jof-07-00962" ref-type="bibr"><u>36</u></xref>].<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34753499">34753499</a> <b>(1)</b> The genetic case for cardiorespiratory fitness as a clinical vital sign and the routine prescription of physical activity in healthcare<ul><li>For data analysis and figure creation, we used both R [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] and Python 3 [<xref ref-type="bibr" rid="CR32"><u>32</u></xref>], in conjunction with the bioinformatic pipeline software <b>Snakemake</b> [<xref ref-type="bibr" rid="CR33"><u>33</u></xref>].<li> [31](R Core Team. R: a language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. 2021. <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/">https://www.R-project.org/</ext-link>.) [32](          <person-group person-group-type="author">            <name>              <surname>Van Rossum</surname>              <given-names>G</given-names>            </name>            <name>              <surname>Drake</surname>              <given-names>FL</given-names>            </name>          </person-group>          <source>Python 3 reference manual</source>          <year>2009</year>          <publisher-loc>CreateSpace</publisher-loc>          <publisher-name>Scotts Valley, CA</publisher-name>        ) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34751401">34751401</a> <b>(1)</b> ReMap 2022: a database of Human, Mouse, Drosophila and Arabidopsis regulatory regions from an integrative analysis of DNA-binding sequencing experiments<ul><li> The pipeline uses <b>Snakemake</b> (<xref rid="B30" ref-type="bibr"><u>30</u></xref>) in a Conda (<ext-link xlink:href="https://conda.io" ext-link-type="uri">https://conda.io</ext-link>) or Singularity (<ext-link xlink:href="https://sylabs.io/" ext-link-type="uri">https://sylabs.io/</ext-link>) environment, depending on the High-Performance Computing (HPC) resources available, and both Torque and Slurm managers are supported.<li> [30](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J.</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S.</given-names></string-name></person-group><article-title>Snakemake&#x2013;a scalable bioinformatics workflow engine</article-title>. <source>Bioinforma. Oxf. Engl.</source><year>2012</year>; <volume>28</volume>:<fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34752502">34752502</a> <b>(7)</b> Impact of extrinsic incubation temperature on natural selection during Zika virus infection of <italic toggle="yes">Aedes aegypti</italic> and <italic toggle="yes">Aedes albopictus</italic><ul><li> this workflow was generated using <b>Snakemake</b> [<xref rid="ppat.1009433.ref061" ref-type="bibr"><u>61</u></xref>] and the workflow and related documentation can be found at <ext-link xlink:href="https://bitbucket.org/murrieta/<b>snakemake</b>/src" ext-link-type="uri">https://bitbucket.org/murrieta/<b>snakemake</b>/src</ext-link>.<li>All genetic diversity calculations were incorporated into Python and R code located at <ext-link xlink:href="https://bitbucket.org/murrieta/<b>snakemake</b>/src/master/scripts/" ext-link-type="uri">https://bitbucket.org/murrieta/<b>snakemake</b>/src/master/scripts/</ext-link>.<li>Intrahost selection was estimated by the ratio of nonsynonymous (<italic toggle="yes">d</italic><sub><italic toggle="yes">N</italic></sub>) to synonymous (<italic toggle="yes">d</italic><sub><italic toggle="yes">S</italic></sub>) SNVs per site (<italic toggle="yes">d</italic><sub><italic toggle="yes">N</italic></sub><italic toggle="yes">/d</italic><sub><italic toggle="yes">S</italic></sub>) using the Jukes-Cantor formula as previously described [<xref rid="ppat.1009433.ref062" ref-type="bibr"><u>62</u></xref>], and incorporated into custom Python code found at <ext-link xlink:href="https://bitbucket.org/murrieta/<b>snakemake</b>/src/master/scripts/" ext-link-type="uri">https://bitbucket.org/murrieta/<b>snakemake</b>/src/master/scripts/</ext-link>.<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [62](<name><surname>Martin</surname><given-names>M.</given-names></name><article-title>Cutadapt Removes Adapter Sequences From High-Throughput Sequencing Reads</article-title>. <source>EMBnetjournal</source>. <volume>17</volume>(<issue>1</issue>). <pub-id pub-id-type="doi">10.14806/ej.17.1.200</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34754035">34754035</a> <b>(3)</b> Gene-corrected p.A30P <italic>SNCA</italic> patient-derived isogenic neurons rescue neuronal branching and function<ul><li>Data was processed using an in-house <b>snakemake</b> workflow<sup><xref ref-type="bibr" rid="CR73"><u>73</u></xref></sup>, available as a git repository <ext-link ext-link-type="uri" xlink:href="https://git-r3lab.uni.lu/aurelien.ginolhac/<b>snakemake</b>-rna-seq">https://git-r3lab.uni.lu/aurelien.ginolhac/<b>snakemake</b>-rna-seq</ext-link> (release v0.2.3, and singularity image v0.4).<li> [73](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34749640">34749640</a> <b>(1)</b> VPMBench: a test bench for variant prioritization methods<ul><li> The prioritization methods further differ in terms of their required input format for the variants and their software stack, e.g., fathmm-MKL [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] requires Python version 2.7 and uses tabix [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] whereas CADD [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] relies on bioconda [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>] and <b>snakemake</b> [<xref ref-type="bibr" rid="CR19"><u>19</u></xref>] to orchestrate a combination of various scripts to retrieve the scores.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25583119">25583119</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21208982">21208982</a>) [17](Kircher M, Witten DM, Jain P, O&#x2019;Roak BJ, Cooper GM, Shendure J. A general framework for estimating the relative pathogenicity of human genetic variants. Nat Genet. 2014;46(3):310&#x2013;5.) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34769461">34769461</a> <b>(1)</b> From Forensics to Clinical Research: Expanding the Variant Calling Pipeline for the Precision ID mtDNA Whole Genome Panel<ul><li> We use a workflow based on <b>Snakemake</b> [<xref rid="B78-ijms-22-12031" ref-type="bibr"><u>78</u></xref>] that uses: (<bold>I</bold>) Awk, for SAM file editing [<xref rid="B79-ijms-22-12031" ref-type="bibr"><u>79</u></xref>]; (<bold>II</bold>) BEDTools, for BAM to FASTQ conversion [<xref rid="B80-ijms-22-12031" ref-type="bibr"><u>80</u></xref>]; (<bold>III</bold>) BWA-MEM, for read alignment [<xref rid="B81-ijms-22-12031" ref-type="bibr"><u>81</u></xref>]; (<bold>IV</bold>) Pycision, for amplicon delimitation and selection [<xref rid="B34-ijms-22-12031" ref-type="bibr"><u>34</u></xref>]; (<bold>V</bold>) SAMtools for BAM conversion, sorting, indexing, and merging [<xref rid="B82-ijms-22-12031" ref-type="bibr"><u>82</u></xref>]; and (<bold>VI</bold>) Trimmomatic for read quality control and trimming [<xref rid="B83-ijms-22-12031" ref-type="bibr"><u>83</u></xref>] (<xref rid="ijms-22-12031-f001" ref-type="fig"><u>Figure 1</u></xref>B).<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [79](          <person-group person-group-type="author"><name><surname>Aho</surname><given-names>A.V.</given-names></name><name><surname>Kernighan</surname><given-names>B.W.</given-names></name><name><surname>Weinberger</surname><given-names>P.J.</given-names></name></person-group>          <article-title>Awk&#x2014;A Pattern Scanning and Processing Language</article-title>          <source>Softw. Pract. Exper.</source>          <year>1979</year>          <volume>9</volume>          <fpage>267</fpage>          <lpage>279</lpage>          <pub-id pub-id-type="doi">10.1002/spe.4380090403</pub-id>        ) [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [81](          <person-group person-group-type="author"><name><surname>Li</surname><given-names>H.</given-names></name></person-group>          <article-title>Aligning Sequence Reads, Clone Sequences and Assembly Contigs with BWA-MEM</article-title>          <source>arXiv</source>          <year>2013</year>          <pub-id pub-id-type="arxiv">1303.3997</pub-id>        ) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30059903">30059903</a>) [82](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [83](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24695404">24695404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737285">34737285</a> <b>(1)</b> Benchmarking pipelines for subclonal deconvolution of bulk tumour sequencing data<ul><li> TITAN (v1.23.1, ichorCNA v0.3.2, HMMcopy v1.26.0) was run using the provided <b>snakemake</b> workflow with default parameters, with the following exceptions: parameters were set for use with the hg38 human reference genome; a BED file of regions covered by the S04380219 Agilent SureSelect All Exon v5+UTR probes was provided to define target regions (ichorCNA_exons); chromosome X was removed from the analysis; initial normal contamination values were set to &#x201C;c(0,0.1,0.2,0.3,0.4,0.5,0.6)&#x201D;; the maximum number of clusters was set to 8; a statistical parameter was adjusted to accommodate WES datasets (TitanCNA_alphaK: 2500).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34737383">34737383</a> <b>(5)</b> Design considerations for workflow management systems use in production genomics research and the clinic<ul><li> Two other popular systems not examined in this study are <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR57"><u>57</u></xref></sup> and Galaxy<sup><xref ref-type="bibr" rid="CR58"><u>58</u></xref></sup>.<li> GA4GH TES support was only added to <b>Snakemake</b> in their November 2020 release, and Galaxy follows a rather different philosophy focusing on graphical user interface (while having its own CLI) and hence were both excluded.<li> It is also not clear how comparable the implementation was in the systems compared: <b>Snakemake</b> (python-based), Pegasus, Nextflow (java-based), Toil+CWL (python-based), and Cromwell+WDL (java-based).<li> <b>Snakemake</b> was best in terms of I/O wait time and idle time, and Cromwell did the best in terms of number of voluntary and involuntary context switches, but was the worst in most the other categories (closely followed by Toil).<li> By rapid prototyping, they quickly evaluated <b>Snakemake</b>, CWL+CWLtool, CWL+Toil, and Nextflow on a subset of RiboVis<sup><xref ref-type="bibr" rid="CR90"><u>90</u></xref></sup> workflows.<li> [57](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [90](          <person-group person-group-type="author">            <name>              <surname>Carja</surname>              <given-names>O</given-names>            </name>            <name>              <surname>Xing</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Wallace</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Plotkin</surname>              <given-names>JB</given-names>            </name>            <name>              <surname>Shah</surname>              <given-names>P</given-names>            </name>          </person-group>          <article-title>riboviz: Analysis and visualization of ribosome profiling datasets</article-title>          <source>BMC Bioinf.</source>          <year>2017</year>          <volume>18</volume>          <fpage>1</fpage>          <lpage>4</lpage>          <pub-id pub-id-type="doi">10.1186/s12859-017-1873-8</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34805683">34805683</a> <b>(1)</b> uPIC&#x2013;M: Efficient and Scalable Preparationof Clonal Single Mutant Libraries for High-Throughput Protein Biochemistry<ul><li>NGS data were processedusing open source software tools, executed with the <b>Snakemake</b> workflowtool.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34724897">34724897</a> <b>(1)</b> FASTAFS: file system virtualisation of random access compressed FASTA files<ul><li> This will allow the researcher to integrate FASTAFS with workflow management systems such as <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR20"><u>20</u></xref>] or Nextflow&#xA0;[<xref ref-type="bibr" rid="CR21"><u>21</u></xref>] as well as software dependencies by including dependency management configurations.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34723794">34723794</a> <b>(1)</b> Fast and accurate annotation of acoustic signals with deep neural networks<ul><li><b>snakemake</b> <xref rid="bib48" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>.<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34724984">34724984</a> <b>(1)</b> A mixture model for signature discovery from sparse mutation data<ul><li> The workflow is managed by <b>Snakemake</b> [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>].<li> [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34718748">34718748</a> <b>(1)</b> MVIP: multi-omics portal of viral infection<ul><li>We managed the data analyses with <b>Snakemake</b> program, a reproducible workflow management system (<xref rid="B50" ref-type="bibr"><u>50</u></xref>), and executed the pipelines on Linux servers.<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34715899">34715899</a> <b>(1)</b> Over 1000 tools reveal trends in the single-cell RNA-seq analysis landscape<ul><li> In most cases, the platform is a programming language but a minority of tools is built around an existing framework including workflow managers such as <b>Snakemake</b> [<xref ref-type="bibr" rid="CR25"><u>25</u></xref>, <xref ref-type="bibr" rid="CR26"><u>26</u></xref>] and Nextflow [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34700384">34700384</a> <b>(1)</b> Plastic-Degrading Potential across the Global Microbiome Correlates with Recent Pollution Trends<ul><li><b>Snakemake</b> v5.10.0 (<xref rid="B121" ref-type="bibr"><u>121</u></xref>), Python v3.6 (<ext-link xlink:href="http://www.python.org" ext-link-type="uri">www.python.org</ext-link>), and R v3.6 (<ext-link xlink:href="http://www.r-project.org" ext-link-type="uri">www.r-project.org</ext-link>) were used for computations.<li> [121](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34827375">34827375</a> <b>(1)</b> Reduction in Epigenetic Age Acceleration Is Related to Empathy in Mothers with Neglectful Caregiving<ul><li> <b>Snakemake</b> was used to treat bioinformatics workflow [<xref rid="B43-brainsci-11-01376" ref-type="bibr"><u>43</u></xref>].<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34669418">34669418</a> <b>(1)</b> Pseudonatural Products Occur Frequently in BiologicallyRelevant Compounds<ul><li> 2.5)for representing fragment connectivity as graphs, and <b>Snakemake</b><sup><xref ref-type="bibr" rid="ref25"><u>25</u></xref></sup> (v.<li> [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34681040">34681040</a> <b>(1)</b> FA-nf: A Functional Annotation Pipeline for Proteins from Non-Model Organisms Implemented in Nextflow<ul><li> During recent years, the bioinformatics community has been progressively adopting various management systems, such as <b>Snakemake</b> [<xref rid="B24-genes-12-01645" ref-type="bibr"><u>24</u></xref>], Galaxy [<xref rid="B25-genes-12-01645" ref-type="bibr"><u>25</u></xref>], Nextflow [<xref rid="B21-genes-12-01645" ref-type="bibr"><u>21</u></xref>].<li> [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32479607">32479607</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34645491">34645491</a> <b>(1)</b> Artificial intelligence enables comprehensive genome interpretation and nomination of candidate diagnoses for rare genetic diseases<ul><li>We used the <b>Snakemake</b> software [<xref ref-type="bibr" rid="CR76"><u>76</u></xref>] to create a workflow that analyzes cases with the VAAST, Phevor, and Exomiser algorithms.<li> [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34647802">34647802</a> <b>(1)</b> Chromosome-Scale Assembly of the Complete Genome Sequence of <named-content content-type="genus-species">Porcisia hertigi</named-content>, Isolate C119, Strain LV43<ul><li>The analysis workflow for assembly and annotation was performed using <b>Snakemake</b> (<xref rid="B15" ref-type="bibr"><u>15</u></xref>) and is available online for reproducibility purposes (<ext-link xlink:href="https://github.com/hatimalmutairi/LGAAP" ext-link-type="uri">https://github.com/hatimalmutairi/LGAAP</ext-link>), including the software versions and parameters used (<xref rid="B16" ref-type="bibr"><u>16</u></xref>).<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34292068">34292068</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34643507">34643507</a> <b>(3)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> For example, the <b>Snakemake</b> and Nextflow workflow engines facilitate execution of workflows and are used widely for scientific research (<xref rid="bib15" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>; <xref rid="bib28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> For example, to use <b>Snakemake</b> or Nextflow, a researcher can specify a workflow using a custom programming language that is specific to either workflow engine.<li> Indeed, CWL documents are already recognized by many workflow engines, including <b>Snakemake</b> (<xref rid="bib28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), <italic toggle="yes">cwltool</italic>, Toil (<xref rid="bib46" ref-type="bibr"><u>Vivian et al., 2017</u></xref>), Apache Airflow (<xref rid="bib29" ref-type="bibr"><u>Kotliar et al., 2019</u></xref>), Tibanna (<xref rid="bib32" ref-type="bibr"><u>Lee et al., 2019</u></xref>), and Arvados (<xref rid="bib1" ref-type="bibr"><u>Amstutz, 2015</u></xref>).<li> [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Vivian et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [Kotliar et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31321430">31321430</a>) [Lee et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31077294">31077294</a>) [Amstutz, 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26309727">26309727</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26309727">26309727</a> <b>(1)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> A reviewer notes, "I think it would be useful to beginners to see more information on the benefits of using CWL over some of the alternatives (Nextflow, <b>Snakemake</b>).<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26309727">26309727</a> <b>(2)</b> Simplifying the development of portable, scalable, and reproducible workflows<ul><li> A reviewer notes, "I think it would be useful to beginners to see more information on the benefits of using CWL over some of the alternatives (Nextflow, <b>Snakemake</b>).<li> [&#x2026;] Indeed, CWL documents are already recognized by many workflow engines, including <b>Snakemake</b> (K&#xF6;ster and Rahmann, 2012), cwltool, Toil (Vivian et al., 2017), Apache Airflow (Kotliar et al., 2019), Tibanna (Lee et al., 2019), and Arvados (Amstutz, 2015).".</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34721349">34721349</a> <b>(1)</b> Strategy and Performance Evaluation of Low-Frequency Variant Calling for SARS-CoV-2 Using Targeted Deep Illumina Sequencing<ul><li> The workflow was built using the <b>Snakemake</b> workflow management system using python 3.6.9 (<xref rid="B46" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> [M&#xF6;lder et al., 2021](<person-group person-group-type="author"><name><surname>M&#xF6;lder</surname><given-names>F.</given-names></name><name><surname>Jablonski</surname><given-names>K. P.</given-names></name><name><surname>Letcher</surname><given-names>B.</given-names></name><name><surname>Hall</surname><given-names>M. B.</given-names></name><name><surname>Tomkins-Tinch</surname><given-names>C. H.</given-names></name><name><surname>Sochat</surname><given-names>V.</given-names></name><etal/></person-group> (<year>2021</year>). <article-title>Sustainable data analysis with Snakemake.</article-title><source><italic toggle="yes">F1000 Res.</italic></source><volume>10</volume>:<issue>33</issue>. <pub-id pub-id-type="doi">10.12688/f1000research.29032.1</pub-id>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34636840">34636840</a> <b>(1)</b> <italic toggle="yes">vcf2gwas</italic>: Python API for comprehensive GWAS analysis using GEMMA<ul><li> <italic toggle="yes">vcf2gwas</italic> avoids steps such as setting up a configuration file that are common in Nextflow, <b>Snakemake</b>-based pipelines, and also avoids commonly experienced issues with root privileges necessary for running an application inside a Docker container.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34636876">34636876</a> <b>(1)</b> Integration features of intact latent HIV-1 in CD4<sup>+</sup> T cell clones contribute to viral persistence<ul><li>Parallel execution and computing resource management were performed by the <b>Snakemake</b> framework (<xref rid="bib32" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>).<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34682274">34682274</a> <b>(1)</b> Evidence for Lignocellulose-Decomposing Enzymes in the Genome and Transcriptome of the Aquatic Hyphomycete <italic toggle="yes">Clavariopsis aquatica</italic><ul><li> All of the following analyses were implemented as a <b>snakemake</b> (version 3.5.4, [<xref rid="B51-jof-07-00854" ref-type="bibr"><u>51</u></xref>]) workflow that can be found at <uri xlink:href="www.github.com/f-heeger/caquatica_expression">www.github.com/f-heeger/caquatica_expression</uri>.<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34634032">34634032</a> <b>(2)</b> Selective sorting of ancestral introgression in maize and teosinte along an elevational cline<ul><li>First, we checked read quality using fastQ Screen (v0.14.0 [<xref rid="pgen.1009810.ref092" ref-type="bibr"><u>92</u></xref>]) and trimmed out adapter content from raw sequencing reads using the trimmomatic wrapper for <b>snakemake</b> (0.59.1/bio/trimmomatric/pe) [<xref rid="pgen.1009810.ref093" ref-type="bibr"><u>93</u></xref>].<li>We constructed and ran bioinformatics pipelines using <b>snakemake</b> (v.5.17.0 [<xref rid="pgen.1009810.ref110" ref-type="bibr"><u>110</u></xref>]) within a python conda environment (v3.6).<li> [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30254741">30254741</a>) [93](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24695404">24695404</a>) [110](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34729487">34729487</a> <b>(1)</b> Intraoperative DNA methylation classification of brain tumors impacts neurosurgical strategy<ul><li> FAST5 and FASTQ files were analyzed using an adapted <b>snakemake</b><sup><xref rid="CIT0019" ref-type="bibr"><u>19</u></xref></sup> v5.4.0 workflow from the nanoDx pipeline<sup><xref rid="CIT0020" ref-type="bibr"><u>20</u></xref></sup> that enabled analysis on a local laptop computer.<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<person-group person-group-type="author"><string-name><surname>Kuschel</surname><given-names>LP</given-names></string-name>, <string-name><surname>Hench</surname><given-names>J</given-names></string-name>, <string-name><surname>Frank</surname><given-names>S</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Robust methylation-based classification of brain tumors using nanopore sequencing</article-title>. <source>medRxiv</source>. doi:<pub-id pub-id-type="doi">10.1101/2021.03.06.21252627</pub-id>, March 8, <year>2021</year>, preprint: not peer reviewed.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34690970">34690970</a> <b>(1)</b> Sheltering Role of Well-Decayed Conifer Logs for Forest Floor Fungi in Long-Term Polluted Boreal Forests<ul><li> All steps of the bioinformatics analysis workflow were managed with <b>Snakemake</b> v.5.31.1 (<xref rid="B69" ref-type="bibr"><u>M&#xF6;lder et al., 2021</u></xref>).<li> [M&#xF6;lder et al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34614189">34614189</a> <b>(10)</b> metaGEM: reconstruction of genome scale metabolic models directly from metagenomes<ul><li> The metaGEM pipeline is implemented in <b>Snakemake</b> (<xref rid="B26" ref-type="bibr"><u>26</u></xref>), an open-source, community driven, and scalable bioinformatics workflow engine, supporting major popular high-performance-computing cluster environments as well as standalone systems (<xref rid="sup1" ref-type="supplementary-material"><u>Supplementary Figure S1</u></xref>).<li><b>Snakemake</b> v5.10.0 (<xref rid="B26" ref-type="bibr"><u>26</u></xref>).<li><ext-link xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>" ext-link-type="uri">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>.<li><b>Snakemake</b> and HPCCs.<li>metaGEM was implemented in <b>Snakemake</b> v5.10.0, and makes use of the bioinformatic tools listed in Table <xref rid="tbl1" ref-type="table"><u>1</u></xref>.<li> All <b>Snakemake</b> jobs were run on the Chalmers Center for Computational Science and Engineering (C3SE) and the European Molecular Biology Laboratory (EMBL) Heidelberg high performance computer clusters (HPCC).<li> The flexibility of the <b>Snakemake</b> framework allows easy expansion of the metaGEM toolset to add new features or alternative tools.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34614167">34614167</a> <b>(1)</b> A replication-linked mutational gradient drives somatic mutation accumulation and influences germline polymorphisms and genome composition in mitochondrial DNA<ul><li>After obtaining the raw data, we processed all Duplex-Seq data using v1.1.4 of our <b>Snakemake</b>-based Duplex-Seq-Pipeline to ensure that all data were uniformly processed with the exception that different data sets had different unique molecular identifier (UMI) and read lengths (<xref rid="B31" ref-type="bibr"><u>31</u></xref>).<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25299156">25299156</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34608132">34608132</a> <b>(2)</b> Orchestrating and sharing large multimodal data for transparent and reproducible research<ul><li> There have been multiple community efforts in creating standardized workflow languages, such as the Common Workflow Language (CWL) and the Workflow Definition Language (WDL), along with associated workflow management systems such as <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR8"><u>8</u></xref></sup> and Nextflow<sup><xref ref-type="bibr" rid="CR9"><u>9</u></xref></sup>, in order to promote reproducibility<sup><xref ref-type="bibr" rid="CR10"><u>10</u></xref>,<xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> All of the RNA-seq raw data have been pre-processed with Kallisto and Salmon <b>Snakemake</b> pipelines using an HPC environment, and subsequently pushed to assigned data repositories on Pachyderm, allowing for specified selection from the web-app (transcriptome and tool version).<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](          <person-group person-group-type="author">            <name>              <surname>Goble</surname>              <given-names>C</given-names>            </name>            <etal/>          </person-group>          <article-title>FAIR computational workflows</article-title>          <source>Data Intell.</source>          <year>2020</year>          <volume>2</volume>          <fpage>108</fpage>          <lpage>121</lpage>          <pub-id pub-id-type="doi">10.1162/dint_a_00033</pub-id>        ) [11](          <person-group person-group-type="author">            <name>              <surname>Kulkarni</surname>              <given-names>N</given-names>            </name>            <etal/>          </person-group>          <article-title>Reproducible bioinformatics project: a community for reproducible bioinformatics analysis pipelines</article-title>          <source>BMC Bioinforma.</source>          <year>2018</year>          <volume>19</volume>          <fpage>349</fpage>          <pub-id pub-id-type="doi">10.1186/s12859-018-2296-x</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34462593">34462593</a> <b>(1)</b> Reproducibility standards for machine learning in the life sciences<ul><li>The analysis workflow for assembly and annotation was performed using <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>) and is available online for reproducibility purposes (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/hatimalmutairi/LGAAP" ext-link-type="uri">https://github.com/hatimalmutairi/LGAAP</ext-link>), including the software versions and parameters used (<xref rid="B20" ref-type="bibr"><u>20</u></xref>).<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [20](<person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> Genome Assembly and Annotation Pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34594038">34594038</a> <b>(1)</b> Single-cell multimodal glioma analyses identify epigenetic regulators of cellular plasticity and environmental stress response<ul><li> Bulk tissue sequencing computational pipelines were developed using <b>snakemake</b> (5.2.2)<sup><xref rid="R82" ref-type="bibr"><u>82</u></xref></sup>.<li> [82](R82) <li> [82](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34586393">34586393</a> <b>(1)</b> MuWU: Mutant-seq library analysis and annotation<ul><li> As the workflow is built on <b>Snakemake</b> (<xref rid="btab679-B3" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), the addition of further analyses to the automatic handling of Mu-seq libraries allows for modular expansion of its current state.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34712396">34712396</a> <b>(2)</b> MOSGA 2: Comparative genomics and validation tools<ul><li> Although MOSGA 2 grows in complexity, it relies on our established architecture that allows high scalability and reproducibility thanks to the multiple-layer design and the <b>Snakemake</b> workflow engine <xref rid="b0105" ref-type="bibr"><u>[21]</u></xref>.<li> We implemented several <b>Snakemake</b> and comprehensive data analysis tests that GitLab performs on source-code contribution to maintain the code quality.<li> [[21]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33258916">33258916</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34696378">34696378</a> <b>(1)</b> Phylogenomic Analysis of Human Papillomavirus Type 31 and Cervical Carcinogenesis: A Study of 2093 Viral Genomes<ul><li> These analyses were executed using <b>Snakemake</b> [<xref rid="B45-viruses-13-01948" ref-type="bibr"><u>45</u></xref>].<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(18)</b> Phylogenomic Analysis of Human Papillomavirus Type 31 and Cervical Carcinogenesis: A Study of 2093 Viral Genomes<ul><li> We provide a reproducible <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>33</u></xref>] workflow based on the best-performing combination of tools for estimating a genotype reference list of SNPs and demultiplexing samples identified in our benchmark, to facilitate experimental design efforts.<li> The <b>Snakemake</b> workflow is modular, allowing users to substitute alternative tools.<li> All code for the benchmark evaluation and <b>Snakemake</b> workflow is freely accessible on GitHub [<xref rid="bib45" ref-type="bibr"><u>34</u></xref>].<li>We evaluated the performance of genetic demultiplexing algorithms for scRNA-seq samples from HGSOC (high CNV) and lung adenocarcinoma (high SNV) using a set of benchmark evaluations and <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>33</u></xref>] workflow built around freely available tools including Cell Ranger [<xref rid="bib34" ref-type="bibr"><u>35</u></xref>], samtools [<xref rid="bib35" ref-type="bibr"><u>36</u></xref>], bcftools [<xref rid="bib36" ref-type="bibr"><u>37</u></xref>], Unix string manipulation tools (sed and awk), cellSNP [<xref rid="bib37" ref-type="bibr"><u>38</u></xref>], and Vireo [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>] (Methods and Fig.&#xA0;<xref rid="fig1" ref-type="fig"><u>1</u></xref>).<li>Schematic illustrating the steps in the <b>Snakemake</b> workflow.<li> The <b>Snakemake</b> workflow runs a complete analysis for 1 dataset (HGSOC) and doublet simulation scenario (20% doublets).<li>We evaluated the computational runtimes for the various components in our benchmark scenarios and <b>Snakemake</b> workflow using the HGSOC data.<li> In the <b>Snakemake</b> workflow (Fig.&#xA0;<xref rid="fig4" ref-type="fig"><u>4A</u></xref>), the slowest steps were running Cell Ranger (&#x223C;6 hours per sample using 10 cores) and parsing the merged BAM file containing aligned reads to combine cell barcodes into simulated doublets (&#x223C;1 day).<li> <bold>(A)</bold> Runtimes for steps in the complete <b>Snakemake</b> workflow, for a single dataset (HGSOC) and doublet simulation scenario (20%).<li> We provide a freely available, modular <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>33</u></xref>] workflow implementing the best-performing scenario from our benchmark, built around cellSNP/Vireo [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>, <xref rid="bib37" ref-type="bibr"><u>38</u></xref>] and other freely accessible tools, as well as additional R and shell scripts to reproduce all analyses in our benchmark evaluations and additional analyses [<xref rid="bib45" ref-type="bibr"><u>34</u></xref>], to allow other researchers to perform similar analyses for experimental design, planning, and budgeting purposes in their own datasets.<li> In this study, we have built our <b>Snakemake</b> workflow around the best-performing tools (cellSNP/Vireo [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>, <xref rid="bib37" ref-type="bibr"><u>38</u></xref>] for demultiplexing and using matched bulk RNA-seq samples for genotyping) and compared against demuxlet [<xref rid="bib5" ref-type="bibr"><u>5</u></xref>] and baseline scenarios (no doublets), but we have not performed a comprehensive benchmark evaluation of all available tools, such as additional tools for demultiplexing (e.g., scSplit [<xref rid="bib4" ref-type="bibr"><u>4</u></xref>], souporcell [<xref rid="bib19" ref-type="bibr"><u>19</u></xref>], and freemuxlet [<xref rid="bib20" ref-type="bibr"><u>20</u></xref>]).<li> However, we have implemented the <b>Snakemake</b> workflow to be modular so that other users can substitute alternative tools if they prefer.<li> Our <b>Snakemake</b> workflow can be used to demultiplex up to &#x223C;12 pooled samples without a genotype reference (limited by the demultiplexing algorithm Vireo); beyond this, the demultiplexing performance of the Vireo algorithm has been shown to decrease [<xref rid="bib3" ref-type="bibr"><u>3</u></xref>].<li> Finally, the <b>Snakemake</b> workflow is relatively computationally intensive and requires access to a high-performance Linux computing cluster or server.<li>We begin by describing in detail our benchmark evaluation framework, and we note that our additional <b>Snakemake</b> [<xref rid="bib33" ref-type="bibr"><u>33</u></xref>] workflow is built around the combination of tools that resulted in the best performance from the benchmark evaluation.<li> The <b>Snakemake</b> workflow is designed to be modular, allowing other alternative or new tools to be substituted.<li> All code for the benchmark evaluation and <b>Snakemake</b> workflow is freely available on GitHub [<xref rid="bib45" ref-type="bibr"><u>34</u></xref>].<li>All code and scripts to reproduce the benchmark evaluations, supplementary analyses, generate figures in the manuscript, and run the <b>Snakemake</b> workflow are freely accessible from GitHub [<xref rid="bib45" ref-type="bibr"><u>34</u></xref>].<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [34](<person-group person-group-type="author"><string-name><surname>Weber</surname><given-names>LM</given-names></string-name></person-group>. <comment>snp-dmx-cancer. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/lmweber/snp-dmx-cancer" ext-link-type="uri">https://github.com/lmweber/snp-dmx-cancer</ext-link>. Accessed 26 August 2021</comment>.) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<comment>10x Genomics. Cell Ranger. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://support10xgenomics.com/single-cell-gene-expression/software/overview/welcome" ext-link-type="uri">https://support10xgenomics.com/single-cell-gene-expression/software/overview/welcome</ext-link></comment>. <year>2020</year>.) [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19505943">19505943</a>) [37](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21903627">21903627</a>) [38](<person-group person-group-type="author"><string-name><surname>Huang</surname><given-names>Y</given-names></string-name></person-group>. <comment>cellSNP</comment>. <year>2020</year>. <comment><ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/single-cell-genetics/cellSNP" ext-link-type="uri">https://github.com/single-cell-genetics/cellSNP</ext-link></comment>.) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31836005">31836005</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31836005">31836005</a>) [38](<person-group person-group-type="author"><string-name><surname>Huang</surname><given-names>Y</given-names></string-name></person-group>. <comment>cellSNP</comment>. <year>2020</year>. <comment><ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/single-cell-genetics/cellSNP" ext-link-type="uri">https://github.com/single-cell-genetics/cellSNP</ext-link></comment>.) [34](<person-group person-group-type="author"><string-name><surname>Weber</surname><given-names>LM</given-names></string-name></person-group>. <comment>snp-dmx-cancer. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/lmweber/snp-dmx-cancer" ext-link-type="uri">https://github.com/lmweber/snp-dmx-cancer</ext-link>. Accessed 26 August 2021</comment>.) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31836005">31836005</a>) [38](<person-group person-group-type="author"><string-name><surname>Huang</surname><given-names>Y</given-names></string-name></person-group>. <comment>cellSNP</comment>. <year>2020</year>. <comment><ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/single-cell-genetics/cellSNP" ext-link-type="uri">https://github.com/single-cell-genetics/cellSNP</ext-link></comment>.) [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29227470">29227470</a>) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31856883">31856883</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32366989">32366989</a>) [20](<person-group person-group-type="author"><string-name><surname>Zhang</surname><given-names>F</given-names></string-name>, <string-name><surname>Kang</surname><given-names>HM</given-names></string-name></person-group>. <comment>popscle: A suite of population scale analysis tools for single-cell genomics data (freemuxlet). Software package</comment>. <year>2020</year>. <comment><ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/statgen/popscle" ext-link-type="uri">https://github.com/statgen/popscle</ext-link></comment>.) [3](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31836005">31836005</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [34](<person-group person-group-type="author"><string-name><surname>Weber</surname><given-names>LM</given-names></string-name></person-group>. <comment>snp-dmx-cancer. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/lmweber/snp-dmx-cancer" ext-link-type="uri">https://github.com/lmweber/snp-dmx-cancer</ext-link>. Accessed 26 August 2021</comment>.) [34](<person-group person-group-type="author"><string-name><surname>Weber</surname><given-names>LM</given-names></string-name></person-group>. <comment>snp-dmx-cancer. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/lmweber/snp-dmx-cancer" ext-link-type="uri">https://github.com/lmweber/snp-dmx-cancer</ext-link>. Accessed 26 August 2021</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34551439">34551439</a> <b>(2)</b> Genomic impact of stress-induced transposable element mobility in Arabidopsis<ul><li> To identify single nucleotide polymorphism (SNP) and short insertions/deletions (indels), we used an implementation of the GATK4 pipeline (<xref rid="B28" ref-type="bibr"><u>28</u></xref>) available here: <ext-link xlink:href="https://github.com/<b>snakemake</b>-workflows/dna-seq-gatk-variant-calling" ext-link-type="uri">https://github.com/<b>snakemake</b>-workflows/dna-seq-gatk-variant-calling</ext-link>.<li> [28](<person-group person-group-type="author"><string-name><surname>O&#x2019;Connor</surname><given-names>B.D.</given-names></string-name>, <string-name><surname>Auwera</surname><given-names>G.V.D.</given-names></string-name></person-group><year>2020</year>; <comment>O&#x2019;Reilly Media</comment>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34548335">34548335</a> <b>(1)</b> A BRCA1 coiled-coil domain variant disrupting PALB2 interaction promotes the development of mammary tumors and confers a targetable defect in homologous recombination repair<ul><li> The pipeline was implemented in <italic toggle="yes"><b>Snakemake</b></italic> v5.8.1 [<xref rid="B47-genes-12-01443" ref-type="bibr"><u>47</u></xref>] and containers were built for all the software using <italic toggle="yes">Singularity</italic> v3.5.3 [<xref rid="B48-genes-12-01443" ref-type="bibr"><u>48</u></xref>].<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34533457">34533457</a> <b>(2)</b> Structural characterization of NrnC identifies unifying features of dinucleases<ul><li><b>SnakeMake</b>.<li>All commands and code from this analysis were constructed as a reproducible <b>SnakeMake</b> pipeline (<xref rid="bib33" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>) available at <ext-link xlink:href="https://github.com/jgoodson/rnases" ext-link-type="uri">https://github.com/jgoodson/rnases</ext-link>, (copy archived at <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:948afe1bbb90435abda675896f596c93577ee420;origin=https://github.com/jgoodson/rnases;visit=swh:1:snp:26ecae94f2928de36d0bdac09fe4eb492c208138;anchor=swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220" ext-link-type="uri">swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220</ext-link>;<xref rid="bib21" ref-type="bibr"><u>Goodson, 2021</u></xref><xref rid="bib21" ref-type="bibr"><u>Goodson, 2021</u></xref>) (commit 66b2664 used in current versions of the figures).<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [Goodson, 2021](          <person-group person-group-type="author"><name><surname>Goodson</surname><given-names>J</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <data-title>Rnases</data-title>          <version designator="swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220">swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220</version>          <source>Software Heritage</source>          <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:948afe1bbb90435abda675896f596c93577ee420;origin=https://github.com/jgoodson/rnases;visit=swh:1:snp:26ecae94f2928de36d0bdac09fe4eb492c208138;anchor=swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:948afe1bbb90435abda675896f596c93577ee420;origin=https://github.com/jgoodson/rnases;visit=swh:1:snp:26ecae94f2928de36d0bdac09fe4eb492c208138;anchor=swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220</ext-link>        ) [Goodson, 2021](          <person-group person-group-type="author"><name><surname>Goodson</surname><given-names>J</given-names></name></person-group>          <year iso-8601-date="2021">2021</year>          <data-title>Rnases</data-title>          <version designator="swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220">swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220</version>          <source>Software Heritage</source>          <ext-link xlink:href="https://archive.softwareheritage.org/swh:1:dir:948afe1bbb90435abda675896f596c93577ee420;origin=https://github.com/jgoodson/rnases;visit=swh:1:snp:26ecae94f2928de36d0bdac09fe4eb492c208138;anchor=swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:948afe1bbb90435abda675896f596c93577ee420;origin=https://github.com/jgoodson/rnases;visit=swh:1:snp:26ecae94f2928de36d0bdac09fe4eb492c208138;anchor=swh:1:rev:66b266487bfafb3dfc8917dbfd710c9ef7dc0220</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34535742">34535742</a> <b>(2)</b> Circulating tumor DNA is readily detectable among Ghanaian breast cancer patients supporting non-invasive cancer genomic studies in Africa<ul><li>For 0.1x ULPS analysis, ichorCNA (v0.2.0) release version (<ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/ichorCNA">https://github.com/broadinstitute/ichorCNA</ext-link>) was used with default settings (<ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/ichorCNA/blob/master/scripts/<b>snakemake</b>/config/config.yaml">https://github.com/broadinstitute/ichorCNA/blob/master/scripts/<b>snakemake</b>/config/config.yaml</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34534350">34534350</a> <b>(1)</b> TISMO: syngeneic mouse tumor database to model tumor immunity and immunotherapy response<ul><li> RIMA is an automated <b>Snakemake</b> pipeline developed by our group to streamline the processing of RNA-seq data, including but not limited to read alignment, quality control, expression qualification, batch effect removal, and immune cell infiltration inference.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34534350">34534350</a> <b>(1)</b> TISMO: syngeneic mouse tumor database to model tumor immunity and immunotherapy response<ul><li> grinch is a python-based tool, the analysis pipeline of which is built on a <b>snakemake</b> backbone<sup><xref rid="ref-13" ref-type="bibr"><u>13</u></xref></sup>.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(1)</b> TISMO: syngeneic mouse tumor database to model tumor immunity and immunotherapy response<ul><li> The scripts directory could be better organised by separating the <b>snakemake</b> files from the regular python files.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(1)</b> TISMO: syngeneic mouse tumor database to model tumor immunity and immunotherapy response<ul><li> The scripts directory could be better organised by separating the <b>snakemake</b> files from the regular python files.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(7)</b> TISMO: syngeneic mouse tumor database to model tumor immunity and immunotherapy response<ul><li> Struo2 uses <b>snakemake</b> and conda (<xref rid="ref-12" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>), and so there are no dependencies that must be installed prior to pipeline execution besides <b>snakemake</b>, conda, and pandas (for input table loading).<li> Moreover, <b>snakemake</b> allows for efficient job execution and easy scaling on to high performance computing systems.<li> Prior to annotation, the sequence queries are split into <italic toggle="yes">n</italic> batches and run in parallel for faster distributed searching with <b>snakemake</b> (<italic toggle="yes">n</italic> is user-defined).<li> 2B</xref> are the sum of all CPU hours for all <b>snakemake</b> jobs, as recorded <italic toggle="yes">via</italic> <b>snakemake</b>&#x2019;s benchmarking feature.<li> Struo2 utilizes <b>snakemake</b> and conda, which allows for easy installation of all dependencies and simplified scaling to high performance computing systems (<xref rid="ref-12" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34525942">34525942</a> <b>(1)</b> Finding functional associations between prokaryotic virus orthologous groups: a proof of concept<ul><li> We expect that the automated, reproducible <b>snakemake</b>-based [<xref ref-type="bibr" rid="CR36"><u>36</u></xref>] workflow provided through the GitHub repository (see Methods) will help users to readily implement these and other additions and further improve the prediction of functional associations between bacteriophage proteins.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Finding functional associations between prokaryotic virus orthologous groups: a proof of concept<ul><li> Tag-Seq underwent preprocessing <italic>via</italic> a reproducible pipeline using <italic><b>snakemake</b></italic> (<xref rid="B38" ref-type="bibr"><u>38</u></xref>).<li> [38](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake&#x2014;a Scalable Bioinformatics Workflow Engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>19</issue>):<page-range>2520&#x2013;2</page-range>. doi:&#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34521442">34521442</a> <b>(3)</b> PRINCESS: comprehensive detection of haplotype resolved SNVs, SVs, and methylation<ul><li> PRINCESS consists of different modules that are managed by <b>Snakemake</b> [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] enabling straightforward adaptability to local machines, cluster, and cloud environments.<li>PRINCESS is implemented in Python (3.7.6) and uses <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR37"><u>37</u></xref>] version 5.7.1 as its core.<li> Additionally, the Python wrapper we developed around <b>Snakemake</b> accepts inputs from the user to override the current parameter used in the configuration file.<li> [37](K&#xF6;ster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine [Internet]. Bioinformatics. 2018:3600&#x2013;0 Available from: 10.1093/bioinformatics/bty350.) [37](K&#xF6;ster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine [Internet]. Bioinformatics. 2018:3600&#x2013;0 Available from: 10.1093/bioinformatics/bty350.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34521456">34521456</a> <b>(2)</b> Pandora: nucleotide-resolution bacterial pan-genomics with reference graphs<ul><li> A <italic><b>snakemake</b></italic> [<xref ref-type="bibr" rid="CR79"><u>79</u></xref>] pipeline to run the <italic>pandora</italic> workflow with and without de novo discovery (see Fig.<li> A <italic><b>snakemake</b></italic> pipeline to run <italic>snippy</italic>, <italic>SAMtools</italic>, <italic>nanopolish</italic>, and <italic>medaka</italic> on all pairwise combinations of 20 samples and 24 references is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/iqbal-lab-org/variant_callers_pipeline">https://github.com/iqbal-lab-org/variant_callers_pipeline</ext-link>.<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34622162">34622162</a> <b>(1)</b> NK-B cell cross talk induces CXCR5 expression on natural killer cells<ul><li>The analysis workflow for assembly, repeat masking, and annotation was performed using <b>Snakemake</b> (<xref rid="B21" ref-type="bibr"><u>21</u></xref>); it is available online for reproducibility purposes (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/hatimalmutairi/LGAAP" ext-link-type="uri">https://github.com/hatimalmutairi/LGAAP</ext-link>), including the software versions and parameters used (<xref rid="B22" ref-type="bibr"><u>22</u></xref>).<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [22](<person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>."><person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>.</a> <b>(1)</b> NK-B cell cross talk induces CXCR5 expression on natural killer cells<ul><li>The analysis workflow for assembly, repeat masking, and annotation was performed using <b>Snakemake</b> (<xref rid="B16" ref-type="bibr"><u>16</u></xref>); it is available online for reproducibility purposes (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/hatimalmutairi/LGAAP" ext-link-type="uri">https://github.com/hatimalmutairi/LGAAP</ext-link>), including the software versions and parameters used (<xref rid="B17" ref-type="bibr"><u>17</u></xref>).<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [17](<person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>."><person-group person-group-type="author"><string-name><surname>Almutairi</surname><given-names>H</given-names></string-name>, <string-name><surname>Urbaniak</surname><given-names>MD</given-names></string-name>, <string-name><surname>Bates</surname><given-names>MD</given-names></string-name>, <string-name><surname>Jariyapan</surname><given-names>N</given-names></string-name>, <string-name><surname>Kwakye-Nuako</surname><given-names>G</given-names></string-name>, <string-name><surname>Thomaz-Soccol</surname><given-names>V</given-names></string-name>, <string-name><surname>Al-Salem</surname><given-names>WS</given-names></string-name>, <string-name><surname>Dillon</surname><given-names>RJ</given-names></string-name>, <string-name><surname>Bates</surname><given-names>PA</given-names></string-name>, <string-name><surname>Gatherer</surname><given-names>D</given-names></string-name></person-group>. <year>2021</year>. <article-title>LGAAP: <italic toggle="yes">Leishmaniinae</italic> genome assembly and annotation pipeline</article-title>. <source>Microbiol Resour Announc</source><volume>10</volume>:<elocation-id>e00439-21</elocation-id>. doi:<pub-id pub-id-type="doi">10.1128/MRA.00439-21</pub-id>.</a> <b>(1)</b> NK-B cell cross talk induces CXCR5 expression on natural killer cells<ul><li>After initial QCs using FastQC (<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="www.bioinformatics.babraham.ac.uk/projects/fastqc/">www.bioinformatics.babraham.ac.uk/projects/fastqc/</uri>, accessed on 19 March 2021) and FastQ Screen (<uri xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="www.bioinformatics.babraham.ac.uk/projects/fastq_screen/">www.bioinformatics.babraham.ac.uk/projects/fastq_screen/</uri>, accessed on 19 March 2021), fastq files were processed using a local <b>Snakemake</b> workflow including the following main steps.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34804501">34804501</a> <b>(1)</b> Perspectives on automated composition of workflows in the life sciences<ul><li> This enables workflows and associated components to be exchanged between the WorkflowHub, workflow management systems like Galaxy, <b>Snakemake</b> and Nextflow and their repositories, and workflow utilities like OpenEBench and LifeMonitor.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34567846">34567846</a> <b>(5)</b> VGEA: an RNA viral assembly toolkit<ul><li> VGEA makes use of existing bioinformatics pipeline/tools to carry out various viral genome analysis tasks and is built on an advanced workflow management system, <b>Snakemake</b> (<xref rid="ref-20" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li>VGEA makes use of <b>Snakemake</b>&#x2019;s benchmarking feature which allows the measurement of the CPU usage and wall clock time of each rule in the pipeline.<li> Knowledge of this can help the user decide on the number of threads to dedicate to each rule as VGEA also makes use of <b>Snakemake</b>&#x2019;s multi-threading feature.<li>VGEA is built on the <b>snakemake</b> workflow management system (<xref rid="ref-20" ref-type="bibr"><u>BK&#xF6;ster &amp; Rahmann, 2012</u></xref>), a workflow management system that allows the effortless deployment and execution of complex distributed computational workflows in any UNIX-based system, from local machines to high-performance computing clusters.<li> VGEA capitalizes on <b>Snakemake</b>&#x2019;s multi-threading feature so that makes it possible for it to be deployed on laptops with greater computing performance or a computing server to improve its speed.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [BK&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34489462">34489462</a> <b>(8)</b> Chromosome-scale genome sequencing, assembly and annotation of six genomes from subfamily <italic>Leishmaniinae</italic><ul><li>Tools used in analysis workflow with conda or docker link.</p></caption><table frame="hsides" rules="groups"><thead><tr><th>Tool</th><th>Website</th><th>conda or docker link</th></tr></thead><tbody><tr><td>AGAT</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/AGAT">https://github.com/NBISweden/AGAT</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/conda-forge/agate">https://anaconda.org/conda-forge/agate</ext-link></td></tr><tr><td>AUGUSTUS</td><td><ext-link ext-link-type="uri" xlink:href="http://bioinf.uni-greifswald.de/webaugustus/about">http://bioinf.uni-greifswald.de/webaugustus/about</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>BCFtools</td><td><ext-link ext-link-type="uri" xlink:href="http://samtools.github.io/bcftools/">http://samtools.github.io/bcftools/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bcftools">https://anaconda.org/bioconda/bcftools</ext-link></td></tr><tr><td>bedtools</td><td><ext-link ext-link-type="uri" xlink:href="https://bedtools.readthedocs.io/en/latest/">https://bedtools.readthedocs.io/en/latest/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bedtools">https://anaconda.org/bioconda/bedtools</ext-link></td></tr><tr><td>blast+</td><td><ext-link ext-link-type="uri" xlink:href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">https://blast.ncbi.nlm.nih.gov/Blast.cgi</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/blast">https://anaconda.org/bioconda/blast</ext-link></td></tr><tr><td>FastQC</td><td><ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/fastqc">https://anaconda.org/bioconda/fastqc</ext-link></td></tr><tr><td>Flye</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/fenderglass/Flye">https://github.com/fenderglass/Flye</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/flye">https://anaconda.org/bioconda/flye</ext-link></td></tr><tr><td>funannotate</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nextgenusfs/funannotate">https://github.com/nextgenusfs/funannotate</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/funannotate">https://anaconda.org/bioconda/funannotate</ext-link></td></tr><tr><td>GAAS</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/GAAS">https://github.com/NBISweden/GAAS</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/gaas">https://anaconda.org/bioconda/gaas</ext-link></td></tr><tr><td>GeneMark</td><td><ext-link ext-link-type="uri" xlink:href="http://exon.gatech.edu/GeneMark/">http://exon.gatech.edu/GeneMark/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>Genometools</td><td><ext-link ext-link-type="uri" xlink:href="http://genometools.org/">http://genometools.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/genometools-genometools">https://anaconda.org/bioconda/genometools-genometools</ext-link></td></tr><tr><td>interproscan</td><td><ext-link ext-link-type="uri" xlink:href="https://www.ebi.ac.uk/interpro/search/sequence/">https://www.ebi.ac.uk/interpro/search/sequence/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/blaxterlab/interproscan">https://hub.docker.com/r/blaxterlab/interproscan</ext-link></td></tr><tr><td>MAKER2</td><td><ext-link ext-link-type="uri" xlink:href="https://www.yandell-lab.org/software/maker.html">https://www.yandell-lab.org/software/maker.html</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>minimap2</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/lh3/minimap2">https://github.com/lh3/minimap2</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/minimap2">https://anaconda.org/bioconda/minimap2</ext-link></td></tr><tr><td>MultiQC</td><td><ext-link ext-link-type="uri" xlink:href="https://multiqc.info/">https://multiqc.info/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/multiqc">https://anaconda.org/bioconda/multiqc</ext-link></td></tr><tr><td>MUMmer</td><td><ext-link ext-link-type="uri" xlink:href="http://mummer.sourceforge.net/">http://mummer.sourceforge.net/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/mummer">https://anaconda.org/bioconda/mummer</ext-link></td></tr><tr><td>Pilon</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/pilon/wiki">https://github.com/broadinstitute/pilon/wiki</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pilon">https://anaconda.org/bioconda/pilon</ext-link></td></tr><tr><td>pycoQC</td><td><ext-link ext-link-type="uri" xlink:href="https://pypi.org/project/pycoQC/">https://pypi.org/project/pycoQC/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pycoqc">https://anaconda.org/bioconda/pycoqc</ext-link></td></tr><tr><td>RaGOO</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/malonge/RaGOO">https://github.com/malonge/RaGOO</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/imperial-college-research-computing/ragoo">https://anaconda.org/imperial-college-research-computing/ragoo</ext-link></td></tr><tr><td>RepeatMasker</td><td><ext-link ext-link-type="uri" xlink:href="http://www.repeatmasker.org/">http://www.repeatmasker.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>SAMtools</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/samtools">https://anaconda.org/bioconda/samtools</ext-link></td></tr><tr><td><b>Snakemake</b>.<li>Tools used in analysis workflow with conda or docker link.</p></caption><table frame="hsides" rules="groups"><thead><tr><th>Tool</th><th>Website</th><th>conda or docker link</th></tr></thead><tbody><tr><td>AGAT</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/AGAT">https://github.com/NBISweden/AGAT</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/conda-forge/agate">https://anaconda.org/conda-forge/agate</ext-link></td></tr><tr><td>AUGUSTUS</td><td><ext-link ext-link-type="uri" xlink:href="http://bioinf.uni-greifswald.de/webaugustus/about">http://bioinf.uni-greifswald.de/webaugustus/about</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>BCFtools</td><td><ext-link ext-link-type="uri" xlink:href="http://samtools.github.io/bcftools/">http://samtools.github.io/bcftools/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bcftools">https://anaconda.org/bioconda/bcftools</ext-link></td></tr><tr><td>bedtools</td><td><ext-link ext-link-type="uri" xlink:href="https://bedtools.readthedocs.io/en/latest/">https://bedtools.readthedocs.io/en/latest/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bedtools">https://anaconda.org/bioconda/bedtools</ext-link></td></tr><tr><td>blast+</td><td><ext-link ext-link-type="uri" xlink:href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">https://blast.ncbi.nlm.nih.gov/Blast.cgi</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/blast">https://anaconda.org/bioconda/blast</ext-link></td></tr><tr><td>FastQC</td><td><ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/fastqc">https://anaconda.org/bioconda/fastqc</ext-link></td></tr><tr><td>Flye</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/fenderglass/Flye">https://github.com/fenderglass/Flye</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/flye">https://anaconda.org/bioconda/flye</ext-link></td></tr><tr><td>funannotate</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nextgenusfs/funannotate">https://github.com/nextgenusfs/funannotate</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/funannotate">https://anaconda.org/bioconda/funannotate</ext-link></td></tr><tr><td>GAAS</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/GAAS">https://github.com/NBISweden/GAAS</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/gaas">https://anaconda.org/bioconda/gaas</ext-link></td></tr><tr><td>GeneMark</td><td><ext-link ext-link-type="uri" xlink:href="http://exon.gatech.edu/GeneMark/">http://exon.gatech.edu/GeneMark/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>Genometools</td><td><ext-link ext-link-type="uri" xlink:href="http://genometools.org/">http://genometools.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/genometools-genometools">https://anaconda.org/bioconda/genometools-genometools</ext-link></td></tr><tr><td>interproscan</td><td><ext-link ext-link-type="uri" xlink:href="https://www.ebi.ac.uk/interpro/search/sequence/">https://www.ebi.ac.uk/interpro/search/sequence/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/blaxterlab/interproscan">https://hub.docker.com/r/blaxterlab/interproscan</ext-link></td></tr><tr><td>MAKER2</td><td><ext-link ext-link-type="uri" xlink:href="https://www.yandell-lab.org/software/maker.html">https://www.yandell-lab.org/software/maker.html</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>minimap2</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/lh3/minimap2">https://github.com/lh3/minimap2</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/minimap2">https://anaconda.org/bioconda/minimap2</ext-link></td></tr><tr><td>MultiQC</td><td><ext-link ext-link-type="uri" xlink:href="https://multiqc.info/">https://multiqc.info/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/multiqc">https://anaconda.org/bioconda/multiqc</ext-link></td></tr><tr><td>MUMmer</td><td><ext-link ext-link-type="uri" xlink:href="http://mummer.sourceforge.net/">http://mummer.sourceforge.net/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/mummer">https://anaconda.org/bioconda/mummer</ext-link></td></tr><tr><td>Pilon</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/pilon/wiki">https://github.com/broadinstitute/pilon/wiki</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pilon">https://anaconda.org/bioconda/pilon</ext-link></td></tr><tr><td>pycoQC</td><td><ext-link ext-link-type="uri" xlink:href="https://pypi.org/project/pycoQC/">https://pypi.org/project/pycoQC/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pycoqc">https://anaconda.org/bioconda/pycoqc</ext-link></td></tr><tr><td>RaGOO</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/malonge/RaGOO">https://github.com/malonge/RaGOO</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/imperial-college-research-computing/ragoo">https://anaconda.org/imperial-college-research-computing/ragoo</ext-link></td></tr><tr><td>RepeatMasker</td><td><ext-link ext-link-type="uri" xlink:href="http://www.repeatmasker.org/">http://www.repeatmasker.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>SAMtools</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/samtools">https://anaconda.org/bioconda/samtools</ext-link></td></tr><tr><td><b>Snakemake</b></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li>Tools used in analysis workflow with conda or docker link.</p></caption><table frame="hsides" rules="groups"><thead><tr><th>Tool</th><th>Website</th><th>conda or docker link</th></tr></thead><tbody><tr><td>AGAT</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/AGAT">https://github.com/NBISweden/AGAT</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/conda-forge/agate">https://anaconda.org/conda-forge/agate</ext-link></td></tr><tr><td>AUGUSTUS</td><td><ext-link ext-link-type="uri" xlink:href="http://bioinf.uni-greifswald.de/webaugustus/about">http://bioinf.uni-greifswald.de/webaugustus/about</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>BCFtools</td><td><ext-link ext-link-type="uri" xlink:href="http://samtools.github.io/bcftools/">http://samtools.github.io/bcftools/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bcftools">https://anaconda.org/bioconda/bcftools</ext-link></td></tr><tr><td>bedtools</td><td><ext-link ext-link-type="uri" xlink:href="https://bedtools.readthedocs.io/en/latest/">https://bedtools.readthedocs.io/en/latest/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bedtools">https://anaconda.org/bioconda/bedtools</ext-link></td></tr><tr><td>blast+</td><td><ext-link ext-link-type="uri" xlink:href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">https://blast.ncbi.nlm.nih.gov/Blast.cgi</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/blast">https://anaconda.org/bioconda/blast</ext-link></td></tr><tr><td>FastQC</td><td><ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/fastqc">https://anaconda.org/bioconda/fastqc</ext-link></td></tr><tr><td>Flye</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/fenderglass/Flye">https://github.com/fenderglass/Flye</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/flye">https://anaconda.org/bioconda/flye</ext-link></td></tr><tr><td>funannotate</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nextgenusfs/funannotate">https://github.com/nextgenusfs/funannotate</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/funannotate">https://anaconda.org/bioconda/funannotate</ext-link></td></tr><tr><td>GAAS</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/NBISweden/GAAS">https://github.com/NBISweden/GAAS</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/gaas">https://anaconda.org/bioconda/gaas</ext-link></td></tr><tr><td>GeneMark</td><td><ext-link ext-link-type="uri" xlink:href="http://exon.gatech.edu/GeneMark/">http://exon.gatech.edu/GeneMark/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>Genometools</td><td><ext-link ext-link-type="uri" xlink:href="http://genometools.org/">http://genometools.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/genometools-genometools">https://anaconda.org/bioconda/genometools-genometools</ext-link></td></tr><tr><td>interproscan</td><td><ext-link ext-link-type="uri" xlink:href="https://www.ebi.ac.uk/interpro/search/sequence/">https://www.ebi.ac.uk/interpro/search/sequence/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/blaxterlab/interproscan">https://hub.docker.com/r/blaxterlab/interproscan</ext-link></td></tr><tr><td>MAKER2</td><td><ext-link ext-link-type="uri" xlink:href="https://www.yandell-lab.org/software/maker.html">https://www.yandell-lab.org/software/maker.html</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>minimap2</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/lh3/minimap2">https://github.com/lh3/minimap2</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/minimap2">https://anaconda.org/bioconda/minimap2</ext-link></td></tr><tr><td>MultiQC</td><td><ext-link ext-link-type="uri" xlink:href="https://multiqc.info/">https://multiqc.info/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/multiqc">https://anaconda.org/bioconda/multiqc</ext-link></td></tr><tr><td>MUMmer</td><td><ext-link ext-link-type="uri" xlink:href="http://mummer.sourceforge.net/">http://mummer.sourceforge.net/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/mummer">https://anaconda.org/bioconda/mummer</ext-link></td></tr><tr><td>Pilon</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/broadinstitute/pilon/wiki">https://github.com/broadinstitute/pilon/wiki</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pilon">https://anaconda.org/bioconda/pilon</ext-link></td></tr><tr><td>pycoQC</td><td><ext-link ext-link-type="uri" xlink:href="https://pypi.org/project/pycoQC/">https://pypi.org/project/pycoQC/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/pycoqc">https://anaconda.org/bioconda/pycoqc</ext-link></td></tr><tr><td>RaGOO</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/malonge/RaGOO">https://github.com/malonge/RaGOO</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/imperial-college-research-computing/ragoo">https://anaconda.org/imperial-college-research-computing/ragoo</ext-link></td></tr><tr><td>RepeatMasker</td><td><ext-link ext-link-type="uri" xlink:href="http://www.repeatmasker.org/">http://www.repeatmasker.org/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/hatimalmutairi/lmgaap-maker">https://hub.docker.com/r/hatimalmutairi/lmgaap-maker</ext-link></td></tr><tr><td>SAMtools</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/samtools">https://anaconda.org/bioconda/samtools</ext-link></td></tr><tr><td><b>Snakemake</b></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link></td><td><ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/<b>snakemake</b>">https://anaconda.org/bioconda/<b>snakemake</b></ext-link>.<li>To make sure that all assemblies and annotations are reproducible by future investigators, the entire process from obtaining the SRAs<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref>&#x2013;<xref ref-type="bibr" rid="CR91"><u>91</u></xref></sup> to the annotation assignments<sup><xref ref-type="bibr" rid="CR92"><u>92</u></xref>&#x2013;<xref ref-type="bibr" rid="CR97"><u>97</u></xref></sup> has been made available<sup><xref ref-type="bibr" rid="CR29"><u>29</u></xref></sup> using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR98"><u>98</u></xref></sup>.<li> This <b>Snakemake</b> pipeline ought to be easily adaptable to the sequencing of further similar parasite genomes, throughout the parasitology community<sup><xref ref-type="bibr" rid="CR30"><u>30</u></xref></sup>.<li> We checked reproducibility of the assemblies and annotations using <b>Snakemake</b>.<li> [49](          <year>2021</year>          <source>NCBI Sequence Read Archive</source>          <pub-id pub-id-type="accession" assigning-authority="ncbi" xlink:href="https://identifiers.org/ncbi/insdc.sra:SRX9957074">SRX9957074</pub-id>        ) [91](          <year>2021</year>          <source>NCBI Sequence Read Archive</source>          <pub-id pub-id-type="accession" assigning-authority="ncbi" xlink:href="https://identifiers.org/ncbi/insdc.sra:SRX9957044">SRX9957044</pub-id>        ) [92](          <name>            <surname>Almutairi</surname>            <given-names>H</given-names>          </name>          <year>2021</year>          <data-title><italic>L</italic>. (<italic>Mundinia</italic>) <italic>martiniquensis</italic>: functional annotations</data-title>          <source>Lancaster University</source>          <pub-id pub-id-type="doi">10.17635/lancaster/researchdata/446</pub-id>        ) [97](          <name>            <surname>Almutairi</surname>            <given-names>H</given-names>          </name>          <year>2021</year>          <data-title><italic>Porcisia hertigi</italic>: functional annotations</data-title>          <source>Lancaster University</source>          <pub-id pub-id-type="doi">10.17635/lancaster/researchdata/461</pub-id>        ) [29](          <name>            <surname>Almutairi</surname>            <given-names>H</given-names>          </name>          <year>2021</year>          <source>hatimalmutairi/LGAAP.</source>          <pub-id pub-id-type="doi">10.5281/zenodo.4663265</pub-id>        ) [98](M&#xF6;lder, F. <italic>et al</italic>. Sustainable data analysis with Snakemake. <italic>F1000Research</italic><bold>10</bold>, <ext-link ext-link-type="uri" xlink:href="https://f1000research.com/articles/10-33/v2">https://f1000research.com/articles/10-33/v2</ext-link> (2021).) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34292068">34292068</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34292068">34292068</a> <b>(3)</b> Chromosome-scale genome sequencing, assembly and annotation of six genomes from subfamily <italic>Leishmaniinae</italic><ul><li> Additionally, the CoronaSV workflow is defined using <b>Snakemake</b>.<li> Running the CoronaSV.smk <b>snakemake</b> pipeline handles downloading all specified data and processing of sequencing data to variant calls.<li> CoronaSV requires python (version 3.6 or newer) and <b>snakemake</b>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34292068">34292068</a> <b>(1)</b> Chromosome-scale genome sequencing, assembly and annotation of six genomes from subfamily <italic>Leishmaniinae</italic><ul><li><bold>CoronaSV</bold>: The method proposes a <b>Snakemake</b> pipeline for SV discovery in SARS-CoV-2 genomes.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34292068">34292068</a> <b>(1)</b> Chromosome-scale genome sequencing, assembly and annotation of six genomes from subfamily <italic>Leishmaniinae</italic><ul><li><bold>Comment 4: </bold>For the CoronaSV Developers: The method proposes a <b>Snakemake</b> pipeline for SV discovery in SARS-CoV-2 genomes.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34539433">34539433</a> <b>(1)</b> Regulation of the Muscarinic M<sub>3</sub> Receptor by Myocardin-Related Transcription Factors<ul><li> All the code, including a conda environment, <b>Snakemake</b> file, and R markdown notebooks, are available at <ext-link xlink:href="https://github.com/LUDC-bioinformatics/SMC_MYOCD" ext-link-type="uri">https://github.com/LUDC-bioinformatics/SMC_MYOCD</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34539433">34539433</a> <b>(2)</b> Regulation of the Muscarinic M<sub>3</sub> Receptor by Myocardin-Related Transcription Factors<ul><li> To fill this critical gap, we systematically study two large public gene expression repositories, Gene Expression Omnibus (GEO) (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) and ARCHS4 (<xref rid="B19" ref-type="bibr"><u>19</u></xref>), using the previously reported R package <italic toggle="yes">BioQC</italic>, and a reproducible, open-source <b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) workflow employing a new Python package <italic toggle="yes">pygenesig</italic> developed for this study.<li> The analysis is wrapped into a reproducible pipeline built on <italic toggle="yes"><b>Snakemake</b></italic> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=11752295">11752295</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29636450">29636450</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34380046">34380046</a> <b>(10)</b> Whole-body integration of gene expression and single-cell morphology<ul><li> <xref rid="bib87" ref-type="bibr"><u>Shamonin et&#xA0;al., 2014</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/SuperElastix/elastix" id="intref0155">https://github.com/SuperElastix/elastix</ext-link></td></tr><tr><td>ElastixWrapper</td><td><xref rid="bib93" ref-type="bibr"><u>Tischer, 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/embl-cba/elastixWrapper" id="intref0160">https://github.com/embl-cba/elastixWrapper</ext-link></td></tr><tr><td>ImageJ</td><td><xref rid="bib83" ref-type="bibr"><u>Schindelin et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0165">https://imagej.nih.gov/ij/</ext-link></td></tr><tr><td>Ilastik</td><td><xref rid="bib11" ref-type="bibr"><u>Berg et&#xA0;al., 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.ilastik.org/" id="intref0170">https://www.ilastik.org/</ext-link></td></tr><tr><td>Python Louvain</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/taynaud/python-louvain" id="intref0175">https://github.com/taynaud/python-louvain</ext-link></td></tr><tr><td>UMAP</td><td><xref rid="bib56" ref-type="bibr"><u>McInnes et&#xA0;al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://umap-learn.readthedocs.io/en/latest/" id="intref0180">https://umap-learn.readthedocs.io/en/latest/</ext-link></td></tr><tr><td>scikit-image</td><td><xref rid="bib99" ref-type="bibr"><u>van der Walt et&#xA0;al., 2014</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://scikit-image.org/" id="intref0185">https://scikit-image.org/</ext-link></td></tr><tr><td>scikit-learn</td><td><xref rid="bib66" ref-type="bibr"><u>Pedregosa et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://scikit-learn.org" id="intref0190">https://scikit-learn.org</ext-link></td></tr><tr><td>vigra</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="http://ukoethe.github.io/vigra/" id="intref0195">http://ukoethe.github.io/vigra/</ext-link></td></tr><tr><td>mahotas</td><td><xref rid="bib22" ref-type="bibr"><u>Coelho, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://mahotas.readthedocs.io/en/latest/" id="intref0200">https://mahotas.readthedocs.io/en/latest/</ext-link></td></tr><tr><td>networkx</td><td><xref rid="bib31" ref-type="bibr"><u>Hagberg et&#xA0;al., 2008</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://networkx.org/" id="intref0205">https://networkx.org/</ext-link></td></tr><tr><td>pandas</td><td><xref rid="bib57" ref-type="bibr"><u>McKinney, 2010</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://pandas.pydata.org/" id="intref0210">https://pandas.pydata.org/</ext-link></td></tr><tr><td>scipy</td><td><xref rid="bib103" ref-type="bibr"><u>Virtanen et&#xA0;al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.scipy.org/" id="intref0215">https://www.scipy.org/</ext-link></td></tr><tr><td>numpy</td><td><xref rid="bib98" ref-type="bibr"><u>van der Walt et&#xA0;al., 2011</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://numpy.org/" id="intref0220">https://numpy.org/</ext-link></td></tr><tr><td><b>snakemake</b>.<li> <xref rid="bib87" ref-type="bibr"><u>Shamonin et&#xA0;al., 2014</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/SuperElastix/elastix" id="intref0155">https://github.com/SuperElastix/elastix</ext-link></td></tr><tr><td>ElastixWrapper</td><td><xref rid="bib93" ref-type="bibr"><u>Tischer, 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/embl-cba/elastixWrapper" id="intref0160">https://github.com/embl-cba/elastixWrapper</ext-link></td></tr><tr><td>ImageJ</td><td><xref rid="bib83" ref-type="bibr"><u>Schindelin et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0165">https://imagej.nih.gov/ij/</ext-link></td></tr><tr><td>Ilastik</td><td><xref rid="bib11" ref-type="bibr"><u>Berg et&#xA0;al., 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.ilastik.org/" id="intref0170">https://www.ilastik.org/</ext-link></td></tr><tr><td>Python Louvain</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/taynaud/python-louvain" id="intref0175">https://github.com/taynaud/python-louvain</ext-link></td></tr><tr><td>UMAP</td><td><xref rid="bib56" ref-type="bibr"><u>McInnes et&#xA0;al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://umap-learn.readthedocs.io/en/latest/" id="intref0180">https://umap-learn.readthedocs.io/en/latest/</ext-link></td></tr><tr><td>scikit-image</td><td><xref rid="bib99" ref-type="bibr"><u>van der Walt et&#xA0;al., 2014</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://scikit-image.org/" id="intref0185">https://scikit-image.org/</ext-link></td></tr><tr><td>scikit-learn</td><td><xref rid="bib66" ref-type="bibr"><u>Pedregosa et&#xA0;al., 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://scikit-learn.org" id="intref0190">https://scikit-learn.org</ext-link></td></tr><tr><td>vigra</td><td>N/A</td><td><ext-link ext-link-type="uri" xlink:href="http://ukoethe.github.io/vigra/" id="intref0195">http://ukoethe.github.io/vigra/</ext-link></td></tr><tr><td>mahotas</td><td><xref rid="bib22" ref-type="bibr"><u>Coelho, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://mahotas.readthedocs.io/en/latest/" id="intref0200">https://mahotas.readthedocs.io/en/latest/</ext-link></td></tr><tr><td>networkx</td><td><xref rid="bib31" ref-type="bibr"><u>Hagberg et&#xA0;al., 2008</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://networkx.org/" id="intref0205">https://networkx.org/</ext-link></td></tr><tr><td>pandas</td><td><xref rid="bib57" ref-type="bibr"><u>McKinney, 2010</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://pandas.pydata.org/" id="intref0210">https://pandas.pydata.org/</ext-link></td></tr><tr><td>scipy</td><td><xref rid="bib103" ref-type="bibr"><u>Virtanen et&#xA0;al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://www.scipy.org/" id="intref0215">https://www.scipy.org/</ext-link></td></tr><tr><td>numpy</td><td><xref rid="bib98" ref-type="bibr"><u>van der Walt et&#xA0;al., 2011</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://numpy.org/" id="intref0220">https://numpy.org/</ext-link></td></tr><tr><td><b>snakemake</b></td><td><xref rid="bib50" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="intref0225">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li>All the morphological clustering analysis was performed in Python with scikit-image (<xref rid="bib99" ref-type="bibr"><u>van der Walt et&#xA0;al., 2014</u></xref>), vigra (<ext-link ext-link-type="uri" xlink:href="http://ukoethe.github.io/vigra/" id="intref0325">http://ukoethe.github.io/vigra/</ext-link>), scipy (<xref rid="bib103" ref-type="bibr"><u>Virtanen et&#xA0;al., 2020</u></xref>), mahotas (<xref rid="bib22" ref-type="bibr"><u>Coelho, 2012</u></xref>), scikit-learn (<xref rid="bib66" ref-type="bibr"><u>Pedregosa et&#xA0;al., 2012</u></xref>), networkx (<xref rid="bib31" ref-type="bibr"><u>Hagberg et&#xA0;al., 2008</u></xref>), python-louvain (<ext-link ext-link-type="uri" xlink:href="https://github.com/taynaud/python-louvain" id="intref0330">https://github.com/taynaud/python-louvain</ext-link>), umap-learn (<xref rid="bib56" ref-type="bibr"><u>McInnes et&#xA0;al., 2018</u></xref>), pandas (<xref rid="bib57" ref-type="bibr"><u>McKinney, 2010</u></xref>), numpy (<xref rid="bib98" ref-type="bibr"><u>van der Walt et&#xA0;al., 2011</u></xref>) and <b>snakemake</b> (<xref rid="bib50" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> All code is freely available in the github repository (the clustering code as a <b>snakemake</b> workflow).<li> The <b>snakemake</b> workflow also contains code for various other analyses e.g., clustering of different subsets of morphological features, plotting of <italic>Platynereis</italic> regions on the UMAP, plotting gene expression on the UMAP, or construction of various heatmaps.<li>The script for calculation of morphological features is available here: <ext-link ext-link-type="uri" xlink:href="https://github.com/mobie/platybrowser-datasets/blob/master/mmpb/extension/attributes/morphology_impl.py" id="intref0335">https://github.com/mobie/platybrowser-datasets/blob/master/mmpb/extension/attributes/morphology_impl.py</ext-link> and the <b>Snakemake</b> workflow for clustering analysis is available here: <ext-link ext-link-type="uri" xlink:href="https://github.com/mobie/platybrowser-datasets/tree/master/analysis/morphology_clustering" id="intref0340">https://github.com/mobie/platybrowser-datasets/tree/master/analysis/morphology_clustering</ext-link>.<li> All the gene expression clustering analysis was performed in Python with scikit-learn (<xref rid="bib66" ref-type="bibr"><u>Pedregosa et&#xA0;al., 2012</u></xref>), networkx (<xref rid="bib31" ref-type="bibr"><u>Hagberg et&#xA0;al., 2008</u></xref>), python-louvain (<ext-link ext-link-type="uri" xlink:href="https://github.com/taynaud/python-louvain" id="intref0380">https://github.com/taynaud/python-louvain</ext-link>), umap-learn (<xref rid="bib56" ref-type="bibr"><u>McInnes et&#xA0;al., 2018</u></xref>), pandas (<xref rid="bib57" ref-type="bibr"><u>McKinney, 2010</u></xref>), numpy (<xref rid="bib98" ref-type="bibr"><u>van der Walt et&#xA0;al., 2011</u></xref>) and <b>snakemake</b> (<xref rid="bib50" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> The code is freely available as a <b>snakemake</b> workflow: <ext-link ext-link-type="uri" xlink:href="https://github.com/mobie/platybrowser-datasets/tree/master/analysis/gene_clustering" id="intref0385">https://github.com/mobie/platybrowser-datasets/tree/master/analysis/gene_clustering</ext-link>.<li> The <b>snakemake</b> workflow also contains code for various other analyses e.g., clustering of different kinds of gene expression (binarised by a threshold, assignment by virtual cells etc), plotting of <italic>Platynereis</italic> regions on the UMAP, plotting morphology statistics on the UMAP, or construction of various heatmaps.<li> [Shamonin et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24474917">24474917</a>) [Tischer, 2019](          <person-group person-group-type="author">            <name>              <surname>Tischer</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>ElastixWrapper: Fiji plugin for 3D image registration with elastix</article-title>          <source>Zenodo</source>          <year>2019</year>        ) [Schindelin et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22743772">22743772</a>) [Berg et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31570887">31570887</a>) [McInnes et&#xA0;al., 2018](          <person-group person-group-type="author">            <name>              <surname>McInnes</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Healy</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Melville</surname>              <given-names>J.</given-names>            </name>          </person-group>          <article-title>UMAP: uniform manifold approximation and projection for dimension reduction</article-title>          <source>arXiv</source>          <year>2018</year>          <comment>arXiv:1802.03426v3</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1802.03426" id="intref0480">https://arxiv.org/abs/1802.03426</ext-link>        ) [van der Walt et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25024921">25024921</a>) [Pedregosa et&#xA0;al., 2012](          <person-group person-group-type="author">            <name>              <surname>Pedregosa</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Gramfort</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michel</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Thirion</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Grisel</surname>              <given-names>O.</given-names>            </name>            <name>              <surname>Blondel</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>M&#xFC;ller</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Nothman</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Louppe</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>Scikit-learn: machine learning in Python</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1201.0490v4</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1201.0490" id="intref0485">https://arxiv.org/abs/1201.0490</ext-link>        ) [Coelho, 2012](          <person-group person-group-type="author">            <name>              <surname>Coelho</surname>              <given-names>L.P.</given-names>            </name>          </person-group>          <article-title>Mahotas: open source software for scriptable computer vision</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1211.4907v2</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1211.4907" id="intref0460">https://arxiv.org/abs/1211.4907</ext-link>        ) [Hagberg et&#xA0;al., 2008](          <person-group person-group-type="author">            <name>              <surname>Hagberg</surname>              <given-names>A.A.</given-names>            </name>            <name>              <surname>Schult</surname>              <given-names>D.A.</given-names>            </name>            <name>              <surname>Swart</surname>              <given-names>P.J.</given-names>            </name>          </person-group>          <person-group person-group-type="editor">            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Vaught</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <source>Exploring Network Structure, Dynamics, and Function using NetworkX</source>          <year>2008</year>          <publisher-name>USDOE</publisher-name>          <fpage>11</fpage>          <lpage>15</lpage>        ) [McKinney, 2010](          <person-group person-group-type="author">            <name>              <surname>McKinney</surname>              <given-names>W.</given-names>            </name>          </person-group>          <part-title>Data structures for statistical computing in Python</part-title>          <person-group person-group-type="editor">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <series>Proceedings of the 9th Python in Science Conference</series>          <year>2010</year>          <publisher-name>SciPy</publisher-name>          <fpage>51</fpage>          <lpage>56</lpage>        ) [Virtanen et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [van der Walt et&#xA0;al., 2011](          <person-group person-group-type="author">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Colbert</surname>              <given-names>S.C.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>The NumPy Array: a structure for efficient numerical computation</article-title>          <source>Comput. Sci. Eng.</source>          <volume>13</volume>          <year>2011</year>          <fpage>22</fpage>          <lpage>30</lpage>        ) [Shamonin et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24474917">24474917</a>) [Tischer, 2019](          <person-group person-group-type="author">            <name>              <surname>Tischer</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>ElastixWrapper: Fiji plugin for 3D image registration with elastix</article-title>          <source>Zenodo</source>          <year>2019</year>        ) [Schindelin et&#xA0;al., 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22743772">22743772</a>) [Berg et&#xA0;al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31570887">31570887</a>) [McInnes et&#xA0;al., 2018](          <person-group person-group-type="author">            <name>              <surname>McInnes</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Healy</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Melville</surname>              <given-names>J.</given-names>            </name>          </person-group>          <article-title>UMAP: uniform manifold approximation and projection for dimension reduction</article-title>          <source>arXiv</source>          <year>2018</year>          <comment>arXiv:1802.03426v3</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1802.03426" id="intref0480">https://arxiv.org/abs/1802.03426</ext-link>        ) [van der Walt et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25024921">25024921</a>) [Pedregosa et&#xA0;al., 2012](          <person-group person-group-type="author">            <name>              <surname>Pedregosa</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Gramfort</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michel</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Thirion</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Grisel</surname>              <given-names>O.</given-names>            </name>            <name>              <surname>Blondel</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>M&#xFC;ller</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Nothman</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Louppe</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>Scikit-learn: machine learning in Python</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1201.0490v4</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1201.0490" id="intref0485">https://arxiv.org/abs/1201.0490</ext-link>        ) [Coelho, 2012](          <person-group person-group-type="author">            <name>              <surname>Coelho</surname>              <given-names>L.P.</given-names>            </name>          </person-group>          <article-title>Mahotas: open source software for scriptable computer vision</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1211.4907v2</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1211.4907" id="intref0460">https://arxiv.org/abs/1211.4907</ext-link>        ) [Hagberg et&#xA0;al., 2008](          <person-group person-group-type="author">            <name>              <surname>Hagberg</surname>              <given-names>A.A.</given-names>            </name>            <name>              <surname>Schult</surname>              <given-names>D.A.</given-names>            </name>            <name>              <surname>Swart</surname>              <given-names>P.J.</given-names>            </name>          </person-group>          <person-group person-group-type="editor">            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Vaught</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <source>Exploring Network Structure, Dynamics, and Function using NetworkX</source>          <year>2008</year>          <publisher-name>USDOE</publisher-name>          <fpage>11</fpage>          <lpage>15</lpage>        ) [McKinney, 2010](          <person-group person-group-type="author">            <name>              <surname>McKinney</surname>              <given-names>W.</given-names>            </name>          </person-group>          <part-title>Data structures for statistical computing in Python</part-title>          <person-group person-group-type="editor">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <series>Proceedings of the 9th Python in Science Conference</series>          <year>2010</year>          <publisher-name>SciPy</publisher-name>          <fpage>51</fpage>          <lpage>56</lpage>        ) [Virtanen et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [van der Walt et&#xA0;al., 2011](          <person-group person-group-type="author">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Colbert</surname>              <given-names>S.C.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>The NumPy Array: a structure for efficient numerical computation</article-title>          <source>Comput. Sci. Eng.</source>          <volume>13</volume>          <year>2011</year>          <fpage>22</fpage>          <lpage>30</lpage>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [van der Walt et&#xA0;al., 2014](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25024921">25024921</a>) [Virtanen et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32015543">32015543</a>) [Coelho, 2012](          <person-group person-group-type="author">            <name>              <surname>Coelho</surname>              <given-names>L.P.</given-names>            </name>          </person-group>          <article-title>Mahotas: open source software for scriptable computer vision</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1211.4907v2</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1211.4907" id="intref0460">https://arxiv.org/abs/1211.4907</ext-link>        ) [Pedregosa et&#xA0;al., 2012](          <person-group person-group-type="author">            <name>              <surname>Pedregosa</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Gramfort</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michel</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Thirion</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Grisel</surname>              <given-names>O.</given-names>            </name>            <name>              <surname>Blondel</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>M&#xFC;ller</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Nothman</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Louppe</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>Scikit-learn: machine learning in Python</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1201.0490v4</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1201.0490" id="intref0485">https://arxiv.org/abs/1201.0490</ext-link>        ) [Hagberg et&#xA0;al., 2008](          <person-group person-group-type="author">            <name>              <surname>Hagberg</surname>              <given-names>A.A.</given-names>            </name>            <name>              <surname>Schult</surname>              <given-names>D.A.</given-names>            </name>            <name>              <surname>Swart</surname>              <given-names>P.J.</given-names>            </name>          </person-group>          <person-group person-group-type="editor">            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Vaught</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <source>Exploring Network Structure, Dynamics, and Function using NetworkX</source>          <year>2008</year>          <publisher-name>USDOE</publisher-name>          <fpage>11</fpage>          <lpage>15</lpage>        ) [McInnes et&#xA0;al., 2018](          <person-group person-group-type="author">            <name>              <surname>McInnes</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Healy</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Melville</surname>              <given-names>J.</given-names>            </name>          </person-group>          <article-title>UMAP: uniform manifold approximation and projection for dimension reduction</article-title>          <source>arXiv</source>          <year>2018</year>          <comment>arXiv:1802.03426v3</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1802.03426" id="intref0480">https://arxiv.org/abs/1802.03426</ext-link>        ) [McKinney, 2010](          <person-group person-group-type="author">            <name>              <surname>McKinney</surname>              <given-names>W.</given-names>            </name>          </person-group>          <part-title>Data structures for statistical computing in Python</part-title>          <person-group person-group-type="editor">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <series>Proceedings of the 9th Python in Science Conference</series>          <year>2010</year>          <publisher-name>SciPy</publisher-name>          <fpage>51</fpage>          <lpage>56</lpage>        ) [van der Walt et&#xA0;al., 2011](          <person-group person-group-type="author">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Colbert</surname>              <given-names>S.C.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>The NumPy Array: a structure for efficient numerical computation</article-title>          <source>Comput. Sci. Eng.</source>          <volume>13</volume>          <year>2011</year>          <fpage>22</fpage>          <lpage>30</lpage>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Pedregosa et&#xA0;al., 2012](          <person-group person-group-type="author">            <name>              <surname>Pedregosa</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Gramfort</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michel</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Thirion</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Grisel</surname>              <given-names>O.</given-names>            </name>            <name>              <surname>Blondel</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>M&#xFC;ller</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Nothman</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Louppe</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>Scikit-learn: machine learning in Python</article-title>          <source>arXiv</source>          <year>2012</year>          <comment>arXiv:1201.0490v4</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1201.0490" id="intref0485">https://arxiv.org/abs/1201.0490</ext-link>        ) [Hagberg et&#xA0;al., 2008](          <person-group person-group-type="author">            <name>              <surname>Hagberg</surname>              <given-names>A.A.</given-names>            </name>            <name>              <surname>Schult</surname>              <given-names>D.A.</given-names>            </name>            <name>              <surname>Swart</surname>              <given-names>P.J.</given-names>            </name>          </person-group>          <person-group person-group-type="editor">            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>            <name>              <surname>Vaught</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <source>Exploring Network Structure, Dynamics, and Function using NetworkX</source>          <year>2008</year>          <publisher-name>USDOE</publisher-name>          <fpage>11</fpage>          <lpage>15</lpage>        ) [McInnes et&#xA0;al., 2018](          <person-group person-group-type="author">            <name>              <surname>McInnes</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Healy</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Melville</surname>              <given-names>J.</given-names>            </name>          </person-group>          <article-title>UMAP: uniform manifold approximation and projection for dimension reduction</article-title>          <source>arXiv</source>          <year>2018</year>          <comment>arXiv:1802.03426v3</comment>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1802.03426" id="intref0480">https://arxiv.org/abs/1802.03426</ext-link>        ) [McKinney, 2010](          <person-group person-group-type="author">            <name>              <surname>McKinney</surname>              <given-names>W.</given-names>            </name>          </person-group>          <part-title>Data structures for statistical computing in Python</part-title>          <person-group person-group-type="editor">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Millman</surname>              <given-names>J.</given-names>            </name>          </person-group>          <series>Proceedings of the 9th Python in Science Conference</series>          <year>2010</year>          <publisher-name>SciPy</publisher-name>          <fpage>51</fpage>          <lpage>56</lpage>        ) [van der Walt et&#xA0;al., 2011](          <person-group person-group-type="author">            <name>              <surname>van der Walt</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Colbert</surname>              <given-names>S.C.</given-names>            </name>            <name>              <surname>Varoquaux</surname>              <given-names>G.</given-names>            </name>          </person-group>          <article-title>The NumPy Array: a structure for efficient numerical computation</article-title>          <source>Comput. Sci. Eng.</source>          <volume>13</volume>          <year>2011</year>          <fpage>22</fpage>          <lpage>30</lpage>        ) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34380047">34380047</a> <b>(3)</b> TDP-43 condensation properties specify its RNA-binding and regulatory repertoire<ul><li><hr/></td></tr><tr><td>Fiji/ImageJ</td><td><xref rid="bib59" ref-type="bibr"><u>Rueden et&#xA0;al., 2017</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0025">https://imagej.nih.gov/ij/</ext-link></td></tr><tr><td>GraphPad Prism 5</td><td>GraphPad Software</td><td><ext-link ext-link-type="uri" xlink:href="https://www.graphpad.com/scientific-software/prism/" id="intref0030">https://www.graphpad.com/scientific-software/prism/</ext-link></td></tr><tr><td>R v 4.0.3</td><td>The R Project for Statistical Computing</td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0035">https://www.r-project.org/</ext-link></td></tr><tr><td>Python v 3.7</td><td>Python Software Foundation</td><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org" id="intref0040">https://www.python.org</ext-link></td></tr><tr><td><b>Snakemake</b> v 5.31.1.<li><hr/></td></tr><tr><td>Fiji/ImageJ</td><td><xref rid="bib59" ref-type="bibr"><u>Rueden et&#xA0;al., 2017</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://imagej.nih.gov/ij/" id="intref0025">https://imagej.nih.gov/ij/</ext-link></td></tr><tr><td>GraphPad Prism 5</td><td>GraphPad Software</td><td><ext-link ext-link-type="uri" xlink:href="https://www.graphpad.com/scientific-software/prism/" id="intref0030">https://www.graphpad.com/scientific-software/prism/</ext-link></td></tr><tr><td>R v 4.0.3</td><td>The R Project for Statistical Computing</td><td><ext-link ext-link-type="uri" xlink:href="https://www.r-project.org/" id="intref0035">https://www.r-project.org/</ext-link></td></tr><tr><td>Python v 3.7</td><td>Python Software Foundation</td><td><ext-link ext-link-type="uri" xlink:href="https://www.python.org" id="intref0040">https://www.python.org</ext-link></td></tr><tr><td><b>Snakemake</b> v 5.31.1</td><td><xref rid="bib47" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io/" id="intref0045">https://<b>snakemake</b>.github.io/</ext-link>.<li> [Rueden et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29187165">29187165</a>) [Rueden et&#xA0;al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29187165">29187165</a>) [M&#xF6;lder et&#xA0;al., 2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34470617">34470617</a> <b>(2)</b> A globally diverse reference alignment and panel for imputation of mitochondrial DNA variants<ul><li> In addition, we describe a <b>SnakeMake</b> pipeline, which we developed for easy imputation of mtSNVs through the IMPUTE2 framework [<xref ref-type="bibr" rid="CR20"><u>20</u></xref>].<li>Pipelines for implementing our imputation pipeline and reproducing our results were initially created in BASH shell scripts then lifted over into <b>SnakeMake</b> [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>] for the MitoImpute pipeline.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19543373">19543373</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34469576">34469576</a> <b>(1)</b> <italic toggle="yes">Drosophila</italic> Evolution over Space and Time (DEST): A New PopulationGenomics Resource<ul><li> The SNP callingstep is built using the <italic toggle="yes"><b>snakemake</b></italic> (<xref rid="msab259-B97" ref-type="bibr">M&#xF6;lder et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34469717">34469717</a> <b>(1)</b> Intra- and extra-cellular environments contribute to the fate of HIV-1 infection<ul><li> RNaseq analysis was performed using the VIPER <b>snakemake</b> pipeline (<xref rid="R10" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34465897">34465897</a> <b>(1)</b> Genome-driven elucidation of phage-host interplay and impact of phage resistance evolution on bacterial fitness<ul><li> The analysis pipeline was automated using the tool <b>Snakemake</b> v5.4.4 [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>].<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34461978">34461978</a> <b>(5)</b> GenTB: A user-friendly genome-based predictor for tuberculosis resistance powered by machine learning<ul><li> The GenTB analysis pipeline is also available for command-line use wrapped in <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>].<li> For batch processing of larger numbers of raw sequence data, we provide a command-line GenTB workflow based on <italic><b>Snakemake</b></italic> v5.20.1 [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>] where dependent software will be sourced via conda [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>].<li> The <italic><b>Snakemake</b></italic> workflow can be accessed via Github (<ext-link ext-link-type="uri" xlink:href="https://github.com/farhat-lab/gentb-<b>snakemake</b>">https://github.com/farhat-lab/gentb-<b>snakemake</b></ext-link>) [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>].<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [27](Matthias Gr&#xF6;schel and Martin Owens. gentb-snakemake. 2021. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/farhat-lab/gentb-snakemake">https://github.com/farhat-lab/gentb-snakemake</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34454437">34454437</a> <b>(1)</b> Associations between the gut microbiome and metabolome in early life<ul><li> <b>Snakemake</b> was used to construct the computational pipeline [<xref ref-type="bibr" rid="CR99"><u>99</u></xref>].<li> [99](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34453168">34453168</a> <b>(1)</b> Functional meta-omics provide critical insights into long- and short-read assemblies<ul><li><b>Snakemake</b> (v.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34534285">34534285</a> <b>(2)</b> Insm1, Neurod1, and Pax6 promote murine pancreatic endocrine cell development through overlapping yet distinct RNA transcription and splicing programs<ul><li> <b>Snakemake</b> (5.2.4), a workflow management system, was used to manage the data processing.<li> <b>Snakemake</b> (5.2.4), a workflow management system, was used to call five programs necessary to process the samples.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34868553">34868553</a> <b>(1)</b> A signaling pathway-driven bioinformatics pipeline for predicting therapeutics against emerging infectious diseases<ul><li> Briefly, this automated <b>snakemake</b>-based workflow performs quality control of the reads with fastQC (<ext-link xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" ext-link-type="uri">www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>), trims the adapters and poor-quality regions with TrimGalore! (<ext-link xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" ext-link-type="uri">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</ext-link>), maps &amp; quantifies reads to the human transcriptome using Salmon<sup><xref rid="ref-26" ref-type="bibr"><u>26</u></xref></sup>, performs differential expression using edgeR<sup><xref rid="ref-27" ref-type="bibr"><u>27</u></xref></sup>, identifies differential transcript usage with DRIMseq<sup><xref rid="ref-28" ref-type="bibr"><u>28</u></xref></sup>, and enriches on Gene Ontology terms<sup><xref rid="ref-29" ref-type="bibr"><u>29</u></xref></sup> and Hallmark gene sets<sup><xref rid="ref-30" ref-type="bibr"><u>30</u></xref></sup> using the Camera algorithm to adjust for inter-gene correlation<sup><xref rid="ref-31" ref-type="bibr"><u>31</u></xref></sup>.<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28263959">28263959</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19910308">19910308</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28105305">28105305</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30395331">30395331</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26771021">26771021</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22638577">22638577</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34489998">34489998</a> <b>(1)</b> Direct and Indirect Transcriptional Effects of Abiotic Stress in <italic>Zea mays</italic> Plants Defective in RNA-Directed DNA Methylation<ul><li> To ensure a consistent and re-producible computation environment, the workflow was containerized using Singularity (3.6.4) (Kurtzer and Sochat, <xref rid="B22" ref-type="bibr"><u>2017</u></xref>) and the data workflow steps were defined using <b>Snakemake</b> (5.30.1) (Koster and Rahmann, <xref rid="B21" ref-type="bibr"><u>2018</u></xref>) and read quality control was assessed using fastp (0.21.0) (Chen et al., <xref rid="B4" ref-type="bibr"><u>2018</u></xref>).<li> [2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30423086">30423086</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34467227">34467227</a> <b>(4)</b> Chromatin and transcription factor profiling in rare stem cell populations using CUT&amp;Tag<ul><li>A pipeline has been established in Github for users to analyze their CUT&amp;Tag sequencing data (<ext-link ext-link-type="uri" xlink:href="https://github.com/dilworthlab/CnT_pipeline_<b>snakemake</b>" id="intref0015">https://github.com/dilworthlab/CnT_pipeline_<b>snakemake</b></ext-link>).<li>A pipeline has been established in Github for users to analyze their CUT&amp;Tag sequencing data (<ext-link ext-link-type="uri" xlink:href="https://github.com/dilworthlab/CnT_pipeline_<b>snakemake</b>" id="intref0070">https://github.com/dilworthlab/CnT_pipeline_<b>snakemake</b></ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34407780">34407780</a> <b>(1)</b> The CRCbiome study: a large prospective cohort study examining the role of lifestyle and the gut microbiome in colorectal cancer screening participants<ul><li> The availability of a Conda package, which no other related deep-learning tool offers to our knowledge, also allows for easy integration into larger workflows, such as <b>Snakemake</b> pipelines [<xref rid="bib22" ref-type="bibr"><u>22</u></xref>] or inside the Galaxy framework [<xref rid="bib23" ref-type="bibr"><u>23</u></xref>].<li> [22](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34404355">34404355</a> <b>(1)</b> NOREC4DNA: using near-optimal rateless erasure codes for DNA storage<ul><li> While most modern sequencing pipelines perform these steps accordingly, we use the <b>Snakemake</b>-based pipeline &#x2018;Natrix&#x2019;&#xA0;[<xref ref-type="bibr" rid="CR33"><u>33</u></xref>].<li> [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33198651">33198651</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34484161">34484161</a> <b>(2)</b> DAnIEL: A User-Friendly Web Server for Fungal ITS Amplicon Sequencing Data<ul><li> The back-end is built as a <b>Snakemake</b> workflow (<xref rid="ref17" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li>Using the <b>Snakemake</b> workflow engine, DAnIEL can be easily extended by other steps (<xref rid="ref17" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> DAnIEL: A User-Friendly Web Server for Fungal ITS Amplicon Sequencing Data<ul><li>Analysis of ribosome profiling data was performed using an in-house <b>snakemake</b> based pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34398234">34398234</a> <b>(2)</b> Recovery of Deleted Deep Sequencing Data Sheds More Light on the Early Wuhan SARS-CoV-2 Epidemic<ul><li> This GitHub repository includes a <b>Snakemake</b> (<xref rid="msab246-B26" ref-type="bibr">M&#xF6;lder et al.<li> 2018</xref>) to trim reads and remove low-quality ones (the exact settings using in this preprocessing are specified in the <b>Snakemake</b> file in the GitHub repository).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34388963">34388963</a> <b>(10)</b> OVarFlow: a resource optimized GATK&#xA0;4 based Open source Variant calling workFlow<ul><li> <b>Snakemake</b> is utilized as a workflow management engine to orchestrate the various data processing steps [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>].<li> Here, <b>Snakemake</b> helps to achieve optimized resource utilization and parallelization from high-end desktops to server to cluster usage [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>].<li> If present, this file is automatically picked up by the <b>Snakemake</b> workflow management system.<li> Parallelization solely relied on <b>Snakemake</b>&#x2019;s scheduling of various processes to be executed simultaneously, and by bwa mem to use six threads for each distinct mapping process.<li> The <b>Snakemake</b> workflow management system, on the other hand, is only capable of calculating and scheduling entire threads.<li> With this setting, the <b>Snakemake</b> scheduler only needed to allocate a single thread per HaplotypeCaller, effectively doubling the number of HaplotypeCaller processes executed in parallel.<li> In this case <b>Snakemake</b> automatically schedules two threads per HaplotypeCaller process.<li> Nevertheless, it can be configured to use just a single native pairHMM thread, in which case <b>Snakemake</b> reserves only a single thread.<li> <b>Snakemake</b> [<xref ref-type="bibr" rid="CR31"><u>31</u></xref>] version 5.26.1 served as a workflow management engine.<li> <b>Snakemake</b>; Bash.<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34389059">34389059</a> <b>(1)</b> Metapangenomics reveals depth-dependent shifts in metabolic potential for the ubiquitous marine bacterial SAR324 lineage<ul><li>Metagenomes were prepared and assembled using a custom workflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/jmeppley/workflows/tree/Aloha_2_0.02">https://github.com/jmeppley/workflows/tree/Aloha_2_0.02</ext-link>) executed by <b>Snakemake</b> ([<xref ref-type="bibr" rid="CR65"><u>65</u></xref>] v3.11.0).<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34458349">34458349</a> <b>(1)</b> Exploring With Transcriptomic Approaches the Underlying Mechanisms of an Essential Oil-Based Phytogenic in the Small Intestine and Liver of Pigs<ul><li>The RNA-Seq pipeline was implemented <italic>via</italic> the <b>Snakemake</b> workflow management system [version 5.2.4; (<xref rid="B51" ref-type="bibr"><u>51</u></xref>)].<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(2)</b> Exploring With Transcriptomic Approaches the Underlying Mechanisms of an Essential Oil-Based Phytogenic in the Small Intestine and Liver of Pigs<ul><li> Nextflow (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) or <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>)), (iii) is straightforward to install via static binaries available for most platforms and architectures, and (iv) provides a public API accessible to developers wanting to manipulate phylogenetic trees and multiple sequence&#xA0;alignments in Go.<li>The Gotree/Goalign toolkit has been developed to be easily integrated in phylogenetic workflows, using main workflow managers such as Nextflow (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) or <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>).<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34369258">34369258</a> <b>(1)</b> Disentangling tumorigenesis-associated DNA methylation changes in colorectal tissues from those associated with ageing<ul><li><b>Snakemake</b> [<xref rid="ppat.1009786.ref038" ref-type="bibr"><u>38</u></xref>] pipelines and ad hoc Python3 scripts used for the analyses described above are available on <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://github.com/evolbioinfo/HIV1-CRF19_Cuba" ext-link-type="uri">github.com/evolbioinfo/HIV1-CRF19_Cuba</ext-link>.<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34353948">34353948</a> <b>(1)</b> <italic>De novo</italic> assembly, annotation, and comparative analysis of 26 diverse maize genomes<ul><li> A solution for this is to use a workflow language such as Workflow Definition Language (WDL) [<xref rid="pcbi.1009207.ref013" ref-type="bibr"><u>13</u></xref>], Common Workflow Language (CWL) [<xref rid="pcbi.1009207.ref014" ref-type="bibr"><u>14</u></xref>], Nextflow [<xref rid="pcbi.1009207.ref015" ref-type="bibr"><u>15</u></xref>], or <b>Snakemake</b>, an extension to the Python language [<xref rid="pcbi.1009207.ref016" ref-type="bibr"><u>16</u></xref>].<li> [13](Broad Institute. WDL (Workflow Definition Language) specification and documentation. 2016. Available from: <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://software.broadinstitute.org/wdl/documentation/spec" ext-link-type="uri">https://software.broadinstitute.org/wdl/documentation/spec</ext-link>.) [14](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Andeer</surname><given-names>R</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Valls Guimera</surname><given-names>R</given-names></name>, <etal>et al</etal>. <source>Common Workflow Language</source>, Draft 3. <year>2016</year>.) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [16](Python. Snakemake documentation. 2021. Available from: <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://snakemake.readthedocs.io/en/stable/index.html" ext-link-type="uri">https://snakemake.readthedocs.io/en/stable/index.html</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34342226">34342226</a> <b>(2)</b> ppx: Programmatic access to proteomics data repositories<ul><li> The adoption of workflow engines such as <b>Snakemake</b> [<xref rid="R26" ref-type="bibr"><u>26</u></xref>], Nextflow [<xref rid="R27" ref-type="bibr"><u>27</u></xref>], and Cromwell [<xref rid="R28" ref-type="bibr"><u>28</u></xref>] provide frameworks to run analysis pipelines reproducibly and orchestrate the necessary environments or containers.<li>Critically, every step of our reanalysis&#x2014;downloading the raw data from PRIDE with ppx, searching with ANN-SoLo, downloading the original analysis results from PRIDE with ppx, and the creation of our figures with Python&#x2014;is fully encapsulated in a <b>Snakemake</b> workflow that is publicly available at <ext-link ext-link-type="uri" xlink:href="https://github.com/Noble-Lab/ppx-workflow">https://github.com/Noble-Lab/ppx-workflow</ext-link>.<li> [26](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> and <name><surname>Rahmann</surname><given-names>S</given-names></name> &#x201C;<article-title>Snakemake&#x2013;a Scalable Bioinformatics Workflow Engine</article-title>.&#x201D; eng. In: <source>Bioinformatics (Oxford, England)</source><volume>28</volume>.<issue>19</issue> (<month>10</month>. <year>2012</year>), pp. <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>.) [27](<name><surname>Di Tommaso</surname><given-names>P</given-names></name>, <name><surname>Chatzou</surname><given-names>M</given-names></name>, <name><surname>Floden</surname><given-names>EW</given-names></name>, <name><surname>Barja</surname><given-names>PP</given-names></name>, <etal/> &#x201C;<article-title>Nextflow Enables Reproducible Computational Workflows</article-title>.&#x201D; eng. In: <source>Nature Biotechnology</source><volume>35</volume>.<issue>4</issue> (<month>4</month>. <year>2017</year>), pp. <fpage>316</fpage>&#x2013;<lpage>319</lpage>.) [28](<name><surname>Voss</surname><given-names>K</given-names></name>, <name><surname>Auwera</surname><given-names>GVD</given-names></name>, and <name><surname>Gentry</surname><given-names>J</given-names></name><article-title>Full-Stack Genomics Pipelining with GATK4 + WDL + Cromwell [Version 1; Not Peer Reviewed]</article-title>. Vol. <volume>6</volume>. <source>F1000Research</source>, <year>2017</year>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34343564">34343564</a> <b>(1)</b> Integrated multiomics analysis identifies molecular landscape perturbations during hyperammonemia in skeletal muscle and myotubes<ul><li> In brief, <b>Snakemake</b> was used to manage the bioinformatics workflow (<xref rid="bib58" ref-type="bibr"><u>58</u></xref>).<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34439964">34439964</a> <b>(4)</b> Sequencing and Chromosome-Scale Assembly of Plant Genomes, <italic>Brassica rapa</italic> as a Use Case<ul><li> Then, we used the Pore-C <b>Snakemake</b> [<xref rid="B43-biology-10-00732" ref-type="bibr"><u>43</u></xref>] pipeline (git commit 6b2f762) developed by ONT to generate the necessary files for scaffolding with the Salsa2 [<xref rid="B44-biology-10-00732" ref-type="bibr"><u>44</u></xref>] scaffolder.<li> Both Salsa2 and the Pore-C <b>Snakemake</b> pipeline were launched with default parameters.<li> As advised by the developers of the Pore-C <b>Snakemake</b> pipeline, we removed reads that were longer than 100 Kb and obtained a resulting dataset composed of 5.8 million reads with an N50 of 3.97 Kb, for a total cumulative size of 15.50 Gb.<li>We applied the Pore-C <b>Snakemake</b> pipeline released by Oxford Nanopore, as well as Salsa2 to scaffold the Necat assembly (<xref rid="biology-10-00732-t001" ref-type="table"><u>Table 1</u></xref>), and obtained 253 scaffolds for a total cumulative size of 443 Mb, with a scaffold N50 of 20.1 Mb.<li> [43](          <person-group person-group-type="author">            <collab>Github</collab>          </person-group>          <article-title>Pore-C-Snakemake</article-title>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://github.com/nanoporetech/Pore-C-Snakemake">https://github.com/nanoporetech/Pore-C-Snakemake</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2021-07-30">(accessed on 30 July 2021)</date-in-citation>        ) [44](          <person-group person-group-type="author">            <name>              <surname>Ghurye</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Pop</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Koren</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Bickhart</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Chin</surname>              <given-names>C.-S.</given-names>            </name>          </person-group>          <article-title>Scaffolding of Long Read Assemblies Using Long Range Contact Information</article-title>          <source>BMC Genom.</source>          <year>2017</year>          <volume>18</volume>          <elocation-id>527</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12864-017-3879-z</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author">            <name>              <surname>Ghurye</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Pop</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Koren</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Bickhart</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Chin</surname>              <given-names>C.-S.</given-names>            </name>          </person-group>          <article-title>Scaffolding of Long Read Assemblies Using Long Range Contact Information</article-title>          <source>BMC Genom.</source>          <year>2017</year>          <volume>18</volume>          <elocation-id>527</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12864-017-3879-z</pub-id>        ">          <person-group person-group-type="author">            <name>              <surname>Ghurye</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Pop</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Koren</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Bickhart</surname>              <given-names>D.</given-names>            </name>            <name>              <surname>Chin</surname>              <given-names>C.-S.</given-names>            </name>          </person-group>          <article-title>Scaffolding of Long Read Assemblies Using Long Range Contact Information</article-title>          <source>BMC Genom.</source>          <year>2017</year>          <volume>18</volume>          <elocation-id>527</elocation-id>          <pub-id pub-id-type="doi">10.1186/s12864-017-3879-z</pub-id>        </a> <b>(1)</b> Sequencing and Chromosome-Scale Assembly of Plant Genomes, <italic>Brassica rapa</italic> as a Use Case<ul><li>Pangolin is a python-based assignment pipeline built using <b>Snakemake</b> (<xref rid="R8" ref-type="bibr">Koster et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34394104">34394104</a> <b>(2)</b> RGC-32 Acts as a Hub to Regulate the Transcriptomic Changes Associated With Astrocyte Development and Reactive Astrocytosis<ul><li><b>Snakemake</b><sup><xref rid="bibr45-11779322211035921" ref-type="bibr"><u>45</u></xref>,<xref rid="bibr46-11779322211035921" ref-type="bibr"><u>46</u></xref></sup>.<li>Automatic translation of any CWL workflow definition into a<b>Snakemake</b> workflow not yet implemented.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(1)</b> RGC-32 Acts as a Hub to Regulate the Transcriptomic Changes Associated With Astrocyte Development and Reactive Astrocytosis<ul><li> We provide an example using <b>Snakemake</b> (<xref rid="msab199-B10" ref-type="bibr">M&#xF6;lder et al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34508652">34508652</a> <b>(4)</b> Emergence of an early SARS-CoV-2 epidemic in the United States<ul><li><hr/></td></tr><tr><td>Pangolin v2.0</td><td><xref rid="bib50" ref-type="bibr"><u>Rambaut et&#xA0;al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/cov-lineages/pangolin" id="intref0025">https://github.com/cov-lineages/pangolin</ext-link></td></tr><tr><td>NextClade v0.12.0</td><td><xref rid="bib64" ref-type="bibr"><u>Hadfield et al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nextstrain/nextclade" id="intref0030">https://github.com/nextstrain/nextclade</ext-link></td></tr><tr><td>IQtree2</td><td><xref rid="bib65" ref-type="bibr"><u>Minh et al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/iqtree/iqtree2" id="intref0035">https://github.com/iqtree/iqtree2</ext-link></td></tr><tr><td>BEASTv1.10.5pre</td><td><xref rid="bib54" ref-type="bibr"><u>Suchard et&#xA0;al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/beast-dev/beast-mcmc/tree/v1.10.5pre_thorney_v0.1.0" id="intref0040">https://github.com/beast-dev/beast-mcmc/tree/v1.10.5pre_thorney_v0.1.0</ext-link></td></tr><tr><td>BEAGLE</td><td><xref rid="bib66" ref-type="bibr"><u>Ayres et al., 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://faculty.washington.edu/browning/beagle/beagle.html#download" id="intref0045">https://faculty.washington.edu/browning/beagle/beagle.html#download</ext-link></td></tr><tr><td>Baltic</td><td>GitHub</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/evogytis/baltic" id="intref0050">https://github.com/evogytis/baltic</ext-link></td></tr><tr><td><b>Snakemake</b>.<li><hr/></td></tr><tr><td>Pangolin v2.0</td><td><xref rid="bib50" ref-type="bibr"><u>Rambaut et&#xA0;al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/cov-lineages/pangolin" id="intref0025">https://github.com/cov-lineages/pangolin</ext-link></td></tr><tr><td>NextClade v0.12.0</td><td><xref rid="bib64" ref-type="bibr"><u>Hadfield et al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/nextstrain/nextclade" id="intref0030">https://github.com/nextstrain/nextclade</ext-link></td></tr><tr><td>IQtree2</td><td><xref rid="bib65" ref-type="bibr"><u>Minh et al., 2020</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/iqtree/iqtree2" id="intref0035">https://github.com/iqtree/iqtree2</ext-link></td></tr><tr><td>BEASTv1.10.5pre</td><td><xref rid="bib54" ref-type="bibr"><u>Suchard et&#xA0;al., 2018</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/beast-dev/beast-mcmc/tree/v1.10.5pre_thorney_v0.1.0" id="intref0040">https://github.com/beast-dev/beast-mcmc/tree/v1.10.5pre_thorney_v0.1.0</ext-link></td></tr><tr><td>BEAGLE</td><td><xref rid="bib66" ref-type="bibr"><u>Ayres et al., 2019</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://faculty.washington.edu/browning/beagle/beagle.html#download" id="intref0045">https://faculty.washington.edu/browning/beagle/beagle.html#download</ext-link></td></tr><tr><td>Baltic</td><td>GitHub</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/evogytis/baltic" id="intref0050">https://github.com/evogytis/baltic</ext-link></td></tr><tr><td><b>Snakemake</b></td><td><xref rid="bib28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="intref0055">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li>Consensus sequences were assembled using an inhouse <b>Snakemake</b> (<xref rid="bib28" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) pipeline with bwa-mem (<xref rid="bib35" ref-type="bibr"><u>Li, 2013</u></xref>) and iVar v1.2.2 (<xref rid="bib24" ref-type="bibr"><u>Grubaugh et&#xA0;al., 2019b</u></xref>; <xref rid="bib35" ref-type="bibr"><u>Li, 2013</u></xref>).<li> [Rambaut et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32669681">32669681</a>) [Hadfield et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) [Minh et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32011700">32011700</a>) [Suchard et&#xA0;al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29942656">29942656</a>) [Ayres et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31034053">31034053</a>) [Rambaut et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32669681">32669681</a>) [Hadfield et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) [Minh et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32011700">32011700</a>) [Suchard et&#xA0;al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29942656">29942656</a>) [Ayres et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31034053">31034053</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Li, 2013](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</part-title>          <year>2013</year>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1303.3997" id="intref0155">https://arxiv.org/abs/1303.3997</ext-link>        ) [Grubaugh et&#xA0;al., 2019b](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30621750">30621750</a>) [Li, 2013](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</part-title>          <year>2013</year>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1303.3997" id="intref0155">https://arxiv.org/abs/1303.3997</ext-link>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</part-title>          <year>2013</year>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1303.3997" id="intref0155">https://arxiv.org/abs/1303.3997</ext-link>        ">          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</part-title>          <year>2013</year>          <ext-link ext-link-type="uri" xlink:href="https://arxiv.org/abs/1303.3997" id="intref0155">https://arxiv.org/abs/1303.3997</ext-link>        </a> <b>(2)</b> Emergence of an early SARS-CoV-2 epidemic in the United States<ul><li> VarCA is implemented as a <b>Snakemake</b> (<xref rid="B11" ref-type="bibr"><u>11</u></xref>) pipeline with two components, which we refer to as the <italic toggle="yes">prepare</italic> and <italic toggle="yes">classify</italic> subworkflows (Figure <xref rid="F1" ref-type="fig"><u>1</u></xref>).<li> In addition, the exact versions of each variant caller are specified in a YAML file, parsed by <b>Snakemake</b> upon execution and used for installing these callers from their conda distributions.<li> [11](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34311761">34311761</a> <b>(1)</b> STRONG: metagenomics strain resolution on assembly graphs<ul><li>The <b>Snakemake</b> based pipeline begins with the recovery of metagenome-assembled genomes (MAGs) [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>].<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34312399">34312399</a> <b>(1)</b> Rapid ecosystem-scale consequences of acute deoxygenation on a Caribbean coral reef<ul><li> Within the anvi&#x2019;o environment, we built a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup> (v.5.10.0) workflow for co-assembly analysis.<li> [66](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34391075">34391075</a> <b>(1)</b> Target capture sequencing of SARS-CoV-2 genomes using the ONETest Coronaviruses Plus<ul><li> The pipeline was implemented in C/C++ and Python using a combination of in-house software and third-party tools, including Biopython v1.78 (<xref rid="bib0005" ref-type="bibr"><u>Cock&#xA0;et&#xA0;al., 2009</u></xref>), bedtools v2.29.2 (<xref rid="bib0019" ref-type="bibr"><u>Quinlan&#xA0;and Hall,&#xA0;2010</u></xref>), pybedtools v0.8.1 (<xref rid="bib0007" ref-type="bibr"><u>Dale&#xA0;et&#xA0;al., 2011</u></xref>), samtools/bcftools/htslib v1.11 (<xref rid="bib0017" ref-type="bibr"><u>Li&#xA0;et&#xA0;al., 2009</u></xref>), pysam v0.16.0.1, pandas v1.1.3, and <b>Snakemake</b> v5.26.1 (<xref rid="bib0013" ref-type="bibr"><u>Koster&#xA0;and Rahmann,&#xA0;2012</u></xref>).<li> [Cock&#xA0;et&#xA0;al., 2009](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19304878">19304878</a>) [Quinlan&#xA0;and Hall,&#xA0;2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20110278">20110278</a>) [Dale&#xA0;et&#xA0;al., 2011](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21949271">21949271</a>) [Li&#xA0;et&#xA0;al., 2009](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Handsaker</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Wysoker</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Fennell</surname>              <given-names>T</given-names>            </name>            <name>              <surname>Ruan</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Homer</surname>              <given-names>N</given-names>            </name>            <etal/>          </person-group>          <part-title>The sequence alignment/map format and SAMtools</part-title>          <series>Bioinformatics</series>          <volume>25</volume>          <year>2009</year>          <fpage>2078</fpage>          <lpage>2079</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btp352</pub-id>        ) [Koster&#xA0;and Rahmann,&#xA0;2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34301987">34301987</a> <b>(1)</b> Gene expression analysis method integration and co-expression module detection applied to rare glucide metabolism disorders using ExpHunterSuite<ul><li>The pipeline was written and executed using the <b>Snakemake</b> (<xref rid="B2" ref-type="bibr"><u>2</u></xref>) workflow management system and consists of a total of 314 computational steps, divided into 21 sequential processes in two main phases (<xref rid="fig1" ref-type="fig"><u>Fig.&#xA0;1</u></xref>).<li> [2](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(1)</b> Gene expression analysis method integration and co-expression module detection applied to rare glucide metabolism disorders using ExpHunterSuite<ul><li>RMarkdown files, <b>snakemake</b> files, and R scripts used in the manuscript are available at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/troyrowan/gpsm_envgwas" ext-link-type="uri">https://github.com/troyrowan/gpsm_envgwas</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34401670">34401670</a> <b>(1)</b> LiquidCNA: Tracking subclonal evolution from longitudinal liquid biopsies using somatic copy number alterations<ul><li> We used the <b>snakemake</b> workflow available from within ichorCNA, using bin sizes of 500 kb with corresponding GC and mappability files, no matched normal panel, maximum allowed CN of 6, and default parameter values otherwise.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34401670">34401670</a> <b>(1)</b> LiquidCNA: Tracking subclonal evolution from longitudinal liquid biopsies using somatic copy number alterations<ul><li> RDscan is implemented as a custom <b>Snakemake</b> (<xref rid="B65" ref-type="bibr"><u>65</u></xref>) workflow.<li> [65](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34298235">34298235</a> <b>(1)</b> Detection of circulating tumor DNA without a tumor-informed search using next-generation sequencing is a prognostic biomarker in pancreatic ductal adenocarcinoma<ul><li> Each of the tools used for alignment and consensus calling are publicly available in USeq tools (<ext-link ext-link-type="uri" xlink:href="https://github.com/HuntsmanCancerInstitute/USeq" id="interref0001">https://github.com/HuntsmanCancerInstitute/USeq</ext-link>) and integrated in several containerized <b>snakemake</b> workflows (<ext-link ext-link-type="uri" xlink:href="https://github.com/HuntsmanCancerInstitute/Workflows/tree/master/Hg38RunnerWorkflows/DnaConsensusAlignQC" id="interref0002">https://github.com/HuntsmanCancerInstitute/Workflows/tree/master/Hg38RunnerWorkflows/DnaConsensusAlignQC</ext-link>, CtSomaticCaller, and Annotator).</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34284719">34284719</a> <b>(2)</b> Pre-capture multiplexing provides additional power to detect copy number variation in exome sequencing<ul><li> Full <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] pipeline provided in supplemental materials.<li> Full <b>Snakemake</b>&#xA0;[<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] pipeline provided in supplemental materials.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34289358">34289358</a> <b>(1)</b> Reprogramming of bivalent chromatin states in <italic>NRAS</italic> mutant melanoma suggests PRC2 inhibition as a therapeutic strategy<ul><li>Raw fastq reads for all ChIP-seq experiments were processed using a <b>snakemake</b> based pipeline <ext-link ext-link-type="uri" xlink:href="https://github.com/crazyhottommy/pyflow-ChIPseq">https://github.com/crazyhottommy/pyflow-ChIPseq</ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34279213">34279213</a> <b>(1)</b> Harmonization of whole-genome sequencing for outbreak surveillance of <italic>Enterobacteriaceae</italic> and <italic>Enterococci</italic><ul><li> All data preprocessing for the SNP-based data analysis was performed using <b>Snakemake</b> [<xref rid="R29" ref-type="bibr"><u>29</u></xref>] as the workflow manager.<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34284136">34284136</a> <b>(6)</b> CoBRA: Containerized Bioinformatics Workflow for Reproducible ChIP/ATAC-seq Analysis<ul><li>CoBRA uses <b>Snakemake</b> <xref rid="b0050" ref-type="bibr"><u>[10]</u></xref>, a workflow management system to create the computational pipeline.<li> Using the <b>Snakemake</b> system enables the reproducibility and scalability of CoBRA.<li>The CoBRA pipeline is implemented using the <b>Snakemake</b> workflow management system <xref rid="b0050" ref-type="bibr"><u>[10]</u></xref> and is described via a human-readable, Python-based language.<li> With the built-in <b>Snakemake</b> reference rule, CoBRA automatically downloads all needed reference files, if they have not been downloaded before.<li> Additionally, the highly modular <b>Snakemake</b> framework allows for rapid integration of new approaches or replacement of existing tools.<li> Modules can be added simply by adding a new <b>Snakemake</b> &#x201C;rule&#x201D; and adding a flag in the config file (<xref rid="s0060" ref-type="sec"><u>Figure S1</u></xref>A&#x2013;C) to turn the analysis on.<li> [[10]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [[10]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34273949">34273949</a> <b>(14)</b> ARPEGGIO: Automated Reproducible Polyploid EpiGenetic GuIdance workflOw<ul><li>ARPEGGIO&#x2019;s implementation combines the <b>Snakemake</b> workflow management system [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>] with the Conda package manager [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>] and Singularity containers&#xA0;[<xref ref-type="bibr" rid="CR48"><u>48</u></xref>] to ensure both ease of use and reproducibility.<li> ARPEGGIO requires the users to install the Conda package management system [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>], then <b>Snakemake</b> [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>] via Conda and, optionally, Singularity [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>].<li>ARPEGGIO is written in <b>Snakemake</b>, a Python based language for workflow development [<xref ref-type="bibr" rid="CR45"><u>45</u></xref>].<li> With <b>Snakemake</b>, a workflow is broken down into a series of rules.<li> Once all the rules are set, to run a <b>Snakemake</b> workflow, a target file (or multiple) needs to be requested.<li> <b>Snakemake</b> will automatically build the workflow to obtain the target file based on the input/output relationships between rules (dependencies).<li>In addition to the core features of <b>Snakemake</b>, ARPEGGIO takes advantage of the integrated Conda package management system [<xref ref-type="bibr" rid="CR46"><u>46</u></xref>].<li> The integration of Conda in <b>Snakemake</b> allows rules to be run within a specific environment and during the execution of a workflow, <b>Snakemake</b> takes care of switching between environments if different rules require different environments.<li> From a user perspective, once Conda and <b>Snakemake</b> are installed, ARPEGGIO will take care of installing all the tools needed for the analyses, running them and switching automatically between environments when needed (Fig.<li>Making the workflow specific for allopolyploids presented major challenges with both <b>Snakemake</b> and Conda.<li> <b>Snakemake</b> rules in ARPEGGIO had to be structured to allow for any combination between sequencing strategies and experimental designs.<li>In practice, any user can take advantage of all the Conda and <b>Snakemake</b> features discussed above with a central configuration file.<li> ARPEGGIO provides a convenient implementation of the selected tools and its architecture allows future modifications as long as the input/output structure of the <b>Snakemake</b> rules is preserved.<li> [45](Koster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics. 2012;28(19):2520&#x2013;2.) [46](Anaconda. Anaconda Software Distribution. 2014. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>.) [48](Kurtzer GM, Sochat V, Bauer MW. Singularity: scientific containers for mobility of compute. PLoS One. 2017;12(5):e0177459.) [46](Anaconda. Anaconda Software Distribution. 2014. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>.) [45](Koster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics. 2012;28(19):2520&#x2013;2.) [48](Kurtzer GM, Sochat V, Bauer MW. Singularity: scientific containers for mobility of compute. PLoS One. 2017;12(5):e0177459.) [45](Koster J, Rahmann S. Snakemake--a scalable bioinformatics workflow engine. Bioinformatics. 2012;28(19):2520&#x2013;2.) [46](Anaconda. Anaconda Software Distribution. 2014. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=Anaconda. Anaconda Software Distribution. 2014. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>.">Anaconda. Anaconda Software Distribution. 2014. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>.</a> <b>(1)</b> ARPEGGIO: Automated Reproducible Polyploid EpiGenetic GuIdance workflOw<ul><li> Potential strain numbers in individual clams/metagenomes were calculated using DESMAN v2.1.1 (<xref rid="r61" ref-type="bibr"><u>61</u></xref>) and the <b>snakemake</b> workflow (<xref rid="r62" ref-type="bibr"><u>62</u></xref>) on the program (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://www.pnas.org/lookup/suppl/doi:10.1073/pnas.2104378118/-/DCSupplemental" ext-link-type="uri" xlink:show="new"><italic toggle="yes">SI Appendix</italic>, <italic toggle="yes">SI Methods</italic></ext-link>).<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28934976">28934976</a>) [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> ARPEGGIO: Automated Reproducible Polyploid EpiGenetic GuIdance workflOw<ul><li>Chimeric reads covering two predicted transcripts were extracted using the ChimerID scripts (<xref rid="bib46" ref-type="bibr"><u>Parker et al., 2020</u></xref>) (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/bartongroup/Simpson_Barton_Nanopore_1" ext-link-type="uri">https://github.com/bartongroup/Simpson_Barton_Nanopore_1</ext-link>, <xref rid="bib15" ref-type="bibr"><u>Farhat, 2021a</u></xref>&#xA0;copy archived at <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://archive.softwareheritage.org/swh:1:dir:7ca31bd3bb216187e167a091d1c559f8726f15a1;origin=https://github.com/bartongroup/Simpson_Barton_Nanopore_1;visit=swh:1:snp:25103e8204f657ea935189da396f5028ff0b8e5c;anchor=swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5" ext-link-type="uri">swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5</ext-link>) from native RNA Nanopore aligned reads within a Anaconda environment containing the following packages (<b>snakemake</b> ;pysam ; samtools ; scipy ; statsmodels ; py-bgzip ; tabix ; pytables ; bedtools ; minimap2 ; panda=0.25.3).<li> [Parker et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31931956">31931956</a>) [Farhat, 2021a](          <person-group person-group-type="author">            <name>              <surname>Farhat</surname>              <given-names>DC</given-names>            </name>          </person-group>          <year iso-8601-date="2021">2021a</year>          <data-title>Simpson/Barton group nanopore paper</data-title>          <source>Software Heritage</source>          <version designator="swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5">swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5</version>          <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://archive.softwareheritage.org/swh:1:dir:7ca31bd3bb216187e167a091d1c559f8726f15a1;origin=https://github.com/bartongroup/Simpson_Barton_Nanopore_1;visit=swh:1:snp:25103e8204f657ea935189da396f5028ff0b8e5c;anchor=swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5" ext-link-type="uri">https://archive.softwareheritage.org/swh:1:dir:7ca31bd3bb216187e167a091d1c559f8726f15a1;origin=https://github.com/bartongroup/Simpson_Barton_Nanopore_1;visit=swh:1:snp:25103e8204f657ea935189da396f5028ff0b8e5c;anchor=swh:1:rev:1b509454a9e25a8c81be5092f8e525ca00e7b5a5</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34273562">34273562</a> <b>(1)</b> From Reads to Insights: Integrative Pipelines for Biological Interpretation of ATAC-seq Data<ul><li> Based on the <b>snakemake</b> system <xref rid="b0085" ref-type="bibr"><u>[17]</u></xref>, CoBRA can also easily integrate additional analytical tools in the future.<li> [[17]](Koster J, Rahmann S. Snakemake&#x2014;a scalable bioinformatics workflow engine. Bioinformatics 2018;34:3600.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34255713">34255713</a> <b>(1)</b> Predicting Depression From Smartphone Behavioral Markers Using Machine Learning Methods, Hyperparameter Optimization, and Feature Importance Analysis: Exploratory Study<ul><li>Data preprocessing and feature extraction pipeline were created with Python (version 3.7.6) and R (version 4.0.2) programming languages, using <b>Snakemake</b> [<xref rid="ref77" ref-type="bibr"><u>77</u></xref>] and RAPIDS [<xref rid="ref78" ref-type="bibr"><u>78</u></xref>] for workflow management.<li> [77](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [78](          <person-group person-group-type="author">            <name>              <surname>Vega</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Aguillera</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Goel</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Joshl</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Durica</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Kunta</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Low</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Reproducible Analysis Pipeline for Data Streams (RAPIDS): Open Source Software to Process Data Collected with Mobile Devices Internet</article-title>          <source>JMIR Preprints</source>          <year>2021</year>          <date-in-citation>2021-06-28</date-in-citation>          <comment>            <ext-link ext-link-type="uri" xlink:href="https://preprints.jmir.org/preprint/23246">https://preprints.jmir.org/preprint/23246</ext-link>          </comment>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author">            <name>              <surname>Vega</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Aguillera</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Goel</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Joshl</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Durica</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Kunta</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Low</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Reproducible Analysis Pipeline for Data Streams (RAPIDS): Open Source Software to Process Data Collected with Mobile Devices Internet</article-title>          <source>JMIR Preprints</source>          <year>2021</year>          <date-in-citation>2021-06-28</date-in-citation>          <comment>            <ext-link ext-link-type="uri" xlink:href="https://preprints.jmir.org/preprint/23246">https://preprints.jmir.org/preprint/23246</ext-link>          </comment>        ">          <person-group person-group-type="author">            <name>              <surname>Vega</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Li</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Aguillera</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Goel</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Joshl</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Durica</surname>              <given-names>K</given-names>            </name>            <name>              <surname>Kunta</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Low</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Reproducible Analysis Pipeline for Data Streams (RAPIDS): Open Source Software to Process Data Collected with Mobile Devices Internet</article-title>          <source>JMIR Preprints</source>          <year>2021</year>          <date-in-citation>2021-06-28</date-in-citation>          <comment>            <ext-link ext-link-type="uri" xlink:href="https://preprints.jmir.org/preprint/23246">https://preprints.jmir.org/preprint/23246</ext-link>          </comment>        </a> <b>(1)</b> Predicting Depression From Smartphone Behavioral Markers Using Machine Learning Methods, Hyperparameter Optimization, and Feature Importance Analysis: Exploratory Study<ul><li>Workflow management systems (WMS) address some of the challenges associated with the execution of complex pipelines, and range from simple command line tools like <b>SnakeMake</b> [<xref rid="pone.0254363.ref009" ref-type="bibr"><u>9</u></xref>] to fully-managed cloud-based enterprise solutions like Closha [<xref rid="pone.0254363.ref010" ref-type="bibr"><u>10</u></xref>] and Terra [<xref rid="pone.0254363.ref011" ref-type="bibr"><u>11</u></xref>].<li> [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29504905">29504905</a>) [11](Garfinkel T, Pfaff B, Chow J, Rosenblum M, Boneh D. Terra: A virtual machine-based platform for trusted computing. In: Operating Systems Review (ACM); 2003.) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34238254">34238254</a> <b>(1)</b> Inference of kinase-signaling networks in human myeloid cell line models by Phosphoproteomics using kinase activity enrichment analysis (KAEA)<ul><li> <italic><b>Snakemake</b></italic> [<xref ref-type="bibr" rid="CR62"><u>62</u></xref>], a python-based workflow engine, was used as a wrapper to the <italic>R</italic> scripts to create a reproducible frame for the pipeline.<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34252764">34252764</a> <b>(1)</b> A direct capture method for purification and detection of viral nucleic acid enables epidemiological surveillance of SARS-CoV-2<ul><li> <b>Snakemake</b> v5.17.0 was used for workflow management on a Microsoft Azure CycleCloud instance.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34229612">34229612</a> <b>(1)</b> Consensus clustering applied to multi-omics disease subtyping<ul><li> To facilitate its use, ClustOmics can be run through the <b>Snakemake</b> workflow management system.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34226720">34226720</a> <b>(2)</b> PaintSHOP enables the interactive design of transcriptome- and genome-scale oligonucleotide FISH experiments<ul><li> Primary probes can be discovered <italic>de novo</italic> with the PaintSHOP pipeline (<ext-link ext-link-type="uri" xlink:href="http://github.com/beliveau-lab/PaintSHOP_pipeline">github.com/beliveau-lab/PaintSHOP_pipeline</ext-link>), which uses OligoMiner<sup><xref rid="R33" ref-type="bibr"><u>33</u></xref></sup> scripts wrapped in <b>Snakemake</b><sup><xref rid="R38" ref-type="bibr"><u>38</u></xref></sup>.<li>All PaintSHOP probe collections were generated using a pipeline implemented with <b>Snakemake</b><sup><xref rid="R38" ref-type="bibr"><u>38</u></xref></sup>.<li> [33](<name><surname>Beliveau</surname><given-names>BJ</given-names></name><etal/><article-title>OligoMiner provides a rapid, flexible environment for the design of genome-scale oligonucleotide in situ hybridization probes</article-title>. <source>Proc. Natl. Acad. Sci. U. S. A</source>. (<year>2018</year>) doi:<pub-id pub-id-type="doi">10.1073/pnas.1714530115</pub-id><comment>.</comment>) [38](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> &amp; <name><surname>Rahmann</surname><given-names>S</given-names></name><article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id><comment>.</comment>) [38](<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> &amp; <name><surname>Rahmann</surname><given-names>S</given-names></name><article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id><comment>.</comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> &amp; <name><surname>Rahmann</surname><given-names>S</given-names></name><article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id><comment>.</comment>"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name> &amp; <name><surname>Rahmann</surname><given-names>S</given-names></name><article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) doi:<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id><comment>.</comment></a> <b>(1)</b> PaintSHOP enables the interactive design of transcriptome- and genome-scale oligonucleotide FISH experiments<ul><li> Except for the nasal swab (NSPP) dataset [<xref rid="R27" ref-type="bibr"><u>27</u></xref>], all datasets were processed using a <b>snakemake</b> pipeline that performs the following steps: samples are downloaded from the SRA, quality-control is performed using fastp [<xref rid="R30" ref-type="bibr"><u>30</u></xref>], alignment to Genome Reference Consortium Human Build 38 version 91 is done using STAR [<xref rid="R31" ref-type="bibr"><u>31</u></xref>], and isoform quantification carried out by RSEM [<xref rid="R32" ref-type="bibr"><u>32</u></xref>].<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33712587">33712587</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30013175">30013175</a>) [31](<name><surname>Dobin</surname><given-names>Alexander</given-names></name>, <name><surname>Davis</surname><given-names>Carrie A</given-names></name>, <name><surname>Schlesinger</surname><given-names>Felix</given-names></name>, <name><surname>Drenkow</surname><given-names>Jorg</given-names></name>, <name><surname>Zaleski</surname><given-names>Chris</given-names></name>, <name><surname>Jha</surname><given-names>Sonali</given-names></name>, <name><surname>Batut</surname><given-names>Philippe</given-names></name>, <name><surname>Chaisson</surname><given-names>Mark</given-names></name>, and <name><surname>Gingeras</surname><given-names>Thomas R</given-names></name>. <article-title>STAR: ultrafast universal RNA-seq aligner</article-title>. <source>Bioinformatics (Oxford, England)</source>, <volume>29</volume>:<fpage>15</fpage>&#x2013;<lpage>21</lpage>, <month>January</month><year>2013</year>.) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=21816040">21816040</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34210356">34210356</a> <b>(2)</b> CLIMB-COVID: continuous integration supporting decentralised sequencing for SARS-CoV-2 genomic surveillance<ul><li> Grapevine is a <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>] used to build a phylogenetic tree that captures the evolutionary relationships between the sampled viruses, placing UK sequences in the global context (Fig.<li> Civet is written in Python and uses <b>Snakemake</b> [<xref ref-type="bibr" rid="CR13"><u>13</u></xref>] to orchestrate its analysis steps.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34281175">34281175</a> <b>(2)</b> Single Copy Oligonucleotide Fluorescence In Situ Hybridization Probe Design Platforms: Development, Application and Evaluation<ul><li> All these processes in the pipeline are integrated into an automated <b>Snakemake</b> workflow [<xref rid="B79-ijms-22-07124" ref-type="bibr"><u>79</u></xref>].<li> By contrast, the PaintSHOP resources host FISH probe sets of minor species, and de novo probe design using PaintSHOP requires the knowledge of bioinformatics, such as <b>Snakemake</b>.<li> [79](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34183036">34183036</a> <b>(1)</b> Using the longest run subsequence problem within homology-based scaffolding<ul><li> The algorithms are implemented in Python and executed via <b>Snakemake</b> [<xref ref-type="bibr" rid="CR10"><u>10</u></xref>] using Python 3.9.1 and PuLP version 2.3.1.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34262593">34262593</a> <b>(1)</b> RNA-Seq Data for Reliable SNP Detection and Genotype Calling: Interest for Coding Variant Characterization and <italic>Cis</italic>-Regulation Analysis by Allele-Specific Expression in Livestock Species<ul><li>For all samples, RNA-seq variants were detected using the <b>snakemake</b> (<xref rid="B26" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) pipeline, available at this reference: (<xref rid="B19" ref-type="bibr"><u>GitLab, 2019</u></xref>).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [GitLab, 2019](<collab>GitLab</collab> (<year>2019</year>). <source><italic>Snakemake/1000RNASeq_chicken/calling &#x22C5; master &#x22C5; bios4biol / workflows.</italic></source> Available online at: <ext-link ext-link-type="uri" xlink:href="https://forgemia.inra.fr/bios4biol/workflows/tree/master/Snakemake/1000RNASeq_chicken/calling">https://forgemia.inra.fr/bios4biol/workflows/tree/master/Snakemake/1000RNASeq_chicken/calling</ext-link><comment>(accessed August 28, 2019)</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34172000">34172000</a> <b>(1)</b> MicroPIPE: validating an end-to-end workflow for high-quality complete bacterial genome construction<ul><li> Katuali is an ONT-developed assembly pipeline implemented in <b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34249907">34249907</a> <b>(1)</b> An Early Diagnostic Clue for <italic>COL18A1-</italic> and <italic>LAMA1-</italic>Associated Diseases: High Myopia With Alopecia Areata in the Cranial Midline<ul><li> The RNA sequencing data were checked for quality using FastQC and mapped and analyzed using the VIPER <b>Snakemake</b> pipeline (<xref rid="B10" ref-type="bibr"><u>Cornwell et al., 2018</u></xref>).<li> [Cornwell et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34202069">34202069</a> <b>(1)</b> Predicting Lifestyle from Positive Selection Data and Genome Properties in Oomycetes<ul><li>We developed a pipeline for whole genome positive selection analysis in Python using the <b>Snakemake</b> modular workflow framework [<xref rid="B41-pathogens-10-00807" ref-type="bibr"><u>41</u></xref>].<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34167443">34167443</a> <b>(1)</b> Genetic architecture of transmission stage production and virulence in schistosome parasites<ul><li> All these steps were automated using <b>Snakemake</b> (v5.14.0) [<xref rid="cit0058" ref-type="bibr"><u>58</u></xref>].<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34146095">34146095</a> <b>(10)</b> Draft <italic toggle="yes">de novo</italic> Genome Assembly of the Elusive Jaguarundi, <italic toggle="yes">Puma yagouaroundi</italic><ul><li> To ensure transparency and reproducibility of our study, we arranged most of the activities in 2 pipelines implemented in <b>Snakemake</b> (<xref rid="CIT0021" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>).<li>The <b>Snakemake</b> pipelines contain scripts in Python and R that process the genome sequence and annotation data.<li> R scripts of the <b>Snakemake</b> pipelines use routines from packages GenomicRanges (<xref rid="CIT0024" ref-type="bibr">Lawrence et al.<li><b>Snakemake</b>.<li> Joint distribution of GC content and length for scaffolds shorter than 1 Mbp was visualized in a separate figure using the JDGA package and the R script from the <b>Snakemake</b> pipeline.<li> We compared repeat annotations for jaguarundi by RepeatMasker and WindowMasker using the R script from the <b>Snakemake</b> pipeline.<li>Repeat annotations for the jaguarundi, cheetah, and puma genome assemblies were summarized by the R script from the <b>Snakemake</b> pipeline.<li> The procedure to identify the scaffolds was implemented in the R script from the <b>Snakemake</b> pipeline.<li> Genotypes of genomic variants from the 2 jaguarundis were compared to each other using routines from the JDGA package and the R script from the <b>Snakemake</b> pipeline.<li>Statistics of the identified genomic variants, their predicted effects, estimated heterozygosity and transition-transversion ratios were obtained using the JDGA package and the R scripts from the <b>Snakemake</b> pipeline.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Draft <italic toggle="yes">de novo</italic> Genome Assembly of the Elusive Jaguarundi, <italic toggle="yes">Puma yagouaroundi</italic><ul><li> We processed the data with the <b>snakemake</b> pipeline SalamiSnake [<xref rid="bib7" ref-type="bibr"><u>7</u></xref>] (v0.0.1) for eCLIP data.<li> [7](<person-group person-group-type="author"><string-name><surname>Heyl</surname><given-names>F</given-names></string-name></person-group>. <source>SalamiSnake</source>. <year>2018</year>. <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/BackofenLab/SalamiSnake" ext-link-type="uri">https://github.com/BackofenLab/SalamiSnake</ext-link>. <comment>(Last accessed: 10.12.2019)</comment>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33827898">33827898</a> <b>(1)</b> A Whole-Genome-Based Gene-by-Gene Typing System for Standardized High-Resolution Strain Typing of <named-content content-type="genus-species">Bacillus anthracis</named-content><ul><li> The <b>snakemake</b> (<xref rid="B35" ref-type="bibr"><u>35</u></xref>) application of chewBBACA, called chewieSnake v.2.0.0-52-g01c32bb (accessed July 2020 [<ext-link ext-link-type="uri" xlink:href="https://gitlab.com/bfr_bioinformatics/chewieSnake">https://gitlab.com/bfr_bioinformatics/chewieSnake</ext-link>]) was used with default options, including 0.6 as the minimum BLAST score ratio for locus similarity (&#x2013;bsr 0.6) and 20% length boundaries for allele definition (&#x2013;st 0.2).<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34239974">34239974</a> <b>(8)</b> Compi: a framework for portable and reproducible pipelines<ul><li> Furthermore, command-line based applications such as Nextflow (<xref rid="ref-2" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), <b>Snakemake</b> (<xref rid="ref-6" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>), or SciPipe (<xref rid="ref-7" ref-type="bibr"><u>Lampa et al., 2019</u></xref>), provide feature-rich workflow engines oriented to bioinformaticians with medium-to-high programming skills.<li> Other frameworks like Galaxy or Taverna made significant progress to support the execution of workflows defined in CWL and other tools allow to export their workflows into CWL (e.g.,&#xA0;<b>Snakemake</b>) or import them from CWL.<li>Unlike <b>Snakemake</b>, the Nextflow and SciPipe tools allow dynamic scheduling, that is, the ability to change the pipeline structure dynamically to schedule a different number of tasks based on the results of a previous step or any other parameter.<li>Compi, <b>Snakemake</b>, and Nextflow are language independent, allowing external scripts written in any programming language to be invoked.<li> Similarly, <b>Snakemake</b> has built-in support to execute complete pipelines on Docker images and individual rules in isolated Conda environments.<li> <b>Snakemake</b> does not provide a similar public repository, but a dedicated GitHub project exists (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>-workflows/docs">https://github.com/<b>snakemake</b>-workflows/docs</ext-link>).<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di&#xA0;Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <year>2017</year>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <volume>35</volume>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Lampa et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34137631">34137631</a> <b>(1)</b> Chromosome-Scale Assembly of the Complete Genome Sequence of <italic>Leishmania</italic> (<italic>Mundinia</italic>) <italic>martiniquensis</italic>, Isolate LSCM1, Strain LV760<ul><li>The analysis workflow for assembly and annotation, including software versions and parameters used, was performed using <b>Snakemake</b> (<xref rid="B19" ref-type="bibr"><u>19</u></xref>) and is available online for reproducibility purposes (<ext-link ext-link-type="uri" xlink:href="https://github.com/hatimalmutairi/LGAAP">https://github.com/hatimalmutairi/LGAAP</ext-link>).<li> [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34134617">34134617</a> <b>(1)</b> Downregulated luteolytic pathways in the transcriptome of early pregnancy bovine corpus luteum are mimicked by interferon-tau in vitro<ul><li> The full pipeline was run using <b>snakemake</b> [<xref ref-type="bibr" rid="CR78"><u>78</u></xref>].<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34211429">34211429</a> <b>(1)</b> A Homozygous Deep Intronic Mutation Alters the Splicing of Nebulin Gene in a Patient With Nemaline Myopathy<ul><li> Additionally, a custom-automated bioinformatics pipeline built using GATK best practices and the <b>Snakemake</b> workflow, DNA-SEQ-GATK variant calling, was used to validate the previous bioinformatics analyses.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34127860">34127860</a> <b>(1)</b> Fine-mapping, trans-ancestral, and genomic analyses identify causal variants, cells, genes, and drug targets for type 1 diabetes<ul><li> Workflows were implemented using conda and <b>snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34110280">34110280</a> <b>(1)</b> rMAP: the Rapid Microbial Analysis Pipeline for ESKAPE bacterial group whole-genome sequence data<ul><li>As a significant limitation, rMAP is coded exclusively in Bash and is not implemented within a modern workflow language manager, such as <b>Snakemake</b> or Nextflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34108460">34108460</a> <b>(1)</b> Translation and codon usage regulate Argonaute slicer activity to trigger small RNA biogenesis<ul><li> Unless otherwise stated, computations were done using Python and UNIX utilities, either as standalone scripts or as steps implemented in a <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR71"><u>71</u></xref></sup> workflow.<li> [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34168993">34168993</a> <b>(2)</b> DIVIS: Integrated and Customizable Pipeline for Cancer Genome Sequencing Analysis and Interpretation<ul><li> Workflow management software such as <b>Snakemake</b> (<xref rid="B23" ref-type="bibr"><u>23</u></xref>), Bpipe (<xref rid="B24" ref-type="bibr"><u>24</u></xref>), Ruffus (<xref rid="B25" ref-type="bibr"><u>25</u></xref>), Nextflow (<xref rid="B26" ref-type="bibr"><u>26</u></xref>), and Galaxy (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) sufficiently satisfy this requirement by automatically generating analysis scripts through graphical operation, which simplifies human intervention.<li> <b>Snakemake</b>, inspired by GNU Make, describes a workflow by using its Domain Specific Language (DSL) and executes the workflow through Python scripts (<xref rid="B23" ref-type="bibr"><u>23</u></xref>).<li> [23](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake&#x2014;a Scalable Bioinformatics Workflow Engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>19</issue>):<page-range>2520&#x2013;2</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) [24](<person-group person-group-type="author"><name><surname>Sadedin</surname><given-names>SP</given-names></name><name><surname>Pope</surname><given-names>B</given-names></name><name><surname>Oshlack</surname><given-names>A</given-names></name></person-group>. <article-title>Bpipe: A Tool for Running and Managing Bioinformatics Pipelines</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>11</issue>):<page-range>1525&#x2013;6</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts167</pub-id>) [25](<person-group person-group-type="author"><name><surname>Goodstadt</surname><given-names>L</given-names></name></person-group>. <article-title>Ruffus: A Lightweight Python Library for Computational Pipelines</article-title>. <source>Bioinformatics</source> (<year>2010</year>) <volume>26</volume>(<issue>21</issue>):<page-range>2778&#x2013;9</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/btq524</pub-id>) [26](<person-group person-group-type="author"><name><surname>Di Tommaso</surname><given-names>P</given-names></name><name><surname>Chatzou</surname><given-names>M</given-names></name><name><surname>Floden</surname><given-names>EW</given-names></name><name><surname>Barja</surname><given-names>PP</given-names></name><name><surname>Palumbo</surname><given-names>E</given-names></name><name><surname>Notredame</surname><given-names>C</given-names></name></person-group>. <article-title>Nextflow Enables Reproducible Computational Workflows</article-title>. <source>Nat Biotechnol</source> (<year>2017</year>) <volume>35</volume>(<issue>4</issue>):<page-range>316&#x2013;9</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [23](<person-group person-group-type="author"><name><surname>K&#xF6;ster</surname><given-names>J</given-names></name><name><surname>Rahmann</surname><given-names>S</given-names></name></person-group>. <article-title>Snakemake&#x2014;a Scalable Bioinformatics Workflow Engine</article-title>. <source>Bioinformatics</source> (<year>2012</year>) <volume>28</volume>(<issue>19</issue>):<page-range>2520&#x2013;2</page-range>. &#xA0;<pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34103501">34103501</a> <b>(9)</b> Systematic benchmarking of tools for CpG methylation detection from nanopore sequencing<ul><li>A standardized workflow was developed to obtain 5mC calls at CpG sites from Nanopolish<sup><xref ref-type="bibr" rid="CR8"><u>8</u></xref></sup>, Megalodon<sup><xref ref-type="bibr" rid="CR12"><u>12</u></xref></sup>, DeepSignal<sup><xref ref-type="bibr" rid="CR13"><u>13</u></xref></sup>, Guppy<sup><xref ref-type="bibr" rid="CR14"><u>14</u></xref></sup>, Tombo<sup><xref ref-type="bibr" rid="CR15"><u>15</u></xref></sup>, and DeepMod<sup><xref ref-type="bibr" rid="CR16"><u>16</u></xref></sup> (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>), using <b>Snakemake</b> pipelines<sup><xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup> (<ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>)<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> <b>Snakemake</b> pipelines and command lines used are available at <ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link><sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> A <b>Snakemake</b> pipeline was not developed for Megalodon and DeepMod as they can be run with a single command.<li> <b>Snakemake</b> pipelines to run these tools are available at GitHub<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> To include all the predictions per read and per site, such CpG groups were split up into the constituent sites with the same log-likelihood ratio using a Python script incorporated in our <b>Snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>)<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> These files were converted to the same TSV format used by other tools with a Python script included in our <b>Snakemake</b> pipeline, with the score and coverage for each mapped CpG site for downstream analyses.<li> In addition, a Python script was added to our <b>Snakemake</b> pipeline to calculate the methylation frequency at each CpG site.<li> These scripts were incorporated into our <b>Snakemake</b> pipeline (<ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>)<sup><xref ref-type="bibr" rid="CR11"><u>11</u></xref></sup>.<li> Coverage plots for target regions were generated using the <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR17"><u>17</u></xref></sup> workflow developed by Oxford Nanopore Technologies<sup><xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup>.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28218898">28218898</a>) [12](Oxford Nanopore Technologies. <italic>GitHub&#x2014;Megalodon</italic> (Oxford Nanopore Technologies, 2020). <ext-link ext-link-type="uri" xlink:href="https://github.com/nanoporetech/megalodon">https://github.com/nanoporetech/megalodon</ext-link>.) [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30994904">30994904</a>) [14](Oxford Nanopore Technologies. <italic>GitHub</italic> (Oxford Nanopore Technologies, 2020). <ext-link ext-link-type="uri" xlink:href="https://github.com/nanoporetech">https://github.com/nanoporetech</ext-link>.) [15](Stoiber, M. et al. De novo identification of dna modifications enabled by genome-guided nanopore signal processing. 10.1101/094672 (2017).) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31164644">31164644</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [11](Yuen, Z. W.-S., Srivastava, A., Jack, C. &amp; Eyras, E. <italic>Systematic benchmarking of tools for CpG methylation detecgtion from Nanopore sequencing.</italic><ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>, 10.5281/zenodo.4748319 (2021).) [11](Yuen, Z. W.-S., Srivastava, A., Jack, C. &amp; Eyras, E. <italic>Systematic benchmarking of tools for CpG methylation detecgtion from Nanopore sequencing.</italic><ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>, 10.5281/zenodo.4748319 (2021).) [11](Yuen, Z. W.-S., Srivastava, A., Jack, C. &amp; Eyras, E. <italic>Systematic benchmarking of tools for CpG methylation detecgtion from Nanopore sequencing.</italic><ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>, 10.5281/zenodo.4748319 (2021).) [11](Yuen, Z. W.-S., Srivastava, A., Jack, C. &amp; Eyras, E. <italic>Systematic benchmarking of tools for CpG methylation detecgtion from Nanopore sequencing.</italic><ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>, 10.5281/zenodo.4748319 (2021).) [11](Yuen, Z. W.-S., Srivastava, A., Jack, C. &amp; Eyras, E. <italic>Systematic benchmarking of tools for CpG methylation detecgtion from Nanopore sequencing.</italic><ext-link ext-link-type="uri" xlink:href="https://github.com/comprna/METEORE">https://github.com/comprna/METEORE</ext-link>, 10.5281/zenodo.4748319 (2021).) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [33](Oxford Nanopore Technologies. <italic>Evaluation of Read-mapping Characteristics from a Cas-Mediated PCR-Free Enrichment</italic> (Oxford Nanopore Technologies, 2019). <ext-link ext-link-type="uri" xlink:href="https://community.nanoporetech.com/knowledge/bioinformatics/evaluation-of-read-mapping/tutorial">https://community.nanoporetech.com/knowledge/bioinformatics/evaluation-of-read-mapping/tutorial</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34090325">34090325</a> <b>(1)</b> SnoRNA copy regulation affects family size, genomic location and family abundance levels<ul><li> Datasets were then analyzed using a succession of bioinformatics tools implemented in a reproducible <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR61"><u>61</u></xref>] available at <ext-link ext-link-type="uri" xlink:href="http://gitlabscottgroup.med.usherbrooke.ca/lafced/Stringent_RNA_seq/tree/master">http://gitlabscottgroup.med.usherbrooke.ca/lafced/Stringent_RNA_seq/tree/master</ext-link>.<li> [61](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34088344">34088344</a> <b>(2)</b> Annotation of snoRNA abundance across human tissues reveals complex snoRNA-host gene relationships<ul><li>All RNA abundance datasets were generated using a succession of bioinformatics tools regrouped in a reproducible <b>Snakemake</b> workflow [<xref ref-type="bibr" rid="CR48"><u>48</u></xref>].<li> All details about parameters and tools used can be found in the <b>Snakemake</b> workflow at <ext-link ext-link-type="uri" xlink:href="https://github.com/etiennefc/TGIRT_Seq_pipeline.git">https://github.com/etiennefc/TGIRT_Seq_pipeline.git</ext-link> [<xref ref-type="bibr" rid="CR49"><u>49</u></xref>], but are also briefly described below.<li> [48](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [49](Fafard-Couture &#xC9;. TGIRT-Seq pipeline. GitHub. 2021; Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/etiennefc/TGIRT_Seq_pipeline">https://github.com/etiennefc/TGIRT_Seq_pipeline</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=Fafard-Couture &#xC9;. TGIRT-Seq pipeline. GitHub. 2021; Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/etiennefc/TGIRT_Seq_pipeline">https://github.com/etiennefc/TGIRT_Seq_pipeline</ext-link>.">Fafard-Couture &#xC9;. TGIRT-Seq pipeline. GitHub. 2021; Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/etiennefc/TGIRT_Seq_pipeline">https://github.com/etiennefc/TGIRT_Seq_pipeline</ext-link>.</a> <b>(1)</b> Annotation of snoRNA abundance across human tissues reveals complex snoRNA-host gene relationships<ul><li>These computational steps were implemented with the <b>Snakemake</b> workflow engine (<xref rid="B44" ref-type="bibr"><u>44</u></xref>) to create a documented and reproducible pipeline which is freely available (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/jcmcnch/MGPrimerEval" ext-link-type="uri">https://github.com/jcmcnch/MGPrimerEval</ext-link>).<li> [44](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(5)</b> Annotation of snoRNA abundance across human tissues reveals complex snoRNA-host gene relationships<ul><li> <monospace>pyrpipe</monospace> pipelines can be easily scaled using a workflow manager, including the popular <b>Snakemake</b> (<xref rid="B7" ref-type="bibr"><u>7</u></xref>), NextFlow (<xref rid="B8" ref-type="bibr"><u>8</u></xref>) or Toil (<xref rid="B14" ref-type="bibr"><u>14</u></xref>).<li>We implemented <monospace>pyrpipe</monospace> within two workflow management systems, <b>Snakemake</b> (<xref rid="B7" ref-type="bibr"><u>7</u></xref>) and NextFlow (<xref rid="B8" ref-type="bibr"><u>8</u></xref>), selecting these specifically because they are widely used by the bioinformatics community (<xref rid="B27" ref-type="bibr"><u>27</u></xref>).<li> <b>Snakemake</b> and NextFlow were independently used to implement, manage and execute the pipeline for multiple RNA-Seq samples in parallel on a single cluster.<li> The data were downloaded from NCBI-SRA and processed using <monospace>pyrpipe</monospace> integrated into the workflow manager, <b>Snakemake</b>.<li> For example, <b>Snakemake</b> uses a &#x2018;pull-based&#x2019; strategy to check for specific output files and schedule jobs accordingly (<xref rid="B6" ref-type="bibr"><u>6</u></xref>,<xref rid="B7" ref-type="bibr"><u>7</u></xref>), whereas Nextflow uses a &#x2018;push-based&#x2019; scheme in which a &#x2018;process&#x2019; defined in the workflow pushes its outputs to downstream &#x2018;processes&#x2019; (<xref rid="B8" ref-type="bibr"><u>8</u></xref>,<xref rid="B38" ref-type="bibr"><u>38</u></xref>).<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a>) [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31029061">31029061</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [38](<person-group person-group-type="author"><string-name><surname>Strozzi</surname><given-names>F.</given-names></string-name>, <string-name><surname>Janssen</surname><given-names>R.</given-names></string-name>, <string-name><surname>Wurmus</surname><given-names>R.</given-names></string-name>, <string-name><surname>Crusoe</surname><given-names>M.R.</given-names></string-name>, <string-name><surname>Githinji</surname><given-names>G.</given-names></string-name>, <string-name><surname>Di&#xA0;Tommaso</surname><given-names>P.</given-names></string-name>, <string-name><surname>Belhachemi</surname><given-names>D.</given-names></string-name>, <string-name><surname>M&#xF6;ller</surname><given-names>S.</given-names></string-name>, <string-name><surname>Smant</surname><given-names>G.</given-names></string-name>, <string-name><surname>de&#xA0;Ligt</surname><given-names>J.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Scalable workflows and reproducible data analysis for genomics</article-title>. <source>Evolutionary Genomics</source>. <year>2019</year>; <publisher-name>Springer</publisher-name><fpage>723</fpage>&#x2013;<lpage>745</lpage>.) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34205926">34205926</a> <b>(1)</b> Microbiota and Metabolite Modifications after Dietary Exclusion of Dairy Products and Reduced Consumption of Fermented Food in Young and Older Men<ul><li> Reads were processed as described previously [<xref rid="B32-nutrients-13-01905" ref-type="bibr"><u>32</u></xref>], using an in-house <b>Snakemake</b> pipeline (<uri xlink:href="https://github.com/metagenlab/microbiome16S_pipeline">https://github.com/metagenlab/microbiome16S_pipeline</uri> accessed on 1 March 2021, release v.0.9.15, [<xref rid="B33-nutrients-13-01905" ref-type="bibr"><u>33</u></xref>]).<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28558854">28558854</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35224210">35224210</a> <b>(1)</b> Activation of regulatory T cells triggers specific changes in glycosylation associated with Siglec-1-dependent inflammatory responses<ul><li> The sequencing data were processed using a <b>snakemake</b> script in a conda environment.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34220740">34220740</a> <b>(6)</b> Decentralized Investigation of Bacterial Outbreaks Based on Hashed cgMLST<ul><li><b>Snakemake</b> workflow.<li>The workflow consists of a <b>Snakemake</b> pipeline (<xref rid="B17" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) with a set of assemblies (or sequencing reads) as input listed in a sample sheet.<li> <b>Snakemake</b> resolves all dependencies in a directed acyclic graph (DAG), runs all necessary components for each sample, merges all results, and renders the final HTML report using Rmarkdown (<xref rid="B29" ref-type="bibr"><u>rmarkdown, 2020</u></xref>) (see also <xref ref-type="supplementary-material" rid="FS1"><u>Supplementary Figure 1</u></xref>).<li> For the usage of chewBBACA in a <b>Snakemake</b> workflow&#x2014;and in particular to allow the continuous addition of samples&#x2014;a fixed version (2.12.0) was included in the <italic>chewieSnake</italic> repository with minor adaptations to their output: chewBBACA is called for one sample at a time, and the output is stored in a folder with the sample&#x2019;s name.<li>Importantly, if only reads are available for analysis, <italic>chewieSnake</italic> can also be given a set of reads as input, which it then trims using fastp (<xref rid="B3" ref-type="bibr"><u>Chen et al., 2018</u></xref>) and subsequently assembles into draft genomes using shovill<sup><xref ref-type="fn" rid="footnote11"><u>11</u></xref></sup>, before proceeding with the <b>Snakemake</b> workflow.<li> The reason for these differences lies in the different order of execution of the sample allele calling (which is not pre-determined by <b>Snakemake</b> but randomly assigned on each execution) and the treatment of novel alleles by chewBBACA.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [rmarkdown, 2020](<collab>rmarkdown</collab> (<year>2020</year>). <source><italic>Dynamic rmarkdown: Dynamic Documents for R. R package version 2.5..</italic></source> Available online at: <ext-link ext-link-type="uri" xlink:href="https://rmarkdown.rstudio.com">https://rmarkdown.rstudio.com</ext-link>) [Chen et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30423086">30423086</a>) [11](<a href="https://github.com/tseemann/shovill">https://github.com/tseemann/shovill</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34050149">34050149</a> <b>(1)</b> Retention time prediction using neural networks increases identifications in crosslinking mass spectrometry<ul><li> The searches were run with the workflow system <b>snakemake</b><sup><xref ref-type="bibr" rid="CR53"><u>53</u></xref></sup>.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34072237">34072237</a> <b>(1)</b> Multi-Omics Model Applied to Cancer Genetics<ul><li> Examples of such systems include Galaxy [<xref rid="B50-ijms-22-05751" ref-type="bibr"><u>50</u></xref>], <b>Snakemake</b> [<xref rid="B51-ijms-22-05751" ref-type="bibr"><u>51</u></xref>], Nextflow [<xref rid="B52-ijms-22-05751" ref-type="bibr"><u>52</u></xref>], and the general-purpose Common Workflow Language [<xref rid="B53-ijms-22-05751" ref-type="bibr"><u>53</u></xref>].<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=20738864">20738864</a>) [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [53](          <person-group person-group-type="author">            <name>              <surname>Peter</surname>              <given-names>A.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>R.C.</given-names>            </name>            <name>              <surname>Neboj&#x161;a</surname>              <given-names>T.</given-names>            </name>            <name>              <surname>Brad</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>John</surname>              <given-names>C.</given-names>            </name>            <name>              <surname>Michael</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Andrey</surname>              <given-names>K.</given-names>            </name>            <name>              <surname>Dan</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Herv&#xE9;</surname>              <given-names>M.</given-names>            </name>            <name>              <surname>Maya</surname>              <given-names>N.</given-names>            </name>            <etal/>          </person-group>          <article-title>Common Workflow Language, v1.0</article-title>          <year>2016</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://escholarship.org/uc/item/25z538jj">https://escholarship.org/uc/item/25z538jj</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-05-30">(accessed on 30 May 2020)</date-in-citation>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34031412">34031412</a> <b>(3)</b> The transcriptional landscape of a rewritten bacterial genome reveals control elements and genome design principles<ul><li> The pipeline itself as well as the algorithms in particular have been implemented as a modular <b>Snakemake</b> v5.19.0 (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) pipeline.<li> The benchmark is set up as a high-throughput, large-scale <b>Snakemake</b> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) workflow.<li> Each meta node is a <b>Snakemake</b> pipeline itself, exposing a defined application programming interface (API), thus, making them interchangeable and reusable.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34063961">34063961</a> <b>(3)</b> Anchored Phylogenomics, Evolution and Systematics of Elateridae: Are All Bioluminescent Elateroidea Derived Click Beetles?<ul><li>A bioinformatics pipeline, heavily drawing upon elements of the Phyluce package [<xref rid="B31-biology-10-00451" ref-type="bibr"><u>31</u></xref>], was developed using <b>Snakemake</b> [<xref rid="B32-biology-10-00451" ref-type="bibr"><u>32</u></xref>] to input raw Illumina reads and output aligned target loci for various target enrichment projects.<li> This pipeline is available at <uri xlink:href="https://github.com/AAFC-BICoE/<b>snakemake</b>-partial-genome-pipeline">https://github.com/AAFC-BICoE/<b>snakemake</b>-partial-genome-pipeline</uri> (accessed on 19 May 2021).<li> [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26530724">26530724</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34055480">34055480</a> <b>(1)</b> ProkEvo: an automated, reproducible, and scalable framework for high-throughput bacterial population genomics analyses<ul><li> While Pegasus is a versatile WMS, other WMS such as Nextflow (<xref rid="ref-33" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>) and <b>Snakemake</b> (<xref rid="ref-60" ref-type="bibr"><u>Koster &amp; Rahmann, 2012</u></xref>) are more commonly used for bioinformatics applications.<li> [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Koster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34016042">34016042</a> <b>(1)</b> Genomic epidemiology of SARS-CoV-2 in Esteio, Rio Grande do Sul, Brazil<ul><li>Reference mapping and consensus calling was performed using an in-house developed pipeline managed with <b>Snakemake</b> [<xref ref-type="bibr" rid="CR80"><u>80</u></xref>].<li> [80](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34011275">34011275</a> <b>(3)</b> Profile hidden Markov model sequence analysis can help remove putative pseudogenes from DNA barcoding and metabarcoding datasets<ul><li> We have also made available a&#xA0;multi-marker metabarcode <b>snakemake</b>&#xA0;pipeline that processes paired-end Illumina reads that provides a pseudogene filtering step for protein coding markers called MetaWorks that can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/terrimporter/MetaWorks">https://github.com/terrimporter/MetaWorks</ext-link>.<li> This pipeline runs in a conda environment with a <b>snakemake</b> pipeline.<li> <b>Snakemake</b> is a python-based workflow manager [<xref ref-type="bibr" rid="CR92"><u>92</u></xref>].<li> [92](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34011274">34011274</a> <b>(2)</b> Investigating the impact of reference assembly choice on genomic analyses in a cattle breed<ul><li> Alignment, coverage, variant calling, imputation, annotation and analysis workflows were implemented as described below using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR54"><u>54</u></xref>] (version 5.10.0).<li> Python 3.7.4 has been used for running custom scripts as well as for submission and generation of <b>Snakemake</b> workflows.<li> [54](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34011996">34011996</a> <b>(1)</b> Unique evolutionary trajectories of breast cancers with distinct genomic and spatial heterogeneity<ul><li>We implemented the complete assembly and analysis pipeline, containing all steps described above, in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR63"><u>63</u></xref></sup> and used Bioconda<sup><xref ref-type="bibr" rid="CR64"><u>64</u></xref></sup> to manage software.<li> [63](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34069990">34069990</a> <b>(1)</b> Dadaist2: A Toolkit to Automate and Simplify Statistical Analysis and Plotting of Metabarcoding Experiments<ul><li>DadaSnake is a workflow written in <b>SnakeMake</b> [<xref rid="B16-ijms-22-05309" ref-type="bibr"><u>16</u></xref>] which wraps DADA2 denoising, but involves a more complex installation and mainly focuses on the primary analysis (with limited support for downstream numerical ecology).<li> [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34006306">34006306</a> <b>(1)</b> Infection, dissemination, and transmission efficiencies of Zika virus in <italic>Aedes aegypti</italic> after serial passage in mosquito or mammalian cell lines or alternating passage in both cell types<ul><li>Next-generation sequencing data were analyzed using a custom <b>Snakemake</b> [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>] workflow, designated as &#x201C;RNA virus Population Genetics (RPG) Workflow.&#x201D; Briefly, Read 1 and Read 2.fastq files from paired-end Illumina MiSeq data were trimmed to remove adaptor sequences and quality trimming of phred scores&#x2009;&lt;&#x2009;30 from the 3&#x2032; and 5&#x2032; read ends using cutadapt [<xref ref-type="bibr" rid="CR39"><u>39</u></xref>].<li> [38](K&#xF6;ster J, Rahmann S. Corrigendum: Snakemake&#x2014;a scalable bioinformatics workflow engine (Bioinformatics (2012) 28:19 (2520&#x2013;2522) 10.1093/bioinformatics/bts480). Bioinformatics. 2018;34(20):3600.) [39](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23066108">23066108</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34414351">34414351</a> <b>(4)</b> mikropml: User-Friendly R Package for Supervised Machine Learning Pipelines<ul><li> We also provide <ext-link ext-link-type="uri" xlink:href="http://www.schlosslab.org/mikropml/articles/index.html">vignettes</ext-link> and an <ext-link ext-link-type="uri" xlink:href="https://github.com/SchlossLab/mikropml-<b>snakemake</b>-workflow">example <b>Snakemake</b> workflow</ext-link> (<xref rid="R10" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>) to showcase how to run an ideal ML pipeline with multiple different train/test data splits.<li> <xref rid="R22" ref-type="bibr"><u>Top&#xE7;uo&#x11F;lu et al., 2020</u></xref>), we provide examples of how to run_ml() many times with different train/test splits and how to get summary information about model performance on <ext-link ext-link-type="uri" xlink:href="http://www.schlosslab.org/mikropml/articles/parallel.html">a local computer</ext-link> or on a high-performance computing cluster using a <ext-link ext-link-type="uri" xlink:href="https://github.com/SchlossLab/mikropml-<b>snakemake</b>-workflow"><b>Snakemake</b> workflow</ext-link>.<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Top&#xE7;uo&#x11F;lu et al., 2020](<name><surname>Top&#xE7;uo&#x11F;lu</surname><given-names>BD</given-names></name>, <name><surname>Lesniak</surname><given-names>NA</given-names></name>, <name><surname>Ruffin</surname><given-names>MT</given-names></name>, <name><surname>Wiens</surname><given-names>J</given-names></name>, &amp; <name><surname>Schloss</surname><given-names>PD</given-names></name> (<year>2020</year>). <article-title>A Framework for Effective Application of Machine Learning to Microbiome-Based Classification Problems</article-title>. <source>mBio</source>, <volume>11</volume>(<issue>3</issue>). <pub-id pub-id-type="doi">10.1128/mBio.00434-20</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<name><surname>Top&#xE7;uo&#x11F;lu</surname><given-names>BD</given-names></name>, <name><surname>Lesniak</surname><given-names>NA</given-names></name>, <name><surname>Ruffin</surname><given-names>MT</given-names></name>, <name><surname>Wiens</surname><given-names>J</given-names></name>, &amp; <name><surname>Schloss</surname><given-names>PD</given-names></name> (<year>2020</year>). <article-title>A Framework for Effective Application of Machine Learning to Microbiome-Based Classification Problems</article-title>. <source>mBio</source>, <volume>11</volume>(<issue>3</issue>). <pub-id pub-id-type="doi">10.1128/mBio.00434-20</pub-id>"><name><surname>Top&#xE7;uo&#x11F;lu</surname><given-names>BD</given-names></name>, <name><surname>Lesniak</surname><given-names>NA</given-names></name>, <name><surname>Ruffin</surname><given-names>MT</given-names></name>, <name><surname>Wiens</surname><given-names>J</given-names></name>, &amp; <name><surname>Schloss</surname><given-names>PD</given-names></name> (<year>2020</year>). <article-title>A Framework for Effective Application of Machine Learning to Microbiome-Based Classification Problems</article-title>. <source>mBio</source>, <volume>11</volume>(<issue>3</issue>). <pub-id pub-id-type="doi">10.1128/mBio.00434-20</pub-id></a> <b>(1)</b> mikropml: User-Friendly R Package for Supervised Machine Learning Pipelines<ul><li> Refgenie has already been adapted for easy use in Galaxy (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) and <b>Snakemake</b> (<xref rid="B21" ref-type="bibr"><u>21</u></xref>) workflow systems, and we are interested in continuing to develop integrations with similar systems to make it easier for users to develop workflows that make use of refgenie reference data.<li> [20](<person-group person-group-type="author"><string-name><surname>VijayKrishna</surname><given-names>N.</given-names></string-name>, <string-name><surname>Joshi</surname><given-names>J.</given-names></string-name>, <string-name><surname>Coraor</surname><given-names>N.</given-names></string-name>, <string-name><surname>Hillman-Jackson</surname><given-names>J.</given-names></string-name>, <string-name><surname>Bouvier</surname><given-names>D.</given-names></string-name>, <string-name><surname>Beek</surname><given-names>M.</given-names></string-name>, <string-name><surname>van&#xA0;den&#xA0;Eguinoa</surname><given-names>I.</given-names></string-name>, <string-name><surname>Coppens</surname><given-names>F.</given-names></string-name>, <string-name><surname>Golitsynskiy</surname><given-names>S.</given-names></string-name>, <string-name><surname>Stolarczyk</surname><given-names>M.</given-names></string-name><etal>et al</etal>.</person-group><article-title>Expanding the galaxy&#x2019;s reference data</article-title>. <year>2020</year>; <comment>bioRxiv doi:</comment><comment>10 October 2020, preprint: not peer reviewed</comment><pub-id pub-id-type="doi">10.1101/2020.10.09.327114</pub-id>.) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33930288">33930288</a> <b>(4)</b> The genomic history of the Aegean palatial civilizations<ul><li> <xref rid="bib116" ref-type="bibr"><u>Gouy et&#xA0;al., 2010</u></xref></td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://doua.prabi.fr/software/seaview" id="intref0205">http://doua.prabi.fr/software/seaview</ext-link>                  </td>                </tr>                <tr>                  <td><b>Snakemake</b> v5.3.0.<li> <xref rid="bib116" ref-type="bibr"><u>Gouy et&#xA0;al., 2010</u></xref></td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="http://doua.prabi.fr/software/seaview" id="intref0205">http://doua.prabi.fr/software/seaview</ext-link>                  </td>                </tr>                <tr>                  <td><b>Snakemake</b> v5.3.0</td>                  <td>(<xref rid="bib211" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>)</td>                  <td>                    <ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/v5.3.0/" id="intref0210">https://<b>snakemake</b>.readthedocs.io/en/v5.3.0/</ext-link>                  .<li>Sequencing reads from screening, WGS, nuclear and mtDNA capture experiments, as well as those from eight published genomes (<xref rid="mmc2" ref-type="supplementary-material"><u>Table S1</u></xref>), were mapped with an in-house mapping pipeline implemented in the workflow manager <b>snakemake</b> (<xref rid="bib211" ref-type="bibr"><u>M&#xF6;lder et&#xA0;al., 2021</u></xref>) and consisting of the following steps.<li> [Gouy et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19854763">19854763</a>) [Gouy et&#xA0;al., 2010](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19854763">19854763</a>) [M&#xF6;lder et&#xA0;al., 2021](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>K.P.</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M.B.</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>C.H.</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>S.O.</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A.</given-names>            </name>          </person-group>          <article-title>Sustainable data analysis with Snakemake</article-title>          <source>F1000Res.</source>          <volume>10</volume>          <year>2021</year>          <fpage>33</fpage>        ) [M&#xF6;lder et&#xA0;al., 2021](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F.</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>K.P.</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B.</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M.B.</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>C.H.</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V.</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S.</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>S.O.</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A.</given-names>            </name>          </person-group>          <article-title>Sustainable data analysis with Snakemake</article-title>          <source>F1000Res.</source>          <volume>10</volume>          <year>2021</year>          <fpage>33</fpage>        ) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33976176">33976176</a> <b>(1)</b> Taxonomic signatures of cause-specific mortality risk in human gut microbiome<ul><li>We analysed shotgun metagenomic sequences using a pipeline built with the <b>Snakemake</b> bioinformatics workflow library<sup><xref ref-type="bibr" rid="CR32"><u>32</u></xref>,<xref ref-type="bibr" rid="CR33"><u>33</u></xref></sup>.<li> [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=15040260">15040260</a>) [33](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18482721">18482721</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33971810">33971810</a> <b>(2)</b> The diversity of opsins in Lake Baikal amphipods (Amphipoda: Gammaridae)<ul><li> and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR100"><u>100</u></xref>, <xref ref-type="bibr" rid="CR101"><u>101</u></xref>].<li> Prerequisites are general-purpose software packages Python3, conda, and <b>Snakemake</b> [<xref ref-type="bibr" rid="CR100"><u>100</u></xref>].<li> [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [101](          <person-group person-group-type="author">            <name>              <surname>M&#xF6;lder</surname>              <given-names>F</given-names>            </name>            <name>              <surname>Jablonski</surname>              <given-names>KP</given-names>            </name>            <name>              <surname>Letcher</surname>              <given-names>B</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>MB</given-names>            </name>            <name>              <surname>Tomkins-Tinch</surname>              <given-names>CH</given-names>            </name>            <name>              <surname>Sochat</surname>              <given-names>V</given-names>            </name>            <name>              <surname>Forster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Lee</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Twardziok</surname>              <given-names>SO</given-names>            </name>            <name>              <surname>Kanitz</surname>              <given-names>A</given-names>            </name>            <etal/>          </person-group>          <article-title>Sustainable data analysis with snakemake</article-title>          <source>Research</source>          <year>2021</year>          <volume>10</volume>          <issue>33</issue>          <fpage>33</fpage>        ) [100](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33972668">33972668</a> <b>(1)</b> Mitofusin-2 boosts innate immunity through the maintenance of aerobic glycolysis and activation of xenophagy in mice<ul><li> We used the <b>Snakemake</b> workflow to process the raw reads<sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup>; FastQC (version 0.11.9) was used to preprocess and control quality of raw reads<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>; Cutadapt (version 2.1) was used to trim of adapter sequence<sup><xref ref-type="bibr" rid="CR51"><u>51</u></xref></sup>; Trimmed reads in FASTQ format were quantified at the transcript level using Salmon<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup> against an Ensemble reference mouse genome (GRCm38 March 2020).<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [50](Andrews, S. <italic>FastQC: A Quality Control Tool for High Throughput Sequence Data</italic>. Babraham Bioinformatics, Babraham Institute (2019).) [51](Martin, M. Cutadapt removes adapter sequences from high-throughput sequencing reads.&#xA0;<italic>EMBnet.journal</italic>,&#xA0;<bold>17</bold>, 3 (2011).) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28263959">28263959</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34040632">34040632</a> <b>(1)</b> A Benchmark of Genetic Variant Calling Pipelines Using Metagenomic Short-Read Sequencing<ul><li>The pipelines used for simulation and variant calling in simulated data and for variant calling in real data were written in <b>Snakemake</b> (v5.9.1) (<xref rid="B15" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and can be found, together with the plotting and statistical analyses scripts, in our Github repository<sup><xref ref-type="fn" rid="footnote5"><u>5</u></xref></sup>.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [5](<a href="https://www.frontiersin.org/articles/10.3389/fgene.2021.648229/full#supplementary-material">https://www.frontiersin.org/articles/10.3389/fgene.2021.648229/full#supplementary-material</a>) </ul></p>
</ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34093996">34093996</a> <b>(1)</b> RiboDoc: A Docker-based package for ribosome profiling analysis<ul><li> The tool itself requires no additional installation and all the pipeline is managed by <b>snakemake</b> (v6.1.1).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34027400">34027400</a> <b>(1)</b> Scalable Declarative HEP Analysis Workflows for Containerised Compute Clouds<ul><li> Several such systems are actively used in experimental particle physics, e.g., <b>Snakemake</b> (M&#xF6;lder et al., <xref rid="B26" ref-type="bibr"><u>2021</u></xref>).<li> [2021](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31157322">31157322</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33961543">33961543</a> <b>(1)</b> <italic>bla</italic><sub>OXA-48</sub>-like genome architecture among carbapenemase-producing <italic>Escherichia coli</italic> and <italic>Klebsiella pneumoniae</italic> in the Netherlands<ul><li>For annotation a Conda environment was set up with packages to facilitate a <b>Snakemake</b> pipeline which could process samples in bulk, and perform initial annotation with Prokka and enhancement with <sc>blast</sc>+ [<xref rid="R26" ref-type="bibr"><u>26, 27</u></xref>].<li> [26, 27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33951731">33951731</a> <b>(1)</b> Comparison of sparse biclustering algorithms for gene expression datasets<ul><li> To make the code easier to understand and run, we organised the workflow using the bioinformatics pipeline tool <b>snakemake</b> [<xref rid="ref20" ref-type="bibr"><u>20</u></xref>] and kept track of dependencies using the open source package management tool conda [<xref rid="ref2" ref-type="bibr"><u>2</u></xref>].<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [2](Anaconda Software Distribution. Anaconda Documentation.Anaconda Inc. 2020. Available from: <ext-link xlink:href="https://docs.anaconda.com/" ext-link-type="uri">https://docs.anaconda.com/</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33953362">33953362</a> <b>(1)</b> Ecologically coherent population structure of uncultivated bacterioplankton<ul><li> Briefly (more information provided in the online documentation), Input_POGENOM is a <b>Snakemake</b> [<xref ref-type="bibr" rid="CR84"><u>84</u></xref>] pipeline that uses Bowtie2 [<xref ref-type="bibr" rid="CR76"><u>76</u></xref>] for read mapping to the reference genome and Freebayes [<xref ref-type="bibr" rid="CR80"><u>80</u></xref>] for variant calling.<li> [84](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [76](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22388286">22388286</a>) [80](Garrison E, Marth G. Haplotype-based variant detection from short-read sequencing. 2012; arXiv:1207.3907 [q-bioGN].) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33953180">33953180</a> <b>(8)</b> Multiple roles of H2A.Z in regulating promoter chromatin architecture in human cells<ul><li> Detailed analysis scripts implemented in &#x201C;<b>snakemake</b>&#x201D; and R scripts can be found at: <ext-link ext-link-type="uri" xlink:href="https://github.com/JCSMR-Tremethick-Lab/MCF10APromoterAnalysis">https://github.com/JCSMR-Tremethick-Lab/MCF10APromoterAnalysis</ext-link><sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup>.<li>A &#x201C;<b>snakemake</b>&#x201D; implementation of the peak calling pipeline can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/JCSMR-TremethickLab/MCF10APromoterAnalysis/blob/master/<b>snakemake</b>/rules/shortFragments_macs2_peak_calling.py">https://github.com/JCSMR-TremethickLab/MCF10APromoterAnalysis/blob/master/<b>snakemake</b>/rules/shortFragments_macs2_peak_calling.py</ext-link><sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup>.<li> The &#x201C;<b>snakemake</b>&#x201D; script driving this abridged MEME analysis pipeline can be found at <ext-link ext-link-type="uri" xlink:href="https://github.com/JCSMR-Tremethick-Lab/MCF10APromoterAnalysis/blob/master/<b>snakemake</b>/rules/shortFragments_meme_processing.py">https://github.com/JCSMR-Tremethick-Lab/MCF10APromoterAnalysis/blob/master/<b>snakemake</b>/rules/shortFragments_meme_processing.py</ext-link><sup><xref ref-type="bibr" rid="CR66"><u>66</u></xref></sup>.<li> A <b>snakemake</b> workflow implementing this pipeline can be accessed at <ext-link ext-link-type="uri" xlink:href="https://github.com/skurscheid/mcf10_promoter_profiling">https://github.com/skurscheid/mcf10_promoter_profiling</ext-link><sup><xref ref-type="bibr" rid="CR72"><u>72</u></xref></sup>.<li> [66](Cole, L. et al. Multiple roles of H2A.Z in regulating promoter chromatin architecture in human cells. <italic>Zenodo</italic>&#xA0;10.5281/zenodo.4478453 (2021).) [66](Cole, L. et al. Multiple roles of H2A.Z in regulating promoter chromatin architecture in human cells. <italic>Zenodo</italic>&#xA0;10.5281/zenodo.4478453 (2021).) [66](Cole, L. et al. Multiple roles of H2A.Z in regulating promoter chromatin architecture in human cells. <italic>Zenodo</italic>&#xA0;10.5281/zenodo.4478453 (2021).) [72](Cole, L. et al. Multiple roles of H2A.Z in regulating promoter chromatin architecture in human cells.<italic> Zenodo</italic>10.5281/zenodo.4478429 (2021).) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33932985">33932985</a> <b>(2)</b> SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor-powered RNA-seq analyses<ul><li> For example, <b>snakemake</b>-based [<xref ref-type="bibr" rid="CR56"><u>56</u></xref>] pipelines [<xref ref-type="bibr" rid="CR21"><u>21</u></xref>, <xref ref-type="bibr" rid="CR22"><u>22</u></xref>] allow specification of the total number of CPU cores to allocate, behaving identically on a local machine as on an arbitrary computing cluster.<li> In the case of Nextflow and <b>snakemake</b>-based workflows, individual jobs are internally submitted for each pipeline component, and typically it is implicitly left to the user to worry about time specification for every component.<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) [22](S. Orjuela, R. Huang, K. M. Hembach, M. D. Robinson, C. Soneson, ARMOR: An Automated Reproducible MOdular Workflow for Preprocessing and Differential Analysis of RNA-seq Data. <italic>G3 (Bethesda)</italic>. <bold>9</bold>, 2089&#x2013;2096 (2019).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33900177">33900177</a> <b>(1)</b> Monitoring SARS-CoV-2 Circulation and Diversity through Community Wastewater Sequencing, the Netherlands and Belgium<ul><li> We used a <b>snakemake</b> script to demultiplex fastq raw reads by using Porechop (<ext-link xlink:href="https://github.com/rrwick/Porechop" ext-link-type="uri">https://github.com/rrwick/Porechop</ext-link>), to trim primers by using Cutadapt (<xref rid="R28" ref-type="bibr"><italic>28</italic></xref>), and to perform a reference-based alignment by using minimap2 to GISAID sequence EPI_ISL_412973 (<ext-link xlink:href="https://www.gisaid.org" ext-link-type="uri">https://www.gisaid.org</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33875482">33875482</a> <b>(1)</b> The <italic>Enterprise</italic>, a massive transposon carrying <italic>Spok</italic> meiotic drive genes<ul><li> Main bioinformatic pipelines were written in <b>Snakemake</b> v.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33875482">33875482</a> <b>(1)</b> The <italic>Enterprise</italic>, a massive transposon carrying <italic>Spok</italic> meiotic drive genes<ul><li> Our ChIP-seq bioinformatic pipeline is freely available for use in <b>Snakemake</b> format (<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/mydennislab/snake-chipseq" ext-link-type="uri">https://github.com/mydennislab/snake-chipseq; last accessed May 7, 2021</ext-link>), allowing the analysis to be replicated in any cell or tissue type of interest.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33911078">33911078</a> <b>(1)</b> Extended haplotype-phasing of long-read de novo genome assemblies using Hi-C<ul><li> We implemented FALCON-Phase using the <b>Snakemake</b> language to provide flexibility and pipeline robustness<sup><xref ref-type="bibr" rid="CR35"><u>35</u></xref></sup>.<li> [35](K&#xF6;ster, J. &amp; Rahmann, S. Snakemake-a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic>10.1093/bioinformatics/bts480 (2012).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=K&#xF6;ster, J. &amp; Rahmann, S. Snakemake-a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic>10.1093/bioinformatics/bts480 (2012).">K&#xF6;ster, J. &amp; Rahmann, S. Snakemake-a scalable bioinformatics workflow engine. <italic>Bioinformatics</italic>10.1093/bioinformatics/bts480 (2012).</a> <b>(1)</b> Extended haplotype-phasing of long-read de novo genome assemblies using Hi-C<ul><li> Pipelines for processing Nanopore DRS data were built using <b>Snakemake</b> (<xref rid="bib43" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Extended haplotype-phasing of long-read de novo genome assemblies using Hi-C<ul><li><b>SnakeMake</b>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33926025">33926025</a> <b>(6)</b> Species-Specific Quality Control, Assembly and Contamination Detection in Microbial Isolate Sequences with AQUAMIS<ul><li>For this purpose, we have developed AQUAMIS, a <b>Snakemake</b> pipeline for Assembly-based Quality Assessment for Microbial Isolate Sequencing.<li>AQUAMIS is a <b>Snakemake</b> [<xref rid="B34-genes-12-00644" ref-type="bibr"><u>34</u></xref>] workflow for the user-friendly, structured primary and secondary analysis of bacterial sequencing datasets.<li> It is highly configurable using both command line options and a <b>Snakemake</b> config file.<li> <b>Snakemake</b> manages the execution of all required modules until the desired final output.<li> Through the use of <b>Snakemake</b> and additional wrappers to catch business logic errors, the pipeline is extremely failsafe and will resume automatically where failures of individual modules, samples or manual interaction have disrupted the workflow.<li> As the pipeline is based on <b>Snakemake</b>, it is also possible to run AQUAMIS in cluster and cloud environments (untested to date).<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a> <b>(1)</b> Species-Specific Quality Control, Assembly and Contamination Detection in Microbial Isolate Sequences with AQUAMIS<ul><li> Cruz-Garcia <italic toggle="yes">et&#xA0;al.</italic> (<xref rid="B17" ref-type="bibr"><u>17</u></xref>) used a <b>Snakemake</b> pipeline which included the <italic toggle="yes">edgeR</italic> (<xref rid="B18" ref-type="bibr"><u>18</u></xref>) quasi-likelihood method to identify a radiation exposure signature consisting of 46 transcripts from a high coverage (40&#x2013;75 million reads per sample) ONT PromethION PCR cDNA dataset of human blood cells.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31829904">31829904</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=19910308">19910308</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33903675">33903675</a> <b>(1)</b> Chromatin occupancy and target genes of the haematopoietic master transcription factor MYB<ul><li>All the above analysis was performed in an automated workflow using <b>snakemake</b> version (version 5.5.4)<sup><xref ref-type="bibr" rid="CR75"><u>75</u></xref></sup>.<li> [75](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33901668">33901668</a> <b>(1)</b> Diagnostic accuracy of loop-mediated isothermal amplification coupled to nanopore sequencing (LamPORE) for the detection of SARS-CoV-2 infection at scale in symptomatic and asymptomatic populations<ul><li> A proprietary Guppy/VSEARCH/<b>SnakeMake</b> pipeline (algorithmic details as per James et&#xA0;al.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33901668">33901668</a> <b>(1)</b> Diagnostic accuracy of loop-mediated isothermal amplification coupled to nanopore sequencing (LamPORE) for the detection of SARS-CoV-2 infection at scale in symptomatic and asymptomatic populations<ul><li> In particular, we used the RNA-seq pipeline (v0.9.16, <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/sequana/sequana_rnaseq" ext-link-type="uri">https://github.com/sequana/sequana_rnaseq</ext-link>) built on top of <b>Snakemake</b> 5.8.1 (<xref rid="B64" ref-type="bibr"><u>64</u></xref>).<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34035898">34035898</a> <b>(144)</b> Sustainable data analysis with <b>Snakemake</b><ul><li>Third, with systems like Nextflow<sup><xref rid="ref-17" ref-type="bibr"><u>17</u></xref></sup>, <b>Snakemake</b><sup><xref rid="ref-18" ref-type="bibr"><u>18</u></xref></sup>, BioQueue<sup><xref rid="ref-19" ref-type="bibr"><u>19</u></xref></sup>, Bpipe<sup><xref rid="ref-20" ref-type="bibr"><u>20</u></xref></sup>, ClusterFlow<sup><xref rid="ref-21" ref-type="bibr"><u>21</u></xref></sup>, Cylc<sup><xref rid="ref-22" ref-type="bibr"><u>22</u></xref></sup>, and BigDataScript<sup><xref rid="ref-23" ref-type="bibr"><u>23</u></xref></sup>, workflows are specified using a domain specific language (DSL).<li> For Nextflow and <b>Snakemake</b>, since the DSL is implemented as an extension to a generic programming language (Groovy and Python), access to the full power of the underlying programming language is maintained (e.g.<li> For example, <b>Snakemake</b> workflows can (within limits) be automatically exported to CWL, and <b>Snakemake</b> can make use of CWL tool definitions.<li> An automatic translation of any CWL workflow definition into a <b>Snakemake</b> workflow is planned as well.<li> Galaxy, Nextflow, <b>Snakemake</b>, WDL, CWL), by allowing the definition and scalable execution of each involved step, including deployment of the software stack needed for each step (e.g.<li>In this work, we show how data analysis sustainability in terms of these aspects is supported by the open source workflow management system <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io">https://<b>snakemake</b>.github.io</ext-link>).<li> Since its original publication in 2012, <b>Snakemake</b> has seen hundreds of releases and contributions (<xref ref-type="fig" rid="f2"><u>Figure 2c</u></xref>).<li> This makes <b>Snakemake</b> one of the most widely used workflow management systems in science.<li>Citations and development of <b>Snakemake</b>.<li>(<bold>a</bold>) citations by year of the original <b>Snakemake</b> article (note that the year 2020 is still incomplete at the time of writing).<li>In order to address the requirements of a potentially diverse readership, we decided to split the following content into two parts.<xref ref-type="other" rid="s2"><u>Section 2</u></xref> concisely presents <b>Snakemake</b> in terms of the aspects shown in<xref ref-type="fig" rid="f1"><u>Figure 1</u></xref>, whereas<xref ref-type="other" rid="s3"><u>section 3</u></xref> provides further details for the particularly interested reader, including how to compose multiple workflows for integrative data analysis, advanced design patterns, and additional technical details.<li>We present how <b>Snakemake</b> enables the researcher to conduct data analyses that have all the properties leading to reproducibility, transparency and adaptability.<li> Several of them are shared with other tools, while others are (at the time of writing) exclusive to <b>Snakemake</b>.<li> Finally, there are features that other workflow management systems provide while <b>Snakemake</b> does not (or not yet) offer them.<li>The central idea of <b>Snakemake</b> is that workflows are specified through decomposition into steps represented as<italic>rules</italic> (<xref ref-type="fig" rid="f3"><u>Figure 3</u></xref>).<li> Depending on the computing platform used and how <b>Snakemake</b> is configured, input and output files are either stored on disk, or in a remote storage (e.g.<li>Example <b>Snakemake</b> workflow.<li> Note that this is solely meant as a short yet comprehensive demonstration of the <b>Snakemake</b> syntax.<li>When using script integration instead of shell commands, <b>Snakemake</b> automatically inserts an object giving access to all properties of the job (e.g.<li>When using script integration instead of shell commands, <b>Snakemake</b> automatically inserts an object giving access to all properties of the job (e.g.<monospace><b>snakemake</b>.output[0]</monospace>, see<xref ref-type="fig" rid="f3"><u>Figure 3c</u></xref>).<li> By replacing wildcards with concrete values, <b>Snakemake</b> turns any rule into a job which will be executed in order to generate the defined output files.<li> For each input file of a job, <b>Snakemake</b> determines a rule that can generate it&#x2014; for example by replacing wildcards again (ambiguity can be resolved by prioritization or constraining wildcards)&#x2014; yielding another job.<li> Then, <b>Snakemake</b> goes on recursively for the latter, until all input files of all jobs are either generated by another job or already present in the used storage (e.g., on disk).<li>From this inference, <b>Snakemake</b> obtains a directed acyclic graph of jobs (DAG, see<xref ref-type="fig" rid="f3"><u>Figure 3b</u></xref>).<li><xref ref-type="fig" rid="f3"><u>Figure 3a</u></xref> illustrates all major design patterns needed to define workflows with <b>Snakemake</b>: workflow configuration (line 1), aggregations (line 5&#x2013;8), specific (line 33&#x2013;43) and generic (line 45&#x2013;53) transformations, target rules (line 3&#x2013;8), log file definition, software stack definition, as well as shell command, script, and wrapper integration.<li> From a given source workflow with already computed results that have been checked for correctness, <b>Snakemake</b> can automatically generate a suite of unit tests, which can be executed via the Pytest framework (<ext-link ext-link-type="uri" xlink:href="https://pytest.org">https://pytest.org</ext-link>).<li>The workflow definition language of <b>Snakemake</b> is designed to allow maximum readability, which is crucial for transparency and adaptability.<li>needs <b>Snakemake</b> knowledge,.<li> In particular, only six lines need <b>Snakemake</b>-specific knowledge (<xref ref-type="fig" rid="f3"><u>Figure 3d</u></xref>).<li> Since <b>Snakemake</b> supports modularization of workflow definitions (<xref ref-type="other" rid="s2.2.1"><u>subsubsection 2.2.1</u></xref>), it is however possible to hide more technical parts of the workflow definition (e.g.<li> While the used way of definition requires specific knowledge about <b>Snakemake</b> (and Python) when trying to understand the line, it can be simplified for a reader that just wants to get an overview of the workflow by replacing the statement with a function name, for example<monospace>get_bins</monospace> and moving the actual function into a separate file which is included into the main workflow definition (see<xref ref-type="other" rid="s2.2.1"><u>subsubsection 2.2.1</u></xref>).<li> Therefore, <b>Snakemake</b> provides mechanisms that help with understanding dependencies on a global level (e.g.<li>In summary, the readability of the example in<xref ref-type="fig" rid="f3"><u>Figure 3</u></xref> should be seen as an optimum a <b>Snakemake</b> workflow developer should aim for.<li> Further, such difficulties can be diminished by <b>Snakemake</b>&#x2019;s ability to automatically generate interactive reports that combine codebase and results in a visual way (<xref ref-type="other" rid="s2.4"><u>subsection 2.4</u></xref>) and thereby help to explore specific parts of the codebase (e.g.<li> <b>Snakemake</b> offers various levels of modularization that help to design a workflow in a way that ensures that a reader is not distracted from the aspects relevant for her or his interest.<li><bold>Snakefile inclusion.</bold> A <b>Snakemake</b> workflow definition (a so-called Snakefile) can include other Snakefiles via a simple<monospace>include</monospace> statement that defines their path or URL.<li><bold>Workflow composition.</bold> By declaring so-called workflow modules, <b>Snakemake</b> allows to compose multiple external workflows together, while modifying and extending them on the fly and documenting those changes transparently.<li> For the latter, <b>Snakemake</b> provides the ability to deposit and use<italic>tool wrappers</italic> in/from a central repository.<li> <b>Snakemake</b> allows the user to modularize such steps either into scripts or to craft them interactively by integrating with Jupyter notebooks (<ext-link ext-link-type="uri" xlink:href="https://jupyter.org">https://jupyter.org</ext-link>).<li> The referred script does not need any boilerplate code, and can instead directly use all properties of the job (input files, output files, wildcard values, parameters, etc.), which are automatically inserted as a global<monospace><b>snakemake</b></monospace> object before the script is executed (see<xref ref-type="fig" rid="f3"><u>Figure 3c</u></xref>).<li> Via the command line interface, it is possible to instruct <b>Snakemake</b> to open a Jupyter notebook server for editing a notebook in the context of a specific job derived from the rule that refers to the notebook.<li> Upon saving the notebook, <b>Snakemake</b> generalizes it such that other jobs from the same rule can subsequently re-use it automatically without the need for another interactive notebook session.<li><bold>Tool wrappers.</bold> Reoccurring tools or libraries can be shared between workflows via <b>Snakemake</b> tool wrappers (see<xref ref-type="fig" rid="f3"><u>Figure 3a</u></xref>, line 52&#x2013;53).<li> A central public repository (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>-wrappers.readthedocs.io">https://<b>snakemake</b>-wrappers.readthedocs.io</ext-link>) allows the community to share wrappers with each other.<li> <b>Snakemake</b> provides automatic code formatting (via the tool<monospace>snakefmt</monospace>) of workflows, together with any contained Python code.<li> Snakefmt formats plain Python parts of the codebase with the Python code formatter Black (<ext-link ext-link-type="uri" xlink:href="https://black.readthedocs.io">https://black.readthedocs.io</ext-link>), while providing its own formatting for any <b>Snakemake</b> specific syntax.<li>In addition, <b>Snakemake</b> has a built in<italic>code linter</italic> that detects code violating best practices and provides suggestions on how to improve the code.<li> As such, there are preconfigured Github actions available for both Snakefmt (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/snakefmt#github-actions">https://github.com/<b>snakemake</b>/snakefmt#github-actions</ext-link>) and the code linter (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>github-action#example-usage">https://github.com/<b>snakemake</b>/<b>snakemake</b>github-action#example-usage</ext-link>).<li> <b>Snakemake</b> itself is easily deployable via the<monospace>Conda</monospace> package manager (<ext-link ext-link-type="uri" xlink:href="https://conda.io">https://conda.io</ext-link>), as a Python package (<ext-link ext-link-type="uri" xlink:href="https://pypi.io">https://pypi.io</ext-link>), or a Docker container (<ext-link ext-link-type="uri" xlink:href="https://hub.docker.com/r/<b>snakemake</b>/<b>snakemake</b>">https://hub.docker.com/r/<b>snakemake</b>/<b>snakemake</b></ext-link>).<li> Instructions and further information can be found at<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io">https://<b>snakemake</b>.github.io</ext-link>.<li>The management of software stacks needed for individual rules is directly integrated into <b>Snakemake</b> itself, via two complementary mechanisms.<li> Upon execution, <b>Snakemake</b> will pull the requested container image and run a job inside that container using Singularity<sup><xref rid="ref-30" ref-type="bibr"><u>30</u></xref></sup>.<li> Containers already exist in centralized repositories for a wide range of scientific software applications, allowing easy integration info <b>Snakemake</b> workflows.<li> Once a workflow becomes production ready or is published, <b>Snakemake</b> offers the ability to automatically generate a containerized version.<li> For this, <b>Snakemake</b> generates a Dockerfile that deploys all defined Conda environments into the container.<li> Upon workflow execution, <b>Snakemake</b> will then use the Conda environments that are found in the container, instead of having to recreate them.<li>While processing a workflow, <b>Snakemake</b> tracks input files, output files, parameters, software, and code of each executed job.<li> Since results are included into the report, their presentation does not depend on availability of server backends, making <b>Snakemake</b> reports easily portable and archivable.<li> In the future, <b>Snakemake</b> reports will be extended to additionally follow the RO-crate standard, which will make them machine-readable and allow an integration with web services like<ext-link ext-link-type="uri" xlink:href="https://workflowhub.eu">https://workflowhub.eu</ext-link>.<li> Like many other state-of-the-art workflow management systems, <b>Snakemake</b> allows workflow execution to scale to various computational platforms (but not to combine multiple of them in a single run), ranging from single workstations to large compute servers, any common cluster middleware (like Slurm, PBS, etc.), grid computing, and cloud computing (with native support for Kubernetes, the Google Cloud Life Sciences API, Amazon AWS, TES (<ext-link ext-link-type="uri" xlink:href="https://www.ga4gh.org">https://www.ga4gh.org</ext-link>), and Microsoft Azure, the latter two in an upcoming release).<li><b>Snakemake</b>&#x2019;s design ensures that scaling a workflow to a specific platform should only entail the modification of command line parameters.<li> Via configuration profiles, it is possible to persist and share the command line setup of <b>Snakemake</b> for any computing platform (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>-profiles/doc">https://github.com/<b>snakemake</b>-profiles/doc</ext-link>).<li> The scheduling problem a workflow manager like <b>Snakemake</b> has to solve is to select the subset<italic>E</italic> &#x2286;<italic>J</italic> that leads to an efficient execution of the workflow, while not exceeding the given resources like hard drive space, I/O capacity and CPU cores.<li> <b>Snakemake</b> solves the scheduling problem at the beginning of the workflow execution and whenever a job has finished and new jobs become pending.<li><b>Snakemake</b> scheduling problem.<li> Second, high-priority jobs should be preferred (<b>Snakemake</b> allows prioritization of jobs via the workflow definition and the command line).<li> Third, temporary output files should be quickly deleted (<b>Snakemake</b> allows output files to be marked as temporary, which leads to their automatic deletion once all consuming jobs have been finished).<li>When running <b>Snakemake</b> in combination with cluster or cloud middleware (Slurm, PBS, LSF, Kubernetes, etc.), <b>Snakemake</b> does not need to govern available resources since that is handled by the middleware (hence, constraint (<xref ref-type="other" rid="M2"><u>2</u></xref>) in<xref rid="T1" ref-type="table"><u>Table 1</u></xref> can be dropped).<li> Instead, <b>Snakemake</b> can pass all information about job resource requirements (threads, memory, and disk) to the middleware, which can use this information to choose the best fitting machine for the job.<li> Nevertheless, the number of jobs that shall be queued/running at a time is usually restricted in such systems, so that <b>Snakemake</b> still has to select a subset of jobs<italic>E</italic> &#x2286;<italic>J</italic> as outlined above.<li>Mixed integer linear program for <b>Snakemake</b>&#x2019;s scheduling problem.<li> By default, <b>Snakemake</b> only considers CPU cores which we indicate with c, i.e.,<italic>R</italic> = {c}.<li> Further, let<italic>U<sub>r</sub></italic> be the free capacity of resource<italic>r</italic> &#x2208;<italic>R</italic> (initially what is provided to <b>Snakemake</b> on the command line; later what is left, given resources already used in running jobs).<li>Additional considerations and alternatives, which may be implemented in subsequent releases of <b>Snakemake</b>, are discussed in<xref ref-type="other" rid="s3.4"><u>subsection 3.4</u></xref>.<li><b>Snakemake</b> provides a new, generic approach to the problem which does not have this downside (see<xref ref-type="fig" rid="f5"><u>Figure 5</u></xref>).<li> Leveraging workflow-inherent information, <b>Snakemake</b> can calculate a hash value for each job that unambiguously captures exactly how an output file is generated, prior to actually generating the file.<li> For any output file in a workflow, if the corresponding rule is marked as eligible for caching, <b>Snakemake</b> can obtain the file from the cache if it has been created before in a different workflow or by a different user on the same system, thereby saving computation time, as well as disk space (on local machines, the file can be linked instead of copied).<li>Blockchain-hashing based between workflow caching scheme of <b>Snakemake</b>.<li> When executing a <b>Snakemake</b> workflow in a cluster or cloud setting, by default, every job will be submitted separately to the underlying queuing system.<li> To minimize such overhead, <b>Snakemake</b> offers the ability to partition the DAG of jobs into subgraphs that will be submitted together, as a single cluster or cloud job.<li> Upon execution, <b>Snakemake</b> determines connected subgraphs with the same assigned group for each job and submits such subgraphs together (as a so called<italic>group job</italic>) instead of submitting each job separately.<li> Apart from the option to mark such files as temporary so that <b>Snakemake</b> will automatically delete them once no longer needed, it is also possible to instruct <b>Snakemake</b> to never store them on disk at all by directly streaming their content from the producing job to to the consuming job.<li> <b>Snakemake</b> ensures this by submitting producer and consumer as a group job (see<xref ref-type="other" rid="s2.5.3"><u>subsubsection 2.5.3</u></xref>).<li> While a framework like <b>Snakemake</b> easily allows to develop a large integrative analyses over such diverse data types, such an analysis can also become very specific to a particular scientific project.<li> Via the declaration of external workflows as modules, integration of such separately maintained workflows is well supported in <b>Snakemake</b> (<xref ref-type="fig" rid="f7"><u>Figure 7</u></xref>).<li>Workflow composition capabilities of <b>Snakemake</b>.<li> Hence, <b>Snakemake</b>&#x2019;s workflow composition mechanism is also appropriate for the simple application of a published data analysis pipeline on new data.<li>Additional design patterns for <b>Snakemake</b> workflows.<li> <b>Snakemake</b>&#x2019;s ability to employ arbitrary Python code for defining a rule&#x2019;s input and output files already enables any kind of scattering, gathering, and aggregations in workflows.<li> Nevertheless, it can be more readable and scalable to use <b>Snakemake</b>&#x2019;s explicit support for scatter/gather processes.<li> A <b>Snakemake</b> workflow can have any number of such processes, each of which has a name (here<monospace>someprocess</monospace>).<li> <b>Snakemake</b> allows to stream output between jobs, instead of writing it to disk (see<xref ref-type="other" rid="s2.5.4"><u>subsubsection 2.5.4</u></xref>).<li> <b>Snakemake</b> allows to model defining by setting the iteration count variable as a wildcard (here<monospace>{i}</monospace>, line 16).<li> <b>Snakemake</b> workflows can be directly integrated with PEPs, thereby allowing to configure them via meta-information that is contained in the sample sheets defined by the PEP.<li> By default, <b>Snakemake</b> determines the entire DAG of jobs upfront, before the first job is executed.<li> Alternatively, if QC shall happen automatically as part of the whole workflow, one can make use of <b>Snakemake</b>&#x2019;s conditional execution capabilities.<li> <b>Snakemake</b> directly supports such benchmarking by defining a<monospace>benchmark</monospace> directive in a rule (line 7).<li> Upon execution of a job spawned from such a rule, <b>Snakemake</b> will constantly measure CPU and memory consumption, and store averaged results together with runtime information into the given TSV file.<li> In Python (and therefore also with <b>Snakemake</b>), large parameter spaces can be represented very well via Pandas<sup><xref rid="ref-38" ref-type="bibr"><u>38</u></xref>,<xref rid="ref-39" ref-type="bibr"><u>39</u></xref></sup> data frames.<li> When such a parameter space shall be explored by the application of a set of rules to each instance of the space (i.e., each row of the data frame), the idiomatic approach in <b>Snakemake</b> is to encode each data frame column as a wildcard and request all occuring combinations of values (i.e., the data frame rows), by some consuming rule.<li> Therefore, <b>Snakemake</b> provides a helper called<monospace>Paramspace</monospace>, which can wrap a Pandas data frame (this functionality was inspired by the JUDI workflow management system<sup><xref rid="ref-16" ref-type="bibr"><u>16</u></xref></sup><ext-link ext-link-type="uri" xlink:href="https://pyjudi.readthedocs.io">https://pyjudi.readthedocs.io</ext-link>).<li> The method<monospace>instance</monospace> of the<monospace>Paramspace</monospace> object, automatically returns the corresponding data frame row (as a Python<monospace>dict</monospace>) for given wildcard values (here, that method is automatically evaluated by <b>Snakemake</b> for each instance of the rule<monospace>simulate</monospace>, line 17).<li>Statements in <b>Snakemake</b> workflow definitions fall into seven categories:.<li> Only for the wrapper directive (<monospace>wrapper:</monospace>) one needs to have the <b>Snakemake</b> specific knowledge that it is possible to refer to publicly available tool wrappers.<li>Wrapper identifiers can be understood with <b>Snakemake</b> knowledge only, since one needs to know about the central tool wrapper repository of <b>Snakemake</b>.<li>Python statements will either need technical knowledge or <b>Snakemake</b> knowledge (when using the <b>Snakemake</b> API, as it happens here with the expand command, which allows to aggregate over a combination of wildcard values).<li>While the first releases of <b>Snakemake</b> used a greedy scheduler, the current implementation aims at using more efficient schedules by solving a mixed integer linear program (MILP) whenever there are free resources.<li> Nevertheless, <b>Snakemake</b> has to process dependencies between jobs, which can incur some startup time until the actual workflow is executed.<li> Then, we benchmarked runtime and memory usage of <b>Snakemake</b> for computing the entire graph of jobs on these on a single core of an Intel Core i5 CPU with 1.6 GHz, 8 GB RAM and a Lenovo PCIe SSD (LENSE20512GMSP34MEAT2TA) (<xref ref-type="fig" rid="f9"><u>Figure 9</u></xref>).<li>Runtime and memory usage of <b>Snakemake</b> while building the graph of jobs depending on the number of jobs in the workflow.<li>The <b>Snakemake</b> workflow generating the results along with a self-contained <b>Snakemake</b> report that connects results and provenance information is available at<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.4244143">https://doi.org/10.5281/zenodo.4244143</ext-link>.<li>For future releases of <b>Snakemake</b>, we plan to further improve performance, for example by making use of PyPy (<ext-link ext-link-type="uri" xlink:href="https://www.pypy.org">https://www.pypy.org</ext-link>), and by caching dependency resolution results between subsequent invocations of <b>Snakemake</b>.<li>By analyzing its readability and presenting its modularization, portability, reporting, scheduling, caching, partitioning, and streaming abilities, we have shown how <b>Snakemake</b> supports all these aspects, thereby providing a comprehensive framework for sustainable data analysis, and enabling an ergonomic, unified, combined representation of any kind of analysis step, from raw data processing, to quality control and fine-grained, interactive exploration and plotting of final results.<li><b>Snakemake</b> is available as MIT licensed open source software (homepage:<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io">https://<b>snakemake</b>.github.io</ext-link>, repository:<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>) and can be installed via Bioconda<sup><xref rid="ref-40" ref-type="bibr"><u>40</u></xref></sup>.<li>The <b>Snakemake</b> workflow generating the results presented in this work, along with the corresponding <b>Snakemake</b> report connecting results and provenance information is available at<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.4244143">https://doi.org/10.5281/zenodo.4244143</ext-link><sup><xref rid="ref-28" ref-type="bibr"><u>28</u></xref></sup>.<li> [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [19](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28633441">28633441</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28299179">28299179</a>) [22](<person-group person-group-type="author"><name><surname>Oliver</surname><given-names>HJ</given-names></name><name><surname>Shin</surname><given-names>M</given-names></name><name><surname>Sanders</surname><given-names>O</given-names></name></person-group>:<article-title>Cylc: A Work&#xFB02;ow Engine for Cycling Systems.</article-title><source><italic toggle="yes">J Open Source Softw.</italic></source><year>2018</year>;<volume>3</volume>(<issue>27</issue>):<fpage>737</fpage>.<pub-id pub-id-type="doi">10.21105/joss.00737</pub-id>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25189778">25189778</a>) [30](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28494014">28494014</a>) [38](<person-group person-group-type="author"><name><surname>McKinney</surname><given-names>W</given-names></name></person-group>:<article-title>Data Structures for Statistical Computing in Python.</article-title><source><italic toggle="yes">Proceedings of the 9th Python in Science Conference.</italic></source>Ed. by St&#xE9;fan van der Walt and Jarrod Millman.<year>2010</year>;<fpage>56</fpage>&#x2013;<lpage>61</lpage>.<pub-id pub-id-type="doi">10.25080/Majora-92bf1922-00a</pub-id>) [39](<collab>The pandas development team</collab>:<article-title>pandas-dev/pandas: Pandas</article-title>. Version latest.<year>2020</year>.<pub-id pub-id-type="doi">10.5281/zenodo.3509134</pub-id>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31882996">31882996</a>) [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a> <b>(8)</b> Sustainable data analysis with <b>Snakemake</b><ul><li>The authors present an updated version of their workflow management system <b>snakemake</b>.<li> Apart from these issues, the authors present their case well on what they consider to be important for sustainable data analysis and show that <b>snakemake</b> is both well maintained, and commonly used, not only in the bioinformatics community.<li> The section on job scheduling is very extensive, but I am also missing some details on how <b>snakemake</b> interacts with cluster scheduling software like SLURM etc.<li> The main issue I have with the manuscript is that the authors overstate the readability of <b>snakemake</b> workflows.<li> However, this workflow would also be pretty easy to read as a simple bash script, so I do not think this is an appropriate example to show how readable <b>snakemake</b> workflows are.<li> Furthermore, several of the lines they consider "trivial", in my opinion, still require some understanding of <b>snakemake</b>, make or programming.<li> I have previously worked with <b>snakemake</b>, though not recently, and in my experience the rule structure is not easy to mentally parse if one is not familiar with it.<li> I think the article would benefit if instead of trying to quantify the readability of the simple workflow, the authors would focus more on the approaches they included in <b>snakemake</b> for improving readability of workflows, i.e.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a> <b>(10)</b> Sustainable data analysis with <b>Snakemake</b><ul><li> Apart from these issues, the authors present their case well on what they consider to be important for sustainable data analysis and show that <b>snakemake</b> is both well maintained, and commonly used, not only in the bioinformatics community.<li> The section on job scheduling is very extensive, but I am also missing some details on how <b>snakemake</b> interacts with cluster scheduling software like SLURM etc.<li> For addressing execution on multiple machines, <b>Snakemake</b> entirely relies on cluster or cloud middleware.<li> The main issue I have with the manuscript is that the authors overstate the readability of <b>snakemake</b> workflows.<li> However, this workflow would also be pretty easy to read as a simple bash script, so I do not think this is an appropriate example to show how readable <b>snakemake</b> workflows are.<li> Furthermore, several of the lines they consider "trivial", in my opinion, still require some understanding of <b>snakemake</b>, make or programming.<li> I have previously worked with <b>snakemake</b>, though not recently, and in my experience the rule structure is not easy to mentally parse if one is not familiar with it.<li> I think the article would benefit if instead of trying to quantify the readability of the simple workflow, the authors would focus more on the approaches they included in <b>snakemake</b> for improving readability of workflows, i.e.<li> While it is indeed simple, a bash script that would contain all the work that <b>Snakemake</b> is doing behind the scenes (checking file consistency, scheduling, various execution backends, parallelization, software stack deployment, etc.) would be much longer and less readable.<li> Here, it is important to mention <b>Snakemake</b>'s ability to visualize dependencies and to automatically generate interactive reports.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a> <b>(18)</b> Sustainable data analysis with <b>Snakemake</b><ul><li>The authors describe <b>Snakemake</b>, a text-based pipeline execution system that builds on the principles of the Unix 'make' utility to include optimized job scheduling, extensive specification of compute resources, advanced workflow features, integration with package managers and container technologies, and result caching.<li> The authors identify where <b>Snakemake</b> exists within the large ecosystem of pipeline management systems and describe the motivating principles of <b>Snakemake</b> in terms of a hierarchy of sustainable data analysis that includes reproducibility, accessibility, transparency, and other objectives.<li> This paper updates and extends the original <b>Snakemake</b> publication of K&#xF6;ster<italic>et al</italic>., Bioinformatics 2012<sup><xref rid="rep-ref-77869-1" ref-type="bibr"><u>1</u></xref></sup>.<li> The value of <b>Snakemake</b> to the field of bioinformatics is well-established, and the paper provides usage statistics and citations to reinforce this point.<li> The paper thoroughly describes the features of <b>Snakemake</b> and the necessary background to understand their use.<li> While it is possible to understand a <b>Snakemake</b> workflow at a high level with little programming knowledge, experience with Unix, Python, and an understanding of how 'make' works are necessary to author <b>Snakemake</b> workflows.<li> They also describe in detail the job scheduling algorithm that <b>snakemake</b> uses.<li> The paper would benefit from a brief discussion of how <b>snakemake</b> interacts with load-balancing software (such as SLURM, Torque, LSF, etc.), because of the focus on the details of executing jobs.<li> This information is in the documentation on the<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.github.io/"><b>Snakemake</b> web site</ext-link>, but a conceptual overview in the paper would help the reader to understand this relationship.<li> Information necessary to install and use <b>snakemake</b> is available on the <b>snakemake</b> web site as referenced in the paper.<li> The instructions and tutorial are comprehensive and understandable, and the progressive exercises give a good feel for the <b>snakemake</b> paradigm.<li> Given the ease and popularity of container technology, it would be useful to include how to run <b>snakemake</b> from a Docker container in the tutorial.<li> I was able to do this easily from the official <b>snakemake</b> container on Dockerhub and think it would be a helpful alternative in the documentation to installing Miniconda.<li> In general, this paper provides a comprehensive introduction and overview of <b>Snakemake</b> and serves as an effective entry point to this popular system.<li> [1](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(8)</b> Sustainable data analysis with <b>Snakemake</b><ul><li> The paper would benefit from a brief discussion of how <b>snakemake</b> interacts with load-balancing software (such as SLURM, Torque, LSF, etc.), because of the focus on the details of executing jobs.<li> This information is in the documentation on the <b>Snakemake</b> web site, but a conceptual overview in the paper would help the reader to understand this relationship.<li> Information necessary to install and use <b>snakemake</b> is available on the <b>snakemake</b> web site as referenced in the paper.<li> The instructions and tutorial are comprehensive and understandable, and the progressive exercises give a good feel for the <b>snakemake</b> paradigm.<li> Given the ease and popularity of container technology, it would be useful to include how to run <b>snakemake</b> from a Docker container in the tutorial.<li> I was able to do this easily from the official <b>snakemake</b> container on Dockerhub and think it would be a helpful alternative in the documentation to installing Miniconda.<li> However, we now instead provide a Gitpod setup for the tutorial, which allows to directly run it, without any setup needed, from inside a browser window which offers both an editor and a terminal based on the Eclipse Theia IDE (see<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/tutorial/setup.html#run-tutorial-for-free-in-the-cloud-via-gitpod">here</ext-link>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Sustainable data analysis with <b>Snakemake</b><ul><li> The job queue is handled with Celery 4.4.7 and Redis, and the jobs are executed via <b>Snakemake</b> 5.31.1 (<xref rid="B29" ref-type="bibr"><u>29</u></xref>).<li> [29](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33872308">33872308</a> <b>(1)</b> MRLocus: Identifying causal genes mediating a trait through Bayesian estimation of allelic heterogeneity<ul><li> The <b>Snakemake</b> software [<xref rid="pgen.1009455.ref043" ref-type="bibr"><u>43</u></xref>] was used for automation of simulation scripts including specification of random seed for each of the simulations, in order to assist with computational reproducibility of simulations.<li> [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33942021">33942021</a> <b>(1)</b> Loss of <italic>stra8</italic> Increases Germ Cell Apoptosis but Is Still Compatible With Sperm Production in Atlantic Salmon (<italic>Salmo salar</italic>)<ul><li> FastQ sequences were analyzed with CRISPResso2 (<xref rid="B15" ref-type="bibr"><u>Clement et al., 2019</u></xref>) using the CRISPRessoPooled command in a customized <b>Snakemake</b> workflow (<xref rid="B40" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [Clement et al., 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30809026">30809026</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33858413">33858413</a> <b>(2)</b> Multi-omic profiling of pituitary thyrotropic cells and progenitors<ul><li> The RNA was checked for quality using FastQC and mapped and analyzed using the VIPER <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR81"><u>81</u></xref>].<li> The quality of the reads was checked using FastQC, aligned to the mm9 genome, and had its peaks called using the Parker lab&#x2019;s <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR85"><u>85</u></xref>].<li> [81](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) [85](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32029221">32029221</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33845886">33845886</a> <b>(3)</b> Niche partitioning in the <italic>Rimicaris exoculata</italic> holobiont: the case of the first symbiotic <italic>Zetaproteobacteria</italic><ul><li>We trimmed adaptors using bbduk from bbmap 38.22 [<xref ref-type="bibr" rid="CR26"><u>26</u></xref>] after which we processed reads using a <b>snakemake</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] workflow implemented in anvi&#x2019;o v5 (<ext-link ext-link-type="uri" xlink:href="http://merenlab.org/2018/07/09/anvio-<b>snakemake</b>-workflows/">http://merenlab.org/2018/07/09/anvio-<b>snakemake</b>-workflows/</ext-link>) [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>].<li> [26](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29073143">29073143</a>) [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26500826">26500826</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33846645">33846645</a> <b>(1)</b> Single-cell CUT&amp;Tag profiles histone modifications and transcription factors in complex tissues<ul><li>Processing pipeline was build using <b>Snakemake</b> platoform<sup><xref rid="R47" ref-type="bibr"><u>47</u></xref></sup>.<li> [47](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33846360">33846360</a> <b>(1)</b> Uncovering transcriptional dark matter via gene annotation independent single-cell RNA sequencing analysis<ul><li>The <b>snakemake</b><sup><xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup> pipeline along with the bash and R scripts used in the TAR-scRNA-seq tool is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/fw262/TAR-scRNA-seq">https://github.com/fw262/TAR-scRNA-seq</ext-link> with the identifier (10.5281/zenodo.4567436).<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33846313">33846313</a> <b>(2)</b> Go Get Data (GGD) is a framework that facilitates reproducible access to genomic data<ul><li> Furthermore, GGD can easily be integrated into reproducible processing and analysis workflow systems such as Nextflow<sup><xref ref-type="bibr" rid="CR23"><u>23</u></xref></sup> and <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR24"><u>24</u></xref></sup>.<li> (Useful for workflows like <b>Snakemake</b>).<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [24](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33823809">33823809</a> <b>(1)</b> The transcriptional landscape of a hepatoma cell line grown on scaffolds of extracellular matrix proteins<ul><li> adapter and poly(T) stretches, indexing the reference transcriptome, mapping the RNA-Seq reads to transcripts and counting the reads assigned to individual transcripts were performed with a <b>Snakemake</b> framework [<xref ref-type="bibr" rid="CR50"><u>50</u></xref>].<li> [50](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33821792">33821792</a> <b>(1)</b> Cerebrospinal fluid proteome maps detect pathogen-specific host response patterns in meningitis<ul><li>A <b>snakemake</b> workflow was created for building generic transition library (GTL) from Thermo DDA experiments.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33823902">33823902</a> <b>(2)</b> Simplitigs as an efficient and scalable representation of de Bruijn graphs<ul><li>Individual experiments were run in parallel on the Harvard Medical School O2&#xA0;cluster using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR94"><u>94</u></xref>] and SLURM.<li> <bold>Supplementary Table S7.</bold> CPU time and memory consumption (both measured by GNU Time and <b>Snakemake</b>) as a function of species, method, number of threads, and <italic>k</italic>-mer length, including the used computational node.<li> [94](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33981780">33981780</a> <b>(5)</b> CRISPAltRations: a validated cloud-based approach for interrogation of double-strand break repair mediated by CRISPR genome editing<ul><li>We developed the CRISPRAltRations software tool in python, and&#xA0;it&#xA0;plus other software tools are together managed by a <b>snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>" id="interref0010">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>) or CWL (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-language/common-workflow-language" id="interref0015">https://github.com/common-workflow-language/common-workflow-language</ext-link>) workflow manager (<xref rid="mmc1" ref-type="supplementary-material"><u>Figures S1</u></xref> and <xref rid="mmc1" ref-type="supplementary-material"><u>S14</u></xref>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33722935">33722935</a> <b>(1)</b> Subgenomic RNA identification in SARS-CoV-2 genomic sequencing data<ul><li>Periscope is a wrapper for a <b>Snakemake</b> (<xref rid="GR268110PARC19" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) workflow with a package written in Python to implement read filtering and classification and is provided with a conda environment definition.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33376190">33376190</a> <b>(7)</b> Identification of m<sup>6</sup>A residues at single-nucleotide resolution using eCLIP and an accessible custom analysis pipeline<ul><li>Straightforward analysis pipeline using <b>Snakemake</b>.<li> In comparison, our approach implements the workflow management system <b>Snakemake</b> (<xref rid="RNA078543ROBC15" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>) to streamline the process, requiring only a single configuration file and no manual installation of software libraries.<li> Specifically, as opposed to running multiple scripts one at a time, <b>Snakemake</b> workflows combine the execution of all the component commands into a human readable file that is easily modifiable.<li> If specific modifications such as METTL16-dependent m<sup>6</sup>A and m<sup>6</sup>Am are of interest, the motif filtering step and conversion event of interest can be easily modified within the <b>Snakemake</b> workflow.<li>Overview of <b>Snakemake</b> workflow.<li> The resulting reads are analyzed via a modified computational pipeline based on the original eCLIP strategy that has been converted into a <b>Snakemake</b> workflow (accessible via GitHub at <uri xlink:href="https://github.com/ajlabuc/meCLIP">https://github.com/ajlabuc/meCLIP</uri>).<li> It can be executed according to <b>Snakemake</b> guidelines using a configuration file detailing the location of the respective sequencing files and relevant genomes.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33850640">33850640</a> <b>(2)</b> Bioinformatic strategies for the analysis of genomic aberrations detected by targeted NGS panels with clinical application<ul><li> The execution of several programs chained through their inputs and outputs is managed by workflow engines such as <b>Snakemake</b> (<xref rid="ref-143" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2018</u></xref>), Bpipe (<xref rid="ref-144" ref-type="bibr"><u>Sadedin, Pope &amp; Oshlack, 2012</u></xref>), reflow (<uri xlink:href="https://github.com/grailbio/reflow">https://github.com/grailbio/reflow</uri>) and nextflow (<xref rid="ref-33" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), which also offers a set of curated pipelines under the nf-core project (<xref rid="ref-37" ref-type="bibr"><u>Ewels et al., 2020</u></xref>).<li>The in-house bioinformatic data analysis workflow consists of two independent branches: (1) a pipeline for detection of SNVs/Indels and CNVs implemented in <b>Snakemake</b>, and (2) a pipeline for the identification of IG/TR gene rearrangements and translocations.<li> [K&#xF6;ster &amp; Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [Sadedin, Pope &amp; Oshlack, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22500002">22500002</a>) [Di Tommaso et al., 2017](          <person-group person-group-type="author">            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Chatzou</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Floden</surname>              <given-names>EW</given-names>            </name>            <name>              <surname>Barja</surname>              <given-names>PP</given-names>            </name>            <name>              <surname>Palumbo</surname>              <given-names>E</given-names>            </name>            <name>              <surname>Notredame</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>Nextflow enables reproducible computational workflows</article-title>          <source>Nature Biotechnology</source>          <year>2017</year>          <volume>35</volume>          <issue>4</issue>          <fpage>316</fpage>          <lpage>319</lpage>          <pub-id pub-id-type="doi">10.1038/nbt.3820</pub-id>        ) [Ewels et al., 2020](          <person-group person-group-type="author">            <name>              <surname>Ewels</surname>              <given-names>PA</given-names>            </name>            <name>              <surname>Peltzer</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Fillinger</surname>              <given-names>S</given-names>            </name>            <name>              <surname>Patel</surname>              <given-names>H</given-names>            </name>            <name>              <surname>Alneberg</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Wilm</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Garcia</surname>              <given-names>MU</given-names>            </name>            <name>              <surname>Di Tommaso</surname>              <given-names>P</given-names>            </name>            <name>              <surname>Nahnsen</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>The nf-core framework for community-curated bioinformatics pipelines</article-title>          <source>Nature Biotechnology</source>          <year>2020</year>          <volume>38</volume>          <issue>3</issue>          <fpage>276</fpage>          <lpage>278</lpage>          <pub-id pub-id-type="doi">10.1038/s41587-020-0439-x</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33861950">33861950</a> <b>(4)</b> Emergence and rapid transmission of SARS-CoV-2 B.1.1.7 in the United States<ul><li> variant calling</td><td><xref rid="bib11" ref-type="bibr"><u>Deng et&#xA0;al., 2020</u></xref></td><td>N/A</td></tr><tr><td>Andersen Lab consensus calling for nanopore data</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://github.com/artic-network/artic-ncov2019" id="intref0055">https://github.com/artic-network/artic-ncov2019</ext-link></td></tr><tr><td><b>Snakemake</b>.<li> variant calling</td><td><xref rid="bib11" ref-type="bibr"><u>Deng et&#xA0;al., 2020</u></xref></td><td>N/A</td></tr><tr><td>Andersen Lab consensus calling for nanopore data</td><td/><td><ext-link ext-link-type="uri" xlink:href="https://github.com/artic-network/artic-ncov2019" id="intref0055">https://github.com/artic-network/artic-ncov2019</ext-link></td></tr><tr><td><b>Snakemake</b></td><td><xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref></td><td><ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/" id="intref0060">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>.<li> Consensus sequences from Illumina data were assembled using an inhouse <b>Snakemake</b> (K&#xF6;ster and Rahmann, 2012) pipeline with bwa-mem (<xref rid="bib21" ref-type="bibr"><u>Li, 2013</u></xref>) and iVar v1.2.2 (<xref rid="bib17" ref-type="bibr"><u>Grubaugh et&#xA0;al., 2019b</u></xref>).<li> [Deng et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32513865">32513865</a>) [Deng et&#xA0;al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32513865">32513865</a>) [K&#xF6;ster and Rahmann, 2012](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J.</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S.</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>          <source>Bioinformatics</source>          <year>2012</year>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) [Li, 2013](          <person-group person-group-type="author">            <name>              <surname>Li</surname>              <given-names>H.</given-names>            </name>          </person-group>          <part-title>Aligning Sequence Reads, Clone Sequences and Assembly Contigs with BWA-MEM</part-title>          <year>2013</year>          <publisher-name>Broad Institute</publisher-name>        ) [Grubaugh et&#xA0;al., 2019b](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30621750">30621750</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33785741">33785741</a> <b>(1)</b> Reprogramming of the FOXA1 cistrome in treatment-emergent neuroendocrine prostate cancer<ul><li>Sample&#x2013;sample clustering, principal component analysis, and identification of lineage-enriched peaks were performed using Cobra v2.0<sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> (<ext-link ext-link-type="uri" xlink:href="https://bitbucket.org/cfce/cobra/src/master/">https://bitbucket.org/cfce/cobra/src/master/</ext-link>), a ChIP-seq analysis pipeline implemented with <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR46"><u>46</u></xref></sup>.<li> [45](Qiu, X. et al. CoBRA: Containerized bioinformatics workflow for reproducible chip/atac-seq analysis-from differential peak calling to pathway analysis. Preprint at <italic>bioRxiv</italic>10.1101/2020.11.06.367409 (2020).) [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(10)</b> Reprogramming of the FOXA1 cistrome in treatment-emergent neuroendocrine prostate cancer<ul><li> We created <b>snakemake</b> (<xref rid="B27" ref-type="bibr"><u>27</u></xref>) pipelines for both experiments with synthetic data to enable reproducibility: <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/SilvioWeging/kASA_<b>snakemake</b>" ext-link-type="uri">https://github.com/SilvioWeging/kASA_<b>snakemake</b></ext-link> and <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/SilvioWeging/kASA_cami" ext-link-type="uri">https://github.com/SilvioWeging/kASA_cami</ext-link>, see Figure <xref rid="F4" ref-type="fig"><u>4</u></xref> for a flowchart of the general workflow inside both benchmarks.<li>Workflow of both <b>snakemake</b> benchmarks.<li>To evaluate the accuracy, performance, and robustness of kASA and mentioned state of the art tools, we designed a synthetic benchmark with <b>snakemake</b>.<li> Times were gathered with the benchmark option in <b>snakemake</b> and sorted in the same manner as in Figure <xref rid="F5" ref-type="fig"><u>5</u></xref>.<li> Times were gathered with the benchmark option in <b>snakemake</b> and sorted in the same manner as in Figure <xref rid="F5" ref-type="fig"><u>5</u></xref>.<li> The <b>Snakemake</b> pipelines are available on <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/SilvioWeging/kASA_<b>snakemake</b>" ext-link-type="uri">https://github.com/SilvioWeging/kASA_<b>snakemake</b></ext-link> and <ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://github.com/SilvioWeging/kASA_cami" ext-link-type="uri">https://github.com/SilvioWeging/kASA_cami</ext-link>.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33869975">33869975</a> <b>(1)</b> Quantum Mechanical Methods Predict Accurate Thermodynamicsof Biochemical Reactions<ul><li> Forthis, we have used <b>snakemake</b><sup><xref ref-type="bibr" rid="ref51"><u>51</u></xref></sup> as an underlyingpackage for workflow management to bind together different Python-basedcomponents ensuring scalability, portability, and fault tolerance.Within the app, the ModelSEED database is used for describing thebiochemical reactions that are typically utilized to construct a newgenome-scale model for a given organism.<li> [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33758076">33758076</a> <b>(9)</b> CellMixS: quantifying and visualizing batch effects in single-cell RNA-seq data<ul><li>The various cell-specific metrics are implemented in the CellMixS R/Bioconductor package and all code for analyses here are available at <ext-link ext-link-type="uri" xlink:href="https://github.com/almutlue/batch_<b>snakemake</b>">https://github.com/almutlue/batch_<b>snakemake</b></ext-link> and <ext-link ext-link-type="uri" xlink:href="https://github.com/almutlue/mixing_benchmark">https://github.com/almutlue/mixing_benchmark</ext-link> (<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.4312672">DOI:10.5281/zenodo.4312672</ext-link>).<li> Detailed analysis of each reference and corresponding simulations are provided at <ext-link ext-link-type="uri" xlink:href="https://almutlue.github.io/batch_<b>snakemake</b>/index.html">https://almutlue.github.io/batch_<b>snakemake</b>/index.html</ext-link>.<li> Batch characterization and simulation was run as a <b>snakemake</b> (<xref rid="bib39" ref-type="bibr"><u>39</u></xref>) pipeline.<li> Code and deployment of results are available at <ext-link ext-link-type="uri" xlink:href="https://github.com/almutlue/batch_<b>snakemake</b>">https://github.com/almutlue/batch_<b>snakemake</b></ext-link> (<ext-link ext-link-type="uri" xlink:href="https://doi.org/10.5281/zenodo.4312603">DOI:10.5281/zenodo.4312603</ext-link>) and <ext-link ext-link-type="uri" xlink:href="https://almutlue.github.io/batch_<b>snakemake</b>/index.html">https://almutlue.github.io/batch_<b>snakemake</b>/index.html</ext-link>.<li> [39](<person-group person-group-type="author"><string-name><surname>K&#xF6;ster</surname><given-names>J</given-names></string-name>, <string-name><surname>Rahmann</surname><given-names>S</given-names></string-name></person-group> (<year>2012</year>) <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>. <source>Bioinformatics</source><volume>18</volume>: <fpage>2520</fpage>&#x2013;<lpage>2522</lpage>. <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33855304">33855304</a> <b>(9)</b> A protocol for transposon insertion sequencing in <italic>Schizosaccharomyces pombe</italic> to identify factors that maintain heterochromatin<ul><li> HTtools is a <b>Snakemake</b>-based pipeline (<xref rid="bib4" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>).<li>Since HTtools is based on <b>Snakemake</b>, the entire workflow can be executed on a single machine, submitted to a cluster, or run on cloud platforms (see the <b>Snakemake</b> documentation for details on these execution methods).<li>bash test_<b>snakemake</b>.sh<list list-type="simple" id="olist0170"><list-item id="o0565"><label>6.<li>Run the workflow:</p>            </list-item>          </list>          <disp-formula id="ufd2"><b>snakemake</b> --config fn=config.yaml --cores=1</disp-formula>        .<li> Run the following (note that the command uses backticks (&#x2019;), not single quote (&#x2018;)):</p></list-item></list><disp-formula id="ufd3"><b>snakemake</b> --config fn=config.yaml -R `<b>snakemake</b> --config fn=config.yaml -- list-params-changes` --cores=1</disp-formula>.<li>The command <b>snakemake</b> --config fn=config.yaml --list-params-changes lists the files affected by any parameter changes done in the config.yaml file since the last <b>snakemake</b> execution.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33749660">33749660</a> <b>(1)</b> Genomic diversity of SARS-CoV-2 during early introduction into the Baltimore&#x2013;Washington metropolitan area<ul><li>We used a customized Nextstrain <b>Snakemake</b> pipeline (<xref rid="B12" ref-type="bibr"><u>12</u></xref>) using augur v9.0.0 on the final data set, which included the 886 Global 1K data set plus all removed regional groups (including all 114 JHHS sequences and all published Virginia sequences), resulting in 1279 sequences used in our phylogenetic analysis.<li> [12](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33744862">33744862</a> <b>(1)</b> Evolution of mammalian longevity: age-related increase in autophagy in bats compared to other mammals<ul><li> All selection tests were implemented using the Optimised High-throughput <b>Snakemake</b> Automisation of PAML (OHSNAP) pipeline [<xref rid="r5" ref-type="bibr"><u>5</u></xref>].<li> [5](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29441358">29441358</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33741650">33741650</a> <b>(1)</b> Structural variants at the <italic>BRCA1/2</italic> loci are a common source of homologous repair deficiency in high grade serous ovarian carcinoma<ul><li> Annotation of resistance genes (<italic toggle="yes">penA, porB, ponA, gyrA, parC</italic>, 23S rRNA, <italic toggle="yes">mtrA/R/C/D/E, rplD, rplV, rpmH, rpoB, rpoD</italic>) was done using either the allele annotation available in the PubMLST database or by manually aligning allele sequences from PubMLST and identifying previously reported resistance mutations or mosaicism.<xref rid="dkab082-B9" ref-type="bibr"><sup>9</sup></xref><sup>,</sup><xref rid="dkab082-B26" ref-type="bibr"><sup>26&#x2013;29</sup></xref> Raw reads were mapped against all 23S rRNA reference sequences in the PubMLST database to identify heterogeneous A2058/2059G/C2611T mutations in the four different alleles using Ariba v2.14.4.<xref rid="dkab082-B30" ref-type="bibr"><sup>30</sup></xref> <b>Snakemake</b> v5.6.0 was used for workflow management.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33736596">33736596</a> <b>(15)</b> WASP: a versatile, web-accessible single cell RNA-Seq processing platform<ul><li> WASP comprises an automated pre-processing pipeline based on <b>Snakemake</b> [<xref ref-type="bibr" rid="CR15"><u>15</u></xref>] and interactive websites implemented with R Shiny (<ext-link ext-link-type="uri" xlink:href="https://shiny.rstudio.com">https://shiny.rstudio.com</ext-link>) for result visualizations and post-processing.<li>WASP is separated into two modules for primary data processing and result visualization, which are implemented in <b>Snakemake</b> and R Shiny.<li> <b>Snakemake</b> is a workflow management system which utilizes a Python-based description language.<li> Thanks to the usage of Conda (<ext-link ext-link-type="uri" xlink:href="https://docs.conda.io/en/latest/">https://docs.conda.io/en/latest/</ext-link>) environments, <b>Snakemake</b> ensures reproducibility and facilitates scalable execution of analyses adaptable to the used hardware.<li>Reproducible pre-processing of raw data using <b>Snakemake</b>.<li> This module is implemented as a <b>Snakemake</b> workflow in order to enable reproducibility of results, and users are provided with JSON-based files documenting the quality metrics as well as results of the analysis.<li> The <b>Snakemake</b> workflow utilizes a variety of established tools for the processing including FastQC (<ext-link ext-link-type="uri" xlink:href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</ext-link>) to verify the quality of the reads, STAR [<xref ref-type="bibr" rid="CR16"><u>16</u></xref>] to map reads to the reference genome, featureCounts [<xref ref-type="bibr" rid="CR17"><u>17</u></xref>] to extract reads mapping to exons and UMI-tools [<xref ref-type="bibr" rid="CR18"><u>18</u></xref>] to remove duplicate reads originating from the same mRNA fragment (Fig.&#xA0;<xref rid="Fig1" ref-type="fig"><u>1</u></xref>a).<li> <bold>a</bold> Shows the four main pre-processing steps of the <b>Snakemake</b> workflow as orange squares and the tools used for the processing in blue squares.<li> To enhance reproducibility, users may also download a table containing all parameters of the tools used during the <b>Snakemake</b> analysis as well as the selected number of barcodes.<li> As a first step, users start the <b>Snakemake</b> workflow on a Linux-based system, providing the FASTQ file with the reads and a reference genome with corresponding annotation.<li> In order to speed up the demultiplexing, we implemented this step as a parallel process while maintaining reproducibility using <b>Snakemake</b> (Fig.<li> After preparing the input data, the <b>Snakemake</b> pipeline can be started by a single command line, with the two mandatory parameters being the number of cores to be used&#xA0;and the&#xA0;scRNA-seq protocol.<li>Following the <b>Snakemake</b> run, the result data is visualized using the dedicated Shiny application.<li> generated with the <b>Snakemake</b> workflow.<li><bold>Programming language:</bold> R, Python/<b>Snakemake</b>.<li> [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [16](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23104886">23104886</a>) [17](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24227677">24227677</a>) [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28100584">28100584</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33735276">33735276</a> <b>(1)</b> Application of a bioinformatics training delivery method for reaching dispersed and distant trainees<ul><li>Introduction to <b>Snakemake</b> and Nextflow (June 2019).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33735208">33735208</a> <b>(2)</b> Principles for data analysis workflows<ul><li> These inform project workflow, and tools, such as <b>Snakemake</b> and Nextflow, can begin to be used to improve the flexibility and reproducibility of the project [<xref rid="pcbi.1008770.ref021" ref-type="bibr"><u>21</u></xref>&#x2013;<xref rid="pcbi.1008770.ref023" ref-type="bibr"><u>23</u></xref>].<li> Tools such as Dr.Watson (<ext-link ext-link-type="uri" xlink:href="https://github.com/JuliaDynamics/DrWatson.jl">https://github.com/JuliaDynamics/DrWatson.jl</ext-link>) and <b>Snakemake</b> [<xref rid="pcbi.1008770.ref023" ref-type="bibr"><u>23</u></xref>] are designed to assist researchers with organization and reproducibility and can inform the polishing process for products emerging from any phase of the analysis (see [<xref rid="pcbi.1008770.ref022" ref-type="bibr"><u>22</u></xref>] for more discussion of reproducible workflow design and tools).<li> [21](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [22](<name><surname>Reiter</surname><given-names>T</given-names></name>, <name><surname>Brooks</surname><given-names>PT</given-names></name>, <name><surname>Irber</surname><given-names>L</given-names></name>, <name><surname>Joslin</surname><given-names>SEK</given-names></name>, <name><surname>Reid</surname><given-names>CM</given-names></name>, <name><surname>Scott</surname><given-names>C</given-names></name>, <etal>et al</etal>. <article-title>Streamlining Data-Intensive Biology With Workflow Systems</article-title>. <source>Cold Spring Harb Laboratory</source>. <year>2020</year>. <pub-id pub-id-type="doi">10.1101/2020.06.30.178673</pub-id>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33809074">33809074</a> <b>(3)</b> The Uncovered Function of the <italic>Drosophila GBA1a</italic>-Encoded Protein<ul><li> Pipeline was constructed using <b>Snakemake</b> (<uri xlink:href="https://<b>snakemake</b>.readthedocs.io/en/v3.9.1/">https://<b>snakemake</b>.readthedocs.io/en/v3.9.1/</uri>).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33712682">33712682</a> <b>(1)</b> Nutrient-driven genome evolution revealed by comparative genomics of chrysomonad flagellates<ul><li> An automated workflow with <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> processed the sequencing data.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33712636">33712636</a> <b>(1)</b> A showcase study on personalized in silico drug response prediction based on the genetic landscape of muscle invasive bladder cancer<ul><li> The pipeline for WES analysis from the retrieved bam files to unannotated variant calls is based on the framework described in Singer et al.<sup><xref ref-type="bibr" rid="CR40"><u>40</u></xref></sup>, employing the <b>Snakemake</b> workflow environment<sup><xref ref-type="bibr" rid="CR41"><u>41</u></xref></sup>.<li> [40](          <person-group person-group-type="author">            <name>              <surname>Singer</surname>              <given-names>J</given-names>            </name>            <etal/>          </person-group>          <article-title>NGS-pipe: A flexible, easily extendable and highly configurable framework for NGS analysis</article-title>          <source>Bioinf. Oxf. Engl.</source>          <year>2018</year>          <volume>34</volume>          <fpage>107</fpage>          <lpage>108</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btx540</pub-id>        ) [41](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake&#x2014;a scalable bioinformatics workflow engine</article-title>          <source>Bioinf. Oxf. Engl.</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33705383">33705383</a> <b>(1)</b> Ten simple rules for quick and dirty scientific programming<ul><li> Such scripts can be made using shell programming (Bash), computational notebooks (e.g., Jupyter [<xref rid="pcbi.1008549.ref030" ref-type="bibr"><u>30</u></xref>]), or workflow description languages such as NextFlow [<xref rid="pcbi.1008549.ref031" ref-type="bibr"><u>31</u></xref>] or <b>Snakemake</b> [<xref rid="pcbi.1008549.ref032" ref-type="bibr"><u>32</u></xref>].<li> [30](<name><surname>Kluyver</surname><given-names>T</given-names></name>, <name><surname>Ragan-Kelley</surname><given-names>B</given-names></name>, <name><surname>P&#xE9;rez</surname><given-names>F</given-names></name>, <name><surname>Granger</surname><given-names>BE</given-names></name>, <name><surname>Bussonnier</surname><given-names>M</given-names></name>, <name><surname>Frederic</surname><given-names>J</given-names></name>, <etal>et al</etal>. <source>Jupyter Notebooks-a publishing format for reproducible computational workflows</source>. In: <publisher-name>ELPUB</publisher-name>; <year>2016</year>. p. <fpage>87</fpage>&#x2013;<lpage>90</lpage>.) [31](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [32](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33815688">33815688</a> <b>(1)</b> Nanopore sequencing and its application to the study of microbial communities<ul><li> Features taxonomic and functional analysis.</td></tr><tr><td>NanoCLUST</td><td><xref rid="b0125" ref-type="bibr"><u>[25]</u></xref></td><td>Targeted</td><td>Analysis pipeline for UMAP-based classification of amplicon-based full-length 16S rRNA nanopore reads.</td></tr><tr><td>Reticulatus</td><td><ext-link ext-link-type="uri" xlink:href="https://github.com/SamStudio8/reticulatus" id="ir015">https://github.com/SamStudio8/reticulatus</ext-link></td><td>Shotgun</td><td><b>Snakemake</b>-based pipeline for assembly and polishing of long genomes from long nanopore reads.<li> [[25]](          <person-group person-group-type="author">            <name>              <surname>Rodr&#xED;guez-P&#xE9;rez</surname>              <given-names>H.</given-names>            </name>            <name>              <surname>Ciuffreda</surname>              <given-names>L.</given-names>            </name>            <name>              <surname>Flores</surname>              <given-names>C.</given-names>            </name>          </person-group>          <article-title>NanoCLUST: a species-level analysis of 16S rRNA nanopore sequencing data</article-title>          <comment>btaa900</comment>          <source>Bioinformatics</source>          <year>2020</year>          <pub-id pub-id-type="doi">10.1093/bioinformatics/btaa900</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33674615">33674615</a> <b>(1)</b> psiCLIP reveals dynamic RNA binding by DEAH-box helicases before and after exon ligation<ul><li>A custom psiCLIP software pipeline was developed in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR60"><u>60</u></xref></sup> and is available from <ext-link ext-link-type="uri" xlink:href="http://github.com/luslab/psiclip">github.com/luslab/psiclip</ext-link>.<li> [60](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33352111">33352111</a> <b>(1)</b> Integrative Single-Cell RNA-Seq and ATAC-Seq Analysis of Human Developmental Hematopoiesis<ul><li>To run SCENIC workflow on our raw scRNA-Seq data, we used an in-house constructed <b>Snakemake</b> pipeline via combining Arboreto package GRNBoost2 and SCENIC algorithms with default parameters.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33664143">33664143</a> <b>(1)</b> ViroMatch: A Computational Pipeline for the Detection of Viral Sequences from Complex Metagenomic Data<ul><li> ViroMatch uses <b>Snakemake</b> (<xref rid="B20" ref-type="bibr"><u>20</u></xref>) to transparently organize and run all of its steps.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33658077">33658077</a> <b>(1)</b> Accurate and sensitive detection of microbial eukaryotes from whole metagenome shotgun sequencing<ul><li>The EukDetect pipeline uses the <b>Snakemake</b> workflow engine [<xref ref-type="bibr" rid="CR71"><u>71</u></xref>].<li> [71](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33713597">33713597</a> <b>(1)</b> The SARS-CoV-2 subgenome landscape and its novel regulatory features<ul><li> The source codes for all the analysis including workflows in <b>Snakemake</b> (<xref rid="bib16" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2018</u></xref>) and scripts in Python and R are available at the <ext-link ext-link-type="uri" xlink:href="https://github.com/zhouyulab/cov2sg/" id="intref0115">https://github.com/zhouyulab/cov2sg/</ext-link> in GitHub.<li> [K&#xF6;ster and Rahmann, 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33662624">33662624</a> <b>(2)</b> CharPlant: A <italic>De Novo</italic> Open Chromatin Region Prediction Tool for Plant Genomes<ul><li> Additionally, a widely used workflow management system, <b>Snakemake</b>, is employed to combine a series of steps into a single pipeline that can be run by an inexperienced user using simple command line entries <xref rid="b0190" ref-type="bibr"><u>[38]</u></xref>.<li> and &#x201C;<b>Snakemake</b>&#x201D; files.<li> [[38]](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33648554">33648554</a> <b>(1)</b> 2passtools: two-pass alignment using machine-learning-filtered splice junctions increases the accuracy of intron detection in long-read RNA sequencing<ul><li>Pipelines for processing of data were written using <b>snakemake</b> version 5.10.0 [<xref ref-type="bibr" rid="CR38"><u>38</u></xref>].<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33637709">33637709</a> <b>(1)</b> Chromatin information content landscapes inform transcription factor and DNA interactions<ul><li> All data was processed uniformly using <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup>.<li> [45](          <person-group person-group-type="author">            <name>              <surname>K&#xF6;ster</surname>              <given-names>J</given-names>            </name>            <name>              <surname>Rahmann</surname>              <given-names>S</given-names>            </name>          </person-group>          <article-title>Snakemake-a scalable bioinformatics workflow engine</article-title>          <source>Bioinforma. Oxf. Engl.</source>          <year>2012</year>          <volume>28</volume>          <fpage>2520</fpage>          <lpage>2522</lpage>          <pub-id pub-id-type="doi">10.1093/bioinformatics/bts480</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33638990">33638990</a> <b>(4)</b> An epigenetic pathway in rice connects genetic variation to anaerobic germination and seedling establishment<ul><li><italic toggle="yes">Small RNA-seq data analysis.</italic> Raw fastq reads were analyzed using the sRNA_<b>snakemake</b>_workflow (<ext-link xlink:href="https://github.com/boseHere/sRNA_<b>snakemake</b>_workflow" ext-link-type="uri">https://github.com/boseHere/sRNA_<b>snakemake</b>_workflow</ext-link>).<li> <xref rid="kiab100-B21" ref-type="bibr"><u>Johnson et&#xA0;al., 2016</u></xref>) using the sRNA_<b>snakemake</b>_workflow parameters.<li> [Johnson et&#xA0;al., 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27175019">27175019</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33630841">33630841</a> <b>(34)</b> Using prototyping to choose a bioinformatics workflow management system<ul><li> Four candidates were shortlisted: <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://github.com/<b>snakemake</b>/<b>snakemake</b>">https://github.com/<b>snakemake</b>/<b>snakemake</b></ext-link>) [<xref rid="pcbi.1008622.ref006" ref-type="bibr"><u>6</u></xref>]; cwltool (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-language/cwltool">https://github.com/common-workflow-language/cwltool</ext-link>), a reference implementation of the Common Workflow Language (CWL) [<xref rid="pcbi.1008622.ref007" ref-type="bibr"><u>7</u></xref>], a language for describing workflows; Toil (<ext-link ext-link-type="uri" xlink:href="https://github.com/DataBiosphere/toil">https://github.com/DataBiosphere/toil</ext-link>) [<xref rid="pcbi.1008622.ref008" ref-type="bibr"><u>8</u></xref>], a production implementation of CWL; and Nextflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/nextflow-io/nextflow">https://github.com/nextflow-io/nextflow</ext-link>) [<xref rid="pcbi.1008622.ref009" ref-type="bibr"><u>9</u></xref>].<li> repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/riboviz/workflows">https://github.com/riboviz/workflows</ext-link>) [<xref rid="pcbi.1008622.ref010" ref-type="bibr"><u>10</u></xref>] contains notes made during the selection and the <b>Snakemake</b>, Nextflow, and CWL workflows that were prototyped.<li>To avoid being rendered indecisive through over-choice, and, as we did not have time to review every popular option, even for shortlisting, we decided to choose only 2 systems: <b>Snakemake</b> and Nextflow.<li><b>Snakemake</b>.<li><b>Snakemake</b></th>                  <th align="left" rowspan="1" colspan="1">cwltool</th>                  <th align="left" rowspan="1" colspan="1">Toil</th>                  <th align="left" rowspan="1" colspan="1">Nextflow</th>                </tr>              </thead>              <tbody>                <tr>                  <td align="left" rowspan="1" colspan="1">Installation method used.<li><b>Snakemake</b>.<li><b>Snakemake</b> was easy to download and install, via the conda (<ext-link ext-link-type="uri" xlink:href="https://docs.conda.io/en/latest/">https://docs.conda.io/en/latest/</ext-link>) package manager, and had a comprehensive tutorial.<li> <b>Snakemake</b> adopts the same model of operation as the GNU Make automated build tool (<ext-link ext-link-type="uri" xlink:href="https://www.gnu.org/software/make/">https://www.gnu.org/software/make/</ext-link>)&#x2014;users specify the output files they want to build, <b>Snakemake</b> looks for rules to create these output files and runs the commands (in <b>Snakemake</b>, bash commands or Python scripts) specified in these rules to create the output files.<li> If these files do not exist, then <b>Snakemake</b> looks for rules to create these, and so on.<li><b>Snakemake</b> is implemented in Python.<li> This, coupled with there being no requirement to implement any additional code beyond the <b>Snakemake</b> workflow itself, meant that a functional version of the complete RiboViz workflow of 14 steps (everything bar steps specifically to handle multiplexed files) was implemented in less than a person-day.<li><b>Snakemake</b> provided all the required and useful functionality listed in our evaluation criteria.<li> <b>Snakemake</b> provides a &#x201C;keep going&#x201D; configuration parameter which can be used to continue processing other samples if processing of 1 sample fails.<li> Like Make, <b>Snakemake</b> supports incremental build and re-entrancy.<li> Step-specific log files can be implemented, but <b>Snakemake</b> does not automatically capture these&#x2014;the bash commands executed by each step explicitly need to redirect standard output and standard error streams into log files.<li> Like Make, <b>Snakemake</b> supports a &#x201C;dry run&#x201D; option that can check that input files exist and that displays the commands that would be run, without running these.<li>While <b>Snakemake</b> does not output a bash script that can be run standalone, it can output a summary file with the commands submitted to bash for execution.<li><b>Snakemake</b> has support for running its jobs within containers, HPC systems, and clouds.<li> Unlike <b>Snakemake</b>, a user does not specify the files they want to create, rather, they declare their input files and related configuration, and Nextflow continues to invoke processes until no process has any outstanding inputs.<li> Nextflow, like <b>Snakemake</b>, supports both incremental build and re-entrancy, via a &#x201C;resume&#x201D; option.<li> Unlike <b>Snakemake</b>, it is not possible to specify the exact files to build, which can make debugging more challenging.<li> analogous to that supported by <b>Snakemake</b> and Make, has been suggested in a Nextflow issue [<xref rid="pcbi.1008622.ref033" ref-type="bibr"><u>33</u></xref>], but has not progressed due to challenges in implementing such a feature within a dataflow model.<li> The fact that Nextflow was based on Java incurs no additional installation overhead for either users or developers compared to <b>Snakemake</b>&#x2014;each can be installed using the conda package manager using a single command.<li> Based on our impressions of their documentation, Nextflow&#x2019;s built-in support for, and documentation around, containers, HPC systems, and cloud, seemed more thorough than that of <b>Snakemake</b> (although we appreciate that this may change as both tools evolve).<li> It was our subjective impression that Nextflow felt far richer than <b>Snakemake</b> both in terms of features and expressivity, and it was felt that these outweighed its lack of a dry run feature.<li> Our positive experiences with <b>Snakemake</b>, and the small effort that would be required to complete the implementation of RiboViz into <b>Snakemake</b>, give us confidence that if we need to migrate from Nextflow to <b>Snakemake</b> in future, then this would be a relatively straightforward migration to undertake.<li> [6](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [7](<name><surname>Amstutz</surname><given-names>P</given-names></name>, <name><surname>Crusoe</surname><given-names>MR</given-names></name>, <name><surname>Tijanic</surname><given-names>N</given-names></name>, <name><surname>Chapman</surname><given-names>B</given-names></name>, <name><surname>Chilton</surname><given-names>J</given-names></name>, <name><surname>Heuer</surname><given-names>M</given-names></name>, <etal>et al</etal>. <article-title>Common Workflow Language, v1.0. Specification</article-title>. <year>2016</year> [cited 2020 Aug 4]. <source>Database: figshare</source> [Internet]. <pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398314">28398314</a>) [9](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [10](<name><surname>Jackson</surname><given-names>M</given-names></name>, <name><surname>Wallace</surname><given-names>E</given-names></name>, <name><surname>Kavoussanakis</surname><given-names>K</given-names></name>, <name><surname>Anderson</surname><given-names>F</given-names></name>. <article-title>Options for RiboViz workflow management</article-title>. [cited 2020 Oct 28] <source>Database: figshare</source> [Internet]. <year>2020</year>. <pub-id pub-id-type="doi">10.6084/m9.figshare.13147979</pub-id>) [33](<collab>Nextflow</collab>. <article-title>suggestion: run -dry #31</article-title>. <source>GitHub</source> [Internet]. <year>2015</year> [updated 2020 Jan 14; cited 2020 Aug 4]. Available from: <ext-link ext-link-type="uri" xlink:href="https://github.com/nextflow-io/nextflow/issues/31">https://github.com/nextflow-io/nextflow/issues/31</ext-link>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33633271">33633271</a> <b>(1)</b> The influence of human genetic variation on Epstein&#x2013;Barr virus sequence diversity<ul><li> The whole pipeline is written in <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR53"><u>53</u></xref></sup> and Python<sup><xref ref-type="bibr" rid="CR54"><u>54</u></xref></sup>.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [54](<italic>Python Language Reference</italic>. (Python Software Foundation).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=36299497">36299497</a> <b>(6)</b> <italic toggle="yes">RP-REP </italic>Ribosomal Profiling Reports: an open-source cloud-enabled framework for reproducible ribosomal profiling data processing, analysis, and result reporting<ul><li> (<bold>B</bold>) and (<bold>C</bold>) depict the respective <b>Snakemake</b> workflow in the form of a directed graph indicating sequential and parallel processing of certain pre-processing components.<li> Processing of samples within each stage is parallelized using the<ext-link xlink:href="https://<b>snakemake</b>.github.io/" ext-link-type="uri"><b>Snakemake</b></ext-link> workflow management system<sup><xref rid="ref-10" ref-type="bibr"><u>10</u></xref></sup>.<li> Additionally, for both Stage 1 and Stage 2, the workflow will track program arguments, program return codes, input and output file names, file size, MDS checksum, wall clock time, CPU time and memory consumption using the built-in <b>Snakemake</b> benchmarking utility.<li> This script also summarizes key run time statistics that were collected in the <b>Snakemake</b> benchmarking Step 2 in graphical form.<li>Another benefit is that the software was designed to handle large data volumes via utilizing the <b>Snakemake</b> workflow system for parallel data processing.<li> [10](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33625229">33625229</a> <b>(1)</b> BioContainers Registry: searching bioinformatics and proteomics tools, packages, and containers<ul><li> These tools are used in combinations with workflows such as Galaxy <sup><xref rid="R13" ref-type="bibr"><u>13</u></xref></sup>, Nextflow <sup><xref rid="R14" ref-type="bibr"><u>14</u></xref></sup>, or <b>Snakemake</b> <sup><xref rid="R15" ref-type="bibr"><u>15</u></xref></sup> which probably are the reason why they are more used than other more windows-based tools such as MaxQuant or Skyline.<li> [13](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790989">29790989</a>) [14](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [15](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33627664">33627664</a> <b>(1)</b> Alternative lengthening of telomeres in childhood neuroblastoma from genome to proteome<ul><li>ChIP-sequencing data was analyzed using a custom pipeline implemented in <b>Snakemake</b>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33620031">33620031</a> <b>(2)</b> COVID-19 CG enables SARS-CoV-2 mutation and lineage tracking by locations and dates of interest<ul><li>Our data processing pipeline is written with the <b>Snakemake</b> scalable bioinformatics workflow engine (<xref rid="bib16" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), which modularizes our workflow and enables reproducibility and compatibility with cloud-computing.<li>Data analysis is broken up into two <b>Snakemake</b> pipelines: (1) ingestion and (2) main.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33623128">33623128</a> <b>(1)</b> Melanocortin receptor activation alleviates amyloid pathology and glial reactivity in an Alzheimer&#x2019;s disease transgenic mouse model<ul><li> The pipeline to run the analysis was generated by <b>Snakemake</b> workflow management system.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33609027">33609027</a> <b>(2)</b> Whole-genome Sequencing of SARS-CoV-2: Using Phylogeny and Structural Modeling to Contextualize Local Viral Evolution<ul><li>The Nextstrain conda environment, Auger (bioinformatics tooling) and Auspice (the Nextstrain open-source visualization tool (<ext-link ext-link-type="uri" xlink:href="https://auspice.us">https://auspice.us</ext-link>), was downloaded and installed locally.<sup><xref rid="R18" ref-type="bibr"><u>18</u></xref></sup> The Nextstrain SARS-CoV-2 tutorial, including <b>snakemake</b> file with auger commands, was downloaded (<ext-link ext-link-type="uri" xlink:href="https://nextstrain.github.io/ncov">https://nextstrain.github.io/ncov</ext-link>) and utilized as the foundation of the CIRS SARS-CoV-2 pathogen build.<li>To build phylogenetic trees in Nextstrain, the <b>Snakemake</b> command was run, referencing CIRS sequences and the Nextstrain CIRS SARS-CoV-2 pathogen build, and Auspice was used to visual the output .json file in the Nextstrain environment.<li> [18](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29790939">29790939</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33608040">33608040</a> <b>(2)</b> ReSeq simulates realistic Illumina high-throughput sequencing data<ul><li>Except for the parts that are specifically mentioned in this paragraph, everything for this paper was run through <b>snakemake</b> [<xref ref-type="bibr" rid="CR55"><u>55</u></xref>] with the scripts available on GitHub [<xref ref-type="bibr" rid="CR56"><u>56</u></xref>].<li> Due to the hard-coded flags, the creation of the intermediate values is not included in the <b>snakemake</b> pipeline.<li> [55](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [56](Schmeing S. ReSeq-paper. GitHub. <ext-link ext-link-type="uri" xlink:href="https://github.com/schmeing/ReSeq-paper">https://github.com/schmeing/ReSeq-paper</ext-link>. Accessed 06 Jan 2021.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33602918">33602918</a> <b>(2)</b> Tumour gene expression signature in primary melanoma predicts long-term outcomes<ul><li> R v4.0.2, <b>snakemake</b><sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> v5.17.0.<li> The parallel processing was conducted using <b>snakemake</b><sup><xref ref-type="bibr" rid="CR45"><u>45</u></xref></sup> v5.17.0.<li> [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [45](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33597026">33597026</a> <b>(5)</b> PathoFact: a pipeline for the prediction of virulence factors and antimicrobial resistance genes in metagenomic data<ul><li> Our tool is written in Python (version 3.6) and uses the <b>Snakemake</b> (version 5.5.4) workflow management software [<xref ref-type="bibr" rid="CR28"><u>28</u></xref>].<li> Each step of the pipeline is implemented as a rule in the <b>Snakemake</b> framework specifying the input needed and the output files generated.<li> Moreover, the use of conda environments makes it possible to incorporate prediction tools dependent on older Python versions incompatible with version 5.5 of <b>Snakemake</b>.<li> As such, Python, <b>Snakemake</b>, and (mini)conda (version 4.7) [<xref ref-type="bibr" rid="CR29"><u>29</u></xref>] installations are required.<li><bold>Other requirements:</bold> <b>snakemake</b> (version &gt; = 5.5), conda (version &gt; = 4.7).<li> [28](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [29](Anaconda INC. Conda. [cited 2018]. Available from: <ext-link ext-link-type="uri" xlink:href="https://anaconda.com">https://anaconda.com</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33596194">33596194</a> <b>(1)</b> Data-driven method to infer the seizure propagation patterns in an epileptic brain from intracranial electroencephalography<ul><li> The entire computational pipeline was constructed and run using the <b>Snakemake</b> workflow manager [<xref rid="pcbi.1008689.ref083" ref-type="bibr"><u>83</u></xref>].<li> [83](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33594001">33594001</a> <b>(1)</b> Comparative Fungal Community Analyses Using Metatranscriptomics and Internal Transcribed Spacer Amplicon Sequencing from Norway Spruce<ul><li>Preprocessing and analysis of metatranscriptomic data were implemented in a <b>Snakemake</b> workflow available on Bitbucket (<xref rid="B54" ref-type="bibr"><u>54</u></xref>); we ensured complete, hardware-agnostic reproducibility through implementation in both docker and singularity containers.<li> [54](<person-group person-group-type="author"><string-name><surname>Sundh</surname><given-names>J</given-names></string-name></person-group>. <year>2020</year>. <article-title>N_Street_1801. Bitbucket</article-title>. <ext-link xlink:href="https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/" ext-link-type="uri">https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/</ext-link>. <date-in-citation content-type="access-date">Accessed 25 June 2020</date-in-citation>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=<person-group person-group-type="author"><string-name><surname>Sundh</surname><given-names>J</given-names></string-name></person-group>. <year>2020</year>. <article-title>N_Street_1801. Bitbucket</article-title>. <ext-link xlink:href="https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/" ext-link-type="uri">https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/</ext-link>. <date-in-citation content-type="access-date">Accessed 25 June 2020</date-in-citation>."><person-group person-group-type="author"><string-name><surname>Sundh</surname><given-names>J</given-names></string-name></person-group>. <year>2020</year>. <article-title>N_Street_1801. Bitbucket</article-title>. <ext-link xlink:href="https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/" ext-link-type="uri">https://bitbucket.org/scilifelab-lts/n_street_1801/src/master/</ext-link>. <date-in-citation content-type="access-date">Accessed 25 June 2020</date-in-citation>.</a> <b>(1)</b> Comparative Fungal Community Analyses Using Metatranscriptomics and Internal Transcribed Spacer Amplicon Sequencing from Norway Spruce<ul><li>Here, we introduce IsoTV (<underline>Iso</underline>form <underline>T</underline>ranscript <underline>V</underline>isualizer), a versatile <b>Snakemake</b> (<xref rid="btab103-B9399841" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) pipeline to analyze and visualize the functional features of translated transcript isoforms.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33580198">33580198</a> <b>(1)</b> Molecular and phenotypic analysis of rodent models reveals conserved and species-specific modulators of human sarcopenia<ul><li> adapters and poly(A)/poly(T) stretches, indexing reference transcriptomes, mapping the RNA-Seq reads to transcripts, and counting reads assigned to individual transcripts were performed with a <b>Snakemake</b> framework<sup><xref ref-type="bibr" rid="CR83"><u>83</u></xref></sup>.<li> [83](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33580208">33580208</a> <b>(1)</b> Efficient manipulation of gene dosage in human iPSCs using CRISPR/Cas9 nickases<ul><li> We created the pipeline to run the analysis using the <b>Snakemake</b> workflow management system<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup>.<li> [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33643254">33643254</a> <b>(1)</b> Toward an Integrated Genome-Based Surveillance of <italic>Salmonella enterica</italic> in Germany<ul><li> This <b>snakemake</b> pipeline (<xref rid="B44" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) encompasses trimming and read-quality control (QC) using fastp (<xref rid="B15" ref-type="bibr"><u>Chen et al., 2018</u></xref>), assembly using shovill (see text footnote 8), automated reference search of complete NCBI Refseq genomes using mash (<xref rid="B53" ref-type="bibr"><u>Ondov et al., 2016</u></xref>), and assembly-quality analysis using Quast (see text footnote 9).<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Chen et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30423086">30423086</a>) [Ondov et al., 2016](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27323842">27323842</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33563206">33563206</a> <b>(4)</b> Impact of concurrency on the performance of a whole exome sequencing pipeline<ul><li>We decided to exploit a WMS called <b>Snakemake</b> [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>], which adapts the GNU Make concept reimplemented leveraging Python language.<li> Furthermore, <b>Snakemake</b> is able to use the Conda environment management system [<xref ref-type="bibr" rid="CR14"><u>14</u></xref>] both for creation and activation of several environments at runtime.<li> Further, our pipeline benchmarks everyone of its steps thanks to <b>Snakemake</b>, which relies on <italic>psutil</italic> to retrieve information about each task, such as execution time and memory usage.<li>We tested the following processors configurations by using <b>Snakemake</b>&#x2019;s <italic>cores</italic> option.<li> [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [14](Anaconda software distribution. <ext-link ext-link-type="uri" xlink:href="https://anaconda.com/">https://anaconda.com/</ext-link>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33567556">33567556</a> <b>(2)</b> Dynamic Molecular Epidemiology Reveals Lineage-Associated Single-Nucleotide Variants That Alter RNA Structure in Chikungunya Virus<ul><li>We employed the workflow management system <monospace><b>Snakemake</b></monospace> [<xref rid="B53-genes-12-00239" ref-type="bibr"><u>53</u></xref>] to build a pipeline for rapid deployment and reproducibility of the CHIKV Nextstrain build.<li> In the first step of the <monospace><b>Snakemake</b></monospace> workflow, we retrieved metadata including collection date, country, and place of isolation (if available) from the Genbank records of all available CHIKV isolates.<li> [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33564781">33564781</a> <b>(1)</b> Emergence of an early SARS-CoV-2 epidemic in the United States<ul><li>Consensus sequences were assembled using an inhouse <b>Snakemake</b> <sup><xref rid="R41" ref-type="bibr"><u>41</u></xref></sup> pipeline with bwa-mem <sup><xref rid="R42" ref-type="bibr"><u>42</u></xref></sup> and iVar v1.2.2<sup><xref rid="R42" ref-type="bibr"><u>42</u></xref>,<xref rid="R43" ref-type="bibr"><u>43</u></xref></sup>.<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [42](<name><surname>Li</surname><given-names>H.</given-names></name><source>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</source>. (<year>2013</year>).) [42](<name><surname>Li</surname><given-names>H.</given-names></name><source>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</source>. (<year>2013</year>).) [43](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30606230">30606230</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33564780">33564780</a> <b>(1)</b> Genomic epidemiology identifies emergence and rapid transmission of SARS-CoV-2 B.1.1.7 in the United States<ul><li> Consensus sequences from Illumina data were assembled using an inhouse <b>Snakemake</b> (K&#xF6;ster and Rahmann, 2012) pipeline with bwa-mem (<xref rid="R21" ref-type="bibr"><u>Li, 2013</u></xref>) and iVar v1.2.2 (<xref rid="R17" ref-type="bibr"><u>Grubaugh et al., 2019b</u></xref>).<li> [Li, 2013](<name><surname>Li</surname><given-names>H.</given-names></name> (<year>2013</year>). <source>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</source> (<publisher-name>Broad Institute</publisher-name>).) [Grubaugh et al., 2019b](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=30621750">30621750</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33633776">33633776</a> <b>(2)</b> Gene Banks as Reservoirs to Detect Recent Selection: The Example of the Asturiana de los Valles Bovine Breed<ul><li> The SNP calling pipelines (DNA Sequencing and Bioinformatics Analysis) can be found at <ext-link ext-link-type="uri" xlink:href="https://forgemia.inra.fr/bios4biol/workflows/-/tree/master/<b>Snakemake</b>">https://forgemia.inra.fr/bios4biol/workflows/-/tree/master/<b>Snakemake</b></ext-link>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33526904">33526904</a> <b>(1)</b> Bisulfite-free epigenomics and genomics of single cells through methylation-sensitive restriction<ul><li>For the extraction of DNA methylation from epi-gSCAR data, NGS reads were analyzed using a custom bioinformatic pipeline (Supplementary Fig.&#xA0;<xref rid="MOESM2" ref-type="media"><u>3</u></xref>), which was automated using <b>Snakemake</b> (version 5.3.0) in Python (version 3.6).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33522966">33522966</a> <b>(1)</b> VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses<ul><li> <xref rid="MOESM2" ref-type="media"><u>S8</u></xref>), but VirSorter2&#x2019;s internal processes are highly parallelizable, and written with <b>snakemake</b>, a cutting-edge pipeline management tool designed for computer clusters [<xref ref-type="bibr" rid="CR58"><u>58</u></xref>].<li> [58](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33514375">33514375</a> <b>(2)</b> XenoCell: classification of cellular barcodes in single cell experiments from xenograft samples<ul><li>XenoCell is implemented in Python and made available as a Docker image, which contains all third-party software dependencies as a pre-configured system in order to facilitate portability across platforms and easier integration into existing workflows, such as those built with the <b>Snakemake</b> [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] or Nextflow [<xref ref-type="bibr" rid="CR8"><u>8</u></xref>] frameworks.<li> We also provide a minimal <b>Snakemake</b>-based [<xref ref-type="bibr" rid="CR7"><u>7</u></xref>] XenoCell pipeline that allows to run the previously described steps on multiple samples in an automated and parallelized fashion, ideally performed on High Performance Computing (HPC) clusters due to high memory requirements.<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33506762">33506762</a> <b>(1)</b> The rise and fall of the ancient northern pike master sex-determining gene<ul><li> The analysis was performed with a <b>snakemake</b> workflow available at <ext-link ext-link-type="uri" xlink:href="https://github.com/SexGenomicsToolkit/PSASS-workflow">https://github.com/SexGenomicsToolkit/PSASS-workflow</ext-link>.<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33584388">33584388</a> <b>(1)</b> Predicting Symptoms of Depression and Anxiety Using Smartphone and Wearable Data<ul><li>The pre-processing and feature extraction were performed using Python (version 3.7.6), R (version 4.0.2) (<xref rid="B61" ref-type="bibr"><u>61</u></xref>), <b>snakemake</b> (version 5.20) (<xref rid="B62" ref-type="bibr"><u>62</u></xref>) and following workflow examples from RAPIDS (<xref rid="B63" ref-type="bibr"><u>63</u></xref>).<li> [61](<person-group person-group-type="author"><collab>R Core Team</collab></person-group><source>R: A Language and Environment for Statistical Computing.</source> (<year>2017</year>). Available online at: <ext-link ext-link-type="uri" xlink:href="https://www.R-project.org/">https://www.R-project.org/</ext-link>.) [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [63](<person-group person-group-type="author"><collab>RAPIDS, &#x2014; RAPIDS 0,.1 documentation</collab></person-group> Available online at: <ext-link ext-link-type="uri" xlink:href="https://rapidspitt.readthedocs.io/en/latest/#">https://rapidspitt.readthedocs.io/en/latest/#</ext-link> (accessed July 24, 2020).) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33511417">33511417</a> <b>(1)</b> Repression of a large number of genes requires interplay between homologous recombination and HIRA<ul><li> Data analysis processing was automated using pipelines implemented in the SnakeLines framework (manuscript in preparation) running on the <b>Snakemake</b> workflow engine (version 5.2.2) (<xref rid="B68" ref-type="bibr"><u>68</u></xref>).<li> [68](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33503429">33503429</a> <b>(2)</b> Sexual and asexual development: two distinct programs producing the same tunicate<ul><li>Following sequencing, reads were processed using a <b>Snakemake</b> (<xref rid="R42" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) pipeline: they were trimmed to remove low quality bases and primers, merged if the reads from both ends overlapped, and aligned to a database of <italic>B.<li> These were processed using a <b>Snakemake</b> pipeline in a similar way to the <italic>B.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33482885">33482885</a> <b>(8)</b> MicroExonator enables systematic discovery and quantification of microexons across mouse embryonic development<ul><li>To ensure that analyses are fully reproducible, MicroExonator was implemented using the <b>SnakeMake</b> workflow manager [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>].<li> As MicroExonator may require significant computational resources, <b>SnakeMake</b> also facilitates running computational analyses on high-performance computer clusters by automating the scheduling of interdependent jobs.<li> <b>SnakeMake</b> itself can be installed from Bioconda [<xref ref-type="bibr" rid="CR35"><u>35</u></xref>], and it can initiate MicroExonator directly after downloading the code from our GitHub repository (<ext-link ext-link-type="uri" xlink:href="https://github.com/hemberg-lab/MicroExonator">https://github.com/hemberg-lab/MicroExonator</ext-link>).<li> Here, we applied MicroExonator to GABA-ergic and glutamatergic cells from the visual cortex, and to increase the power, we developed a downstream <b>SnakeMake</b> workflow, snakepool.<li>MicroExonator was implemented over the <b>SnakeMake</b> workflow engine [<xref ref-type="bibr" rid="CR34"><u>34</u></xref>], to facilitate reproducible processing of large numbers of RNA-seq samples.<li> After configuration, <b>SnakeMake</b> enables coordination with cluster schedulers used on high-performance computing platforms or direct process management on a single computer.<li> Moreover, we provide additional <b>SnakeMake</b> workflows to integrate MicroExonator with downstream quantification steps and to optimize analyses of single-cell RNA-seq, which are often much noisier than bulk RNA-seq data.<li>To identify differentially included microexons across cell populations profiled using scRNA-seq, we have developed snakepool, a <b>SnakeMake</b> framework that works as a downstream module of MicroExonator.<li> [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29967506">29967506</a>) [34](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33479378">33479378</a> <b>(1)</b> Metagenomic analysis of the cow, sheep, reindeer and red deer rumen<ul><li> This <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR49"><u>49</u></xref></sup> pipeline uses CheckM (v.1.0.5); prodigal (v2.6.3)<sup><xref ref-type="bibr" rid="CR50"><u>50</u></xref></sup>; Pfam_Scan (v.1.6)<sup><xref ref-type="bibr" rid="CR51"><u>51</u></xref></sup>; DIAMOND (v.0.9.22.123)<sup><xref ref-type="bibr" rid="CR52"><u>52</u></xref></sup> searching against UniProt TrEMBL<sup><xref ref-type="bibr" rid="CR53"><u>53</u></xref></sup>; PhyloPhlAn (v.0.99)<sup><xref ref-type="bibr" rid="CR54"><u>54</u></xref></sup> and sourmash (v.2.0.0)<sup><xref ref-type="bibr" rid="CR55"><u>55</u></xref></sup> searching against all public bacterial genomes.<li> [49](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [50](          <person-group person-group-type="author">            <name>              <surname>Hyatt</surname>              <given-names>D</given-names>            </name>            <etal/>          </person-group>          <article-title>Prodigal: Prokaryotic gene recognition and translation initiation site identification</article-title>          <source>BMC Bioinform.atics</source>          <year>2010</year>          <volume>11</volume>          <fpage>11</fpage>          <pub-id pub-id-type="doi">10.1186/1471-2105-11-119</pub-id>        ) [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24288371">24288371</a>) [52](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25402007">25402007</a>) [53](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=27899622">27899622</a>) [54](          <person-group person-group-type="author">            <name>              <surname>Segata</surname>              <given-names>N</given-names>            </name>            <name>              <surname>Bornigen</surname>              <given-names>D</given-names>            </name>            <name>              <surname>Morgan</surname>              <given-names>XC</given-names>            </name>            <name>              <surname>Huttenhower</surname>              <given-names>C</given-names>            </name>          </person-group>          <article-title>PhyloPhlAn is a new method for improved phylogenetic and taxonomic placement of microbes</article-title>          <source>Nat. Commun.</source>          <year>2013</year>          <volume>4</volume>          <fpage>11</fpage>          <pub-id pub-id-type="doi">10.1038/ncomms3304</pub-id>        ) [55](          <person-group person-group-type="author">            <name>              <surname>Brown</surname>              <given-names>CT</given-names>            </name>            <name>              <surname>Irber</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>Sourmash: A library for MinHash sketching of DNA</article-title>          <source>J. Open Source Softw.</source>          <year>2016</year>          <volume>1</volume>          <fpage>27</fpage>          <pub-id pub-id-type="doi">10.21105/joss.00027</pub-id>        ) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=          <person-group person-group-type="author">            <name>              <surname>Brown</surname>              <given-names>CT</given-names>            </name>            <name>              <surname>Irber</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>Sourmash: A library for MinHash sketching of DNA</article-title>          <source>J. Open Source Softw.</source>          <year>2016</year>          <volume>1</volume>          <fpage>27</fpage>          <pub-id pub-id-type="doi">10.21105/joss.00027</pub-id>        ">          <person-group person-group-type="author">            <name>              <surname>Brown</surname>              <given-names>CT</given-names>            </name>            <name>              <surname>Irber</surname>              <given-names>L</given-names>            </name>          </person-group>          <article-title>Sourmash: A library for MinHash sketching of DNA</article-title>          <source>J. Open Source Softw.</source>          <year>2016</year>          <volume>1</volume>          <fpage>27</fpage>          <pub-id pub-id-type="doi">10.21105/joss.00027</pub-id>        </a> <b>(1)</b> Metagenomic analysis of the cow, sheep, reindeer and red deer rumen<ul><li> The splitSwish function was also added to the package, and allows most of the <italic toggle="yes">Swish</italic> computations to be distributed across cores using <italic toggle="yes"><b>Snakemake</b></italic> (K<bold>&#xF6;</bold>ster and Rahmann, <xref rid="btab001-B18" ref-type="bibr"><u>2012</u></xref>).<li> [2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a> <b>(1)</b> Metagenomic analysis of the cow, sheep, reindeer and red deer rumen<ul><li>V-pipe uses the <b>Snakemake</b> workflow management system (<xref rid="btab015-B20" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>), which enables the well-controlled and scalable execution of the pipeline in local as well as in high-performance computing (HPC) environments.<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33463525">33463525</a> <b>(1)</b> Evidence for adaptive evolution in the receptor-binding domain of seasonal coronaviruses OC43 and 229e<ul><li> A <b>Snakemake</b> file (<xref rid="bib22" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) within each HCoV directory follows the general outline of a Nextstrain build (Nextstrain, RRID:<ext-link xlink:href="https://identifiers.org/RRID/RRID:SCR_018223" ext-link-type="uri">SCR_018223</ext-link>) and was used to align each gene to a reference strain and build a time-resolved phylogeny with IQ-Tree v1 (<xref rid="bib35" ref-type="bibr"><u>Nguyen et al., 2015</u></xref>) and TimeTree (<xref rid="bib41" ref-type="bibr"><u>Sagulenko et al., 2018</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Nguyen et al., 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25371430">25371430</a>) [Sagulenko et al., 2018](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29340210">29340210</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33461660">33461660</a> <b>(1)</b> Freshwater monitoring by nanopore sequencing<ul><li>The described data processing and read classification steps were implemented using the <b>Snakemake</b> workflow management system (<xref rid="bib37" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) and are available on Github - together with all necessary downstream analysis scripts to reproduce the results of this manuscript (<ext-link ext-link-type="uri" xlink:href="https://github.com/d-j-k/puntseq">https://github.com/d-j-k/puntseq</ext-link>; <xref rid="bib77" ref-type="bibr"><u>Urban et al., 2020</u></xref>; copy archived at <ext-link ext-link-type="uri" xlink:href="https://archive.softwareheritage.org/swh:1:dir:2a7babaf1eb34f3a7a8a91d9297cc28a0d3f187e;origin=https://github.com/d-j-k/puntseq;visit=swh:1:snp:8e8c60432f23b1876b83e059ec7283010fdaad94;anchor=swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a/">swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a</ext-link>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [Urban et al., 2020](          <person-group person-group-type="author">            <name>              <surname>Urban</surname>              <given-names>L</given-names>            </name>            <name>              <surname>Holzer</surname>              <given-names>A</given-names>            </name>            <name>              <surname>Hall</surname>              <given-names>M</given-names>            </name>            <name>              <surname>Kunz</surname>              <given-names>DJ</given-names>            </name>            <name>              <surname>Stammnitz</surname>              <given-names>MR</given-names>            </name>          </person-group>          <year>2020</year>          <data-title>puntseq</data-title>          <source>Software Heritage</source>          <version designator="swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a">swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a</version>          <ext-link ext-link-type="uri" xlink:href="https://archive.softwareheritage.org/swh:1:dir:2a7babaf1eb34f3a7a8a91d9297cc28a0d3f187e;origin=https://github.com/d-j-k/puntseq;visit=swh:1:snp:8e8c60432f23b1876b83e059ec7283010fdaad94;anchor=swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a/">https://archive.softwareheritage.org/swh:1:dir:2a7babaf1eb34f3a7a8a91d9297cc28a0d3f187e;origin=https://github.com/d-j-k/puntseq;visit=swh:1:snp:8e8c60432f23b1876b83e059ec7283010fdaad94;anchor=swh:1:rev:1408d508c807b88e0989a5252c5d904072dc3c4a/</ext-link>        ) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33477885">33477885</a> <b>(1)</b> Intra-Host Diversity of SARS-Cov-2 Should Not Be Neglected: Case of the State of Victoria, Australia<ul><li> This pipeline was semi-automated with <b>Snakemake</b> [<xref rid="B38-viruses-13-00133" ref-type="bibr"><u>38</u></xref>].<li> [38](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33446098">33446098</a> <b>(2)</b> Genetic variation for tolerance to the downy mildew pathogen <italic>Peronospora variabilis</italic> in genetic resources of quinoa (<italic>Chenopodium quinoa</italic>)<ul><li>Processing of the raw reads, mapping, and variant calling were done with a custom <b>Snakemake</b> pipeline [<xref ref-type="bibr" rid="CR84"><u>84</u></xref>].<li> We implemented the method in a <b>Snakemake</b> pipeline using the following parameters: <italic>k</italic>-mer length of =31 nucleotides, minor allele count =3 minor allele frequency =0.05.<li> [84](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33519720">33519720</a> <b>(1)</b> Distribution of Beta-Lactamase Producing Gram-Negative Bacterial Isolates in Isabela River of Santo Domingo, Dominican Republic<ul><li>Genomes were assembled using the Assembly HiSeq Pipeline, a <b>SnakeMake</b> pipeline to assemble sequencing data produced by Illumina (Miranda and Ramos, <xref rid="B46" ref-type="bibr"><u>2020</u></xref>).<li> [2020](<person-group person-group-type="author"><name><surname>Miranda</surname><given-names>F.</given-names></name><name><surname>Ramos</surname><given-names>R.</given-names></name></person-group> (<year>2020</year>). <source>Assembly Hiseq</source>. <publisher-loc>Bel&#xE9;m Do Par&#xE1;</publisher-loc>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33438730">33438730</a> <b>(6)</b> Streamlining data-intensive biology with workflow systems<ul><li> Our laboratory has adopted <b>Snakemake</b> [<xref rid="bib23" ref-type="bibr"><u>23</u></xref>, <xref rid="bib24" ref-type="bibr"><u>24</u></xref>], in part due to its integration with Python, its flexibility for building and testing new analyses in different languages, and its intuitive integration with software management tools (described below).<li> <b>Snakemake</b> and Nextflow [<xref rid="bib25" ref-type="bibr"><u>25</u></xref>] are commonly used for developing new research pipelines, where flexibility and iterative, branching development are key features.<li><b>Snakemake</b>.<li> For example, you might need to use Python2 to install and run a program (e.g.,&#xA0;older scripts written by members of your laboratory), while also using <b>snakemake</b> to execute your workflows (requires Python &#x2265;3.5).<li> For example, Fig.&#xA0;<xref rid="fig5" ref-type="fig"><u>5</u></xref> exhibits a modified <b>Snakemake</b> workflow visualization from an RNA-seq quantification pipeline [<xref rid="bib47" ref-type="bibr"><u>59</u></xref>].<li> The DAG shown in this figure illustrates a real bioinformatics workflow for RNA-seq quantification that was generated by modifying the default <b>Snakemake</b> workflow DAG.<li> [23](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [24](<person-group person-group-type="author"><string-name><surname>M&#xF6;lder</surname><given-names>F</given-names></string-name>, <string-name><surname>Jablonski</surname><given-names>KP</given-names></string-name>, <string-name><surname>Letcher</surname><given-names>B</given-names></string-name>, <etal>et al.</etal></person-group><article-title>Sustainable data analysis with Snakemake</article-title>. <source>Zenodo</source>. <year>2020</year>. <comment><ext-link xlink:href="https://doi.org/ghdx9x" ext-link-type="uri">https://doi.org/ghdx9x</ext-link></comment>.) [25](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [59](<comment>ngs-docs/2020-ggg-201b-rnaseq. GitHub. <ext-link xlink:href="https://github.com/ngs-docs/2020-ggg-201b-rnaseq" ext-link-type="uri">https://github.com/ngs-docs/2020-ggg-201b-rnaseq</ext-link>, accessed: 10/01/2020</comment>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33427200">33427200</a> <b>(2)</b> Selfing is the safest sex for <italic>Caenorhabditis tropicalis</italic><ul><li> A <b>SnakeMake</b> pipeline (<xref rid="bib78" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>) implementing this procedure is on github.<li>A <b>SnakeMake</b> pipeline implementing variant calling and filtering is available from github (<xref rid="bib78" ref-type="bibr"><u>K&#xF6;ster and Rahmann, 2012</u></xref>).<li> [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [K&#xF6;ster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33429418">33429418</a> <b>(1)</b> Circuits between infected macrophages and T cells in SARS-CoV-2 pneumonia<ul><li> Computations were automated with <b>snakemake</b> v5.5.4<sup><xref ref-type="supplementary-material" rid="SD13"><u>66</u></xref></sup>.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33420778">33420778</a> <b>(1)</b> Significantly improving the quality of genome assemblies through curation<ul><li> Loading of the analyses into gEVAL and subsequent assembly analyses are pipelined using <b>snakemake</b> and vr-runner [<xref rid="bib42" ref-type="bibr"><u>41</u></xref>, <xref rid="bib43" ref-type="bibr"><u>42</u></xref>].<li> [41](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) [42](<person-group person-group-type="author"><name name-style="western"><surname>Danecek</surname><given-names>P</given-names></name>, <name name-style="western"><surname>McCarthy</surname><given-names>S</given-names></name>, <name name-style="western"><surname>Randall</surname><given-names>JC</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<comment>vr-runner: A lightweight pipeline framework.</comment><ext-link ext-link-type="uri" xlink:href="https://github.com/VertebrateResequencing/vr-runner">https://github.com/VertebrateResequencing/vr-runner</ext-link>. <comment>Accessed 17 July 2020.</comment>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=34189396">34189396</a> <b>(1)</b> Augur: a bioinformatics toolkit for phylogenetic analyses of human pathogens<ul><li>By implementing the core components of Augur as a command line tool, we were able to rewrite our existing pathogen analyses as straightforward bioinformatics workflows using existing workflow management software like <b>Snakemake</b> (<xref rid="R8" ref-type="bibr"><u>K&#xF6;ster &amp; Rahmann, 2012</u></xref>).<li> [K&#xF6;ster &amp; Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33390587">33390587</a> <b>(1)</b> Sequential fate-switches in stem-like cells drive the tumorigenic trajectory from human neural stem cells to malignant glioma<ul><li> All analysis procedures were integrated by <b>Snakemake</b> (v5.5.2).</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33397413">33397413</a> <b>(1)</b> Reference flow: reducing reference bias using multiple population genomes<ul><li> Since the wrapper is written using <b>Snakemake</b> [<xref ref-type="bibr" rid="CR40"><u>40</u></xref>], it is easily deployable both in one-sample single-computer scenarios and in scenarios involving many samples or a collection of networked computers.<li> [40](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=35083472">35083472</a> <b>(1)</b> Transcription factor motif activity as a biomarker of muscle aging<ul><li> All steps were performed with a <b>Snakemake</b> framework [<xref rid="R7" ref-type="bibr"><u>7</u></xref>].<li> [7](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33315488">33315488</a> <b>(1)</b> Micronutrient supplementation affects transcriptional and epigenetic regulation of lipid metabolism in a dose-dependent manner<ul><li> For RRBS, R and Python scripts were coordinated in a pipeline by using <b>Snakemake</b> [<xref rid="cit0072" ref-type="bibr"><u>72</u></xref>].<li> [72](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33381832">33381832</a> <b>(1)</b> Inferring signaling pathways with probabilistic programming<ul><li> It includes a <b>Snakemake</b> workflow (<xref rid="btaa861-B28" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>) for our full evaluation pipeline, enabling the reader to reproduce our results.<li> [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33367849">33367849</a> <b>(1)</b> HAPHPIPE: Haplotype Reconstruction and Phylodynamics for Deep Sequencing of Intrahost Viral Populations<ul><li> to use <b>Snakemake</b> (<xref rid="msaa315-B33" ref-type="bibr"><u>K&#xF6;ster and Rahmann 2012</u></xref>), which is a workflow management system.<li> [K&#xF6;ster and Rahmann 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33342424">33342424</a> <b>(2)</b> 29th Annual Computational Neuroscience Meeting: CNS*2020<ul><li> The model is available open-source [https://inm-6.github.io/multi-area-model/] and uses the tool <b>Snakemake</b> [9] for formalizing the workflow from the experimental data to simulation, analysis, and visualization.<li> <b>Snakemake</b>&#x2014;a scalable bioinformatics workflow engine.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33339831">33339831</a> <b>(1)</b> Glioblastoma epigenome profiling identifies SOX10 as a master regulator of molecular tumour subtype<ul><li>ATAC-seq and ChIP-seq datasets were processed using a custom pipeline implemented in <b>Snakemake</b> (v.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33334308">33334308</a> <b>(4)</b> Inferring time-dependent population growth rates in cell cultures undergoing adaptation<ul><li>The program runs on linux and has a simple command line usage powered by a <b>snakemake</b> [<xref ref-type="bibr" rid="CR27"><u>27</u></xref>] workflow that automatically processes input data, runs simulations and compiles an output report.<li>Using data/minimal.csv and data/minimal.toml as input, the <b>snakemake</b> workflow automatically is used to run all simulation steps by calling <b>snakemake</b> results/minimal.pdf results/minimal.fit.csv which yields plots and a table of the inferred growth rates.<li>Other requirements: Python 3.6 or higher, pyABC 9.14 or higher, <b>snakemake</b> 4.4 or higher.<li> [27](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33327929">33327929</a> <b>(1)</b> Selected abstracts of &#x201C;Bioinformatics: from Algorithms to Applications 2020&#x201D; conference<ul><li> To ensure data analysis reproducibility, JAMIRA workflow was designed using the <b>Snakemake</b> framework.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33325479">33325479</a> <b>(5)</b> aCLImatise: automated generation of tool definitions for bioinformatics workflows<ul><li> Four popular workflow languages in bioinformatics are Nextflow, <b>Snakemake</b>, Workflow Definition Language (WDL) and Common Workflow Language (CWL) (<xref rid="btaa1033-B2" ref-type="bibr"><u>Bed&#x151;, 2019</u></xref>).<li> Such repositories exist for WDL (<ext-link ext-link-type="uri" xlink:href="https://github.com/biowdl/tasks">https://github.com/biowdl/tasks</ext-link>), <b>Snakemake</b> (<ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>-wrappers.readthedocs.io/en/stable/">https://<b>snakemake</b>-wrappers.readthedocs.io/en/stable/</ext-link>), Nextflow (<ext-link ext-link-type="uri" xlink:href="https://github.com/nf-core/modules">https://github.com/nf-core/modules</ext-link>) and CWL (<ext-link ext-link-type="uri" xlink:href="https://github.com/common-workflow-library/bio-cwl-tools">https://github.com/common-workflow-library/bio-cwl-tools</ext-link>), while registries such as Dockstore cater to multiple workflow languages simultaneously (<xref rid="btaa1033-B13" ref-type="bibr">O&#x2019;Connor <italic>et al.<li> We envisage that Galaxy (<xref rid="btaa1033-B1" ref-type="bibr">Afgan <italic>et al.</italic>, 2018</xref>), Nextflow and <b>Snakemake</b> could be supported in the future.<li> [Bed&#x151;, 2019](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=31328031">31328031</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33323122">33323122</a> <b>(1)</b> Functional and genetic markers of niche partitioning among enigmatic members of the human oral microbiome<ul><li> which uses <b>Snakemake</b> [<xref ref-type="bibr" rid="CR83"><u>83</u></xref>] to execute the steps described above for our metagenomic read recruitment analysis.<li> [83](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33377120">33377120</a> <b>(1)</b> Optimized workflow for single-cell transcriptomics on infectious diseases including COVID-19<ul><li> The protocol is based on the Drop-seq tools initially developed for processing of drop-seq data (<xref rid="bib4" ref-type="bibr"><u>Macosko et&#xA0;al., 2015</u></xref>) and has been developed into a <b>Snakemake</b> workflow for simple and reproducible use (<ext-link ext-link-type="uri" xlink:href="https://github.com/Hoohm/dropSeqPipe" id="intref0020">https://github.com/Hoohm/dropSeqPipe</ext-link>).<li> [Macosko et&#xA0;al., 2015](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=26000488">26000488</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33319904">33319904</a> <b>(3)</b> Can reproducibility be improved in clinical natural language processing? A study of 7 clinical NLP suites<ul><li> Nextflow,<xref rid="ocaa261-B16" ref-type="bibr"><sup>16</sup></xref> <b>SnakeMake</b>,<xref rid="ocaa261-B17" ref-type="bibr"><sup>17</sup></xref> and Galaxy<xref rid="ocaa261-B18" ref-type="bibr"><sup>18</sup></xref> are the most used workflow management systems within the bioinformatics community.<li> Today, <b>Snakemake</b>, Galaxy, and Nextflow are adopted by the bioinformatics community.<li> The developer community uses all WMS with a preference for <b>Snakemake</b> and Nextflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33362871">33362871</a> <b>(1)</b> Editorial: Computational Methods for Microbiome Analysis<ul><li> Cascabel wraps around existing and well-established tools in the field of amplicon analysis, connecting them by means of a <b>Snakemake</b> workflow, thus allowing for an easy and flexible execution of a common amplicon analysis pipeline.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33311468">33311468</a> <b>(1)</b> Aberrant interaction of FUS with the U1 snRNA provides a molecular mechanism of FUS induced amyotrophic lateral sclerosis<ul><li> The entire data analysis process was executed as a workflow which was created with <b>snakemake</b> version 4.3.0<sup><xref ref-type="bibr" rid="CR78"><u>78</u></xref></sup>.<li> [78](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33362867">33362867</a> <b>(4)</b> State of the Field in Multi-Omics Research: From Computational Needs to Data Mining and Sharing<ul><li><b>Snakemake</b>.<li>                <ext-link ext-link-type="uri" xlink:href="https://<b>snakemake</b>.readthedocs.io/en/stable/">https://<b>snakemake</b>.readthedocs.io/en/stable/</ext-link>              .<li>As multi-omic analyses are often multi-step with each output being the input of another, in order to increase the efficiency workflows can be written with Domain Specific Languages (DSL) such as: (a) Common Workflow Language (CWL) (<xref rid="B2" ref-type="bibr"><u>Amstutz et al., 2016</u></xref>), (b) Nextflow (<xref rid="B23" ref-type="bibr"><u>Di Tommaso et al., 2017</u></xref>), (c) <b>Snakemake</b> (<xref rid="B48" ref-type="bibr"><u>Koster and Rahmann, 2012</u></xref>), and (d) Galaxy-workflows (<xref rid="B58" ref-type="bibr"><u>McGowan et al., 2020</u></xref>).<li> [Amstutz et al., 2016](<person-group person-group-type="author"><name><surname>Amstutz</surname><given-names>P.</given-names></name><name><surname>Chapman</surname><given-names>B.</given-names></name><name><surname>Chilton</surname><given-names>J.</given-names></name><name><surname>Heuer</surname><given-names>M.</given-names></name><name><surname>Stojanovic</surname><given-names>E.</given-names></name></person-group> (<year>2016</year>). <source><italic>Common Workflow Language, v1.0 Common Workflow Language (CWL) Command Line Tool Description, v1.0.</italic></source><pub-id pub-id-type="doi">10.6084/m9.figshare.3115156.v2</pub-id>) [Di Tommaso et al., 2017](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28398311">28398311</a>) [Koster and Rahmann, 2012](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [McGowan et al., 2020](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=32236523">32236523</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33322033">33322033</a> <b>(4)</b> RNAflow: An Effective and Simple RNA-Seq Differential Gene Expression Pipeline Using Nextflow<ul><li> A&#xA0;recent study compared the currently most commonly used workflow management systems (<b>Snakemake</b>, Nextflow, CWL) with a focus on rapid prototyping for the analysis of ribosome profiling data and, finally, chose Nextflow, which best met the authors&#x2019; requirements [<xref rid="B14-genes-11-01487" ref-type="bibr"><u>14</u></xref>].<li> <monospace>VIPER</monospace> [<xref rid="B4-genes-11-01487" ref-type="bibr"><u>4</u></xref>] and <monospace>RASflow</monospace> [<xref rid="B50-genes-11-01487" ref-type="bibr"><u>50</u></xref>] make use of <b>Snakemake</b> [<xref rid="B51-genes-11-01487" ref-type="bibr"><u>51</u></xref>], another popular bioinformatic workflow management tool.<li> In&#xA0;contrast, <monospace>hppRNA</monospace> [<xref rid="B8-genes-11-01487" ref-type="bibr"><u>8</u></xref>] mainly uses Perl (<uri xlink:href="https://www.perl.org/">https://www.perl.org/</uri>) scripts to create a <b>Snakemake</b> file, but&#xA0;it lacks a true dependency management system.<li> The&#xA0;back-end is implemented within the <b>Snakemake</b> framework.<li> [14](          <person-group person-group-type="author">            <name>              <surname>Jackson</surname>              <given-names>M.J.</given-names>            </name>            <name>              <surname>Wallace</surname>              <given-names>E.</given-names>            </name>            <name>              <surname>Kavoussanakis</surname>              <given-names>K.</given-names>            </name>          </person-group>          <article-title>Using rapid prototyping to choose a bioinformatics workflow management system</article-title>          <source>bioRxiv</source>          <year>2020</year>          <comment>Available online: <ext-link ext-link-type="uri" xlink:href="https://www.biorxiv.org/content/early/2020/08/05/2020.08.04.236208.full.pdf">https://www.biorxiv.org/content/early/2020/08/05/2020.08.04.236208.full.pdf</ext-link></comment>          <date-in-citation content-type="access-date" iso-8601-date="2020-12-09">(accessed on 9 December 2020)</date-in-citation>        ) [4](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29649993">29649993</a>) [50](          <person-group person-group-type="author">            <name>              <surname>Zhang</surname>              <given-names>X.</given-names>            </name>            <name>              <surname>Jonassen</surname>              <given-names>I.</given-names>            </name>          </person-group>          <article-title>RASflow: An RNA-Seq analysis workflow with Snakemake</article-title>          <source>BMC Bioinform.</source>          <year>2020</year>          <volume>21</volume>          <fpage>1</fpage>          <lpage>9</lpage>          <pub-id pub-id-type="doi">10.1186/s12859-020-3433-x</pub-id>        ) [51](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [8](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=28096075">28096075</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33297942">33297942</a> <b>(1)</b> ideal: an R/Bioconductor package for interactive differential expression analysis<ul><li> in combined use with <b>Snakemake</b> [<xref ref-type="bibr" rid="CR62"><u>62</u></xref>], with binaries available at <ext-link ext-link-type="uri" xlink:href="https://anaconda.org/bioconda/bioconductor-ideal">https://anaconda.org/bioconda/bioconductor-ideal</ext-link>.<li> [62](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33291140">33291140</a> <b>(1)</b> Autozygosity mapping and time-to-spontaneous delivery in Norwegian parent-offspring trios<ul><li>Code for data manipulation and analysis was structured using <b>Snakemake</b> (<xref rid="ref64" ref-type="bibr"><u>64</u></xref>) and is available at <ext-link ext-link-type="uri" xlink:href="https://github.com/PerinatalLab/ROH">https://github.com/PerinatalLab/ROH</ext-link>.<li> [64](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33302542">33302542</a> <b>(1)</b> Comparative Genomics of <italic>Xanthomonas citri</italic> pv. <italic>citri</italic> A* Pathotype Reveals Three Distinct Clades with Varying Plasmid Distribution<ul><li> All tasks were automated with a <b>Snakemake</b> v5.5.0 [<xref rid="B46-microorganisms-08-01947" ref-type="bibr"><u>46</u></xref>] workflow.<li> [46](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33288906">33288906</a> <b>(1)</b> Fully phased human genome assembly without parental data using single-cell strand sequencing and long reads<ul><li>The basic workflow of our study is implemented in a reproducible and scalable <b>Snakemake</b><sup><xref ref-type="bibr" rid="CR56"><u>56</u></xref></sup> pipeline that has been successfully tested in compute environments ranging from single servers to high-performance cluster setups (&#x2018;Code availability&#x2019;).<li> [56](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33268597">33268597</a> <b>(1)</b> Meta-analysis of the gut microbiota in predicting response to cancer immunotherapy in metastatic melanoma<ul><li> All scripts, files, and data used for the analysis are publicly available on GitHub at <ext-link ext-link-type="uri" xlink:href="https://github.com/angelolimeta/Gut-microbiome-immunotherapy">https://github.com/angelolimeta/Gut-microbiome-immunotherapy</ext-link> Processing steps utilizing command line tools were performed in a reproducible Conda environment (Anaconda, Version 2-2.4.0, <ext-link ext-link-type="uri" xlink:href="https://www.anaconda.com">https://www.anaconda.com</ext-link>) with the <b>Snakemake</b> workflow engine (<xref rid="B36" ref-type="bibr"><u>36</u></xref>) on a computational cluster.<li> [36](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a>) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=29788404">29788404</a> <b>(1)</b> Meta-analysis of the gut microbiota in predicting response to cancer immunotherapy in metastatic melanoma<ul><li> The workflow is managed by <b>snakemake</b> v5.19 and is tailored for a Kubernetes cluster with AWS Spot Instances, which ensures optimal cost-effectiveness.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33335013">33335013</a> <b>(1)</b> Characterization of an unbalanced translocation causing 3q28qter duplication and 10q26.2qter deletion in a patient with global developmental delay and self-injury<ul><li> 2013</xref>) in a <b>Snakemake</b> (version 4.2.0) workflow.</ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33252655">33252655</a> <b>(8)</b> Dadasnake, a <b>Snakemake</b> implementation of DADA2 to process amplicon sequencing data for microbial ecology<ul><li> dadasnake is implemented in <b>Snakemake</b> [<xref rid="bib20" ref-type="bibr"><u>20</u></xref>] using the conda package management system.<li>Reproducibility, user-friendliness, and modular design are facilitated by the <b>Snakemake</b> framework, a popular workflow manager for reproducible and scalable data analyses (<b>Snakemake</b>, <ext-link ext-link-type="uri" xlink:href="https://scicrunch.org/resolver/RRID:SCR_003475">RRID:SCR_003475</ext-link>) [<xref rid="bib20" ref-type="bibr"><u>20</u></xref>].<li> <b>Snakemake</b> also generates HTML reports, which store code, version numbers, the workflow, and links to results.<li> <b>Snakemake</b> also ensures flexible use as single-threaded local workflow or efficient deployment on a batch scheduling system.<li> dadasnake is highly configurable compared with other <b>Snakemake</b>-based amplicon sequencing workflows, e.g., Hundo [<xref rid="bib34" ref-type="bibr"><u>35</u></xref>].<li> The <b>Snakemake</b>-generated HTML report contains all software versions and settings to facilitate the publication of the workflow's results (see supporting material [<xref rid="bib59" ref-type="bibr"><u>60</u></xref>]).<li><b>Snakemake</b> provides detailed error reports, and the logs of each step are recorded during runs.<li> [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [20](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) [35](<person-group person-group-type="author"><name name-style="western"><surname>Brown</surname><given-names>J</given-names></name>, <name name-style="western"><surname>Zavoshy</surname><given-names>N</given-names></name>, <name name-style="western"><surname>Brislawn</surname><given-names>CJ</given-names></name>, <etal>et al.</etal></person-group> &#xA0;<article-title>Hundo: a Snakemake workflow for microbial community sequence data</article-title>. <source>PeerJ Preprints</source>. <year>2018</year>:<fpage>e27272v1</fpage>.) [60](<person-group person-group-type="author"><name name-style="western"><surname>Heintz-Buschart</surname><given-names>A</given-names></name></person-group><article-title>Supplementary Datasets for dadasnake workflow</article-title>. <comment>Zenodo</comment> &#xA0;<year>2020</year><pub-id pub-id-type="doi">10.5281/zenodo.4181260</pub-id>.) </ul></p>
<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=33257685">33257685</a> <b>(1)</b> <italic>muscat</italic> detects subpopulation-specific state transitions from multi-sample multi-condition single-cell transcriptomics data<ul><li>All methods tested are available within the <italic>muscat</italic> R package and a <italic><b>Snakemake</b></italic><sup><xref ref-type="bibr" rid="CR35"><u>35</u></xref></sup> workflow was built to run simulation replicates.<li> [35](<a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=22908215">22908215</a>) </ul></p>
</body>
